--- caja-1.28.0/src/Makefile.am.orig	2024-02-20 01:30:36.000000000 +0100
+++ caja-1.28.0/src/Makefile.am	2024-02-26 08:42:41.085295973 +0100
@@ -35,6 +35,8 @@
 	$(EXIF_LIBS) \
 	$(EXEMPI_LIBS) \
 	$(POPT_LIBS) \
+        $(SCF_LIBS) \
+        $(NVPAIR_LIBS) \
     -lnotify
 	$(NULL)
 
@@ -159,6 +161,10 @@
 	caja-zoom-action.h \
 	caja-zoom-control.c \
 	caja-zoom-control.h \
+	caja-zfs-bar.c \
+	caja-zfs-bar.h \
+	timescale.h \
+	timescale.c \
 	$(NULL)
 
 nodist_caja_SOURCES = \
