--- brasero-2.30.3/libbrasero-burn/Makefile.am.orig	2013-08-13 20:06:36.308434572 +0530
+++ brasero-2.30.3/libbrasero-burn/Makefile.am	2013-08-13 20:07:07.329887471 +0530
@@ -69,7 +69,8 @@ libbrasero_burn_la_LIBADD =					\
 	$(BRASERO_GTK_LIBS)					\
 	$(BRASERO_GSTREAMER_LIBS)	\
 	$(BRASERO_DBUS_LIBS)                                    \
-	$(BRASERO_CANBERRA_LIBS)
+	$(BRASERO_CANBERRA_LIBS)				\
+	-lm
 
 libbrasero_burn_la_LDFLAGS =					\
 	-version-info $(LIBBRASERO_LT_VERSION)			\
