diff -urN brasero-2.30.3.orig/nautilus/brasero-nautilus.desktop.in.in brasero-2.30.3.hacked/nautilus/brasero-nautilus.desktop.in.in
--- brasero-2.30.3.orig/nautilus/brasero-nautilus.desktop.in.in	2010-06-11 19:49:21.000000000 +0000
+++ brasero-2.30.3.hacked/nautilus/brasero-nautilus.desktop.in.in	2011-04-15 11:09:22.829647013 +0000
@@ -10,6 +10,7 @@
 Terminal=false
 Type=Application
 OnlyShowIn=GNOME;
+NoDisplay=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=brasero
 X-GNOME-Bugzilla-Component=general
