--- brasero-2.30.3/configure.in.ori	2010-09-28 11:29:24.261642100 +0100
+++ brasero-2.30.3/configure.in	2010-09-28 11:31:03.972116600 +0100
@@ -2,6 +2,8 @@
 AC_INIT([brasero],[2.30.3])
 AC_CONFIG_SRCDIR(src/main.c)
 
+AC_PROG_LIBTOOL
+
 BRASERO_MAJOR_VERSION=2
 BRASERO_MINOR_VERSION=30
 BRASERO_SUB=3
--- brasero-2.30.3/src/Makefile.am-orig	2011-03-14 17:55:39.288360429 -0500
+++ brasero-2.30.3/src/Makefile.am	2011-03-14 17:56:02.707660708 -0500
@@ -169,7 +169,7 @@ brasero_LDADD =						\
 	$(BRASERO_SEARCH_LIBS)		\
 	$(BRASERO_PL_PARSER_LIBS)	\
 	$(BRASERO_LIBUNIQUE_LIBS)	\
-	$(BRASERO_SM_LIBS)
+	$(BRASERO_SM_LIBS) -lICE
 
 EXTRA_DIST =			\
 	brasero-marshal.list
