--- qt-everywhere-src-5.15.18/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro.orig
+++ qt-everywhere-src-5.15.18/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
@@ -2,8 +2,7 @@
 
 load(qt_helper_lib)
 
-CONFIG += qt c++14 exceptions warn_off staticlib object_parallel_to_source
-CONFIG -= c++1z
+CONFIG += qt c++17 exceptions warn_off staticlib object_parallel_to_source
 
 QT += network-private \
       gui-private \
