--- qt-everywhere-src-5.15.3/qtbase/src/3rdparty/forkfd/forkfd_atomic.h	Thu Mar 18 23:29:29 2021
+++ qt-everywhere-src-5.15.3_patched/qtbase/src/3rdparty/forkfd/forkfd_atomic.h	Wed Mar 23 23:03:29 2022
@@ -24,15 +24,7 @@
 
 #if !defined(FFD_ATOMIC_H) & !defined(FFD_ATOMIC_RELAXED)
 #define FFD_ATOMIC_H
-
-#if defined(__cplusplus) && __cplusplus >= 201103L
-#  include "forkfd_c11.h"
-#elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
-#  include "forkfd_c11.h"
-#elif defined(__GNUC__)
 #  include "forkfd_gcc.h"
-#endif
-
 #endif /* FFD_ATOMIC_h && FFD_ATOMIC_RELAXED */
 #ifndef FFD_ATOMIC_RELAXED
 #  error "Could not determine atomics for this platform"
