--- qt-everywhere-src-5.12.9/qtmultimedia/src/gsttools/gsttools.old	Wed Jun  3 05:51:21 2020
+++ qt-everywhere-src-5.12.9/qtmultimedia/src/gsttools/gsttools.pro	Fri Oct 16 19:52:25 2020
@@ -2,7 +2,7 @@
 MODULE = multimediagsttools
 CONFIG += internal_module
 
-QT = core-private multimedia-private gui-private
+QT = core-private multimedia-private gui-private opengl
 
 !static:DEFINES += QT_MAKEDLL
 DEFINES += GLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_26
