--- qt-everywhere-src-5.15.18/qtmultimedia/src/gsttools/gsttools.pro.orig
+++ qt-everywhere-src-5.15.18/qtmultimedia/src/gsttools/gsttools.pro
@@ -2,7 +2,7 @@
 MODULE = multimediagsttools
 CONFIG += internal_module
 
-QT = core-private multimedia-private gui-private
+QT = core-private multimedia-private gui-private opengl
 
 !static:DEFINES += QT_MAKEDLL
 DEFINES += GLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_26
