--- qt-everywhere-src-5.15.18/qttools/src/qdoc/configure.pri.orig
+++ qt-everywhere-src-5.15.18/qttools/src/qdoc/configure.pri
@@ -52,6 +52,14 @@
             isEmpty(LLVM_INSTALL_DIR): LLVM_INSTALL_DIR = $$(LLVM_INSTALL_DIR_MINGW)
         }
     }
+    solaris-g++: {
+        qtLog("Clang on this platform is 64-bit only, not building qdoc")
+        return(false)
+    }
+    solaris-g++-64: {
+        qtLog("Building qdoc")
+        continue
+     } 
     isEmpty(LLVM_INSTALL_DIR): LLVM_INSTALL_DIR = $$(LLVM_INSTALL_DIR)
 
     # Assume libclang is installed on the target system
