--- qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/peakcan/peakcan.pro.orig
+++ qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/peakcan/peakcan.pro
@@ -1,6 +1,6 @@
 TARGET = qtpeakcanbus
 
-QT = core serialbus
+QT = core network serialport serialbus
 
 HEADERS += \
     peakcanbackend.h \
--- qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/tinycan/tinycan.pro.orig
+++ qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/tinycan/tinycan.pro
@@ -1,6 +1,6 @@
 TARGET = qttinycanbus
 
-QT = core serialbus
+QT = core network serialport serialbus
 
 HEADERS += \
     tinycanbackend.h \
--- qt-everywhere-src-5.15.18/qtserialbus/src/tools/canbusutil/canbusutil.pro.orig
+++ qt-everywhere-src-5.15.18/qtserialbus/src/tools/canbusutil/canbusutil.pro
@@ -1,4 +1,4 @@
-QT = core serialbus
+QT = core network serialport serialbus
 
 SOURCES += main.cpp \
     readtask.cpp \
--- qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/virtualcan/virtualcan.pro.orig
+++ qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/virtualcan/virtualcan.pro
@@ -1,6 +1,6 @@
 TARGET = qtvirtualcanbus
 
-QT = core network serialbus
+QT = core network serialport serialbus
 
 HEADERS += \
     virtualcanbackend.h
--- qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/passthrucan/passthrucan.pro.orig
+++ qt-everywhere-src-5.15.18/qtserialbus/src/plugins/canbus/passthrucan/passthrucan.pro
@@ -1,4 +1,4 @@
-QT = serialbus
+QT = serialbus network serialport
 
 TARGET = qtpassthrucanbus
 
