--- qt-everywhere-opensource-src-5.8.0/qtserialbus/src/plugins/canbus/peakcan/peakcan.pro.orig	2017-01-25 02:43:19.243668208 +0100
+++ qt-everywhere-opensource-src-5.8.0/qtserialbus/src/plugins/canbus/peakcan/peakcan.pro	2017-01-25 02:43:31.025786865 +0100
@@ -1,6 +1,6 @@
 TARGET = qtpeakcanbus
 
-QT = core-private serialbus
+QT = core-private serialport serialbus
 
 HEADERS += \
     peakcanbackend.h \
--- qt-everywhere-opensource-src-5.8.0/qtserialbus/src/plugins/canbus/tinycan/tinycan.pro.orig	2017-01-25 02:42:40.618025200 +0100
+++ qt-everywhere-opensource-src-5.8.0/qtserialbus/src/plugins/canbus/tinycan/tinycan.pro	2017-01-25 02:42:58.180358147 +0100
@@ -1,6 +1,6 @@
 TARGET = qttinycanbus
 
-QT = core-private serialbus
+QT = core-private serialport serialbus
 
 HEADERS += \
     tinycanbackend.h \
--- qt-everywhere-opensource-src-5.8.0/qtserialbus/src/tools/canbusutil/canbusutil.pro.orig	2017-01-25 02:44:37.215270069 +0100
+++ qt-everywhere-opensource-src-5.8.0/qtserialbus/src/tools/canbusutil/canbusutil.pro	2017-01-25 02:44:52.714369839 +0100
@@ -1,4 +1,4 @@
-QT = core serialbus
+QT = core serialport serialbus
 
 SOURCES += main.cpp \
     readtask.cpp \
