--- qt-everywhere-src-5.15.3/qtdeclarative/src/qml/jit/qv4assemblercommon_p.h	Thu Mar 18 20:46:04 2021
+++ qt-everywhere-src-5.15.3_patched/qtdeclarative/src/qml/jit/qv4assemblercommon_p.h	Wed Mar 23 22:32:00 2022
@@ -66,7 +66,7 @@
 namespace JIT {
 
 #if defined(Q_PROCESSOR_X86_64) || defined(ENABLE_ALL_ASSEMBLERS_FOR_REFACTORING_PURPOSES)
-#if defined(Q_OS_LINUX) || defined(Q_OS_QNX) || defined(Q_OS_FREEBSD) || defined(Q_OS_DARWIN)
+#if defined(Q_OS_LINUX) || defined(Q_OS_QNX) || defined(Q_OS_FREEBSD) || defined(Q_OS_DARWIN) || defined(Q_OS_SOLARIS)
 
 class PlatformAssembler_X86_64_SysV : public JSC::MacroAssembler<JSC::MacroAssemblerX86_64>
 {
