diff -rupN areca-7.1-orig/areca.sh areca-7.1/areca.sh
--- areca-7.1-orig/areca.sh	2009-04-20 01:44:14.000000000 -0700
+++ areca-7.1/areca.sh	2012-11-19 16:13:05.288917000 -0800
@@ -7,14 +7,7 @@
 #Options
 JAVADIR=/usr/java
 
-#Getting Areca's directory
-PROGRAM_DIR=`dirname "$0"`
-#Solaris path
-if [ -f ../../usr/bin/areca ]; then
-	PROGRAM_DIR=../../usr/share/areca
-fi
-
-PROGRAM_DIR=`cd "$PROGRAM_DIR"; pwd`
+PROGRAM_DIR=/usr/share/areca
 cd "${PROGRAM_DIR}"
 
 #Configured directories
