--- tcsh-6.18.01/configure.in.orig	Wed Sep  5 11:52:15 2012
+++ tcsh-6.18.01/configure.in	Wed Sep  5 11:52:40 2012
@@ -303,7 +303,6 @@
 AC_SEARCH_LIBS(gethostbyname, nsl)
 AC_SEARCH_LIBS(connect, socket)
 AC_SEARCH_LIBS(catgets, catgets)
-AM_ICONV
 
 dnl Checks for header files
 AC_CHECK_HEADERS([auth.h crypt.h features.h inttypes.h paths.h] dnl
--- tcsh-6.18.01/Makefile.in.orig	Wed Sep  5 11:56:12 2012
+++ tcsh-6.18.01/Makefile.in	Wed Sep  5 11:57:51 2012
@@ -12,6 +12,7 @@
 BUILD=tcsh$(EXEEXT)
 VPATH=@srcdir@
 srcdir=@srcdir@
+localedir=@localedir@
 
 ################################################################
 ## CFLAGS.  For various -D things, see config.h
@@ -145,6 +146,7 @@
 #DFLAGS=-D_PATH_TCSHELL='"${DESTBIN}/tcsh"'
 ## The following is set by autoconf.
 DFLAGS = -D_PATH_TCSHELL='"${bindir}/tcsh"' @DFLAGS@ @CPPFLAGS@
+DFLAGS += -DLOCALEDIR='"${localedir}"'
 
 
 ################################################################
@@ -296,7 +298,7 @@
 #
 
 EXTRAFLAGS = @HESDEF@ $(AFSDEF)
-EXTRALIBS = @HESLIB@ $(AFSLIB) @LIBICONV@
+EXTRALIBS = @HESLIB@ $(AFSLIB)
 
 
 
--- tcsh-6.18.01/sh.c.orig	Thu Nov 15 14:26:18 2012
+++ tcsh-6.18.01/sh.c	Thu Nov 15 14:26:37 2012
@@ -175,17 +175,10 @@
     size_t len;
     int add_LOC = 1;
     int add_lang = 1;
-    char trypath[MAXPATHLEN];
-    struct stat st;
 
     if (path == NULL)
         return;
 
-    (void) xsnprintf(trypath, sizeof(trypath), "%s/en/LC_MESSAGES/tcsh.cat",
-	path);
-    if (stat(trypath, &st) == -1)
-	return;
-
     if ((old = getenv("NLSPATH")) != NULL)
         len = strlen(old) + 1;	/* don't forget the colon. */
     else
--- tcsh-6.18.01/sh.func.c.orig	Mon Dec  3 13:09:33 2012
+++ tcsh-6.18.01/sh.func.c	Mon Dec  3 13:09:54 2012
@@ -2627,6 +2627,8 @@
 
     if (adrof(STRcatalog) != NULL)
 	catalog = xasprintf("tcsh.%s", short2str(varval(STRcatalog)));
+/* for compatibility */
+#undef NL_CAT_LOCALE
 #ifdef NL_CAT_LOCALE /* POSIX-compliant. */
     /*
      * Check if LC_MESSAGES is set in the environment and use it, if so.
