--- tcsh-6.18.01/host.defs.orig	Mon Mar  4 10:31:07 2013
+++ tcsh-6.18.01/host.defs	Mon Mar  4 10:42:45 2013
@@ -41,6 +41,7 @@
 macro	: M_mips64eb : (defined(mips64) || defined(__mips64)) && (defined(MIPSEB) || defined(__MIPSEB))
 macro	: M_mipsel : (!defined(M_mips64el)) && (defined(mips) || defined(__mips)) && (defined(MIPSEL) || defined(__MIPSEL))
 macro	: M_mipseb : (!defined(M_mips64eb)) && (defined(mips) || defined(__mips)) && (defined(MIPSEB) || defined(__MIPSEB))
+macro	: M_amd64 : (defined(__amd64))
 macro	: M_i386 : (defined(i386) || defined(__i386__))
 macro	: M_i486 : (defined(i486) || defined(__i486__))
 macro	: M_i586 : (defined(i586) || defined(__i586__))
@@ -333,6 +334,7 @@
 vendor	:						: "sun"
 hosttype: defined(M_i386) && !defined(__SVR4)		: "sun386i"
 hosttype: defined(M_i386) && defined(__SVR4)		: "i86pc"
+hosttype: defined(M_amd64) 				: "i86pc"
 hosttype: defined(mc68010)				: "sun2"
 hosttype: defined(mc68020)				: "sun3"
 hosttype: defined(sparc)				: "sun4"
@@ -345,6 +347,7 @@
 machtype: defined(sparcv9)				: "sparcv9"
 machtype: defined(sparc)				: "sparc"
 machtype: defined(M_i386)				: "i386"
+machtype: defined(M_amd64)				: "amd64"
 enddef	:
 
 
