--- tcsh-6.18.01/sh.char.h.orig	Thu Apr 17 10:20:22 2014
+++ tcsh-6.18.01/sh.char.h	Thu Apr 17 10:20:38 2014
@@ -167,9 +167,9 @@
 #define iscmdmeta(c)	cmap((c), _CMD)
 #ifdef WIDE_STRINGS
 #define letter(c)	(((c) & QUOTE) ? 0 :  \
-			 (iswalpha((tcshuc) (c)) || (c) == '_'))
+			 (iswalpha((c)) || (c) == '_'))
 #define alnum(c)	(((c) & QUOTE) ? 0 :  \
-		         (iswalnum((tcshuc) (c)) || (c) == '_'))
+		         (iswalnum((c)) || (c) == '_'))
 #else
 #define letter(c)	(((Char)(c) & QUOTE) ? 0 :  \
 			 ((isalpha((tcshuc) (c)) && !(cmap((c), _PUN))) \
