# This patch has been fed upstream.
--- sendmail-8.14.7/libmilter/Makefile.m4.orig	2013-04-16 13:19:54.000000000 -0700
+++ sendmail-8.14.7/libmilter/Makefile.m4	2013-06-18 20:04:41.169213184 -0700
@@ -9,7 +9,7 @@
 SMSRCDIR=ifdef(`confSMSRCDIR', `confSMSRCDIR', `${SRCDIR}/sendmail')
 PREPENDDEF(`confINCDIRS', `-I${SMSRCDIR} ')
 
-bldPRODUCT_START(`library', `libmilter')
+bldPRODUCT_START(`shared-library', `libmilter')
 define(`bldINSTALLABLE', `true')
 define(`LIBMILTER_EXTRAS', `errstring.c strl.c')
 APPENDDEF(`confENVDEF', `-DNOT_SENDMAIL -Dsm_snprintf=snprintf')
