--- sendmail-8.17.1/libsm/Makefile.m4	2021-07-14 07:34:51.000000000 +0000
+++ sendmail-8.17.1/libsm/Makefile.m4.new	2022-02-01 12:47:29.126524073 +0000
@@ -45,7 +45,7 @@
 smcheck(`t-ixlen', `compile')
 smcheck(`t-ixlen.sh', `run')
 smcheck(`t-streq', `compile')
-smcheck(`t-utf8_valid', `compile')
+dnl ? smcheck(`t-utf8_valid', `compile')
 smcheck(`t-streq.sh', `run')
 divert(bldTARGETS_SECTION)
 divert(0)
