--- open-vm-tools-stable-9.4.6/lib/misc/codesetOld.c.orig	2016-05-26 12:35:43.465450741 +0300
+++ open-vm-tools-stable-9.4.6/lib/misc/codesetOld.c	2016-05-26 12:36:29.023823413 +0300
@@ -934,10 +934,10 @@
        * change bufIn so a simple cast is safe. --plangdale
        */

-#ifdef __linux__
+#if defined (__linux_) || defined (__sun)
       status = iconv(cd, (char **)&bufIn, &sizeIn, &out, &outLeft);
 #else
       status = iconv(cd, &bufIn, &sizeIn, &out, &outLeft);
 #endif

       DynBuf_SetSize(db, size + out - outOrig);
