--- open-vm-tools-stable-10.1.15/services/plugins/vmbackup/Makefile.in.orig	2017-12-03 10:46:34.975583757 +0000
+++ open-vm-tools-stable-10.1.15/services/plugins/vmbackup/Makefile.in	2017-12-03 10:46:59.763164767 +0000
@@ -238,7 +238,7 @@
 CAF_SUBSYS_LDFLAGS = @CAF_SUBSYS_LDFLAGS@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -Wno-unused-variable
 COMMON_PLUGIN_INSTALLDIR = @COMMON_PLUGIN_INSTALLDIR@
 COMMON_XLIBS = @COMMON_XLIBS@
 CPP = @CPP@
