--- open-vm-tools-stable-10.2.5/hgfsmounter/hgfsmounter.c.orig	2018-04-30 16:23:56.361064863 +0000
+++ open-vm-tools-stable-10.2.5/hgfsmounter/hgfsmounter.c	2018-04-30 16:24:28.813729272 +0000
@@ -103,7 +103,7 @@
 #include "hgfsmounter_version.h"
 
 /* XXX embed_version.h does not currently support Mach-O binaries (OS X). */
-#if defined(__linux__) || defined(__FreeBSD__)
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__sun)
 #  include "vm_version.h"
 #  include "embed_version.h"
    VM_EMBED_VERSION(HGFSMOUNTER_VERSION_STRING);
