--- open-vm-tools-stable-10.3.0/lib/include/vmxrpc.h.~1~	Fri Jul 13 14:54:23 2018
+++ open-vm-tools-stable-10.3.0/lib/include/vmxrpc.h	Sun Aug 19 14:38:37 2018
@@ -39,7 +39,7 @@
  * slightly different names than all other platforms. Provide macros to
  * translate the names to the more common ones.
  */
-#if defined(__GLIBC__) || defined(sun)
+#if defined(__GLIBC__) || defined(__sun)
 #  define xdr_u_int16_t(x, i)    xdr_uint16_t(x, i)
 #  define xdr_u_int32_t(x, i)    xdr_uint32_t(x, i)
 #  define xdr_u_int64_t(x, i)    xdr_uint64_t(x, i)
