--- open-vm-tools-stable-10.3.0/lib/include/iovector.h.~1~	Fri Jul 13 14:54:23 2018
+++ open-vm-tools-stable-10.3.0/lib/include/iovector.h	Sun Aug 19 15:48:13 2018
@@ -36,7 +36,7 @@
 /*
  * Ugly definition of struct iovec.
  */
-#if defined(__linux__) || defined(sun) || defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__linux__) || defined(__sun) || defined(__APPLE__) || defined(__FreeBSD__)
 #include <sys/uio.h>    // for struct iovec
 #else
 
