Index: open-vm-tools-2013.04.16-1098359/open-vm-tools/lib/misc/timeutil.c
===================================================================
--- open-vm-tools-2013.04.16-1098359/lib/misc/timeutil.c.orig
+++ open-vm-tools-2013.04.16-1098359/lib/misc/timeutil.c
@@ -855,7 +855,7 @@ TimeUtil_GetTimeFormat(int64 utcTime,  /
    const time_t t = (time_t) utcTime;  // Implicit narrowing on 32-bit

 #if defined sun
-   str = Util_SafeStrdup(ctime_r(&t, buf, sizeof buf));
+   str = Util_SafeStrdup(ctime_r(&t, buf));
 #else
    str = Util_SafeStrdup(ctime_r(&t, buf));
 #endif
Index: open-vm-tools-2013.04.16-1098359/open-vm-tools/lib/procMgr/procMgrSolaris.c
===================================================================
--- open-vm-tools-2013.04.16-1098359/lib/procMgr/procMgrSolaris.c.orig
+++ open-vm-tools-2013.04.16-1098359/lib/procMgr/procMgrSolaris.c
@@ -647,7 +647,7 @@ ProcMgr_ImpersonateUserStart(const char
    char *userLocal;

    ppw = &pw;
-   if ((ppw = getpwuid_r(0, &pw, buffer, sizeof buffer)) == NULL) {
+   if (getpwuid_r(0, &pw, buffer, sizeof buffer, &ppw) == NULL) {
       return FALSE;
    }

@@ -660,7 +660,7 @@ ProcMgr_ImpersonateUserStart(const char
        return FALSE;
    }

-   ppw = getpwnam_r(userLocal, &pw, buffer, sizeof buffer);
+   getpwnam_r(userLocal, &pw, buffer, sizeof buffer, &ppw);

    free(userLocal);

@@ -759,7 +759,7 @@ ProcMgr_ImpersonateUserStop(void)
    int ret;

    ppw = &pw;
-   if ((ppw = getpwuid_r(0, &pw, buffer, sizeof buffer)) == NULL) {
+   if (getpwuid_r(0, &pw, buffer, sizeof buffer, &ppw) == NULL) {
       return FALSE;
    }

@@ -823,7 +823,7 @@ ProcMgr_GetImpersonatedUserInfo(char **u
    *homeDir = NULL;

    ppw = &pw;
-   if ((ppw = getpwuid_r(Id_GetEUid(), &pw, buffer, sizeof buffer)) == NULL) {
+   if (getpwuid_r(Id_GetEUid(), &pw, buffer, sizeof buffer, &ppw) == NULL) {
       return FALSE;
    }

Index: open-vm-tools-2013.04.16-1098359/open-vm-tools/lib/include/vmci_sockets.h
===================================================================
--- open-vm-tools-2013.04.16-1098359/lib/include/vmci_sockets.h.orig
+++ open-vm-tools-2013.04.16-1098359/lib/include/vmci_sockets.h
@@ -487,7 +487,7 @@ struct uuid_2_cid {
       }
 #  endif // !NT_INCLUDED
 #else // _WIN32
-#if (defined(linux) && !defined(VMKERNEL)) || (defined(__APPLE__))
+#if (defined(linux) && !defined(VMKERNEL)) || (defined(__APPLE__)) || (defined(__sun))
 #  if defined(linux) && defined(__KERNEL__)
    void VMCISock_KernelRegister(void);
    void VMCISock_KernelDeregister(void);
@@ -507,7 +507,7 @@ struct uuid_2_cid {
 /** \cond PRIVATE */
 #  define VMCI_SOCKETS_DEFAULT_DEVICE      "/dev/vsock"
 #  define VMCI_SOCKETS_CLASSIC_ESX_DEVICE  "/vmfs/devices/char/vsock/vsock"
-#  if defined(linux)
+#  if defined(linux) || defined(__sun)
 #     define VMCI_SOCKETS_VERSION       1972
 #     define VMCI_SOCKETS_GET_AF_VALUE  1976
 #     define VMCI_SOCKETS_GET_LOCAL_CID 1977

