https://github.com/PerlAlien/Alien-Build/pull/427

--- Alien-Build-2.84/lib/Alien/Build/Plugin/Core/Setup.pm.orig
+++ Alien-Build-2.84/lib/Alien/Build/Plugin/Core/Setup.pm
@@ -199,6 +199,7 @@
     | \b i386 \b   # freebsd-i386
     | \b i486 \b   # i486-linux
     | \b i686 \b   # i686-cygwin
+    | \b i86pc.*64 \b # i86pc-solaris-thread-multi-64
     /ix ) {
     $arch = { name => 'x86' };
   } elsif( $Config{archname} =~ m/
