--- apache-tomcat-6.0.37-src/build.properties.default	Mon Apr 29 02:36:52 2013
+++ apache-tomcat-6.0.37-src/build.properties.default	Wed Jun 12 01:31:42 2013
@@ -54,6 +54,12 @@
 base-sf.loc=http://downloads.sourceforge.net
 base-maven.loc=http://repo2.maven.org/maven2
 
+# In Userland we really don't want Tomcat to download its build
+# dependencies itself but instead we will do it ourself in advance.
+# Therefore those dependencies which are really required need to be
+# redirected to local file system.
+userland-files.loc=file://__DISTFILES_LOCATION__
+
 # ----- Commons Logging, version 1.1 or later -----
 # If this version is updated, check the versions required for the deps
 # - avalon-framework
@@ -114,7 +120,7 @@
 jdt.jar=${jdt.home}/ecj-${jdt.version}.jar
 # The download will be moved to the archive area eventually. We are taking care of that in advance.
 # Note older JARs were called ecj.jar. Newer JARs are called ecj-${jdt.version}.jar
-jdt.loc.1=http://archive.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
+jdt.loc.1=${userland-files.loc}/ecj-${jdt.version}.jar
 jdt.loc.2=http://download.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
 
 # ----- Tomcat native library -----
@@ -126,16 +132,16 @@
 tomcat-native.win.1=${base-tomcat.loc.1}/tomcat-connectors/native/${tomcat-native.version}/binaries/tomcat-native-${tomcat-native.version}-win32-bin.zip
 tomcat-native.win.2=${base-tomcat.loc.2}/tomcat-connectors/native/${tomcat-native.version}/binaries/tomcat-native-${tomcat-native.version}-win32-bin.zip
 
-# ----- Commons DBCP, version 1.1 or later -----
-commons-dbcp.version=1.3
+# ----- Commons DBCP, version 1.4 (for JDK 1.6) -----
+commons-dbcp.version=1.4
 commons-dbcp.home=${base.path}/commons-dbcp-${commons-dbcp.version}-src
-commons-dbcp-src.loc.1=${base-commons.loc.1}/dbcp/source/commons-dbcp-${commons-dbcp.version}-src.tar.gz
+commons-dbcp-src.loc.1=${userland-files.loc}/commons-dbcp-${commons-dbcp.version}-src.tar.gz
 commons-dbcp-src.loc.2=${base-commons.loc.2}/dbcp/source/commons-dbcp-${commons-dbcp.version}-src.tar.gz
 
 # ----- Commons Pool, version 1.1 or later -----
 commons-pool.version=1.5.7
 commons-pool.home=${base.path}/commons-pool-${commons-pool.version}-src
-commons-pool-src.loc.1=${base-commons.loc.1}/pool/source/commons-pool-${commons-pool.version}-src.tar.gz
+commons-pool-src.loc.1=${userland-files.loc}/commons-pool-${commons-pool.version}-src.tar.gz
 commons-pool-src.loc.2=${base-commons.loc.2}/pool/source/commons-pool-${commons-pool.version}-src.tar.gz
 
 # ----- NSIS, version 2.0 or later -----
