--- ./src/lib/openjpip/openjpip.h.orig	2015-02-11 06:20:47.740450785 +0000
+++ ./src/lib/openjpip/openjpip.h	2015-02-11 06:24:12.124327493 +0000
@@ -276,23 +276,20 @@
  *  test the format of index (cidx) box in JP2 file
  */
 
-/** Redefinition of index parameters*/
-typedef index_param_t index_t;
-
 /**
  * Parse JP2 file and get index information from cidx box inside
  * 
  * @param[in] fd file descriptor of the JP2 file
  * @return       pointer to the generated structure of index parameters
  */
-OPJ_API index_t * OPJ_CALLCONV get_index_from_JP2file( int fd);
+OPJ_API index_param_t * OPJ_CALLCONV get_index_from_JP2file( int fd);
 
 /**
  * Destroy index parameters
  *
  * @param[in,out] idx addressof the index pointer
  */
-OPJ_API void OPJ_CALLCONV destroy_index( index_t **idx);
+OPJ_API void OPJ_CALLCONV destroy_index( index_param_t **idx);
 
 
 /**
@@ -300,7 +297,7 @@
  *
  * @param[in] index index parameters
  */
-OPJ_API void OPJ_CALLCONV output_index( index_t *index);
+OPJ_API void OPJ_CALLCONV output_index( index_param_t *index);
 
 #endif /*SERVER*/
 
--- ./src/lib/openjpip/openjpip.c.orig	2015-02-11 06:20:19.227751378 +0000
+++ ./src/lib/openjpip/openjpip.c	2015-02-11 06:23:37.274451134 +0000
@@ -409,7 +409,7 @@
   opj_free( *dec);
 }
 
-index_t * OPJ_CALLCONV get_index_from_JP2file( int fd)
+index_param_t * OPJ_CALLCONV get_index_from_JP2file( int fd)
 {
   char *data;
  
@@ -437,12 +437,12 @@
   return parse_jp2file( fd);
 }
 
-void OPJ_CALLCONV destroy_index( index_t **idx)
+void OPJ_CALLCONV destroy_index( index_param_t **idx)
 {
   delete_index( idx);
 }
 
-void OPJ_CALLCONV output_index( index_t *index)
+void OPJ_CALLCONV output_index( index_param_t *index)
 {
   print_index( *index);
 }
--- ./src/bin/jpip/opj_jpip_test.c.orig	2015-02-11 06:21:44.574812169 +0000
+++ ./src/bin/jpip/opj_jpip_test.c	2015-02-11 06:22:17.156907327 +0000
@@ -52,7 +52,7 @@
 main(int argc, char *argv[])
 {
   int fd;
-  index_t *jp2idx;
+  index_param_t *jp2idx;
   if( argc < 2 ) return 1;
   
   if( (fd = open( argv[1], O_RDONLY)) == -1){
