--- xscreensaver-6.08/hacks/glx/papercube.c.orig	2023-08-02 20:45:42.291318997 +0200
+++ xscreensaver-6.08/hacks/glx/papercube.c	2023-10-26 22:08:16.913136092 +0200
@@ -858,7 +858,7 @@
   double angle = 90.0;
   double fold = FOLD_DURATION / speed;
   double pause = PAUSE_DURATION / speed;
-  double sun = SUN_DURATION / speed;
+  double xxxsun = SUN_DURATION / speed;
   double spin = SPIN_DURATION / speed;
   double spin_rps = SPIN_RPS;
   double time = 0;
@@ -866,7 +866,7 @@
   struct field_move *field_move;
   int i;
 
-  set_move(&moves->sunrise, time, sun, 0.0, brightness);
+  set_move(&moves->sunrise, time, xxxsun, 0.0, brightness);
 
   time = moves->sunrise.stop + pause;
 
@@ -901,12 +901,12 @@
 
   time = field_move->move.stop + pause;
 
-  set_move(&moves->spin, time, time + spin + sun,
-           0.0, (spin + sun) * spin_rps * 360);
+  set_move(&moves->spin, time, time + spin + xxxsun,
+           0.0, (spin + xxxsun) * spin_rps * 360);
 
   time += spin;
 
-  set_move(&moves->sunset, time, time + sun, brightness, 0.0);
+  set_move(&moves->sunset, time, time + xxxsun, brightness, 0.0);
 
   moves->stage = Sunrise;
   set_move_time_big_bang(&moves->move_time);
