--- xscreensaver-6.16/driver/dialog.c.~1~	2025-12-02 04:17:44.694436494 +0100
+++ xscreensaver-6.16/driver/dialog.c	2026-09-27 16:21:19.820528900 +0200
@@ -1197,7 +1197,8 @@
                   xgwa.your_event_mask | SubstructureNotifyMask);
   }
 
-
+  /* switching users is broken as of 2022-08-26 */
+  ws->newlogin_button_state.disabled_p = True;
   ws->newlogin_button_state.cmd = ws->newlogin_cmd;
   ws->demo_button_state.cmd =
     get_string_resource (ws->dpy, "demoCommand", "Command");
