--- a/driver/dialog.c	Fri Aug 26 19:59:18 2022
+++ b/driver/dialog.c	Fri Aug 26 19:59:01 2022
@@ -1159,7 +1159,8 @@
                   xgwa.your_event_mask | SubstructureNotifyMask);
   }
 
-
+  /* switching users is broken as of 2022-08-26 */
+  ws->newlogin_button_state.disabled_p = True;
   ws->newlogin_button_state.cmd = ws->newlogin_cmd;
   ws->demo_button_state.cmd =
     get_string_resource (ws->dpy, "demoCommand", "Command");
