--- xscreensaver-6.08/driver/xscreensaver-auth.c.orig	2022-09-12 04:33:40.607094356 +0200
+++ xscreensaver-6.08/driver/xscreensaver-auth.c	2023-10-26 21:32:25.594969717 +0200
@@ -340,7 +340,8 @@
   if (!splash_p && !init_p)
     lock_priv_init ();
 
-  disavow_privileges ();
+  // Unable to complete the pam conversation unless we're setuid the whole time
+  //  disavow_privileges ();
 
   if (!splash_p && !init_p)
     lock_init ();
