--- xscreensaver-6.16/driver/xscreensaver.c.~2~	2026-09-27 16:21:19.734642300 +0200
+++ xscreensaver-6.16/driver/xscreensaver.c	2026-09-27 16:21:19.873846200 +0200
@@ -1454,8 +1454,11 @@
   if (!why)
     {
       uid_t u = getuid();
-      if (u == 0 || u == (uid_t) -1 || u == (uid_t) -2)
-        why = "cannot lock when running as root";
+      if (u == 0 || u == (uid_t) -1 || u == (uid_t) -2) {
+        fprintf(stderr, "%s: Logging in as root is a completely unnecessary security risk.\n",blurb());
+        fprintf(stderr, "%s: xscreensaver's locking functionality is normally disabled when the \n",blurb());
+        fprintf(stderr, "%s: daemon is run as root. Please log in as a normal user.\n",blurb());
+      }
     }
 
   if (!why && getenv ("RUNNING_UNDER_GDM"))
@@ -1489,8 +1492,9 @@
       else
         {
           locking_disabled_p = True;
-          if (lock_p || verbose_p)
+          if (lock_p || verbose_p) {
             fprintf (stderr, "%s: locking disabled: %s\n", blurb(), why);
+          }
         }
     }
 }
