diff -Nur driver/Makefile.in driver/Makefile.in
--- driver/Makefile.in	2017-01-06 09:29:12.165724605 -0500
+++ driver/Makefile.in	2017-01-06 09:33:44.727774491 -0500
@@ -45,7 +45,7 @@
 CONF_DEFS	= -DHACK_CONFIGURATION_PATH='"$(HACK_CONF_DIR)"'
 LOCK_DEFS	= $(DEFS) -DLOCKDIR=\"$(LOCK_DIR)\"
 
-LIBS		= @LIBS@
+LIBS		= @LIBS@ -lglib-2.0
 INTL_LIBS	= @INTLLIBS@
 JPEG_LIBS	= @JPEG_LIBS@
 PERL		= @PERL@
