--- ptlib-2.6.7.orig/include/ptlib/array.h
+++ ptlib-2.6.7/include/ptlib/array.h
@@ -523,7 +523,7 @@ template <class T> class PScalarArray :
       T t;
       stream >> t;
       if (!stream.fail())
-        SetAt(index, t);
+        this->SetAt(index, t);
     }
 };

--- ptlib-2.6.7.orig/include/ptlib/safecoll.h
+++ ptlib-2.6.7/include/ptlib/safecoll.h
@@ -782,7 +782,7 @@
      */
     PSafePtr & operator=(T * obj)
       {
-        Assign(obj);
+        this->Assign(obj);
         return *this;
       }

