--- doc/lighttpd.conf-orig	Thu Apr  9 13:02:00 2009
+++ doc/lighttpd.conf	Wed Nov  4 03:16:09 2009
@@ -36,10 +36,10 @@
 
 ## A static document-root. For virtual hosting take a look at the
 ## mod_simple_vhost module.
-server.document-root        = "/srv/www/htdocs/"
+server.document-root = "/var/lighttpd/1.4/docroot"
 
 ## where to send error-messages to
-server.errorlog             = "/var/log/lighttpd/error.log"
+server.errorlog = "/var/lighttpd/1.4/logs/error.log"
 
 # files to check for if .../ is requested
 index-file.names            = ( "index.php", "index.html",
@@ -46,8 +46,10 @@
                                 "index.htm", "default.htm" )
 
 ## set the event-handler (read the performance section in the manual)
-# server.event-handler = "freebsd-kqueue" # needed on OS X
+server.event-handler = "solaris-devpoll"
+server.network-backend = "writev"
 
+
 # mimetype mapping
 mimetype.assign             = (
   ".pdf"          =>      "application/pdf",
@@ -115,7 +117,7 @@
 # server.tag                 = "lighttpd"
 
 #### accesslog module
-accesslog.filename          = "/var/log/lighttpd/access.log"
+accesslog.filename = "/var/lighttpd/1.4/logs/access.log"
 
 ## deny access the file-extensions
 #
@@ -147,7 +149,7 @@
 #server.error-handler-404   = "/error-handler.php"
 
 ## to help the rc.scripts
-#server.pid-file            = "/var/run/lighttpd.pid"
+server.pid-file = "/var/run/lighttpd14.pid"
 
 
 ###### virtual hosts
@@ -160,7 +162,7 @@
 ## or
 ##   virtual-server-root + http-host + virtual-server-docroot
 ##
-#simple-vhost.server-root   = "/srv/www/vhosts/"
+#simple-vhost.server-root   = "/var/lighttpd/1.4/vhosts/"
 #simple-vhost.default-host  = "www.example.org"
 #simple-vhost.document-root = "/htdocs/"
 
@@ -168,8 +170,7 @@
 ##
 ## Format: <errorfile-prefix><status-code>.html
 ## -> ..../status-404.html for 'File not found'
-#server.errorfile-prefix    = "/usr/share/lighttpd/errors/status-"
-#server.errorfile-prefix    = "/srv/www/errors/status-"
+#server.errorfile-prefix    = "/var/lighttpd/1.4/errors/status-"
 
 ## virtual directory listings
 #dir-listing.activate       = "enable"
@@ -188,10 +189,10 @@
 #server.chroot              = "/"
 
 ## change uid to <uid> (default: don't care)
-#server.username            = "wwwrun"
+server.username = "webservd"
 
 ## change uid to <uid> (default: don't care)
-#server.groupname           = "wwwrun"
+server.groupname           = "webservd"
 
 #### compress module
 #compress.cache-dir         = "/var/cache/lighttpd/compress/"
@@ -214,8 +215,8 @@
 #fastcgi.server             = ( ".php" =>
 #                               ( "localhost" =>
 #                                 (
-#                                   "socket" => "/var/run/lighttpd/php-fastcgi.socket",
-#                                   "bin-path" => "/usr/local/bin/php-cgi"
+#                                   "socket" => "/tmp/lighttpd/php-fastcgi.socket",
+#                                   "bin-path" => "/usr/php/bin/php-cgi"
 #                                 )
 #                               )
 #                            )
@@ -274,7 +275,7 @@
 # %3 => subdomain 1 name
 # %4 => subdomain 2 name
 #
-#evhost.path-pattern        = "/srv/www/vhosts/%3/htdocs/"
+#evhost.path-pattern        = "/var/lighttpd/1.4/vhosts/%3/htdocs/"
 
 #### expire module
 #expire.url                 = ( "/buggy/" => "access 2 hours", "/asdhas/" => "access plus 1 seconds 2 minutes")
@@ -312,8 +313,8 @@
 #index-file.names += (foo + ".php")
 
 #### include
-#include /etc/lighttpd/lighttpd-inc.conf
-## same as above if you run: "lighttpd -f /etc/lighttpd/lighttpd.conf"
+#include /etc/lighttpd/1.4/lighttpd-inc.conf
+## same as above if you run: "lighttpd -f /etc/lighttpd/1.4/lighttpd.conf"
 #include "lighttpd-inc.conf"
 
 #### include_shell
@@ -320,3 +321,5 @@
 #include_shell "echo var.a=1"
 ## the above is same as:
 #var.a=1
+# Set the directory used for file uploads to /tmp
+server.upload-dirs = ( "/tmp" )
