--- libglvnd-1.3.0/src/GLdispatch/Makefile.am.orig	2020-02-02 23:46:00.969122296 +0000
+++ libglvnd-1.3.0/src/GLdispatch/Makefile.am	2020-02-02 23:46:52.158088657 +0000
@@ -45,7 +45,7 @@
 VERSION_SCRIPT=$(top_srcdir)/src/GLdispatch/export_list_tsd.ver
 endif
 EXTRA_libGLdispatch_la_DEPENDENCIES = $(VERSION_SCRIPT)
-libGLdispatch_la_LDFLAGS += -Xlinker --version-script=$(VERSION_SCRIPT)
+libGLdispatch_la_LDFLAGS += -Xlinker -Wl,-M$(VERSION_SCRIPT)
 
 libGLdispatch_la_SOURCES = \
 	GLdispatch.c
