--- engrampa-1.14.1/src/fr-command-unarchiver.c.1	2016-09-02 09:23:35.023209440 +0300
+++ engrampa-1.14.1/src/fr-command-unarchiver.c	2016-09-02 09:23:53.823127470 +0300
@@ -19,7 +19,9 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifndef __sun__
 #define _XOPEN_SOURCE       /* See feature_test_macros(7) */
+#endif
 #include <time.h>
 #include <stdio.h>
 #include <stdlib.h>
