--- sqlite-src-3430200/Makefile.in.orig	2023-10-10 17:08:53.000000000 +0200
+++ sqlite-src-3430200/Makefile.in	2023-10-10 21:36:31.072280140 +0200
@@ -1339,7 +1339,7 @@
 # This is the common case.  Run many tests that do not take too long,
 # including fuzzcheck, sqlite3_analyzer, and sqldiff tests.
 #
-test:	srctree-check fuzztest sourcetest $(TESTPROGS) tcltest
+test:	srctree-check fuzztest sourcetest $(HAVE_TCL:1=tcltest)
 
 # Run a test using valgrind.  This can take a really long time
 # because valgrind is so much slower than a native machine.
