Force use of -fPIC

--- sqlite-src-3500000/main.mk.orig	2025-05-29 16:34:54.000000000 +0200
+++ sqlite-src-3500000/main.mk	2025-06-03 12:04:47.939594778 +0200
@@ -1665,7 +1665,7 @@
 #
 # $(CFLAGS.tclextension) = CFLAGS for the tclextension* targets.
 #
-CFLAGS.tclextension = $(CFLAGS.intree_includes) $(CFLAGS.env) $(OPT_FEATURE_FLAGS) $(OPTS)
+CFLAGS.tclextension = $(CFLAGS.intree_includes) $(CFLAGS.env) $(OPT_FEATURE_FLAGS) $(OPTS) -fPIC
 #
 # Build the SQLite TCL extension in a way that make it compatible
 # with whatever version of TCL is running as $TCLSH_CMD, possibly defined
