--- sqlite-src-3530000/test/testrunner.tcl.orig	2026-04-09 15:11:37.000000000 +0200
+++ sqlite-src-3530000/test/testrunner.tcl	2026-04-19 13:04:19.204770656 +0200
@@ -279,6 +279,15 @@
     set TRG(run)         run.sh
     set TRG(runcmd)      "sh run.sh"
   }
+  *SunOS* {
+    set TRG(platform)    linux
+    set TRG(make)        make.sh
+    set TRG(makecmd)     "sh make.sh"
+    set TRG(testfixture) testfixture
+    set TRG(shell)       sqlite3
+    set TRG(run)         run.sh
+    set TRG(runcmd)      "sh run.sh"
+  }
   *win* {
     set TRG(platform)    win
     set TRG(make)        make.bat
