--- sqlite-src-3260000/src/os_unix.c	2018-12-01 14:38:18.000000000 +0000
+++ sqlite-src-3260000/src/os_unix.c.new	2018-12-15 21:14:25.966602925 +0000
@@ -444,7 +444,11 @@ static struct unix_syscall {
 #define osFchmod    ((int(*)(int,mode_t))aSyscall[14].pCurrent)
 
 #if defined(HAVE_POSIX_FALLOCATE) && HAVE_POSIX_FALLOCATE
+#if !defined(_LP64) && _FILE_OFFSET_BITS == 64
+  { "fallocate",    (sqlite3_syscall_ptr)posix_fallocate64,  0 },
+#else
   { "fallocate",    (sqlite3_syscall_ptr)posix_fallocate,  0 },
+#endif
 #else
   { "fallocate",    (sqlite3_syscall_ptr)0,                0 },
 #endif
