--- sqlite-autoconf-3130000/Makefile.am.1	2016-07-21 10:16:49.840352588 +0300
+++ sqlite-autoconf-3130000/Makefile.am	2016-07-21 10:17:14.420719978 +0300
@@ -9,6 +9,7 @@
 sqlite3_SOURCES = shell.c sqlite3.h
 EXTRA_sqlite3_SOURCES = sqlite3.c
 sqlite3_LDADD = @EXTRA_SHELL_OBJ@ @READLINE_LIBS@
+sqlite3_LDFLAGS = -Wl,-zrecord
 sqlite3_DEPENDENCIES = @EXTRA_SHELL_OBJ@
 sqlite3_CFLAGS = $(AM_CFLAGS) -DSQLITE_ENABLE_EXPLAIN_COMMENTS
 
