--- sqlite-src-3260000/Makefile.in	2018-12-01 14:38:18.000000000 +0000
+++ sqlite-src-3260000/Makefile.in.new	2018-12-15 21:10:39.413857908 +0000
@@ -627,7 +627,7 @@ sqlite3.pc: $(TOP)/sqlite3.pc.in
 
 libsqlite3.la:	$(LIBOBJ)
 	$(LTLINK) -no-undefined -o $@ $(LIBOBJ) $(TLIBS) \
-		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
+		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8" -Wl,-M,mapfile -Wl,-zrecord
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -no-undefined -o $@ tclsqlite.lo \
