https://github.com/agronholm/exceptiongroup/issues/148

--- exceptiongroup-1.3.0/tests/test_exceptions.py.orig
+++ exceptiongroup-1.3.0/tests/test_exceptions.py
@@ -374,7 +374,7 @@
 class DeepRecursionInSplitAndSubgroup(unittest.TestCase):
     def make_deep_eg(self):
         e = TypeError(1)
-        for _ in range(10000):
+        for _ in range(150_000):
             e = ExceptionGroup("eg", [e])
         return e
 
