From 07e227ed00d0fb787c06cf03069b2904e692e770 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Thu, 7 Sep 2017 18:55:46 +0200
Subject: Fix memory leak in LZMA decompressor

---
 xzlib.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xzlib.c b/xzlib.c
index f43632b..8bafbd3 100644
--- a/xzlib.c
+++ b/xzlib.c
@@ -797,6 +797,8 @@ __libxml2_xzclose(xzFile file)
         xmlFree(state->in);
     }
     xmlFree(state->path);
+    if ((state->msg != NULL) && (state->err != LZMA_MEM_ERROR))
+        xmlFree(state->msg);
     ret = close(state->fd);
     xmlFree(state);
     return ret ? ret : LZMA_OK;
-- 
cgit v0.12

