https://github.com/fastmail/mail-dkim/issues/28

--- Mail-DKIM-1.20240923/Makefile.PL.orig
+++ Mail-DKIM-1.20240923/Makefile.PL
@@ -37,7 +37,6 @@
     "Test::More" => 0,
     "Test::RequiresInternet" => 0,
     "Test::Simple" => 0,
-    "YAML" => 0,
     "YAML::XS" => 0,
     "lib" => 0
   },
@@ -64,7 +63,6 @@
   "Test::More" => 0,
   "Test::RequiresInternet" => 0,
   "Test::Simple" => 0,
-  "YAML" => 0,
   "YAML::XS" => 0,
   "base" => 0,
   "lib" => 0,
