--- storage/innobase/include/univ.i	Mon Jul 13 16:12:14 2009
+++ storage/innobase/include/univ.i.org	Thu May 12 23:27:42 2011
@@ -56,9 +56,9 @@
 # endif
 
 /* We only try to do explicit inlining of functions with gcc and
-Microsoft Visual C++ */
+Microsoft Visual C++  and Oracle Compilers */
 
-# if !defined(__GNUC__)
+# if !defined(__GNUC__) && !defined(__SUNWSPRO_C)
 #  undef  UNIV_MUST_NOT_INLINE			/* Remove compiler warning */
 #  define UNIV_MUST_NOT_INLINE
 # endif
