--- aiohttp-3.9.0/requirements/test.in.orig
+++ aiohttp-3.9.0/requirements/test.in
@@ -1,5 +1,4 @@
--r base.in
--c broken-projects.in
+-r requirements/base.in
 
 coverage
 mypy; implementation_name == "cpython"
--- aiohttp-3.9.0/requirements/base.in.orig
+++ aiohttp-3.9.0/requirements/base.in
@@ -1,5 +1,5 @@
--r typing-extensions.in
--r runtime-deps.in
+-r requirements/typing-extensions.in
+-r requirements/runtime-deps.in
 
 gunicorn
 uvloop; platform_system != "Windows" and implementation_name == "cpython"  # MagicStack/uvloop#14
