--- aiohttp-3.10.1/requirements/test.in.orig
+++ aiohttp-3.10.1/requirements/test.in
@@ -1,5 +1,5 @@
 -c broken-projects.in
--r base.in
+-r requirements/base.in
 
 coverage
 freezegun
--- aiohttp-3.10.1/requirements/base.in.orig
+++ aiohttp-3.10.1/requirements/base.in
@@ -1,4 +1,4 @@
--r runtime-deps.in
+-r requirements/runtime-deps.in
 
 gunicorn
 uvloop; platform_system != "Windows" and implementation_name == "cpython"  # MagicStack/uvloop#14
