We have no isal packaged yet.

--- aiohttp-3.12.4/requirements/test.in.orig
+++ aiohttp-3.12.4/requirements/test.in
@@ -3,7 +3,6 @@
 blockbuster
 coverage
 freezegun
-isal
 mypy; implementation_name == "cpython"
 pkgconfig
 proxy.py >= 2.4.4rc5
--- aiohttp-3.12.4/tests/conftest.py.orig
+++ aiohttp-3.12.4/tests/conftest.py
@@ -12,7 +12,6 @@
 from unittest import mock
 from uuid import uuid4
 
-import isal.isal_zlib
 import pytest
 import zlib_ng.zlib_ng
 from blockbuster import blockbuster_ctx
@@ -331,7 +330,7 @@
         s.close()
 
 
-@pytest.fixture(params=[zlib, zlib_ng.zlib_ng, isal.isal_zlib])
+@pytest.fixture(params=[zlib, zlib_ng.zlib_ng])
 def parametrize_zlib_backend(
     request: pytest.FixtureRequest,
 ) -> Generator[None, None, None]:
