--- aiohttp-3.8.6/tests/test___all__.py.orig
+++ aiohttp-3.8.6/tests/test___all__.py
@@ -8,7 +8,7 @@
             from aiohttp import *
         """
     )
-    result = pytester.runpytest("-vv")
+    result = pytester.runpytest("-vv", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter")
     result.assert_outcomes(passed=0, errors=0)
 
 
@@ -18,5 +18,5 @@
             from aiohttp.web import *
         """
     )
-    result = pytester.runpytest("-vv")
+    result = pytester.runpytest("-vv", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter")
     result.assert_outcomes(passed=0, errors=0)
--- aiohttp-3.8.6/tests/test_pytest_plugin.py.orig
+++ aiohttp-3.8.6/tests/test_pytest_plugin.py
@@ -149,7 +149,7 @@
 """
     )
     testdir.makeconftest(CONFTEST)
-    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop")
+    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter")
     result.assert_outcomes(passed=12)
 
 
@@ -170,7 +170,7 @@
     )
     testdir.makeconftest(CONFTEST)
     result = testdir.runpytest(
-        "-p", "no:sugar", "-s", "-W", "default", "--aiohttp-loop=pyloop"
+        "-p", "no:sugar", "-s", "-W", "default", "--aiohttp-loop=pyloop", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter"
     )
     expected_outcomes = (
         {"failed": 0, "passed": 2}
@@ -235,7 +235,7 @@
 """
     )
     testdir.makeconftest(CONFTEST)
-    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop")
+    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter")
     result.assert_outcomes(passed=3, errors=1)
     result.stdout.fnmatch_lines(
         "*Asynchronous fixtures must depend on the 'loop' fixture "
@@ -282,5 +282,5 @@
 """
     )
     testdir.makeconftest(CONFTEST)
-    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop")
+    result = testdir.runpytest("-p", "no:sugar", "--aiohttp-loop=pyloop", "-p", "no:black", "-p", "no:mypy", "-p", "no:reporter")
     result.assert_outcomes(passed=2)
--- aiohttp-3.8.6/tests/test_test_utils.py.orig
+++ aiohttp-3.8.6/tests/test_test_utils.py
@@ -312,7 +312,7 @@
                 pass
         """
     )
-    result = testdir.runpytest()
+    result = testdir.runpytest("-p", "no:reporter")
     result.stdout.fnmatch_lines(["*RuntimeError*"])
 
 
