We have no zlib_ng packaged yet.

--- aiohttp-3.12.2/requirements/test.in.orig
+++ aiohttp-3.12.2/requirements/test.in
@@ -16,4 +16,3 @@
 setuptools-git
 trustme; platform_machine != "i686"  # no 32-bit wheels
 wait-for-it
-zlib_ng
--- aiohttp-3.12.2/tests/conftest.py.orig
+++ aiohttp-3.12.2/tests/conftest.py
@@ -13,7 +13,6 @@
 from uuid import uuid4
 
 import pytest
-import zlib_ng.zlib_ng
 from blockbuster import blockbuster_ctx
 
 from aiohttp import payload
@@ -322,7 +321,7 @@
         s.close()
 
 
-@pytest.fixture(params=[zlib, zlib_ng.zlib_ng])
+@pytest.fixture(params=[zlib])
 def parametrize_zlib_backend(
     request: pytest.FixtureRequest,
 ) -> Generator[None, None, None]:
