We have no blockbuster packaged yet.

--- aiohttp-3.12.14/requirements/test.in.orig
+++ aiohttp-3.12.14/requirements/test.in
@@ -1,6 +1,5 @@
 -r requirements/base.in
 
-blockbuster
 coverage
 freezegun
 mypy; implementation_name == "cpython"
--- aiohttp-3.12.14/tests/conftest.py.orig
+++ aiohttp-3.12.14/tests/conftest.py
@@ -13,7 +13,6 @@
 from uuid import uuid4
 
 import pytest
-from blockbuster import blockbuster_ctx
 
 from aiohttp import payload
 from aiohttp.client_proto import ResponseHandler
@@ -48,6 +47,9 @@
 
 @pytest.fixture(autouse=True)
 def blockbuster(request: pytest.FixtureRequest) -> Iterator[None]:
+    yield
+    return
+
     # Allow selectively disabling blockbuster for specific tests
     # using the @pytest.mark.skip_blockbuster marker.
     if "skip_blockbuster" in request.node.keywords:
