We do not have these projects packaged yet.

--- aiohttp-3.12.11/requirements/test.in.orig
+++ aiohttp-3.12.11/requirements/test.in
@@ -4,14 +4,9 @@
 freezegun
 mypy; implementation_name == "cpython"
 pkgconfig
-proxy.py >= 2.4.4rc5
 pytest
 pytest-cov
 pytest-mock
 pytest-xdist
-pytest_codspeed
-python-on-whales
 re-assert
-setuptools-git
 trustme; platform_machine != "i686"  # no 32-bit wheels
-wait-for-it
--- aiohttp-3.12.11/requirements/base.in.orig
+++ aiohttp-3.12.11/requirements/base.in
@@ -1,5 +1,4 @@
 -r requirements/runtime-deps.in
 
 gunicorn
-uvloop; platform_system != "Windows" and implementation_name == "cpython"  # MagicStack/uvloop#14
 winloop; platform_system == "Windows" and implementation_name == "cpython"
--- aiohttp-3.12.11/requirements/runtime-deps.in.orig
+++ aiohttp-3.12.11/requirements/runtime-deps.in
@@ -1,6 +1,5 @@
 # Extracted from `setup.cfg` via `make sync-direct-runtime-deps`
 
-aiodns >= 3.3.0
 aiohappyeyeballs >= 2.5.0
 aiosignal >= 1.1.2
 async-timeout >= 4.0, < 6.0 ; python_version < "3.11"
