--- php-5.4.16/ext/mysqli/config.m4	2013-06-05 09:03:57.000000000 +0400
+++ php-5.4.16/ext/mysqli/config.m4	2013-07-08 14:52:56.245692052 +0400
@@ -34,7 +34,7 @@
   
   if test -x "$MYSQL_CONFIG" && $MYSQL_CONFIG $MYSQL_LIB_CFG > /dev/null 2>&1; then
     MYSQLI_INCLINE=`$MYSQL_CONFIG --cflags | $SED -e "s/'//g"`
-    MYSQLI_LIBLINE=`$MYSQL_CONFIG $MYSQL_LIB_CFG | $SED -e "s/'//g"`
+    MYSQLI_LIBLINE=`$MYSQL_CONFIG $MYSQL_LIB_CFG | $SED -e "s/'//g" -e "s:-L/opt/SUNWspro/lib::g" -e "s:-lCrun::g"`
   else
     AC_MSG_RESULT([mysql_config not found])
     AC_MSG_ERROR([Please reinstall the mysql distribution])
