--- boost_1_58_0/libs/log/build/Jamfile.v2.orig	2015-06-21 03:57:28.173374685 +0000
+++ boost_1_58_0/libs/log/build/Jamfile.v2	2015-06-21 03:57:44.553816608 +0000
@@ -178,7 +178,7 @@
         <target-os>linux:<define>_XOPEN_SOURCE=600
         <target-os>linux:<define>_GNU_SOURCE=1
 
-        <target-os>solaris:<define>_XOPEN_SOURCE=500
+        <target-os>solaris:<define>_XOPEN_SOURCE=600
         <target-os>solaris:<define>__EXTENSIONS__
         <target-os>solaris:<library>socket
         <target-os>solaris:<library>nsl
