https://rt.cpan.org/Ticket/Display.html?id=168226

--- Math-Base-Convert-0.11/Convert.pm.orig
+++ Math-Base-Convert-0.11/Convert.pm
@@ -254,7 +254,7 @@
 # if a large base and digit string will fit in a single 32 bit register
     elsif ( $fblen > 32 &&		# big base
 #	    exists $maxdlen{$fbase} &&	# has to exist
-	  ! $slen > $maxdlen{$fbase}) {
+	not $slen > $maxdlen{$fbase}) {
       $bc->useFROMbaseto32wide;		# CalcPP is faster
     }
     else {				# shortcuts faster for big numbers
