https://github.com/wntrblm/nox/issues/793

--- nox-2024.3.2/tests/test_sessions.py.orig
+++ nox-2024.3.2/tests/test_sessions.py
@@ -946,7 +946,6 @@
                 nox.virtualenv.VirtualEnv,
             ),
             ("nox.virtualenv.VirtualEnv.create", "venv", nox.virtualenv.VirtualEnv),
-            ("nox.virtualenv.CondaEnv.create", "conda", nox.virtualenv.CondaEnv),
         ],
     )
     def test__create_venv_options(self, create_method, venv_backend, expected_backend):
