illumos doesn't support netstat -l.
--- gnutls-3.7.8/tests/scripts/common.sh.orig	2022-09-27 14:42:22.000000000 +0000
+++ gnutls-3.7.8/tests/scripts/common.sh	2022-10-31 11:28:56.517033635 +0000
@@ -80,7 +80,7 @@
 check_if_port_listening() {
 	local PORT=$1
 	have_port_finder
-	$PFCMD -anl|grep "[\:\.]$PORT"|grep LISTEN >/dev/null 2>&1
+	$PFCMD -an|grep "[\:\.]$PORT"|grep LISTEN >/dev/null 2>&1
 }
 
 # Find a port number not currently in use.
