We have no pytest-retry packaged yet.

--- sphobjinv-2.3.1.3/tox.ini.orig
+++ sphobjinv-2.3.1.3/tox.ini
@@ -72,7 +72,6 @@
 
     pytest>=4.4.0
     pytest-check>=1.1.2
-    pytest-retry
     pytest-timeout
     stdio-mgr>=1.0.1
     sphinx-issues
--- sphobjinv-2.3.1.3/tests/test_cli_nonlocal.py.orig
+++ sphobjinv-2.3.1.3/tests/test_cli_nonlocal.py
@@ -45,7 +45,6 @@
 pytestmark = [
     pytest.mark.cli,
     pytest.mark.nonloc,
-    pytest.mark.flaky(retries=2, delay=5),
 ]
 
 
--- sphobjinv-2.3.1.3/tests/test_api_good_nonlocal.py.orig
+++ sphobjinv-2.3.1.3/tests/test_api_good_nonlocal.py
@@ -37,7 +37,6 @@
 pytestmark = [
     pytest.mark.api,
     pytest.mark.nonloc,
-    pytest.mark.flaky(retries=2, delay=5),
 ]
 
 
