Index: trunk/capplet/vino-preferences.desktop.in.in
===================================================================
--- trunk/capplet/vino-preferences.desktop.in.in	(revision 1106)
+++ trunk/capplet/vino-preferences.desktop.in.in	(working copy)
@@ -1,5 +1,6 @@
 [Desktop Entry]
-_Name=Remote Desktop
+# SUN_BRANDING
+_Name=Desktop Sharing
 _Comment=Choose how other users can remotely view your desktop
 Exec=vino-preferences
 Icon=preferences-desktop-remote-desktop
