Index: trunk/server/vino-input.c
===================================================================
--- trunk/server/vino-input.c	(revision 1041)
+++ trunk/server/vino-input.c	(working copy)
@@ -1134,7 +1134,6 @@
 				   VinoServer *server)
 {
   GtkClipboard *cb;
-  gsize a, b;
   GtkTargetEntry targets[] = {
 				{"UTF8_STRING", 0, 0},
 				{"COMPOUND_TEXT", 0, 0},
@@ -1146,7 +1145,7 @@
     return;
 
   g_free (cb_str);
-  cb_str = g_convert (text, len, "utf-8", "iso8859-1", &a, &b, NULL);
+  cb_str = g_strdup (text);
   if (cb_str)
     {
       cb = gtk_clipboard_get_for_display (gdk_screen_get_display (screen),
Index: trunk/server/vino-server.c
===================================================================
--- trunk/server/vino-server.c	(revision 1041)
+++ trunk/server/vino-server.c	(working copy)
@@ -889,8 +889,7 @@
                           GdkEvent     *event,
                           VinoServer   *server)
 {
-  gchar *text, *out;
-  size_t a, b;
+  gchar *text;
 
   if (!server->priv->clients)
     return;
@@ -905,14 +904,9 @@
   if (!text)
     return;
 
-  out = g_convert (text, -1, "iso8859-1", "utf-8", &a, &b, NULL);
-  if (out)
-    {
-      rfbSendServerCutText (server->priv->rfb_screen,
-                            out,
-                            strlen (out));
-      g_free (out);
-    }
+  rfbSendServerCutText (server->priv->rfb_screen,
+                        text,
+                        strlen (text));
 
   g_free (text);
 }
