--- vino-2.32.2/server/libvncserver/rfbserver.c.orig	2013-01-02 19:49:45.591938225 +0530
+++ vino-2.32.2/server/libvncserver/rfbserver.c	2013-01-02 19:58:19.594866751 +0530
@@ -1519,7 +1519,8 @@ rfbSendBell(rfbScreenInfoPtr rfbScreen)
 
 
 /*
- * rfbSendServerCutText sends a ServerCutText message to all the clients.
+ * rfbSendServerCutText sends a ServerCutText message to all the authenticated
+ * clients.
  */
 
 void
@@ -1531,6 +1532,8 @@ rfbSendServerCutText(rfbScreenInfoPtr rf
 
     iterator = rfbGetClientIterator(rfbScreen);
     while ((cl = rfbClientIteratorNext(iterator)) != NULL) {
+	if (cl->state != RFB_NORMAL) /* Client is not authenticated, ignore. */
+		continue;
         sct.type = rfbServerCutText;
         sct.length = Swap32IfLE(len);
         if (WriteExact(cl, (char *)&sct,
