--- openjdk-jdk11u-jdk-11.0.10-9-1/make/autoconf/configure.orig	Sat May  1 14:54:56 2021
+++ openjdk-jdk11u-jdk-11.0.10-9-1/make/autoconf/configure	Sat May  1 14:55:04 2021
@@ -22,6 +22,8 @@
 # questions.
 #
 
+cd build
+
 if test "x$1" != xCHECKME; then
   echo "ERROR: Calling this wrapper script directly is not supported."
   echo "Use the 'configure' script in the top-level directory instead."
