$NetBSD$

Support SunOS/gcc.  XXX: retain solstudio support.

--- a/make/lib/Awt2dLibraries.gmk.orig	2019-01-08 09:40:28.000000000 +0000
+++ b/make/lib/Awt2dLibraries.gmk
@@ -384,7 +384,6 @@ $(eval $(call SetupJdkLibrary, BUILD_LIB
     OPTIMIZATION := HIGHEST, \
     CFLAGS := $(LCMS_CFLAGS_JDKLIB) \
         $(LCMS_CFLAGS), \
-    CFLAGS_solaris := -xc99=no_lib, \
     CFLAGS_windows := -DCMS_IS_WINDOWS_, \
     EXTRA_HEADER_DIRS := \
         common/awt/debug \
@@ -554,7 +553,11 @@ ifneq (, $(findstring $(OPENJDK_TARGET_O
     HARFBUZZ_CFLAGS += -DHAVE_INTEL_ATOMIC_PRIMITIVES -DHB_NO_VISIBILITY
   endif
   ifeq ($(call isTargetOs, solaris), true)
+   ifeq ($(TOOLCHAIN_TYPE), gcc)
+    HARFBUZZ_CFLAGS += -DHAVE_INTEL_ATOMIC_PRIMITIVES -DHB_NO_VISIBILITY
+   else
     HARFBUZZ_CFLAGS += -DHAVE_SOLARIS_ATOMIC_OPS
+   endif
   endif
   # hb-ft.cc is not presently needed, and requires freetype 2.4.2 or later.
   LIBFONTMANAGER_EXCLUDE_FILES += libharfbuzz/hb-ft.cc
