diff -Nru jdk17u-jdk-17.0.4-8.orig/make/Install.gmk jdk17u-jdk-17.0.4-8/make/Install.gmk
--- jdk17u-jdk-17.0.4-8.orig/make/Install.gmk	2022-08-16 18:22:12.399540371 +0000
+++ jdk17u-jdk-17.0.4-8/make/Install.gmk	2022-08-16 18:23:06.601016006 +0000
@@ -33,6 +33,9 @@
 # Install the jdk image, in a very crude way. Not taking into
 # account, how to install properly on macosx or windows etc.
 install:
+	echo Dummy
+
+installRude:
 	echo Installing jdk image into $(INSTALL_PREFIX)/jvm/$(INSTALLDIR)
 	echo and creating $(words $(BINARIES)) links from $(INSTALL_PREFIX)/bin into the jdk.
 	$(MKDIR) -p $(INSTALL_PREFIX)/jvm/$(INSTALLDIR)
