$NetBSD$

Support SunOS/gcc.

--- a/make/lib/Lib-jdk.crypto.ec.gmk.orig	2019-01-08 09:40:29.000000000 +0000
+++ b/make/lib/Lib-jdk.crypto.ec.gmk
@@ -38,6 +38,12 @@ ifeq ($(ENABLE_INTREE_EC), true)
     BUILD_LIBSUNEC_CXXFLAGS_JDKLIB := $(CXXFLAGS_JDKLIB)
   endif
 
+  ifeq ($(OPENJDK_TARGET_OS), solaris)
+    ifeq ($(TOOLCHAIN_TYPE), gcc)
+      BUILD_LIBSUNEC_FLAGS += -D__EXTENSIONS__
+    endif
+  endif
+
   $(eval $(call SetupJdkLibrary, BUILD_LIBSUNEC, \
       NAME := sunec, \
       TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
