$NetBSD$

Support for SunOS/gcc.

--- a/make/launcher/LauncherCommon.gmk.orig	2019-01-08 09:40:28.000000000 +0000
+++ b/make/launcher/LauncherCommon.gmk
@@ -44,7 +44,9 @@ endif
 # Also provide an override for non-conformant libraries.
 ifeq ($(TOOLCHAIN_TYPE), gcc)
   LAUNCHER_CFLAGS += -fvisibility=hidden
+ ifneq ($(OPENJDK_TARGET_OS), solaris)
   LDFLAGS_JDKEXE += -Wl,--exclude-libs,ALL
+ endif
 else ifeq ($(TOOLCHAIN_TYPE), clang)
   ifneq ($(OPENJDK_TARGET_OS), macosx)
     LAUNCHER_CFLAGS += -fvisibility=hidden
