--- a/src/java.desktop/share/classes/javax/swing/UIManager.java.orig	Mon Nov 27 19:15:37 2023
+++ b/src/java.desktop/share/classes/javax/swing/UIManager.java	Mon Nov 27 19:45:00 2023
@@ -657,9 +657,12 @@
         if (osType == OSInfo.OSType.WINDOWS) {
             return "com.sun.java.swing.plaf.windows.WindowsLookAndFeel";
         } else {
+            String xdgCurrentDesktop = System.getenv("XDG_CURRENT_DESKTOP");
             String desktop = AccessController.doPrivileged(new GetPropertyAction("sun.desktop"));
             Toolkit toolkit = Toolkit.getDefaultToolkit();
-            if ("gnome".equals(desktop) &&
+            if (("mate".equalsIgnoreCase(xdgCurrentDesktop) || 
+                    "gnome".equalsIgnoreCase(xdgCurrentDesktop) || 
+                    "gnome".equals(desktop)) &&
                     toolkit instanceof SunToolkit &&
                     ((SunToolkit) toolkit).isNativeGTKAvailable()) {
                 // May be set on Linux and Solaris boxs.
