diff -wpruN '--exclude=*.orig' a~/src/hotspot/os/solaris/attachListener_solaris.cpp a/src/hotspot/os/solaris/attachListener_solaris.cpp
--- a~/src/hotspot/os/solaris/attachListener_solaris.cpp	1970-01-01 00:00:00
+++ a/src/hotspot/os/solaris/attachListener_solaris.cpp	1970-01-01 00:00:00
@@ -422,7 +422,7 @@ int SolarisAttachListener::create_door()
     if (::rename(initial_path, door_path) == -1) {
         ::close(dd);
         ::fdetach(initial_path);
-        log_debug(attach)("unable to create door - rename %s to %s failed (%d)", errno);
+        log_debug(attach)("unable to create door - rename %s to %s failed (%d)", initial_path, door_path, errno);
         dd = -1;
     }
   }
