--- a/make/autoconf/flags-cflags.m4~	Wed Apr  8 19:47:07 2020
+++ b/make/autoconf/flags-cflags.m4	Thu Apr  9 17:26:51 2020
@@ -499,8 +499,8 @@
   fi
 
   if test "x$TOOLCHAIN_TYPE" = xgcc; then
-    TOOLCHAIN_CFLAGS_JVM="$TOOLCHAIN_CFLAGS_JVM -fcheck-new -fstack-protector"
-    TOOLCHAIN_CFLAGS_JDK="-pipe -fstack-protector"
+    TOOLCHAIN_CFLAGS_JVM="$TOOLCHAIN_CFLAGS_JVM -fcheck-new"
+    TOOLCHAIN_CFLAGS_JDK="-pipe"
     # reduce lib size on s390x in link step, this needs also special compile flags
     if test "x$OPENJDK_TARGET_CPU" = xs390x; then
       TOOLCHAIN_CFLAGS_JVM="$TOOLCHAIN_CFLAGS_JVM -ffunction-sections -fdata-sections"
