diff -wpruN '--exclude=*.orig' a~/make/launcher/LauncherCommon.gmk a/make/launcher/LauncherCommon.gmk
--- a~/make/launcher/LauncherCommon.gmk	1970-01-01 00:00:00
+++ a/make/launcher/LauncherCommon.gmk	1970-01-01 00:00:00
@@ -178,7 +178,7 @@ define SetupBuildLauncherBody
           -DLAUNCHER_NAME='"$$(LAUNCHER_NAME)"' \
           -DPROGNAME='"$1"' \
           $$($1_CFLAGS), \
-      CFLAGS_solaris := -KPIC, \
+      CFLAGS_solaris := -fPIC, \
       CFLAGS_windows := $$($1_CFLAGS_windows), \
       LDFLAGS := $$(LDFLAGS_JDKEXE) \
           $$(ORIGIN_ARG) \
