diff -wpruN '--exclude=*.orig' a~/make/launcher/Launcher-jdk.pack.gmk a/make/launcher/Launcher-jdk.pack.gmk
--- a~/make/launcher/Launcher-jdk.pack.gmk	1970-01-01 00:00:00
+++ a/make/launcher/Launcher-jdk.pack.gmk	1970-01-01 00:00:00
@@ -86,7 +86,7 @@ $(eval $(call SetupJdkExecutable, BUILD_
     CFLAGS := $(UNPACKEXE_CFLAGS) $(CXXFLAGS_JDKEXE) -DFULL, \
     CFLAGS_release := -DPRODUCT, \
     CFLAGS_linux := -fPIC, \
-    CFLAGS_solaris := -KPIC, \
+    CFLAGS_solaris := -fPIC, \
     CFLAGS_macosx := -fPIC, \
     LDFLAGS := $(LDFLAGS_JDKEXE) $(LDFLAGS_CXX_JDK) \
         $(call SET_SHARED_LIBRARY_ORIGIN), \
