--- a/make/lib/Lib-jdk.hotspot.agent.gmk~	Mon Apr 13 16:53:05 2020
+++ b/make/lib/Lib-jdk.hotspot.agent.gmk	Wed Apr 15 13:54:06 2020
@@ -63,7 +63,7 @@
     EXTRA_SRC := $(LIBSA_EXTRA_SRC), \
     LDFLAGS := $(LDFLAGS_JDKLIB) $(SA_LDFLAGS), \
     LIBS_linux := $(LIBDL), \
-    LIBS_solaris := -ldl -ldemangle -lthread -lproc, \
+    LIBS_solaris := -ldl -lstdc++ -lthread -lproc, \
     LIBS_macosx := -framework Foundation \
         -framework JavaRuntimeSupport -framework Security -framework CoreFoundation, \
     LIBS_windows := dbgeng.lib, \
