--- a/make/launcher/LauncherCommon.gmk~	Thu Apr  9 18:18:09 2020
+++ b/make/launcher/LauncherCommon.gmk	Tue Nov 26 00:55:54 2019
@@ -178,7 +178,7 @@
           -DLAUNCHER_NAME='"$$(LAUNCHER_NAME)"' \
           -DPROGNAME='"$1"' \
           $$($1_CFLAGS), \
-      CFLAGS_solaris := -KPIC, \
+      CFLAGS_solaris := -fPIC, \
       CFLAGS_windows := $$($1_CFLAGS_windows), \
       LDFLAGS := $$(LDFLAGS_JDKEXE) \
           $$(ORIGIN_ARG) \
