--- elinks-0.11.7/Makefile.lib.orig   Wed Jan 26 13:36:42 2011
+++ elinks-0.11.7/Makefile.lib        Wed Jan 26 13:37:29 2011
@@ -39,7 +39,7 @@
 
 quiet_cmd_compile = '      [$(CC_COLOR)CC$(END_COLOR)]   $(RELPATH)$@'
  masq_cmd_compile = $(COMPILE) -o $(@) -c $< $(2)
-      cmd_compile = $(COMPILE) -o $(@) -Wp,-MD,.deps/$(*F).pp -c $< $(2)
+      cmd_compile = $(COMPILE) -o $(@) -c $< $(2)
 
 # Rule to compile a set of .o files into one .o file
 quiet_cmd_ld_objs = "      [$(LD_COLOR)LD$(END_COLOR)]   $(RELPATH)$@"

