--- exiv2-0.27.4/cmake/mainSetup.cmake.orig	2021-04-09 19:21:40.000000000 +0000
+++ exiv2-0.27.4/cmake/mainSetup.cmake	2021-05-21 22:49:15.935070998 +0000
@@ -23,8 +23,8 @@
     if (APPLE)
         set(CMAKE_MACOSX_RPATH ON)
         set(CMAKE_INSTALL_RPATH "@loader_path")
-    else()
-        join_paths(CMAKE_INSTALL_RPATH "$ORIGIN" ".." "${CMAKE_INSTALL_LIBDIR}")
+        #    else()
+        #        join_paths(CMAKE_INSTALL_RPATH "$ORIGIN" ".." "${CMAKE_INSTALL_LIBDIR}")
     endif()
 endif()
 
