--- exiv2-0.27.3-Source/cmake/mainSetup.cmake.1	2020-08-01 13:01:24.421051179 +0000
+++ exiv2-0.27.3-Source/cmake/mainSetup.cmake	2020-08-01 13:01:40.862593461 +0000
@@ -22,8 +22,8 @@
     if (APPLE)
         set(CMAKE_MACOSX_RPATH ON)
         set(CMAKE_INSTALL_RPATH "@loader_path")
-    else()
-        set(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
+#    else()
+#        set(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
     endif()
 endif()
 
