--- exiv2-0.25/src/Makefile.~1~	2015-06-21 17:20:11.000000000 +0000
+++ exiv2-0.25/src/Makefile	2018-01-29 22:53:38.767986956 +0000
@@ -249,11 +249,11 @@
 	@touch lib
 
 $(BINARY): %: %.o lib
-	@$(LIBTOOL) --mode=link $(LINK.cc) -o $@ $(LIBRARY) $@.o -rpath $(libdir)
+	@$(LIBTOOL) --mode=link $(LINK.cc) -o $@ $(LIBRARY) $@.o
 
 $(EXIV2BIN): lib $(EXIV2OBJ) $(EXIV2COBJ)
 	mkdir -pv ../bin 2>&1 > /dev/null
-	@$(LIBTOOL) --mode=link $(LINK.cc) -o ../bin/$@ $(LIBRARY) $(EXIV2OBJ) $(EXIV2COBJ) -rpath $(libdir)
+	@$(LIBTOOL) --mode=link $(LINK.cc) -o ../bin/$@ $(LIBRARY) $(EXIV2OBJ) $(EXIV2COBJ)
 
 install-header:
 	$(INSTALL_DIRS) $(DESTDIR)$(incdir)
