--- fs-uae-3.1.66.orig/configure.ac	So. Dez. 19 04:24:42 2021
+++ fs-uae-3.1.66/configure.ac	So. Juni 12 13:37:51 2022
@@ -390,6 +390,20 @@
     OS_CPPFLAGS="-DDRAGONFLYBSD"
     LOADABLE_MODULE_EXT=".so"
 ],
+[sunos*], [
+    OS_NAME="sunos"
+    OS_NAME_2="SunOS"
+    OS_NAME_3="SunOS"
+    OS_CPPFLAGS="-DSOLARIS"
+    LOADABLE_MODULE_EXT=".so"
+],
+[solaris*], [
+    OS_NAME="solaris"
+    OS_NAME_2="Solaris"
+    OS_NAME_3="Solaris"
+    OS_CPPFLAGS="-DSOLARIS"
+    LOADABLE_MODULE_EXT=".so"
+],
 [
     OS_NAME="unknown"
     OS_NAME_2="Unknown OS"
@@ -800,6 +814,7 @@
 
 AC_CONFIG_FILES([
     dist/linux/Makefile
+    dist/solaris/Makefile
     dist/macos/Info.plist
     dist/macos/Makefile
     dist/steamos/Makefile
