--- fs-uae-3.1.66.orig/src/slirp/ip_icmp.h	So. Dez. 19 04:24:39 2021
+++ fs-uae-3.1.66/src/slirp/ip_icmp.h	Di. Juni 14 08:24:46 2022
@@ -157,9 +157,9 @@
 
 void icmp_init(void);
 void icmp_cleanup(void);
-void icmp_input _P((struct mbuf *, int));
-void icmp_error _P((struct mbuf *, u_char, u_char, int, const char *));
-void icmp_reflect _P((struct mbuf *));
+void icmp_input (struct mbuf *, int);
+void icmp_error (struct mbuf *, u_char, u_char, int, const char *);
+void icmp_reflect (struct mbuf *);
 void icmp_receive(struct socket *so);
 void icmp_detach(struct socket *so);
 
