--- fs-uae-3.1.66.orig/src/slirp/misc.h	So. Dez. 19 04:24:39 2021
+++ fs-uae-3.1.66/src/slirp/misc.h	Di. Juni 14 08:39:28 2022
@@ -19,15 +19,15 @@
 extern struct ex_list *exec_list;
 extern u_int curtime, time_fasttimo, last_slowtimo, detach_time, detach_wait;
 
-extern int (*lprint_print) _P((void *, const char *, va_list));
+extern int (*lprint_print) (void *, const char *, va_list);
 extern char *lprint_ptr, *lprint_ptr2, **lprint_arg;
 extern struct sbuf *lprint_sb;
 
 #ifndef HAVE_STRDUP
-char *strdup _P((const char *));
+char *strdup (const char *);
 #endif
 
-void do_wait _P((int));
+void do_wait (int);
 
 #define EMU_NONE 0x0
 
@@ -67,21 +67,21 @@
 
 extern int x_port, x_server, x_display;
 
-int show_x _P((char *, struct socket *));
-void redir_x _P((u_int32_t, int, int, int));
-void getouraddr _P((void));
-void slirp_insque  _P((void *, void *));
-void slirp_remque  _P((void *));
-int add_exec _P((struct ex_list **, int, char *, int, int));
-int slirp_openpty _P((int *, int *));
-int fork_exec _P((struct socket *, char *, int));
-void snooze_hup _P((int));
-void snooze _P((void));
-void relay _P((int));
-void add_emu _P((char *));
-void u_sleep _P((int));
-void fd_nonblock _P((SLIRP_SOCKET));
-void fd_block _P((SLIRP_SOCKET));
-int rsh_exec _P((struct socket *, struct socket *, char *, char *, char *));
+int show_x (char *, struct socket *);
+void redir_x (u_int32_t, int, int, int);
+void getouraddr (void);
+void slirp_insque  (void *, void *);
+void slirp_remque  (void *);
+int add_exec (struct ex_list **, int, char *, int, int);
+int slirp_openpty (int *, int *);
+int fork_exec (struct socket *, char *, int);
+void snooze_hup (int);
+void snooze (void);
+void relay (int);
+void add_emu (char *);
+void u_sleep (int);
+void fd_nonblock (SLIRP_SOCKET);
+void fd_block (SLIRP_SOCKET);
+int rsh_exec (struct socket *, struct socket *, char *, char *, char *);
 
 #endif
