--- fs-uae-3.1.66.orig/src/slirp/tcp_timer.h	So. Dez. 19 04:24:39 2021
+++ fs-uae-3.1.66/src/slirp/tcp_timer.h	Di. Juni 14 08:21:19 2022
@@ -130,9 +130,9 @@
 
 struct tcpcb;
 
-void tcp_fasttimo _P((void));
-void tcp_slowtimo _P((void));
-void tcp_canceltimers _P((struct tcpcb *));
-struct tcpcb * tcp_timers _P((register struct tcpcb *, int));
+void tcp_fasttimo (void);
+void tcp_slowtimo (void);
+void tcp_canceltimers (struct tcpcb *);
+struct tcpcb * tcp_timers (register struct tcpcb *, int);
 
 #endif
