--- fs-uae-3.1.66.orig/src/slirp/udp.h	So. Dez. 19 04:24:39 2021
+++ fs-uae-3.1.66/src/slirp/udp.h	Di. Juni 14 08:22:43 2022
@@ -93,15 +93,15 @@
 extern struct socket udb;
 struct mbuf;
 
-void udp_init _P((void));
-void udp_cleanup _P((void));
-void udp_input _P((register struct mbuf *, int));
-int udp_output _P((struct socket *, struct mbuf *, struct sockaddr_in *));
-SLIRP_SOCKET udp_attach _P((struct socket *));
-void udp_detach _P((struct socket *));
-u_int8_t udp_tos _P((struct socket *));
-void udp_emu _P((struct socket *, struct mbuf *));
-struct socket * udp_listen _P((u_int, u_int32_t, u_int, int));
+void udp_init (void);
+void udp_cleanup (void);
+void udp_input (register struct mbuf *, int);
+int udp_output (struct socket *, struct mbuf *, struct sockaddr_in *);
+SLIRP_SOCKET udp_attach (struct socket *);
+void udp_detach (struct socket *);
+u_int8_t udp_tos (struct socket *);
+void udp_emu (struct socket *, struct mbuf *);
+struct socket * udp_listen (u_int, u_int32_t, u_int, int);
 int udp_output2(struct socket *so, struct mbuf *m, 
                 struct sockaddr_in *saddr, struct sockaddr_in *daddr,
                 int iptos);
