--- argon2_cffi-23.1.0/tox.ini.orig
+++ argon2_cffi-23.1.0/tox.ini
@@ -16,14 +16,14 @@
 package = wheel
 wheel_build_env = .pkg
 extras =
-    tests: tests
+    tests
     mypy: typing
 pass_env =
     FORCE_COLOR
     NO_COLOR
 commands =
-    tests: pytest {posargs}
-    tests: python -Im argon2 -n 1 -t 1 -m 8 -p 1
+    pytest {posargs}
+    python -Im argon2 -n 1 -t 1 -m 8 -p 1
     mypy: mypy tests/typing
 
 
