diff -Nur geeqie-1.1-orig/src/editors.c geeqie-1.1/src/editors.c
--- geeqie-1.1-orig/src/editors.c	2012-08-12 22:13:41.000000000 +0200
+++ geeqie-1.1/src/editors.c	2013-01-27 17:01:45.986293363 +0100
@@ -759,7 +759,7 @@
 	if (output)
 		result = g_string_new("");
 
-	if (editor->exec[0] == '\0')
+	if (editor->exec == NULL || editor->exec[0] == '\0')
 		{
 		flags |= EDITOR_ERROR_EMPTY;
 		goto err;
