--- firefox-43.0b4/ipc/app/moz.build.1	2015-11-23 14:10:51.595534157 +0300
+++ firefox-43.0b4/ipc/app/moz.build	2015-11-23 14:11:02.066488370 +0300
@@ -131,7 +131,6 @@
          '$(DEPTH)/security/nss/lib/nss/libnss3.so',
          '$(DEPTH)/config/external/sqlite/libmozsqlite3.so',
          '$(DEPTH)/config/external/lgpllibs/liblgpllibs.so',
-         '-lsendfile',
-         '-last'
+         '-lsendfile'
    ]
 
--- firefox-43.0b4/js/xpconnect/shell/moz.build.1	2015-11-23 15:32:07.587916209 +0300
+++ firefox-43.0b4/js/xpconnect/shell/moz.build	2015-11-23 15:32:14.437920731 +0300
@@ -48,6 +48,5 @@
          '$(DEPTH)/security/nss/lib/nss/libnss3.so',
          '$(DEPTH)/config/external/sqlite/libmozsqlite3.so',
 	 '$(DEPTH)/config/external/lgpllibs/liblgpllibs.so',
-         '-lsendfile',
-         '-last'
+         '-lsendfile'
    ]
