https://www.illumos.org/issues/5876

--- firefox-43.0/js/src/asmjs/AsmJSSignalHandlers.cpp.1	2015-12-16 09:24:58.022749462 +0300
+++ firefox-43.0/js/src/asmjs/AsmJSSignalHandlers.cpp	2015-12-16 09:26:10.494605469 +0300
@@ -215,6 +215,11 @@
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 # include <sys/ucontext.h> // for ucontext_t, mcontext_t
 #endif
+#if defined(SOLARIS)
+# include <ucontext.h> // for ucontext_t, etc
+# include <sys/regset.h>   // for REG_PC, etc
+#endif
+
 
 #if defined(JS_CPU_X64)
 # if defined(__DragonFly__)
