diff --git a/js/src/jit/RegisterSets.h b/js/src/jit/RegisterSets.h
--- a/js/src/jit/RegisterSets.h
+++ b/js/src/jit/RegisterSets.h
@@ -682,6 +682,8 @@
     AnyRegister reg() const { return kind_ == GPR ? AnyRegister(gpr()) : AnyRegister(fpu()); }
 };
 
+JS_STATIC_ASSERT(jit::AnyRegister::Total < UINT8_MAX);
+
 class AsmJSHeapAccess
 {
     uint32_t offset_;
@@ -692,8 +694,6 @@
     uint8_t isFloat32Load_;
     jit::AnyRegister::Code loadedReg_ : 8;
 
-    JS_STATIC_ASSERT(jit::AnyRegister::Total < UINT8_MAX);
-
   public:
 #if defined(JS_CPU_X86)
     AsmJSHeapAccess(uint32_t cmp, uint32_t offset, uint32_t after, ArrayBufferView::ViewType vt,
