--- mozilla-esr10/xpcom/glue/Makefile.in.~1~	2013-01-03 21:44:00.000000000 +0400
+++ mozilla-esr10/xpcom/glue/Makefile.in	2013-12-18 19:44:32.632530413 +0400
@@ -44,6 +44,7 @@
 include $(srcdir)/objs.mk
 
 EXTRA_DEPS += $(srcdir)/objs.mk
+VISIBILITY_FLAGS =
 
 DIRS            = standalone nomozalloc
 
diff --git a/xpcom/build/Makefile.in b/xpcom/build/Makefile.in
index 2ff9f82..481d720 100644
--- a/xpcom/build/Makefile.in
+++ b/xpcom/build/Makefile.in
@@ -51,6 +51,7 @@
 SHORT_LIBNAME	= xpcomcor
 LIBXUL_LIBRARY = 1
 EXPORT_LIBRARY = 1
+VISIBILITY_FLAGS =
 
 GRE_MODULE	= 1
 MOZILLA_INTERNAL_API = 1
