diff --git a/gfx/2d/unittest/TestBase.h b/gfx/2d/unittest/TestBase.h
--- a/gfx/2d/unittest/TestBase.h
+++ b/gfx/2d/unittest/TestBase.h
@@ -4,6 +4,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef TestBase_h__
+#define TestBase_h__
 
 #include <string>
 #include <vector>
@@ -52,3 +54,4 @@
   // This doesn't really work with our generic member pointer trick.
   TestBase(const TestBase &aOther);
 };
+#endif
