diff --git a/dom/plugins/ipc/PluginInstanceChild.cpp b/dom/plugins/ipc/PluginInstanceChild.cpp
index a7b7653..1654acf 100644
--- a/dom/plugins/ipc/PluginInstanceChild.cpp
+++ b/dom/plugins/ipc/PluginInstanceChild.cpp
@@ -2747,7 +2747,7 @@ PluginInstanceChild::PaintRectToSurface(const nsIntRect& aRect,
         if (!mMaemoImageRendering ||
             renderSurface->GetType() != gfxASurface::SurfaceTypeImage)
 #endif
-        renderSurface = mHelperSurface;
+        renderSurface = mHelperSurface ? mHelperSurface : mCurrentSurface;
     }
 #endif
 
