diff --git a/netwerk/cache/nsCacheEntryDescriptor.h b/netwerk/cache/nsCacheEntryDescriptor.h
--- a/netwerk/cache/nsCacheEntryDescriptor.h
+++ b/netwerk/cache/nsCacheEntryDescriptor.h
@@ -74,12 +74,14 @@ private:
       *************************************************************************/
      class nsInputStreamWrapper : public nsIInputStream {
          friend class nsCacheEntryDescriptor;
-
-     private:
+       
+        public:
          nsCacheEntryDescriptor    * mDescriptor;
+     private:
          nsCOMPtr<nsIInputStream>    mInput;
          uint32_t                    mStartOffset;
          bool                        mInitialized;
+        public:
          mozilla::Mutex              mLock;
      public:
          NS_DECL_ISUPPORTS
@@ -98,11 +100,11 @@
              NS_IF_RELEASE(mDescriptor);
          }
 
+         nsresult Read_Locked(char *buf, uint32_t count, uint32_t *countRead);
+         nsresult Close_Locked();
      private:
          nsresult LazyInit();
          nsresult EnsureInit();
-         nsresult Read_Locked(char *buf, uint32_t count, uint32_t *countRead);
-         nsresult Close_Locked();
          void CloseInternal();
      };
 
@@ -173,14 +175,14 @@
              NS_ASSERTION(!mDescriptor, "Bad state");
          }
 
+         nsresult Write_Locked(const char * buf,
+                               uint32_t count,
+                               uint32_t * result);
+         nsresult Close_Locked();
      private:
          nsresult LazyInit();
          nsresult EnsureInit();
          nsresult OnWrite(uint32_t count);
-         nsresult Write_Locked(const char * buf,
-                               uint32_t count,
-                               uint32_t * result);
-         nsresult Close_Locked();
          void CloseInternal();
      };
 
