--- a/toolkit/library/Makefile.in.old	2013-05-30 16:50:42.336042009 +0530
+++ b/toolkit/library/Makefile.in	2013-05-30 16:51:42.837462375 +0530
@@ -498,7 +498,7 @@ endif
 endif
 endif
 
-EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS) $(MKSHLIB_FORCE_ALL) $(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(MKSHLIB_UNFORCE_ALL)
+EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS) $(MKSHLIB_FORCE_ALL) ../../mfbt/libmfbt.a $(LIBXUL_DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) $(MKSHLIB_UNFORCE_ALL)
 
 
 DEFINES += -DIMPL_XREAPI
