diff --git a/gfx/skia/include/gpu/GrSurface.h b/gfx/skia/include/gpu/GrSurface.h
--- a/gfx/skia/include/gpu/GrSurface.h
+++ b/gfx/skia/include/gpu/GrSurface.h
@@ -12,17 +12,17 @@
 #include "GrTypes.h"
 #include "GrResource.h"
 
 class GrTexture;
 class GrRenderTarget;
 
 class GrSurface : public GrResource {
 public:
-    SK_DECLARE_INST_COUNT(GrSurface);
+    SK_DECLARE_INST_COUNT(GrSurface)
 
     /**
      * Retrieves the width of the surface.
      *
      * @return the width in texels
      */
     int width() const { return fDesc.fWidth; }
 
diff --git a/gfx/skia/src/gpu/GrGeometryBuffer.h b/gfx/skia/src/gpu/GrGeometryBuffer.h
--- a/gfx/skia/src/gpu/GrGeometryBuffer.h
+++ b/gfx/skia/src/gpu/GrGeometryBuffer.h
@@ -14,17 +14,17 @@
 
 class GrGpu;
 
 /**
  * Parent class for vertex and index buffers
  */
 class GrGeometryBuffer : public GrResource {
 public:
-    SK_DECLARE_INST_COUNT(GrGeometryBuffer);
+    SK_DECLARE_INST_COUNT(GrGeometryBuffer)
 
     /**
      *Retrieves whether the buffer was created with the dynamic flag
      *
      * @return true if the buffer was created with the dynamic flag
      */
     bool dynamic() const { return fDynamic; }
 
diff --git a/gfx/skia/include/gpu/GrContext.h b/gfx/skia/include/gpu/GrContext.h
--- a/gfx/skia/include/gpu/GrContext.h
+++ b/gfx/skia/include/gpu/GrContext.h
@@ -337,17 +337,17 @@ public:
 
     /**
      * Clear the entire or rect of the render target, ignoring any clips.
      * @param rect  the rect to clear or the whole thing if rect is NULL.
      * @param color the color to clear to.
      * @param target if non-NULL, the render target to clear otherwise clear
      *               the current render target
      */
-    void clear(const GrIRect* rect, GrColor color,
+    void clear(const GrIRect* rect, const GrColor color,
                GrRenderTarget* target = NULL);
 
     /**
      *  Draw everywhere (respecting the clip) with the paint.
      */
     void drawPaint(const GrPaint& paint);
 
     /**
diff --git a/gfx/skia/src/gpu/GrPath.h b/gfx/skia/src/gpu/GrPath.h
--- a/gfx/skia/src/gpu/GrPath.h
+++ b/gfx/skia/src/gpu/GrPath.h
@@ -14,7 +14,7 @@
 
 class GrPath : public GrResource {
 public:
-    SK_DECLARE_INST_COUNT(GrPath);
+    SK_DECLARE_INST_COUNT(GrPath)
 
     GrPath(GrGpu* gpu) : INHERITED(gpu) {}
 
diff --git a/gfx/skia/src/gpu/GrStencilBuffer.h b/gfx/skia/src/gpu/GrStencilBuffer.h
--- a/gfx/skia/src/gpu/GrStencilBuffer.h
+++ b/gfx/skia/src/gpu/GrStencilBuffer.h
@@ -20,7 +20,7 @@ class GrResourceKey;
 
 class GrStencilBuffer : public GrResource {
 public:
-    SK_DECLARE_INST_COUNT(GrStencilBuffer);
+    SK_DECLARE_INST_COUNT(GrStencilBuffer)
     GR_DECLARE_RESOURCE_CACHE_TYPE()
 
     virtual ~GrStencilBuffer() {
diff --git a/gfx/skia/src/gpu/effects/GrTextureStripAtlas.h b/gfx/skia/src/gpu/effects/GrTextureStripAtlas.h
--- a/gfx/skia/src/gpu/effects/GrTextureStripAtlas.h
+++ b/gfx/skia/src/gpu/effects/GrTextureStripAtlas.h
@@ -117,6 +117,7 @@
      **/
     int searchByKey(uint32_t key);
 
+public:
     /**
      * Compare two atlas rows by key, so we can sort/search by key
      */
@@ -124,6 +125,8 @@
         return lhs->fKey - rhs->fKey;
     }
 
+private:
+
 #ifdef SK_DEBUG
     void validate();
 #endif
