--- a/js/src/configure	2012-07-19 10:31:29.007161000 +0800
+++ b/js/src/configure	2012-07-19 10:31:43.807921000 +0800
@@ -1180,7 +1180,7 @@
   fi
 
     fi
-    IS_GCC=$($CC -v 2>&1 | grep gcc)
+    IS_GCC=`$CC -v 2>&1 | grep gcc`
     if test -n "$IS_GCC"
     then
       echo gcc is known to be broken on OS X, please use clang.
@@ -4126,7 +4126,7 @@
 for base in $MOZILLA_CENTRAL_PATH $_topsrcdir $_topsrcdir/../..; do
   possible=$base/build/virtualenv/populate_virtualenv.py
 
-  if test -e $possible; then
+  if test -f $possible; then
     _virtualenv_topsrcdir=$base
     _virtualenv_populate_path=$possible
     break
