diff --git a/netwerk/dash/mpd/AdaptationSet.cpp b/netwerk/dash/mpd/AdaptationSet.cpp
--- a/netwerk/dash/mpd/AdaptationSet.cpp
+++ b/netwerk/dash/mpd/AdaptationSet.cpp
@@ -107,7 +107,7 @@
 }
 
 void
-AdaptationSet::EnableBitstreamSwitching(bool aEnable)
+AdaptationSet::EnableBitstreamSwitching(const bool aEnable)
 {
   mIsBitstreamSwitching = aEnable;
 }
diff --git a/netwerk/dash/mpd/Representation.cpp b/netwerk/dash/mpd/Representation.cpp
--- a/netwerk/dash/mpd/Representation.cpp
+++ b/netwerk/dash/mpd/Representation.cpp
@@ -53,7 +53,7 @@
 }
 
 void
-Representation::SetBitrate(int64_t aBitrate)
+Representation::SetBitrate(const int64_t aBitrate)
 {
   mBitrate = aBitrate;
 }
@@ -71,7 +71,7 @@
 }
 
 void
-Representation::SetHeight(int32_t aHeight)
+Representation::SetHeight(const int32_t aHeight)
 {
   mHeight = aHeight;
 }
