--- a/js/src/ctypes/CTypes.h	2013-12-07 08:33:10.624155000 +0800
+++ b/js/src/ctypes/CTypes.h	2013-12-07 08:33:27.713041000 +0800
@@ -60,8 +60,8 @@
 template<class T, size_t N = 0>
 class Array : public Vector<T, N, SystemAllocPolicy>
 {
-  MOZ_STATIC_ASSERT((!mozilla::IsSame<T, JS::Value>::value),
-                    "use JS::AutoValueVector instead");
+//  MOZ_STATIC_ASSERT((!mozilla::IsSame<T, JS::Value>::value),
+//                    "use JS::AutoValueVector instead");
 };
 
 // String and AutoString classes, based on Vector.
