diff --git a/widget/xpwidgets/GfxInfoX11.cpp b/widget/xpwidgets/GfxInfoX11.cpp
--- a/widget/xpwidgets/GfxInfoX11.cpp
+++ b/widget/xpwidgets/GfxInfoX11.cpp
@@ -12,6 +12,7 @@
 #include <sys/utsname.h>
 #include "nsCRTGlue.h"
 #include "prenv.h"
+#include "plstr.h"
 
 #include "GfxInfoX11.h"
 
@@ -210,13 +211,13 @@
         // with Mesa, the version string contains "Mesa major.minor" and that's all the version information we get:
         // there is no actual driver version info.
         whereToReadVersionNumbers = Mesa_in_version_string + strlen("Mesa");
-        if (strcasestr(mVendor.get(), "nouveau"))
+        if (PL_strcasestr(mVendor.get(), "nouveau"))
             mIsNouveau = true;
-        if (strcasestr(mRenderer.get(), "intel")) // yes, intel is in the renderer string
+        if (PL_strcasestr(mRenderer.get(), "intel")) // yes, intel is in the renderer string
             mIsIntel = true;
-        if (strcasestr(mRenderer.get(), "llvmpipe"))
+        if (PL_strcasestr(mRenderer.get(), "llvmpipe"))
             mIsLlvmpipe = true;
-        if (strcasestr(mRenderer.get(), "software rasterizer"))
+        if (PL_strcasestr(mRenderer.get(), "software rasterizer"))
             mIsOldSwrast = true;
     } else if (strstr(mVendor.get(), "NVIDIA Corporation")) {
         mIsNVIDIA = true;
