--- mozilla-esr10/content/media/ogg/Makefile.in.~1~	2013-12-17 18:24:47.083655596 +0400
+++ mozilla-esr10/content/media/ogg/Makefile.in	2013-12-17 18:25:18.787483891 +0400
@@ -44,6 +44,7 @@
 MODULE		= content
 LIBRARY_NAME	= gkconogg_s
 LIBXUL_LIBRARY 	= 1
+VISIBILITY_FLAGS =
 
 
 EXPORTS		+= \
--- mozilla-esr10/content/media/webm/Makefile.in.~1~	2013-12-17 18:23:32.693516182 +0400
+++ mozilla-esr10/content/media/webm/Makefile.in	2013-12-17 18:24:15.248491871 +0400
@@ -44,6 +44,7 @@
 MODULE		= content
 LIBRARY_NAME	= gkconwebm_s
 LIBXUL_LIBRARY 	= 1
+VISIBILITY_FLAGS =
 
 
 EXPORTS		+= \
