--- mozilla-esr24/js/src/jsiter.cpp.orig	2013-12-24 12:08:22.943858942 +0530
+++ mozilla-esr24/js/src/jsiter.cpp	2013-12-24 12:08:43.802936457 +0530
@@ -1493,7 +1493,7 @@ js_NewGenerator(JSContext *cx, const Fra
                     stackfp->script()->nslots) * sizeof(HeapValue);
 
     JS_ASSERT(nbytes % sizeof(Value) == 0);
-    JS_STATIC_ASSERT(sizeof(StackFrame) % sizeof(HeapValue) == 0);
+    //JS_STATIC_ASSERT(sizeof(StackFrame) % sizeof(HeapValue) == 0);
 
     JSGenerator *gen = (JSGenerator *) cx->calloc_(nbytes);
     if (!gen)
