--- mozilla-esr24/layout/media/Makefile.in.~1~	2014-07-06 13:17:20.241608647 +0400
+++ mozilla-esr24/layout/media/Makefile.in	2014-07-06 13:18:44.829565495 +0400
@@ -32,21 +32,17 @@
 
 ifdef MOZ_VORBIS
 SHARED_LIBRARY_LIBS 	+= \
-	$(DEPTH)/media/libvorbis/lib/$(LIB_PREFIX)vorbis.$(LIB_SUFFIX) \
-	$(DEPTH)/media/libogg/src/$(LIB_PREFIX)ogg.$(LIB_SUFFIX) \
 	$(NULL)
 endif
 
 ifdef MOZ_TREMOR
 SHARED_LIBRARY_LIBS	+= \
 	$(DEPTH)/media/libtremor/lib/$(LIB_PREFIX)tremor.$(LIB_SUFFIX) \
-	$(DEPTH)/media/libogg/src/$(LIB_PREFIX)ogg.$(LIB_SUFFIX) \
 	$(NULL)
 endif
 
 ifdef MOZ_OGG
 SHARED_LIBRARY_LIBS 	+= \
-	$(DEPTH)/media/libtheora/lib/$(LIB_PREFIX)theora.$(LIB_SUFFIX) \
 	$(NULL)
 endif
 
diff --git a/toolkit/library/Makefile.in b/toolkit/library/Makefile.in
index afdcb02..e5204b9 100644
--- a/toolkit/library/Makefile.in
+++ b/toolkit/library/Makefile.in
@@ -240,6 +240,10 @@ EXTRA_DSO_LDOPTS += -lbe -ltracker
 endif
 
 ifeq ($(OS_ARCH),SunOS)
+EXTRA_DSO_LDOPTS += -lvorbis -ltheora -logg
+endif
+
+ifeq ($(OS_ARCH),SunOS)
 ifdef GNU_CC
 EXTRA_DSO_LDOPTS += -lelf
 else
--- mozilla-esr24/toolkit/toolkit.mozbuild.1	2014-07-08 15:56:25.490821343 +0400
+++ mozilla-esr24/toolkit/toolkit.mozbuild	2014-07-08 15:57:01.210673969 +0400
@@ -64,9 +64,6 @@
 if CONFIG['MOZ_JSDEBUGGER']:
     add_tier_dir('platform', 'js/jsd')
 
-if CONFIG['MOZ_VORBIS']:
-    add_tier_dir('platform', 'media/libvorbis')
-
 if CONFIG['MOZ_TREMOR']:
     add_tier_dir('platform', 'media/libtremor')
 
@@ -81,9 +78,6 @@
 if CONFIG['MOZ_VP8'] and not CONFIG['MOZ_NATIVE_LIBVPX']:
     add_tier_dir('platform', 'media/libvpx')
 
-if CONFIG['MOZ_OGG']:
-    add_tier_dir('platform', ['media/libogg', 'media/libtheora'])
-
 if CONFIG['MOZ_WEBRTC']:
     add_tier_dir('platform', [
         'media/webrtc',
