diff --git a/accessible/src/base/nsAccessibilityService.cpp b/accessible/src/base/nsAccessibilityService.cpp
--- a/accessible/src/base/nsAccessibilityService.cpp
+++ b/accessible/src/base/nsAccessibilityService.cpp
@@ -656,6 +656,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 // nsAccessibilityService public
 
+#pragma weak "__1cWnsAccessibilityServiceVGetOrCreateAccessible6MpnHnsINode_pnHmozillaEa11yKAccessible_pb_6_" = "__1cWnsAccessibilityServiceVGetOrCreateAccessible6MpnHnsINode_pnHmozillaEa11yKAccessible_pb_p5_"
 Accessible*
 nsAccessibilityService::GetOrCreateAccessible(nsINode* aNode,
                                               Accessible* aContext,
