--- mozilla-esr24/xpcom/glue/standalone/nsXPCOMGlue.cpp.orig	2014-01-03 11:31:22.583321889 +0530
+++ mozilla-esr24/xpcom/glue/standalone/nsXPCOMGlue.cpp	2014-01-03 11:32:39.562717966 +0530
@@ -453,6 +453,8 @@ XPCOMGlueLoad(const char *xpcomFile)
         }
     }
 
+	static void* gthreadso = dlopen("libgthread-2.0.so.0", RTLD_GLOBAL | RTLD_LAZY);
+
     GetFrozenFunctionsFunc sym =
         (GetFrozenFunctionsFunc) GetSymbol(sTop->libHandle, LEADING_UNDERSCORE "NS_GetFrozenFunctions");
 
