--- mozilla-esr24/configure.~2~	2014-07-06 13:51:50.884024959 +0400
+++ mozilla-esr24/configure	2014-07-06 13:57:10.141533765 +0400
@@ -9593,8 +9593,8 @@
 fi
 
 if test "$GNU_CC"; then
-    CFLAGS="$CFLAGS -ffunction-sections -fdata-sections"
-    CXXFLAGS="$CXXFLAGS -ffunction-sections -fdata-sections -fno-exceptions"
+    CFLAGS="$CFLAGS"
+    CXXFLAGS="$CXXFLAGS -fno-exceptions"
 fi
 
 
@@ -9622,7 +9622,7 @@
               'int main() {return foo() - bar();}' > conftest.${ac_ext}
         # If the linker supports ICF, foo and bar symbols will have
         # the same address
-        if { ac_try='${CC-cc} -o conftest${ac_exeext} $LDFLAGS -Wl,--icf=safe -ffunction-sections conftest.${ac_ext} $LIBS 1>&2'; { (eval echo configure:9626: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }; } &&
+        if { ac_try='${CC-cc} -o conftest${ac_exeext} $LDFLAGS -Wl,--icf=safe conftest.${ac_ext} $LIBS 1>&2'; { (eval echo configure:9626: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }; } &&
            test -s conftest${ac_exeext} &&
            objdump -t conftest${ac_exeext} | awk '{a[$6] = $1} END {if (a["foo"] && (a["foo"] != a["bar"])) { exit 1 }}'; then
             LD_SUPPORTS_ICF=yes
