diff --git a/configure b/configure
index 19028c1..794c6f0 100755
--- a/configure
+++ b/configure
@@ -21032,7 +21031,7 @@ fi
 # Check whether --enable-update-channel or --disable-update-channel was given.
 if test "${enable_update_channel+set}" = set; then
   enableval="$enable_update_channel"
-  MOZ_UPDATE_CHANNEL=`echo $enableval | tr A-Z a-z`
+  MOZ_UPDATE_CHANNEL=`echo $enableval | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 
@@ -24121,7 +24120,7 @@ fi
 # Check whether --enable-chrome-format or --disable-chrome-format was given.
 if test "${enable_chrome_format+set}" = set; then
   enableval="$enable_chrome_format"
-  MOZ_CHROME_FILE_FORMAT=`echo $enableval | tr A-Z a-z`
+  MOZ_CHROME_FILE_FORMAT=`echo $enableval | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 
@@ -26253,7 +26252,7 @@ EOF
 # - MOZ_EXTENSION_MANAGER: When set, enabled extension manager.
 
 if test -z "$MOZ_APP_NAME"; then
-   MOZ_APP_NAME=`echo $MOZ_APP_BASENAME | tr A-Z a-z`
+   MOZ_APP_NAME=`echo $MOZ_APP_BASENAME | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 # For extensions and langpacks, we require a max version that is compatible
