diff --git a/mfbt/decimal/Decimal.cpp b/mfbt/decimal/Decimal.cpp
--- a/mfbt/decimal/Decimal.cpp
+++ b/mfbt/decimal/Decimal.cpp
@@ -33,6 +33,7 @@
 
 #include <algorithm>
 #include <float.h>
+#include <ieeefp.h>
 
 namespace WebCore {
 
@@ -684,13 +685,13 @@
 
 Decimal Decimal::fromDouble(double doubleValue)
 {
-    if (std::isfinite(doubleValue))
+    if (isnand(doubleValue))
+        return nan();
+
+    if (finite(doubleValue))
         return fromString(mozToString(doubleValue));
 
-    if (std::isinf(doubleValue))
-        return infinity(doubleValue < 0 ? Negative : Positive);
-
-    return nan();
+    return infinity(doubleValue < 0 ? Negative : Positive);
 }
 
 Decimal Decimal::fromString(const String& str)
