diff --git a/gfx/graphite2/include/graphite2/Font.h b/gfx/graphite2/include/graphite2/Font.h
--- a/gfx/graphite2/include/graphite2/Font.h
+++ b/gfx/graphite2/include/graphite2/Font.h
@@ -28,6 +28,9 @@
 
 #include "graphite2/Types.h"
 
+#ifndef GRAPHITE2_FONTS_H
+#define GRAPHITE2_FONTS_H
+
 #define GR2_VERSION_MAJOR   1
 #define GR2_VERSION_MINOR   2
 #define GR2_VERSION_BUGFIX  3
@@ -387,3 +390,4 @@
 }
 #endif
 
+#endif
diff --git a/gfx/graphite2/include/graphite2/Segment.h b/gfx/graphite2/include/graphite2/Segment.h
--- a/gfx/graphite2/include/graphite2/Segment.h
+++ b/gfx/graphite2/include/graphite2/Segment.h
@@ -26,9 +26,13 @@
 */
 #pragma once
 
+
 #include "graphite2/Types.h"
 #include "graphite2/Font.h"
 
+#ifndef GRAPHITE2_SEGMENT_H
+#define GRAPHITE2_SEGMENT_H
+
 #ifdef __cplusplus
 extern "C"
 {
@@ -398,3 +402,5 @@
 #ifdef __cplusplus
 }
 #endif
+
+#endif
diff --git a/gfx/graphite2/include/graphite2/Types.h b/gfx/graphite2/include/graphite2/Types.h
--- a/gfx/graphite2/include/graphite2/Types.h
+++ b/gfx/graphite2/include/graphite2/Types.h
@@ -26,6 +26,9 @@
 */
 #pragma once
 
+#ifndef GRAPHITE2_TYPES_H
+#define GRAPHITE2_TYPES_H
+
 #include <stddef.h>
 
 typedef unsigned char   gr_uint8;
@@ -67,3 +70,5 @@
     #define GR2_LOCAL
   #endif
 #endif
+
+#endif
diff --git a/gfx/graphite2/include/graphite2/XmlLog.h b/gfx/graphite2/include/graphite2/XmlLog.h
--- a/gfx/graphite2/include/graphite2/XmlLog.h
+++ b/gfx/graphite2/include/graphite2/XmlLog.h
@@ -25,6 +25,8 @@
     either version 2 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_XMLLOG_H
+#define GRAPHITE2_XMLLOG_H
 
 #include <graphite2/Types.h>
 #include <stdio.h>
@@ -53,3 +55,5 @@
 #ifdef __cplusplus
 }
 #endif
+
+#endif
diff --git a/gfx/graphite2/src/Bidi.cpp b/gfx/graphite2/src/Bidi.cpp
--- a/gfx/graphite2/src/Bidi.cpp
+++ b/gfx/graphite2/src/Bidi.cpp
@@ -30,6 +30,9 @@
 
 using namespace graphite2;
 
+#undef ES
+#undef CS
+
 enum DirCode {  // Hungarian: dirc
         Unk        = -1,
         N          =  0,   // other neutrals (default) - ON
diff --git a/gfx/graphite2/src/Pass.cpp b/gfx/graphite2/src/Pass.cpp
--- a/gfx/graphite2/src/Pass.cpp
+++ b/gfx/graphite2/src/Pass.cpp
@@ -72,7 +72,7 @@
     delete [] m_rules;
 }
 
-bool Pass::readPass(const byte * const pass_start, size_t pass_length, size_t subtable_base, GR_MAYBE_UNUSED const Face & face)
+bool Pass::readPass(const byte * pass_start, size_t pass_length, size_t subtable_base, GR_MAYBE_UNUSED const Face & face)
 {
     const byte *                p = pass_start,
                * const pass_end   = p + pass_length;
diff --git a/gfx/graphite2/src/inc/CharInfo.h b/gfx/graphite2/src/inc/CharInfo.h
--- a/gfx/graphite2/src/inc/CharInfo.h
+++ b/gfx/graphite2/src/inc/CharInfo.h
@@ -25,6 +25,9 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_CHARINFO_H
+#define GRAPHITE2_SRC_CHARINFO_H
+
 #include "inc/Main.h"
 
 
@@ -65,3 +68,5 @@
 
 struct gr_char_info : public graphite2::CharInfo {};
 
+#endif
+
diff --git a/gfx/graphite2/src/inc/Code.h b/gfx/graphite2/src/inc/Code.h
--- a/gfx/graphite2/src/inc/Code.h
+++ b/gfx/graphite2/src/inc/Code.h
@@ -30,6 +30,8 @@
 // Author: Tim Eves
 
 #pragma once
+#ifndef GRAPHITE2_SRC_CODE_H
+#define GRAPHITE2_SRC_CODE_H
 
 #include <cassert>
 #include <graphite2/Types.h>
@@ -173,3 +175,5 @@
 
 } // namespace vm
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/Face.h b/gfx/graphite2/src/inc/Face.h
--- a/gfx/graphite2/src/inc/Face.h
+++ b/gfx/graphite2/src/inc/Face.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_FACE_H
+#define GRAPHITE2_SRC_FACE_H
 
 #include <stdio.h>
 
@@ -204,3 +206,5 @@
 } // namespace graphite2
 
 struct gr_face : public graphite2::Face {};
+
+#endif
diff --git a/gfx/graphite2/src/inc/FeatureVal.h b/gfx/graphite2/src/inc/FeatureVal.h
--- a/gfx/graphite2/src/inc/FeatureVal.h
+++ b/gfx/graphite2/src/inc/FeatureVal.h
@@ -25,6 +25,9 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_FEATUREVAL_H
+#define GRAPHITE2_SRC_FEATUREVAL_H
+
 #include <cstring>
 #include <cassert>
 #include "inc/Main.h"
@@ -66,3 +69,5 @@
 
 
 struct gr_feature_val : public graphite2::FeatureVal {};
+
+#endif
diff --git a/gfx/graphite2/src/inc/Font.h b/gfx/graphite2/src/inc/Font.h
--- a/gfx/graphite2/src/inc/Font.h
+++ b/gfx/graphite2/src/inc/Font.h
@@ -25,6 +25,9 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_FONT_H
+#define GRAPHITE2_SRC_FONT_H
+
 #include <cassert>
 #include "graphite2/Font.h"
 #include "inc/Main.h"
@@ -87,3 +90,5 @@
 } // namespace graphite2
 
 struct gr_font : public graphite2::Font {};
+
+#endif
diff --git a/gfx/graphite2/src/inc/GlyphFace.h b/gfx/graphite2/src/inc/GlyphFace.h
--- a/gfx/graphite2/src/inc/GlyphFace.h
+++ b/gfx/graphite2/src/inc/GlyphFace.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_GLYPHFACE_H
+#define GRAPHITE2_SRC_GLYPHFACE_H
 
 #include "inc/Main.h"
 #include "inc/Position.h"
@@ -81,3 +83,5 @@
 }
 
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/GlyphFaceCache.h b/gfx/graphite2/src/inc/GlyphFaceCache.h
--- a/gfx/graphite2/src/inc/GlyphFaceCache.h
+++ b/gfx/graphite2/src/inc/GlyphFaceCache.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_GLYPHFACECACHE_H
+#define GRAPHITE2_SRC_GLYPHFACECACHE_H
 
 #include "inc/GlyphFace.h"
 #include "graphite2/Font.h"
@@ -101,3 +103,5 @@
 };
 
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/List.h b/gfx/graphite2/src/inc/List.h
--- a/gfx/graphite2/src/inc/List.h
+++ b/gfx/graphite2/src/inc/List.h
@@ -27,6 +27,8 @@
 
 // designed to have a limited subset of the std::vector api
 #pragma once
+#ifndef GRAPHITE2_SRC_LIST_H
+#define GRAPHITE2_SRC_LIST_H
 
 #include <cstddef>
 #include <cassert>
@@ -153,3 +155,5 @@
 }
 
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/Machine.h b/gfx/graphite2/src/inc/Machine.h
--- a/gfx/graphite2/src/inc/Machine.h
+++ b/gfx/graphite2/src/inc/Machine.h
@@ -35,6 +35,9 @@
 #include <graphite2/Types.h>
 #include "inc/Main.h"
 
+#ifndef GRAPHITE2_SRC_MACHINE_H
+#define GRAPHITE2_SRC_MACHINE_H
+
 #if defined(__GNUC__)
 #if defined(__clang__) || (__GNUC__ * 100 + __GNUC_MINOR__ * 10) < 430
 #define     HOT
@@ -193,5 +196,5 @@
 } // namespace vm
 } // namespace graphite2
 
+#endif
 
-
diff --git a/gfx/graphite2/src/inc/Main.h b/gfx/graphite2/src/inc/Main.h
--- a/gfx/graphite2/src/inc/Main.h
+++ b/gfx/graphite2/src/inc/Main.h
@@ -29,6 +29,9 @@
 #include <cstdlib>
 #include "graphite2/Types.h"
 
+#ifndef GRAPHITE2_SRC_MAIN_H
+#define GRAPHITE2_SRC_MAIN_H
+
 #ifdef GRAPHITE2_CUSTOM_HEADER
 #include GRAPHITE2_CUSTOM_HEADER
 #endif
@@ -73,3 +76,5 @@
 #else
 #define GR_MAYBE_UNUSED
 #endif
+
+#endif
diff --git a/gfx/graphite2/src/inc/Pass.h b/gfx/graphite2/src/inc/Pass.h
--- a/gfx/graphite2/src/inc/Pass.h
+++ b/gfx/graphite2/src/inc/Pass.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_PASS_H
+#define GRAPHITE2_SRC_PASS_H
 
 #include <cstdlib>
 #include "inc/Code.h"
@@ -95,3 +97,5 @@
 };
 
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/Position.h b/gfx/graphite2/src/inc/Position.h
--- a/gfx/graphite2/src/inc/Position.h
+++ b/gfx/graphite2/src/inc/Position.h
@@ -26,6 +26,9 @@
 */
 #pragma once
 
+#ifndef Position_h
+#define Position_h
+
 namespace graphite2 {
 
 class Position
@@ -57,3 +60,4 @@
 };
 
 } // namespace graphite2
+#endif
diff --git a/gfx/graphite2/src/inc/Segment.h b/gfx/graphite2/src/inc/Segment.h
--- a/gfx/graphite2/src/inc/Segment.h
+++ b/gfx/graphite2/src/inc/Segment.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_SEGMENT_H
+#define GRAPHITE2_SRC_SEGMENT_H
 
 #include "inc/Main.h"
 
@@ -268,3 +270,5 @@
 
 struct gr_segment : public graphite2::Segment {};
 
+#endif
+
diff --git a/gfx/graphite2/src/inc/Silf.h b/gfx/graphite2/src/inc/Silf.h
--- a/gfx/graphite2/src/inc/Silf.h
+++ b/gfx/graphite2/src/inc/Silf.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_SILF_H
+#define GRAPHITE2_SRC_SILF_H
 
 #include "graphite2/Font.h"
 #include "inc/Main.h"
@@ -124,3 +126,5 @@
 };
 
 } // namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/Slot.h b/gfx/graphite2/src/inc/Slot.h
--- a/gfx/graphite2/src/inc/Slot.h
+++ b/gfx/graphite2/src/inc/Slot.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_SLOT_H
+#define GRAPHITE2_SRC_SLOT_H
 
 #include "graphite2/Types.h"
 #include "graphite2/Segment.h"
@@ -165,3 +167,5 @@
 } // namespace graphite2
 
 struct gr_slot : public graphite2::Slot {};
+
+#endif
diff --git a/gfx/graphite2/src/inc/TtfUtil.h b/gfx/graphite2/src/inc/TtfUtil.h
--- a/gfx/graphite2/src/inc/TtfUtil.h
+++ b/gfx/graphite2/src/inc/TtfUtil.h
@@ -25,6 +25,8 @@
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_TTFUTIL_H
+#define GRAPHITE2_SRC_TTFUTIL_H
 /*--------------------------------------------------------------------*//*:Ignore this sentence.
 
 File: TtfUtil.h
@@ -204,3 +206,5 @@
 
 } // end of namespace TtfUtil
 } // end of namespace graphite2
+
+#endif
diff --git a/gfx/graphite2/src/inc/FeatureMap.h b/gfx/graphite2/src/inc/FeatureMap.h
--- a/gfx/graphite2/src/inc/FeatureMap.h
+++ b/gfx/graphite2/src/inc/FeatureMap.h
@@ -20,16 +20,19 @@
     internet at http://www.fsf.org/licenses/lgpl.html.
 
 Alternatively, the contents of this file may be used under the terms of the
 Mozilla Public License (http://mozilla.org/MPL) or the GNU General Public
 License, as published by the Free Software Foundation, either version 2
 of the License or (at your option) any later version.
 */
 #pragma once
+#ifndef GRAPHITE2_SRC_FEATUREMAP_H
+#define GRAPHITE2_SRC_FEATUREMAP_H
+
 #include "inc/Main.h"
 #include "inc/FeatureVal.h"
 
 namespace graphite2 {
 
 // Forward declarations for implmentation types
 class FeatureMap;
 class Face;
@@ -177,8 +180,10 @@ private:
 private:        //defensive on m_langFeats
     SillMap(const SillMap&);
     SillMap& operator=(const SillMap&);
 };
 
 } // namespace graphite2
 
 struct gr_feature_ref : public graphite2::FeatureRef {};
+
+#endif
