diff --git a/content/media/webaudio/ThreeDPoint.h b/content/media/webaudio/ThreeDPoint.h
--- a/content/media/webaudio/ThreeDPoint.h
+++ b/content/media/webaudio/ThreeDPoint.h
@@ -8,6 +8,7 @@
 #define ThreeDPoint_h_
 
 #include <cmath>
+#include <math.h>
 
 namespace mozilla {
 
diff --git a/content/media/webaudio/blink/DenormalDisabler.h b/content/media/webaudio/blink/DenormalDisabler.h
--- a/content/media/webaudio/blink/DenormalDisabler.h
+++ b/content/media/webaudio/blink/DenormalDisabler.h
@@ -27,6 +27,7 @@
 
 #define _USE_MATH_DEFINES
 #include <cmath>
+#include <math.h>
 #include <float.h>
 
 namespace WebCore {
