diff --git a/content/media/webm/WebMBufferedParser.h b/content/media/webm/WebMBufferedParser.h
--- a/content/media/webm/WebMBufferedParser.h
+++ b/content/media/webm/WebMBufferedParser.h
@@ -5,17 +5,21 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 #if !defined(WebMBufferedParser_h_)
 #define WebMBufferedParser_h_
 
 #include "nsISupportsImpl.h"
 #include "nsTArray.h"
 #include "mozilla/ReentrantMonitor.h"
 
+namespace mozilla {
+namespace dom {
 class TimeRanges;
+}
+}
 
 namespace mozilla {
 
 // Stores a stream byte offset and the scaled timecode of the block at
 // that offset.  The timecode must be scaled by the stream's timecode
 // scale before use.
 struct WebMTimeDataOffset
 {
