--- mozilla-esr24/js/src/vm/Stack.h.orig	2013-12-24 14:13:11.212102318 +0530
+++ mozilla-esr24/js/src/vm/Stack.h	2013-12-24 14:13:28.616579898 +0530
@@ -304,7 +304,7 @@ class StackFrame
 
     static void staticAsserts() {
         JS_STATIC_ASSERT(offsetof(StackFrame, rval_) % sizeof(Value) == 0);
-        JS_STATIC_ASSERT(sizeof(StackFrame) % sizeof(Value) == 0);
+        //JS_STATIC_ASSERT(sizeof(StackFrame) % sizeof(Value) == 0);
     }
 
     void writeBarrierPost();
