diff --git a/dom/indexedDB/ipc/IndexedDBParent.h b/dom/indexedDB/ipc/IndexedDBParent.h
--- a/dom/indexedDB/ipc/IndexedDBParent.h
+++ b/dom/indexedDB/ipc/IndexedDBParent.h
@@ -663,7 +663,7 @@
   // Don't let anyone call this directly, instead go through SendResponse.
   using PIndexedDBRequestParent::Send__delete__;
 
-  typedef ipc::ResponseValue ResponseValue;
+  //typedef ipc::ResponseValue ResponseValue;
   typedef PIndexedDBRequestParent::PBlobParent PBlobParent;
 
   nsRefPtr<IDBRequest> mRequest;
