diff --git a/content/media/webspeech/recognition/EnableWebSpeechRecognitionCheck.h b/content/media/webspeech/recognition/EnableWebSpeechRecognitionCheck.h
--- a/content/media/webspeech/recognition/EnableWebSpeechRecognitionCheck.h
+++ b/content/media/webspeech/recognition/EnableWebSpeechRecognitionCheck.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef EnableWebSpeechRecognitionCheck_h__
+#define EnableWebSpeechRecognitionCheck_h__
 
 namespace mozilla {
 namespace dom {
@@ -17,3 +19,5 @@
 
 }
 }
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechGrammar.h b/content/media/webspeech/recognition/SpeechGrammar.h
--- a/content/media/webspeech/recognition/SpeechGrammar.h
+++ b/content/media/webspeech/recognition/SpeechGrammar.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechGrammar_h__
+#define SpeechGrammar_h__
 
 #include "nsCOMPtr.h"
 #include "nsCycleCollectionParticipant.h"
@@ -54,3 +56,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechGrammarList.h b/content/media/webspeech/recognition/SpeechGrammarList.h
--- a/content/media/webspeech/recognition/SpeechGrammarList.h
+++ b/content/media/webspeech/recognition/SpeechGrammarList.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechGrammarList_h__
+#define SpeechGrammarList_h__
 
 #include "nsCOMPtr.h"
 #include "nsCycleCollectionParticipant.h"
@@ -55,3 +57,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechRecognitionAlternative.h b/content/media/webspeech/recognition/SpeechRecognitionAlternative.h
--- a/content/media/webspeech/recognition/SpeechRecognitionAlternative.h
+++ b/content/media/webspeech/recognition/SpeechRecognitionAlternative.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechRecognitionAlternative_h__
+#define SpeechRecognitionAlternative_h__
 
 #include "nsCycleCollectionParticipant.h"
 #include "nsString.h"
@@ -49,3 +51,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechRecognitionResult.h b/content/media/webspeech/recognition/SpeechRecognitionResult.h
--- a/content/media/webspeech/recognition/SpeechRecognitionResult.h
+++ b/content/media/webspeech/recognition/SpeechRecognitionResult.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechRecognitionResult_h__
+#define SpeechRecognitionResult_h__
 
 #include "nsCOMPtr.h"
 #include "nsCycleCollectionParticipant.h"
@@ -52,3 +54,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechRecognitionResultList.h b/content/media/webspeech/recognition/SpeechRecognitionResultList.h
--- a/content/media/webspeech/recognition/SpeechRecognitionResultList.h
+++ b/content/media/webspeech/recognition/SpeechRecognitionResultList.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechRecognitionResultList_h__
+#define SpeechRecognitionResultList_h__
 
 #include "nsCycleCollectionParticipant.h"
 #include "nsWrapperCache.h"
@@ -51,3 +53,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/recognition/SpeechStreamListener.h b/content/media/webspeech/recognition/SpeechStreamListener.h
--- a/content/media/webspeech/recognition/SpeechStreamListener.h
+++ b/content/media/webspeech/recognition/SpeechStreamListener.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechStreamListener_h__
+#define SpeechStreamListener_h__
 
 #include "MediaStreamGraph.h"
 #include "AudioSegment.h"
@@ -38,3 +40,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/synth/EnableSpeechSynthesisCheck.h b/content/media/webspeech/synth/EnableSpeechSynthesisCheck.h
--- a/content/media/webspeech/synth/EnableSpeechSynthesisCheck.h
+++ b/content/media/webspeech/synth/EnableSpeechSynthesisCheck.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef EnableSpeechSynthesisCheck_h__
+#define EnableSpeechSynthesisCheck_h__
 
 namespace mozilla {
 namespace dom {
@@ -19,3 +21,5 @@
 
 }
 }
+
+#endif
diff --git a/content/media/webspeech/synth/SpeechSynthesis.h b/content/media/webspeech/synth/SpeechSynthesis.h
--- a/content/media/webspeech/synth/SpeechSynthesis.h
+++ b/content/media/webspeech/synth/SpeechSynthesis.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechSynthesis_h__
+#define SpeechSynthesis_h__
 
 #include "nsCOMPtr.h"
 #include "nsString.h"
@@ -71,3 +73,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/synth/SpeechSynthesisUtterance.h b/content/media/webspeech/synth/SpeechSynthesisUtterance.h
--- a/content/media/webspeech/synth/SpeechSynthesisUtterance.h
+++ b/content/media/webspeech/synth/SpeechSynthesisUtterance.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechSynthesisUtterance_h__
+#define SpeechSynthesisUtterance_h__
 
 #include "nsCOMPtr.h"
 #include "nsDOMEventTargetHelper.h"
@@ -117,3 +119,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/synth/SpeechSynthesisVoice.h b/content/media/webspeech/synth/SpeechSynthesisVoice.h
--- a/content/media/webspeech/synth/SpeechSynthesisVoice.h
+++ b/content/media/webspeech/synth/SpeechSynthesisVoice.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef SpeechSynthesisVoice_h__
+#define SpeechSynthesisVoice_h__
 
 #include "nsCOMPtr.h"
 #include "nsString.h"
@@ -59,3 +61,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
diff --git a/content/media/webspeech/synth/nsSpeechTask.h b/content/media/webspeech/synth/nsSpeechTask.h
--- a/content/media/webspeech/synth/nsSpeechTask.h
+++ b/content/media/webspeech/synth/nsSpeechTask.h
@@ -5,6 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #pragma once
+#ifndef nsSpeechTask_h__
+#define nsSpeechTask_h__
 
 #include "MediaStreamGraph.h"
 #include "SpeechSynthesisUtterance.h"
@@ -86,3 +88,5 @@
 
 } // namespace dom
 } // namespace mozilla
+
+#endif
