--- mozilla-esr17/toolkit/library/Makefile.in.~1~	2013-12-21 02:32:15.373074676 +0400
+++ mozilla-esr17/toolkit/library/Makefile.in	2013-12-21 02:33:55.715822750 +0400
@@ -451,7 +451,7 @@
 endif
 endif
 
-EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS)
+EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS) $(MKSHLIB_FORCE_ALL) ../../mfbt/libmfbt.a $(MKSHLIB_UNFORCE_ALL) 
 
 DEFINES += -DIMPL_XREAPI
 
