--- mozilla-esr17/browser/branding/official/pref/firefox-branding.js.~1~	2013-12-20 11:41:33.871499664 +0400
+++ mozilla-esr17/browser/branding/official/pref/firefox-branding.js	2013-12-20 11:42:04.666610205 +0400
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 pref("startup.homepage_override_url","");
-pref("startup.homepage_welcome_url","http://www.mozilla.com/%LOCALE%/%APP%/%VERSION%/firstrun/");
+pref("startup.homepage_welcome_url","file:///usr/share/doc/openindiana-welcome/html/index.html");
 // Interval: Time between checks for a new version (in seconds)
 pref("app.update.interval", 43200); // 12 hours
 // The time interval between the downloading of mar file chunks in the
