--- mozilla-esr24/browser/branding/official/pref/firefox-branding.js.~1~	2014-07-06 14:03:41.892246822 +0400
+++ mozilla-esr24/browser/branding/official/pref/firefox-branding.js	2014-07-06 14:05:41.524422720 +0400
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 pref("startup.homepage_override_url","");
-pref("startup.homepage_welcome_url","https://www.mozilla.org/%LOCALE%/firefox/%VERSION%/firstrun/");
+pref("startup.homepage_welcome_url","file:///usr/share/doc/openindiana-welcome/html/index.html");
 // Interval: Time between checks for a new version (in seconds)
 pref("app.update.interval", 43200); // 12 hours
 // The time interval between the downloading of mar file chunks in the
