--- mozilla-esr17/security/nss/lib/softoken/manifest.mn	2013-12-08 18:32:45.201684187 +0400
+++ mozilla-esr17/security/nss/lib/softoken/manifest.mn	2013-12-08 18:32:58.542403987 +0400
@@ -9,7 +9,7 @@
 
 LIBRARY_NAME = softokn
 LIBRARY_VERSION = 3
-MAPFILE = $(OBJDIR)/softokn.def
+MAPFILE_SOURCE = $(OBJDIR)/softokn.def
 
 DEFINES += -DSHLIB_SUFFIX=\"$(DLL_SUFFIX)\" -DSHLIB_PREFIX=\"$(DLL_PREFIX)\" -DSOFTOKEN_LIB_NAME=\"$(notdir $(SHARED_LIBRARY))\" -DSHLIB_VERSION=\"$(LIBRARY_VERSION)\"
 
