https://gitlab.freedesktop.org/xorg/lib/libxfont/-/issues/14

--- libXfont-1.5.4/configure.ac.orig
+++ libXfont-1.5.4/configure.ac
@@ -56,6 +56,7 @@
 # If the first PKG_CHECK_MODULES appears inside a conditional, pkg-config
 # must first be located explicitly.
 PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
 
 #
 # select libraries to include
--- libXfont-1.5.4/Makefile.am.orig
+++ libXfont-1.5.4/Makefile.am
@@ -40,7 +40,6 @@
 nodist_libXfontinclude_HEADERS = \
 	include/X11/fonts/fontconf.h
 
-pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xfont.pc
 
 MAINTAINERCLEANFILES = ChangeLog INSTALL
