--- FreeRDP-3.31.0/winpr/libwinpr/sspi/Kerberos/krb5glue_heimdal.c.old	2026-08-26 16:21:52.063177449 -0400
+++ FreeRDP-3.31.0/winpr/libwinpr/sspi/Kerberos/krb5glue_heimdal.c	2026-08-26 16:22:34.049646939 -0400
@@ -147,7 +147,7 @@
 
 krb5_error_code krb5glue_get_init_creds(krb5_context ctx, krb5_principal princ, krb5_ccache ccache,
                                         krb5_prompter_fct prompter, char* password,
-                                        SEC_WINPR_KERBEROS_SETTINGS* krb_settings)
+                                        SEC_WINPR_KERBEROS_SETTINGS_V2* krb_settings)
 {
 	krb5_error_code rv = 0;
 	krb5_deltat start_time = 0;
