There is no definition of a simple MIN macro

diff --git a/src/vis.c.original b/src/vis.c
index a454874..b60b59d 100644
--- a/src/vis.c.original
+++ b/src/vis.c
@@ -85,6 +85,8 @@ __weak_alias(strvisx,_strvisx)
 #include <stdio.h>
 #include <string.h>
 
+#define MIN(a,b) ((a<b)?(a):(b))
+
 /*
  * The reason for going through the trouble to deal with character encodings
  * in vis(3), is that we use this to safe encode output of commands. This
