--- libedit-20120601-3.0/src/sys.h	Fri Jun  1 20:53:49 2012
+++ libedit-20120601-3.0/src/sys.h	Sat Jun  2 14:09:45 2012
@@ -130,13 +130,4 @@
 #define	REGEX		/* Use POSIX.2 regular expression functions */
 #undef	REGEXP		/* Use UNIX V8 regular expression functions */
 
-#if defined(__sun)
-extern int tgetent(char *, const char *);
-extern int tgetflag(char *);
-extern int tgetnum(char *);
-extern int tputs(const char *, int, int (*)(int));
-extern char* tgoto(const char*, int, int);
-extern char* tgetstr(char*, char**);
-#endif
-
 #endif /* _h_sys */
--- libedit-20120601-3.0/src/terminal.h	Tue May 22 19:43:49 2012
+++ libedit-20120601-3.0/src/terminal.h	Sat Jun  2 14:23:21 2012
@@ -82,15 +82,6 @@
 #define	A_K_DE		6
 #define	A_K_NKEYS	7
 
-#ifdef __sun
-extern int tgetent(char *, const char *);
-extern int tgetflag(char *);
-extern int tgetnum(char *);
-extern int tputs(const char *, int, int (*)(int));
-extern char* tgoto(const char*, int, int);
-extern char* tgetstr(char*, char**);
-#endif
-
 protected void	terminal_move_to_line(EditLine *, int);
 protected void	terminal_move_to_char(EditLine *, int);
 protected void	terminal_clear_EOL(EditLine *, int);
