--- swipl-9.2.4/packages/xpce/src/ker/self.c.orig	2024-04-12 21:35:29.000000000 +0200
+++ swipl-9.2.4/packages/xpce/src/ker/self.c	2024-05-09 22:02:40.933140747 +0200
@@ -57,10 +57,6 @@
 #define OS ws_os()
 #endif
 
-#if (defined(__sun__) && !STDC_HEADERS)
-extern int gethostname(char *__name, size_t __len);
-#endif
-
 static void	callExitMessagesPce(int stat, Pce pce);
 static void	exit_pce(int);
 #ifdef HAVE_ON_EXIT
