diff -Nru rsyslog-8.22208.0.orig/runtime/rsyslog.h rsyslog-8.2208.0/runtime/rsyslog.h
--- rsyslog-8.2208.0.orig/runtime/rsyslog.h	2022-08-09 09:08:31.000000000 +0000
+++ rsyslog-8.2208.0/runtime/rsyslog.h	2022-09-28 10:13:45.174499668 +0000
@@ -215,6 +215,8 @@
 #endif
 #define	LOG_AUTHPRIV	(10<<3)	/* security/authorization messages (private) */
 #define	LOG_FTP		(11<<3)	/* ftp daemon */
+#define	LOG_AUDIT	(13<<3)	/* rfc5424 */
+#define	LOG_CRON2	(15<<3)	/* rfc5424 */
 #if defined(_AIX)		/* AIXPORT : These are necessary for AIX */
 #define	LOG_ASO		(12<<3) /* Active System Optimizer. Reserved for internal use */
 #define	LOG_CAA		(15<<3) /* Cluster aware AIX subsystem */
