--- llvm-3.8.1.src/Makefile.rules.orig	2016-08-10 23:42:36.970414278 +0200
+++ llvm-3.8.1.src/Makefile.rules	2016-08-10 23:45:12.899913722 +0200
@@ -644,7 +644,7 @@
     LD.Flags += -Wl,-dead_strip
   else
     ifeq ($(HOST_OS),SunOS)
-      LD.Flags += -Wl,-z -Wl,discard-unused=sections
+      LD.Flags += -Wl,-z -Wl,ignore
     else
       ifneq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
         LD.Flags += -Wl,--gc-sections
--- llvm-3.8.1.src/cmake/modules/AddLLVM.cmake.orig	2016-08-10 23:42:42.590840775 +0200
+++ llvm-3.8.1.src/cmake/modules/AddLLVM.cmake	2016-08-10 23:45:33.068719692 +0200
@@ -176,7 +176,7 @@
                      LINK_FLAGS " -Wl,-dead_strip")
       elseif(${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
         set_property(TARGET ${target_name} APPEND_STRING PROPERTY
-                     LINK_FLAGS " -Wl,-z -Wl,discard-unused=sections")
+                     LINK_FLAGS " -Wl,-z -Wl,ignore")
       elseif(NOT WIN32 AND NOT LLVM_LINKER_IS_GOLD)
         # Object files are compiled with -ffunction-data-sections.
         # Versions of bfd ld < 2.23.1 have a bug in --gc-sections that breaks
