--- res/res_crypto.c	2022-10-19 16:31:39.000000000 +0000
+++ res/res_crypto.c.new	2022-11-04 20:05:43.695055005 +0000
@@ -772,6 +772,11 @@
 	DIR *dir = NULL;
 	struct dirent *ent;
 	int note = 0;
+#ifdef SOLARIS
+	int len = 0;
+	char *filename = NULL;
+	struct stat t_statbuf;
+#endif
 
 	AST_RWLIST_WRLOCK(&keys);
 
@@ -786,10 +791,25 @@
 			if (!strcmp(ent->d_name, ".") || !strcmp(ent->d_name, "..")) {
 				continue;
 			}
+#ifndef SOLARIS
 			if (ent->d_type != DT_REG) {
 				ast_log(LOG_WARNING, "Non-regular file '%s' in keys directory\n", ent->d_name);
 				continue;
 			}
+#else
+			len = strlen(ast_config_AST_KEY_DIR) + strlen(ent->d_name) + 2;
+			if ((filename = (char *) ast_malloc(sizeof (char) * len)) == NULL) {
+				ast_log(LOG_ERROR, "No memory available to create filename string '%s'\n", filename);
+				exit(ENOMEM);
+			}
+			snprintf(filename, len, "%s/%s", ast_config_AST_KEY_DIR, ent->d_name);
+			if (lstat(filename, &t_statbuf) == 0) {
+                       	if (! S_ISREG(t_statbuf.st_mode)) {
+					ast_log(LOG_WARNING, "Non-regular file '%s' in keys directory\n", ent->d_name);
+					continue;
+				}
+			}
+#endif
 			try_load_key(ast_config_AST_KEY_DIR, ent->d_name, ifd, ofd, &note);
 		}
 		closedir(dir);
