--- compiz-fusion-plugins-main-0.7.8/src/animation/animation.c.ori	2009-03-11 08:58:13.532394579 +0000
+++ compiz-fusion-plugins-main-0.7.8/src/animation/animation.c	2009-03-11 09:01:16.600066248 +0000
@@ -806,7 +806,7 @@
     CompVector coordsTransformed;
     
     matrixMultiplyVector (&coordsTransformed, coords, transformMat);
-    expandBoxWithPoint (target, coordsTransformed.x, coordsTransformed.y);
+    expandBoxWithPoint (target, coordsTransformed.a.x, coordsTransformed.a.y);
 }
 
 // Either points or objects should be non-NULL.
@@ -910,10 +910,10 @@
 	    {
 		CompVector coords;
     
-		coords.x = object->position.x;
-		coords.y = object->position.y;
-		coords.z = 0;
-		coords.w = 1;
+		coords.a.x = object->position.x;
+		coords.a.y = object->position.y;
+		coords.a.z = 0;
+		coords.a.w = 1;
 
 		expandBoxWithPoint2DTransform (w->screen,
 					       BB,
