https://github.com/gevent/gevent/pull/1950

--- gevent-22.10.2/src/gevent/tests/test__socket.py.orig
+++ gevent-22.10.2/src/gevent/tests/test__socket.py
@@ -396,13 +396,13 @@ def accept_once():
 
     def test_attributes(self):
         s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
-        self.assertEqual(socket.AF_INET, s.type)
-        self.assertEqual(socket.SOCK_DGRAM, s.family)
+        self.assertEqual(socket.AF_INET, s.family)
+        self.assertEqual(socket.SOCK_DGRAM, s.type)
         self.assertEqual(0, s.proto)
 
         if hasattr(socket, 'SOCK_NONBLOCK'):
             s.settimeout(1)
-            self.assertEqual(socket.AF_INET, s.type)
+            self.assertEqual(socket.AF_INET, s.family)
 
             s.setblocking(0)
             std_socket = monkey.get_original('socket', 'socket')(socket.AF_INET, socket.SOCK_DGRAM, 0)
