--- gevent-22.10.2/src/gevent/testing/testrunner.py.orig
+++ gevent-22.10.2/src/gevent/testing/testrunner.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 from __future__ import print_function, absolute_import, division
 
 import re
--- gevent-22.10.2/src/gevent/tests/test__socket_ssl.py.orig
+++ gevent-22.10.2/src/gevent/tests/test__socket_ssl.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 from gevent import monkey
 monkey.patch_all()
 
