Curio is abandonware.
https://github.com/dabeaz/curio/issues/367

--- sniffio-1.3.1/test-requirements.txt.orig
+++ sniffio-1.3.1/test-requirements.txt
@@ -1,2 +1 @@
 pytest
-curio
--- sniffio-1.3.1/sniffio/_tests/test_sniffio.py.orig
+++ sniffio-1.3.1/sniffio/_tests/test_sniffio.py
@@ -58,8 +58,7 @@
         current_async_library()
 
 
-@pytest.mark.skipif(
-    sys.version_info >= (3, 12),
+@pytest.mark.skip(
     reason=
     "curio broken on 3.12 (https://github.com/python-trio/sniffio/pull/42)",
 )
