Override Solaris-specific settings => not for upstream.

--- ImageMagick-6.9.4/PerlMagick/Makefile.PL.in.~1~	2016-05-09 10:28:57.000000000 -0700
+++ ImageMagick-6.9.4/PerlMagick/Makefile.PL.in	2016-05-10 14:46:18.533378790 -0700
@@ -214,6 +214,8 @@
    # C compiler flags (e.g. -O -g)
    'CCFLAGS' => $CCFLAGS_magick,
 
+   'CCCDLFLAGS' => '-fPIC',
+
    # Linker
    #'LD' => $Config{'ld'} == $Config{'cc'} ? '@CC@' : $Config{'ld'},
 
@@ -232,8 +234,7 @@
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
 
-   # Let CFLAGS drive optimization flags by setting OPTIMIZE to empty
-   # 'OPTIMIZE'	=> '',
+   'OPTIMIZE'	=> '-O3',
 
    # Use same compiler as ImageMagick
    'PERLMAINCC'	=> '@PERLMAINCC@ @OPENMP_CFLAGS@',
--- ImageMagick-6.9.4/PerlMagick/quantum/Makefile.PL.in.~1~	2016-05-09 10:28:57.000000000 -0700
+++ ImageMagick-6.9.4/PerlMagick/quantum/Makefile.PL.in	2016-05-10 13:42:57.219875745 -0700
@@ -211,6 +211,8 @@
    # C compiler flags (e.g. -O -g)
    'CCFLAGS' => $CCFLAGS_magick,
 
+   'CCCDLFLAGS' => '-fPIC',
+
    # Linker
    #'LD' => $Config{'ld'} == $Config{'cc'} ? '@CC@' : $Config{'ld'},
 
@@ -229,8 +231,7 @@
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
 
-   # Let CFLAGS drive optimization flags by setting OPTIMIZE to empty
-   # 'OPTIMIZE'	=> '',
+   'OPTIMIZE'	=> '-O3',
 
    # Use same compiler as ImageMagick
    'PERLMAINCC'	=> '@PERLMAINCC@ @OPENMP_CFLAGS@',
