diff --git java/src/com/tigervnc/vncviewer/Makefile java/src/com/tigervnc/vncviewer/Makefile
index f434441..00323f6 100644
--- java/src/com/tigervnc/vncviewer/Makefile
+++ java/src/com/tigervnc/vncviewer/Makefile
@@ -9,6 +9,7 @@ JAR = jar
 ARCHIVE = VncViewer.jar
 MANIFEST = MANIFEST.MF
 PAGES = index.vnc index.html
+ICON = tigervnc_16.gif
 INSTALL_DIR = /usr/local/vnc/classes
 
 CLASSES = VncViewer.class RfbProto.class AuthPanel.class VncCanvas.class \
@@ -44,9 +45,12 @@ $(ARCHIVE): $(CLASSES) $(MANIFEST)
 		com/tigervnc/decoder/*.class \
 		com/tigervnc/decoder/common/*.class
 
-install: $(PAGES) README $(ARCHIVE)
+$(ICON):
+	convert -background transparent ../../media/tigervnc_16.svg $@
+
+install: $(PAGES) README $(ARCHIVE) $(ICON)
 	mkdir -p $(INSTALL_DIR)
-	$(CP) $(ARCHIVE) $(PAGES) README $(INSTALL_DIR)
+	$(CP) $(ARCHIVE) $(PAGES) README $(ICON) $(INSTALL_DIR)
 
 export:: $(CLASSES) $(ARCHIVE) $(PAGES)
 	@$(ExportJavaClasses)
diff --git java/src/com/tigervnc/vncviewer/index.html java/src/com/tigervnc/vncviewer/index.html
index b17cf70..c982196 100644
--- java/src/com/tigervnc/vncviewer/index.html
+++ java/src/com/tigervnc/vncviewer/index.html
@@ -19,12 +19,13 @@
 <TITLE>
 TigerVNC desktop
 </TITLE>
+<link rel="icon" type="image/gif" href="tigervnc_16.gif">
 </HEAD><BODY>
 <APPLET CODE="com.tigervnc.vncviewer.VncViewer" ARCHIVE="VncViewer.jar"
         WIDTH="800" HEIGHT="632">
 <PARAM NAME="PORT" VALUE="5901">
 </APPLET>
 <BR>
-<A href="http://www.tigervnc.org/">TigerVNC site</A>
+<A href="http://www.tigervnc.org/"><IMG border="0" hspace="4" src="tigervnc_16.gif">TigerVNC site</A>
 </BODY>
 </HTML>
diff --git java/src/com/tigervnc/vncviewer/index.vnc java/src/com/tigervnc/vncviewer/index.vnc
index d71234f..948e506 100644
--- java/src/com/tigervnc/vncviewer/index.vnc
+++ java/src/com/tigervnc/vncviewer/index.vnc
@@ -14,6 +14,7 @@
 <TITLE>
 $USER's $DESKTOP desktop ($DISPLAY)
 </TITLE>
+<link rel="icon" type="image/gif" href="tigervnc_16.gif">
 </HEAD><BODY>
 <APPLET CODE="com.tigervnc.vncviewer.VncViewer" ARCHIVE="VncViewer.jar"
         WIDTH="$APPLETWIDTH" HEIGHT="$APPLETHEIGHT">
@@ -21,6 +22,6 @@ $USER's $DESKTOP desktop ($DISPLAY)
 $PARAMS
 </APPLET>
 <BR>
-<A href="http://www.tigervnc.org/">TigerVNC site</A>
+<A href="http://www.tigervnc.org/"><IMG border="0" hspace="4" src="tigervnc_16.gif">TigerVNC site</A>
 </BODY>
 </HTML>
