--- unix/xserver/hw/vnc/xvnc.cc.~1~	2014-04-29 15:59:02.141105648 +0400
+++ unix/xserver/hw/vnc/xvnc.cc	2014-04-29 15:59:34.373783442 +0400
@@ -746,7 +746,7 @@
     switch (fbmemtype) {
 #ifdef HAS_SHM
     case SHARED_MEMORY_FB:
-        if (-1 == shmdt(pfb->pfbMemory)) {
+        if (-1 == shmdt((char*)pfb->pfbMemory)) {
             perror("shmdt");
             ErrorF("shmdt failed, errno %d", errno);
         }
