--- unix/xserver/hw/vnc/Makefile.am	Thu Dec  4 14:53:42 2014
+++ unix/xserver/hw/vnc/Makefile.am	Thu Dec  4 14:54:08 2014
@@ -17,7 +17,7 @@
 
 libvnccommon_la_CPPFLAGS = -DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
 	-DVENDOR_STRING="\"$(VENDOR_STRING)\"" -I$(TIGERVNC_SRCDIR)/common -UHAVE_CONFIG_H \
-	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) $(PIXMAN_CFLAGS) -I$(includedir)
+	-I$(TIGERVNC_SRCDIR)/unix/vncconfig $(XVNC_CPPFLAGS) $(PIXMAN_CFLAGS) -I$(top_srcdir)$(includedir)
 
 bin_PROGRAMS = Xvnc
 
