--- unix/xserver/hw/vnc/xorg-version.h.orig	2016-09-12 22:18:33.654752000 +0200
+++ unix/xserver/hw/vnc/xorg-version.h	2016-09-12 22:23:57.165646149 +0200
@@ -48,6 +48,8 @@
 #define XORG 116
 #elif XORG_VERSION_CURRENT < ((1 * 10000000) + (17 * 100000) + (99 * 1000))
 #define XORG 117
+#elif XORG_VERSION_CURRENT < ((1 * 10000000) + (18 * 100000) + (99 * 1000))
+#define XORG 118
 #else
 #error "X.Org newer than 1.10 is not supported"
 #endif
--- unix/xserver/hw/vnc/Input.cc.orig	2016-09-12 22:31:26.090115571 +0200
+++ unix/xserver/hw/vnc/Input.cc	2016-09-12 22:31:13.189854166 +0200
@@ -331,8 +331,10 @@
 #if XORG < 111
 	n = GetKeyboardEvents(eventq, dev, action, kc, NULL);
 	enqueueEvents(dev, n);
-#else
+#elif XORG < 118
 	QueueKeyboardEvents(dev, action, kc, NULL);
+#else
+	QueueKeyboardEvents(dev, action, kc);
 #endif
 }
 
