We do not have these projects packaged yet.

--- elastic_transport-8.17.1/setup.py.orig
+++ elastic_transport-8.17.1/setup.py
@@ -65,10 +65,6 @@
             "requests",
             "aiohttp",
             "httpx",
-            "respx",
-            "opentelemetry-api",
-            "opentelemetry-sdk",
-            "orjson",
             # Override Read the Docs default (sphinx<2)
             "sphinx>2",
             "sphinx-autodoc-typehints",
