--- gedit-2.26.3.orig/gedit/gedit-view.c	2009-07-08 12:39:39.441938100 +0530
+++ gedit-2.26.3/gedit/gedit-view.c	2009-07-08 12:46:28.261536300 +0530
@@ -523,9 +523,8 @@ gedit_view_cut_clipboard (GeditView *vie
 	/* FIXME: what is default editability of a buffer? */
   	gtk_text_buffer_cut_clipboard (buffer,
   				       clipboard,
-				       !gedit_document_get_readonly (
-				       		GEDIT_DOCUMENT (buffer)));
-  	
+  				       TRUE); /* Fix Bug 6601741 */
+	
 	gtk_text_view_scroll_to_mark (GTK_TEXT_VIEW (view),
 				      gtk_text_buffer_get_insert (buffer),
 				      GEDIT_VIEW_SCROLL_MARGIN,
@@ -575,8 +574,7 @@ gedit_view_paste_clipboard (GeditView *v
   	gtk_text_buffer_paste_clipboard (buffer,
 					 clipboard,
 					 NULL,
-					 !gedit_document_get_readonly (
-						GEDIT_DOCUMENT (buffer)));
+					 TRUE); /* Fix Bug 6601741 */
 
 	gtk_text_view_scroll_to_mark (GTK_TEXT_VIEW (view),
 				      gtk_text_buffer_get_insert (buffer),
@@ -608,9 +606,8 @@ gedit_view_delete_selection (GeditView *
 	/* FIXME: what is default editability of a buffer? */
 	gtk_text_buffer_delete_selection (buffer,
 					  TRUE,
-					  !gedit_document_get_readonly (
-						GEDIT_DOCUMENT (buffer)));
-						
+					  TRUE);  /* Fix Bug 6601741 */
+	
 	gtk_text_view_scroll_to_mark (GTK_TEXT_VIEW (view),
 				      gtk_text_buffer_get_insert (buffer),
 				      GEDIT_VIEW_SCROLL_MARGIN,
