diff -ruN gedit-2.30.2.orig/configure.ac gedit-2.30.2/configure.ac
--- gedit-2.30.2.orig/configure.ac      2010-04-22 15:42:10.828298608 +0100
+++ gedit-2.30.2/configure.ac   2010-04-22 15:41:53.799049461 +0100
@@ -36,8 +36,8 @@
 AC_PROG_OBJC
 
 # Initialize libtool
-LT_PREREQ([2.2.6])
-LT_INIT(disable-static)
+#LT_PREREQ([2.2.6])
+#LT_INIT(disable-static)
 
 GEDIT_API_VERSION=2.20
 AC_SUBST(GEDIT_API_VERSION)
diff -ruN gedit-2.30.2.orig/Makefile.am gedit-2.30.2/Makefile.am
--- gedit-2.30.2.orig/Makefile.am       2010-04-22 15:51:45.994908303 +0100
+++ gedit-2.30.2/Makefile.am    2010-04-22 15:52:00.187062829 +0100
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = gedit pixmaps po data plugin-loaders plugins docs tests win32 osx
+SUBDIRS = gedit pixmaps po data plugin-loaders plugins docs win32 osx
 
 if !OS_OSX
 SUBDIRS += help
