diff -r 132649e875f3 plugins/time/gedit-time-plugin.c
--- a/plugins/time/gedit-time-plugin.c	Thu May 27 15:32:41 2010 +0800
+++ b/plugins/time/gedit-time-plugin.c	Thu May 27 15:34:14 2010 +0800
@@ -359,12 +359,6 @@
 {
	const gchar * str;

-	if (!gconf_client_key_is_writable (plugin->priv->gconf_client,
-					   PROMPT_TYPE_KEY,
-					   NULL))
-	{
-		return;
-	}

	switch (prompt_type)
	{
@@ -406,12 +400,6 @@
 {
	g_return_if_fail (format != NULL);

-	if (!gconf_client_key_is_writable (plugin->priv->gconf_client,
-					   SELECTED_FORMAT_KEY,
-					   NULL))
-	{
-		return;
-	}

	gconf_client_set_string (plugin->priv->gconf_client,
				 SELECTED_FORMAT_KEY,
@@ -438,10 +426,6 @@
 {
	g_return_if_fail (format != NULL);

-	if (!gconf_client_key_is_writable (plugin->priv->gconf_client,
-					   CUSTOM_FORMAT_KEY,
-					   NULL))
-		return;

	gconf_client_set_string (plugin->priv->gconf_client,
				 CUSTOM_FORMAT_KEY,
