diff -ur tracker-1.8.1-1/src/miners/apps/tracker-miner-apps.desktop.in.in tracker-1.8.1/src/miners/apps/tracker-miner-apps.desktop.in.in
--- tracker-1.8.1-1/src/miners/apps/tracker-miner-apps.desktop.in.in	2017-03-17 08:47:03.750877639 +0300
+++ tracker-1.8.1/src/miners/apps/tracker-miner-apps.desktop.in.in	2017-03-17 08:47:44.396725818 +0300
@@ -7,11 +7,11 @@
 Type=Application
 Categories=Utility;
 X-Meego-Priority=Late
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;X-MEEGO-HS;X-MEEGO-NB;X-IVI;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
diff -ur tracker-1.8.1-1/src/miners/fs/tracker-miner-fs.desktop.in.in tracker-1.8.1/src/miners/fs/tracker-miner-fs.desktop.in.in
--- tracker-1.8.1-1/src/miners/fs/tracker-miner-fs.desktop.in.in	2017-03-17 08:47:03.752376945 +0300
+++ tracker-1.8.1/src/miners/fs/tracker-miner-fs.desktop.in.in	2017-03-17 08:47:44.398196819 +0300
@@ -7,11 +7,11 @@
 Type=Application
 Categories=Utility;
 X-Meego-Priority=Late
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;X-MEEGO-HS;X-MEEGO-NB;X-IVI;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
diff -ur tracker-1.8.1-1/src/miners/rss/tracker-miner-rss.desktop.in.in tracker-1.8.1/src/miners/rss/tracker-miner-rss.desktop.in.in
--- tracker-1.8.1-1/src/miners/rss/tracker-miner-rss.desktop.in.in	2017-03-17 08:47:03.755346800 +0300
+++ tracker-1.8.1/src/miners/rss/tracker-miner-rss.desktop.in.in	2017-03-17 08:47:44.397448258 +0300
@@ -9,8 +9,8 @@
 X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
diff -ur tracker-1.8.1-1/src/miners/user-guides/tracker-miner-user-guides.desktop.in.in tracker-1.8.1/src/miners/user-guides/tracker-miner-user-guides.desktop.in.in
--- tracker-1.8.1-1/src/miners/user-guides/tracker-miner-user-guides.desktop.in.in	2017-03-17 08:47:03.750355435 +0300
+++ tracker-1.8.1/src/miners/user-guides/tracker-miner-user-guides.desktop.in.in	2017-03-17 08:47:44.398907715 +0300
@@ -7,11 +7,11 @@
 Type=Application
 Categories=Utility;
 X-Meego-Priority=Late
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;X-MEEGO-HS;X-MEEGO-NB;X-IVI;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
diff -ur tracker-1.8.1-1/src/tracker-extract/tracker-extract.desktop.in.in tracker-1.8.1/src/tracker-extract/tracker-extract.desktop.in.in
--- tracker-1.8.1-1/src/tracker-extract/tracker-extract.desktop.in.in	2017-03-17 08:47:03.746732661 +0300
+++ tracker-1.8.1/src/tracker-extract/tracker-extract.desktop.in.in	2017-03-17 08:47:44.395980724 +0300
@@ -7,11 +7,11 @@
 Type=Application
 Categories=Utility;
 X-Meego-Priority=Late
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;X-MEEGO-HS;X-MEEGO-NB;X-IVI;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
diff -ur tracker-1.8.1-1/src/tracker-store/tracker-store.desktop.in.in tracker-1.8.1/src/tracker-store/tracker-store.desktop.in.in
--- tracker-1.8.1-1/src/tracker-store/tracker-store.desktop.in.in	2017-03-17 08:47:03.718529366 +0300
+++ tracker-1.8.1/src/tracker-store/tracker-store.desktop.in.in	2017-03-17 08:47:44.395165553 +0300
@@ -6,11 +6,11 @@
 Terminal=false
 Type=Application
 Categories=Utility;
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
-X-KDE-UniqueApplet=true
-NoDisplay=true
+X-KDE-UniqueApplet=false
+NoDisplay=false
 OnlyShowIn=GNOME;KDE;XFCE;X-MEEGO-HS;X-MEEGO-NB;X-IVI;Unity;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=tracker
