--- tracker-0.10.37.1/src/tracker-preferences/tracker-preferences.desktop.in.in.1	2016-04-13 23:25:14.383482369 +0300
+++ tracker-0.10.37.1/src/tracker-preferences/tracker-preferences.desktop.in.in	2016-04-13 23:26:09.199947902 +0300
@@ -1,6 +1,8 @@
 [Desktop Entry]
-_Name=Search and Indexing
-_Comment=Configure file indexing with Tracker
+# SUN_BRANDING
+_Name=File Indexing
+# SUN_BRANDING
+_Comment=Choose how files are indexed for searching
 Icon=tracker
 Exec=tracker-preferences
 Terminal=false
