diff -ur tracker-0.10.37/examples/libtracker-extract/Makefile.am tracker-0.10.37.1/examples/libtracker-extract/Makefile.am
--- tracker-0.10.37/examples/libtracker-extract/Makefile.am	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/examples/libtracker-extract/Makefile.am	2016-04-13 23:43:20.158400645 +0300
@@ -1,6 +1,6 @@
 include $(top_srcdir)/Makefile.decl
 
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version 
 modulesdir = $(libdir)/tracker-$(TRACKER_API_VERSION)/extract-modules
 
 AM_CPPFLAGS =                                          \
diff -ur tracker-0.10.37/src/plugins/evolution/Makefile.am tracker-0.10.37.1/src/plugins/evolution/Makefile.am
--- tracker-0.10.37/src/plugins/evolution/Makefile.am	2011-06-08 12:06:26.000000000 +0400
+++ tracker-0.10.37.1/src/plugins/evolution/Makefile.am	2016-04-13 23:43:20.163167458 +0300
@@ -23,7 +23,7 @@
 
 eplugin_LTLIBRARIES = liborg-freedesktop-Tracker-evolution-plugin.la
 
-# module_flags = -module -avoid-version -no-undefined
+# module_flags = -module -avoid-version 
 
 liborg_freedesktop_Tracker_evolution_plugin_la_SOURCES = \
 	tracker-evolution-plugin.c                     \
diff -ur tracker-0.10.37/src/tracker-extract/Makefile.am tracker-0.10.37.1/src/tracker-extract/Makefile.am
--- tracker-0.10.37/src/tracker-extract/Makefile.am	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-extract/Makefile.am	2016-04-13 23:43:20.162476886 +0300
@@ -1,6 +1,6 @@
 include $(top_srcdir)/Makefile.decl
 
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version 
 modulesdir = $(libdir)/tracker-$(TRACKER_API_VERSION)/extract-modules
 
 AM_CPPFLAGS = \
diff -ur tracker-0.10.37/src/tracker-writeback/Makefile.am tracker-0.10.37.1/src/tracker-writeback/Makefile.am
--- tracker-0.10.37/src/tracker-writeback/Makefile.am	2011-10-06 12:11:23.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-writeback/Makefile.am	2016-04-13 23:43:20.164182894 +0300
@@ -1,6 +1,6 @@
 include $(top_srcdir)/Makefile.decl
 
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version 
 modulesdir = $(libdir)/tracker-$(TRACKER_API_VERSION)/writeback-modules
 
 AM_CPPFLAGS =                                          \
