--- tracker-0.10.37/src/tracker-extract/tracker-extract-gif.c.~2~	2016-10-19 09:21:37.048149628 +0300
+++ tracker-0.10.37/src/tracker-extract/tracker-extract-gif.c	2016-10-19 09:25:42.157499226 +0300
@@ -96,6 +96,7 @@
 	unsigned char *framedata = NULL;
 	GPtrArray *keywords;
 	guint i;
+	gint h;
 	int status;
 	MergeData md = { 0 };
 	GifData   gd = { 0 };
@@ -122,11 +123,12 @@
 			framewidth  = gifFile->Image.Width;
 			frameheight = gifFile->Image.Height;
 
-			framedata = g_malloc (framewidth*frameheight);
-
-			if (DGifGetLine(gifFile, framedata, framewidth*frameheight)==GIF_ERROR) {
-				PrintGifError();
-				return;
+			framedata = g_malloc_n (framewidth, sizeof(GifPixelType));
+			for (h = 0; h < frameheight; h++) {
+				if (DGifGetLine(gifFile, framedata, framewidth)==GIF_ERROR) {
+					PrintGifError();
+					return;
+				}
 			}
 
 			gd.width  = g_strdup_printf ("%d", framewidth);
