--- tracker-0.10.37/configure.ac.1	2016-04-13 23:53:49.068556974 +0300
+++ tracker-0.10.37/configure.ac	2016-04-13 23:54:07.717549052 +0300
@@ -238,8 +238,7 @@
 # Check requirements for libtracker-data
 LIBTRACKER_DATA_REQUIRED="glib-2.0     >= $GLIB_REQUIRED
                           gio-unix-2.0 >= $GLIB_REQUIRED
-                          sqlite3      >= $SQLITE_REQUIRED
-                          uuid"
+                          sqlite3      >= $SQLITE_REQUIRED"
 PKG_CHECK_MODULES(LIBTRACKER_DATA, [$LIBTRACKER_DATA_REQUIRED])
 
 LIBTRACKER_DATA_LIBS="$LIBTRACKER_DATA_LIBS -lz -lm"
@@ -272,16 +271,15 @@
 
 # Check requirements for libtracker-sparql
 LIBTRACKER_SPARQL_REQUIRED="glib-2.0     >= $GLIB_REQUIRED
-                            gio-unix-2.0 >= $GLIB_REQUIRED
-                            uuid"
+                            gio-unix-2.0 >= $GLIB_REQUIRED"
 
 PKG_CHECK_MODULES(LIBTRACKER_SPARQL, [$LIBTRACKER_SPARQL_REQUIRED])
 
 # Check requirements for tracker-control
 TRACKER_CONTROL_REQUIRED="glib-2.0     >= $GLIB_REQUIRED
                           gio-unix-2.0 >= $GLIB_REQUIRED
-                          gthread-2.0  >= $GLIB_REQUIRED
-                          uuid"
+                          gthread-2.0  >= $GLIB_REQUIRED"
+                          
 
 PKG_CHECK_MODULES(TRACKER_CONTROL, [$TRACKER_CONTROL_REQUIRED])
 
