diff -ur tracker-0.10.37/examples/libtracker-miner/tracker-main.c tracker-0.10.37.1/examples/libtracker-miner/tracker-main.c
--- tracker-0.10.37/examples/libtracker-miner/tracker-main.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/examples/libtracker-miner/tracker-main.c	2016-04-14 00:14:23.106484233 +0300
@@ -196,7 +196,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	main_loop = g_main_loop_new (NULL, FALSE);
diff -ur tracker-0.10.37/src/miners/flickr/tracker-miner-flickr.c tracker-0.10.37.1/src/miners/flickr/tracker-miner-flickr.c
--- tracker-0.10.37/src/miners/flickr/tracker-miner-flickr.c	2011-12-02 14:53:49.000000000 +0400
+++ tracker-0.10.37.1/src/miners/flickr/tracker-miner-flickr.c	2016-04-14 00:14:22.995515749 +0300
@@ -6648,7 +6648,6 @@
 
 int main (int argc, char ** argv) {
 #line 950 "/home/juerg/Code/tracker/tracker/src/miners/flickr/tracker-miner-flickr.vala"
-	g_thread_init (NULL);
 #line 950 "/home/juerg/Code/tracker/tracker/src/miners/flickr/tracker-miner-flickr.vala"
 	g_type_init ();
 #line 950 "/home/juerg/Code/tracker/tracker/src/miners/flickr/tracker-miner-flickr.vala"
diff -ur tracker-0.10.37/src/miners/fs/tracker-main.c tracker-0.10.37.1/src/miners/fs/tracker-main.c
--- tracker-0.10.37/src/miners/fs/tracker-main.c	2011-12-02 14:33:49.000000000 +0400
+++ tracker-0.10.37.1/src/miners/fs/tracker-main.c	2016-04-14 00:14:22.933079868 +0300
@@ -849,7 +849,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	setlocale (LC_ALL, "");
diff -ur tracker-0.10.37/src/miners/rss/tracker-main.c tracker-0.10.37.1/src/miners/rss/tracker-main.c
--- tracker-0.10.37/src/miners/rss/tracker-main.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/miners/rss/tracker-main.c	2016-04-14 00:14:22.997261469 +0300
@@ -61,7 +61,6 @@
 	const gchar *error_message;
 
 	g_type_init ();
-	g_thread_init (NULL);
 
 	setlocale (LC_ALL, "");
 
diff -ur tracker-0.10.37/src/tracker-control/tracker-control.c tracker-0.10.37.1/src/tracker-control/tracker-control.c
--- tracker-0.10.37/src/tracker-control/tracker-control.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-control/tracker-control.c	2016-04-14 00:14:23.035965019 +0300
@@ -69,7 +69,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	/* Translators: this messagge will apper immediately after the  */
diff -ur tracker-0.10.37/src/tracker-utils/tracker-import.c tracker-0.10.37.1/src/tracker-utils/tracker-import.c
--- tracker-0.10.37/src/tracker-utils/tracker-import.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-import.c	2016-04-14 00:14:23.010492512 +0300
@@ -105,7 +105,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	connection = tracker_sparql_connection_get (NULL, &error);
diff -ur tracker-0.10.37/src/tracker-utils/tracker-info.c tracker-0.10.37.1/src/tracker-utils/tracker-info.c
--- tracker-0.10.37/src/tracker-utils/tracker-info.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-info.c	2016-04-14 00:14:23.009025815 +0300
@@ -350,7 +350,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	connection = tracker_sparql_connection_get (NULL, &error);
diff -ur tracker-0.10.37/src/tracker-utils/tracker-search.c tracker-0.10.37.1/src/tracker-utils/tracker-search.c
--- tracker-0.10.37/src/tracker-utils/tracker-search.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-search.c	2016-04-14 00:14:23.033081904 +0300
@@ -1378,7 +1378,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 
diff -ur tracker-0.10.37/src/tracker-utils/tracker-sparql.c tracker-0.10.37.1/src/tracker-utils/tracker-sparql.c
--- tracker-0.10.37/src/tracker-utils/tracker-sparql.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-sparql.c	2016-04-14 00:14:23.004524648 +0300
@@ -281,7 +281,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	connection = tracker_sparql_connection_get (NULL, &error);
diff -ur tracker-0.10.37/src/tracker-utils/tracker-stats.c tracker-0.10.37.1/src/tracker-utils/tracker-stats.c
--- tracker-0.10.37/src/tracker-utils/tracker-stats.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-stats.c	2016-04-14 00:14:23.034432017 +0300
@@ -82,7 +82,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	connection = tracker_sparql_connection_get (NULL, &error);
diff -ur tracker-0.10.37/src/tracker-utils/tracker-tag.c tracker-0.10.37.1/src/tracker-utils/tracker-tag.c
--- tracker-0.10.37/src/tracker-utils/tracker-tag.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-utils/tracker-tag.c	2016-04-14 00:14:23.019096722 +0300
@@ -934,7 +934,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	connection = tracker_sparql_connection_get (NULL, &error);
diff -ur tracker-0.10.37/src/tracker-writeback/tracker-main.c tracker-0.10.37.1/src/tracker-writeback/tracker-main.c
--- tracker-0.10.37/src/tracker-writeback/tracker-main.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/src/tracker-writeback/tracker-main.c	2016-04-14 00:14:22.998785070 +0300
@@ -85,7 +85,6 @@
 	gchar *log_filename;
 	guint shutdown_timeout;
 
-	g_thread_init (NULL);
 
 	g_type_init ();
 
diff -ur tracker-0.10.37/tests/libtracker-common/tracker-dbus-test.c tracker-0.10.37.1/tests/libtracker-common/tracker-dbus-test.c
--- tracker-0.10.37/tests/libtracker-common/tracker-dbus-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-common/tracker-dbus-test.c	2016-04-14 00:14:23.038926535 +0300
@@ -289,7 +289,6 @@
 	gint result;
 
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 /*
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-backup-test.c tracker-0.10.37.1/tests/libtracker-data/tracker-backup-test.c
--- tracker-0.10.37/tests/libtracker-data/tracker-backup-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-backup-test.c	2016-04-14 00:14:23.041327133 +0300
@@ -220,7 +220,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-db-journal.c tracker-0.10.37.1/tests/libtracker-data/tracker-db-journal.c
--- tracker-0.10.37/tests/libtracker-data/tracker-db-journal.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-db-journal.c	2016-04-14 00:14:23.056248725 +0300
@@ -366,7 +366,6 @@
 	int result;
 
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 #ifndef DISABLE_JOURNAL
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-ontology-change-test.c tracker-0.10.37.1/tests/libtracker-data/tracker-ontology-change-test.c
--- tracker-0.10.37/tests/libtracker-data/tracker-ontology-change-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-ontology-change-test.c	2016-04-14 00:14:23.052646160 +0300
@@ -315,7 +315,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-ontology-test.c tracker-0.10.37.1/tests/libtracker-data/tracker-ontology-test.c
--- tracker-0.10.37/tests/libtracker-data/tracker-ontology-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-ontology-test.c	2016-04-14 00:14:23.044407341 +0300
@@ -279,7 +279,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-sparql-blank-test.c tracker-0.10.37.1/tests/libtracker-data/tracker-sparql-blank-test.c
--- tracker-0.10.37/tests/libtracker-data/tracker-sparql-blank-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-sparql-blank-test.c	2016-04-14 00:14:23.045921519 +0300
@@ -128,7 +128,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-data/tracker-sparql-test.c tracker-0.10.37.1/tests/libtracker-data/tracker-sparql-test.c
--- tracker-0.10.37/tests/libtracker-data/tracker-sparql-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-data/tracker-sparql-test.c	2016-04-14 00:14:23.049455471 +0300
@@ -327,7 +327,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-extract/tracker-test-xmp.c tracker-0.10.37.1/tests/libtracker-extract/tracker-test-xmp.c
--- tracker-0.10.37/tests/libtracker-extract/tracker-test-xmp.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-extract/tracker-test-xmp.c	2016-04-14 00:14:23.095718309 +0300
@@ -587,7 +587,6 @@
 	gint result;
 
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 	g_test_message ("Testing XMP");
diff -ur tracker-0.10.37/tests/libtracker-fts/tracker-fts-test.c tracker-0.10.37.1/tests/libtracker-fts/tracker-fts-test.c
--- tracker-0.10.37/tests/libtracker-fts/tracker-fts-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-fts/tracker-fts-test.c	2016-04-14 00:14:23.097729855 +0300
@@ -182,7 +182,6 @@
 	g_type_init ();
 
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	g_test_init (&argc, &argv, NULL);
diff -ur tracker-0.10.37/tests/libtracker-fts/tracker-parser-test.c tracker-0.10.37.1/tests/libtracker-fts/tracker-parser-test.c
--- tracker-0.10.37/tests/libtracker-fts/tracker-parser-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-fts/tracker-parser-test.c	2016-04-14 00:14:23.104146486 +0300
@@ -350,7 +350,6 @@
 
 	g_type_init ();
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 	g_test_init (&argc, &argv, NULL);
 
diff -ur tracker-0.10.37/tests/libtracker-fts/tracker-parser.c tracker-0.10.37.1/tests/libtracker-fts/tracker-parser.c
--- tracker-0.10.37/tests/libtracker-fts/tracker-parser.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-fts/tracker-parser.c	2016-04-14 00:14:23.100117740 +0300
@@ -222,7 +222,6 @@
 {
 	g_type_init ();
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	/* Setup locale */
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-crawler-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-crawler-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-crawler-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-crawler-test.c	2016-04-14 00:14:23.086556232 +0300
@@ -320,7 +320,6 @@
       char **argv)
 {
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 	g_test_message ("Testing filesystem crawler");
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-miner-manager-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-miner-manager-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-miner-manager-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-miner-manager-test.c	2016-04-14 00:14:23.062175240 +0300
@@ -138,7 +138,6 @@
         gint result;
 
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
         
 	g_test_message ("Testing miner manager");
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-monitor-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-monitor-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-monitor-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-monitor-test.c	2016-04-14 00:14:23.080647280 +0300
@@ -1874,7 +1874,6 @@
 main (gint    argc,
       gchar **argv)
 {
-	g_thread_init (NULL);
 	g_type_init ();
 	g_test_init (&argc, &argv, NULL);
 
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-password-provider-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-password-provider-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-password-provider-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-password-provider-test.c	2016-04-14 00:14:23.083548226 +0300
@@ -106,7 +106,6 @@
 
 	g_type_init ();
 
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 	/* Set test environment up */
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-priority-queue-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-priority-queue-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-priority-queue-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-priority-queue-test.c	2016-04-14 00:14:23.058625867 +0300
@@ -226,7 +226,6 @@
       char **argv)
 {
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 	g_test_add_func ("/libtracker-miner/tracker-priority-queue/emptiness",
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-task-pool-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-task-pool-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-task-pool-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-task-pool-test.c	2016-04-14 00:14:23.060385058 +0300
@@ -161,7 +161,6 @@
 main (gint argc, gchar **argv)
 {
         g_type_init ();
-        g_thread_init (NULL);
         g_test_init (&argc, &argv, NULL);
 
         g_test_add_func ("/libtracker-miner/tracker-task-pool/limit_set",
diff -ur tracker-0.10.37/tests/libtracker-miner/tracker-thumbnailer-test.c tracker-0.10.37.1/tests/libtracker-miner/tracker-thumbnailer-test.c
--- tracker-0.10.37/tests/libtracker-miner/tracker-thumbnailer-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-miner/tracker-thumbnailer-test.c	2016-04-14 00:14:23.082229942 +0300
@@ -133,7 +133,6 @@
       char **argv)
 {
 	g_type_init ();
-	g_thread_init (NULL);
 	g_test_init (&argc, &argv, NULL);
 
 	g_test_message ("Testing thumbnailer");
diff -ur tracker-0.10.37/tests/libtracker-sparql/tracker-test.c tracker-0.10.37.1/tests/libtracker-sparql/tracker-test.c
--- tracker-0.10.37/tests/libtracker-sparql/tracker-test.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/tests/libtracker-sparql/tracker-test.c	2016-04-14 00:14:23.090195337 +0300
@@ -357,7 +357,6 @@
 {
 	int result;
 
-	g_thread_init (NULL);
 	g_type_init ();
 	g_test_init (&argc, &argv, NULL);
 
diff -ur tracker-0.10.37/utils/mtp/mtp-dummy.c tracker-0.10.37.1/utils/mtp/mtp-dummy.c
--- tracker-0.10.37/utils/mtp/mtp-dummy.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/utils/mtp/mtp-dummy.c	2016-04-14 00:14:22.921002671 +0300
@@ -530,7 +530,6 @@
 int main (int argc, char **argv)
 {
 	/* Initialize stuff */
-	g_thread_init (NULL);
 	g_type_init ();
 
 	/* Initialize locale support! */
diff -ur tracker-0.10.37/utils/tracker-sql/tracker-sql.c tracker-0.10.37.1/utils/tracker-sql/tracker-sql.c
--- tracker-0.10.37/utils/tracker-sql/tracker-sql.c	2011-12-02 13:50:04.000000000 +0400
+++ tracker-0.10.37.1/utils/tracker-sql/tracker-sql.c	2016-04-14 00:14:22.923138348 +0300
@@ -93,7 +93,6 @@
 
 	g_type_init ();
 	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
 	}
 
 	if (file) {
