diff -ur tracker-0.10.37/data/tracker-miner-flickr.desktop.in.in tracker-0.10.37.1/data/tracker-miner-flickr.desktop.in.in
--- tracker-0.10.37/data/tracker-miner-flickr.desktop.in.in	2011-06-08 12:06:26.000000000 +0400
+++ tracker-0.10.37.1/data/tracker-miner-flickr.desktop.in.in	2016-04-13 23:22:59.279620131 +0300
@@ -6,7 +6,7 @@
 Terminal=false
 Type=Application
 Categories=Utility;
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
 X-KDE-UniqueApplet=true
diff -ur tracker-0.10.37/data/tracker-miner-rss.desktop.in.in tracker-0.10.37.1/data/tracker-miner-rss.desktop.in.in
--- tracker-0.10.37/data/tracker-miner-rss.desktop.in.in	2011-06-08 12:06:26.000000000 +0400
+++ tracker-0.10.37.1/data/tracker-miner-rss.desktop.in.in	2016-04-13 23:22:59.278874949 +0300
@@ -6,7 +6,7 @@
 Terminal=false
 Type=Application
 Categories=Utility;
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
 X-KDE-UniqueApplet=true
