--- polib-1.2.0/tests/tests.py.orig
+++ polib-1.2.0/tests/tests.py
@@ -628,7 +628,7 @@
         Test for the POFile.save_as_mofile() method.
         """
         import distutils.spawn
-        msgfmt = distutils.spawn.find_executable('msgfmt')
+        msgfmt = distutils.spawn.find_executable('gmsgfmt')
         if msgfmt is None:
             try:
                 return unittest.skip('msgfmt is not installed')
