diff -rupN sudo-1.8.4p5.orig/compat/Makefile.in sudo-1.8.4p5/compat/Makefile.in
--- sudo-1.8.4p5.orig/compat/Makefile.in	2012-03-29 10:37:00.000000000 -0700
+++ sudo-1.8.4p5/compat/Makefile.in	2012-05-17 11:19:30.429987000 -0700
@@ -181,6 +181,8 @@ mktemp.lo: $(srcdir)/mktemp.c $(top_buil
 nanosleep.lo: $(srcdir)/nanosleep.c $(top_builddir)/config.h \
               $(top_srcdir)/compat/timespec.h $(incdir)/missing.h
 	$(LIBTOOL) --mode=compile $(CC) -c -o $@ $(CPPFLAGS) $(CFLAGS) $(DEFS) $(srcdir)/nanosleep.c
+pw_dup.lo: $(srcdir)/pw_dup.c $(top_builddir)/config.h
+	$(LIBTOOL) --mode=compile $(CC) -c -o $@ $(CPPFLAGS) $(CFLAGS) $(DEFS) $(srcdir)/pw_dup.c
 setenv.lo: $(srcdir)/setenv.c $(top_builddir)/config.h $(incdir)/missing.h
 	$(LIBTOOL) --mode=compile $(CC) -c -o $@ $(CPPFLAGS) $(CFLAGS) $(DEFS) $(srcdir)/setenv.c
 siglist.lo: siglist.c $(top_builddir)/config.h $(incdir)/missing.h
