--- sudo-1.7.4p4.orig/sudo.man.in	Wed Jun  1 16:45:17 2011
+++ sudo-1.7.4p4/sudo.man.in	Wed Jun  1 16:49:12 2011
@@ -1,3 +1,7 @@
+'\" t
+.\" Modified for Solaris to to add 
+.\" a note about auditing and source availability
+.\" 
 .\" Copyright (c) 1994-1996, 1998-2005, 2007-2010
 .\" 	Todd C. Miller <Todd.Miller@courtesan.com>
 .\" 
@@ -800,3 +804,8 @@
 and fitness for a particular purpose are disclaimed.  See the \s-1LICENSE\s0
 file distributed with \fBsudo\fR or http://www.sudo.ws/sudo/license.html
 for complete details.
+.\" Begin Solaris update
+.SH NOTES
+\fBsudo\fR does not create \fBaudit\fR(2) records; for a Role Based administration
+solution that performs auditing of all actions, please refer to \fBrbac\fR(5).
+.\" End Solaris update
