--- gcc-4.5.2/gcc/config/i386/sol2-10.h	Wed Jul 14 01:47:54 2010
+++ gcc-4.5.2/gcc/config/i386/sol2-10.h	Fri Aug  3 22:03:59 2012
@@ -108,13 +108,8 @@
   "%{G:-G} \
    %{YP,*} \
    %{R*} \
-   %{compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,/usr/ucblib/64:/usr/lib/libp/64:/lib/64:/usr/lib/64} \
-             %{!p:%{!pg:-Y P,/usr/ucblib/64:/lib:/usr/lib/64}}} \
-             -R /usr/ucblib/64} \
-   %{!compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,/usr/lib/libp/64:/lib/64:/usr/lib/64} \
-             %{!p:%{!pg:-Y P,/lib/64:/usr/lib/64}}}}"
+   %{!YP,*:%{p|pg:-Y P,%R/usr/lib/libp/64:%R/lib/64:%R/usr/lib/64} \
+           %{!p:%{!pg:-Y P,%R/lib/64:%R/usr/lib/64}}}"
 
 #undef LINK_ARCH64_SPEC
 #define LINK_ARCH64_SPEC LINK_ARCH64_SPEC_BASE
