--- gcc-4.5.2/gcc/config/sparc/sol2-bi.h	Fri Apr  2 12:54:46 2010
+++ gcc-4.5.2/gcc/config/sparc/sol2-bi.h	Fri Aug  3 21:59:09 2012
@@ -194,13 +194,8 @@
    %{G:-G} \
    %{YP,*} \
    %{R*} \
-   %{compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,%R/usr/ucblib/sparcv9:%R/usr/lib/libp/sparcv9:%R/usr/lib/sparcv9} \
-       %{!p:%{!pg:-Y P,%R/usr/ucblib/sparcv9:%R/usr/lib/sparcv9}}} \
-     -R %R/usr/ucblib/sparcv9} \
-   %{!compat-bsd: \
-     %{!YP,*:%{p|pg:-Y P,%R/usr/lib/libp/sparcv9:%R/usr/lib/sparcv9} \
-       %{!p:%{!pg:-Y P,%R/usr/lib/sparcv9}}}}"
+   %{!YP,*:%{p|pg:-Y P,%R/usr/lib/libp/64:%R/lib/64:%R/usr/lib/64} \
+     %{!p:%{!pg:-Y P,%R/lib/64:%R/usr/lib/64}}}"
 
 #define LINK_ARCH64_SPEC LINK_ARCH64_SPEC_BASE
 
