diff -Nur openttd-1.4.3.orig/src/core/alloc_func.hpp openttd-1.4.3/src/core/alloc_func.hpp
--- openttd-1.4.3.orig/src/core/alloc_func.hpp	2014-10-04 09:11:06.657504387 -0400
+++ openttd-1.4.3/src/core/alloc_func.hpp	2014-10-04 09:11:55.360884737 -0400
@@ -12,6 +12,8 @@
 #ifndef ALLOC_FUNC_HPP
 #define ALLOC_FUNC_HPP
 
+#include <alloca.h>
+
 /*
  * Functions to exit badly with an error message.
  * It has to be linked so the error messages are not
