diff -urN -x '*.orig' -x '*.rej' gnomecc.orig/capplets/accessibility/at-properties/main.c gnomecc.new/capplets/accessibility/at-properties/main.c
--- gnomecc.orig/capplets/accessibility/at-properties/main.c	2009-10-30 14:09:58.677822645 +0000
+++ gnomecc.new/capplets/accessibility/at-properties/main.c	2009-10-30 14:10:14.795670128 +0000
@@ -18,12 +18,29 @@
 #include "capplet-util.h"
 #include "gconf-property-editor.h"
 #include "activate-settings-daemon.h"
+#include <zone.h>
 
 #define ACCESSIBILITY_KEY       "/desktop/gnome/interface/accessibility"
 #define ACCESSIBILITY_KEY_DIR   "/desktop/gnome/interface"
 
 static gboolean initial_state;
 
+static gboolean
+tx_session ()
+{
+	static int ret = -1;
+
+	if (ret == -1) {
+		if (getzoneid () == 0 && getenv ("TRUSTED_SESSION")) {
+			ret = 1;
+		} else {
+			ret = 0;
+		}
+	}
+
+	return ret ? TRUE : FALSE;
+}
+
 static GtkBuilder *
 create_builder (void)
 {
@@ -54,6 +71,12 @@
 		}
 
 		g_free (prog);
+
+		if (tx_session ()) {
+			object = gtk_builder_get_object (builder, 
+				 			 "at_pref_button");
+			gtk_widget_hide (GTK_WIDGET (object));
+		}
 	} else {
 		g_warning ("Could not load UI: %s", error->message);
 		g_error_free (error);
diff -urN -x '*.orig' -x '*.rej' gnomecc.orig/capplets/appearance/appearance-desktop.c gnomecc.new/capplets/appearance/appearance-desktop.c
--- gnomecc.orig/capplets/appearance/appearance-desktop.c	2009-10-30 14:09:58.653599709 +0000
+++ gnomecc.new/capplets/appearance/appearance-desktop.c	2009-10-30 14:10:26.173294668 +0000
@@ -31,6 +31,8 @@
 #include <libgnomeui/gnome-desktop-thumbnail.h>
 #include <libgnomeui/gnome-bg.h>
 
+#include <tsol/label.h>
+
 enum {
   TARGET_URI_LIST,
   TARGET_BGIMAGE
@@ -1162,7 +1164,7 @@
 
   w = appearance_capplet_get_widget (data, "more_backgrounds_linkbutton");
   url = gconf_client_get_string (data->client, MORE_BACKGROUNDS_URL_KEY, NULL);
-  if (url != NULL && url[0] != '\0') {
+  if (!is_system_labeled () && url != NULL && url[0] != '\0') {
     gtk_link_button_set_uri (GTK_LINK_BUTTON (w), url);
     gtk_widget_show (w);
   } else {
diff -urN -x '*.orig' -x '*.rej' gnomecc.orig/capplets/appearance/appearance-themes.c gnomecc.new/capplets/appearance/appearance-themes.c
--- gnomecc.orig/capplets/appearance/appearance-themes.c	2009-10-30 14:09:58.651652280 +0000
+++ gnomecc.new/capplets/appearance/appearance-themes.c	2009-10-30 14:10:26.174570283 +0000
@@ -33,6 +33,8 @@
 #include <string.h>
 #include <libgnomeui/gnome-desktop-thumbnail.h>
 
+#include <tsol/label.h>
+
 #define CUSTOM_THEME_NAME "__custom__"
 
 enum
@@ -1130,7 +1132,7 @@
 
   w = appearance_capplet_get_widget (data, "more_themes_linkbutton");
   url = gconf_client_get_string (data->client, MORE_THEMES_URL_KEY, NULL);
-  if (url != NULL && url[0] != '\0') {
+  if (!is_system_labeled () && url != NULL && url[0] != '\0') {
     gtk_link_button_set_uri (GTK_LINK_BUTTON (w), url);
     gtk_widget_show (w);
   } else {
