diff -up gnome-control-center-2.24.0.1/capplets/appearance/appearance-font.c.orig gnome-control-center-2.24.0.1/capplets/appearance/appearance-font.c
--- gnome-control-center-2.24.0.1/capplets/appearance/appearance-font.c.orig	2008-11-14 10:46:31.000000000 +0000
+++ gnome-control-center-2.24.0.1/capplets/appearance/appearance-font.c	2008-11-14 11:08:45.694981062 +0000
@@ -864,7 +864,7 @@ dpi_load (GConfClient   *client,
   GConfValue *value;
   gdouble dpi;
 
-  value = gconf_client_get_without_default (client, FONT_DPI_KEY, NULL);
+  value = gconf_client_get(client, FONT_DPI_KEY, NULL);
 
   if (value) {
     dpi = gconf_value_get_float (value);

