--- hyperscan-5.4.1/examples/pcapscan.cc	2023-02-26 17:13:53.853385829 +0100
+++ hyperscan-5.4.1/examples/pcapscan.cc.new	2023-02-26 17:57:39.048349531 +0100
@@ -68,7 +68,8 @@
 #include <netinet/tcp.h>
 #include <netinet/udp.h>
 #include <netinet/ip_icmp.h>
-#include <net/ethernet.h>
+#include <libnet.h>
+#include <sys/ethernet.h>
 #include <arpa/inet.h>
 
 #include <pcap.h>
--- hyperscan-5.4.1/examples/patbench.cc	2022-09-13 14:57:08.000000000 +0200
+++ hyperscan-5.4.1/examples/patbench.cc.new	2023-02-26 18:25:39.051761823 +0100
@@ -134,7 +134,8 @@
 #include <netinet/tcp.h>
 #include <netinet/udp.h>
 #include <netinet/ip_icmp.h>
-#include <net/ethernet.h>
+#include <libnet.h>
+#include <sys/ethernet.h>
 #include <arpa/inet.h>
 
 #include <pcap.h>
