--- coreutils-9.10/tests/df/no-mtab-status.sh.orig
+++ coreutils-9.10/tests/df/no-mtab-status.sh
@@ -35,10 +35,14 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <mntent.h>
 #include <string.h>
 #include <stdarg.h>
 #include <dlfcn.h>
+#ifdef __illumos__
+#   include <sys/mntent.h>
+#else
+#   include <mntent.h>
+#endif
 
 static FILE *(*fopen_func) (const char *, const char *);
 
@@ -104,7 +108,11 @@
   return open_func(path, flags, mode);
 }
 
+#ifdef __illumos__
+int getmntent (FILE *fp, struct mnttab *mp)
+#else
 struct mntent *getmntent (FILE *fp)
+#end
 {
   /* Prove that LD_PRELOAD works. */
   static int done = 0;
@@ -115,7 +123,11 @@
     }
   /* Now simulate the failure. */
   errno = ENOENT;
+#ifdef __illumos__
+  return -1;
+#else
   return NULL;
+#endif
 }
 EOF
 
