--- coreutils-9.6/tests/df/no-mtab-status.sh.orig	2025-01-15 18:07:19.000000000 +0100
+++ coreutils-9.6/tests/df/no-mtab-status.sh	2025-01-19 18:25:45.042646204 +0100
@@ -34,10 +34,14 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <mntent.h>
 #include <string.h>
 #include <stdarg.h>
 #include <dlfcn.h>
+#ifdef __illumos__
+#   include <sys/mntent.h>
+#else
+#   include <mntent.h>
+#endif
 
 #define STREQ(a, b) (strcmp (a, b) == 0)
 
@@ -76,7 +80,11 @@
     return open_func(path, flags, mode);
 }
 
+#ifdef __illumos__
+int getmntent (FILE *fp, struct mnttab *mp)
+#else
 struct mntent *getmntent (FILE *fp)
+#end
 {
   /* Prove that LD_PRELOAD works. */
   static int done = 0;
@@ -87,7 +95,11 @@
     }
   /* Now simulate the failure. */
   errno = ENOENT;
+#ifdef __illumos__
+  return -1;
+#else
   return NULL;
+#endif
 }
 EOF
 
