diff -Nru ejabberd-22.05.orig/configure.ac ejabberd-22.05/configure.ac
--- ejabberd-22.05.orig/configure.ac	2022-05-06 15:12:22.000000000 +0000
+++ ejabberd-22.05/configure.ac	2022-06-06 15:52:22.296964977 +0000
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT(ejabberd, m4_esyscmd([echo `git describe --tags 2>/dev/null || echo 22.05` | sed 's/-g.*//;s/-/./' | tr -d '\012']), [ejabberd@process-one.net], [ejabberd])
+AC_INIT(ejabberd, [22.05], [ejabberd@process-one.net], [ejabberd])
 REQUIRE_ERLANG_MIN="8.3 (Erlang/OTP 19.3)"
 REQUIRE_ERLANG_MAX="100.0.0 (No Max)"
 
