--- ejabberd-24.02/configure.ac.orig	2024-02-27 10:32:55.000000000 +0100
+++ ejabberd-24.02/configure.ac	2024-04-02 10:43:54.514925690 +0200
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT(ejabberd, m4_esyscmd([echo `git describe --tags 2>/dev/null || echo 24.02` | sed 's/-g.*//;s/-/./' | tr -d '\012']), [ejabberd@process-one.net], [ejabberd])
+AC_INIT(ejabberd, [24.02], [ejabberd@process-one.net], [ejabberd])
 REQUIRE_ERLANG_MIN="9.0.5 (Erlang/OTP 20.0)"
 REQUIRE_ERLANG_MAX="100.0.0 (No Max)"
 
