--- OpenColorIO-2.1.3/src/OpenColorIO/CMakeLists.txt.orig	2023-01-06 02:36:46.000000000 +0100
+++ OpenColorIO-2.1.3/src/OpenColorIO/CMakeLists.txt	2023-11-10 11:40:57.981433469 +0100
@@ -305,8 +305,8 @@
     # Also hide all the symbols of dependent libraries to prevent clashes if
     # an app using this project is linked against other versions of our
     # dependencies.
-    set_property (TARGET OpenColorIO
-                  APPEND PROPERTY LINK_FLAGS "-Wl,--exclude-libs,ALL")
+    #    set_property (TARGET OpenColorIO
+    #             APPEND PROPERTY LINK_FLAGS "-Wl,--exclude-libs,ALL")
 elseif(APPLE)
     # Hide the expat symbols.
     set_property (TARGET OpenColorIO
