--- OpenColorIO-tags-v2.1.1/share/cmake/modules/Findyaml-cpp.cmake	週五 12月 17 03:19:03 2021
+++ OpenColorIO-tags-v2.1.1/share/cmake/modules/Findyaml-cpp_.cmake	週六 4月 23 07:19:08 2022
@@ -132,7 +132,7 @@
     # TODO: yaml-cpp master is using GNUInstallDirs to define include and lib 
     #       dir names. Once that change is released and OCIO updates the 
     #       minimum yaml-cpp version, toggle the three disabled lines below.
-    #include(GNUInstallDirs)
+    include(GNUInstallDirs)
 
     set(_EXT_DIST_ROOT "${CMAKE_BINARY_DIR}/ext/dist")
     set(_EXT_BUILD_ROOT "${CMAKE_BINARY_DIR}/ext/build")
@@ -150,8 +150,10 @@
             string(APPEND _yaml-cpp_LIB_SUFFIX "d")
         endif()
     endif()
+	set(CMAKE_INSTALL_LIBDIR "lib/amd64")
+	set(yaml-cpp_CXX_FLAGS "-m64 -fPIC")
     set(yaml-cpp_LIBRARY
-        "${_EXT_DIST_ROOT}/lib/libyaml-cpp${_yaml-cpp_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+        "${_EXT_DIST_ROOT}/lib/amd64/libyaml-cpp${_yaml-cpp_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
 
     if(_yaml-cpp_TARGET_CREATE)
         if(MSVC)
@@ -179,6 +180,7 @@
             -DCMAKE_INSTALL_MESSAGE=${CMAKE_INSTALL_MESSAGE}
             -DCMAKE_INSTALL_PREFIX=${_EXT_DIST_ROOT}
             -DCMAKE_OBJECT_PATH_MAX=${CMAKE_OBJECT_PATH_MAX}
+            -DCMAKE_INSTALL_LIBDIR=lib/amd64
             -DBUILD_SHARED_LIBS=OFF
             -DYAML_BUILD_SHARED_LIBS=OFF
             -DYAML_CPP_BUILD_TESTS=OFF
