--- OpenColorIO-tags-v2.1.1/share/cmake/modules/Findpystring.cmake	週五 12月 17 03:19:03 2021
+++ OpenColorIO-tags-v2.1.1/share/cmake/modules/Findpystring_.cmake	週六 4月 23 07:45:35 2022
@@ -66,6 +66,7 @@
 
 if(NOT pystring_FOUND)
     include(ExternalProject)
+       include(GNUInstallDirs)
 
     set(_EXT_DIST_ROOT "${CMAKE_BINARY_DIR}/ext/dist")
     set(_EXT_BUILD_ROOT "${CMAKE_BINARY_DIR}/ext/build")
@@ -75,8 +75,9 @@
     set(pystring_VERSION ${pystring_FIND_VERSION})
     set(pystring_INCLUDE_DIR "${_EXT_DIST_ROOT}/include")
     set(pystring_LIBRARY 
-        "${_EXT_DIST_ROOT}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}pystring${CMAKE_STATIC_LIBRARY_SUFFIX}")
-
+        "${_EXT_DIST_ROOT}/lib/amd64/${CMAKE_STATIC_LIBRARY_PREFIX}pystring${CMAKE_STATIC_LIBRARY_SUFFIX}")
+	set(CMAKE_INSTALL_LIBDIR "lib/amd64")
+	set(pystring_CXX_FLAGS "-m64 -fPIC")
     if(_pystring_TARGET_CREATE)
         if(MSVC)
             set(pystring_CXX_FLAGS "${pystring_CXX_FLAGS} /EHsc")
@@ -95,6 +95,8 @@
             -DCMAKE_INSTALL_MESSAGE=${CMAKE_INSTALL_MESSAGE}
             -DCMAKE_INSTALL_PREFIX=${_EXT_DIST_ROOT}
+            -DCMAKE_INSTALL_COMPONENT=file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/amd64" TYPE STATIC_LIBRARY MESSAGE_NEVER FILES "${_EXT_BUILD_ROOT}/amd64/ext/build/pystring/src/pystring_install-build/libpystring.a")
             -DCMAKE_OBJECT_PATH_MAX=${CMAKE_OBJECT_PATH_MAX}
+            -DCMAKE_INSTALL_LIBDIR=lib/amd64
         )
 
         if(CMAKE_TOOLCHAIN_FILE)
