OpenIndiana has its python packages in vendor-packages folder.

--- OpenColorIO-2.1.3/src/bindings/python/CMakeLists.txt.orig	2023-01-06 02:36:46.000000000 +0100
+++ OpenColorIO-2.1.3/src/bindings/python/CMakeLists.txt	2023-11-10 12:09:45.056930447 +0100
@@ -228,7 +228,7 @@
 if(WIN32)
     set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/site-packages")
 else()
-    set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages")
+    set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/vendor-packages")
 endif()
 
 # Create an internal global variable to access it in another scope but not publicly visible
