OpenIndiana has its python packages in vendor-packages folder.

--- OpenColorIO-2.4.0/src/bindings/python/CMakeLists.txt.orig
+++ OpenColorIO-2.4.0/src/bindings/python/CMakeLists.txt
@@ -241,7 +241,7 @@
 if(WIN32)
     set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/site-packages")
 else()
-    set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages")
+    set(_Python_VARIANT_PATH "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/vendor-packages")
 endif()
 
 # Create an internal global variable to access it in another scope but not publicly visible.
