--- gnu-cobol-1.1/configure.orig	2017-06-04 00:39:11.023473329 +0000
+++ gnu-cobol-1.1/configure	2017-06-04 00:39:37.315548643 +0000
@@ -26647,8 +26647,8 @@
 
 if test "`basename $COB_CC`" = "gcc"
 then
-	cob_temp_flags="`${COB_CC} ${srcdir}/cpucheck.c -o cpucheck 1>/dev/null 2>&1 && ./cpucheck 2>/dev/null`"
-	rm -f cpucheck${EXEEXT}
+	#cob_temp_flags="`${COB_CC} ${srcdir}/cpucheck.c -o cpucheck 1>/dev/null 2>&1 && ./cpucheck 2>/dev/null`"
+	#rm -f cpucheck${EXEEXT}
 	if test "x$cob_temp_flags" != "x"
 	then
 		CFLAGS="$CFLAGS $cob_temp_flags -finline-functions"
--- gnu-cobol-1.1/configure.ac.1	2018-01-28 21:03:00.667171771 +0000
+++ gnu-cobol-1.1/configure.ac	2018-01-28 21:03:27.962670315 +0000
@@ -650,8 +650,8 @@
 
 if test "`basename $COB_CC`" = "gcc"
 then
-	cob_temp_flags="`${COB_CC} ${srcdir}/cpucheck.c -o cpucheck 1>/dev/null 2>&1 && ./cpucheck 2>/dev/null`"
-	rm -f cpucheck${EXEEXT}
+#	cob_temp_flags="`${COB_CC} ${srcdir}/cpucheck.c -o cpucheck 1>/dev/null 2>&1 && ./cpucheck 2>/dev/null`"
+#	rm -f cpucheck${EXEEXT}
 	if test "x$cob_temp_flags" != "x"
 	then
 		CFLAGS="$CFLAGS $cob_temp_flags -finline-functions"
