--- gnu-cobol-1.1/tests/Makefile.am.orig	2017-06-04 00:33:03.202756316 +0000
+++ gnu-cobol-1.1/tests/Makefile.am	2017-06-04 00:33:46.086966909 +0000
@@ -24,11 +24,11 @@
 
 SUBDIRS = cobol85
 
-TESTS = $(srcdir)/syntax \
-	$(srcdir)/run \
-	$(srcdir)/run-O \
-	$(srcdir)/data-rep \
-	$(srcdir)/data-rep-O
+TESTS = syntax \
+	run \
+	run-O \
+	data-rep \
+	data-rep-O
 
 syntax_DEPENDENCIES = \
 	syntax.at \
