--- scons-src-2.4.1/src/engine/SCons/Tool/__init__.py.1	2017-08-11 13:17:37.318200642 +0000
+++ scons-src-2.4.1/src/engine/SCons/Tool/__init__.py	2017-08-11 13:20:59.209866346 +0000
@@ -1020,13 +1020,13 @@
         fortran_compilers = ['f95', 'f90', 'f77', 'g77', 'fortran']
         ars = ['sgiar']
     elif str(platform) == 'sunos':
-        "prefer Forte tools on SunOS"
+        "prefer GCC tools on SunOS"
         linkers = ['sunlink', 'gnulink']
-        c_compilers = ['suncc', 'gcc', 'cc']
-        cxx_compilers = ['sunc++', 'g++', 'c++']
-        assemblers = ['as', 'gas']
-        fortran_compilers = ['sunf95', 'sunf90', 'sunf77', 'f95', 'f90', 'f77',
-                             'gfortran', 'g77', 'fortran']
+        c_compilers = ['gcc', 'cc','suncc']
+        cxx_compilers = ['g++', 'c++', 'sunc++']
+        assemblers = ['gas', 'as']
+        fortran_compilers = ['gfortran', 'g77', 'f95', 'f90', 'f77',
+                              'fortran', 'sunf95', 'sunf90', 'sunf77']
         ars = ['sunar']
     elif str(platform) == 'hpux':
         "prefer aCC tools on HP-UX"
