--- scons-src-3.1.0/src/engine/SCons/Tool/__init__.py.~1~	2019-07-29 15:20:52.592718204 +0000
+++ scons-src-3.1.0/src/engine/SCons/Tool/__init__.py	2019-07-29 16:01:13.620013678 +0000
@@ -1216,13 +1216,13 @@
         fortran_compilers = ['f95', 'f90', 'f77', 'g77', 'fortran']
         ars = ['sgiar']
     elif str(platform) == 'sunos':
-        "prefer Forte tools on SunOS"
-        linkers = ['sunlink', 'gnulink']
-        c_compilers = ['suncc', 'gcc', 'cc']
-        cxx_compilers = ['suncxx', 'g++', 'cxx']
-        assemblers = ['as', 'gas']
-        fortran_compilers = ['sunf95', 'sunf90', 'sunf77', 'f95', 'f90', 'f77',
-                             'gfortran', 'g77', 'fortran']
+        "prefer GCC tools on SunOS"
+        linkers = ['gnulink', 'sunlink']
+        c_compilers = ['gcc', 'cc', 'suncc']
+        cxx_compilers = ['g++', 'cxx', 'suncxx']
+        assemblers = ['gas', 'as']
+        fortran_compilers = ['gfortran', 'g77', 'f95', 'f90', 'f77',
+                             'fortran', 'sunf95', 'sunf90', 'sunf77']
         ars = ['sunar']
     elif str(platform) == 'hpux':
         "prefer aCC tools on HP-UX"
