# parfait bug 17813405 
--- librabbitmq-1.0.1/clib/librabbitmq/amqp_private.h.orig	Mon Mar 10 13:37:33 2014
+++ librabbitmq-1.0.1/clib/librabbitmq/amqp_private.h	Mon Mar 10 13:37:44 2014
@@ -51,7 +51,7 @@
 # elif defined(__GNUC__) && !defined(__GNUC_STDC_INLINE__)
 #  define _AMQP_INLINE extern __inline
 # else
-#  define _AMQP_INLINE __inline
+#  define _AMQP_INLINE static inline
 # endif
 #endif
 
