--- gtk+-2.20.1/modules/printbackends/papi/gtkprintbackendpapi.c.~1~	2014-02-24 11:19:29.064783467 +0400
+++ gtk+-2.20.1/modules/printbackends/papi/gtkprintbackendpapi.c	2014-02-24 11:20:27.194411952 +0400
@@ -352,7 +352,7 @@
   const gchar *title;
   char *prtnm = NULL;
   GtkPrintDuplex val;
-  papi_status_t pstatus = NULL;
+  papi_status_t pstatus;
   papi_attribute_t **attrs = NULL;
   papi_job_ticket_t *ticket = NULL;
   
