--- ../i386/gtk+-2.16.0/gtk/gtkscalebutton.c	2009-03-13 14:40:33.000000000 +0100
+++ gtk+-2.16.0/gtk/gtkscalebutton.c	2009-03-17 18:35:24.927048961 +0100
@@ -423,6 +423,10 @@
   return object;
 }
 
+void
+gtk_scale_button_set_orientation (GtkScaleButton *button,
+                                  GtkOrientation  orientation);
+
 static void
 gtk_scale_button_set_property (GObject       *object,
 			       guint          prop_id,
