--- claws-mail-3.13.2/src/Makefile.in.orig	2016-04-26 20:19:06.408524973 +0300
+++ claws-mail-3.13.2/src/Makefile.in	2016-04-26 20:19:13.728311248 +0300
@@ -1130,7 +1130,7 @@
 	-I$(srcdir)/gtk \
 	-I$(srcdir)/etpan
 
-@PLATFORM_WIN32_FALSE@arch_ldflags := -Wl,--export-dynamic
+@PLATFORM_WIN32_FALSE@arch_ldflags :=
 @PLATFORM_WIN32_TRUE@arch_ldflags := -Wl,--export-all-symbols
 claws_mail_LDFLAGS = \
 	$(arch_ldflags) $(claws_mail_res_ldflag)
