--- claws-mail-4.1.1/get-git-version.orig	2022-09-23 17:27:46.000000000 +0000
+++ claws-mail-4.1.1/get-git-version	2022-10-22 10:23:32.190943851 +0000
@@ -9,9 +9,9 @@
     exit 1
   fi
 elif test \( -s ./version \); then
-  version=`sh -c ". ./version"`
+  version=`bash -c ". ./version"`
 else
   echo "Error: *** invalid tree. Neither '.git' nor 'version' were found"
   exit 1
 fi
-echo -n $version
+gecho -n $version
