https://github.com/joke2k/faker/issues/2019

--- Faker-24.7.1/tests/providers/test_phone_number.py.orig
+++ Faker-24.7.1/tests/providers/test_phone_number.py
@@ -73,16 +73,6 @@
             assert pattern.fullmatch(phone_number)
 
 
-class TestSkSk:
-    """Test sk_SK phone number provider methods"""
-
-    def test_phone_number(self, faker, num_samples):
-        pattern: Pattern = re.compile(r"^(00421|\+421)? ?[2-9][0-9]{2} ?[0-9]{3} ?[0-9]{3}$")
-        for _ in range(num_samples):
-            phone_number = faker.phone_number()
-            assert pattern.fullmatch(phone_number)
-
-
 class TestPtBr:
     """Test pt_BR phone number provider methods"""
 
