--- boost_1_63_0/libs/log/build/Jamfile.v2.orig	2017-04-09 06:21:11.924221737 +0200
+++ boost_1_63_0/libs/log/build/Jamfile.v2	2017-04-09 06:22:17.784445352 +0200
@@ -253,7 +253,7 @@
         <target-os>linux:<define>_XOPEN_SOURCE=600
         <target-os>linux:<library>rt
 
-        <target-os>solaris:<define>_XOPEN_SOURCE=500 # Solaris headers are broken and cannot be included in C++ when _XOPEN_SOURCE=600
+        <target-os>solaris:<define>_XOPEN_SOURCE=600 # If it were true why would we need to patch your ass all over again...
         <target-os>solaris:<define>__EXTENSIONS__
         <target-os>solaris:<library>socket
         <target-os>solaris:<library>nsl
