--- fish-4.0.1/src/wutil/dir_iter.rs.old	2025-04-13 23:32:47.449072755 -0400
+++ fish-4.0.1/src/wutil/dir_iter.rs	2025-04-13 23:34:42.689522160 -0400
@@ -20,7 +20,7 @@
     chr,      // character device
     dir,      // directory
     blk,      // block device
-    reg,      // regular file
+    regr,      // regular file
     lnk,      // symlink
     sock,     // socket
     whiteout, // whiteout (from BSD)
@@ -140,7 +140,7 @@
         DT_CHR => Some(DirEntryType::chr),
         DT_DIR => Some(DirEntryType::dir),
         DT_BLK => Some(DirEntryType::blk),
-        DT_REG => Some(DirEntryType::reg),
+        DT_REG => Some(DirEntryType::regr),
         DT_LNK => Some(DirEntryType::lnk),
         DT_SOCK => Some(DirEntryType::sock),
         // todo!("whiteout")
@@ -154,7 +154,7 @@
         S_IFCHR => Some(DirEntryType::chr),
         S_IFDIR => Some(DirEntryType::dir),
         S_IFBLK => Some(DirEntryType::blk),
-        S_IFREG => Some(DirEntryType::reg),
+        S_IFREG => Some(DirEntryType::regr),
         S_IFLNK => Some(DirEntryType::lnk),
         S_IFSOCK => Some(DirEntryType::sock),
         _ => {
@@ -447,9 +447,9 @@
         let expected = if entry.name == dirname {
             Some(DirEntryType::dir)
         } else if entry.name == regname {
-            Some(DirEntryType::reg)
+            Some(DirEntryType::regr)
         } else if entry.name == reglinkname {
-            Some(DirEntryType::reg)
+            Some(DirEntryType::regr)
         } else if entry.name == dirlinkname {
             Some(DirEntryType::dir)
         } else if entry.name == badlinkname {
--- fish-4.0.1/src/wildcard.rs.old	2025-04-13 23:39:07.237912731 -0400
+++ fish-4.0.1/src/wildcard.rs	2025-04-13 23:39:26.301942052 -0400
@@ -379,7 +379,7 @@
     // regular file *excludes* broken links - we have no use for them as commands.
     let is_regular_file = entry
         .check_type()
-        .map(|x| x == DirEntryType::reg)
+        .map(|x| x == DirEntryType::regr)
         .unwrap_or(false);
     if executables_only && (!is_regular_file || waccess(filepath, X_OK) != 0) {
         return false;
