--- fish-4.2.0/src/builtins/ulimit.rs.old	2025-11-11 08:52:53.735806386 -0500
+++ fish-4.2.0/src/builtins/ulimit.rs	2025-11-11 08:55:03.968404454 -0500
@@ -21,7 +21,7 @@
         pub const DATA: libc::c_int = libc::RLIMIT_DATA as _;
         pub const FSIZE: libc::c_int = libc::RLIMIT_FSIZE as _;
         cfg_if!(
-            if #[cfg(cygwin)] {
+            if #[cfg(any(cygwin,illumos))] {
                 pub const MEMLOCK: libc::c_int = -1;
             } else {
                 pub const MEMLOCK: libc::c_int = libc::RLIMIT_MEMLOCK as _;
@@ -31,7 +31,7 @@
         pub const STACK: libc::c_int = libc::RLIMIT_STACK as _;
         pub const CPU: libc::c_int = libc::RLIMIT_CPU as _;
         cfg_if!(
-            if #[cfg(cygwin)] {
+            if #[cfg(any(cygwin,illumos))] {
                 pub const NPROC: libc::c_int = -1;
             } else {
                 pub const NPROC: libc::c_int = libc::RLIMIT_NPROC as _;
