--- xcowsay-1.5.1/src/Makefile.am.orig	2020-02-02 10:44:18.000000000 +0000
+++ xcowsay-1.5.1/src/Makefile.am	2020-10-18 17:42:30.337575868 +0000
@@ -4,7 +4,7 @@
 GTK3_CHECK = -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -DGSEAL_ENABLE
 GTK3_CHECK += -DGDK_DISABLE_DEPRECATED
 
-AM_CFLAGS = $(XCOWSAY_CFLAGS) -Wall $(GTK3_CHECK)
+AM_CFLAGS = $(XCOWSAY_CFLAGS) $(GTK3_CHECK)
 LDADD = $(XCOWSAY_LIBS)
 
 xcowsay_SOURCES = xcowsay.c display_cow.c display_cow.h floating_shape.h \
--- xcowsay-1.5.1/src/Makefile.in.orig	2020-10-01 06:27:05.000000000 +0000
+++ xcowsay-1.5.1/src/Makefile.in	2020-10-18 17:42:19.682031165 +0000
@@ -327,7 +327,7 @@
 bin_SCRIPTS = xcowfortune xcowdream xcowthink
 GTK3_CHECK = -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES \
 	-DGSEAL_ENABLE -DGDK_DISABLE_DEPRECATED
-AM_CFLAGS = $(XCOWSAY_CFLAGS) -Wall $(GTK3_CHECK)
+AM_CFLAGS = $(XCOWSAY_CFLAGS) $(GTK3_CHECK)
 LDADD = $(XCOWSAY_LIBS)
 xcowsay_SOURCES = xcowsay.c display_cow.c display_cow.h floating_shape.h \
 	floating_shape.c settings.h settings.c Cowsay_glue.h xcowsayd.h \
