--- xcowsay-1.4/src/Makefile.in.orig	2017-05-28 16:59:55.302311143 +0000
+++ xcowsay-1.4/src/Makefile.in	2017-05-28 17:00:22.334676739 +0000
@@ -324,7 +324,7 @@
 GTK3_CHECK = #-DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED \
 	-DGTK_DISABLE_SINGLE_INCLUDES -DGSEAL_ENABLE
 
-AM_CFLAGS = $(XCOWSAY_CFLAGS) -Wall $(GTK3_CHECK)
+AM_CFLAGS = $(XCOWSAY_CFLAGS) $(GTK3_CHECK)
 LDADD = $(XCOWSAY_LIBS)
 xcowsay_SOURCES = xcowsay.c display_cow.c display_cow.h floating_shape.h \
 	floating_shape.c settings.h settings.c Cowsay_glue.h xcowsayd.h \
--- xcowsay-1.4/src/Makefile.am.orig	2017-05-28 17:00:01.627186175 +0000
+++ xcowsay-1.4/src/Makefile.am	2017-05-28 17:00:29.501125109 +0000
@@ -3,7 +3,7 @@
 
 GTK3_CHECK = #-DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED \
 	-DGTK_DISABLE_SINGLE_INCLUDES -DGSEAL_ENABLE
-AM_CFLAGS = $(XCOWSAY_CFLAGS) -Wall $(GTK3_CHECK)
+AM_CFLAGS = $(XCOWSAY_CFLAGS) $(GTK3_CHECK)
 LDADD = $(XCOWSAY_LIBS)
 
 xcowsay_SOURCES = xcowsay.c display_cow.c display_cow.h floating_shape.h \
