--- git-crypt-0.7.0/Makefile.orig	2022-04-21 19:08:16.000000000 +0000
+++ git-crypt-0.7.0/Makefile	2022-11-17 14:03:24.825736229 +0000
@@ -6,11 +6,11 @@
 
 CXXFLAGS ?= -Wall -pedantic -Wno-long-long -O2
 CXXFLAGS += -std=c++11
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR ?= $(PREFIX)/bin
 MANDIR ?= $(PREFIX)/share/man
 
-ENABLE_MAN ?= no
+ENABLE_MAN ?= yes
 DOCBOOK_XSL ?= http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl
 
 OBJFILES = \
