--- a/modules/om/generic/omGeneric.c	Mon Mar 16 09:27:28 2015
+++ b/modules/om/generic/omGeneric.c	Mon Mar 16 09:29:59 2015
@@ -849,7 +849,8 @@
     if ((font_data->xlfd_name = get_font_name(oc, buf)) != NULL)
 	return True;
 
-    /* This may mot be needed anymore as XListFonts() takes care of this */
+    /* This may not be needed anymore as XListFonts() takes care of this */
+#ifndef SUNSOFT
     if (num_fields < 12) {
 	if ((last - buf) > (XLFD_MAX_LEN - 2))
 	    return -1;
@@ -861,8 +862,8 @@
 	if ((font_data->xlfd_name = get_font_name(oc, buf)) != NULL)
 	    return True;
     }
+#endif /* SUNSOFT */
 
-
     return False;
 }
 
