--- uhttpmock-0.5.0/m4/ax_compiler_flags_ldflags.m4.1	2018-04-04 10:36:59.568613400 +0000
+++ uhttpmock-0.5.0/m4/ax_compiler_flags_ldflags.m4	2018-04-04 10:37:45.164259138 +0000
@@ -48,11 +48,6 @@
     ])
 
     # Base flags
-    AX_APPEND_COMPILE_FLAGS([ dnl
-        -Wl,--no-as-needed dnl
-        $3 dnl
-    ],ax_warn_ldflags_variable,[$ax_compiler_flags_test])
-
     AS_IF([test "$ax_enable_compile_warnings" != "no"],[
         # "minimum" flags
         AX_APPEND_COMPILE_FLAGS([$4],
