--- logrotate-3.17.0/logrotate.8	2020-07-10 11:05:38.000000000 +0000
+++ logrotate-3.17.0/logrotate.8.bak	2020-12-03 16:45:12.653322906 +0000
@@ -1,4 +1,4 @@
-.TH LOGROTATE 8 "3.17.0" "Linux" "System Administrator's Manual"
+.TH LOGROTATE 8 "3.17.0" "System Administrator's Manual"
 .\" Per groff_man(7), the TQ macro should be copied from an-ext.tmac when
 .\" not running under groff.  That's not quite right; not all groff
 .\" installations include this macro.  So bring it in with another name
@@ -727,6 +727,7 @@ l l l.
 .BR chmod (2),
 .BR gunzip (1),
 .BR gzip (1),
+.BR logadm (1M),
 .BR mail (1),
 .BR shred (1),
 .BR strftime (3),

--- logrotate-3.15.0/logrotate.conf.5	2017-12-06 15:39:03.000000000 +0000
+++ logrotate-3.15.0/logrotate.conf.5.new	2019-03-07 19:42:50.619161542 +0000
@@ -1 +1 @@
-.so man8/logrotate.8
+.so man1/logrotate.1
