--- logrotate-3.15.0/examples/wtmp	2018-05-25 14:31:21.000000000 +0000
+++ logrotate-3.15.0/examples/wtmp.new	2019-03-07 18:44:14.390703981 +0000
@@ -1,8 +1,8 @@
-# no packages own wtmp -- we'll rotate it here
-/var/log/wtmp {
+# no packages own wtmpx -- we'll rotate it here
+/var/adm/wtmpx {
     missingok
     monthly
-    create 0664 root utmp
+    create 0664 root bin
     minsize 1M
     rotate 1
 }
--- logrotate-3.15.1/examples/btmp	2019-08-30 15:04:51.000000000 +0000
+++ logrotate-3.15.1/examples/btmp.new	2019-08-30 16:52:24.045593850 +0000
@@ -1,7 +1,7 @@
-# no packages own btmp -- we'll rotate it here
-/var/log/btmp {
+# no packages own utmpx -- we'll rotate it here
+/var/adm/utmpx {
     missingok
     monthly
-    create 0660 root utmp
+    create 0660 adm adm
     rotate 1
 }
