--- pkgtree-1.1/bin/pkgtree	2013-11-22 06:33:03.000000000 +0000
+++ pkgtree	2017-10-25 16:44:13.137477038 +0000
@@ -146,7 +146,6 @@
 delete $args{'cache-file'};
 delete $args{'force-cache'};
 delete $args{'no-cache'};
-delete $args{'no-pager'};
 delete $args{'no-colour'};
 
 my @output;
@@ -171,7 +170,7 @@
 # If STDOUT is connected to a terminal and --no-pager was not specified
 # then send the result to the pager
 #
-if (-t STDOUT and !$args{'--no-pager'}) {
+if (-t STDOUT and !$args{'no-pager'}) {
     my $pager = $ENV{PAGER} || 'less';
 
     #
