--- llvm-8.0.1.src/include/llvm/Demangle/MicrosoftDemangleNodes.h.orig	2020-05-13 11:06:54.758406084 +0000
+++ llvm-8.0.1.src/include/llvm/Demangle/MicrosoftDemangleNodes.h	2020-05-13 11:07:06.660524025 +0000
@@ -4,6 +4,7 @@
 #include "llvm/Demangle/Compiler.h"
 #include "llvm/Demangle/StringView.h"
 #include <array>
+#include <string>
 
 class OutputStream;
 
@@ -602,4 +603,4 @@
 } // namespace ms_demangle
 } // namespace llvm
 
-#endif
\ No newline at end of file
+#endif
