http://bugs.proftpd.org/show_bug.cgi?id=4132

diff --git a/src/log.c b/src/log.c
index 4b83413..c83b8ee 100644
--- a/src/log.c
+++ b/src/log.c
@@ -36,7 +36,7 @@
 static int syslog_open = FALSE;
 static int syslog_discard = FALSE;
 static int logstderr = TRUE;
-static int debug_level = DEBUG0;	/* Default is no debug logging */
+static int debug_level = DEBUG0 - 1;	/* Default is no debug logging */
 static int facility = LOG_DAEMON;
 static int set_facility = -1;
 static char systemlog_fn[PR_TUNABLE_PATH_MAX] = {'\0'};
