--- a/modules/mod_ctrls.c
+++ b/modules/mod_ctrls.c
@@ -1157,7 +1157,8 @@ static void ctrls_shutdown_ev(const void *event_data, void *user_data) {
 }
 
 static void ctrls_postparse_ev(const void *event_data, void *user_data) {
-  if (!ctrls_engine) {
+  if (ctrls_engine == FALSE ||
+      ServerType == SERVER_INETD) {
     return;
   }
 
diff --git a/src/ctrls.c b/src/ctrls.c
index 232bb7d..1f47c73 100644
--- a/src/ctrls.c
+++ b/src/ctrls.c
@@ -1975,7 +1975,7 @@ void init_ctrls(void) {
 
   sockfd = socket(AF_UNIX, SOCK_STREAM, 0);
   if (sockfd < 0) {
-    pr_log_pri(PR_LOG_DEBUG, "unable to create Unix domain socket: %s",
+    pr_log_debug(DEBUG10, "unable to create Unix domain socket: %s",
       strerror(errno));
     return;
   }
@@ -1986,7 +1986,7 @@ void init_ctrls(void) {
   socklen = sizeof(struct sockaddr_un);
 
   if (bind(sockfd, (struct sockaddr *) &sockun, socklen) < 0) {
-    pr_log_pri(PR_LOG_DEBUG,
+    pr_log_debug(DEBUG10,
       "unable to bind to Unix domain socket at '%s': %s",
       sockpath, strerror(errno));
     (void) close(sockfd);
@@ -1995,7 +1995,7 @@ void init_ctrls(void) {
   }
 
   if (fstat(sockfd, &st) < 0) {
-    pr_log_pri(PR_LOG_DEBUG,
+    pr_log_debug(DEBUG10,
       "unable to stat Unix domain socket at '%s': %s",
       sockpath, strerror(errno));
     (void) close(sockfd);

