--- a/src/inet.c
+++ b/src/inet.c
@@ -1238,7 +1238,7 @@ conn_t *pr_inet_openrw(pool *p, conn_t *c, pr_netaddr_t *addr, int strm_type,
   res->mode = CM_OPEN;
 
 #if defined(HAVE_STROPTS_H) && defined(I_SRDOPT) && defined(RPROTDIS) && \
-    defined(SOLARIS2)
+    defined(SOLARIS2) && !defined(SOLARIS2_11)
   /* This is needed to work around control messages in STREAMS devices
    * (as on Solaris 9/NFS).
    */

