--- yelp-3.20.1/m4/ax_compiler_flags_ldflags.m4.1	2016-05-24 12:38:23.174847774 +0300
+++ yelp-3.20.1/m4/ax_compiler_flags_ldflags.m4	2016-05-24 12:38:38.400271278 +0300
@@ -48,10 +48,6 @@
     ])
 
     # Base flags
-    AX_APPEND_COMPILE_FLAGS([ dnl
-        -Wl,--no-as-needed dnl
-        $3 dnl
-    ],ax_warn_ldflags_variable,[$ax_compiler_flags_test])
 
     AS_IF([test "$ax_enable_compile_warnings" != "no"],[
         # "minimum" flags
