--- terminator-2.1.3/terminatorlib/layoutlauncher.py.orig
+++ terminator-2.1.3/terminatorlib/layoutlauncher.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """layoutlauncher.py - class for the Layout Launcher window"""
--- terminator-2.1.3/terminatorlib/prefseditor.py.orig
+++ terminator-2.1.3/terminatorlib/prefseditor.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 """Preferences Editor for Terminator. 
 
 Load a UIBuilder config file, display it,
