--- terminator-0.92/terminatorlib/config.py.orig	2009-12-07 14:11:11.005441328 +0800
+++ terminator-0.92/terminatorlib/config.py	2009-12-07 14:11:18.610994113 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    TerminatorConfig - layered config classes
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/translation.py.orig	2009-12-07 14:12:55.054782550 +0800
+++ terminator-0.92/terminatorlib/translation.py	2009-12-07 14:13:02.884510662 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    Terminator - multiple gnome terminals in one window
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/freebsd.py.orig	2009-12-07 14:14:19.472054807 +0800
+++ terminator-0.92/terminatorlib/freebsd.py	2009-12-07 14:14:26.621901549 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #
 # Copyright (c) 2008, Thomas Hurst <tom@hur.st>
 #
--- terminator-0.92/terminatorlib/encoding.py.orig	2009-12-07 14:16:51.666580691 +0800
+++ terminator-0.92/terminatorlib/encoding.py	2009-12-07 14:16:59.960487436 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    TerminatorEncoding - charset encoding classes
 #    Copyright (C) 2006-2010  chantra@debuntu.org
 #
--- terminator-0.92/terminatorlib/keybindings.py.orig	2009-12-07 14:17:34.659188298 +0800
+++ terminator-0.92/terminatorlib/keybindings.py	2009-12-07 14:17:42.913028120 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #  Terminator - multiple gnome terminals in one window
 #   Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/terminator.py.orig	2009-12-07 14:18:09.188014065 +0800
+++ terminator-0.92/terminatorlib/terminator.py	2009-12-07 14:18:14.705972666 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """terminator.py - class for the master Terminator singleton"""
--- terminator-0.92/terminatorlib/__init__.py.orig	2009-12-07 14:18:41.540480577 +0800
+++ terminator-0.92/terminatorlib/__init__.py	2009-12-07 14:18:47.314041144 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    Terminator - multiple gnome terminals in one window
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/version.py.orig	2009-12-07 14:19:06.732667557 +0800
+++ terminator-0.92/terminatorlib/version.py	2009-12-07 14:19:13.754397904 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    TerminatorVersion - version number
 #    Copyright (C) 2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/debugserver.py.orig	2009-12-07 14:19:40.741382009 +0800
+++ terminator-0.92/terminatorlib/debugserver.py	2009-12-07 14:19:47.491126856 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #
 # Copyright (c) 2008, Thomas Hurst <tom@hur.st>
 #
--- terminator-0.92/terminatorlib/util.py-orig	2010-04-15 13:36:18.055257212 -0500
+++ terminator-0.92/terminatorlib/util.py	2010-04-15 13:38:17.193266957 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    Terminator.util - misc utility functions
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/searchbar.py-orig	2010-04-15 13:36:14.488157062 -0500
+++ terminator-0.92/terminatorlib/searchbar.py	2010-04-15 13:38:13.105156516 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """searchbar.py - classes necessary to provide a terminal search bar"""
--- terminator-0.92/terminatorlib/titlebar.py-orig	2010-04-15 13:36:27.678884075 -0500
+++ terminator-0.92/terminatorlib/titlebar.py	2010-04-15 13:38:30.289117539 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """titlebar.py - classes necessary to provide a terminal title bar"""
--- terminator-0.92/terminatorlib/cwd.py-orig	2010-04-15 13:36:44.872632630 -0500
+++ terminator-0.92/terminatorlib/cwd.py	2010-04-15 13:38:51.961893298 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """cwd.py - function necessary to get the cwd for a given pid on various OSes
--- terminator-0.92/terminatorlib/window.py-orig	2010-04-15 13:36:31.341248379 -0500
+++ terminator-0.92/terminatorlib/window.py	2010-04-15 13:38:36.185277498 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """window.py - class for the main Terminator window"""
--- terminator-0.92/terminatorlib/prefseditor.py-orig	2010-04-15 13:36:05.322715956 -0500
+++ terminator-0.92/terminatorlib/prefseditor.py	2010-04-15 13:38:04.545944916 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """Preferences Editor for Terminator. 
 
 Load a UIBuilder config file, display it,
--- terminator-0.92/terminatorlib/notebook.py-orig	2010-04-15 13:37:03.055988855 -0500
+++ terminator-0.92/terminatorlib/notebook.py	2010-04-15 13:39:02.745910962 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """notebook.py - classes for the notebook widget"""
--- terminator-0.92/terminatorlib/borg.py-orig	2010-04-15 13:36:09.554656310 -0500
+++ terminator-0.92/terminatorlib/borg.py	2010-04-15 13:38:08.304506542 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """borg.py - We are the borg. Resistance is futile.
--- terminator-0.92/terminatorlib/plugin.py-orig	2010-04-15 13:36:23.189163250 -0500
+++ terminator-0.92/terminatorlib/plugin.py	2010-04-15 13:38:21.745117689 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """plugin.py - Base plugin system
--- terminator-0.92/terminatorlib/optionparse.py-orig	2010-04-15 13:36:36.233483456 -0500
+++ terminator-0.92/terminatorlib/optionparse.py	2010-04-15 13:38:42.361578009 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #    Terminator.optionparse - Parse commandline options
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/tests/testsignalman.py-orig	2010-04-15 13:37:32.603268428 -0500
+++ terminator-0.92/terminatorlib/tests/testsignalman.py	2010-04-15 13:39:28.066678475 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """testsignalman.py - Test the signalman class
--- terminator-0.92/terminatorlib/tests/testborg.py-orig	2010-04-15 13:37:25.413536971 -0500
+++ terminator-0.92/terminatorlib/tests/testborg.py	2010-04-15 13:39:24.042466233 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """testborg.py - We are the borg. Resistance is futile.
--- terminator-0.92/terminatorlib/paned.py-orig	2010-04-15 13:35:59.843231712 -0500
+++ terminator-0.92/terminatorlib/paned.py	2010-04-15 13:37:59.545188619 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """paned.py - a base Paned container class and the vertical/horizontal
--- terminator-0.92/terminatorlib/container.py-orig	2010-04-15 13:37:17.427838385 -0500
+++ terminator-0.92/terminatorlib/container.py	2010-04-15 13:39:16.658120145 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """container.py - classes necessary to contain Terminal widgets"""
--- terminator-0.92/terminatorlib/signalman.py-orig	2010-04-15 13:37:08.478736726 -0500
+++ terminator-0.92/terminatorlib/signalman.py	2010-04-15 13:39:06.793875755 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """Simple management of Gtk Widget signal handlers"""
--- terminator-0.92/terminatorlib/factory.py-orig	2010-04-15 13:37:12.974882079 -0500
+++ terminator-0.92/terminatorlib/factory.py	2010-04-15 13:39:11.442203532 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """factory.py - Maker of objects
--- terminator-0.92/terminatorlib/terminal.py-orig	2010-04-15 13:36:40.380236186 -0500
+++ terminator-0.92/terminatorlib/terminal.py	2010-04-15 13:38:47.283020365 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """terminal.py - classes necessary to provide Terminal widgets"""
--- terminator-0.92/terminatorlib/terminal_popup_menu.py-orig	2010-04-15 13:36:53.995294119 -0500
+++ terminator-0.92/terminatorlib/terminal_popup_menu.py	2010-04-15 13:38:57.930231108 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """terminal_popup_menu.py - classes necessary to provide a terminal context 
--- terminator-0.92/terminatorlib/plugins/custom_commands.py-orig	2010-04-15 13:37:40.586250827 -0500
+++ terminator-0.92/terminatorlib/plugins/custom_commands.py	2010-04-15 13:39:34.154626664 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # Terminator by Chris Jones <cmsj@tenshu.net>
 # GPL v2 only
 """custom_commands.py - Terminator Plugin to add custom command menu entries"""
--- terminator-0.92/terminator-orig	2010-04-15 13:43:21.444039059 -0500
+++ terminator-0.92/terminator	2010-04-15 13:43:26.664025137 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #    Terminator - multiple gnome terminals in one window
 #    Copyright (C) 2006-2010  cmsj@tenshu.net
 #
--- terminator-0.92/terminatorlib/editablelabel.py-orig	2010-04-15 13:43:03.705918270 -0500
+++ terminator-0.92/terminatorlib/editablelabel.py	2010-04-15 13:43:10.055749519 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # vim: tabstop=4 softtabstop=4 shiftwidth=4 expandtab
 #
 # Copyright (c) 2009, Emmanuel Bretelle <chantra@debuntu.org>
--- terminator-0.92/setup.py-orig	2010-04-15 13:44:03.846371877 -0500
+++ terminator-0.92/setup.py	2010-04-15 13:44:10.233062053 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python-2.6
 
 from distutils.core import setup
 from distutils.dist import Distribution
