From e68e88d2de7449b53c98ad7fc3e1ff169313bef3 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Wed, 5 Aug 2015 12:25:15 -0700
Subject: [PATCH 21/34] Try to create privsep chroot dir if it doesn't exist
 yet

diff -wpruN '--exclude=*.orig' a~/sshd.c a/sshd.c
--- a~/sshd.c	1970-01-01 00:00:00
+++ a/sshd.c	1970-01-01 00:00:00
@@ -1823,11 +1823,30 @@ main(int ac, char **av)
 
 	if (privsep_chroot) {
 		struct stat st;
+		int rc;
 
 		if ((stat(_PATH_PRIVSEP_CHROOT_DIR, &st) == -1) ||
-		    (S_ISDIR(st.st_mode) == 0))
-			fatal("Missing privilege separation directory: %s",
-			    _PATH_PRIVSEP_CHROOT_DIR);
+		    (S_ISDIR(st.st_mode) == 0)) {
+			rc = mkdir(_PATH_PRIVSEP_CHROOT_DIR, 0755);
+			if (rc == 0) {
+				/*
+				 * If mkdir works, try stat again, so the
+				 * permissions check below can work.
+				 */
+				rc = stat(_PATH_PRIVSEP_CHROOT_DIR, &st);
+				if (rc == 0 && S_ISDIR(st.st_mode) == 0) {
+					rc = -1;
+					errno = ENOTDIR;
+				}
+			}
+
+			if (rc != 0) {
+				fatal("Failed to create privilege separation "
+				    "directory %s: %s",
+				    _PATH_PRIVSEP_CHROOT_DIR,
+				    strerror(errno));
+			}
+		}
 
 #ifdef HAVE_CYGWIN
 		if (check_ntsec(_PATH_PRIVSEP_CHROOT_DIR) &&
