From 2ddf64a6bfa74df030fa7b765c315377d9b5bc04 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 21 Aug 2015 18:47:46 -0700
Subject: [PATCH 29/36] Don't use krb5-config to check for GSSAPI on Illumos

---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8c0abd6..e251f9b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4064,6 +4064,11 @@ AC_ARG_WITH([kerberos5],
 		AC_PATH_PROG([KRB5CONF], [krb5-config],
 			     [$KRB5ROOT/bin/krb5-config],
 			     [$KRB5ROOT/bin:$PATH])
+		# Illumos has GSS but krb5-config doesn't talk about it
+		# Fall back to regular probing for libs
+		if uname -rs | grep 'SunOS 5.11' ; then
+			KRB5CONF=nonexistent
+		fi
 		if test -x $KRB5CONF ; then
 			K5CFLAGS="`$KRB5CONF --cflags`"
 			K5LIBS="`$KRB5CONF --libs`"
@@ -4105,7 +4110,7 @@ AC_ARG_WITH([kerberos5],
 					 AC_CHECK_LIB([des], [des_cbc_encrypt],
 					   [K5LIBS="$K5LIBS -ldes"])
 				       ], [ AC_MSG_RESULT([no])
-					 K5LIBS="-lkrb5 -lk5crypto -lcom_err"
+					 K5LIBS="-lkrb5"
 				       
 			])
 			AC_SEARCH_LIBS([dn_expand], [resolv])
-- 
2.5.4 (Apple Git-61)

