configure checks for mblen and mbtowc with wrong CFLAGS

--- openssh-7.4p1/configure.ac.1	2017-11-15 20:03:47.708528641 +0000
+++ openssh-7.4p1/configure.ac	2017-11-15 20:05:11.547475245 +0000
@@ -1863,7 +1863,7 @@
 
 dnl Wide character support.  Linux man page says it needs _XOPEN_SOURCE.
 saved_CFLAGS="$CFLAGS"
-CFLAGS="$CFLAGS -D_XOPEN_SOURCE"
+#CFLAGS="$CFLAGS -D_XOPEN_SOURCE"
 AC_CHECK_FUNCS([mblen mbtowc nl_langinfo wcwidth])
 CFLAGS="$saved_CFLAGS"
 
