--- openssh-7.2p2/Makefile.in.1	2016-04-14 14:17:58.130093926 +0300
+++ openssh-7.2p2/Makefile.in	2016-04-14 14:19:11.759175242 +0300
@@ -26,7 +26,7 @@
 SFTP_SERVER=$(libexecdir)/sftp-server
 SSH_KEYSIGN=$(libexecdir)/ssh-keysign
 SSH_PKCS11_HELPER=$(libexecdir)/ssh-pkcs11-helper
-ROOTDLIBDIR64=$(DESTDIR)/usr/lib/dtrace/64
+ROOTDLIBDIR=$(DESTDIR)/usr/lib/dtrace
 PRIVSEP_PATH=@PRIVSEP_PATH@
 SSH_PRIVSEP_USER=@SSH_PRIVSEP_USER@
 STRIP_OPT=@STRIP_OPT@
@@ -234,7 +234,7 @@
 	    -o $(srcdir)/sftp_provider.h
 
 sftp_provider.o: sftp_provider.d sftp_provider.h sftp-server.o
-	/usr/sbin/dtrace -G -64 -xnolibs -s $(srcdir)/sftp_provider.d \
+	/usr/sbin/dtrace -G -xnolibs -s $(srcdir)/sftp_provider.d \
 	    sftp-server.o -o sftp_provider.o
 
 # special case for sftp-server.o, it includes sftp_provider.h
@@ -352,7 +352,7 @@
 	$(INSTALL) -m 644 sftp-server.8.out $(DESTDIR)$(mandir)/$(mansubdir)8/sftp-server.8
 	$(INSTALL) -m 644 ssh-keysign.8.out $(DESTDIR)$(mandir)/$(mansubdir)8/ssh-keysign.8
 	$(INSTALL) -m 644 ssh-pkcs11-helper.8.out $(DESTDIR)$(mandir)/$(mansubdir)8/ssh-pkcs11-helper.8
-	mkdir -p $(ROOTDLIBDIR64) && cp $(srcdir)/sftp64.d $(ROOTDLIBDIR64)/sftp64.d
+	mkdir -p $(ROOTDLIBDIR) && cp $(srcdir)/sftp64.d $(ROOTDLIBDIR)/sftp.d
 
 install-sysconf:
 	if [ ! -d $(DESTDIR)$(sysconfdir) ]; then \
