diff -wpruN --no-dereference '--exclude=*.orig' a~/sshd_config a/sshd_config
--- a~/sshd_config	1970-01-01 00:00:00
+++ a/sshd_config	1970-01-01 00:00:00
@@ -29,7 +29,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
@@ -88,7 +88,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
