--- mutt-1.5.21/muttbug.sh.in.orig	Thu Dec  1 09:37:14 2011
+++ mutt-1.5.21/muttbug.sh.in	Fri Dec  2 10:52:31 2011
@@ -251,14 +251,9 @@
 echo 
 echo "-- Build environment information"
 echo
-echo "(Note: This is the build environment installed on the system"
-echo "muttbug is run on.  Information may or may not match the environment"
-echo "used to build mutt.)"
+echo "- C compiler version information"
+echo "XXX-CUSAGE-XXX"
 echo
-echo "- gcc version information"
-echo "@CC@"
-@CC@ -v 2>&1
-echo
 echo "- CFLAGS"
 echo @CFLAGS@
 
