diff --git a/nsprpub/pr/include/md/_solaris.h b/nsprpub/pr/include/md/_solaris.h
index 4ebf77f..a6b384f 100644
--- a/nsprpub/pr/include/md/_solaris.h
+++ b/nsprpub/pr/include/md/_solaris.h
@@ -52,6 +52,12 @@
 #define _PR_HAVE_ATOMIC_OPS
 #endif
 
+/* Earlier these settings were detected by nsprpub configure, now it doesn't run */
+#define HAVE_FCNTL_FILE_LOCKING
+#define _PR_HAVE_OFF64_T
+#define _PR_INET6
+/* End of configure-detected settings */
+
 #define _PR_POLL_AVAILABLE
 #define _PR_USE_POLL
 #define _PR_STAT_HAS_ST_ATIM

