--- gecko-dev-master/ipc/chromium/src/third_party/libevent/solaris/event2/event-config.h.~1~	2022-05-06 15:04:20.000000000 +0000
+++ gecko-dev-master/ipc/chromium/src/third_party/libevent/solaris/event2/event-config.h	2022-05-10 15:18:57.904722373 +0000
@@ -33,7 +33,7 @@
 #define EVENT__HAVE_ARC4RANDOM 1
 
 /* Define to 1 if you have the `arc4random_addrandom' function. */
-#define EVENT__HAVE_ARC4RANDOM_ADDRANDOM 1
+/* #undef EVENT__HAVE_ARC4RANDOM_ADDRANDOM 1 */
 
 /* Define to 1 if you have the `arc4random_buf' function. */
 #define EVENT__HAVE_ARC4RANDOM_BUF 1
@@ -218,7 +218,7 @@
 #define EVENT__HAVE_SELECT 1
 
 /* Define to 1 if you have the `sendfile' function. */
-#define EVENT__HAVE_SENDFILE 1
+#undef EVENT__HAVE_SENDFILE
 
 /* Define to 1 if you have the `setenv' function. */
 #define EVENT__HAVE_SETENV 1
