diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index ecd41b3..c2835ac 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3550,6 +3550,12 @@ pref("toolkit.rust-components.logging.internal-level", "Warn");
 // Settings Redesign 2025 prefs
 pref("browser.settings-redesign.enabled", false);
 
+// Audio/video doesn't work with new Utility system
+pref("media.allow-audio-non-utility",true);
+
+// Disable privacy-perserving ads
+pref("dom.private-attribution.submission.enabled", false);
+
 // A preference that will be locked to reflect whether this build has support
 // for XDG Config Home handling. Mostly used to be able to keep tests around
 // in case of a backout of the feature
