diff --git a/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp b/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp
index 379e5ce..85e0269 100644
--- a/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp
+++ b/gfx/angle/checkout/src/common/third_party/smhasher/src/PMurHash.cpp
@@ -92,9 +92,9 @@ on big endian machines, or a byte-by-byte read if the endianess is unknown.
  * or even _LITTLE_ENDIAN or _BIG_ENDIAN (Note the single _ prefix) */
 #if !defined(__BYTE_ORDER)
 #    if defined(__LITTLE_ENDIAN__) && __LITTLE_ENDIAN__ == 1 || \
-        defined(_LITTLE_ENDIAN) && _LITTLE_ENDIAN == 1
+        defined(_LITTLE_ENDIAN)
 #        define __BYTE_ORDER __LITTLE_ENDIAN
-#    elif defined(__BIG_ENDIAN__) && __BIG_ENDIAN__ == 1 || defined(_BIG_ENDIAN) && _BIG_ENDIAN == 1
+#    elif defined(__BIG_ENDIAN__) && __BIG_ENDIAN__ == 1 || defined(_BIG_ENDIAN)
 #        define __BYTE_ORDER __BIG_ENDIAN
 #    endif
 #endif
