--- firefox-16f979184458f531fce47706d2fa7f2902e75ad3/config/system-headers.mozbuild.old	2025-09-11 14:37:37.662739263 -0400
+++ firefox-16f979184458f531fce47706d2fa7f2902e75ad3/config/system-headers.mozbuild	2025-09-11 14:38:12.370010345 -0400
@@ -864,7 +864,6 @@
     "sys/shm.h",
     "sys/siginfo.h",
     "sys/signal.h",
-    "sys/single_threaded.h",
     "sys/socket.h",
     "sys/sockio.h",
     "sys/sparc/frame.h",
