--- gecko-dev-master/security/nss/lib/freebl/verified/internal/Hacl_Bignum_Base.h.old	2023-10-04 14:49:29.605062111 -0400
+++ gecko-dev-master/security/nss/lib/freebl/verified/internal/Hacl_Bignum_Base.h	2023-10-04 14:52:16.004563990 -0400
@@ -39,6 +39,10 @@
 #include "Hacl_Krmllib.h"
 #include "lib_intrinsics.h"
 
+#if defined(__sun)
+#include <alloca.h>
+#endif
+
 static inline uint32_t
 Hacl_Bignum_Base_mul_wide_add2_u32(uint32_t a, uint32_t b, uint32_t c_in, uint32_t *out)
 {
