--- firefox-107.0/build/moz.configure/toolchain.configure.old   2022-11-27 00:43:21.686745320 +0000
+++ firefox-107.0/build/moz.configure/toolchain.configure       2022-11-27 00:43:55.202798351 +0000
@@ -1805,7 +1805,7 @@
     # The oldest versions of supported compilers default to DWARF-4, but
     # newer versions may default to DWARF-5 or newer (e.g. clang 14), which
     # Valgrind doesn't support. Force-use DWARF-4.
-    return "-gdwarf-4"
+    return ""
 
 
 option(env="MOZ_DEBUG_FLAGS", nargs=1, help="Debug compiler flags")
