diff --git a/python/mozbuild/mozbuild/frontend/data.py b/python/mozbuild/mozbuild/frontend/data.py
index 7ba6c23..d8b478e 100644
--- a/python/mozbuild/mozbuild/frontend/data.py
+++ b/python/mozbuild/mozbuild/frontend/data.py
@@ -913,7 +913,7 @@ class SharedLibrary(Library):
                 )
             elif os_target == "SunOS":
                 self.symbols_link_arg = (
-                    "-z gnu-version-script-compat -Wl,--version-script,"
+                    "-Wl,-M,"
                     + self.symbols_file
                 )
             elif os_target == "WINNT":
