--- mongodb-src-r3.3.12/src/mongo/gotools/vendor/src/github.com/spacemonkeygo/openssl/build.go.1	2016-09-09 19:03:08.734300807 +0300
+++ mongodb-src-r3.3.12/src/mongo/gotools/vendor/src/github.com/spacemonkeygo/openssl/build.go	2016-09-09 19:03:18.218838944 +0300
@@ -17,6 +17,7 @@
 package openssl
 
 // #cgo linux pkg-config: openssl
+// #cgo solaris pkg-config: openssl
 // #cgo windows CFLAGS: -DWIN32_LEAN_AND_MEAN
 // #cgo darwin CFLAGS: -Wno-deprecated-declarations
 // #cgo darwin LDFLAGS: -lssl -lcrypto
