--- minidlna-1.2.1/minidlna.conf.1	2020-01-25 14:34:20.034314401 +0000
+++ minidlna-1.2.1/minidlna.conf	2020-01-25 14:35:08.168353536 +0000
@@ -28,7 +28,7 @@
 #db_dir=/var/cache/minidlna
 
 # set this if you would like to specify the directory where you want MiniDLNA to store its log file
-#log_dir=/var/log
+#log_dir=/var/log/minidlna
 
 # set this to change the verbosity of the information that is logged
 # each section can use a different level: off, fatal, error, warn, info, or debug
