--- a/src/hotspot/share/runtime/osThread.hpp	Wed Jul 13 20:15:53 2022
+++ b/src/hotspot/share/runtime/osThread.hpp	Sun Jul 24 14:21:39 2022
@@ -32,7 +32,7 @@
 #include "runtime/suspendedThreadTask.hpp"
 #include "utilities/macros.hpp"
 
-#if defined(LINUX) || defined(AIX) || defined(BSD)
+#if defined(LINUX) || defined(AIX) || defined(BSD) || defined(SOLARIS)
 #include "suspendResume_posix.hpp"
 #endif
 
