--- gnutls-2.8.0/doc/Makefile.in.orig	2009-05-31 14:58:01.039912000 +0800
+++ gnutls-2.8.0/doc/Makefile.in	2009-05-31 14:59:04.854878000 +0800
@@ -196,7 +196,7 @@
 	distdir
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = examples cyclo scripts manpages credentials reference
+DIST_SUBDIRS = cyclo scripts manpages credentials reference
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
   dir0=`pwd`; \
@@ -746,7 +746,7 @@
 EXTRA_DIST = TODO README.gaa certtool.cfg gnutls.ps gnutls.pdf \
 	gnutls.html extract-guile-c-doc.scm doxygen/Doxyfile.in \
 	doxygen/Doxyfile.orig texinfo.css $(guile_texi)
-SUBDIRS = examples cyclo scripts manpages credentials $(am__append_1)
+SUBDIRS = cyclo scripts manpages credentials $(am__append_1)
 info_TEXINFOS = gnutls.texi
 
 # Generated texinfos.
