From 53556e7d196e6324ea8d25a8c8b52658bc5a691f Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Wed, 28 Apr 2021 11:40:25 +0000
Subject: 13726 distinguish ourselves with a macro (__illumos__)

---
 gcc/config/sol2.h | 1 +
 1 file changed, 1 insertion(+)

--- gcc-releases-gcc-14.3.0/gcc/config/sol2.h.orig
+++ gcc-releases-gcc-14.3.0/gcc/config/sol2.h
@@ -85,6 +85,7 @@
   do {							\
     builtin_define_std ("unix");			\
     builtin_define_std ("sun");				\
+    builtin_define ("__illumos__");			\
     builtin_define ("__svr4__");			\
     builtin_define ("__SVR4");				\
     builtin_assert ("system=unix");			\
