diff -urN a/plugin/Makefile.am b/plugin/Makefile.am
--- a/plugin/Makefile.am	2012-03-21 14:15:38.054202916 +0000
+++ b/plugin/Makefile.am	2012-03-21 14:16:29.114046538 +0000
@@ -12,7 +12,7 @@
 	-module -avoid-version
 gtk_vnc_plugin_la_CFLAGS = \
 	-I$(top_srcdir)/src @GTK_CFLAGS@ \
-	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@ @WERROR_CFLAGS@
+	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@
 
 all-local: gtk-vnc-plugin.so
 
--- gtk-vnc-0.3.10/plugin/Makefile.in.~1~	2013-12-21 21:01:41.534401286 +0400
+++ gtk-vnc-0.3.10/plugin/Makefile.in	2013-12-21 21:02:20.229465299 +0400
@@ -503,7 +503,7 @@
 
 @ENABLE_PLUGIN_TRUE@gtk_vnc_plugin_la_CFLAGS = \
 @ENABLE_PLUGIN_TRUE@	-I$(top_srcdir)/src @GTK_CFLAGS@ \
-@ENABLE_PLUGIN_TRUE@	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@ @WERROR_CFLAGS@
+@ENABLE_PLUGIN_TRUE@	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@
 
 @ENABLE_PLUGIN_TRUE@EXTRA_DIST = README
 @ENABLE_PLUGIN_TRUE@CLEANFILES = gtk-vnc-plugin.so
