--- alacarte-0.12.3/data/alacarte.desktop.in.in.ori	2009-09-14 06:29:36.139144044 +0100
+++ alacarte-0.12.3/data/alacarte.desktop.in.in	2009-09-14 06:30:05.281579551 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=Main Menu
-_Comment=Change which applications are shown on the main menu
+# SUN_BRANDING
+_Comment=Add or remove entries from the Applications menu
 Exec=alacarte
 Terminal=false
 Type=Application
