diff -u a/sql/mysqld.cc b/sql/mysqld.cc
--- a/sql/mysqld.cc	Wed Aug 17 23:01:56 2022
+++ b/sql/mysqld.cc	Wed Aug 17 23:45:44 2022
@@ -5660,7 +5657,6 @@
     SYSVAR_AUTOSIZE(my_thread_stack_size, new_thread_stack_size);
   }
 
-  (void) thr_setconcurrency(concurrency);	// 10 by default
 
   select_thread=pthread_self();
   select_thread_in_use=1;
diff -u a/sql/mysqld.h b/sql/mysqld.h
--- a/sql/mysqld.h	Wed Aug 10 11:31:54 2022
+++ b/sql/mysqld.h	Wed Aug 17 23:45:10 2022
@@ -258,7 +258,6 @@
 extern ulong executed_events;
 extern char language[FN_REFLEN];
 extern "C" MYSQL_PLUGIN_IMPORT ulong server_id;
-extern ulong concurrency;
 extern time_t server_start_time, flush_status_time;
 extern char *opt_mysql_tmpdir, mysql_charsets_dir[];
 extern size_t mysql_unpacked_real_data_home_len;
--- a/libmysqld/lib_sql.cc	2022-08-10 11:31:48.000000000 +0000
+++ b/libmysqld/lib_sql.cc	2022-08-18 20:38:30.006538457 +0000
@@ -622,7 +622,6 @@
     udf_init();
 #endif
 
-  (void) thr_setconcurrency(concurrency);	// 10 by default
 
   if (flush_time && flush_time != ~(ulong) 0L)
     start_handle_manager();
