--- mariadb-10.6.11/libmariadb/plugins/auth/CMakeLists.txt	2022-11-03 11:07:25.000000000 +0000
+++ mariadb-10.6.11/libmariadb/plugins/auth/CMakeLists.txt.new	2022-11-30 21:06:27.127641383 +0000
@@ -99,7 +99,7 @@
   INCLUDE(${CC_SOURCE_DIR}/cmake/FindGSSAPI.cmake)
   IF(GSSAPI_FOUND)
     SET(GSSAPI_SOURCES ${AUTH_DIR}/auth_gssapi_client.c ${AUTH_DIR}/gssapi_client.c ${AUTH_DIR}/gssapi_errmsg.c)
-    CHECK_C_COMPILER_FLAG(-Wl,--as-needed have_C__Wl___as_needed)
+    #CHECK_C_COMPILER_FLAG(-Wl,--as-needed have_C__Wl___as_needed)
   ENDIF()
 ELSE()
   SET(GSSAPI_LIBS secur32)
