diff --git a/m4/pthread.m4 b/m4/pthread.m4
index de8001e..51c8afd 100644
--- a/m4/pthread.m4
+++ b/m4/pthread.m4
@@ -100,7 +100,7 @@ AS_HELP_STRING([--with-libpthread@<:@=DIR@:>@], [use libpthread from given base
 	)
 
 	AC_CHECK_HEADER([pthread.h],[found_libpthread=yes])
-	LIBPTHREAD_LIBS="-lpthread"
+	LIBPTHREAD_LIBS=""
 	AC_MSG_CHECKING(for process shared libpthread support)
 
 	if test -n "$_libpthread_dir_set" -o "x$found_libpthread" = xyes; then
