See: https://support.zabbix.com/browse/ZBX-18311

diff --git a/src/libs/zbxsysinfo/solaris/proc.c b/src/libs/zbxsysinfo/solaris/proc.c
index ebad565..beb991a 100644
--- a/src/libs/zbxsysinfo/solaris/proc.c
+++ b/src/libs/zbxsysinfo/solaris/proc.c
@@ -709,6 +709,9 @@ int	proc_num(AGENT_REQUEST *request, AGENT_RESULT *result)
 		if (SUCCEED != proc_get_process_info(entries->d_name, proc_props, &proc, &psinfo))
 			continue;
 
+		if (SUCCEED != check_procstate(&psinfo, zbx_proc_stat))
+			continue;
+
 		if (SUCCEED == proc_match_props(&proc, usrinfo, procname, proccomm_rxp))
 		{
 #ifdef HAVE_ZONE_H
