the include path for libevent2 is missing

diff --git a/src/libs/zbxdiscoverer/Makefile.am b/src/libs/zbxdiscoverer/Makefile.am
index 4ec0939..17d1329 100644
--- a/src/libs/zbxdiscoverer/Makefile.am
+++ b/src/libs/zbxdiscoverer/Makefile.am
@@ -21,4 +21,5 @@ libzbxdiscoverer_a_SOURCES = \
 	discoverer_int.h
 
 libzbxdiscoverer_a_CFLAGS = \
-	$(TLS_CFLAGS)
+	$(TLS_CFLAGS)	\
+	$(LIBEVENT_CFLAGS)
diff --git a/src/libs/zbxpoller/Makefile.am b/src/libs/zbxpoller/Makefile.am
index 13e9e14..5c4ff5a 100644
--- a/src/libs/zbxpoller/Makefile.am
+++ b/src/libs/zbxpoller/Makefile.am
@@ -51,7 +51,8 @@ libzbxpoller_a_SOURCES = \
 libzbxpoller_a_CFLAGS = \
 	$(TLS_CFLAGS) \
 	$(LIBXML2_CFLAGS) \
-	$(SNMP_CFLAGS)
+	$(SNMP_CFLAGS) \
+	$(LIBEVENT_CFLAGS)
 
 if HAVE_SSH
 libzbxpoller_a_SOURCES += ssh_run.c
