--- gdl-1.0.2/src/gdl.cpp.orig	2023-01-06 18:39:47.000000000 +0100
+++ gdl-1.0.2/src/gdl.cpp	2023-02-05 16:23:03.404578067 +0100
@@ -239,7 +239,7 @@
     #include <unistd.h>
 #endif
 
-#ifdef __linux__
+#ifdef __sun
     #include <limits.h>
     #include <libgen.h>
     #include <unistd.h>
@@ -274,7 +274,7 @@
 
 #endif
 
-#ifdef __linux__
+#ifdef __sun
 
 std::string getExecutablePath() {
    char rawPathName[PATH_MAX];
