--- nmap-7.95/charpool.cc.orig	2024-02-28 20:46:45.000000000 +0200
+++ nmap-7.95/charpool.cc	2025-01-03 19:45:55.140063616 +0200
@@ -62,6 +62,7 @@
 /* $Id: charpool.cc 38790 2024-02-28 18:46:45Z dmiller $ */
 
 #include <stddef.h>
+#include <climits>
 #undef NDEBUG
 #include <assert.h>
 
--- nmap-7.95/nping/NpingOps.cc.orig	2024-02-28 20:46:45.000000000 +0200
+++ nmap-7.95/nping/NpingOps.cc	2025-01-03 19:55:15.811101239 +0200
@@ -59,6 +59,8 @@
  *
  ***************************************************************************/
 
+#include <climits>
+
 #ifdef WIN32
 #include "winfix.h"
 #endif
