Avoid hardcoding runpath

--- nmap-7.99/libdnet-stripped/src/Makefile.in.old
+++ nmap-7.99/libdnet-stripped/src/Makefile.in
@@ -410,7 +410,7 @@
 	}
 
 libdnet.la: $(libdnet_la_OBJECTS) $(libdnet_la_DEPENDENCIES) $(EXTRA_libdnet_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libdnet_la_LINK) -rpath $(libdir) $(libdnet_la_OBJECTS) $(libdnet_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libdnet_la_LINK) $(libdnet_la_OBJECTS) $(libdnet_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
