Allow building in zones with shared IP stack
--- nmap-7.99/libdnet-stripped/configure.old
+++ nmap-7.99/libdnet-stripped/configure
@@ -15246,7 +15246,7 @@
   printf %s "(cached) " >&6
 else case e in #(
   e)
-        if test -f /usr/include/inet/mib2.h -a '(' -c /dev/ip -o -c /dev/arp ')' ; then
+        if test -f /usr/include/inet/mib2.h ; then
         ac_cv_dnet_streams_mib2=yes
     else
         ac_cv_dnet_streams_mib2=no
