Allow building in zones with shared IP stack
--- nmap-7.80/libdnet-stripped/config/acinclude.m4.~1~	2020-07-30 15:39:51.648028516 +0000
+++ nmap-7.80/libdnet-stripped/config/acinclude.m4	2020-07-30 15:42:42.397525112 +0000
@@ -162,7 +162,7 @@
 AC_DEFUN(AC_DNET_STREAMS_MIB2,
     [AC_MSG_CHECKING(for SNMP MIB2 STREAMS)
     AC_CACHE_VAL(ac_cv_dnet_streams_mib2,[
-        if test -f /usr/include/inet/mib2.h -a '(' -c /dev/ip -o -c /dev/arp ')' ; then
+        if test -f /usr/include/inet/mib2.h ; then
             ac_cv_dnet_streams_mib2=yes
         else
             ac_cv_dnet_streams_mib2=no
