--- nmap-7.95/liblinear/Makefile.orig	2024-02-28 20:18:35.000000000 +0200
+++ nmap-7.95/liblinear/Makefile	2025-01-03 19:41:04.853958913 +0200
@@ -35,9 +35,9 @@
 	$(CXX) $(CFLAGS) -c -o linear.o linear.cpp
 
 blas/blas.a: blas/*.c blas/*.h
-	make -C blas OPTFLAGS='$(CFLAGS)' CC='$(CC)';
+	gmake -C blas OPTFLAGS='$(CFLAGS)' CC='$(CC)';
 
 clean:
-	make -C blas clean
-	make -C matlab clean
+	gmake -C blas clean
+	gmake -C matlab clean
 	rm -f *~ newton.o linear.o train predict liblinear.so.$(SHVER)
