--- util-linux-2.33.2/Makefile.in	2019-04-09 15:02:11.266936132 +0000
+++ util-linux-2.33.2/Makefile.in.new	2019-05-16 08:52:01.147843878 +0000
@@ -279,9 +279,7 @@ check_PROGRAMS = test_blkdev$(EXEEXT) te
 @BUILD_LIBMOUNT_TRUE@am__append_54 = libmount/mount.pc
 @BUILD_LIBMOUNT_TRUE@am__append_55 = libmount/mount.pc
 @BUILD_LIBSMARTCOLS_TRUE@am__append_56 = libsmartcols.la
-@BUILD_LIBSMARTCOLS_TRUE@@HAVE_VSCRIPT_TRUE@am__append_57 = $(VSCRIPT_LDFLAGS),$(top_srcdir)/libsmartcols/src/libsmartcols.sym
 @BUILD_LIBSMARTCOLS_TRUE@am__append_58 =  \
-@BUILD_LIBSMARTCOLS_TRUE@	libsmartcols/src/libsmartcols.sym \
 @BUILD_LIBSMARTCOLS_TRUE@	libsmartcols/COPYING
 @BUILD_LIBSMARTCOLS_TRUE@am__append_59 = install-exec-hook-libsmartcols
 @BUILD_LIBSMARTCOLS_TRUE@am__append_60 = uninstall-hook-libsmartcols
@@ -4818,9 +4816,6 @@ test_pwdutils_CFLAGS = $(AM_CFLAGS) -DTE
 @BUILD_LIBSMARTCOLS_TRUE@	-I$(ul_libsmartcols_incdir) \
 @BUILD_LIBSMARTCOLS_TRUE@	-I$(top_srcdir)/libsmartcols/src
 
-@BUILD_LIBSMARTCOLS_TRUE@EXTRA_libsmartcols_la_DEPENDENCIES = \
-@BUILD_LIBSMARTCOLS_TRUE@	libsmartcols/src/libsmartcols.sym
-
 @BUILD_LIBSMARTCOLS_TRUE@libsmartcols_la_LDFLAGS = $(SOLIB_LDFLAGS) \
 @BUILD_LIBSMARTCOLS_TRUE@	$(am__append_57) -version-info \
 @BUILD_LIBSMARTCOLS_TRUE@	$(LIBSMARTCOLS_VERSION_INFO)
