--- gnutls-3.8.10/src/gl/tests/localename-unsafe.c.orig	2025-07-09 06:36:59.000000000 +0200
+++ gnutls-3.8.10/src/gl/tests/localename-unsafe.c	2025-09-19 21:44:08.617451968 +0200
@@ -48,7 +48,7 @@
 # if defined __sun
 #  if HAVE_GETLOCALENAME_L
 /* Solaris >= 12.  */
-extern char * getlocalename_l(int, locale_t);
+extern const char * getlocalename_l(int, locale_t);
 #  elif HAVE_SOLARIS114_LOCALES
 #   include <sys/localedef.h>
 #  endif
