--- gnutls-2.12.24/doc/Makefile.in.~1~	2016-11-04 09:55:28.000000000 +0300
+++ gnutls-2.12.24/doc/Makefile.in	2016-11-25 15:04:09.206411814 +0300
@@ -362,7 +362,7 @@
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = examples cyclo scripts manpages credentials reference
+DIST_SUBDIRS = cyclo scripts manpages credentials reference
 am__DIST_COMMON = $(gnutls_TEXINFOS) $(srcdir)/Makefile.in \
 	$(top_srcdir)/build-aux/depcomp \
 	$(top_srcdir)/build-aux/mdate-sh \
@@ -1295,7 +1295,7 @@
 EXTRA_DIST = TODO certtool.cfg gnutls.ps gnutls.pdf gnutls.html \
 	doxygen/Doxyfile.in doxygen/Doxyfile.orig texinfo.css \
 	$(guile_texi) extract-guile-c-doc.scm
-SUBDIRS = examples cyclo scripts manpages credentials $(am__append_1)
+SUBDIRS = cyclo scripts manpages credentials $(am__append_1)
 info_TEXINFOS = gnutls.texi
 
 # Examples.
