--- gnutls-2.12.21/doc/Makefile.in.~1~	2012-11-09 02:43:20.000000000 +0400
+++ gnutls-2.12.21/doc/Makefile.in	2016-05-06 15:37:51.063922134 +0300
@@ -268,7 +268,7 @@
 	distdir
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = examples cyclo scripts manpages credentials reference
+DIST_SUBDIRS = cyclo scripts manpages credentials reference
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
   dir0=`pwd`; \
@@ -1192,7 +1192,7 @@
 EXTRA_DIST = TODO certtool.cfg gnutls.ps gnutls.pdf gnutls.html \
 	doxygen/Doxyfile.in doxygen/Doxyfile.orig texinfo.css \
 	$(guile_texi) extract-guile-c-doc.scm
-SUBDIRS = examples cyclo scripts manpages credentials $(am__append_1)
+SUBDIRS = cyclo scripts manpages credentials $(am__append_1)
 info_TEXINFOS = gnutls.texi
 
 # Examples.
