--- gnutls-2.12.24/doc/Makefile.in.~2~	2016-11-25 15:07:11.563940567 +0300
+++ gnutls-2.12.24/doc/Makefile.in	2016-11-25 15:07:57.913288748 +0300
@@ -282,7 +282,7 @@
 am__v_texidevnull_ = $(am__v_texidevnull_@AM_DEFAULT_V@)
 am__v_texidevnull_0 = > /dev/null
 am__v_texidevnull_1 = 
-INFO_DEPS = $(srcdir)/gnutls.info
+INFO_DEPS =
 TEXINFO_TEX = $(top_srcdir)/build-aux/texinfo.tex
 am__TEXINFO_TEX_DIR = $(top_srcdir)/build-aux
 DVIS = gnutls.dvi
