--- a/build-aux/update-copyright	2013-09-22 06:15:20.000000000 +0000
+++ b/build-aux/update-copyright	2015-11-22 08:30:37.968983827 +0000
@@ -124,7 +124,7 @@
 use warnings;
 
 my $copyright_re = 'Copyright';
-my $circle_c_re = '(?:\([cC]\)|@copyright{}|&copy;)';
+my $circle_c_re = '(?:\([cC]\)|@copyright\{}|\\\\\(co|&copy;)';
 my $holder = $ENV{UPDATE_COPYRIGHT_HOLDER};
 $holder ||= 'Free Software Foundation, Inc.';
 my $prefix_max = 5;
