#see https://bugs.php.net/bug.php?id=81658
--- php-8.1.1/ext/mysqli/mysqli_api.c	2021-12-15 03:00:45.000000000 +0000
+++ php-8.1.1/ext/mysqli/mysqli_api.c.new	2022-01-18 13:58:01.795367965 +0000
@@ -1720,7 +1720,7 @@
 #if MYSQL_VERSION_ID > 50605 || defined(MYSQLI_USE_MYSQLND)
 		case MYSQL_SERVER_PUBLIC_KEY:
 #endif
-#if MYSQL_VERSION_ID >= 80021 || defined(MYSQLI_USE_MYSQLND)
+#if (MYSQL_VERSION_ID >= 80021 && !defined(MARIADB_BASE_VERSION)) || defined(MYSQLI_USE_MYSQLND)
 		case MYSQL_OPT_LOAD_DATA_LOCAL_DIR:
 #endif
 			return IS_STRING;
