--- node-v14.14.0/configure.orig	2020-10-16 01:50:41.000000000 +0000
+++ node-v14.14.0/configure	2020-10-16 19:01:00.202122289 +0000
@@ -5,12 +5,12 @@
 # as is the fact that the ] goes on a new line.
 _=[ 'exec' '/bin/sh' '-c' '''
 test ${FORCE_PYTHON2} && exec python2 "$0" "$@"  # workaround for gclient
+command -v python2.7 >/dev/null && exec python2.7 "$0" "$@"
 command -v python3.9 >/dev/null && exec python3.9 "$0" "$@"
 command -v python3.8 >/dev/null && exec python3.8 "$0" "$@"
 command -v python3.7 >/dev/null && exec python3.7 "$0" "$@"
 command -v python3.6 >/dev/null && exec python3.6 "$0" "$@"
 command -v python3.5 >/dev/null && exec python3.5 "$0" "$@"
-command -v python2.7 >/dev/null && exec python2.7 "$0" "$@"
 exec python "$0" "$@"
 ''' "$0" "$@"
 ]
