--- xdebug-2.1.3/usefulstuff.c_orig	Thu Jan 26 01:48:14 2012
+++ xdebug-2.1.3/usefulstuff.c	Thu Feb  9 15:33:34 2012
@@ -429,7 +429,6 @@
 	FILE *fh;
 	struct stat buf;
 	char *tmp_fname = NULL;
-	int   filename_len = 0;
 
 	/* We're not doing any tricks for append mode... as that has atomic writes
 	 * anyway. And we ignore read mode as well. */
@@ -437,14 +436,6 @@
 		return xdebug_open_file(fname, mode, extension, new_fname);
 	}
 
-	/* Make sure we don't open a file with a path that's too long */
-	filename_len += (fname ? strlen(fname) : 0); /* filename */
-	filename_len += (extension ? strlen(extension) : 0) + 1; /* extension (+ ".") */
-	filename_len += 8; /* possible random extension (+ ".") */
-	if (filename_len > NAME_MAX) {
-		fname[NAME_MAX - (extension ? strlen(extension) : 0 )] = '\0';
-	}
-
 	/* In write mode however we do have to do some stuff. */
 	/* 1. Check if the file exists */
 	if (extension) {
