--- vte-0.30.0/src/vte.c-orig	2011-10-05 00:05:58.598942571 -0500
+++ vte-0.30.0/src/vte.c	2011-10-05 00:06:24.232849441 -0500
@@ -9246,7 +9246,7 @@ vte_terminal_determine_colors (VteTermin
 			       gboolean highlight,
 			       guint *fore, guint *back)
 {
-	return vte_terminal_determine_colors_internal (terminal, cell,
+	vte_terminal_determine_colors_internal (terminal, cell,
 						       highlight, FALSE,
 						       fore, back);
 }
@@ -9257,7 +9257,7 @@ vte_terminal_determine_cursor_colors (Vt
 				      gboolean highlight,
 				      guint *fore, guint *back)
 {
-	return vte_terminal_determine_colors_internal (terminal, cell,
+	vte_terminal_determine_colors_internal (terminal, cell,
 						       highlight, TRUE,
 						       fore, back);
 }
