diff -urN vte.orig/src/vte.c vte.new/src/vte.c
--- vte.orig/src/vte.c	2011-05-20 11:09:40.653834969 +0100
+++ vte.new/src/vte.c	2011-05-20 11:10:11.113623701 +0100
@@ -12350,9 +12350,9 @@ vte_terminal_class_init(VteTerminalClass
 	binding_set = gtk_binding_set_by_class(klass);
 
 	/* Bind Copy, Paste, Cut keys */
-	gtk_binding_entry_add_signal(binding_set, GDK_KEY (F16), 0, "copy-clipboard",0);
-	gtk_binding_entry_add_signal(binding_set, GDK_KEY (F18), 0, "paste-clipboard", 0);
-	gtk_binding_entry_add_signal(binding_set, GDK_KEY (F20), 0, "copy-clipboard",0);
+	gtk_binding_entry_add_signal(binding_set, GDK_KEY (Copy), 0, "copy-clipboard",0);
+	gtk_binding_entry_add_signal(binding_set, GDK_KEY (Paste), 0, "paste-clipboard", 0);
+	gtk_binding_entry_add_signal(binding_set, GDK_KEY (Cut), 0, "copy-clipboard",0);
 
 	process_timer = g_timer_new ();
 }
