--- vte-0.28.2/configure.in.~1~	2011-08-29 01:32:53.000000000 +0000
+++ vte-0.28.2/configure.in	2017-11-15 16:53:28.404981338 +0000
@@ -202,7 +202,7 @@
 -Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
 -Wdeclaration-after-statement -Wold-style-definition \
 -Wno-missing-field-initializers -Wno-unused-parameter \
--fno-common -Wno-switch-enum"
+-Wno-switch-enum"
 
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$vte_cv_warn_maybe" != "x$MAYBE_WARN"; then
@@ -320,8 +320,6 @@
 
 case $host in
 *-*-solaris*)
-	AC_DEFINE(_XOPEN_SOURCE_EXTENDED,1,[Needed to get declarations for msg_control and msg_controllen on Solaris])
-	AC_DEFINE(_XOPEN_SOURCE,1,[Needed to get declarations for msg_control and msg_controllen on Solaris])
 	AC_DEFINE(__EXTENSIONS__,1,[Needed to get declarations for msg_control and msg_controllen on Solaris])
 	;;
 esac
