--- pygobject-3.18.0/Makefile.am.1	2016-07-26 22:15:25.381195518 +0300
+++ pygobject-3.18.0/Makefile.am	2016-07-26 22:15:39.665336215 +0300
@@ -2,7 +2,7 @@
 AUTOMAKE_OPTIONS = 1.7
 
 # Limit PyFlakes and PEP8 to these directories.
-pycheck_dirs = examples gi tests pygtkcompat
+pycheck_dirs = examples gi pygtkcompat
 
 # Part of the gi subdirectory is handled with non-recursive make to avoid
 # py-compile getting confused between gi/types.py and Python's standard
