--- gnome-commander-1.2.7/data/gnome-commander.desktop.in.in.orig       2007-03-04 22:46:25.000000000 +0000
+++ gnome-commander-1.2.7/data/gnome-commander.desktop.in.in    2008-09-05 09:17:37.304957392 +0100
@@ -6,7 +6,7 @@ Exec=gnome-commander
 Icon=gnome-commander
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Utility;FileManager;
+Categories=GNOME;System;FileManager;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-commander
