--- libjpeg-turbo-1.5.1/release/libjpeg.pc.in.orig	2017-03-02 23:32:25.537068631 +0100
+++ libjpeg-turbo-1.5.1/release/libjpeg.pc.in	2017-03-02 23:33:34.182713980 +0100
@@ -6,5 +6,5 @@
 Name: libjpeg
 Description: A SIMD-accelerated JPEG codec that provides the libjpeg API
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ljpeg
+Libs: -L${libdir} -R${libdir} -ljpeg
 Cflags: -I${includedir}
--- libjpeg-turbo-1.5.1/release/libturbojpeg.pc.in.orig	2017-03-02 23:33:09.858744986 +0100
+++ libjpeg-turbo-1.5.1/release/libturbojpeg.pc.in	2017-03-02 23:33:50.072086016 +0100
@@ -6,5 +6,5 @@
 Name: libturbojpeg
 Description: A SIMD-accelerated JPEG codec that provides the TurboJPEG API
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lturbojpeg
+Libs: -L${libdir} -R${libdir} -lturbojpeg
 Cflags: -I${includedir}
