We have no pytest-mypy-testing packaged yet.

--- traitlets-5.15.0/pyproject.toml.orig
+++ traitlets-5.15.0/pyproject.toml
@@ -38,7 +38,6 @@
     "mypy>=1.7.0,<1.19 ; platform_python_implementation == 'PyPy'",
     "pre-commit",
     "pytest-mock",
-    "pytest-mypy-testing",
     "pytest>=7.0,<8.2",
 ]
 docs = [
@@ -102,7 +101,7 @@
 log_cli_level = "info"
 addopts = [
   "-ra", "--durations=10", "--color=yes", "--doctest-modules",
-   "--showlocals", "--strict-markers", "--strict-config",
+   "--showlocals", "--strict-config",
    "--ignore=examples/docs/configs"
 ]
 testpaths = [
