We have no pytest-mypy-testing packaged yet.

--- traitlets-5.15.1/pyproject.toml.orig
+++ traitlets-5.15.1/pyproject.toml
@@ -40,7 +40,6 @@
     "mypy>=1.17.0,<1.19",
     "pre-commit",
     "pytest-mock",
-    "pytest-mypy-testing",
     "pytest>=7.0,<8.2",
 ]
 docs = [
@@ -104,7 +103,7 @@
 log_cli_level = "info"
 addopts = [
   "-ra", "--durations=10", "--color=yes", "--doctest-modules",
-   "--showlocals", "--strict-markers", "--strict-config",
+   "--showlocals", "--strict-config",
    "--ignore=examples/docs/configs"
 ]
 testpaths = [
