We have no pytest-mypy-testing packaged yet.

--- traitlets-5.14.3/pyproject.toml.orig
+++ traitlets-5.14.3/pyproject.toml
@@ -36,7 +36,6 @@
     "mypy>=1.7.0",
     "pre-commit",
     "pytest-mock",
-    "pytest-mypy-testing",
     "pytest>=7.0,<8.2",
 ]
 docs = [
@@ -100,7 +99,7 @@
 log_cli_level = "info"
 addopts = [
   "-ra", "--durations=10", "--color=yes", "--doctest-modules",
-   "--showlocals", "--strict-markers", "--strict-config",
+   "--showlocals", "--strict-config",
    "--ignore=examples/docs/configs"
 ]
 testpaths = [
