--- polkit-0.113/configure.ac.~1~	2015-06-19 23:31:02.000000000 +0300
+++ polkit-0.113/configure.ac	2016-08-02 09:40:52.194690032 +0300
@@ -160,10 +145,6 @@
 
 AC_CHECK_FUNCS(clearenv fdatasync)
 
-if test "x$GCC" = "xyes"; then
-  LDFLAGS="-Wl,--as-needed $LDFLAGS"
-fi
-
 dnl ---------------------------------------------------------------------------
 dnl - Check whether setnetgrent has a return value
 dnl ---------------------------------------------------------------------------
