--- httpcore-1.0.0/tests/_async/test_socks_proxy.py.orig
+++ httpcore-1.0.0/tests/_async/test_socks_proxy.py
@@ -4,6 +4,7 @@
 
 
 @pytest.mark.anyio
+@pytest.mark.skip(reason="we have no socksio")
 async def test_socks5_request():
     """
     Send an HTTP request via a SOCKS proxy.
@@ -62,6 +63,7 @@
 
 
 @pytest.mark.anyio
+@pytest.mark.skip(reason="we have no socksio")
 async def test_authenticated_socks5_request():
     """
     Send an HTTP request via a SOCKS proxy.
@@ -109,6 +111,7 @@
 
 
 @pytest.mark.anyio
+@pytest.mark.skip(reason="we have no socksio")
 async def test_socks5_request_connect_failed():
     """
     Attempt to send an HTTP request via a SOCKS proxy, resulting in a connect failure.
@@ -138,6 +141,7 @@
 
 
 @pytest.mark.anyio
+@pytest.mark.skip(reason="we have no socksio")
 async def test_socks5_request_failed_to_provide_auth():
     """
     Attempt to send an HTTP request via an authenticated SOCKS proxy,
@@ -166,6 +170,7 @@
 
 
 @pytest.mark.anyio
+@pytest.mark.skip(reason="we have no socksio")
 async def test_socks5_request_incorrect_auth():
     """
     Attempt to send an HTTP request via an authenticated SOCKS proxy,
--- httpcore-1.0.0/tests/_sync/test_socks_proxy.py.orig
+++ httpcore-1.0.0/tests/_sync/test_socks_proxy.py
@@ -4,6 +4,7 @@
 
 
 
+@pytest.mark.skip(reason="we have no socksio")
 def test_socks5_request():
     """
     Send an HTTP request via a SOCKS proxy.
@@ -62,6 +63,7 @@
 
 
 
+@pytest.mark.skip(reason="we have no socksio")
 def test_authenticated_socks5_request():
     """
     Send an HTTP request via a SOCKS proxy.
@@ -109,6 +111,7 @@
 
 
 
+@pytest.mark.skip(reason="we have no socksio")
 def test_socks5_request_connect_failed():
     """
     Attempt to send an HTTP request via a SOCKS proxy, resulting in a connect failure.
@@ -138,6 +141,7 @@
 
 
 
+@pytest.mark.skip(reason="we have no socksio")
 def test_socks5_request_failed_to_provide_auth():
     """
     Attempt to send an HTTP request via an authenticated SOCKS proxy,
@@ -166,6 +170,7 @@
 
 
 
+@pytest.mark.skip(reason="we have no socksio")
 def test_socks5_request_incorrect_auth():
     """
     Attempt to send an HTTP request via an authenticated SOCKS proxy,
