https://github.com/python-coincidence/importcheck/issues/60

--- importcheck-0.5.0/tests/conftest.py.orig
+++ importcheck-0.5.0/tests/conftest.py
@@ -22,6 +22,7 @@
 			'"platform_system == \\"Windows\\"" = [ "msvcrt",]',
 			'"platform_system == \\"Linux\\"" = [ "posix",]',
 			'"platform_system == \\"Darwin\\"" = [ "plistlib",]',
+			'"platform_system == \\"SunOS\\"" = [ "posix",]',
 			'',
 			"[tool.importcheck.config]",
 			"show = true",
