--- transmission-2.42/gtk/main.c.~1~	2015-03-16 15:06:33.259728585 +0300
+++ transmission-2.42/gtk/main.c	2015-03-16 15:10:45.404076792 +0300
@@ -590,6 +590,8 @@
         { NULL, 0,   0, 0, NULL, NULL, NULL }
     };
 
+    setenv( "EVENT_NOEVPORT", "1", 1 );
+
     /* default settings */
     memset( &cbdata, 0, sizeof( struct cbdata ) );
     cbdata.config_dir = (char*) tr_getDefaultConfigDir( MY_CONFIG_NAME );
