--- transmission-2.52/gtk/transmission-gtk.desktop.in.~1~	2015-03-16 15:26:11.900706664 +0300
+++ transmission-2.52/gtk/transmission-gtk.desktop.in	2015-03-16 15:27:56.434449700 +0300
@@ -9,7 +9,7 @@
 TryExec=transmission-gtk
 Type=Application
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
-Categories=Network;FileTransfer;P2P;GTK;
+Categories=Applications;Network;FileTransfer;P2P;GTK;
 X-Ubuntu-Gettext-Domain=transmission
 X-AppInstall-Keywords=torrent 
 Actions=Pause;Minimize;
