--- transmission-2.52/third-party/miniupnp/miniupnpc.c.~1~	2015-03-16 15:38:15.065447560 +0300
+++ transmission-2.52/third-party/miniupnp/miniupnpc.c	2015-03-16 15:38:29.757787391 +0300
@@ -17,7 +17,7 @@
 #endif
 #endif
 
-#if !defined(__OpenBSD__) && !defined(__NetBSD__) && !defined(MACOSX) && !defined(_WIN32)
+#if !defined(__OpenBSD__) && !defined(__NetBSD__) && !defined(MACOSX) && !defined(_WIN32) && !defined(__sun)
 #define HAS_IP_MREQN
 #endif
 
