--- pecl.ORIG	Tue Sep  9 19:45:00 2008
+++ pecl	Tue Sep  9 19:45:33 2008
@@ -25,4 +25,15 @@
   fi
 fi
 
+# define CC if not defined
+if test -x "/usr/bin/gcc"; then
+  CC="/usr/bin/gcc"
+  LD=$CC
+  export CC LD
+elif test -x "/usr/sfw/bin/gcc"; then
+  CC="/usr/sfw/bin/gcc"
+  LD=$CC
+  export CC LD
+fi
+
 exec $PHP -C -n -q $INCARG -d output_buffering=1 -d variables_order=EGPCS -d safe_mode=0 -d register_argc_argv="On" $INCDIR/peclcmd.php "$@"
