--- PEAR/Builder.php.ORIG	Tue Jun 30 16:44:29 2009
+++ PEAR/Builder.php	Tue Jun 30 16:44:37 2009
@@ -379,7 +379,7 @@
         $built_files = array();
         $prefix = exec($this->config->get('php_prefix')
                         . "php-config" .
-                       $this->config->get('php_suffix') . " --prefix");
+                       $this->config->get('php_suffix') . " --pear-prefix");
         $this->_harvestInstDir($prefix, $inst_dir . DIRECTORY_SEPARATOR . $prefix, $built_files);
         chdir($old_cwd);
         return $built_files;
