--- dash-0.5.11.1/configure.ac.orig	2020-07-08 09:06:36.000000000 +0000
+++ dash-0.5.11.1/configure.ac	2020-08-09 12:31:54.601331852 +0000
@@ -140,14 +140,16 @@
 fi
 
 dnl Check for stat64 (dietlibc/klibc).
-AC_CHECK_FUNC(stat64,, [
-	AC_DEFINE(fstat64, fstat, [64-bit operations are the same as 32-bit])
-	AC_DEFINE(lstat64, lstat, [64-bit operations are the same as 32-bit])
-	AC_DEFINE(stat64, stat, [64-bit operations are the same as 32-bit])
-	AC_DEFINE(readdir64, readdir,
-		  [64-bit operations are the same as 32-bit])
-	AC_DEFINE(dirent64, dirent,
-		  [64-bit operations are the same as 32-bit])
+AC_CHECK_DECL(stat64,, [
+    AC_CHECK_FUNC(stat64,, [
+	    AC_DEFINE(fstat64, fstat, [64-bit operations are the same as 32-bit])
+    	AC_DEFINE(lstat64, lstat, [64-bit operations are the same as 32-bit])
+	    AC_DEFINE(stat64, stat, [64-bit operations are the same as 32-bit])
+    	AC_DEFINE(readdir64, readdir,
+		    [64-bit operations are the same as 32-bit])
+	    AC_DEFINE(dirent64, dirent,
+		    [64-bit operations are the same as 32-bit])
+    ])
 ])
 
 dnl OS X apparently has stat64 but not open64.
