--- glib-2.86.0/meson.build.~1~	2025-09-05 08:42:14.000000000 -0400
+++ glib-2.86.0/meson.build	2025-09-14 19:26:07.323353559 -0400
@@ -2549,26 +2549,6 @@
 # Some installed tests require a custom environment
 env_program = find_program('env', required: installed_tests_enabled)
 
-# illumos & Solaris may need extra definitions to expose some SUS/POSIX
-# interfaces in headers that conflict with previous Solaris headers.
-# But if we define them to request an older version of the standards,
-# we may hide things introduced in newer versions.  We only check the
-# versions that are supported on systems new enough that meson runs on them.
-if host_system == 'sunos'
-  xopen_test_code = '''#include <unistd.h>
-  #if _XOPEN_VERSION != _XOPEN_SOURCE
-  #error "XOPEN_SOURCE of _XOPEN_SOURCE not supported"
-  #endif'''
-  foreach std : ['800', '700', '600']
-    if cc.compiles(xopen_test_code, args: '-D_XOPEN_SOURCE=' + std, name: 'building with _XOPEN_SOURCE=' + std)
-      xopen_version = std
-      break
-    endif
-  endforeach
-  glib_conf.set('_XOPEN_SOURCE', xopen_version)
-  glib_conf.set('__EXTENSIONS__',1)
-endif
-
 # Sadly Meson does not expose this value:
 # https://github.com/mesonbuild/meson/pull/3460
 if host_system == 'windows'
