--- glib-2.62.3/glib/glibconfig.h.in.1	2019-12-18 22:04:52.612866967 +0000
+++ glib-2.62.3/glib/glibconfig.h.in	2019-12-18 22:09:08.401976414 +0000
@@ -74,12 +74,21 @@
 #define GLIB_SIZEOF_SIZE_T @glib_size_t@
 #define GLIB_SIZEOF_SSIZE_T @glib_ssize_t@
 
-typedef signed @glib_size_type_define@ gssize;
-typedef unsigned @glib_size_type_define@ gsize;
-#define G_GSIZE_MODIFIER @gsize_modifier@
-#define G_GSSIZE_MODIFIER @gssize_modifier@
-#define G_GSIZE_FORMAT @gsize_format@
-#define G_GSSIZE_FORMAT @gssize_format@
+#if (defined (__amd64) || defined(__sparcv9))
+typedef signed long gssize;
+typedef unsigned long gsize;
+#define G_GSIZE_MODIFIER "l"
+#define G_GSSIZE_MODIFIER "l"
+#define G_GSIZE_FORMAT "lu"
+#define G_GSSIZE_FORMAT "li"
+#else
+typedef signed int gssize;
+typedef unsigned int gsize;
+#define G_GSIZE_MODIFIER ""
+#define G_GSSIZE_MODIFIER ""
+#define G_GSIZE_FORMAT "u"
+#define G_GSSIZE_FORMAT "i"
+#endif
 
 #define G_MAXSIZE	G_MAXU@glib_msize_type@
 #define G_MINSSIZE	G_MIN@glib_msize_type@
