--- glib-2.25.1/configure.ac-orig	2011-09-14 10:46:42.436850348 -0500
+++ glib-2.25.1/configure.ac	2011-09-14 10:47:05.373926999 -0500
@@ -1089,7 +1089,7 @@ if test $glib_native_win32 = no; then
 	       res_query("test", 0, 0, (void *)0, 0);
               ],[AC_MSG_RESULT([yes])],
               [save_libs="$LIBS"
-               LIBS="-lresolv $LIBS"
+               LIBS="-lresolv -lsocket $LIBS"
                AC_TRY_LINK([#include <sys/types.h>
 			    #include <netinet/in.h>
 			    #include <arpa/nameser.h>
@@ -2011,7 +2011,7 @@ if test x"$have_threads" = xposix; then
       # Sun Studio expands -mt to -D_REENTRANT and -lthread
       # gcc expands -pthreads to -D_REENTRANT -D_PTHREADS -lpthread
       G_THREAD_CFLAGS="-D_REENTRANT -D_PTHREADS"
-      G_THREAD_LIBS="-lpthread -lthread"
+      G_THREAD_LIBS="-lpthread"
       ;;
     *)
       for flag in pthread pthreads mt; do
