--- glib-2.75.2/meson.build	2023-01-25 16:18:55.570622511 +0000
+++ glib-2.75.2/meson.build.new	2023-01-25 17:10:36.757782083 +0000
@@ -2274,9 +2274,8 @@
 
 glib_conf.set('HAVE_PROC_SELF_CMDLINE', have_proc_self_cmdline)
 
-python = import('python').find_installation()
-# used for '#!/usr/bin/env <name>'
-python_name = 'python3'
+python = import('python').find_installation('python3.9')
+python_name = '/usr/bin/python3.9'
 
 python_version = python.language_version()
 python_version_req = '>=3.5'
