diff -ur glib-2.55.2-orig/gio/gdbus-2.0/codegen/gdbus-codegen.in glib-2.55.2/gio/gdbus-2.0/codegen/gdbus-codegen.in
--- glib-2.55.2-orig/gio/gdbus-2.0/codegen/gdbus-codegen.in	2018-02-16 17:18:51.213354835 +0000
+++ glib-2.55.2/gio/gdbus-2.0/codegen/gdbus-codegen.in	2018-02-16 17:19:19.687640248 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # GDBus - GLib D-Bus Library
 #
diff -ur glib-2.55.2-orig/gobject/glib-genmarshal.in glib-2.55.2/gobject/glib-genmarshal.in
--- glib-2.55.2-orig/gobject/glib-genmarshal.in	2018-02-16 17:18:51.225945754 +0000
+++ glib-2.55.2/gobject/glib-genmarshal.in	2018-02-16 17:19:48.804553286 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # pylint: disable=too-many-lines, missing-docstring, invalid-name
 
diff -ur glib-2.55.2-orig/gobject/glib-mkenums.in glib-2.55.2/gobject/glib-mkenums.in
--- glib-2.55.2-orig/gobject/glib-mkenums.in	2018-02-16 17:18:51.226243036 +0000
+++ glib-2.55.2/gobject/glib-mkenums.in	2018-02-16 17:20:01.125266201 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # If the code below looks horrible and unpythonic, do not panic.
 #
