--- glib-2.62.3/meson.build.~2~	2019-12-19 12:55:14.141812814 +0000
+++ glib-2.62.3/meson.build	2019-12-19 12:57:07.818095148 +0000
@@ -1992,9 +1992,8 @@
 
 glib_conf.set('HAVE_PROC_SELF_CMDLINE', have_proc_self_cmdline)
 
-python = import('python').find_installation('python3')
-# used for '#!/usr/bin/env <name>'
-python_name = 'python3'
+python = import('python').find_installation('python3.5')
+python_name = '/usr/bin/python3.5'
 
 python_version = python.language_version()
 python_version_req = '>=3.4'
diff -ur glib-2.55.2-orig/gio/gdbus-2.0/codegen/gdbus-codegen.in glib-2.55.2/gio/gdbus-2.0/codegen/gdbus-codegen.in
--- glib-2.55.2-orig/gio/gdbus-2.0/codegen/gdbus-codegen.in	2018-02-16 17:18:51.213354835 +0000
+++ glib-2.55.2/gio/gdbus-2.0/codegen/gdbus-codegen.in	2018-02-16 17:19:19.687640248 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # GDBus - GLib D-Bus Library
 #
diff -ur glib-2.55.2-orig/gobject/glib-genmarshal.in glib-2.55.2/gobject/glib-genmarshal.in
--- glib-2.55.2-orig/gobject/glib-genmarshal.in	2018-02-16 17:18:51.225945754 +0000
+++ glib-2.55.2/gobject/glib-genmarshal.in	2018-02-16 17:19:48.804553286 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # pylint: disable=too-many-lines, missing-docstring, invalid-name
 
diff -ur glib-2.55.2-orig/gobject/glib-mkenums.in glib-2.55.2/gobject/glib-mkenums.in
--- glib-2.55.2-orig/gobject/glib-mkenums.in	2018-02-16 17:18:51.226243036 +0000
+++ glib-2.55.2/gobject/glib-mkenums.in	2018-02-16 17:20:01.125266201 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!@PYTHON@
 
 # If the code below looks horrible and unpythonic, do not panic.
 #
--- glib-2.62.3/glib/gtester-report.in.1	2019-12-19 12:52:05.595160008 +0000
+++ glib-2.62.3/glib/gtester-report.in	2019-12-19 12:52:15.716208884 +0000
@@ -1,4 +1,4 @@
-#! /usr/bin/env @PYTHON@
+#!@PYTHON@
 # GLib Testing Framework Utility			-*- Mode: python; -*-
 # Copyright (C) 2007 Imendio AB
 # Authors: Tim Janik
