diff --git a/glib/tests/mapping.c b/glib/tests/mapping.c
index ea3495c..996e4da 100644
--- a/glib/tests/mapping.c
+++ b/glib/tests/mapping.c
@@ -225,7 +225,7 @@ test_child_private (void)
   signal (SIGUSR1, handle_usr1);
 #endif
 
-  g_snprintf (pid, sizeof(pid), "%d", getpid ());
+  g_snprintf (pid, sizeof(pid), "%ld", (long) getpid ());
   child_argv[0] = local_argv[0];
   child_argv[1] = "mapchild";
   child_argv[2] = pid;
