--- glib-2.85.4/glib/tests/mapping.c.~1~	2025-08-22 08:29:34.000000000 -0400
+++ glib-2.85.4/glib/tests/mapping.c	2025-09-03 13:40:32.045301689 -0400
@@ -234,7 +234,7 @@
   spawn_flags |= G_SPAWN_DO_NOT_REAP_CHILD;
 #endif
 
-  g_snprintf (pid, sizeof (pid), "%d", (int) getpid ());
+  g_snprintf (pid, sizeof (pid), "%ld", (long) getpid ());
   child_argv[0] = local_argv[0];
   child_argv[1] = "mapchild";
   child_argv[2] = pid;
