--- glib-2.62.3/tests/mapping-test.c.1	2019-12-19 00:36:15.376515079 +0000
+++ glib-2.62.3/tests/mapping-test.c	2019-12-19 00:42:04.596913255 +0000
@@ -208,7 +208,7 @@
   signal (SIGUSR1, handle_usr1);
 #endif
 
-  g_snprintf (pid, sizeof(pid), "%d", getpid ());
+  g_snprintf (pid, sizeof(pid), "%ld", (long) getpid ());
   child_argv[0] = argv0;
   child_argv[1] = "mapchild";
   child_argv[2] = pid;
