--- llvm-20.1.6.src/tools/clang/lib/Basic/Targets/OSTargets.h.orig	2025-05-28 00:33:46.000000000 +0200
+++ llvm-20.1.6.src/tools/clang/lib/Basic/Targets/OSTargets.h	2025-06-03 17:47:49.194655635 +0200
@@ -615,6 +615,7 @@
   void getOSDefines(const LangOptions &Opts, const llvm::Triple &Triple,
                     MacroBuilder &Builder) const override {
     DefineStd(Builder, "sun", Opts);
+    Builder.defineMacro("__illumos__");
     DefineStd(Builder, "unix", Opts);
     Builder.defineMacro("__svr4__");
     Builder.defineMacro("__SVR4");
