--- cdrtools-3.00/DEFAULTS/Defaults.sunos5.orig	Mon Apr  2 12:22:58 2012
+++ cdrtools-3.00/DEFAULTS/Defaults.sunos5	Mon Apr  2 12:23:37 2012
@@ -29,9 +29,9 @@
 CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
+LDPATH=		
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	
 
 ###########################################################################
 #
@@ -38,7 +38,7 @@
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/opt/schily
+INS_BASE=	/usr
 INS_KBASE=	/
 INS_RBASE=	/
 #
