--- cdrtools-3.00/mkisofs/iso9660.h.orig	Tue Jun 28 15:30:50 2011
+++ cdrtools-3.00/mkisofs/iso9660.h	Tue Jun 28 15:32:46 2011
@@ -169,6 +169,29 @@
 	char pad2			[ISODCL(13,  32)];
 };
 
+/* El Torito section header entry in boot catalog */
+struct eltorito_sectionheader_entry {
+	char flags			[ISODCL(1,    1)]; /* 711 */
+#define	ELTORITO_SHDR_FLAG_SHDR		0x90
+#define	ELTORITO_SHDR_FLAG_LAST_SHDR	0x91
+	char platform_id		[ISODCL(2,    2)];
+	char entry_count		[ISODCL(3,    4)]; /* 711 */
+	char id				[ISODCL(5,   32)];
+};
+
+/* El Torito initial/default entry in boot catalog */
+struct eltorito_section_entry {
+	char boot_id			[ISODCL(1,    1)]; /* 711 */
+	char boot_media			[ISODCL(2,    2)];
+	char loadseg			[ISODCL(3,    4)]; /* 711 */
+	char sys_type			[ISODCL(5,    5)];
+	char pad1			[ISODCL(6,    6)];
+	char nsect			[ISODCL(7,    8)];
+	char bootoff			[ISODCL(9,   12)];
+	 char sel_criteria		[ISODCL(13,  13)];
+	char vendor_sel_criteria	[ISODCL(14,  32)];
+};
+
 /*
  * XXX JS: The next two structures have odd lengths!
  * Some compilers (e.g. on Sun3/mc68020) padd the structures to even length.
