--- cdrtools-3.00/mkisofs/mkisofs.h.orig	Tue Jun 28 15:31:14 2011
+++ cdrtools-3.00/mkisofs/mkisofs.h	Tue Jun 28 15:32:46 2011
@@ -287,6 +287,10 @@
 
 struct eltorito_boot_entry_info {
 	struct eltorito_boot_entry_info *next;
+	int		type;
+#define	ELTORITO_DEFAULT_ENTRY  0
+#define	ELTORITO_SECTION_HEADER 1
+
 	char		*boot_image;
 	int		not_bootable;
 	int		no_emul_boot;
@@ -294,6 +298,10 @@
 	int		boot_info_table;
 	int		load_size;
 	int		load_addr;
+
+/* When type == ELTORITO_SECTION_HEADER */
+	int		boot_platid;
+
 };
 
 extern int	goof;
