--- cdrtools-3.00/mkisofs/write.c.orig	Tue Jun 28 15:31:20 2011
+++ cdrtools-3.00/mkisofs/write.c	Tue Jun 28 15:32:46 2011
@@ -1978,6 +1978,7 @@
 	int		i;
 	int		s;
 	Uchar		*cp;
+	extern		char *modification_date;
 
 
 	time(&begun);
@@ -2079,7 +2080,11 @@
 	FILL_SPACE(application_data);
 
 	memcpy(vol_desc.creation_date, iso_time, 17);
-	memcpy(vol_desc.modification_date, iso_time, 17);
+	if (modification_date == NULL) {
+		memcpy(vol_desc.modification_date, iso_time, 17);
+	} else {
+		memcpy(vol_desc.modification_date, modification_date, 17);
+	}
 	memcpy(vol_desc.expiration_date, "0000000000000000", 17);
 	memcpy(vol_desc.effective_date, iso_time, 17);
 
