--- FreeImage/Source/FreeImage/PluginWebP.cpp	Wed Jun 15 15:48:12 2016
+++ FreeImage/Source/FreeImage/PluginWebP_.cpp	Tue May 24 01:23:27 2022
@@ -24,9 +24,9 @@
 
 #include "../Metadata/FreeImageTag.h"
 
-#include "../LibWebP/src/webp/decode.h"
-#include "../LibWebP/src/webp/encode.h"
-#include "../LibWebP/src/webp/mux.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include "webp/mux.h"
 
 // ==========================================================
 // Plugin Interface
