--- FreeImage/Source/FreeImage/PluginTIFF.cpp	Sun Jul 29 00:24:44 2018
+++ FreeImage/Source/FreeImage/PluginTIFF_.cpp	Tue May 24 01:20:01 2022
@@ -37,9 +37,9 @@
 
 #include "FreeImage.h"
 #include "Utilities.h"
-#include "../LibTIFF4/tiffiop.h"
+#include "tiffio.h"
 #include "../Metadata/FreeImageTag.h"
-#include "../OpenEXR/Half/half.h"
+#include "half.h"
 
 #include "FreeImageIO.h"
 #include "PSDParser.h"
