# Patch required for Solaris.
# Will contribute upstream if possible.

--- gdb-8.0-orig/gdb/solib-svr4.c	2017-06-04 08:54:54.000000000 -0700
+++ gdb-8.0/gdb/solib-svr4.c	2017-07-21 09:51:56.459712610 -0700
@@ -174,8 +174,12 @@ svr4_same_1 (const char *gdb_so_name, co
       && strcmp (inferior_so_name, "/lib/ld.so.1") == 0)
     return 1;
 
-  /* Similarly, we observed the same issue with sparc64, but with
+  /* Similarly, we observed the same issue with amd64 and sparc64, but with
      different locations.  */
+  if (strcmp (gdb_so_name, "/usr/lib/amd64/ld.so.1") == 0
+      && strcmp (inferior_so_name, "/lib/amd64/ld.so.1") == 0)
+    return 1;
+
   if (strcmp (gdb_so_name, "/usr/lib/sparcv9/ld.so.1") == 0
       && strcmp (inferior_so_name, "/lib/sparcv9/ld.so.1") == 0)
     return 1;
