diff -wpruN '--exclude=*.orig' a~/gdb/osabi.c a/gdb/osabi.c
--- a~/gdb/osabi.c	1970-01-01 00:00:00
+++ a/gdb/osabi.c	1970-01-01 00:00:00
@@ -64,7 +64,7 @@ static const struct osabi_names gdb_osab
 
   { "SVR4", NULL },
   { "GNU/Hurd", NULL },
-  { "Solaris", NULL },
+  { "illumos", "solaris[2-9]\.[0-9][0-9]" },
   { "GNU/Linux", "linux(-gnu[^-]*)?" },
   { "FreeBSD", NULL },
   { "NetBSD", NULL },
