--- gdb-7.10.1/gdb/Makefile.in.~1~	Sat Dec  5 07:16:45 2015
+++ gdb-7.10.1/gdb/Makefile.in	Sat Feb 20 21:26:58 2016
@@ -160,7 +160,7 @@
 # Where is the READLINE library?  Typically in ../readline.
 READLINE_DIR = ../readline
 READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE = @READLINE@
+READLINE = $(READLINE_DIR)/libreadline.a $(READLINE_DIR)/libhistory.a
 READLINE_DEPS = @READLINE_DEPS@
 READLINE_CFLAGS = @READLINE_CFLAGS@
 
@@ -187,7 +187,7 @@
 GDB_WARN_CFLAGS_NO_DEFS = `echo " $(GDB_WARN_CFLAGS) " \
 		   | sed "s/ -Wold-style-definition / -Wno-old-style-definition /g"`
 
-RDYNAMIC = @RDYNAMIC@
+RDYNAMIC =
 
 # Where is the INTL library?  Typically in ../intl.
 INTL = @LIBINTL@
@@ -524,7 +524,7 @@
 CONFIG_OBS= @CONFIG_OBS@
 CONFIG_SRCS= @CONFIG_SRCS@
 CONFIG_DEPS= @CONFIG_DEPS@
-CONFIG_LDFLAGS = @CONFIG_LDFLAGS@
+CONFIG_LDFLAGS =
 ENABLE_CFLAGS= @ENABLE_CFLAGS@
 CONFIG_ALL= @CONFIG_ALL@
 CONFIG_CLEAN= @CONFIG_CLEAN@
@@ -1088,7 +1088,7 @@
 
 TSOBS = inflow.o
 
-SUBDIRS = doc @subdirs@ data-directory $(GNULIB_BUILDDIR)
+SUBDIRS = @subdirs@ data-directory $(GNULIB_BUILDDIR)
 CLEANDIRS = $(SUBDIRS)
 
 # List of subdirectories in the build tree that must exist.
