--- /dev/null	2013-10-05 19:42:45.000000000 -0700
+++ gdb-7.6/gdb/features/sparc-fpu.c	2013-08-28 00:21:38.000000000 -0700
@@ -0,0 +1,16 @@
+/* THIS FILE IS GENERATED.  -*- buffer-read-only: t -*- vi:set ro:
+  Original: sparc-fpu.xml */
+
+#include "defs.h"
+#include "osabi.h"
+#include "target-descriptions.h"
+
+struct target_desc *tdesc_sparc_fpu;
+static void
+initialize_tdesc_sparc_fpu (void)
+{
+  struct target_desc *result = allocate_target_description ();
+  struct tdesc_feature *feature;
+
+  tdesc_sparc_fpu = result;
+}
