--- gdb-7.10.1/gdb/data-directory/Makefile.in.~1~	Mon Jul  6 12:57:46 2015
+++ gdb-7.10.1/gdb/data-directory/Makefile.in	Sat Feb 20 21:01:42 2016
@@ -49,7 +49,9 @@
 	arm-linux.xml aarch64-linux.xml \
 	ppc-linux.xml ppc64-linux.xml \
 	i386-linux.xml amd64-linux.xml \
+	i386-solaris.xml amd64-solaris.xml \
 	sparc-linux.xml sparc64-linux.xml \
+	sparc-solaris.xml sparc64-solaris.xml \
 	mips-o32-linux.xml mips-n32-linux.xml mips-n64-linux.xml \
 	s390-linux.xml s390x-linux.xml
 
