diff -ur cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc1/ha_logger.xml.in cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/ha_logger.xml.in
--- cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc1/ha_logger.xml.in	2017-09-17 09:34:08.893024952 +0000
+++ cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/ha_logger.xml.in	2017-09-17 09:35:11.721935313 +0000
@@ -103,8 +103,8 @@
   <refsection id="rs-ha_logger-seealso">
     <title>See also</title>
     <para>
-      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
-      <citerefentry><refentrytitle>ha_logd</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>1m</manvolnum></citerefentry>,
+      <citerefentry><refentrytitle>ha_logd</refentrytitle><manvolnum>1m</manvolnum></citerefentry>
     </para>
   </refsection>
 </refentry>
diff -ur cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc1/stonith.xml.in cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/stonith.xml.in
--- cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc1/stonith.xml.in	2017-09-17 09:34:08.893240899 +0000
+++ cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/stonith.xml.in	2017-09-17 09:35:49.046442523 +0000
@@ -28,7 +28,7 @@
   </refentryinfo>
   <refmeta>
     <refentrytitle>stonith</refentrytitle>
-    <manvolnum>8</manvolnum>
+    <manvolnum>1m</manvolnum>
     <refmiscinfo class="manual">System administration utilities</refmiscinfo>
   </refmeta>
   <refnamediv>
@@ -319,8 +319,8 @@
   <refsection id="rs-stonith-seealso">
     <title>See also</title>
     <para>
-      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
-      <citerefentry><refentrytitle>meatclient</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>1m</manvolnum></citerefentry>,
+      <citerefentry><refentrytitle>meatclient</refentrytitle><manvolnum>1m</manvolnum></citerefentry>
     </para>
   </refsection>
 </refentry>
--- cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/ha_logd.xml.in.~1~	2017-08-03 11:41:13.000000000 +0000
+++ cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/ha_logd.xml.in	2017-09-17 09:40:07.250355358 +0000
@@ -34,7 +34,7 @@
   </refentryinfo>
   <refmeta>
     <refentrytitle>ha_logd</refentrytitle>
-    <manvolnum>8</manvolnum>
+    <manvolnum>1m</manvolnum>
     <refmiscinfo class="manual">System administration utilities</refmiscinfo>
   </refmeta>
   <refnamediv>
@@ -127,7 +127,7 @@
   <refsection id="rs-ha_logd-seealso">
     <title>See also</title>
     <para>
-      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>1m</manvolnum></citerefentry>,
       <citerefentry><refentrytitle>ha_logger</refentrytitle><manvolnum>1</manvolnum></citerefentry>
     </para>
   </refsection>
--- cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/meatclient.xml.in.1	2017-09-17 09:42:27.607796329 +0000
+++ cluster-glue-bdd95fd0f89f31b19fa46ffa0ea3f5022ebcd858/doc/meatclient.xml.in	2017-09-17 09:42:43.369887187 +0000
@@ -34,7 +34,7 @@
   </refentryinfo>
   <refmeta>
     <refentrytitle>meatclient</refentrytitle>
-    <manvolnum>8</manvolnum>
+    <manvolnum>1m</manvolnum>
     <refmiscinfo class="manual">System administration utilities</refmiscinfo>
   </refmeta>
   <refnamediv>
@@ -70,8 +70,8 @@
   <refsection id="rs-meatclient-seealso">
     <title>See also</title>
     <para>
-      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
-      <citerefentry><refentrytitle>stonith</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+      <citerefentry><refentrytitle>heartbeat</refentrytitle><manvolnum>1m</manvolnum></citerefentry>,
+      <citerefentry><refentrytitle>stonith</refentrytitle><manvolnum>1m</manvolnum></citerefentry>
     </para>
   </refsection>
 </refentry>
