diff -Nura a/usr/lib/stonith/plugins/external/dracmc-telnet b/usr/lib/stonith/plugins/external/dracmc-telnet
--- a/lib/plugins/stonith/external/dracmc-telnet	2016-10-28 08:26:12.544448553 +0200
+++ b/lib/plugins/stonith/external/dracmc-telnet	2016-10-28 08:27:07.576243323 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # vim: set filetype=python
 #######################################################################
 #
diff -Nura a/usr/lib/stonith/plugins/external/ibmrsa-telnet b/usr/lib/stonith/plugins/external/ibmrsa-telnet
--- a/lib/plugins/stonith/external/ibmrsa-telnet	2016-10-28 08:25:57.526113047 +0200
+++ b/lib/plugins/stonith/external/ibmrsa-telnet	2016-10-28 08:27:34.908188863 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.7
 # vim: set filetype=python
 #######################################################################
 #
diff -Nura a/usr/lib/stonith/plugins/external/riloe b/usr/lib/stonith/plugins/external/riloe
--- a/lib/plugins/stonith/external/riloe	2016-10-28 08:25:42.467974465 +0200
+++ b/lib/plugins/stonith/external/riloe	2016-10-28 08:27:50.022642344 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # Stonith module for RILOE Stonith device
 #
diff -Nura a/usr/lib/stonith/plugins/stonith2/ribcl.py b/usr/lib/stonith/plugins/stonith2/ribcl.py
--- a/lib/plugins/stonith/ribcl.py.in	2016-10-28 08:25:23.193133907 +0200
+++ b/lib/plugins/stonith/ribcl.py.in	2016-10-28 08:28:16.882225009 +0200
@@ -1,4 +1,4 @@
-#!@PYTHON@
+#!/usr/bin/python2.7

 
 
 #
