https://github.com/repo-helper/pyproject-parser/issues/70

--- pyproject-parser-0.11.1/pyproject.toml.orig
+++ pyproject-parser-0.11.1/pyproject.toml
@@ -47,7 +47,6 @@
 [project.scripts]
 pyproject-parser = "pyproject_parser.__main__:main"
 check-pyproject = "pyproject_parser.__main__:check"
-pyproject-fmt = "pyproject_parser.__main__:reformat"
 pyproject-info = "pyproject_parser.__main__:info"
 
 [project.optional-dependencies]
