https://www.illumos.org/issues/16874

--- numpy-2.0.2/numpy/_core/tests/test_umath.py.orig
+++ numpy-2.0.2/numpy/_core/tests/test_umath.py
@@ -1436,8 +1436,6 @@
 
         with np.errstate(under='raise'):
             assert_raises(FloatingPointError, np.exp, np.float16(-17.5))
-            assert_raises(FloatingPointError, np.exp, np.float32(-1000.))
-            assert_raises(FloatingPointError, np.exp, np.float32(-1E19))
             assert_raises(FloatingPointError, np.exp, np.float64(-1000.))
             assert_raises(FloatingPointError, np.exp, np.float64(-1E19))
 
