https://github.com/numpy/numpy/pull/27685

--- numpy-2.0.2/numpy/random/_examples/cython/extending.pyx.orig
+++ numpy-2.0.2/numpy/random/_examples/cython/extending.pyx
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 #cython: language_level=3
 
 from libc.stdint cimport uint32_t
--- numpy-2.0.2/numpy/random/_examples/cython/extending_distributions.pyx.orig
+++ numpy-2.0.2/numpy/random/_examples/cython/extending_distributions.pyx
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 #cython: language_level=3
 """
 This file shows how the to use a BitGenerator to create a distribution.
--- numpy-2.0.2/numpy/random/c_distributions.pxd.orig
+++ numpy-2.0.2/numpy/random/c_distributions.pxd
@@ -1,4 +1,3 @@
-#!python
 #cython: wraparound=False, nonecheck=False, boundscheck=False, cdivision=True, language_level=3
 from numpy cimport npy_intp
 
--- numpy-2.0.2/numpy/f2py/__init__.py.orig
+++ numpy-2.0.2/numpy/f2py/__init__.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """Fortran to Python Interface Generator.
 
 Copyright 1999 -- 2011 Pearu Peterson all rights reserved.
--- numpy-2.0.2/numpy/f2py/cfuncs.py.orig
+++ numpy-2.0.2/numpy/f2py/cfuncs.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """
 C declarations, CPP macros, and C functions for f2py2e.
 Only required declarations/macros/functions will be used.
--- numpy-2.0.2/numpy/f2py/crackfortran.py.orig
+++ numpy-2.0.2/numpy/f2py/crackfortran.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """
 crackfortran --- read fortran (77,90) code and extract declaration information.
 
--- numpy-2.0.2/numpy/f2py/f2py2e.py.orig
+++ numpy-2.0.2/numpy/f2py/f2py2e.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """
 
 f2py2e - Fortran to Python C/API generator. 2nd Edition.
--- numpy-2.0.2/numpy/f2py/rules.py.orig
+++ numpy-2.0.2/numpy/f2py/rules.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """
 
 Rules for building C/API module with f2py2e.
--- numpy-2.0.2/numpy/distutils/system_info.py.orig
+++ numpy-2.0.2/numpy/distutils/system_info.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 """
 This file defines a set of system_info classes for getting
 information about various resources (libraries, library directories,
