--- xf86-input-mouse-1.9.1/src/mouse.c.~1~	2014-08-12 10:36:34.000000000 +0400
+++ xf86-input-mouse-1.9.1/src/mouse.c	2015-02-14 18:59:12.095956075 +0300
@@ -1753,13 +1753,14 @@
         ErrorF("assigning %p name=%s\n", device, pInfo->name);
 #endif
         MouseInitProperties(device);
+        pInfo->fd = xf86OpenSerial(pInfo->options);
+        if (pInfo->fd == -1)
+	    xf86Msg(X_WARNING, "%s: cannot open input device\n", pInfo->name);
+
         break;
 
     case DEVICE_ON:
-        pInfo->fd = xf86OpenSerial(pInfo->options);
-        if (pInfo->fd == -1)
-            xf86Msg(X_WARNING, "%s: cannot open input device\n", pInfo->name);
-        else {
+        if (pInfo->fd != -1) {
 #if defined(__NetBSD__) && defined(WSCONS_SUPPORT) && defined(WSMOUSEIO_SETVERSION)
             int version = WSMOUSE_EVENT_VERSION;
             if (ioctl(pInfo->fd, WSMOUSEIO_SETVERSION, &version) == -1)
@@ -1820,8 +1821,6 @@
                 XisbFree(pMse->buffer);
                 pMse->buffer = NULL;
             }
-            xf86CloseSerial(pInfo->fd);
-            pInfo->fd = -1;
             if (pMse->emulate3Buttons || pMse->emulate3ButtonsSoft)
             {
                 RemoveBlockAndWakeupHandlers (MouseBlockHandler,
@@ -1832,6 +1831,10 @@
         device->public.on = FALSE;
         break;
     case DEVICE_CLOSE:
+        if (pInfo->fd != -1) {
+            xf86CloseSerial(pInfo->fd);
+            pInfo->fd = -1;
+        }
         free(pMse->mousePriv);
         pMse->mousePriv = NULL;
         break;
