--- ./src/sun_mouse.c.orig	2016-05-21 23:08:17.372636430 +0200
+++ ./src/sun_mouse.c	2016-05-21 23:09:21.192957910 +0200
@@ -51,6 +50,8 @@
 #include <xorg-config.h>
 #endif
 
+#include <sys/param.h>
+
 #include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSlib.h"
@@ -58,6 +59,10 @@
 #include "xisb.h"
 #include "mipointer.h"
 #include "xf86Crtc.h"
+
+#include <unistd.h>
+#include <stropts.h>
+
 #include <sys/stropts.h>
 #include <sys/vuid_event.h>
 #include <sys/msio.h>
