*** tcsh-6.17.00/config_f.h.orig	Thu Aug 20 12:31:33 2009
--- tcsh-6.17.00/config_f.h	Thu Aug 20 12:31:42 2009
***************
*** 131,137 ****
   *		variable "mbytemap".
   *		(use for multi-byte table check)
   */
! #undef MBYTEDEBUG
  
  /*
   * NEWGRP	Provide a newgrp builtin.
--- 131,137 ----
   *		variable "mbytemap".
   *		(use for multi-byte table check)
   */
! #define MBYTEDEBUG
  
  /*
   * NEWGRP	Provide a newgrp builtin.
