--- webkit-1.2.7/JavaScriptCore/parser/Grammar.y.~1~	2010-09-10 17:20:23.000000000 +0400
+++ webkit-1.2.7/JavaScriptCore/parser/Grammar.y	2017-03-07 00:01:20.578705167 +0300
@@ -1,4 +1,6 @@
-%pure_parser
+%pure-parser
+%parse-param { void* globalPtr }
+%lex-param { void* globalPtr }
 
 %{
 
@@ -49,7 +51,7 @@
 #define YYERROR_VERBOSE
 #endif
 
-int jscyyerror(const char*);
+int jscyyerror(void*, void*, const char*);
 
 static inline bool allowAutomaticSemicolon(JSC::Lexer&, int);
 
@@ -86,8 +88,6 @@
 
 #endif
 
-#define YYPARSE_PARAM globalPtr
-#define YYLEX_PARAM globalPtr
 
 template <typename T> inline NodeDeclarationInfo<T> createNodeDeclarationInfo(T node,
     ParserArenaData<DeclarationStacks::VarStack>* varDecls,
@@ -2066,7 +2066,7 @@
 }
 
 // Called by yyparse on error.
-int yyerror(const char*)
+int yyerror(void*, void *, const char*)
 {
     return 1;
 }
