--- avahi-0.8/avahi-python/Makefile.am.~1~	2020-02-12 10:47:14.336807694 +0200
+++ avahi-0.8/avahi-python/Makefile.am	2025-04-19 21:49:12.731324541 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 EXTRA_DIST = \
 	avahi-bookmarks.in
 
--- avahi-0.8/avahi-common/Makefile.am.~1~	2015-04-01 07:58:14.145727222 +0300
+++ avahi-0.8/avahi-common/Makefile.am	2025-04-19 21:49:12.731574702 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 avahi_commonincludedir=$(includedir)/avahi-common
 
 avahi_commoninclude_HEADERS = \
--- avahi-0.8/avahi-client/Makefile.am.~1~	2015-04-01 07:58:14.145727222 +0300
+++ avahi-0.8/avahi-client/Makefile.am	2025-04-19 21:49:12.731791662 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 if HAVE_DBUS
 
 avahi_clientincludedir=$(includedir)/avahi-client
--- avahi-0.8/avahi-compat-libdns_sd/Makefile.am.~1~	2015-04-01 07:58:14.149727123 +0300
+++ avahi-0.8/avahi-compat-libdns_sd/Makefile.am	2025-04-19 21:49:12.731998542 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 if HAVE_DBUS
 if ENABLE_COMPAT_LIBDNS_SD
 
--- avahi-0.8/avahi-daemon/Makefile.am.~1~	2017-05-20 10:36:29.335062232 +0300
+++ avahi-0.8/avahi-daemon/Makefile.am	2025-04-19 21:49:12.732200593 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 if HAVE_LIBDAEMON
 if HAVE_XML
 
--- avahi-0.8/avahi-glib/Makefile.am.~1~	2015-04-01 07:58:14.153727024 +0300
+++ avahi-0.8/avahi-glib/Makefile.am	2025-04-19 21:49:12.732423614 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 if HAVE_GLIB
 
 avahiglibincludedir=$(includedir)/avahi-glib
--- avahi-0.8/avahi-compat-howl/Makefile.am.~1~	2015-04-01 07:58:14.145727222 +0300
+++ avahi-0.8/avahi-compat-howl/Makefile.am	2025-04-19 21:49:12.732611237 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 SUBDIRS = . samples
 
 if HAVE_DBUS
--- avahi-0.8/avahi-compat-howl/samples/Makefile.am.~1~	2015-04-01 07:58:14.149727123 +0300
+++ avahi-0.8/avahi-compat-howl/samples/Makefile.am	2025-04-19 21:49:12.732800254 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir) -I$(top_srcdir)/avahi-compat-howl/include
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 if HAVE_DBUS
 if ENABLE_COMPAT_HOWL
 
--- avahi-0.8/avahi-core/Makefile.am.~1~	2015-04-01 07:58:14.149727123 +0300
+++ avahi-0.8/avahi-core/Makefile.am	2025-04-19 21:49:12.733032492 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 avahiincludedir=$(includedir)/avahi-core
 
 avahiinclude_HEADERS = \
@@ -39,14 +36,12 @@
 	conformance-test \
 	avahi-reflector \
 	dns-test \
-	dns-spin-test \
 	timeeventq-test \
 	hashmap-test \
 	querier-test \
 	update-test
 
 TESTS = \
-	dns-spin-test \
 	dns-test \
 	hashmap-test
 endif
@@ -146,11 +141,6 @@
 dns_test_CFLAGS = $(AM_CFLAGS)
 dns_test_LDADD = $(AM_LDADD) ../avahi-common/libavahi-common.la
 
-dns_spin_test_SOURCES = \
-	dns-spin-test.c
-dns_spin_test_CFLAGS = $(AM_CFLAGS)
-dns_spin_test_LDADD = $(AM_LDADD) libavahi-core.la
-
 timeeventq_test_SOURCES = \
 	timeeventq-test.c \
 	timeeventq.h timeeventq.c \
--- avahi-0.8/avahi-utils/Makefile.am.~1~	2015-04-01 07:58:14.157726925 +0300
+++ avahi-0.8/avahi-utils/Makefile.am	2025-04-19 21:49:12.733268305 +0300
@@ -17,9 +17,6 @@
 
 AM_CFLAGS=-I$(top_srcdir)
 
-# This cool debug trap works on i386/gcc only
-AM_CFLAGS+='-DDEBUG_TRAP=__asm__("int $$3")'
-
 pkglibdatadir=$(libdir)/avahi
 
 if HAVE_DBUS
