--- avahi-0.6.30/configure.ac.~2~	2014-05-19 15:33:22.806928364 +0400
+++ avahi-0.6.30/configure.ac	2014-05-19 15:33:55.481490968 +0400
@@ -367,7 +367,7 @@
  # Solaris stuff
  AC_SEARCH_LIBS([inet_ntop],[nsl])
  AC_SEARCH_LIBS([recv],[socket])
- AC_CHECK_DECL([CMSG_SPACE],,CFLAGS="$CFLAGS -D_XOPEN_SOURCE=500 -D__EXTENSIONS__", [[#include <sys/socket.h>]])
+ AC_CHECK_DECL([CMSG_SPACE],,CFLAGS="$CFLAGS -D_XOPEN_SOURCE=600 -D__EXTENSIONS__", [[#include <sys/socket.h>]])
 
 # Checks for library functions.
 AC_FUNC_MEMCMP
