--- gdbm-1.23/configure.orig	2022-02-04 16:03:12.000000000 +0200
+++ gdbm-1.23/configure	2023-02-19 11:00:02.339740123 +0200
@@ -12668,7 +12668,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_sys_file_offset_bits" >&5
 $as_echo "$ac_cv_sys_file_offset_bits" >&6; }
 case $ac_cv_sys_file_offset_bits in #(
-  no | unknown) ;;
+  no | unknown | 64) ;;
   *)
 cat >>confdefs.h <<_ACEOF
 #define _FILE_OFFSET_BITS $ac_cv_sys_file_offset_bits
