https://github.com/jazzband/contextlib2/issues/54

--- contextlib2-21.6.0/tox.ini.orig
+++ contextlib2-21.6.0/tox.ini
@@ -7,8 +7,6 @@
     coverage run -m unittest discover -t . -s test
     coverage report
     coverage xml
-    # mypy won't install on PyPy, so only run the typechecking on CPython
-    !pypy3: python -m mypy.stubtest --allowlist dev/mypy.allowlist contextlib2
 deps =
     coverage
     !pypy3: mypy
