Index: trunk/trackerd.desktop.in.in
===================================================================
--- trunk/trackerd.desktop.in.in	(revision 1092)
+++ trunk/data/trackerd.desktop.in.in	(working copy)
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 Categories=
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false
 X-KDE-autostart-after=panel
 X-KDE-StartupNotify=false
 X-KDE-UniqueApplet=true
Index: trunk/src/tracker-applet/tracker-applet.desktop.in.in
===================================================================
--- trunk/src/tracker-applet/tracker-applet.desktop.in.in	(revision 1092)
+++ trunk/src/tracker-applet/tracker-applet.desktop.in.in	(working copy)
@@ -13,4 +13,4 @@
 X-GNOME-Bugzilla-Product=tracker
 X-GNOME-Bugzilla-Component=Tracker Applet
 X-GNOME-Bugzilla-Version=@VERSION@
-X-GNOME-Autostart-enabled=true
+X-GNOME-Autostart-enabled=false


