diff -ruN tracker-0.6.96.orig/Makefile.am tracker-0.6.96/Makefile.am
--- tracker-0.6.96.orig/Makefile.am	2010-02-04 10:25:44.000000000 +0000
+++ tracker-0.6.96/Makefile.am	2014-02-14 14:26:41.116540822 +0000
@@ -40,18 +40,3 @@
 	--disable-deskbar-applet		\
 	--enable-gtk-doc
 
-if OLD_EXEC_REMOVE_ALL
-install-exec-hook:
-	rm -Rf $(DESTDIR)$(bindir)/trackerd
-	rm -Rf $(DESTDIR)$(bindir)/tracker-indexer
-	rm -Rf $(DESTDIR)$(bindir)/tracker-thumbnailer
-	rm -Rf $(DESTDIR)$(bindir)/tracker-extract
-endif
-
-if OLD_DATA_REMOVE_ALL
-install-data-hook:
-	rm -Rf $(DESTDIR)$(DBUS_SERVICES_DIR)/tracker.service
-	rm -Rf $(DESTDIR)$(datadir)/tracker/sqlite-service-stored-procs.sql
-	rm -Rf $(DESTDIR)$(datadir)/tracker/tracker-introspect.xml
-endif	
-
