--- tracker-0.6.96/src/tracker-search-tool/Makefile.am.~1~	2015-03-24 17:36:51.084137854 +0300
+++ tracker-0.6.96/src/tracker-search-tool/Makefile.am	2015-03-24 17:37:00.095524832 +0300
@@ -3,7 +3,6 @@
 @INTLTOOL_DESKTOP_RULE@
 
 INCLUDES = 	\
-	-DG_DISABLE_DEPRECATED					\
 	-DTRACKER_LOCALEDIR=\""$(localedir)"\" 			\
 	-DTRACKER_DATADIR=\""$(datadir)"\"			\
 	-DGREP_COMMAND=\""$(GREP_COMMAND)"\"			\
