--- tracker-0.6.96/src/libtracker-common/tracker-hal.c.orig	2013-12-23 15:44:47.352226950 +0400
+++ tracker-0.6.96/src/libtracker-common/tracker-hal.c	2013-12-23 15:45:23.253309303 +0400
@@ -718,8 +718,6 @@
 		return "LIBHAL_DRIVE_TYPE_JAZ";
 	case LIBHAL_DRIVE_TYPE_FLASHKEY:
 		return "LIBHAL_DRIVE_TYPE_FLASHKEY";
-	case LIBHAL_DRIVE_TYPE_MO:
-		return "LIBHAL_DRIVE_TYPE_MO";
 	default:
 		return "";
 	}
