--- tracker-0.6.95/src/tracker-extract/Makefile.am.ori	2012-08-08 12:16:18.165884342 +0800
+++ tracker-0.6.95/src/tracker-extract/Makefile.am	2012-08-08 12:16:27.011073672 +0800
@@ -1,6 +1,6 @@
 include $(top_srcdir)/Makefile.decl
 
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version
 modulesdir = $(libdir)/tracker/extract-modules
 
 INCLUDES = 								\
--- tracker-0.6.95/src/tracker-indexer/modules/Makefile.am.ori	2012-08-09 14:28:44.497805072 +0800
+++ tracker-0.6.95/src/tracker-indexer/modules/Makefile.am	2012-08-09 14:28:55.392174945 +0800
@@ -1,6 +1,6 @@
 include $(top_srcdir)/Makefile.decl
 
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version
 indexer_modulesdir = $(libdir)/tracker/indexer-modules
 
 INCLUDES =								\
