diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.presentation-template_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.presentation-template_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.presentation-template_filter	2009-03-04 17:23:20.162042000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.presentation-template_filter	2009-03-04 17:33:08.464041000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.presentation_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.presentation_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.presentation_filter	2009-03-04 17:23:20.157329000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.presentation_filter	2009-03-04 17:33:01.127878000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt 
diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.spreadsheet-template_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.spreadsheet-template_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.spreadsheet-template_filter	2009-03-04 17:23:20.161104000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.spreadsheet-template_filter	2009-03-04 17:34:13.639905000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.spreadsheet_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.spreadsheet_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.spreadsheet_filter	2009-03-04 17:23:20.161854000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.spreadsheet_filter	2009-03-04 17:34:27.647737000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.text-template_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.text-template_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.text-template_filter	2009-03-04 17:23:20.157829000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.text-template_filter	2009-03-04 17:26:19.439909000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.text_filter tracker-0.6.90/filters/application/vnd.oasis.opendocument.text_filter
--- tracker-0.6.90-orig/filters/application/vnd.oasis.opendocument.text_filter	2009-03-04 17:23:20.159874000 +0800
+++ tracker-0.6.90/filters/application/vnd.oasis.opendocument.text_filter	2009-03-04 17:24:34.271984000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.stardivision.writer_filter tracker-0.6.90/filters/application/vnd.stardivision.writer_filter
--- tracker-0.6.90-orig/filters/application/vnd.stardivision.writer_filter	2009-03-04 17:23:20.161298000 +0800
+++ tracker-0.6.90/filters/application/vnd.stardivision.writer_filter	2009-03-04 17:35:09.791972000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.calc.template_filter tracker-0.6.90/filters/application/vnd.sun.xml.calc.template_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.calc.template_filter	2009-03-04 17:23:20.158224000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.calc.template_filter	2009-03-04 17:35:34.527931000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.calc_filter tracker-0.6.90/filters/application/vnd.sun.xml.calc_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.calc_filter	2009-03-04 17:23:20.161494000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.calc_filter	2009-03-04 17:35:50.639531000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.draw_filter tracker-0.6.90/filters/application/vnd.sun.xml.draw_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.draw_filter	2009-03-04 17:23:20.158990000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.draw_filter	2009-03-04 17:36:08.615914000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.impress.template_filter tracker-0.6.90/filters/application/vnd.sun.xml.impress.template_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.impress.template_filter	2009-03-04 17:23:20.160042000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.impress.template_filter	2009-03-04 17:36:49.127822000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.impress_filter tracker-0.6.90/filters/application/vnd.sun.xml.impress_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.impress_filter	2009-03-04 17:23:20.159352000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.impress_filter	2009-03-04 17:37:04.711650000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.writer.template_filter tracker-0.6.90/filters/application/vnd.sun.xml.writer.template_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.writer.template_filter	2009-03-04 17:23:20.158809000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.writer.template_filter	2009-03-04 17:37:25.695996000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
diff -ur tracker-0.6.90-orig/filters/application/vnd.sun.xml.writer_filter tracker-0.6.90/filters/application/vnd.sun.xml.writer_filter
--- tracker-0.6.90-orig/filters/application/vnd.sun.xml.writer_filter	2009-03-04 17:23:20.157128000 +0800
+++ tracker-0.6.90/filters/application/vnd.sun.xml.writer_filter	2009-03-04 17:37:55.352063000 +0800
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-nice -n19 odt2txt "$1"
+nice -n19 unzip -p "$1" content.xml | o3totxt
