diff -ur tracker-0.6.6-orig//src/tracker-preferences/tracker-preferences.desktop.in.in tracker-0.6.6//src/tracker-preferences/tracker-preferences.desktop.in.in
--- tracker-0.6.6-orig//src/tracker-preferences/tracker-preferences.desktop.in.in	Fri Sep 12 14:45:31 2008
+++ tracker-0.6.6//src/tracker-preferences/tracker-preferences.desktop.in.in	Fri Sep 12 15:00:03 2008
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Search and Indexing
-_Comment=Configure file indexing with Tracker
+# SUN_BRANDING
+_Name=File Indexing
+# SUN_BRANDING
+_Comment=Choose how files are indexed for searching
 Icon=tracker
 Exec=tracker-preferences
 Terminal=false
diff -ur tracker-0.6.6-orig//src/tracker-search-tool/tracker-search-tool.desktop.in.in tracker-0.6.6//src/tracker-search-tool/tracker-search-tool.desktop.in.in
--- tracker-0.6.6-orig//src/tracker-search-tool/tracker-search-tool.desktop.in.in	Fri Sep 12 14:45:31 2008
+++ tracker-0.6.6//src/tracker-search-tool/tracker-search-tool.desktop.in.in	Fri Sep 12 14:46:24 2008
@@ -12,4 +12,4 @@
 X-GNOME-Bugzilla-Product=tracker
 X-GNOME-Bugzilla-Component=Tracker Search Tool
 X-GNOME-Bugzilla-Version=@VERSION@
-
+NoDisplay=true
