--- jdk/make/sun/font/Makefile.~1~	2017-11-09 11:05:01.390861758 +0000
+++ jdk/make/sun/font/Makefile	2017-11-09 11:08:31.377155441 +0000
@@ -206,7 +206,7 @@
 ifeq ($(PLATFORM), solaris)
   # Note that on Solaris, fontmanager is built against the headless library.
   LDFLAGS      += -L$(LIBDIR)/$(LIBARCH)/headless
-  OTHER_LDLIBS += -lawt -L$(LIBDIR)/$(LIBARCH)/xawt -lmawt -lc $(LIBM) $(LIBCXX)
+  OTHER_LDLIBS += -lawt -L$(LIBDIR)/$(LIBARCH)/xawt -R/usr/jdk/instances/openjdk1.7.0/jre/lib/$(LIBARCH)/xawt -lmawt -lc $(LIBM) $(LIBCXX)
 else
 ifeq ($(PLATFORM), aix)
   LDFLAGS      += -L$(LIBDIR)/$(LIBARCH)/headless
--- jdk/make/sun/font/t2k/Makefile.~1~	2014-01-02 11:10:51.425535643 +0400
+++ jdk/make/sun/font/t2k/Makefile	2014-01-02 11:14:35.399856446 +0400
@@ -99,7 +99,7 @@
     endif                           
   else
     ifeq ($(PLATFORM), solaris)
-      OTHER_LDLIBS += -lawt -L$(LIBDIR)/$(LIBARCH)/xawt -lmawt
+      OTHER_LDLIBS += -lawt -L$(LIBDIR)/$(LIBARCH)/xawt -R/usr/jdk/instances/openjdk1.7.0/jre/lib/$(LIBARCH)/xawt -lmawt
     endif
   endif
 endif
--- jdk/make/sun/jawt/Makefile.~1~	2014-01-02 11:11:03.852717421 +0400
+++ jdk/make/sun/jawt/Makefile	2014-01-02 11:15:43.613185103 +0400
@@ -116,9 +116,9 @@
 #
 ifeq ($(PLATFORM), solaris)
   ifndef BUILD_HEADLESS_ONLY
-    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/xawt -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
+    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/xawt -R/usr/jdk/instances/openjdk1.7.0/jre/lib/$(LIBARCH)/xawt -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
   else
-    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/headless -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
+    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/headless -R/usr/jdk/instances/openjdk1.7.0/jre/lib/$(LIBARCH)/headless -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
   endif
 endif # PLATFORM
 
