https://github.com/OpenIndiana/oi-userland/pull/2915#issuecomment-289797801

--- rhythmbox-3.4.1/plugins/artsearch/artsearch.py.1	2017-03-28 18:31:36.755414692 +0300
+++ rhythmbox-3.4.1/plugins/artsearch/artsearch.py	2017-03-28 18:34:27.073617432 +0300
@@ -46,7 +46,10 @@
 
 	def next_search(self):
 		if len(self.searches) == 0:
-			key = RB.ExtDBKey.create_storage("album", self.key.get_field("album"))
+			album = self.key.get_field("album")
+			if not album:
+				return False
+			key = RB.ExtDBKey.create_storage("album", album)
 			key.add_field("artist", self.key.get_field("artist"))
 			self.store.store(key, RB.ExtDBSourceType.NONE, None)
 			return False
