https://github.com/steve-m-hay/Text-Balanced/pull/8

--- Text-Balanced-2.06/lib/Text/Balanced.pm.orig
+++ Text-Balanced-2.06/lib/Text/Balanced.pm
@@ -1005,7 +1005,8 @@
             {
                 $unkpos = pos($$textref)-1
                     unless $igunk || defined $unkpos;
-                _update_patvalid($textref, substr $$textref, $unkpos, pos($$textref)-$unkpos);
+                _update_patvalid($textref, substr $$textref, defined $unkpos ? $unkpos : 0,
+                                 pos($$textref) - (defined $unkpos ? $unkpos : 0));
             }
         }
 
