--- a/src/hotspot/share/nmt/vmatree.cpp	Thu Sep 25 00:07:44 2025
+++ b/src/hotspot/share/nmt/vmatree.cpp	Mon Oct  6 13:05:13 2025
@@ -28,6 +28,10 @@
 #include "utilities/globalDefinitions.hpp"
 #include "utilities/growableArray.hpp"
 
+// leak from ctype.h, but used in hotspot
+#ifdef _B
+#undef _B
+#endif
 
 // Semantics
 // This tree is used to store and track the state of virtual memory regions.
