--- gnome-commander-1.16.1/data/org.gnome.gnome-commander.desktop.in.orig	2023-07-08 17:33:10.000000000 +0200
+++ gnome-commander-1.16.1/data/org.gnome.gnome-commander.desktop.in	2023-10-28 17:56:18.236506112 +0200
@@ -8,5 +8,5 @@
 Icon=gnome-commander
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Utility;FileManager;System;FileTools;
+Categories=GNOME;System;FileManager;System;FileTools;
 StartupNotify=true
