diff -Nuraw otp_src_26.2.2_O/lib/crypto/configure otp_src_26.2.2/lib/crypto/configure
--- otp_src_26.2.2_O/lib/crypto/configure	2024-02-07 15:00:00.000000000 +0100
+++ otp_src_26.2.2/lib/crypto/configure	2024-02-18 05:14:02.000000000 +0100
@@ -5594,21 +5594,8 @@
 
 
 
-std_ssl_locations="\
-/usr/local \
-/usr/sfw \
-/usr \
-/opt/local \
-/usr/pkg \
-/usr/local/openssl \
-/usr/local/opt/openssl \
-/opt/homebrew/opt/openssl \
-/usr/lib/openssl \
-/usr/openssl \
-/usr/local/ssl \
-/usr/lib/ssl \
-/usr/ssl \
-/"
+std_ssl_locations="/usr"
+
 
 
 # Check whether --with-ssl was given.
@@ -5736,7 +5723,7 @@
 # Setup subdirectories to search in...
 search_subdirs="lib"
 if test "$ac_cv_sizeof_void_p" = "8"; then
-   search_subdirs="$search_subdirs lib64 lib/64"
+   search_subdirs="lib/amd64"
 else
    search_subdirs="$search_subdirs lib32 lib/32"
 fi
@@ -7251,7 +7238,7 @@
                       # If none of the above links, we skip runtime
                       # library path...
                       #
-                      for type in std x_std curr; do
+                      for type in curr; do
 
                           ded_ld_rpath="$ded_ld_rflg$SSL_RUNTIME_LIBDIR"
                           rpath="$SSL_RUNTIME_LIBDIR"
diff -Nuraw otp_src_26.2.2_O/lib/crypto/configure.ac otp_src_26.2.2/lib/crypto/configure.ac
--- otp_src_26.2.2_O/lib/crypto/configure.ac	2024-02-07 15:00:00.000000000 +0100
+++ otp_src_26.2.2/lib/crypto/configure.ac	2024-02-18 05:12:06.000000000 +0100
@@ -54,6 +54,7 @@
 LM_WINDOWS_ENVIRONMENT
 
 ERL_DED
+DED_EXT=so
 
 AC_MSG_CHECKING([for multiarch directory])
 multiarch_dir=
@@ -81,21 +82,7 @@
 
 AC_CHECK_SIZEOF(void *)
 
-std_ssl_locations="\
-/usr/local \
-/usr/sfw \
-/usr \
-/opt/local \
-/usr/pkg \
-/usr/local/openssl \
-/usr/local/opt/openssl \
-/opt/homebrew/opt/openssl \
-/usr/lib/openssl \
-/usr/openssl \
-/usr/local/ssl \
-/usr/lib/ssl \
-/usr/ssl \
-/"
+std_ssl_locations="/usr"
 			
 AC_ARG_WITH(ssl,
 AS_HELP_STRING([--with-ssl=PATH], [base location of OpenSSL include and lib directories])
@@ -395,7 +382,7 @@
 # Setup subdirectories to search in...
 search_subdirs="lib"
 if test "$ac_cv_sizeof_void_p" = "8"; then
-   search_subdirs="$search_subdirs lib64 lib/64"
+   search_subdirs="lib/amd64"
 else
    search_subdirs="$search_subdirs lib32 lib/32"
 fi
@@ -747,7 +734,7 @@
                       # If none of the above links, we skip runtime
                       # library path...
                       #
-                      for type in std x_std curr; do
+                      for type in curr; do
 
                           ded_ld_rpath="$ded_ld_rflg$SSL_RUNTIME_LIBDIR"
                           rpath="$SSL_RUNTIME_LIBDIR"
