--- planner-0.14.2-orig/acinclude.m4	2006-11-25 07:11:10.000000000 -0800
+++ planner-0.14.2/acinclude.m4	2008-09-05 13:38:46.377330002 -0700
@@ -87,9 +87,11 @@
         AC_MSG_CHECKING([for Python library path])
         python_path=`echo $PYTHON | sed "s,/bin.*$,,"`
         for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
-                python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
-                if test -n "$python_path" ; then
-                        break
+		if test -d "$i" ; then
+			python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
+			if test -n "$python_path" ; then
+				break
+			fi
                 fi
         done
         python_path=`echo $python_path | sed "s,/libpython.*$,,"`
