--- libxslt-1.1.26/libxslt.pc.in.orig	2012-02-28 08:52:25.350596427 -0800
+++ libxslt-1.1.26/libxslt.pc.in	2012-02-28 08:53:36.436859739 -0800
@@ -8,5 +8,5 @@
 Version: @VERSION@
 Description: XSLT library version 2.
 Requires: libxml-2.0
-Libs: @XSLT_LIBDIR@ @XSLT_LIBS@
+Libs: -L${libdir} -lxslt
 Cflags: @XSLT_INCLUDEDIR@
--- libxslt-1.1.26/libexslt.pc.in.orig	2012-02-28 08:52:18.790178751 -0800
+++ libxslt-1.1.26/libexslt.pc.in	2012-02-28 08:53:21.141163457 -0800
@@ -8,5 +8,5 @@
 Version: @LIBEXSLT_VERSION@
 Description: EXSLT Extension library
 Requires: libxml-2.0
-Libs: @EXSLT_LIBDIR@ @EXSLT_LIBS@
+Libs: -L${libdir} -lexslt
 Cflags: @EXSLT_INCLUDEDIR@
