diff -uNr shairplay/shairplay.pc shairplay2/shairplay.pc
--- shairplay/shairplay.pc  1970-01-01 08:00:00.000000000 +0800
+++ shairplay2/shairplay.pc 2016-01-15 00:54:51.773077123 +0800
@@ -0,0 +1,10 @@
+prefix=/usr/
+exec_prefix=${prefix}
+libdir=%LIBDIR%
+includedir=${prefix}/include
+
+Name: shairplay
+Description: Apple airplay and raop protocol server
+Version: %VERSION%
+Libs: -L${libdir} -lshairplay
+Cflags: -I${includedir} 
