--- wget-1.12/doc/sample.wgetrc.orig	Mon Sep 21 19:53:58 2009
+++ wget-1.12/doc/sample.wgetrc	Wed Sep 30 07:42:08 2009
@@ -7,7 +7,7 @@
 ## not contain a comprehensive list of commands -- look at the manual
 ## to find out what you can put into this file.
 ##
-## Wget initialization file can reside in /usr/local/etc/wgetrc
+## Wget initialization file can reside in /etc/wgetrc
 ## (global, for all users) or $HOME/.wgetrc (for a single user).
 ##
 ## To use the settings in this file, you will have to uncomment them,
@@ -16,7 +16,7 @@
 
 
 ##
-## Global settings (useful for setting up in /usr/local/etc/wgetrc).
+## Global settings (useful for setting up in /etc/wgetrc).
 ## Think well before you change them, since they may reduce wget's
 ## functionality, and make it behave contrary to the documentation:
 ##
--- wget-1.12/doc/wget.texi.orig	Fri Feb  4 07:28:46 2011
+++ wget-1.12/doc/wget.texi	Fri Feb  4 07:31:24 2011
@@ -190,12 +190,12 @@
 Most of the features are fully configurable, either through command line
 options, or via the initialization file @file{.wgetrc} (@pxref{Startup
 File}).  Wget allows you to define @dfn{global} startup files
-(@file{/usr/local/etc/wgetrc} by default) for site settings.
+(@file{/etc/wgetrc} by default) for site settings.
 
 @ignore
 @c man begin FILES
 @table @samp
-@item /usr/local/etc/wgetrc
+@item /etc/wgetrc
 Default location of the @dfn{global} startup file.
 
 @item .wgetrc
