diff -Nru bind-9.16.33.orig/bin/python/dnssec-checkds.py.in bind-9.16.33/bin/python/dnssec-checkds.py.in
--- bind-9.16.33.orig/bin/python/dnssec-checkds.py.in	2022-09-08 15:01:23.000000000 +0000
+++ bind-9.16.33/bin/python/dnssec-checkds.py.in	2022-09-28 11:10:25.444562876 +0000
@@ -22,7 +22,7 @@
         sys.path.insert(
             1,
             os.path.join(
-                "@prefix@", "lib", "python" + sys.version[:3], "site-packages"
+                "@prefix@", "lib", "python" + sys.version[:3], "vendor-packages"
             ),
         )
 
diff -Nru bind-9.16.33.orig/bin/python/dnssec-coverage.py.in bind-9.16.33/bin/python/dnssec-coverage.py.in
--- bind-9.16.33.orig/bin/python/dnssec-coverage.py.in	2022-09-08 15:01:23.000000000 +0000
+++ bind-9.16.33/bin/python/dnssec-coverage.py.in	2022-09-28 11:10:33.515339741 +0000
@@ -22,7 +22,7 @@
         sys.path.insert(
             1,
             os.path.join(
-                "@prefix@", "lib", "python" + sys.version[:3], "site-packages"
+                "@prefix@", "lib", "python" + sys.version[:3], "vendor-packages"
             ),
         )
 
diff -Nru bind-9.16.33.orig/bin/python/dnssec-keymgr.py.in bind-9.16.33/bin/python/dnssec-keymgr.py.in
--- bind-9.16.33.orig/bin/python/dnssec-keymgr.py.in	2022-09-08 15:01:23.000000000 +0000
+++ bind-9.16.33/bin/python/dnssec-keymgr.py.in	2022-09-28 11:10:42.164733437 +0000
@@ -22,7 +22,7 @@
         sys.path.insert(
             1,
             os.path.join(
-                "@prefix@", "lib", "python" + sys.version[:3], "site-packages"
+                "@prefix@", "lib", "python" + sys.version[:3], "vendor-packages"
             ),
         )
 
