--- bind-9.20.26/bin/tests/system/ifconfig.sh.in.orig	2026-07-22 22:25:59.924671233 +0200
+++ bind-9.20.26/bin/tests/system/ifconfig.sh.in	2026-07-22 22:26:30.890141633 +0200
@@ -71,7 +71,7 @@
     *-*-solaris2.1[1-9])
       [ "$a" ] && {
         /sbin/ipadm create-addr -t -T static \
-          -a $a lo0/bind9v4$int \
+          -a $a/32 lo0/bind9v4$int \
           || echo failed lo0/bind9v4$int
       }
       [ "$aaaa" ] && {
