--- bind-9.16.10/config.threads.in.orig	2020-12-07 09:16:53.000000000 +0000
+++ bind-9.16.10/config.threads.in	2020-12-18 12:54:21.544651854 +0000
@@ -88,7 +88,7 @@
 	# Search for / configure pthreads in a system-dependent fashion.
 	#
 	case "$host" in
-		*-freebsd*)
+		*-freebsd* | *-solaris*)
 			# We don't want to set -lpthread as that break
 			# the ability to choose threads library at final
 			# link time and is not valid for all architectures.
