--- ./Makefile.system.orig	2015-02-09 16:28:21.406004454 +0000
+++ ./Makefile.system	2015-02-09 16:31:16.427027007 +0000
@@ -385,8 +385,7 @@
 
 ifeq ($(DYNAMIC_ARCH), 1)
 ifeq ($(ARCH), x86)
-DYNAMIC_CORE = KATMAI COPPERMINE NORTHWOOD PRESCOTT BANIAS \
-	       CORE2 PENRYN DUNNINGTON NEHALEM ATHLON OPTERON OPTERON_SSE3 BARCELONA BOBCAT ATOM NANO
+DYNAMIC_CORE = PRESCOTT CORE2 PENRYN DUNNINGTON NEHALEM ATHLON OPTERON OPTERON_SSE3 BARCELONA BOBCAT ATOM NANO 
 endif
 
 ifeq ($(ARCH), x86_64)
diff -Nru OpenBLAS-0.3.21.orig/driver/others/dynamic.c OpenBLAS-0.3.21/driver/others/dynamic.c
--- OpenBLAS-0.3.21.orig/driver/others/dynamic.c	2022-08-07 22:36:26.000000000 +0000
+++ OpenBLAS-0.3.21/driver/others/dynamic.c	2022-10-02 10:12:23.036339663 +0000
@@ -43,6 +43,8 @@
 #define strcasecmp _stricmp
 #endif
 
+// Disable 32bit-only architectures
+#undef ARCH_X86
 #ifdef ARCH_X86
 #define EXTERN extern
 #else