--- ./driver/others/memory.c.orig	2015-12-14 13:15:29.425758330 +0100
+++ ./driver/others/memory.c	2015-12-14 13:17:50.242607056 +0100
@@ -142,7 +142,7 @@
 #if defined(_MSC_VER) && !defined(__clang__)
 #define CONSTRUCTOR __cdecl
 #define DESTRUCTOR __cdecl
-#elif defined(OS_DARWIN) && defined(C_GCC)
+#elif (defined(OS_DARWIN)||defined(OS_SUNOS)) && defined(C_GCC)
 #define CONSTRUCTOR	__attribute__ ((constructor))
 #define DESTRUCTOR	__attribute__ ((destructor))
 #else
