--- gftp-2.0.18/lib/gftp.h.orig	2008-11-21 15:22:37.740720956 +0800
+++ gftp-2.0.18/lib/gftp.h	2008-11-21 15:22:56.292234508 +0800
@@ -198,7 +198,7 @@
 #define GFTP_ETIMEDOUT		-5		/* Connected timed out */
 
 /* Some general settings */
-#define BASE_CONF_DIR		"~/.gftp"
+#define BASE_CONF_DIR		"~/.gnome2/gftp"
 #define CONFIG_FILE		BASE_CONF_DIR "/gftprc"
 #define BOOKMARKS_FILE		BASE_CONF_DIR "/bookmarks"
 #define LOG_FILE		BASE_CONF_DIR "/gftp.log"
