--- gftp-2.9.1b/src/text/gftp-text.c.~1~	2022-02-15 01:37:55.000000000 -0500
+++ gftp-2.9.1b/src/text/gftp-text.c	2024-07-01 22:45:13.402115453 -0400
@@ -61,13 +61,16 @@
       savechar = *endpos;
       *endpos = '\0';
 
+#if 0
       if (strlen (stpos) <= sw)
         {
+#endif
           printf ("%s%c", stpos, savechar);
           *endpos = savechar;
           if (savechar == '\0')
             break;
           stpos = endpos + 1;
+#if 0
         }
       else
         {
@@ -82,6 +85,7 @@
           printf ("%s\n", stpos);
           stpos = endpos + 1;
         }
+#endif
     }
   while (stpos != endpos);
 
