--- emacs-30.1/configure.ac.orig	2025-02-20 00:46:53.000000000 +0100
+++ emacs-30.1/configure.ac	2025-05-20 21:54:34.769059898 +0200
@@ -4458,7 +4458,7 @@
 if test "$with_xwidgets" != "no"; then
   if test "$USE_GTK_TOOLKIT" = "GTK3" && test "$window_system" != "none"; then
     WEBKIT_REQUIRED=2.12
-    WEBKIT_BROKEN=2.41.92
+    WEBKIT_BROKEN=2.49.92
     WEBKIT_MODULES="webkit2gtk-4.1 >= $WEBKIT_REQUIRED webkit2gtk-4.1 < $WEBKIT_BROKEN"
     EMACS_CHECK_MODULES([WEBKIT], [$WEBKIT_MODULES])
     if test "$HAVE_WEBKIT" = "no"; then
