Fixes error: missing sentinel in function call [-Werror=format=]
--- logrotate-r3-9-1/logrotate.c.~1~	2015-09-21 17:23:10.385648887 +0300
+++ logrotate-r3-9-1/logrotate.c	2015-09-21 17:25:03.727947187 +0300
@@ -356,7 +356,7 @@
 				DOEXIT(1);
 			}
 		}
-		execl("/bin/sh", "sh", "-c", script, "logrotate_script", logfn, NULL);
+		execl("/bin/sh", "sh", "-c", script, "logrotate_script", logfn, (char *) NULL);
 		DOEXIT(1);
 	}
 
@@ -671,7 +671,7 @@
 				DOEXIT(1);
 			}
 
-			execlp(uncompressCommand, uncompressCommand, NULL);
+			execlp(uncompressCommand, uncompressCommand, (char *) NULL);
 			DOEXIT(1);
 		}
 
