diff -urN hexchat-2.10.2.orig/src/common/util.c hexchat-2.10.2/src/common/util.c
--- hexchat-2.10.2.orig/src/common/util.c	2014-11-25 18:42:44.000000000 +0100
+++ hexchat-2.10.2/src/common/util.c	2015-12-13 20:10:54.902023916 +0100
@@ -70,6 +70,10 @@
 #endif
 #endif
 
+#if defined (__SVR4) && defined (__sun)
+#include <kstat.h>
+#endif
+
 #ifndef HAVE_SNPRINTF
 #define snprintf g_snprintf
 #endif
@@ -554,7 +558,7 @@
 	return len;
 }
 
-#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__)
+#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__) || (defined (__SVR4) && defined (__sun))
 
 static void
 get_cpu_info (double *mhz, int *cpus)
@@ -644,6 +648,37 @@
 	*mhz = (freq / 1000000);
 
 #endif
+#if defined (__SVR4) && defined (__sun)
+
+	kstat_t       *ksp;
+	kstat_ctl_t   *kc;
+	kstat_named_t *knp = NULL;
+	int	      numcpu = 0;
+	kc = kstat_open();
+	if (kc == NULL) {
+		fprintf (stderr, "get speed: kstat error %d\n", errno);
+		return;
+	}
+
+	ksp = kstat_lookup(kc, "cpu_info", -1, NULL);
+	for (; ksp != NULL; ksp = ksp->ks_next) {
+                if (strcmp(ksp->ks_module, "cpu_info"))
+                        continue;
+	        numcpu++;
+
+	        kstat_read(kc, ksp, NULL);
+
+		if (knp == NULL) {
+		        knp = kstat_data_lookup(ksp, "clock_MHz");
+		}
+	}
+
+	*cpus = numcpu;
+	if (knp) {
+		*mhz = knp->value.i32;
+	}
+
+#endif
 
 }
 #endif
@@ -794,7 +829,7 @@
 char *
 get_sys_str (int with_cpu)
 {
-#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__)
+#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__) || (defined (__SVR4) && defined (__sun))
 	double mhz;
 #endif
 	int cpus = 1;
@@ -808,7 +843,7 @@
 
 	uname (&un);
 
-#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__)
+#if defined (USING_LINUX) || defined (USING_FREEBSD) || defined (__APPLE__) || (defined (__SVR4) && defined (__sun))
 	get_cpu_info (&mhz, &cpus);
 	if (mhz && with_cpu)
 	{
diff -urN hexchat-2.10.2.orig/src/fe-gtk/Makefile.in hexchat-2.10.2/src/fe-gtk/Makefile.in
--- hexchat-2.10.2.orig/src/fe-gtk/Makefile.in	2015-12-13 20:10:34.366521832 +0100
+++ hexchat-2.10.2/src/fe-gtk/Makefile.in	2015-12-13 20:12:30.308832191 +0100
@@ -487,7 +487,7 @@
 
 hexchat$(EXEEXT): $(hexchat_OBJECTS) $(hexchat_DEPENDENCIES) $(EXTRA_hexchat_DEPENDENCIES) 
 	@rm -f hexchat$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(hexchat_OBJECTS) $(hexchat_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(LINK) $(hexchat_OBJECTS) $(hexchat_LDADD) $(LIBS) -lkstat
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
