diff -urN hexchat-2.10.2.orig/src/fe-gtk/Makefile.in hexchat-2.10.2/src/fe-gtk/Makefile.in
--- hexchat-2.10.2.orig/src/fe-gtk/Makefile.in	2014-11-25 19:04:10.000000000 +0100
+++ hexchat-2.10.2/src/fe-gtk/Makefile.in	2015-12-13 01:15:00.841337712 +0100
@@ -765,8 +765,8 @@
 	tags tags-am uninstall uninstall-am uninstall-binPROGRAMS
 
 
-resources.c: ../../data/hexchat.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=../../data --generate-dependencies ../../data/hexchat.gresource.xml)
-	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=../../data --generate-source $<
+resources.c: ../../data/hexchat.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(top_srcdir)/data --generate-dependencies $(top_srcdir)/data/hexchat.gresource.xml)
+	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(top_srcdir)/data --generate-source $(top_srcdir)/data/hexchat.gresource.xml
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
