diff -urN hexchat-2.12.0.orig/src/fe-gtk/Makefile.in hexchat-2.12.0/src/fe-gtk/Makefile.in
--- hexchat-2.12.0.orig/src/fe-gtk/Makefile.in	2016-03-12 09:05:22.000000000 +0100
+++ hexchat-2.12.0/src/fe-gtk/Makefile.in	2016-04-04 00:16:13.688933195 +0200
@@ -577,7 +577,7 @@
 
 hexchat$(EXEEXT): $(hexchat_OBJECTS) $(hexchat_DEPENDENCIES) $(EXTRA_hexchat_DEPENDENCIES) 
 	@rm -f hexchat$(EXEEXT)
-	$(AM_V_OBJCLD)$(hexchat_LINK) $(hexchat_OBJECTS) $(hexchat_LDADD) $(LIBS) -lkstat
+	$(AM_V_OBJCLD)$(hexchat_LINK) $(hexchat_OBJECTS) $(hexchat_LDADD) $(LIBS) -lkstat -lssp -lssp_nonshared
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
