--- hexchat-2.14.3/plugins/sysinfo/meson.build	2019-12-21 07:43:47.652401700 +0000
+++ hexchat-2.14.3/plugins/sysinfo/meson.build.new	2020-01-02 12:27:17.764611458 +0000
@@ -47,7 +47,7 @@ elif system == 'windows'
     '../../src/common/sysinfo/win32/backend.c'
   ]
 else
-  error('sysinfo: Unknown system?')
+  warning('sysinfo: Unknown system?')
 endif
 
 shared_module('sysinfo', sysinfo_sources,
