--- ./examples/ExampleBrowser/GwenGUISupport/GwenParameterInterface.cpp.orig	2015-07-14 23:17:44.656776490 +0000
+++ ./examples/ExampleBrowser/GwenGUISupport/GwenParameterInterface.cpp	2015-07-14 23:18:09.472151012 +0000
@@ -1,7 +1,7 @@
 #include "GwenParameterInterface.h"
 #include "gwenInternalData.h"
 
-
+#include <string.h>
 
 template<typename T>
 struct MySliderEventHandler : public Gwen::Event::Handler
@@ -164,4 +164,4 @@
 	m_paramInternalData->m_textLabels.clear();
 	
 	m_gwenInternalData->m_curYposition = this->m_paramInternalData->m_savedYposition;
-}
\ No newline at end of file
+}
