--- gst-libav-1.10.4/configure.ac.1	2017-03-16 23:37:02.319114126 +0300
+++ gst-libav-1.10.4/configure.ac	2017-03-16 23:38:04.936112390 +0300
@@ -369,7 +369,8 @@
         --disable-audiotoolbox --disable-videotoolbox --disable-vaapi --disable-crystalhd \
         --disable-mediacodec --disable-nvenc --disable-mmal --disable-omx \
         --disable-omx-rpi --disable-cuda --disable-cuvid --disable-libmfx \
-        --disable-libnpp --disable-iconv --disable-jni --enable-optimizations"
+        --disable-libnpp --disable-iconv --disable-jni --enable-optimizations \
+	--disable-yasm --disable-asm"
 
   if test "x$enable_gpl" == "xyes"; then
     emblibav_configure_args="$emblibav_configure_args --enable-gpl"
