--- pyftpdlib-1.5.7.orig/pyftpdlib/test/test_authorizers.py
+++ pyftpdlib-1.5.7/pyftpdlib/test/test_authorizers.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 
 # Copyright (C) 2007 Giampaolo Rodola' <g.rodola@gmail.com>.
 # Use of this source code is governed by MIT license that can be
--- pyftpdlib-1.5.7.orig/pyftpdlib/test/test_functional.py
+++ pyftpdlib-1.5.7/pyftpdlib/test/test_functional.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 
 # Copyright (C) 2007 Giampaolo Rodola' <g.rodola@gmail.com>.
 # Use of this source code is governed by MIT license that can be
--- pyftpdlib-1.5.7.orig/pyftpdlib/test/test_misc.py
+++ pyftpdlib-1.5.7/pyftpdlib/test/test_misc.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 
 # Copyright (C) 2007 Giampaolo Rodola' <g.rodola@gmail.com>.
 # Use of this source code is governed by MIT license that can be
