--- fcron-3.3.0/Makefile.in	Sun Jun 26 19:56:27 2016
+++ p0/fcron-3.3.0/Makefile.in	Sun Apr 25 13:05:14 2021
@@ -152,7 +152,7 @@
 # needed by boot-install script :
 	echo `fcron -V 2>&1 | grep "^fcron "` > PREVIOUS_VERSION && chmod a+rw PREVIOUS_VERSION
 
-	$(SRCDIR)/script/user-group $(USERNAME) $(GROUPNAME) $(ANSWERALL) $(SRCDIR)
+#	$(SRCDIR)/script/user-group $(USERNAME) $(GROUPNAME) $(ANSWERALL) $(SRCDIR)
 
 # check if the directories we use exist, and if not, create them
 	if test ! -d $(DESTDIR)$(DESTSBIN); then $(INSTALL) -m 755 -d $(DESTDIR)$(DESTSBIN) ; fi
