--- apache-couchdb-2.1.0/rel/overlay/bin/couchup.orig	2017-10-07 13:09:47.591823720 +0000
+++ apache-couchdb-2.1.0/rel/overlay/bin/couchup	2017-10-07 13:11:25.777635346 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/amd64/python2.7
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
 # the License at
