--- apache-couchdb-2.1.0/rel/overlay/bin/couchdb.~1~	2017-07-31 00:11:20.000000000 +0000
+++ apache-couchdb-2.1.0/rel/overlay/bin/couchdb	2017-10-12 07:08:33.840286976 +0000
@@ -12,7 +12,7 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
-COUCHDB_BIN_DIR=$(cd "${0%/*}" && pwd)
+COUCHDB_BIN_DIR=/usr/lib/amd64/couchdb/2.1/bin
 ERTS_BIN_DIR=$COUCHDB_BIN_DIR/../
 cd "$COUCHDB_BIN_DIR/../"
