diff -ur apache-couchdb-2.1.1-1/build-aux/logfile-uploader.py apache-couchdb-2.1.1/build-aux/logfile-uploader.py
--- apache-couchdb-2.1.1-1/build-aux/logfile-uploader.py	2020-08-17 10:21:11.641904589 +0000
+++ apache-couchdb-2.1.1/build-aux/logfile-uploader.py	2020-08-17 10:22:02.702079770 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3.5
 #
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
diff -ur apache-couchdb-2.1.1-1/dev/run apache-couchdb-2.1.1/dev/run
--- apache-couchdb-2.1.1-1/dev/run	2020-08-17 10:21:11.727788113 +0000
+++ apache-couchdb-2.1.1/dev/run	2020-08-17 10:22:02.701665274 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3.5
 #
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
diff -ur apache-couchdb-2.1.1-1/rel/overlay/bin/couchup apache-couchdb-2.1.1/rel/overlay/bin/couchup
--- apache-couchdb-2.1.1-1/rel/overlay/bin/couchup	2020-08-17 10:21:11.643965013 +0000
+++ apache-couchdb-2.1.1/rel/overlay/bin/couchup	2020-08-17 10:22:02.700659828 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3.5
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
 # the License at
diff -ur apache-couchdb-2.1.1-1/src/rebar/pr2relnotes.py apache-couchdb-2.1.1/src/rebar/pr2relnotes.py
--- apache-couchdb-2.1.1-1/src/rebar/pr2relnotes.py	2020-08-17 10:21:11.412800715 +0000
+++ apache-couchdb-2.1.1/src/rebar/pr2relnotes.py	2020-08-17 10:22:02.702793000 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3.5
 
 ## Install info
 ##   $ virtualenv env
diff -ur apache-couchdb-2.1.1-1/test/javascript/run apache-couchdb-2.1.1/test/javascript/run
--- apache-couchdb-2.1.1-1/test/javascript/run	2020-08-17 10:21:11.747173706 +0000
+++ apache-couchdb-2.1.1/test/javascript/run	2020-08-17 10:22:02.701133363 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3.5
 #
 # Licensed under the Apache License, Version 2.0 (the "License"); you may not
 # use this file except in compliance with the License. You may obtain a copy of
