--- opendbx-1.4.6/test/Makefile.in	2012-06-10 23:33:53.000000000 +0000
+++ opendbx-1.4.6/test/Makefile.in.new	2022-01-16 19:35:00.347203649 +0000
@@ -72,7 +72,7 @@
 odbxtest_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(odbxtest_LDFLAGS) \
 	$(LDFLAGS) -o $@
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_srcdir)/lib
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
