--- opendbx-1.4.6/utils/Makefile.in	2012-06-10 23:33:53.000000000 +0000
+++ opendbx-1.4.6/utils/Makefile.in.new	2022-01-16 19:37:07.132537362 +0000
@@ -67,7 +67,7 @@
 @ENABLEUTILS_TRUE@odbx_sql_DEPENDENCIES = ../lib/.libs/$(LIBPREFIX)opendbx$(LIBSUFFIX) \
 @ENABLEUTILS_TRUE@	../lib/.libs/$(LIBPREFIX)opendbxplus$(LIBSUFFIX) \
 @ENABLEUTILS_TRUE@	$(am__DEPENDENCIES_1)
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_srcdir)/lib
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
