--- opendbx-1.4.6/backends/pgsql/Makefile.in	2012-06-10 23:33:53.000000000 +0000
+++ opendbx-1.4.6/backends/pgsql/Makefile.in.new	2022-01-16 20:16:02.781680577 +0000
@@ -83,7 +83,7 @@
 	$(libpgsqlbackend_la_LDFLAGS) $(LDFLAGS) -o $@
 @SINGLELIB_FALSE@am_libpgsqlbackend_la_rpath = -rpath $(pkglibdir)
 @SINGLELIB_TRUE@am_libpgsqlbackend_la_rpath =
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_srcdir)/lib
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
