--- opendbx-1.4.6/backends/sqlite3/Makefile.in	2012-06-10 23:33:53.000000000 +0000
+++ opendbx-1.4.6/backends/sqlite3/Makefile.in.new	2022-01-16 20:17:37.404544035 +0000
@@ -84,7 +84,7 @@
 	$(libsqlite3backend_la_LDFLAGS) $(LDFLAGS) -o $@
 @SINGLELIB_FALSE@am_libsqlite3backend_la_rpath = -rpath $(pkglibdir)
 @SINGLELIB_TRUE@am_libsqlite3backend_la_rpath =
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_srcdir)/lib
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
