diff -urN not.orig/nautilus-open-terminal.schemas.in not.new/nautilus-open-terminal.schemas.in
--- not.orig/nautilus-open-terminal.schemas.in	2009-08-24 14:33:53.146276561 +0100
+++ not.new/nautilus-open-terminal.schemas.in	2009-08-24 14:36:58.443201447 +0100
@@ -20,7 +20,7 @@
       <applyto>/apps/nautilus-open-terminal/display_mc_item</applyto>
       <owner>nautilus-open-terminal</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Whether a Midnight Commander menu item should be displayed</short>
          <long>
