--- /usr/tmp/clean/nautilus-open-terminal-0.9/nautilus-open-terminal.schemas.in	2007-08-23 09:51:58.000000000 +0100
+++ nautilus-open-terminal-0.9/nautilus-open-terminal.schemas.in	2008-08-08 10:47:18.377501000 +0100
@@ -5,7 +5,7 @@
       <applyto>/apps/nautilus-open-terminal/desktop_opens_home_dir</applyto>
       <owner>nautilus-open-terminal</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>Whether opening a terminal on the desktop opens a terminal in the home directory</short>
          <long>
