Set some reasonable defaults.  Not suitable for upstream.

--- sys/unix/sysconf	2015-12-07 00:26:01.000000000 -0800
+++ sys/unix/sysconf	2016-04-14 18:33:58.537973707 -0700
@@ -28,14 +28,14 @@
 # occasionally get stuck outside the game by accidentally typing '!' or
 # '^Z' during play and not knowing how to go back.)
 # Uses the same syntax as the WIZARDS and EXPLORERS options above.
-#SHELLERS=
+SHELLERS=*
 
 # Limit the number of simultaneous games (see also nethack.sh).
-MAXPLAYERS=10
+MAXPLAYERS=25
 
 # If not null, added to string "To get local support, " in the support
 # information help.
-#SUPPORT=call Izchak at extension 42.
+SUPPORT=call Izchak at extension 42.
 
 # If not null, displayed at the end of a panic-save sequence.
 #RECOVER=Run the recover program.
@@ -51,14 +51,14 @@
 # CAUTION: changing these after people have started playing games can
 #  lead to lost high scores!
 # Maximum entries for one person.
-#PERSMAX=10
+PERSMAX=100
 # Maximum entries in the record file.
-#ENTRYMAX=100
+ENTRYMAX=1000
 # Minimum points to get an entry.
 #POINTSMIN=1
 # Determine identity of "person" in the score file with name (0) or
 # numeric (1) user id.
-#PERS_IS_UID=1
+PERS_IS_UID=0
 
 # Maximum number of score file entries to use for random statue names
 #MAX_STATUENAME_RANK=10
