--- libwps-0.4.6/src/conv/wks2csv/wks2csv.cpp.orig	Sat May  6 18:56:29 2017
+++ libwps-0.4.6/src/conv/wks2csv/wks2csv.cpp	Sat May  6 18:56:41 2017
@@ -34,7 +34,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 using namespace libwps;
 
--- libwps-0.4.6/src/conv/raw/wps2raw.cpp.orig	Sat May  6 18:55:28 2017
+++ libwps-0.4.6/src/conv/raw/wps2raw.cpp	Sat May  6 18:55:41 2017
@@ -30,7 +30,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
--- libwps-0.4.6/src/conv/wks2text/wks2text.cpp.orig	Sat May  6 18:57:19 2017
+++ libwps-0.4.6/src/conv/wks2text/wks2text.cpp	Sat May  6 18:57:31 2017
@@ -33,7 +33,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
--- libwps-0.4.6/src/conv/text/wps2text.cpp.orig	Sat May  6 18:56:03 2017
+++ libwps-0.4.6/src/conv/text/wps2text.cpp	Sat May  6 18:56:16 2017
@@ -29,7 +29,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 using namespace libwps;
 
--- libwps-0.4.6/src/conv/html/wps2html.cpp.orig	Sat May  6 18:53:16 2017
+++ libwps-0.4.6/src/conv/html/wps2html.cpp	Sat May  6 18:54:50 2017
@@ -31,7 +31,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 using namespace libwps;
 
--- libwps-0.4.6/src/conv/wks2raw/wks2raw.cpp.orig	Sat May  6 18:56:55 2017
+++ libwps-0.4.6/src/conv/wks2raw/wks2raw.cpp	Sat May  6 18:57:07 2017
@@ -30,7 +30,7 @@
 
 #include <libwps/libwps.h>
 
-#include "helper.h"
+#include "../helper/helper.h"
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
