--- a/data/gnome-power-manager.schemas.in.org	2007-12-12 18:10:33.063268000 +0800
+++ b/data/gnome-power-manager.schemas.in	2007-12-12 18:12:08.408882000 +0800
@@ -1021,7 +1021,7 @@
       <applyto>/apps/gnome-power-manager/ui/icon_policy</applyto>
       <owner>gnome-power-manager</owner>
       <type>string</type>
-      <default>present</default>
+      <default>always</default>
       <locale name="C">
         <short>When to show the notification icon</short>
         <long>Display options for the notification icon. Valid options are "never", "critical", "charge", "present" and "always".</long>
