--- a/src/gpm-prefs-core.c.orig	2008-06-03 15:51:05.336557000 +0800
+++ b/src/gpm-prefs-core.c	2008-06-03 15:54:06.524979000 +0800
@@ -1081,7 +1081,11 @@
 	prefs->priv->has_lcd = ((caps & GPM_PREFS_SERVER_BACKLIGHT) > 0);
 	prefs->priv->has_ambient = ((caps & GPM_PREFS_SERVER_AMBIENT) > 0);
 	prefs->priv->has_button_lid = ((caps & GPM_PREFS_SERVER_LID) > 0);
+#if defined(sun) && defined(__SVR4)
+	prefs->priv->has_button_suspend = FALSE;
+#else
 	prefs->priv->has_button_suspend = TRUE;
+#endif
 	prefs->priv->can_shutdown = gpm_dbus_method_bool ("CanShutdown");
 	prefs->priv->can_suspend = gpm_dbus_method_bool ("CanSuspend");
 	prefs->priv->can_hibernate = gpm_dbus_method_bool ("CanHibernate");
