--- gnome-power-manager-2.23.3/data/gnome-power-statistics.desktop.in.in.orig	2008-08-26 13:55:38.924206000 +0800
+++ gnome-power-manager-2.23.3/data/gnome-power-statistics.desktop.in.in	2008-08-26 13:56:45.754123000 +0800
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 _Name=Power Statistics
-_Comment=Observe power management
+# SUN_BRANDING
+_Comment=Show graphs of recent battery performance
 Icon=gnome-power-statistics
 Exec=gnome-power-statistics
 Terminal=false
@@ -9,6 +9,7 @@
 Categories=GNOME;GTK;System;X-Red-Hat-Base;
 OnlyShowIn=GNOME;
 StartupNotify=true
+NoDisplay=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-power-manager
 X-GNOME-Bugzilla-Component=gnome-power-statistics
diff -ur gnome-power-manager-2.23.91-orig//data/gnome-power-preferences.desktop.in.in gnome-power-manager-2.23.91//data/gnome-power-preferences.desktop.in.in
--- gnome-power-manager-2.23.91-orig//data/gnome-power-preferences.desktop.in.in	Thu Sep 11 13:53:50 2008
+++ gnome-power-manager-2.23.91//data/gnome-power-preferences.desktop.in.in	Thu Sep 11 13:58:04 2008
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 _Name=Power Management
-_Comment=Configure power management
+# SUN_BRANDING
+_Comment=Optimize your computer's power usage and CPU speed
 Icon=gnome-power-manager
 Exec=gnome-power-preferences
 Terminal=false
