--- gnome-power-manager-2.24.4-orig/src/gpm-profile.h	2009-05-05 11:57:32.751152059 +0800
+++ gnome-power-manager-2.24.4/src/gpm-profile.h	2009-05-05 12:17:23.919643732 +0800
@@ -34,7 +34,7 @@ G_BEGIN_DECLS
 #define GPM_IS_PROFILE_CLASS(k)		(G_TYPE_CHECK_CLASS_TYPE ((k), GPM_TYPE_PROFILE))
 #define GPM_PROFILE_GET_CLASS(o)	(G_TYPE_INSTANCE_GET_CLASS ((o), GPM_TYPE_PROFILE, GpmProfileClass))
 
-#define GPM_PROFILE_GOOD_TRUST	40
+#define GPM_PROFILE_GOOD_TRUST	0
 
 typedef struct GpmProfilePrivate GpmProfilePrivate;
 
