Remove "-Werror" as it breaks compilation with recent gcc
--- gnome-power-manager-2.24.4/configure.ac.1	2015-01-30 12:38:19.585818509 +0300
+++ gnome-power-manager-2.24.4/configure.ac	2015-01-30 12:38:30.758539622 +0300
@@ -22,7 +22,7 @@
 dnl - Extra verbose warning switches
 dnl ---------------------------------------------------------------------------
 if test "$GCC" = "yes"; then
-    CPPFLAGS="$CPPFLAGS -Werror -Wcast-align -Wno-uninitialized"
+    CPPFLAGS="$CPPFLAGS -Wcast-align -Wno-uninitialized"
     CPPFLAGS="$CPPFLAGS -Wall -Wformat-security -Wno-strict-aliasing"
 fi
 
