--- libcouchbase-2.5.7/cmake/Modules/GetVersionInfo.cmake.orig	2016-04-12 16:14:41.685784470 +0300
+++ libcouchbase-2.5.7/cmake/Modules/GetVersionInfo.cmake	2016-04-12 16:14:51.147374852 +0300
@@ -16,7 +16,7 @@
         OUTPUT_STRIP_TRAILING_WHITESPACE)
 ENDMACRO()
 
-if (GIT_EXECUTABLE)
+if (GIT_EXECUTABLE AND FALSE)
     RUNGIT(LCB_REVDESCRIBE describe --long)
     RUNGIT(LCB_VERSION describe)
     STRING(REPLACE "-" "_" LCB_VERSION "${LCB_VERSION}")
