--- libcouchbase-2.5.7/packaging/libcouchbase.pc.in.orig	2016-04-12 20:02:25.787457960 +0300
+++ libcouchbase-2.5.7/packaging/libcouchbase.pc.in	2016-04-12 20:02:53.516700103 +0300
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}/bin
+exec_prefix=${prefix}/@CMAKE_INSTALL_BINDIR@
 includedir=${prefix}/include
-libdir=${prefix}/lib
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
 
 Name: libcouchbase @LCB_VERSION@
 Description: Couchbase client library
