--- build/amd64/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nix-0.30.1/src/sys/resource.rs.~1~	2006-07-23 21:21:28.000000000 -0400
+++ build/amd64/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nix-0.30.1/src/sys/resource.rs	2025-10-09 23:49:44.754622732 -0400
@@ -19,6 +19,7 @@
         bsd,
         target_os = "android",
         target_os = "aix",
+        target_os = "illumos",
         all(target_os = "linux", not(target_env = "gnu")),
         target_os = "cygwin"
     ))]{
@@ -50,6 +51,7 @@
             bsd,
             target_os = "android",
             target_os = "aix",
+            target_os = "illumos",
             all(target_os = "linux", not(any(target_env = "gnu", target_env = "uclibc"))),
             target_os = "cygwin"
         ), repr(i32))]
--- build/amd64/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nix-0.30.1/src/sys/mod.rs.~1~	2006-07-23 21:21:28.000000000 -0400
+++ build/amd64/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/nix-0.30.1/src/sys/mod.rs	2025-10-09 23:55:13.805322108 -0400
@@ -96,7 +96,6 @@
 #[cfg(not(any(
     target_os = "redox",
     target_os = "fuchsia",
-    solarish,
     target_os = "haiku"
 )))]
 feature! {
