Description: Corrected depreciated string exceptions in python 2.6
Origin: Debian
Author: Charlie Smotherman <cjsmo@cableone.net>
Forwarded: to upstreams bug tracker

Index: coherence-0.6.6.3/coherence/extern/galleryremote/gallery.py
===================================================================
--- coherence-0.6.6.3.orig/coherence/extern/galleryremote/gallery.py	2010-06-07 12:10:53.440177834 -0500
+++ coherence-0.6.6.3/coherence/extern/galleryremote/gallery.py	2010-06-07 12:11:39.520707119 -0500
@@ -137,7 +137,7 @@
 
         # make sure the 1st line is #__GR2PROTO__
         if string.find( line, '#__GR2PROTO__' ) == -1:
-            raise "Bad response: \r\n" + response
+            raise Exception("Bad response: \r\n" + response)
 
         resDict = {}
 
