*** doc/bash.1.orig	Wed Oct 12 06:45:05 2011
--- doc/bash.1	Wed Oct 12 06:45:46 2011
***************
*** 3843,3849 ****
  .B \fIexpr\fP?\fIexpr\fP:\fIexpr\fP
  conditional operator
  .TP
! .B = *= /= %= += \-= <<= >>= &= ^= |=
  assignment
  .TP
  .B \fIexpr1\fP , \fIexpr2\fP
--- 3843,3849 ----
  .B \fIexpr\fP?\fIexpr\fP:\fIexpr\fP
  conditional operator
  .TP
! .B "= *= /= %= += \-= <<= >>= &= ^= |="
  assignment
  .TP
  .B \fIexpr1\fP , \fIexpr2\fP
***************
*** 5235,5245 ****
  .IP \fB$include\fP
  This directive takes a single filename as an argument and reads commands
  and bindings from that file.  For example, the following directive
! would read \fI/etc/inputrc\fP:
  .sp 1
  .RS
  .nf
! \fB$include\fP \^ \fI/etc/inputrc\fP
  .fi
  .RE
  .SS Searching
--- 5235,5245 ----
  .IP \fB$include\fP
  This directive takes a single filename as an argument and reads commands
  and bindings from that file.  For example, the following directive
! would read \fI/etc/bash/inputrc\fP:
  .sp 1
  .RS
  .nf
! \fB$include\fP \^ \fI/etc/bash/inputrc\fP
  .fi
  .RE
  .SS Searching
