--- libayatana-appindicator-0.5.94/src/CMakeLists.txt.orig	2025-03-12 09:30:06.000000000 +0100
+++ libayatana-appindicator-0.5.94/src/CMakeLists.txt	2025-03-13 12:39:43.421682452 +0100
@@ -125,7 +125,7 @@
 target_include_directories("${ayatana_appindicator_gtkver}" PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
 target_include_directories("${ayatana_appindicator_gtkver}" PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries("${ayatana_appindicator_gtkver}" ${PROJECT_DEPS_LIBRARIES})
-if(NOT APPLE)
+if(NOT UNIX)
     target_link_options ("${ayatana_appindicator_gtkver}" PRIVATE "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/app-indicator.symbols")
 endif()
 install(TARGETS "${ayatana_appindicator_gtkver}" LIBRARY DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
