--- percona-server-5.6.49-89.0/mysys_ssl/CMakeLists.txt.~1~	2020-08-11 17:57:49.072815600 +0000
+++ percona-server-5.6.49-89.0/mysys_ssl/CMakeLists.txt	2020-08-11 17:58:28.742866772 +0000
@@ -48,7 +48,8 @@
 IF(WIN32)
   TARGET_LINK_LIBRARIES(mysys_ssl crypt32)
 ENDIF()
-DTRACE_INSTRUMENT(mysys_ssl)
+# Percona doesn't provide dtrace probes in mysys_ssl
+# DTRACE_INSTRUMENT(mysys_ssl)
 
 IF(MSVC)
    INSTALL_DEBUG_TARGET(mysys_ssl DESTINATION ${INSTALL_LIBDIR}/debug)
