https://bugs.launchpad.net/percona-server/+bug/1329986

--- percona-server-5.6.28-76.1/plugin/HandlerSocket-Plugin-for-MySQL/handlersocket/hstcpsvr_worker.cpp.1	2016-01-21 14:28:14.307404841 +0300
+++ percona-server-5.6.28-76.1/plugin/HandlerSocket-Plugin-for-MySQL/handlersocket/hstcpsvr_worker.cpp	2016-01-21 14:28:29.136108977 +0300
@@ -14,6 +14,7 @@
 #include <stdexcept>
 #include <signal.h>
 #include <list>
+#include <alloca.h>
 #if __linux__
 #include <sys/epoll.h>
 #endif
