We ship event_*-2.0 libraries.

--- percona-server-5.6.44-85.0/cmake/libevent.cmake	2019-05-18 10:34:52.000000000 +0000
+++ percona-server-5.6.44-85.0/cmake/libevent.cmake	2019-05-20 14:08:07.971557193 +0000
@@ -60,9 +60,9 @@ MACRO (MYSQL_CHECK_LIBEVENT)
     ENDIF()
 
     ## libevent.so is historical, use libevent_core.so if found.
-    FIND_LIBRARY(LIBEVENT_CORE event_core PATHS ${LIBEVENT_LIB_PATHS})
-    FIND_LIBRARY(LIBEVENT_EXTRA event_extra PATHS ${LIBEVENT_LIB_PATHS})
-    FIND_LIBRARY(LIBEVENT_LIB event PATHS ${LIBEVENT_LIB_PATHS})
+    FIND_LIBRARY(LIBEVENT_CORE event_core-2.0 PATHS ${LIBEVENT_LIB_PATHS})
+    FIND_LIBRARY(LIBEVENT_EXTRA event_extra-2.0 PATHS ${LIBEVENT_LIB_PATHS})
+    FIND_LIBRARY(LIBEVENT_LIB event-2.0 PATHS ${LIBEVENT_LIB_PATHS})
 
     IF (NOT LIBEVENT_LIB AND NOT LIBEVENT_CORE)
         MESSAGE(SEND_ERROR "Cannot find appropriate event lib in /usr/local/lib or /opt/local/lib. Use bundled libevent")
