--- gthumb-2.11.1/data/gthumb.desktop.in.in.old	2010-01-25 16:19:25.699198591 +0800
+++ gthumb-2.11.1/data/gthumb.desktop.in.in	2010-01-25 16:35:19.456397533 +0800
@@ -1,8 +1,9 @@
 [Desktop Entry]
-_Name=gThumb
-_GenericName=Image Viewer
-_X-GNOME-FullName=gThumb Image Viewer
-_Comment=View and organize your images
+#SUN_BRANDING
+_Name=Image Organizer
+_GenericName=gthumb Image Organizer
+_X-GNOME-FullName=gThumb Image Organizer
+_Comment=Arrange, tag and search your image library
 Categories=GNOME;GTK;Graphics;Viewer;RasterGraphics;2DGraphics;Photography;
 Exec=gthumb %U
 Icon=gthumb
