--- gthumb-2.14.4/configure.ac	2014-04-11 13:47:04.495233208 +0200
+++ gthumb-2.14.4/configure.ac	2014-04-11 13:46:27.355254331 +0200
@@ -63,7 +63,7 @@
 dnl ===========================================================================
 
 CFLAGS=${CFLAGS-:""}
-CFLAGS="$CFLAGS -fPIC -DPIC -Wl,--as-needed"
+CFLAGS="$CFLAGS -fPIC -DPIC"
 
 AC_ARG_ENABLE(deprecated,
 	      AS_HELP_STRING([--disable-deprecated], [do not use deprecated API]),
@@ -139,7 +139,6 @@
 	gtk+-2.0 		>= $GTK_REQUIRED
 	gconf-2.0 		>= $GCONF_REQUIRED
 	libpng
-	zlib
         ice
 ])
 AC_SUBST(GTHUMB_LIBS)
