lockfile is not needed to run tests.

--- apeye-1.4.1/pyproject.toml.orig
+++ apeye-1.4.1/pyproject.toml
@@ -46,7 +46,7 @@
 Documentation = "https://apeye.readthedocs.io/en/latest"
 
 [project.optional-dependencies]
-limiter = [ "cachecontrol[filecache]>=0.12.6", "lockfile>=0.12.2",]
+limiter = [ "cachecontrol[filecache]>=0.12.6",]
 all = [ "cachecontrol[filecache]>=0.12.6", "lockfile>=0.12.2",]
 
 [tool.mkrecipe]
