--- apeye-1.4.1/tox.ini.orig
+++ apeye-1.4.1/tox.ini
@@ -238,13 +238,7 @@
 deps = -r{toxinidir}/tests/requirements.txt
 extras = limiter
 commands =
-    python --version
-    python -m pip install "pyyaml>=5.3.1" ruamel.yaml
     python -m pytest --cov=apeye -r aR tests/ {posargs}
-    python -m pip uninstall pyyaml -y
-    python -m pytest --cov=apeye -r aR tests/ --cov-append {posargs}
-    python -m pip uninstall ruamel.yaml -y
-    python -m pytest --cov=apeye -r aR tests/ --cov-append {posargs}
 
 [dep_checker]
 allowed_unused = lockfile
