diff -ru ChezScheme-9.5.2/c/Mf-a6s2 /builds/richlowe/ChezScheme-9.5.2/c/Mf-a6s2
--- ChezScheme-9.5.2/c/Mf-a6s2	2019-03-21 22:05:24.000000000 +0000
+++ /builds/richlowe/ChezScheme-9.5.2/c/Mf-a6s2	2019-08-06 00:16:29.279014957 +0000
@@ -16,7 +16,7 @@
 m = a6s2
 Cpu = X86_64
 
-mdclib = -lnsl -ldl -lm -lcurses -lrt
+mdclib = -lnsl -ldl -lm -lcurses -lrt -luuid
 C = ${CC} ${CPPFLAGS} -m64 -Wpointer-arith -Wextra -Werror -O ${CFLAGS}
 o = o
 mdsrc = i3le.c
--- ChezScheme-9.5.2/c/Mf-ta6s2	2019-03-21 22:05:24.000000000 +0000
+++ /builds/richlowe/ChezScheme-9.5.2/c/Mf-ta6s2	2019-08-06 00:16:36.818329868 +0000
@@ -16,7 +16,7 @@
 m = ta6s2
 Cpu = X86_64
 
-mdclib = -lnsl -ldl -lm -lpthread -lcurses -lrt
+mdclib = -lnsl -ldl -lm -lpthread -lcurses -lrt -luuid
 C = ${CC} ${CPPFLAGS} -m64 -Wpointer-arith -Wextra -Werror -O2 -D_REENTRANT ${CFLAGS}
 o = o
 mdsrc = i3le.c
diff -ru ChezScheme-9.5.2.orig/c/Mf-i3s2 ChezScheme-9.5.2/c/Mf-i3s2
--- ChezScheme-9.5.2.orig/c/Mf-i3s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-i3s2	2019-08-25 16:24:13.152327983 +0000
@@ -16,7 +16,7 @@
 m = i3s2
 Cpu = I386
 
-mdclib = -lnsl -ldl -lm -lcurses -lrt
+mdclib = -lnsl -ldl -lm -lcurses -lrt -luuid
 C = ${CC} ${CFLAGS} -m32 -Wpointer-arith -Wextra -Werror -O ${CPPFLAGS}
 o = o
 mdsrc = i3le.c
diff -ru ChezScheme-9.5.2.orig/c/Mf-ti3s2 ChezScheme-9.5.2/c/Mf-ti3s2
--- ChezScheme-9.5.2.orig/c/Mf-ti3s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-ti3s2	2019-08-25 16:24:36.560269634 +0000
@@ -16,7 +16,7 @@
 m = ti3s2
 Cpu = I386
 
-mdclib = -lnsl -ldl -lm -lpthread -lcurses -lrt
+mdclib = -lnsl -ldl -lm -lpthread -lcurses -lrt -luuid
 C = ${CC} ${CPPFLAGS} -m32 -Wpointer-arith -Wextra -Werror -O2 -D_REENTRANT ${CFLAGS}
 o = o
 mdsrc = i3le.c
