diff -ru ChezScheme-9.5.2.orig/c/Mf-a6s2 ChezScheme-9.5.2/c/Mf-a6s2
--- ChezScheme-9.5.2.orig/c/Mf-a6s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-a6s2	2019-08-25 16:30:45.196593078 +0000
@@ -31,7 +31,7 @@
 include Mf-base
 
 ${Kernel}: ${kernelobj} ../zlib/libz.a
-	ld -melf_x86_64 -r -X -o ${Kernel} ${kernelobj} ../zlib/libz.a
+	/bin/ld -r -o ${Kernel} ${kernelobj} ../zlib/libz.a
 
 ${Scheme}: ${Kernel} ${Main}
 	$C -o ${Scheme} ${Kernel} ${Main} ${mdclib} ${LDFLAGS}
diff -ru ChezScheme-9.5.2.orig/c/Mf-i3s2 ChezScheme-9.5.2/c/Mf-i3s2
--- ChezScheme-9.5.2.orig/c/Mf-i3s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-i3s2	2019-08-25 16:31:04.882010980 +0000
@@ -31,7 +31,7 @@
 include Mf-base
 
 ${Kernel}: ${kernelobj} ../zlib/libz.a
-	ld -melf_i386 -r -X -o ${Kernel} ${kernelobj} ../zlib/libz.a
+	/bin/ld -r -o ${Kernel} ${kernelobj} ../zlib/libz.a
 
 ${Scheme}: ${Kernel} ${Main}
 	$C -o ${Scheme} ${Kernel} ${Main} ${mdclib} ${LDFLAGS}
diff -ru ChezScheme-9.5.2.orig/c/Mf-ta6s2 ChezScheme-9.5.2/c/Mf-ta6s2
--- ChezScheme-9.5.2.orig/c/Mf-ta6s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-ta6s2	2019-08-25 16:30:45.196775283 +0000
@@ -31,7 +31,7 @@
 include Mf-base
 
 ${Kernel}: ${kernelobj} ../zlib/libz.a
-	ld -melf_x86_64 -r -X -o ${Kernel} ${kernelobj} ../zlib/libz.a
+	/bin/ld -r -o ${Kernel} ${kernelobj} ../zlib/libz.a
 
 ${Scheme}: ${Kernel} ${Main}
 	$C -o ${Scheme} ${Kernel} ${Main} ${mdclib} ${LDFLAGS}
diff -ru ChezScheme-9.5.2.orig/c/Mf-ti3s2 ChezScheme-9.5.2/c/Mf-ti3s2
--- ChezScheme-9.5.2.orig/c/Mf-ti3s2	2019-03-21 18:05:24.000000000 +0000
+++ ChezScheme-9.5.2/c/Mf-ti3s2	2019-08-25 16:31:18.894981878 +0000
@@ -31,7 +31,7 @@
 include Mf-base
 
 ${Kernel}: ${kernelobj} ../zlib/libz.a
-	ld -melf_i386 -r -X -o ${Kernel} ${kernelobj} ../zlib/libz.a
+	/bin/ld -r -o ${Kernel} ${kernelobj} ../zlib/libz.a
 
 ${Scheme}: ${Kernel} ${Main}
 	$C -o ${Scheme} ${Kernel} ${Main} ${mdclib} ${LDFLAGS}
