diff -wpruN --no-dereference '--exclude=*.orig' a~/src/main/tls.c a/src/main/tls.c
--- a~/src/main/tls.c	1970-01-01 00:00:00
+++ a/src/main/tls.c	1970-01-01 00:00:00
@@ -1824,7 +1824,7 @@ static int load_dh_params(SSL_CTX *ctx,
 	 *
 	 * Change suggested by @t8m
 	 */
-#if OPENSSL_VERSION_NUMBER >= 0x10101000L
+#if (OPENSSL_VERSION_NUMBER >= 0x10101000L) && (OPENSSL_VERSION_NUMBER < 0x30000000L)
 	if (FIPS_mode() > 0) {
 		WARN(LOG_PREFIX ": Ignoring user-selected DH parameters in FIPS mode. Using defaults.");
 		file = NULL;
