--- gtk+-3.1.12/configure.ac-orig	2011-08-18 07:12:43.425834135 -0500
+++ gtk+-3.1.12/configure.ac	2011-08-18 07:13:06.216930259 -0500
@@ -177,6 +177,7 @@ dnl Check for a working C++ compiler, bu
 dnl We use this for an automated test for C++ header correctness.
 dnl
 AC_CHECK_TOOLS(CXX, [$CCC c++ g++ gcc CC cxx cc++ cl], gcc)
+AC_PROG_CXX
 AC_LANG_PUSH([C++])
 
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]],
@@ -191,7 +192,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[@in
                   [OBJC=yes],[OBJC=no])
 AM_CONDITIONAL(HAVE_OBJC, test "$OBJC" = "yes")
 CXXFLAGS="$gtk_save_cxxflags"
-AC_LANG_POP([C++])
+#AC_LANG_POP([C++])
 
 if test "$os_win32" = "yes"; then
   if test x$enable_static = xyes -o x$enable_static = x; then
