--- gtk+-2.12.1/gtk/gtkfilechooserdialog.c.old	2007-10-16 21:20:38.000000000 +0800
+++ gtk+-2.12.1/gtk/gtkfilechooserdialog.c	2007-12-27 16:20:18.388325000 +0800
@@ -465,6 +465,12 @@
     {
       response_id = va_arg (varargs, gint);
       gtk_dialog_add_button (GTK_DIALOG (result), button_text, response_id);
+     
+      if ((response_id == GTK_RESPONSE_OK) ||
+	  (response_id == GTK_RESPONSE_ACCEPT) ||
+	  (response_id == GTK_RESPONSE_YES) ||
+	  (response_id == GTK_RESPONSE_APPLY))
+              gtk_dialog_set_default_response (GTK_DIALOG (result), response_id);	
       button_text = va_arg (varargs, const gchar *);
     }
 
