--- xmlschema-2.3.0/tests/test_typing.py.orig
+++ xmlschema-2.3.0/tests/test_typing.py
@@ -34,6 +34,7 @@
         cls.cases_dir = Path(__file__).parent.joinpath('test_cases/mypy')
         cls.config_file = Path(__file__).parent.parent.joinpath('mypy.ini')
 
+    @unittest.skip("fails for Python 3.7: https://github.com/sissaschool/xmlschema/issues/348")
     def test_schema_source(self):
         result = mypy_api.run([
             '--strict',
@@ -43,6 +44,7 @@
         ])
         self.assertEqual(result[2], 0, msg=result[1] or result[0])
 
+    @unittest.skip("fails for Python 3.7: https://github.com/sissaschool/xmlschema/issues/348")
     def test_simple_types(self):
         result = mypy_api.run([
             '--strict',
@@ -52,6 +54,7 @@
         ])
         self.assertEqual(result[2], 0, msg=result[1] or result[0])
 
+    @unittest.skip("fails for Python 3.7: https://github.com/sissaschool/xmlschema/issues/348")
     def test_extra_validator__issue_291(self):
         result = mypy_api.run([
             '--strict',
