diff -Naur mintmenu-5.5.2/usr/bin/mintmenu mintmenu-5.5.2-patched/usr/bin/mintmenu
--- mintmenu-5.5.2/usr/bin/mintmenu	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/bin/mintmenu	2016-03-29 22:44:56.879949897 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.7
 
 import sys, os
 
diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/capi.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/capi.py
--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/capi.py	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/capi.py	2016-03-29 22:45:11.019507067 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 
 import gi
 import ctypes
diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/compile.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/compile.py
--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/compile.py	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/compile.py	2016-03-29 22:45:17.995536922 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 
 import compileall
 
diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintMenu.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintMenu.py
--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintMenu.py	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintMenu.py	2016-03-29 22:45:26.890692058 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 
 import gi
 gi.require_version("Gtk", "2.0")
diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintMenuConfig.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintMenuConfig.py
--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintMenuConfig.py	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintMenuConfig.py	2016-03-29 22:45:32.507373476 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 
 import sys
 
diff -Naur mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintRemove.py mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintRemove.py
--- mintmenu-5.5.2/usr/lib/linuxmint/mintMenu/mintRemove.py	2014-03-18 12:50:46.000000000 +0100
+++ mintmenu-5.5.2-patched/usr/lib/linuxmint/mintMenu/mintRemove.py	2016-03-29 22:45:37.903789626 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 
 import gi
 gi.require_version("Gtk", "2.0")
