diff -Npur gdm-old/daemon/gdm-session-worker.c gdm-new/daemon/gdm-session-worker.c
--- gdm-old/daemon/gdm-session-worker.c	2009-08-20 16:28:09.000000000 +0800
+++ gdm-new/daemon/gdm-session-worker.c	2009-08-20 16:31:11.000000000 +0800
@@ -955,7 +955,7 @@ gdm_session_worker_process_pam_message (
                 res = gdm_session_worker_ask_for_secret (worker, utf8_msg, &user_answer);
                 break;
         case PAM_TEXT_INFO:
-                res = gdm_session_worker_report_info (worker, utf8_msg);
+                res = gdm_session_worker_report_problem (worker, utf8_msg);
                 break;
         case PAM_ERROR_MSG:
                 res = gdm_session_worker_report_problem (worker, utf8_msg);
diff -Npur gdm-old/gui/simple-greeter/gdm-greeter-login-window.c gdm-new/gui/simple-greeter/gdm-greeter-login-window.c
--- gdm-old/gui/simple-greeter/gdm-greeter-login-window.c	2009-08-20 16:28:26.000000000 +0800
+++ gdm-new/gui/simple-greeter/gdm-greeter-login-window.c	2009-08-20 16:31:36.000000000 +0800
@@ -862,9 +862,68 @@ gdm_greeter_login_window_problem (GdmGre
 
         g_debug ("GdmGreeterLoginWindow: problem: %s", text);
 
-        set_message (GDM_GREETER_LOGIN_WINDOW (login_window), text);
+        _gdm_greeter_login_window_set_interactive (login_window, FALSE);
+
+        GtkWidget *dialog;
+        GtkWidget *dialog_vbox;
+        GtkWidget *dialog_action_area;
+        GtkWidget *hbox;
+        GtkWidget *vbox;
+        GtkWidget *label;
+        GtkWidget *button;
+        GtkWidget *image;
+        gchar     *title;
+        gchar     *msg;
+
+        dialog = gtk_dialog_new ();
+
+        gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
+        gtk_container_set_border_width (GTK_CONTAINER (dialog), 5);
+        gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
+        gtk_window_set_title (GTK_WINDOW (dialog), "");
+
+        dialog_vbox = GTK_DIALOG (dialog)->vbox;
+        gtk_box_set_spacing (GTK_BOX (dialog_vbox), 12);
+
+        hbox = gtk_hbox_new (FALSE, 12);
+        gtk_box_pack_start (GTK_BOX (dialog_vbox), hbox, FALSE, FALSE, 0);
+        gtk_container_set_border_width (GTK_CONTAINER (hbox), 5);
+        gtk_widget_show (hbox);
+
+        image = gtk_image_new_from_stock ("gtk-dialog-warning", GTK_ICON_SIZE_DIALOG);
+        gtk_box_pack_start (GTK_BOX (hbox), image, FALSE, FALSE, 0);
+        gtk_widget_show (image);
+
+        vbox = gtk_vbox_new (FALSE, 6);
+        gtk_box_pack_start (GTK_BOX (hbox), vbox, TRUE, TRUE, 0);
+        gtk_widget_show (vbox);
+
+        title = g_strconcat ("<b>", text, "</b>", NULL);
+        label = gtk_label_new (title);
+        gtk_box_pack_start (GTK_BOX (vbox), label, FALSE, FALSE, 0);
+        gtk_label_set_use_markup (GTK_LABEL (label), TRUE);
+        gtk_label_set_justify (GTK_LABEL (label), GTK_JUSTIFY_LEFT);
+        gtk_label_set_line_wrap (GTK_LABEL (label), TRUE);
+        gtk_misc_set_alignment (GTK_MISC (label), 0, 0.5);
+        gtk_widget_show (label);
+        g_free (title);
+
+        button = gtk_button_new_from_stock (GTK_STOCK_OK);
+        gtk_dialog_add_action_widget (GTK_DIALOG (dialog), button, GTK_RESPONSE_OK);
+        GTK_WIDGET_SET_FLAGS (button, GTK_CAN_DEFAULT);
+        gtk_widget_show (button);
+
+        gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (login_window));
+
+        gtk_dialog_set_default_response (GTK_DIALOG (dialog),
+                                        GTK_RESPONSE_OK);
+        gtk_dialog_run (GTK_DIALOG (dialog));
+        gtk_widget_destroy (dialog);
+
         gdk_window_beep (GTK_WIDGET (login_window)->window);
 
+        _gdm_greeter_login_window_set_interactive (login_window, TRUE);
+
         return TRUE;
 }
 
