--- gdm-2.30.5/daemon/gdm-slave.c-orig	2011-03-29 04:40:11.034342290 -0500
+++ gdm-2.30.5/daemon/gdm-slave.c	2011-03-29 04:41:19.979405546 -0500
@@ -1343,7 +1343,7 @@ gdm_slave_get_primary_session_id_for_use
                                  G_TYPE_BOOLEAN, &can_activate_sessions,
                                  G_TYPE_INVALID);
         if (! res) {
-                g_warning ("unable to determine if seat can activate sessions: %s",
+                g_debug ("unable to determine if seat can activate sessions: %s",
                            error->message);
                 g_error_free (error);
                 goto out;
