--- ospm-1.4.11/applet/ospm-status-dialog.c.1	2016-04-14 01:20:14.360705025 +0300
+++ ospm-1.4.11/applet/ospm-status-dialog.c	2016-04-14 01:23:51.940803072 +0300
@@ -139,7 +139,7 @@
         if (!notify_init (PACKAGE))
                 exit (1);
 
-        notify = notify_notification_new_with_status_icon ("ospm-applet", NULL, NULL, icon);
+        notify = notify_notification_new ("ospm-applet", NULL, NULL);
 
         notify_notification_add_action (notify, "default", "default action", notify_action_cb, icon, NULL);
 
