diff -Nru mono-6.13.0.1237.orig/mono/metadata/w32file-unix.c mono-6.13.0.1237/mono/metadata/w32file-unix.c
--- mono-6.13.0.1237.orig/mono/metadata/w32file-unix.c	2022-06-15 00:54:53.000000000 +0000
+++ mono-6.13.0.1237/mono/metadata/w32file-unix.c	2022-07-13 18:41:43.425599234 +0000
@@ -4766,7 +4766,7 @@
 	return (drive_type);
 }
 
-#if defined (HOST_DARWIN) || defined (__linux__) || defined(HOST_BSD) || defined(__FreeBSD_kernel__) || defined(__HAIKU__) || defined(_AIX)
+#if defined (HOST_DARWIN) || defined (__linux__) || defined(HOST_BSD) || defined(__FreeBSD_kernel__) || defined(__HAIKU__) || defined(_AIX) || defined(__sun)
 static gchar*
 get_fstypename (gchar *utfpath)
 {
