--- a/src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c	Thu Jun  8 15:06:27 2023
+++ b/src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c	Mon Jun 12 09:15:47 2023
@@ -37,6 +37,10 @@
 #include "gtk_interface.h"
 #include "gtk3_interface.h"
 
+#ifndef MAX
+#define MAX(a,b) ((a) > (b) ? (a) : (b))
+#endif
+
 int DEBUG_SCREENCAST_ENABLED = FALSE;
 
 #define EXCEPTION_CHECK_DESCRIBE() if ((*env)->ExceptionCheck(env)) { \
