--- poppler-0.32.0/glib/Makefile.am.~1~	2015-03-24 08:52:53.005146994 +0300
+++ poppler-0.32.0/glib/Makefile.am	2015-03-24 08:55:45.958272694 +0300
@@ -108,6 +108,7 @@
 Poppler_0_18_gir_FILES = $(introspection_files)
 Poppler_0_18_gir_SCANNERFLAGS = --c-include poppler.h
 Poppler_0_18_gir_EXPORT_PACKAGES = poppler-glib
+Poppler_0_18_gir_LDFLAGS = -lgmodule-2.0
 
 INTROSPECTION_GIRS += Poppler-0.18.gir
 
