--- mpc-0.9/configure.orig	Wed Jan 25 11:05:39 2012
+++ mpc-0.9/configure	Wed Jan 25 11:05:46 2012
@@ -10989,7 +10989,7 @@
 
 fi
 
-for ac_header in complex.h locale.h inttypes.h stdint.h limits.h unistd.h sys/time.h
+for ac_header in locale.h inttypes.h stdint.h limits.h unistd.h sys/time.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
