https://github.com/jelmer/dulwich/issues/1293

--- dulwich-dulwich-0.22.1/tests/test_greenthreads.py.orig
+++ dulwich-dulwich-0.22.1/tests/test_greenthreads.py
@@ -35,7 +35,7 @@
     gevent_support = False
 
 if gevent_support:
-    from ..greenthreads import GreenThreadsMissingObjectFinder
+    from dulwich.greenthreads import GreenThreadsMissingObjectFinder
 
 skipmsg = "Gevent library is not installed"
 
