--- openbox-3.6.1/configure-orig	Wed Jul  1 04:23:47 2015
+++ openbox-3.6.1/configure	Wed May  5 12:36:30 2021
@@ -18551,9 +18551,9 @@
 # Flag that enforces no undefined symbols.
 no_undefined_flag=$lt_no_undefined_flag
 
-# Flag to hardcode \$libdir into a binary during linking.
-# This must work even if \$libdir does not exist
-hardcode_libdir_flag_spec=$lt_hardcode_libdir_flag_spec
+# Omit flag to hardcode \$libdir into a binary during linking.
+# The default directories are never added to the runtime search path on OI
+hardcode_libdir_flag_spec=""
 
 # Whether we need a single "-rpath" flag with a separated argument.
 hardcode_libdir_separator=$lt_hardcode_libdir_separator
