Disable all utilities but watch.

--- procps-ng-3.3.15/Makefile.am.~1~	2018-05-19 23:35:23.067591570 +0000
+++ procps-ng-3.3.15/Makefile.am	2019-02-16 15:13:07.738914886 +0000
@@ -30,24 +30,13 @@ transform =
 
 if !CYGWIN
 transform += s/pscommand/ps/; $(program_transform_name)
-sbin_PROGRAMS = \
-	sysctl
+sbin_PROGRAMS = 
 else
 transform += s/pscommand/procps/; $(program_transform_name)
 endif
 
 if !CYGWIN
-bin_PROGRAMS = \
-	ps/pscommand \
-	free \
-	pgrep \
-	pkill \
-	pmap \
-	pwdx \
-	tload \
-	uptime \
-	vmstat \
-	w
+bin_PROGRAMS = 
 else
 usrbin_exec_PROGRAMS += \
 	ps/pscommand \
@@ -63,15 +52,7 @@ endif
 lib_LTLIBRARIES = \
 	proc/libprocps.la
 
-dist_man_MANS = \
-	free.1 \
-	pgrep.1 \
-	pkill.1 \
-	pmap.1 \
-	uptime.1 \
-	vmstat.8 \
-	w.1 \
-	ps/procps.1
+dist_man_MANS = 
 
 if !CYGWIN
 dist_man_MANS += \
@@ -133,9 +114,7 @@ endif
 if WITH_NCURSES
 if !CYGWIN
 bin_PROGRAMS += \
-	slabtop \
-	watch \
-	top/top
+	watch
 else
 usrbin_exec_PROGRAMS += \
 	watch \
@@ -143,8 +122,7 @@ usrbin_exec_PROGRAMS += \
 endif
 
 dist_man_MANS += \
-	watch.1 \
-	top/top.1
+	watch.1
 watch_SOURCES = watch.c lib/strutils.c lib/fileutils.c
 watch_LDADD = @WATCH_NCURSES_LIBS@ $(CYGWINFLAGS)
 watch_CFLAGS = @WATCH_NCURSES_CFLAGS@
@@ -197,8 +175,7 @@ pwdx_SOURCES = pwdx.c lib/fileutils.c
 pwdx_LDADD= $(CYGWINFLAGS)
 sysctl_SOURCES = \
 	sysctl.c \
-	lib/fileutils.c \
-	procio.c
+	lib/fileutils.c
 endif
 tload_SOURCES = tload.c lib/strutils.c lib/fileutils.c
 uptime_SOURCES = uptime.c lib/fileutils.c
@@ -224,8 +201,7 @@ endif
 
 proc_libprocps_la_LDFLAGS = \
 	-version-info $(LIBprocps_CURRENT):$(LIBprocps_REVISION):$(LIBprocps_AGE) \
-	-no-undefined \
-	-Wl,--version-script=$(top_srcdir)/proc/libprocps.sym
+	-no-undefined
 
 proc_libprocps_la_SOURCES = \
 	proc/alloc.c \
@@ -246,7 +222,6 @@ proc_libprocps_la_SOURCES = \
 	proc/sig.h \
 	proc/slab.c \
 	proc/slab.h \
-	proc/sysinfo.c \
 	proc/sysinfo.h \
 	proc/version.c \
 	proc/version.h \
