$NetBSD$

--- procps-3.2.8/Makefile.orig	2009-05-10 18:50:48.000000000 +0000
+++ procps-3.2.8/Makefile
@@ -32,16 +32,15 @@ install  := install -D --owner 0 --group
 # Lame x86-64 /lib64 and /usr/lib64 abomination:
 lib64    := lib$(shell [ -d /lib64 ] && echo 64)
 
-usr/bin                  := $(DESTDIR)/usr/bin/
-bin                      := $(DESTDIR)/bin/
-sbin                     := $(DESTDIR)/sbin/
-usr/proc/bin             := $(DESTDIR)/usr/bin/
-man1                     := $(DESTDIR)/usr/share/man/man1/
-man5                     := $(DESTDIR)/usr/share/man/man5/
-man8                     := $(DESTDIR)/usr/share/man/man8/
-lib                      := $(DESTDIR)/$(lib64)/
-usr/lib                  := $(DESTDIR)/usr/$(lib64)/
-usr/include              := $(DESTDIR)/usr/include/
+usr/bin                  := $(DESTDIR)$(PREFIX)/bin/
+bin                      := $(DESTDIR)$(PREFIX)/bin/
+sbin                     := $(DESTDIR)$(PREFIX)/sbin/
+man1                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man1/
+man5                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man5/
+man8                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man8/
+lib                      := $(DESTDIR)$(PREFIX)/lib/
+usr/lib                  := $(DESTDIR)$(PREFIX)/lib/
+usr/include              := $(DESTDIR)$(PREFIX)/include/
 
 #SKIP     := $(bin)kill $(man1)kill.1
 
@@ -105,8 +104,8 @@ PKG_CFLAGS   := -fno-common -ffast-math
 CFLAGS       := -O2 -s
 ALL_CFLAGS   := $(PKG_CFLAGS) $(CFLAGS)
 
-PKG_LDFLAGS  := -Wl,-warn-common
-LDFLAGS      :=
+PKG_LDFLAGS  := 
+#LDFLAGS      :=
 ALL_LDFLAGS  := $(PKG_LDFLAGS) $(LDFLAGS)
 
 ############ Add some extra flags if gcc allows
