--- procps-3.2.8/Makefile-orig	   :: 
+++ procps-3.2.8/Makefile	   :: 
@@ -32,16 +32,15 @@
 # Lame x86-64 /lib64 and /usr/lib64 abomination:
 lib64    := lib$(shell [ -d /lib64 ] && echo 64)
 
-usr/bin                  := $(DESTDIR)/usr/bin/
-bin                      := $(DESTDIR)/bin/
-sbin                     := $(DESTDIR)/sbin/
-usr/proc/bin             := $(DESTDIR)/usr/bin/
-man1                     := $(DESTDIR)/usr/share/man/man1/
-man5                     := $(DESTDIR)/usr/share/man/man5/
-man8                     := $(DESTDIR)/usr/share/man/man8/
-lib                      := $(DESTDIR)/$(lib64)/
-usr/lib                  := $(DESTDIR)/usr/$(lib64)/
-usr/include              := $(DESTDIR)/usr/include/
+usr/bin                  := $(DESTDIR)$(PREFIX)/bin/
+bin                      := $(DESTDIR)$(PREFIX)/bin/
+sbin                     := $(DESTDIR)$(PREFIX)/sbin/
+man1                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man1/
+man5                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man5/
+man8                     := $(DESTDIR)$(PREFIX)/$(PKGMANDIR)/man8/
+lib                      := $(DESTDIR)$(PREFIX)/lib/
+usr/lib                  := $(DESTDIR)$(PREFIX)/lib/
+usr/include              := $(DESTDIR)$(PREFIX)/include/
 
 #SKIP     := $(bin)kill $(man1)kill.1
 
@@ -102,11 +101,11 @@
   -Wstrict-prototypes -Wmissing-prototypes
 # Note that some stuff below is conditional on CFLAGS containing
 # an option that starts with "-g". (-g, -g2, -g3, -ggdb, etc.)
-CFLAGS       := -O2 -s
+#CFLAGS       := -O2 -s
 ALL_CFLAGS   := $(PKG_CFLAGS) $(CFLAGS)
 
-PKG_LDFLAGS  := -Wl,-warn-common
-LDFLAGS      :=
+PKG_LDFLAGS  := 
+#LDFLAGS      :=
 ALL_LDFLAGS  := $(PKG_LDFLAGS) $(LDFLAGS)
 
 ############ Add some extra flags if gcc allows
@@ -127,7 +126,7 @@
 # produce separate executables for ppc and ppc64, s390 and s390x,
 # i386 and x86-64, mips and mips64, sparc and sparc64, and so on.
 # Failure to do so will cause data corruption.
-m64 := $(call check_gcc,-m64,$(call check_gcc,-mabi=64,))
+
 ALL_CFLAGS += $(m64)
 
 ALL_CFLAGS += $(call check_gcc,-Wdeclaration-after-statement,)
