--- libgnome-2.10.0/schemas/desktop_gnome_interface.schemas.in.in	2005-07-18 15:54:51.821099000 +0600
+++ libgnome-2.10.0-new/schemas/desktop_gnome_interface.schemas.in.in	2005-07-18 16:30:02.215039000 +0600
@@ -185,7 +185,7 @@
       <applyto>/desktop/gnome/interface/gtk-im-status-style</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>callback</default>
+      <default>nothing</default>
       <locale name="C">
 	<short>GTK IM Status Style</short>
 	<long>Name of the GTK+ input method Status Style used by gtk+.</long>
