--- libgnome-2.10.0/schemas/desktop_gnome_interface.schemas.in.in	Mon Jul 18 15:46:43 2005
+++ libgnome-2.10.0-new/schemas/desktop_gnome_interface.schemas.in.in	Mon Jul 18 15:45:59 2005
@@ -130,7 +130,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>nimbus</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
@@ -141,7 +141,7 @@
       <applyto>/desktop/gnome/interface/gtk_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>nimbus</default>
       <locale name="C">
 	<short>Gtk+ Theme</short>
 	<long>Basename of the default theme used by gtk+.</long>
