--- brltty-4.4/Bindings/Python/bindings.m4.~1~	2014-12-03 14:46:47.374925879 +0300
+++ brltty-4.4/Bindings/Python/bindings.m4	2014-12-03 14:47:14.353183467 +0300
@@ -121,7 +121,7 @@
 
 if test "${GCC}" = "yes"
 then
-   PYREXC_CFLAGS="-Wno-parentheses -Wno-unused -fno-strict-aliasing -U_POSIX_C_SOURCE -U_XOPEN_SOURCE"
+   PYREXC_CFLAGS="-Wno-parentheses -Wno-unused -fno-strict-aliasing -U_POSIX_C_SOURCE -U_XOPEN_SOURCE ${CFLAGS}"
 else
    case "${host_os}"
    in
