Avoid dependency on gnome/accessibility/at-spi
--- brltty-5.5/configure.ac.1	2017-08-02 14:05:31.262686345 +0000
+++ brltty-5.5/configure.ac	2017-08-02 14:06:11.373618729 +0000
@@ -1802,16 +1802,16 @@
    BRLTTY_SCREEN_DRIVER([sc], [Screen])
 ])
 
-if test "${brltty_enabled_x}" = "yes"
-then
-   BRLTTY_HAVE_PACKAGE([cspi], [cspi-1.0], [dnl
-      AC_CHECK_HEADER([X11/keysym.h], [dnl
-         AC_CHECK_HEADER([X11/Xlib.h], [dnl
-            BRLTTY_SCREEN_DRIVER([as], [AtSpi], [$(X11_LIBS) $(CSPI_LIBS)])
-         ])
-      ])
-   ])
-fi
+#if test "${brltty_enabled_x}" = "yes"
+#then
+#   BRLTTY_HAVE_PACKAGE([cspi], [cspi-1.0], [dnl
+#      AC_CHECK_HEADER([X11/keysym.h], [dnl
+#         AC_CHECK_HEADER([X11/Xlib.h], [dnl
+#            BRLTTY_SCREEN_DRIVER([as], [AtSpi], [$(X11_LIBS) $(CSPI_LIBS)])
+#         ])
+#      ])
+#   ])
+#fi
 
 BRLTTY_HAVE_PACKAGE([dbus], ["dbus-1 >= 1.0"], [dnl
    AC_CHECK_HEADER([X11/keysym.h], [dnl
