--- p7zip-17.05/C/zstd/xxhash.h.~1~	2023-03-03 07:16:28.000000000 -0500
+++ p7zip-17.05/C/zstd/xxhash.h	2024-11-16 13:54:19.631821723 -0500
@@ -211,8 +211,8 @@
 #  define restrict   /* disable restrict */
 #endif
 
-XXH_PUBLIC_API void XXH32_copyState(XXH32_state_t* restrict dst_state, const XXH32_state_t* restrict src_state);
-XXH_PUBLIC_API void XXH64_copyState(XXH64_state_t* restrict dst_state, const XXH64_state_t* restrict src_state);
+XXH_PUBLIC_API void XXH32_copyState(XXH32_state_t* dst_state, const XXH32_state_t* src_state);
+XXH_PUBLIC_API void XXH64_copyState(XXH64_state_t* dst_state, const XXH64_state_t* src_state);
 
 
 /* **************************
