--- p7zip-17.05/install.sh.old	2024-11-19 13:36:29.702314413 -0500
+++ p7zip-17.05/install.sh	2024-11-19 13:37:13.412063857 -0500
@@ -159,10 +159,6 @@
    chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 fi
 
-gzip "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-gzip "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-gzip "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-
 if [ -f README ]
 then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/README"
--- p7zip-17.05/makefile.common.old	2024-11-19 13:52:42.646052274 -0500
+++ p7zip-17.05/makefile.common	2024-11-19 13:53:05.302728216 -0500
@@ -4,7 +4,7 @@
 DEST_BIN=$(DEST_HOME)/bin
 DEST_SHARE=$(DEST_HOME)/lib/p7zip
 DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
-DEST_MAN=$(DEST_HOME)/man
+DEST_MAN=$(DEST_HOME)/share/man
 
 .PHONY: default all all2 7za 7zG 7zFM sfx 7z 7zr Client7z common common7z LangAndHelp tar_bin depend test test_7z test_7zr test_7zG test_Client7z LzmaCon test_LzmaCon  all_test app cppcheck lzham test_lib other
 
