*** gcc-3.4.3/gcc/Makefile.in.orig	Sat Oct 27 14:36:42 2007
--- gcc-3.4.3/gcc/Makefile.in	Sat Oct 27 14:37:25 2007
***************
*** 3859,3865 ****
  # script does that.
  .po.gmo:
  	-test -d po || mkdir po
! 	$(GMSGFMT) --statistics -o $@ $<
  
  # The new .po has to be gone over by hand, so we deposit it into
  # build/po with a different extension.
--- 3859,3865 ----
  # script does that.
  .po.gmo:
  	-test -d po || mkdir po
! 	$(GMSGFMT) -o $@ $<
  
  # The new .po has to be gone over by hand, so we deposit it into
  # build/po with a different extension.
