*** gcc-3.4.3/gcc/config/sparc/sol2-bi.h.orig	Tue Jul 13 04:18:39 2004
--- gcc-3.4.3/gcc/config/sparc/sol2-bi.h	Mon Jan 28 13:37:41 2008
***************
*** 150,161 ****
     %{YP,*} \
     %{R*} \
     %{compat-bsd: \
!      %{!YP,*:%{p|pg:-Y P,/usr/ucblib/sparcv9:/usr/lib/libp/sparcv9:/usr/lib/sparcv9} \
!        %{!p:%{!pg:-Y P,/usr/ucblib/sparcv9:/usr/lib/sparcv9}}} \
!      -R /usr/ucblib/sparcv9} \
     %{!compat-bsd: \
!      %{!YP,*:%{p|pg:-Y P,/usr/lib/libp/sparcv9:/usr/lib/sparcv9} \
!        %{!p:%{!pg:-Y P,/usr/lib/sparcv9}}}}"
  
  #define LINK_ARCH64_SPEC LINK_ARCH64_SPEC_BASE
  
--- 150,161 ----
     %{YP,*} \
     %{R*} \
     %{compat-bsd: \
!      %{!YP,*:%{p|pg:-Y P,/usr/gcc/3.4/lib/64:/usr/lib/libp/64:/lib/64:/usr/lib/64 -R /usr/gcc/3.4/lib/64:/usr/lib/libp/64} \
!        %{!p:%{!pg:-Y P,/usr/gcc/3.4/lib/64:/lib/64:/usr/lib/64 -R /usr/gcc/3.4/lib/64}}} \
!      } \
     %{!compat-bsd: \
!      %{!YP,*:%{p|pg:-Y P,/usr/gcc/3.4/lib/64:/usr/lib/libp/64:/lib/64:/usr/lib/64 -R /usr/gcc/3.4/lib/64:/usr/lib/libp/64} \
!        %{!p:%{!pg:-Y P,/usr/gcc/3.4/lib/64:/lib/64:/usr/lib/64 -R /usr/gcc/3.4/lib/64}}}}"
  
  #define LINK_ARCH64_SPEC LINK_ARCH64_SPEC_BASE
  
