*** gcc-3.4.3/gcc/version.c.orig	Tue Nov  9 17:51:34 2004
--- gcc-3.4.3/gcc/version.c	Tue Nov  9 17:52:08 2004
***************
*** 5,11 ****
     please modify this string to indicate that, e.g. by putting your
     organization's name in parentheses at the end of the string.  */
  
! const char version_string[] = "3.4.3";
  
  /* This is the location of the online document giving instructions for
     reporting bugs.  If you distribute a modified version of GCC,
--- 5,11 ----
     please modify this string to indicate that, e.g. by putting your
     organization's name in parentheses at the end of the string.  */
  
! const char version_string[] = "3.4.3 (csl-sol210-3_4-20050802)";
  
  /* This is the location of the online document giving instructions for
     reporting bugs.  If you distribute a modified version of GCC,
