diff -wpruN --no-dereference '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -189,7 +189,7 @@ AC_EGREP_CPP(yes,
 ], AC_NOTE(- you have a SVR4 system) AC_DEFINE(SVR4) svr4=1)
 if test -n "$svr4" ; then
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
+LIBS="$LIBS"
 AC_CHECKING(SVR4)
 AC_TRY_LINK([
     #include <utmpx.h>
@@ -206,7 +206,7 @@ AC_EGREP_CPP(YES_IS_DEFINED,
 [#if defined(SVR4) && defined(sun)
   YES_IS_DEFINED;
 #endif
-], LIBS="$LIBS -lsocket -lnsl -lkstat")
+], LIBS="$LIBS -lsocket")
 
 dnl
 dnl    ****  typedefs ****
