--- deskbar-applet-2.30.1/deskbar/handlers/evolution/evolution.c	2010-05-02 12:09:39.000000000 +0100
+++ deskbar-applet-2.30.1/deskbar/handlers/evolution/evolution.c	2014-03-07 19:38:10.275763470 +0000
@@ -1,5 +1,4 @@
 #include <glib.h>
-#include <glib/gstring.h>
 #include <libebook/e-book.h>
 #include <pango/pango.h>
 #include <string.h>
--- deskbar-applet-2.30.1/deskbar/handlers/evolution/evolution.h	2010-05-02 12:09:39.000000000 +0100
+++ deskbar-applet-2.30.1/deskbar/handlers/evolution/evolution.h	2014-03-07 19:38:10.278030660 +0000
@@ -2,7 +2,7 @@
 #ifndef __DESKBAR_EVOLUTION_H__
 #define __DESKBAR_EVOLUTION_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
 G_BEGIN_DECLS
