--- a/src/hotspot/os/posix/signals_posix.cpp	Wed Oct  7 17:21:51 2020
+++ b/src/hotspot/os/posix/signals_posix.cpp	Sat Oct 10 16:46:21 2020
@@ -548,6 +548,8 @@
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_aix_signal
 #elif defined(LINUX)
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_linux_signal
+#elif defined(SOLARIS)
+#define JVM_HANDLE_XXX_SIGNAL JVM_handle_solaris_signal
 #else
 #error who are you?
 #endif
