This is called extensively from the test suite.

--- a/src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/services/Services.java	Tue Aug 20 16:57:04 2024
+++ b/src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/services/Services.java	Mon Sep 23 11:04:21 2024
@@ -258,6 +258,7 @@
                 }
                 return "Windows";
             }
+            case SOLARIS: return "Solaris";
             default: throw new InternalError("missing case for " + os);
         }
     }
@@ -272,6 +273,7 @@
             case ARM: return "arm";
             case S390: return "s390";
             case PPC64: return "ppc64";
+            case SPARCV9: return "sparcv9";
             case OTHER: return "other";
             default: throw new InternalError("missing case for " + arch);
         }
