Avoid core dump when gconf schema is not updated

--- evolution-2.32.3/shell/e-shell-settings.c.~1~	2015-01-11 14:23:00.095580494 +0300
+++ evolution-2.32.3/shell/e-shell-settings.c	2015-01-11 14:23:52.677223463 +0300
@@ -65,6 +65,8 @@
		return NULL;
	}

+	g_return_val_if_fail (entry != NULL, NULL);
+
	schema_name = gconf_entry_get_schema_name (entry);
	g_return_val_if_fail (schema_name != NULL, NULL);
