--- evolution-2.27.91/data/evolution.desktop.in.in.orig	Mon Aug 24 14:08:06 2009
+++ evolution-2.27.91/data/evolution.desktop.in.in	Mon Aug 24 14:09:21 2009
@@ -6,7 +6,7 @@
 Icon=evolution
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Office;Email;Calendar;ContactManagement;X-Red-Hat-Base;
+Categories=GNOME;GTK;Network;Email;Calendar;ContactManagement;X-Red-Hat-Base;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Evolution
