--- evolution-2.23.3.1/mail/mail-config.c.old	Fri May 30 15:23:14 2008
+++ evolution-2.23.3.1/mail/mail-config.c	Wed Jun 11 15:48:42 2008
@@ -184,7 +184,7 @@
 	spell_color = gconf_client_get_string (client, key, NULL);
 
 	fprintf (rc, "style \"evolution-mail-custom-fonts\" {\n");
-	fprintf (rc, "        GtkHTML::spell_error_color = \"%s\"\n", spell_color);
+	fprintf (rc, "        GtkHTML::spell_error_color = \"%s\"\n", spell_color ? spell_color : "");
 	g_free (spell_color);
 
 	if (gconf_client_get_bool (config->gconf, "/apps/evolution/mail/display/mark_citations", NULL))
