It is hard to normalize pytest-xdist test results.

--- pyhamcrest-2.1.0/pyproject.toml.orig
+++ pyhamcrest-2.1.0/pyproject.toml
@@ -50,7 +50,6 @@
 docs = ["sphinx~=4.0", "alabaster~=0.7"]
 tests = [
   "pytest>=5.0",
-  "pytest-xdist",
   "coverage[toml]",
   # No point on Pypy thanks to https://github.com/python/typed_ast/issues/111
   "pytest-mypy-plugins; platform_python_implementation != 'PyPy'",
