diff -urN gnome-vfs-2.10.1/libgnomevfs/xdgmime.c gnome-vfs-2.10.1-hacked/libgnomevfs/xdgmime.c
--- gnome-vfs-2.10.1/libgnomevfs/xdgmime.c	2004-12-10 04:43:48.000000000 +1300
+++ gnome-vfs-2.10.1-hacked/libgnomevfs/xdgmime.c	2005-04-26 14:46:23.988504256 +1200
@@ -207,7 +207,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = GNOME_VFS_DATADIR":/usr/share/gnome/:/usr/sfw/share:/usr/local/share/:/usr/share/";
 
   ptr = xdg_data_dirs;
 
