--- gtk-vnc-0.7.1/src/libgtk-vnc_sym.version.1	2017-12-01 13:45:00.921530132 +0000
+++ gtk-vnc-0.7.1/src/libgtk-vnc_sym.version	2017-12-01 13:45:41.185497884 +0000
@@ -71,9 +71,10 @@
 
     vnc_display_request_update;
 
-    vnc_image_framebuffer_get_type;
-    vnc_image_framebuffer_new;
-    vnc_image_framebuffer_get_image;
+# Not present in GTK3 version
+#    vnc_image_framebuffer_get_type;
+#    vnc_image_framebuffer_new;
+#    vnc_image_framebuffer_get_image;
 
     vnc_cairo_framebuffer_get_type;
     vnc_cairo_framebuffer_new;
