--- http-parser-2.9.4/test.c.orig	   :: 
+++ http-parser-2.9.4/test.c	   :: 
@@ -4221,7 +4221,9 @@
   printf("http_parser v%u.%u.%u (0x%06lx)\n", major, minor, patch, version);
 
   printf("sizeof(http_parser) = %u\n", (unsigned int)sizeof(http_parser));
-  assert(sizeof(http_parser) == 4 + 4 + 8 + 2 + 2 + 4 + sizeof(void *));
+  //// On sparcv7, the field alignment/padding will result http_parser
+  //// to be 32 bytes, yet pointer size is still 32-bit.
+  //// assert(sizeof(http_parser) == 4 + 4 + 8 + 2 + 2 + 4 + sizeof(void *));
 
   //// API
   test_preserve_data();
