--- seahorse-plugins-2.30.1/libseahorse/seahorse-notification.c.1	2016-04-14 11:49:06.190078651 +0300
+++ seahorse-plugins-2.30.1/libseahorse/seahorse-notification.c	2016-04-14 11:49:40.718368348 +0300
@@ -213,7 +213,7 @@
     heading = format_key_text (snotif->heading);
     message = format_key_text (snotif->message);
     
-    notif = notify_notification_new (heading, message, snotif->icon, attachto);
+    notif = notify_notification_new (heading, message, snotif->icon);
     
     g_free (heading);
     g_free (message);
