Disable fuzz tests, as they require fmemopen

--- wget-1.21/Makefile.in	2020-12-31 16:54:36.000000000 +0000
+++ wget-1.21/Makefile.in	2021-01-01 20:23:27.118043752 +0000
@@ -1637,7 +1637,7 @@ distuninstallcheck_listfiles = find . -t
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po gnulib_po util fuzz tests testenv
+SUBDIRS = lib src doc po gnulib_po util tests testenv
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
              msdos/Makefile.WC ABOUT-NLS \
