Disable fuzz tests, as they require fmemopen

--- wget-1.19.5/Makefile.in.1	2018-05-14 13:12:56.368147267 +0000
+++ wget-1.19.5/Makefile.in	2018-05-14 13:13:10.470098770 +0000
@@ -1443,7 +1443,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po util fuzz tests testenv
+SUBDIRS = lib src doc po util tests testenv
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
              msdos/Makefile.WC ABOUT-NLS \
