diff -u gDesklets-0.36.1.old/ctrlinfo gDesklets-0.36.1/ctrlinfo
--- gDesklets-0.36.1.old/ctrlinfo	2009-12-09 14:48:14.041042000 +0800
+++ gDesklets-0.36.1/ctrlinfo	2009-12-09 14:54:08.396962000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 
 # This tool loads a given control and outputs information about the control.
 #
diff -u gDesklets-0.36.1.old/gdesklets gDesklets-0.36.1/gdesklets
--- gDesklets-0.36.1.old/gdesklets	2009-12-09 14:48:14.041210000 +0800
+++ gDesklets-0.36.1/gdesklets	2009-12-09 14:54:20.452467000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 
 from main import client, DISPLAY, HOME, USERHOME
 from main import LOGFILE, LOG_PATH, PID_PATH, REGISTRY_PATH, SOCKET_PATH
diff -u gDesklets-0.36.1.old/gdesklets-daemon gDesklets-0.36.1/gdesklets-daemon
--- gDesklets-0.36.1.old/gdesklets-daemon	2009-12-09 14:48:14.041345000 +0800
+++ gDesklets-0.36.1/gdesklets-daemon	2009-12-09 14:54:42.524281000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -*- coding: UTF-8 -*-
 
 import __builtin__
diff -u gDesklets-0.36.1.old/gdesklets-logview gDesklets-0.36.1/gdesklets-logview
--- gDesklets-0.36.1.old/gdesklets-logview	2009-12-09 14:48:14.041477000 +0800
+++ gDesklets-0.36.1/gdesklets-logview	2009-12-09 14:54:53.596152000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 
 #
 # Log file viewer for gDesklets.
diff -u gDesklets-0.36.1.old/gdesklets-migration-tool gDesklets-0.36.1/gdesklets-migration-tool
--- gDesklets-0.36.1.old/contrib/gdesklets-migration-tool	2009-12-09 14:48:14.041612000 +0800
+++ gDesklets-0.36.1/contrib/gdesklets-migration-tool	2009-12-09 14:55:13.364014000 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # TODO:
 #   -Add selectors for previous and current gDesklets versions
diff -u gDesklets-0.36.1.old/gdesklets-shell gDesklets-0.36.1/gdesklets-shell
--- gDesklets-0.36.1.old/gdesklets-shell	2009-12-09 14:48:14.041739000 +0800
+++ gDesklets-0.36.1/gdesklets-shell	2009-12-09 14:55:25.715918000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 
 from utils import i18n
 import __builtin__
diff -u gDesklets-0.36.1.old/test-control.py gDesklets-0.36.1/test-control.py
--- gDesklets-0.36.1.old/test-control.py	2009-12-09 14:48:14.053620000 +0800
+++ gDesklets-0.36.1/test-control.py	2009-12-09 14:55:50.883623000 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # A simple script to test controls interactively.
 
 from plugin.Interface import Interface
