--- openjade-1.3.2/config/configure.in.bak	Tue Jan 11 03:00:04 2022
+++ openjade-1.3.2/config/configure.in	Tue Jan 11 03:01:01 2022
@@ -12,9 +12,8 @@
 dnl  Credits: this autoconf script was largely "inspired" <g> by the
 dnl  autoconf script around SP made by Henry Thompson.
 dnl
-AC_INIT(dsssl)
+AC_INIT( openjade, 1.3.2)
 AC_CONFIG_AUX_DIR(config)
-AM_INIT_AUTOMAKE( openjade, 1.3.2)
 TOP=`pwd`
 AC_SUBST(TOP)
 dnl
--- openjade-1.3.2/include/config.h.in.bak	Tue Jan 11 03:22:53 2022
+++ openjade-1.3.2/include/config.h.in	Tue Jan 11 03:23:27 2022
@@ -18,8 +18,8 @@
 #endif
 #endif
 
-#define OPENJADE_PACKAGE PACKAGE
-#define OPENJADE_VERSION VERSION
+#define OPENJADE_PACKAGE "@PACKAGE_NAME@"
+#define OPENJADE_VERSION "@PACKAGE_VERSION@"
 #ifndef OPENJADE_MESSAGE_DOMAIN
 #define OPENJADE_MESSAGE_DOMAIN ""
 #endif
