diff -wpruN '--exclude=*.orig' a~/doc/Makefile.in a/doc/Makefile.in
--- a~/doc/Makefile.in	1970-01-01 00:00:00
+++ a/doc/Makefile.in	1970-01-01 00:00:00
@@ -317,7 +317,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-@WITH_DOCBOOK_TRUE@dist_man_MANS = xmlwf.1
+dist_man_MANS = xmlwf.1
 EXTRA_DIST = \
     expat.png \
     reference.html \
