#
# Submitted upstream 2017-01-23, will be fixed in the next release.
#
diff --git a/dwarfdump/dwarf_tsearchbal.c b/dwarfdump/dwarf_tsearchbal.c
index 06628f8..08d8ff3 100644
--- a/dwarfdump/dwarf_tsearchbal.c
+++ b/dwarfdump/dwarf_tsearchbal.c
@@ -522,6 +522,7 @@ dwarf_tsearch(const void *key, void **headin,
         }
         root = allocate_ts_entry(key);
         if(!root) {
+            free(head);
             return NULL;
         }
         head->rlink = root;
