--- gnome-applets-2.23.90.old/accessx-status/applet.c	2008-08-28 15:49:16.141662000 +0100
+++ gnome-applets-2.23.90/accessx-status/applet.c	2008-08-28 15:49:25.404358000 +0100
@@ -379,7 +379,7 @@
 	PangoLayout *layout;
 	PangoRectangle ink, logic;
 	PangoContext *pango_context;
-	GdkColormap *cmap;
+	GdkColormap *cmap = NULL;
 	GdkGC       *gc;
 	GdkVisual   *visual = gdk_visual_get_best ();
 	gint w = gdk_pixbuf_get_width (base);
