diff -ur system-config-printer-1.5.12/applet.py system-config-printer-py3/applet.py
--- system-config-printer-1.5.12/applet.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/applet.py	2020-03-11 13:46:31.475660069 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014 Red Hat, Inc.
 ## Author: Tim Waugh <twaugh@redhat.com>
diff -ur system-config-printer-1.5.12/asyncconn.py system-config-printer-py3/asyncconn.py
--- system-config-printer-1.5.12/asyncconn.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/asyncconn.py	2020-03-11 13:46:31.462683776 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2012, 2013 Red Hat, Inc.
 ## Copyright (C) 2008 Novell, Inc.
diff -ur system-config-printer-1.5.12/asyncipp.py system-config-printer-py3/asyncipp.py
--- system-config-printer-1.5.12/asyncipp.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/asyncipp.py	2020-03-11 13:46:31.473695625 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2015 Red Hat, Inc.
 ## Copyright (C) 2008 Novell, Inc.
diff -ur system-config-printer-1.5.12/asyncpk1.py system-config-printer-py3/asyncpk1.py
--- system-config-printer-1.5.12/asyncpk1.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/asyncpk1.py	2020-03-11 13:46:31.449979658 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2012, 2013, 2014 Red Hat, Inc.
 ## Copyright (C) 2008 Novell, Inc.
diff -ur system-config-printer-1.5.12/authconn.py system-config-printer-py3/authconn.py
--- system-config-printer-1.5.12/authconn.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/authconn.py	2020-03-11 13:46:31.464297151 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2011, 2013, 2014 Red Hat, Inc.
 ## Author: Tim Waugh <twaugh@redhat.com>
diff -ur system-config-printer-1.5.12/check-device-ids.py system-config-printer-py3/check-device-ids.py
--- system-config-printer-1.5.12/check-device-ids.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/check-device-ids.py	2020-03-11 13:46:31.472954285 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## check-device-ids
 
diff -ur system-config-printer-1.5.12/cupshelpers/installdriver.py system-config-printer-py3/cupshelpers/installdriver.py
--- system-config-printer-1.5.12/cupshelpers/installdriver.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/cupshelpers/installdriver.py	2020-03-11 13:46:31.460494058 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/cupshelpers/openprinting.py system-config-printer-py3/cupshelpers/openprinting.py
--- system-config-printer-1.5.12/cupshelpers/openprinting.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/cupshelpers/openprinting.py	2020-03-11 13:46:31.461667781 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/cupshelpers/ppds.py system-config-printer-py3/cupshelpers/ppds.py
--- system-config-printer-1.5.12/cupshelpers/ppds.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/cupshelpers/ppds.py	2020-03-11 13:46:31.461250203 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/cupshelpers/xmldriverprefs.py system-config-printer-py3/cupshelpers/xmldriverprefs.py
--- system-config-printer-1.5.12/cupshelpers/xmldriverprefs.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/cupshelpers/xmldriverprefs.py	2020-03-11 13:46:31.460174572 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/debug.py system-config-printer-py3/debug.py
--- system-config-printer-1.5.12/debug.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/debug.py	2020-03-11 13:46:31.463289923 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2008, 2010 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/dnssdresolve.py system-config-printer-py3/dnssdresolve.py
--- system-config-printer-1.5.12/dnssdresolve.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/dnssdresolve.py	2020-03-11 13:46:31.463003822 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2010, 2011, 2012, 2013, 2014 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/errordialogs.py system-config-printer-py3/errordialogs.py
--- system-config-printer-1.5.12/errordialogs.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/errordialogs.py	2020-03-11 13:46:31.469031467 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/firewallsettings.py system-config-printer-py3/firewallsettings.py
--- system-config-printer-1.5.12/firewallsettings.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/firewallsettings.py	2020-03-11 13:46:31.475286746 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/gtkinklevel.py system-config-printer-py3/gtkinklevel.py
--- system-config-printer-1.5.12/gtkinklevel.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/gtkinklevel.py	2020-03-11 13:46:31.469796230 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2009, 2010, 2012 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/gui.py system-config-printer-py3/gui.py
--- system-config-printer-1.5.12/gui.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/gui.py	2020-03-11 13:46:31.465669590 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/install-printerdriver.py system-config-printer-py3/install-printerdriver.py
--- system-config-printer-1.5.12/install-printerdriver.py	2019-09-30 19:56:24.000000000 +0000
+++ system-config-printer-py3/install-printerdriver.py	2020-03-11 13:46:31.473232880 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 import gi
 gi.require_version('PackageKitGlib', '1.0')
diff -ur system-config-printer-1.5.12/installpackage.py system-config-printer-py3/installpackage.py
--- system-config-printer-1.5.12/installpackage.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/installpackage.py	2020-03-11 13:46:31.465920551 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/killtimer.py system-config-printer-py3/killtimer.py
--- system-config-printer-1.5.12/killtimer.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/killtimer.py	2020-03-11 13:46:31.466672484 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/monitor.py system-config-printer-py3/monitor.py
--- system-config-printer-1.5.12/monitor.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/monitor.py	2020-03-11 13:46:31.470303277 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2011, 2012, 2014 Red Hat, Inc.
 ## Author: Tim Waugh <twaugh@redhat.com>
diff -ur system-config-printer-1.5.12/newprinter.py system-config-printer-py3/newprinter.py
--- system-config-printer-1.5.12/newprinter.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/newprinter.py	2020-03-11 13:46:31.468713779 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/PhysicalDevice.py system-config-printer-py3/PhysicalDevice.py
--- system-config-printer-1.5.12/PhysicalDevice.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/PhysicalDevice.py	2020-03-11 13:46:31.470653416 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2008, 2009, 2010, 2012, 2014 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/ppdcache.py system-config-printer-py3/ppdcache.py
--- system-config-printer-1.5.12/ppdcache.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/ppdcache.py	2020-03-11 13:46:31.474299281 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2010, 2011, 2012, 2013, 2014 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/ppdippstr.py system-config-printer-py3/ppdippstr.py
--- system-config-printer-1.5.12/ppdippstr.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/ppdippstr.py	2020-03-11 13:46:31.462034405 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/ppdsloader.py system-config-printer-py3/ppdsloader.py
--- system-config-printer-1.5.12/ppdsloader.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/ppdsloader.py	2020-03-11 13:46:31.459742945 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/printerproperties.py system-config-printer-py3/printerproperties.py
--- system-config-printer-1.5.12/printerproperties.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/printerproperties.py	2020-03-11 13:46:31.472288393 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/profile-ppds.py system-config-printer-py3/profile-ppds.py
--- system-config-printer-1.5.12/profile-ppds.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/profile-ppds.py	2020-03-11 13:46:31.474975208 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 import cups
 import cupshelpers
 import hotshot
diff -ur system-config-printer-1.5.12/pysmb.py system-config-printer-py3/pysmb.py
--- system-config-printer-1.5.12/pysmb.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/pysmb.py	2020-03-11 13:46:31.470946359 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 ## CUPS backend
diff -ur system-config-printer-1.5.12/scp-dbus-service.py system-config-printer-py3/scp-dbus-service.py
--- system-config-printer-1.5.12/scp-dbus-service.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/scp-dbus-service.py	2020-03-11 13:46:31.474690624 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/serversettings.py system-config-printer-py3/serversettings.py
--- system-config-printer-1.5.12/serversettings.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/serversettings.py	2020-03-11 13:46:31.466354589 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/smburi.py system-config-printer-py3/smburi.py
--- system-config-printer-1.5.12/smburi.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/smburi.py	2020-03-11 13:46:31.462347048 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/statereason.py system-config-printer-py3/statereason.py
--- system-config-printer-1.5.12/statereason.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/statereason.py	2020-03-11 13:46:31.463611411 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2007, 2008, 2009, 2010, 2012, 2013, 2014 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/system-config-printer.py system-config-printer-py3/system-config-printer.py
--- system-config-printer-1.5.12/system-config-printer.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/system-config-printer.py	2020-03-11 13:46:31.465344820 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
diff -ur system-config-printer-1.5.12/test_PhysicalDevice.py system-config-printer-py3/test_PhysicalDevice.py
--- system-config-printer-1.5.12/test_PhysicalDevice.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/test_PhysicalDevice.py	2020-03-11 13:46:31.463897666 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2015 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/timedops.py system-config-printer-py3/timedops.py
--- system-config-printer-1.5.12/timedops.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/timedops.py	2020-03-11 13:46:31.471572502 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2008, 2009, 2010, 2012, 2014 Red Hat, Inc.
 ## Authors:
diff -ur system-config-printer-1.5.12/troubleshoot/__init__.py system-config-printer-py3/troubleshoot/__init__.py
--- system-config-printer-1.5.12/troubleshoot/__init__.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/__init__.py	2020-03-11 13:46:31.455538470 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/base.py system-config-printer-py3/troubleshoot/base.py
--- system-config-printer-1.5.12/troubleshoot/base.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/base.py	2020-03-11 13:46:31.452511380 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckLocalServerPublishing.py system-config-printer-py3/troubleshoot/CheckLocalServerPublishing.py
--- system-config-printer-1.5.12/troubleshoot/CheckLocalServerPublishing.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckLocalServerPublishing.py	2020-03-11 13:46:31.457843649 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckNetworkServerSanity.py system-config-printer-py3/troubleshoot/CheckNetworkServerSanity.py
--- system-config-printer-1.5.12/troubleshoot/CheckNetworkServerSanity.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckNetworkServerSanity.py	2020-03-11 13:46:31.450745033 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckPPDSanity.py system-config-printer-py3/troubleshoot/CheckPPDSanity.py
--- system-config-printer-1.5.12/troubleshoot/CheckPPDSanity.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckPPDSanity.py	2020-03-11 13:46:31.458997235 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckPrinterSanity.py system-config-printer-py3/troubleshoot/CheckPrinterSanity.py
--- system-config-printer-1.5.12/troubleshoot/CheckPrinterSanity.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckPrinterSanity.py	2020-03-11 13:46:31.453912538 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckSELinux.py system-config-printer-py3/troubleshoot/CheckSELinux.py
--- system-config-printer-1.5.12/troubleshoot/CheckSELinux.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckSELinux.py	2020-03-11 13:46:31.451121904 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/CheckUSBPermissions.py system-config-printer-py3/troubleshoot/CheckUSBPermissions.py
--- system-config-printer-1.5.12/troubleshoot/CheckUSBPermissions.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/CheckUSBPermissions.py	2020-03-11 13:46:31.457525778 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ChooseNetworkPrinter.py system-config-printer-py3/troubleshoot/ChooseNetworkPrinter.py
--- system-config-printer-1.5.12/troubleshoot/ChooseNetworkPrinter.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ChooseNetworkPrinter.py	2020-03-11 13:46:31.459336546 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ChoosePrinter.py system-config-printer-py3/troubleshoot/ChoosePrinter.py
--- system-config-printer-1.5.12/troubleshoot/ChoosePrinter.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ChoosePrinter.py	2020-03-11 13:46:31.454529078 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/DeviceListed.py system-config-printer-py3/troubleshoot/DeviceListed.py
--- system-config-printer-1.5.12/troubleshoot/DeviceListed.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/DeviceListed.py	2020-03-11 13:46:31.457189755 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ErrorLogCheckpoint.py system-config-printer-py3/troubleshoot/ErrorLogCheckpoint.py
--- system-config-printer-1.5.12/troubleshoot/ErrorLogCheckpoint.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ErrorLogCheckpoint.py	2020-03-11 13:46:31.451866437 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ErrorLogFetch.py system-config-printer-py3/troubleshoot/ErrorLogFetch.py
--- system-config-printer-1.5.12/troubleshoot/ErrorLogFetch.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ErrorLogFetch.py	2020-03-11 13:46:31.458209865 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ErrorLogParse.py system-config-printer-py3/troubleshoot/ErrorLogParse.py
--- system-config-printer-1.5.12/troubleshoot/ErrorLogParse.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ErrorLogParse.py	2020-03-11 13:46:31.454221846 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/Locale.py system-config-printer-py3/troubleshoot/Locale.py
--- system-config-printer-1.5.12/troubleshoot/Locale.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/Locale.py	2020-03-11 13:46:31.453244804 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/LocalOrRemote.py system-config-printer-py3/troubleshoot/LocalOrRemote.py
--- system-config-printer-1.5.12/troubleshoot/LocalOrRemote.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/LocalOrRemote.py	2020-03-11 13:46:31.456484205 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/NetworkCUPSPrinterShared.py system-config-printer-py3/troubleshoot/NetworkCUPSPrinterShared.py
--- system-config-printer-1.5.12/troubleshoot/NetworkCUPSPrinterShared.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/NetworkCUPSPrinterShared.py	2020-03-11 13:46:31.452867568 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/PrinterStateReasons.py system-config-printer-py3/troubleshoot/PrinterStateReasons.py
--- system-config-printer-1.5.12/troubleshoot/PrinterStateReasons.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/PrinterStateReasons.py	2020-03-11 13:46:31.455178617 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/PrintTestPage.py system-config-printer-py3/troubleshoot/PrintTestPage.py
--- system-config-printer-1.5.12/troubleshoot/PrintTestPage.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/PrintTestPage.py	2020-03-11 13:46:31.458675600 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/QueueNotEnabled.py system-config-printer-py3/troubleshoot/QueueNotEnabled.py
--- system-config-printer-1.5.12/troubleshoot/QueueNotEnabled.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/QueueNotEnabled.py	2020-03-11 13:46:31.450423917 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/QueueRejectingJobs.py system-config-printer-py3/troubleshoot/QueueRejectingJobs.py
--- system-config-printer-1.5.12/troubleshoot/QueueRejectingJobs.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/QueueRejectingJobs.py	2020-03-11 13:46:31.454850109 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/RemoteAddress.py system-config-printer-py3/troubleshoot/RemoteAddress.py
--- system-config-printer-1.5.12/troubleshoot/RemoteAddress.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/RemoteAddress.py	2020-03-11 13:46:31.456139171 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/SchedulerNotRunning.py system-config-printer-py3/troubleshoot/SchedulerNotRunning.py
--- system-config-printer-1.5.12/troubleshoot/SchedulerNotRunning.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/SchedulerNotRunning.py	2020-03-11 13:46:31.453580772 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/ServerFirewalled.py system-config-printer-py3/troubleshoot/ServerFirewalled.py
--- system-config-printer-1.5.12/troubleshoot/ServerFirewalled.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/ServerFirewalled.py	2020-03-11 13:46:31.451431014 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/Shrug.py system-config-printer-py3/troubleshoot/Shrug.py
--- system-config-printer-1.5.12/troubleshoot/Shrug.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/Shrug.py	2020-03-11 13:46:31.456848961 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/VerifyPackages.py system-config-printer-py3/troubleshoot/VerifyPackages.py
--- system-config-printer-1.5.12/troubleshoot/VerifyPackages.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/VerifyPackages.py	2020-03-11 13:46:31.452203130 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/troubleshoot/Welcome.py system-config-printer-py3/troubleshoot/Welcome.py
--- system-config-printer-1.5.12/troubleshoot/Welcome.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/troubleshoot/Welcome.py	2020-03-11 13:46:31.455830266 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Printing troubleshooter
 
diff -ur system-config-printer-1.5.12/udev/udev-add-printer system-config-printer-py3/udev/udev-add-printer
--- system-config-printer-1.5.12/udev/udev-add-printer	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/udev/udev-add-printer	2020-03-11 13:46:31.472608242 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3 -sB
+#!/usr/bin/python3.5 -sB
 
 ## udev-add-printer
 
diff -ur system-config-printer-1.5.12/userdefault.py system-config-printer-py3/userdefault.py
--- system-config-printer-1.5.12/userdefault.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/userdefault.py	2020-03-11 13:46:31.473994586 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## Copyright (C) 2006, 2007, 2008, 2010, 2012, 2014 Red Hat, Inc.
 ## Author: Tim Waugh <twaugh@redhat.com>
diff -ur system-config-printer-1.5.12/xml/validate.py system-config-printer-py3/xml/validate.py
--- system-config-printer-1.5.12/xml/validate.py	2019-09-16 14:55:16.000000000 +0000
+++ system-config-printer-py3/xml/validate.py	2020-03-11 13:46:31.471253558 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/python3.5
 
 ## system-config-printer
 
