--- system-config-printer-1.5.18/cupshelpers/installdriver.py.orig
+++ system-config-printer-1.5.18/cupshelpers/installdriver.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python3
 
 ## system-config-printer
 
--- system-config-printer-1.5.18/cupshelpers/openprinting.py.orig
+++ system-config-printer-1.5.18/cupshelpers/openprinting.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python3
 
 ## system-config-printer
 
--- system-config-printer-1.5.18/cupshelpers/ppds.py.orig
+++ system-config-printer-1.5.18/cupshelpers/ppds.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python3
 
 ## system-config-printer
 
--- system-config-printer-1.5.18/cupshelpers/xmldriverprefs.py.orig
+++ system-config-printer-1.5.18/cupshelpers/xmldriverprefs.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python3
 
 ## system-config-printer
 
