--- system-config-printer-1.0.9.p3/my-default-printer.in	2008-08-28 14:30:27.000000000 +0100
+++ system-config-printer-1.0.9.p3/my-default-printer.in	2008-11-13 11:31:47.197136000 +0000
@@ -1,3 +1,2 @@
 #!/bin/sh
-prefix=@prefix@
-exec @datarootdir@/@PACKAGE@/my-default-printer.py "$@"
+exec @libdir@/@PACKAGE@/my-default-printer.py "$@"
diff -urN ./system-config-printer-applet.in ../system-config-printer-1.0.9.p3/system-config-printer-applet.in
--- system-config-printer-1.0.9.p3/system-config-printer-applet.in	2008-08-28 14:30:27.000000000 +0100
+++ system-config-printer-1.0.9.p3/system-config-printer-applet.in	2008-11-13 11:30:59.924220000 +0000
@@ -1,3 +1,2 @@
 #!/bin/sh
-prefix=@prefix@
-exec @datarootdir@/@PACKAGE@/applet.py "$@"
+exec @libdir@/@PACKAGE@/applet.py "$@"
diff -urN ./system-config-printer.in ../system-config-printer-1.0.9.p3/system-config-printer.in
--- system-config-printer-1.0.9.p3/system-config-printer.in	2008-08-28 14:30:27.000000000 +0100
+++ system-config-printer-1.0.9.p3/system-config-printer.in	2008-11-13 11:31:16.561683000 +0000
@@ -1,3 +1,2 @@
 #!/bin/sh
-prefix=@prefix@
-exec @datarootdir@/@PACKAGE@/system-config-printer.py "$@"
+exec @libdir@/@PACKAGE@/system-config-printer.py "$@"
