diff -urN system-config-printer-1.0.16/Makefile.am ../SUNWcups-manager-1.0.16.hacked/system-config-printer-1.0.16/Makefile.am
--- system-config-printer-1.4.8/Makefile.am.~1~	2016-09-14 15:42:40.567222700 +0300
+++ system-config-printer-1.4.8/Makefile.am	2016-09-14 15:44:47.242294642 +0300
@@ -233,15 +233,16 @@
 install-desktopDATA: $(desktop_DATA)
 	mkdir -p $(DESTDIR)$(desktopdir)
 	mkdir -p $(DESTDIR)$(desktopdir)
-	desktop-file-install $(DESKTOP_VENDOR)	\
+	desktop-file-install 	\
 	  --dir $(DESTDIR)$(desktopdir)		\
 	  --add-category System			\
 	  --add-category Settings		\
 	  --add-category HardwareSettings	\
 	  --add-category Printing		\
 	  --add-category GTK			\
+	  --add-only-show-in="MATE"		\
 	  system-config-printer.desktop
-	desktop-file-install $(DESKTOP_VENDOR)	\
+	desktop-file-install 	\
 	  --dir $(DESTDIR)$(autostartdir)	\
 	  --add-category System			\
 	  --add-category Monitor		\
diff -urN system-config-printer-1.0.16/print-applet.desktop.in ../SUNWcups-manager-1.0.16.hacked/system-config-printer-1.0.16/print-applet.desktop.in
--- system-config-printer-1.0.16/print-applet.desktop.in	2009-02-04 16:14:06.000000000 +0000
+++ ../SUNWcups-manager-1.0.16.hacked/system-config-printer-1.0.16/print-applet.desktop.in	2011-08-18 12:21:10.749136057 +0000
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 _Name=Print Queue Applet
 _Comment=System tray icon for managing print jobs
-Exec=system-config-printer-applet
+Exec=/usr/lib/system-config-printer/system-config-printer-applet
 Terminal=false
 Type=Application
 Icon=printer
--- system-config-printer-1.4.8/system-config-printer.desktop.in.~1~	2015-03-17 13:32:23.000000000 +0000
+++ system-config-printer-1.4.8/system-config-printer.desktop.in	2018-06-13 18:44:44.934043844 +0000
@@ -1,8 +1,9 @@
 [Desktop Entry]
-_Name=Print Settings
-_Comment=Configure printers
+_Name=Print Manager
+_Comment=Manage your printer queues and jobs
 Exec=system-config-printer
 Terminal=false
 Type=Application
 Icon=printer
 StartupNotify=true
+OnlyShowIn=MATE;
