--- llvm-15.0.6.src/tools/clang/tools/clang-shlib/CMakeLists.txt	2022-11-29 11:05:58.000000000 +0000
+++ llvm-15.0.6.src/tools/clang/tools/clang-shlib/CMakeLists.txt.new	2023-01-06 12:30:04.711112582 +0000
@@ -50,6 +50,6 @@
                   ${_DEPS})
 # Optimize function calls for default visibility definitions to avoid PLT and
 # reduce dynamic relocations.
-if (NOT APPLE AND NOT MINGW)
-  target_link_options(clang-cpp PRIVATE LINKER:-Bsymbolic-functions)
-endif()
+#if (NOT APPLE AND NOT MINGW)
+#  target_link_options(clang-cpp PRIVATE LINKER:-Bsymbolic-functions)
+#endif()
