diff -ruN libunique-1.1.6.orig/doc/reference/Makefile.am libunique-1.1.6/doc/reference/Makefile.am
--- libunique-1.1.6.orig/doc/reference/Makefile.am      2010-01-05 00:02:16.493854262 +0000
+++ libunique-1.1.6/doc/reference/Makefile.am   2010-01-05 00:03:47.003603531 +0000
@@ -34,7 +34,7 @@
 
 # Extra options to supply to gtkdoc-fixref. Not normally needed.
 # e.g. FIXXREF_OPTIONS=--extra-dir=../gdk-pixbuf/html --extra-dir=../gdk/html
-FIXXREF_OPTIONS =
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
 
 # Used for dependencies. The docs will be rebuilt if any of these change.
 # e.g. HFILE_GLOB=$(top_srcdir)/gtk/*.h
