--- libunique-1.1.6/unique/bacon/uniquebackend-bacon.c.~1~	2015-03-17 16:24:19.457635546 +0300
+++ libunique-1.1.6/unique/bacon/uniquebackend-bacon.c	2015-03-17 16:24:32.791997755 +0300
@@ -186,7 +186,7 @@
 
   g_free (basename);
 
-  basename = g_strdup_printf ("%s.%s.%d", name, token, getpid ());
+  basename = g_strdup_printf ("%s.%s.%d", name, token, (int)getpid ());
   
   path = g_build_filename (tmpdir, basename, NULL);
 
--- libunique-1.1.6/unique/dbus/uniquebackend-dbus.c.~1~	2015-03-17 16:25:35.346917674 +0300
+++ libunique-1.1.6/unique/dbus/uniquebackend-dbus.c	2015-03-17 16:27:51.243028927 +0300
@@ -84,7 +84,6 @@
 static gboolean
 unique_backend_dbus_request_name (UniqueBackend *backend)
 {
-  UniqueBackendDBus *backend_dbus;
   const gchar *name;
   DBusGConnection *connection;
   DBusGProxy *proxy;
@@ -97,7 +96,6 @@
   if (!connection)
     return FALSE;
 
-  backend_dbus = UNIQUE_BACKEND_DBUS (backend);
 
   retval = TRUE;
   name = unique_backend_get_name (backend);
@@ -186,7 +184,6 @@
   GValueArray *data;
   gchar *cmd;
   gchar *resp;
-  gboolean res;
   GError *error;
   UniqueResponse response;
 
@@ -203,7 +200,7 @@
   resp = NULL;
  
   error = NULL;
-  res = org_gtk_UniqueApp_send_message (backend_dbus->proxy,
+  org_gtk_UniqueApp_send_message (backend_dbus->proxy,
                                         cmd, data, time_,
                                         &resp,
                                         &error);
