--- automat-24.8.0/tox.ini.orig
+++ automat-24.8.0/tox.ini
@@ -5,8 +5,8 @@
 
 [testenv]
 deps =
-    extras: graphviz>=0.4.9
-    extras: Twisted>=16.2.0
+    graphviz>=0.4.9
+    Twisted>=16.2.0
 
     mypy: mypy
     mypy: graphviz>=0.4.9
@@ -16,7 +16,7 @@
     pytest
 
 commands =
-    {extras,noextras}: coverage run --parallel --source src -m pytest -s -rfEsx src/automat/_test
+    coverage run --parallel --source src -m pytest -s -rfEsx src/automat/_test
     mypy: mypy {posargs:src/automat}
 
 depends =
