--- webkitgtk-2.10.4/Source/WebKit2/CMakeLists.txt.~1~	2015-11-11 10:42:51.000000000 +0300
+++ webkitgtk-2.10.4/Source/WebKit2/CMakeLists.txt	2015-12-11 08:37:48.768110988 +0300
@@ -777,6 +777,8 @@
 # Suppress unused parameter warnings for sources in WebKit2.
 ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-Wno-unused-parameter")
 
+ADD_TARGET_PROPERTIES(WebKit2 COMPILE_FLAGS "-DGTK_API_VERSION_2")
+
 if (WebKit2_VERSION_SCRIPT)
     ADD_TARGET_PROPERTIES(WebKit2 LINK_FLAGS "${WebKit2_VERSION_SCRIPT}")
 endif ()
