--- openvpn-2.4.1/src/openvpn/console_builtin.c.~1~	2017-04-05 23:22:38.118462490 +0300
+++ openvpn-2.4.1/src/openvpn/console_builtin.c	2017-04-05 23:24:15.915462559 +0300
@@ -240,7 +240,12 @@
     }
     else
     {
+#ifdef __sun
+	/* On Solaris/illumos getpass() returns up to 8 symbols */
+	char *gp = getpassphrase (prompt);
+#else
         char *gp = getpass(prompt);
+#endif
         if (gp)
         {
             strncpynt(input, gp, capacity);
