--- libgnomeui-2.24.5/libgnomeui/gnome-dateedit.c.orig	Fri Apr 28 19:30:02 2017
+++ libgnomeui-2.24.5/libgnomeui/gnome-dateedit.c	Fri Apr 28 19:30:47 2017
@@ -29,8 +29,6 @@
  * Author: Miguel de Icaza
  */
 
-#define _XOPEN_SOURCE
-
 #undef GTK_DISABLE_DEPRECATED
 
 #include <config.h>
--- libgnomeui-2.24.5/libgnomeui/gnome-client.c.orig	Fri Apr 28 19:29:51 2017
+++ libgnomeui-2.24.5/libgnomeui/gnome-client.c	Fri Apr 28 19:30:32 2017
@@ -23,8 +23,6 @@
   @NOTATION@
 */
 
-#define _XOPEN_SOURCE 500
-
 #include <config.h>
 #include <stdlib.h>
 #include <stdarg.h>
--- libgnomeui-2.24.5/libgnomeui/gnome-file-entry.c.orig	Fri Apr 28 19:29:44 2017
+++ libgnomeui-2.24.5/libgnomeui/gnome-file-entry.c	Fri Apr 28 19:30:19 2017
@@ -32,8 +32,6 @@
 
 #undef GTK_DISABLE_DEPRECATED
 
-#define _XOPEN_SOURCE 500
-
 #include <config.h>
 #include <glib.h>
 #include <sys/param.h>
