diff --git a/completions-core/Makefile.am b/completions-core/Makefile.am
index 3828893..ee5ba92 100644
--- a/completions-core/Makefile.am
+++ b/completions-core/Makefile.am
@@ -441,7 +441,15 @@ cross_platform = 2to3.bash \
 		zopfli.bash \
 		zopflipng.bash
 
-solaris_extra = pkgutil.bash
+solaris_extra = beadm \
+		bootadm \
+		dladm \
+		pfexec \
+		pkg \
+		svcadm \
+		zfs \
+		zones \
+		zpool
 
 bsd_extra = pkg_delete.bash
 
