Add bash completions from openindiana-completions project.

--- bash-completion-2.8/completions/Makefile.am	2018-03-17 10:16:03.000000000 +0000
+++ bash-completion-2.8/completions/Makefile.am	2018-12-02 01:32:54.847267869 +0000
@@ -433,7 +433,15 @@ bashcomp_DATA = 2to3 \
 		_yum \
 		yum-arch \
 		zopfli \
-		zopflipng
+		zopflipng \
+		beadm \
+		bootadm \
+		dladm \
+		pfexec \
+		pkg \
+		zfs \
+		zones \
+		zpool
 
 EXTRA_DIST = $(bashcomp_DATA)
 
