Add bash completions from openindiana-completions project.

--- bash-completion-2.12.0/completions/Makefile.am.orig	2024-02-21 07:55:15.177887345 +0100
+++ bash-completion-2.12.0/completions/Makefile.am	2024-03-04 08:03:14.045199103 +0100
@@ -498,7 +498,15 @@
 		_yum \
 		yum-arch \
 		zopfli \
-		zopflipng
+		zopflipng \
+		beadm \
+		bootadm \
+		dladm \
+		pfexec \
+		pkg \
+		zfs \
+		zones \
+		zpool
 
 EXTRA_DIST = $(bashcomp_DATA)
 
