diff --git a/completions/Makefile.am b/completions/Makefile.am
index 726252e..2d8b0cb 100644
--- a/completions/Makefile.am
+++ b/completions/Makefile.am
@@ -1,6 +1,20 @@
 # We include all the files in the tarball
 
 bashcompdir = $(datadir)/$(PACKAGE)/completions
-bashcomp_DATA = README.md
+bashcomp_DATA = README.md \
+	beadm.bash \
+	bootadm.bash \
+	dladm.bash \
+	flowadm.bash \
+	fmadm.bash \
+	ipadm.bash \
+	pfexec.bash \
+	pkg.bash \
+	svcadm.bash \
+	zfs.bash \
+	zlogin.bash \
+	zoneadm.bash \
+	zonecfg.bash \
+	zpool.bash
 
 EXTRA_DIST = $(bashcomp_DATA)
diff --git a/startup/Makefile.am b/startup/Makefile.am
index 0da318b..114478c 100644
--- a/startup/Makefile.am
+++ b/startup/Makefile.am
@@ -1,4 +1,5 @@
 startupdir = $(datadir)/$(PACKAGE)/startup
-startup_DATA = README.md
+startup_DATA = README.md \
+	zones.bash
 
 EXTRA_DIST = $(startup_DATA)
