--- bash-completion-2.16.0/completions/tar.orig
+++ bash-completion-2.16.0/completions/tar
@@ -143,7 +143,7 @@
 
             _comp_cmd_gtar__parse_help_line "$str" "$arg"
         fi
-    done <<<"$(tar --help)"
+    done <<<"$(gtar --help)"
 
     long_opts="\
         $long_arg_none        $long_arg_opt        $long_arg_req"
@@ -156,7 +156,7 @@
 {
     local -a warnings=()
     local lines line
-    _comp_split -l lines "$(LC_ALL=C exec tar --warning= 2>&1)"
+    _comp_split -l lines "$(LC_ALL=C exec gtar --warning= 2>&1)"
     for line in "${line[@]}"; do
         if [[ $line =~ ^[[:blank:]]*-[[:blank:]]*[\`\']([a-zA-Z0-9-]+)\'$ ]]; then
             warnings+=("${BASH_REMATCH[1]}" "no-${BASH_REMATCH[1]}")
