--- percona-server-8.0.13-3/CMakeLists.txt	2018-12-19 17:31:11.000000000 +0000
+++ percona-server-8.0.13-3/CMakeLists.txt.new	2019-01-12 21:58:19.483524728 +0000
@@ -800,9 +800,7 @@ ADD_DEFINITIONS(-DHAVE_OPENSSL)
 ADD_DEFINITIONS(-D__STDC_LIMIT_MACROS)     # Enable C99 limit macros
 ADD_DEFINITIONS(-D__STDC_FORMAT_MACROS)    # Enable C99 printf format macros
 ADD_DEFINITIONS(-D_USE_MATH_DEFINES)       # Get access to M_PI, M_E, etc. in math.h
-IF(APPLE)
-  ADD_DEFINITIONS(-D__STDC_WANT_LIB_EXT1__=1) # Get access to memset_s
-ENDIF()
+ADD_DEFINITIONS(-D__STDC_WANT_LIB_EXT1__=1) # Get access to memset_s
 
 OPTION(ENABLE_EXPERIMENT_SYSVARS "Expose ussually hidden system variables to allow experiments" OFF)
 IF(ENABLE_EXPERIMENT_SYSVARS)
