--- planner-0.14.2-orig/python/Makefile.am	2006-06-11 08:53:15.000000000 -0700
+++ planner-0.14.2/python/Makefile.am	2008-09-05 13:48:03.947177662 -0700
@@ -10,13 +10,13 @@
 
 pkgpyexec_LTLIBRARIES = planner.la plannerui.la
 
-planner_la_CFLAGS = -Wno-missing-declarations -Wno-missing-prototypes
+planner_la_CFLAGS = 
 planner_la_LDFLAGS = -module -avoid-version -export-symbols-regex initplanner
 planner_la_LIBADD = $(LIBPLANNER_LIBS) $(top_builddir)/libplanner/libplanner-1.la
 planner_la_SOURCES = plannermodule.c
 nodist_planner_la_SOURCES = planner.c
 
-plannerui_la_CFLAGS = -Wno-missing-declarations -Wno-missing-prototypes
+plannerui_la_CFLAGS = 
 plannerui_la_LDFLAGS = -module -avoid-version -export-symbols-regex initplannerui
 plannerui_la_LIBADD = $(PLANNER_LIBS) $(top_builddir)/libplanner/libplanner-1.la
 plannerui_la_SOURCES = planneruimodule.c
