https://github.com/pytest-dev/pytest-xprocess/issues/155

--- pytest-xprocess-1.0.2/tox.ini.orig
+++ pytest-xprocess-1.0.2/tox.ini
@@ -20,7 +20,7 @@
     coverage
 commands=
     coverage run -m pytest -v
-    coverage report --omit="*/.tox/*,*/test_functional_workflow.py" --fail-under=90
+    coverage report --omit="*/.tox/*,*/test_functional_work_flow.py" --fail-under=90
 usedevelop = True
 
 [testenv:linting]
