--- fsvs-1.2.5/configure.in.orig	2013-08-02 18:24:41.482817815 +0400
+++ fsvs-1.2.5/configure.in	2013-08-02 18:27:10.595330042 +0400
@@ -100,6 +100,13 @@
 			LIBDIRS="$LIBDIRS $withval"
 		])
 
+AC_ARG_WITH(aprutillib,
+		AC_HELP_STRING([--with-aprutillib=PATH],
+			[Specify a directory containing APR libraries.]),
+		[
+			LIBDIRS="$LIBDIRS $withval"
+		])
+
 AC_ARG_WITH(svnlib,
 		AC_HELP_STRING([--with-svnlib=PATH],
 			[Specify a directory containing subversion libraries.]),
