--- fsvs-1.2.5/configure.in	2013-08-07 18:41:11.340013489 +0400
+++ fsvs-1.2.5.patched/configure.in	2013-08-07 18:42:52.078710996 +0400
@@ -3,7 +3,7 @@
 
 AC_PREREQ(2.60)
 AC_INIT(fsvs, 
-	[esyscmd(make --quiet --no-print-directory -f Makefile.in version-nnl 2>/dev/null)], 
+	[esyscmd(gmake --quiet --no-print-directory -f Makefile.in version-nnl 2>/dev/null)], 
 	http://fsvs.tigris.org/)
 AC_GNU_SOURCE
 
