diff -ruN gconf-editor-2.23.91.orig/data/gconf-editor.desktop.in.in gconf-editor-2.23.91/data/gconf-editor.desktop.in.in
--- gconf-editor-2.23.91.orig/data/gconf-editor.desktop.in.in      2008-09-01 16:02:40.988509831 +0
100
+++ gconf-editor-2.23.91/data/gconf-editor.desktop.in.in   2008-09-01 16:03:33.105449238 +0100
@@ -7,6 +7,7 @@ Terminal=false
 Type=Application
 Icon=gconf-editor
 StartupNotify=true
+NoDisplay=true
 Categories=GNOME;GTK;System;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gconf-editor
