https://rt.cpan.org/Public/Bug/Display.html?id=133363

--- Crypt-DES-2.09/_des.h.orig
+++ Crypt-DES-2.09/_des.h
@@ -5,3 +5,5 @@
 void perl_des_crypt( des_cblock in, des_cblock out, des_ks key, int encrypt );
 void perl_des_expand_key( des_user_key userKey, des_ks key );
 
+void perl_des_expand_key(des_user_key userKey, des_ks ks);
+void perl_des_crypt( des_cblock input, des_cblock output, des_ks ks, int encrypt );
