--- a/man/iceauth.man	Thu Apr 23 12:11:32 2015
+++ b/man/iceauth.man	Thu Apr 23 12:11:45 2015
@@ -27,7 +27,7 @@
 iceauth \- ICE authority file utility
 .SH SYNOPSIS
 .B iceauth
-[ \fB\-f\fP \fIauthfile\fP ] [ \fB\-vqib\fP ] [ \fIcommand arg ...\fP ]
+[ \fB\-f\fP \fIauthfile\fP ] [ \fB\-vqibuV\fP ] [ \fIcommand arg ...\fP ]
 .SH DESCRIPTION
 .PP
 The \fIiceauth\fP program is used to edit and display the authorization
