--- aoo-4.1.0/main/configure.in.~1~	2014-07-28 16:19:23.551638775 +0400
+++ aoo-4.1.0/main/configure.in	2014-07-28 16:19:44.096425930 +0400
@@ -1712,13 +1712,13 @@
    else
       AC_MSG_RESULT([checked (gcc $_gcc_version)])
    fi
-   if test "$_os" = "SunOS"; then
-      AC_MSG_CHECKING([gcc linker])
-      if $CC -Wl,--version 2>&1 |head -n 1| grep -v GNU > /dev/null;then
-          AC_MSG_ERROR([failed (not GNU ld). Use GNU ld instead of Sun ld on Solaris])
-      fi
-      AC_MSG_RESULT([ok (GNU ld)])
-   fi
+#   if test "$_os" = "SunOS"; then
+#      AC_MSG_CHECKING([gcc linker])
+#      if $CC -Wl,--version 2>&1 |head -n 1| grep -v GNU > /dev/null;then
+#          AC_MSG_ERROR([failed (not GNU ld). Use GNU ld instead of Sun ld on Solaris])
+#      fi
+#      AC_MSG_RESULT([ok (GNU ld)])
+#   fi
 fi 
 AC_SUBST(GCCVER)
 
