--- aoo-4.1.1/main/sc/util/makefile.pmk	2014-02-25 12:27:57.000000000 +0400
+++ aoo-4.1.1/main/sc/util/makefile.pmk	2014-10-03 18:40:42.540927212 +0400
@@ -25,4 +25,4 @@
 # define SC_DLLIMPLEMENTATION (see @ scdllapi.h)
 CDEFS += -DSC_DLLIMPLEMENTATION
 
-VISIBILITY_HIDDEN=TRUE
+VISIBILITY_HIDDEN=
--- aoo-4.1.1/main/sd/util/makefile.pmk.~1~	2014-10-03 10:20:02.647625184 +0400
+++ aoo-4.1.1/main/sd/util/makefile.pmk	2014-10-03 10:20:50.401869732 +0400
@@ -25,4 +25,4 @@
 # define SD_DLLIMPLEMENTATION (see @ sddllapi.h)
 CDEFS += -DSD_DLLIMPLEMENTATION
 
-VISIBILITY_HIDDEN=TRUE
+VISIBILITY_HIDDEN=
