--- aoo-4.1.0/main/sd/source/ui/slidesorter/controller/makefile.mk.~1~	2014-07-25 11:15:19.718921349 +0400
+++ aoo-4.1.0/main/sd/source/ui/slidesorter/controller/makefile.mk	2014-07-25 11:15:27.983210418 +0400
@@ -35,6 +35,7 @@
 
 .INCLUDE :  settings.mk
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
+VISIBILITY_HIDDEN=
 
 # --- Files --------------------------------------------------------
 
--- aoo-4.1.0/main/sc/source/filter/xml/makefile.mk.~1~	2014-07-25 17:22:15.408787581 +0400
+++ aoo-4.1.0/main/sc/source/filter/xml/makefile.mk	2014-07-25 17:22:55.391760383 +0400
@@ -38,6 +38,8 @@
 
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
+VISIBILITY_HIDDEN=
+
 .INCLUDE :  scpre.mk
 .INCLUDE :  settings.mk
 .INCLUDE :  sc.mk
--- aoo-4.1.0/main/sc/source/core/data/makefile.mk.~1~	2014-07-25 14:56:11.379754439 +0400
+++ aoo-4.1.0/main/sc/source/core/data/makefile.mk	2014-07-25 14:56:23.294858821 +0400
@@ -39,6 +39,7 @@
 .INCLUDE :  sc.mk
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
+VISIBILITY_HIDDEN=
 # --- Files --------------------------------------------------------
 
 SLOFILES =  \
--- aoo-4.1.0/main/sc/source/core/tool/makefile.mk.~1~	2014-07-25 16:02:58.759143795 +0400
+++ aoo-4.1.0/main/sc/source/core/tool/makefile.mk	2014-07-25 16:03:37.665850344 +0400
@@ -38,6 +38,7 @@
 .INCLUDE :  settings.mk
 .INCLUDE :  sc.mk
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
+VISIBILITY_HIDDEN=
 
 # --- Files --------------------------------------------------------
 
