--- aoo-4.1.0/main/comphelper/inc/comphelper/locale.hxx.~1~	2014-07-23 12:12:53.241887403 +0400
+++ aoo-4.1.0/main/comphelper/inc/comphelper/locale.hxx	2014-07-23 12:26:56.868638074 +0400
@@ -70,10 +70,12 @@
         static const Locale& DE_CH();
         static const Locale& DE_AT();
         static const Locale& AR();
+#undef  CS
         static const Locale& CA();
         static const Locale& CS();
         static const Locale& DA();
         static const Locale& EL();
+#undef  ES
         static const Locale& ES();
         static const Locale& FI();
         static const Locale& FR();
--- aoo-4.1.0/main/connectivity/source/drivers/odbcbase/ODriver.cxx.~1~	2014-07-24 17:44:54.372402955 +0400
+++ aoo-4.1.0/main/connectivity/source/drivers/odbcbase/ODriver.cxx	2014-07-24 17:45:21.117640182 +0400
@@ -69,6 +69,7 @@
 		// Please take care when changing it.
 }
 
+#undef SS
 typedef Sequence< ::rtl::OUString > SS;
 //------------------------------------------------------------------------------
 SS ODBCDriver::getSupportedServiceNames_Static(  ) throw (RuntimeException)
