--- aoo-4.1.0/main/soltools/mkdepend/makefile.mk.~1~	2014-07-22 14:06:37.867115746 +0400
+++ aoo-4.1.0/main/soltools/mkdepend/makefile.mk	2014-07-22 14:07:03.301660002 +0400
@@ -64,7 +64,7 @@
 
 .IF "$(OS)"=="SOLARIS"
 #APP1STDLIBS+=-lstlport
-APP1STDLIBS+=-lCstd
+APP1STDLIBS+=-lstdc++
 .ENDIF
 
 
--- aoo-4.1.0/main/soltools/adjustvisibility/makefile.mk.~1~	2014-07-22 14:15:44.694918310 +0400
+++ aoo-4.1.0/main/soltools/adjustvisibility/makefile.mk	2014-07-22 14:15:51.234491888 +0400
@@ -46,7 +46,7 @@
 APP1RPATH   =   NONE
 
 #APP1STDLIBS+=-lstlport
-APP1STDLIBS+=-lCstd
+APP1STDLIBS+=-lstdc++
 
 .ENDIF "$(OS)"=="SOLARIS"
 
