--- aoo-4.1.0/main/sal/osl/unx/util.c.~1~	2014-07-22 16:19:14.702447603 +0400
+++ aoo-4.1.0/main/sal/osl/unx/util.c	2014-07-22 16:19:59.201718086 +0400
@@ -313,7 +313,7 @@
 #endif
 
 #if    ( defined(__GNUC__) && (defined(X86) || defined(X86_64)) )\
-    || ( defined(SOLARIS) && defined (__SUNPRO_C) && defined(__i386) )
+    || ( defined(SOLARIS) && (defined (__SUNPRO_C) || defined(__GNUC__)) && defined(__i386) )
 
 /* Safe default */
 int osl_isSingleCPU = 0;
--- aoo-4.1.0/main/sal/osl/unx/file.cxx.~1~	2014-07-22 16:11:30.346297137 +0400
+++ aoo-4.1.0/main/sal/osl/unx/file.cxx	2014-07-22 16:12:13.954038091 +0400
@@ -1085,14 +1085,14 @@
         // OS simultaneously pages in the rest); on other platforms, it remains
         // to be evaluated whether madvise or equivalent is available and
         // actually useful:
-#if defined MACOSX
+#if defined MACOSX || defined SOLARIS
         int e = posix_madvise(p, nLength, POSIX_MADV_WILLNEED);
         if (e != 0)
         {
             OSL_TRACE(
                 "posix_madvise(..., POSIX_MADV_WILLNEED) failed with %d", e);
         }
-#elif defined SOLARIS
+#elif defined notSOLARIS
         if (madvise(static_cast< caddr_t >(p), nLength, MADV_WILLNEED) != 0)
         {
             OSL_TRACE("madvise(..., MADV_WILLNEED) failed with %d", errno);
