--- aoo-4.1.0/main/cppu/source/uno/data.cxx.~1~	2014-08-12 12:58:22.017486428 +0400
+++ aoo-4.1.0/main/cppu/source/uno/data.cxx	2014-08-12 12:59:37.293556407 +0400
@@ -357,7 +357,7 @@
 
 #if defined(INTEL) \
     && (defined(__GNUC__) && (defined(LINUX) || defined(FREEBSD) || defined(OS2)) || defined(MACOSX) \
-        || defined(__SUNPRO_CC) && defined(SOLARIS))
+        || ( defined(__SUNPRO_CC) || defined(__GNUC__)) && defined(SOLARIS))
 #define MAX_ALIGNMENT_4
 #endif
 
