--- opal-3.6.4/plugins/video/H.261-vic/h261vic.cxx.old	2009-09-18 18:21:30.590839000 +0800
+++ opal-3.6.4/plugins/video/H.261-vic/h261vic.cxx	2009-09-18 18:22:17.867751000 +0800
@@ -509,7 +509,7 @@
       int wraptime = now ^ 0x80;
       u_char * ts = rvts;
       int k;
-      for (k = nblk; --k >= 0; ++ts) {
+      for (k = nblk; --k > 0; ++ts) {
         if (*ts == wraptime)
           *ts = (u_char)now;
       }
