--- tcsh-6.22.04/sh.c.orig	Mon Apr 26 16:54:25 2021
+++ tcsh-6.22.04/sh.c	Mon May 17 20:23:51 2021
@@ -173,17 +173,10 @@
     size_t len;
     int add_LOC = 1;
     int add_lang = 1;
-    char trypath[MAXPATHLEN];
-    struct stat st;
 
     if (path == NULL)
         return;
 
-    (void) xsnprintf(trypath, sizeof(trypath), "%s/C/LC_MESSAGES/tcsh.cat",
-	path);
-    if (stat(trypath, &st) == -1)
-	return;
-
     if ((old = getenv("NLSPATH")) != NULL)
         len = strlen(old) + 1;	/* don't forget the colon. */
     else
--- tcsh-6.22.04/sh.func.c.orig	Mon Apr 26 16:54:25 2021
+++ tcsh-6.22.04/sh.func.c	Mon May 17 20:26:57 2021
@@ -2671,6 +2671,8 @@
 
     if (adrof(STRcatalog) != NULL)
 	catalog = xasprintf("tcsh.%s", short2str(varval(STRcatalog)));
+/* for compatibility */
+#undef NL_CAT_LOCALE
 #ifdef NL_CAT_LOCALE /* POSIX-compliant. */
     /*
      * Check if LC_MESSAGES is set in the environment and use it, if so.
