--- tcsh-6.24.08/Makefile.in.orig	2023-04-10 11:26:45.000000000 +0200
+++ tcsh-6.24.08/Makefile.in	2023-04-11 22:08:08.291769614 +0200
@@ -723,6 +723,7 @@ $(srcdir)/stamp-h.in: $(srcdir)/configur
 
 check test: atconfig atlocal $(srcdir)/tests/testsuite $(BUILD)
 	$(ENVCMD) - \
+		PATH=/usr/bin \
 		TESTSUITEREGEN="$(TESTSUITEREGEN)" \
 		USER="$(USER)" \
 		$(SHELL) $(srcdir)/tests/testsuite $(TESTSUITEFLAGS)
