https://github.com/jaraco/inflect/issues/177

--- inflect-6.0.2/inflect/__init__.py.orig
+++ inflect-6.0.2/inflect/__init__.py
@@ -3628,7 +3628,7 @@
     # NUMERICAL INFLECTIONS
 
     @validate_arguments
-    def ordinal(self, num: Union[int, Word]) -> str:  # noqa: C901
+    def ordinal(self, num: Union[int, Word]) -> str:
         """
         Return the ordinal of num.
 
@@ -3787,7 +3787,7 @@
             num = ONE_DIGIT_WORD.sub(self.unitsub, num, 1)
         return num
 
-    @validate_arguments(config=dict(arbitrary_types_allowed=True))  # noqa: C901
+    @validate_arguments(config=dict(arbitrary_types_allowed=True))
     def number_to_words(  # noqa: C901
         self,
         num: Union[Number, Word],
