--- llvm-16.0.0.src/tools/clang/tools/libclang/CMakeLists.txt	2023-03-17 13:26:07.000000000 +0100
+++ llvm-16.0.0.src/tools/clang/tools/libclang/CMakeLists.txt.orig	2023-03-22 18:37:02.933769276 +0100
@@ -184,21 +184,21 @@
         )
     endif()
   endif()
-  if (USE_VERSION_SCRIPT)
-    target_link_options(libclang PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libclang.map")
+#  if (USE_VERSION_SCRIPT)
+#    target_link_options(libclang PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libclang.map")
     # The Solaris 11.4 linker supports a subset of GNU ld version scripts,
     # but requires a special option to enable it.
-    if (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
-      target_link_options(libclang PRIVATE "-Wl,-z,gnu-version-script-compat")
-    endif()
+#    if (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
+#      target_link_options(libclang PRIVATE "-Wl,-z,gnu-version-script-compat")
+#    endif()
     # Ensure that libclang.so gets rebuilt when the linker script changes.
-    set_property(SOURCE ARCMigrate.cpp APPEND PROPERTY
-                 OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/libclang.map)
+#    set_property(SOURCE ARCMigrate.cpp APPEND PROPERTY
+#                 OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/libclang.map)
 
-    set_target_properties(libclang PROPERTIES
-                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
-                          ${LIBCLANG_SOVERSION_ARG})
-  endif()
+#    set_target_properties(libclang PROPERTIES
+#                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
+#                          ${LIBCLANG_SOVERSION_ARG})
+#  endif()
 endif()
 
 if(INTERNAL_INSTALL_PREFIX)
