--- llvm-16.0.6.src/tools/clang/tools/clang-shlib/CMakeLists.txt.~1~	2024-11-20 00:31:31.176341874 -0500
+++ llvm-16.0.6.src/tools/clang/tools/clang-shlib/CMakeLists.txt	2024-11-20 00:34:12.888848288 -0500
@@ -50,6 +50,6 @@
                   ${_DEPS})
 # Optimize function calls for default visibility definitions to avoid PLT and
 # reduce dynamic relocations.
-if (NOT APPLE AND NOT MINGW)
-  target_link_options(clang-cpp PRIVATE LINKER:-Bsymbolic-functions)
-endif()
+# if (NOT APPLE AND NOT MINGW)
+#   target_link_options(clang-cpp PRIVATE LINKER:-Bsymbolic-functions)
+# endif()
