--- p7zip-17.05/install.sh.old	2024-11-19 15:03:25.603308860 -0500
+++ p7zip-17.05/install.sh	2024-11-19 15:03:52.660451532 -0500
@@ -10,7 +10,7 @@
     chmod 755 "${DEST_DIR}${DEST_SHARE}/${prg}"
     strip     "${DEST_DIR}${DEST_SHARE}/${prg}"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/${prg}"
-    echo "\"${DEST_SHARE}/${prg}\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/${prg}"
+    echo "\"${DEST_SHARE}/amd64/${prg}\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/${prg}"
     chmod 755 "${DEST_DIR}${DEST_BIN}/${prg}"
   fi
 }
