--- fsvs-1.2.5/src/options.c	2011-04-29 16:20:18.000000000 +0400
+++ fsvs-1.2.5_corrected/src/options.c	2013-08-02 21:14:48.832738924 +0400
@@ -263,7 +263,7 @@
 		.name="diff_prg", .cp_val="diff", .parse=opt___store_string, 
 	},
 	[OPT__DIFF_OPT] = {
-		.name="diff_opt", .cp_val="-pu", .parse=opt___store_string,
+		.name="diff_opt", .cp_val="-u", .parse=opt___store_string,
 	},
 	[OPT__DIFF_EXTRA] = {
 		.name="diff_extra", .cp_val=NULL, .parse=opt___store_string,
