--- fsvs-1.2.5/src/Makefile.in.orig	2013-08-02 19:11:14.135673921 +0400
+++ fsvs-1.2.5/src/Makefile.in	2013-08-02 19:11:31.232331089 +0400
@@ -15,6 +15,7 @@
 HEADREV	:= "$Revision: 2457 $"
 VERSION	= $(shell  perl -e '($$r) = (q( $(HEADREV) ) =~ m:(\d+):); $$t= q( $(HEADURL) ) =~ m:/tags/([^/]+): ? $$1 : "trunk"; print "$$t:$$r\n";' )
 
+CC	:= @CC@
 CFLAGS	:= @CFLAGS@ @NEED_FNESTED_FUNCTIONS@
 CFLAGS	+= -Wall -funsigned-char -Os -DFSVS_VERSION='"$(VERSION)"'
 LDFLAGS	:= @LDFLAGS@
