--- fsvs-1.2.5/src/Makefile.in	2013-08-02 19:55:01.319329538 +0400
+++ fsvs-1.2.5/src/Makefile.in	2013-08-02 20:02:14.290355242 +0400
@@ -90,7 +90,7 @@
 mandir	= @mandir@
 install:
 	mkdir -p $(DESTDIR)/etc/fsvs $(DESTDIR)/var/spool/fsvs $(DESTDIR)/$(bindir) $(DESTDIR)/etc/fsvs/svn/auth/svn.{simple,ssl.server,ssl.client-passphrase}
-	install -m 0755 $(DEST) $(DESTDIR)/$(bindir)
+	ginstall -m 0755 $(DEST) $(DESTDIR)/$(bindir)
 #	install -m 0644 ../doc/fsvs.1 $(DESTDIR)/(mandir)
 
 # No automatic rebuild (?)
