--- incremental-24.11.0/tox.ini.orig
+++ incremental-24.11.0/tox.ini
@@ -22,7 +22,7 @@
     pindeps: true
 deps =
     lint: -rrequirements_lint.txt
-    tests: -rrequirements_tests.txt
+    -rrequirements_tests.txt
     mypy: -rrequirements_mypy.txt
     apidocs: pydoctor
     pindeps: pip-tools
@@ -31,10 +31,10 @@
     ; Suppress pointless chatter in the output.
     PIP_DISABLE_PIP_VERSION_CHECK=yes
 
-    tests: COVERAGE_PROCESS_START={toxinidir}/.coveragerc
+    COVERAGE_PROCESS_START={toxinidir}/.coveragerc
     ; This must be an absolute path because the example tests
     ; run Python processes with alternative working directories.
-    tests: COVERAGE_FILE={toxinidir}/.coverage
+    COVERAGE_FILE={toxinidir}/.coverage
 
 commands =
     python -V
@@ -43,15 +43,15 @@
 
     apidocs: pydoctor -q --project-name incremental src/incremental
 
-    tests: coverage --version
-    tests: {envbindir}/trial --version
-    tests: coverage erase
-    tests: coverage run {envbindir}/trial incremental
-    tests: coverage run {envbindir}/trial tests/test_examples.py
-    tests: coverage combine
-    tests: coverage report
-    tests: coverage html
-    tests: coverage xml
+    coverage --version
+    {envbindir}/trial --version
+    coverage erase
+    coverage run {envbindir}/trial incremental
+    coverage run {envbindir}/trial tests/test_examples.py
+    coverage combine
+    coverage report
+    coverage html
+    coverage xml
 
     mypy: mypy src
 
