Do not try to reinstall system packages.

--- incremental-24.7.2/tests/test_examples.py.orig
+++ incremental-24.7.2/tests/test_examples.py
@@ -43,7 +43,7 @@
             pkgfile = builder.build("wheel", output_directory=dist_dir)
 
         # Force reinstall in case tox reused the venv.
-        run(["pip", "install", "--force-reinstall", pkgfile], check=True)
+        run(["pip", "install", pkgfile], check=True)
 
 
 class ExampleTests(TestCase):
