--- build-1.2.2/tests/test_util.py.orig
+++ build-1.2.2/tests/test_util.py
@@ -22,7 +22,7 @@
 @pytest.mark.pypy3323bug
 def test_wheel_metadata_isolation(package_test_flit):
     if importlib.util.find_spec('flit_core'):
-        pytest.xfail('flit_core is available -- we want it missing!')  # pragma: no cover
+        pytest.skip('flit_core is the build backend so it is always installed')  # pragma: no cover
 
     metadata = build.util.project_wheel_metadata(package_test_flit)
 
