https://github.com/giampaolo/psutil/pull/2546

--- psutil-7.0.0/psutil/_psutil_sunos.c.orig
+++ psutil-7.0.0/psutil/_psutil_sunos.c
@@ -1327,7 +1327,7 @@
         }
 #endif
         // UDPv4
-        else if (mibhdr.level == MIB2_UDP || mibhdr.level == MIB2_UDP_ENTRY) {
+        else if (mibhdr.level == MIB2_UDP && mibhdr.name == MIB2_UDP_ENTRY) {
             num_ent = mibhdr.len / sizeof(mib2_udpEntry_t);
             assert(num_ent * sizeof(mib2_udpEntry_t) == mibhdr.len);
             for (i = 0; i < num_ent; i++) {
