pytest-instafail is not needed to run tests.

--- psutil-7.2.1/setup.py.orig
+++ psutil-7.2.1/setup.py
@@ -79,7 +79,6 @@
 TEST_DEPS = [
     "psleak",
     "pytest",
-    "pytest-instafail",
     "pytest-xdist",
     "setuptools",
 ]
--- psutil-7.2.1/pyproject.toml.orig
+++ psutil-7.2.1/pyproject.toml
@@ -242,13 +242,11 @@
 [tool.pytest]
 addopts = [
     "--capture=no",
-    "--instafail",
     "--no-header",
     "--strict-config",
     "--strict-markers",
     "--tb=short",
     "--verbose",
-    "-p instafail",
     "-p no:doctest",
     "-p no:junitxml",
     "-p no:nose",
