https://github.com/giampaolo/psutil/issues/2317

--- psutil-5.9.6/psutil/tests/test_connections.py.orig
+++ psutil-5.9.6/psutil/tests/test_connections.py
@@ -52,13 +52,13 @@
 class ConnectionTestCase(PsutilTestCase):
 
     def setUp(self):
-        if not (NETBSD or FREEBSD):
+        if not (NETBSD or FREEBSD or SUNOS):
             # process opens a UNIX socket to /var/log/run.
             cons = thisproc.connections(kind='all')
             assert not cons, cons
 
     def tearDown(self):
-        if not (FREEBSD or NETBSD):
+        if not (FREEBSD or NETBSD or SUNOS):
             # Make sure we closed all resources.
             # NetBSD opens a UNIX socket to /var/log/run.
             cons = thisproc.connections(kind='all')
