--- psutil-6.0.0/psutil/tests/__init__.py.orig
+++ psutil-6.0.0/psutil/tests/__init__.py
@@ -230,7 +230,7 @@
 HAS_SENSORS_FANS = hasattr(psutil, "sensors_fans")
 HAS_SENSORS_TEMPERATURES = hasattr(psutil, "sensors_temperatures")
 HAS_THREADS = hasattr(psutil.Process, "threads")
-SKIP_SYSCONS = (MACOS or AIX) and os.getuid() != 0
+SKIP_SYSCONS = (MACOS or AIX or SUNOS) and os.getuid() != 0
 
 # --- misc
 
