https://github.com/giampaolo/psutil/pull/2688

--- psutil-7.2.1/tests/test_system.py.orig
+++ psutil-7.2.1/tests/test_system.py
@@ -493,7 +493,7 @@
                     return None
 
     @pytest.mark.skipif(
-        (CI_TESTING and OPENBSD) or MACOS, reason="unreliable on OPENBSD + CI"
+        (CI_TESTING and OPENBSD) or MACOS or SUNOS, reason="unreliable on OPENBSD + CI"
     )
     @retry_on_failure(30)
     def test_cpu_times_comparison(self):
