https://github.com/giampaolo/psutil/pull/2594

--- psutil-7.0.0/psutil/__init__.py.orig
+++ psutil-7.0.0/psutil/__init__.py
@@ -426,7 +426,7 @@
         # on PID and creation time.
         if not isinstance(other, Process):
             return NotImplemented
-        if OPENBSD or NETBSD:  # pragma: no cover
+        if OPENBSD or NETBSD or SUNOS:  # pragma: no cover
             # Zombie processes on Open/NetBSD have a creation time of
             # 0.0. This covers the case when a process started normally
             # (so it has a ctime), then it turned into a zombie. It's
--- psutil-7.0.0/psutil/tests/__init__.py.orig
+++ psutil-7.0.0/psutil/tests/__init__.py
@@ -1041,7 +1041,7 @@
         # Cloned zombie on Open/NetBSD has null creation time, see:
         # https://github.com/giampaolo/psutil/issues/2287
         assert proc == clone
-        if not (OPENBSD or NETBSD):
+        if not (OPENBSD or NETBSD or SUNOS):
             assert hash(proc) == hash(clone)
         # Its status always be querable.
         assert proc.status() == psutil.STATUS_ZOMBIE
