--- nano-5.3/doc/sample.nanorc.in.orig	2020-09-28 14:57:36.000000000 +0000
+++ nano-5.3/doc/sample.nanorc.in	2020-10-14 19:34:16.916949183 +0000
@@ -231,7 +231,7 @@
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
 
 ## To include most of the existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 ## Or you can select just the ones you need.  For example:
 # include "@PKGDATADIR@/html.nanorc"
