--- mod_perl-2.0.12/xs/APR/aprext/Makefile.PL	2022-01-30 14:02:24.000000000 +0000
+++ mod_perl-2.0.12/xs/APR/aprext/Makefile.PL.new	2022-06-10 11:10:59.522295310 +0000
@@ -28,7 +28,7 @@
 my %args = (NAME          => 'lib' . $build->{MP_APR_LIB},
             VERSION_FROM  => '../APR/APR.pm',
             SKIP          =>  [ @skip ] ,
-            LINKTYPE      =>  'static',
+            LINKTYPE      =>  'dynamic',
             OBJECT        => "@obj",
             clean         => { FILES => "@clean" },
 	   );
