diff -ruN gedit-2.27.5.orig/data/gedit.desktop.in.in gedit-2.27.5/data/gedit.desktop.in.in
--- gedit-2.27.5.orig/data/gedit.desktop.in.in	2009-08-29 00:27:09.628798889 +0100
+++ gedit-2.27.5/data/gedit.desktop.in.in	2009-08-29 00:27:36.912900983 +0100
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=gedit
 _GenericName=Text Editor
-_Comment=Edit text files
+# SUN_BRANDING
+_Comment=Edit text or source files, with syntax highlighting
 Exec=gedit %U
 Terminal=false
 Type=Application
