--- pcs-0.10.1/pcs/settings_default.py-orig	Fri Nov 23 06:57:53 2018
+++ pcs-0.10.1/pcs/settings_default.py	Thu Feb 11 17:00:17 2021
@@ -29,10 +29,10 @@
 booth_authkey_bytes = 64
 cluster_conf_file = "/etc/cluster/cluster.conf"
 fence_agent_binaries = "/usr/sbin/"
-pacemaker_schedulerd = "/usr/libexec/pacemaker/pacemaker-schedulerd"
-pacemaker_controld = "/usr/libexec/pacemaker/pacemaker-controld"
-pacemaker_based = "/usr/libexec/pacemaker/pacemaker-based"
-pacemaker_fenced = "/usr/libexec/pacemaker/pacemaker-fenced"
+pacemaker_schedulerd = "/usr/lib/pacemaker/pacemaker-schedulerd"
+pacemaker_controld = "/usr/lib/pacemaker/pacemaker-controld"
+pacemaker_based = "/usr/lib/pacemaker/pacemaker-based"
+pacemaker_fenced = "/usr/lib/pacemaker/pacemaker-fenced"
 pcs_version = "0.10.1"
 crm_report = os.path.join(pacemaker_binaries, "crm_report")
 crm_verify = os.path.join(pacemaker_binaries, "crm_verify")
