--- pcs-0.9.168/Makefile.~1~	2019-08-02 08:12:11.000000000 +0000
+++ pcs-0.9.168/Makefile	2019-08-19 13:21:24.635540714 +0000
@@ -109,10 +109,10 @@
 	Overpass-Bold.ttf;Overpass:style=Bold
 
 
-install: install_bundled_libs
+install: install_bundled_libs install_pcsd
 	# make Python interpreter execution sane (via -Es flags)
 	printf "[build]\nexecutable = $(PYTHON) -Es\n" > setup.cfg
-	$(PYTHON) setup.py install --root=$(or ${DESTDIR}, /) ${EXTRA_SETUP_OPTS}
+	$(PYTHON) setup.py install --root=$(or ${DESTDIR}, /) ${EXTRA_SETUP_OPTS} --install-lib=$(PYTHON_SITELIB)
 	# fix excessive script interpreting "executable" quoting with old setuptools:
 	# https://github.com/pypa/setuptools/issues/188
 	# https://bugzilla.redhat.com/1353934
@@ -148,9 +148,9 @@
 
 install_pcsd:
 ifeq ($(BUILD_GEMS),true)
-	make -C pcsd build_gems
+	DESTDIR= make -C pcsd build_gems
 endif
-	mkdir -p ${DESTDIR}/var/log/pcsd
+	DESTDIR= mkdir -p ${DESTDIR}/var/log/pcsd
 ifeq ($(IS_DEBIAN),true)
 	mkdir -p ${DESTDIR}/usr/share/
 	cp -r pcsd ${DESTDIR}/usr/share/
@@ -172,23 +172,23 @@
   endif
 else
 	mkdir -p ${DESTDIR}${PCSD_PARENT_DIR}/
-	cp -r pcsd ${DESTDIR}${PCSD_PARENT_DIR}/
+	cp -L -r pcsd ${DESTDIR}${PCSD_PARENT_DIR}/
 	install -m 644 -D pcsd/pcsd.conf ${DESTDIR}/etc/sysconfig/pcsd
 	install -d ${DESTDIR}/etc/pam.d
 	install -m 644 pcsd/pcsd.pam ${DESTDIR}/etc/pam.d/pcsd
   ifeq ($(IS_SYSTEMCTL),true)
-	install -d ${DESTDIR}/${systemddir}/system/
-	install -m 644 pcsd/pcsd.service ${DESTDIR}/${systemddir}/system/
+	install -d ${DESTDIR}${systemddir}/system/
+	install -m 644 pcsd/pcsd.service ${DESTDIR}${systemddir}/system/
 # ${DESTDIR}${PCSD_PARENT_DIR}/pcsd/pcsd holds the selinux context
 	install -m 755 pcsd/pcsd.service-runner ${DESTDIR}${PCSD_PARENT_DIR}/pcsd/pcsd
 	rm ${DESTDIR}${PCSD_PARENT_DIR}/pcsd/pcsd.service-runner
   else
-	install -m 755 -D pcsd/pcsd ${DESTDIR}/${initdir}/pcsd
+	install -m 755 -D pcsd/pcsd ${DESTDIR}${initdir}/pcsd
   endif
 endif
 	install -m 700 -d ${DESTDIR}/var/lib/pcsd
 	install -m 644 -D pcsd/pcsd.logrotate ${DESTDIR}/etc/logrotate.d/pcsd
-	install -m644 -D pcsd/pcsd.8 ${DESTDIR}/${MANDIR}/man8/pcsd.8
+	install -m644 -D pcsd/pcsd.8 ${DESTDIR}${MANDIR}/man8/pcsd.8
 	$(foreach font,$(pcsd_fonts),\
 		$(eval font_file = $(word 1,$(subst ;, ,$(font)))) \
 		$(eval font_def = $(word 2,$(subst ;, ,$(font)))) \
