--- pcs-0.10.1/Makefile-orig	Fri Nov 23 06:57:53 2018
+++ pcs-0.10.1/Makefile	Sun Feb 21 08:51:28 2021
@@ -208,7 +208,7 @@
 install_python_part: install_bundled_libs
 	# make Python interpreter execution sane (via -Es flags)
 	printf "[build]\nexecutable = $(PYTHON) -Es\n" > setup.cfg
-	$(PYTHON) setup.py install --root=$(or ${DESTDIR}, /) ${EXTRA_SETUP_OPTS}
+	$(PYTHON) setup.py install --root=$(or ${DESTDIR}, /) ${EXTRA_SETUP_OPTS} --install-lib=$(PYTHON_SITELIB)
 	# fix excessive script interpreting "executable" quoting with old setuptools:
 	# https://github.com/pypa/setuptools/issues/188
 	# https://bugzilla.redhat.com/1353934
@@ -244,7 +244,8 @@
 endif
 	install -d -m 700 ${DESTDIR}/var/log/pcsd
 	mkdir -p ${DEST_LIB}
-	cp -r pcsd ${DEST_LIB}
+	# Copy files, not symlinks to files
+	cp -L -r pcsd ${DEST_LIB}
 	install -m 644 -D pcsd/pcsd.conf ${DEST_CONF}/pcsd
 	install -d ${DESTDIR}/etc/pam.d
 	install -m 644 pcsd/pcsd.pam ${DESTDIR}/etc/pam.d/pcsd
