--- pcs-0.10.1/pcs/settings_default.py-orig	Fri Feb 12 10:18:23 2021
+++ pcs-0.10.1/pcs/settings_default.py	Fri Feb 12 12:40:24 2021
@@ -44,7 +44,7 @@
 pcsd_users_conf_location = "/var/lib/pcsd/pcs_users.conf"
 pcsd_settings_conf_location = "/var/lib/pcsd/pcs_settings.conf"
 pcsd_exec_location = "/usr/lib/pcsd/"
-pcsd_log_location = "/var/log/pcsd/pcsd.log"
+pcsd_log_location = "/var/log/hacluster/pcsd.log"
 pcsd_default_port = 2224
 cib_dir = "/var/lib/pacemaker/cib/"
 pacemaker_uname = "hacluster"
--- pcs-0.9.168/pcsd/pcsd.logrotate.~1~	2019-08-19 13:22:09.723145661 +0000
+++ pcs-0.9.168/pcsd/pcsd.logrotate	2019-08-19 13:23:34.600518821 +0000
@@ -1,4 +1,4 @@
-/var/log/pcsd/*.log {
+/var/log/hacluster/*.log {
     rotate 5
     weekly
     missingok
