diff -Nura a/ssl.rb b/ssl.rb
--- a/pcsd/ssl.rb	2016-10-26 16:05:27.469339703 +0200
+++ b/pcsd/ssl.rb	2016-10-26 16:10:33.750086914 +0200
@@ -13,7 +13,7 @@
 end
 
 server_name = WEBrick::Utils::getservername
-$logger = configure_logger('/var/log/pcsd/pcsd.log')
+$logger = configure_logger('/var/log/hacluster/pcsd.log')
 
 def generate_cert_key_pair(server_name)
   name = "/C=US/ST=MN/L=Minneapolis/O=pcsd/OU=pcsd/CN=#{server_name}"
--- pcs-0.9.168/pcsd/pcsd.rb.~1~	2019-08-19 13:22:09.723894638 +0000
+++ pcs-0.9.168/pcsd/pcsd.rb	2019-08-19 13:24:35.330533055 +0000
@@ -128,12 +128,12 @@
   # File.open(path, mode, options)
   # File.open(path, mode, perm, options)
   # In order to set permissions, the method must be called with 4 arguments.
-  logger = File.open("/var/log/pcsd/pcsd.log", "a+", 0600, {})
+  logger = File.open("/var/log/hacluster/pcsd.log", "a+", 0600, {})
   STDOUT.reopen(logger)
   STDERR.reopen(logger)
   STDOUT.sync = true
   STDERR.sync = true
-  $logger = configure_logger('/var/log/pcsd/pcsd.log')
+  $logger = configure_logger('/var/log/hacluster/pcsd.log')
   $semaphore_cfgsync = Mutex.new
 
   capabilities, capabilities_pcsd = get_capabilities($logger)
--- pcs-0.9.168/pcsd/pcsd.logrotate.~1~	2019-08-19 13:22:09.723145661 +0000
+++ pcs-0.9.168/pcsd/pcsd.logrotate	2019-08-19 13:23:34.600518821 +0000
@@ -1,4 +1,4 @@
-/var/log/pcsd/*.log {
+/var/log/hacluster/*.log {
     rotate 5
     weekly
     missingok
