--- pcs-0.10.1/pcs/settings_default.py-orig	Fri Feb 12 12:46:33 2021
+++ pcs-0.10.1/pcs/settings_default.py	Fri Feb 12 15:37:18 2021
@@ -81,7 +81,7 @@
 ])
 # Set pcsd_gem_path to None if there are no bundled ruby gems and the path does
 # not exists.
-pcsd_gem_path = "vendor/bundle/ruby"
+pcsd_gem_path = "vendor/bundle/ruby/2.3.0"
 ruby_executable = "/usr/bin/ruby"
 
 gui_session_lifetime_seconds = 60 * 60
