--- pcs-0.9.159/pcsd/Makefile.orig	2017-09-23 07:45:08.178514033 +0000
+++ pcs-0.9.159/pcsd/Makefile	2017-09-23 07:46:13.770853713 +0000
@@ -1,16 +1,16 @@
 FFI_VERSION="1.9.17"
-FFI_C_DIR=vendor/bundle/ruby/gems/ffi-${FFI_VERSION}/ext/ffi_c
+FFI_C_DIR=vendor/bundle/ruby/2.3.0/gems/ffi-${FFI_VERSION}/ext/ffi_c
 
 build_gems: get_gems
-	bundle install --local --deployment
+	bundle install --local --deployment --path vendor/bundle --standalone
 	#ffi makes symlink with absolute path. Let's change it to relative path.
-	for fname in `ls ${FFI_C_DIR}/libffi-*/include/ffitarget.h`; do \
-		if [[ -L "$$fname" ]]; then \
-			target=$$(readlink $$fname | sed "s~.*/${FFI_C_DIR}\(/libffi/src/[^/]\+/ffitarget.h\)~../..\1~"); \
-			rm $$fname; \
-			ln -s $$target $$fname; \
-		fi; \
-	done;
+#	for fname in `ls ${FFI_C_DIR}/libffi-*/include/ffitarget.h`; do \
+#		if [[ -L "$$fname" ]]; then \
+#			target=$$(readlink $$fname | sed "s~.*/${FFI_C_DIR}\(/libffi/src/[^/]\+/ffitarget.h\)~../..\1~"); \
+#			echo rm $$fname; \
+#			echo ln -s $$target $$fname; \
+#		fi; \
+#	done;
 
 # RHEL6 needs special rpam-ruby19 gem to work with 1.8.7
 # also bundler is not available on RHEL6 in rpm
@@ -34,7 +34,7 @@
 	-- '--with-ldflags="-Wl,-z,now -Wl,-z,relro"'
 
 get_gems:
-	bundle package
+	bundle package --path vendor/cache
 
 clean:
 	rm -rfv vendor/
