--- pcs-0.10.1/pcsd/bootstrap.rb.orig
+++ pcs-0.10.1/pcsd/bootstrap.rb
@@ -4,6 +4,9 @@
 
 require 'settings.rb'
 
+def is_solaris()
+   return (`uname -s`.chomp.eql? "SunOS")
+end
 
 def is_systemctl()
   systemctl_paths = [
@@ -43,6 +46,7 @@
 PCS_VERSION = '0.10.1'
 # unique instance signature, allows detection of dameon restarts
 COROSYNC = COROSYNC_BINARIES + "corosync"
+ISSOLARIS = is_solaris
 ISSYSTEMCTL = is_systemctl
 COROSYNC_CMAPCTL = COROSYNC_BINARIES + "corosync-cmapctl"
 COROSYNC_QUORUMTOOL = COROSYNC_BINARIES + "corosync-quorumtool"
