diff -Nur a/pcs/settings_default.py b/pcs/settings_default.py
--- a/pcs/settings_default.py	2016-11-28 11:13:46.023251786 +0100
+++ b/pcs/settings_default.py	2016-11-28 11:14:26.005641239 +0100
@@ -21,10 +21,10 @@
 corosync_qdevice_net_client_ca_file_name = "qnetd-cacert.crt"
 cluster_conf_file = "/etc/cluster/cluster.conf"
 fence_agent_binaries = "/usr/sbin/"
-pengine_binary = "/usr/libexec/pacemaker/pengine"
-crmd_binary = "/usr/libexec/pacemaker/crmd"
-cib_binary = "/usr/libexec/pacemaker/cib"
-stonithd_binary = "/usr/libexec/pacemaker/stonithd"
+pengine_binary = "/usr/lib/pacemaker/pengine"
+crmd_binary = "/usr/lib/pacemaker/crmd"
+cib_binary = "/usr/lib/pacemaker/cib"
+stonithd_binary = "/usr/lib/pacemaker/stonithd"
 pcs_version = "0.9.155"
 crm_report = pacemaker_binaries + "crm_report"
 crm_verify = pacemaker_binaries + "crm_verify"
