--- a/pcsd/settings.rb	2016-10-26 07:46:42.246359217 +0200
+++ b/pcsd/settings.rb	2016-10-26 07:47:25.880345727 +0200
@@ -9,8 +9,8 @@
 HEARTBEAT_AGENTS_DIR = "/usr/lib/ocf/resource.d/heartbeat/"
 PACEMAKER_AGENTS_DIR = "/usr/lib/ocf/resource.d/pacemaker/"
 NAGIOS_METADATA_DIR = '/usr/share/pacemaker/nagios/plugins-metadata/'
-PENGINE = "/usr/libexec/pacemaker/pengine"
-CIB_BINARY = '/usr/libexec/pacemaker/cib'
+PENGINE = "/usr/lib/pacemaker/pengine"
+CIB_BINARY = '/usr/lib/pacemaker/cib'
 CRM_MON = "/usr/sbin/crm_mon"
 CRM_NODE = "/usr/sbin/crm_node"
 CRM_ATTRIBUTE = "/usr/sbin/crm_attribute"
