diff -Nur a/pcsd/settings.rb b/pcsd/settings.rb
--- a/pcsd/settings.rb	2016-11-28 11:06:09.495592984 +0100
+++ b/pcsd/settings.rb	2016-11-28 11:06:41.053749634 +0100
@@ -5,8 +5,8 @@
 KEY_FILE = PCSD_VAR_LOCATION + 'pcsd.key'
 COOKIE_FILE = PCSD_VAR_LOCATION + 'pcsd.cookiesecret'
 
-PENGINE = "/usr/libexec/pacemaker/pengine"
-CIB_BINARY = '/usr/libexec/pacemaker/cib'
+PENGINE = "/usr/lib/pacemaker/pengine"
+CIB_BINARY = '/usr/lib/pacemaker/cib'
 CRM_MON = "/usr/sbin/crm_mon"
 CRM_NODE = "/usr/sbin/crm_node"
 CRM_ATTRIBUTE = "/usr/sbin/crm_attribute"
