--- a/pcs/settings_default.py	2016-10-26 07:43:57.391309989 +0200
+++ b/pcs/settings_default.py	2016-10-26 07:45:20.522677585 +0200
@@ -18,10 +18,10 @@
 corosync_qdevice_net_client_ca_file_name = "qnetd-cacert.crt"
 cluster_conf_file = "/etc/cluster/cluster.conf"
 fence_agent_binaries = "/usr/sbin/"
-pengine_binary = "/usr/libexec/pacemaker/pengine"
-crmd_binary = "/usr/libexec/pacemaker/crmd"
-cib_binary = "/usr/libexec/pacemaker/cib"
-stonithd_binary = "/usr/libexec/pacemaker/stonithd"
+pengine_binary = "/usr/lib/pacemaker/pengine"
+crmd_binary = "/usr/lib/pacemaker/crmd"
+cib_binary = "/usr/lib/pacemaker/cib"
+stonithd_binary = "/usr/lib/pacemaker/stonithd"
 pcs_version = "0.9.154"
 crm_report = pacemaker_binaries + "crm_report"
 crm_verify = pacemaker_binaries + "crm_verify"
