--- pytest_order-1.3.0/tox.ini.orig
+++ pytest_order-1.3.0/tox.ini
@@ -12,10 +12,10 @@
     pytest624: pytest>=6.2.4,<6.3
     pytest70: pytest>=7.0,<7.1
     pytest74: pytest>=7.3,<8.0
-    pytest80: pytest==8.0.1
+    pytest==8.0.1
     pytest-cov<2.10
     pytest{50,54}: pytest-xdist<2.0.0
-    pytest{60,62,624,70,74,80}: pytest-xdist
+    pytest-xdist
     pytest-dependency>=0.5.1
     pytest-mock
 
