--- pytest_order-1.4.0/tox.ini.orig
+++ pytest_order-1.4.0/tox.ini
@@ -9,10 +9,10 @@
     pytest74: pytest>=7.3,<8.0
     pytest80: pytest==8.0.1
     pytest83: pytest>=8.3,<9.0
-    pytest90: pytest>=9.0,<10.0
+    pytest>=9.0,<10.0
     pytest-cov<2.10
     pytest{50,54}: pytest-xdist<2.0.0
-    pytest{60,62,624,70,74,80,83,90}: pytest-xdist
+    pytest-xdist
     pytest-dependency>=0.5.1
     pytest-mock
 
