We do not have pytest-pylint packaged.

--- StrEnum-0.4.15/setup.py.orig
+++ StrEnum-0.4.15/setup.py
@@ -22,7 +22,6 @@
             "pytest",
             "pytest-black",
             "pytest-cov",
-            "pytest-pylint",
             "pylint",
         ],
         "docs": [
--- StrEnum-0.4.15/pytest.ini.orig
+++ StrEnum-0.4.15/pytest.ini
@@ -1,3 +1,3 @@
 [pytest]
-addopts = --cov=strenum --cov-report term-missing --black --pylint --ignore=versioneer.py --ignore=strenum/_version.py --ignore=docs/
+addopts = --cov=strenum --cov-report term-missing --black --ignore=versioneer.py --ignore=strenum/_version.py --ignore=docs/
 filterwarnings = ignore::DeprecationWarning
