https://github.com/akiym/Crypt-OpenSSL-Guess/pull/15

--- Crypt-OpenSSL-Guess-0.15/lib/Crypt/OpenSSL/Guess.pm.orig
+++ Crypt-OpenSSL-Guess-0.15/lib/Crypt/OpenSSL/Guess.pm
@@ -47,7 +47,7 @@
     return '' unless -x $exec;
 
     my @lib_paths;
-    for ($prefix, "$prefix/lib64", "$prefix/lib", "$prefix/out32dll") {
+    for ($prefix, "$prefix/lib64", "$prefix/lib/64", "$prefix/lib", "$prefix/out32dll") {
         push @lib_paths, $_ if -d $_;
     }
 
