--- gexif-0.5/gexif/Makefile.am.orig	2008-07-10 18:01:14.647857800 +0100
+++ gexif-0.5/gexif/Makefile.am	2008-07-10 18:01:19.841533900 +0100
@@ -1,7 +1,6 @@
 INCLUDES =						\
 	-I$(top_srcdir)					\
 	-I$(top_srcdir)/intl				\
-	-DGTK_DISABLE_DEPRECATED			\
 	-DGEXIF_LOCALEDIR="\"$(datadir)/locale\""	\
 	$(GEXIF_CFLAGS)
 
--- gexif-0.5/gexif/gexif-thumbnail.h.orig	2008-07-10 18:00:53.254827600 +0100
+++ gexif-0.5/gexif/gexif-thumbnail.h	2008-07-10 18:01:04.356548300 +0100
@@ -43,7 +43,7 @@
 	GtkDialogClass parent_class;
 };
 
-GtkType    gexif_thumbnail_get_type (void);
+GType      gexif_thumbnail_get_type (void);
 GtkWidget *gexif_thumbnail_new      (const guchar *data, guint size);
 
 #endif /* __GEXIF_THUMBNAIL_H__ */
