--- gexif-0.5/gexif/gexif.c.~1~	2014-12-26 16:09:05.870048917 +0300
+++ gexif-0.5/gexif/gexif.c	2014-12-26 16:21:33.548143466 +0300
@@ -54,7 +54,7 @@
 	textdomain (PACKAGE);
 
 	gtk_init (&argc, &argv);
-	g_log_set_always_fatal (G_LOG_LEVEL_CRITICAL | G_LOG_LEVEL_WARNING);
+/*	g_log_set_always_fatal (G_LOG_LEVEL_CRITICAL | G_LOG_LEVEL_WARNING); */
 
 	w = gexif_main_new ();
 	gtk_widget_show (w);
