--- evolution-2.30.3/plugins/caldav/caldav-browse-server.c.orig	2013-11-19 16:36:27.082184523 +0400
+++ evolution-2.30.3/plugins/caldav/caldav-browse-server.c	2013-11-19 16:37:02.071331457 +0400
@@ -975,7 +975,7 @@
 
 	soup_message_headers_append (message->request_headers, "User-Agent", "Evolution/" VERSION);
 	soup_message_headers_append (message->request_headers, "Depth", depth_1 ? "1" : "0");
-	soup_message_set_request (message, "application/xml", SOUP_MEMORY_COPY, (const gchar *) buf->buffer->content, buf->buffer->use);
+	soup_message_set_request (message, "application/xml", SOUP_MEMORY_COPY, (const gchar *) xmlBufContent(buf->buffer), xmlBufUse(buf->buffer));
 
 	/* Clean up the memory */
 	xmlOutputBufferClose (buf);
