--- evolution-2.4.1/plugin.mk	2006-07-21 11:01:02.841254000 +0800
+++ evolution-2.4.1/plugin.mk.bak	2006-07-21 11:11:37.458252000 +0800
@@ -1,8 +1,8 @@
 %.eplug: %.eplug.in
-	sed -e 's|\@PLUGINDIR\@|$(plugindir)|'		\
+	/usr/xpg4/bin/sed -e 's|\@PLUGINDIR\@|$(plugindir)|'		\
 	-e 's|\@SOEXT\@|$(SOEXT)|'			\
 	-e 's|\@GETTEXT_PACKAGE\@|$(GETTEXT_PACKAGE)|'	\
-	-e 's|\@LOCALEDIR\@|$(localedir)|' $< > $@
+	-e 's|\@LOCALEDIR\@|$(localedir)|' $< > $@ 2>/dev/null;echo ""
 
 %.eplug.in: %.eplug.xml
 	LC_ALL=C $(INTLTOOL_MERGE) -x -u /tmp $< $@
