--- evolution-2.32.3/modules/addressbook/addressbook-config.c.~1~	2015-01-11 13:37:43.327181148 +0300
+++ evolution-2.32.3/modules/addressbook/addressbook-config.c	2015-01-11 13:40:39.090196842 +0300
@@ -921,8 +921,7 @@
 	gtk_entry_set_text((GtkEntry *)sdialog->search_filter, lud && lud->lud_filter ? lud->lud_filter : "");
 	g_signal_connect (sdialog->search_filter, "changed",  G_CALLBACK (search_filter_changed_cb), sdialog);
 
-	g_signal_connect (e_builder_get_widget(builder, "rootdn-button"), "clicked",
-			  G_CALLBACK (query_for_supported_bases), sdialog);
+	gtk_widget_set_sensitive (e_builder_get_widget(builder, "rootdn-button"), FALSE);
 
 	if (lud)
 		ldap_free_urldesc (lud);
