--- evolution-2.32.3/configure.ac.1	2015-11-09 12:57:13.536426763 +0300
+++ evolution-2.32.3/configure.ac	2015-11-09 12:57:29.439194682 +0300
@@ -23,7 +23,7 @@
 dnl Autoconf / Automake Initialization
 AC_PREREQ(2.58)
 AC_INIT([evolution],[evo_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=Evolution])
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror foreign])
+AM_INIT_AUTOMAKE([1.10 -Wall foreign])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR(README)
