--- evolution-2.32.3/calendar/gui/alarm-notify/alarm-queue.c.1	2016-04-14 00:53:51.770362097 +0300
+++ evolution-2.32.3/calendar/gui/alarm-notify/alarm-queue.c	2016-04-14 00:54:09.611416736 +0300
@@ -1606,7 +1606,7 @@
 			body = g_strdup_printf ("%s %s", start_str, time_str);
 	}
 
-	n = notify_notification_new (summary, body, "stock_appointment-reminder", NULL);
+	n = notify_notification_new (summary, body, "stock_appointment-reminder");
 	if (!notify_notification_show(n, NULL))
 	    g_warning ("Could not send notification to daemon\n");
 
--- evolution-2.32.3/plugins/publish-calendar/publish-calendar.c.~1~	2011-03-29 22:38:48.000000000 +0400
+++ evolution-2.32.3/plugins/publish-calendar/publish-calendar.c	2016-04-14 01:14:36.175497291 +0300
@@ -150,8 +150,7 @@
 				return;
 			}
 
-			notify  = notify_notification_new (_("Calendar Publishing"), actual_msg->str, stock_name, NULL);
-			notify_notification_attach_to_status_icon (notify, status_icon);
+			notify  = notify_notification_new (_("Calendar Publishing"), actual_msg->str, stock_name);
 			notify_notification_set_urgency (notify, NOTIFY_URGENCY_NORMAL);
 			notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 			g_timeout_add (500, show_notify_cb, NULL);
--- evolution-2.32.3/plugins/mail-notification/mail-notification.c.1	2016-04-14 01:15:39.093843235 +0300
+++ evolution-2.32.3/plugins/mail-notification/mail-notification.c	2016-04-14 01:16:47.565296259 +0300
@@ -564,9 +564,7 @@
 
 			notify  = notify_notification_new (
 				_("New email"), safetext,
-				"mail-unread", NULL);
-			notify_notification_attach_to_status_icon (
-				notify, status_icon);
+				"mail-unread");
 
 			/* Check if actions are supported */
 			if (can_support_actions ()) {
