--- evolution-2.23.92/composer/e-msg-composer.c.old	Fri Sep 19 16:25:16 2008
+++ evolution-2.23.92//composer/e-msg-composer.c	Fri Sep 19 16:26:01 2008
@@ -248,7 +248,7 @@
 static CamelTransferEncoding
 best_encoding (GByteArray *buf, const gchar *charset)
 {
-	gchar *in, *out, outbuf[256], *ch;
+	gchar *in, *out, outbuf[4096], *ch;
 	gsize inlen, outlen;
 	gint status, count = 0;
 	iconv_t cd;
@@ -289,11 +289,6 @@
 {
 	gchar *charset;
 
-	/* First try US-ASCII */
-	*encoding = best_encoding (buf, "US-ASCII");
-	if (*encoding == CAMEL_TRANSFER_ENCODING_7BIT)
-		return NULL;
-
 	/* Next try the user-specified charset for this message */
 	*encoding = best_encoding (buf, default_charset);
 	if (*encoding != -1)
