We have no a2wsgi packaged yet.

--- uvicorn-0.30.6/requirements.txt.orig
+++ uvicorn-0.30.6/requirements.txt
@@ -5,7 +5,6 @@
 h11 @ git+https://github.com/python-hyper/h11.git@master
 
 # Explicit optionals
-a2wsgi==1.10.6
 wsproto==1.2.0
 websockets==12.0
 
--- uvicorn-0.30.6/tests/middleware/test_wsgi.py.orig
+++ uvicorn-0.30.6/tests/middleware/test_wsgi.py
@@ -4,7 +4,6 @@
 import sys
 from typing import AsyncGenerator, Callable
 
-import a2wsgi
 import httpx
 import pytest
 
@@ -52,7 +51,7 @@
         return [output]
 
 
-@pytest.fixture(params=[wsgi._WSGIMiddleware, a2wsgi.WSGIMiddleware])
+@pytest.fixture(params=[wsgi._WSGIMiddleware])
 def wsgi_middleware(request: pytest.FixtureRequest) -> Callable:
     return request.param
 
