We have no a2wsgi packaged yet.

--- uvicorn-0.51.0/pyproject.toml.orig
+++ uvicorn-0.51.0/pyproject.toml
@@ -69,7 +69,6 @@
     # check dist
     "twine==6.2.0",
     # Explicit optionals,
-    "a2wsgi==1.10.10",
     "wsproto==1.3.2",
 ]
 docs = [
--- uvicorn-0.51.0/tests/middleware/test_wsgi.py.orig
+++ uvicorn-0.51.0/tests/middleware/test_wsgi.py
@@ -4,7 +4,6 @@
 import sys
 from collections.abc import AsyncGenerator, Callable
 
-import a2wsgi
 import httpx
 import pytest
 
@@ -52,7 +51,7 @@
         return [output]
 
 
-@pytest.fixture(params=[wsgi._WSGIMiddleware, a2wsgi.WSGIMiddleware])
+@pytest.fixture(params=[wsgi._WSGIMiddleware])
 def wsgi_middleware(request: pytest.FixtureRequest) -> Callable:
     return request.param
 
