We have no wsproto packaged yet.

--- uvicorn-0.30.6/requirements.txt.orig
+++ uvicorn-0.30.6/requirements.txt
@@ -5,7 +5,6 @@
 h11 @ git+https://github.com/python-hyper/h11.git@master
 
 # Explicit optionals
-wsproto==1.2.0
 websockets==12.0
 
 # Packaging
--- uvicorn-0.30.6/tests/protocols/test_websocket.py.orig
+++ uvicorn-0.30.6/tests/protocols/test_websocket.py
@@ -1089,12 +1089,8 @@
         response = await wsresponse(url)
         assert response.status_code == 403
         assert response.content == b"hardbody"
-        if ws_protocol_cls == _WSProtocol:
-            # wsproto automatically makes the message chunked
-            assert response.headers["transfer-encoding"] == "chunked"
-        else:
-            # websockets automatically adds a content-length
-            assert response.headers["content-length"] == "8"
+        # websockets automatically adds a content-length
+        assert response.headers["content-length"] == "8"
 
     config = Config(
         app=app,
