These projects are not needed to run tests.

--- uvicorn-0.51.0/pyproject.toml.orig
+++ uvicorn-0.51.0/pyproject.toml
@@ -42,7 +42,6 @@
     "httptools>=0.8.0",
     "python-dotenv>=0.13",
     "PyYAML>=5.1",
-    "uvloop>=0.15.1; sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')",
     "watchfiles>=0.20",
     "websockets>=13.0",
 ]
@@ -51,7 +50,6 @@
 dev = [
     # We add uvicorn[standard] so `uv sync` considers the extras.
     "uvicorn[standard]",
-    "ruff>=0.15.15",
     "pytest==9.0.3",
     "pytest-mock==3.15.1",
     "pytest-xdist[psutil]==3.8.0",
@@ -66,8 +64,6 @@
     "coverage-enable-subprocess==1.0",
     "httpx==0.28.1",
     "anyio",
-    # check dist
-    "twine==6.2.0",
 ]
 docs = [
     "zensical==0.0.43",
