https://github.com/Kludex/uvicorn/discussions/2755

--- uvicorn-0.51.0/pyproject.toml.orig
+++ uvicorn-0.51.0/pyproject.toml
@@ -111,6 +111,7 @@
     "error",
     "ignore:Uvicorn's native WSGI implementation is deprecated.*:DeprecationWarning",
     "ignore: 'cgi' is deprecated and slated for removal in Python 3.13:DeprecationWarning",
+    "ignore: websockets.exceptions.InvalidStatusCode is deprecated:DeprecationWarning",
     # httptools in Python 3.14t needs the `PYTHON_GIL=0` environment variable, or raises a RuntimeWarning.
     "ignore: The global interpreter lock (GIL)*:RuntimeWarning"
 ]
