--- cryptography-3.2.1/src/cryptography/hazmat/bindings/openssl/binding.py.orig
+++ cryptography-3.2.1/src/cryptography/hazmat/bindings/openssl/binding.py
@@ -167,26 +167,7 @@
 
 
 def _verify_openssl_version(lib):
-    if (
-        lib.CRYPTOGRAPHY_OPENSSL_LESS_THAN_110
-        and not lib.CRYPTOGRAPHY_IS_LIBRESSL
-    ):
-        if os.environ.get("CRYPTOGRAPHY_ALLOW_OPENSSL_102"):
-            warnings.warn(
-                "OpenSSL version 1.0.2 is no longer supported by the OpenSSL "
-                "project, please upgrade. The next version of cryptography "
-                "will completely remove support for it.",
-                utils.CryptographyDeprecationWarning,
-            )
-        else:
-            raise RuntimeError(
-                "You are linking against OpenSSL 1.0.2, which is no longer "
-                "supported by the OpenSSL project. To use this version of "
-                "cryptography you need to upgrade to a newer version of "
-                "OpenSSL. For this version only you can also set the "
-                "environment variable CRYPTOGRAPHY_ALLOW_OPENSSL_102 to "
-                "allow OpenSSL 1.0.2."
-            )
+    pass
 
 
 def _verify_package_version(version):
