--- cryptography-43.0.1/pyproject.toml.orig
+++ cryptography-43.0.1/pyproject.toml
@@ -8,7 +8,7 @@
     "cffi>=1.12; platform_python_implementation != 'PyPy'",
     # Needed because cffi imports distutils, and in Python 3.12, distutils has
     # been removed from the stdlib, but installing setuptools puts it back.
-    "setuptools!=74.0.0,!=74.1.0,!=74.1.1",
+    "setuptools",
 ]
 build-backend = "maturin"
 
