diff --git a/runtime/syntax/c.vim b/runtime/syntax/c.vim
index 68b6778..2dc6ccd 100644
--- a/runtime/syntax/c.vim
+++ b/runtime/syntax/c.vim
@@ -412,6 +412,8 @@ if !exists("c_no_ansi") || exists("c_ansi_constants") || exists("c_gnu")
   " Add POSIX signals as well...
   syn keyword cConstant SIGABRT SIGALRM SIGCHLD SIGCONT SIGFPE SIGHUP SIGILL SIGINT SIGKILL SIGPIPE SIGQUIT SIGSEGV
   syn keyword cConstant SIGSTOP SIGTERM SIGTRAP SIGTSTP SIGTTIN SIGTTOU SIGUSR1 SIGUSR2
+  " Add illumos-specific signals
+  syn keyword cConstant SIGCANCEL SIGCLD SIGEMT SIGFREEZE SIGIO SIGIOT SIGJVM1 SIGJVM2 SIGLOST SIGLWP SIGPWR SIGRTMAX SIGRTMIN SIGTHAW SIGWAITING SIGXRES
   syn keyword cConstant _IOFBF _IOLBF _IONBF BUFSIZ EOF WEOF FOPEN_MAX FILENAME_MAX L_tmpnam
   syn keyword cConstant SEEK_CUR SEEK_END SEEK_SET TMP_MAX EXIT_FAILURE EXIT_SUCCESS RAND_MAX
   syn keyword cConstant stdin stdout stderr
