diff -wpruN --no-dereference '--exclude=*.orig' a~/runtime/syntax/dtrace.vim a/runtime/syntax/dtrace.vim
--- a~/runtime/syntax/dtrace.vim	1970-01-01 00:00:00
+++ a/runtime/syntax/dtrace.vim	1970-01-01 00:00:00
@@ -77,7 +77,7 @@ syn match dtraceOption contained "\%(def
 
 " Turn reserved but unspecified keywords into errors
 syn keyword dtraceReservedKeyword auto break case continue counter default do
-syn keyword dtraceReservedKeyword else for goto if import probe provider
+syn keyword dtraceReservedKeyword for goto import probe provider
 syn keyword dtraceReservedKeyword register restrict return static switch while
 
 " Add dtrace-specific stuff
