--- llvm-16.0.0.src/tools/clang/CMakeLists.txt	2023-03-17 13:26:07.000000000 +0100
+++ llvm-16.0.0.src/tools/clang/CMakeLists.txt.orig	2023-03-24 10:19:42.035659075 +0100
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.13.4)
 
 if(NOT DEFINED LLVM_COMMON_CMAKE_UTILS)
-  set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+  set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 endif()
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
--- llvm-16.0.3.src/CMakeLists.txt	2023-05-02 19:50:56.000000000 +0200
+++ llvm-16.0.3.src/CMakeLists.txt.new	2023-05-06 21:50:18.070375923 +0200
@@ -9,8 +9,8 @@
     "at least 3.20.0 now to avoid issues in the future!")
 endif()
 
-set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
-include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
+set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
 
 set(CMAKE_BUILD_WITH_INSTALL_NAME_DIR ON)
--- llvm-17.0.1.src/cmake/modules/HandleLLVMOptions.cmake	2023-09-19 11:05:13.000000000 +0200
+++ llvm-17.0.1.src/cmake/modules/HandleLLVMOptions.cmake.new	2023-09-23 19:21:33.854537851 +0200
@@ -1324,7 +1324,7 @@
   add_flag_if_supported("-no-canonical-prefixes" NO_CANONICAL_PREFIXES)
 endif()
 
-set(LLVM_THIRD_PARTY_DIR  ${CMAKE_CURRENT_SOURCE_DIR}/../third-party CACHE STRING
+set(LLVM_THIRD_PARTY_DIR  ${CMAKE_CURRENT_SOURCE_DIR}/third-party CACHE STRING
     "Directory containing third party software used by LLVM (e.g. googletest)")
 
 set(LLVM_UNITTEST_LINK_FLAGS "" CACHE STRING
