--- samba-4.17.4//source3/modules/vfs_solarisacl.c	2022-08-08 16:15:39.408192400 +0000
+++ samba-4.17.4//source3/modules/vfs_solarisacl.c.new	2022-12-21 12:23:41.015910848 +0000
@@ -62,7 +62,7 @@
 
 /* public functions - the api */
 
-static SMB_ACL_T solarisacl_sys_acl_get_file(vfs_handle_struct *handle,
+SMB_ACL_T solarisacl_sys_acl_get_file(vfs_handle_struct *handle,
 				const struct smb_filename *smb_fname,
 				SMB_ACL_TYPE_T type,
 				TALLOC_CTX *mem_ctx)
--- samba-4.17.0/source3/modules/vfs_solarisacl.c	2022-08-08 07:15:39.408192400 +0000
+++ samba-4.17.0/source3/modules/vfs_solarisacl.c.new	2022-09-17 09:16:38.514619461 +0000
@@ -252,7 +252,7 @@
 	}
 	ret = acl(fsp->fsp_name->base_name, SETACL, count, solaris_acl);
 	if (ret != 0) {
-		DBG_DEBG("settinge file acl failed!\n");
+		DBG_DEBUG("setting file acl failed!\n");
 	}
 
  done:
--- samba-4.20.1/source3/modules/vfs_zfsacl.c	2024-01-29 15:41:26.313078200 +0100
+++ samba-4.20.1/source3/modules/vfs_zfsacl.c.new	2024-05-19 18:45:05.340824249 +0200
@@ -28,7 +28,7 @@
 #include "nfs4_acls.h"
 
 #ifdef HAVE_FREEBSD_SUNACL_H
-#include "sunacl.h"
+#include "sys/acl.h"
 #endif
 
 #undef DBGC_CLASS
--- samba-4.20.1/source3/modules/vfs_posixacl.c	2024-01-29 15:41:26.309078000 +0100
+++ samba-4.20.1/source3/modules/vfs_posixacl.c.new	2024-05-20 19:01:42.541478333 +0200
@@ -17,6 +17,7 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include "sys/acl.h"
 #include "includes.h"
 #include "system/filesys.h"
 #include "smbd/smbd.h"
