diff -Naur samba-4.10.0.old/buildtools/wafsamba/samba_conftests.py samba-4.10.0/buildtools/wafsamba/samba_conftests.py
--- samba-4.10.0.old/buildtools/wafsamba/samba_conftests.py	2019-03-19 20:23:41.107626056 +0000
+++ samba-4.10.0/buildtools/wafsamba/samba_conftests.py	2019-03-19 20:26:38.211897527 +0000
@@ -487,7 +487,7 @@
     if not conf.CONFIG_SET('XSLTPROC'):
         return False
 
-    s='http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl'
+    s='file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl'
     conf.CHECK_COMMAND('%s --nonet %s 2> /dev/null' % (conf.env.get_flat('XSLTPROC'), s),
                              msg='Checking for stylesheet %s' % s,
                              define='XSLTPROC_MANPAGES', on_target=False,
diff -Naur samba-4.10.0.old/buildtools/wafsamba/wafsamba.py samba-4.10.0/buildtools/wafsamba/wafsamba.py
--- samba-4.10.0.old/buildtools/wafsamba/wafsamba.py	2019-03-19 20:23:41.108093940 +0000
+++ samba-4.10.0/buildtools/wafsamba/wafsamba.py	2019-03-19 20:27:29.508298852 +0000
@@ -903,7 +903,7 @@
 
 def MANPAGES(bld, manpages, install):
     '''build and install manual pages'''
-    bld.env.MAN_XSL = 'http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl'
+    bld.env.MAN_XSL = 'file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl'
     for m in manpages.split():
         source = m + '.xml'
         bld.SAMBA_GENERATOR(m,
diff -Naur samba-4.10.0.old/docs-xml/xslt/man.xsl samba-4.10.0/docs-xml/xslt/man.xsl
--- samba-4.10.0.old/docs-xml/xslt/man.xsl	2019-03-19 20:23:40.789515250 +0000
+++ samba-4.10.0/docs-xml/xslt/man.xsl	2019-03-19 20:28:10.503880732 +0000
@@ -4,7 +4,7 @@
 	        xmlns:xlink='http://www.w3.org/1999/xlink'
                 version='1.0'>
 
-<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"/>
+<xsl:import href="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"/>
 
 <xsl:param name="chunk.section.depth" select="0"/>
 <xsl:param name="chunk.first.sections" select="1"/>
diff -Naur samba-4.10.0.old/lib/ldb/docs/builddocs.sh samba-4.10.0/lib/ldb/docs/builddocs.sh
--- samba-4.10.0.old/lib/ldb/docs/builddocs.sh	2019-03-19 20:23:43.084361105 +0000
+++ samba-4.10.0/lib/ldb/docs/builddocs.sh	2019-03-19 20:29:15.932143106 +0000
@@ -10,7 +10,7 @@
     exit 0
 fi
 
-MANXSL="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"
+MANXSL="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"
 HTMLXSL="http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl"
 
 mkdir -p man
