--- samba-4.15.6/source3/libads/kerberos_keytab.c	2021-08-09 15:38:36.387380800 +0000
+++ samba-4.15.6/source3/libads/kerberos_keytab.c.new	2022-03-16 14:57:33.195601566 +0000
@@ -590,11 +590,11 @@
 		}
 	}
 
-#if 0	/* don't create the CIFS/... keytab entries since no one except smbd
+#if 1	/* don't create the CIFS/... keytab entries since no one except smbd
 	   really needs them and we will fall back to verifying against
 	   secrets.tdb */
 
-	ret = ads_keytab_add_entry(ads, "cifs", false));
+	ret = ads_keytab_add_entry(ads, "cifs", false);
 	if (ret != 0 ) {
 		DEBUG(1, (__location__ ": ads_keytab_add_entry failed while "
 			  "adding 'cifs'.\n"));
