--- samba-4.17.0/lib/tsocket/tsocket_bsd.c	2022-08-08 07:15:39.152190400 +0000
+++ samba-4.17.0/lib/tsocket/tsocket_bsd.c.new	2022-09-17 05:56:02.043645009 +0000
@@ -457,7 +457,7 @@
 	char *buf = NULL;
 	uint64_t port = 0;
 	int ret;
-	char *s_addr = NULL;
+	char *so_addr = NULL;
 	uint16_t s_port = default_port;
 	bool conv_ret;
 	bool is_ipv6_by_squares = false;
@@ -486,7 +486,7 @@
 		/* IPv6 possibly with port - squares detected */
 		port_sep = pr_sq + 1;
 		if (*port_sep == '\0') {
-			s_addr = pl_sq + 1;
+			so_addr = pl_sq + 1;
 			*pr_sq = 0;
 			s_port = default_port;
 			goto get_addr;
@@ -508,7 +508,7 @@
 		s_port = (uint16_t)port;
 		*port_sep = 0;
 		*pr_sq = 0;
-		s_addr = pl_sq + 1;
+		so_addr = pl_sq + 1;
 		*pl_sq = 0;
 		goto get_addr;
 	} else if (pl_period != NULL && port_sep != NULL) {
@@ -525,18 +525,18 @@
 		}
 		s_port = (uint16_t)port;
 		*port_sep = 0;
-		s_addr = buf;
+		so_addr = buf;
 		goto get_addr;
 	} else {
 		/* Everything else, let tsocket_address_inet_from string() */
 		/* find parsing errors */
-		s_addr = buf;
+		so_addr = buf;
 		s_port = default_port;
 		goto get_addr;
 	}
 get_addr:
 	ret = _tsocket_address_inet_from_strings(
-	    mem_ctx, fam, s_addr, s_port, _addr, location);
+	    mem_ctx, fam, so_addr, s_port, _addr, location);
 
 	return ret;
 }
