diff -Naur samba-4.10.0.old/source4/scripting/bin/samba_dnsupdate samba-4.10.0/source4/scripting/bin/samba_dnsupdate
--- samba-4.10.0.old/source4/scripting/bin/samba_dnsupdate	2019-03-19 20:23:41.912133734 +0000
+++ samba-4.10.0/source4/scripting/bin/samba_dnsupdate	2019-03-19 20:41:48.309912230 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.5
 # vim: expandtab
 #
 # update our DNS names using TSIG-GSS
diff -Naur samba-4.10.0.old/source4/scripting/bin/samba_kcc samba-4.10.0/source4/scripting/bin/samba_kcc
--- samba-4.10.0.old/source4/scripting/bin/samba_kcc	2019-03-19 20:23:41.907199129 +0000
+++ samba-4.10.0/source4/scripting/bin/samba_kcc	2019-03-19 20:42:08.538954480 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.5
 #
 # Compute our KCC topology
 #
diff -Naur samba-4.10.0.old/source4/scripting/bin/samba_spnupdate samba-4.10.0/source4/scripting/bin/samba_spnupdate
--- samba-4.10.0.old/source4/scripting/bin/samba_spnupdate	2019-03-19 20:23:41.908194895 +0000
+++ samba-4.10.0/source4/scripting/bin/samba_spnupdate	2019-03-19 20:42:27.552396666 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.5
 #
 # update our servicePrincipalName names from spn_update_list
 #
diff -Naur samba-4.10.0.old/source4/scripting/bin/samba_upgradedns samba-4.10.0/source4/scripting/bin/samba_upgradedns
--- samba-4.10.0.old/source4/scripting/bin/samba_upgradedns	2019-03-19 20:23:41.909145111 +0000
+++ samba-4.10.0/source4/scripting/bin/samba_upgradedns	2019-03-19 20:42:42.891104487 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.5
 #
 # Unix SMB/CIFS implementation.
 # Copyright (C) Amitay Isaacs <amitay@gmail.com> 2012
diff -Naur samba-4.10.0.old/source4/scripting/bin/samba-tool samba-4.10.0/source4/scripting/bin/samba-tool
--- samba-4.10.0.old/source4/scripting/bin/samba-tool	2019-03-19 20:23:41.908369641 +0000
+++ samba-4.10.0/source4/scripting/bin/samba-tool	2019-03-19 20:43:00.784339006 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.5
 
 # Unix SMB/CIFS implementation.
 # Copyright (C) Jelmer Vernooij <jelmer@samba.org> 2008-2012
