diff -ur samba-4.1.17/buildtools/wafsamba/samba_conftests.py samba-4.1.17.mod/buildtools/wafsamba/samba_conftests.py
--- samba-4.1.17/buildtools/wafsamba/samba_conftests.py	2013-12-05 13:16:48.000000000 +0400
+++ samba-4.1.17.mod/buildtools/wafsamba/samba_conftests.py	2015-08-20 15:49:55.727324116 +0300
@@ -498,7 +498,7 @@
     if not conf.CONFIG_SET('XSLTPROC'):
         return False
 
-    s='http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl'
+    s='file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl'
     conf.CHECK_COMMAND('%s --nonet %s 2> /dev/null' % (conf.env.XSLTPROC, s),
                              msg='Checking for stylesheet %s' % s,
                              define='XSLTPROC_MANPAGES', on_target=False,
diff -ur samba-4.1.17/buildtools/wafsamba/wafsamba.py samba-4.1.17.mod/buildtools/wafsamba/wafsamba.py
--- samba-4.1.17/buildtools/wafsamba/wafsamba.py	2015-01-12 10:43:53.000000000 +0300
+++ samba-4.1.17.mod/buildtools/wafsamba/wafsamba.py	2015-08-20 15:49:55.704547426 +0300
@@ -832,7 +832,7 @@
 
 def MANPAGES(bld, manpages, install):
     '''build and install manual pages'''
-    bld.env.MAN_XSL = 'http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl'
+    bld.env.MAN_XSL = 'file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl'
     for m in manpages.split():
         source = m + '.xml'
         bld.SAMBA_GENERATOR(m,
diff -ur samba-4.1.17/docs-xml/xslt/man.xsl samba-4.1.17.mod/docs-xml/xslt/man.xsl
--- samba-4.1.17/docs-xml/xslt/man.xsl	2013-06-13 13:21:02.000000000 +0400
+++ samba-4.1.17.mod/docs-xml/xslt/man.xsl	2015-08-20 15:49:55.733590328 +0300
@@ -4,7 +4,7 @@
 	        xmlns:xlink='http://www.w3.org/1999/xlink'
                 version='1.0'>
 
-<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"/>
+<xsl:import href="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"/>
 
 <xsl:param name="chunk.section.depth" select="0"/>
 <xsl:param name="chunk.first.sections" select="1"/>
diff -ur samba-4.1.17/lib/ldb/docs/builddocs.sh samba-4.1.17.mod/lib/ldb/docs/builddocs.sh
--- samba-4.1.17/lib/ldb/docs/builddocs.sh	2013-06-13 13:21:02.000000000 +0400
+++ samba-4.1.17.mod/lib/ldb/docs/builddocs.sh	2015-08-20 15:49:55.740040198 +0300
@@ -10,7 +10,7 @@
     exit 0
 fi
 
-MANXSL="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"
+MANXSL="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"
 HTMLXSL="http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl"
 
 mkdir -p man
