--- libmicrohttpd-0.9.52/configure.ac.~1~	2016-10-17 20:08:16.000000000 +0300
+++ libmicrohttpd-0.9.52/configure.ac	2016-12-13 18:56:13.282524062 +0300
@@ -36,7 +36,7 @@
 
 # check for gettext
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT_VERSION([0.19.7])
 
 AC_MSG_CHECKING([[whether z/OS special settings are required]])
 if test `uname -s` = "OS/390"
@@ -1025,7 +1025,7 @@
           [
            AC_MSG_RESULT([[yes]])
            have_gnutls=yes
-           GNUTLS_CPPLAGS="$GNUTLS_CFLAGS"
+           GNUTLS_CPPFLAGS="$GNUTLS_CFLAGS"
            GNUTLS_LDFLAGS="$GNUTLS_LIBS"
            AC_MSG_CHECKING([[for gnutls_privkey_import_x509_raw()]])
            AC_LINK_IFELSE([
