--- mypy-1.1.1/mypyc/test/test_external.py.orig
+++ mypy-1.1.1/mypyc/test/test_external.py
@@ -24,7 +24,7 @@
             cppflags += ["-mmacosx-version-min=10.10", "-stdlib=libc++"]
         env["CPPFLAGS"] = " ".join(cppflags)
         subprocess.check_call(
-            ["make", "libgtest.a"],
+            ["gmake", "libgtest.a"],
             env=env,
             cwd=os.path.join(base_dir, "mypyc", "external", "googletest", "make"),
         )
