https://github.com/python/mypy/issues/14522

--- mypy-0.991/tox.ini.orig
+++ mypy-0.991/tox.ini
@@ -14,7 +14,10 @@
 
 [testenv]
 description = run the test driver with {basepython}
-passenv = PYTEST_XDIST_WORKER_COUNT PROGRAMDATA PROGRAMFILES(X86)
+passenv =
+    PYTEST_XDIST_WORKER_COUNT
+    PROGRAMDATA
+    PROGRAMFILES(X86)
 deps = -rtest-requirements.txt
 commands = python -m pytest {posargs}
 
