It is hard to normalize pytest-xdist test results

--- mypy-1.1.1/pytest.ini.orig
+++ mypy-1.1.1/pytest.ini
@@ -18,10 +18,9 @@
 python_classes =
 python_functions =
 
-# always run in parallel (requires pytest-xdist, see test-requirements.txt)
-# and enable strict mode: require all markers
+# enable strict mode: require all markers
 # to be defined and raise on invalid config values
-addopts = -nauto --strict-markers --strict-config
+addopts = --strict-markers --strict-config
 
 # treat xpasses as test failures so they get converted to regular tests as soon as possible
 xfail_strict = true
