https://github.com/python/mypy/issues/16255

--- mypy-1.6.1/pyproject.toml.orig
+++ mypy-1.6.1/pyproject.toml
@@ -105,13 +105,13 @@
 
 [tool.coverage.run]
 branch = true
-source = "mypy"
+source = ["mypy"]
 parallel = true
 
 [tool.coverage.report]
 show_missing = true
 skip_covered = true
-omit = 'mypy/test/*'
+omit = ['mypy/test/*']
 exclude_lines = [
   '\#\s*pragma: no cover',
   '^\s*raise AssertionError\b',
