https://github.com/pypa/setuptools_scm/issues/953

--- setuptools-scm-8.0.4/src/setuptools_scm/_integration/setuptools.py.orig
+++ setuptools-scm-8.0.4/src/setuptools_scm/_integration/setuptools.py
@@ -116,6 +116,6 @@
     try:
         config = _config.Configuration.from_file(dist_name=dist_name)
     except LookupError as e:
-        log.warning(e)
+        log.info(e)
     else:
         _assign_version(dist, config)
