diff -ur cmake-2.8.12.orig/Modules/CMakeDetermineJavaCompiler.cmake cmake-2.8.12/Modules/CMakeDetermineJavaCompiler.cmake
--- cmake-2.8.12.orig/Modules/CMakeDetermineJavaCompiler.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/CMakeDetermineJavaCompiler.cmake	2015-06-03 09:26:30.105901688 -0400
@@ -55,9 +55,10 @@
     /usr/bin
     /usr/lib/java/bin
     /usr/share/java/bin
-    /usr/local/bin
-    /usr/local/java/bin
-    /usr/local/java/share/bin
+# Uncomment the following to enable /usr/local search
+#    /usr/local/bin
+#    /usr/local/java/bin
+#    /usr/local/java/share/bin
     /usr/java/j2sdk1.4.2_04
     /usr/lib/j2sdk1.4-sun/bin
     /usr/java/j2sdk1.4.2_09/bin
diff -ur cmake-2.8.12.orig/Modules/CMakeDetermineSystem.cmake cmake-2.8.12/Modules/CMakeDetermineSystem.cmake
--- cmake-2.8.12.orig/Modules/CMakeDetermineSystem.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/CMakeDetermineSystem.cmake	2015-06-03 09:26:30.107633536 -0400
@@ -43,7 +43,7 @@
 
 # find out on which system cmake runs
 if(CMAKE_HOST_UNIX)
-  find_program(CMAKE_UNAME uname /bin /usr/bin /usr/local/bin )
+  find_program(CMAKE_UNAME uname /bin /usr/bin )
   if(CMAKE_UNAME)
     exec_program(uname ARGS -s OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_NAME)
     exec_program(uname ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
diff -ur cmake-2.8.12.orig/Modules/FindFLTK.cmake cmake-2.8.12/Modules/FindFLTK.cmake
--- cmake-2.8.12.orig/Modules/FindFLTK.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindFLTK.cmake	2015-06-03 09:26:30.108627970 -0400
@@ -101,9 +101,12 @@
 
     PATHS
     # Look in standard UNIX install locations.
-    /usr/local/lib/fltk
+
+# Uncomment the following to enable /usr/local search
+#    /usr/local/lib/fltk
     /usr/lib/fltk
-    /usr/local/fltk
+# Uncomment the following to enable /usr/local search
+#    /usr/local/fltk
     /usr/X11R6/include
 
     # Read from the CMakeSetup registry entries.  It is likely that
@@ -217,7 +220,6 @@
     endif()
 
     set(FLTK_INCLUDE_SEARCH_PATH ${FLTK_INCLUDE_SEARCH_PATH}
-      /usr/local/fltk
       /usr/X11R6/include
       ${_FLTK_POSSIBLE_INCLUDE_DIRS}
       )
@@ -238,7 +240,6 @@
     endif()
 
     set(FLTK_LIBRARY_SEARCH_PATH ${FLTK_LIBRARY_SEARCH_PATH}
-      /usr/local/fltk/lib
       /usr/X11R6/lib
       ${FLTK_INCLUDE_DIR}/lib
       ${_FLTK_POSSIBLE_LIBRARY_DIR}
diff -ur cmake-2.8.12.orig/Modules/FindFLTK2.cmake cmake-2.8.12/Modules/FindFLTK2.cmake
--- cmake-2.8.12.orig/Modules/FindFLTK2.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindFLTK2.cmake	2015-06-03 09:26:30.109546757 -0400
@@ -79,9 +79,12 @@
     ${FLTK2_DIR_SEARCH}
 
     # Look in standard UNIX install locations.
-    /usr/local/lib/fltk2
+
+# Uncomment the following to enable /usr/local search
+#    /usr/local/lib/fltk2
     /usr/lib/fltk2
-    /usr/local/fltk2
+# Uncomment the following to enable /usr/local search
+#    /usr/local/fltk2
     /usr/X11R6/include
 
     # Read from the CMakeSetup registry entries.  It is likely that
@@ -187,14 +190,12 @@
     endif()
 
     set(FLTK2_INCLUDE_SEARCH_PATH ${FLTK2_INCLUDE_SEARCH_PATH}
-      /usr/local/fltk2
       /usr/X11R6/include
       )
 
     find_path(FLTK2_INCLUDE_DIR fltk/run.h ${FLTK2_INCLUDE_SEARCH_PATH})
 
     set(FLTK2_LIBRARY_SEARCH_PATH ${FLTK2_LIBRARY_SEARCH_PATH}
-      /usr/local/fltk2/lib
       /usr/X11R6/lib
       ${FLTK2_INCLUDE_DIR}/lib
       )
diff -ur cmake-2.8.12.orig/Modules/FindFreetype.cmake cmake-2.8.12/Modules/FindFreetype.cmake
--- cmake-2.8.12.orig/Modules/FindFreetype.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindFreetype.cmake	2015-06-03 09:28:02.238736903 -0400
@@ -47,8 +47,6 @@
     ENV FREETYPE_DIR
   PATHS
     /usr/X11R6
-    /usr/local/X11R6
-    /usr/local/X11
     /usr/freeware
     ENV GTKMM_BASEPATH
     [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]
@@ -61,8 +59,6 @@
     ENV FREETYPE_DIR
   PATHS
     /usr/X11R6
-    /usr/local/X11R6
-    /usr/local/X11
     /usr/freeware
     ENV GTKMM_BASEPATH
     [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]
@@ -77,8 +73,6 @@
   PATH_SUFFIXES lib
   PATHS
     /usr/X11R6
-    /usr/local/X11R6
-    /usr/local/X11
     /usr/freeware
     ENV GTKMM_BASEPATH
     [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]
diff -ur cmake-2.8.12.orig/Modules/FindGDAL.cmake cmake-2.8.12/Modules/FindGDAL.cmake
--- cmake-2.8.12.orig/Modules/FindGDAL.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindGDAL.cmake	2015-06-03 09:26:30.112333376 -0400
@@ -53,7 +53,8 @@
       /Library/Frameworks/gdal.framework/Headers
       /sw # Fink
       /opt/local # DarwinPorts
-      /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#      /opt/csw # Blastwave
       /opt
 )
 
@@ -70,7 +71,8 @@
         PATHS
             /sw # Fink
             /opt/local # DarwinPorts
-            /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#            /opt/csw # Blastwave
             /opt
     )
 
@@ -95,7 +97,8 @@
   PATHS
     /sw
     /opt/local
-    /opt/csw
+# Uncomment the following to enable /opt/csw search
+#    /opt/csw
     /opt
     /usr/freeware
 )
diff -ur cmake-2.8.12.orig/Modules/FindGTK.cmake cmake-2.8.12/Modules/FindGTK.cmake
--- cmake-2.8.12.orig/Modules/FindGTK.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindGTK.cmake	2015-06-03 09:26:30.113396597 -0400
@@ -36,9 +36,11 @@
   find_path( GTK_glibconfig_INCLUDE_PATH NAMES glibconfig.h
     PATHS
     /usr/openwin/share/include
-    /usr/local/include/glib12
+# Uncomment the following to enable /usr/local search
+#    /usr/local/include/glib12
     /usr/lib/glib/include
-    /usr/local/lib/glib/include
+# Uncomment the following to enable /usr/local search
+#    /usr/local/lib/glib/include
     /opt/gnome/include
     /opt/gnome/lib/glib/include
   )
diff -ur cmake-2.8.12.orig/Modules/FindGTK2.cmake cmake-2.8.12/Modules/FindGTK2.cmake
--- cmake-2.8.12.orig/Modules/FindGTK2.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindGTK2.cmake	2015-06-03 09:26:30.115038103 -0400
@@ -217,8 +217,6 @@
     find_path(${_var}_INCLUDE_DIR ${_hdr}
         PATHS
             ${_gtk2_arch_dir}
-            /usr/local/lib64
-            /usr/local/lib
             /usr/lib64
             /usr/lib
             /opt/gnome/include
diff -ur cmake-2.8.12.orig/Modules/FindJNI.cmake cmake-2.8.12/Modules/FindJNI.cmake
--- cmake-2.8.12.orig/Modules/FindJNI.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindJNI.cmake	2015-06-03 09:26:30.116199579 -0400
@@ -102,12 +102,14 @@
   ${_JAVA_HOME}/lib
   ${_JAVA_HOME}
   /usr/lib
-  /usr/local/lib
+# Uncomment the following to enable /usr/local search
+#  /usr/local/lib
   /usr/lib/jvm/java/lib
   /usr/lib/java/jre/lib/{libarch}
   /usr/lib/jvm/jre/lib/{libarch}
-  /usr/local/lib/java/jre/lib/{libarch}
-  /usr/local/share/java/jre/lib/{libarch}
+# Uncomment the following to enable /usr/local search
+#  /usr/local/lib/java/jre/lib/{libarch}
+#  /usr/local/share/java/jre/lib/{libarch}
   /usr/lib/j2sdk1.4-sun/jre/lib/{libarch}
   /usr/lib/j2sdk1.5-sun/jre/lib/{libarch}
   /opt/sun-jdk-1.5.0.04/jre/lib/{libarch}
@@ -144,15 +146,12 @@
   "[HKEY_LOCAL_MACHINE\\SOFTWARE\\JavaSoft\\Java Development Kit\\${java_install_version};JavaHome]/include"
   ${_JAVA_HOME}/include
   /usr/include
-  /usr/local/include
   /usr/lib/java/include
-  /usr/local/lib/java/include
   /usr/lib/jvm/java/include
   /usr/lib/jvm/java-6-sun/include
   /usr/lib/jvm/java-1.5.0-sun/include
   /usr/lib/jvm/java-6-sun-1.6.0.00/include       # can this one be removed according to #8821 ? Alex
   /usr/lib/jvm/java-6-openjdk/include
-  /usr/local/share/java/include
   /usr/lib/j2sdk1.4-sun/include
   /usr/lib/j2sdk1.5-sun/include
   /opt/sun-jdk-1.5.0.04/include
diff -ur cmake-2.8.12.orig/Modules/FindJava.cmake cmake-2.8.12/Modules/FindJava.cmake
--- cmake-2.8.12.orig/Modules/FindJava.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindJava.cmake	2015-06-03 09:26:30.117271669 -0400
@@ -73,8 +73,9 @@
 set(_JAVA_PATHS
   /usr/lib/java/bin
   /usr/share/java/bin
-  /usr/local/java/bin
-  /usr/local/java/share/bin
+# Uncomment the following to enable /usr/local search
+#  /usr/local/java/bin
+#  /usr/local/java/share/bin
   /usr/java/j2sdk1.4.2_04
   /usr/lib/j2sdk1.4-sun/bin
   /usr/java/j2sdk1.4.2_09/bin
diff -ur cmake-2.8.12.orig/Modules/FindKDE3.cmake cmake-2.8.12/Modules/FindKDE3.cmake
--- cmake-2.8.12.orig/Modules/FindKDE3.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindKDE3.cmake	2015-06-03 09:26:30.118154306 -0400
@@ -141,7 +141,8 @@
   /opt/kde3/include
   /opt/kde/include
   /usr/include/kde
-  /usr/local/include/kde
+# Uncomment the following to enable /usr/local search
+#  /usr/local/include/kde
   )
 
 #now the KDE library directory
diff -ur cmake-2.8.12.orig/Modules/FindMatlab.cmake cmake-2.8.12/Modules/FindMatlab.cmake
--- cmake-2.8.12.orig/Modules/FindMatlab.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindMatlab.cmake	2015-06-03 09:26:30.118979980 -0400
@@ -60,7 +60,6 @@
   if(CMAKE_SIZEOF_VOID_P EQUAL 4)
     # Regular x86
     set(MATLAB_ROOT
-      /usr/local/matlab-7sp1/bin/glnx86/
       /opt/matlab-7sp1/bin/glnx86/
       $ENV{HOME}/matlab-7sp1/bin/glnx86/
       $ENV{HOME}/redhat-matlab/bin/glnx86/
@@ -68,7 +67,6 @@
   else()
     # AMD64:
     set(MATLAB_ROOT
-      /usr/local/matlab-7sp1/bin/glnxa64/
       /opt/matlab-7sp1/bin/glnxa64/
       $ENV{HOME}/matlab7_64/bin/glnxa64/
       $ENV{HOME}/matlab-7sp1/bin/glnxa64/
@@ -89,7 +87,8 @@
     )
   find_path(MATLAB_INCLUDE_DIR
     "mex.h"
-    "/usr/local/matlab-7sp1/extern/include/"
+# Uncomment the following to enable /usr/local search
+#    "/usr/local/matlab-7sp1/extern/include/"
     "/opt/matlab-7sp1/extern/include/"
     "$ENV{HOME}/matlab-7sp1/extern/include/"
     "$ENV{HOME}/redhat-matlab/extern/include/"
diff -ur cmake-2.8.12.orig/Modules/FindOpenThreads.cmake cmake-2.8.12/Modules/FindOpenThreads.cmake
--- cmake-2.8.12.orig/Modules/FindOpenThreads.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindOpenThreads.cmake	2015-06-03 09:26:30.119800475 -0400
@@ -70,7 +70,8 @@
     PATHS
         /sw # Fink
         /opt/local # DarwinPorts
-        /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#        /opt/csw # Blastwave
         /opt
         /usr/freeware
     PATH_SUFFIXES include
@@ -92,7 +93,8 @@
     PATHS
         /sw
         /opt/local
-        /opt/csw
+# Uncomment the following to enable /opt/csw search
+#        /opt/csw
         /opt
         /usr/freeware
     PATH_SUFFIXES lib
@@ -114,7 +116,8 @@
     PATHS
         /sw
         /opt/local
-        /opt/csw
+# Uncomment the following to enable /opt/csw search
+#        /opt/csw
         /opt
         /usr/freeware
     PATH_SUFFIXES lib
diff -ur cmake-2.8.12.orig/Modules/FindPHP4.cmake cmake-2.8.12/Modules/FindPHP4.cmake
--- cmake-2.8.12.orig/Modules/FindPHP4.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindPHP4.cmake	2015-06-03 09:26:30.120587298 -0400
@@ -22,10 +22,12 @@
 
 set(PHP4_POSSIBLE_INCLUDE_PATHS
   /usr/include/php4
-  /usr/local/include/php4
+# Uncomment the following to enable /usr/local search
+#  /usr/local/include/php4
   /usr/include/php
-  /usr/local/include/php
-  /usr/local/apache/php
+# Uncomment the following to enable /usr/local search
+#  /usr/local/include/php
+#  /usr/local/apache/php
   )
 
 set(PHP4_POSSIBLE_LIB_PATHS
diff -ur cmake-2.8.12.orig/Modules/FindPike.cmake cmake-2.8.12/Modules/FindPike.cmake
--- cmake-2.8.12.orig/Modules/FindPike.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindPike.cmake	2015-06-03 09:26:30.121336015 -0400
@@ -22,7 +22,9 @@
 
 file(GLOB PIKE_POSSIBLE_INCLUDE_PATHS
   /usr/include/pike/*
-  /usr/local/include/pike/*)
+# Uncomment the following to enable /usr/local search
+#  /usr/local/include/pike/*)
+  )
 
 find_path(PIKE_INCLUDE_PATH program.h
   ${PIKE_POSSIBLE_INCLUDE_PATHS})
diff -ur cmake-2.8.12.orig/Modules/FindQt.cmake cmake-2.8.12/Modules/FindQt.cmake
--- cmake-2.8.12.orig/Modules/FindQt.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindQt.cmake	2015-06-03 09:26:30.122129492 -0400
@@ -98,8 +98,9 @@
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/include/Qt"
   ${qt_headers}/Qt
   $ENV{QTDIR}/include/Qt
-  /usr/local/qt/include/Qt
-  /usr/local/include/Qt
+# Uncomment the following to enable /usr/local search
+#  /usr/local/qt/include/Qt
+#  /usr/local/include/Qt
   /usr/lib/qt/include/Qt
   /usr/include/Qt
   /usr/share/qt4/include/Qt
@@ -118,8 +119,9 @@
   C:/Qt/3.3.3Educational/include
   $ENV{QTDIR}/include
   /usr/include/qt3/Qt
-  /usr/local/qt/include
-  /usr/local/include
+# Uncomment the following to enable /usr/local search
+#  /usr/local/qt/include
+#  /usr/local/include
   /usr/lib/qt/include
   /usr/include
   /usr/share/qt3/include
diff -ur cmake-2.8.12.orig/Modules/FindQt3.cmake cmake-2.8.12/Modules/FindQt3.cmake
--- cmake-2.8.12.orig/Modules/FindQt3.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindQt3.cmake	2015-06-03 09:26:30.123039938 -0400
@@ -57,14 +57,12 @@
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]/include/Qt"
   $ENV{QTDIR}/include
   ${GLOB_PATHS}
-  /usr/local/qt/include
   /usr/lib/qt/include
   /usr/lib/qt3/include
   /usr/include/qt
   /usr/share/qt3/include
   C:/Progra~1/qt/include
   /usr/include/qt3
-  /usr/local/include/X11/qt3
   )
 
 # if qglobal.h is not in the qt_include_dir then set
@@ -97,7 +95,6 @@
       "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]"
       ENV QTDIR
       ${GLOB_PATHS_LIB}
-      /usr/local/qt
       /usr/lib/qt
       /usr/lib/qt3
       /usr/share/qt3
@@ -118,7 +115,6 @@
       "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]"
       ENV QTDIR
       ${GLOB_PATHS_LIB}
-      /usr/local/qt
       /usr/lib/qt
       /usr/lib/qt3
       /usr/share/qt3
@@ -137,7 +133,6 @@
     "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]"
     ENV QTDIR
     ${GLOB_PATHS_LIB}
-    /usr/local/qt
     /usr/lib/qt3
     /usr/share/qt3
     C:/Progra~1/qt
@@ -155,8 +150,6 @@
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.2.0;InstallDir]/include/Qt"
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]/include/Qt"
   ${GLOB_PATHS_BIN}
-    /usr/local/lib/qt3
-    /usr/local/qt
     /usr/lib/qt
     /usr/lib/qt3
     /usr/share/qt3
@@ -180,7 +173,6 @@
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.2.0;InstallDir]/include/Qt"
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\3.1.0;InstallDir]/include/Qt"
   ${GLOB_PATHS_BIN}
-    /usr/local/qt
     /usr/lib/qt
     /usr/lib/qt3
     /usr/share/qt3
diff -ur cmake-2.8.12.orig/Modules/FindSDL.cmake cmake-2.8.12/Modules/FindSDL.cmake
--- cmake-2.8.12.orig/Modules/FindSDL.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindSDL.cmake	2015-06-03 09:26:30.124243581 -0400
@@ -96,7 +96,8 @@
       PATHS
       /sw
       /opt/local
-      /opt/csw
+# Uncomment the following to enable /opt/csw search
+#      /opt/csw
       /opt
     )
   endif()
diff -ur cmake-2.8.12.orig/Modules/FindSDL_sound.cmake cmake-2.8.12/Modules/FindSDL_sound.cmake
--- cmake-2.8.12.orig/Modules/FindSDL_sound.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindSDL_sound.cmake	2015-06-03 09:26:30.125202500 -0400
@@ -196,7 +196,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -216,7 +215,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -238,7 +236,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -256,7 +253,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -277,7 +273,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -298,7 +293,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -322,7 +316,6 @@
            ENV SDLDIR
            /sw
            /opt/local
-           /opt/csw
            /opt
          PATH_SUFFIXES
            lib
@@ -344,7 +337,6 @@
              ENV SDLDIR
              /sw
              /opt/local
-             /opt/csw
              /opt
            PATH_SUFFIXES lib
            )
diff -ur cmake-2.8.12.orig/Modules/FindSelfPackers.cmake cmake-2.8.12/Modules/FindSelfPackers.cmake
--- cmake-2.8.12.orig/Modules/FindSelfPackers.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindSelfPackers.cmake	2015-06-03 09:26:30.125987069 -0400
@@ -25,7 +25,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 
@@ -34,7 +33,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 
diff -ur cmake-2.8.12.orig/Modules/FindUnixCommands.cmake cmake-2.8.12/Modules/FindUnixCommands.cmake
--- cmake-2.8.12.orig/Modules/FindUnixCommands.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/FindUnixCommands.cmake	2015-06-03 09:26:30.126838902 -0400
@@ -22,7 +22,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
@@ -34,7 +33,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
@@ -46,7 +44,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
@@ -58,7 +55,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
@@ -70,7 +66,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
@@ -85,7 +80,6 @@
   ${CYGWIN_INSTALL_PATH}/bin
   /bin
   /usr/bin
-  /usr/local/bin
   /sbin
 )
 mark_as_advanced(
diff -ur cmake-2.8.12.orig/Modules/Findosg_functions.cmake cmake-2.8.12/Modules/Findosg_functions.cmake
--- cmake-2.8.12.orig/Modules/Findosg_functions.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/Findosg_functions.cmake	2015-06-03 09:26:30.127655247 -0400
@@ -37,7 +37,8 @@
        PATHS
             /sw # Fink
             /opt/local # DarwinPorts
-            /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#            /opt/csw # Blastwave
             /opt
             /usr/freeware
    )
@@ -63,7 +64,8 @@
        PATHS
             /sw # Fink
             /opt/local # DarwinPorts
-            /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#            /opt/csw # Blastwave
             /opt
             /usr/freeware
    )
@@ -81,7 +83,8 @@
        PATHS
             /sw # Fink
             /opt/local # DarwinPorts
-            /opt/csw # Blastwave
+# Uncomment the following to enable /opt/csw search
+#            /opt/csw # Blastwave
             /opt
             /usr/freeware
     )
diff -ur cmake-2.8.12.orig/Modules/GetPrerequisites.cmake cmake-2.8.12/Modules/GetPrerequisites.cmake
--- cmake-2.8.12.orig/Modules/GetPrerequisites.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/GetPrerequisites.cmake	2015-06-03 09:26:30.128590038 -0400
@@ -575,7 +575,8 @@
     "C:/Program Files (x86)/Microsoft Visual Studio 8/VC/BIN"
     "C:/Program Files/Microsoft Visual Studio .NET 2003/VC7/BIN"
     "C:/Program Files (x86)/Microsoft Visual Studio .NET 2003/VC7/BIN"
-    "/usr/local/bin"
+# Uncomment the following to enable /usr/local search
+#    "/usr/local/bin"
     "/usr/bin"
     )
 
diff -ur cmake-2.8.12.orig/Modules/UseJava.cmake cmake-2.8.12/Modules/UseJava.cmake
--- cmake-2.8.12.orig/Modules/UseJava.cmake	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Modules/UseJava.cmake	2015-06-03 09:26:30.129584327 -0400
@@ -534,7 +534,8 @@
     set(_jar_versions)
     set(_jar_paths
         /usr/share/java/
-        /usr/local/share/java/
+# Uncomment the following to enable /usr/local search
+#        /usr/local/share/java/
         ${Java_JAR_PATHS})
     set(_jar_doc "NOTSET")
 
diff -ur cmake-2.8.12.orig/Source/cmDocumentation.cxx cmake-2.8.12/Source/cmDocumentation.cxx
--- cmake-2.8.12.orig/Source/cmDocumentation.cxx	2013-10-07 11:31:00.000000000 -0400
+++ cmake-2.8.12/Source/cmDocumentation.cxx	2015-06-03 09:26:30.130866568 -0400
@@ -1674,6 +1674,7 @@
 bool cmDocumentation::PrintDocumentationUsage(std::ostream& os)
 {
   this->ClearSections();
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Usage");
   this->AddSectionToPrint("Options");
   if(this->ShowGenerators)
@@ -1700,6 +1701,7 @@
   this->ClearSections();
   this->CreateModulesSection();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(Modules));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Description");
   this->AddSectionToPrint("Modules");
   this->AddSectionToPrint("Copyright");
@@ -1716,6 +1718,7 @@
   this->ClearSections();
   this->CreateCustomModulesSection();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(CustomModules));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Description");
   this->AddSectionToPrint("Custom CMake Modules");
 // the custom modules are most probably not under Kitware's copyright, Alex
@@ -1733,6 +1736,7 @@
 {
   this->ClearSections();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(Policies));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Description");
   this->AddSectionToPrint("Policies");
   this->AddSectionToPrint("Copyright");
@@ -1749,6 +1753,7 @@
 {
   this->ClearSections();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(Properties));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Properties Description");
   for (std::vector<std::string>::iterator i =
          this->PropertySections.begin();
@@ -1769,6 +1774,7 @@
 {
   this->ClearSections();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(Variables));
+  this->AddSectionToPrint("Name");
   for (std::vector<std::string>::iterator i =
          this->VariableSections.begin();
        i != this->VariableSections.end(); ++i)
@@ -1788,6 +1794,7 @@
 {
   this->ClearSections();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(Commands));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Commands");
   this->AddSectionToPrint("Copyright");
   this->AddSectionToPrint("Standard See Also");
@@ -1802,6 +1809,7 @@
 {
   this->ClearSections();
   this->AddDocumentIntroToPrint(GET_DOCUMENT_INTRO(CompatCommands));
+  this->AddSectionToPrint("Name");
   this->AddSectionToPrint("Compatibility Commands Description");
   this->AddSectionToPrint("Compatibility Commands");
   this->AddSectionToPrint("Copyright");
