--- jxrlib-0.2.4/java/cli/build.gradle.orig	2020-05-11 11:07:07.000000000 +0200
+++ jxrlib-0.2.4/java/cli/build.gradle	2025-05-17 15:35:45.000000000 +0200
@@ -4,7 +4,7 @@
 }
 
 group = 'ome.jxrlib-cli'
-version = '0.2.4'
+version = '0.3.0-SNAPSHOT'
 
 mainClassName = 'ome.jxrlib.Main'
 
@@ -29,7 +29,7 @@
 
 dependencies {
     implementation 'ch.qos.logback:logback-classic:1.1.7'
-    implementation 'ome:jxrlib-all:0.2.4'
+    implementation 'ome:jxrlib-all:0.3.0-SNAPSHOT'
     implementation 'args4j:args4j:2.33'
     implementation 'javax.xml.bind:jaxb-api:2.3.0'
     implementation 'com.sun.xml.bind:jaxb-core:2.3.0'
