commit 04ae90d45173f3dbcc3cf1194271e5ba9694bacf
Author: Andreas Grueninger <andreas.grueninger@example.com>
Date:   Tue Oct 25 12:31:44 2016 +0200

    Fix: add header for flocking

diff --git a/qdevices/utils.c b/qdevices/utils.c
index 41850b8..d61e0f2 100644
--- a/qdevices/utils.c
+++ b/qdevices/utils.c
@@ -34,6 +34,8 @@
 
 #include <sys/types.h>
 #include <sys/file.h>
+#include <sys/flock_impl.h>
+#include <fcntl.h>
 #include <arpa/inet.h>
 #include <sys/stat.h>
 
