diff -Nur a/man/cmap_context_get.3.in b/man/cmap_context_get.3.in
--- a/man/cmap_context_get.3.in	2017-09-26 20:23:12.072051528 +0000
+++ b/man/cmap_context_get.3.in	2017-09-26 20:25:14.938961089 +0000
@@ -55,4 +55,4 @@
 
 .SH "SEE ALSO"
 .BR cmap_context_set (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_context_set.3.in b/man/cmap_context_set.3.in
--- a/man/cmap_context_set.3.in	2017-09-26 20:23:12.072081390 +0000
+++ b/man/cmap_context_set.3.in	2017-09-26 20:25:14.974389521 +0000
@@ -57,4 +57,4 @@
 
 .SH "SEE ALSO"
 .BR cmap_context_get (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_dec.3.in b/man/cmap_dec.3.in
--- a/man/cmap_dec.3.in	2017-09-26 20:23:12.072111436 +0000
+++ b/man/cmap_dec.3.in	2017-09-26 20:25:14.967804747 +0000
@@ -89,4 +89,4 @@
 .BR cmap_get (3),
 .BR cmap_set (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_delete.3.in b/man/cmap_delete.3.in
--- a/man/cmap_delete.3.in	2017-09-26 20:23:12.072141728 +0000
+++ b/man/cmap_delete.3.in	2017-09-26 20:25:14.973303335 +0000
@@ -67,4 +67,4 @@
 .SH "SEE ALSO"
 .BR cmap_initialize (3),
 .BR cmap_set (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_dispatch.3.in b/man/cmap_dispatch.3.in
--- a/man/cmap_dispatch.3.in	2017-09-26 20:23:12.072171698 +0000
+++ b/man/cmap_dispatch.3.in	2017-09-26 20:25:14.938122172 +0000
@@ -88,4 +88,4 @@
 .SH "SEE ALSO"
 .BR cmap_track_add (3),
 .BR cmap_track_delete (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_fd_get.3.in b/man/cmap_fd_get.3.in
--- a/man/cmap_fd_get.3.in	2017-09-26 20:23:12.072203609 +0000
+++ b/man/cmap_fd_get.3.in	2017-09-26 20:25:14.965141391 +0000
@@ -64,4 +64,4 @@
 
 .SH "SEE ALSO"
 .BR cmap_dispatch (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_finalize.3.in b/man/cmap_finalize.3.in
--- a/man/cmap_finalize.3.in	2017-09-26 20:23:12.072232330 +0000
+++ b/man/cmap_finalize.3.in	2017-09-26 20:25:14.958140726 +0000
@@ -58,4 +58,4 @@
 .SH "SEE ALSO"
 .BR cmap_initialize (3),
 .BR cmap_dispatch (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_get.3.in b/man/cmap_get.3.in
--- a/man/cmap_get.3.in	2017-09-26 20:23:12.072262630 +0000
+++ b/man/cmap_get.3.in	2017-09-26 20:25:14.966905016 +0000
@@ -150,4 +150,4 @@
 .SH "SEE ALSO"
 .BR cmap_set (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_inc.3.in b/man/cmap_inc.3.in
--- a/man/cmap_inc.3.in	2017-09-26 20:23:12.072291295 +0000
+++ b/man/cmap_inc.3.in	2017-09-26 20:25:14.951567373 +0000
@@ -89,4 +89,4 @@
 .BR cmap_get (3),
 .BR cmap_set (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_initialize.3.in b/man/cmap_initialize.3.in
--- a/man/cmap_initialize.3.in	2017-09-26 20:23:12.072320713 +0000
+++ b/man/cmap_initialize.3.in	2017-09-26 20:25:14.948489199 +0000
@@ -60,4 +60,4 @@
 
 .SH "SEE ALSO"
 .BR cmap_finalize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_iter_finalize.3.in b/man/cmap_iter_finalize.3.in
--- a/man/cmap_iter_finalize.3.in	2017-09-26 20:23:12.072350101 +0000
+++ b/man/cmap_iter_finalize.3.in	2017-09-26 20:25:14.954250299 +0000
@@ -69,4 +69,4 @@
 .SH "SEE ALSO"
 .BR cmap_iter_init (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_iter_init.3.in b/man/cmap_iter_init.3.in
--- a/man/cmap_iter_init.3.in	2017-09-26 20:23:12.072380915 +0000
+++ b/man/cmap_iter_init.3.in	2017-09-26 20:25:14.956859968 +0000
@@ -77,4 +77,4 @@
 .BR cmap_iter_next (3),
 .BR cmap_iter_finalize (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_iter_next.3.in b/man/cmap_iter_next.3.in
--- a/man/cmap_iter_next.3.in	2017-09-26 20:23:12.072411511 +0000
+++ b/man/cmap_iter_next.3.in	2017-09-26 20:25:14.939684069 +0000
@@ -79,4 +79,4 @@
 .BR cmap_iter_finalize (3),
 .BR cmap_initialize (3),
 .BR cmap_get (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_keys.8 b/man/cmap_keys.8
--- a/man/cmap_keys.8	2017-09-26 20:23:12.072444255 +0000
+++ b/man/cmap_keys.8	2017-09-26 20:25:14.964631728 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH "CMAP_KEYS" 8 "10/10/2012" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH "CMAP_KEYS" 1m "10/10/2012" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 
 .SH NAME
 .P
@@ -365,6 +365,6 @@
 * Change the configuration file on all nodes.
 
 .SH "SEE ALSO"
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .BR corosync.conf (5),
-.BR corosync-cmapctl (8)
+.BR corosync-cmapctl (1m)
diff -Nur a/man/cmap_overview.8 b/man/cmap_overview.8
--- a/man/cmap_overview.8	2017-09-26 20:23:12.072473683 +0000
+++ b/man/cmap_overview.8	2017-09-26 20:25:14.973580490 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH "CMAP_OVERVIEW" 8 "03/02/2012" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH "CMAP_OVERVIEW" 1m "03/02/2012" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 
 .SH NAME
 .P
@@ -54,7 +54,7 @@
 .PP
 * Track changes on keys
 
-Description of most keys created by corosync itself can be found in cmap_keys (8).
+Description of most keys created by corosync itself can be found in cmap_keys (1m).
 
 .SH BUGS
 .SH "SEE ALSO"
@@ -74,4 +74,4 @@
 .BR cmap_iter_finalize (3),
 .BR cmap_track_add (3),
 .BR cmap_track_delete (3),
-.BR cmap_keys (8)
+.BR cmap_keys (1m)
diff -Nur a/man/cmap_set.3.in b/man/cmap_set.3.in
--- a/man/cmap_set.3.in	2017-09-26 20:23:12.072502941 +0000
+++ b/man/cmap_set.3.in	2017-09-26 20:25:14.948911132 +0000
@@ -124,4 +124,4 @@
 .SH "SEE ALSO"
 .BR cmap_get (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_track_add.3.in b/man/cmap_track_add.3.in
--- a/man/cmap_track_add.3.in	2017-09-26 20:23:12.072533186 +0000
+++ b/man/cmap_track_add.3.in	2017-09-26 20:25:14.949429664 +0000
@@ -161,4 +161,4 @@
 .BR cmap_initialize (3),
 .BR cmap_get (3),
 .BR cmap_dispatch (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/cmap_track_delete.3.in b/man/cmap_track_delete.3.in
--- a/man/cmap_track_delete.3.in	2017-09-26 20:23:12.072562127 +0000
+++ b/man/cmap_track_delete.3.in	2017-09-26 20:25:14.961143778 +0000
@@ -67,4 +67,4 @@
 .SH "SEE ALSO"
 .BR cmap_track_add (3),
 .BR cmap_initialize (3),
-.BR cmap_overview (8)
+.BR cmap_overview (1m)
diff -Nur a/man/corosync-blackbox.8 b/man/corosync-blackbox.8
--- a/man/corosync-blackbox.8	2017-09-26 20:23:12.072591048 +0000
+++ b/man/corosync-blackbox.8	2017-09-26 20:25:14.952860300 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-BLACKBOX 8 2010-05-30
+.TH COROSYNC-BLACKBOX 1m 2010-05-30
 .SH NAME
 corosync-blackbox \- Dump live "flight data" from the corosync "blackbox".
 .SH SYNOPSIS
@@ -62,8 +62,8 @@
 Finishing replay: records found [2609]
 .br
 .SH SEE ALSO
-.BR qb-blackbox (8),
-.BR corosync-cmapctl (8)
+.BR qb-blackbox (1m),
+.BR corosync-cmapctl (1m)
 .SH AUTHOR
 Angus Salkeld
 .PP
diff -Nur a/man/corosync-cfgtool.8 b/man/corosync-cfgtool.8
--- a/man/corosync-cfgtool.8	2017-09-26 20:23:12.072620217 +0000
+++ b/man/corosync-cfgtool.8	2017-09-26 20:25:14.972765673 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH "COROSYNC-CFGTOOL" "8" "2010-05-30" "" ""
+.TH "COROSYNC-CFGTOOL" "1m" "2010-05-30" "" ""
 .SH "NAME"
 corosync-cfgtool \- An administrative tool for corosync.
 .SH "SYNOPSIS"
@@ -74,7 +74,7 @@
 .B -H
 Shutdown corosync cleanly on this node.
 .SH "SEE ALSO"
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .SH "AUTHOR"
 Angus Salkeld
 .PP 
diff -Nur a/man/corosync-cmapctl.8 b/man/corosync-cmapctl.8
--- a/man/corosync-cmapctl.8	2017-09-26 20:23:12.072649419 +0000
+++ b/man/corosync-cmapctl.8	2017-09-26 20:25:14.957656214 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-CMAP 8 2012-01-23
+.TH COROSYNC-CMAP 1m 2012-01-23
 .SH NAME
 corosync-cmapctl: \- A tool for accessing the object database.
 .SH DESCRIPTION
@@ -81,5 +81,5 @@
 corosync\-cmapctl [\-b] \fB\-T\fR key_prefix
 
 .SH "SEE ALSO"
-.BR cmap_overview (8),
-.BR cmap_keys (8)
+.BR cmap_overview (1m),
+.BR cmap_keys (1m)
diff -Nur a/man/corosync-cpgtool.8 b/man/corosync-cpgtool.8
--- a/man/corosync-cpgtool.8	2017-09-26 20:23:12.072681157 +0000
+++ b/man/corosync-cpgtool.8	2017-09-26 20:25:14.947067984 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-CPGTOOL 8 2010-05-30
+.TH COROSYNC-CPGTOOL 1m 2010-05-30
 .SH NAME
 corosync-cpgtool \- A tool for displaying cpg groups and members.
 .SH SYNOPSIS
@@ -79,7 +79,7 @@
 .br
 example-group
 .SH SEE ALSO
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .SH AUTHOR
 Angus Salkeld
 .PP
diff -Nur a/man/corosync-keygen.8 b/man/corosync-keygen.8
--- a/man/corosync-keygen.8	2017-09-26 20:23:12.072710463 +0000
+++ b/man/corosync-keygen.8	2017-09-26 20:25:14.946768645 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-KEYGEN 8 2010-05-30
+.TH COROSYNC-KEYGEN 1m 2010-05-30
 .SH NAME
 corosync-keygen \- Generate an authentication key for Corosync.
 .SH SYNOPSIS
@@ -91,7 +91,7 @@
 Writing corosync key to /tmp/authkey.
 .br
 .SH SEE ALSO
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .BR corosync.conf (5),
 .SH AUTHOR
 Angus Salkeld
diff -Nur a/man/corosync-notifyd.8 b/man/corosync-notifyd.8
--- a/man/corosync-notifyd.8	2017-09-26 20:23:12.072739182 +0000
+++ b/man/corosync-notifyd.8	2017-09-26 20:25:14.954713360 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-NOTIFYD 8 2011-01-14
+.TH COROSYNC-NOTIFYD 1m 2011-01-14
 .SH NAME
 corosync-notifyd \- Listen for important corosync events and send dbus and/or snmp traps.
 .SH SYNOPSIS
@@ -137,8 +137,8 @@
 .br
    string "left"
 .SH SEE ALSO
-.BR corosync (8),
-.BR corosync-cmapctl (8),
+.BR corosync (1m),
+.BR corosync-cmapctl (1m),
 .BR dbus-monitor (1),
 .SH AUTHOR
 Angus Salkeld
diff -Nur a/man/corosync-qdevice-net-certutil.8 b/man/corosync-qdevice-net-certutil.8
--- a/man/corosync-qdevice-net-certutil.8	2017-09-26 20:23:12.072777444 +0000
+++ b/man/corosync-qdevice-net-certutil.8	2017-09-26 20:25:14.973880531 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QDEVICE-NET-CERTUTIL 8 2016-06-28
+.TH COROSYNC-QDEVICE-NET-CERTUTIL 1m 2016-06-28
 .SH NAME
 corosync-qdevice-net-certutil - tool to generate qdevice model net TLS certificates
 .SH SYNOPSIS
@@ -78,8 +78,8 @@
 .B -n
 Name of the cluster.
 .SH SEE ALSO
-.BR corosync-qnetd (8)
-.BR corosync-qdevice (8)
+.BR corosync-qnetd (1m)
+.BR corosync-qdevice (1m)
 .SH AUTHOR
 Jan Friesse
 .PP
diff -Nur a/man/corosync-qdevice-tool.8 b/man/corosync-qdevice-tool.8
--- a/man/corosync-qdevice-tool.8	2017-09-26 20:23:12.072808914 +0000
+++ b/man/corosync-qdevice-tool.8	2017-09-26 20:25:14.942794927 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QDEVICE-TOOL 8 2016-06-24
+.TH COROSYNC-QDEVICE-TOOL 1m 2016-06-24
 .SH NAME
 corosync-qdevice-tool \- corosync-qdevice control interface.
 .SH SYNOPSIS
@@ -118,8 +118,8 @@
 algorithm decides that the current node shouldn't get a vote),
 voting (without cast vote) and voting (with cast vote).
 .SH SEE ALSO
-.BR corosync-qnetd (8)
-.BR corosync-qdevice (8)
+.BR corosync-qnetd (1m)
+.BR corosync-qdevice (1m)
 .SH AUTHOR
 Jan Friesse
 .PP
diff -Nur a/man/corosync-qdevice.8 b/man/corosync-qdevice.8
--- a/man/corosync-qdevice.8	2017-09-26 20:23:12.072845931 +0000
+++ b/man/corosync-qdevice.8	2017-09-26 20:25:14.943989174 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QDEVICE 8 2016-06-29
+.TH COROSYNC-QDEVICE 1m 2016-06-29
 .SH NAME
 corosync-qdevice \- QDevice daemon
 .SH SYNOPSIS
@@ -309,9 +309,9 @@
 .B net_test_algorithm_enabled
 Enable test algorithm. (if built with --enable-debug on, otherwise off)
 .SH SEE ALSO
-.BR corosync-qdevice-tool (8)
-.BR corosync-qdevice-net-certutil (8)
-.BR corosync-qnetd (8)
+.BR corosync-qdevice-tool (1m)
+.BR corosync-qdevice-net-certutil (1m)
+.BR corosync-qnetd (1m)
 .BR corosync.conf (5)
 .SH AUTHOR
 Jan Friesse
diff -Nur a/man/corosync-qnetd-certutil.8 b/man/corosync-qnetd-certutil.8
--- a/man/corosync-qnetd-certutil.8	2017-09-26 20:23:12.072876250 +0000
+++ b/man/corosync-qnetd-certutil.8	2017-09-26 20:25:14.975407305 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QNETD-CERTUTIL 8 2016-06-28
+.TH COROSYNC-QNETD-CERTUTIL 1m 2016-06-28
 .SH NAME
 corosync-qnetd-certutil - tool to generate qnetd TLS certificates
 .SH SYNOPSIS
@@ -66,8 +66,8 @@
 .B corosync-qnetd-certutil
 is executed as root it tries to copy the owner and group of /etc/corosync/qnetd to all of the created files.
 .SH SEE ALSO
-.BR corosync-qnetd (8)
-.BR corosync-qdevice (8)
+.BR corosync-qnetd (1m)
+.BR corosync-qdevice (1m)
 .SH AUTHOR
 Jan Friesse
 .PP
diff -Nur a/man/corosync-qnetd-tool.8 b/man/corosync-qnetd-tool.8
--- a/man/corosync-qnetd-tool.8	2017-09-26 20:23:12.072906049 +0000
+++ b/man/corosync-qnetd-tool.8	2017-09-26 20:25:14.944412311 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QNETD-TOOL 8 2016-06-23
+.TH COROSYNC-QNETD-TOOL 1m 2016-06-23
 .SH NAME
 corosync-qnetd-tool \- corosync-qnetd control interface.
 .SH SYNOPSIS
@@ -121,8 +121,8 @@
 .B corosync-qdevice
 client. The last ACK/NACK vote (if it exists) is in parentheses.
 .SH SEE ALSO
-.BR corosync-qnetd (8)
-.BR corosync-qdevice (8)
+.BR corosync-qnetd (1m)
+.BR corosync-qdevice (1m)
 .SH AUTHOR
 Jan Friesse
 .PP
diff -Nur a/man/corosync-qnetd.8 b/man/corosync-qnetd.8
--- a/man/corosync-qnetd.8	2017-09-26 20:23:12.072936814 +0000
+++ b/man/corosync-qnetd.8	2017-09-26 20:25:14.965848601 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QNETD 8 2016-06-29
+.TH COROSYNC-QNETD 1m 2016-06-29
 .SH NAME
 corosync-qnetd \- QNet daemon
 .SH SYNOPSIS
@@ -219,9 +219,9 @@
 .B ipc_max_send_size
 Maximum size of a message sent to an IPC client. (10485760)
 .SH SEE ALSO
-.BR corosync-qnetd-tool (8)
-.BR corosync-qnetd-certutil (8)
-.BR corosync-qdevice (8)
+.BR corosync-qnetd-tool (1m)
+.BR corosync-qnetd-certutil (1m)
+.BR corosync-qdevice (1m)
 .SH AUTHOR
 Jan Friesse
 .PP
diff -Nur a/man/corosync-quorumtool.8 b/man/corosync-quorumtool.8
--- a/man/corosync-quorumtool.8	2017-09-26 20:23:12.072967935 +0000
+++ b/man/corosync-quorumtool.8	2017-09-26 20:25:14.949737724 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-QUORUMTOOL 8 2012-01-12
+.TH COROSYNC-QUORUMTOOL 1m 2012-01-12
 .SH NAME
 corosync-quorumtool \- Set and display quorum settings.
 .SH SYNOPSIS
@@ -80,8 +80,8 @@
 .PP
 * Starred items only work if votequorum is the quorum provider for corosync
 .SH SEE ALSO
-.BR corosync_overview (8),
-.BR votequorum_overview (8),
+.BR corosync_overview (1m),
+.BR votequorum_overview (1m),
 .SH AUTHOR
 Angus Salkeld
 .PP
diff -Nur a/man/corosync-xmlproc.8 b/man/corosync-xmlproc.8
--- a/man/corosync-xmlproc.8	2017-09-26 20:23:12.072997040 +0000
+++ b/man/corosync-xmlproc.8	2017-09-26 20:25:14.953677480 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC-XMLPROC 8 2011-12-16
+.TH COROSYNC-XMLPROC 1m 2011-12-16
 .SH NAME
 corosync-xmlproc \- Converts corosync.xml to the Corosync configuration file.
 .SH SYNOPSIS
diff -Nur a/man/corosync.8 b/man/corosync.8
--- a/man/corosync.8	2017-09-26 20:23:12.073024879 +0000
+++ b/man/corosync.8	2017-09-26 20:25:14.966101497 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC 8 2010-05-30
+.TH COROSYNC 1m 2010-05-30
 .SH NAME
 corosync \- The Corosync Cluster Engine.
 .SH SYNOPSIS
@@ -56,13 +56,13 @@
 .B -v
 Display version and SVN revision of Corosync and exit.
 .SH SEE ALSO
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .BR corosync.conf (5),
-.BR cpg_overview (8),
-.BR votequorum_overview (8),
-.BR sam_overview (8),
-.BR cmap_overview (8),
-.BR quorum_overview (8)
+.BR cpg_overview (1m),
+.BR votequorum_overview (1m),
+.BR sam_overview (1m),
+.BR cmap_overview (1m),
+.BR quorum_overview (1m)
 .SH AUTHOR
 Angus Salkeld
 .PP
diff -Nur a/man/corosync.conf.5 b/man/corosync.conf.5
--- a/man/corosync.conf.5	2017-09-26 20:23:12.073062661 +0000
+++ b/man/corosync.conf.5	2017-09-26 20:25:14.941790752 +0000
@@ -554,7 +554,7 @@
 
 The default is syslog and stderr.
 
-Please note, if you are using to_logfile and want to rotate the file, use logrotate(8)
+Please note, if you are using to_logfile and want to rotate the file, use logrotate(1m)
 with the option 
 .B
 copytruncate.
@@ -686,8 +686,8 @@
 The corosync executive configuration file.
 
 .SH "SEE ALSO"
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .BR votequorum (5),
-.BR corosync-qdevice (8),
-.BR logrotate (8)
+.BR corosync-qdevice (1m),
+.BR logrotate (1m)
 .PP
diff -Nur a/man/corosync.xml.5 b/man/corosync.xml.5
--- a/man/corosync.xml.5	2017-09-26 20:23:12.073092512 +0000
+++ b/man/corosync.xml.5	2017-09-26 20:25:14.946153499 +0000
@@ -40,7 +40,7 @@
 
 .SH DESCRIPTION
 The corosync.xml is XML version of corosync.conf file. It is processed by
-.B corosync-xmlproc (8)
+.B corosync-xmlproc (1m)
 to generate valid conf file.
 
 File must be valid XML. File starts with root element
@@ -66,7 +66,7 @@
 XML version of the corosync executive configuration file.
 
 .SH "SEE ALSO"
-.BR corosync_overview (8),
+.BR corosync_overview (1m),
 .BR corosync.conf (5),
-.BR corosync-xmlproc (8)
+.BR corosync-xmlproc (1m)
 .PP
diff -Nur a/man/corosync_overview.8 b/man/corosync_overview.8
--- a/man/corosync_overview.8	2017-09-26 20:23:12.073122837 +0000
+++ b/man/corosync_overview.8	2017-09-26 20:25:14.962792220 +0000
@@ -32,7 +32,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH COROSYNC_OVERVIEW 8 2012-02-13 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH COROSYNC_OVERVIEW 1m 2012-02-13 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 .SH NAME
 corosync_overview \- Corosync overview
 .SH OVERVIEW
@@ -189,7 +189,7 @@
 
 .SH "SEE ALSO"
 .BR corosync.conf (5),
-.BR corosync-keygen (8),
-.BR cpg_overview (8),
-.BR sam_overview (8)
+.BR corosync-keygen (1m),
+.BR cpg_overview (1m),
+.BR sam_overview (1m)
 .PP
diff -Nur a/man/cpg_context_get.3.in b/man/cpg_context_get.3.in
--- a/man/cpg_context_get.3.in	2017-09-26 20:23:12.073150983 +0000
+++ b/man/cpg_context_get.3.in	2017-09-26 20:25:14.945887298 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_context_set.3.in b/man/cpg_context_set.3.in
--- a/man/cpg_context_set.3.in	2017-09-26 20:23:12.073182003 +0000
+++ b/man/cpg_context_set.3.in	2017-09-26 20:25:14.976410524 +0000
@@ -51,7 +51,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_dispatch.3.in b/man/cpg_dispatch.3.in
--- a/man/cpg_dispatch.3.in	2017-09-26 20:23:12.073211135 +0000
+++ b/man/cpg_dispatch.3.in	2017-09-26 20:25:14.972182561 +0000
@@ -93,7 +93,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_fd_get.3.in b/man/cpg_fd_get.3.in
--- a/man/cpg_fd_get.3.in	2017-09-26 20:23:12.073243564 +0000
+++ b/man/cpg_fd_get.3.in	2017-09-26 20:25:14.974662330 +0000
@@ -56,7 +56,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_finalize.3.in b/man/cpg_finalize.3.in
--- a/man/cpg_finalize.3.in	2017-09-26 20:23:12.073272902 +0000
+++ b/man/cpg_finalize.3.in	2017-09-26 20:25:14.961693422 +0000
@@ -52,7 +52,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_initialize.3.in b/man/cpg_initialize.3.in
--- a/man/cpg_initialize.3.in	2017-09-26 20:23:12.073301548 +0000
+++ b/man/cpg_initialize.3.in	2017-09-26 20:25:14.955746229 +0000
@@ -155,7 +155,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_iteration_finalize.3.in b/man/cpg_iteration_finalize.3.in
--- a/man/cpg_iteration_finalize.3.in	2017-09-26 20:23:12.073330494 +0000
+++ b/man/cpg_iteration_finalize.3.in	2017-09-26 20:25:14.956574636 +0000
@@ -64,4 +64,4 @@
 
 .SH "SEE ALSO"
 .BR cpg_iteration_initialize (3),
-.BR cpg_overview (8)
+.BR cpg_overview (1m)
diff -Nur a/man/cpg_iteration_initialize.3.in b/man/cpg_iteration_initialize.3.in
--- a/man/cpg_iteration_initialize.3.in	2017-09-26 20:23:12.073359792 +0000
+++ b/man/cpg_iteration_initialize.3.in	2017-09-26 20:25:14.967237660 +0000
@@ -93,4 +93,4 @@
 .BR cpg_iteration_next (3),
 .BR cpg_iteration_finalize (3),
 .BR cpg_initialize (3),
-.BR cpg_overview (8)
+.BR cpg_overview (1m)
diff -Nur a/man/cpg_iteration_next.3.in b/man/cpg_iteration_next.3.in
--- a/man/cpg_iteration_next.3.in	2017-09-26 20:23:12.073389072 +0000
+++ b/man/cpg_iteration_next.3.in	2017-09-26 20:25:14.972481108 +0000
@@ -82,4 +82,4 @@
 
 .SH "SEE ALSO"
 .BR cpg_iteration_initialize (3),
-.BR cpg_overview (8)
+.BR cpg_overview (1m)
diff -Nur a/man/cpg_join.3.in b/man/cpg_join.3.in
--- a/man/cpg_join.3.in	2017-09-26 20:23:12.073419286 +0000
+++ b/man/cpg_join.3.in	2017-09-26 20:25:14.947868925 +0000
@@ -93,7 +93,7 @@
 .SH ERRORS
 Not all errors are documented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_leave.3.in b/man/cpg_leave.3.in
--- a/man/cpg_leave.3.in	2017-09-26 20:23:12.073448380 +0000
+++ b/man/cpg_leave.3.in	2017-09-26 20:25:14.974160546 +0000
@@ -58,7 +58,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_local_get.3.in b/man/cpg_local_get.3.in
--- a/man/cpg_local_get.3.in	2017-09-26 20:23:12.073476888 +0000
+++ b/man/cpg_local_get.3.in	2017-09-26 20:25:14.959478703 +0000
@@ -53,7 +53,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_mcast_joined.3.in b/man/cpg_mcast_joined.3.in
--- a/man/cpg_mcast_joined.3.in	2017-09-26 20:23:12.073506507 +0000
+++ b/man/cpg_mcast_joined.3.in	2017-09-26 20:25:14.959931594 +0000
@@ -122,7 +122,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_membership_get.3.in b/man/cpg_membership_get.3.in
--- a/man/cpg_membership_get.3.in	2017-09-26 20:23:12.073535304 +0000
+++ b/man/cpg_membership_get.3.in	2017-09-26 20:25:14.961433635 +0000
@@ -62,7 +62,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_model_initialize.3.in b/man/cpg_model_initialize.3.in
--- a/man/cpg_model_initialize.3.in	2017-09-26 20:23:12.073564670 +0000
+++ b/man/cpg_model_initialize.3.in	2017-09-26 20:25:14.971810834 +0000
@@ -211,7 +211,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_overview.8 b/man/cpg_overview.8
--- a/man/cpg_overview.8	2017-09-26 20:23:12.073592507 +0000
+++ b/man/cpg_overview.8	2017-09-26 20:25:14.975698282 +0000
@@ -31,7 +31,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH CPG_OVERVIEW 8 2009-4-15 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH CPG_OVERVIEW 1m 2009-4-15 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 .SH NAME
 cpg_overview \- CPG library overview
 .SH OVERVIEW
@@ -60,7 +60,7 @@
 access the corosync services.
 
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_zcb_alloc.3.in b/man/cpg_zcb_alloc.3.in
--- a/man/cpg_zcb_alloc.3.in	2017-09-26 20:23:12.073620745 +0000
+++ b/man/cpg_zcb_alloc.3.in	2017-09-26 20:25:14.960886674 +0000
@@ -66,7 +66,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_zcb_free.3.in b/man/cpg_zcb_free.3.in
--- a/man/cpg_zcb_free.3.in	2017-09-26 20:23:12.073651058 +0000
+++ b/man/cpg_zcb_free.3.in	2017-09-26 20:25:14.958417545 +0000
@@ -58,7 +58,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/cpg_zcb_mcast_joined.3.in b/man/cpg_zcb_mcast_joined.3.in
--- a/man/cpg_zcb_mcast_joined.3.in	2017-09-26 20:23:12.073679857 +0000
+++ b/man/cpg_zcb_mcast_joined.3.in	2017-09-26 20:25:14.951984152 +0000
@@ -108,7 +108,7 @@
 .SH ERRORS
 The errors are undocumented.
 .SH "SEE ALSO"
-.BR cpg_overview (8),
+.BR cpg_overview (1m),
 .BR cpg_initialize (3),
 .BR cpg_finalize (3),
 .BR cpg_fd_get (3),
diff -Nur a/man/index.html b/man/index.html
--- a/man/index.html	2017-09-26 20:23:12.073709281 +0000
+++ b/man/index.html	2017-09-26 20:25:14.951248601 +0000
@@ -19,7 +19,7 @@
     <p>
       <h2>Daemon, tools and configuration</h2>
 
-      <a href="corosync_overview.8.html">corosync_overview(8)</a>:
+      <a href="corosync_overview.8.html">corosync_overview(1m)</a>:
       Overview of the corosync system.
       <br>
 
@@ -35,43 +35,43 @@
       Description of configuration options for votequorum module in corosync.conf
       <br>
 
-      <a href="corosync.8.html">corosync(8)</a>:
+      <a href="corosync.8.html">corosync(1m)</a>:
       Description of corosync daemon.
       <br>
 
-      <a href="corosync-blackbox.8.html">corosync-blackbox(8)</a>:
+      <a href="corosync-blackbox.8.html">corosync-blackbox(1m)</a>:
       Description of corosync-blackbox tool.
       <br>
 
-      <a href="corosync-cfgtool.8.html">corosync-cfgtool(8)</a>:
+      <a href="corosync-cfgtool.8.html">corosync-cfgtool(1m)</a>:
       Description of corosync-cfgtool tool.
       <br>
 
-      <a href="corosync-cpgtool.8.html">corosync-cpgtool(8)</a>:
+      <a href="corosync-cpgtool.8.html">corosync-cpgtool(1m)</a>:
       Description of corosync-cpgtool tool.
       <br>
 
-      <a href="corosync-keygen.8.html">corosync-keygen(8)</a>:
+      <a href="corosync-keygen.8.html">corosync-keygen(1m)</a>:
       Description of corosync-keygen tool.
       <br>
 
-      <a href="corosync-notifyd.8.html">corosync-notifyd(8)</a>:
+      <a href="corosync-notifyd.8.html">corosync-notifyd(1m)</a>:
       Description of corosync-notifyd tool.
       <br>
 
-      <a href="corosync-cmapctl.8.html">corosync-cmapctl(8)</a>:
+      <a href="corosync-cmapctl.8.html">corosync-cmapctl(1m)</a>:
       Description of corosync-cmapctl tool.
       <br>
 
-      <a href="cmap_keys.8.html">cmap_keys(8)</a>:
+      <a href="cmap_keys.8.html">cmap_keys(1m)</a>:
       Overview of keys stored in the Configuration Map.
       <br>
 
-      <a href="corosync-quorumtool.8.html">corosync-quorumtool(8)</a>:
+      <a href="corosync-quorumtool.8.html">corosync-quorumtool(1m)</a>:
       Description of corosync-quorumtool tool.
       <br>
 
-      <a href="corosync-xmlproc.8.html">corosync-xmlproc(8)</a>:
+      <a href="corosync-xmlproc.8.html">corosync-xmlproc(1m)</a>:
       Description of corosync-xmlproc tool.
       <br>
     </p>
@@ -81,7 +81,7 @@
 
       <h3>CPG service</h3>
 
-      <a href="cpg_overview.8.html">cpg_overview(8)</a>:
+      <a href="cpg_overview.8.html">cpg_overview(1m)</a>:
       Overview of the cpg extended virtual synchrony group
       communication toolkit.
       <br>
@@ -160,7 +160,7 @@
 
       <h3>SAM service</h3>
 
-      <a href="sam_overview.8.html">sam_overview(8)</a>:
+      <a href="sam_overview.8.html">sam_overview(1m)</a>:
       Description of sam_overview interface.
       <br>
 
@@ -214,7 +214,7 @@
 
       <h3>QUORUM service</h3>
 
-      <a href="quorum_overview.8.html">quorum_overview(8)</a>:
+      <a href="quorum_overview.8.html">quorum_overview(1m)</a>:
       An overview of the quorum service
       <br>
 
@@ -256,7 +256,7 @@
 
       <h3>VOTEQUORUM service</h3>
 
-      <a href="votequorum_overview.8.html">votequorum_overview(8)</a>:
+      <a href="votequorum_overview.8.html">votequorum_overview(1m)</a>:
       An overview of the vote-based quorum service
       <br>
 
@@ -326,7 +326,7 @@
 
       <h3>CMAP service</h3>
 
-      <a href="cmap_overview.8.html">cmap_overview(8)</a>:
+      <a href="cmap_overview.8.html">cmap_overview(1m)</a>:
       An overview of the configuration map service.
       <br>
 
diff -Nur a/man/quorum_context_get.3.in b/man/quorum_context_get.3.in
--- a/man/quorum_context_get.3.in	2017-09-26 20:23:12.073767446 +0000
+++ b/man/quorum_context_get.3.in	2017-09-26 20:25:14.943029646 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/quorum_context_set.3.in b/man/quorum_context_set.3.in
--- a/man/quorum_context_set.3.in	2017-09-26 20:23:12.073800570 +0000
+++ b/man/quorum_context_set.3.in	2017-09-26 20:25:14.975138280 +0000
@@ -51,7 +51,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/quorum_dispatch.3.in b/man/quorum_dispatch.3.in
--- a/man/quorum_dispatch.3.in	2017-09-26 20:23:12.073829903 +0000
+++ b/man/quorum_dispatch.3.in	2017-09-26 20:25:14.938467308 +0000
@@ -85,7 +85,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/quorum_fd_get.3.in b/man/quorum_fd_get.3.in
--- a/man/quorum_fd_get.3.in	2017-09-26 20:23:12.073858097 +0000
+++ b/man/quorum_fd_get.3.in	2017-09-26 20:25:14.957907388 +0000
@@ -56,7 +56,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/quorum_finalize.3.in b/man/quorum_finalize.3.in
--- a/man/quorum_finalize.3.in	2017-09-26 20:23:12.073888444 +0000
+++ b/man/quorum_finalize.3.in	2017-09-26 20:25:14.959216981 +0000
@@ -53,7 +53,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_getquorate (3),
 .BR quorum_trackstart (3),
diff -Nur a/man/quorum_getquorate.3.in b/man/quorum_getquorate.3.in
--- a/man/quorum_getquorate.3.in	2017-09-26 20:23:12.073917613 +0000
+++ b/man/quorum_getquorate.3.in	2017-09-26 20:25:14.963525364 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_trackstart (3),
diff -Nur a/man/quorum_initialize.3.in b/man/quorum_initialize.3.in
--- a/man/quorum_initialize.3.in	2017-09-26 20:23:12.073948200 +0000
+++ b/man/quorum_initialize.3.in	2017-09-26 20:25:14.948250060 +0000
@@ -101,7 +101,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
 .BR quorum_trackstart (3),
diff -Nur a/man/quorum_overview.8 b/man/quorum_overview.8
--- a/man/quorum_overview.8	2017-09-26 20:23:12.073975910 +0000
+++ b/man/quorum_overview.8	2017-09-26 20:25:14.937773112 +0000
@@ -32,7 +32,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH QUORUM_OVERVIEW 8 2012-02-09 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH QUORUM_OVERVIEW 1m 2012-02-09 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 .SH NAME
 quorum_overview \- Quorum Library Overview
 .SH OVERVIEW
@@ -49,7 +49,7 @@
 .SH BUGS
 No known bugs at the time of writing. The authors are from outerspace. Deal with it.
 .SH "SEE ALSO"
-.BR corosync-quorumtool (8),
+.BR corosync-quorumtool (1m),
 .BR corosync.conf (5),
 .BR votequorum (5),
 .BR quorum_initialize (3),
diff -Nur a/man/quorum_trackstart.3.in b/man/quorum_trackstart.3.in
--- a/man/quorum_trackstart.3.in	2017-09-26 20:23:12.074004351 +0000
+++ b/man/quorum_trackstart.3.in	2017-09-26 20:25:14.973043059 +0000
@@ -65,7 +65,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/quorum_trackstop.3.in b/man/quorum_trackstop.3.in
--- a/man/quorum_trackstop.3.in	2017-09-26 20:23:12.074032593 +0000
+++ b/man/quorum_trackstop.3.in	2017-09-26 20:25:14.962173121 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR quorum_overview (8),
+.BR quorum_overview (1m),
 .BR quorum_initialize (3),
 .BR quorum_finalize (3),
 .BR quorum_getquorate (3),
diff -Nur a/man/sam_overview.8 b/man/sam_overview.8
--- a/man/sam_overview.8	2017-09-26 20:23:12.074293435 +0000
+++ b/man/sam_overview.8	2017-09-26 20:25:14.971107853 +0000
@@ -32,7 +32,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH "SAM_OVERVIEW" 8 "21/05/2010" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH "SAM_OVERVIEW" 1m "21/05/2010" "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 
 .SH NAME
 .P
diff -Nur a/man/votequorum.5 b/man/votequorum.5
--- a/man/votequorum.5	2017-09-26 20:23:12.074438260 +0000
+++ b/man/votequorum.5	2017-09-26 20:25:14.945631782 +0000
@@ -401,9 +401,9 @@
 .SH BUGS
 No known bugs at the time of writing. The authors are from outerspace. Deal with it.
 .SH "SEE ALSO"
-.BR corosync (8),
+.BR corosync (1m),
 .BR corosync.conf (5),
-.BR corosync-quorumtool (8),
-.BR corosync-qdevice (8),
-.BR votequorum_overview (8)
+.BR corosync-quorumtool (1m),
+.BR corosync-qdevice (1m),
+.BR votequorum_overview (1m)
 .PP
diff -Nur a/man/votequorum_context_get.3.in b/man/votequorum_context_get.3.in
--- a/man/votequorum_context_get.3.in	2017-09-26 20:23:12.074466835 +0000
+++ b/man/votequorum_context_get.3.in	2017-09-26 20:25:14.952605830 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_context_set.3.in b/man/votequorum_context_set.3.in
--- a/man/votequorum_context_set.3.in	2017-09-26 20:23:12.074495597 +0000
+++ b/man/votequorum_context_set.3.in	2017-09-26 20:25:14.963040141 +0000
@@ -51,7 +51,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_dispatch.3.in b/man/votequorum_dispatch.3.in
--- a/man/votequorum_dispatch.3.in	2017-09-26 20:23:12.074532194 +0000
+++ b/man/votequorum_dispatch.3.in	2017-09-26 20:25:14.942379029 +0000
@@ -85,7 +85,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_fd_get.3.in b/man/votequorum_fd_get.3.in
--- a/man/votequorum_fd_get.3.in	2017-09-26 20:23:12.074562262 +0000
+++ b/man/votequorum_fd_get.3.in	2017-09-26 20:25:14.938733904 +0000
@@ -56,7 +56,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_finalize.3.in b/man/votequorum_finalize.3.in
--- a/man/votequorum_finalize.3.in	2017-09-26 20:23:12.074594393 +0000
+++ b/man/votequorum_finalize.3.in	2017-09-26 20:25:14.957109128 +0000
@@ -53,7 +53,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_getinfo (3),
 .BR votequorum_trackstart (3),
diff -Nur a/man/votequorum_getinfo.3.in b/man/votequorum_getinfo.3.in
--- a/man/votequorum_getinfo.3.in	2017-09-26 20:23:12.074624260 +0000
+++ b/man/votequorum_getinfo.3.in	2017-09-26 20:25:14.952362058 +0000
@@ -97,7 +97,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_trackstart (3),
diff -Nur a/man/votequorum_initialize.3.in b/man/votequorum_initialize.3.in
--- a/man/votequorum_initialize.3.in	2017-09-26 20:23:12.074652832 +0000
+++ b/man/votequorum_initialize.3.in	2017-09-26 20:25:14.976154270 +0000
@@ -125,7 +125,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
 .BR votequorum_trackstart (3),
diff -Nur a/man/votequorum_overview.8 b/man/votequorum_overview.8
--- a/man/votequorum_overview.8	2017-09-26 20:23:12.074680540 +0000
+++ b/man/votequorum_overview.8	2017-09-26 20:25:14.953161472 +0000
@@ -32,7 +32,7 @@
 .\" * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
 .\" * THE POSSIBILITY OF SUCH DAMAGE.
 .\" */
-.TH VOTEQUORUM_OVERVIEW 8 2012-01-12 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
+.TH VOTEQUORUM_OVERVIEW 1m 2012-01-12 "corosync Man Page" "Corosync Cluster Engine Programmer's Manual"
 .SH NAME
 votequorum_overview \- Votequorum Library Overview
 .SH OVERVIEW
@@ -68,7 +68,7 @@
 .SH BUGS
 No known bugs at the time of writing. The authors are from outerspace. Deal with it.
 .SH "SEE ALSO"
-.BR corosync-quorumtool (8),
+.BR corosync-quorumtool (1m),
 .BR votequorum (5),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
diff -Nur a/man/votequorum_qdevice_master_wins.3.in b/man/votequorum_qdevice_master_wins.3.in
--- a/man/votequorum_qdevice_master_wins.3.in	2017-09-26 20:23:12.074709563 +0000
+++ b/man/votequorum_qdevice_master_wins.3.in	2017-09-26 20:25:14.956325224 +0000
@@ -59,7 +59,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_qdevice_poll.3.in b/man/votequorum_qdevice_poll.3.in
--- a/man/votequorum_qdevice_poll.3.in	2017-09-26 20:23:12.074748972 +0000
+++ b/man/votequorum_qdevice_poll.3.in	2017-09-26 20:25:14.956043214 +0000
@@ -64,7 +64,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_qdevice_register.3.in b/man/votequorum_qdevice_register.3.in
--- a/man/votequorum_qdevice_register.3.in	2017-09-26 20:23:12.074779042 +0000
+++ b/man/votequorum_qdevice_register.3.in	2017-09-26 20:25:14.947366913 +0000
@@ -64,7 +64,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_qdevice_unregister.3.in b/man/votequorum_qdevice_unregister.3.in
--- a/man/votequorum_qdevice_unregister.3.in	2017-09-26 20:23:12.074807289 +0000
+++ b/man/votequorum_qdevice_unregister.3.in	2017-09-26 20:25:14.963291019 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_qdevice_update.3.in b/man/votequorum_qdevice_update.3.in
--- a/man/votequorum_qdevice_update.3.in	2017-09-26 20:23:12.074838765 +0000
+++ b/man/votequorum_qdevice_update.3.in	2017-09-26 20:25:14.964891526 +0000
@@ -50,7 +50,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
@@ -65,5 +65,5 @@
 .BR votequorum_qdevice_register (3),
 .BR votequorum_qdevice_poll (3),
 .BR votequorum_qdevice_master_wins (3),
-.BR corosync-quorumtool (8)
+.BR corosync-quorumtool (1m)
 .PP
diff -Nur a/man/votequorum_setexpected.3.in b/man/votequorum_setexpected.3.in
--- a/man/votequorum_setexpected.3.in	2017-09-26 20:23:12.074867467 +0000
+++ b/man/votequorum_setexpected.3.in	2017-09-26 20:25:14.961940200 +0000
@@ -52,7 +52,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_setvotes.3.in b/man/votequorum_setvotes.3.in
--- a/man/votequorum_setvotes.3.in	2017-09-26 20:23:12.074895029 +0000
+++ b/man/votequorum_setvotes.3.in	2017-09-26 20:25:14.970525268 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_trackstart.3.in b/man/votequorum_trackstart.3.in
--- a/man/votequorum_trackstart.3.in	2017-09-26 20:23:12.074923820 +0000
+++ b/man/votequorum_trackstart.3.in	2017-09-26 20:25:14.950054625 +0000
@@ -69,7 +69,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
diff -Nur a/man/votequorum_trackstop.3.in b/man/votequorum_trackstop.3.in
--- a/man/votequorum_trackstop.3.in	2017-09-26 20:23:12.074951907 +0000
+++ b/man/votequorum_trackstop.3.in	2017-09-26 20:25:14.974900164 +0000
@@ -49,7 +49,7 @@
 .SH ERRORS
 @COMMONIPCERRORS@
 .SH "SEE ALSO"
-.BR votequorum_overview (8),
+.BR votequorum_overview (1m),
 .BR votequorum_initialize (3),
 .BR votequorum_finalize (3),
 .BR votequorum_getinfo (3),
