https://debbugs.gnu.org/cgi/bugreport.cgi?bug=68832

--- automake-1.16.5/t/yacc-cxx.sh.orig
+++ automake-1.16.5/t/yacc-cxx.sh
@@ -18,7 +18,7 @@
 # headers are not involved).
 # Keep in sync with sister test 'yacc-basic.sh'.
 
-required='c++ yacc'
+required='c++ bison'
 . test-init.sh
 
 cat >> configure.ac << 'END'
--- automake-1.16.5/t/yacc-clean-cxx.sh.orig
+++ automake-1.16.5/t/yacc-clean-cxx.sh
@@ -20,7 +20,7 @@
 # "make maintainer-clean".
 # See also sister test 'yacc-clean.sh'.
 
-required='c++ yacc'
+required='c++ bison'
 . test-init.sh
 
 cat >> configure.ac << 'END'
--- automake-1.16.5/t/yacc-d-cxx.sh.orig
+++ automake-1.16.5/t/yacc-d-cxx.sh
@@ -18,7 +18,7 @@
 # (i.e., '-d' in *YFLAGS).
 # Keep in sync with sister test 'yacc-d-basic.sh'.
 
-required='c++ yacc'
+required='c++ bison'
 . test-init.sh
 
 write_parse ()
--- automake-1.16.5/t/yacc-mix-c-cxx.sh.orig
+++ automake-1.16.5/t/yacc-mix-c-cxx.sh
@@ -17,7 +17,7 @@
 # Check that many different Yacc parsers (both C and C++) can co-exists
 # in the same directory.
 
-required='cc c++ yacc'
+required='cc c++ bison'
 . test-init.sh
 
 cat >> configure.ac << 'END'
