Make sure we test trove-classifiers from proto area.

--- trove_classifiers-2026.5.22.10/tests/test_cli.py.orig
+++ trove_classifiers-2026.5.22.10/tests/test_cli.py
@@ -16,7 +16,7 @@
 
 def test_entry_point():
     """Simple test for no error when calling the entry point. Output is not validated."""
-    subprocess.check_call(f"{BINDIR}/trove-classifiers")
+    subprocess.check_call(f"trove-classifiers")
 
 
 def test_module_run_is_entry_point():
@@ -27,6 +27,6 @@
         encoding="utf-8",
     )
     entry_point_proc = subprocess.run(
-        f"{BINDIR}/trove-classifiers", capture_output=True, encoding="utf-8"
+        f"trove-classifiers", capture_output=True, encoding="utf-8"
     )
     assert module_run_proc.stdout == entry_point_proc.stdout
