--- gnome-system-tools-2.14.0/configure.in.orig	2008-01-11 17:24:01.366288759 +0000
+++ gnome-system-tools-2.14.0/configure.in	2008-01-11 17:24:47.343602233 +0000
@@ -278,6 +278,10 @@
 AC_SUBST(NAUTILUS_LIBS)
 AC_SUBST(NAUTILUS_CFLAGS)
 
+dnl Determine the install dir for nautilus extension.
+NAUTILUSDIR=`pkg-config --variable=extensiondir libnautilus-extension`
+AC_SUBST(NAUTILUSDIR)
+
 dnl ===========================
 dnl END: NAUTILUS EXTENSION
 dnl ===========================
--- gnome-system-tools-2.14.0/src/shares/nautilus/Makefile.am.orig	2008-01-11 17:23:35.662704014 +0000
+++ gnome-system-tools-2.14.0/src/shares/nautilus/Makefile.am	2008-01-11 17:23:52.962974203 +0000
@@ -5,7 +5,7 @@
 	-I$(top_builddir)
 
 # FIXME: get this from the .pc file
-nautilus_extensiondir=$(libdir)/nautilus/extensions-1.0
+nautilus_extensiondir=$(NAUTILUSDIR)
 
 nautilus_extension_LTLIBRARIES=libnautilus-gst-shares.la
 
