--- gnome-system-tools-2.14.0/configure.in.orig	2007-08-30 14:27:11.715539285 +0100
+++ gnome-system-tools-2.14.0/configure.in	2007-08-30 14:27:20.889088552 +0100
@@ -200,10 +200,7 @@
 dnl Translations
 dnl ========================================================
 
-ALL_LINGUAS="am ar az bg bn ca cs cy da de el en_CA en_GB es et eu fi fr
-ga gl gu he hi hr hu id it ja ko ku lt lv mk ml mn ms nb ne nl nn no pa
-pl pt pt_BR ro ru rw sk sq sr sr@Latn sv ta th tr uk vi wa xh zh_CN
-zh_HK zh_TW"
+ALL_LINGUAS="am ar az bg bn ca cs cy da de el en_CA en_GB es et eu fi fr ga gl gu he hi hr hu id it ja ko ku lt lv mk ml mn ms nb ne nl nn no pa pl pt pt_BR ro ru rw sk sq sr sr@Latn sv ta th tr uk vi wa xh zh_CN zh_HK zh_TW"
 GETTEXT_PACKAGE=gnome-system-tools
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [gettext package name])
