--- mate-applets-1.22.2/multiload/linux-proc.c	2019-09-19 14:17:35.000000000 +0000
+++ mate-applets-1.22.2/multiload/linux-proc.c	2019-09-19 21:04:08.314342879 +0000
@@ -151,6 +151,13 @@ GetDiskLoad (int Maximum, int data [3],
     }
     else
     {
+
+        /*
+          At the moment, Solaris can't get the read/write information for filesystem
+          so we would like to use disk to get them, later when some unstable interfaces
+          become stable or public, will use them
+        */
+        /*
         glibtop_mountlist mountlist;
         glibtop_mountentry *mountentries;
 
@@ -179,6 +186,10 @@ GetDiskLoad (int Maximum, int data [3],
         }
 
         g_free(mountentries);
+        */
+        glibtop_fsusage fsusage;
+        glibtop_get_fsusage(&fsusage,"/"); /* "/" will not use, just as a input */
+        read = fsusage.read; write = fsusage.write;
     }
 
     readdiff  = read - lastread;
@@ -376,6 +387,7 @@ GetNet (int Maximum, int data [4], LoadG
         autoscaler_init(&scaler, 60, 501);
     }
 
+    netlist.flags = 1;
     devices = glibtop_get_netlist(&netlist);
 
     for(i = 0; i < netlist.number; ++i)
@@ -385,7 +397,12 @@ GetNet (int Maximum, int data [4], LoadG
 
         glibtop_get_netload(&netload, devices[i]);
 
-        g_return_if_fail((netload.flags & needed_netload_flags) == needed_netload_flags);
+        /* this is not correct, because after run get netlist, there are lots of net devices,
+           not all of them meet the requirement of needed_netload_flags, but we don't need to
+           exit, we can just ignore them, and try to continue with the next one */
+        if ((netload.flags & needed_netload_flags) != needed_netload_flags)
+            continue;
+
 
         if (!(netload.if_flags & (1L << GLIBTOP_IF_FLAGS_UP)))
             continue;
