Our iov_base is caddr_t, not void*

--- memcached-1.6.38/storage.c.~1~	2025-03-19 23:47:32.000000000 +0200
+++ memcached-1.6.38/storage.c	2025-06-17 13:01:51.780686964 +0300
@@ -311,7 +311,7 @@
         }
 
         // fill the header so we can get the full data + crc back.
-        eio->iov[0].iov_base = new_it;
+        eio->iov[0].iov_base = (void *)new_it;
         eio->iov[0].iov_len = ITEM_ntotal(new_it) - new_it->nbytes;
         ciovcnt++;
 
