# bzero is a deprecated funtion

--- memcached-1.6.38/tls.c.~1~	2025-02-04 03:04:42.000000000 +0200
+++ memcached-1.6.38/tls.c	2025-06-17 13:01:51.795734773 +0300
@@ -306,7 +306,8 @@
         *errmsg = NULL;
         return false;
     }
-    bzero(ssl_err_msg, SSL_ERROR_MSG_SIZE);
+    //bzero(ssl_err_msg, SSL_ERROR_MSG_SIZE);
+    memset(ssl_err_msg, 0, sizeof(SSL_ERROR_MSG_SIZE));
     size_t err_msg_size = 0;
 
     SSL_LOCK();
