--- memcached-1.6.22/configure.ac	2023-10-16 08:03:25.000000000 +0200
+++ memcached-1.6.22/configure.ac.new	2023-12-25 19:29:02.691348520 +0100
@@ -369,7 +369,7 @@
   le_found=no
   for ledir in $trylibeventdir "" $prefix /usr/local ; do
     LDFLAGS="$saved_LDFLAGS"
-    LIBS="-levent $saved_LIBS"
+    LIBS="-levent-2.1 $saved_LIBS"
 
     # Skip the directory if it isn't there.
     if test ! -z "$ledir" -a ! -d "$ledir" ; then
@@ -417,7 +417,7 @@
 ])
   fi
 ])
-LIBS="-levent $LIBS"
+LIBS="-levent-2.1 $LIBS"
 if test $ac_cv_libevent_dir != "(system)"; then
   if test -d "$ac_cv_libevent_dir/lib" ; then
     LDFLAGS="-L$ac_cv_libevent_dir/lib $LDFLAGS"
