--- hdf5-1.8.18/config/solaris.orig	2017-02-03 21:05:39.824699478 +0100
+++ hdf5-1.8.18/config/solaris	2017-02-03 21:07:52.636271572 +0100
@@ -105,7 +105,6 @@
 
 # Try solaris native compiler flags
 if test -z "$cxx_flags_set"; then
-  H5_CXXFLAGS="$H5_CXXFLAGS -instances=static"
   H5_CPPFLAGS="$H5_CPPFLAGS -LANG:std"
   # -g produces rather slow code. "-g -O" produces much faster code with some
   # loss of debugger functions such as not able to print local variables.
