--- gnome-terminal-2.30.2/configure.ac.ori	2011-03-16 09:23:25.315008093 +0000
+++ gnome-terminal-2.30.2/configure.ac	2011-03-16 09:25:02.110567006 +0000
@@ -69,7 +69,7 @@
   [AS_HELP_STRING([--with-smclient-backend],[which smclient backend to use (xsmp/win32/quartz)])],
   [],
   [case "$GDK_TARGET" in
-    x11) with_smclient=xsmp SMCLIENT_PKGS="sm" ;;
+    x11) with_smclient=xsmp SMCLIENT_PKGS="sm ice" ;;
     win32|quartz) with_smclient=$GDK_TARGET ;;
     *) with_smclient=no ;;
    esac])
