diff -ruN gnome-terminal-2.24.0.orig/gnome-terminal.desktop.in.in gnome-terminal-2.24.0/gnome-terminal.desktop.in.in
--- gnome-terminal-2.24.0.orig/gnome-terminal.desktop.in.in	2008-09-24 09:23:01.463901647 +0100
+++ gnome-terminal-2.24.0/gnome-terminal.desktop.in.in	2008-09-24 09:23:28.472220048 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=Terminal
-_Comment=Use the command line
+# SUN_BRANDING
+_Comment=Run a command line shell
 TryExec=gnome-terminal
 Exec=gnome-terminal
 Icon=utilities-terminal
