--- node-v6.11.3/deps/uv/src/unix/sunos.c.orig	Sat Sep 16 13:39:06 2017
+++ node-v6.11.3/deps/uv/src/unix/sunos.c	Sat Sep 16 13:39:34 2017
@@ -783,6 +784,10 @@
     if (ent->ifa_addr == NULL)
       continue;
 
+    if (ent->ifa_addr->sa_family != AF_INET &&
+        ent->ifa_addr->sa_family != AF_INET6)
+      continue;
+
     address->name = uv__strdup(ent->ifa_name);
 
     if (ent->ifa_addr->sa_family == AF_INET6) {
