--- node-v6.1.0/deps/openssl/openssl/crypto/rand/randfile.c.1   2016-05-13 08:40:00.387386356 +0300
+++ node-v6.1.0/deps/openssl/openssl/crypto/rand/randfile.c     2016-05-13 08:40:24.880983892 +0300
@@ -57,7 +57,7 @@
  */
 
 /* We need to define this to get macros like S_IFBLK and S_IFCHR */
-#if !defined(OPENSSL_SYS_VXWORKS)
+#if !defined(OPENSSL_SYS_VXWORKS) && !defined(__sun__)
 # define _XOPEN_SOURCE 500
 #endif
 