--- imap-2007f/src/osdep/unix/flocksim.c_orig	Fri Jul 22 17:20:11 2011
+++ imap-2007f/src/osdep/unix/flocksim.c	Mon Nov  7 14:24:05 2011
@@ -52,7 +52,9 @@
   char tmp[MAILTMPLEN];
   int logged = 0;
   struct stat sbuf;
+#if !defined(NOUSTAT)
   struct ustat usbuf;
+#endif
   struct flock fl;
 				/* lock zero bytes at byte 0 */
   fl.l_whence = SEEK_SET; fl.l_start = fl.l_len = 0;
@@ -129,7 +131,9 @@
 	((vsbuf.f_basetype[0] == 'n') || (vsbuf.f_basetype[0] == 'a')))
       return 0;
 #endif		/* NOFSTATVFS */
+#if !defined(NOUSTAT)
     if (!ustat (sbuf.st_dev,&usbuf) && !++usbuf.f_tinode) return 0;
+#endif
   }
 
 				/* do the lock */
