diff -ruN uw-imap-2006b.dfsg.orig/src/osdep/unix/env_unix.c uw-imap-2006b.dfsg/src/osdep/unix/env_unix.c
--- uw-imap-2006b.dfsg.orig/src/osdep/unix/env_unix.c	2006-10-09 18:48:56.000000000 +0200
+++ uw-imap-2006b.dfsg/src/osdep/unix/env_unix.c	2006-10-09 19:23:13.000000000 +0200
@@ -77,7 +77,7 @@
 static short anonymous = NIL;	/* is anonymous */
 static short blackBox = NIL;	/* is a black box */
 static short closedBox = NIL;	/* is a closed box (uses chroot() jail) */
-static short restrictBox = NIL;	/* is a restricted box */
+static short restrictBox = -1;	/* is a restricted box */
 static short has_no_life = NIL;	/* is a cretin with no life */
 				/* block environment init */
 static short block_env_init = NIL;
