--- imap-2007e/Makefile.orig	2011-05-10 14:16:22.710615591 +0100
+++ imap-2007e/Makefile	2011-05-10 14:16:30.256456227 +0100
@@ -642,7 +642,6 @@
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 	@echo
 	@echo Do you want to build with IPv6 anyway?  Type y or n please:
-	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make noip6;exit 1);; esac'
 	@echo OK, I will remember that you really want to build with IPv6.
 	@echo You will not see this message again.
 	@$(TOUCH) ip6
