--- htop-220-sunos_11-p4/CRT.h.orig	2020-05-15 08:28:45.295864748 +0000
+++ htop-220-sunos_11-p4/CRT.h	2020-05-15 08:29:20.326745819 +0000
@@ -147,7 +147,7 @@
 
 extern int CRT_delay;
 
-int* CRT_colors;
+extern int* CRT_colors;
 
 extern int CRT_colorSchemes[LAST_COLORSCHEME][LAST_COLORELEMENT];
 
@@ -157,13 +157,13 @@
 
 extern int CRT_scrollWheelVAmount;
 
-char* CRT_termType;
+extern char* CRT_termType;
 
 // TODO move color scheme to Settings, perhaps?
 
 extern int CRT_colorScheme;
 
-void *backtraceArray[128];
+extern void *backtraceArray[128];
 
 #if HAVE_SETUID_ENABLED
 
--- htop-220-sunos_11-p4/solaris/SolarisProcessList.h.orig	2020-05-15 08:33:01.988946407 +0000
+++ htop-220-sunos_11-p4/solaris/SolarisProcessList.h	2020-05-15 08:33:11.030177935 +0000
@@ -24,7 +24,7 @@
 #include <sys/swap.h>
 
 #define ZONE_ERRMSGLEN 1024
-char zone_errmsg[ZONE_ERRMSGLEN];
+extern char zone_errmsg[ZONE_ERRMSGLEN];
 
 typedef struct CPUData_ {
    double userPercent;
