--- DBD-Pg-3.7.0/Makefile.PL.1	2017-11-30 11:56:29.295061879 +0000
+++ DBD-Pg-3.7.0/Makefile.PL	2017-11-30 11:56:55.426326827 +0000
@@ -215,7 +215,7 @@
      VERSION_FROM   => 'Pg.pm',
      INC            => "-I$POSTGRES_INCLUDE -I$dbi_arch_dir",
      OBJECT         => "Pg\$(OBJ_EXT) dbdimp\$(OBJ_EXT) quote\$(OBJ_EXT) types\$(OBJ_EXT)",
-     LIBS           => ["-L$POSTGRES_LIB -lpq -lm"],
+     LIBS           => ["-lpq -lm"],
      AUTHOR         => 'Greg Sabino Mullane',
      ABSTRACT       => 'PostgreSQL database driver for the DBI module',
      PREREQ_PM      => {
