--- gtk+-3.24.43/modules/input/imxim.c.orig
+++ gtk+-3.24.43/modules/input/imxim.c
@@ -26,7 +26,8 @@
   NC_("input method menu", "X Input Method"), /* Human readable name */
   GETTEXT_PACKAGE,		   /* Translation domain */
   GTK_LOCALEDIR,		   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "ko:ja:th:zh"		           /* Languages for which this module is the default */
+  /* Sun's local patch to make XIM default input method for all locales */
+  "*"		  		   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
