--- gtk+-2.4.1/modules/input/imxim.c.orig	Mon May 17 13:00:49 2004
+++ gtk+-2.4.1/modules/input/imxim.c	Mon May 17 13:01:52 2004
@@ -28,7 +28,8 @@
   N_("X Input Method"),            /* Human readable name */
   GETTEXT_PACKAGE,		   /* Translation domain */
   GTK_LOCALEDIR,		   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "ko:ja:th:zh"		           /* Languages for which this module is the default */
+  /* Sun's local patch to make XIM default input method for all locales */
+  "*"		  		   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
