--- gtk+-3.19.3/gtk/gtkprintoperation-unix.c.~1~	2015-03-02 22:34:03.000000000 +0300
+++ gtk+-3.19.3/gtk/gtkprintoperation-unix.c	2015-12-01 09:25:21.793540932 +0300
@@ -111,7 +111,7 @@
               w = gtk_page_setup_get_paper_width (page_setup, GTK_UNIT_POINTS);
               h = gtk_page_setup_get_paper_height (page_setup, GTK_UNIT_POINTS);
             }
-          cairo_pdf_surface_set_size (op_unix->surface, w, h);
+          cairo_ps_surface_set_size (op_unix->surface, w, h);
         }
     }
 }
@@ -792,7 +792,7 @@
   cairo_surface_t *surface;
   static cairo_user_data_key_t key;
   
-  filename = g_build_filename (g_get_tmp_dir (), "previewXXXXXX.pdf", NULL);
+  filename = g_build_filename (g_get_tmp_dir (), "previewXXXXXX.ps", NULL);
   fd = g_mkstemp (filename);
 
   if (fd < 0)
@@ -808,7 +808,7 @@
   h = gtk_paper_size_get_height (paper_size, GTK_UNIT_POINTS);
     
   *dpi_x = *dpi_y = 72;
-  surface = cairo_pdf_surface_create_for_stream (write_preview, GINT_TO_POINTER (fd), w, h);
+  surface = cairo_ps_surface_create_for_stream (write_preview, GINT_TO_POINTER (fd), w, h);
  
   cairo_surface_set_user_data (surface, &key, GINT_TO_POINTER (fd), close_preview);
 
@@ -839,7 +839,7 @@
   
   w = gtk_page_setup_get_paper_width (page_setup, GTK_UNIT_POINTS);
   h = gtk_page_setup_get_paper_height (page_setup, GTK_UNIT_POINTS);
-  cairo_pdf_surface_set_size (surface, w, h);
+  cairo_ps_surface_set_size (surface, w, h);
 }
 
 
