diff -urN gtk+-2.20.1/configure.ac ../../SUNWgtk2-2.20.1.hacked/i386/gtk+-2.20.1/configure.ac
--- gtk+-2.20.1/configure.ac	2011-08-11 16:24:07.371964160 +0000
+++ ../../SUNWgtk2-2.20.1.hacked/i386/gtk+-2.20.1/configure.ac	2011-08-11 17:05:56.394292246 +0000
@@ -1759,7 +1759,6 @@ modules/printbackends/Makefile
 modules/printbackends/cups/Makefile
 modules/printbackends/lpr/Makefile
 modules/printbackends/file/Makefile
-modules/printbackends/papi/Makefile
 modules/printbackends/test/Makefile
 perf/Makefile
 ])
diff -urN gtk+-2.20.1/modules/printbackends/Makefile.am ../../SUNWgtk2-2.20.1.hacked/i386/gtk+-2.20.1/modules/printbackends/Makefile.am
--- gtk+-3.19.3/modules/printbackends/Makefile.am.~1~	2015-12-01 09:26:14.893987658 +0300
+++ gtk+-3.19.3/modules/printbackends/Makefile.am	2015-12-01 09:27:12.292480891 +0300
@@ -14,10 +14,6 @@
 SUBDIRS += test
 endif
 
-if HAVE_PAPI
-SUBDIRS += papi
-endif
-
-DIST_SUBDIRS = cloudprint cups file lpr test papi
+DIST_SUBDIRS = cloudprint cups file lpr test
 
 -include $(top_srcdir)/git.mk
