diff -Nru gtk+-3.24.37.orig/testsuite/gtk/meson.build gtk+-3.24.37/testsuite/gtk/meson.build
--- gtk+-3.24.37.orig/testsuite/gtk/meson.build	2023-03-03 01:31:52.000000000 +0100
+++ gtk+-3.24.37/testsuite/gtk/meson.build	2023-03-21 10:23:06.489271609 +0100
@@ -1,7 +1,7 @@
 gtk_tests_export_dynamic_ldflag = []
 
 if cc.get_id() != 'msvc' and cc.get_id() != 'clang'
-  gtk_tests_export_dynamic_ldflag = ['-Wl,--export-dynamic']
+  gtk_tests_export_dynamic_ldflag = ['-W']
 endif
 
 tests = [
