https://github.com/python/cpython/pull/137927

--- Python-3.14.7/Include/pythread.h.orig
+++ Python-3.14.7/Include/pythread.h
@@ -42,7 +42,8 @@
 #if (defined(__APPLE__) || defined(__linux__) || defined(_WIN32) \
      || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) \
      || defined(__OpenBSD__) || defined(__NetBSD__) \
-     || defined(__DragonFly__) || defined(_AIX))
+     || defined(__DragonFly__) || defined(_AIX) \
+     || (defined(__sun__) && SIZEOF_LONG >= 8))
 #define PY_HAVE_THREAD_NATIVE_ID
 PyAPI_FUNC(unsigned long) PyThread_get_thread_native_id(void);
 #endif
--- Python-3.14.7/Python/thread_pthread.h.orig
+++ Python-3.14.7/Python/thread_pthread.h
@@ -30,6 +30,8 @@
 #   include <lwp.h>             /* _lwp_self() */
 #elif defined(__DragonFly__)
 #   include <sys/lwp.h>         /* lwp_gettid() */
+#elif defined(__sun__) && SIZEOF_LONG >= 8
+#   include <thread.h>
 #endif
 
 /* The POSIX spec requires that use of pthread_attr_setstacksize
@@ -409,6 +411,8 @@
 #elif defined(__DragonFly__)
     // lwp_gettid() is only unique within a process, so combine it with the pid.
     unsigned long native_id = (unsigned long)getpid() << 32 | lwp_gettid();
+#elif defined(__sun__) && SIZEOF_LONG >= 8
+    unsigned long native_id = (unsigned long)getpid() << 32 | thr_self();
 #endif
     return (unsigned long) native_id;
 }
