--- Python-3.14.6/Lib/test/test_email/test_utils.py.orig
+++ Python-3.14.6/Lib/test/test_email/test_utils.py
@@ -151,6 +151,7 @@
         t2 = utils.localtime(t0.replace(tzinfo=None))
         self.assertEqual(t1, t2)
 
+    @unittest.skip("https://www.illumos.org/issues/17608")
     @test.support.run_with_tz('Europe/Kyiv')
     def test_variable_tzname(self):
         t0 = datetime.datetime(1984, 1, 1, tzinfo=datetime.timezone.utc)
