--- gnome-system-tools-2.13.2/src/shares/share-export.c-orig	2006-03-03 16:16:05.157378000 +0000
+++ gnome-system-tools-2.13.2/src/shares/share-export.c	2006-03-03 16:16:26.132960000 +0000
@@ -181,5 +181,5 @@
 	if (GST_SHARE_GET_CLASS (share)->get_xml == NULL)
 		return;
 
-	return GST_SHARE_GET_CLASS (share)->get_xml (share, parent);
+	GST_SHARE_GET_CLASS (share)->get_xml (share, parent);
 }

diff -u gnome-system-tools-2.14.0/src/shares/callbacks.c-orig gnome-system-tools-2.14.0/src/shares/callbacks.c
--- gnome-system-tools-2.14.0/src/shares/callbacks.c-orig	2005-08-08 20:00:03.000000000 +0100
+++ gnome-system-tools-2.14.0/src/shares/callbacks.c	2006-10-23 15:31:55.258228000 +0100
@@ -128,7 +128,7 @@
 			path = g_strdup (gnome_vfs_uri_get_path (uris->data));
 
 			if (g_file_test (path, G_FILE_TEST_IS_DIR)) {
-				share_settings_dialog_run (path, FALSE);
+				share_settings_dialog_run (path, ANY_SHARE, FALSE);
 				gtk_drag_finish (context, TRUE, FALSE, time);
 			} else {
 				gtk_drag_finish (context, FALSE, FALSE, time);
@@ -143,7 +143,7 @@
 void
 on_add_share_clicked (GtkWidget *widget, gpointer data)
 {
-	share_settings_dialog_run (NULL, FALSE);
+	share_settings_dialog_run (NULL, ANY_SHARE, FALSE);
 }
 
 void
@@ -154,6 +154,7 @@
 	GtkTreeModel     *model;
 	GtkTreeIter       iter;
 	gchar            *path;
+	gint              share_type;
 
 	table = gst_dialog_get_widget (tool->main_dialog, "shares_table");
 	selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (table));
@@ -161,9 +162,9 @@
 	if (!gtk_tree_selection_get_selected (selection, &model, &iter))
 		return;
 
-	gtk_tree_model_get (model, &iter, COL_PATH, &path, -1);
+	gtk_tree_model_get (model, &iter, COL_PATH, &path, COL_SHARETYPE, &share_type, -1);
 
-	share_settings_dialog_run (path, FALSE);
+	share_settings_dialog_run (path, share_type, FALSE);
 	g_free (path);
 }
 

diff -u gnome-system-tools-2.14.0/src/shares/main.c-orig gnome-system-tools-2.14.0/src/shares/main.c
--- gnome-system-tools-2.14.0/src/shares/main.c-orig	2005-08-08 20:00:03.000000000 +0100
+++ gnome-system-tools-2.14.0/src/shares/main.c	2006-10-23 15:31:55.362018000 +0100
@@ -70,7 +70,7 @@
 init_standalone_dialog (const gchar *path)
 {
 	gst_tool_main_with_hidden_dialog (tool, TRUE);
-	share_settings_dialog_run (path, TRUE);
+	share_settings_dialog_run (path, ANY_SHARE, TRUE);
 }
 
 void

diff -u gnome-system-tools-2.14.0/src/shares/share-export-nfs.c-orig gnome-system-tools-2.14.0/src/shares/share-export-nfs.c
--- gnome-system-tools-2.14.0/src/shares/share-export-nfs.c-orig	2006-11-13 13:30:16.525878000 +0800
+++ gnome-system-tools-2.14.0/src/shares/share-export-nfs.c	2006-11-13 13:13:57.350210000 +0800
@@ -137,9 +137,10 @@
 	GstShareNFS *share;
 
 	share = g_object_new (GST_TYPE_SHARE_NFS,
-			      "share-path", path,
 			      NULL);
 
+	gst_share_set_path    (GST_SHARE (share), path);
+
 	return share;
 }
 
@@ -173,7 +174,6 @@
 	}
 
 	g_free (path);
-	g_free (type);
 
 	return share;
 }

diff -u gnome-system-tools-2.14.0/src/shares/share-settings.c-orig gnome-system-tools-2.14.0/src/shares/share-settings.c
--- gnome-system-tools-2.14.0/src/shares/share-settings.c-orig	2006-11-13 13:31:05.341437000 +0800
+++ gnome-system-tools-2.14.0/src/shares/share-settings.c	2006-11-13 13:26:19.086675000 +0800
@@ -244,7 +244,7 @@
 	gint         flags = 0;
 
 	widget  = gst_dialog_get_widget (tool->main_dialog, "share_path");
-	path    = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (widget));
+	path    = gtk_file_chooser_get_filename (GTK_FILE_CHOOSER (widget));
 
 	widget  = gst_dialog_get_widget (tool->main_dialog, "share_smb_name");
 	name    = gtk_entry_get_text (GTK_ENTRY (widget));
@@ -274,7 +274,7 @@
 	GstShareNFS *share;
 
 	widget  = gst_dialog_get_widget (tool->main_dialog, "share_path");
-	path    = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (widget));
+	path    = gtk_file_chooser_get_filename (GTK_FILE_CHOOSER (widget));
 
 	share = gst_share_nfs_new (path);
 	nfs_acl_table_insert_elements (share);
@@ -334,7 +334,7 @@
 }
 
 void
-share_settings_dialog_run (const gchar *path, gboolean standalone)
+share_settings_dialog_run (const gchar *path, gint share_type, gboolean standalone)
 {
 	GtkWidget   *dialog;
 	gint         response;
@@ -346,7 +346,7 @@
 	dialog = share_settings_prepare_dialog (path, standalone);
 
 	/* check whether the path already exists */
-	path_exists = table_get_iter_with_path (path, &iter);
+	path_exists = table_get_iter_with_path (path, share_type, &iter);
 
 	if (path_exists) {
 		share = table_get_share_at_iter (&iter);

diff -u gnome-system-tools-2.14.0/src/shares/share-settings.h-orig gnome-system-tools-2.14.0/src/shares/share-settings.h
--- gnome-system-tools-2.14.0/src/shares/share-settings.h-orig	2005-04-06 23:21:44.000000000 +0100
+++ gnome-system-tools-2.14.0/src/shares/share-settings.h	2006-10-23 15:31:55.555242000 +0100
@@ -33,7 +33,7 @@
 	SHARE_THROUGH_NFS
 };
 
-void          share_settings_dialog_run        (const gchar*, gboolean);
+void          share_settings_dialog_run        (const gchar*, gint, gboolean);
 void          share_settings_create_combo      (void);
 gboolean      share_settings_validate          (void);
 

diff -u gnome-system-tools-2.14.0/src/shares/table.c-orig gnome-system-tools-2.14.0/src/shares/table.c
--- gnome-system-tools-2.14.0/src/shares/table.c-orig	2005-08-08 20:00:03.000000000 +0100
+++ gnome-system-tools-2.14.0/src/shares/table.c	2006-10-23 15:31:55.679417000 +0100
@@ -85,7 +85,8 @@
 	store = gtk_list_store_new (COL_LAST,
 				    GDK_TYPE_PIXBUF,
 				    G_TYPE_STRING,
-				    G_TYPE_OBJECT);
+				    G_TYPE_OBJECT,
+				    G_TYPE_INT);
 
 	return GTK_TREE_MODEL (store);
 }
@@ -185,6 +186,7 @@
 	GtkWidget    *table = gst_dialog_get_widget (tool->main_dialog, "shares_table");
 	GtkTreeModel *model;
 	GdkPixbuf    *pixbuf = NULL;
+        gint          share_type = ANY_SHARE;
 
 	g_return_if_fail (share != NULL);
 	g_return_if_fail (GST_IS_SHARE (share));
@@ -195,17 +197,20 @@
 		pixbuf = gtk_icon_theme_load_icon (icon_theme,
 						   "gnome-fs-smb",
 						   48, 0, NULL);
+                share_type = SMB_SHARE;
 	} else if (GST_IS_SHARE_NFS (share)) {
 		pixbuf = gtk_icon_theme_load_icon (icon_theme,
 						   "gnome-fs-nfs",
 						   48, 0, NULL);
+                share_type = NFS_SHARE;
 	}
 
 	gtk_list_store_set (GTK_LIST_STORE (model),
 			    iter,
-			    0, pixbuf,
-			    1, gst_share_get_path (share),
-			    2, share,
+			    COL_PIXBUF, pixbuf,
+			    COL_PATH, gst_share_get_path (share),
+			    COL_POINTER, share,
+			    COL_SHARETYPE, share_type,
 			    -1);
 
 	g_object_unref (share);
@@ -241,12 +246,13 @@
 }
 
 gboolean
-table_get_iter_with_path (const gchar *path, GtkTreeIter *iter)
+table_get_iter_with_path (const gchar *path, gint share_type, GtkTreeIter *iter)
 {
 	GtkWidget    *table = gst_dialog_get_widget (tool->main_dialog, "shares_table");
 	GtkTreeModel *model;
 	gboolean      valid, found;
 	gchar        *iter_path;
+	gint          iter_share_type;
 
 	if (!path)
 		return FALSE;
@@ -258,9 +264,11 @@
 	while (valid) {
 		gtk_tree_model_get (model, iter,
 				    COL_PATH, &iter_path,
+				    COL_SHARETYPE, &iter_share_type,
 				    -1);
 
-		if (strcmp (iter_path, path) == 0) {
+		if (strcmp (iter_path, path) == 0 &&
+                   ( iter_share_type == share_type || iter_share_type == ANY_SHARE ) ) {
 			found = TRUE;
 			valid = FALSE;
 		} else

diff -u gnome-system-tools-2.14.0/src/shares/table.h-orig gnome-system-tools-2.14.0/src/shares/table.h
--- gnome-system-tools-2.14.0/src/shares/table.h-orig	2004-09-20 16:30:44.000000000 +0100
+++ gnome-system-tools-2.14.0/src/shares/table.h	2006-10-23 15:31:55.698077000 +0100
@@ -30,17 +30,24 @@
 	   COL_PIXBUF,
 	   COL_PATH,
 	   COL_POINTER,
+	   COL_SHARETYPE,
 	   COL_LAST
 };
 
 enum {
+    ANY_SHARE,
+    SMB_SHARE,
+    NFS_SHARE
+};
+
+enum {
 	SHARES_DND_URI_LIST
 };
 
 void      table_create               (void);
 void      table_add_share_from_node  (xmlNodePtr);
 void      table_add_share            (GstShare*);
-gboolean  table_get_iter_with_path   (const gchar*, GtkTreeIter*);
+gboolean  table_get_iter_with_path   (const gchar*, gint, GtkTreeIter*);
 GstShare* table_get_share_at_iter    (GtkTreeIter*);
 void      table_modify_share_at_iter (GtkTreeIter*, GstShare*);
 void      table_delete_share_at_iter (GtkTreeIter*);

