--- gnome-system-tools-2.14.0/src/common/zone_tab.h.orig	2009-04-06 10:59:45.191464000 +0900
+++ gnome-system-tools-2.14.0/src/common/zone_tab.h	2009-04-06 15:07:45.909805000 +0900
@@ -0,0 +1,809 @@
+#ifndef __ZONE_TAB_H__
+#define __ZONE_TAB_H__
+
+#include <config.h>
+#include <glib/gi18n.h>
+static const char *dummy_zone_location [] = {
+  /* SUN_BRANDING */
+  N_("Europe/Andorra"),
+  /* SUN_BRANDING */
+  N_("Asia/Dubai"),
+  /* SUN_BRANDING */
+  N_("Asia/Kabul"),
+  /* SUN_BRANDING */
+  N_("America/Antigua"),
+  /* SUN_BRANDING */
+  N_("America/Anguilla"),
+  /* SUN_BRANDING */
+  N_("Europe/Tirane"),
+  /* SUN_BRANDING */
+  N_("Asia/Yerevan"),
+  /* SUN_BRANDING */
+  N_("America/Curacao"),
+  /* SUN_BRANDING */
+  N_("Africa/Luanda"),
+  /* SUN_BRANDING */
+  N_("Antarctica/McMurdo"),
+  /* SUN_BRANDING */
+  N_("Antarctica/South_Pole"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Rothera"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Palmer"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Mawson"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Davis"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Casey"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Vostok"),
+  /* SUN_BRANDING */
+  N_("Antarctica/DumontDUrville"),
+  /* SUN_BRANDING */
+  N_("Antarctica/Syowa"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Buenos_Aires"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Cordoba"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Jujuy"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Tucuman"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Catamarca"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/La_Rioja"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/San_Juan"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Mendoza"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Rio_Gallegos"),
+  /* SUN_BRANDING */
+  N_("America/Argentina/Ushuaia"),
+  /* SUN_BRANDING */
+  N_("Pacific/Pago_Pago"),
+  /* SUN_BRANDING */
+  N_("Europe/Vienna"),
+  /* SUN_BRANDING */
+  N_("Australia/Lord_Howe"),
+  /* SUN_BRANDING */
+  N_("Australia/Hobart"),
+  /* SUN_BRANDING */
+  N_("Australia/Currie"),
+  /* SUN_BRANDING */
+  N_("Australia/Melbourne"),
+  /* SUN_BRANDING */
+  N_("Australia/Sydney"),
+  /* SUN_BRANDING */
+  N_("Australia/Broken_Hill"),
+  /* SUN_BRANDING */
+  N_("Australia/Brisbane"),
+  /* SUN_BRANDING */
+  N_("Australia/Lindeman"),
+  /* SUN_BRANDING */
+  N_("Australia/Adelaide"),
+  /* SUN_BRANDING */
+  N_("Australia/Darwin"),
+  /* SUN_BRANDING */
+  N_("Australia/Perth"),
+  /* SUN_BRANDING */
+  N_("America/Aruba"),
+  /* SUN_BRANDING */
+  N_("Europe/Mariehamn"),
+  /* SUN_BRANDING */
+  N_("Asia/Baku"),
+  /* SUN_BRANDING */
+  N_("Europe/Sarajevo"),
+  /* SUN_BRANDING */
+  N_("America/Barbados"),
+  /* SUN_BRANDING */
+  N_("Asia/Dhaka"),
+  /* SUN_BRANDING */
+  N_("Europe/Brussels"),
+  /* SUN_BRANDING */
+  N_("Africa/Ouagadougou"),
+  /* SUN_BRANDING */
+  N_("Europe/Sofia"),
+  /* SUN_BRANDING */
+  N_("Asia/Bahrain"),
+  /* SUN_BRANDING */
+  N_("Africa/Bujumbura"),
+  /* SUN_BRANDING */
+  N_("Africa/Porto-Novo"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Bermuda"),
+  /* SUN_BRANDING */
+  N_("Asia/Brunei"),
+  /* SUN_BRANDING */
+  N_("America/La_Paz"),
+  /* SUN_BRANDING */
+  N_("America/Noronha"),
+  /* SUN_BRANDING */
+  N_("America/Belem"),
+  /* SUN_BRANDING */
+  N_("America/Fortaleza"),
+  /* SUN_BRANDING */
+  N_("America/Recife"),
+  /* SUN_BRANDING */
+  N_("America/Araguaina"),
+  /* SUN_BRANDING */
+  N_("America/Maceio"),
+  /* SUN_BRANDING */
+  N_("America/Bahia"),
+  /* SUN_BRANDING */
+  N_("America/Sao_Paulo"),
+  /* SUN_BRANDING */
+  N_("America/Campo_Grande"),
+  /* SUN_BRANDING */
+  N_("America/Cuiaba"),
+  /* SUN_BRANDING */
+  N_("America/Porto_Velho"),
+  /* SUN_BRANDING */
+  N_("America/Boa_Vista"),
+  /* SUN_BRANDING */
+  N_("America/Manaus"),
+  /* SUN_BRANDING */
+  N_("America/Eirunepe"),
+  /* SUN_BRANDING */
+  N_("America/Rio_Branco"),
+  /* SUN_BRANDING */
+  N_("America/Nassau"),
+  /* SUN_BRANDING */
+  N_("Asia/Thimphu"),
+  /* SUN_BRANDING */
+  N_("Africa/Gaborone"),
+  /* SUN_BRANDING */
+  N_("Europe/Minsk"),
+  /* SUN_BRANDING */
+  N_("America/Belize"),
+  /* SUN_BRANDING */
+  N_("America/St_Johns"),
+  /* SUN_BRANDING */
+  N_("America/Halifax"),
+  /* SUN_BRANDING */
+  N_("America/Glace_Bay"),
+  /* SUN_BRANDING */
+  N_("America/Moncton"),
+  /* SUN_BRANDING */
+  N_("America/Goose_Bay"),
+  /* SUN_BRANDING */
+  N_("America/Blanc-Sablon"),
+  /* SUN_BRANDING */
+  N_("America/Montreal"),
+  /* SUN_BRANDING */
+  N_("America/Toronto"),
+  /* SUN_BRANDING */
+  N_("America/Nipigon"),
+  /* SUN_BRANDING */
+  N_("America/Thunder_Bay"),
+  /* SUN_BRANDING */
+  N_("America/Iqaluit"),
+  /* SUN_BRANDING */
+  N_("America/Pangnirtung"),
+  /* SUN_BRANDING */
+  N_("America/Atikokan"),
+  /* SUN_BRANDING */
+  N_("America/Rankin_Inlet"),
+  /* SUN_BRANDING */
+  N_("America/Winnipeg"),
+  /* SUN_BRANDING */
+  N_("America/Rainy_River"),
+  /* SUN_BRANDING */
+  N_("America/Regina"),
+  /* SUN_BRANDING */
+  N_("America/Swift_Current"),
+  /* SUN_BRANDING */
+  N_("America/Edmonton"),
+  /* SUN_BRANDING */
+  N_("America/Cambridge_Bay"),
+  /* SUN_BRANDING */
+  N_("America/Yellowknife"),
+  /* SUN_BRANDING */
+  N_("America/Inuvik"),
+  /* SUN_BRANDING */
+  N_("America/Dawson_Creek"),
+  /* SUN_BRANDING */
+  N_("America/Vancouver"),
+  /* SUN_BRANDING */
+  N_("America/Whitehorse"),
+  /* SUN_BRANDING */
+  N_("America/Dawson"),
+  /* SUN_BRANDING */
+  N_("Indian/Cocos"),
+  /* SUN_BRANDING */
+  N_("Africa/Kinshasa"),
+  /* SUN_BRANDING */
+  N_("Africa/Lubumbashi"),
+  /* SUN_BRANDING */
+  N_("Africa/Bangui"),
+  /* SUN_BRANDING */
+  N_("Africa/Brazzaville"),
+  /* SUN_BRANDING */
+  N_("Europe/Zurich"),
+  /* SUN_BRANDING */
+  N_("Africa/Abidjan"),
+  /* SUN_BRANDING */
+  N_("Pacific/Rarotonga"),
+  /* SUN_BRANDING */
+  N_("America/Santiago"),
+  /* SUN_BRANDING */
+  N_("Pacific/Easter"),
+  /* SUN_BRANDING */
+  N_("Africa/Douala"),
+  /* SUN_BRANDING */
+  N_("Asia/Shanghai"),
+  /* SUN_BRANDING */
+  N_("America/Bogota"),
+  /* SUN_BRANDING */
+  N_("America/Costa_Rica"),
+  /* SUN_BRANDING */
+  N_("America/Havana"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Cape_Verde"),
+  /* SUN_BRANDING */
+  N_("Indian/Christmas"),
+  /* SUN_BRANDING */
+  N_("Asia/Nicosia"),
+  /* SUN_BRANDING */
+  N_("Europe/Prague"),
+  /* SUN_BRANDING */
+  N_("Europe/Berlin"),
+  /* SUN_BRANDING */
+  N_("Africa/Djibouti"),
+  /* SUN_BRANDING */
+  N_("Europe/Copenhagen"),
+  /* SUN_BRANDING */
+  N_("America/Dominica"),
+  /* SUN_BRANDING */
+  N_("America/Santo_Domingo"),
+  /* SUN_BRANDING */
+  N_("Africa/Algiers"),
+  /* SUN_BRANDING */
+  N_("America/Guayaquil"),
+  /* SUN_BRANDING */
+  N_("Pacific/Galapagos"),
+  /* SUN_BRANDING */
+  N_("Europe/Tallinn"),
+  /* SUN_BRANDING */
+  N_("Africa/Cairo"),
+  /* SUN_BRANDING */
+  N_("Africa/El_Aaiun"),
+  /* SUN_BRANDING */
+  N_("Europe/Madrid"),
+  /* SUN_BRANDING */
+  N_("Africa/Ceuta"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Canary"),
+  /* SUN_BRANDING */
+  N_("Africa/Addis_Ababa"),
+  /* SUN_BRANDING */
+  N_("Europe/Helsinki"),
+  /* SUN_BRANDING */
+  N_("Pacific/Fiji"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Stanley"),
+  /* SUN_BRANDING */
+  N_("Pacific/Truk"),
+  /* SUN_BRANDING */
+  N_("Pacific/Ponape"),
+  /* SUN_BRANDING */
+  N_("Pacific/Kosrae"),
+  /* SUN_BRANDING */
+  N_("Europe/Paris"),
+  /* SUN_BRANDING */
+  N_("Africa/Libreville"),
+  /* SUN_BRANDING */
+  N_("Europe/London"),
+  /* SUN_BRANDING */
+  N_("America/Grenada"),
+  /* SUN_BRANDING */
+  N_("Asia/Tbilisi"),
+  /* SUN_BRANDING */
+  N_("America/Cayenne"),
+  /* SUN_BRANDING */
+  N_("Europe/Guernsey"),
+  /* SUN_BRANDING */
+  N_("Africa/Accra"),
+  /* SUN_BRANDING */
+  N_("Europe/Gibraltar"),
+  /* SUN_BRANDING */
+  N_("America/Godthab"),
+  /* SUN_BRANDING */
+  N_("America/Danmarkshavn"),
+  /* SUN_BRANDING */
+  N_("America/Scoresbysund"),
+  /* SUN_BRANDING */
+  N_("America/Thule"),
+  /* SUN_BRANDING */
+  N_("Africa/Banjul"),
+  /* SUN_BRANDING */
+  N_("Africa/Conakry"),
+  /* SUN_BRANDING */
+  N_("America/Guadeloupe"),
+  /* SUN_BRANDING */
+  N_("Africa/Malabo"),
+  /* SUN_BRANDING */
+  N_("Europe/Athens"),
+  /* SUN_BRANDING */
+  N_("Atlantic/South_Georgia"),
+  /* SUN_BRANDING */
+  N_("America/Guatemala"),
+  /* SUN_BRANDING */
+  N_("Pacific/Guam"),
+  /* SUN_BRANDING */
+  N_("Africa/Bissau"),
+  /* SUN_BRANDING */
+  N_("America/Guyana"),
+  /* SUN_BRANDING */
+  N_("Asia/Hong_Kong"),
+  /* SUN_BRANDING */
+  N_("America/Tegucigalpa"),
+  /* SUN_BRANDING */
+  N_("Europe/Zagreb"),
+  /* SUN_BRANDING */
+  N_("America/Port-au-Prince"),
+  /* SUN_BRANDING */
+  N_("Europe/Budapest"),
+  /* SUN_BRANDING */
+  N_("Asia/Jakarta"),
+  /* SUN_BRANDING */
+  N_("Asia/Pontianak"),
+  /* SUN_BRANDING */
+  N_("Asia/Makassar"),
+  /* SUN_BRANDING */
+  N_("Asia/Jayapura"),
+  /* SUN_BRANDING */
+  N_("Europe/Dublin"),
+  /* SUN_BRANDING */
+  N_("Asia/Jerusalem"),
+  /* SUN_BRANDING */
+  N_("Europe/Isle_of_Man"),
+  /* SUN_BRANDING */
+  N_("Indian/Chagos"),
+  /* SUN_BRANDING */
+  N_("Asia/Baghdad"),
+  /* SUN_BRANDING */
+  N_("Asia/Tehran"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Reykjavik"),
+  /* SUN_BRANDING */
+  N_("Europe/Rome"),
+  /* SUN_BRANDING */
+  N_("Europe/Jersey"),
+  /* SUN_BRANDING */
+  N_("America/Jamaica"),
+  /* SUN_BRANDING */
+  N_("Asia/Amman"),
+  /* SUN_BRANDING */
+  N_("Asia/Tokyo"),
+  /* SUN_BRANDING */
+  N_("Africa/Nairobi"),
+  /* SUN_BRANDING */
+  N_("Asia/Bishkek"),
+  /* SUN_BRANDING */
+  N_("Asia/Phnom_Penh"),
+  /* SUN_BRANDING */
+  N_("Pacific/Tarawa"),
+  /* SUN_BRANDING */
+  N_("Pacific/Enderbury"),
+  /* SUN_BRANDING */
+  N_("Pacific/Kiritimati"),
+  /* SUN_BRANDING */
+  N_("Indian/Comoro"),
+  /* SUN_BRANDING */
+  N_("America/St_Kitts"),
+  /* SUN_BRANDING */
+  N_("Asia/Pyongyang"),
+  /* SUN_BRANDING */
+  N_("Asia/Seoul"),
+  /* SUN_BRANDING */
+  N_("Asia/Kuwait"),
+  /* SUN_BRANDING */
+  N_("America/Cayman"),
+  /* SUN_BRANDING */
+  N_("Asia/Almaty"),
+  /* SUN_BRANDING */
+  N_("Asia/Qyzylorda"),
+  /* SUN_BRANDING */
+  N_("Asia/Aqtobe"),
+  /* SUN_BRANDING */
+  N_("Asia/Aqtau"),
+  /* SUN_BRANDING */
+  N_("Asia/Oral"),
+  /* SUN_BRANDING */
+  N_("Asia/Vientiane"),
+  /* SUN_BRANDING */
+  N_("Asia/Beirut"),
+  /* SUN_BRANDING */
+  N_("America/St_Lucia"),
+  /* SUN_BRANDING */
+  N_("Europe/Vaduz"),
+  /* SUN_BRANDING */
+  N_("Asia/Colombo"),
+  /* SUN_BRANDING */
+  N_("Africa/Monrovia"),
+  /* SUN_BRANDING */
+  N_("Africa/Maseru"),
+  /* SUN_BRANDING */
+  N_("Europe/Vilnius"),
+  /* SUN_BRANDING */
+  N_("Europe/Luxembourg"),
+  /* SUN_BRANDING */
+  N_("Europe/Riga"),
+  /* SUN_BRANDING */
+  N_("Africa/Tripoli"),
+  /* SUN_BRANDING */
+  N_("Africa/Casablanca"),
+  /* SUN_BRANDING */
+  N_("Europe/Monaco"),
+  /* SUN_BRANDING */
+  N_("Europe/Chisinau"),
+  /* SUN_BRANDING */
+  N_("Europe/Podgorica"),
+  /* SUN_BRANDING */
+  N_("Indian/Antananarivo"),
+  /* SUN_BRANDING */
+  N_("Pacific/Majuro"),
+  /* SUN_BRANDING */
+  N_("Pacific/Kwajalein"),
+  /* SUN_BRANDING */
+  N_("Europe/Skopje"),
+  /* SUN_BRANDING */
+  N_("Africa/Bamako"),
+  /* SUN_BRANDING */
+  N_("Asia/Rangoon"),
+  /* SUN_BRANDING */
+  N_("Asia/Ulaanbaatar"),
+  /* SUN_BRANDING */
+  N_("Asia/Hovd"),
+  /* SUN_BRANDING */
+  N_("Asia/Choibalsan"),
+  /* SUN_BRANDING */
+  N_("Asia/Macau"),
+  /* SUN_BRANDING */
+  N_("Pacific/Saipan"),
+  /* SUN_BRANDING */
+  N_("America/Martinique"),
+  /* SUN_BRANDING */
+  N_("Africa/Nouakchott"),
+  /* SUN_BRANDING */
+  N_("America/Montserrat"),
+  /* SUN_BRANDING */
+  N_("Europe/Malta"),
+  /* SUN_BRANDING */
+  N_("Indian/Mauritius"),
+  /* SUN_BRANDING */
+  N_("Indian/Maldives"),
+  /* SUN_BRANDING */
+  N_("Africa/Blantyre"),
+  /* SUN_BRANDING */
+  N_("America/Mexico_City"),
+  /* SUN_BRANDING */
+  N_("America/Cancun"),
+  /* SUN_BRANDING */
+  N_("America/Merida"),
+  /* SUN_BRANDING */
+  N_("America/Monterrey"),
+  /* SUN_BRANDING */
+  N_("America/Mazatlan"),
+  /* SUN_BRANDING */
+  N_("America/Chihuahua"),
+  /* SUN_BRANDING */
+  N_("America/Hermosillo"),
+  /* SUN_BRANDING */
+  N_("America/Tijuana"),
+  /* SUN_BRANDING */
+  N_("Asia/Kuala_Lumpur"),
+  /* SUN_BRANDING */
+  N_("Asia/Kuching"),
+  /* SUN_BRANDING */
+  N_("Africa/Maputo"),
+  /* SUN_BRANDING */
+  N_("Africa/Windhoek"),
+  /* SUN_BRANDING */
+  N_("Pacific/Noumea"),
+  /* SUN_BRANDING */
+  N_("Africa/Niamey"),
+  /* SUN_BRANDING */
+  N_("Pacific/Norfolk"),
+  /* SUN_BRANDING */
+  N_("Africa/Lagos"),
+  /* SUN_BRANDING */
+  N_("America/Managua"),
+  /* SUN_BRANDING */
+  N_("Europe/Amsterdam"),
+  /* SUN_BRANDING */
+  N_("Europe/Oslo"),
+  /* SUN_BRANDING */
+  N_("Asia/Katmandu"),
+  /* SUN_BRANDING */
+  N_("Pacific/Nauru"),
+  /* SUN_BRANDING */
+  N_("Pacific/Niue"),
+  /* SUN_BRANDING */
+  N_("Pacific/Auckland"),
+  /* SUN_BRANDING */
+  N_("Pacific/Chatham"),
+  /* SUN_BRANDING */
+  N_("Asia/Muscat"),
+  /* SUN_BRANDING */
+  N_("America/Panama"),
+  /* SUN_BRANDING */
+  N_("America/Lima"),
+  /* SUN_BRANDING */
+  N_("Pacific/Tahiti"),
+  /* SUN_BRANDING */
+  N_("Pacific/Marquesas"),
+  /* SUN_BRANDING */
+  N_("Pacific/Gambier"),
+  /* SUN_BRANDING */
+  N_("Pacific/Port_Moresby"),
+  /* SUN_BRANDING */
+  N_("Asia/Manila"),
+  /* SUN_BRANDING */
+  N_("Asia/Karachi"),
+  /* SUN_BRANDING */
+  N_("Europe/Warsaw"),
+  /* SUN_BRANDING */
+  N_("America/Miquelon"),
+  /* SUN_BRANDING */
+  N_("Pacific/Pitcairn"),
+  /* SUN_BRANDING */
+  N_("America/Puerto_Rico"),
+  /* SUN_BRANDING */
+  N_("Asia/Gaza"),
+  /* SUN_BRANDING */
+  N_("Europe/Lisbon"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Madeira"),
+  /* SUN_BRANDING */
+  N_("Atlantic/Azores"),
+  /* SUN_BRANDING */
+  N_("Pacific/Palau"),
+  /* SUN_BRANDING */
+  N_("America/Asuncion"),
+  /* SUN_BRANDING */
+  N_("Asia/Qatar"),
+  /* SUN_BRANDING */
+  N_("Indian/Reunion"),
+  /* SUN_BRANDING */
+  N_("Europe/Bucharest"),
+  /* SUN_BRANDING */
+  N_("Europe/Belgrade"),
+  /* SUN_BRANDING */
+  N_("Europe/Kaliningrad"),
+  /* SUN_BRANDING */
+  N_("Europe/Moscow"),
+  /* SUN_BRANDING */
+  N_("Europe/Volgograd"),
+  /* SUN_BRANDING */
+  N_("Europe/Samara"),
+  /* SUN_BRANDING */
+  N_("Asia/Yekaterinburg"),
+  /* SUN_BRANDING */
+  N_("Asia/Omsk"),
+  /* SUN_BRANDING */
+  N_("Asia/Novosibirsk"),
+  /* SUN_BRANDING */
+  N_("Asia/Krasnoyarsk"),
+  /* SUN_BRANDING */
+  N_("Asia/Irkutsk"),
+  /* SUN_BRANDING */
+  N_("Asia/Yakutsk"),
+  /* SUN_BRANDING */
+  N_("Asia/Vladivostok"),
+  /* SUN_BRANDING */
+  N_("Asia/Sakhalin"),
+  /* SUN_BRANDING */
+  N_("Asia/Magadan"),
+  /* SUN_BRANDING */
+  N_("Asia/Kamchatka"),
+  /* SUN_BRANDING */
+  N_("Asia/Anadyr"),
+  /* SUN_BRANDING */
+  N_("Africa/Kigali"),
+  /* SUN_BRANDING */
+  N_("Asia/Riyadh"),
+  /* SUN_BRANDING */
+  N_("Pacific/Guadalcanal"),
+  /* SUN_BRANDING */
+  N_("Indian/Mahe"),
+  /* SUN_BRANDING */
+  N_("Africa/Khartoum"),
+  /* SUN_BRANDING */
+  N_("Europe/Stockholm"),
+  /* SUN_BRANDING */
+  N_("Asia/Singapore"),
+  /* SUN_BRANDING */
+  N_("Atlantic/St_Helena"),
+  /* SUN_BRANDING */
+  N_("Europe/Ljubljana"),
+  /* SUN_BRANDING */
+  N_("Arctic/Longyearbyen"),
+  /* SUN_BRANDING */
+  N_("Europe/Bratislava"),
+  /* SUN_BRANDING */
+  N_("Africa/Freetown"),
+  /* SUN_BRANDING */
+  N_("Europe/San_Marino"),
+  /* SUN_BRANDING */
+  N_("Africa/Dakar"),
+  /* SUN_BRANDING */
+  N_("Africa/Mogadishu"),
+  /* SUN_BRANDING */
+  N_("America/Paramaribo"),
+  /* SUN_BRANDING */
+  N_("Africa/Sao_Tome"),
+  /* SUN_BRANDING */
+  N_("America/El_Salvador"),
+  /* SUN_BRANDING */
+  N_("Asia/Damascus"),
+  /* SUN_BRANDING */
+  N_("Africa/Mbabane"),
+  /* SUN_BRANDING */
+  N_("America/Grand_Turk"),
+  /* SUN_BRANDING */
+  N_("Africa/Ndjamena"),
+  /* SUN_BRANDING */
+  N_("Indian/Kerguelen"),
+  /* SUN_BRANDING */
+  N_("Africa/Lome"),
+  /* SUN_BRANDING */
+  N_("Asia/Bangkok"),
+  /* SUN_BRANDING */
+  N_("Asia/Dushanbe"),
+  /* SUN_BRANDING */
+  N_("Pacific/Fakaofo"),
+  /* SUN_BRANDING */
+  N_("Asia/Dili"),
+  /* SUN_BRANDING */
+  N_("Asia/Ashgabat"),
+  /* SUN_BRANDING */
+  N_("Africa/Tunis"),
+  /* SUN_BRANDING */
+  N_("Pacific/Tongatapu"),
+  /* SUN_BRANDING */
+  N_("Europe/Istanbul"),
+  /* SUN_BRANDING */
+  N_("America/Port_of_Spain"),
+  /* SUN_BRANDING */
+  N_("Pacific/Funafuti"),
+  /* SUN_BRANDING */
+  N_("Asia/Taipei"),
+  /* SUN_BRANDING */
+  N_("Africa/Dar_es_Salaam"),
+  /* SUN_BRANDING */
+  N_("Europe/Kiev"),
+  /* SUN_BRANDING */
+  N_("Europe/Uzhgorod"),
+  /* SUN_BRANDING */
+  N_("Europe/Zaporozhye"),
+  /* SUN_BRANDING */
+  N_("Europe/Simferopol"),
+  /* SUN_BRANDING */
+  N_("Africa/Kampala"),
+  /* SUN_BRANDING */
+  N_("Pacific/Johnston"),
+  /* SUN_BRANDING */
+  N_("Pacific/Midway"),
+  /* SUN_BRANDING */
+  N_("Pacific/Wake"),
+  /* SUN_BRANDING */
+  N_("America/New_York"),
+  /* SUN_BRANDING */
+  N_("America/Detroit"),
+  /* SUN_BRANDING */
+  N_("America/Kentucky/Louisville"),
+  /* SUN_BRANDING */
+  N_("America/Kentucky/Monticello"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Indianapolis"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Vincennes"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Marengo"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Petersburg"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Vevay"),
+  /* SUN_BRANDING */
+  N_("America/Chicago"),
+  /* SUN_BRANDING */
+  N_("America/Indiana/Knox"),
+  /* SUN_BRANDING */
+  N_("America/Menominee"),
+  /* SUN_BRANDING */
+  N_("America/North_Dakota/Center"),
+  /* SUN_BRANDING */
+  N_("America/North_Dakota/New_Salem"),
+  /* SUN_BRANDING */
+  N_("America/Denver"),
+  /* SUN_BRANDING */
+  N_("America/Boise"),
+  /* SUN_BRANDING */
+  N_("America/Shiprock"),
+  /* SUN_BRANDING */
+  N_("America/Phoenix"),
+  /* SUN_BRANDING */
+  N_("America/Los_Angeles"),
+  /* SUN_BRANDING */
+  N_("America/Anchorage"),
+  /* SUN_BRANDING */
+  N_("America/Juneau"),
+  /* SUN_BRANDING */
+  N_("America/Yakutat"),
+  /* SUN_BRANDING */
+  N_("America/Nome"),
+  /* SUN_BRANDING */
+  N_("America/Adak"),
+  /* SUN_BRANDING */
+  N_("Pacific/Honolulu"),
+  /* SUN_BRANDING */
+  N_("America/Montevideo"),
+  /* SUN_BRANDING */
+  N_("Asia/Samarkand"),
+  /* SUN_BRANDING */
+  N_("Asia/Tashkent"),
+  /* SUN_BRANDING */
+  N_("Europe/Vatican"),
+  /* SUN_BRANDING */
+  N_("America/St_Vincent"),
+  /* SUN_BRANDING */
+  N_("America/Caracas"),
+  /* SUN_BRANDING */
+  N_("America/Tortola"),
+  /* SUN_BRANDING */
+  N_("America/St_Thomas"),
+  /* SUN_BRANDING */
+  N_("Pacific/Efate"),
+  /* SUN_BRANDING */
+  N_("Pacific/Wallis"),
+  /* SUN_BRANDING */
+  N_("Pacific/Apia"),
+  /* SUN_BRANDING */
+  N_("Asia/Aden"),
+  /* SUN_BRANDING */
+  N_("Indian/Mayotte"),
+  /* SUN_BRANDING */
+  N_("Africa/Johannesburg"),
+  /* SUN_BRANDING */
+  N_("Africa/Lusaka"),
+  /* SUN_BRANDING */
+  N_("Africa/Harare"),
+  /* SUN_BRANDING */
+  N_("Europe/"),
+  /* SUN_BRANDING */
+  N_("Europe/"),
+  /* SUN_BRANDING */
+  N_("Europe/"),
+  /* SUN_BRANDING */
+  N_("America/Buenos_Aires"),
+  /* SUN_BRANDING */
+  N_("America/Cordoba"),
+  /* SUN_BRANDING */
+  N_("America/Jujuy"),
+  /* SUN_BRANDING */
+  N_("America/Catamarca"),
+  /* SUN_BRANDING */
+  N_("America/Mendoza"),
+  /* SUN_BRANDING */
+  N_("Australia/LHI"),
+  /* SUN_BRANDING */
+  N_("Australia/NSW"),
+  /* SUN_BRANDING */
+  N_("PRC"),
+  /* SUN_BRANDING */
+  N_("Hongkong"),
+  /* SUN_BRANDING */
+  N_("Japan"),
+  /* SUN_BRANDING */
+  N_("ROK"),
+  /* SUN_BRANDING */
+  N_("Poland"),
+  /* SUN_BRANDING */
+  N_("US/Pacific"),
+  NULL
+};
+#endif
--- gnome-system-tools-2.14.0/src/common/zone_tab.sh.orig	2009-04-06 15:29:09.628838000 +0900
+++ gnome-system-tools-2.14.0/src/common/zone_tab.sh	2009-04-06 20:57:32.178223000 +0900
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+# This script creates a xml file from /usr/share/zoneinfo/zone.tab
+# to localize time zone info.
+# static const char *third_party_app_menu [] = {
+#  N_("This is a test message to check the .mo file."),
+
+OS=`uname -s`
+if [ "$OS" != "SunOS" ] ; then
+  TZ_DATA_FILE=/usr/share/zoneinfo/zone.tab
+else
+  TZ_DATA_FILE=/usr/share/lib/zoneinfo/tab/zone_sun.tab
+fi
+ZONE_H_FILE=`basename $0 .sh`.h
+
+echo "#ifndef __ZONE_TAB_H__" > $ZONE_H_FILE
+echo "#define __ZONE_TAB_H__" >> $ZONE_H_FILE
+echo "" >> $ZONE_H_FILE
+echo '#include <config.h>' >> $ZONE_H_FILE
+echo '#include <glib/gi18n.h>' >> $ZONE_H_FILE
+echo "static const char *dummy_zone_location [] = {" >> $ZONE_H_FILE
+cat $TZ_DATA_FILE | awk '$1 !~ /^#/{
+  COMMENT="/* SUN_BRANDING */"
+  printf ("  %s\n  N_(\"%s\"),\n", COMMENT, $3)}' \
+>> $ZONE_H_FILE
+cat $TZ_DATA_FILE | awk '$1 !~ /^#/{
+  COMMENT="/* SUN_BRANDING */"
+  if ($4 !~ /-/) {
+    printf ("  %s\n  N_(\"%s\"),\n", COMMENT, $4)
+  }}' |\
+  grep -v 'N_("-")' \
+>> $ZONE_H_FILE
+echo "  NULL" >> $ZONE_H_FILE
+echo "};"     >> $ZONE_H_FILE
+echo "#endif" >> $ZONE_H_FILE
+
--- gnome-system-tools-2.14.0/src/time/main.c.orig	2009-04-06 18:36:28.194077000 +0900
+++ gnome-system-tools-2.14.0/src/time/main.c	2009-04-06 18:38:06.171271000 +0900
@@ -208,7 +208,7 @@ gst_time_init_timezone (GstTimeTool *tim
 	locs = tz_get_locations (e_tz_map_get_tz_db (tzmap));
 
 	for (i = 0; g_ptr_array_index (locs, i); i++)
-		gtk_combo_box_append_text (GTK_COMBO_BOX (w), tz_location_get_zone (g_ptr_array_index (locs, i)));
+		gtk_combo_box_append_text (GTK_COMBO_BOX (w), _(tz_location_get_zone (g_ptr_array_index (locs, i))));
 }
 
 #define is_leap_year(yyy) ((((yyy % 4) == 0) && ((yyy % 100) != 0)) || ((yyy % 400) == 0));
@@ -433,7 +433,7 @@ update_tz (GstTimeTool *time_tool)
 	l = gst_dialog_get_widget (GST_DIALOG (GST_TOOL (time_tool)->main_dialog), "tzlabel");
 
 	if (time_tool->time_zone_name) {
-		gtk_label_set_text (GTK_LABEL (l), time_tool->time_zone_name);
+		gtk_label_set_text (GTK_LABEL (l), _(time_tool->time_zone_name));
 	}
 }      
 
--- gnome-system-tools-2.14.0/src/time/tz-map.c.orig	2009-04-06 10:21:39.385846000 +0900
+++ gnome-system-tools-2.14.0/src/time/tz-map.c	2009-04-06 13:07:22.899948000 +0900
@@ -21,6 +21,7 @@
  */
 
 #include "config.h"
+#include <glib/gi18n.h>
 #include <glade/glade.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -30,6 +31,7 @@
 #include <string.h>
 
 #include "gst.h"
+#include "zone_tab.h"
 #include "time-tool.h"
 
 #include "tz.h"
@@ -147,7 +149,7 @@ e_tz_map_set_location_text (ETzMap *tzma
 	while (valid) {
 		gtk_tree_model_get (model, &iter, 0, &location, -1);
 
-		if (strcmp (location, name) == 0) {
+		if (strcmp (location, _(name)) == 0) {
 			gtk_combo_box_set_active_iter (GTK_COMBO_BOX (location_combo), &iter);
 			valid = FALSE;
 		} else {
@@ -190,6 +192,18 @@ e_tz_map_set_tz_from_name (ETzMap *tzmap
 	e_tz_map_set_location_text (tzmap, tz_location_get_zone (e_tz_map_location_from_point (tzmap, tzmap->point_selected)));
 }
 
+static gchar *
+get_en_location (gchar *localized_location)
+{
+	int i;
+	for (i = 0; localized_location && dummy_zone_location[i]; i++) {
+		if (!strcmp (localized_location, _(dummy_zone_location[i])))
+			return dummy_zone_location[i];
+	}
+
+	return localized_location;
+}
+
 static gboolean
 update_map (GtkWidget *w, gpointer data)
 {
@@ -203,7 +217,7 @@ update_map (GtkWidget *w, gpointer data)
 	if (gtk_combo_box_get_active_iter (GTK_COMBO_BOX (w), &iter)) {
 		model = gtk_combo_box_get_model (GTK_COMBO_BOX (w));
 		gtk_tree_model_get (model, &iter, 0, &location, -1);
-		e_tz_map_set_tz_from_name (tzmap, location);
+		e_tz_map_set_tz_from_name (tzmap, get_en_location (location));
 	}
 
 	return TRUE;
@@ -224,7 +238,7 @@ e_tz_map_get_selected_tz_name (ETzMap *t
 		gtk_tree_model_get (model, &iter, 0, &location, -1);
 	}
 
-	return location;
+	return get_en_location (location);
 }
 
 
@@ -307,7 +321,7 @@ motion (GtkWidget *widget, GdkEventMotio
 	 * practice there are no reasons why it should */
 
 	gtk_label_set_text (GTK_LABEL (((GstTimeTool *) tzmap->tool)->map_hover_label),
-			    tz_location_get_zone (e_tz_map_location_from_point (tzmap, tzmap->point_hover)));
+			    _(tz_location_get_zone (e_tz_map_location_from_point (tzmap, tzmap->point_hover))));
 
 	return TRUE;
 }
