diff -ur /export/home/malin/gnome-system-tools-2.14.0/src/network/network.desktop.in.in gnome-system-tools-2.14.0/src/network/network.desktop.in.in
--- /export/home/malin/gnome-system-tools-2.14.0/src/network/network.desktop.in.in	2006-01-31 06:18:45.000000000 +0800
+++ gnome-system-tools-2.14.0/src/network/network.desktop.in.in	2010-02-26 13:30:36.665000735 +0800
@@ -1,5 +1,7 @@
 [Desktop Entry]
-_Name=Networking
+# SUN_BRANDING
+_Name=Network
+NoDisplay=true
 _Comment=Configure network devices and connections
 Exec=network-admin
 Icon=@pixmapsdir@/network.png
diff -ur /export/home/malin/gnome-system-tools-2.14.0/src/services/services.desktop.in.in gnome-system-tools-2.14.0/src/services/services.desktop.in.in
--- /export/home/malin/gnome-system-tools-2.14.0/src/services/services.desktop.in.in	2006-01-31 06:18:45.000000000 +0800
+++ gnome-system-tools-2.14.0/src/services/services.desktop.in.in	2010-02-26 13:29:32.091911012 +0800
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=Services
-_Comment=Configure which services will be run when the system starts
+# SUN_BRANDING
+_Comment=Start and stop daemons and system services
 Exec=services-admin
 Icon=@pixmapsdir@/services.png
 Terminal=false
diff -ur /export/home/malin/gnome-system-tools-2.14.0/src/shares/shares.desktop.in.in gnome-system-tools-2.14.0/src/shares/shares.desktop.in.in
--- /export/home/malin/gnome-system-tools-2.14.0/src/shares/shares.desktop.in.in	2006-01-31 06:18:46.000000000 +0800
+++ gnome-system-tools-2.14.0/src/shares/shares.desktop.in.in	2010-02-26 13:29:32.092038440 +0800
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=Shared Folders
-_Comment=Configure which folders are available for your network neighborhood
+# SUN_BRANDING
+_Comment=Choose which folders are visible to other users on your network
 Exec=shares-admin
 Icon=@pixmapsdir@/shares.png
 Terminal=false
diff -ur /export/home/malin/gnome-system-tools-2.14.0/src/users/users.desktop.in.in gnome-system-tools-2.14.0/src/users/users.desktop.in.in
--- /export/home/malin/gnome-system-tools-2.14.0/src/users/users.desktop.in.in	2006-01-31 06:18:46.000000000 +0800
+++ gnome-system-tools-2.14.0/src/users/users.desktop.in.in	2010-02-26 13:29:32.092165471 +0800
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=Users and Groups
-_Comment=Add or remove users and groups
+# SUN_BRANDING
+_Comment=Add, remove or edit the users and groups that may access this computer
 Exec=users-admin
 Icon=@pixmapsdir@/users.png
 Terminal=false
