--- evolution-data-server-2.32.3/camel/camel-store.c.~1~	2011-04-21 23:35:36.000000000 +0400
+++ evolution-data-server-2.32.3/camel/camel-store.c	2015-01-11 12:45:02.239953971 +0300
@@ -152,6 +152,7 @@
 	CamelServiceClass *service_class;
 	CamelStore *store = CAMEL_STORE(service);
 	gchar *store_db_path, *store_path = NULL;
+	int loop = 3;
 
 	/* Chain up to parent's construct() method. */
 	service_class = CAMEL_SERVICE_CLASS (camel_store_parent_class);
@@ -198,7 +199,12 @@
 	}
 	g_free (store_db_path);
 
-	if (camel_db_create_folders_table (store->cdb_r, error)) {
+	while (loop && camel_db_create_folders_table (store->cdb_r, error) ) {
+		usleep (100000);
+		g_clear_error (error);
+		loop--;
+	}
+	if (!loop) {
 		g_warning ("something went wrong terribly during db creation \n");
 		return FALSE;
 	}
