--- evolution-data-server-2.27.91/libedataserverui/e-name-selector-entry.c.orig	2009-09-03 16:55:01.816776000 +0800
+++ evolution-data-server-2.27.91/libedataserverui/e-name-selector-entry.c	2009-09-03 16:55:04.896892000 +0800
@@ -438,7 +438,7 @@
 		comma_str = NULL;
 	}
 
-	g_string_append (out, " (beginswith ");
+	g_string_append (out, " (contains ");
 	e_sexp_encode_string (out, field);
 	e_sexp_encode_string (out, spaced_str);
 	g_string_append (out, ")");
