--- evolution-data-server-2.29.6/servers/groupwise/Makefile.am.orig	2010-01-27 14:41:01.201200494 +0800
+++ evolution-data-server-2.29.6/servers/groupwise/Makefile.am	2010-01-27 14:40:42.520525244 +0800
@@ -1,4 +1,4 @@
-#noinst_PROGRAMS = create-account soap-test
+noinst_PROGRAMS = create-account soap-test
 
 SERVERS_GROUPWISE_CPPFLAGS= \
 	-DG_LOG_DOMAIN=\"libegroupwise\"		\
@@ -20,7 +20,8 @@
 	$(top_builddir)/libedataserver/libedataserver-1.2.la	\
 	$(SOUP_LIBS)						\
 	libegroupwise-1.2.la					\
-	$(E_DATA_SERVER_LIBS)
+	$(E_DATA_SERVER_LIBS)					\
+	-lgthread-2.0
 
 build-timestamp.h : $(top_builddir)/config.h
 	echo "const char *build_timestamp = \"Evolution-Data-Server-@BASE_VERSION@-`date +%F`\";" > $@
