--- evolution-data-server-2.32.3/camel/camel-sasl-gssapi.c.~1~	2015-01-11 12:38:53.805121705 +0300
+++ evolution-data-server-2.32.3/camel/camel-sasl-gssapi.c	2015-01-11 12:40:31.369715982 +0300
@@ -44,21 +44,8 @@
 #include "camel-session.h"
 
 #ifdef HAVE_KRB5
-
-#ifdef HAVE_HEIMDAL_KRB5
-#include <krb5.h>
-#else
-#include <krb5/krb5.h>
-#endif /* HAVE_HEIMDAL_KRB5 */
-
-#ifdef HAVE_ET_COM_ERR_H
-#include <et/com_err.h>
-#else
-#ifdef HAVE_COM_ERR_H
-#include <com_err.h>
-#endif /* HAVE_COM_ERR_H */
-#endif /* HAVE_ET_COM_ERR_H */
-
+#include <kerberosv5/krb5.h>
+#include <kerberosv5/com_err.h>
 #ifdef HAVE_MIT_KRB5
 #include <gssapi/gssapi.h>
 #include <gssapi/gssapi_generic.h>
