--- evolution-data-server-2.29.6/configure.ac.orig	2010-01-27 15:25:33.184240914 +0800
+++ evolution-data-server-2.29.6/configure.ac	2010-01-27 15:29:47.431166006 +0800
@@ -1374,6 +1374,7 @@
 AM_CONDITIONAL(ENABLE_LARGEFILE, [test "x$enable_largefile" = "xyes"])
 
 EVO_SET_COMPILE_FLAGS(CAMEL, $mozilla_nss gio-2.0 sqlite3 >= sqlite_minimum_version gthread-2.0, $THREADS_CFLAGS $KRB5_CFLAGS $MANUAL_NSS_CFLAGS $LARGEFILE_CFLAGS, -lz $THREADS_LIBS $KRB5_LIBS $MANUAL_NSS_LIBS)
+CAMEL_LIBS="$CAMEL_LIBS -lsocket -lnsl"
 AC_SUBST(CAMEL_CFLAGS)
 AC_SUBST(CAMEL_LIBS)
 AC_SUBST(CAMEL_GROUPWISE_CFLAGS)
