--- node-v24.0.0/deps/v8/src/base/platform/platform-posix.cc.orig
+++ node-v24.0.0/deps/v8/src/base/platform/platform-posix.cc
@@ -82,7 +82,7 @@
 
 #if defined(V8_OS_SOLARIS)
 #if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE > 2) || defined(__EXTENSIONS__)
-extern "C" int madvise(caddr_t, size_t, int);
+extern "C" int madvise(void *, size_t, int);
 #else
 extern int madvise(caddr_t, size_t, int);
 #endif
