--- OpenSP-1.5.2/configure.in.orig
+++ OpenSP-1.5.2/configure.in
@@ -238,7 +238,7 @@
 dnl Checks for libraries.
 AC_CHECK_LIB(pthread,pthread_create,,AC_CHECK_LIB(threads,cthread_fork))
 
-AM_GNU_GETTEXT
+AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION([0.14.5])
 AC_CHECK_HEADERS(locale.h)
 AC_DEFINE_DIR(SP_LOCALE_DIR, datadir/locale, [location of message catalogs])
@@ -491,7 +491,7 @@
 	lib/Makefile nsgmls/Makefile spent/Makefile
 	spam/Makefile sgmlnorm/Makefile sx/Makefile spcat/Makefile
 	include/Makefile include/config.h.old generic/Makefile doc/Makefile
-	pubtext/Makefile tests/Makefile unicode/Makefile intl/Makefile
+	pubtext/Makefile tests/Makefile unicode/Makefile
 	po/Makefile.in])
 
 BUILDDOCS=yes
--- OpenSP-1.5.2/Makefile.am.orig
+++ OpenSP-1.5.2/Makefile.am
@@ -4,7 +4,7 @@
 
 SUBDIRS = pubtext unicode doc \
 	include generic \
-	intl po \
+	po \
 	lib nsgmls spent spam sgmlnorm sx spcat \
 	tests @DOCSRC@
 
