--- OpenSP-1.5.1~/include/InternalInputSource.h  2003-07-17 
14:14:00.000000000 -0400
+++ OpenSP-1.5.1/include/InternalInputSource.h  2008-11-13 
12:24:04.591406546 -0500
@@ -34,7 +34,7 @@
   void pushCharRef(Char ch, const NamedCharRef &);
   Boolean rewind(Messenger &);
   const StringC *contents();
-  InternalInputSource *InternalInputSource::asInternalInputSource();
+  InternalInputSource *asInternalInputSource();
   ~InternalInputSource();
 private:
   InternalInputSource(const InternalInputSource &); // undefined
