--- openal-soft-1.16.0/CMakeLists.txt.~1~	2015-07-21 14:20:26.610939157 +0300
+++ openal-soft-1.16.0/CMakeLists.txt	2015-07-21 14:21:19.153155895 +0300
@@ -1013,7 +1013,7 @@
 SET(prefix ${CMAKE_INSTALL_PREFIX})
 SET(exec_prefix "\${prefix}")
 SET(libdir "\${exec_prefix}/lib${LIB_SUFFIX}")
-SET(bindir "\${exec_prefix}/bin")
+SET(bindir "\${exec_prefix}/bin${LIB_SUFFIX}")
 SET(includedir "\${prefix}/include")
 SET(PACKAGE_VERSION "${LIB_VERSION}")
 
@@ -1095,7 +1095,7 @@
 
 # Add an install target here
 INSTALL(TARGETS ${LIBNAME}
-        RUNTIME DESTINATION bin
+        RUNTIME DESTINATION "bin${LIB_SUFFIX}"
         LIBRARY DESTINATION "lib${LIB_SUFFIX}"
         ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
 )
@@ -1160,7 +1160,7 @@
     ENDIF()
 
     INSTALL(TARGETS openal-info makehrtf
-            RUNTIME DESTINATION bin
+            RUNTIME DESTINATION "bin${LIB_SUFFIX}"
             LIBRARY DESTINATION "lib${LIB_SUFFIX}"
             ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
     )
@@ -1200,7 +1200,7 @@
                                                                            ${SDL_SOUND_INCLUDE_DIR})
 
         INSTALL(TARGETS alstream alreverb allatency alloopback
-                RUNTIME DESTINATION bin
+                RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                 LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                 ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
         )
@@ -1236,7 +1236,7 @@
                                                                              ${FFMPEG_INCLUDE_DIRS})
 
             INSTALL(TARGETS alffplay
-                    RUNTIME DESTINATION bin
+                    RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                     LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                     ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
             )
