From solaris-xorg

+++ b/configure.ac	2017-04-17 18:40:39.007084697 +0000
@@ -230,7 +230,7 @@
 	    GLDSONAME="libfltk_gl.so.$FL_DSO_VERSION"
 	    IMGDSONAME="libfltk_images.so.$FL_DSO_VERSION"
 	    CAIRODSONAME="libfltk_cairo.so.$FL_DSO_VERSION"
-	    DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -h \$@ \$(LDLIBS) -G $DEBUGFLAG -o"
+	    DSOCOMMAND="\$(CXX) \$(LDFLAGS) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
 	    if test "x$libdir" != "x/usr/lib"; then
 		    DSOLINK="-R$libdir"
 	    fi
--- a/configure	2017-04-17 18:49:30.345562062 +0000
+++ b/configure	2017-04-17 18:50:04.812193611 +0000
@@ -3837,7 +3837,7 @@
 	    GLDSONAME="libfltk_gl.so.$FL_DSO_VERSION"
 	    IMGDSONAME="libfltk_images.so.$FL_DSO_VERSION"
 	    CAIRODSONAME="libfltk_cairo.so.$FL_DSO_VERSION"
-	    DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -h \$@ \$(LDLIBS) -G $DEBUGFLAG -o"
+	    DSOCOMMAND="\$(CXX) \$(LDFLAGS) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
 	    if test "x$libdir" != "x/usr/lib"; then
 		    DSOLINK="-R$libdir"
 	    fi
