--- mkdocs-1.5.2/mkdocs/tests/plugin_tests.py.orig
+++ mkdocs-1.5.2/mkdocs/tests/plugin_tests.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 from __future__ import annotations
 
 import os
--- mkdocs-1.5.2/mkdocs/tests/cli_tests.py.orig
+++ mkdocs-1.5.2/mkdocs/tests/cli_tests.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 import io
 import logging
--- mkdocs-1.5.2/mkdocs/__main__.py.orig
+++ mkdocs-1.5.2/mkdocs/__main__.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 from __future__ import annotations
 
--- mkdocs-1.5.2/mkdocs/tests/build_tests.py.orig
+++ mkdocs-1.5.2/mkdocs/tests/build_tests.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 from __future__ import annotations
 
 import contextlib
