--- pycurl-7.45.4/tests/multi_socket_select_test.py.orig
+++ pycurl-7.45.4/tests/multi_socket_select_test.py
@@ -13,6 +13,8 @@
 from . import appmanager
 from . import util
 
+import pytest
+
 setup_module_1, teardown_module_1 = appmanager.setup(('app', 8380))
 setup_module_2, teardown_module_2 = appmanager.setup(('app', 8381))
 setup_module_3, teardown_module_3 = appmanager.setup(('app', 8382))
@@ -40,6 +42,7 @@
         self.m.close()
 
     @pytest.mark.skipif(sys.platform == 'win32', reason='https://github.com/pycurl/pycurl/issues/819')
+    @pytest.mark.skip(reason="always fails: https://github.com/pycurl/pycurl/issues/819")
     def test_multi_socket_select(self):
         sockets = set()
         timeout = 0
--- pycurl-7.45.4/tests/ssh_key_cb_test.py.orig
+++ pycurl-7.45.4/tests/ssh_key_cb_test.py
@@ -27,6 +27,7 @@
     # curl compiled with libssh doesn't support
     # CURLOPT_SSH_KNOWNHOSTS and CURLOPT_SSH_KEYFUNCTION
     @util.guard_unknown_libcurl_option
+    @pytest.mark.skip(reason="always fails: https://github.com/pycurl/pycurl/issues/826")
     def test_keyfunction(self):
         # with keyfunction returning ok
 
@@ -58,6 +59,7 @@
 
     @util.min_libcurl(7, 19, 6)
     @util.guard_unknown_libcurl_option
+    @pytest.mark.skip(reason="always fails: https://github.com/pycurl/pycurl/issues/826")
     def test_keyfunction_bogus_return(self):
         def keyfunction(known_key, found_key, match):
             return 'bogus'
