--- libgksu-2.0.12/libgksu/Makefile.am.1	2014-04-02 17:44:42.387706021 +0400
+++ libgksu-2.0.12/libgksu/Makefile.am	2014-04-02 17:46:52.555056419 +0400
@@ -22,8 +22,8 @@
 pkgconfigdir = ${libdir}/pkgconfig
 pkgconfig_DATA = libgksu2.pc
 
-pkglibdir = ${libdir}/${PACKAGE}
-pkglib_PROGRAMS = gksu-run-helper
+pkglibexecdir = ${libdir}/${PACKAGE}
+pkglibexec_PROGRAMS = gksu-run-helper
 gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
 gksu_run_helper_SOURCES = gksu-run-helper.c
 
