--- clean-compizconfig-python-0.8.4/configure.ac	2009-12-15 14:20:12.883953756 +0100
+++ compizconfig-python-0.8.4//configure.ac	2009-12-17 14:10:08.856864880 +0100
@@ -1,5 +1,5 @@
 AC_PREREQ(2.57)
-AC_INIT([compizconfig-python],esyscmd(. ./VERSION;echo -n $VERSION), [livinglatexkali@gmail.com])
+AC_INIT([compizconfig-python],esyscmd(. ./VERSION;echo $VERSION | tr -d "\n"), [livinglatexkali@gmail.com])
 
 AM_INIT_AUTOMAKE([1.9 dist-bzip2])
 AC_CONFIG_HEADER([config.h])
@@ -21,15 +21,10 @@ AC_SUBST(CCS_REQUIRES)
 
 dnl PYTHON SPECIFIC
 
-m4_define(required_python_abi,2.4)
+m4_define(required_python_abi,2.6)
 
-AM_PATH_PYTHON
-if test "$PYTHON_VERSION" = "2.4"
-then
-	AM_PATH_PYTHON_VERSION(2.4,2.4.0)
-else
-	AM_PATH_PYTHON_VERSION(2.5,2.5.0)
-fi
+AM_PATH_PYTHON(2.6,,PYTHON="/usr/bin/python2.6")
+AM_PATH_PYTHON_VERSION(2.6,2.6.4)
 
 AC_CHECK_PROGS(PYREX, pyrexc)
 if test -z "$PYREX"
