--- ./configure.ac.orig	2015-06-15 18:18:35.739567907 +0000
+++ ./configure.ac	2015-06-15 18:20:05.612627569 +0000
@@ -2,6 +2,7 @@
 AC_INIT([compizconfig-python],esyscmd(. ./VERSION;echo $VERSION | tr -d "\n"), [livinglatexkali@gmail.com])
 
 AM_INIT_AUTOMAKE([1.9 dist-bzip2])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADER([config.h])
 AM_MAINTAINER_MODE
 
--- ./Makefile.am.orig	2015-06-15 18:18:54.923942509 +0000
+++ ./Makefile.am	2015-06-15 18:20:45.757654039 +0000
@@ -1,3 +1,5 @@
+ACLOCAL_AMFLAGS = -I m4 --install
+
 SUBDIRS = src
 
 EXTRA_DIST = \
