--- compizconfig-python/src/Makefile.am.orig	2015-06-15 18:39:36.455930609 +0000
+++ compizconfig-python/src/Makefile.am	2015-06-15 18:40:19.591678968 +0000
@@ -3,7 +3,7 @@
 		 @CCS_CFLAGS@
 
 compizconfig.c: compizconfig.pyx
-	-pyrexc compizconfig.pyx
+	-pyrexc $(srcdir)/compizconfig.pyx -o $(builddir)/compizconfig.c
 
 compizconfig_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initcompizconfig -lX11
 compizconfig_la_LIBADD = @CCS_LIBS@
