diff -wpruN --no-dereference '--exclude=*.orig' a~/sql/mysqld.cc a/sql/mysqld.cc
--- a~/sql/mysqld.cc	1970-01-01 00:00:00
+++ a/sql/mysqld.cc	1970-01-01 00:00:00
@@ -469,7 +469,7 @@ Atomic_counter<uint32_t> slave_open_temp
 */
 Atomic_counter<ulonglong> sending_new_binlog_file;
 ulong thread_created;
-ulong back_log, connect_timeout, server_id;
+ulong back_log, connect_timeout, concurrency, server_id;
 ulong what_to_log;
 ulong slow_launch_time;
 ulong open_files_limit, max_binlog_size;
