diff --git a/js/src/jsfun.h b/js/src/jsfun.h
index a52092c..9555a03 100644
--- a/jsfun.h
+++ b/jsfun.h
@@ -144,6 +144,7 @@ struct JSFunction : public JSObject_Slots2
         void            *nativeOrScript;
     } u;
     JSAtom          *atom;        /* name for diagnostics and decompiling */
+    void            *dummy;       /* for building with Solaris Studio */
 
     bool optimizedClosure()  const { return FUN_KIND(this) > JSFUN_INTERPRETED; }
     bool needsWrapper()      const { return FUN_NULL_CLOSURE(this) && u.i.skipmin != 0; }
