diff --git a/js/src/configure b/js/src/configure
index 0e98daf..11bb34f 100755
--- a/configure
+++ b/configure
@@ -14560,10 +14560,10 @@ if test -n "$INTEL_CC"; then
 fi
 
 if test "$SOLARIS_SUNPRO_CC"; then
-  PROFILE_GEN_CFLAGS="-xprofile=collect:$_objdir/$enable_application"
-  PROFILE_GEN_LDFLAGS="-xprofile=collect:$_objdir/$enable_application"
-  PROFILE_USE_CFLAGS="-xprofile=use:$_objdir/$enable_application"
-  PROFILE_USE_LDFLAGS="-xprofile=use:$_objdir/$enable_application"
+  PROFILE_GEN_CFLAGS="-xprofile=collect:$_objdir/../../$enable_application"
+  PROFILE_GEN_LDFLAGS="-xprofile=collect:$_objdir/../../$enable_application"
+  PROFILE_USE_CFLAGS="-xtemp=$_objdir -xlinkopt=2 -xprofile=use:$_objdir/../../$enable_application"
+  PROFILE_USE_LDFLAGS="-xtemp=$_objdir -xlinkopt=2 -xprofile=use:$_objdir/../../$enable_application"
 fi
 
