--- c_icap_modules-0.5.7/services/url_check/Makefile.am.orig	2023-09-28 11:56:46.000000000 +0200
+++ c_icap_modules-0.5.7/services/url_check/Makefile.am	2024-02-09 22:52:44.880721721 +0100
@@ -40,8 +40,8 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(CONFIGDIR)
-	$(INSTALL) $(srcdir)/srv_url_check.conf $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf.default
-	if test ! -f $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf; then $(INSTALL) srv_url_check.conf $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf; fi
+	$(INSTALL) -D $(srcdir)/srv_url_check.conf $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf.default
+	if test ! -f $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf; then $(INSTALL) $(srcdir)/srv_url_check.conf $(DESTDIR)$(CONFIGDIR)/srv_url_check.conf; fi
 	$(mkinstalldirs) $(DESTDIR)$(cicapdatadir)/templates/srv_url_check/en
 	for f in $(TMPLS); do  $(INSTALL) $(srcdir)/$$f $(DESTDIR)$(cicapdatadir)/templates/srv_url_check/en/; done
 
