Solaris-specific; not suitable for upstream

--- /tmp/Makefile.shared	Mon Feb 14 14:39:29 2011
+++ openssl-1.0.0d/Makefile.shared	Mon Feb 14 14:50:52 2011
@@ -108,7 +108,7 @@
     LD_LIBRARY_PATH=$$LIBPATH:$$LD_LIBRARY_PATH \
     $${SHAREDCMD} $${SHAREDFLAGS} \
 	-o $$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX \
-	$$ALLSYMSFLAGS $$SHOBJECTS $$NOALLSYMSFLAGS $$LIBDEPS \
+	$$ALLSYMSFLAGS $$SHOBJECTS $$NOALLSYMSFLAGS $$LIBDEPS -lc \
   ) && $(SYMLINK_SO)
 
 SYMLINK_SO=	\
