--- libevent-1.3e/configure.orig	Wed Apr 27 12:01:29 2011
+++ libevent-1.3e/configure	Wed Apr 27 12:01:48 2011
@@ -9129,8 +9129,8 @@
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
-  soname_spec='${libname}${release}${shared_ext}$major'
+  library_names_spec='${libname}${shared_ext}$versuffix ${libname}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
   hardcode_into_libs=yes
