# This patch has been fed upstream.
--- /dev/null						2013-06-18 19:31:00.000000000 -0700
+++ sendmail-8.14.7/devtools/M4/UNIX/shared-library.m4	2013-06-18 20:02:40.335623219 -0700
@@ -0,0 +1,35 @@
+divert(-1)
+#
+# Copyright (c) 1999-2001, 2006 Sendmail, Inc. and its suppliers.
+#	All rights reserved.
+#
+# By using this file, you agree to the terms and conditions set
+# forth in the LICENSE file which can be found at the top level of
+# the sendmail distribution.
+#
+#
+#  Definitions for Makefile construction for sendmail
+#
+#	$Id: library.m4,v 8.11 2006/09/05 22:56:53 ca Exp $
+#
+divert(0)dnl
+include(confBUILDTOOLSDIR`/M4/'bldM4_TYPE_DIR`/links.m4')dnl
+bldLIST_PUSH_ITEM(`bldC_PRODUCTS', bldCURRENT_PRODUCT)dnl
+bldPUSH_TARGET(bldCURRENT_PRODUCT`.so')dnl
+bldPUSH_INSTALL_TARGET(`install-'bldCURRENT_PRODUCT)dnl
+bldPUSH_CLEAN_TARGET(bldCURRENT_PRODUCT`-clean')dnl
+
+include(confBUILDTOOLSDIR`/M4/'bldM4_TYPE_DIR`/defines.m4')
+divert(bldTARGETS_SECTION)
+bldCURRENT_PRODUCT.so: ${BEFORE} ${bldCURRENT_PRODUCT`OBJS'}
+	${CCLINK} ${LDOPTS} -G -o bldCURRENT_PRODUCT.so ${bldCURRENT_PRODUCT`OBJS'} -lc
+ifdef(`bldLINK_SOURCES', `bldMAKE_SOURCE_LINKS(bldLINK_SOURCES)')
+
+install-`'bldCURRENT_PRODUCT: bldCURRENT_PRODUCT.so
+ifdef(`bldINSTALLABLE', `	ifdef(`confMKDIR', `if [ ! -d ${DESTDIR}${LIBDIR} ]; then confMKDIR -p ${DESTDIR}${LIBDIR}; else :; fi ')
+	${INSTALL} -c -o ${LIBOWN} -g ${LIBGRP} -m ${LIBMODE} bldCURRENT_PRODUCT.so ${DESTDIR}${LIBDIR}')
+
+bldCURRENT_PRODUCT-clean:
+	rm -f ${OBJS} bldCURRENT_PRODUCT.so ${MANPAGES}
+
+divert(0)
