--- git-1.7.3.2/git_remote_helpers/Makefile.orig	Tue Nov 15 05:25:37 2011
+++ git-1.7.3.2/git_remote_helpers/Makefile	Tue Nov 15 06:35:06 2011
@@ -23,13 +23,15 @@
 
 PYLIBDIR=$(shell $(PYTHON_PATH) -c \
 	 "import sys; \
-	 print 'lib/python%i.%i/site-packages' % sys.version_info[:2]")
+	 print 'lib/python%i.%i/vendor-packages' % sys.version_info[:2]")
 
 all: $(pysetupfile)
 	$(QUIET)$(PYTHON_PATH) $(pysetupfile) $(QUIETSETUP) build
 
 install: $(pysetupfile)
-	$(PYTHON_PATH) $(pysetupfile) install --prefix $(DESTDIR_SQ)$(prefix)
+	$(PYTHON_PATH) $(pysetupfile) install \
+	--install-lib=$(DESTDIR_SQ)$(prefix)/$(PYLIBDIR) \
+	--prefix $(DESTDIR_SQ)$(prefix)
 
 instlibdir: $(pysetupfile)
 	@echo "$(DESTDIR_SQ)$(prefix)/$(PYLIBDIR)"
