--- zope_proxy-6.1/src/zope/proxy/tests/test_proxy.py.orig
+++ zope_proxy-6.1/src/zope/proxy/tests/test_proxy.py
@@ -1114,6 +1114,7 @@
         self.assertFalse(self._callFUT(proxy1, proxy2))
 
 
+@unittest.skip("always fails with zope.security 7.2: https://github.com/zopefoundation/zope.proxy/issues/72")
 class Test_sameProxiedObjects(Test_py_sameProxiedObjects):
 
     def _callFUT(self, *args):
@@ -1337,6 +1338,7 @@
         self.assertIs(self._callFUT(proxy), c)
 
 
+@unittest.skip("always fails with zope.security 7.2: https://github.com/zopefoundation/zope.proxy/issues/72")
 class Test_removeAllProxies(Test_py_removeAllProxies):
 
     def _callFUT(self, *args):
