--- opal-3.10.10/plugins/plugin-inc.mak.in.orig	2016-07-31 15:42:10.608990027 +0200
+++ opal-3.10.10/plugins/plugin-inc.mak.in	2016-07-31 15:44:02.151612438 +0200
@@ -30,7 +30,7 @@
 
 CC            := @CC@
 CXX           := @CXX@
-CFLAGS        += @CFLAGS@ -I@prefix@/include -I$(PLUGINDIR)/../include -I$(PLUGINDIR)
+CFLAGS        += @CFLAGS@ -fPIC -I@prefix@/include -I$(PLUGINDIR)/../include -I$(PLUGINDIR)
 LDFLAGS       += @LDFLAGS@ @LDSO@
 PLUGINEXT     :=@PLUGINEXT@
 
