diff -Nurb opal-3.10.10.orig/src/codec/opalpluginmgr.cxx opal-3.10.10/src/codec/opalpluginmgr.cxx
--- opal-3.10.10.orig/src/codec/opalpluginmgr.cxx	2013-02-20 03:18:46.000000000 +0100
+++ opal-3.10.10/src/codec/opalpluginmgr.cxx	2016-07-31 15:45:34.810717429 +0200
@@ -53,6 +53,8 @@
 #include <codec/vidcodec.h>
 #endif
 
+#include <alloca.h>
+
 
 PFACTORY_CREATE_SINGLETON(PFactory<PPluginModuleManager>, OpalPluginCodecManager);
 
diff -Nurb opal-3.10.10.orig/src/codec/opalwavfile.cxx opal-3.10.10/src/codec/opalwavfile.cxx
--- opal-3.10.10.orig/src/codec/opalwavfile.cxx	2013-02-20 03:18:46.000000000 +0100
+++ opal-3.10.10/src/codec/opalwavfile.cxx	2016-07-31 15:45:34.811283666 +0200
@@ -36,6 +36,7 @@
 
 #include <codec/g711codec.h>
 
+#include <alloca.h>
 
 
 #define new PNEW
diff -Nurb opal-3.10.10.orig/src/opal/opalmixer.cxx opal-3.10.10/src/opal/opalmixer.cxx
--- opal-3.10.10.orig/src/opal/opalmixer.cxx	2013-02-20 03:18:31.000000000 +0100
+++ opal-3.10.10/src/opal/opalmixer.cxx	2016-07-31 15:45:34.774532717 +0200
@@ -41,6 +41,7 @@
 #include <ptlib/vconvert.h>
 #include <ptclib/pwavfile.h>
 
+#include <alloca.h>
 
 #define DETAIL_LOG_LEVEL 6
 
