--- opal-3.10.10/configure.ac.orig	2016-07-31 03:54:30.197989095 +0200
+++ opal-3.10.10/configure.ac	2016-07-31 03:54:39.969795127 +0200
@@ -139,7 +139,7 @@
 CFLAGS="$CFLAGS $PTLIB_CFLAGS"
 LDFLAGS="$LDFLAGS $PTLIB_CFLAGS"
 CXXFLAGS="$CXXFLAGS $PTLIB_CFLAGS $PTLIB_CXXFLAGS"
-#LIBS="$LIBS $PTLIB_LIBS"
+LIBS="$LIBS $PTLIB_LIBS"
 
 
 dnl ########################################################################
