--- libxml2-2.6.31.orig/Makefile.in	Fri Jan 11 01:10:54 2008
+++ libxml2-2.6.31/Makefile.in	Wed Jan 16 14:33:06 2008
@@ -486,7 +486,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = include . doc example xstc @PYTHON_SUBDIR@
+SUBDIRS = include . @PYTHON_SUBDIR@
 DIST_SUBDIRS = include . doc example python xstc
 INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
 bin_SCRIPTS = xml2-config
