--- libxml2-2.7.6/relaxng.c.proto-mismatch	Thu Oct  8 00:39:20 2009
+++ libxml2-2.7.6/relaxng.c	Thu Oct  8 00:41:57 2009
@@ -5369,7 +5369,7 @@
     } else {
         xmlRngPErr(ctxt, node, XML_RNGP_CHOICE_CONTENT,
                    "expecting name, anyName, nsName or choice : got %s\n",
-                   (node == NULL ? "nothing" : node->name), NULL);
+                   (node == NULL ? (const unsigned char *)"nothing" : node->name), NULL);
         return (NULL);
     }
     if (ret != def) {
