--- qt-creator-opensource-src-8.0.2/src/plugins/clangformat/CMakeLists.txt.~1~	2022-10-19 16:33:27.000000000 +0300
+++ qt-creator-opensource-src-8.0.2/src/plugins/clangformat/CMakeLists.txt	2025-05-28 19:14:17.778590282 +0300
@@ -24,11 +24,6 @@
 endif()
 
 extend_qtc_plugin(ClangFormat
-  CONDITION UNIX AND NOT APPLE
-  PROPERTIES LINK_FLAGS "-Wl,--exclude-libs,ALL"
-)
-
-extend_qtc_plugin(ClangFormat
   CONDITION WITH_TESTS
   SOURCES
     tests/clangformat-test.cpp
