--- qt-creator-opensource-src-4.11.2/src/tools/clangpchmanagerbackend/source/collectusedmacroactionfactory.h.orig	Thu Mar  3 19:47:21 2022
+++ qt-creator-opensource-src-4.11.2/src/tools/clangpchmanagerbackend/source/collectusedmacroactionfactory.h	Thu Mar  3 14:18:05 2022
@@ -61,13 +61,13 @@
                                                                     diagnosticConsumer);
     }
 
-    clang::FrontendAction *create() override
+    clang::ASTFrontendAction create() override
     {
-        return new CollectUsedMacrosAction(m_usedMacros,
+        return (std::unique_ptr<clang::FrontendAction>)(new CollectUsedMacrosAction(m_usedMacros,
                                            m_filePathCache,
                                            m_sourceDependencies,
                                            m_sourceFiles,
-                                           m_fileStatuses);
+                                           m_fileStatuses) );
     }
 
 private:
