--- qt-creator-opensource-src-4.11.2/src/plugins/clangformat/clangformatutils.cpp.orig	Thu Mar  3 19:47:50 2022
+++ qt-creator-opensource-src-4.11.2/src/plugins/clangformat/clangformatutils.cpp	Tue Mar  1 13:32:07 2022
@@ -51,13 +51,13 @@
     style.Language = FormatStyle::LK_Cpp;
     style.AccessModifierOffset = -4;
     style.AlignAfterOpenBracket = FormatStyle::BAS_Align;
-    style.AlignConsecutiveAssignments = false;
-    style.AlignConsecutiveDeclarations = false;
+    style.AlignConsecutiveAssignments = FormatStyle::ACS_Consecutive; //false;
+    style.AlignConsecutiveDeclarations = FormatStyle::ACS_Consecutive; //false;
     style.AlignEscapedNewlines = FormatStyle::ENAS_DontAlign;
-    style.AlignOperands = true;
+    style.AlignOperands = FormatStyle::OAS_DontAlign; // true;
     style.AlignTrailingComments = true;
     style.AllowAllParametersOfDeclarationOnNextLine = true;
-    style.AllowShortBlocksOnASingleLine = false;
+    style.AllowShortBlocksOnASingleLine = FormatStyle::SBS_Empty; //= false;
     style.AllowShortCaseLabelsOnASingleLine = false;
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
 #if LLVM_VERSION_MAJOR >= 9
@@ -72,7 +72,7 @@
     style.BinPackArguments = false;
     style.BinPackParameters = false;
     style.BraceWrapping.AfterClass = true;
-    style.BraceWrapping.AfterControlStatement = false;
+    style.BraceWrapping.AfterControlStatement = FormatStyle::BWACS_Never; // false;
     style.BraceWrapping.AfterEnum = false;
     style.BraceWrapping.AfterFunction = true;
     style.BraceWrapping.AfterNamespace = false;
@@ -128,7 +128,7 @@
     style.PenaltyReturnTypeOnItsOwnLine = 300;
     style.PointerAlignment = FormatStyle::PAS_Right;
     style.ReflowComments = false;
-    style.SortIncludes = true;
+    style.SortIncludes = FormatStyle::SI_Never; // true;
     style.SortUsingDeclarations = true;
     style.SpaceAfterCStyleCast = true;
     style.SpaceAfterTemplateKeyword = false;
@@ -136,7 +136,7 @@
     style.SpaceBeforeParens = FormatStyle::SBPO_ControlStatements;
     style.SpaceInEmptyParentheses = false;
     style.SpacesBeforeTrailingComments = 1;
-    style.SpacesInAngles = false;
+    style.SpacesInAngles = FormatStyle::SIAS_Never; // false;
     style.SpacesInContainerLiterals = false;
     style.SpacesInCStyleCastParentheses = false;
     style.SpacesInParentheses = false;
