--- qt-creator-opensource-src-4.11.2/src/tools/clangrefactoringbackend/source/symbolscollector.cpp.orig	Thu Mar  3 19:48:05 2022
+++ qt-creator-opensource-src-4.11.2/src/tools/clangrefactoringbackend/source/symbolscollector.cpp	Tue Mar  1 20:52:33 2022
@@ -74,10 +74,11 @@
             : m_action(consumerFactory)
         {}
 
-        clang::FrontendAction *create() override { return new AdaptorAction(m_action); }
+        std::unique_ptr<clang::FrontendAction> create() override { 
+			return( (std::unique_ptr<clang::FrontendAction>)(new AdaptorAction(m_action)) ); }
 
     private:
-        class AdaptorAction : public clang::ASTFrontendAction
+        class AdaptorAction : public clang::ASTFrontendAction // class ASTFrontendAction : public FrontendAction {...
         {
         public:
             AdaptorAction(CollectSymbolsAction *action)
