--- qt-creator-opensource-src-4.11.2/src/shared/qbs/src/lib/corelib/tools/processutils.cpp.orig	Thu Mar  3 19:48:09 2022
+++ qt-creator-opensource-src-4.11.2/src/shared/qbs/src/lib/corelib/tools/processutils.cpp	Mon Feb 28 20:08:01 2022
@@ -57,13 +57,25 @@
 # if !defined(Q_OS_NETBSD)
 #   include <sys/user.h>
 # endif
+#elif defined(Q_OS_UNIX)
+#include <iostream>
+#include <cstdlib>
+#include <procfs.h>
+#include <fstream>
+#include <string>
 #else
 #   error Missing implementation of processNameByPid for this platform.
 #endif
 
+#include <sys/procfs.h>
+
 namespace qbs {
 namespace Internal {
 
+#ifdef Q_OS_UNIX
+
+#endif
+
 QString processNameByPid(qint64 pid)
 {
 #if defined(Q_OS_WIN)
@@ -94,6 +106,20 @@
     sprintf(exePath, "/proc/%lld/exe", pid);
     readlink(exePath, buf, sizeof(buf));
     return FileInfo::fileName(QString::fromUtf8(buf));
+#elif defined(Q_OS_UNIX)
+
+	char filePath[64];
+	psinfo_t info;
+    info.pr_fname[0] = '\0';
+    sprintf(filePath, "/proc/%lld/psinfo", pid);
+	std::ifstream in(filePath);
+    if (in)
+    {
+        in.read((char*)&info, sizeof(psinfo_t));
+        in.close();
+    }
+    return QString::fromUtf8(info.pr_fname);
+
 #elif defined(Q_OS_BSD4)
 # if defined(Q_OS_NETBSD)
     struct kinfo_proc2 kp;
