--- qt-creator-opensource-src-4.11.2/src/plugins/clangformat/clangformatbaseindenter.cpp.orig	Thu Mar  3 19:47:50 2022
+++ qt-creator-opensource-src-4.11.2/src/plugins/clangformat/clangformatbaseindenter.cpp	Tue Mar  1 13:05:16 2022
@@ -42,7 +42,7 @@
                                    ReplacementsToKeep replacementsToKeep)
 {
     style.MaxEmptyLinesToKeep = 100;
-    style.SortIncludes = false;
+    style.SortIncludes = clang::format::FormatStyle::SortIncludesOptions::SI_Never; // false;
     style.SortUsingDeclarations = false;
 
     // This is a separate pass, don't do it unless it's the full formatting.
@@ -343,7 +343,7 @@
                                     .size();
         convertedReplacements.emplace_back(utf16Offset,
                                            utf16Length,
-                                           QString::fromStdString(replacement.getReplacementText()));
+                                           QString::fromStdString( replacement.getReplacementText().str() ) );
     }
 
     return convertedReplacements;
