--- qt-creator-opensource-src-4.11.2/src/libs/utils/process_stub.pro.orig	Thu Mar  3 19:48:18 2022
+++ qt-creator-opensource-src-4.11.2/src/libs/utils/process_stub.pro	Thu Mar  3 08:00:04 2022
@@ -1,5 +1,6 @@
 CONFIG          += warn_on console use_c_linker
 CONFIG          -= qt app_bundle
+LIBS += -lsocket
 
 include(../../../qtcreator.pri)
 
@@ -14,7 +15,7 @@
 
 unix {
     SOURCES += process_stub_unix.c
-    solaris-.*: LIBS += -lsocket
+    solaris*: LIBS += -lsocket
 } else {
     SOURCES += process_stub_win.c
     LIBS += -lshell32
