--- qt-creator-opensource-src-4.11.2/src/tools/clangrefactoringbackend/source/collectsymbolsaction.h.orig	Thu Mar  3 19:47:22 2022
+++ qt-creator-opensource-src-4.11.2/src/tools/clangrefactoringbackend/source/collectsymbolsaction.h	Tue Mar  1 14:29:16 2022
@@ -48,7 +48,7 @@
 public:
     CollectSymbolsAction(std::shared_ptr<IndexDataConsumer> indexDataConsumer)
         : clang::WrapperFrontendAction(
-            clang::index::createIndexingAction(indexDataConsumer, createIndexingOptions(), nullptr))
+            clang::index::createIndexingAction(indexDataConsumer, createIndexingOptions()))
         , m_indexDataConsumer(indexDataConsumer)
     {}
 
