--- sntp/m4/ntp_libevent.m4.~1~	2015-04-13 20:37:19.822964156 +0300
+++ sntp/m4/ntp_libevent.m4	2015-04-13 20:38:33.318954409 +0300
@@ -100,9 +100,9 @@
 	    done
 	    case "$LIBISC_PTHREADS_NOTHREADS" in
 	     pthreads)
-		LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_pthreads"
+		LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_pthreads-2.0"
 	    esac
-	    LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_core"
+	    LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_core-2.0"
 	    AC_MSG_RESULT([yes])
 	else
 	    ntp_use_local_libevent=yes
