--- sntp/header.h.orig	Thu May 14 06:34:41 2009
+++ sntp/header.h	Thu May 14 06:15:12 2009
@@ -19,10 +19,10 @@
 #define MAX_SOCKETS        10          /* Maximum number of addresses */
 
 #ifndef LOCKNAME
-#    define LOCKNAME "/etc/sntp.pid"  /* Stores the pid */
+#    define LOCKNAME "/var/run/sntp.pid"  /* Stores the pid */
 #endif
 #ifndef SAVENAME
-#    define SAVENAME "/etc/sntp.state" /* Stores the recovery state */
+#    define SAVENAME "/var/ntp/sntp.state" /* Stores the recovery state */
 #endif
 
 
--- include/ntp_config.h.orig	Thu May 14 06:34:41 2009
+++ include/ntp_config.h	Thu May 14 06:15:12 2009
@@ -10,7 +10,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE	"%windir%\\system32\\drivers\\etc\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.conf"
--- libntp/audio.c.orig	Thu May 14 06:34:41 2009
+++ libntp/audio.c	Thu May 14 06:15:12 2009
@@ -53,7 +53,7 @@
 static struct audio_device device; /* audio device ident */
 #endif /* HAVE_SYS_AUDIOIO_H */
 #ifdef PCM_STYLE_SOUND
-# define INIT_FILE "/etc/ntp.audio"
+# define INIT_FILE "/etc/inet/ntp.audio"
 int agc =	SOUND_MIXER_WRITE_RECLEV; /* or IGAIN or LINE */
 int monitor =	SOUND_MIXER_WRITE_VOLUME; /* or OGAIN */
 int devmask = 0;
--- ntpdate/ntpdate.c.orig	Thu May 14 06:34:41 2009
+++ ntpdate/ntpdate.c	Thu May 14 06:15:12 2009
@@ -192,7 +192,7 @@
 
 #ifndef KEYFILE
 # ifndef SYS_WINNT
-#define KEYFILE 	"/etc/ntp.keys"
+#define KEYFILE 	"/etc/inet/ntp.keys"
 # else
 #define KEYFILE 	"%windir%\\ntp.keys"
 # endif /* SYS_WINNT */
--- ntpdate/ntptimeset.c.orig	Thu May 14 06:34:41 2009
+++ ntpdate/ntptimeset.c	Thu May 14 06:15:12 2009
@@ -270,7 +270,7 @@
 
 #ifndef KEYFILE
 # ifndef SYS_WINNT
-#define KEYFILE 	"/etc/ntp.keys"
+#define KEYFILE 	"/etc/inet/ntp.keys"
 # else
 #define KEYFILE 	"%windir%\\ntp.keys"
 # endif /* SYS_WINNT */
--- ntpdate/ntptime_config.c.orig	Thu May 14 06:34:41 2009
+++ ntpdate/ntptime_config.c	Thu May 14 06:15:12 2009
@@ -37,7 +37,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE 	"%windir%\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.ini"
--- ntpd/refclock_oncore.c.orig	Thu May 14 06:34:41 2009
+++ ntpd/refclock_oncore.c	Thu May 14 06:15:12 2009
@@ -1188,7 +1188,7 @@
 
 	FILE	*fd;
 	char	*cp, *cc, *ca, line[100], units[2], device[20], Msg[160], **cpp;
-	char	*dirs[] = { "/etc/ntp", "/etc", 0 };
+	char	*dirs[] = { "/etc/inet", "/etc", 0 };
 	int	i, sign, lat_flg, long_flg, ht_flg, mode, mask;
 	double	f1, f2, f3;
 
