--- ntpd/Makefile.in.fcs	Mon Mar 28 10:52:07 2011
+++ ntpd/Makefile.in	Mon Mar 28 10:54:13 2011
@@ -243,11 +243,11 @@
 am__include = @am__include@
 am__leading_dot = @am__leading_dot@
 am__quote = @am__quote@
 am__tar = @am__tar@
 am__untar = @am__untar@
-bindir = ${exec_prefix}/${BINSUBDIR}
+bindir = @libexecdir@
 build = @build@
 build_alias = @build_alias@
 build_cpu = @build_cpu@
 build_os = @build_os@
 build_vendor = @build_vendor@
