*** ntpd/ntp_io.c.orig	Fri Jul 30 06:57:58 2010
--- ntpd/ntp_io.c	Fri Jul 30 06:58:46 2010
***************
*** 1869,1875 ****
  
  	case AF_INET:
  		if (setsockopt(iface->fd, IPPROTO_IP, IP_MULTICAST_IF,
! 			       (char *)NSRCADR(&iface->sin),
  			       sizeof(NSRCADR(&iface->sin)))) {
  
  			msyslog(LOG_ERR,
--- 1869,1875 ----
  
  	case AF_INET:
  		if (setsockopt(iface->fd, IPPROTO_IP, IP_MULTICAST_IF,
! 			       (void *)&NSRCADR(&iface->sin),
  			       sizeof(NSRCADR(&iface->sin)))) {
  
  			msyslog(LOG_ERR,
