--- scripts/ntp-wait.in.orig	Thu May 14 06:34:42 2009
+++ scripts/ntp-wait.in	Thu May 14 06:15:13 2009
@@ -20,8 +20,8 @@
     while(<Q>) {
       chomp;
       # the first line should be similar to:
-      # associd=0 status=0645 leap_none, sync_ntp, ...
-      if (/^associd=0 status=(\S{4}) (\S+), (\S+),/) {
+      # status=0645 leap_none, sync_ntp, ...
+      if (/status=(\S{4}) (\S+), (\S+),/) {
 	my $status = $1;
 	my $leap = $2;
 	my $sync = $3;
