--- php-8.3.22/configure	2025-06-03 19:17:57.000000000 +0200
+++ php-8.3.22/configure.new	2025-07-06 19:23:48.197625774 +0200
@@ -46815,7 +46815,7 @@
       ln -s "$IMAP_DIR/$PHP_LIBDIR/c-client.a" "$IMAP_DIR/$PHP_LIBDIR/libc-client.a" >/dev/null 2>&1
     fi
 
-    for lib in c-client4 c-client imap; do
+    for lib in  c-client ; do
       IMAP_LIB=$lib
 
   str="$IMAP_DIR/$PHP_LIBDIR/lib$lib.*"
@@ -46824,9 +46824,9 @@
   done
 
 
-  str="$IMAP_DIR/c-client/lib$lib.*"
+  str="$IMAP_DIR/lib$lib.*"
   for i in `echo $str`; do
-    test -r $i && IMAP_LIBDIR=$IMAP_DIR/c-client && break 2
+    test -r $i && IMAP_LIBDIR=$IMAP_DIR && break 2
   done
 
     done
