--- Makefile.in	2010-11-25 09:55:59.000000000 -0800
+++ Makefile.in	2011-04-14 11:19:09.472192719 -0700
@@ -76,8 +76,8 @@
 swig_rb_libdir = @libdir@
 
 ### these possibly need further discussion
-swig_pydir = @libdir@/svn-python/libsvn
-swig_pydir_extra = @libdir@/svn-python/svn
+swig_pydir = /usr/lib/python2.6/vendor-packages/libsvn
+swig_pydir_extra = /usr/lib/python2.6/vendor-packages/svn
 swig_pldir = @libdir@/svn-perl
 swig_rbdir = $(SWIG_RB_SITE_ARCH_DIR)/svn/ext
 contribdir = @bindir@/svn-contrib
