--- configure	2012-05-10 11:52:02.000000000 -0700
+++ configure	2013-03-20 14:20:04.139519952 -0700
@@ -4686,14 +4686,14 @@
   fi
 
   if test "$enable_all_static" = "yes"; then
-    SVN_APR_LIBS="`$apr_config --link-ld --libs`"
+    SVN_APR_LIBS="`$apr_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apr-config --link-ld --libs failed" "$LINENO" 5
+      as_fn_error $? "apr-config --link-libtool --libs failed" "$LINENO" 5
     fi
   else
-    SVN_APR_LIBS="`$apr_config --link-ld`"
+    SVN_APR_LIBS="`$apr_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apr-config --link-ld failed" "$LINENO" 5
+      as_fn_error $? "apr-config --link-libtool failed" "$LINENO" 5
     fi
   fi
   SVN_APR_LIBS="`
@@ -5089,14 +5089,14 @@
   fi
 
   if test "$enable_all_static" = "yes"; then
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld --libs`"
+    SVN_APRUTIL_LIBS="`$apu_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apu-config --link-ld --libs failed" "$LINENO" 5
+      as_fn_error $? "apu-config --link-libtool --libs failed" "$LINENO" 5
     fi
   else
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld`"
+    SVN_APRUTIL_LIBS="`$apu_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apu-config --link-ld failed" "$LINENO" 5
+      as_fn_error $? "apu-config --link-libtool --libs failed" "$LINENO" 5
     fi
   fi
   SVN_APRUTIL_LIBS="`
@@ -5300,6 +5300,8 @@
               old_CFLAGS="$CFLAGS"
               old_LIBS="$LIBS"
               NEON_LIBS=`$PKG_CONFIG neon --libs`
+              NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+              NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
               CFLAGS="$CFLAGS $SVN_NEON_INCLUDES"
               LIBS="$LIBS $NEON_LIBS"
               neon_test_code="
@@ -5320,6 +5322,8 @@
     conftest$ac_exeext conftest.$ac_ext
               if test "$shared_linking" = "no"; then
                 NEON_LIBS=`$PKG_CONFIG neon --libs --static`
+		NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+		NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
                 LIBS="$LIBS $NEON_LIBS"
                 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -5481,6 +5485,8 @@
               old_CFLAGS="$CFLAGS"
               old_LIBS="$LIBS"
               NEON_LIBS=`$PKG_CONFIG neon --libs`
+	      NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+	      NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
               CFLAGS="$CFLAGS $SVN_NEON_INCLUDES"
               LIBS="$LIBS $NEON_LIBS"
               neon_test_code="
@@ -5501,6 +5507,8 @@
     conftest$ac_exeext conftest.$ac_ext
               if test "$shared_linking" = "no"; then
                 NEON_LIBS=`$PKG_CONFIG neon --libs --static`
+		NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+		NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
                 LIBS="$LIBS $NEON_LIBS"
                 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -5520,6 +5528,8 @@
               SVN_NEON_INCLUDES=`$neon_config --cflags | $SED -e 's/-D[^ ]*//g'`
               CFLAGS="$CFLAGS `$neon_config --cflags | $SED -e 's/-I[^ ]*//g'`"
               NEON_LIBS=`$neon_config --libs`
+	      NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+	      NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
             fi
             svn_lib_neon="yes"
             break
@@ -5576,8 +5586,9 @@
 fi
 
 
-
-
+NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-m32##g'`
+NEON_LIBS=`echo ${NEON_LIBS} | sed -e 's#-L/usr/lib##g'`
+export NEON_LIBS
 
 
 if test "$svn_lib_neon" = "yes"; then
