--- pgloader-c603221ae0948d02580e6fcfddfa90f3e6d2e6b0/pgloader/options.py.old	2013-11-02 21:48:56.466895159 +0400
+++ pgloader-c603221ae0948d02580e6fcfddfa90f3e6d2e6b0/pgloader/options.py	2013-11-02 21:49:32.620786698 +0400
@@ -40,7 +40,7 @@
 UDC_PREFIX = 'udc_'
 
 REFORMAT_PATH = None
-DEFAULT_REFORMAT_PATH = ['/usr/share/python-support/pgloader/reformat']
+DEFAULT_REFORMAT_PATH = ['/usr/lib/python2.6/reformat']
 
 DEFAULT_MAX_PARALLEL_SECTIONS = 1
 DEFAULT_SECTION_THREADS       = 1
