--- a/xfontsel.c	2022-07-12 14:08:44.281585500 +0000
+++ b/xfontsel.c	2022-07-12 14:08:51.804374434 +0000
@@ -53,6 +53,9 @@
 #include <X11/Xlib.h>
 #include "ULabel.h"
 
+#include <libintl.h>
+#include <X11/Xlocale.h>
+
 #define MIN_APP_DEFAULTS_VERSION 1
 #define FIELD_COUNT 14
 #define DELIM '-'
@@ -154,6 +157,7 @@
 static void Syntax(const char *call)
 {
     fprintf (stderr, "usage:  %s [-options ...] -fn font\n\n%s\n", call,
+       gettext(
 	"where options include:\n"
 	"    -display dpy           X server to contact\n"
 	"    -geometry geom         size and location of window\n"
@@ -163,7 +167,7 @@
 	"    -sample16 string       sample text to use for 2-byte fonts\n"
 	"    -sampleUCS string      sample text to use for ISO10646 fonts\n"
 	"    -scaled                use scaled instances of fonts\n"
-	"plus any standard toolkit options\n");
+	"plus any standard toolkit options\n"));
     exit (1);
 }
 
@@ -258,10 +262,28 @@
 {
     Widget topLevel, pane;
 
+    /*
+     * Set up internationalize message for bug: 4363521   Jhobson 8/23/00
+     */
+    char *domaindir;
+
     XtSetLanguageProc(NULL, (XtLanguageProc) NULL, NULL);
 
     topLevel = XtAppInitialize(&appCtx, "XFontSel", options, XtNumber(options),
 			       &argc, argv, NULL, NULL, 0);
+    /*
+     * Set up internationalize message for bug: 4363521    Jhobson 8/23/00
+     *
+     * Do this after the AppInitialize since setlocale is setup
+     * by XtSetLanguageProc, but does not occur until XtAppInitialize
+     * happens. (Goofy X server)
+     */
+    textdomain("xfontsel");
+
+    if((domaindir = getenv("TEXTDOMAINDIR")) == NULL){
+        domaindir = getenv("XLOCALEDIR");
+    }
+    bindtextdomain("xfontsel", domaindir);
 
     if (argc != 1) Syntax(argv[0]);
 
@@ -272,12 +294,15 @@
     XtGetApplicationResources( topLevel, (XtPointer)&AppRes,
 			       resources, XtNumber(resources), NZ );
     if (AppRes.app_defaults_version < MIN_APP_DEFAULTS_VERSION) {
+        char message[200];
+        char full_message[300];
 	XrmDatabase rdb = XtDatabase(XtDisplay(topLevel));
-	XtWarning( "app-defaults file not properly installed." );
-	XrmPutLineResource( &rdb,
-"*sampleText*UCSLabel:XFontSel app-defaults file not properly installed;\\n\
-see 'xfontsel' manual page."
-			  );
+        sprintf( message, gettext("app-defaults file not properly installed.") );
+        sprintf( full_message, gettext("%s %s;\\nsee '%s' manual page."),
+                "*sampleText*UCSLabel:XFontSel", message, "xfontsel" );
+
+        XtWarning( message );
+	XrmPutLineResource( &rdb, full_message);
     }
 
     ScheduleWork(GetFontNames, (XtPointer)topLevel, 0);
@@ -519,7 +544,7 @@
 	    }
 	    else
 		XtAppWarning( appCtx,
-		    "internal error; pattern didn't match first font" );
+		    gettext("internal error; pattern didn't match first font" ));
 	}
 	else {
 	    SetNoFonts();
@@ -1056,11 +1081,11 @@
     char label[80];
     Arg args[1];
     if (matchingFontCount == 1)
-	strcpy( label, "1 name matches" );
+	strcpy( label, gettext("1 name matches") );
     else if (matchingFontCount)
-	snprintf( label, sizeof(label), "%d names match", matchingFontCount );
+	snprintf( label, sizeof(label), gettext("%d names match"), matchingFontCount);
     else
-	strcpy( label, "no names match" );
+	strcpy( label, gettext("no names match") );
     XtSetArg( args[0], XtNlabel, label );
     XtSetValues( countLabel, args, ONE );
 }
@@ -1071,9 +1096,9 @@
     char label[80];
     Arg args[1];
     if (count == 1)
-	strcpy( label, "1 name to parse" );
+	strcpy( label, gettext("1 name to parse") );
     else
-	snprintf( label, sizeof(label), "%d names to parse", count );
+	snprintf( label, sizeof(label), gettext("%d names to parse"), count );
     XtSetArg( args[0], XtNlabel, label );
     XtSetValues( countLabel, args, ONE );
     FlushXqueue(XtDisplay(countLabel));
@@ -1279,9 +1304,9 @@
     int field = (long)closure;
     FieldValue *values = fieldValues[field]->value;
     int count = fieldValues[field]->count;
-    printf( "field %d:\n", field );
+    printf(gettext("field %d:\n"), field );
     while (count--) {
-	printf( " %s: %d fonts\n", values->string, values->count );
+	printf( gettext(" %s: %d fonts\n"), values->string, values->count );
 	values++;
     }
     printf( "\n" );
