Do not run tests that fail.

--- gnupg-2.4.8/tests/openpgp/Makefile.am.orig
+++ gnupg-2.4.8/tests/openpgp/Makefile.am
@@ -88,14 +88,12 @@
 	trust-pgp-1.scm \
 	trust-pgp-2.scm \
 	trust-pgp-3.scm \
-	trust-pgp-4.scm \
 	gpgtar.scm \
 	use-exact-key.scm \
 	default-key.scm \
 	export.scm \
 	ssh-import.scm \
 	ssh-export.scm \
-	quick-key-manipulation.scm \
 	key-selection.scm \
 	delete-keys.scm \
 	gpgconf.scm \
--- gnupg-2.4.8/g10/Makefile.am.orig
+++ gnupg-2.4.8/g10/Makefile.am
@@ -183,7 +183,7 @@
 
 
 t_common_ldadd =
-module_tests = t-rmd160 t-keydb t-keydb-get-keyblock t-stutter t-keyid
+module_tests = t-keydb t-keydb-get-keyblock t-stutter t-keyid
 t_rmd160_SOURCES = t-rmd160.c rmd160.c
 t_rmd160_LDADD = $(t_common_ldadd)
 t_keydb_SOURCES = t-keydb.c test-stubs.c $(common_source)
