--- docutils-0.23/test/alltests.py.orig
+++ docutils-0.23/test/alltests.py
@@ -106,7 +106,7 @@
     print(f'Working directory: {Path.cwd()}')
     print(f'Docutils package: {Path(docutils.__file__).parent}')
     sys.stdout.flush()
-    result = unittest.TextTestRunner(resultclass=NumbersTestResult).run(suite)
+    result = unittest.TextTestRunner(resultclass=NumbersTestResult,verbosity=2).run(suite)
     finish = time.time()
     print(f'Elapsed time: {finish - start:.3f} seconds')
     sys.exit(not result.wasSuccessful())
