--- docutils-0.20/test/alltests.py.orig
+++ docutils-0.20/test/alltests.py
@@ -89,7 +89,7 @@
     print(f'Working directory: {os.getcwd()}')
     print(f'Docutils package: {os.path.dirname(docutils.__file__)}')
     sys.stdout.flush()
-    result = unittest.TextTestRunner(resultclass=NumbersTestResult).run(suite)
+    result = unittest.TextTestRunner(resultclass=NumbersTestResult,verbosity=2).run(suite)
     finish = time.time()
     print(f'Elapsed time: {finish - start:.3f} seconds')
     sys.exit(not result.wasSuccessful())
