https://sourceforge.net/p/docutils/bugs/486/

--- /dev/null
+++ docutils-0.21.1/tox.ini
@@ -0,0 +1,17 @@
+[tox]
+minversion = 2.0
+envlist = py{39,310,311,312}
+
+[testenv]
+allowlist_externals =
+  find
+changedir = test
+commands =
+  find . -type f -name "*.pyc" -delete
+  python alltests.py
+
+[testenv:style]
+deps =
+  flake8
+commands =
+  flake8 {toxinidir}
--- /dev/null
+++ docutils-0.21.1/test/functional/output/README.txt
@@ -0,0 +1,3 @@
+This is the directory where the actual output is stored.
+
+This README.txt is just a placeholder to make CVS create the directory.
