--- tmux-1.5/configure.old	2011-07-09 17:12:35.000000000 +0100
+++ tmux-1.5/configure	2011-08-23 17:13:39.706752362 +0100
@@ -4986,7 +4986,7 @@
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $found_cmsg_data" >&5
 $as_echo "$found_cmsg_data" >&6; }
 	if test "x$found_cmsg_data" = xyes; then
-		XOPEN_DEFINES="-D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED"
+		XOPEN_DEFINES="" #-xc99 -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED"
 	else
     		as_fn_error "\"CMSG_DATA not found\"" "$LINENO" 5
 	fi
