--- pkgbuild-1.3.105/configure.in.old	2013-10-08 22:31:22.833908710 +0400
+++ pkgbuild-1.3.105/configure.in	2013-10-08 22:31:42.066133406 +0400
@@ -36,7 +36,7 @@
 SED_IS_GNU=no
 $GNU_SED --version 2>/dev/null | grep 'Free Software Foundation' > /dev/null \
        && SED_IS_GNU=yes
-$GNU_SED --version 2>/dev/null | head -1 | grep 'GNU sed ' > /dev/null \
+$GNU_SED --version 2>/dev/null | head -1 | grep 'GNU sed' > /dev/null \
        || SED_IS_GNU=no
 AC_MSG_RESULT([$SED_IS_GNU])
 if test $SED_IS_GNU = no; then
