--- eom-1.14.1/configure.ac.1	2016-08-01 10:11:38.245985146 +0300
+++ eom-1.14.1/configure.ac	2016-08-01 10:12:24.202195623 +0300
@@ -396,7 +396,7 @@
 if test "x$have_python" != "xno"; then
 	PY_PREFIX=`$PYTHON -c 'import sys ; print sys.prefix'`
 	PY_EXEC_PREFIX=`$PYTHON -c 'import sys ; print sys.exec_prefix'`
-	PYTHON_LIBS="-lpython$PYTHON_VERSION -lutil"
+	PYTHON_LIBS="-lpython$PYTHON_VERSION"
 	PYTHON_LIB_LOC="-L$PY_EXEC_PREFIX/lib/python$PYTHON_VERSION/config"
 	PYTHON_CFLAGS="-I$PY_PREFIX/include/python$PYTHON_VERSION"
 	PYTHON_MAKEFILE="$libdir/python$PYTHON_VERSION/config/Makefile"
