--- eom-1.16.0/src/eom-util.c.orig	Sun May  7 11:03:33 2017
+++ eom-1.16.0/src/eom-util.c	Sun May  7 11:04:07 2017
@@ -28,7 +28,9 @@
 
 #include <sys/time.h>
 #ifdef HAVE_STRPTIME
+#ifndef __sun
 #define _XOPEN_SOURCE
+#endif
 #endif /* HAVE_STRPTIME */
 
 #include <time.h>
--- eom-1.16.0/src/eom-exif-util.c.orig	Sun May  7 11:03:42 2017
+++ eom-1.16.0/src/eom-exif-util.c	Sun May  7 11:05:38 2017
@@ -29,9 +29,11 @@
 #endif
 
 #ifdef HAVE_STRPTIME
+#ifndef __sun
 #define _XOPEN_SOURCE
 #define _XOPEN_SOURCE_EXTENDED 1
 #endif
+#endif
 #include <time.h>
 
 #include "eom-exif-util.h"
