--- configure.orig	2012-05-29 06:47:27.467999971 -0700
+++ configure	2012-05-29 06:59:52.994079022 -0700
@@ -2708,7 +2708,7 @@
 
 
 
-ac_config_headers="$ac_config_headers lib/curl_config.h include/curl/curlbuild.h"
+ac_config_headers="$ac_config_headers lib/curl_config.h"
 
 :
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to enable maintainer-specific portions of Makefiles" >&5
@@ -3207,12 +3207,6 @@
 fi
 
 
-
-
-if test -f ${srcdir}/include/curl/curlbuild.h; then
-  rm -f ${srcdir}/include/curl/curlbuild.h
-fi
-
 CURLVERSION=`$SED -ne 's/^#define LIBCURL_VERSION "\(.*\)"/\1/p' ${srcdir}/include/curl/curlver.h`
 am__api_version="1.9"
 ac_aux_dir=
@@ -20377,15 +20371,15 @@
            LIBS="$LIBS $gss_libs"
         elif test "$GSSAPI_ROOT" != "yes"; then
            LDFLAGS="$LDFLAGS -L$GSSAPI_ROOT/lib$libsuff"
-           LIBS="$LIBS -lgssapi"
+           LIBS="$LIBS -lgss"
         else
-           LIBS="$LIBS -lgssapi"
+           LIBS="$LIBS -lgss"
         fi
         ;;
      esac
   else
      LDFLAGS="$LDFLAGS $GSSAPI_LIB_DIR"
-     LIBS="$LIBS -lgssapi"
+     LIBS="$LIBS -lgss"
   fi
 else
   CPPFLAGS="$save_CPPFLAGS"
@@ -20411,7 +20405,7 @@
         if test x$cross_compiling != xyes; then
             PKGTEST="yes"
     fi
-    PREFIX_OPENSSL=/usr/local/ssl
+    PREFIX_OPENSSL="/usr"
     LIB_OPENSSL="$PREFIX_OPENSSL/lib$libsuff"
     ;;
   off)
@@ -25375,20 +25369,13 @@
 
 
 
-curl_preprocess_callconv="\
-/* preprocess start */
-#ifdef HAVE_WINDOWS_H
-#  define FUNCALLCONV __stdcall
-#else
-#  define FUNCALLCONV
-#endif
-/* preprocess end */"
+curl_preprocess_callconv=""
 
 
         #
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for curl_socklen_t data type" >&5
 $as_echo_n "checking for curl_socklen_t data type... " >&6; }
-  curl_typeof_curl_socklen_t="unknown"
+  curl_typeof_curl_socklen_t="socklen_t"
   for arg1 in int SOCKET; do
     for arg2 in 'struct sockaddr' void; do
       for t in socklen_t int size_t 'unsigned int' long 'unsigned long' void; do
@@ -39129,7 +39116,6 @@
 do
   case $ac_config_target in
     "lib/curl_config.h") CONFIG_HEADERS="$CONFIG_HEADERS lib/curl_config.h" ;;
-    "include/curl/curlbuild.h") CONFIG_HEADERS="$CONFIG_HEADERS include/curl/curlbuild.h" ;;
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
