--- lib/Makefile.in	2010-10-01 13:49:18.000000000 -0700
+++ lib/Makefile.in	2010-10-18 08:05:00.803406622 -0700
@@ -364,7 +364,7 @@
 @USE_EMBEDDED_ARES_TRUE@           -I$(top_builddir)/ares         \
 @USE_EMBEDDED_ARES_TRUE@           -I$(top_srcdir)/ares
 
-@SONAME_BUMP_FALSE@VERSIONINFO = -version-info 6:0:2
+@SONAME_BUMP_FALSE@VERSIONINFO = -version-info 3:0:0
 
 #
 # Bumping of SONAME conditionally may seem like a weird thing to do, and yeah
@@ -375,7 +375,7 @@
 #
 # This conditional soname bump SHOULD be removed at next "proper" bump.
 #
-@SONAME_BUMP_TRUE@VERSIONINFO = -version-info 7:0:2
+@SONAME_BUMP_TRUE@VERSIONINFO = -version-info 3:0:0
 
 # This flag accepts an argument of the form current[:revision[:age]]. So,
 # passing -version-info 3:12:1 sets current to 3, revision to 12, and age to
