--- lib/http_negotiate.c	2010-09-18 14:00:21.000000000 -0700
+++ lib/http_negotiate.c	2011-07-05 08:16:14.333605202 -0700
@@ -242,7 +242,7 @@
                                       &neg_ctx->context,
                                       neg_ctx->server_name,
                                       GSS_C_NO_OID,
-                                      GSS_C_DELEG_FLAG,
+                                      0,
                                       0,
                                       GSS_C_NO_CHANNEL_BINDINGS,
                                       &input_token,
