--- tk8.5.9/unix/Makefile.in.~1~	2010-09-02 11:12:14.000000000 -0700
+++ tk8.5.9/unix/Makefile.in	2011-04-17 15:41:11.333992473 -0700
@@ -80,11 +80,11 @@
 
 # Directory in which to install manual entries for Tk's C library
 # procedures:
-MAN3_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man3
+MAN3_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man3tk
 
 # Directory in which to install manual entries for the built-in
 # Tcl commands implemented by Tk:
-MANN_INSTALL_DIR	= $(MAN_INSTALL_DIR)/mann
+MANN_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man1t
 
 # Path to the html documentation dir:
 HTML_DIR		= @HTML_DIR@
