--- tk8.5.9/doc/label.n.orig	Tue Dec 20 08:29:11 2011
+++ tk8.5.9/doc/label.n	Tue Dec 20 08:34:08 2011
@@ -114,7 +114,7 @@
 \fBlabel\fR .l \-text "Left\enHand\enSide"
 \fBlabel\fR .r \-text "Right\enHand\enSide"
 text .mid
-\.mid insert end "This layout is like Java's BorderLayout"
+\&\.mid insert end "This layout is like Java's BorderLayout"
 # Lay them out
 pack .t   \-side top    \-fill x
 pack .b   \-side bottom \-fill x
--- tk8.5.9/doc/pack.n.orig	Tue Dec 20 08:28:56 2011
+++ tk8.5.9/doc/pack.n	Tue Dec 20 08:28:00 2011
@@ -268,7 +268,7 @@
 label .l \-text "Left\enHand\enSide"
 label .r \-text "Right\enHand\enSide"
 text .mid
-\.mid insert end "This layout is like Java's BorderLayout"
+\&\.mid insert end "This layout is like Java's BorderLayout"
 # Lay them out
 \fBpack\fR .t   \-side top    \-fill x
 \fBpack\fR .b   \-side bottom \-fill x
--- tk8.5.9/doc/text.n.orig	Tue Dec 20 08:29:19 2011
+++ tk8.5.9/doc/text.n	Tue Dec 20 08:34:42 2011
@@ -2294,8 +2294,8 @@
 a rare problem, but it can occur, for example:
 .CS
 pack [text .t]
-\.t insert 1.0 "aaaa\enbbbb\encccc\enbbbb\enaaaa\en"
-\.t search \-regexp \-\- {(a+|b+\enc+\enb+)+\ena+} 1.0
+\&\.t insert 1.0 "aaaa\enbbbb\encccc\enbbbb\enaaaa\en"
+\&\.t search \-regexp \-\- {(a+|b+\enc+\enb+)+\ena+} 1.0
 .CE
 will not find a match when one exists of 19
 characters starting from the first
@@ -2312,8 +2312,8 @@
 For example:
 .CS
 pack [text .t]
-\.t insert 1.0 "aaaa\enbbbb\enbbbb\enbbbb\enbbbb\\n"
-\.t search \-regexp \-backward \-\- {b+\en|a+\en(b+\en)+} end
+\&\.t insert 1.0 "aaaa\enbbbb\enbbbb\enbbbb\enbbbb\\n"
+\&\.t search \-regexp \-backward \-\- {b+\en|a+\en(b+\en)+} end
 .CE
 matches at
 .QW 5.0
--- tk8.5.9/doc/ttk_notebook.n.orig	Tue Dec 20 08:29:25 2011
+++ tk8.5.9/doc/ttk_notebook.n	Tue Dec 20 08:35:01 2011
@@ -201,9 +201,9 @@
 .SH "EXAMPLE"
 .CS
 pack [\fBttk::notebook\fR .nb]
-\.nb add [frame .nb.f1] \-text "First tab"
-\.nb add [frame .nb.f2] \-text "Second tab"
-\.nb select .nb.f2
+\&\.nb add [frame .nb.f1] \-text "First tab"
+\&\.nb add [frame .nb.f2] \-text "Second tab"
+\&\.nb select .nb.f2
 ttk::notebook::enableTraversal .nb
 .CE
 .SH "SEE ALSO"
