We do not have this code coverage tool available.

--- pyproject_fmt_rust-1.1.6/tox.ini.orig
+++ pyproject_fmt_rust-1.1.6/tox.ini
@@ -23,7 +23,6 @@
     COVERAGE_PROCESS_START = {toxinidir}/setup.cfg
     _COVERAGE_SRC = {envsitepackagesdir}/sphinx_argparse_cli
 commands =
-    cargo tarpaulin --no-default-features --out Html --output-dir {env_dir}
     python -m pytest {tty:--color=yes} {posargs: \
       --junitxml {toxworkdir}{/}junit.{envname}.xml --cov {envsitepackagesdir}{/}pyproject_fmt_rust \
       --cov {toxinidir}{/}tests --cov-fail-under=100 \
