--- autogen-5.18.16/autoopts/mk-autoopts-pc.in	2018-08-26 19:44:39.000000000 +0000
+++ autogen-5.18.16/autoopts/mk-autoopts-pc.in.new	2019-08-23 12:50:08.677533797 +0000
@@ -31,7 +31,7 @@
       cflags="-I${includedir}"
 test 'X@ENABLE_STATIC@' = Xno && static_libs=''
 case "${libdir}" in
-/lib | /lib64 | /usr/lib | /usr/lib64 | /usr/lib/* )
+/lib | /lib64 | /usr/lib | /usr/lib64 | /usr/lib/amd64 | /usr/lib/* )
     ldopts=''
     ldflags=-lopts
     ;;
