Index: autogen-5.18.16/agen5/Makefile.am
===================================================================
--- autogen-5.18.16.orig/agen5/Makefile.am
+++ autogen-5.18.16/agen5/Makefile.am
@@ -63,7 +63,7 @@ SNV_LIB         = $(top_builddir)/snprin
 nodist_autogen_SOURCES = ag.c
 autogen_SOURCES = $(gen_csrc)
 autogen_LDADD   = $(LO_LIB) $(SNV_LIB) $(GUILE_LIBS)
-autogen_LDFLAGS = $(DYNAMIC_AG) $(AG_STATIC_AUTOGEN) -no-install
+autogen_LDFLAGS = $(DYNAMIC_AG) $(AG_STATIC_AUTOGEN)
 autogen_CFLAGS  = $(GUILE_CFLAGS)
 stamp_script    = $(srcdir)/mk-stamps.sh
 
Index: autogen-5.18.16/agen5/Makefile.in
===================================================================
--- autogen-5.18.16.orig/agen5/Makefile.in
+++ autogen-5.18.16/agen5/Makefile.in
@@ -482,7 +482,7 @@ SNV_LIB = $(top_builddir)/snprintfv/libs
 nodist_autogen_SOURCES = ag.c
 autogen_SOURCES = $(gen_csrc)
 autogen_LDADD = $(LO_LIB) $(SNV_LIB) $(GUILE_LIBS)
-autogen_LDFLAGS = $(DYNAMIC_AG) $(AG_STATIC_AUTOGEN) -no-install
+autogen_LDFLAGS = $(DYNAMIC_AG) $(AG_STATIC_AUTOGEN)
 autogen_CFLAGS = $(GUILE_CFLAGS)
 stamp_script = $(srcdir)/mk-stamps.sh
 AM_YFLAGS = -d
Index: autogen-5.18.16/columns/Makefile.am
===================================================================
--- autogen-5.18.16.orig/columns/Makefile.am
+++ autogen-5.18.16/columns/Makefile.am
@@ -20,7 +20,6 @@
 ## with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 bin_PROGRAMS    = columns
-columns_LDFLAGS = -no-install
 csrc            = opts.h columns.c opts.c
 nodist_columns_SOURCES = cols.c
 
Index: autogen-5.18.16/columns/Makefile.in
===================================================================
--- autogen-5.18.16.orig/columns/Makefile.in
+++ autogen-5.18.16/columns/Makefile.in
@@ -128,7 +128,7 @@ am__v_lt_0 = --silent
 am__v_lt_1 = 
 columns_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(columns_LDFLAGS) $(LDFLAGS) -o $@
+	$(LDFLAGS) -o $@
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -389,7 +389,6 @@ target_vendor = @target_vendor@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-columns_LDFLAGS = -no-install
 csrc = opts.h columns.c opts.c
 nodist_columns_SOURCES = cols.c
 LO_LIB = $(top_builddir)/autoopts/libopts.la
Index: autogen-5.18.16/getdefs/Makefile.am
===================================================================
--- autogen-5.18.16.orig/getdefs/Makefile.am
+++ autogen-5.18.16/getdefs/Makefile.am
@@ -23,7 +23,6 @@
 TARG = getdefs
 
 bin_PROGRAMS    = getdefs
-getdefs_LDFLAGS = -no-install
 gdsrcs          = getdefs.h proto.h gdemit.c gdinit.c getdefs.c
 getdefs_SOURCES = proto.h
 BUILT_SOURCES   = gd.c
Index: autogen-5.18.16/getdefs/Makefile.in
===================================================================
--- autogen-5.18.16.orig/getdefs/Makefile.in
+++ autogen-5.18.16/getdefs/Makefile.in
@@ -130,7 +130,7 @@ am__v_lt_0 = --silent
 am__v_lt_1 = 
 getdefs_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(getdefs_LDFLAGS) $(LDFLAGS) -o $@
+	$(LDFLAGS) -o $@
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -434,7 +434,6 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 TARG = getdefs
-getdefs_LDFLAGS = -no-install
 gdsrcs = getdefs.h proto.h gdemit.c gdinit.c getdefs.c
 getdefs_SOURCES = proto.h
 BUILT_SOURCES = gd.c
