--- autogen-5.18.16/autoopts/tpl/texi2man.sh.orig	2022-07-15 14:16:01.045961975 +0000
+++ autogen-5.18.16/autoopts/tpl/texi2man.sh	2022-07-15 14:16:54.567772628 +0000
@@ -31,7 +31,7 @@
 ##
 ## And run the entire output through "sed" to convert texi-isms
 
-nl='
+newline='
 '
 sedcmd=
 bracket='{\([^}]*\)}'
@@ -40,12 +40,12 @@
 
 for f in code command var env dvn samp option strong
 do
-    sedcmd="${sedcmd}s;@${f}${bracket};${replB};g${nl}"
+    sedcmd="${sedcmd}s;@${f}${bracket};${replB};g${newline}"
 done
 
 for f in i file emph kbd key abbr acronym email indicateurl
 do
-    sedcmd="${sedcmd}s;@${f}${bracket};${replI};g${nl}"
+    sedcmd="${sedcmd}s;@${f}${bracket};${replI};g${newline}"
 done
 
 sed \
--- autogen-5.18.16/autoopts/tpl/texi2mdoc.sh.orig	2022-07-15 14:15:50.486064487 +0000
+++ autogen-5.18.16/autoopts/tpl/texi2mdoc.sh	2022-07-15 14:16:27.801098197 +0000
@@ -151,7 +151,7 @@
 }
 
 
-nl='
+newline='
 '
 sedcmd=
 bracket='{\([^}]*\)}'
@@ -160,12 +160,12 @@
 
 for f in code command var env dvn samp option strong
 do
-    sedcmd="${sedcmd}s;@${f}${bracket};${replB};g${nl}"
+    sedcmd="${sedcmd}s;@${f}${bracket};${replB};g${newline}"
 done
 
 for f in i file emph kbd key abbr acronym email indicateurl
 do
-    sedcmd="${sedcmd}s;@${f}${bracket};${replI};g${nl}"
+    sedcmd="${sedcmd}s;@${f}${bracket};${replI};g${newline}"
 done
 
 fixfont="${sedcmd}"'
