--- autogen-5.16.2/autoopts/mk-autoopts-pc.in.1	2016-08-23 20:44:24.696042196 +0300
+++ autogen-5.16.2/autoopts/mk-autoopts-pc.in	2016-08-23 20:45:12.591276387 +0300
@@ -31,7 +31,7 @@
       cflags="-I${includedir}"
 test 'X@ENABLE_STATIC@' = Xno && static_libs=''
 case "${libdir}" in
-/lib | /lib64 | /usr/lib | /usr/lib64 )
+/lib | /lib64 | /usr/lib | /usr/lib64 | /usr/lib/amd64)
     ldopts=''
     ldflags=-lopts
     ;;
