https://github.com/radiator-software/p5-net-ssleay/pull/489

--- Net-SSLeay-1.94/Makefile.PL.orig
+++ Net-SSLeay-1.94/Makefile.PL
@@ -244,7 +244,7 @@
     # phase fails.
     my @try_lib_paths = (
 	["$prefix/lib64", "$prefix/lib", "$prefix/out32dll", $prefix] => sub {$OSNAME eq 'darwin' },
-	[$prefix, "$prefix/lib64", "$prefix/lib", "$prefix/out32dll"] => sub { 1 },
+	[$prefix, "$prefix/lib64", "$prefix/lib/64", "$prefix/lib", "$prefix/out32dll"] => sub { 1 },
 	);
 
     while (
