Allow building in zones with shared IP stack
--- nmap-6.25/libdnet-stripped/config/acinclude.m4	2013-07-27 13:36:57.396713132 +0400
+++ nmap-6.25/libdnet-stripped/config/acinclude.m4	2013-07-27 13:37:29.025349176 +0400
@@ -162,7 +162,7 @@
 AC_DEFUN(AC_DNET_STREAMS_MIB2,
     [AC_MSG_CHECKING(for SNMP MIB2 STREAMS)
     AC_CACHE_VAL(ac_cv_dnet_streams_mib2,
-        if test -f /usr/include/inet/mib2.h -a -c /dev/ip ; then
+        if test -f /usr/include/inet/mib2.h ; then
             ac_cv_dnet_streams_mib2=yes
         else
             ac_cv_dnet_streams_mib2=no
