--- nmap-5.21/nbase/Makefile.in	2009-03-25 05:49:32.000000000 -0700
+++ nmap-5.21/nbase/Makefile.in	2010-02-23 10:25:33.136566614 -0800
@@ -6,7 +6,7 @@
 srcdir = @srcdir@
 
 CC = @CC@
-AR = ar
+CXX = @CXX@
 RANLIB = @RANLIB@
 CCOPT = 
 DEFS = @DEFS@
@@ -14,12 +14,14 @@
 DEFS += -D_FORTIFY_SOURCE=2
 CPPFLAGS = @CPPFLAGS@
 CFLAGS = @CFLAGS@ $(CCOPT) $(GLIB_CFLAGS) $(DEFS) $(INCLS)
+CXXFLAGS = @CXXFLAGS@
 STATIC = 
 LDFLAGS = @LDFLAGS@ $(STATIC)
 LIBS =  @LIBS@ 
 SHTOOL = ./shtool
 INSTALL = $(SHTOOL) install 
 MAKEDEPEND = @MAKEDEPEND@
+AR = /usr/bin/ar cq
 
 TARGET = libnbase.a
 
@@ -30,7 +32,7 @@
 
 $(TARGET): $(DEPS) $(OBJS)
 	rm -f $@
-	$(AR) cr $@ $(OBJS)
+	$(AR) $@ $(OBJS)
 	$(RANLIB) $@
 
 clean:
@@ -51,8 +53,10 @@
 config.status: configure
 	./config.status --recheck
 
+.c.o:
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 .cc.o:
-	$(CC) -c $(CFLAGS) $*.cc
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< o $@
 
 # DO NOT DELETE -- Needed by makedepend
 
