--- nmap-5.21/ncat/Makefile.in.orig	2011-11-01 11:54:32.150993236 -0700
+++ nmap-5.21/ncat/Makefile.in	2011-11-01 11:56:09.158430172 -0700
@@ -126,11 +126,11 @@
 
 .PHONY: uninstall all clean distclean
 
-$(NBASEDIR)/libnbase.a: $(NBASEDIR)/Makefile
+$(NBASEDIR)/libnbase.a:
 	@echo Compiling libnbase;
 	cd $(NBASEDIR) && $(MAKE)
 
-$(NSOCKDIR)/libnsock.a: $(NSOCKDIR)/Makefile
+$(NSOCKDIR)/libnsock.a:
 	@echo Compiling libnsock;
 	cd $(NSOCKDIR) && $(MAKE)
 
@@ -172,6 +172,6 @@
 .NOEXPORT:
 
 makefile.dep:
-	$(CC) -MM $(CPPFLAGS) $(SRCS) > $@
+	$(CC) $(CPPFLAGS) $(CFLAGS) -MM $(SRCS) > $@
 include makefile.dep
 
