--- nmap-5.21/zenmap/setup.py	2010-01-26 16:57:31.000000000 -0800
+++ nmap-5.21/zenmap/setup.py	2010-02-24 10:12:10.652841785 -0800
@@ -324,7 +324,6 @@
             if not re_sys.match(line):
                 break
             uline += 1
-        ucontent.insert(uline, "\nimport sys\nsys.path.append(%s)\n" % repr(modules))
 
         ufile = open(app_file_name, "w")
         ufile.writelines(ucontent)
