--- gcc-5.4.0/gcc/config.gcc.~1~	2015-09-10 17:17:53.000000000 +0000
+++ gcc-5.4.0/gcc/config.gcc	2017-08-31 00:19:51.521347790 +0000
@@ -820,7 +820,7 @@
   sol2_tm_file_tail="${cpu_type}/sol2.h sol2.h"
   sol2_tm_file="${sol2_tm_file_head} ${sol2_tm_file_tail}"
   case ${target} in
-    *-*-solaris2.1[2-9]*)
+    *-*-solaris2.1[1-9]*)
       # __cxa_atexit was introduced in Solaris 12.
       default_use_cxa_atexit=yes
       ;;
