--- factory_boy-3.3.0/Makefile.orig
+++ factory_boy-3.3.0/Makefile
@@ -64,7 +64,7 @@
 		-Wdefault:"set_output_charset() is deprecated":DeprecationWarning:: \
 		-Wdefault:"parameter codeset is deprecated":DeprecationWarning:: \
 		-Wdefault:"'cgi' is deprecated and slated for removal in Python 3.13":DeprecationWarning:: \
-		-m unittest
+		-m unittest -v
 
 # DOC: Test the examples
 example-test:
