--- ncurses-5.9/configure.orig	2013-06-24 22:55:46.932166210 +0200
+++ ncurses-5.9/configure	2013-06-24 22:57:02.209637590 +0200
@@ -7096,7 +7096,7 @@
 	# setting _XOPEN_SOURCE breaks Lynx on SCO Unix / OpenServer
 	;;
 solaris2.1[0-9]) #(vi
-	cf_xopen_source="-D__EXTENSIONS__ -D_XOPEN_SOURCE=$cf_XOPEN_SOURCE"
+	cf_xopen_source="-D__EXTENSIONS__"
 	;;
 solaris2.[1-9]) #(vi
 	cf_xopen_source="-D__EXTENSIONS__"
