Comment from ncurses.spec file:
date:2009-06-06 owner:jouby type:branding bugster:6754653

--- ncurses-5.7/configure.orig	Mon Jul 20 14:54:49 2009
+++ ncurses-5.7/configure	Mon Jul 20 14:57:21 2009
@@ -5242,7 +5242,7 @@
 		# tested with SunOS 5.5.1 (solaris 2.5.1) and gcc 2.7.2
 		# tested with SunOS 5.10 (solaris 10) and gcc 3.4.3
 		if test "$DFT_LWR_MODEL" = "shared" ; then
-			LOCAL_LDFLAGS="-R \$(LOCAL_LIBDIR):\${libdir}"
+			LOCAL_LDFLAGS="-R \${libdir}"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
 		fi
 		if test "$cf_cv_ld_rpath" = yes ; then
