Comment from Desktop ncurses.spec file:
date:2008-09-12 owner:jouby type:branding

--- ncurses-5.7/configure.orig	Mon Jul 20 14:35:32 2009
+++ ncurses-5.7/configure	Mon Jul 20 14:37:18 2009
@@ -6824,7 +6824,7 @@
 echo "$as_me:6824: result: $with_widec" >&5
 echo "${ECHO_T}$with_widec" >&6
 if test "$with_widec" = yes ; then
-	LIB_SUFFIX="w${LIB_SUFFIX}"
+	#LIB_SUFFIX="w${LIB_SUFFIX}"
 	cat >>confdefs.h <<\EOF
 #define USE_WIDEC_SUPPORT 1
 EOF
