Coverage fails so do not do it.

--- yarl-1.15.2/pytest.ini.orig
+++ yarl-1.15.2/pytest.ini
@@ -19,14 +19,6 @@
   # Dump the test results in junit format:
   # --junitxml=.test-results/pytest/results.xml
 
-  # Pre-load the `pytest-cov` plugin early:
-  -p pytest_cov
-
-  # `pytest-cov`:
-  --cov
-  --cov-context=test
-  --cov-config=.coveragerc
-
   # Fail on config parsing warnings:
   # --strict-config
 
