--- node-v0.12.15/deps/openssl/openssl/crypto/rand/randfile.c.1	2016-06-23 18:46:33.000000000 +0200
+++ node-v0.12.15/deps/openssl/openssl/crypto/rand/randfile.c	2016-07-10 16:05:53.866250571 +0200
@@ -57,7 +57,7 @@
  */
 
 /* We need to define this to get macros like S_IFBLK and S_IFCHR */
-#if !defined(OPENSSL_SYS_VXWORKS)
+#if !defined(OPENSSL_SYS_VXWORKS) && !defined(__sun__)
 # define _XOPEN_SOURCE 500
 #endif
 
