This should be sent upstream to add Solaris

--- a/configure	2018-04-18 21:08:53.994022527 -0700
+++ b/configure	2018-04-18 21:16:29.261782547 -0700
@@ -518,7 +518,7 @@
         # here rather than at option parse time because the target auto-detect
         # magic happens after the command line has been parsed.
         case "${tgt_os}" in
-        linux|os2|darwin*|iphonesimulator*)
+        linux|os2|solaris|darwin*|iphonesimulator*)
             # Supported platforms
             ;;
         *)
