--- libvpx-1.6.1/configure.orig	2017-06-09 19:27:10.353541785 +0000
+++ libvpx-1.6.1/configure	2017-06-09 19:28:03.679168380 +0000
@@ -496,7 +496,7 @@
         # here rather than at option parse time because the target auto-detect
         # magic happens after the command line has been parsed.
         case "${tgt_os}" in
-        linux|os2|darwin*|iphonesimulator*)
+        linux|os2|darwin*|solaris|iphonesimulator*)
             # Supported platforms
             ;;
         *)
