https://github.com/yt-dlp/yt-dlp/pull/9906

--- yt_dlp-2024.4.9/test/test_networking.py.orig
+++ yt_dlp-2024.4.9/test/test_networking.py
@@ -550,7 +550,7 @@
             with pytest.raises(TransportError):
                 validate_and_send(
                     rh, Request(connect_timeout_url))
-            assert 0.01 <= time.time() - now < 20
+            assert time.time() - now < 20
 
         with handler() as rh:
             with pytest.raises(TransportError):
@@ -558,7 +558,7 @@
                 now = time.time()
                 validate_and_send(
                     rh, Request(connect_timeout_url, extensions={'timeout': 0.01}))
-                assert 0.01 <= time.time() - now < 20
+                assert time.time() - now < 20
 
     @pytest.mark.parametrize('handler', ['Urllib', 'Requests', 'CurlCFFI'], indirect=True)
     def test_source_address(self, handler):
