--- ruby-2.6.3/ext/fiddle/closure.c.1	2019-06-28 16:57:01.417889071 +0000
+++ ruby-2.6.3/ext/fiddle/closure.c	2019-06-28 16:57:34.126439943 +0000
@@ -14,7 +14,7 @@
 } fiddle_closure;
 
 #if defined(USE_FFI_CLOSURE_ALLOC)
-#elif defined(__OpenBSD__) || defined(__APPLE__) || defined(__linux__)
+#elif defined(__OpenBSD__) || defined(__APPLE__) || defined(__linux__) || defined(__sun__) || defined (__sun)
 # define USE_FFI_CLOSURE_ALLOC 0
 #elif defined(RUBY_LIBFFI_MODVERSION) && RUBY_LIBFFI_MODVERSION < 3000005 && \
 	(defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_AMD64))
