--- vinagre-3.22.0/Makefile.am.1	2017-12-01 14:08:11.889674381 +0000
+++ vinagre-3.22.0/Makefile.am	2017-12-01 14:08:35.518231413 +0000
@@ -258,10 +258,6 @@
 mimedir = $(datadir)/mime/packages
 nodist_mime_DATA = data/vinagre-mime.xml
 
-@APPSTREAM_XML_RULES@
-appstream_in_files = data/vinagre.appdata.xml.in
-appstream_XML = $(appstream_in_files:.appdata.xml.in=.appdata.xml)
-
 gsettings_SCHEMAS = data/org.gnome.Vinagre.gschema.xml
 
 @GSETTINGS_RULES@
