We have no pre-commit packaged.

--- re-assert-1.1.0/requirements-dev.txt.orig
+++ re-assert-1.1.0/requirements-dev.txt
@@ -1,3 +1,2 @@
 coverage
-pre-commit
 pytest
--- re-assert-1.1.0/tox.ini.orig
+++ re-assert-1.1.0/tox.ini
@@ -7,7 +7,6 @@
     coverage erase
     coverage run -m pytest {posargs:tests}
     coverage report --fail-under 100
-    pre-commit install
 
 [testenv:pre-commit]
 skip_install = true
