diff -ruN ntfs-3g_ntfsprogs-2016.2.22AR.1.orig/libntfs-3g/dir.c ntfs-3g_ntfsprogs-2016.2.22AR.1/libntfs-3g/dir.c
--- ntfs-3g_ntfsprogs-2016.2.22AR.1.orig/libntfs-3g/dir.c	2016-04-01 08:37:45.000000000 +0200
+++ ntfs-3g_ntfsprogs-2016.2.22AR.1/libntfs-3g/dir.c	2016-05-04 18:16:08.517144523 +0200
@@ -40,9 +40,15 @@
 #include <sys/stat.h>
 #endif
 
+#if defined(__sun) && defined (__SVR4)
+#ifdef HAVE_SYS_MKDEV_H
+#include <sys/mkdev.h>
+#endif
+#else /* defined(__sun) && defined (__SVR4) */
 #ifdef HAVE_SYS_SYSMACROS_H
 #include <sys/sysmacros.h>
 #endif
+#endif /* defined(__sun) && defined (__SVR4) */
 
 #include "param.h"
 #include "types.h"
