--- ntfs-3g_ntfsprogs-2015.3.14/src/usermap.c.~1~	2015-03-14 17:10:12.000000000 +0300
+++ ntfs-3g_ntfsprogs-2015.3.14/src/usermap.c	2015-09-03 14:43:20.076520339 +0300
@@ -79,8 +79,8 @@
 #else
 #define USESTUBS 0 /* direct calls to API, based on following definitions */
 #define ENVNTFS3G "NTFS3G"
-#define LIBFILE64 "/lib64/libntfs-3g.so.491"
-#define LIBFILE "/lib/libntfs-3g.so.491"
+#define LIBFILE64 "/usr/lib/64/libntfs-3g.so.491"
+#define LIBFILE "/usr/lib/libntfs-3g.so.491"
 #endif
 
 #define GET_FILE_SECURITY "ntfs_get_file_security"
