--- a/tools/Makefile.am.bk	2011-05-11 17:06:41.263948182 +0800
+++ b/tools/Makefile.am	2011-05-11 17:06:57.885923460 +0800
@@ -30,11 +30,6 @@
 	fi
 	${INSTALL} -d -m 1770 $(DESTDIR)$(localstatedir)/lib/stunnel
 	-chgrp $(DEFAULT_GROUP) $(DESTDIR)$(localstatedir)/lib/stunnel
-	if uname | grep SunOS; then \
-		${INSTALL} -d -m 755 $(DESTDIR)$(localstatedir)/lib/stunnel/dev; \
-		mknod $(DESTDIR)$(localstatedir)/lib/stunnel/dev/zero c 13 12; \
-		chmod 666 $(DESTDIR)$(localstatedir)/lib/stunnel/dev/zero; \
-	fi
 
 clean-local:
 	-rm -f stunnel.rnd
