Allow builds with clang

--- a/make/autoconf/toolchain.m4	Tue Apr  1 14:13:45 2025
+++ b/make/autoconf/toolchain.m4	Wed Apr  2 12:02:47 2025
@@ -39,7 +39,7 @@
 
 # These toolchains are valid on different platforms
 VALID_TOOLCHAINS_linux="gcc clang"
-VALID_TOOLCHAINS_solaris="gcc"
+VALID_TOOLCHAINS_solaris="gcc clang"
 VALID_TOOLCHAINS_macosx="clang"
 VALID_TOOLCHAINS_aix="clang"
 VALID_TOOLCHAINS_windows="microsoft"
