--- gc-7.6.0/configure.ac.orig	2017-06-20 22:12:50.372904107 +0000
+++ gc-7.6.0/configure.ac	2017-06-20 22:14:44.338388609 +0000
@@ -254,11 +254,7 @@
           AC_DEFINE(PARALLEL_MARK)
         fi
         AC_DEFINE(THREAD_LOCAL_ALLOC)
-        # Need to use alternate thread library, otherwise gctest hangs
-        # on Solaris 8.
-        multi_os_directory=`$CC -print-multi-os-directory`
-        THREADDLLIBS="-L/usr/lib/lwp/$multi_os_directory \
-                      -R/usr/lib/lwp/$multi_os_directory -lpthread -lrt"
+        THREADDLLIBS="-lpthread -lrt"
         ;;
      *-*-irix*)
         AC_DEFINE(GC_IRIX_THREADS)
