--- ipmitool-1.8.17/lib/ipmi_main.c-orig	   :: 
+++ ipmitool-1.8.17/lib/ipmi_main.c	   :: 
@@ -29,7 +29,7 @@
  * LIABILITY, ARISING OUT OF THE USE OF OR INABILITY TO USE THIS SOFTWARE,
  * EVEN IF SUN HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
  */
-#define _XOPEN_SOURCE 700
+#define _XOPEN_SOURCE 600
 #define _BSD_SOURCE || \
 	(_XOPEN_SOURCE >= 500 || \
 	_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) && \
--- ipmitool-1.8.17/src/ipmievd.c-orig	   :: 
+++ ipmitool-1.8.17/src/ipmievd.c	   :: 
@@ -29,7 +29,7 @@
  * LIABILITY, ARISING OUT OF THE USE OF OR INABILITY TO USE THIS SOFTWARE,
  * EVEN IF SUN HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
  */
-#define _XOPEN_SOURCE 700
+#define _XOPEN_SOURCE 600
 #define _BSD_SOURCE
 
 #include <stdio.h>
