https://github.com/ipmitool/ipmitool/issues/377

--- a/Makefile.am.~1~	Thu Sep  1 20:42:31 2022
+++ a/Makefile.am	Tue Jul 30 17:21:51 2024
@@ -41,7 +41,7 @@
 	$(distdir).tar.gz $(distdir).tar.bz2
 
 SUBDIRS = lib src include doc contrib control
-IANA_PEN = http://www.iana.org/assignments/enterprise-numbers
+IANA_PEN = http://www.iana.org/assignments/enterprise-numbers/enterprise-numbers
 
 dist-hook:
 	cp control/ipmitool.spec $(distdir)
