https://github.com/pexpect/pexpect/issues/766

--- pexpect-4.8.0/tests/test_performance.py.orig
+++ pexpect-4.8.0/tests/test_performance.py
@@ -103,7 +103,7 @@
         print("100000 calls to faster_range:", (time.time() - start_time))
 
     def test_large_stdout_stream(self):
-        e = pexpect.spawn('openssl rand -base64 {}'.format(1024*1024*25), searchwindowsize=1000)
+        e = pexpect.spawn('openssl rand -base64 {}'.format(1024*1024*25), searchwindowsize=1000, timeout=40)
         resp = e.expect(['Password:', pexpect.EOF, pexpect.TIMEOUT])
         assert resp == 1  # index 1 == EOF
 
