--- ./src/runtime/runtime-gdb_test.go.orig	2016-04-28 15:00:59.194196644 +0200
+++ ./src/runtime/runtime-gdb_test.go	2016-04-28 15:01:27.275109628 +0200
@@ -14,7 +14,7 @@
 )
 
 func checkGdbPython(t *testing.T) {
-	cmd := exec.Command("gdb", "-nx", "-q", "--batch", "-iex", "python import sys; print('go gdb python support')")
+	cmd := exec.Command("gdb", "-nx", "-q", "--batch", "-iex", "/usr/bin/amd64/python import sys; print('go gdb python support')")
 	out, err := cmd.CombinedOutput()
 
 	if err != nil {
