Illumos has dlsym(), let meson detect it.

--- dav1d-1.5.0/meson.build.orig	2024-10-18 01:02:57.000000000 +0200
+++ dav1d-1.5.0/meson.build	2024-11-16 13:33:35.261561302 +0100
@@ -190,7 +190,7 @@
 
 libdl_dependency = []
 have_dlsym = false
-if host_machine.system() == 'linux'
+if host_machine.system() == 'linux' or host_machine.system() == 'sunos'
     libdl_dependency = cc.find_library('dl', required : false)
     have_dlsym = cc.has_function('dlsym', prefix : '#include <dlfcn.h>', args : test_args, dependencies : libdl_dependency)
 endif
