--- gammu-1.37.4/libgammu/device/usb/usb.c.1	2016-10-21 09:11:55.925264840 +0300
+++ gammu-1.37.4/libgammu/device/usb/usb.c	2016-10-21 09:14:25.891997596 +0300
@@ -43,18 +43,18 @@
  * Union of CDC descriptor.
  */
 struct cdc_union_desc {
-    u_int8_t      bLength;
-    u_int8_t      bDescriptorType;
-    u_int8_t      bDescriptorSubType;
+    uint8_t      bLength;
+    uint8_t      bDescriptorType;
+    uint8_t      bDescriptorSubType;
 
-    u_int8_t      bMasterInterface0;
-    u_int8_t      bSlaveInterface0;
+    uint8_t      bMasterInterface0;
+    uint8_t      bSlaveInterface0;
 } __attribute__ ((packed));
 
 struct cdc_extra_desc {
-    u_int8_t      bLength;
-    u_int8_t      bDescriptorType;
-    u_int8_t      bDescriptorSubType;
+    uint8_t      bLength;
+    uint8_t      bDescriptorType;
+    uint8_t      bDescriptorSubType;
 } __attribute__ ((packed));
 
 /* CDC header types (bDescriptorSubType) */
