--- ghostscript-9.25/Resource/Init/pdf_sec.ps	2018-09-13 12:02:01.000000000 +0000
+++ ghostscript-9.25/Resource/Init/pdf_sec.ps.new	2018-09-23 11:54:12.754858110 +0000
@@ -505,12 +505,12 @@ def
     } ifelse
   } ifelse
 
-%   Trailer /Encrypt oget /P oget 4 and 0 eq #? and
-%    { (   ****This owner of this file has requested you do not print it.\n)
-%      pdfformaterror printProducer
-%      /pdf_process_Encrypt cvx /invalidfileaccess signalerror
-%    }
-%   if
+  Trailer /Encrypt oget /P oget 4 and 0 eq #? and
+   { (   ****This owner of this file has requested you do not print it.\n)
+     pdfformaterror printProducer
+     /pdf_process_Encrypt cvx /invalidfileaccess signalerror
+   }
+  if
 } bind executeonly def
 
 % Calculate the key used to decrypt an object (to pass to .decpdfrun or
