--- ghostscript-8.64/base/configure.ac.orig	Fri Oct 15 02:02:36 2010
+++ ghostscript-8.64/base/configure.ac	Fri Oct 15 02:12:00 2010
@@ -1191,7 +1191,6 @@
 		SunOS)
 		DYNAMIC_DEVS="\$(GLOBJDIR)/X11.so"
 		DYNAMIC_FLAGS="-DGS_DEVS_SHARED -DGS_DEVS_SHARED_DIR=\\\"\$(gssharedir)\\\""
-		X_DEVS=""
 		;;
 		*)
 		AC_MSG_ERROR([Sorry, dynamic driver support not available on this platform!])
@@ -1233,6 +1232,10 @@
 	# These font directories are used by Solaris...
 	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type1"
 	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/TrueType"
+	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type3"
+	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type1"
+	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type3"
+	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/TrueType"
 
 	# This font directory is used by CUPS...
 	if test "x$CUPSCONFIG" != x; then
