From de930266a0bb47b6cb8e89a8eff2ae6f9fe40523 Mon Sep 17 00:00:00 2001
From: Josh Moyer <JMoyer@NODOMAIN.NET>
Date: Sun, 15 Feb 2026 05:11:55 -0800
Subject: [PATCH 3/3] gdevadmp-][ rev B: expose legacy driver suite and fix A4
 printing

---
 contrib/gdevadmp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/gdevadmp.c b/contrib/gdevadmp.c
index 435edaba20..fac47e36b5 100644
--- a/contrib/gdevadmp.c
+++ b/contrib/gdevadmp.c
@@ -77,7 +77,7 @@
  *   bugs or deficiencies.
  * + Changed devtype to dev_type for better consistency with dev_rez & etc.
  * 
- * + * Fixes in version ][ revision B (2.0.2):
+ * Fixes in version ][ revision B (2.0.2):
  * + Enable legacy (pre-10.06.0) driver compatibility mode by appending l to device names.
  * 
  * Known issues include:
-- 
2.52.0.windows.1

