--- ghostscript-9.00/Resource/Init/gs_init.ps.orig	Tue Sep 14 07:56:26 2010
+++ ghostscript-9.00/Resource/Init/gs_init.ps	Fri Oct 15 03:50:03 2010
@@ -2150,7 +2150,7 @@
 {
   SAFETY /safe get not {
     <<
-      /PermitFileReading [ ]
+      /PermitFileReading [ (/usr/openwin/lib/*) (/usr/X11/lib/*) (/usr/share/ghostscript/*) ]
       /PermitFileWriting [ ]
       /PermitFileControl [ ]
     >> setuserparams
