--- ghostscript-9.23/base/unixinst.mak.1	2018-05-04 22:34:21.902804453 +0000
+++ ghostscript-9.23/base/unixinst.mak	2018-05-04 22:34:56.664314825 +0000
@@ -165,6 +165,7 @@
 	$(SH) -c 'for f in $(DOC_PAGES) ;\
 	do if ( test -f $(PSDOCDIR)/$$f ); then $(INSTALL_DATA) $(PSDOCDIR)/$$f $(DESTDIR)$(docdir); fi;\
 	done'
+	rm -f $(DESTDIR)$(gsdatadir)/doc
 	ln -s $(DESTDIR)$(docdir) $(DESTDIR)$(gsdatadir)/doc
 
 # install the man pages for each locale
