*** ghostscript-8.64/base/unixinst.mak.orig	Tue Aug 12 14:58:30 2008
--- ghostscript-8.64/base/unixinst.mak	Wed Aug 13 11:47:54 2008
***************
*** 108,114 ****
  # install html documentation
  DOC_PAGES=PUBLIC README index.html gs.css \
  	   API.htm Bug-form.htm Bug-info.htm \
! 	   C-style.htm Changes.htm Commprod.htm Copying.htm \
  	   Current.htm Deprecated.htm \
  	   DLL.htm Deprecated.htm Develop.htm Devices.htm Drivers.htm \
  	   Fonts.htm Helpers.htm Hershey.htm \
--- 108,114 ----
  # install html documentation
  DOC_PAGES=PUBLIC README index.html gs.css \
  	   API.htm Bug-form.htm Bug-info.htm \
! 	   C-style.htm Changes.htm Commprod.htm Copying.htm COPYING README.sharedlibs \
  	   Current.htm Deprecated.htm \
  	   DLL.htm Deprecated.htm Develop.htm Devices.htm Drivers.htm \
  	   Fonts.htm Helpers.htm Hershey.htm \
***************
*** 119,126 ****
  	   Lib.htm Maintain.htm Make.htm New-user.htm \
  	   News.htm Projects.htm Ps-style.htm Ps2epsi.htm Ps2pdf.htm \
  	   Psfiles.htm Public.htm Readme.htm Release.htm \
! 	   Source.htm Testing.htm Unix-lpr.htm \
! 	   Use.htm Xfonts.htm
  install-doc: $(PSDOCDIR)/News.htm
  	-mkdir -p $(DESTDIR)$(docdir)
  	$(SH) -c 'for f in $(DOC_PAGES) ;\
--- 119,126 ----
  	   Lib.htm Maintain.htm Make.htm New-user.htm \
  	   News.htm Projects.htm Ps-style.htm Ps2epsi.htm Ps2pdf.htm \
  	   Psfiles.htm Public.htm Readme.htm Release.htm \
! 	   Deprecated.htm Details.htm Details8.htm gs-vms.hlp gsdoc.el \
! 	   Source.htm Tester.htm Testing.htm Unix-lpr.htm Use.htm Xfonts.htm
  install-doc: $(PSDOCDIR)/News.htm
  	-mkdir -p $(DESTDIR)$(docdir)
  	$(SH) -c 'for f in $(DOC_PAGES) ;\
