--- ghostscript-9.00/Resource/Init/cidfmap.orig	Wed Jun  2 01:37:07 2010
+++ ghostscript-9.00/Resource/Init/cidfmap	Fri Oct 15 02:20:40 2010
@@ -31,3 +31,133 @@
 %   /Ryumin-Medium << /FileType /TrueType /Path (H:/AuxFiles/Fonts/BATANG.TTC) /SubfontID 3 /CSI [(Japan1) 2] >> ;
 %
 
+%
+% Mapping for Asian Language Fonts delivered with Openwin.
+%
+
+%	Japanese Fonts
+%
+/HG-MinchoL <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/ja/X11/fonts/TT/hgmlsun.ttf)
+	/SubfontID 0
+	/CSI [(Japan1) 1] >> ;
+/Ryumin-Light		/HG-MinchoL	;
+/Adobe-Japan1		/HG-MinchoL	;
+/Adobe-Japan2		/HG-MinchoL	;
+
+/HG-GothicB <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/ja/X11/fonts/TT/hggbsun.ttf)
+	/SubfontID 0
+	/CSI [(Japan1) 1] >> ;
+/GothicBBB-Medium       /HG-GothicB	;
+
+%	CNS Fonts
+%
+/hei <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/hei.ttf)
+	/SubfontID 0
+	/CSI [(CNS1) 1] >> ;
+/MHei-Medium            /hei 	;
+
+/kei <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/kei.ttf)
+	/SubfontID 0
+	/CSI [(CNS1) 1] >> ;
+/MKei-Medium            /kei 	;
+
+/ming <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/ming.ttf)
+	/SubfontID 0
+	/CSI [(CNS1) 1] >> ;
+/MMing-Medium		/ming	;
+/Adobe-CNS1		/ming	;
+
+%	GB Fonts
+%
+/zycjkfangs <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/fangsongti.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+/STFangsong-Light	/zycjkfangs	;
+
+/zycjkhei <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/heiti.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+/STHeiti-Regular	/zycjkhei	;
+
+/zycjkkai <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/kaiti.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+/STKaiti-Regular	/zycjkkai	;
+
+/zycjksun <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/songti.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+/STSong-Light           /zycjksun	;
+/Adobe-GB1		/zycjksun	;
+
+/CFangSongGB-Light <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/cfsgbl.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+
+/MHeiGB-Bold <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/mhgbb.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+
+/MSungGB-Medium <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/mhgbm.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+
+/MSungGB-Light <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/mhgbl.ttf)
+	/SubfontID 0
+	/CSI [(GB1) 1] >> ;
+
+%	Korean Fonts
+%
+/Gothic <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/h2gtrm.ttf)
+	/SubfontID 0
+	/CSI [(Korea1) 1] >> ;
+/HYGoThic-Medium	/Gothic        	;
+
+/RoundedGothic <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/h2drrm.ttf)
+	/SubfontID 0
+	/CSI [(Korea1) 1] >> ;
+/HYRGoThic-Medium	/RoundedGothic	;
+
+/Haeseo <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/h2hsrm.ttf)
+	/SubfontID 0
+	/CSI [(Korea1) 1] >> ;
+
+/Myengjo <<
+	/FileType /TrueType
+	/Path (/usr/openwin/lib/locale/zh_TW.BIG5/X11/fonts/TT/h2mjsm.ttf)
+	/SubfontID 0
+	/CSI [(Korea1) 1] >> ;
+/Adobe-Korea1		/Myeongjo	;
+/HYSMyeongJo-Medium	/Myeongjo	;
