--- ghostscript-9.00/base/configure.ac.~1~	2014-04-24 11:40:19.621823448 +0400
+++ ghostscript-9.00/base/configure.ac	2014-04-24 11:45:07.848885201 +0400
@@ -1159,9 +1159,15 @@
 	DYNAMIC_LIBS=""
 	;;
 	SunOS)
-	DYNAMIC_CFLAGS="-KPIC"
-	DYNAMIC_LDFLAGS="-G"
-	DYNAMIC_LIBS=""
+	if test "$GCC" = yes; then
+    	  DYNAMIC_CFLAGS="-fPIC"
+          DYNAMIC_LDFLAGS="-fPIC -shared"
+          DYNAMIC_LIBS=""
+        else
+	  DYNAMIC_CFLAGS="-KPIC"
+	  DYNAMIC_LDFLAGS="-G"
+	  DYNAMIC_LIBS=""
+        fi
 	;;
 esac
 
