--- a/configure.~1~	2017-08-05 19:22:48.000000000 +0000
+++ b/configure	2017-08-11 10:06:36.599227732 +0000
@@ -4896,14 +4896,14 @@
   fi
 
   if test "$enable_all_static" = "yes"; then
-    SVN_APR_LIBS="`$apr_config --link-ld --libs`"
+    SVN_APR_LIBS="`$apr_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apr-config --link-ld --libs failed" "$LINENO" 5
+      as_fn_error $? "apr-config --link-libtool --libs failed" "$LINENO" 5
     fi
   else
-    SVN_APR_LIBS="`$apr_config --link-ld`"
+    SVN_APR_LIBS="`$apr_config --link-libtool`"
     if test $? -ne 0; then
-      as_fn_error $? "apr-config --link-ld failed" "$LINENO" 5
+      as_fn_error $? "apr-config --link-libtool failed" "$LINENO" 5
     fi
   fi
   SVN_APR_LIBS="`
@@ -5140,14 +5140,14 @@
   fi
 
   if test "$enable_all_static" = "yes"; then
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld --libs`"
+    SVN_APRUTIL_LIBS="`$apu_config --link-libtool --libs`"
     if test $? -ne 0; then
-      as_fn_error $? "apu-config --link-ld --libs failed" "$LINENO" 5
+      as_fn_error $? "apu-config --link-libtool --libs failed" "$LINENO" 5
     fi
   else
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld`"
+    SVN_APRUTIL_LIBS="`$apu_config --link-libtool`"
     if test $? -ne 0; then
-      as_fn_error $? "apu-config --link-ld failed" "$LINENO" 5
+      as_fn_error $? "apu-config --link-libtool failed" "$LINENO" 5
     fi
   fi
   SVN_APRUTIL_LIBS="`
