--- a/configure.1	2016-03-25 19:17:20.210122569 +0300
+++ b/configure	2016-03-25 19:18:36.742280283 +0300
@@ -21446,7 +21446,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERL_VERSION" >&5
 $as_echo "$PERL_VERSION" >&6; }
     if test "$PERL_VERSION" -ge "5008000"; then
-      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts`"
+      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts|sed -e 's:-fstack-protector-strong::'`"
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: perl bindings require perl 5.8.0 or newer." >&5
 $as_echo "$as_me: WARNING: perl bindings require perl 5.8.0 or newer." >&2;}
@@ -21911,7 +21911,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERL_VERSION" >&5
 $as_echo "$PERL_VERSION" >&6; }
     if test "$PERL_VERSION" -ge "5008000"; then
-      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts`"
+      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts|sed -e 's:-fstack-protector-strong::'`"
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: perl bindings require perl 5.8.0 or newer." >&5
 $as_echo "$as_me: WARNING: perl bindings require perl 5.8.0 or newer." >&2;}
@@ -22376,7 +22376,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERL_VERSION" >&5
 $as_echo "$PERL_VERSION" >&6; }
     if test "$PERL_VERSION" -ge "5008000"; then
-      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts`"
+      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts|sed -e 's:-fstack-protector-strong::'`"
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: perl bindings require perl 5.8.0 or newer." >&5
 $as_echo "$as_me: WARNING: perl bindings require perl 5.8.0 or newer." >&2;}
@@ -22844,7 +22844,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERL_VERSION" >&5
 $as_echo "$PERL_VERSION" >&6; }
     if test "$PERL_VERSION" -ge "5008000"; then
-      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts`"
+      SWIG_PL_INCLUDES="\$(SWIG_INCLUDES) `$PERL -MExtUtils::Embed -e ccopts|sed -e 's:-fstack-protector-strong::'`"
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: perl bindings require perl 5.8.0 or newer." >&5
 $as_echo "$as_me: WARNING: perl bindings require perl 5.8.0 or newer." >&2;}
