--- putty-0.74/cmdgen.c.orig	2020-06-21 19:30:43.000000000 +0000
+++ putty-0.74/cmdgen.c	2020-08-12 21:59:46.262592636 +0000
@@ -219,8 +219,8 @@
     Filename *infilename = NULL, *outfilename = NULL;
     enum { NOKEYGEN, RSA1, RSA2, DSA, ECDSA, ED25519 } keytype = NOKEYGEN;
     char *outfile = NULL, *outfiletmp = NULL;
-    enum { PRIVATE, PUBLIC, PUBLICO, FP, OPENSSH_AUTO,
-           OPENSSH_NEW, SSHCOM } outtype = PRIVATE;
+    enum { PPRIVATE, PUBLIC, PUBLICO, FP, OPENSSH_AUTO,
+           OPENSSH_NEW, SSHCOM } outtype = PPRIVATE;
     int bits = -1;
     const char *comment = NULL;
     char *origcomment = NULL;
@@ -432,7 +432,7 @@
                         else if (!strcmp(p, "public-openssh"))
                             outtype = PUBLICO;
                         else if (!strcmp(p, "private"))
-                            outtype = PRIVATE;
+                            outtype = PPRIVATE;
                         else if (!strcmp(p, "fingerprint"))
                             outtype = FP;
                         else if (!strcmp(p, "private-openssh"))
@@ -549,7 +549,7 @@
      * We must save the private part when generating a new key.
      */
     if (keytype != NOKEYGEN &&
-        (outtype != PRIVATE && outtype != OPENSSH_AUTO &&
+        (outtype != PPRIVATE && outtype != OPENSSH_AUTO &&
          outtype != OPENSSH_NEW && outtype != SSHCOM)) {
         fprintf(stderr, "puttygen: this would generate a new key but "
                 "discard the private part\n");
@@ -612,8 +612,8 @@
      * Also in this code, we bomb out if the input and output file
      * formats are the same and no other action is performed.
      */
-    if ((intype == SSH_KEYTYPE_SSH1 && outtype == PRIVATE) ||
-        (intype == SSH_KEYTYPE_SSH2 && outtype == PRIVATE) ||
+    if ((intype == SSH_KEYTYPE_SSH1 && outtype == PPRIVATE) ||
+        (intype == SSH_KEYTYPE_SSH2 && outtype == PPRIVATE) ||
         (intype == SSH_KEYTYPE_OPENSSH_PEM && outtype == OPENSSH_AUTO) ||
         (intype == SSH_KEYTYPE_OPENSSH_NEW && outtype == OPENSSH_NEW) ||
         (intype == SSH_KEYTYPE_SSHCOM && outtype == SSHCOM)) {
@@ -633,7 +633,7 @@
              * Bomb out rather than automatically choosing to write
              * a private key file to stdout.
              */
-            if (outtype == PRIVATE || outtype == OPENSSH_AUTO ||
+            if (outtype == PPRIVATE || outtype == OPENSSH_AUTO ||
                 outtype == OPENSSH_NEW || outtype == SSHCOM) {
                 fprintf(stderr, "puttygen: need to specify an output file\n");
                 return 1;
@@ -647,7 +647,7 @@
      * out a private key format, or (b) the entire input key file
      * is encrypted.
      */
-    if (outtype == PRIVATE || outtype == OPENSSH_AUTO ||
+    if (outtype == PPRIVATE || outtype == OPENSSH_AUTO ||
         outtype == OPENSSH_NEW || outtype == SSHCOM ||
         intype == SSH_KEYTYPE_OPENSSH_PEM ||
         intype == SSH_KEYTYPE_OPENSSH_NEW ||
@@ -950,7 +950,7 @@
         bool ret;
         int real_outtype;
 
-      case PRIVATE:
+      case PPRIVATE:
         random_ref(); /* we'll need a few random bytes in the save file */
         if (sshver == 1) {
             assert(ssh1key);
