--- metacity-2.25.144/src/metacity.schemas.in.in.old	2009-03-05 17:09:17.883154000 +0800
+++ metacity-2.25.144/src/metacity.schemas.in.in	2009-03-05 17:10:22.423735000 +0800
@@ -247,7 +247,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>nimbus</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
