diff -ruN metacity-2.28.1.orig/src/core/delete.c metacity-2.28.1/src/core/delete.c
--- metacity-2.28.1.orig/src/core/delete.c	2010-02-02 06:39:27.349000933 +0000
+++ metacity-2.28.1/src/core/delete.c	2010-02-02 06:40:26.190105767 +0000
@@ -38,6 +38,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <sys/wait.h>
 
 static void meta_window_present_delete_dialog (MetaWindow *window,
                                                guint32     timestamp);
--- metacity-2.30.3/src/core/bell.c.~1~	2010-09-04 20:09:52.000000000 +0400
+++ metacity-2.30.3/src/core/bell.c	2014-12-22 22:39:02.230066926 +0300
@@ -311,15 +311,8 @@
 
       ca_proplist_destroy (p);
 
-      if (res != CA_SUCCESS && res != CA_ERROR_DISABLED)
-        {      
-          /* ...and in case that failed we use the classic X11 bell. */
-          XkbForceDeviceBell (display->xdisplay, 
-                              xkb_bell_event->device, 
-                              xkb_bell_event->bell_class, 
-                              xkb_bell_event->bell_id, 
-                              xkb_bell_event->percent);
-        }
+      XkbForceBell(display->xdisplay, xkb_bell_event->percent);
+
     }
 }
 #endif /* HAVE_XKB */
