--- otp_src_R12B-5/lib/erl_interface/configure.orig	Wed Jan 25 05:47:46 2012
+++ otp_src_R12B-5/lib/erl_interface/configure	Wed Jan 25 05:51:11 2012
@@ -5043,16 +5043,6 @@
    { (exit 1); exit 1; }; }
 fi
 
-case $host in
-  *-*-solaris*|free_source)
-    if test -x /usr/ucb/install; then
-      INSTALL="/usr/ucb/install -c"
-    fi
-    ;;
-  *)
-    ;;
-esac
-
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
 # incompatible versions:
--- otp_src_R12B-5/erts/configure.orig	Wed Jan 25 06:18:20 2012
+++ otp_src_R12B-5/erts/configure	Wed Jan 25 06:18:34 2012
@@ -5283,15 +5283,6 @@
 
 _search_path=
 
-case $host in
-  *-*-solaris*|free_source)
-    if test -x /usr/ucb/install; then
-      INSTALL="/usr/ucb/install -c"
-    fi
-    ;;
-  *)
-    ;;
-esac
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
 # incompatible versions:
