--- nautilus-sendto-2.28.5/src/plugins/gajim/Makefile.am	Sat Sep  7 11:28:42 2013
+++ nautilus-sendto-2.28.5/src/plugins/gajim/Makefile.am	Sat Sep  7 11:28:45 2013
@@ -7,6 +7,7 @@
         -I$(top_srcdir)/src                     \
         -I$(top_builddir)                       \
         $(NAUTILUS_SENDTO_CFLAGS)               \
+	$(DBUS_CFLAGS)				\
         $(DISABLE_DEPRECATED)			\
 	$(WARN_CFLAGS)
 
