--- typeguard-4.4.4/pyproject.toml.orig
+++ typeguard-4.4.4/pyproject.toml
@@ -104,7 +104,7 @@
 requires = ["tox >= 4.22"]
 
 [tool.tox.env_run_base]
-commands = [["coverage", "run", "-m", "pytest", { replace = "posargs", extend = true }]]
+commands = [["python", "-m", "coverage", "run", "-m", "pytest", { replace = "posargs", extend = true }]]
 package = "editable"
 dependency_groups = ["test"]
 
