--- tar-1.32/gnu/sys-limits.h.orig	2020-03-20 11:03:58.634448830 +0000
+++ tar-1.32/gnu/sys-limits.h	2020-03-20 11:04:27.578798808 +0000
@@ -15,8 +15,8 @@
    You should have received a copy of the GNU General Public License
    along with this program; if not, see <https://www.gnu.org/licenses/>.  */
 
-#ifndef _SYS_LIMITS_H
-#define _SYS_LIMITS_H
+#ifndef GTAR_SYS_LIMITS_H
+#define GTAR_SYS_LIMITS_H
 
 #include <limits.h>
 
