--- gnome-mount-0.8/src/gnome-mount.c.1	2017-12-25 13:00:01.385522600 +0000
+++ gnome-mount-0.8/src/gnome-mount.c	2017-12-25 13:00:29.437695673 +0000
@@ -1577,8 +1577,7 @@
 		message = g_strdup_printf (_("There is data that needs to be written to the device %s before it can be removed. Please do not remove the media or disconnect the drive."), unmount_note_drive_name);
 		unmount_note = notify_notification_new (summary,
 							message,
-							"gnome-dev-harddisk", /* TODO: use appropriate icon */
-							NULL);
+							"gnome-dev-harddisk" /* TODO: use appropriate icon */ );
 		if (unmount_note == NULL) {
 			g_warning ("Cannot create note for unmount cache sync");
 		} else {
