diff -uprN gnome-power-manager-2.24.4-orig/src/gpm-dpms.c gnome-power-manager-2.24.4/src/gpm-dpms.c
--- gnome-power-manager-2.24.4-orig/src/gpm-dpms.c	2009-03-19 14:37:38.856619450 +0800
+++ gnome-power-manager-2.24.4/src/gpm-dpms.c	2009-03-19 14:39:35.232623388 +0800
@@ -343,6 +343,7 @@ static gboolean
 sync_settings (GpmDpms *dpms,
 	       GError **error)
 {
+#if 0
 	guint    standby;
 	guint    suspend;
 	guint    off;
@@ -378,6 +379,8 @@ sync_settings (GpmDpms *dpms,
 					      suspend,
 					      off,
 					      error);
+#endif
+	return TRUE;
 }
 
 gboolean
