--- gnome-power-manager-2.24.4/data/gnome-power-manager.desktop.in.orig	2009-04-22 16:15:07.025188065 +0800
+++ gnome-power-manager-2.24.4/data/gnome-power-manager.desktop.in	2009-04-22 16:15:15.312355348 +0800
@@ -13,3 +13,4 @@ X-GNOME-Bugzilla-Product=gnome-power-man
 X-GNOME-Bugzilla-Component=gnome-power-manager
 X-GNOME-Bugzilla-Version=2.24.4
 X-GNOME-Bugzilla-ExtraInfoScript=gnome-power-bugreport.sh
+X-GNOME-AutoRestart=true
