--- gnome-power-manager-2.24.4/src/gpm-notify.c.1	2016-04-13 22:26:13.262439131 +0300
+++ gnome-power-manager-2.24.4/src/gpm-notify.c	2016-04-13 22:28:29.766327259 +0300
@@ -103,14 +103,7 @@
 //		notify->priv->libnotify = NULL;
 	}
 
-	if (notify->priv->status_icon != NULL &&
-	    gtk_status_icon_get_visible (notify->priv->status_icon)) {
-		notify->priv->libnotify = notify_notification_new_with_status_icon (title, content,
-										    msgicon,
-										    notify->priv->status_icon);
-	} else {
-		notify->priv->libnotify = notify_notification_new (title, content, msgicon, NULL);
-	}
+	notify->priv->libnotify = notify_notification_new (title, content, msgicon);
 
 	if (timeout == GPM_NOTIFY_TIMEOUT_NEVER) {
 		notify_notification_set_timeout (notify->priv->libnotify, 0);
