We have no pytest-subket packaged yet.

--- pip-25.2/pyproject.toml.orig
+++ pip-25.2/pyproject.toml
@@ -59,7 +59,6 @@
     "pytest-cov",
     "pytest-rerunfailures",
     "pytest-xdist",
-    "pytest-subket",
     "scripttest",
     "setuptools",
     # macOS (darwin) arm64 always uses virtualenv >= 20.0
@@ -304,9 +303,6 @@
     "--color=yes",
     "--ignore=src/pip/_vendor",
     "--ignore=tests/tests_cache",
-    "--disable-socket",
-    "--allow-unix-socket",
-    "--allow-hosts=localhost",
 ]
 xfail_strict = true
 markers = [
