https://github.com/PyO3/setuptools-rust/issues/461

--- setuptools_rust-1.10.1/setuptools_rust/build.py.orig
+++ setuptools_rust-1.10.1/setuptools_rust/build.py
@@ -37,7 +37,7 @@
 
 
 try:
-    from wheel.bdist_wheel import bdist_wheel as CommandBdistWheel
+    from setuptools.command.bdist_wheel import bdist_wheel as CommandBdistWheel
 except ImportError:  # wheel installation might be deferred in PEP 517
     from setuptools import Command as CommandBdistWheel
 
--- setuptools_rust-1.10.1/setuptools_rust/setuptools_ext.py.orig
+++ setuptools_rust-1.10.1/setuptools_rust/setuptools_ext.py
@@ -19,7 +19,7 @@
 from .extension import Binding, RustBin, RustExtension, Strip
 
 try:
-    from wheel.bdist_wheel import bdist_wheel
+    from setuptools.command.bdist_wheel import bdist_wheel
 except ImportError:
     bdist_wheel = None
 
