--- gstreamer-0.10.32/gst/gstregistry.c-orig	2012-04-03 12:52:13.923734456 -0500
+++ gstreamer-0.10.32/gst/gstregistry.c	2012-04-03 12:53:56.632467757 -0500
@@ -1592,7 +1592,11 @@ ensure_current_registry (GError ** error
   registry_file = g_strdup (g_getenv ("GST_REGISTRY"));
   if (registry_file == NULL) {
     registry_file = g_build_filename (g_get_home_dir (),
+#if defined (__amd64)
+        ".gstreamer-" GST_MAJORMINOR, "registry." "amd64" ".bin", NULL);
+#else
         ".gstreamer-" GST_MAJORMINOR, "registry." HOST_CPU ".bin", NULL);
+#endif
   }
 
   if (!_gst_disable_registry_cache) {
