--- gstreamer-0.10.36/libs/gst/base/gsttypefindhelper.c-orig	2012-05-02 17:46:10.988641464 -0500
+++ gstreamer-0.10.36/libs/gst/base/gsttypefindhelper.c	2012-05-02 17:47:32.187031547 -0500
@@ -43,7 +43,8 @@
 /* ********************** typefinding in pull mode ************************ */
 
 static void
-helper_find_suggest (gpointer data, guint probability, const GstCaps * caps);
+helper_find_suggest (gpointer data, GstTypeFindProbability probability,
+    const GstCaps * caps);
 
 typedef struct
 {
