https://github.com/sphinx-doc/sphinxcontrib-applehelp/pull/9

--- sphinxcontrib-applehelp-1.0.2/tox.ini.orig
+++ sphinxcontrib-applehelp-1.0.2/tox.ini
@@ -4,7 +4,7 @@
 [testenv]
 deps=
     pytest
-    git+https://github.com/sphinx-doc/sphinx
+    sphinx @ git+https://github.com/sphinx-doc/sphinx
 extras=
     test
 setenv =
@@ -15,7 +15,7 @@
 [testenv:py39]
 deps =
     {[testenv]deps}
-    git+https://github.com/html5lib/html5lib-python
+    html5lib @ git+https://github.com/html5lib/html5lib-python
 
 [testenv:flake8]
 description =
