--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/osc/osc/OscHostEndianness.h.orig	2022-08-10 16:19:40.415029036 +0000
+++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/osc/osc/OscHostEndianness.h	2022-08-10 16:20:27.584673043 +0000
@@ -51,7 +51,7 @@
 
 #else
 
-    #if defined(__GLIBC__) || defined(__ANDROID__) || defined(__CYGWIN__)
+    #if defined(__GLIBC__) || defined(__ANDROID__) || defined(__CYGWIN__) || ((__sun) && (__SVR4))
         #include <endian.h>
         #if (__BYTE_ORDER == __LITTLE_ENDIAN)
             #ifndef __LITTLE_ENDIAN__
