--- llvm-19.1.7.src/tools/clang/CMakeLists.txt.orig
+++ llvm-19.1.7.src/tools/clang/CMakeLists.txt
@@ -2,7 +2,7 @@
 set(LLVM_SUBPROJECT_TITLE "Clang")
 
 if(NOT DEFINED LLVM_COMMON_CMAKE_UTILS)
-  set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+  set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 endif()
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
--- llvm-19.1.7.src/CMakeLists.txt.orig
+++ llvm-19.1.7.src/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
 
--- llvm-19.1.7.src/cmake/modules/HandleLLVMOptions.cmake.orig
+++ llvm-19.1.7.src/cmake/modules/HandleLLVMOptions.cmake
@@ -1388,7 +1388,7 @@
   endif()
 endif()
 
-set(LLVM_THIRD_PARTY_DIR  ${CMAKE_CURRENT_SOURCE_DIR}/../third-party CACHE STRING
+set(LLVM_THIRD_PARTY_DIR  ${CMAKE_CURRENT_SOURCE_DIR}/third-party CACHE STRING
     "Directory containing third party software used by LLVM (e.g. googletest)")
 
 set(LLVM_UNITTEST_LINK_FLAGS "" CACHE STRING
