--- gamin-0.1.10/configure.in.1	2016-07-05 13:20:32.204942553 +0300
+++ gamin-0.1.10/configure.in	2016-07-05 14:38:45.765920210 +0300
@@ -506,7 +506,6 @@
 PYTHON_INCLUDES=
 PYTHON_SITE_PACKAGES=
 PYTHON_TESTS=
-pythondir=
 if test "$with_python" != "no" ; then
     if test -x "$with_python/bin/python"
     then
@@ -559,11 +558,13 @@
 	    fi
 	fi
     fi
-    if test "$with_python" != ""
-    then
-        pythondir='$(PYTHON_SITE_PACKAGES)'
-    else
-        pythondir='$(libdir)/python$(PYTHON_VERSION)/vendor-packages'
+    if test -z "$pythondir"; then 
+        if test "$with_python" != ""
+        then
+            pythondir='$(PYTHON_SITE_PACKAGES)'
+        else
+            pythondir='$(libdir)/python$(PYTHON_VERSION)/vendor-packages'
+        fi
     fi
 else
     PYTHON=
