diff -ur devhelp-0.19.1-orig/data/devhelp.desktop.in.in devhelp-0.19.1/data/devhelp.desktop.in.in
--- devhelp-0.19.1-orig/data/devhelp.desktop.in.in	2008-08-15 16:39:24.632145267 +0800
+++ devhelp-0.19.1/data/devhelp.desktop.in.in	2008-08-15 16:44:39.824572320 +0800
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Devhelp
-_Comment=Developer's Help program
+# SUN_BRANDING
+_Name=GNOME Developer Documentation
+# SUN_BRANDING
+_Comment=Browse and search gtk+ and GNOME API documentation
 _GenericName=Documentation Browser
 Exec=devhelp
 Terminal=false
