--- devhelp-2.32.0/data/devhelp.desktop.in.in.~1~	2014-07-17 23:14:14.975712197 +0400
+++ devhelp-2.32.0/data/devhelp.desktop.in.in	2014-07-17 23:15:13.788172031 +0400
@@ -1,6 +1,8 @@
 [Desktop Entry]
-_Name=Devhelp
-_Comment=Developer's Help program
+# SUN_BRANDING
+_Name=GNOME Developer Documentation
+# SUN_BRANDING
+_Comment=Browse and search gtk+ and GNOME API documentation
 _GenericName=Documentation Browser
 Exec=devhelp
 Terminal=false
