diff -ruN liboil-0.3.16.orig/doc/Makefile.am liboil-0.3.16/doc/Makefile.am
--- liboil-0.3.16.orig/doc/Makefile.am  2010-01-05 10:03:59.816025670 +0000
+++ liboil-0.3.16/doc/Makefile.am       2010-01-05 10:04:20.180786367 +0000
@@ -33,7 +33,7 @@
 
 # Extra options to supply to gtkdoc-fixref. Not normally needed.
 # e.g. FIXXREF_OPTIONS=--extra-dir=../gdk-pixbuf/html --extra-dir=../gdk/html
-FIXXREF_OPTIONS=
+FIXXREF_OPTIONS=--module=$(DOC_MODULE)
 
 # Used for dependencies. The docs will be rebuilt if any of these change.
 # e.g. HFILE_GLOB=$(top_srcdir)/gtk/*.h
