--- coreutils-8.21/configure.orig	2013-02-14 10:54:18.000000000 -0500
+++ coreutils-8.21/configure		2013-07-19 08:06:46.984375000 -0400
@@ -62759,7 +62759,6 @@
 }
 fi
 
-# Limit stdbuf to ELF systems with GCC
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether this is an ELF system" >&5
 $as_echo_n "checking whether this is an ELF system... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -62778,15 +62777,17 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $elf_sys" >&5
 $as_echo "$elf_sys" >&6; }
-if test "$elf_sys" = "yes" && \
-   test "$GCC" = "yes"; then
-  {
-  if test -z "$optional_bin_progs"; then
-    optional_bin_progs=stdbuf
-  else
-    optional_bin_progs="$optional_bin_progs stdbuf"
-  fi
-}
+
+if test -z "$optional_bin_progs"; then
+  optional_bin_progs=stdbuf
+else
+  optional_bin_progs="$optional_bin_progs stdbuf"
+ fi
+
+if test -z "$optional_pkglib_progs"; then
+  optional_pkglib_progs=libstdbuf.so
+else
+  optional_pkglib_progs="$optional_pkglib_progs libstdbuf.so"
 fi
 
 ############################################################################
