--- coreutils-8.16/configure.orig	2012-04-18 07:35:04.318609521 -0700
+++ coreutils-8.16/configure	2012-04-18 07:36:27.319985971 -0700
@@ -62005,7 +62005,6 @@
 }
 fi
 
-# Limit stdbuf to ELF systems with GCC
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether this is an ELF system" >&5
 $as_echo_n "checking whether this is an ELF system... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -62024,15 +62023,17 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $elf_sys" >&5
 $as_echo "$elf_sys" >&6; }
-if test "$elf_sys" = "yes" && \
-   test "$GCC" = "yes"; then
-  {
-  if test -z "$optional_bin_progs"; then
-    optional_bin_progs=stdbuf
-  else
-    optional_bin_progs="$optional_bin_progs stdbuf"
-  fi
-}
+
+if test -z "$optional_bin_progs"; then
+  optional_bin_progs=stdbuf
+else
+  optional_bin_progs="$optional_bin_progs stdbuf"
+ fi
+
+if test -z "$optional_pkglib_progs"; then
+  optional_pkglib_progs=libstdbuf.so
+else
+  optional_pkglib_progs="$optional_pkglib_progs libstdbuf.so"
 fi
 
 ############################################################################
