--- coreutils-8.21/lib/mountlist.c	2013-02-07 11:44:47.000000000 -0500
+++ coreutils-8.21/lib/mountlist.c.new	2013-07-19 08:06:37.671875000 -0400
@@ -207,10 +207,11 @@
 #endif
 
 #ifndef ME_REMOTE
-/* A file system is "remote" if its Fs_name contains a ':'
-   or if (it is of type (smbfs or cifs) and its Fs_name starts with '//').  */
+/* A file system is `remote' if its Fs_name contains a `:' (and isn't of 
+   type pcfs), or if (it is of type (smbfs or cifs) and its Fs_name starts
+   with `//').  */
 # define ME_REMOTE(Fs_name, Fs_type)            \
-    (strchr (Fs_name, ':') != NULL              \
+    ((strchr (Fs_name, ':') != NULL && strcmp (Fs_type, "pcfs") != 0) \
      || ((Fs_name)[0] == '/'                    \
          && (Fs_name)[1] == '/'                 \
          && (strcmp (Fs_type, "smbfs") == 0     \
