--- coreutils-8.21/src/rm.c.orig	2013-01-30 19:46:24.000000000 -0500
+++ coreutils-8.21/src/rm.c		2013-07-19 11:51:45.526576400 -0400
@@ -339,9 +339,13 @@
     {
       fprintf (stderr,
                (x.recursive
-                ? _("%s: remove all arguments recursively? ")
-                : _("%s: remove all arguments? ")),
-               program_name);
+                ? ngettext ("%s: remove %zu argument recursively? ",
+                            "%s: remove %zu arguments recursively? ",
+                            select_plural (n_files))
+                : ngettext ("%s: remove %zu argument? ",
+                            "%s: remove %zu arguments? ",
+                            select_plural (n_files))),
+               program_name, n_files);
       if (!yesno ())
         exit (EXIT_SUCCESS);
     }
