--- coreutils-8.22/tests/df/no-mtab-status.sh.~1~	2013-12-04 18:48:30.000000000 +0400
+++ coreutils-8.22/tests/df/no-mtab-status.sh	2015-01-15 17:33:26.371656533 +0300
@@ -27,9 +27,13 @@
 cat > k.c <<'EOF' || framework_failure_
 #include <stdio.h>
 #include <errno.h>
-#include <mntent.h>
-
+#ifdef __sun
+#include <sys/mnttab.h>
+int getmntent (FILE *fp, struct mnttab *mp)
+#else
+#include <mntent.h>
 struct mntent *getmntent (FILE *fp)
+#endif
 {
   /* Prove that LD_PRELOAD works. */
   static int done = 0;
@@ -40,7 +44,11 @@
     }
   /* Now simulate the failure. */
   errno = ENOENT;
+#ifdef __sun
+  return -1;
+#else
   return NULL;
+#endif
 }
 EOF
 
