--- coreutils-8.22/tests/df/skip-duplicates.sh.~1~	2013-12-04 18:48:30.000000000 +0400
+++ coreutils-8.22/tests/df/skip-duplicates.sh	2015-01-15 18:32:03.508343752 +0300
@@ -28,6 +28,43 @@
 cat > k.c <<'EOF' || framework_failure_
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef __sun
+#include <sys/mnttab.h>
+
+int getmntent (FILE *fp, struct mnttab *mp)
+{
+  /* Prove that LD_PRELOAD works. */
+  static int done = 0;
+  if (!done)
+    {
+      fclose (fopen ("x", "w"));
+      ++done;
+    }
+
+  static struct mnttab mntents[] = {
+    {.mnt_special="/short",  .mnt_mountp="/invalid/mount/dir"},
+    {.mnt_special="fsname",  .mnt_mountp="/",},
+    {.mnt_special="/fsname", .mnt_mountp="/root"},
+    {.mnt_special="/fsname", .mnt_mountp="/"},
+  };
+
+  if (!getenv ("CU_TEST_DUPE_INVALID") && done == 1)
+    done++;  /* skip the first entry.  */
+
+  while (done++ <= 4)
+    {
+      mp->mnt_special=mntents[done-2].mnt_special;
+      mp->mnt_mountp=mntents[done-2].mnt_mountp;
+      mp->mnt_fstype = "-";
+      mp->mnt_mntopts="";
+      mp->mnt_time="";
+
+      return 0;
+    }
+
+  return -1;
+}
+#else
 #include <mntent.h>
 
 struct mntent *getmntent (FILE *fp)
@@ -57,6 +94,7 @@
     }
   return NULL;
 }
+#endif
 EOF
 
 # Then compile/link it:
