--- coreutils-8.21/src/who.c.orig	2013-01-30 19:46:24.000000000 -0500
+++ coreutils-8.21/src/who.c		2013-07-19 08:07:58.500000000 -0400
@@ -514,8 +514,8 @@
   unsigned char curr = UT_PID (utmp_ent) % 256;
 
   if (!runlevline)
-    runlevline = xmalloc (strlen (_("run-level")) + 3);
-  sprintf (runlevline, "%s %c", _("run-level"), curr);
+    runlevline = xmalloc (sizeof (utmp_ent->ut_line));
+  sprintf (runlevline, "%s", utmp_ent->ut_line);
 
   if (!comment)
     comment = xmalloc (strlen (_("last=")) + 2);
