--- giflib-5.2.2/doc/Makefile.orig	2024-02-18 20:15:05.000000000 +0200
+++ giflib-5.2.2/doc/Makefile	2025-06-17 18:46:18.724644866 +0300
@@ -43,7 +43,7 @@
 
 # Logo image file for HTML docs
 giflib-logo.gif: ../pic/gifgrid.gif
-	convert $^ -resize 50x50 $@
+	magick $^ -resize 50x50 $@
 
 # Philosophical choice: the website gets the internal manual pages
 allhtml: $(XMLALL:.xml=.html) giflib-logo.gif
