https://gitlab.isc.org/isc-projects/dhcp/issues/83

--- dhcp-4.4.2/keama/keama.c.1	2020-01-23 11:45:09.902945953 +0000
+++ dhcp-4.4.2/keama/keama.c	2020-01-23 11:45:21.227152034 +0000
@@ -21,9 +21,9 @@
  *
  */
 
-#include <sys/errno.h>
 #include <arpa/inet.h>
 #include <assert.h>
+#include <errno.h>
 #include <fcntl.h>
 #include <stdarg.h>
 #include <stdlib.h>
diff -ur dhcp-4.4.2-1/keama/confparse.c dhcp-4.4.2/keama/confparse.c
--- dhcp-4.4.2-1/keama/confparse.c	2020-01-23 11:52:07.125812139 +0000
+++ dhcp-4.4.2/keama/confparse.c	2020-01-23 11:53:01.302995421 +0000
@@ -25,10 +25,10 @@
 
 #include "keama.h"
 
-#include <sys/errno.h>
 #include <arpa/inet.h>
 #include <assert.h>
 #include <ctype.h>
+#include <errno.h>
 #include <fcntl.h>
 #include <time.h>
 #include <stdlib.h>
diff -ur dhcp-4.4.2-1/keama/eval.c dhcp-4.4.2/keama/eval.c
--- dhcp-4.4.2-1/keama/eval.c	2020-01-23 11:52:07.074672343 +0000
+++ dhcp-4.4.2/keama/eval.c	2020-01-23 11:53:16.607161657 +0000
@@ -23,10 +23,10 @@
 
 #include "keama.h"
 
-#include <sys/errno.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
 #include <ctype.h>
+#include <errno.h>
 #include <netdb.h>
 #include <stdarg.h>
 #include <stdio.h>
diff -ur dhcp-4.4.2-1/keama/print.c dhcp-4.4.2/keama/print.c
--- dhcp-4.4.2-1/keama/print.c	2020-01-23 11:52:07.074901688 +0000
+++ dhcp-4.4.2/keama/print.c	2020-01-23 11:52:38.214309421 +0000
@@ -23,10 +23,10 @@
 
 #include "keama.h"
 
-#include <sys/errno.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
 #include <ctype.h>
+#include <errno.h>
 #include <netdb.h>
 #include <stdarg.h>
 #include <stdio.h>
diff -ur dhcp-4.4.2-1/keama/reduce.c dhcp-4.4.2/keama/reduce.c
--- dhcp-4.4.2-1/keama/reduce.c	2020-01-23 11:52:07.075218446 +0000
+++ dhcp-4.4.2/keama/reduce.c	2020-01-23 11:52:26.230241016 +0000
@@ -23,10 +23,10 @@
 
 #include "keama.h"
 
-#include <sys/errno.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
 #include <ctype.h>
+#include <errno.h>
 #include <netdb.h>
 #include <stdarg.h>
 #include <stdio.h>
