These modules are deprecated.
We do not have libmaxminddb.

--- Mail-SpamAssassin-4.0.0/Makefile.PL.orig
+++ Mail-SpamAssassin-4.0.0/Makefile.PL
@@ -374,8 +374,6 @@
         'Net::LibIDN2' =>              0,
         'Net::LibIDN' =>               0,
         'Mail::SPF' =>                 0,
-        'MaxMind::DB::Reader' =>       0,
-        'MaxMind::DB::Reader::XS' =>   0,
         'Geo::IP' =>                   0,
         'IP::Country::DB_File' =>      0,
         'IP::Country::Fast' =>         0,
--- Mail-SpamAssassin-4.0.0/META.json.orig
+++ Mail-SpamAssassin-4.0.0/META.json
@@ -53,8 +53,6 @@
             "Mail::DKIM" : "0.37",
             "Mail::DMARC" : "0",
             "Mail::SPF" : "0",
-            "MaxMind::DB::Reader" : "0",
-            "MaxMind::DB::Reader::XS" : "0",
             "Net::CIDR::Lite" : "0",
             "Net::LibIDN" : "0",
             "Net::LibIDN2" : "0",
