--- Mail-SpamAssassin-4.0.0/t/dnsbl_subtests.t.orig
+++ Mail-SpamAssassin-4.0.0/t/dnsbl_subtests.t
@@ -15,6 +15,7 @@
 use Errno qw(EADDRINUSE EACCES);
 
 use constant HAS_NET_DNS_NAMESERVER => eval { require Net::DNS::Nameserver; };
+plan skip_all => "Net tests disabled" unless conf_bool('run_net_tests');
 plan skip_all => "Net::DNS::Nameserver in unavailable on this system" unless (HAS_NET_DNS_NAMESERVER);
 plan  tests => 46;
 
