The IO::Socket::INET6 module is deprecated.

--- Mail-SpamAssassin-4.0.1/Makefile.PL.orig
+++ Mail-SpamAssassin-4.0.1/Makefile.PL
@@ -378,7 +378,6 @@
         'IP::Country::Fast' =>         0,
         'Razor2::Client::Agent' =>     2.61,
         'IO::Socket::IP' =>            0.09,
-        'IO::Socket::INET6' =>         0,
         'IO::Socket::SSL' =>           1.76,
         'Compress::Zlib' =>            0,
         'Mail::DKIM' =>                0.37,
--- Mail-SpamAssassin-4.0.1/META.json.orig
+++ Mail-SpamAssassin-4.0.1/META.json
@@ -41,7 +41,6 @@
             "Email::Address::XS" : "0",
             "Encode::Detect::Detector" : "0",
             "Geo::IP" : "0",
-            "IO::Socket::INET6" : "0",
             "IO::Socket::IP" : "0.09",
             "IO::Socket::SSL" : "1.76",
             "IO::String" : "0",
