--- openscap-0.8.1/configure.ac.orig	2012-05-09 09:09:41.227540183 -0700
+++ openscap-0.8.1/configure.ac	2012-05-09 15:05:59.163156764 -0700
@@ -837,11 +837,11 @@
 
 AC_PATH_PROG(PERL, perl)
 PERL_INCLUDES="`$PERL -e 'use Config; print $Config{archlib}'`/CORE"
-vendorlib="$(  $PERL -e 'use Config; print $Config{vendorlib}'  | sed "s|$($PERL -e 'use Config; print $Config{prefix}')||" )"
-vendorarch="$( $PERL -e 'use Config; print $Config{vendorarch}' | sed "s|$($PERL -e 'use Config; print $Config{prefix}')||" )"
+vendorlib="$(  $PERL -e 'use Config; print $Config{vendorlib}')"
+vendorarch="$( $PERL -e 'use Config; print $Config{vendorarch}')"
 AC_SUBST([PERL_INCLUDES], ["$PERL_INCLUDES"])
-AC_SUBST([perl_vendorlibdir], ['${prefix}'$vendorlib])
-AC_SUBST([perl_vendorarchdir], ['${prefix}'$vendorarch])
+AC_SUBST([perl_vendorlibdir], [$vendorlib])
+AC_SUBST([perl_vendorarchdir], [$vendorarch])
 
 AC_ARG_ENABLE([oval],
      [AC_HELP_STRING([--enable-oval], [include support for OVAL (default=yes)])],
