--- openscap-0.9.7/src/OVAL/probes/unix/solaris/isainfo.c.~1~	2013-04-23 06:21:51.478000001 -0700
+++ openscap-0.9.7/src/OVAL/probes/unix/solaris/isainfo.c	2013-09-10 11:06:13.488839049 -0700
@@ -47,6 +47,7 @@
 #include <arpa/inet.h>
 #include <regex.h>
 #include <sys/systeminfo.h>
+#include "../../../../common/debug_priv.h"
 
 /* man sysinfo (2) recommends using 257 for this size */
 #define MAX_STR_RESULT 257
@@ -92,8 +93,12 @@
 	if (sysinfo(SI_ARCHITECTURE_K, result.kernel_isa, MAX_STR_RESULT) == -1) {
 		return err;
 	}
-
+#if defined(__SVR4) && defined(__sun)
+	if ((sysinfo(SI_ARCHITECTURE_32, result.application_isa, MAX_STR_RESULT) == -1) &&
+	   (sysinfo(SI_ARCHITECTURE_64, result.application_isa, MAX_STR_RESULT) == -1)) {
+#else
 	if (sysinfo(SI_ARCHITECTURE_NATIVE, result.application_isa, MAX_STR_RESULT) == -1) {
+#endif
 		return err;
 	}
 
