--- openscap-0.8.1/src/OVAL/oval_probe.c.orig	2012-11-19 10:04:14.596320954 -0800
+++ openscap-0.8.1/src/OVAL/oval_probe.c	2012-11-19 09:56:27.562728645 -0800
@@ -84,7 +84,8 @@
         OVAL_PROBE_EXTERNAL(OVAL_UNIX_PROCESS58, "process58"),
         OVAL_PROBE_EXTERNAL(OVAL_UNIX_FILEEXTENDEDATTRIBUTE, "fileextendedattribute"),
         OVAL_PROBE_EXTERNAL(OVAL_UNIX_GCONF, "gconf"),
-        OVAL_PROBE_EXTERNAL(OVAL_UNIX_ROUTINGTABLE, "routingtable")
+        OVAL_PROBE_EXTERNAL(OVAL_UNIX_ROUTINGTABLE, "routingtable"),
+        OVAL_PROBE_EXTERNAL(OVAL_SOLARIS_SMF, "smf")
 };
 
 #define __PROBE_META_COUNT (sizeof OSCAP_GSYM(__probe_meta)/sizeof OSCAP_GSYM(__probe_meta)[0])
