--- dmenu-4.9/config.mk.orig	2019-02-02 13:55:02.000000000 +0000
+++ dmenu-4.9/config.mk	2020-08-09 15:02:29.080054631 +0000
@@ -2,7 +2,7 @@
 VERSION = 4.9
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = $(PREFIX)/share/man
 
 X11INC = /usr/X11R6/include
@@ -27,5 +27,3 @@
 CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
 LDFLAGS  = $(LIBS)
 
-# compiler and linker
-CC = cc
