--- dmenu-5.3/config.mk.orig	2024-03-19 12:25:09.000000000 +0100
+++ dmenu-5.3/config.mk	2025-06-29 13:24:14.964917587 +0200
@@ -2,7 +2,7 @@
 VERSION = 5.3
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = $(PREFIX)/share/man
 
 X11INC = /usr/X11R6/include
@@ -25,8 +25,6 @@
 
 # flags
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
-CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
+CFLAGS   = -std=c99 -pedantic -Wall -Wno-error=implicit-function-declaration -Os $(INCS) $(CPPFLAGS)
 LDFLAGS  = $(LIBS)
 
-# compiler and linker
-CC = cc
