--- dmenu-4.7/config.mk.orig	2017-06-08 21:45:08.602819712 +0000
+++ dmenu-4.7/config.mk	2017-06-08 21:45:29.864223449 +0000
@@ -2,7 +2,7 @@
 VERSION = 4.7
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr
 MANPREFIX = ${PREFIX}/share/man
 
 X11INC = /usr/X11R6/include
@@ -24,8 +24,8 @@
 
 # flags
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS  = -s ${LIBS}
+CFLAGS   = ${CC_BITS} -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+LDFLAGS  = ${LD_BITS} -s ${LIBS}
 
 # compiler and linker
 CC = cc
