--- dmenu-4.6.orig/Makefile	2016-06-22 12:18:41.989120448 +0300
+++ dmenu-4.6/Makefile	2016-06-22 12:18:20.589834653 +0300
@@ -47,13 +47,13 @@
 	@rm -rf dmenu-${VERSION}
 
 install: all
-	@echo installing executables to ${DESTDIR}${PREFIX}/bin
-	@mkdir -p ${DESTDIR}${PREFIX}/bin
-	@cp -f dmenu dmenu_path dmenu_run stest ${DESTDIR}${PREFIX}/bin
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/dmenu
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/dmenu_path
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/dmenu_run
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/stest
+	@echo installing executables to ${DESTDIR}${PREFIX}/bin/${MACH64}
+	@mkdir -p ${DESTDIR}${PREFIX}/bin/${MACH64}
+	@cp -f dmenu dmenu_path dmenu_run stest ${DESTDIR}${PREFIX}/bin/${MACH64}
+	@chmod 755 ${DESTDIR}${PREFIX}/bin/${MACH64}/dmenu
+	@chmod 755 ${DESTDIR}${PREFIX}/bin/${MACH64}/dmenu_path
+	@chmod 755 ${DESTDIR}${PREFIX}/bin/${MACH64}/dmenu_run
+	@chmod 755 ${DESTDIR}${PREFIX}/bin/${MACH64}/stest
 	@echo installing manual pages to ${DESTDIR}${MANPREFIX}/man1
 	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
 	@sed "s/VERSION/${VERSION}/g" < dmenu.1 > ${DESTDIR}${MANPREFIX}/man1/dmenu.1
