--- libproxy-0.4.18/libmodman/test/CMakeLists.txt.orig	2017-01-09 15:23:39.000000000 +0100
+++ libproxy-0.4.18/libmodman/test/CMakeLists.txt	2023-10-21 12:07:24.701047968 +0200
@@ -28,10 +28,6 @@
     set_property(TARGET ${name} PROPERTY 
                  COMPILE_DEFINITIONS
                  EXTTYPE=${EXTTYPE}_extension;SYMB=1)
-    if(NOT WIN32 AND NOT APPLE)
-      set_property(TARGET ${name} PROPERTY
-                   LINK_FLAGS -Wl,--no-as-needed)
-    endif()
   endif()
 endfunction(mm_create_program)
 
