--- libproxy-0.3.1/src/modules/Makefile.am.ori	Wed Aug  8 11:04:12 2012
+++ libproxy-0.3.1/src/modules/Makefile.am	Wed Aug  8 11:09:01 2012
@@ -93,7 +93,7 @@
 pacrunner_mozjs_la_SOURCES         = pacrunner_mozjs.c
 pacrunner_mozjs_la_CPPFLAGS        = -I$(top_srcdir)/src/lib @MOZJS_CFLAGS@
 pacrunner_mozjs_la_LIBADD          = ../lib/libproxy.la
-pacrunner_mozjs_la_LDFLAGS         = $(AM_LDFLAGS) @MOZJS_LIBS@
+pacrunner_mozjs_la_LDFLAGS         = $(AM_LDFLAGS) @MOZJS_LIBS@ -lnsl
 nodist_pacrunner_mozjs_la_SOURCES  = pacutils.h
 
 # NetworkManager Module
