--- ctags-5.8.orig/Makefile.in	2016-06-21 15:10:25.569889751 +0300
+++ ctags-5.8/Makefile.in	2016-06-21 15:12:42.003919977 +0300
@@ -6,8 +6,8 @@
 # These are the names of the installed programs, in case you wish to change
 # them.
 #
-CTAGS_PROG = ctags
-ETAGS_PROG = etags
+CTAGS_PROG = exuberant-ctags
+ETAGS_PROG = exuberant-etags

 # Set this to the path to your shell (must run Bourne shell commands).
 #
