diff -up ctags-5.7/Makefile.in.destdir ctags-5.7/Makefile.in
--- ctags-5.7/Makefile.in.destdir	2006-10-12 05:26:40.000000000 +0200
+++ ctags-5.7/Makefile.in	2008-02-15 17:10:23.000000000 +0100
@@ -59,7 +59,7 @@ include $(srcdir)/source.mak
 
 VPATH	= $(srcdir)
 
-INSTALL		= cp
+INSTALL		= cp -p
 INSTALL_PROG	= $(INSTALL)
 INSTALL_DATA	= $(INSTALL)
 
@@ -138,7 +141,8 @@ install-ebin: $(DEST_ETAGS)
 install-lib: $(DEST_READ_LIB) $(DEST_READ_INC)
 
 $(DEST_CTAGS): $(CTAGS_EXEC) $(bindir) FORCE
-	$(INSTALL_PROG) $(CTAGS_EXEC) $@  &&  chmod 755 $@
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
+	$(INSTALL_PROG) $(CTAGS_EXEC) $(DESTDIR)$@  &&  chmod 755 $(DESTDIR)$@
 
 $(DEST_ETAGS):
 	- if [ -x $(DEST_CTAGS) ]; then \
@@ -153,7 +157,8 @@ install-cman: $(DEST_CMAN)
 install-eman: $(DEST_EMAN)
 
 $(DEST_CMAN): $(man1dir) $(MANPAGE) FORCE
-	- $(INSTALL_DATA) $(srcdir)/$(MANPAGE) $@  &&  chmod 644 $@
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(man1dir)
+	- $(INSTALL_DATA) $(srcdir)/$(MANPAGE) $(DESTDIR)$@  &&  chmod 644 $(DESTDIR)$@
 
 $(DEST_EMAN):
 	- if [ -f $(DEST_CMAN) ]; then \
@@ -164,9 +169,11 @@ $(DEST_EMAN):
 # install the library
 #
 $(DEST_READ_LIB): $(READ_LIB) $(libdir) FORCE
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(libdir)
 	$(INSTALL_PROG) $(READ_LIB) $@  &&  chmod 644 $@
 
 $(DEST_READ_INC): $(READ_INC) $(incdir) FORCE
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(incdir)
 	$(INSTALL_PROG) $(READ_INC) $@  &&  chmod 644 $@
 
 
