From b56f7d2cce83d39a26ef3b5a509435353d9825b6 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Tue, 4 Aug 2015 15:50:09 -0700
Subject: [PATCH 21/32] privsep_preauth_child should drop Illumos privileges

---
 sshd.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/sshd.c b/sshd.c
index 1e99488..20ae24a 100644
--- a/sshd.c
+++ b/sshd.c
@@ -72,6 +72,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <limits.h>
+#include <priv.h>
 
 #ifdef WITH_OPENSSL
 #include <openssl/dh.h>
@@ -610,6 +611,7 @@ privsep_preauth_child(void)
 {
 	u_int32_t rnd[256];
 	gid_t gidset[1];
+	priv_set_t *pset = NULL;
 
 	/* Enable challenge-response authentication for privilege separation */
 	privsep_challenge_enable();
@@ -649,6 +651,26 @@ privsep_preauth_child(void)
 		fatal("setgroups: %.100s", strerror(errno));
 	permanently_set_uid(privsep_pw);
 #endif
+	/* Drop Illumos privileges. */
+	if ((pset = priv_allocset()) == NULL)
+		fatal("priv_allocset failed");
+	/* Start with "basic" and drop everything we don't need. */
+	priv_basicset(pset);
+	(void) priv_delset(pset, PRIV_FILE_LINK_ANY);
+	(void) priv_delset(pset, PRIV_PROC_INFO);
+	(void) priv_delset(pset, PRIV_PROC_SESSION);
+	(void) priv_delset(pset, PRIV_PROC_FORK);
+	(void) priv_delset(pset, PRIV_FILE_READ);
+	(void) priv_delset(pset, PRIV_FILE_WRITE);
+	(void) priv_delset(pset, PRIV_NET_ACCESS);
+	(void) priv_delset(pset, PRIV_PROC_EXEC);
+
+	if (setppriv(PRIV_SET, PRIV_PERMITTED, pset))
+		fatal("setppriv failed: %s", strerror(errno));
+	if (setppriv(PRIV_SET, PRIV_LIMIT, pset))
+		fatal("setppriv failed: %s", strerror(errno));
+	if (setppriv(PRIV_SET, PRIV_INHERITABLE, pset))
+		fatal("setppriv failed: %s", strerror(errno));
 }
 
 static int
-- 
2.3.2 (Apple Git-55)

