From a510026de2e6c4f3691549c87227d7e23be16953 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 24 Aug 2015 17:56:45 -0700
Subject: [PATCH 29/30] ssh-agent should drop Illumos privs

---
 ssh-agent.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/ssh-agent.c b/ssh-agent.c
index a335ea3..a0d2682 100644
--- a/ssh-agent.c
+++ b/ssh-agent.c
@@ -67,6 +67,7 @@
 #include <stdlib.h>
 #include <time.h>
 #include <string.h>
+#include <priv.h>
 #include <unistd.h>
 #ifdef HAVE_UTIL_H
 # include <util.h>
@@ -1187,6 +1188,7 @@ main(int ac, char **av)
 	struct timeval *tvp = NULL;
 	size_t len;
 	mode_t prev_mask;
+	priv_set_t *privs;
 
 	/* Ensure that fds 0, 1 and 2 are open or directed to /dev/null */
 	sanitise_stdfd();
@@ -1361,6 +1363,26 @@ main(int ac, char **av)
 	/* child */
 	log_init(__progname, SYSLOG_LEVEL_INFO, SYSLOG_FACILITY_AUTH, 0);
 
+	/*
+	 * Drop unneeded privs, including basic ones like fork/exec.
+	 *
+	 * Idiom: remove from 'basic' privs we know we don't want,
+	 * invert the result and remove the resulting set from P.
+	 *
+	 * None of the priv_delset() calls below, nor the setppriv call
+	 * below can fail, so their return values are not checked.
+	 */
+	if ((privs = priv_str_to_set("basic", ",", NULL)) == NULL)
+		fatal("priv_str_to_set failed: %m");
+	(void) priv_delset(privs, PRIV_PROC_EXEC);
+	(void) priv_delset(privs, PRIV_PROC_FORK);
+	(void) priv_delset(privs, PRIV_FILE_LINK_ANY);
+	(void) priv_delset(privs, PRIV_PROC_INFO);
+	(void) priv_delset(privs, PRIV_PROC_SESSION);
+	priv_inverse(privs);
+	(void) setppriv(PRIV_OFF, PRIV_PERMITTED, privs);
+	(void) priv_freeset(privs);
+
 	if (setsid() == -1) {
 		error("setsid: %s", strerror(errno));
 		cleanup_exit(1);
-- 
2.3.2 (Apple Git-55)

