--- desktop-cache-smf-services-0.2.2/pixbuf-loaders/pixbuf-loaders-installer.orig	2025-06-14 14:21:46.618482739 +0300
+++ desktop-cache-smf-services-0.2.2/pixbuf-loaders/pixbuf-loaders-installer	2025-06-14 14:22:16.967290852 +0300
@@ -58,6 +58,7 @@
   BINDIR='/usr/bin'
   DIR="$ARCH"
   [ "$ARCH" == "i386" ] && BINDIR='/usr/bin/i86' && DIR=''
+  [ "$ARCH" == "sparc" ] && BINDIR='/usr/bin/sparcv7' && DIR=''
 
   test -x $BINDIR/gdk-pixbuf-query-loaders || {
       echo "gdk-pixbuf-query-loaders not installed"
