--- a/src/yelp-db-print.c
+++ b/src/yelp-db-print.c
@@ -477,17 +477,17 @@ dbprint_process (YelpDbprint *dbprint)
 	params_max += 20;
 	params = g_renew (gchar *, params, params_max);
     }
-    params[params_i++] = "db.chunk.extension";
+    params[params_i++] = g_strdup ("db.chunk.extension");
     params[params_i++] = g_strdup ("\"\"");
-    params[params_i++] = "db.chunk.info_basename";
+    params[params_i++] = g_strdup ("db.chunk.info_basename");
     params[params_i++] = g_strdup ("\"index\"");
-    params[params_i++] = "db.chunk.max_depth";
+    params[params_i++] = g_strdup ("db.chunk.max_depth");
     params[params_i++] = g_strdup ("0");
-    params[params_i++] = "db2html.navbar.top";
+    params[params_i++] = g_strdup ("db2html.navbar.top");
     params[params_i++] = g_strdup ("0");
-    params[params_i++] = "db2html.navbar.bottom";
+    params[params_i++] = g_strdup ("db2html.navbar.bottom");
     params[params_i++] = g_strdup ("0");
-    params[params_i++] = "db2html.sidenav";
+    params[params_i++] = g_strdup ("db2html.sidenav");
     params[params_i++] = g_strdup ("0");
 
     params[params_i] = NULL;
