--- yelp/yelp.desktop.in.in-orig	2011-07-12 19:14:33.415601370 -0500
+++ yelp/yelp.desktop.in.in	2011-07-12 19:15:18.916634869 -0500
@@ -1,8 +1,9 @@
 [Desktop Entry]
 _Name=Help
-_Comment=Get help with GNOME
+# SUN_BRANDING
+_Comment=Browse and search GNOME and OpenSolaris documentation
 OnlyShowIn=GNOME;
-Exec=yelp
+Exec=yelp %u
 Icon=help-browser
 StartupNotify=true
 Terminal=false
@@ -12,3 +13,4 @@ X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=Yelp
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
+MimeType=x-scheme-handler/ghelp;x-scheme-handler/help;x-scheme-handler/info;x-scheme-handler/man;
