diff -urN yelp-2.30.2/stylesheets/search2html.xsl yelp-2.30.2.new/stylesheets/search2html.xsl
--- yelp-2.30.2/stylesheets/search2html.xsl	2010-08-04 03:41:38.000000000 +0800
+++ yelp-2.30.2.new/stylesheets/search2html.xsl	2011-06-15 16:01:56.351273774 +0800
@@ -77,7 +77,7 @@
         dl { margin-left: 36px; padding-left: 0em; }
         dt { margin-top: 1em; }
         dd { margin-left: 1em; margin-top: 0.5em; }
-        a { text-decoration: none; }
+        a { text-decoration: none; color: </xsl:text><xsl:if test="$yelp.color.anchor=''">blue</xsl:if><xsl:if test="$yelp.color.anchor!=''"><xsl:value-of select="$yelp.color.anchor"/></xsl:if><xsl:text>;}
         a:hover { text-decoration: underline; }
         </xsl:text></style>
       </head>
