--- libtorrent-rasterbar-v2.0.5/src/enum_net.cpp	Sun Dec  5 06:09:00 2021
+++ libtorrent-rasterbar-v2.0.5_patched/src/enum_net.cpp	Fri Thu  1 01:45:15 2022
@@ -1524,7 +1524,7 @@
 #elif defined TORRENT_ANDROID && __ANDROID_API__ >= 24
 		ec = boost::asio::error::operation_not_supported;
 #else
-#error "don't know how to enumerate network routes on this platform"
+//#error "don't know how to enumerate network routes on this platform"
 #endif
 		return ret;
 	}
