--- readline-5.2/doc/Makefile.in	2004-01-08 07:42:54.000000000 -0800
+++ solaris_nv/readline-5.2/doc/Makefile.in	2007-11-01 03:18:18.154665000 -0700
@@ -47,7 +47,11 @@
 BUILD_DIR = @BUILD_DIR@
 TEXINPUTDIR = $(srcdir)
 
-MAKEINFO    = LANGUAGE= makeinfo
+# modification for solaris nv sfw-workspace nightly build
+# MAKEINFO    = LANGUAGE= makeinfo
+MAKEINFO    = makeinfo
+# end of modification
+
 TEXI2DVI    = $(srcdir)/texi2dvi
 TEXI2HTML   = $(srcdir)/texi2html
 QUIETPS     = #set this to -q to shut up dvips
