--- readline-5.2/readline.h.orig	Wed Nov 30 06:37:51 2011
+++ readline-5.2/readline.h	Wed Nov 30 06:38:26 2011
@@ -375,7 +375,7 @@
 #if defined (USE_VARARGS) && defined (PREFER_STDARG)
 extern int rl_message (const char *, ...)  __attribute__((__format__ (printf, 1, 2)));
 #else
-extern int rl_message ();
+extern int rl_message PARAMS((const char *, ...));
 #endif
 
 extern int rl_show_char PARAMS((int));
