--- pyOpenSSL-0.11/setup.py.orig	Mon Feb 14 23:17:02 2011
+++ pyOpenSSL-0.11/setup.py	Mon Feb 14 23:17:31 2011
@@ -203,7 +203,6 @@
                      'OpenSSL.test.test_crypto',
                      'OpenSSL.test.test_rand',
                      'OpenSSL.test.test_ssl'],
-      zip_safe = False,
       cmdclass = {"build_ext": BuildExtension},
       description = 'Python wrapper module around the OpenSSL library',
       author = 'Martin Sjögren, AB Strakt',
