https://github.com/davidhalter/jedi/pull/1996

--- jedi-0.19.1/test/test_utils.py.orig
+++ jedi-0.19.1/test/test_utils.py
@@ -12,8 +12,8 @@
     class NameSpace(object):
         pass
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
 
         self.namespace = self.NameSpace()
         utils.setup_readline(self.namespace)
