--- avant-window-navigator-0.3.2.1/data/awn-applets-migration-orig	2009-12-01 12:30:52.044855776 -0600
+++ avant-window-navigator-0.3.2.1/data/awn-applets-migration	2009-12-01 12:31:10.679206816 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 #  Copyright (C) 2009 Julien Lavergne <julien.lavergne@gmail.com>
 #
--- avant-window-navigator-0.3.2.1/data/awn-schema-to-gconf-orig	2009-12-01 12:31:51.791787050 -0600
+++ avant-window-navigator-0.3.2.1/data/awn-schema-to-gconf	2009-12-01 12:31:58.313503408 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 #  Copyright (C) 2007 Mark Lee <avant-wn@lazymalevolence.com>
 #
--- avant-window-navigator-0.3.2.1/awn-manager/awnManager.py.in-orig	2009-12-01 12:32:39.168247434 -0600
+++ avant-window-navigator-0.3.2.1/awn-manager/awnManager.py.in	2009-12-01 12:32:45.953740703 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 #
 #  Copyright (C) 2007 Neil Jagdish Patel <njpatel@gmail.com>
 #
