diff -up avant-window-navigator-0.3.2.1/libawn/awn-settings.c.orig avant-window-navigator-0.3.2.1/libawn/awn-settings.c
--- avant-window-navigator-0.3.2.1/libawn/awn-settings.c.orig	2009-05-02 07:05:22.000000000 +0100
+++ avant-window-navigator-0.3.2.1/libawn/awn-settings.c	2009-10-09 12:36:50.464770620 +0100
@@ -139,7 +139,7 @@ awn_settings_new()
 
   AwnSettings *s = NULL;
 
-  s = g_new(AwnSettings, 1);
+  s = g_new0(AwnSettings, 1);
 
   settings = s;
 

