--- Makefile.am.orig	Tue Feb 10 13:53:43 2009
+++ Makefile.am	Fri Jan 16 12:42:53 2009
@@ -15,13 +15,13 @@
 iftop_SOURCES = addr_hash.c edline.c hash.c iftop.c ns_hash.c \
                  options.c resolver.c screenfilter.c serv_hash.c \
                  sorted_list.c threadprof.c ui.c util.c \
-		 addrs_ioctl.c addrs_dlpi.c dlcommon.c \
+		 addrs_ioctl.c addrs_dlpi.c \
 		 stringmap.c cfgfile.c vector.c
 
 noinst_HEADERS = addr_hash.h ether.h ethertype.h extract.h hash.h iftop.h \
                  integers.h ip.h llc.h ns_hash.h options.h resolver.h \
                  screenfilter.h serv_hash.h sll.h sorted_list.h tcp.h \
-                 threadprof.h token.h ui.h dlcommon.h stringmap.h \
+                 threadprof.h token.h ui.h stringmap.h \
 		 vector.h cfgfile.h ppp.h
 
 man_MANS = iftop.8
