--- Makefile.in.orig	Tue Feb 10 13:54:06 2009
+++ Makefile.in	Fri Jan 16 12:42:53 2009
@@ -99,7 +99,7 @@
 iftop_SOURCES = addr_hash.c edline.c hash.c iftop.c ns_hash.c \
                  options.c resolver.c screenfilter.c serv_hash.c \
                  sorted_list.c threadprof.c ui.c util.c \
-		 addrs_ioctl.c addrs_dlpi.c dlcommon.c \
+		 addrs_ioctl.c addrs_dlpi.c \
 		 stringmap.c cfgfile.c vector.c
 
 
@@ -106,7 +106,7 @@
 noinst_HEADERS = addr_hash.h ether.h ethertype.h extract.h hash.h iftop.h \
                  integers.h ip.h llc.h ns_hash.h options.h resolver.h \
                  screenfilter.h serv_hash.h sll.h sorted_list.h tcp.h \
-                 threadprof.h token.h ui.h dlcommon.h stringmap.h \
+                 threadprof.h token.h ui.h stringmap.h \
 		 vector.h cfgfile.h ppp.h
 
 
@@ -134,7 +134,7 @@
 	resolver.$(OBJEXT) screenfilter.$(OBJEXT) serv_hash.$(OBJEXT) \
 	sorted_list.$(OBJEXT) threadprof.$(OBJEXT) ui.$(OBJEXT) \
 	util.$(OBJEXT) addrs_ioctl.$(OBJEXT) addrs_dlpi.$(OBJEXT) \
-	dlcommon.$(OBJEXT) stringmap.$(OBJEXT) cfgfile.$(OBJEXT) \
+	stringmap.$(OBJEXT) cfgfile.$(OBJEXT) \
 	vector.$(OBJEXT)
 iftop_OBJECTS = $(am_iftop_OBJECTS)
 iftop_LDADD = $(LDADD)
@@ -151,7 +151,7 @@
 @AMDEP_TRUE@DEP_FILES = ./$(DEPDIR)/addr_hash.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/addrs_dlpi.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/addrs_ioctl.Po ./$(DEPDIR)/cfgfile.Po \
-@AMDEP_TRUE@	./$(DEPDIR)/dlcommon.Po ./$(DEPDIR)/edline.Po \
+@AMDEP_TRUE@	./$(DEPDIR)/edline.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/hash.Po ./$(DEPDIR)/iftop.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/ns_hash.Po ./$(DEPDIR)/options.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/resolver.Po ./$(DEPDIR)/screenfilter.Po \
@@ -257,7 +257,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/addrs_dlpi.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/addrs_ioctl.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cfgfile.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/dlcommon.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/edline.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hash.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/iftop.Po@am__quote@
