--- graphviz-2.50.0/cmd/gvpr/Makefile.am.orig	2022-02-02 16:03:31.089546040 +0000
+++ graphviz-2.50.0/cmd/gvpr/Makefile.am	2022-02-02 16:03:50.555085542 +0000
@@ -33,7 +33,8 @@
 gvpr_CPPFLAGS = $(AM_CPPFLAGS)
 gvpr_LDADD = $(top_builddir)/lib/gvpr/libgvpr.la \
 	$(top_builddir)/lib/cgraph/libcgraph.la \
-	$(top_builddir)/lib/cdt/libcdt.la
+	$(top_builddir)/lib/cdt/libcdt.la \
+	-lsocket
 
 if ENABLE_MAN_PDFS
 if HAVE_PS2PDF
