--- graphviz-2.28.0/tclpkg/Makefile.in.~1~	2014-04-24 15:50:09.733932907 +0400
+++ graphviz-2.28.0/tclpkg/Makefile.in	2014-04-24 15:51:30.344564076 +0400
@@ -976,6 +976,7 @@
 @WITH_TCL_TRUE@	-mkdir -p $(DESTDIR)@TCL_INSTALL_DIR@;
 @WITH_TCL_TRUE@	if test -w $(DESTDIR)@TCL_INSTALL_DIR@/; then \
 @WITH_TCL_TRUE@		(cd $(DESTDIR)@TCL_INSTALL_DIR@; \
+@WITH_TCL_TRUE@			rm -f @PACKAGE_NAME@; \
 @WITH_TCL_TRUE@			$(LN_S) -f $(pkgtcldir) @PACKAGE_NAME@;) \
 @WITH_TCL_TRUE@	else \
 @WITH_TCL_TRUE@		echo "Warning: @TCL_INSTALL_DIR@ is not writable."; \
