--- graphviz-2.38.0/tclpkg/Makefile.in.1	2016-05-10 11:48:29.633603016 +0300
+++ graphviz-2.38.0/tclpkg/Makefile.in	2016-05-10 11:50:36.617293727 +0300
@@ -1010,6 +1010,7 @@
 @WITH_TCL_TRUE@	-mkdir -p $(DESTDIR)@TCL_INSTALL_DIR@;
 @WITH_TCL_TRUE@	if test -w $(DESTDIR)@TCL_INSTALL_DIR@/; then \
 @WITH_TCL_TRUE@		(cd $(DESTDIR)@TCL_INSTALL_DIR@; \
+@WITH_TCL_TRUE@			rm -f @PACKAGE_NAME@; \
 @WITH_TCL_TRUE@			cp -rf $(DESTDIR)$(pkgtcldir) @PACKAGE_NAME@;) \
 @WITH_TCL_TRUE@	else \
 @WITH_TCL_TRUE@		echo "Warning: @TCL_INSTALL_DIR@ is not writable."; \
