--- graphviz-2.28.0/libltdl/config/ltmain.sh.~1~	2014-01-16 16:59:51.219232636 +0400
+++ graphviz-2.28.0/libltdl/config/ltmain.sh	2014-01-16 17:03:05.408070755 +0400
@@ -4763,9 +4763,10 @@
       # compiler args for GCC
       # -F/path gives path to uninstalled frameworks, gcc on darwin
       # -p, -pg, --coverage, -fprofile-* pass through profiling flag for GCC
+      # -fstack-protector*   stack protector flags for GCC
       # @file GCC response files
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
-      -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*)
+      -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|-fstack-protector*|@*)
         func_quote_for_eval "$arg"
 	arg="$func_quote_for_eval_result"
         func_append compile_command " $arg"
