--- graphviz-2.38.0/configure.ac.1	2016-12-02 08:24:36.817166614 +0300
+++ graphviz-2.38.0/configure.ac	2016-12-02 08:24:52.526198058 +0300
@@ -1570,7 +1570,7 @@
           RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb vendorarchdir`"
 	fi
       ],[
-        PKG_CHECK_MODULES([RUBY], [ruby-1.9],[
+        PKG_CHECK_MODULES([RUBY], [ruby-2.3],[
 	  AC_CHECK_PROG(RUBY,ruby,ruby)
           if test "x$RUBY" = "x"; then
             use_ruby="No (ruby not available)"
