--- graphviz-2.38.0/tclpkg/gv/Makefile.am.~1~	2016-05-09 19:07:19.568875688 +0300
+++ graphviz-2.38.0/tclpkg/gv/Makefile.am	2016-05-09 19:08:39.066873834 +0300
@@ -145,7 +145,7 @@
 pkgperldir = $(pkglibdir)/perl
 PERL        ?= perl
 # cleanup for opencsw's perl which adds flags that g++ doesn't like
-PERL_CC      = `$(PERL) -MExtUtils::Embed -e ccopts | $(PERL) -pe 's/-x.*? //g'`
+PERL_CC      = `$(PERL) -MExtUtils::Embed -e ccopts | $(PERL) -pe 's/-x.*? //g' | sed -e 's:-fstack-protector-strong::'`
 PERL_LD      = `$(PERL) -MExtUtils::Embed -e ldopts`
 VENDOR_ARCH  = `$(PERL) -MConfig -e 'print $$Config{installvendorarch};'`
 PERL_data = gv.pm
--- graphviz-2.38.0/tclpkg/gv/Makefile.in.~1~	2016-05-09 19:07:19.571479596 +0300
+++ graphviz-2.38.0/tclpkg/gv/Makefile.in	2016-05-09 19:09:07.591366517 +0300
@@ -957,7 +957,7 @@
 libgv_ocaml_la_CPPFLAGS = $(BASECPPFLAGS) $(OCAML_INCLUDES)
 pkgperldir = $(pkglibdir)/perl
 # cleanup for opencsw's perl which adds flags that g++ doesn't like
-PERL_CC = `$(PERL) -MExtUtils::Embed -e ccopts | $(PERL) -pe 's/-x.*? //g'`
+PERL_CC = `$(PERL) -MExtUtils::Embed -e ccopts | $(PERL) -pe 's/-x.*? //g' |sed -e 's:-fstack-protector-strong::'`
 PERL_LD = `$(PERL) -MExtUtils::Embed -e ldopts`
 VENDOR_ARCH = `$(PERL) -MConfig -e 'print $$Config{installvendorarch};'`
 PERL_data = gv.pm
