To use psinfo_t structure, Tracker needs to also include procfs.h on
Solaris.

Need to file upstream bug

--- tracker-1.4.0/src/tracker/tracker-process.c 2015-01-26 15:58:43.000000000 -0800
+++ tracker-1.4.0/src/tracker/tracker-process.c     2015-07-08 18:28:17.031064501 -0700
@@ -28,6 +28,10 @@
 
 #include "tracker-process.h"
 
+#ifdef __sun /* Solaris */
+#include <sys/procfs.h>
+#endif
+
 static TrackerProcessData *
 process_data_new (gchar *cmd, pid_t pid)
 {
