Don't encode build host information in header files

--- Botan-1.10.9/src/build-data/buildh.in.1	2016-04-06 14:01:29.482555225 +0300
+++ Botan-1.10.9/src/build-data/buildh.in	2016-04-06 14:01:38.601538346 +0300
@@ -2,16 +2,6 @@
 #ifndef BOTAN_BUILD_CONFIG_H__
 #define BOTAN_BUILD_CONFIG_H__
 
-/*
-* This file was automatically generated %{timestamp} UTC by
-* %{user}@%{hostname} running '%{command_line}'
-*
-* Target
-*  - Compiler: %{cc} %{lib_opt} %{mach_opt}
-*  - Arch: %{submodel}/%{arch}
-*  - OS: %{os}
-*/
-
 #define BOTAN_VERSION_MAJOR %{version_major}
 #define BOTAN_VERSION_MINOR %{version_minor}
 #define BOTAN_VERSION_PATCH %{version_patch}
