--- inkscape-INKSCAPE_1_0/src/widgets/desktop-widget.cpp.old	2020-05-06 10:36:38.687041267 +0000
+++ inkscape-INKSCAPE_1_0/src/widgets/desktop-widget.cpp	2020-05-06 10:34:38.886377238 +0000
@@ -578,7 +578,7 @@
 
 
     // Zoom status spinbutton ---------------
-    auto zoom_adj = Gtk::Adjustment::create(100.0, log(SP_DESKTOP_ZOOM_MIN)/log(2), log(SP_DESKTOP_ZOOM_MAX)/log(2), 0.1);
+    auto zoom_adj = Gtk::Adjustment::create(100.0, log(SP_DESKTOP_ZOOM_MIN)/log(2.), log(SP_DESKTOP_ZOOM_MAX)/log(2.), 0.1);
     dtw->_zoom_status = Gtk::manage(new Inkscape::UI::Widget::SpinButton(zoom_adj));
 
     dtw->_zoom_status->set_defocus_widget(dtw->_canvas);
@@ -2050,7 +2050,7 @@
         correction = prefs->getDouble("/options/zoomcorrection/value", 1.0);
     }
     _zoom_status_value_changed_connection.block();
-    _zoom_status->set_value(log(desktop->current_zoom() / correction) / log(2));
+    _zoom_status->set_value(log(desktop->current_zoom() / correction) / log(2.));
     _zoom_status->queue_draw();
     _zoom_status_value_changed_connection.unblock();
 }
