--- inkscape-0.91/src/widgets/desktop-widget.cpp.orig	2016-08-16 09:54:24.538266549 +0200
+++ inkscape-0.91/src/widgets/desktop-widget.cpp	2016-08-16 09:55:11.412038849 +0200
@@ -627,7 +627,7 @@
     gtk_box_pack_end (GTK_BOX (dtw->statusbar), statusbar_tail, FALSE, FALSE, 0);
 
     // zoom status spinbutton
-    dtw->zoom_status = gtk_spin_button_new_with_range (log(SP_DESKTOP_ZOOM_MIN)/log(2), log(SP_DESKTOP_ZOOM_MAX)/log(2), 0.1);
+    dtw->zoom_status = gtk_spin_button_new_with_range (log(SP_DESKTOP_ZOOM_MIN)/log(2.), log(SP_DESKTOP_ZOOM_MAX)/log(2.), 0.1);
     gtk_widget_set_tooltip_text (dtw->zoom_status, _("Zoom"));
     gtk_widget_set_size_request (dtw->zoom_status, STATUS_ZOOM_WIDTH, -1);
     gtk_entry_set_width_chars (GTK_ENTRY (dtw->zoom_status), 6);
@@ -1832,7 +1832,7 @@
 static gdouble
 sp_dtw_zoom_display_to_value (gdouble value)
 {
-    return  log (value / 100.0) / log (2);
+    return  log (value / 100.0) / log (2.);
 }
 
 static gint
@@ -2026,7 +2026,7 @@
     GdkWindow *window = gtk_widget_get_window(GTK_WIDGET(dtw->zoom_status));
 
     g_signal_handlers_block_by_func (G_OBJECT (dtw->zoom_status), (gpointer)G_CALLBACK (sp_dtw_zoom_value_changed), dtw);
-    gtk_spin_button_set_value (GTK_SPIN_BUTTON (dtw->zoom_status), log(dtw->desktop->current_zoom()) / log(2));
+    gtk_spin_button_set_value (GTK_SPIN_BUTTON (dtw->zoom_status), log(dtw->desktop->current_zoom()) / log(2.));
     gtk_widget_queue_draw(GTK_WIDGET(dtw->zoom_status));
     if (window)
         gdk_window_process_updates(window, TRUE);
