--- ./src/live_effects/lpe-lattice.cpp.orig	2015-07-20 03:31:32.150515939 +0200
+++ ./src/live_effects/lpe-lattice.cpp	2015-07-20 03:39:10.713309563 +0200
@@ -150,7 +150,7 @@
                         // corner = actual corner of the rectangle
                         // origin = Upperleft point
                         double dl = dot((*handles[corner+4*i] - (base + origin)), dir)/dot(dir,dir);
-                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0,ui+vi);
+                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0, (int) (ui+vi));
                     }
                 }
             }
--- inkscape-0.92.0.orig/src/live_effects/lpe-ellipse_5pts.cpp	2017-01-23 06:02:12.270692447 +0100
+++ inkscape-0.92.0/src/live_effects/lpe-ellipse_5pts.cpp	2017-01-23 06:23:39.021893455 +0100
@@ -171,14 +171,14 @@
 
     // figure out if we have a slice, guarding against rounding errors
 
-    Path p(Geom::Point(cos(0), sin(0)));
+    Path p(Geom::Point(cos(0.), sin(0.)));
 
     double end = 2 * M_PI;
     for (s = 0; s < end; s += M_PI_2) {
         e = s + M_PI_2;
         if (e > end)
             e = end;
-        len = 4*tan((e - s)/4)/3;
+        len = 4.*tan((e - s)/4.)/3.;
         x0 = cos(s);
         y0 = sin(s);
         x1 = x0 + len * cos(s + M_PI_2);
--- inkscape-0.92.0.orig/src/live_effects/lpe-lattice2.cpp	2017-01-23 06:02:12.272268271 +0100
+++ inkscape-0.92.0/src/live_effects/lpe-lattice2.cpp	2017-01-23 06:25:42.837760506 +0100
@@ -197,7 +197,7 @@
                         // corner = actual corner of the rectangle
                         // origin = Upperleft point
                         double dl = dot((handles[corner+4*i] - (base + origin)), dir)/dot(dir,dir);
-                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0,ui+vi);
+                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0, int(ui+vi));
                     }
                 }
             }
