--- inkscape-0.91/src/2geom/math-utils.h.orig	2016-08-16 10:39:57.225272482 +0200
+++ inkscape-0.91/src/2geom/math-utils.h	2016-08-16 10:52:08.150424165 +0200
@@ -112,7 +112,7 @@
 #elif defined(isnan) || defined(__FreeBSD__) || defined(__osf__)
 # define IS_NAN(_a) (isnan(_a))		/* GNU definition */
 #elif defined (SOLARIS)
-# define IS_NAN(_a) (isnan(_a))		/* GNU definition */
+# define IS_NAN(_a) (std::isnan(_a))		/* GNU definition */
 #else
 # define IS_NAN(_a) (boost::math::isnan(_a))
 #endif
@@ -129,9 +129,9 @@
 # define IS_FINITE(_a) (isfinite(_a))
 #elif defined(__osf__)
 # define IS_FINITE(_a) (finite(_a) && !IS_NAN(_a))
-#elif defined (SOLARIS)
-#include  <ieeefp.h>
-#define IS_FINITE(_a) (finite(_a) && !IS_NAN(_a))
+//#elif defined (SOLARIS)
+//#include  <ieeefp.h>
+//#define IS_FINITE(_a) (finite(_a) && !IS_NAN(_a))
 #else
 # define IS_FINITE(_a) (boost::math::isfinite(_a))
 #endif
--- inkscape-0.91/src/isinf.h.orig	2016-08-16 10:41:14.870539944 +0200
+++ inkscape-0.91/src/isinf.h	2016-08-16 10:48:33.703505113 +0200
@@ -5,12 +5,12 @@
  * Fix for missing std::isnormal with SOLARIS8/GCC3.2
  */
  
-#if defined (SOLARIS)
-
-	#include <ieeefp.h>
-	#define isinf(x) ((fpclass(x) == FP_NINF) || (fpclass(x) == FP_PINF))
-	
-#elif defined(__APPLE__) && __GNUC__ == 3
+//#if defined (SOLARIS)
+//
+//#include <ieeefp.h>
+//#define isinf(x) ((fpclass(x) == FP_NINF) || (fpclass(x) == FP_PINF))
+//	
+#if defined(__APPLE__) && __GNUC__ == 3
 #define isinf(x) __isinf(x)
 #elif __cplusplus >= 201103L
 # include <cmath>
