--- inkscape-1.2.2_2022-12-01_b0a8486541/src/extension/implementation/script.cpp.old	2023-01-03 17:06:12.313343728 +0000
+++ inkscape-1.2.2_2022-12-01_b0a8486541/src/extension/implementation/script.cpp	2023-01-03 17:06:55.713358104 +0000
@@ -86,7 +86,7 @@
     { "python",  {"python-interpreter",  {"python3"           }}},
 #else
     { "perl",    {"perl-interpreter",    {"perl"              }}},
-    { "python",  {"python-interpreter",  {"python3", "python" }}},
+    { "python",  {"python-interpreter",  {"/usr/bin/python3.9", "python" }}},
 #endif
     { "python2", {"python2-interpreter", {"python2", "python" }}},
     { "ruby",    {"ruby-interpreter",    {"ruby"    }}},
