--- inkscape-0.91/src/sp-ellipse.cpp.orig	2016-08-16 09:59:53.804374055 +0200
+++ inkscape-0.91/src/sp-ellipse.cpp	2016-08-16 10:00:36.358170158 +0200
@@ -203,7 +203,7 @@
 
         double const dx = viewbox.width();
         double const dy = viewbox.height();
-        double const dr = hypot(dx, dy) / sqrt(2);
+        double const dr = hypot(dx, dy) / sqrt(2.);
         double const em = this->style->font_size.computed;
         double const ex = em * 0.5; // fixme: get from pango or libnrtype
 
