--- ./src/svg/svg-length.cpp.orig	2015-07-20 04:50:16.246770036 +0200
+++ ./src/svg/svg-length.cpp	2015-07-20 04:51:31.174044537 +0200
@@ -114,13 +114,13 @@
     /* Determine the actual number of fractional digits */
     fprec = MAX(static_cast<int>(fprec), static_cast<int>(tprec) - idigits);
     /* Round value */
-    val += 0.5 / pow(10.0, fprec);
+    val += 0.5 / pow(10.0, static_cast<int>(fprec));
     /* Extract integral and fractional parts */
     double dival = floor(val);
     double fval = val - dival;
     /* Write integra */
     if (idigits > (int)tprec) {
-        i += sp_svg_number_write_ui(buf + i, (unsigned int)floor(dival/pow(10.0, idigits-tprec) + .5));
+        i += sp_svg_number_write_ui(buf + i, (unsigned int)floor(dival/pow(10.0, static_cast<double>(idigits-tprec)) + .5));
         for(unsigned int j=0; j<(unsigned int)idigits-tprec; j++) {
             buf[i+j] = '0';
         }
