--- inkscape-0.92.0.orig/src/gradient-drag.cpp	2017-01-23 06:02:12.237411642 +0100
+++ inkscape-0.92.0/src/gradient-drag.cpp	2017-01-23 06:15:04.967239803 +0100
@@ -1246,7 +1246,7 @@
     int knot_size = sizeTable[size - 1];
     if(knot->shape == SP_KNOT_SHAPE_TRIANGLE){
         knot_size *= 2.2;
-        knot_size = floor(knot_size);
+        knot_size = floor(double(knot_size));
         if ( knot_size % 2 == 0 ){
             knot_size += 1;
         }
