--- ./configure.ac.orig	2015-07-29 23:41:25.551865325 +0200
+++ ./configure.ac	2015-07-29 23:43:48.416230649 +0200
@@ -489,8 +489,10 @@
     if test "x${have_lcms2}" = "xyes"; then
         LIBS="$LIBS $LCMS2_LIBS"
         AC_DEFINE(HAVE_LIBLCMS2, 1, [define to 1 if you have lcms version 2.x])
-	AC_SUBST(LCMS2_CFLAGS)
-	AC_SUBST(LCMS2_LIBS)
+				AC_SUBST(LCMS2_CFLAGS)
+				AC_SUBST(LCMS2_LIBS)
+				LCMS_CFLAGS="$LCMS2_CFLAGS"
+				LCMS__LIBS="$LCMS2_LIBS"
     else
         PKG_CHECK_MODULES(LCMS, lcms >= 1.13, lcms=yes, lcms=no)
         if test "x$lcms" = "xyes"; then
