--- inkscape-1.2_2022-05-15_dc2aedaf03/src/CMakeLists.txt.old	2022-05-17 13:06:41.531571345 +0000
+++ inkscape-1.2_2022-05-15_dc2aedaf03/src/CMakeLists.txt	2022-05-17 13:07:25.568459413 +0000
@@ -402,13 +402,6 @@
 # Build everything except main and inkview.c in a shared library.
 add_library(inkscape_base ${inkscape_SRC} ${sp_SRC})
 
-# added so that locally installed inkscape can find lib2geom if it's not installed systemwide
-if(NOT APPLE)
-    set_target_properties(inkscape_base PROPERTIES INSTALL_RPATH "$ORIGIN/..")
-else()
-    set_target_properties(inkscape_base PROPERTIES INSTALL_RPATH "@loader_path/..")
-endif()
-
 if(WITH_INTERNAL_CAIRO)
 add_dependencies(inkscape_base cairo_ext)
 endif()
