--- inkscape-1.4_2024-10-09_e7c3feb100/src/path-prefix.cpp.orig
+++ inkscape-1.4_2024-10-09_e7c3feb100/src/path-prefix.cpp
@@ -218,6 +218,12 @@
         if (!program_name) {
             g_warning("get_program_name() - g_file_read_link failed");
         }
+#elif defined(__sun)
+        const char *exec_name = getexecname();
+        program_name = (gchar *) exec_name;
+        if (!program_name) {
+            g_warning("get_program_name() - getexecname failed");
+        }
 #elif defined(__NetBSD__)
         static const int name[] = {CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME};
         char path[MAXPATHLEN];
