From 63e0e35a346a66b7260909cf859b3396c280a5a2 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Wed, 28 Apr 2021 11:40:25 +0000
Subject: Add __illumos__ preprocessor definition

---
 gcc/config/sol2.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/sol2.h b/gcc/config/sol2.h
index 72b41ec5e1a..a945e96d4a2 100644
--- a/gcc/config/sol2.h
+++ b/gcc/config/sol2.h
@@ -85,6 +85,7 @@ along with GCC; see the file COPYING3.  If not see
   do {							\
     builtin_define_std ("unix");			\
     builtin_define_std ("sun");				\
+    builtin_define ("__illumos__");			\
     builtin_define ("__svr4__");			\
     builtin_define ("__SVR4");				\
     builtin_assert ("system=unix");			\
