--- libdrm-2.4.58/exynos/exynos_drm.c.backup	2015-01-08 18:34:51.809200717 +0300
+++ libdrm-2.4.58/exynos/exynos_drm.c	2015-01-08 18:35:10.234657246 +0300
@@ -34,7 +34,7 @@
 #include <errno.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 
--- libdrm-2.4.58/exynos/exynos_fimg2d.c.backup	2015-01-08 19:21:22.648231644 +0300
+++ libdrm-2.4.58/exynos/exynos_fimg2d.c	2015-01-08 19:21:42.357339447 +0300
@@ -20,7 +20,7 @@
 #include <errno.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 
--- libdrm-2.4.58/tests/exynos/exynos_fimg2d_test.c.backup	2015-01-08 19:23:24.788944043 +0300
+++ libdrm-2.4.58/tests/exynos/exynos_fimg2d_test.c	2015-01-08 19:23:44.048826244 +0300
@@ -20,7 +20,7 @@
 #include <errno.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 #include <xf86drmMode.h>
