--- libdrm-2.4.70/freedreno/kgsl/kgsl_bo.c.orig	2016-09-14 01:56:02.447145494 +0200
+++ libdrm-2.4.70/freedreno/kgsl/kgsl_bo.c	2016-09-14 01:57:13.859602672 +0200
@@ -32,7 +32,12 @@
 
 #include "kgsl_priv.h"
 
+#if defined(__sun) && defined(__SVR4)
+#include <xorg/xorg-server.h>
+#include <xorg/fb.h>
+#else
 #include <linux/fb.h>
+#endif
 
 static int set_memtype(struct fd_device *dev, uint32_t handle, uint32_t flags)
 {
