diff -ur libdrm-2.4.58-pc/exynos/libdrm_exynos.pc.in libdrm-2.4.58/exynos/libdrm_exynos.pc.in
--- libdrm-2.4.58-pc/exynos/libdrm_exynos.pc.in	2013-02-14 19:14:08.000000000 +0400
+++ libdrm-2.4.58/exynos/libdrm_exynos.pc.in	2015-01-16 11:06:53.912309735 +0300
@@ -6,6 +6,6 @@
 Name: libdrm_exynos
 Description: Userspace interface to exynos kernel DRM services
 Version: 0.7
-Libs: -L${libdir} -ldrm_exynos
+Libs: -L${libdir} -R${libdir} -ldrm_exynos
 Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/exynos
 Requires.private: libdrm
diff -ur libdrm-2.4.58-pc/freedreno/libdrm_freedreno.pc.in libdrm-2.4.58/freedreno/libdrm_freedreno.pc.in
--- libdrm-2.4.58-pc/freedreno/libdrm_freedreno.pc.in	2013-10-09 20:03:23.000000000 +0400
+++ libdrm-2.4.58/freedreno/libdrm_freedreno.pc.in	2015-01-16 11:06:53.934385038 +0300
@@ -6,6 +6,6 @@
 Name: libdrm_freedreno
 Description: Userspace interface to freedreno kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_freedreno
+Libs: -L${libdir} -R${libdir} -ldrm_freedreno
 Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/freedreno
 Requires.private: libdrm
diff -ur libdrm-2.4.58-pc/intel/libdrm_intel.pc.in libdrm-2.4.58/intel/libdrm_intel.pc.in
--- libdrm-2.4.58-pc/intel/libdrm_intel.pc.in	2013-10-09 20:03:25.000000000 +0400
+++ libdrm-2.4.58/intel/libdrm_intel.pc.in	2015-01-16 11:06:53.922157974 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to intel kernel DRM services
 Version: @PACKAGE_VERSION@
 Requires: libdrm
-Libs: -L${libdir} -ldrm_intel
+Libs: -L${libdir} -R${libdir} -ldrm_intel
 Cflags: -I${includedir} -I${includedir}/libdrm
diff -ur libdrm-2.4.58-pc/libdrm.pc.in libdrm-2.4.58/libdrm.pc.in
--- libdrm-2.4.58-pc/libdrm.pc.in	2013-02-14 19:14:08.000000000 +0400
+++ libdrm-2.4.58/libdrm.pc.in	2015-01-16 11:06:53.919163639 +0300
@@ -6,5 +6,5 @@
 Name: libdrm
 Description: Userspace interface to kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm
+Libs: -L${libdir} -R${libdir} -ldrm
 Cflags: -I${includedir} -I${includedir}/libdrm
diff -ur libdrm-2.4.58-pc/libkms/libkms.pc.in libdrm-2.4.58/libkms/libkms.pc.in
--- libdrm-2.4.58-pc/libkms/libkms.pc.in	2013-02-14 19:14:08.000000000 +0400
+++ libdrm-2.4.58/libkms/libkms.pc.in	2015-01-16 11:06:53.916371045 +0300
@@ -6,5 +6,5 @@
 Name: libkms
 Description: Library that abstract aways the different mm interface for kernel drivers
 Version: 1.0.0
-Libs: -L${libdir} -lkms
+Libs: -L${libdir} -R${libdir} -lkms
 Cflags: -I${includedir}/libkms
diff -ur libdrm-2.4.58-pc/nouveau/libdrm_nouveau.pc.in libdrm-2.4.58/nouveau/libdrm_nouveau.pc.in
--- libdrm-2.4.58-pc/nouveau/libdrm_nouveau.pc.in	2013-10-09 20:03:25.000000000 +0400
+++ libdrm-2.4.58/nouveau/libdrm_nouveau.pc.in	2015-01-16 11:06:53.928345775 +0300
@@ -6,6 +6,6 @@
 Name: libdrm_nouveau
 Description: Userspace interface to nouveau kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_nouveau
+Libs: -L${libdir} -R${libdir} -ldrm_nouveau
 Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/libdrm/nouveau
 Requires.private: libdrm
diff -ur libdrm-2.4.58-pc/omap/libdrm_omap.pc.in libdrm-2.4.58/omap/libdrm_omap.pc.in
--- libdrm-2.4.58-pc/omap/libdrm_omap.pc.in	2013-02-14 19:14:08.000000000 +0400
+++ libdrm-2.4.58/omap/libdrm_omap.pc.in	2015-01-16 11:06:53.925430710 +0300
@@ -6,6 +6,6 @@
 Name: libdrm_omap
 Description: Userspace interface to omap kernel DRM services
 Version: 0.6
-Libs: -L${libdir} -ldrm_omap
+Libs: -L${libdir} -R${libdir} -ldrm_omap
 Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/omap
 Requires.private: libdrm
diff -ur libdrm-2.4.58-pc/radeon/libdrm_radeon.pc.in libdrm-2.4.58/radeon/libdrm_radeon.pc.in
--- libdrm-2.4.58-pc/radeon/libdrm_radeon.pc.in	2013-02-14 19:14:08.000000000 +0400
+++ libdrm-2.4.58/radeon/libdrm_radeon.pc.in	2015-01-16 11:06:53.931197646 +0300
@@ -6,5 +6,5 @@
 Name: libdrm_radeon
 Description: Userspace interface to kernel DRM services for radeon
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_radeon
+Libs: -L${libdir} -R${libdir} -ldrm_radeon
 Cflags: -I${includedir} -I${includedir}/libdrm
--- libdrm-2.4.64/tegra/libdrm_tegra.pc.in.~1~	2015-09-14 15:55:25.984198527 +0300
+++ libdrm-2.4.64/tegra/libdrm_tegra.pc.in	2015-09-14 15:55:39.369838244 +0300
@@ -6,6 +6,6 @@
 Name: libdrm_tegra
 Description: Userspace interface to Tegra kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_tegra
+Libs: -L${libdir} -R${libdir} -ldrm_tegra
 Cflags: -I${includedir} -I${includedir}/libdrm
 Requires.private: libdrm
--- libdrm-2.4.64/amdgpu/libdrm_amdgpu.pc.in.~1~	2015-09-14 15:54:50.636837960 +0300
+++ libdrm-2.4.64/amdgpu/libdrm_amdgpu.pc.in	2015-09-14 15:55:02.641516915 +0300
@@ -6,5 +6,5 @@
 Name: libdrm_amdgpu
 Description: Userspace interface to kernel DRM services for amdgpu
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_amdgpu
+Libs: -L${libdir} -R${libdir} -ldrm_amdgpu
 Cflags: -I${includedir} -I${includedir}/libdrm
