diff -wpruN '--exclude=*.orig' a~/mesonbuild/compilers/compilers.py a/mesonbuild/compilers/compilers.py
--- a~/mesonbuild/compilers/compilers.py	1970-01-01 00:00:00
+++ a/mesonbuild/compilers/compilers.py	1970-01-01 00:00:00
@@ -1039,7 +1039,7 @@ class Compiler(HoldableObject, metaclass
 
     def get_largefile_args(self) -> T.List[str]:
         '''Enable transparent large-file-support for 32-bit UNIX systems'''
-        if not (self.get_argument_syntax() == 'msvc' or self.info.is_darwin()):
+        if not (self.get_argument_syntax() == 'msvc' or self.info.is_darwin() or self.info.is_sunos()):
             # Enable large-file support unconditionally on all platforms other
             # than macOS and MSVC. macOS is now 64-bit-only so it doesn't
             # need anything special, and MSVC doesn't have automatic LFS.
diff -wpruN '--exclude=*.orig' a~/mesonbuild/linkers/detect.py a/mesonbuild/linkers/detect.py
--- a~/mesonbuild/linkers/detect.py	1970-01-01 00:00:00
+++ a/mesonbuild/linkers/detect.py	1970-01-01 00:00:00
@@ -44,7 +44,7 @@ if T.TYPE_CHECKING:
     from ..mesonlib import MachineChoice
 
 defaults: T.Dict[str, T.List[str]] = {}
-defaults['static_linker'] = ['ar', 'gar']
+defaults['static_linker'] = ['gar', 'ar']
 defaults['vs_static_linker'] = ['lib']
 defaults['clang_cl_static_linker'] = ['llvm-lib']
 defaults['cuda_static_linker'] = ['nvlink']
