--- mesonbuild/environment.py.orig	2019-08-08 15:14:14.707828920 +0000
+++ mesonbuild/environment.py	2019-08-08 15:18:59.055235264 +0000
@@ -242,7 +242,7 @@
                 trial = 'ppc64'
         if 'powerpc64' in stdo:
             trial = 'ppc64'
-    elif trial in ('amd64', 'x64'):
+    elif trial in ('amd64', 'x64', 'i86pc'):
         trial = 'x86_64'
 
     # On Linux (and maybe others) there can be any mixture of 32/64 bit code in
--- mesonbuild/envconfig.py.org	2019-08-08 15:20:34.110643703 +0000
+++ mesonbuild/envconfig.py	2019-08-08 15:21:34.932065388 +0000
@@ -41,6 +41,7 @@
     'arm',
     'e2k',
     'ia64',
+    'i386pc',
     'mips',
     'mips64',
     'parisc',
