--- "meson-1.6.0/test cases/frameworks/28 gir link order 2/meson.build.orig"
+++ "meson-1.6.0/test cases/frameworks/28 gir link order 2/meson.build"
@@ -1,5 +1,7 @@
 project('gir link order 2', 'c')
 
+error('MESON_SKIP_TEST https://github.com/mesonbuild/meson/issues/5915')
+
 if not dependency('gobject-2.0', required : false).found() or not find_program('g-ir-scanner', required: false).found()
   error('MESON_SKIP_TEST gobject not found.')
 endif
