--- mesonbuild/compilers/compilers.py	2020-01-07 22:52:53.923018916 +0000
+++ mesonbuild/compilers/compilers.py.new	2020-01-08 08:34:45.398904282 +0000
@@ -1113,7 +1113,7 @@ def get_largefile_args(compiler):
     '''
     Enable transparent large-file-support for 32-bit UNIX systems
     '''
-    if not (compiler.info.is_windows() or compiler.info.is_darwin()):
+    if not (compiler.info.is_windows() or compiler.info.is_darwin() or compiler.info.is_sunos()):
         # Enable large-file support unconditionally on all platforms other
         # than macOS and Windows. macOS is now 64-bit-only so it doesn't
         # need anything special, and Windows doesn't have automatic LFS.
