diff -Nru libayatana-indicator-0.9.3.orig/src/CMakeLists.txt libayatana-indicator-0.9.3/src/CMakeLists.txt
--- libayatana-indicator-0.9.3.orig/src/CMakeLists.txt	2022-10-26 00:08:17.000000000 +0000
+++ libayatana-indicator-0.9.3/src/CMakeLists.txt	2023-01-07 19:07:12.206431827 +0000
@@ -139,7 +139,7 @@
 target_include_directories("${ayatana_indicator_gtkver}" PUBLIC ${PROJECT_DEPS_INCLUDE_DIRS})
 target_include_directories("${ayatana_indicator_gtkver}" PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories("${ayatana_indicator_gtkver}" PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
-target_link_options("${ayatana_indicator_gtkver}" PRIVATE "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/indicator.symbols")
+target_link_options("${ayatana_indicator_gtkver}" PRIVATE "-W")
 target_link_libraries("${ayatana_indicator_gtkver}" ${PROJECT_DEPS_LIBRARIES} ${EXTRA_LIBS})
 add_dependencies("${ayatana_indicator_gtkver}" "src-generated")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/lib${ayatana_indicator_gtkver}.so" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
