--- MEGAsync_ori/src/MEGASync/mega/configure.ac	Wed Mar  9 01:46:53 2022
+++ MEGAsync_patched/src/MEGASync/mega/configure.ac	Mon Feb 28 22:19:04 2022
@@ -612,26 +612,26 @@
    *)
 
     # determine if library is installed
-    if test -d "$with_pcre/lib64"; then
-        LDFLAGS="-L$with_pcre/lib64 $LDFLAGS"
-        CXXFLAGS="-I$with_pcre/include $CXXFLAGS"
-        CPPFLAGS="-I$with_pcre/include $CPPFLAGS"
+    if test -d "$with_pcre/lib/amd64"; then
+        LDFLAGS="-L$with_pcre/lib/amd64 $LDFLAGS"
+        CXXFLAGS="-I$with_pcre/include/pcre $CXXFLAGS"
+        CPPFLAGS="-I$with_pcre/include/pcre $CPPFLAGS"
 
         AC_CHECK_HEADERS(pcrecpp.h,
-         PCRE_CXXFLAGS="-I$with_pcre/include"
-         PCRE_CPPFLAGS="-I$with_pcre/include"
-         PCRE_LDFLAGS="-L$with_pcre/lib64",
+         PCRE_CXXFLAGS="-I$with_pcre/include/pcre"
+         PCRE_CPPFLAGS="-I$with_pcre/include/pcre"
+         PCRE_LDFLAGS="-L$with_pcre/lib/amd64",
          AC_MSG_ERROR([pcrecpp.h header not found or not usable])
          )
     else
-        LDFLAGS="-L$with_pcre/lib32 $LDFLAGS"
-        CXXFLAGS="-I$with_pcre/include $CXXFLAGS"
-        CPPFLAGS="-I$with_pcre/include $CPPFLAGS"
+        LDFLAGS="-L$with_pcre/lib $LDFLAGS"
+        CXXFLAGS="-I$with_pcre/include/pcre $CXXFLAGS"
+        CPPFLAGS="-I$with_pcre/include/pcre $CPPFLAGS"
 
         AC_CHECK_HEADERS(pcrecpp.h,
-         PCRE_CXXFLAGS="-I$with_pcre/include"
-         PCRE_CPPFLAGS="-I$with_pcre/include"
-         PCRE_LDFLAGS="-L$with_pcre/lib32",
+         PCRE_CXXFLAGS="-I$with_pcre/include/pcre"
+         PCRE_CPPFLAGS="-I$with_pcre/include/pcre"
+         PCRE_LDFLAGS="-L$with_pcre/lib",
          AC_MSG_ERROR([pcrecpp.h header not found or not usable])
          )
     fi
@@ -1652,13 +1652,13 @@
    *)
 
     # determine if library is installed
-    if test -d "$with_freeimage/lib"; then
-        LDFLAGS="-L$with_freeimage/lib $LDFLAGS"
+    if test -d "$with_freeimage/lib/amd64"; then
+        LDFLAGS="-L$with_freeimage/lib/amd64 $LDFLAGS"
         CXXFLAGS="-I$with_freeimage/include $CXXFLAGS"
         CPPFLAGS="-I$with_freeimage/include $CPPFLAGS"
 
         AC_CHECK_HEADERS([FreeImage.h],[
-         FI_LDFLAGS="-L$with_freeimage/lib"
+         FI_LDFLAGS="-L$with_freeimage/lib/amd64"
          FI_CXXFLAGS="-I$with_freeimage/include"
          FI_CPPFLAGS="-I$with_freeimage/include"],
          AC_MSG_ERROR([FreeImage.h header not found or not usable])
