--- MEGAsync_ori/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.h	Wed Jan 26 20:25:17 2022
+++ MEGAsync_patched/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.h	Sat Mar  5 15:56:15 2022
@@ -36,6 +36,8 @@
 #define TARGET_CPU_SPARC 1
 #elif defined(i386) || defined(__i386)
 #define TARGET_CPU_X86 1
+#elif defined(amd64) || defined(__amd64)
+#define TARGET_CPU_X86_64 1
 #else
 #error "cannot determine cpu type"
 #endif
