We have no html5lib packaged yet.

--- soupsieve-2.8.4/pyproject.toml.orig
+++ soupsieve-2.8.4/pyproject.toml
@@ -112,17 +112,6 @@
     py{39,310,311,312,313,314},
     lint, nolxml, nohtml5lib
 
-[testenv]
-passenv = *
-deps =
-    -rrequirements/tests.txt
-commands =
-    mypy
-    pytest --cov soupsieve --cov-append {toxinidir}
-    coverage html -d {envtmpdir}/coverage
-    coverage xml
-    coverage report --show-missing
-
 [testenv:documents]
 passenv = *
 deps =
@@ -145,7 +134,7 @@
 commands =
     pytest {toxinidir}
 
-[testenv:nohtml5lib]
+[testenv]
 passenv = *
 deps =
     -rrequirements/tests-nohtml5lib.txt
