https://github.com/ssimms/pdfapi2/issues/93

--- PDF-API2-2.048/LICENSE.orig
+++ PDF-API2-2.048/LICENSE
@@ -9,7 +9,7 @@
 
   (The master copy of this license lives on the GNU website.)
 
-Copyright (C) 1991, 1999 Free Software Foundation, Inc. 59
+Copyright (C) 1991, 1999 Free Software Foundation, Inc.
 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 Everyone is permitted to copy and distribute verbatim copies
