diff -ruN libgnomeui-2.24.2.orig/doc/reference/Makefile.am libgnomeui-2.24.2/doc/reference/Makefile.am
--- libgnomeui-2.24.2.orig/doc/reference/Makefile.am	2010-01-04 21:01:03.916799751 +0000
+++ libgnomeui-2.24.2/doc/reference/Makefile.am	2010-01-04 21:02:13.347390505 +0000
@@ -74,7 +74,7 @@
 	$(top_srcdir)/libgnomeui/pixmaps/stock_volume.png
 
 # Extra options to supply to gtkdoc-fixref.
-FIXXREF_OPTIONS=
+FIXXREF_OPTIONS=--module=$(DOC_MODULE)
 
 include $(top_srcdir)/gtk-doc.make
 
