--- pulseaudio-1.0/src/pulsecore/core-scache.h-orig	2011-10-19 09:56:30.443291065 -0500
+++ pulseaudio-1.0/src/pulsecore/core-scache.h	2011-10-19 09:57:15.863329411 -0500
@@ -27,8 +27,6 @@
 #include <pulsecore/memchunk.h>
 #include <pulsecore/sink.h>
 
-#define PA_SCACHE_ENTRY_SIZE_MAX (1024*1024*16)
-
 typedef struct pa_scache_entry {
     uint32_t index;
     pa_core *core;
--- pulseaudio-1.0/src/pulsecore/memchunk.h-orig	2011-10-19 09:56:45.049056686 -0500
+++ pulseaudio-1.0/src/pulsecore/memchunk.h	2011-10-19 09:56:59.243495913 -0500
@@ -53,4 +53,6 @@ pa_memchunk* pa_memchunk_memcpy(pa_memch
 /* Return TRUE if any field is set != 0 */
 pa_bool_t pa_memchunk_isset(pa_memchunk *c);
 
+#define PA_SCACHE_ENTRY_SIZE_MAX (1024*1024*16)
+
 #endif
--- pulseaudio-1.0/src/pulsecore/pstream.c-orig	2011-10-19 09:56:00.968603148 -0500
+++ pulseaudio-1.0/src/pulsecore/pstream.c	2011-10-19 09:57:49.036969685 -0500
@@ -37,7 +37,7 @@
 #include <pulsecore/socket.h>
 #include <pulsecore/queue.h>
 #include <pulsecore/log.h>
-#include <pulsecore/core-scache.h>
+#include <pulsecore/memchunk.h>
 #include <pulsecore/creds.h>
 #include <pulsecore/refcnt.h>
 #include <pulsecore/flist.h>
--- pulseaudio-1.0/src/Makefile.am-orig	2011-10-19 10:26:08.909354110 -0500
+++ pulseaudio-1.0/src/Makefile.am	2011-10-19 10:26:21.545547580 -0500
@@ -510,7 +510,7 @@ usergroup_test_CFLAGS = $(AM_CFLAGS)
 usergroup_test_LDFLAGS = $(AM_LDFLAGS) $(BINLDFLAGS)
 
 connect_stress_SOURCES = tests/connect-stress.c
-connect_stress_LDADD = $(AM_LDADD) libpulse.la libpulsecommon-@PA_MAJORMINOR@.la
+connect_stress_LDADD = $(AM_LDADD) libpulse.la libpulsecommon-@PA_MAJORMINOR@.la libpulsecore-@PA_MAJORMINOR@.la
 connect_stress_CFLAGS = $(AM_CFLAGS)
 connect_stress_LDFLAGS = $(AM_LDFLAGS) $(BINLDFLAGS)
 
