--- pulseaudio-1.1/src/daemon/default.pa.in-orig	2012-05-17 01:30:56.827201316 -0500
+++ pulseaudio-1.1/src/daemon/default.pa.in	2012-05-17 01:31:12.110991968 -0500
@@ -52,7 +52,7 @@ ifelse(@HAVE_ALSA@, 1, [dnl
 #load-module module-alsa-source device=hw:1,0
 ])dnl
 ifelse(@HAVE_OSS_OUTPUT@, 1, [dnl
-#load-module module-oss device="/dev/dsp" sink_name=output source_name=input
+load-module module-oss device="/dev/dsp" sink_name=output source_name=input
 #load-module module-oss-mmap device="/dev/dsp" sink_name=output source_name=input
 ])dnl
 ifelse(@HAVE_WAVEOUT@, 1, [dnl
@@ -120,11 +120,11 @@ ifelse(@OS_IS_WIN32@, 0, [dnl
 ### Load additional modules from GConf settings. This can be configured with the paprefs tool.
 ### Please keep in mind that the modules configured by paprefs might conflict with manually
 ### loaded modules.
-.ifexists module-gconf@PA_SOEXT@
-.nofail
-load-module module-gconf
-.fail
-.endif
+#.ifexists module-gconf@PA_SOEXT@
+#.nofail
+#load-module module-gconf
+#.fail
+#.endif
 
 ### Automatically restore the default sink/source when changed by the user
 ### during runtime
