--- pulseaudio-1.0/src/pulsecore/core-util.c-orig	2011-10-06 19:40:11.876688365 -0500
+++ pulseaudio-1.0/src/pulsecore/core-util.c	2011-10-06 19:42:47.201806884 -0500
@@ -3195,7 +3195,7 @@ char *pa_read_line_from_file(const char 
 
 pa_bool_t pa_running_in_vm(void) {
 
-#if defined(__i386__) || defined(__x86_64__)
+#if defined(__i386__)
 
     /* Both CPUID and DMI are x86 specific interfaces... */
 
--- pulseaudio-1.0/src/pulsecore/remap_mmx.c-orig	2011-10-17 22:11:30.204585354 -0500
+++ pulseaudio-1.0/src/pulsecore/remap_mmx.c	2011-10-17 22:17:22.993499444 -0500
@@ -100,7 +100,7 @@
                 "4:                             \n\t"  \
                 " emms                          \n\t"
 
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 static void remap_mono_to_stereo_mmx(pa_remap_t *m, void *dst, const void *src, unsigned n) {
     pa_reg_x86 temp, temp2;
 
@@ -144,10 +144,10 @@ static void init_remap_mmx(pa_remap_t *m
         pa_log_info("Using MMX mono to stereo remapping");
     }
 }
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 
 void pa_remap_func_init_mmx(pa_cpu_x86_flag_t flags) {
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 
     if (flags & PA_CPU_X86_MMX) {
         pa_log_info("Initialising MMX optimized remappers.");
@@ -155,5 +155,5 @@ void pa_remap_func_init_mmx(pa_cpu_x86_f
         pa_set_init_remap_func((pa_init_remap_func_t) init_remap_mmx);
     }
 
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 }
--- pulseaudio-1.0/src/pulsecore/remap_sse.c-orig	2011-10-17 22:17:37.212080000 -0500
+++ pulseaudio-1.0/src/pulsecore/remap_sse.c	2011-10-17 22:17:52.347912472 -0500
@@ -99,7 +99,7 @@
                 " jne 3b                        \n\t"  \
                 "4:                             \n\t"
 
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 static void remap_mono_to_stereo_sse2(pa_remap_t *m, void *dst, const void *src, unsigned n) {
     pa_reg_x86 temp, temp2;
 
@@ -143,15 +143,15 @@ static void init_remap_sse2(pa_remap_t *
         pa_log_info("Using SSE mono to stereo remapping");
     }
 }
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 
 void pa_remap_func_init_sse(pa_cpu_x86_flag_t flags) {
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 
     if (flags & PA_CPU_X86_SSE2) {
         pa_log_info("Initialising SSE2 optimized remappers.");
         pa_set_init_remap_func ((pa_init_remap_func_t) init_remap_sse2);
     }
 
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 }
--- pulseaudio-1.0/src/pulsecore/svolume_sse.c-orig	2011-10-17 22:18:54.325624955 -0500
+++ pulseaudio-1.0/src/pulsecore/svolume_sse.c	2011-10-17 22:19:08.744843997 -0500
@@ -34,7 +34,7 @@
 
 #include "sample-util.h"
 
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 
 #define VOLUME_32x16(s,v)                  /* .. |   vh  |   vl  | */                   \
       " pxor %%xmm4, %%xmm4          \n\t" /* .. |    0  |    0  | */                   \
@@ -332,10 +332,10 @@ static void run_test(void) {
     pa_assert_se(memcmp(samples_ref, samples, sizeof(samples)) == 0);
 }
 #endif
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 
 void pa_volume_func_init_sse(pa_cpu_x86_flag_t flags) {
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 
 #ifdef RUN_TEST
     run_test();
@@ -347,5 +347,5 @@ void pa_volume_func_init_sse(pa_cpu_x86_
         pa_set_volume_func(PA_SAMPLE_S16NE, (pa_do_volume_func_t) pa_volume_s16ne_sse2);
         pa_set_volume_func(PA_SAMPLE_S16RE, (pa_do_volume_func_t) pa_volume_s16re_sse2);
     }
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 }
--- pulseaudio-1.0/src/pulsecore/svolume_mmx.c-orig	2011-10-17 22:18:23.485381654 -0500
+++ pulseaudio-1.0/src/pulsecore/svolume_mmx.c	2011-10-17 22:18:39.050384384 -0500
@@ -34,7 +34,7 @@
 
 #include "sample-util.h"
 
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 /* in s: 2 int16_t samples
  * in v: 2 int32_t volumes, fixed point 16:16
  * out s: contains scaled and clamped int16_t samples.
@@ -324,11 +324,11 @@ static void run_test(void) {
 }
 #endif
 
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 
 
 void pa_volume_func_init_mmx(pa_cpu_x86_flag_t flags) {
-#if defined (__i386__) || defined (__amd64__)
+#if defined (__i386__)
 
 #ifdef RUN_TEST
     run_test();
@@ -340,5 +340,5 @@ void pa_volume_func_init_mmx(pa_cpu_x86_
         pa_set_volume_func(PA_SAMPLE_S16NE, (pa_do_volume_func_t) pa_volume_s16ne_mmx);
         pa_set_volume_func(PA_SAMPLE_S16RE, (pa_do_volume_func_t) pa_volume_s16re_mmx);
     }
-#endif /* defined (__i386__) || defined (__amd64__) */
+#endif /* defined (__i386__) */
 }
