--- upower-0.99.4/src/up-daemon.c.1	2017-12-28 14:28:27.778699691 +0000
+++ upower-0.99.4/src/up-daemon.c	2017-12-28 14:28:47.012134228 +0000
@@ -435,7 +435,7 @@
 			       GDBusMethodInvocation *invocation,
 			       UpDaemon *daemon)
 {
-	up_exported_daemon_complete_get_display_device (skeleton, invocation,
+	up_exported_daemon_complete_get_critical_action (skeleton, invocation,
 							up_backend_get_critical_action (daemon->priv->backend));
 	return TRUE;
 }
