--- opensmalltalk-vm-sun-v5.0.8/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sat Jul  4 10:53:04 2020
+++ p0/opensmalltalk-vm-sun-v5.0.8/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sat Jul  4 15:06:45 2020
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202007040853-cog $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202007040853-cog $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
