--- opensmalltalk-vm-sun-v5.0.7/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sat May  2 12:55:37 2020
+++ p0/opensmalltalk-vm-sun-v5.0.7/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Tue Jun 16 07:01:52 2020
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202005021055-cog $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202005021055-cog $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
