--- opensmalltalk-vm-sun-v5.0.7/platforms/unix/config/Makefile.install	Sat May  2 12:55:37 2020
+++ p0/opensmalltalk-vm-sun-v5.0.7/platforms/unix/config/Makefile.install	Sun Jul 12 17:11:41 2020
@@ -63,7 +63,6 @@
 	  echo $(INSTALL_DATA) $(topdir)/platforms/unix/doc/$$f $(ROOT)$(docdir); \
 	  $(INSTALL_DATA) $(topdir)/platforms/unix/doc/$$f $(ROOT)$(docdir); \
 	done
-	gzip -f9 $(ROOT)$(docdir)/*
 	$(MKINSTALLDIRS) $(ROOT)$(mandir)/man1
 	$(INSTALL_DATA) squeak.1 $(ROOT)$(mandir)/man1
 	rm -f $(ROOT)$(mandir)/man1/inisqueak.1 $(ROOT)$(mandir)/man1/inisqueak.1.gz
