--- opensmalltalk-vm-sun-v5.0.29/platforms/unix/vm/sqUnixSpurMemory.c	2021-05-05 11:00:40.000000000 +0200
+++ p0/opensmalltalk-vm-sun-v5.0.29/platforms/unix/vm/sqUnixSpurMemory.c	2021-05-21 17:27:04.202017773 +0200
@@ -103,6 +103,8 @@
 usqInt
 sqAllocateMemory(usqInt minHeapSize, usqInt desiredHeapSize)
 {
+	int i,n=1000;
+	char **ptr;
 	char *hint, *address, *alloc;
     unsigned long alignment;
     sqInt allocBytes;
@@ -114,6 +116,20 @@
 	pageSize = getpagesize();
 	pageMask = ~(pageSize - 1);
 
+	/* malloc workaround - before calling sbrk create a malloc freelist */
+	ptr=malloc(sizeof(char*) * (n+1));
+	if (ptr == NULL) {
+		fprintf(stderr, "sqAllocateMemory: malloc failure\n");
+	} else {
+		for(i=0;i<n;i++) {
+			ptr[i]=malloc(pageSize);
+		}
+		for(i=0;i<n;i++) {
+			if (ptr[i]) free(ptr[i]);
+		}
+		free(ptr);
+	}
+
 	hint = sbrk(0);
 
 	alignment = max(pageSize,1024*1024);
