--- java-atk-wrapper-0.33.2/wrapper/Makefile.am.1	2017-03-01 11:35:54.119651372 +0300
+++ java-atk-wrapper-0.33.2/wrapper/Makefile.am	2017-03-01 11:39:55.806727712 +0300
@@ -3,13 +3,8 @@
 ALL_CLASSES=org/GNOME/Accessibility/*.class
 DEP_CLASSES=$(wildcard $(ALL_CLASSES))

-if USER
 java_atk_wrapperdir = $(JAVA_HOME)/lib/ext
 propertiesdir = $(JAVA_HOME)/lib
-else
-java_atk_wrapperdir = $(HOME)/.local/share/java-atk-wrapper$(JAVA_HOME)/lib/ext
-propertiesdir = $(HOME)/.local/share/java-atk-wrapper$(JAVA_HOME)/lib
-endif

 java_atk_wrapper_DATA = $(JARFILES)
 properties_DATA = accessibility.properties
