--- java-atk-wrapper-0.33.2/wrapper/org/GNOME/Accessibility/AtkWrapper.java.1	2017-03-01 12:43:27.924538332 +0300
+++ java-atk-wrapper-0.33.2/wrapper/org/GNOME/Accessibility/AtkWrapper.java	2017-03-01 12:43:58.810222727 +0300
@@ -31,7 +31,7 @@
   static boolean accessibilityEnabled = false;
   static {
     try {
-      Process p = Runtime.getRuntime().exec("/opt/X11/bin/xprop -root");
+      Process p = Runtime.getRuntime().exec("/usr/bin/xprop -root");
       BufferedReader b = new BufferedReader(new InputStreamReader (p.getInputStream ()));
       String result;
       while ((result = b.readLine()) != null) {
