--- cmake-3.30.1/Modules/CMakeDetermineFortranCompiler.cmake.orig
+++ cmake-3.30.1/Modules/CMakeDetermineFortranCompiler.cmake
@@ -13,7 +13,7 @@
 include(Platform/${CMAKE_SYSTEM_NAME}-Determine-Fortran OPTIONAL)
 include(Platform/${CMAKE_SYSTEM_NAME}-Fortran OPTIONAL)
 if(NOT CMAKE_Fortran_COMPILER_NAMES)
-  set(CMAKE_Fortran_COMPILER_NAMES f95)
+  set(CMAKE_Fortran_COMPILER_NAMES gfortran)
 endif()
 
 if(${CMAKE_GENERATOR} MATCHES "Visual Studio")
