--- cmake-3.11.3/Modules/CMakeDetermineFortranCompiler.cmake.orig	2018-06-04 20:51:30.573604966 +0000
+++ cmake-3.11.3/Modules/CMakeDetermineFortranCompiler.cmake	2018-06-04 20:57:50.257210543 +0000
@@ -13,7 +13,7 @@
 include(Platform/${CMAKE_SYSTEM_NAME}-Determine-Fortran OPTIONAL)
 include(Platform/${CMAKE_SYSTEM_NAME}-Fortran OPTIONAL)
 if(NOT CMAKE_Fortran_COMPILER_NAMES)
-  set(CMAKE_Fortran_COMPILER_NAMES f95)
+  set(CMAKE_Fortran_COMPILER_NAMES gfortran)
 endif()
 
 if(${CMAKE_GENERATOR} MATCHES "Visual Studio")
