--- cmake-3.9.1/Utilities/cmlibuv/src/unix/sunos.c.orig	2017-09-07 16:03:58.671432822 +0000
+++ cmake-3.9.1/Utilities/cmlibuv/src/unix/sunos.c	2017-09-07 16:05:57.026055449 +0000
@@ -751,7 +751,8 @@
     return 1;
   if (ent->ifa_addr == NULL)
     return 1;
-  if (ent->ifa_addr->sa_family == PF_PACKET)
+  if (ent->ifa_addr->sa_family != AF_INET &&
+      ent->ifa_addr->sa_family != AF_INET6)
     return 1;
   return 0;
 }
