--- cmake-3.11.3/Modules/FindMPI.cmake.orig	2018-06-04 19:13:02.572174189 +0000
+++ cmake-3.11.3/Modules/FindMPI.cmake	2018-06-04 19:36:02.929469382 +0000
@@ -1247,6 +1247,9 @@
   # FreeBSD ships mpich under the normal system paths - but available openmpi implementations
   # will be found in /usr/local/mpi/<name>
   MPI_search_mpi_prefix_folder("/usr/local/mpi")
+elseif("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "SunOS")
+  # OpenIndiana ships MPI in subdirectories
+  MPI_search_mpi_prefix_folder("/usr/lib/64/mpich/gcc")
 endif()
 
 # Most MPI distributions have some form of mpiexec or mpirun which gives us something we can look for.
--- cmake-3.11.3/Modules/CMakeDetermineJavaCompiler.cmake.orig	2018-06-04 19:13:02.561764067 +0000
+++ cmake-3.11.3/Modules/CMakeDetermineJavaCompiler.cmake	2018-06-04 19:16:19.645223465 +0000
@@ -43,18 +43,9 @@
     "[HKEY_LOCAL_MACHINE\\SOFTWARE\\JavaSoft\\Java Development Kit\\1.3;JavaHome]/bin"
     $ENV{JAVA_HOME}/bin
     /usr/bin
-    /usr/lib/java/bin
-    /usr/share/java/bin
-    /usr/local/bin
-    /usr/local/java/bin
-    /usr/local/java/share/bin
-    /usr/java/j2sdk1.4.2_04
-    /usr/lib/j2sdk1.4-sun/bin
-    /usr/java/j2sdk1.4.2_09/bin
-    /usr/lib/j2sdk1.5-sun/bin
-    /opt/sun-jdk-1.5.0.04/bin
-    /usr/local/jdk-1.7.0/bin
-    /usr/local/jdk-1.6.0/bin
+	/usr/jdk/latest/bin
+	/usr/jdk/openjdk1.8.0/bin
+	/usr/jdk/openjdk1.7.0/bin
     )
   # if no compiler has been specified yet, then look for one
   if(CMAKE_Java_COMPILER_INIT)
--- cmake-3.11.3/Modules/FindJava.cmake.orig	2018-06-04 19:13:02.566853874 +0000
+++ cmake-3.11.3/Modules/FindJava.cmake	2018-06-04 19:23:26.243740666 +0000
@@ -128,17 +128,10 @@
 # Hard-coded guesses should still go in PATHS. This ensures that the user
 # environment can always override hard guesses.
 set(_JAVA_PATHS
-  /usr/lib/java/bin
-  /usr/share/java/bin
-  /usr/local/java/bin
-  /usr/local/java/share/bin
-  /usr/java/j2sdk1.4.2_04
-  /usr/lib/j2sdk1.4-sun/bin
-  /usr/java/j2sdk1.4.2_09/bin
-  /usr/lib/j2sdk1.5-sun/bin
-  /opt/sun-jdk-1.5.0.04/bin
-  /usr/local/jdk-1.7.0/bin
-  /usr/local/jdk-1.6.0/bin
+  /usr/bin
+  /usr/jdk/latest/bin
+  /usr/jdk/openjdk1.8.0/bin
+  /usr/jdk/openjdk1.7.0/bin
   )
 find_program(Java_JAVA_EXECUTABLE
   NAMES java
--- cmake-3.11.3/Modules/FindIce.cmake.orig	2018-06-04 19:13:02.587162341 +0000
+++ cmake-3.11.3/Modules/FindIce.cmake	2018-06-04 19:19:29.168461709 +0000
@@ -408,7 +408,7 @@
   # In common use on Linux, MacOS X (homebrew) and FreeBSD; prefer
   # version-specific dir
   list(APPEND ice_slice_paths
-       /usr/local/share /usr/share)
+       /usr/share)
   list(APPEND ice_slice_suffixes
        "Ice-${Ice_VERSION_SLICE2CPP_FULL}/slice"
        "Ice-${Ice_VERSION_SLICE2CPP_SHORT}/slice"
--- cmake-3.11.3/Modules/CMakeGenericSystem.cmake.orig	2018-06-04 19:20:00.932406860 +0000
+++ cmake-3.11.3/Modules/CMakeGenericSystem.cmake	2018-06-04 19:20:17.905182457 +0000
@@ -163,7 +163,7 @@
 
 # Choose a default install prefix for this platform.
 if(CMAKE_HOST_UNIX)
-  set(CMAKE_INSTALL_PREFIX "/usr/local"
+  set(CMAKE_INSTALL_PREFIX "/usr"
     CACHE PATH "Install path prefix, prepended onto install directories.")
 else()
   GetDefaultWindowsPrefixBase(CMAKE_GENERIC_PROGRAM_FILES)
--- cmake-3.11.3/Modules/FindQt.cmake.orig	2018-06-04 19:13:02.531675148 +0000
+++ cmake-3.11.3/Modules/FindQt.cmake	2018-06-04 19:38:29.010903484 +0000
@@ -105,10 +105,8 @@
   "[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/include/Qt"
   ${qt_headers}/Qt
   $ENV{QTDIR}/include/Qt
+  /usr/lib/qt/4.8/include/Qt
   /usr/lib/qt/include/Qt
-  /usr/share/qt4/include/Qt
-  /usr/local/include/X11/qt4/Qt
-  C:/Progra~1/qt/include/Qt
   PATH_SUFFIXES qt/include/Qt include/Qt)
 
 if(QT4_QGLOBAL_H_FILE)
--- cmake-3.11.3/Modules/UseJava.cmake.orig	2018-06-04 19:13:02.577337570 +0000
+++ cmake-3.11.3/Modules/UseJava.cmake	2018-06-04 19:39:02.131449893 +0000
@@ -896,7 +896,6 @@
     set(_jar_versions)
     set(_jar_paths
         /usr/share/java/
-        /usr/local/share/java/
         ${Java_JAR_PATHS})
     set(_jar_doc "NOTSET")
 
--- cmake-3.25.0/Modules/FindGTK2.cmake.orig	2022-11-16 15:42:03.000000000 +0000
+++ cmake-3.25.0/Modules/FindGTK2.cmake	2022-11-17 11:10:21.464644803 +0000
@@ -301,21 +301,7 @@
     find_path(GTK2_${_var}_INCLUDE_DIR ${_hdr}
         PATHS
             ${_gtk2_arch_dir}
-            /usr/local/libx32
-            /usr/local/lib64
-            /usr/local/lib
-            /usr/libx32
-            /usr/lib64
             /usr/lib
-            /opt/gnome/include
-            /opt/gnome/lib
-            /opt/openwin/include
-            /usr/openwin/lib
-            /sw/lib
-            /opt/local/lib
-            /opt/homebrew/lib
-            /usr/pkg/lib
-            /usr/pkg/include/glib
             $ENV{GTKMM_BASEPATH}/include
             $ENV{GTKMM_BASEPATH}/lib
             [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]/include
--- cmake-3.26.0/Modules/CMakeDetermineSystem.cmake.orig	2023-03-14 15:04:14.000000000 +0100
+++ cmake-3.26.0/Modules/CMakeDetermineSystem.cmake	2023-03-19 19:28:16.553347338 +0100
@@ -33,7 +33,7 @@
 
 # find out on which system cmake runs
 if(CMAKE_HOST_UNIX)
-  find_program(CMAKE_UNAME NAMES uname PATHS /bin /usr/bin /usr/local/bin)
+  find_program(CMAKE_UNAME NAMES uname PATHS /bin /usr/bin)
   if(CMAKE_UNAME)
     if(CMAKE_HOST_SYSTEM_NAME STREQUAL "AIX")
       execute_process(COMMAND ${CMAKE_UNAME} -v
