The 'pkg' package exists.

--- importlib_metadata-8.4.0/tests/test_api.py.orig
+++ importlib_metadata-8.4.0/tests/test_api.py
@@ -50,7 +50,7 @@
                 assert distribution(name).metadata['Name'] == 'pkg.dot'
 
     def test_prefix_not_matched(self):
-        prefixes = 'p', 'pkg', 'pkg.'
+        prefixes = 'p', 'pkg.'
         for prefix in prefixes:
             with self.subTest(prefix):
                 with self.assertRaises(PackageNotFoundError):
