--- qt-everywhere-src-6.2.4/qtopcua/src/3rdparty/open62541/open62541.c	Fri Mar  4 15:17:58 2022
+++ qt-everywhere-src-6.2.4_patched/qtopcua/src/3rdparty/open62541/open62541.c	Sun Mar 27 06:41:32 2022
@@ -20,6 +20,9 @@
 # define MDNSD_DYNAMIC_LINKING
 #endif
 
+#include<netinet/in.h>
+#include<inttypes.h>
+
 /* Disable security warnings for BSD sockets on MSVC */
 #if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_WARNINGS)
 # define _CRT_SECURE_NO_WARNINGS
