--- qt-everywhere-src-6.9.3/qtquick3d/src/runtimerender/graphobjects/qssgrenderparticles.cpp.orig
+++ qt-everywhere-src-6.9.3/qtquick3d/src/runtimerender/graphobjects/qssgrenderparticles.cpp
@@ -29,7 +29,7 @@
     }
     int vec4PerParticle = ceilDivide(particleSize, 16);
     int vec4s = particleCount * vec4PerParticle;
-    int width = divisibleBy(std::sqrt(vec4s), vec4PerParticle);
+    int width = divisibleBy((int)std::sqrt((double)vec4s), vec4PerParticle);
     int height = ceilDivide(vec4s, width);
     m_particlesPerSlice = width / vec4PerParticle;
     m_particleCount = particleCount;
