--- qt-everywhere-src-6.9.3/qttools/src/linguist/lupdate/CMakeLists.txt.orig
+++ qt-everywhere-src-6.9.3/qttools/src/linguist/lupdate/CMakeLists.txt
@@ -11,6 +11,8 @@
     set_property(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" PROPERTY _qt_skip_separate_debug_info ON)
 endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -fno-rtti")
+
 #####################################################################
 ## lupdate Tool:
 #####################################################################
