--- qt-everywhere-src-6.2.4/qttools/src/linguist/lupdate/CMakeLists.txt	Fri Mar  4 15:17:55 2022
+++ qt-everywhere-src-6.2.4_patched/qttools/src/linguist/lupdate/CMakeLists.txt	Sun May 15 07:11:08 2022
@@ -9,6 +9,8 @@
 if (MINGW)
     set_property(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" PROPERTY _qt_skip_separate_debug_info ON)
 endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -fno-rtti")
 # special case end
 
 #####################################################################
