diff -Nru qt-everywhere-src-6.4.2.orig/qtbase/util/testrunner/qt-testrunner.py qt-everywhere-src-6.4.2/qtbase/util/testrunner/qt-testrunner.py
--- qt-everywhere-src-6.4.2.orig/qtbase/util/testrunner/qt-testrunner.py	2022-11-16 08:54:24.000000000 +0000
+++ qt-everywhere-src-6.4.2/qtbase/util/testrunner/qt-testrunner.py	2023-01-12 16:14:36.145376179 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Copyright (C) 2021 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR GPL-3.0-only WITH Qt-GPL-exception-1.0
 
diff -Nru qt-everywhere-src-6.5.1.orig/qtbase/util/testrunner/sanitizer-testrunner.py qt-everywhere-src-6.5.1/qtbase/util/testrunner/sanitizer-testrunner.py
--- qt-everywhere-src-6.5.1.orig/qtbase/util/testrunner/sanitizer-testrunner.py	2023-06-12 08:54:24.000000000 +0000
+++ qt-everywhere-src-6.5.1/qtbase/util/testrunner/sanitizer-testrunner.py	2023-06-12 16:14:36.145376179 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR GPL-3.0-only WITH Qt-GPL-exception-1.0
 
diff -Nru qt-everywhere-src-6.4.2.orig/qtbase/mkspecs/features/uikit/devices.py qt-everywhere-src-6.4.2/qtbase/mkspecs/features/uikit/devices.py
--- qt-everywhere-src-6.4.2.orig/qtbase/mkspecs/features/uikit/devices.py	2022-11-16 08:54:24.000000000 +0000
+++ qt-everywhere-src-6.4.2/qtbase/mkspecs/features/uikit/devices.py	2023-01-12 16:16:13.690349675 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Copyright (C) 2017 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
 
