diff -Nru qt-everywhere-src-6.4.2.orig/qtbase/util/testrunner/qt-testrunner.py qt-everywhere-src-6.4.2/qtbase/util/testrunner/qt-testrunner.py
--- qt-everywhere-src-6.4.2.orig/qtbase/util/testrunner/qt-testrunner.py        2022-11-16 08:54:24.000000000 +0000
+++ qt-everywhere-src-6.4.2/qtbase/util/testrunner/qt-testrunner.py     2023-01-12 16:14:36.145376179 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 
 
 #############################################################################