--- qt-everywhere-src-6.9.3/qtbase/src/corelib/plugin/qplugin.h.orig
+++ qt-everywhere-src-6.9.3/qtbase/src/corelib/plugin/qplugin.h
@@ -142,13 +142,6 @@
 // Since Qt 6.3
 template <auto (&PluginMetaData)> class QPluginMetaDataV2
 {
-    struct ElfNotePayload : QPluginMetaData::ElfNoteHeader {
-        static constexpr size_t HeaderOffset = offsetof(QPluginMetaData::ElfNoteHeader, header);
-        quint8 payload[sizeof(PluginMetaData)] = {};
-        constexpr ElfNotePayload() : ElfNoteHeader(sizeof(PluginMetaData))
-        { QPluginMetaData::copy(payload, PluginMetaData); }
-    };
-
     struct RegularPayload : QPluginMetaData::MagicHeader {
         static constexpr size_t HeaderOffset = offsetof(QPluginMetaData::MagicHeader, header);
         quint8 payload[sizeof(PluginMetaData)] = {};
@@ -165,16 +158,6 @@
 #if defined(QT_STATICPLUGIN)
 #  define QT_PLUGIN_METADATAV2_SECTION
     using Payload = StaticPayload;
-#elif defined(Q_OF_ELF)
-#  ifdef Q_CC_CLANG
-#    define QT_PLUGIN_METADATAV2_SECTION                                                            \
-            __attribute__((section(".note.qt.metadata"), used, aligned(alignof(void *)),            \
-                           no_sanitize("address")))
-#  else
-#    define QT_PLUGIN_METADATAV2_SECTION                                                            \
-            __attribute__((section(".note.qt.metadata"), used, aligned(alignof(void *))))
-#  endif
-    using Payload = ElfNotePayload;
 #else
 #  define QT_PLUGIN_METADATAV2_SECTION      QT_PLUGIN_METADATA_SECTION
     using Payload = RegularPayload;
