--- qt-everywhere-src-6.9.3/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp.orig
+++ qt-everywhere-src-6.9.3/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp
@@ -339,12 +339,12 @@
                             }
                         }
 
-                        if (Q_UNLIKELY(p1 < 0 || p1 > UINT16_MAX
-                                        || p2 < 0 || p2 > UINT16_MAX
-                                        || p3 < 0 || p3 > UINT16_MAX
-                                        || t1 < 0 || t1 > UINT16_MAX
-                                        || t2 < 0 || t2 > UINT16_MAX
-                                        || t3 < 0 || t3 > UINT16_MAX)) {
+                        if (Q_UNLIKELY(p1 < 0 || p1 > INT16_MAX
+                                        || p2 < 0 || p2 > INT16_MAX
+                                        || p3 < 0 || p3 > INT16_MAX
+                                        || t1 < 0 || t1 > INT16_MAX
+                                        || t2 < 0 || t2 > INT16_MAX
+                                        || t3 < 0 || t3 > INT16_MAX)) {
                             setLastError(UnsupportedIndexSizeError);
                             return;
                         }
@@ -377,7 +377,7 @@
                             }
                         }
 
-                        if (Q_UNLIKELY(p4 < 0 || p4 > UINT16_MAX || t4 < 0 || t4 > UINT16_MAX)) {
+                        if (Q_UNLIKELY(p4 < 0 || p4 > INT16_MAX || t4 < 0 || t4 > INT16_MAX)) {
                             setLastError(UnsupportedIndexSizeError);
                             return;
                         }
