diff -Nru qt-everywhere-src-6.3.2.orig/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp qt-everywhere-src-6.3.2/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp
--- qt-everywhere-src-6.3.2.orig/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp	2023-09-19 16:28:08.031043407 +0200
+++ qt-everywhere-src-6.3.2/qtdeclarative/src/labs/wavefrontmesh/qwavefrontmesh.cpp	2022-09-06 10:52:24.000000000 +0200
@@ -375,12 +375,12 @@
                             }
                         }
 
-                        if (Q_UNLIKELY(p1 < 0 || p1 > UINT16_MAX
-                                        || p2 < 0 || p2 > UINT16_MAX
-                                        || p3 < 0 || p3 > UINT16_MAX
-                                        || t1 < 0 || t1 > UINT16_MAX
-                                        || t2 < 0 || t2 > UINT16_MAX
-                                        || t3 < 0 || t3 > UINT16_MAX)) {
+                        if (Q_UNLIKELY(p1 < 0 || p1 > INT16_MAX
+                                        || p2 < 0 || p2 > INT16_MAX
+                                        || p3 < 0 || p3 > INT16_MAX
+                                        || t1 < 0 || t1 > INT16_MAX
+                                        || t2 < 0 || t2 > INT16_MAX
+                                        || t3 < 0 || t3 > INT16_MAX)) {
                             setLastError(UnsupportedIndexSizeError);
                             return;
                         }
@@ -413,7 +413,7 @@
                             }
                         }
 
-                        if (Q_UNLIKELY(p4 < 0 || p4 > UINT16_MAX || t4 < 0 || t4 > UINT16_MAX)) {
+                        if (Q_UNLIKELY(p4 < 0 || p4 > INT16_MAX || t4 < 0 || t4 > INT16_MAX)) {
                             setLastError(UnsupportedIndexSizeError);
                             return;
                         }
