diff -Nru qt-everywhere-src-6.4.2.orig/qtmultimedia/src/multimedia/configure.cmake qt-everywhere-src-6.4.2/qtmultimedia/src/multimedia/configure.cmake
--- qt-everywhere-src-6.4.2.orig/qtmultimedia/src/multimedia/configure.cmake	2022-12-12 20:36:40.000000000 +0000
+++ qt-everywhere-src-6.4.2/qtmultimedia/src/multimedia/configure.cmake	2023-01-13 11:41:44.337836000 +0000
@@ -73,7 +73,7 @@
     LABEL "FFmpeg"
     ENABLE INPUT_ffmpeg STREQUAL 'yes'
     DISABLE INPUT_ffmpeg STREQUAL 'no'
-    CONDITION FFmpeg_FOUND AND (APPLE OR WIN32 OR ANDROID OR QNX OR QT_FEATURE_pulseaudio)
+    CONDITION false
 )
 qt_feature("alsa" PUBLIC PRIVATE
     LABEL "ALSA (experimental)"
@@ -111,7 +111,7 @@
 )
 qt_feature("gstreamer_photography" PRIVATE
     LABEL "GStreamer Photography"
-    CONDITION ( QT_FEATURE_gstreamer_1_0 AND GStreamer_Photography_FOUND )
+    CONDITION false
 )
 qt_feature("gstreamer_gl" PRIVATE
     LABEL "GStreamer OpenGL"
