--- qt-everywhere-src-6.9.3/qtbase/src/corelib/global/qversiontagging.h.orig
+++ qt-everywhere-src-6.9.3/qtbase/src/corelib/global/qversiontagging.h
@@ -117,7 +117,9 @@
 #endif
 
 #ifdef Q_OF_ELF
-#  define QT_VERSION_TAG_SYMBOL(prefix, sym, m, n)      sym
+//#  define QT_VERSION_TAG_SYMBOL(prefix, sym, m, n)      sym
+#  define QT_VERSION_TAG_SYMBOL2(prefix, sym, m, n)     prefix ## sym ## _ ## m ## _ ## n
+#  define QT_VERSION_TAG_SYMBOL(prefix, sym, m, n)      QT_VERSION_TAG_SYMBOL2(prefix, sym, m, n)
 #else
 #  define QT_VERSION_TAG_SYMBOL2(prefix, sym, m, n)     prefix ## sym ## _ ## m ## _ ## n
 #  define QT_VERSION_TAG_SYMBOL(prefix, sym, m, n)      QT_VERSION_TAG_SYMBOL2(prefix, sym, m, n)
