diff -Nru qt-everywhere-src-6.4.2.orig/qtmultimedia/cmake/FindFFmpeg.cmake qt-everywhere-src-6.4.2/qtmultimedia/cmake/FindFFmpeg.cmake
--- qt-everywhere-src-6.4.2.orig/qtmultimedia/cmake/FindFFmpeg.cmake	2022-12-12 20:36:40.000000000 +0000
+++ qt-everywhere-src-6.4.2/qtmultimedia/cmake/FindFFmpeg.cmake	2023-01-12 13:37:20.345900539 +0000
@@ -248,7 +248,7 @@
         endif()
         target_link_libraries(FFmpeg::${_lowerComponent} INTERFACE "${${_component}_LIBRARY}")
         if (UNIX AND NOT APPLE)
-          target_link_options(FFmpeg::${_lowerComponent} INTERFACE  "-Wl,--exclude-libs=lib${_lowerComponent}")
+          target_link_options(FFmpeg::${_lowerComponent} INTERFACE  "-W")
         endif ()
     endif()
     else()
