diff --git a/qtmultimedia/cmake/FindFFmpeg.cmake b/qtmultimedia/cmake/FindFFmpeg.cmake
index decac97..f9286ef 100644
--- a/qtmultimedia/cmake/FindFFmpeg.cmake
+++ b/qtmultimedia/cmake/FindFFmpeg.cmake
@@ -334,7 +334,7 @@ foreach (_component ${FFmpeg_FIND_COMPONENTS})
 
             __ffmpeg_internal_set_dependencies(${_component})
             if (UNIX AND NOT APPLE)
-                target_link_options(${_target} INTERFACE  "-Wl,--exclude-libs=lib${_lowerComponent}")
+                target_link_options(${_target} INTERFACE  "-W")
             endif ()
         endif()
     endif()
