--- a/qtmultimedia/src/multimedia/configure.cmake
+++ b/qtmultimedia/src/multimedia/configure.cmake
@@ -236,9 +236,7 @@ qt_feature("ffmpeg" PRIVATE
     LABEL "FFmpeg"
     ENABLE INPUT_ffmpeg STREQUAL 'yes'
     DISABLE INPUT_ffmpeg STREQUAL 'no'
-    CONDITION FFmpeg_FOUND
-              AND (APPLE OR WIN32 OR ANDROID OR QNX OR QT_FEATURE_pulseaudio OR QT_FEATURE_pipewire)
-              AND QT_FEATURE_thread
+    CONDITION false
 )
 
 # Caveat: FEATURE_ffmpeg_stubs cannot really be used to disable stubbing, it is just used to inform
