diff --git a/src/soapcpp2_lex.l b/src/soapcpp2_lex.l
index 3b2f709..23b4177 100644
--- a/gsoap/src/soapcpp2_lex.l
+++ b/gsoap/src/soapcpp2_lex.l
@@ -50,7 +50,7 @@ A commercial use license is available from Genivia, Inc., contact@genivia.com
 extern int isatty(int); /* silence warnings */
 
 #ifdef WITH_BISON
-YYSTYPE yylval;
+extern YYSTYPE yylval;
 #undef YY_NEED_STRLEN
 #define YY_INPUT(buf, result, max_size) \
         { \
