--- audacity-Audacity-2.1.1/src/TrackPanel.cpp.1	2017-12-11 22:45:26.592506691 +0000
+++ audacity-Audacity-2.1.1/src/TrackPanel.cpp	2017-12-11 22:49:43.280353701 +0000
@@ -3709,7 +3709,7 @@
          wt->GetRate(), logF)
          : rect.y;
       wxInt64 signedBottomDist = int(event.m_y - bottomSel);
-      wxInt64 verticalDist = abs(signedBottomDist);
+      wxInt64 verticalDist = abs((int)signedBottomDist);
       if (bottomSel == topSel)
          // Top and bottom are too close to resolve on screen
          chooseBottom = (signedBottomDist >= 0);
