--- audacity-Audacity-2.1.1/src/effects/Effect.cpp.1	2017-12-12 06:38:58.375325622 +0000
+++ audacity-Audacity-2.1.1/src/effects/Effect.cpp	2017-12-12 06:40:19.477581561 +0000
@@ -34,6 +34,10 @@
 #include <wx/utils.h>
 #include <wx/log.h>
 
+#if defined(__sun)
+#include <alloca.h>
+#endif
+
 #include "audacity/ConfigInterface.h"
 
 #include "Effect.h"
--- audacity-Audacity-2.1.1/src/effects/EffectManager.cpp.1	2017-12-12 06:56:47.435493685 +0000
+++ audacity-Audacity-2.1.1/src/effects/EffectManager.cpp	2017-12-12 06:57:11.065892133 +0000
@@ -15,6 +15,10 @@
 #include <wx/stopwatch.h>
 #include <wx/tokenzr.h>
 
+#if defined(__sun)
+#include <alloca.h>
+#endif
+
 #include "../Experimental.h"
 
 #if defined(EXPERIMENTAL_EFFECTS_RACK)
--- audacity-Audacity-2.1.1/lib-src/libnyquist/nyquist/xlisp/xldmem.c.1	2017-12-12 07:35:21.394569941 +0000
+++ audacity-Audacity-2.1.1/lib-src/libnyquist/nyquist/xlisp/xldmem.c	2017-12-12 07:35:57.394217655 +0000
@@ -17,6 +17,10 @@
 #include "string.h"
 #include "xlisp.h"
 
+#if defined(__sun)
+#include <alloca.h>
+#endif
+
 #ifdef WIN32
 #include "malloc.h" // defines alloca()
 #endif
