--- audacity-Audacity-2.1.1/src/effects/VST/VSTEffect.cpp.~1~	2015-07-11 13:02:22.000000000 +0000
+++ audacity-Audacity-2.1.1/src/effects/VST/VSTEffect.cpp	2017-12-11 22:54:47.404709988 +0000
@@ -2576,7 +2576,7 @@
    // symbols.
    //
    // Once we define a proper external API, the flags can be removed.
-   void *lib = dlopen((const char *)wxString(realPath).ToUTF8(), RTLD_NOW | RTLD_LOCAL | RTLD_DEEPBIND);
+   void *lib = dlopen((const char *)wxString(realPath).ToUTF8(), RTLD_NOW | RTLD_LOCAL);
    if (!lib) 
    {
       return false;
