--- audacity-Audacity-2.3.2/src/widgets/NumericTextCtrl.cpp.orig	lun. janv. 24 18:11:16 2022
+++ audacity-Audacity-2.3.2/src/widgets/NumericTextCtrl.cpp	lun. janv. 24 18:11:54 2022
@@ -722,7 +722,7 @@
    int numFracFields = 0;
    wxString numStr;
    wxString delimStr;
-   unsigned int i;
+   size_t i;
 
    mNtscDrop = false;
    for(i=0; i<format.length(); i++) {
@@ -786,7 +786,7 @@
             numFracFields++;
          }
          else {
-            unsigned int j;
+            size_t j;
             for(j=0; j<mFields.size(); j++)
                mFields[j].base *= range;
             mFields.push_back(NumericField(inFrac, 1, range, zeropad));
