diff -ruN goffice-0.8.1.orig/docs/reference/Makefile.am goffice-0.8.1/docs/reference/Makefile.am
--- goffice-0.8.1.orig/docs/reference/Makefile.am	2010-03-15 04:48:22.085543869 +0000
+++ goffice-0.8.1/docs/reference/Makefile.am	2010-03-15 04:49:26.769854630 +0000
@@ -14,7 +14,7 @@
 
 MKTMPL_OPTIONS=
 
-FIXXREF_OPTIONS=
+FIXXREF_OPTIONS=--module=$(DOC_MODULE)
 
 HFILE_GLOB=$(top_srcdir)/goffice/*.h $(top_srcdir)/goffice/*/*.h
 CFILE_GLOB=$(top_srcdir)/goffice/*.c $(top_srcdir)/goffice/*/*.c
