--- mate-power-manager-1.18.1/data/org.mate.power-manager.gschema.xml.in.1	2018-01-11 12:58:03.505128814 +0000
+++ mate-power-manager-1.18.1/data/org.mate.power-manager.gschema.xml.in	2018-01-11 12:59:58.776390481 +0000
@@ -336,7 +336,7 @@
       <summary>If preferences and statistics items should be shown in the context menu</summary>
     </key>
     <key name="icon-policy" enum="org.mate.power-manager.IconPolicy">
-      <default>'present'</default>
+      <default>'always'</default>
       <summary>When to show the notification icon</summary>
       <description>Display options for the notification icon.</description>
     </key>
