diff -ur notion-3-2014052800/build/rules.mk notion-3-2014052800.mangled/build/rules.mk
--- notion-3-2014052800/build/rules.mk	2014-05-27 15:00:18.000000000 -0700
+++ notion-3-2014052800.mangled/build/rules.mk	2014-08-03 09:55:00.518894966 -0700
@@ -104,8 +104,8 @@
 TO_CLEAN += preload.c
 else # !PRELOAD_MODULES
 LDFLAGS += $(EXPORT_DYNAMIC)
-WHOLEA = -Wl,-whole-archive
-NO_WHOLEA = -Wl,-no-whole-archive
+WHOLEA = #-Wl,-whole-archive
+NO_WHOLEA = #-Wl,-no-whole-archive
 endif # !PRELOAD_MODULES
 
 preload.c:
diff -ur notion-3-2014052800/libtu/Makefile notion-3-2014052800.mangled/libtu/Makefile
--- notion-3-2014052800/libtu/Makefile	2014-05-27 15:00:18.000000000 -0700
+++ notion-3-2014052800.mangled/libtu/Makefile	2014-08-03 10:00:43.998570080 -0700
@@ -16,7 +16,7 @@
 
 #INCLUDES += $(LIBTU_INCLUDES) $(LUA_INCLUDES)
 
-CFLAGS += $(C98_SOURCE) $(POSIX_SOURCE) $(WARN)
+CFLAGS += $(C99_SOURCE) $(POSIX_SOURCE) $(WARN)
 
 SOURCES=iterable.c  map.c  misc.c  obj.c  objlist.c  optparser.c  output.c  parser.c  prefix.c  ptrlist.c  rb.c  setparam.c  stringstore.c  tokenizer.c  util.c errorlog.c
 
diff -ur notion-3-2014052800/mod_notionflux/notionflux/Makefile notion-3-2014052800.mangled/mod_notionflux/notionflux/Makefile
--- notion-3-2014052800/mod_notionflux/notionflux/Makefile	2014-05-27 15:00:18.000000000 -0700
+++ notion-3-2014052800.mangled/mod_notionflux/notionflux/Makefile	2014-08-03 09:49:09.092115438 -0700
@@ -12,7 +12,7 @@
 
 INCLUDES += $(X11_INCLUDES) -I$(TOPDIR)
 
-LIBS += $(X11_LIBS) 
+LIBS += $(X11_LIBS) -lsocket
 
 EXTRA_EXECUTABLE = notionflux
 
diff -ur notion-3-2014052800/system-autodetect.mk notion-3-2014052800.mangled/system-autodetect.mk
--- notion-3-2014052800/system-autodetect.mk	2014-05-27 15:00:18.000000000 -0700
+++ notion-3-2014052800.mangled/system-autodetect.mk	2014-08-03 10:01:18.006992409 -0700
@@ -9,7 +9,7 @@
 # Installation path prefix. Unless you know what you're doing, the default
 # of /usr/local is likely the correct choice.
 #DIST: PREFIX=/usr/local
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 
 # Unless you are creating a package conforming to some OS's standards, you
 # probably do not want to modify the following directories:
@@ -96,7 +96,7 @@
 
 # Remap F11 key to SunF36 and F12 to SunF37? You may want to set this
 # on SunOS.
-#DEFINES += -DCF_SUN_F1X_REMAP
+DEFINES += -DCF_SUN_F1X_REMAP
 
 
 ##
@@ -150,10 +150,10 @@
 WARN=-W -Wall -pedantic 
 
 CFLAGS += -Os $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES) \
-          -DHAS_SYSTEM_ASPRINTF=$(HAS_SYSTEM_ASPRINTF)
+          -DHAS_SYSTEM_ASPRINTF=$(HAS_SYSTEM_ASPRINTF) -std=c99 -D_XPG6
 
-LDFLAGS += -Wl,--as-needed $(LIBS) $(EXTRA_LIBS)
-EXPORT_DYNAMIC=-Xlinker --export-dynamic
+LDFLAGS += $(LIBS) $(EXTRA_LIBS)
+EXPORT_DYNAMIC=#-Xlinker --export-dynamic
 
 # The following options are mainly for development use and can be used
 # to check that the code seems to conform to some standards. Depending
