*** idnkit-1.0-src/include/idn/assert.h.org	Tue Aug 26 16:49:02 2003
--- idnkit-1.0-src/include/idn/assert.h	Tue Sep  2 17:30:59 2003
***************
*** 48,58 ****
--- 48,66 ----
  extern "C" {
  #endif
  
+ #ifdef	SUN_CHANGE
+ #ifndef DEBUG
+ #define NDEBUG
+ #endif
+ 
  #include <assert.h>
+ #else
+ #include <assert.h>
  
  #ifndef DEBUG
  #define NDEBUG
  #endif
+ #endif	/* SUN_CHANGE */
  
  #ifdef __cplusplus
  }
