--- ./ltconfig.orig	Fri Jun  3 04:15:19 2011
+++ ./ltconfig	Fri Jun  3 04:15:41 2011
@@ -2072,7 +2072,7 @@
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so$major $libname.so'
+  library_names_spec='${libname}${release}.so$major $libname.so'
   soname_spec='${libname}${release}.so$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
