--- gst-plugins-good-0.10.31/gst/deinterlace/tvtime/greedyh.c	Tue Jan 24 01:21:20 2012
+++ gst-plugins-good-0.10.31/gst/deinterlace/tvtime/greedyh.c	Sun Sep  1 12:00:15 2013
@@ -49,6 +49,8 @@
 #define GST_DEINTERLACE_METHOD_GREEDY_H_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), GST_TYPE_DEINTERLACE_METHOD_GREEDY_H, GstDeinterlaceMethodGreedyHClass))
 #define GST_DEINTERLACE_METHOD_GREEDY_H_CAST(obj)	((GstDeinterlaceMethodGreedyH*)(obj))
 
+#undef BUILD_X86_ASM
+
 typedef struct
 {
   GstDeinterlaceMethod parent;
--- gst-plugins-good-0.10.31/gst/deinterlace/tvtime/tomsmocomp.c	Sun Dec 11 19:48:11 2011
+++ gst-plugins-good-0.10.31/gst/deinterlace/tvtime/tomsmocomp.c	Sun Sep  1 11:55:02 2013
@@ -37,6 +37,8 @@
 #include "gstdeinterlacemethod.h"
 #include "plugins.h"
 
+#undef BUILD_X86_ASM
+
 #define GST_TYPE_DEINTERLACE_METHOD_TOMSMOCOMP	(gst_deinterlace_method_tomsmocomp_get_type ())
 #define GST_IS_DEINTERLACE_METHOD_TOMSMOCOMP(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), GST_TYPE_DEINTERLACE_METHOD_TOMSMOCOMP))
 #define GST_IS_DEINTERLACE_METHOD_TOMSMOCOMP_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), GST_TYPE_DEINTERLACE_METHOD_TOMSMOCOMP))
