--- brasero-2.32.1/libbrasero-burn/Makefile.am.~1~	2014-12-12 15:23:41.990403481 +0300
+++ brasero-2.32.1/libbrasero-burn/Makefile.am	2014-12-12 15:24:57.371024724 +0300
@@ -65,7 +65,8 @@
 	$(BRASERO_GIO_LIBS)					\
 	$(BRASERO_GTK_LIBS)					\
 	$(BRASERO_GSTREAMER_LIBS)				\
-	$(BRASERO_CANBERRA_LIBS)
+	$(BRASERO_CANBERRA_LIBS)				\
+	-lm
 
 libbrasero_burn@BRASERO_LIBRARY_SUFFIX@_la_LDFLAGS =					\
 	-version-info $(LIBBRASERO_LT_VERSION)			\
