--- php-7.0.24/ext/sqlite3/config0.m4.~1~	2017-10-14 10:34:26.093057208 +0000
+++ php-7.0.24/ext/sqlite3/config0.m4	2017-10-14 10:36:01.845522558 +0000
@@ -86,7 +86,7 @@
     fi
 
     AC_DEFINE(HAVE_SQLITE3_ERRSTR, 1, [have sqlite3_errstr function])
-    PHP_SQLITE3_CFLAGS="-I@ext_srcdir@/libsqlite $other_flags $threadsafe_flags $debug_flags"
+    PHP_SQLITE3_CFLAGS="-std=c99 -I@ext_srcdir@/libsqlite $other_flags $threadsafe_flags $debug_flags"
     PHP_INSTALL_HEADERS([ext/sqlite3/libsqlite/sqlite3.h])
   fi
 
