--- php-7.3.21/ext/intl/tests/locale_filter_matches3.phpt	Tue Aug  4 16:06:22 2020
+++ php-7.3.21_patched/ext/intl/tests/locale_filter_matches3.phpt	Sat Apr  9 06:59:14 2022
@@ -1,8 +1,8 @@
 --TEST--
-locale_filter_matches.phpt() ICU >= 51.2
+locale_filter_matches.phpt() ICU >= 51.2 and < 70.1
 --SKIPIF--
 <?php if( !extension_loaded( 'intl' ) ) print 'skip'; ?>
-<?php if (version_compare(INTL_ICU_VERSION, '51.2') < 0) die('skip for ICU >= 51.2'); ?>
+<?php if (version_compare(INTL_ICU_VERSION, '51.2') < 0 || version_compare(INTL_ICU_VERSION, '70.1') >= 0) die('skip for ICU >= 51.2 and < 70.1'); ?>
 --FILE--
 <?php
 
