--- trealla-2.103.0/Makefile.orig	2026-07-02 03:14:41.000000000 +0200
+++ trealla-2.103.0/Makefile	2026-07-02 07:17:30.627451963 +0200
@@ -1,18 +1,21 @@
 # Installation paths
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR ?= $(PREFIX)/bin
 LIBDIR ?= $(PREFIX)/share/trealla
 MANDIR ?= $(PREFIX)/share/man
 
 EMBED ?= 1
 
-HOST_CC ?= cc
+HOST_CC ?= gcc
 
 GIT_VERSION := "$(shell git describe --abbrev=4 --dirty --always --tags)"
 GIT_VERSION := "$(shell git describe --abbrev=4 --dirty --always --tags)"
 COMPILER_IS_GCC := $(shell $(CC) --version | grep -E -o 'g?cc')
 
-CFLAGS = -Isrc -I/usr/local/include -DVERSION='$(GIT_VERSION)' \
+CFLAGS = -Isrc \
+	-Wno-error=int-conversion \
+	-Wno-error=implicit-function-declaration \
+	$(XPG8MODE) \
 	-DDEFAULT_LIBRARY_PATH='"$(LIBDIR)/library"' \
 	-O3 $(OPT) -D_GNU_SOURCE \
 	-Wall -Wextra \
@@ -25,7 +28,7 @@
 CFLAGS += -DEMBED=1
 endif
 
-LDFLAGS = -L/usr/local/lib -lm
+LDFLAGS = -lm -lsocket -lnsl
 
 UNAME_S := $(shell uname -s)
 
@@ -93,12 +96,12 @@
 endif
 
 ifndef NOFFI
-CFLAGS += -DUSE_FFI=1 -I/usr/local/opt/libffi/include
-LDFLAGS += -lffi -ldl
+CFLAGS += -DUSE_FFI=1 -I/usr/lib/MACH64/libffi-3.2.1/include
+LDFLAGS += -lffi -ldl -L/usr/openssl/3/lib/MACH64
 endif
 
 ifndef NOSSL
-CFLAGS += -DUSE_OPENSSL=1
+CFLAGS += -DUSE_OPENSSL=1 -I/usr/openssl/3/include
 LDFLAGS += -lssl -lcrypto
 endif
 
