--- trealla-2.94.9/src/bif_posix.c.orig	2026-04-30 13:49:45.000000000 +0200
+++ trealla-2.94.9/src/bif_posix.c	2026-04-30 20:07:49.831943275 +0200
@@ -204,7 +204,7 @@
 	time_t when = get_smallint(p1);
 	char tmpbuf[256];
 	cell tmp;
-	make_cstring(&tmp, ctime_r(&when, tmpbuf));
+	make_cstring(&tmp, ctime_r(&when, tmpbuf, 256));
 	bool ok = unify(q, p2, p2_ctx, &tmp, q->st.cur_ctx);
 	unshare_cell(&tmp);
 	return ok;
