--- trealla-3.0.12/src/network.c.orig
+++ trealla-3.0.12/src/network.c
@@ -4,6 +4,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <time.h>
+#include <sys/fcntl.h>
+#include <sys/types.h>
+#include <fcntl.h>
 
 #if USE_OPENSSL
 #include <openssl/ssl.h>
@@ -327,8 +330,8 @@
 void tpl_set_nonblocking(stream *str)
 {
 #if !defined(_WIN32) && !defined(__wasi__)
-	unsigned long flag = 1;
-	ioctl(fileno(str->fp_in), FIONBIO, &flag);
+    int flags = fcntl(fileno(str->fp_in), F_GETFL, 0);
+    fcntl(fileno(str->fp_in), F_SETFL, flags | O_NONBLOCK);
 #endif
 }
 
