--- trealla-2.103.0/src/network.c.orig	2026-07-02 03:14:41.000000000 +0200
+++ trealla-2.103.0/src/network.c	2026-07-02 07:24:10.887000825 +0200
@@ -43,10 +43,13 @@
 #endif
 #include <netinet/in.h>
 #include <netinet/tcp.h>
+#include <sys/fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
+#include <sys/types.h>
 #include <unistd.h>
+#include <fcntl.h>
 #endif
 
 
@@ -311,9 +314,10 @@
 void tpl_set_nonblocking(stream *str)
 {
 #if !defined(_WIN32) && !defined(__wasi__)
-	unsigned long flag = 1;
-	ioctl(fileno(str->fp_in), FIONBIO, &flag);
-	ioctl(fileno(str->fp_out), FIONBIO, &flag);
+    int flags = fcntl(fileno(str->fp_in), F_GETFL, 0);
+    fcntl(fileno(str->fp_in), F_SETFL, flags | O_NONBLOCK);
+    flags = fcntl(fileno(str->fp_out), F_GETFL, 0);
+    fcntl(fileno(str->fp_out), F_SETFL, flags | O_NONBLOCK);
 #endif
 }
 
