--- trealla-2.69.10/src/network.c.orig	2025-05-02 04:04:54.000000000 +0200
+++ trealla-2.69.10/src/network.c	2025-09-01 17:03:12.410016605 +0200
@@ -43,6 +43,7 @@
 #endif
 #include <netinet/in.h>
 #include <netinet/tcp.h>
+#include <sys/fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
@@ -128,7 +129,7 @@
 		int flag = 1;
 		setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (char *)&flag, sizeof(flag));
 		int flag2 = 1;
-		setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, (char *)&flag2, sizeof(flag2));
+//		setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, (char *)&flag2, sizeof(flag2));
 
 		if (connect(fd, rp->ai_addr, rp->ai_addrlen) != -1)
 			break;
@@ -183,7 +184,7 @@
 		int flag = 1;
 		setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (char *)&flag, sizeof(flag));
 		int flag2 = 1;
-		setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, (char *)&flag2, sizeof(flag2));
+//		setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, (char *)&flag2, sizeof(flag2));
 
 		if (bind(fd, rp->ai_addr, rp->ai_addrlen) == 0)
 			break;
@@ -263,8 +264,8 @@
 void net_set_nonblocking(stream *str)
 {
 #if !defined(_WIN32) && !defined(__wasi__)
-	unsigned long flag = 1;
-	ioctl(fileno(str->fp), FIONBIO, &flag);
+	int flags = fcntl(fileno(str->fp), F_GETFL, 0);
+    fcntl(fileno(str->fp), F_SETFL, flags | O_NONBLOCK);
 #endif
 }
 
