Modification for configure file.
Will not send upstream.

--- configure	2016-11-17 11:11:01.452074740 +0000
+++ configure	2016-11-23 11:17:14.362287357 +0000
@@ -10384,7 +10384,7 @@
 	  ;;
 	esac
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec=''
       hardcode_shlibpath_var=no
       case $host_os in
       solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -13607,7 +13607,7 @@
 	    archive_expsym_cmds_CXX='echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~echo "local: *; };" >> $lib.exp~
 	      $CC -G${allow_undefined_flag} ${wl}-M ${wl}$lib.exp -h$soname -o $lib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags~$RM $lib.exp'
 
-	    hardcode_libdir_flag_spec_CXX='-R$libdir'
+	    hardcode_libdir_flag_spec_CXX=''
 	    hardcode_shlibpath_var_CXX=no
 	    case $host_os in
 	      solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -17518,7 +17518,7 @@
 { $as_echo "$as_me:$LINENO: checking for gcc optimization flags" >&5
 $as_echo_n "checking for gcc optimization flags... " >&6; }
 old_cflags=$CFLAGS
-CFLAGS="$CFLAGS -pipe"
+CFLAGS="$CFLAGS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -17552,7 +17552,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest.$ac_objext; then
-   EXTRA_OPT_CFLAGS="-pipe"
+   EXTRA_OPT_CFLAGS=""
 else
   $as_echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
