--- OpenEXR.pc.in	2007-03-30 10:14:05.000000000 -0700
+++ OpenEXR.pc.in	2011-03-21 16:01:21.006971380 -0700
@@ -7,5 +7,5 @@
 Name: OpenEXR
 Description: OpenEXR image library
 Version: @OPENEXR_VERSION@
-Libs: -L${libdir} @ILMBASE_LDFLAGS@ -lIlmImf -lz @ILMBASE_LIBS@
-Cflags: @ILMBASE_CXXFLAGS@ -I${OpenEXR_includedir}
+Libs: -L${libdir} -lIlmImf -lz @ILMBASE_LIBS@
+Cflags: -I${OpenEXR_includedir}
