# Internal patch.
--- IlmImf/ImfAutoArray.h	2015-02-01 16:32:21.274048802 -0800
+++ IlmImf/ImfAutoArray.h	2015-02-01 16:32:54.346820460 -0800
@@ -46,6 +46,8 @@
 
 #include "OpenEXRConfig.h"
 
+#include <cstring>
+
 namespace Imf {
 
 
@@ -58,7 +60,7 @@
       public:
 
 	AutoArray (): _data (new T [size])
-	{ (void) memset(_data, '\0', size * sizeof(T)); }
+	{ (void) std::memset(_data, '\0', size * sizeof(T)); }
 
 	~AutoArray () {delete [] _data;}
 
