--- gdm-2.30.7/daemon/gdm-greeter-server.c-orig	2011-06-24 14:20:01.238188292 -0500
+++ gdm-2.30.7/daemon/gdm-greeter-server.c	2011-06-24 14:21:04.971643276 -0500
@@ -990,6 +990,13 @@ gdm_greeter_server_stop (GdmGreeterServe
 
         g_debug ("GreeterServer: Stopping greeter server...");
 
+        dbus_server_disconnect (greeter_server->priv->server);
+        dbus_server_unref (greeter_server->priv->server);
+        greeter_server->priv->server = NULL;
+
+        g_free (greeter_server->priv->server_address);
+        greeter_server->priv->server_address = NULL;
+
         return ret;
 }
 
