--- gdm-2.30.7/data/greeter-autostart/Makefile.am.1	2017-08-26 10:22:20.314111802 +0000
+++ gdm-2.30.7/data/greeter-autostart/Makefile.am	2017-08-26 10:22:24.400704389 +0000
@@ -17,7 +17,6 @@
 	gdm-simple-greeter.desktop.in		\
 	gnome-power-manager.desktop.in		\
 	gnome-settings-daemon.desktop.in	\
-	gok.desktop.in				\
 	metacity.desktop.in			\
 	orca-screen-reader.desktop.in		\
 	polkit-gnome-authentication-agent-1.desktop.in	\
