--- gdm-2.30.5/daemon/gdm-welcome-session.c-orig	2010-09-23 16:53:08.941432205 -0500
+++ gdm-2.30.5/daemon/gdm-welcome-session.c	2010-09-23 16:55:45.580327687 -0500
@@ -602,7 +602,7 @@ spawn_child_setup (SpawnChildData *data)
         }
 
         g_debug ("GdmWelcomeSession: Setting up run time dir %s", data->runtime_dir);
-        g_mkdir (data->runtime_dir, 0755);
+        g_mkdir_with_parents (data->runtime_dir, 0755);
         res = chown (data->runtime_dir, pwent->pw_uid, pwent->pw_gid);
         if (res == -1) {
                 g_warning ("GdmWelcomeSession: Error setting owner of run time directory: %s",
