--- gdm-2.26.1/data/gconf.path-orig	2009-04-28 20:36:41.977711000 -0500
+++ gdm-2.26.1/data/gconf.path	2009-04-28 20:37:24.422960000 -0500
@@ -1,5 +1,6 @@
 # treat system-wide settings as mandatory
-xml:readonly:/etc/gconf/gconf.xml.system
+# We don't use this default mechanism on opensolaris.
+# xml:readonly:/etc/gconf/gconf.xml.system
 
 # override some settings
 # we treat settings in this location as
