--- gdm-2.30.7/daemon/gdm-session-record.c.orig	2012-09-17 10:20:17.669528894 +0200
+++ gdm-2.30.7/daemon/gdm-session-record.c	2012-09-17 10:21:19.928873229 +0200
@@ -276,6 +276,7 @@ gdm_session_record_logout (GPid         
         UTMP        session_record = { 0 };
         UTMP       *u;
 
+        record_set_username (&session_record, user_name);
         g_debug ("Writing logout record");
 
 #if defined(HAVE_UT_UT_TYPE)
