*** grails-1.0.3/ant/bin/runant.py.orig	Fri Jul  8 09:46:11 2011
--- grails-1.0.3/ant/bin/runant.py	Fri Jul  8 09:46:26 2011
***************
*** 1,4 ****
! #!/usr/bin/python
  # Copyright 2001,2003-2004 The Apache Software Foundation
  #
  #  Licensed under the Apache License, Version 2.0 (the "License");
--- 1,4 ----
! #!/usr/bin/python2.7
  # Copyright 2001,2003-2004 The Apache Software Foundation
  #
  #  Licensed under the Apache License, Version 2.0 (the "License");
