adapt to openssl 1.1 style

--- cxxtools-3.0/src/tcpsocketimpl.cpp	2020-04-11 23:27:22.000000000 +0000
+++ cxxtools-3.0/src/tcpsocketimpl.cpp.new	2022-12-27 14:39:22.797156048 +0000
@@ -253,8 +253,9 @@
 {
     openssl_mutex = new cxxtools::Mutex[CRYPTO_num_locks()];
 
-    CRYPTO_set_id_callback(pthreads_thread_id);
-    CRYPTO_set_locking_callback(pthreads_locking_callback);
+    /*CRYPTO_set_id_callback(pthreads_thread_id);
+    CRYPTO_set_locking_callback(pthreads_locking_callback); */
+    CRYPTO_THREADID_set_callback((unsigned long (*)())thread_id);
 }
 
 void TcpSocketImpl::initSsl()
