--- ncurses-5.9/configure.1	2014-12-16 13:09:31.611329529 +0300
+++ ncurses-5.9/configure	2014-12-16 13:10:38.680798247 +0300
@@ -7864,7 +7864,7 @@
 echo "${ECHO_T}$cf_result" >&6

 if test "$cf_result" = yes ; then
-	CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE_EXTENDED"
+	CPPFLAGS="$CPPFLAGS"
 elif test "x" != "x" ; then
	echo "$as_me:7869: checking checking for compatible value versus " >&5
 echo $ECHO_N "checking checking for compatible value versus ... $ECHO_C" >&6
