--- ncurses-5.9/configure.orig	2016-04-12 10:42:44.808765427 +0300
+++ ncurses-5.9/configure	2016-04-12 10:42:56.607701646 +0300
@@ -3621,6 +3621,9 @@
	echo "$as_me:3621: checking if we should install .pc files for $PKG_CONFIG" >&5
 echo $ECHO_N "checking if we should install .pc files for $PKG_CONFIG... $ECHO_C" >&6

+	if test -n "$PKG_CONFIG_PATH" ; then
+		PKG_CONFIG_LIBDIR=$PKG_CONFIG_PATH
+	fi
	# Leave this as something that can be overridden in the environment.
	if test -z "$PKG_CONFIG_LIBDIR" ; then
		PKG_CONFIG_LIBDIR=`echo "$PKG_CONFIG" | sed -e 's,/[^/]*/[^/]*$,,'`/lib/pkgconfig
