--- tests/t-modlinv.c	2009-05-12 02:12:12.000000000 -0400
+++ tests/t-modlinv.c	2009-09-21 18:25:28.190094000 -0400
@@ -26,24 +26,24 @@
 #include "tests.h"
 
 
-void
+static void
 one (mp_limb_t n)
 {
-  mp_limb_t  inv, prod;
+  mp_limb_t  i, p;
 
-  binvert_limb (inv, n);
-  prod = (inv * n) & GMP_NUMB_MASK;
-  if (prod != 1)
+  binvert_limb (i, n);
+  p = (i * n) & GMP_NUMB_MASK;
+  if (p != 1)
     {
       printf ("binvert_limb wrong\n");
       mp_limb_trace ("  n       ", n);
-      mp_limb_trace ("  got     ", inv);
-      mp_limb_trace ("  product ", prod);
+      mp_limb_trace ("  got     ", i);
+      mp_limb_trace ("  product ", p);
       abort ();
     }
 }
 
-void
+static void
 some (void)
 {
   int  i;
@@ -51,7 +51,7 @@
     one (refmpn_random_limb () | 1);
 }
 
-void
+static void
 all (void)
 {
   mp_limb_t  n;
