--- gmp-h.in	2008-09-10 17:02:01.000000000 -0400
+++ gmp-h.in	2008-09-19 17:10:39.000000000 -0400
@@ -28,10 +28,18 @@
 
 /* Instantiated by configure. */
 #if ! defined (__GMP_WITHIN_CONFIGURE)
-#define __GMP_BITS_PER_MP_LIMB             @BITS_PER_MP_LIMB@
+#if defined(__amd64) || defined(__sparcv9)
+#define __GMP_BITS_PER_MP_LIMB 64
+#else
+#define __GMP_BITS_PER_MP_LIMB 32
+#endif
 #define __GMP_HAVE_HOST_CPU_FAMILY_power   @HAVE_HOST_CPU_FAMILY_power@
 #define __GMP_HAVE_HOST_CPU_FAMILY_powerpc @HAVE_HOST_CPU_FAMILY_powerpc@
-#define GMP_LIMB_BITS                      @BITS_PER_MP_LIMB@
+#if defined(__amd64) || defined(__sparcv9)
+#define GMP_LIMB_BITS 64
+#else
+#define GMP_LIMB_BITS 32
+#endif
 #define GMP_NAIL_BITS                      @GMP_NAIL_BITS@
 #endif
 #define GMP_NUMB_BITS     (GMP_LIMB_BITS - GMP_NAIL_BITS)
@@ -427,6 +435,10 @@
 #define __GMP_EXTERN_INLINE      extern __inline__
 #endif
 #define __GMP_INLINE_PROTOTYPES  1
+#else
+#ifdef __SUNPRO_C
+#define __GMP_EXTERN_INLINE     static inline
+#endif
 #endif
 
 /* DEC C (eg. version 5.9) supports "static __inline foo()", even in -std1
