--- tcl8.4.18/unix/installManPage.~1~	2008-02-06 07:38:26.000000000 -0800
+++ tcl8.4.18/unix/installManPage	2011-04-01 00:04:27.166384166 -0700
@@ -48,7 +48,7 @@
 	q
     }' $MANPAGE`
 
-SECTION=`echo $MANPAGE | sed 's/.*\(.\)$/\1/'`
+SECTION=`echo $MANPAGE | sed 's/.*\.\(.*\)$/\1/'`
 SRCDIR=`dirname $MANPAGE`
 FIRST=""
 for f in $NAMES; do
@@ -57,6 +57,7 @@
 	FIRST=$f
 	rm -f $DIR/$FIRST $DIR/$FIRST.*
 	sed -e "/man\.macros/r $SRCDIR/man.macros" -e "/man\.macros/d" \
+	    -f ../../oracleman-stability \
 	    $MANPAGE > $DIR/$FIRST
 	chmod 444 $DIR/$FIRST
 	$ZIP $DIR/$FIRST
