*** tcl8.4.18/unix/Makefile.in.orig	Thu Jun 14 04:46:38 2007
--- tcl8.4.18/unix/Makefile.in	Thu Jun 14 04:46:49 2007
***************
*** 72,82 ****
  
  # Directory in which to install manual entries for Tcl's C library
  # procedures:
! MAN3_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man3
  
  # Directory in which to install manual entries for the built-in
  # Tcl commands:
! MANN_INSTALL_DIR	= $(MAN_INSTALL_DIR)/mann
  
  # Path to the html documentation dir:
  HTML_DIR		= @HTML_DIR@
--- 72,82 ----
  
  # Directory in which to install manual entries for Tcl's C library
  # procedures:
! MAN3TCL_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man3tcl
  
  # Directory in which to install manual entries for the built-in
  # Tcl commands:
! MAN1T_INSTALL_DIR	= $(MAN_INSTALL_DIR)/man1t
  
  # Path to the html documentation dir:
  HTML_DIR		= @HTML_DIR@
***************
*** 298,304 ****
  
  APP_CC_SWITCHES = $(CC_SWITCHES) @EXTRA_APP_CC_SWITCHES@
  
! LIBS		= @DL_LIBS@ @LIBS@ $(MATH_LIBS)
  
  DEPEND_SWITCHES	= ${CFLAGS} -I${GENERIC_DIR} -I${SRC_DIR} \
  ${AC_FLAGS} ${MATH_FLAGS} \
--- 298,304 ----
  
  APP_CC_SWITCHES = $(CC_SWITCHES) @EXTRA_APP_CC_SWITCHES@
  
! LIBS		= @DL_LIBS@ @LIBS@ $(MATH_LIBS) -lc
  
  DEPEND_SWITCHES	= ${CFLAGS} -I${GENERIC_DIR} -I${SRC_DIR} \
  ${AC_FLAGS} ${MATH_FLAGS} \
***************
*** 711,717 ****
  	@if test ! -x $(UNIX_DIR)/installManPage; then \
  	    chmod +x $(UNIX_DIR)/installManPage; \
  	    fi
! 	@for i in $(MAN_INSTALL_DIR) $(MAN1_INSTALL_DIR) $(MAN3_INSTALL_DIR) $(MANN_INSTALL_DIR) ; \
  	    do \
  	    if [ ! -d $$i ] ; then \
  		echo "Making directory $$i"; \
--- 711,717 ----
  	@if test ! -x $(UNIX_DIR)/installManPage; then \
  	    chmod +x $(UNIX_DIR)/installManPage; \
  	    fi
! 	@for i in $(MAN_INSTALL_DIR) $(MAN1_INSTALL_DIR) $(MAN3TCL_INSTALL_DIR) $(MAN1T_INSTALL_DIR) ; \
  	    do \
  	    if [ ! -d $$i ] ; then \
  		echo "Making directory $$i"; \
***************
*** 725,738 ****
  	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MAN1_INSTALL_DIR); \
  	done
  
! 	@echo "Installing and cross-linking C API (.3) docs";
! 	@for i in $(TOP_DIR)/doc/*.3; do \
! 	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MAN3_INSTALL_DIR); \
  	done
  
! 	@echo "Installing and cross-linking command (.n) docs";
! 	@for i in $(TOP_DIR)/doc/*.n; do \
! 	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MANN_INSTALL_DIR); \
  	done
  
  # Optional target to install private headers
--- 725,738 ----
  	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MAN1_INSTALL_DIR); \
  	done
  
! 	@echo "Installing and cross-linking C API (.3tcl) docs";
! 	@for i in $(TOP_DIR)/doc/*.3tcl; do \
! 	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MAN3TCL_INSTALL_DIR); \
  	done
  
! 	@echo "Installing and cross-linking command (.1t) docs";
! 	@for i in $(TOP_DIR)/doc/*.1t; do \
! 	    $(UNIX_DIR)/installManPage $(MAN_FLAGS) $$i $(MAN1T_INSTALL_DIR); \
  	done
  
  # Optional target to install private headers
