--- tcl8.4.18/unix/tcl.m4.orig	Tue Mar 25 03:48:12 2008
+++ tcl8.4.18/unix/tcl.m4	Tue Mar 25 03:48:42 2008
@@ -1926,7 +1926,7 @@
 	    else
 		case $system in
 		    SunOS-5.[[1-9]][[0-9]]*)
-			SHLIB_LD='${CC} -G -z text';;
+			SHLIB_LD='${CC} -G -z text ${LDFLAGS}';;
 		    *)
 			SHLIB_LD="/usr/ccs/bin/ld -G -z text";;
 		esac
--- tcl8.4.18/unix/dltest/Makefile.in.orig	Tue Mar 25 08:16:31 2008
+++ tcl8.4.18/unix/dltest/Makefile.in	Tue Mar 25 08:18:59 2008
@@ -7,6 +7,7 @@
 CC = @CC@
 LIBS =			@TCL_BUILD_STUB_LIB_SPEC@ @DL_LIBS@ @LIBS@ @MATH_LIBS@
 AC_FLAGS =		@DEFS@
+LDFLAGS =		@LDFLAGS_DEFAULT@ @LDFLAGS@
 SHLIB_CFLAGS =		@SHLIB_CFLAGS@
 SHLIB_LD =		@SHLIB_LD@
 SHLIB_LD_LIBS =		@SHLIB_LD_LIBS@
