--- tcl8.6.7/unix/tcl.m4.1	2018-01-29 16:20:50.229143078 +0000
+++ tcl8.6.7/unix/tcl.m4	2018-01-29 16:22:05.592719535 +0000
@@ -1912,7 +1912,7 @@
 	    DL_LIBS="-ldl"
 	    AS_IF([test "$GCC" = yes], [
 		SHLIB_LD='${CC} -shared'
-		CC_SEARCH_FLAGS='-Wl,-R,${LIB_RUNTIME_DIR}'
+		CC_SEARCH_FLAGS=''
 		LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
 		AS_IF([test "$do64bit_ok" = yes], [
 		    AS_IF([test "$arch" = "sparcv9 sparc"], [
