--- freeipmi-1.6.11/common/portability/freeipmi-argp-fmtstream.h.orig	2023-06-19 18:49:46.522814933 +0200
+++ freeipmi-1.6.11/common/portability/freeipmi-argp-fmtstream.h	2023-06-19 18:50:20.479694482 +0200
@@ -192,7 +192,7 @@
 extern int _argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 extern int __argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 
-#if defined(__OPTIMIZE__) && !defined(__CYGWIN__)
+#if 0
 /* Inline versions of above routines.  */
 
 #if !_LIBC
--- freeipmi-1.6.11/common/portability/freeipmi-argp-fmtstream.c.orig	2023-06-06 00:58:25.000000000 +0200
+++ freeipmi-1.6.11/common/portability/freeipmi-argp-fmtstream.c	2023-06-19 18:46:05.959433472 +0200
@@ -25,6 +25,7 @@
 #include <config.h>
 #endif
 
+#include <alloca.h>
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
