--- freeipmi-1.5.1/common/portability/freeipmi-argp-fmtstream.h.1	2017-11-27 17:55:53.273542494 +0000
+++ freeipmi-1.5.1/common/portability/freeipmi-argp-fmtstream.h	2017-11-27 17:56:10.528739183 +0000
@@ -192,7 +192,7 @@
 extern int _argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 extern int __argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 
-#ifdef __OPTIMIZE__
+#if 0
 /* Inline versions of above routines.  */
 
 #if !_LIBC
--- freeipmi-1.5.1/common/portability/freeipmi-argp-parse.c.1	2017-11-27 17:59:26.172914226 +0000
+++ freeipmi-1.5.1/common/portability/freeipmi-argp-parse.c	2017-11-27 18:01:20.609295362 +0000
@@ -27,7 +27,6 @@
 #endif
 
 /* AIX requires this to be the first thing in the file.  */
-#ifndef __GNUC__
 # if HAVE_ALLOCA_H
 #  include <alloca.h>
 # else
@@ -39,7 +38,6 @@
 #   endif
 #  endif
 # endif
-#endif
 
 #include <stdlib.h>
 #include <string.h>
