--- rss-glx_0.9.1/src/Makefile.in	2010-01-05 05:59:19.000000000 +0300
+++ rss-glx_0.9.1/src/Makefile.in	2010-01-05 05:59:19.000000000 +0300 
@@ -614,7 +614,7 @@
 	$(matrixview_LINK) $(matrixview_OBJECTS) $(matrixview_LDADD) $(LIBS)
 pixelcity$(EXEEXT): $(pixelcity_OBJECTS) $(pixelcity_DEPENDENCIES) 
 	@rm -f pixelcity$(EXEEXT)
-	$(LINK) $(pixelcity_OBJECTS) $(pixelcity_LDADD) $(LIBS)
+	$(CXXLINK) $(pixelcity_OBJECTS) $(pixelcity_LDADD) $(LIBS)
 plasma$(EXEEXT): $(plasma_OBJECTS) $(plasma_DEPENDENCIES) 
 	@rm -f plasma$(EXEEXT)
 	$(CXXLINK) $(plasma_OBJECTS) $(plasma_LDADD) $(LIBS)
