--- rss-glx_0.9.1/utils/rss-glx_install.pl	2013-07-14 12:33:48.092146896 +0400
+++ rss-glx_0.9.1/utils/rss-glx_install.pl	2013-07-14 12:33:48.135019308 +0400
@@ -1,10 +1,10 @@
 #!/usr/bin/env perl
 use strict;
 
-if (`ps x | grep ' xscreensaver ' | grep -v grep`) {
-	print "XScreenSaver seems to be running. Exit it before running this script.\n";
-	exit;
-}
+#if (`ps x | grep ' xscreensaver ' | grep -v grep`) {
+#	print "XScreenSaver seems to be running. Exit it before running this script.\n";
+#	exit;
+#}
 
 my $screensavers = { };
 
@@ -32,15 +32,15 @@
 $screensavers->{'feedback'}	= '  GL:               "Feedback"  feedback --root                             \\n\\';
 $screensavers->{'pixelcity'}	= '  GL:             "Pixel City"  pixelcity --root                            \\n\\';
 
-open XSCREENSAVER, "$ENV{'HOME'}/.xscreensaver";
+open XSCREENSAVER, "$ARGV[0]";
 my @xscreensaver_config_file = <XSCREENSAVER>;
 close XSCREENSAVER;
 
-open XSCREENSAVER, ">$ENV{'HOME'}/.xscreensaver";
+open XSCREENSAVER, ">$ARGV[0]";
 
 my $programs_section_flag = 0;
 foreach my $line (@xscreensaver_config_file) {
-	if ($line =~ /^programs:/) {
+	if ($line =~ /^\*programs:/) {
 		$programs_section_flag = 1;
 	} elsif ($programs_section_flag) {
 		if ($line =~ /\\\s+/) {
