--- ipython-7.34.0/IPython/terminal/ipapp.py.orig
+++ ipython-7.34.0/IPython/terminal/ipapp.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # encoding: utf-8
 """
 The :class:`~traitlets.config.application.Application` object for the command
--- ipython-7.34.0/IPython/testing/plugin/iptest.py.orig
+++ ipython-7.34.0/IPython/testing/plugin/iptest.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 """Nose-based test runner.
 """
 
--- ipython-7.34.0/IPython/testing/plugin/setup.py.orig
+++ ipython-7.34.0/IPython/testing/plugin/setup.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 """A Nose plugin to support IPython doctests.
 """
 
