--- hypothesis-hypothesis-python-6.115.0/hypothesis-python/tox.ini.orig
+++ hypothesis-hypothesis-python-6.115.0/hypothesis-python/tox.ini
@@ -4,7 +4,7 @@
 
 [testenv]
 deps =
-    -r../requirements/test.txt
+    -r../requirements/test.in
 allowlist_externals =
     bash
 passenv=
@@ -22,7 +22,7 @@
     brief: python -bb -X dev -m pytest -n auto tests/cover/test_testdecorators.py {posargs}
     cover: python -bb -X dev -m pytest -n auto tests/cover/ tests/pytest/ tests/conjecture/ {posargs}
     conjecture: python -bb -X dev -m pytest -n auto tests/conjecture/ {posargs}
-    nocover: python -bb -X dev -m pytest -n auto tests/nocover/ {posargs}
+    python -bb -X dev -m pytest -n auto tests/nocover/ {posargs}
     niche: bash scripts/other-tests.sh
     custom: python -bb -X dev -m pytest {posargs}
 
