--- hypothesis-hypothesis-python-6.139.2/hypothesis-python/tox.ini.orig
+++ hypothesis-hypothesis-python-6.139.2/hypothesis-python/tox.ini
@@ -4,7 +4,7 @@
 
 [testenv]
 deps =
-    -r../requirements/test.txt
+    -r../requirements/test.in
 extras =
     zoneinfo
 allowlist_externals =
@@ -26,7 +26,7 @@
     cover: python -bb -X dev -m pytest -n auto tests/cover/ tests/pytest/ tests/conjecture/ {posargs}
     rest: python -bb -X dev -m pytest -n auto tests/ --ignore=tests/cover/ --ignore=tests/pytest/ --ignore=tests/conjecture/ --ignore=tests/nocover/ --ignore=tests/quality/ --ignore=tests/ghostwriter/ --ignore=tests/patching/ --ignore=tests/crosshair/test_crosshair.py {posargs}
     conjecture: python -bb -X dev -m pytest -n auto tests/conjecture/ {posargs}
-    nocover: python -bb -X dev -m pytest -n auto tests/nocover/ {posargs}
+    python -bb -X dev -m pytest -n auto tests/nocover/ {posargs}
     niche: bash scripts/other-tests.sh
     custom: python -bb -X dev -m pytest {posargs}
 
