https://github.com/HypothesisWorks/hypothesis/issues/3641

--- hypothesis-hypothesis-python-6.75.3/hypothesis-python/tests/nocover/test_type_lookup_future_annotations.py.orig
+++ hypothesis-hypothesis-python-6.75.3/hypothesis-python/tests/nocover/test_type_lookup_future_annotations.py
@@ -10,7 +10,11 @@
 
 from __future__ import annotations
 
-from typing import TypedDict, Union
+try:
+    from typing import TypedDict, Union
+except:
+    import pytest
+    pytest.skip(allow_module_level=True)
 
 import pytest
 
