https://github.com/pytest-dev/pytest/issues/12186

--- hypothesis-hypothesis-python-6.100.2/hypothesis-python/tests/nocover/test_baseexception.py.orig
+++ hypothesis-hypothesis-python-6.100.2/hypothesis-python/tests/nocover/test_baseexception.py
@@ -125,6 +125,6 @@
         exception=exc_name, strategy="things()" if use_composite else "st.none()"
     )
     test_file = str(testdir.makepyfile(code))
-    pytest_stdout = str(testdir.runpytest_inprocess(test_file, "--tb=native", "-p", "no:checkdocs", "-p", "no:reporter").stdout)
+    pytest_stdout = str(testdir.runpytest_inprocess(test_file, "--tb=native", "-p", "no:checkdocs", "-p", "no:reporter", "-p", "no:system-statistics", "-p", "no:salt-factories", "-p", "no:salt-factories-event-listener", "-p", "no:salt-factories-factories", "-p", "no:salt-factories-loader-mock", "-p", "no:salt-factories-log-server", "-p", "no:salt-factories-markers").stdout)
     assert "x=101" in pytest_stdout
     assert exc_name in pytest_stdout
--- hypothesis-hypothesis-python-6.100.2/hypothesis-python/tests/nocover/test_targeting.py.orig
+++ hypothesis-hypothesis-python-6.100.2/hypothesis-python/tests/nocover/test_targeting.py
@@ -29,7 +29,7 @@
 @pytest.mark.parametrize("multiple", [False, True])
 def test_reports_target_results(testdir, multiple):
     script = testdir.makepyfile(TESTSUITE.format("" if multiple else "# "))
-    result = testdir.runpytest(script, "--tb=native", "-rN")
+    result = testdir.runpytest(script, "--tb=native", "-rN", "-p", "no:system-statistics", "-p", "no:salt-factories", "-p", "no:salt-factories-event-listener", "-p", "no:salt-factories-factories", "-p", "no:salt-factories-loader-mock", "-p", "no:salt-factories-log-server", "-p", "no:salt-factories-markers")
     out = "\n".join(result.stdout.lines)
     assert "Falsifying example" in out
     assert "x=101" in out
