https://github.com/dateutil/dateutil/issues/1261

--- python-dateutil-2.9.0.post0/tox.ini.orig
+++ python-dateutil-2.9.0.post0/tox.ini
@@ -28,7 +28,7 @@
 [testenv]
 description = run the unit tests with pytest under {basepython}
 setenv = COVERAGE_FILE={toxworkdir}/.coverage.{envname}
-passenv = DATEUTIL_MAY_CHANGE_TZ TOXENV CI TRAVIS TRAVIS_* APPVEYOR APPVEYOR_* CODECOV_* SYSTEM_* AGENT_* BUILD_* TF_BUILD
+passenv = DATEUTIL_MAY_CHANGE_TZ, TOXENV, CI, TRAVIS, TRAVIS_*, APPVEYOR, APPVEYOR_*, CODECOV_*, SYSTEM_*, AGENT_*, BUILD_*, TF_BUILD
 commands = python -m pytest {posargs: "{toxinidir}/tests" "{toxinidir}/docs" --cov-config="{toxinidir}/tox.ini" --cov=dateutil}
 deps = -rrequirements-dev.txt
 
