https://bugs.launchpad.net/subunit/+bug/2136951

--- subunit-1.4.5/python/subunit/tests/__init__.py.orig
+++ subunit-1.4.5/python/subunit/tests/__init__.py
@@ -39,7 +39,6 @@
     test_subunit_tags,
     test_tap2subunit,
     test_test_protocol,
-    test_test_protocol2,
     test_test_results,
 )
 
@@ -52,7 +51,6 @@
     result.addTest(loader.loadTestsFromModule(test_progress_model))
     result.addTest(loader.loadTestsFromModule(test_test_results))
     result.addTest(loader.loadTestsFromModule(test_test_protocol))
-    result.addTest(loader.loadTestsFromModule(test_test_protocol2))
     result.addTest(loader.loadTestsFromModule(test_tap2subunit))
     result.addTest(loader.loadTestsFromModule(test_filter_to_disk))
     result.addTest(loader.loadTestsFromModule(test_subunit_filter))
