diff -urN gnome-control-center-2.30.1-orig/capplets/default-applications/gnome-default-applications.xml.in gnome-control-center-2.30.1/capplets/default-applications/gnome-default-applications.xml.in
--- gnome-control-center-2.30.1-orig/capplets/default-applications/gnome-default-applications.xml.in	2011-03-29 16:16:08.359816379 +0800
+++ gnome-control-center-2.30.1/capplets/default-applications/gnome-default-applications.xml.in	2011-03-29 16:16:31.982465085 +0800
@@ -409,13 +409,6 @@
 
   <a11y-mobility>
     <mobility>
-      <_name>GNOME OnScreen Keyboard</_name>
-      <executable>gok</executable>
-      <command>gok</command>
-      <icon-name>accessibility-keyboard-capplet</icon-name>
-      <run-at-start>false</run-at-start>
-    </mobility>
-    <mobility>
       <_name>Dasher</_name>
       <executable>dasher</executable>
       <command>dasher</command>
