--- gnome-control-center-2.32.1/capplets/display/xrandr-capplet.c	2016-03-04 09:25:52.082991969 +0300
+++ gnome-control-center-2.32.1/capplets/display/xrandr-capplet.c	2016-03-04 09:26:09.398584341 +0300
@@ -2360,7 +2360,7 @@
 
     source_filename = gnome_rr_config_get_intended_filename ();
 
-    command_line = g_strdup_printf ("pkexec %s/gnome-display-properties-install-systemwide %s %s",
+    command_line = g_strdup_printf ("/usr/bin/gksu -u root %s/gnome-display-properties-install-systemwide %s %s",
 				    SBINDIR,
 				    source_filename,
 				    dest_basename);
--- gnome-control-center-2.32.1/capplets/display/gnome-display-properties-install-systemwide.c	2016-03-04 09:27:01.529349845 +0300
+++ gnome-control-center-2.32.1/capplets/display/gnome-display-properties-install-systemwide.c	2016-03-04 09:28:34.042883919 +0300
@@ -130,8 +130,6 @@
 	uid_t uid, euid;
 	const char *source_filename;
 	const char *dest_name;
-	const char *pkexec_uid_str;
-	int pkexec_uid;
 	struct stat statbuf;
 	int err;
 	int source_fd;
@@ -197,25 +195,6 @@
 		return EXIT_FAILURE;
 	}
 
-	/* We only copy files that are really owned by the calling user */
-
-	pkexec_uid_str = g_getenv ("PKEXEC_UID");
-	if (pkexec_uid_str == NULL) {
-		g_print ("%s\n", _("This program must only be run through pkexec(1)"));
-		return EXIT_FAILURE;
-	}
-
-	if (sscanf (pkexec_uid_str, "%d", &pkexec_uid) != 1) {
-		g_print ("%s\n", _("PKEXEC_UID must be set to an integer value"));
-		return EXIT_FAILURE;
-	}
-
-	if (statbuf.st_uid != pkexec_uid) {
-		/* Translators: we are complaining that a file must be really owned by the user who called this program */
-		g_print (_("%s must be owned by you\n"), source_filename);
-		return EXIT_FAILURE;
-	}
-
 	/* We only accept basenames for the destination */
 
 	if (!is_basename (dest_name)) {
