--- control-center-2.10.1/capplets/keybindings/eggcellrendererkeys.c	Thu May 12 16:46:18 2005
+++ control-center-2.10.1-new/capplets/keybindings/eggcellrendererkeys.c	Fri May 13 12:14:24 2005
@@ -430,7 +430,8 @@ grab_key_callback (GtkWidget    *widget,
 			GDK_BUTTON2_MASK |
 			GDK_BUTTON3_MASK |
 			GDK_BUTTON4_MASK |
-			GDK_BUTTON5_MASK;
+			GDK_BUTTON5_MASK |
+			GDK_LOCK_MASK;
 
   /* filter consumed/ignored modifiers */
 
--- gnome-control-center-2.25.2/capplets/keybindings/gnome-keybinding-properties.c.ori	2008-11-18 09:57:04.000000000 +0000
+++ gnome-control-center-2.25.2/capplets/keybindings/gnome-keybinding-properties.c	2008-12-24 08:44:13.101053485 +0000
@@ -23,6 +23,7 @@
 #define MAX_CUSTOM_SHORTCUTS 1000
 #define RESPONSE_ADD 0
 #define RESPONSE_REMOVE 1
+#define ignore_modifiers        ( GDK_LOCK_MASK )
 
 typedef struct {
   char *name;
@@ -993,6 +994,9 @@
 		      -1);
 
   /* no conflict for : blanks, different modifiers, or ourselves */
+
+  new_key->mask = new_key->mask & ~(ignore_modifiers);
+
   if (element == NULL || new_key->mask != element->mask ||
       !strcmp (new_key->gconf_key, element->gconf_key))
     return FALSE;
