diff -ruN gnome-control-center-2.29.6.orig/configure.ac gnome-control-center-2.29.6/configure.ac
--- gnome-control-center-2.29.6.orig/configure.ac	2010-02-02 20:48:09.567574830 +0000
+++ gnome-control-center-2.29.6/configure.ac	2010-02-02 20:48:27.018724373 +0000
@@ -122,7 +122,7 @@
 				   ],
 				   have_libslab_deps=yes,
 				   have_libslab_deps=no)
-	WARN_CFLAGS="-Wall"
+	WARN_CFLAGS=""
 else
 	have_libslab_deps=no
 fi
