diff -up gnome-control-center-2.26.0/capplets/keybindings/01-desktop-key.xml.in-clean gnome-control-center-2.26.0/capplets/keybindings/01-desktop-key.xml.in
--- gnome-control-center-2.26.0/capplets/keybindings/01-desktop-key.xml.in-clean	2009-08-06 09:00:53.402148243 +0100
+++ gnome-control-center-2.26.0/capplets/keybindings/01-desktop-key.xml.in	2009-08-06 09:02:17.170780319 +0100
@@ -23,10 +23,16 @@
 	name="/apps/gnome_settings_daemon/keybindings/screensaver" />
 
 	<KeyListEntry
+    name="/apps/gnome_settings_daemon/keybindings/screensaver_alternative"/>
+
+	<KeyListEntry
 	name="/apps/gnome_settings_daemon/keybindings/home" />
 
 	<KeyListEntry
 	name="/apps/gnome_settings_daemon/keybindings/search" />
 
+	<KeyListEntry
+	name="/apps/gnome_settings_daemon/keybindings/display" />
+
 </KeyListEntries>
 

