--- setuptools-72.1.0/setuptools/_distutils/unixccompiler.py.orig
+++ setuptools-72.1.0/setuptools/_distutils/unixccompiler.py
@@ -116,11 +116,11 @@
     # Python extensions).
     executables = {
         'preprocessor': None,
-        'compiler': ["cc"],
-        'compiler_so': ["cc"],
-        'compiler_cxx': ["cc"],
-        'linker_so': ["cc", "-shared"],
-        'linker_exe': ["cc"],
+        'compiler': ["gcc"],
+        'compiler_so': ["gcc"],
+        'compiler_cxx': ["gcc"],
+        'linker_so': ["gcc", "-shared"],
+        'linker_exe': ["gcc"],
         'archiver': ["ar", "-cr"],
         'ranlib': None,
     }
