--- setuptools-74.1.2/setuptools/_distutils/unixccompiler.py.orig
+++ setuptools-74.1.2/setuptools/_distutils/unixccompiler.py
@@ -116,14 +116,14 @@
     # Python extensions).
     executables = {
         'preprocessor': None,
-        'compiler': ["cc"],
-        'compiler_so': ["cc"],
-        'compiler_cxx': ["c++"],
-        'compiler_so_cxx': ["c++"],
-        'linker_so': ["cc", "-shared"],
-        'linker_so_cxx': ["c++", "-shared"],
-        'linker_exe': ["cc"],
-        'linker_exe_cxx': ["c++", "-shared"],
+        'compiler': ["gcc"],
+        'compiler_so': ["gcc"],
+        'compiler_cxx': ["g++"],
+        'compiler_so_cxx': ["g++"],
+        'linker_so': ["gcc", "-shared"],
+        'linker_so_cxx': ["g++", "-shared"],
+        'linker_exe': ["gcc"],
+        'linker_exe_cxx': ["g++", "-shared"],
         'archiver': ["ar", "-cr"],
         'ranlib': None,
     }
