https://github.com/pypa/setuptools/issues/4050

--- setuptools-69.2.0/setup.cfg.orig
+++ setuptools-69.2.0/setup.cfg
@@ -59,8 +59,6 @@
 	ini2toml[lite]>=0.9
 	tomli-w>=1.0.0
 	pytest-timeout
-	pytest-perf; \
-	sys_platform != "cygwin"
 	jaraco.develop >= 7.21; python_version >= "3.9" and sys_platform != "cygwin"
 	pytest-home >= 0.5
 	mypy==1.9  # pin mypy version so a new version doesn't suddenly cause the CI to fail
