Without this patch meld can't find python3.9.

--- meld-3.21.0/meson.build	2020-04-19 03:11:05.000000000 +0000
+++ meld-3.21.0/meson.build	2022-01-02 15:56:01.696648011 +0000
@@ -15,7 +15,7 @@ endif
 
 # These requirements are kept in both `bin/meld` and `meson.build`. If you
 # update one, update the other.
-dependency('python3', version : '>= 3.6')
+dependency('python-3.9', version : '>= 3.6')
 dependency('gtk+-3.0', version: '>= 3.20')
 dependency('glib-2.0', version: '>= 2.48')
 dependency('gtksourceview-4', version: '>= 4.0.0')

