--- scheduler/Makefile.orig
+++ scheduler/Makefile
@@ -150,13 +150,6 @@
 		echo Creating $(SERVERROOT)/ssl...; \
 		$(INSTALL_DIR) -m 700 -g $(CUPS_GROUP) $(SERVERROOT)/ssl; \
 	fi
-	if test "$(STATEDIR)" != "$(SERVERROOT)"; then \
-		echo Creating $(STATEDIR)...; \
-		$(INSTALL_DIR) -m 755 $(STATEDIR); \
-	fi
-	echo Creating $(STATEDIR)/certs...
-	$(INSTALL_DIR) -m 511 -o $(CUPS_USER) -g $(CUPS_PRIMARY_SYSTEM_GROUP) \
-		$(STATEDIR)/certs
 	echo Creating $(LOGDIR)...
 	$(INSTALL_DIR) -m 755 $(LOGDIR)
 	echo Creating $(REQUESTS)...
