--- config-scripts/cups-gssapi.m4.~1~	2014-04-22 14:59:26.274159949 +0400
+++ config-scripts/cups-gssapi.m4	2014-04-22 15:00:09.613105019 +0400
@@ -42,7 +42,7 @@
 					AC_DEFINE(HAVE_GSSAPI, 1, [Whether GSSAPI is available])
 					CFLAGS="`$KRB5CONFIG --cflags` $CFLAGS"
 					CPPFLAGS="`$KRB5CONFIG --cflags` $CPPFLAGS"
-					LIBGSSAPI="-lgss `$KRB5CONFIG --libs`")
+					LIBGSSAPI="-lgss -lkrb5")
 				;;
 			*)
 				# Other platforms just ask for GSSAPI
