diff -wpruN '--exclude=*.orig' a~/config-scripts/cups-compiler.m4 a/config-scripts/cups-compiler.m4
--- a~/config-scripts/cups-compiler.m4	1970-01-01 00:00:00
+++ a/config-scripts/cups-compiler.m4	1970-01-01 00:00:00
@@ -145,7 +145,7 @@ if test -n "$GCC"; then
 
 			*)
 				CFLAGS="$CFLAGS -fPIE -pie"
-				AC_TRY_COMPILE(,,[
+				AC_TRY_LINK(,,[
 					PIEFLAGS="-fPIE -pie"
 					AC_MSG_RESULT(yes)],
 					AC_MSG_RESULT(no))

