https://github.com/stanislavlevin/pyproject_installer/pull/101

--- pyproject_installer-0.5.5/backend/metadata.py.orig
+++ pyproject_installer-0.5.5/backend/metadata.py
@@ -33,7 +33,8 @@
         and len(child.targets) == 1
         and isinstance(child.targets[0], ast.Name)
         and child.targets[0].id == "version"
-        and isinstance(child.value, ast.Str)
+        and isinstance(child.value, ast.Constant)
+        and isinstance(child.value.value, str)
     ):
         return child.value.value
 
