We do not have pyright packaged yet.

--- inline_snapshot-0.20.5/pyproject.toml.orig
+++ inline_snapshot-0.20.5/pyproject.toml
@@ -151,7 +151,6 @@
   "dirty-equals>=0.9.0",
   "hypothesis>=6.75.5",
   "mypy>=1.2.0",
-  "pyright>=1.1.359",
   "pytest-subtests>=0.11.0",
   "pytest-freezer>=0.4.8",
   "pytest-mock>=3.14.0"
--- inline_snapshot-0.20.5/tests/test_typing.py.orig
+++ inline_snapshot-0.20.5/tests/test_typing.py
@@ -3,7 +3,7 @@
 import pytest
 
 
-@pytest.fixture(params=["mypy", "pyright"])
+@pytest.fixture(params=["mypy"])
 def check_typing(pytester, request):
     prog = request.param
 
