--- icu/source/runConfigureICU.~1~	2010-08-30 23:24:42.000000000 +0400
+++ icu/source/runConfigureICU	2013-10-10 22:24:32.924465822 +0400
@@ -146,25 +146,18 @@
     Solaris)
         THE_OS=SOLARIS
         THE_COMP="Sun's CC"
-        CC=`which cc`; export CC
-        CXX=`which CC`; export CXX
         RELEASE_CFLAGS="-xO4 -xlibmil"
         RELEASE_CXXFLAGS="-O4 -xlibmil"
         ;;
     Solaris/GCC)
         THE_OS=SOLARIS
         THE_COMP="the GNU C++"
-        CC=gcc; export CC
-        CXX=g++; export CXX
         RELEASE_CFLAGS=-O1
         RELEASE_CXXFLAGS=-O3
         ;;
     SolarisX86)
         THE_OS="SOLARIS X86"
         THE_COMP="Sun's CC"
-        CC=`which cc`; export CC
-        CXX=`which CC`; export CXX
-        LDFLAGS="-L -lCrun";export LDFLAGS
         RELEASE_CFLAGS=-xO3
         RELEASE_CXXFLAGS=-O3
         ;;
