--- icu/source/data/mappings/ucmfiles.mk.orig	Fri Nov  5 20:53:41 2010
+++ icu/source/data/mappings/ucmfiles.mk	Tue Dec  7 17:38:50 2010
@@ -115,5 +115,6 @@
 icu-internal-25546.ucm lmb-excp.ucm \
 icu-internal-compound-d1.ucm icu-internal-compound-d2.ucm icu-internal-compound-d3.ucm icu-internal-compound-d4.ucm\
 icu-internal-compound-d5.ucm icu-internal-compound-d6.ucm icu-internal-compound-d7.ucm \
-icu-internal-compound-s1.ucm icu-internal-compound-s2.ucm icu-internal-compound-s3.ucm icu-internal-compound-t.ucm
+icu-internal-compound-s1.ucm icu-internal-compound-s2.ucm icu-internal-compound-s3.ucm icu-internal-compound-t.ucm \
+johab.ucm
 
