--- icu/source/config/mh-solaris.orig	2010-11-08 16:15:55.301588087 +0100
+++ icu/source/config/mh-solaris	2010-11-08 16:16:34.256065061 +0100
@@ -4,8 +4,8 @@
 ## others. All Rights Reserved.
 
 ## Flags for position independent code
-SHAREDLIBCFLAGS = -KPIC
-SHAREDLIBCXXFLAGS = -KPIC
+SHAREDLIBCFLAGS = -DPIC -fPIC
+SHAREDLIBCXXFLAGS = DPIC -fPIC
 SHAREDLIBCPPFLAGS = -DPIC
 
 ## Commands to generate dependency files
