--- gd.h 2007-05-06 13:38:20.000000000 -0800
+++ gd.h 2010-04-15 12:13:00.000000000 -0800
@@ -71,7 +71,28 @@
 /* 2.0.23: more Type 1 fonts */
 #ifndef DEFAULT_FONTPATH
 /* default fontpath for unix systems  - whatever happened to standards ! */
-#define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1:/usr/lib/X11/fonts/Type1:/usr/openwin/lib/X11/fonts/Type1"
+#define DEFAULT_FONTPATH \
+"/usr/share/fonts/TrueType/arphic:" \
+"/usr/share/fonts/TrueType/baekmuk:" \
+"/usr/share/fonts/TrueType/bh-luxi:" \
+"/usr/share/fonts/TrueType/bitstream-vera:" \
+"/usr/share/fonts/TrueType/bpg-georgian:" \
+"/usr/share/fonts/TrueType/core:" \
+"/usr/share/fonts/TrueType/dejavu:" \
+"/usr/share/fonts/TrueType/freefont:" \
+"/usr/share/fonts/TrueType/gentium:" \
+"/usr/share/fonts/TrueType/google-droid:" \
+"/usr/share/fonts/TrueType/hanyang:" \
+"/usr/share/fonts/TrueType/indic-fonts-core:" \
+"/usr/share/fonts/TrueType/ipafont:" \
+"/usr/share/fonts/TrueType/kacst:" \
+"/usr/share/fonts/TrueType/liberation:" \
+"/usr/share/fonts/TrueType/mgopen:" \
+"/usr/share/fonts/TrueType/sil:" \
+"/usr/share/fonts/TrueType/unifont:" \
+"/usr/share/fonts/X11/Type1:" \
+"/usr/share/fonts/sun/TrueType:" \
+"/usr/share/fonts/sun/Type1"
 #endif
 #ifndef PATHSEPARATOR
 #define PATHSEPARATOR ":"
