--- a/src/gdft.c 2015-01-14 13:38:20.000000000 -0800
+++ b/src/gdft.c 2016-04-11 12:13:00.000000000 -0800
@@ -74,7 +74,28 @@
 #    define DEFAULT_FONTPATH "/usr/share/fonts/truetype:/System/Library/Fonts:/Library/Fonts"
 #  else
    /* default fontpath for unix systems  - whatever happened to standards ! */
-#    define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1:/usr/lib/X11/fonts/Type1:/usr/openwin/lib/X11/fonts/Type1"
+#define DEFAULT_FONTPATH \
+"/usr/share/fonts/TrueType/arphic:" \
+"/usr/share/fonts/TrueType/baekmuk:" \
+"/usr/share/fonts/TrueType/bh-luxi:" \
+"/usr/share/fonts/TrueType/bitstream-vera:" \
+"/usr/share/fonts/TrueType/bpg-georgian:" \
+"/usr/share/fonts/TrueType/core:" \
+"/usr/share/fonts/TrueType/dejavu:" \
+"/usr/share/fonts/TrueType/freefont:" \
+"/usr/share/fonts/TrueType/gentium:" \
+"/usr/share/fonts/TrueType/google-droid:" \
+"/usr/share/fonts/TrueType/hanyang:" \
+"/usr/share/fonts/TrueType/indic-fonts-core:" \
+"/usr/share/fonts/TrueType/ipafont:" \
+"/usr/share/fonts/TrueType/kacst:" \
+"/usr/share/fonts/TrueType/liberation:" \
+"/usr/share/fonts/TrueType/mgopen:" \
+"/usr/share/fonts/TrueType/sil:" \
+"/usr/share/fonts/TrueType/unifont:" \
+"/usr/share/fonts/X11/Type1:" \
+"/usr/share/fonts/sun/TrueType:" \
+"/usr/share/fonts/sun/Type1"
 #  endif
 #endif
 
