--- liboil-0.3.16/liboil/liboilcpu-x86.c-orig	2010-04-08 13:52:37.069918066 -0500
+++ liboil-0.3.16/liboil/liboilcpu-x86.c	2010-04-08 13:54:34.368071110 -0500
@@ -273,10 +273,8 @@ oil_cpu_detect_getisax (void)
   if (ui & AV_386_MMX) {
      oil_cpu_flags |= OIL_IMPL_FLAG_MMX;
   }
-  if (ui & AV_386_SSE) {
-     oil_cpu_flags |= OIL_IMPL_FLAG_SSE;
-  }
   if (ui & AV_386_SSE2) {
+     oil_cpu_flags |= OIL_IMPL_FLAG_SSE;
      oil_cpu_flags |= OIL_IMPL_FLAG_SSE2;
      oil_cpu_flags |= OIL_IMPL_FLAG_MMXEXT;
   }
