--- net-snmp-5.4.1/perl/agent/Makefile.PL	Thu May 31 04:23:09 2007
+++ net-snmp-5.4.1.mod/perl/agent/Makefile.PL	Tue Jul  7 06:34:37 2009
@@ -90,7 +90,7 @@
 	    $Params{'LIBS'} = "-L../../snmplib/.libs -L../../snmplib/ -L../../agent/.libs -L../../agent/ -L../../agent/helpers/.libs -L../../agent/helpers/ " . $Params{'LIBS'};
 	    $Params{'CCFLAGS'} = "-I../../include " . $Params{'CCFLAGS'};
 #	} else {
-	    $Params{'LIBS'} = `$opts->{'nsconfig'} --libdir` . $Params{'LIBS'};
+	    $Params{'LIBS'} = $Params{'LIBS'} . `$opts->{'nsconfig'} --libdir` 
 #	    $Params{'PREREQ_PM'} = {'NetSNMP::OID' => '0.1'};
 	}
 	$Params{'CCFLAGS'} =~ s/ -W(all|inline|strict-prototypes|write-strings|cast-qual|no-char-subscripts)//g; # ignore developer warnings
