--- net-snmp-5.9.1/perl/agent/Makefile.PL.orig	2021-05-26 00:19:35.000000000 +0000
+++ net-snmp-5.9.1/perl/agent/Makefile.PL	2022-12-25 13:12:48.480816285 +0000
@@ -88,7 +88,7 @@
                 " -L" . abs_path("../../agent") .
                 " " . $Params{'LIBS'};
 #	} else {
-	    $Params{'LIBS'} = `$opts->{'nsconfig'} --libdir` . $Params{'LIBS'};
+	    $Params{'LIBS'} = $Params{'LIBS'} . `$opts->{'nsconfig'} --libdir` 
 #	    $Params{'PREREQ_PM'} = {'NetSNMP::OID' => '0.1'};
 	}
     }
