--- net-snmp-5.8/python/setup.py.~1~	2018-07-16 16:33:40.000000000 +0000
+++ net-snmp-5.8/python/setup.py	2019-12-10 15:53:58.469966601 +0000
@@ -16,9 +16,13 @@
 
 if intree:
     netsnmp_libs = os.popen(basedir+'/net-snmp-config --libs').read()
+    netsnmp_ldflags = os.popen(basedir+'/net-snmp-config --ldflags').read()
+    netsnmp_cflags = os.popen(basedir+'/net-snmp-config --cflags').read()
     libdir = os.popen(basedir+'/net-snmp-config --build-lib-dirs '+basedir).read()
     incdir = os.popen(basedir+'/net-snmp-config --build-includes '+basedir).read() + " " + os.popen(basedir+'/net-snmp-config --base-cflags '+basedir).read()
     libs = re.findall(r"-l(\S+)", netsnmp_libs)
+    ldflags = re.findall(r"(\S+)", netsnmp_ldflags)
+    cflags = re.findall(r"(\S+)", netsnmp_cflags)
     libdirs = re.findall(r"-L(\S+)", libdir)
     incdirs = re.findall(r"-I(\S+)", incdir)
 else:
@@ -41,6 +45,8 @@
        Extension("netsnmp.client_intf", ["netsnmp/client_intf.c"],
                  library_dirs=libdirs,
                  include_dirs=incdirs,
-                 libraries=libs )
+                 libraries=libs,
+                 extra_compile_args=cflags,
+                 extra_link_args=ldflags)
        ]
     )
