--- net-snmp/agent/mibgroup/kernel_sunos5.c	Wed Aug 17 09:14:14 2011
+++ net-snmp/agent/mibgroup/kernel_sunos5.c	Wed Aug 17 09:17:24 2011
@@ -88,7 +88,7 @@
      0},
     {MIB_IP_ROUTE, 200 * sizeof(mib2_ipRouteEntry_t), (void *) -1, 0, 30,
      0, 0},
-    {MIB_IP_NET, 100 * sizeof(mib2_ipNetToMediaEntry_t), (void *) -1, 0,
+    {MIB_IP_NET, 400 * sizeof(mib2_ipNetToMediaEntry_t), (void *) -1, 0,
      300, 0, 0},
     {MIB_ICMP, sizeof(mib2_icmp_t), (void *) -1, 0, 60, 0, 0},
     {MIB_TCP, sizeof(mib2_tcp_t), (void *) -1, 0, 60, 0, 0},
