--- net-snmp-5.9.4/Makefile.in.orig
+++ net-snmp-5.9.4/Makefile.in
@@ -234,7 +234,7 @@
         fi
 
 pythoninstall:
-	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir --root=$(DESTDIR) --prefix=$(prefix)) ; \
+	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir --root $(DESTDIR) --prefix=$(prefix) --install-lib=$(PYTHON_VENDOR_PACKAGES)) ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
