--- a/agent/mibgroup/util_funcs.c	2007-01-11 23:13:56.000000000 +0100
+++ b/agent/mibgroup/util_funcs.c	2012-06-01 13:20:07.243670066 +0200
@@ -254,7 +254,7 @@
 #if HAVE_EXECV
     char            cachefile[STRMAX];
     char            cache[NETSNMP_MAXCACHESIZE];
-    ssize_t         cachebytes;
+    int             cachebytes;
     int             cfd;
 #ifdef NETSNMP_EXCACHETIME
     long            curtime;
