--- net-snmp-5.4.1/doxygen.conf.orig	2012-09-27 11:27:49.421496871 -0700
+++ net-snmp-5.4.1/doxygen.conf	2012-09-27 11:29:25.510013537 -0700
@@ -422,6 +422,12 @@
 
 HTML_STYLESHEET        = 
 
+# If the HTML_TIMESTAMP tag is set to YES then the footer of
+# each generated HTML page will contain the date and time when the page
+# was generated. Setting this to NO can help when comparing the output of
+# multiple runs.
+HTML_TIMESTAMP         = NO
+
 # If the HTML_ALIGN_MEMBERS tag is set to YES, the members of classes, 
 # files or namespaces will be aligned in HTML using tables. If set to 
 # NO a bullet list will be used.
