diff -Nur a/configure b/configure
--- a/configure	2017-09-12 13:48:29.516214596 +0000
+++ b/configure	2017-09-12 13:49:37.746621912 +0000
@@ -11985,7 +11985,7 @@
       ;;
 
     solaris*)
-      lt_prog_compiler_pic='-KPIC'
+      lt_prog_compiler_pic='-fPIC'
       lt_prog_compiler_static='-Bstatic'
       case $cc_basename in
       f77* | f90* | f95* | sunf77* | sunf90* | sunf95*)
@@ -23410,7 +23410,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Perl cc" >&5
 $as_echo_n "checking for Perl cc... " >&6; }
 
-        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([ -=\w\s\/]+).;\s*/$1/);'`
 
         if test "x$PERLCC" != "x" ; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERLCC" >&5
