Solaris specific patch, not suitable for upstream.
--- net-snmp-5.8.old/man/snmptrapd.conf.5.def	2018-07-16 07:33:40.000000000 +0000
+++ net-snmp-5.8/man/snmptrapd.conf.5.def	2019-03-08 05:22:34.480553145 +0000
@@ -298,7 +298,7 @@
 The default in each case is for the feature to be turned off, so these
 directives are typically only used to enable the appropriate behaviour.
 .SH FILES
-SYSCONFDIR/snmp/snmptrapd.conf
+DATADIR/snmp/snmptrapd.conf
 .SH "SEE ALSO"
 snmp_config(5), snmptrapd(8), syslog(8), traptoemail(1), variables(5), netsnmp_config_api(3).
 
--- net-snmp-5.8.old/man/snmpd.conf.5.def	2018-07-16 07:33:40.000000000 +0000
+++ net-snmp-5.8/man/snmpd.conf.5.def	2019-03-08 05:22:55.475802327 +0000
@@ -1818,7 +1818,7 @@
 detailed example of how the above information is used in real
 examples.
 .SH "FILES"
-SYSCONFDIR/snmp/snmpd.conf
+DATADIR/snmp/snmpd.conf
 .SH "SEE ALSO"
 snmpconf(1), snmpusm(1), snmp.conf(5), snmp_config(5), snmpd(8), EXAMPLE.conf, netsnmp_config_api(3).
 .\" Local Variables:
