diff -Nru net-snmp-5.9.4.orig/configure net-snmp-5.9.4/configure
--- net-snmp-5.9.4.orig/configure	2023-08-15 22:32:24.000000000 +0200
+++ net-snmp-5.9.4/configure	2023-10-08 16:30:14.614812598 +0200
@@ -31670,7 +31670,7 @@
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: ${type}" >&5
 printf "%s\n" "${type}" >&6; }
 
-printf "%s\n" "#define NETSNMP_FD_MASK_TYPE ${type}" >>confdefs.h
+printf "%s\n" "#define NETSNMP_FD_MASK_TYPE long int" >>confdefs.h
 
 
 CFLAGS=$netsnmp_save_CFLAGS
