https://www.illumos.org/issues/9042
--- tmux-2.6/compat/forkpty-sunos.c.1	Mon Feb  5 11:32:16 2018
+++ tmux-2.6/compat/forkpty-sunos.c	Mon Feb  5 11:43:20 2018
@@ -61,11 +61,13 @@
 		if (ioctl(slave, TIOCSCTTY, NULL) == -1)
 			fatal("ioctl failed");
 #endif
+#if !defined(_XPG4)
 
 		if (ioctl(slave, I_PUSH, "ptem") == -1)
 			fatal("ioctl failed");
 		if (ioctl(slave, I_PUSH, "ldterm") == -1)
 			fatal("ioctl failed");
+#endif
 
 		if (tio != NULL && tcsetattr(slave, TCSAFLUSH, tio) == -1)
 			fatal("tcsetattr failed");
