--- wqy-zenhei/44-wqy-zenhei.conf.orig	2016-09-07 23:50:58.417851667 +0200
+++ wqy-zenhei/44-wqy-zenhei.conf	2016-09-07 23:51:46.114888416 +0200
@@ -4,9 +4,13 @@
 <fontconfig>
   <!-- modified by Madsen@SMTH -->
   <match target="font">
-        <test qual="any" name="family">
+        <test name="family">
                 <string>WenQuanYi Zen Hei</string>
+        </test> 
+        <test name="family">
                 <string>文泉驿正黑</string>
+        </test> 
+        <test name="family">
                 <string>文泉驛正黑</string>
         </test> 
         <edit name="globaladvance"><bool>false</bool></edit>
