--- wireshark.desktop.orig	Thu Jun 25 07:50:11 2009
+++ wireshark.desktop	Wed May 20 13:51:51 2009
@@ -69,8 +69,8 @@
 Comment[fr]=Analyseur de trafic réseau
 Comment[sv]=Nätverkstrafikanalysator
 Icon=wireshark.png
-TryExec=wireshark
-Exec=wireshark
+TryExec=/usr/sbin/wireshark
+Exec=/usr/sbin/wireshark
 Terminal=false
 # Categorie entry according to version 1.0 of
 # http://www.freedesktop.org/wiki/Specifications/menu-spec
