--- wireshark.desktop	Tue Nov  1 09:36:05 2011
+++ wireshark.desktop	Mon Nov  7 01:58:14 2011
@@ -69,8 +69,8 @@
 Comment[fr]=Analyseur de trafic réseau
 Comment[sv]=Nätverkstrafikanalysator
 Icon=wireshark.png
-TryExec=wireshark
-Exec=wireshark %f
+TryExec=/usr/sbin/wireshark
+Exec=/usr/sbin/wireshark %f
 Terminal=false
 MimeType=application/vnd.tcpdump.pcap;
 # Categorie entry according to version 1.0 of
