--- doc/tshark.pod.orig	Wed Oct 19 10:42:24 2011
+++ doc/tshark.pod	Wed Oct 19 10:44:12 2011
@@ -406,9 +406,9 @@
 
 Network interface names should match one of the names listed in
 "B<tshark -D>" (described above); a number, as reported by
-"B<tshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> option to B<ifconfig>.
+"B<tshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<TShark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
@@ -996,7 +996,7 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(5) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 READ FILTER SYNTAX
@@ -1228,7 +1228,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), wireshark(1), editcap(1), pcap-filter(4), tcpdump(8),
+wireshark-filter(4), wireshark(1), editcap(1), pcap-filter(5), tcpdump(1),
 pcap(3), dumpcap(1), text2pcap(1), mergecap(1)
 
 =head1 NOTES
--- doc/wireshark.pod.template.orig	Wed Oct 19 10:42:29 2011
+++ doc/wireshark.pod.template	Wed Oct 19 10:43:51 2011
@@ -309,9 +309,9 @@
 
 Network interface names should match one of the names listed in
 "B<wireshark -D>" (described above); a number, as reported by
-"B<wireshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> flag to B<ifconfig>.
+"B<wireshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<Wireshark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
@@ -2343,7 +2343,7 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(5) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 DISPLAY FILTER SYNTAX
@@ -2678,7 +2678,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), tshark(1), editcap(1), pcap-filter(4), tcpdump(8),
+wireshark-filter(4), tshark(1), editcap(1), pcap-filter(5), tcpdump(1),
 pcap(3), dumpcap(1), mergecap(1), text2pcap(1)
 
 =head1 NOTES
--- doc/capinfos.pod.orig	Wed Oct 26 09:10:07 2011
+++ doc/capinfos.pod	Wed Oct 26 09:10:26 2011
@@ -317,7 +317,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), mergecap(1), editcap(1), tshark(1),
+tcpdump(1), pcap(3), wireshark(1), mergecap(1), editcap(1), tshark(1),
 dumpcap(1)
 
 =head1 NOTES
--- doc/dumpcap.pod.orig	Wed Oct 26 09:13:27 2011
+++ doc/dumpcap.pod	Wed Oct 26 09:14:40 2011
@@ -259,13 +259,13 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 SEE ALSO
 
 wireshark(1), tshark(1), editcap(1), mergecap(1), capinfos(1), pcap-filter(4),
-tcpdump(8), pcap(3)
+tcpdump(1), pcap(3)
 
 =head1 NOTES
 
--- doc/editcap.pod.orig	Wed Oct 26 09:13:34 2011
+++ doc/editcap.pod	Wed Oct 26 09:14:47 2011
@@ -340,7 +340,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), tshark(1), mergecap(1), dumpcap(1),
+tcpdump(1), pcap(3), wireshark(1), tshark(1), mergecap(1), dumpcap(1),
 capinfos(1), text2pcap(1), od(1)
 
 =head1 NOTES
--- doc/mergecap.pod.orig	Wed Oct 26 09:13:43 2011
+++ doc/mergecap.pod	Wed Oct 26 09:14:55 2011
@@ -134,7 +134,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), tshark(1), dumpcap(1), editcap(1),
+tcpdump(1), pcap(3), wireshark(1), tshark(1), dumpcap(1), editcap(1),
 text2pcap(1)
 
 =head1 NOTES
--- doc/rawshark.pod.orig	Wed Oct 26 09:13:52 2011
+++ doc/rawshark.pod	Wed Oct 26 09:15:02 2011
@@ -467,7 +467,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), wireshark(1), tshark(1), editcap(1), tcpdump(8),
+wireshark-filter(4), wireshark(1), tshark(1), editcap(1), tcpdump(1),
 pcap(3), dumpcap(1), text2pcap(1)
 
 =head1 NOTES
--- doc/text2pcap.pod.orig	Wed Oct 26 09:14:00 2011
+++ doc/text2pcap.pod	Wed Oct 26 09:15:09 2011
@@ -198,7 +198,7 @@
 
 =head1 SEE ALSO
 
-od(1), tcpdump(8), pcap(3), wireshark(1), tshark(1), dumpcap(1), mergecap(1),
+od(1), tcpdump(1), pcap(3), wireshark(1), tshark(1), dumpcap(1), mergecap(1),
 editcap(1), strptime(3).
 
 =head1 NOTES
--- doc/wireshark-filter.pod.template.orig	Wed Oct 26 09:14:08 2011
+++ doc/wireshark-filter.pod.template	Wed Oct 26 09:15:19 2011
@@ -387,12 +387,12 @@
 
 This manpage does not describe the capture filter syntax, which is
 different. See the manual page of pcap-filter(4) or, if that doesn't exist,
-tcpdump(8), or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>
+tcpdump(1), or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>
 for a description of capture filters.
 
 =head1 SEE ALSO
 
-wireshark(1), tshark(1), editcap(1), pcap-filter(4), tcpdump(8), pcap(3)
+wireshark(1), tshark(1), editcap(1), pcap-filter(4), tcpdump(1), pcap(3)
 
 =head1 AUTHORS
 
