--- doc/tshark.pod	Tue May 31 09:12:34 2011
+++ doc/tshark.pod	Wed Jul 13 07:52:24 2011
@@ -406,9 +406,9 @@
 
 Network interface names should match one of the names listed in
 "B<tshark -D>" (described above); a number, as reported by
-"B<tshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> option to B<ifconfig>.
+"B<tshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<TShark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
--- doc/wireshark.pod.template	Tue May 31 09:12:34 2011
+++ doc/wireshark.pod.template	Wed Jul 13 07:56:06 2011
@@ -309,9 +309,9 @@
 
 Network interface names should match one of the names listed in
 "B<wireshark -D>" (described above); a number, as reported by
-"B<wireshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> flag to B<ifconfig>.
+"B<wireshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<Wireshark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
