--- doc/tshark.pod
+++ doc/tshark.pod
@@ -419,9 +419,9 @@
 
 Network interface names should match one of the names listed in
 "B<tshark -D>" (described above); a number, as reported by
-"B<tshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> option to B<ifconfig>.
+"B<tshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<TShark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
@@ -1102,7 +1102,7 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(5) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 READ FILTER SYNTAX
@@ -1355,7 +1355,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), wireshark(1), editcap(1), pcap-filter(4), tcpdump(8),
+wireshark-filter(4), wireshark(1), editcap(1), pcap-filter(5), tcpdump(1),
 pcap(3), dumpcap(1), text2pcap(1), mergecap(1)
 
 =head1 NOTES
--- doc/wireshark.pod.template
+++ doc/wireshark.pod.template
@@ -309,9 +309,9 @@
 
 Network interface names should match one of the names listed in
 "B<wireshark -D>" (described above); a number, as reported by
-"B<wireshark -D>", can also be used.  If you're using UNIX, "B<netstat
--i>" or "B<ifconfig -a>" might also work to list interface names,
-although not all versions of UNIX support the B<-a> flag to B<ifconfig>.
+"B<wireshark -D>", can also be used.  On Solaris, all network interfaces
+that are displayed with the command "dladm show-link" or "ipadm show-if"
+can be used with the "-i" command line option.
 
 If no interface is specified, B<Wireshark> searches the list of
 interfaces, choosing the first non-loopback interface if there are any
@@ -2349,7 +2349,7 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(5) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 DISPLAY FILTER SYNTAX
@@ -2705,7 +2705,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), tshark(1), editcap(1), pcap-filter(4), tcpdump(8),
+wireshark-filter(4), tshark(1), editcap(1), pcap-filter(5), tcpdump(1),
 pcap(3), dumpcap(1), mergecap(1), text2pcap(1)
 
 =head1 NOTES
--- doc/capinfos.pod
+++ doc/capinfos.pod
@@ -319,7 +319,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), mergecap(1), editcap(1), tshark(1),
+tcpdump(1), pcap(3), wireshark(1), mergecap(1), editcap(1), tshark(1),
 dumpcap(1)
 
 =head1 NOTES
--- doc/dumpcap.pod
+++ doc/dumpcap.pod
@@ -265,13 +265,13 @@
 
 =head1 CAPTURE FILTER SYNTAX
 
-See the manual page of pcap-filter(4) or, if that doesn't exist, tcpdump(8),
+See the manual page of pcap-filter(5) or, if that doesn't exist, tcpdump(1),
 or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>.
 
 =head1 SEE ALSO
 
-wireshark(1), tshark(1), editcap(1), mergecap(1), capinfos(1), pcap-filter(4),
-tcpdump(8), pcap(3)
+wireshark(1), tshark(1), editcap(1), mergecap(1), capinfos(1), pcap-filter(5),
+tcpdump(1), pcap(3)
 
 =head1 NOTES
 
--- doc/editcap.pod
+++ doc/editcap.pod
@@ -368,7 +368,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), tshark(1), mergecap(1), dumpcap(1),
+tcpdump(1), pcap(3), wireshark(1), tshark(1), mergecap(1), dumpcap(1),
 capinfos(1), text2pcap(1), od(1)
 
 =head1 NOTES
--- doc/mergecap.pod
+++ doc/mergecap.pod
@@ -134,7 +134,7 @@
 
 =head1 SEE ALSO
 
-tcpdump(8), pcap(3), wireshark(1), tshark(1), dumpcap(1), editcap(1),
+tcpdump(1), pcap(3), wireshark(1), tshark(1), dumpcap(1), editcap(1),
 text2pcap(1)
 
 =head1 NOTES
--- doc/rawshark.pod
+++ doc/rawshark.pod
@@ -488,7 +488,7 @@
 
 =head1 SEE ALSO
 
-wireshark-filter(4), wireshark(1), tshark(1), editcap(1), tcpdump(8),
+wireshark-filter(4), wireshark(1), tshark(1), editcap(1), tcpdump(1),
 pcap(3), dumpcap(1), text2pcap(1)
 
 =head1 NOTES
--- doc/text2pcap.pod
+++ doc/text2pcap.pod
@@ -202,7 +202,7 @@
 
 =head1 SEE ALSO
 
-od(1), tcpdump(8), pcap(3), wireshark(1), tshark(1), dumpcap(1), mergecap(1),
+od(1), tcpdump(1), pcap(3), wireshark(1), tshark(1), dumpcap(1), mergecap(1),
 editcap(1), strptime(3).
 
 =head1 NOTES
--- doc/wireshark-filter.pod
+++ doc/wireshark-filter.pod
@@ -400,13 +400,13 @@
 the Perl-Compatible Regular Expressions library: see B<http://www.pcre.org/>.
 
 This manpage does not describe the capture filter syntax, which is
-different. See the manual page of pcap-filter(4) or, if that doesn't exist,
-tcpdump(8), or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>
+different. See the manual page of pcap-filter(5) or, if that doesn't exist,
+tcpdump(1), or, if that doesn't exist, L<http://wiki.wireshark.org/CaptureFilters>
 for a description of capture filters.
 
 =head1 SEE ALSO
 
-wireshark(1), tshark(1), editcap(1), pcap-filter(4), tcpdump(8), pcap(3)
+wireshark(1), tshark(1), editcap(1), pcap-filter(5), tcpdump(1), pcap(3)
 
 =head1 AUTHORS
 
