http://sourceforge.net/projects/sccs/files/sccs-5.07-patch01/download
--- a/sccs/sccs/lib/comobj/src/date_bal.c-orig	Do Okt 31 22:38:28 2013
+++ b/sccs/sccs/lib/comobj/src/date_bal.c	        Mo Jan 26 20:12:15 2015
@@ -27,14 +27,14 @@
  * Use is subject to license terms.
  */
 /*
- * This file contains modifications Copyright 2008-2013 J. Schilling
+ * This file contains modifications Copyright 2008-2015 J. Schilling
  *
- * @(#)date_bal.c	1.10 13/10/31 J. Schilling
+ * @(#)date_bal.c	1.11 15/01/26 J. Schilling
  *
  * From Sun: @(#)sccs:lib/comobj/date_ba.c @(#)date_ba.c 1.5 06/12/12
  */
 #if defined(sun)
-#pragma ident "@(#)date_bal.c 1.10 13/10/31 J. Schilling"
+#pragma ident "@(#)date_bal.c 1.11 15/01/26 J. Schilling"
 #endif
 # include	<defines.h>
 
@@ -115,10 +115,12 @@
 		register int	n;
 
 		--p;
-		if (z < 0)
+		if (z < 0) {
 			*p++ = '-';
-		else
+			z = -z;
+		} else {
 			*p++ = '+';
+		}
 		n = z / 60;
 		DO2_(p, n);
 		n = z % 60;
