https://github.com/networkupstools/nut/issues/2181

--- PyNUTClient-2.8.1/setup.py.orig
+++ PyNUTClient-2.8.1/setup.py
@@ -24,7 +24,6 @@
     long_description_content_type="text/plain",	# NOTE: No asciidoc so far, see https://packaging.python.org/en/latest/specifications/core-metadata/
     long_description=long_description,
     packages=find_packages(),
-    install_requires=['telnetlib'],
     keywords=['pypi', 'cicd', 'python'],
     classifiers=[
         "Development Status :: 5 - Production/Stable",
