https://github.com/networkupstools/nut/issues/2193

--- PyNUTClient-2.8.1.post1/setup.py.orig
+++ PyNUTClient-2.8.1.post1/setup.py
@@ -27,7 +27,6 @@
     packages   =	find_packages(),
     #py_modules =	['PyNUT'],
     package_dir =	{'PyNUT': 'PyNUTClient'},
-    data_files =	[('', ['tox.ini'])],
     #scripts    =	['PyNUTClient/test_nutclient.py', 'PyNUTClient/__init__.py'],
     python_requires =	'>=2.6',
     # install_requires =	['telnetlib'],	# NOTE: telnetlib.py is part of Python core for tested 2.x and 3.x versions, not something 'pip' can download
