https://trac.ffmpeg.org/ticket/2016

--- ffmpeg-2.7.2/configure.~2~	2015-07-24 23:41:43.016451238 +0300
+++ ffmpeg-2.7.2/configure	2015-07-24 23:43:45.494055319 +0300
@@ -5113,7 +5113,8 @@
 enabled libopencore_amrwb && require libopencore_amrwb opencore-amrwb/dec_if.h D_IF_init -lopencore-amrwb
 enabled libopencv         && require_pkg_config opencv opencv/cxcore.h cvCreateImageHeader
 enabled libopenh264       && require_pkg_config openh264 wels/codec_api.h WelsGetCodecVersion
-enabled libopenjpeg       && { check_lib openjpeg.h opj_version -lopenmj2 -DOPJ_STATIC ||
+enabled libopenjpeg       && { check_lib openjpeg-2.1/openmj2.h opj_version -lopenmj2 ||
+                               check_lib openjpeg.h opj_version -lopenmj2 -DOPJ_STATIC ||
                                check_lib openjpeg-1.5/openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC ||
                                check_lib openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC ||
                                die "ERROR: libopenjpeg not found"; }
--- ffmpeg-2.7.2/libavcodec/libopenjpegenc.c.~1~	2015-06-19 23:44:39.000000000 +0300
+++ ffmpeg-2.7.2/libavcodec/libopenjpegenc.c	2015-07-24 23:41:43.023290629 +0300
@@ -37,7 +37,11 @@
 #if HAVE_OPENJPEG_1_5_OPENJPEG_H
 # include <openjpeg-1.5/openjpeg.h>
 #else
-# include <openjpeg.h>
+  #if HAVE_OPENJPEG_2_0_OPENJPEG_H
+    # include <openjpeg-2.1/openmj2.h>
+  #else     
+    # include <openjpeg.h>
+  #endif  
 #endif
 
 typedef struct LibOpenJPEGContext {
--- ffmpeg-2.7.2/libavcodec/libopenjpegdec.c.~1~	2015-07-25 00:01:25.945133666 +0300
+++ ffmpeg-2.7.2/libavcodec/libopenjpegdec.c	2015-07-25 00:02:08.560600649 +0300
@@ -39,7 +39,11 @@
 #if HAVE_OPENJPEG_1_5_OPENJPEG_H
 # include <openjpeg-1.5/openjpeg.h>
 #else
-# include <openjpeg.h>
+  #if HAVE_OPENJPEG_2_0_OPENJPEG_H
+    # include <openjpeg-2.1/openmj2.h>
+  #else     
+    # include <openjpeg.h>
+  #endif  
 #endif
 
 #define JP2_SIG_TYPE    0x6A502020
