diff -Nru ffmpeg-6.0.orig/tests/fate/filter-video.mak ffmpeg-6.0/tests/fate/filter-video.mak
--- ffmpeg-6.0.orig/tests/fate/filter-video.mak	2023-02-27 21:43:45.000000000 +0100
+++ ffmpeg-6.0/tests/fate/filter-video.mak	2023-03-21 17:37:57.139581873 +0100
@@ -523,7 +523,7 @@
 tests/pixfmts.mak: TAG = GEN
 tests/pixfmts.mak: ffmpeg$(PROGSSUF)$(EXESUF) | tests
 	$(M)printf "PIXFMTS = " > $@
-	$(Q)$(TARGET_EXEC) $(TARGET_PATH)/$< -nostdin -pix_fmts list 2> /dev/null | awk 'NR > 8 && /^IO/ { printf $$2 " " }' >> $@
+	$(Q)$(TARGET_EXEC) $(TARGET_PATH)/$< -nostdin -pix_fmts list 2> /dev/null | gawk 'NR > 8 && /^IO/ { printf $$2 " " }' >> $@
 	$(Q)printf "\n" >> $@
 
 RUNNING_PIXFMTS_TESTS := $(filter check fate fate-list fate-filter fate-vfilter fate-filter-pixdesc%,$(MAKECMDGOALS))
