diff --git a/data/gdm.schemas.in.in b/data/gdm.schemas.in.in
index 82a2223..069f31b 100644
--- a/data/gdm.schemas.in.in
+++ b/data/gdm.schemas.in.in
@@ -67,7 +67,7 @@
     <schema>
       <key>greeter/IncludeAll</key>
       <signature>b</signature>
-      <default>true</default>
+      <default>false</default>
     </schema>
 
     <schema>
--- gdm-2.27.4/gui/simple-greeter/gdm-simple-greeter.schemas.in-orig	2009-08-21 16:10:05.874005000 -0500
+++ gdm-2.27.4/gui/simple-greeter/gdm-simple-greeter.schemas.in	2009-08-21 16:10:59.867723000 -0500
@@ -71,7 +71,7 @@
       <applyto>/apps/gdm/simple-greeter/disable_user_list</applyto>
       <owner>gdm-simple-greeter</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Do not show known users in the login window</short>
         <long>Set to true to disable showing known users in the login window.</long>
--- gdm-2.28.1/data/session-setup.entries-orig	2009-10-26 16:01:20.789224457 -0500
+++ gdm-2.28.1/data/session-setup.entries	2009-10-26 16:04:34.518041811 -0500
@@ -397,6 +397,18 @@
         <bool>true</bool>
       </value>
     </entry>
+    <entry>
+      <key>/desktop/gnome/accessibility/keyboard/bouncekeys_delay</key>
+      <value>
+        <int>300</int>
+      </value>
+    </entry>
+    <entry>
+      <key>/desktop/gnome/accessibility/keyboard/slowkeys_delay</key>
+      <value>
+        <int>300</int>
+      </value>
+    </entry>
 
     <entry>
       <key>/apps/gnome_settings_daemon/keybindings/eject</key>
