--- gdm-2.30.5/daemon/gdm-session-worker.c-orig	2010-09-14 16:29:03.364190938 -0500
+++ gdm-2.30.5/daemon/gdm-session-worker.c	2010-09-14 16:39:49.210954639 -0500
@@ -1668,6 +1668,12 @@ gdm_session_worker_authorize_user (GdmSe
         /* it's possible that the user needs to change their password or pin code
          */
         if (error_code == PAM_NEW_AUTHTOK_REQD) {
+                char    *utf8_msg;
+
+                utf8_msg = convert_to_utf8 (_("Your password has expired, please change it now"));
+                gdm_session_worker_report_problem (worker, utf8_msg);
+                g_free (utf8_msg);
+
                 error_code = pam_chauthtok (worker->priv->pam_handle, PAM_CHANGE_EXPIRED_AUTHTOK);
 
                 if (error_code != PAM_SUCCESS) {
