diff -rupN INIT.2011-02-08.clean/src/cmd/ksh93/Makefile INIT.2011-02-08/src/cmd/ksh93/Makefile
--- INIT.2011-02-08.clean/src/cmd/ksh93/Makefile	2012-05-14 16:26:08.957879865 +0100
+++ INIT.2011-02-08/src/cmd/ksh93/Makefile	2012-05-14 16:30:00.712605686 +0100
@@ -39,7 +39,7 @@ SHOPT_MULTIBYTE == 1		/* multibyte chara
 SHOPT_NAMESPACE == 1		/* allow namespaces */
 SHOPT_OLDTERMIO ==		/* support both TCGETA and TCGETS */
 SHOPT_OPTIMIZE == 1		/* optimize loop invariants */
-SHOPT_PFSH == 1			/* solaris exec_attr(4) profile execution */
+SHOPT_PFSH == 0			/* solaris exec_attr(4) profile execution */
 SHOPT_P_SUID == 1		/* real uid's that require -p for set[ug]id */
 SHOPT_RAWONLY == 1		/* make viraw the only vi mode */
 SHOPT_REGRESS ==		/* enable __regress__ builtin and instrumented intercepts for testing */
