--- a/src/cmd/ksh93/sh/nvtype.c	Mon Jun 24 07:10:02 2013
+++ b/src/cmd/ksh93/sh/nvtype.c	Mon Jun 24 08:33:32 2013
@@ -925,8 +925,6 @@
 	}
 	offset = roundof(offset,sizeof(char*));
 	nv_setsize(mp,offset);
-	if(nd)
-		nd++;
 	k = roundof(sizeof(Namtype_t),sizeof(Sfdouble_t)) - sizeof(Namtype_t);
 	pp = newof(NiL, Namtype_t, 1, nnodes*NV_MINSZ + offset + size + (nnodes+nd)*sizeof(char*) + iref*sizeof(struct Namref)+k);
 	pp->fun.dsize = sizeof(Namtype_t)+nnodes*NV_MINSZ +offset+k;
