This issue has been fixed in community version: 2013-10-10
---
--- a/src/cmd/ksh93/sh/jobs.c	2013-11-12 10:32:35.378209600 -0800
+++ b/src/cmd/ksh93/sh/jobs.c	2013-11-12 10:35:10.458987800 -0800
@@ -1100,7 +1100,7 @@
 
 int job_kill(register struct process *pw,register int sig)
 {
-	Shell_t	*shp = pw->p_shp;
+	Shell_t	*shp;
 	register pid_t pid;
 	register int r;
 	const char *msg;
@@ -1113,6 +1113,7 @@
 	errno = ECHILD;
 	if(pw==0)
 		goto error;
+	shp = pw->p_shp;
 	pid = pw->p_pid;
 #if SHOPT_COSHELL
 	if(pw->p_cojob)
