diff -rupN INIT.2011-02-08.clean/src/lib/libast/port/mnt.c INIT.2011-02-08/src/lib/libast/port/mnt.c
--- INIT.2011-02-08.clean/src/lib/libast/port/mnt.c	2012-05-07 15:33:03.490042977 +0100
+++ INIT.2011-02-08/src/lib/libast/port/mnt.c	2012-05-07 15:40:58.347078946 +0100
@@ -416,11 +416,6 @@ mntread(void* handle)
 			t = "sfs";
 			break;
 #endif
-#ifdef MNT_CACHEFS
-		case MNT_CACHEFS:
-			t = "cachefs";
-			break;
-#endif
 #ifdef MNT_NFS3
 		case MNT_NFS3:
 			t = "nfs3";
