diff -rupN INIT.2011-02-08.clean/src/cmd/ksh93/sh/nvdisc.c INIT.2011-02-08/src/cmd/ksh93/sh/nvdisc.c
--- INIT.2011-02-08.clean/src/cmd/ksh93/sh/nvdisc.c	2012-05-02 03:46:22.725281017 -0700
+++ INIT.2011-02-08/src/cmd/ksh93/sh/nvdisc.c	2012-05-02 03:54:19.300127633 -0700
@@ -593,7 +593,7 @@ static char *setdisc(register Namval_t* 
 		action = vp->bltins[type];
 		vp->bltins[type] = 0;
 	}
-	return(action?(char*)action:"");
+	return((char*)action);
 }
 
 static void putdisc(Namval_t* np, const char* val, int flag, Namfun_t* fp)
