--- a/src/cmd/ksh93/data/builtins.c	Tue Dec  7 19:23:04 2010
+++ b/src/cmd/ksh93/data/builtins.c	Mon Jul 18 12:08:50 2011
@@ -108,7 +108,6 @@
 	"printf",	NV_BLTIN|BLT_ENV,		bltin(printf),
 	"pwd",		NV_BLTIN,			bltin(pwd),
 	"read",		NV_BLTIN|BLT_ENV,		bltin(read),
-	"sleep",	NV_BLTIN,			bltin(sleep),
 	"alarm",	NV_BLTIN,			bltin(alarm),
 	"ulimit",	NV_BLTIN|BLT_ENV,		bltin(ulimit),
 	"umask",	NV_BLTIN|BLT_ENV,		bltin(umask),

