diff -Naur INIT.2011-02-08.orig/src/cmd/ksh93/sh/init.c INIT.2011-02-08/src/cmd/ksh93/sh/init.c
--- INIT.2011-02-08.orig/src/cmd/ksh93/sh/init.c	2012-06-25 01:05:32.997978071 -0700
+++ INIT.2011-02-08/src/cmd/ksh93/sh/init.c	2012-06-25 01:19:10.000000000 -0700
@@ -1181,7 +1181,7 @@
 	Shell_t	*shp;
 	register int n;
 	int type;
-	static char *login_files[3];
+	static char *login_files[2];
 	memfatal();
 	n = strlen(e_version);
 	if(e_version[n-1]=='$' && e_version[n-2]==' ')
@@ -1468,7 +1468,6 @@
         shp->topscope = (Shscope_t*)shp->st.self;
 	sh_offstate(SH_INIT);
 	login_files[0] = (char*)e_profile;
-	login_files[1] = ".profile";
 	shp->gd->login_files = login_files;
 	shp->bltindata.version = SH_VERSION;
 	shp->bltindata.shp = shp;
