--- a/src/cmd/ksh93/Makefile
+++ b/src/cmd/ksh93/Makefile
@@ -28,7 +28,7 @@ SHOPT_COSHELL == 1		/* build with connection to coshell */
 SHOPT_CRNL ==			/* accept <cr><nl> for <nl> */
 SHOPT_DYNAMIC == 1		/* dynamic loading for builtins */
 SHOPT_ECHOPRINT ==		/* make echo equivalent to print */
-SHOPT_EDPREDICT == 1		/* predictive editing  */
+SHOPT_EDPREDICT == 0		/* predictive editing  */
 SHOPT_ESH == 1			/* emacs/gmacs edit mode */
 SHOPT_FILESCAN == 1		/* fast file scan */
 SHOPT_FIXEDARRAY == 1		/* fixed dimension indexed array */

