diff -rupN INIT.2011-02-08.clean/src/cmd/ksh93/Makefile INIT.2011-02-08/src/cmd/ksh93/Makefile
--- INIT.2011-02-08.clean/src/cmd/ksh93/Makefile	2012-04-11 17:12:31.210695651 +0100
+++ INIT.2011-02-08/src/cmd/ksh93/Makefile	2012-04-11 17:19:23.339887285 +0100
@@ -167,12 +167,6 @@ shell$(RELEASE) $(VERSION) id=shell :LIB
 "sol*.i386*"	:NOOPTIMIZE: main.c
 "win32*"	:NOOPTIMIZE: strdata.c trestore.c
 
-if CC.HOSTTYPE == "win32*"
-:DLL: $(CC.DEBUG)
-else
-:DLL: $(CC.OPTIMIZE)
-end
-
 :READONLY: $(DATAFILES)
 
 $(INCLUDEDIR) :INSTALLPROTO: nval.h shell.h history.h
