--- pacemaker-Pacemaker-2.0.3/Makefile.am.~1~	2020-01-15 11:01:11.359936577 +0000
+++ pacemaker-Pacemaker-2.0.3/Makefile.am	2020-01-15 11:04:19.971572195 +0000
@@ -99,12 +99,12 @@
 	$(INSTALL) -d -m 750 $(DESTDIR)/$(CRM_BLACKBOX_DIR)
 	$(INSTALL) -d -m 770 $(DESTDIR)/$(CRM_LOG_DIR)
 	$(INSTALL) -d -m 770 $(DESTDIR)/$(CRM_BUNDLE_DIR)
-	-chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PACEMAKER_CONFIG_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CONFIG_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CORE_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_BLACKBOX_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_LOG_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_BUNDLE_DIR)
+#	-chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PACEMAKER_CONFIG_DIR)
+#	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CONFIG_DIR)
+#	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CORE_DIR)
+#	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_BLACKBOX_DIR)
+#	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_LOG_DIR)
+#	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_BUNDLE_DIR)
 # Use chown because the user/group may not exist
 
 clean-generic:
--- pacemaker-Pacemaker-2.0.2/daemons/schedulerd/Makefile.am.~1~	2019-08-19 18:20:33.905492526 +0000
+++ pacemaker-Pacemaker-2.0.2/daemons/schedulerd/Makefile.am	2019-08-19 18:20:51.452080978 +0000
@@ -31,9 +31,9 @@

 install-exec-local:
	$(mkinstalldirs) $(DESTDIR)/$(PE_STATE_DIR)
-	-chown $(CRM_DAEMON_USER) $(DESTDIR)/$(PE_STATE_DIR)
-	-chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PE_STATE_DIR)
-	-chmod 750 $(DESTDIR)/$(PE_STATE_DIR)
+#	-chown $(CRM_DAEMON_USER) $(DESTDIR)/$(PE_STATE_DIR)
+#	-chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PE_STATE_DIR)
+#	-chmod 750 $(DESTDIR)/$(PE_STATE_DIR)

 if BUILD_LEGACY_LINKS
 install-exec-hook:
