diff -Nura a/pengine/Makefile.am b/pengine/Makefile.am
--- a/pengine/Makefile.am	2016-11-01 20:39:14.618708339 +0100
+++ b/pengine/Makefile.am	2016-11-01 20:39:43.964137494 +0100
@@ -76,8 +76,8 @@
 
 install-exec-local:
 	$(mkinstalldirs) $(DESTDIR)/$(PE_STATE_DIR)
-	-chown $(CRM_DAEMON_USER) $(DESTDIR)/$(PE_STATE_DIR)
-	-chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PE_STATE_DIR)
+# -chown $(CRM_DAEMON_USER) $(DESTDIR)/$(PE_STATE_DIR)
+# -chgrp $(CRM_DAEMON_GROUP) $(DESTDIR)/$(PE_STATE_DIR)
 	-chmod 750 $(DESTDIR)/$(PE_STATE_DIR)
 
 uninstall-local:
