--- a/configure.ac.1	2017-09-17 08:58:08.773671176 +0000
+++ b/configure.ac	2017-09-17 08:58:37.604466522 +0000
@@ -1086,7 +1086,7 @@
 AC_DEFINE_UNQUOTED(CRM_STATE_DIR,"$CRM_STATE_DIR", Where to keep state files and sockets)
 AC_SUBST(CRM_STATE_DIR)

-CRM_LOG_DIR="${localstatedir}/log/pacemaker"
+CRM_LOG_DIR="${localstatedir}/log/hacluster"
 AC_DEFINE_UNQUOTED(CRM_LOG_DIR,"$CRM_LOG_DIR", Where Pacemaker can store log files)
 AC_SUBST(CRM_LOG_DIR)

--- a/extra/logrotate/pacemaker.in.~1~	2017-09-17 08:53:34.128499112 +0000
+++ b/extra/logrotate/pacemaker.in	2017-09-17 08:56:47.902520059 +0000
@@ -1,4 +1,4 @@
-@localstatedir@/log/pacemaker.log @CRM_LOG_DIR@/bundles/*/pacemaker.log {
+@localstatedir@/log/hacluster/pacemaker.log @CRM_LOG_DIR@/bundles/*/pacemaker.log {
     compress
     dateext
     weekly
