diff -Nura a/Makefile.am b/Makefile.am
--- a/Makefile.am	2016-11-01 20:44:20.989796800 +0100
+++ b/Makefile.am	2016-11-01 20:49:45.320974849 +0100
@@ -52,13 +52,13 @@
 	list='$(CORE)'; for subdir in $$list; do echo "Cleaning $$subdir"; $(MAKE) -C $$subdir clean || exit 1; done
 
 install-exec-local:
-	$(INSTALL) -d $(DESTDIR)/$(LCRSODIR)
-	$(INSTALL) -d -m 750 $(DESTDIR)/$(CRM_CONFIG_DIR)
-	$(INSTALL) -d -m 750 $(DESTDIR)/$(CRM_CORE_DIR)
-	$(INSTALL) -d -m 750 $(DESTDIR)/$(CRM_BLACKBOX_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CONFIG_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_CORE_DIR)
-	-chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)/$(CRM_BLACKBOX_DIR)
+	$(INSTALL) -d $(DESTDIR)$(LCRSODIR)
+	$(INSTALL) -d -m 750 $(DESTDIR)$(CRM_CONFIG_DIR)
+	$(INSTALL) -d -m 750 $(DESTDIR)$(CRM_CORE_DIR)
+	$(INSTALL) -d -m 750 $(DESTDIR)$(CRM_BLACKBOX_DIR)
+# -chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)$(CRM_CONFIG_DIR)
+# -chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)$(CRM_CORE_DIR)
+# -chown $(CRM_DAEMON_USER):$(CRM_DAEMON_GROUP) $(DESTDIR)$(CRM_BLACKBOX_DIR)
 if BUILD_CS_PLUGIN
 	rm -f $(DESTDIR)$(LCRSODIR)/pacemaker.lcrso $(DESTDIR)$(LCRSODIR)/service_crm.so
 	cp $(DESTDIR)$(libdir)/service_crm.so $(DESTDIR)$(LCRSODIR)/pacemaker.lcrso
