--- mongodb-src-r4.0.1/src/mongo/util/time_support.cpp.orig	2018-08-10 13:04:20.915958851 +0000
+++ mongodb-src-r4.0.1/src/mongo/util/time_support.cpp	2018-08-10 13:08:14.392568977 +0000
@@ -47,7 +47,7 @@
 #include "mongo/util/timer.h"
 #include <boost/date_time/filetime_functions.hpp>
 #include <mmsystem.h>
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__sun__)
 #include <time.h>
 #elif defined(__APPLE__)
 #include <mach/clock.h>
@@ -930,7 +930,7 @@
 // Find minimum timer resolution of OS
 Nanoseconds getMinimumTimerResolution() {
     Nanoseconds minTimerResolution;
-#if defined(__linux__) || defined(__FreeBSD__)
+#if defined(__linux__) || defined(__sun__) || defined(__FreeBSD__)
     struct timespec tp;
     clock_getres(CLOCK_REALTIME, &tp);
     minTimerResolution = Nanoseconds{tp.tv_nsec};
