Developed internally, for system-test; not suitable for upstream.

--- pigz-2.3.4/Makefile.orig	2017-03-02 13:38:09.343094775 +0000
+++ pigz-2.3.4/Makefile	2017-03-02 13:48:28.190657548 +0000
@@ -1,5 +1,6 @@
 CC=cc
 CFLAGS=-O3 -Wall -Wextra
+TARGETDIR ?= $(shell pwd)
 LDFLAGS=
 LIBS=-lm -lz
 ZOPFLI=zopfli/src/zopfli/
@@ -53,36 +54,36 @@
 tryn.o: try.c try.h
 	$(CC) $(CFLAGS) -DDEBUG -DNOTHREAD -g -c -o tryn.o try.c
 
-test: pigz
-	./pigz -kf pigz.c ; ./pigz -t pigz.c.gz
-	./pigz -kfb 32 pigz.c ; ./pigz -t pigz.c.gz
-	./pigz -kfp 1 pigz.c ; ./pigz -t pigz.c.gz
-	./pigz -kfz pigz.c ; ./pigz -t pigz.c.zz
-	./pigz -kfK pigz.c ; ./pigz -t pigz.c.zip
-	printf "" | ./pigz -cdf | wc -c | test `cat` -eq 0
-	printf "x" | ./pigz -cdf | wc -c | test `cat` -eq 1
-	printf "xy" | ./pigz -cdf | wc -c | test `cat` -eq 2
-	printf "xyz" | ./pigz -cdf | wc -c | test `cat` -eq 3
-	(printf "w" | gzip ; printf "x") | ./pigz -cdf | wc -c | test `cat` -eq 2
-	(printf "w" | gzip ; printf "xy") | ./pigz -cdf | wc -c | test `cat` -eq 3
-	(printf "w" | gzip ; printf "xyz") | ./pigz -cdf | wc -c | test `cat` -eq 4
-	-@if test "`which compress | grep /`" != ""; then \
-	  echo 'compress -f < pigz.c | ./unpigz | cmp - pigz.c' ;\
-	  compress -f < pigz.c | ./unpigz | cmp - pigz.c ;\
+test: $(TARGETDIR)/pigz
+	$(TARGETDIR)/pigz -kf pigz.c ; $(TARGETDIR)/pigz -t pigz.c.gz
+	$(TARGETDIR)/pigz -kfb 32 pigz.c ; $(TARGETDIR)/pigz -t pigz.c.gz
+	$(TARGETDIR)/pigz -kfp 1 pigz.c ; $(TARGETDIR)/pigz -t pigz.c.gz
+	$(TARGETDIR)/pigz -kfz pigz.c ; $(TARGETDIR)/pigz -t pigz.c.zz
+	$(TARGETDIR)/pigz -kfK pigz.c ; $(TARGETDIR)/pigz -t pigz.c.zip
+	printf "" | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 0
+	printf "x" | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 1
+	printf "xy" | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 2
+	printf "xyz" | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 3
+	(printf "w" | gzip ; printf "x") | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 2
+	(printf "w" | gzip ; printf "xy") | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 3
+	(printf "w" | gzip ; printf "xyz") | $(TARGETDIR)/pigz -cdf | wc -c | test `cat` -eq 4
+	-@if test "`type -f compress | grep /`" != ""; then \
+	  echo 'compress -f < pigz.c | $(TARGETDIR)/unpigz | cmp - pigz.c' ;\
+	  compress -f < pigz.c | $(TARGETDIR)/unpigz | cmp - pigz.c ;\
 	fi
 	@rm -f pigz.c.gz pigz.c.zz pigz.c.zip
 	@rm -rf d/1 d/2
-	(mkdir -p d/1; cd d/1; tar xzf ../../../../pigz-2.3.3.tar.gz; \
-	  cd ..; cp -pr 1 2; ../pigz -rp 4 --index %z 1; \
-	  ../pigz -drp 4 --index %z 1; diff -r 1 2)
+	(mkdir -p d/1; cd d/1; tar xzf ../../../../pigz-2.3.4.tar.gz; \
+	  cd ..; cp -pr 1 2; $(TARGETDIR)/pigz -rp 4 --index %z 1; \
+	  $(TARGETDIR)/pigz -drp 4 --index %z 1; diff -r 1 2)
 	@rm -rf d/1 d/2
-	(mkdir -p d/1; cd d/1; tar xzf ../../../../pigz-2.3.3.tar.gz; \
-	  cd ..; cp -pr 1 2; ../pigz -zrp 4 -X %f.idx 1; \
-	  ../pigz -dzrp 4 -X %f.idx 1; diff -r 1 2)
+	(mkdir -p d/1; cd d/1; tar xzf ../../../../pigz-2.3.4.tar.gz; \
+	  cd ..; cp -pr 1 2; $(TARGETDIR)/pigz -zrp 4 -X %f.idx 1; \
+	  $(TARGETDIR)/pigz -dzrp 4 -X %f.idx 1; diff -r 1 2)
 	@rm -rf d/1 d/2
 
 tests: dev test
-	./pigzn -kf pigz.c ; ./pigz -t pigz.c.gz
+	$(TARGETDIR)/pigzn -kf pigz.c ; $(TARGETDIR)/pigz -t pigz.c.gz
 	@rm -f pigz.c.gz
 
 docs: pigz.pdf
