diff -Nura pigz-2.3.orig/Makefile pigz-2.3/Makefile
--- pigz-2.3.orig/Makefile	2013-03-04 07:06:35.000000000 +0100
+++ pigz-2.3/Makefile	2013-08-17 11:55:29.238402787 +0200
@@ -58,7 +58,7 @@
 	(printf "w" | gzip ; printf "x") | ./pigz -cdf | wc -c | test `cat` -eq 2
 	(printf "w" | gzip ; printf "xy") | ./pigz -cdf | wc -c | test `cat` -eq 3
 	(printf "w" | gzip ; printf "xyz") | ./pigz -cdf | wc -c | test `cat` -eq 4
-	-@if test "`whereis compress | grep /`" != ""; then \
+	-@if test "`type -f compress | grep /`" != ""; then \
 	  echo 'compress -f < pigz.c | ./unpigz | cmp - pigz.c' ;\
 	  compress -f < pigz.c | ./unpigz | cmp - pigz.c ;\
 	fi
