pytest-timeout is not needed to run tests.

--- chardet-6.0.0.post1/pyproject.toml.orig
+++ chardet-6.0.0.post1/pyproject.toml
@@ -58,7 +58,6 @@
 test = [
   "hypothesis>=6.0.0,<7",
   "pytest>=8.3.2,<9",
-  "pytest-timeout>=2.2.0,<3",
   "pytest-xdist>=3.0.0,<4",
 ]
 train = [
@@ -79,7 +78,7 @@
 reportTypedDictNotRequiredAccess = "none"
 
 [tool.pytest.ini_options]
-addopts = "-v -n auto --timeout=60"
+addopts = "-v -n auto"
 norecursedirs = "*"
 python_files = "test.py"
 
