--- unrar/makefile	2017-11-21 22:53:39.000000000 +0000
+++ unrar/makefile.new	2018-05-26 15:19:08.862458981 +0000
@@ -2,13 +2,13 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=c++
-CXXFLAGS=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
+#CXX=c++
+#CXXFLAGS=-O2 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
 LIBFLAGS=-fPIC
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
 STRIP=strip
 AR=ar
-LDFLAGS=-pthread
+#LDFLAGS=-pthread
 DESTDIR=/usr
 
 # Linux using LCC
