https://github.com/neithere/argh/issues/155

diff -urN argh-0.26.2.orig/reqs-dev.txt argh-0.26.2/reqs-dev.txt
--- argh-0.26.2.orig/reqs-dev.txt
+++ argh-0.26.2/reqs-dev.txt
@@ -0,0 +1,6 @@
+mock==1.0.1
+pytest>=2.3.7
+pytest-cov>=1.8.0
+pytest-xdist>=1.11
+tox==1.8.1
+iocapture==0.1.2
diff -urN argh-0.26.2.orig/tox.ini argh-0.26.2/tox.ini
--- argh-0.26.2.orig/tox.ini
+++ argh-0.26.2/tox.ini
@@ -0,0 +1,24 @@
+[tox]
+envlist=py26,py27,py32,py35,pypy
+
+[testenv]
+deps=
+    -rreqs-dev.txt
+commands=
+    py.test []
+
+[testenv:py26]
+deps=
+    {[testenv]deps}
+    argparse        # not shipped with Python < 2.7
+
+[testenv:py32]
+deps=
+    {[testenv]deps}
+    coverage==3.7.1
+
+[testenv:tdd]
+# Special case for active development phase.
+basepython=python3.5
+commands=
+    py.test --exitfirst --looponfail []
