--- illumos-kvm-a82c4d2a9555461ee43e3fd3b6bbff10e6f5458a/Makefile.orig	2022-01-27 22:57:35.000000000 +0100
+++ illumos-kvm-a82c4d2a9555461ee43e3fd3b6bbff10e6f5458a/Makefile	2025-09-27 10:43:48.404147370 +0200
@@ -2,7 +2,7 @@
 # Copyright 2020 Joyent, Inc.
 #
 
-include		$(PWD)/../../../build.env
+#include		$(PWD)/../../../build.env
 
 KERNEL_SOURCE =	$(PWD)/../../illumos
 MDB_SOURCE =	$(KERNEL_SOURCE)/usr/src/cmd/mdb
@@ -131,7 +131,7 @@
 USER_CFLAGS = \
 	-finline \
 	-gdwarf-2 \
-	-std=gnu89 \
+	-std=gnu99 \
 	-Wno-missing-braces \
 	-Wno-sign-compare \
 	-Wno-parentheses \
@@ -177,7 +177,7 @@
 	-m64 \
 	-shared \
 	-nodefaultlibs \
-	-std=gnu89 \
+	-std=gnu99 \
 	-Wl,-M$(KERNEL_SOURCE)/usr/src/common/mapfiles/common/map.pagealign \
 	-Wl,-M$(KERNEL_SOURCE)/usr/src/common/mapfiles/common/map.noexdata \
 	-Wl,-ztext \
