--- mysql_fdw-REL-2_5_0/mysql_fdw.c.~1~	2018-11-22 20:22:06.382590630 +0000
+++ mysql_fdw-REL-2_5_0/mysql_fdw.c	2018-11-22 20:23:29.950029980 +0000
@@ -200,7 +200,7 @@
 bool
 mysql_load_library(void)
 {
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__sun)
 	/*
 	 * Mac OS/BSD does not support RTLD_DEEPBIND, but it still
 	 * works without the RTLD_DEEPBIND
