--- sbcl-1.4.7/src/runtime/Config.x86-sunos.1	2018-05-16 19:07:43.713900529 +0000
+++ sbcl-1.4.7/src/runtime/Config.x86-sunos	2018-05-16 19:09:59.795730115 +0000
@@ -9,8 +9,8 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CC=gcc
-CFLAGS = -g -O2 -Wall -D__EXTENSIONS__ -D_POSIX_C_SOURCE=199506L -DSVR4 -D_REENTRANT -fno-omit-frame-pointer
+CC?=gcc
+CFLAGS = -g -O2 -Wall -D__EXTENSIONS__ -D_POSIX_C_SOURCE=200809L -DSVR4 -D_REENTRANT -fno-omit-frame-pointer
 NM = nm -xgp
 GREP = ggrep
 
