--- sbcl-2.1.1/src/runtime/Config.x86-sunos.orig	2021-01-30 12:22:38.000000000 +0000
+++ sbcl-2.1.1/src/runtime/Config.x86-sunos	2021-02-08 21:14:13.761865293 +0000
@@ -9,8 +9,8 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CC=gcc
-CFLAGS = -g -O2 -Wall -D__EXTENSIONS__ -D_POSIX_C_SOURCE=199506L -DSVR4 -D_REENTRANT -fno-omit-frame-pointer
+CC?=gcc
+CFLAGS = -g -O2 -Wall -D__EXTENSIONS__ -D_POSIX_C_SOURCE=200809L -DSVR4 -D_REENTRANT -fno-omit-frame-pointer
 GREP = ggrep
 
 ASSEM_SRC = x86-assem.S
