Add a define in order to make sigwait() POSIX conformant.

--- sbcl-2.2.5/src/runtime/Config.x86-64-sunos.orig	2022-05-30 15:27:23.000000000 +0000
+++ sbcl-2.2.5/src/runtime/Config.x86-64-sunos	2022-07-09 17:24:32.295128794 +0000
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS = -m64 -g -Wall -std=gnu89 -D__EXTENSIONS__ -DSVR4 -D_REENTRANT -fno-omit-frame-pointer
+CFLAGS = -m64 -g -Wall -std=gnu89 -D__EXTENSIONS__ -DSVR4 -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -fno-omit-frame-pointer
 LINKFLAGS = -m64 -g
 ASFLAGS = -m64 -Wall
 GREP = ggrep
