--- sbcl-2.6.2/src/runtime/thread.c.orig	2026-02-27 19:08:09.000000000 +0100
+++ sbcl-2.6.2/src/runtime/thread.c	2026-03-07 10:38:20.627344285 +0100
@@ -702,6 +702,8 @@
     stack_size = stack.ss_size;
     stack_addr = (void*)((size_t)stack.ss_sp - stack_size);
 # elif defined LISP_FEATURE_SUNOS
+    #include "/usr/include/thread.h"
+    #include <signal.h>
     stack_t stack;
     thr_stksegment(&stack);
     stack_size = stack.ss_size;
