diff -rup sexy-python-0.1.9/sexy/Makefile.am ../sexy-python-0.1.9/sexy/Makefile.am
--- sexy-python-0.1.9/sexy/Makefile.am	2005-12-13 20:12:19.000000000 +0100
+++ ../sexy-python-0.1.9/sexy/Makefile.am	2007-08-24 17:11:50.768871000 +0200
@@ -3,8 +3,8 @@ INCLUDES = $(PYTHON_INCLUDES) $(PYGTK_CF
 defsdir = $(datadir)/pygtk/2.0/defs
 defs_DATA = sexy.defs
 
-pkgpythondir = $(pyexecdir)/gtk-2.0
-pkgpyexecdir = $(pyexecdir)/gtk-2.0
+pkgpythondir = $(pyexecdir)
+pkgpyexecdir = $(pyexecdir)
 
 pkgpyexec_LTLIBRARIES = sexy.la
 sexy_la_CFLAGS = $(LIBSEXY_CFLAGS)
