--- squid-5.8/src/comm/TcpAcceptor.cc	2022-09-05 06:06:48.000000000 +0000
+++ squid-5.8/src/comm/TcpAcceptor.cc.new	2022-10-01 16:39:57.269429909 +0000
@@ -407,6 +407,7 @@
     details->local = *gai;
     Ip::Address::FreeAddr(gai);
 
+#if IPF_TRANSPARENT || LINUX_NETFILTER || IPFW_TRANSPARENT || PF_TRANSPARENT
     // Perform NAT or TPROXY operations to retrieve the real client/dest IP addresses
     if (conn->flags&(COMM_TRANSPARENT|COMM_INTERCEPTION) && !Ip::Interceptor.Lookup(details, conn)) {
         debugs(50, DBG_IMPORTANT, "ERROR: NAT/TPROXY lookup failed to locate original IPs on " << details);
@@ -414,6 +415,7 @@
         PROF_stop(comm_accept);
         return Comm::COMM_ERROR;
     }
+#endif
 
 #if USE_SQUID_EUI
     if (Eui::TheConfig.euiLookup) {
