diff -uNr libvisual-0.4.0.orig/libvisual/map.remove_all libvisual-0.4.0/libvisual/map.remove_all
--- libvisual-0.4.0.orig/libvisual/map.remove_all	1970-01-01 08:00:00.000000000 +0800
+++ libvisual-0.4.0/libvisual/map.remove_all	2009-01-07 11:44:41.487159000 +0800
@@ -0,0 +1 @@
+hwcap_1 = OVERRIDE;
diff -uNr libvisual-0.4.0.orig/map.remove_all libvisual-0.4.0/map.remove_all
--- libvisual-0.4.0.orig/map.remove_all	1970-01-01 08:00:00.000000000 +0800
+++ libvisual-0.4.0/map.remove_all	2009-01-19 14:00:20.369036000 +0800
@@ -0,0 +1 @@
+hwcap_1 = OVERRIDE;
