--- gnome-settings-daemon-2.30.2/plugins/media-keys/gsd-media-keys-manager.c.old	2012-10-18 11:31:46.976301810 +0530
+++ gnome-settings-daemon-2.30.2/plugins/media-keys/gsd-media-keys-manager.c	2012-10-18 11:35:15.855637565 +0530
@@ -581,7 +581,7 @@ do_www_action (GsdMediaKeysManager *mana
 static void
 do_exit_action (GsdMediaKeysManager *manager)
 {
-        execute (manager, "gnome-session-save --shutdown-dialog", FALSE, FALSE);
+        execute (manager, "gnome-session-save --logout-dialog", FALSE, FALSE);
 }
 
 static void
