OpenIndiana specific documentation changes.

--- unbound-1.24.0/doc/unbound.8.in.orig	2025-09-17 11:27:16.000000000 +0200
+++ unbound-1.24.0/doc/unbound.8.in	2025-09-20 09:57:30.677502562 +0200
@@ -50,6 +50,15 @@
 A second query for the same name will then be answered from the cache.
 Unbound can also do DNSSEC validation.
 .sp
+Unbound can be enabled by authorized users within the service management
+facility (SMF), using  the  following  fault  management  resource
+identifier (FMRI):
+.sp
+.RS 6n
+svc:/network/dns/server:unbound
+.RE
+.sp
+.P
 To use a locally running Unbound for resolving put:
 .INDENT 0.0
 .INDENT 3.5
