OpenIndiana specific documentation changes.

--- unbound-1.16.2/doc/unbound.8.in
+++ unbound-1.16.2/doc/unbound.8.in
@@ -35,6 +35,15 @@ certain time (the answer's time\-to\-liv
 A second query for the same name will then be answered from the cache.
 Unbound can also do DNSSEC validation.
 .P
+Unbound can be enabled by authorized users within the service management
+facility (SMF), using  the  following  fault  management  resource
+identifier (FMRI):
+.sp
+.RS 6n
+svc:/network/dns/server:unbound
+.RE
+.sp
+.P
 To use a locally running
 .B Unbound
 for resolving put
