--- modules/filters/config.m4.orig	Mon May  9 09:31:02 2011
+++ modules/filters/config.m4	Mon May  9 09:32:22 2011
@@ -10,6 +10,9 @@
 APACHE_MODULE(filter, Smart Filtering, , , yes)
 APACHE_MODULE(substitute, response content rewrite-like filtering, , , most)
 
+sed_obj="mod_sed.lo sed0.lo sed1.lo regexp.lo"
+APACHE_MODULE(sed, filter request and/or response bodies through sed, $sed_obj)
+
 if test "$ac_cv_ebcdic" = "yes"; then
 # mod_charset_lite can be very useful on an ebcdic system,
 #   so include it by default
--- modules/aaa/config.m4.orig	Mon May  9 13:55:46 2011
+++ modules/aaa/config.m4	Mon May  9 14:07:47 2011
@@ -54,5 +54,9 @@
     enable_auth_digest="no"
   fi
 ])
+APACHE_MODULE(auth_gss, GSS based authentication, , , yes, [
+  MOD_AUTH_GSS_LDADD="-lgss"
+  AC_SUBST(MOD_AUTH_GSS_LDADD)
+])
 
 APACHE_MODPATH_FINISH
--- modules/slotmem/config.m4.~1~	2014-02-25 17:24:49.343535224 +0400
+++ modules/slotmem/config.m4	2014-02-25 17:24:59.535707087 +0400
@@ -4,7 +4,7 @@
 
 APACHE_MODPATH_INIT(slotmem)
 
-APACHE_MODULE(slotmem_shm, slotmem provider that uses shared memory, , , most)
+APACHE_MODULE(slotmem_shm, slotmem provider that uses shared memory, , , yes)
 APACHE_MODULE(slotmem_plain, slotmem provider that uses plain memory, , , )
 
 APACHE_MODPATH_FINISH
