--- libmemcached-1.0.17/m4/ax_memcached.m4  2013-02-20 23:58:10 +0000
+++ libmemcached-1.0.17/m4/ax_memcached.m4  2013-04-11 07:22:14 +0000
@@ -44,15 +44,26 @@
 #  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

-#serial 3
+#serial 4

 AC_DEFUN([AX_PROG_MEMCACHED],
-    [AX_WITH_PROG([MEMCACHED_BINARY],[memcached])
-    AS_IF([test -f "$ac_cv_path_MEMCACHED_BINARY"],
-      [AC_DEFINE([HAVE_MEMCACHED_BINARY], [1], [If Memcached binary is available])
-      AC_DEFINE_UNQUOTED([MEMCACHED_BINARY],"$ac_cv_path_MEMCACHED_BINARY",[Name of the memcached binary used in make test])
-      ax_memcached_version=`$MEMCACHED_BINARY -h | sed 1q | awk '{print \$ 2 }' | sed 's|[\-a-z0-9]*$||' | awk -F. '{printf "%d.%d.%d", $[]1, $[]2, $[]3}'`
-      AC_MSG_RESULT([memcached version "$ax_memcached_version"])
-      AC_DEFINE_UNQUOTED([MEMCACHED_VERSION],"$ax_memcached_version",[Memcached binary version])
-      ])
-    ])
+         [AX_WITH_PROG([MEMCACHED_BINARY],[memcached],[unknown])
+         ax_memcached_success=
+         AS_IF([test x"$MEMCACHED_BINARY" != xunknown],
+               [AS_IF([test -x "$MEMCACHED_BINARY"],
+                      [ax_memcached_version=`$MEMCACHED_BINARY -h | sed 1q | awk '{print \$ 2 }' | sed 's|[\-a-z0-9]*$||' | awk -F. '{printf "%d.%d.%d", $[]1, $[]2, $[]3}'`
+                      AS_IF([test -n "$ax_memcached_version"],
+                            [ax_memcached_success='ok'
+                            AC_MSG_RESULT([memcached version "$ax_memcached_version"])
+                            AC_DEFINE_UNQUOTED([MEMCACHED_VERSION],"$ax_memcached_version",[Memcached binary version])])
+                      ])
+               ])
+
+         AS_IF([test -n "$ax_memcached_success"],
+               [AC_DEFINE([HAVE_MEMCACHED_BINARY], [1], [If Memcached binary is available])
+               AC_DEFINE_UNQUOTED([MEMCACHED_BINARY],"$MEMCACHED_BINARY",[Name of the memcached binary used in make test])
+               ],
+               [AC_DEFINE([HAVE_MEMCACHED_BINARY], [0], [If Memcached binary is available])
+               MEMCACHED_BINARY=
+               ])
+         ])
