--- gcc-7.3.0/libstdc++-v3/libsupc++/new_opa.cc.orig	2018-05-29 08:41:13.791974891 +0000
+++ gcc-7.3.0/libstdc++-v3/libsupc++/new_opa.cc	2018-05-29 08:42:31.369329492 +0000
@@ -31,6 +31,11 @@
 using std::new_handler;
 using std::bad_alloc;
 
+// Workaround issue in illumos headers
+#if defined(__sun) && defined(__SVR4)
+using std::aligned_alloc;
+#endif
+
 #if !_GLIBCXX_HAVE_ALIGNED_ALLOC
 #if _GLIBCXX_HAVE__ALIGNED_MALLOC
 #define aligned_alloc(al,sz) _aligned_malloc(sz,al)
