Without this we will miss libobjc in proto area.

--- gcc-gcc-7.5.0-oi-0/libobjc/Makefile.in.orig
+++ gcc-gcc-7.5.0-oi-0/libobjc/Makefile.in
@@ -324,7 +324,8 @@
 	  $(INSTALL_DATA) $${realfile} $(DESTDIR)$(libsubdir)/$(includedirname)/objc; \
 	done
 
-check uninstall install-strip dist installcheck installdirs:
+check uninstall dist installcheck installdirs:
+install-strip: install
 
 mostlyclean:
 	-$(LIBTOOL_CLEAN) rm -f libobjc$(libsuffix).la libobjc_gc$(libsuffix).la *.lo
