--- setup.py.orig	2013-12-11 13:28:52.780272139 +0000
+++ setup.py	2013-12-11 13:29:17.585007544 +0000
@@ -45,7 +45,7 @@
                                  "Lib/site-packages/numpy/core/include")]
     else:
         arrayobject_h_include = [os.path.join(sys.prefix,
-                                "lib/python%s.%s/site-packages/numpy/core/include"
+                                "lib/python%s.%s/vendor-packages/numpy/core/include"
                                 % sys.version_info [:2])]
 
 math_libraries = []
