diff -wpruN '--exclude=*.orig' a~/tools/libclang/CMakeLists.txt a/tools/libclang/CMakeLists.txt
--- a~/tools/clang/tools/libclang/CMakeLists.txt	1970-01-01 00:00:00
+++ a/tools/clang/tools/libclang/CMakeLists.txt	1970-01-01 00:00:00
@@ -166,21 +166,21 @@ if(ENABLE_SHARED)
         )
     endif()
   endif()
-  if (USE_VERSION_SCRIPT)
-    target_link_options(libclang PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libclang.map")
-    # The Solaris 11.4 linker supports a subset of GNU ld version scripts,
-    # but requires a special option to enable it.
-    if (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
-      target_link_options(libclang PRIVATE "-Wl,-z,gnu-version-script-compat")
-    endif()
-    # Ensure that libclang.so gets rebuilt when the linker script changes.
-    set_property(SOURCE ARCMigrate.cpp APPEND PROPERTY
-                 OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/libclang.map)
-
-    set_target_properties(libclang PROPERTIES
-                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
-                          SOVERSION ${CLANG_SONAME})
-  endif()
+#  if (USE_VERSION_SCRIPT)
+#    target_link_options(libclang PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/libclang.map")
+#    # The Solaris 11.4 linker supports a subset of GNU ld version scripts,
+#    # but requires a special option to enable it.
+#    if (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
+#      target_link_options(libclang PRIVATE "-Wl,-z,gnu-version-script-compat")
+#    endif()
+#    # Ensure that libclang.so gets rebuilt when the linker script changes.
+#    set_property(SOURCE ARCMigrate.cpp APPEND PROPERTY
+#                 OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/libclang.map)
+#
+#    set_target_properties(libclang PROPERTIES
+#                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
+#                          SOVERSION ${CLANG_SONAME})
+#  endif()
 endif()
 
 if(INTERNAL_INSTALL_PREFIX)
