diff -Nru libevent-2.1.12-stable.orig/libevent_core.pc.in libevent-2.1.12-stable/libevent_core.pc.in
--- libevent-2.1.12-stable.orig/libevent_core.pc.in	2019-01-26 10:53:41.000000000 +0100
+++ libevent-2.1.12-stable/libevent_core.pc.in	2023-10-03 18:45:14.330811548 +0200
@@ -10,7 +10,7 @@
 Version: @VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -levent_core
+Libs: -L${libdir} -levent_core-2.1
 Libs.private: @LIBS@
 Cflags: -I${includedir}
 
diff -Nru libevent-2.1.12-stable.orig/libevent_extra.pc.in libevent-2.1.12-stable/libevent_extra.pc.in
--- libevent-2.1.12-stable.orig/libevent_extra.pc.in	2019-01-26 10:53:41.000000000 +0100
+++ libevent-2.1.12-stable/libevent_extra.pc.in	2023-10-03 18:45:22.438496508 +0200
@@ -10,7 +10,7 @@
 Version: @VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -levent_extra
+Libs: -L${libdir} -levent_extra-2.1
 Libs.private: @LIBS@
 Cflags: -I${includedir}
 
diff -Nru libevent-2.1.12-stable.orig/libevent_openssl.pc.in libevent-2.1.12-stable/libevent_openssl.pc.in
--- libevent-2.1.12-stable.orig/libevent_openssl.pc.in	2019-01-26 10:53:41.000000000 +0100
+++ libevent-2.1.12-stable/libevent_openssl.pc.in	2023-10-03 18:45:29.416058466 +0200
@@ -10,7 +10,7 @@
 Version: @VERSION@
 Requires: libevent
 Conflicts:
-Libs: -L${libdir} -levent_openssl
+Libs: -L${libdir} -levent_openssl-2.1
 Libs.private: @LIBS@ @OPENSSL_LIBS@
 Cflags: -I${includedir} @OPENSSL_INCS@
 
diff -Nru libevent-2.1.12-stable.orig/libevent_pthreads.pc.in libevent-2.1.12-stable/libevent_pthreads.pc.in
--- libevent-2.1.12-stable.orig/libevent_pthreads.pc.in	2019-01-26 10:53:41.000000000 +0100
+++ libevent-2.1.12-stable/libevent_pthreads.pc.in	2023-10-03 18:45:36.042336603 +0200
@@ -10,7 +10,7 @@
 Version: @VERSION@
 Requires: libevent
 Conflicts:
-Libs: -L${libdir} -levent_pthreads
+Libs: -L${libdir} -levent_pthreads-2.1
 Libs.private: @LIBS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PTHREAD_CFLAGS@
 
diff -Nru libevent-2.1.12-stable.orig/libevent.pc.in libevent-2.1.12-stable/libevent.pc.in
--- libevent-2.1.12-stable.orig/libevent.pc.in	2019-01-26 10:53:41.000000000 +0100
+++ libevent-2.1.12-stable/libevent.pc.in	2023-10-03 18:44:58.814981627 +0200
@@ -10,7 +10,7 @@
 Version: @VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} -levent
+Libs: -L${libdir} -levent-2.1
 Libs.private: @LIBS@
 Cflags: -I${includedir}
 
