diff -ruN transmission-2.92.orig/gtk/main.c transmission-2.92/gtk/main.c
--- transmission-2.92.orig/gtk/main.c	2016-02-28 07:52:32.021496953 +0300
+++ transmission-2.92/gtk/main.c	2016-05-26 22:34:02.337989639 +0300
@@ -627,6 +627,8 @@
     { NULL, 0,   0, 0, NULL, NULL, NULL }
   };

+  setenv( "EVENT_NOEVPORT", "1", 1 );
+
   /* default settings */
   memset (&cbdata, 0, sizeof (struct cbdata));
   cbdata.config_dir = (char*) tr_getDefaultConfigDir (MY_CONFIG_NAME);
