--- transmission-2.92.orig/gtk/transmission-gtk.desktop.in      2015-11-07 14:11:10.000000000 +0300
+++ transmission-2.92/gtk/transmission-gtk.desktop.in   2016-05-26 22:37:17.352407656 +0300
@@ -10,7 +10,7 @@
 Type=Application
 StartupNotify=true
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
-Categories=Network;FileTransfer;P2P;GTK;
+Categories=Applications;Network;FileTransfer;P2P;GTK;
 X-Ubuntu-Gettext-Domain=transmission
 X-AppInstall-Keywords=torrent
 Actions=Pause;Minimize;
