--- common/autoconf/generated-configure.sh.orig
+++ common/autoconf/generated-configure.sh
@@ -42679,6 +42679,8 @@
     LEGACY_TARGET_CFLAGS="$LEGACY_TARGET_CFLAGS -fstack-protector"
     LEGACY_HOST_CXXFLAGS="$LEGACY_HOST_CXXFLAGS -fstack-protector"
     LEGACY_TARGET_CXXFLAGS="$LEGACY_TARGET_CXXFLAGS -fstack-protector"
+    LDFLAGS_JDK="$LDFLAGS_JDK -lssp"
+    LEGACY_EXTRA_LDFLAGS="$LEGACY_EXTRA_LDFLAGS -lssp"
     if test "x$OPENJDK_TARGET_OS" != xmacosx -a test "x$OPENJDK_TARGET_OS" != xsolaris; then
       LDFLAGS_JDK="$LDFLAGS_JDK -Wl,-z,relro"
       LEGACY_HOST_LDFLAGS="$LEGACY_HOST_LDFLAGS -Wl,-z,relro"
--- hotspot/make/solaris/makefiles/vm.make.orig
+++ hotspot/make/solaris/makefiles/vm.make
@@ -151,6 +151,8 @@
 
 LIBS += -lrt
 
+LIBS += -lssp
+
 # By default, link the *.o into the library, not the executable.
 LINK_INTO$(LINK_INTO) = LIBJVM
 
