--- common/autoconf/generated-configure.sh	2019-03-14 16:21:04.775731966 +0000
+++ common/autoconf/generated-configure.sh.new	2019-03-14 16:23:56.718998837 +0000
@@ -40579,8 +40579,6 @@ $as_echo "$tool_specified" >&6; }
   # FIXME: we should really only export STRIPFLAGS from here, not POST_STRIP_CMD.
   if test "x$OPENJDK_TARGET_OS" = xlinux; then
     STRIPFLAGS="-g"
-  elif test "x$OPENJDK_TARGET_OS" = xsolaris; then
-    STRIPFLAGS="-x"
   elif test "x$OPENJDK_TARGET_OS" = xmacosx; then
     STRIPFLAGS="-S"
   elif test "x$OPENJDK_TARGET_OS" = xaix; then
