--- hotspot/src/os/solaris/vm/thread_solaris.inline.hpp.orig
+++ hotspot/src/os/solaris/vm/thread_solaris.inline.hpp
@@ -44,8 +44,5 @@
   _thr_current = thread;
 }
 
-inline Thread* ThreadLocalStorage::thread()  {
-  return _thr_current;
-}
 
 #endif // OS_SOLARIS_VM_THREAD_SOLARIS_INLINE_HPP
--- hotspot/src/os_cpu/solaris_x86/vm/threadLS_solaris_x86.hpp.orig
+++ hotspot/src/os_cpu/solaris_x86/vm/threadLS_solaris_x86.hpp
@@ -34,6 +34,6 @@
   static bool _initialized;  // needed for shared API
 
 public:
-  static inline Thread* thread();
+  static inline Thread* thread() { return _thr_current; }
 
 #endif // OS_CPU_SOLARIS_X86_VM_THREADLS_SOLARIS_X86_HPP
