--- hotspot/src/cpu/sparc/vm/nativeInst_sparc.hpp.~1~	2024-10-05 03:14:08.000000000 +0200
+++ hotspot/src/cpu/sparc/vm/nativeInst_sparc.hpp	2025-02-07 12:04:37.819451576 +0100
@@ -370,7 +370,7 @@
 
   // Creation
   friend inline NativeCall* nativeCall_at(address instr);
-  friend NativeCall* nativeCall_overwriting_at(address instr, address destination = NULL) {
+  friend NativeCall* nativeCall_overwriting_at(address instr, address destination) {
     // insert a "blank" call:
     NativeCall* call = (NativeCall*)instr;
     call->set_long_at(0 * BytesPerInstWord, call_instruction(destination, instr));
@@ -409,6 +409,7 @@
 
   static void replace_mt_safe(address instr_addr, address code_buffer);
 };
+
 inline NativeCall* nativeCall_at(address instr) {
   NativeCall* call = (NativeCall*)instr;
 #ifdef ASSERT
@@ -481,7 +482,7 @@
     return call;
   }
 
-  friend inline NativeFarCall* nativeFarCall_overwriting_at(address instr, address destination = NULL) {
+  friend inline NativeFarCall* nativeFarCall_overwriting_at(address instr, address destination) {
     Unimplemented();
     NativeFarCall* call = (NativeFarCall*)instr;
     return call;
