--- common/autoconf/generated-configure.sh.1	2016-07-18 01:59:04.492012322 +0300
+++ common/autoconf/generated-configure.sh	2016-07-18 01:59:32.265593286 +0300
@@ -29371,7 +29371,7 @@
     LDEXE="$CC"
     LDCXX="$CXX"
     LDEXECXX="$CXX"
-    POST_STRIP_CMD="$STRIP -g"
+    POST_STRIP_CMD="$STRIP"
 
     # Linking is different on MacOSX
     if test "x$OPENJDK_TARGET_OS" = xmacosx; then
--- common/autoconf/toolchain.m4.1	2016-07-18 02:00:28.460882338 +0300
+++ common/autoconf/toolchain.m4	2016-07-18 02:00:49.676692525 +0300
@@ -623,7 +623,7 @@
     LDEXE="$CC"
     LDCXX="$CXX"
     LDEXECXX="$CXX"
-    POST_STRIP_CMD="$STRIP -g"
+    POST_STRIP_CMD="$STRIP"
 
     # Linking is different on MacOSX
     if test "x$OPENJDK_TARGET_OS" = xmacosx; then
