Index: libgsf-1.14.27/gsf/gsf-output-iconv.c
===================================================================
--- libgsf-1.14.27.orig/gsf/gsf-output-iconv.c	2013-03-03 10:17:11.000000000 -0500
+++ libgsf-1.14.27/gsf/gsf-output-iconv.c	2014-01-21 10:33:51.055020978 -0500
@@ -119,7 +119,7 @@
 			ok = FALSE;
 		} else {
 			ic->buf_len -= bytes_read;
-			g_memmove (ic->buf, ic->buf + bytes_read, ic->buf_len);
+			memmove (ic->buf, ic->buf + bytes_read, ic->buf_len);
 
 			ok = gsf_output_write (ic->sink, bytes_written, data);
 			if (!ok) {
