--- borgbackup-1.0.0/borg/archiver.py.1	2016-03-19 15:44:20.181037884 +0300
+++ borgbackup-1.0.0/borg/archiver.py	2016-03-19 15:46:35.359058018 +0300
@@ -281,6 +281,9 @@
         elif stat.S_ISSOCK(st.st_mode):
             # Ignore unix sockets
             return
+        elif stat.S_ISDOOR(st.st_mode):
+            # Ignore doors
+            return
         else:
             self.print_warning('Unknown file type: %s', path)
             return
