REG_EIP is already defined elsewhere for Linux. 

--- comm-esr31/mozilla/js/src/jit/AsmJSSignalHandlers.cpp.orig	2015-06-04 17:42:51.702781399 -0700
+++ comm-esr31/mozilla/js/src/jit/AsmJSSignalHandlers.cpp	2015-06-04 17:42:51.707794331 -0700
@@ -58,6 +58,9 @@
 # define R14_sig(p) ((p)->sc_r14)
 # define R15_sig(p) ((p)->sc_r15)
 #elif defined(__linux__) || defined(SOLARIS)
+# if defined(SOLARIS)
+#  define REG_EIP EIP
+# endif
 # if defined(__linux__)
 #  define XMM_sig(p,i) ((p)->uc_mcontext.fpregs->_xmm[i])
 #  define EIP_sig(p) ((p)->uc_mcontext.gregs[REG_EIP])
