Already exists upstream.

--- comm-esr31/mozilla/gfx/skia/trunk/include/core/SkPreConfig.h.orig	2015-06-04 17:42:51.687614279 -0700
+++ comm-esr31/mozilla/gfx/skia/trunk/include/core/SkPreConfig.h	2015-06-04 17:42:51.689910035 -0700
@@ -93,7 +93,7 @@
 
 #if !defined(SK_CPU_BENDIAN) && !defined(SK_CPU_LENDIAN)
     #if defined (__ppc__) || defined(__PPC__) || defined(__ppc64__) \
-        || defined(__PPC64__)
+        || defined(__PPC64__) || defined(__sparc) || defined(__sparc__)
         #define SK_CPU_BENDIAN
     #else
         #define SK_CPU_LENDIAN
