--- file-roller-2.30.2/src/actions.c.orig	2012-04-09 14:08:57.245181745 +0800
+++ file-roller-2.30.2/src/actions.c	2012-04-09 14:09:21.523887348 +0800
@@ -240,7 +241,7 @@
 			return NULL;
 		}
 
-		if(!strcmp(data->window->priv->archive_uri, uri))
+		if(!strcmp(data->window->priv->archive_uri?data->window->priv->archive_uri:"", uri))
 		{
 			dialog = _gtk_error_dialog_new (GTK_WINDOW (data->dialog),
 							GTK_DIALOG_MODAL,
@@ -255,22 +256,44 @@
 		return NULL;
 		}
 		
-		g_file_delete (file, NULL, &err);
-		if (err != NULL) {
-			GtkWidget *dialog;
-			dialog = _gtk_error_dialog_new (GTK_WINDOW (data->dialog),
+                GtkWidget *d = _gtk_message_dialog_new (GTK_WINDOW (data->dialog),
+                                             GTK_DIALOG_MODAL,
+                                             GTK_STOCK_DIALOG_QUESTION,
+                                             _("The archive is already present.  Do you want to overwrite it?"),
+                                             NULL,
+                                             GTK_STOCK_NO, GTK_RESPONSE_NO,
+                                             _("_Overwrite"), GTK_RESPONSE_YES,
+                                             NULL);
+
+                 gtk_dialog_set_default_response (GTK_DIALOG (d), GTK_RESPONSE_YES);
+                 int r = gtk_dialog_run (GTK_DIALOG (d));
+                 gtk_widget_destroy (GTK_WIDGET (d));
+
+                 if (r == GTK_RESPONSE_YES) {
+                        g_object_unref(file);
+                        file = g_file_new_for_uri (uri);
+                        g_file_delete (file, NULL, &err);
+                        if (err != NULL) {
+				GtkWidget *dialog;
+				dialog = _gtk_error_dialog_new (GTK_WINDOW (data->dialog),
 							GTK_DIALOG_DESTROY_WITH_PARENT,
 							NULL,
 							_("Could not delete the old archive."),
 							"%s",
-							err->message);
-			gtk_dialog_run (GTK_DIALOG (dialog));
-			gtk_widget_destroy (GTK_WIDGET (dialog));
-			g_error_free (err);
-			g_free (uri);
-			g_object_unref (file);
-			return NULL;
-		}
+							err);
+				gtk_dialog_run (GTK_DIALOG (dialog));
+				gtk_widget_destroy (GTK_WIDGET (dialog));
+				g_error_free (err);
+				g_free (uri);
+				g_object_unref (file);
+				return NULL;
+                       }
+                }
+                else {
+                     g_free (uri);
+                     g_object_unref (file);
+                     return NULL;
+                }
 	}
 
 	g_object_unref (file);
