--- guile-2.2.6.orig/module/web/client.scm	2018-06-23 17:23:28.000000000 +0000
+++ guile-2.2.6/module/web/client.scm	2019-08-23 23:00:47.764220579 +0000
@@ -182,9 +182,9 @@
                     (cond (port => number->string)
                           ((uri-scheme uri) => symbol->string)
                           (else (error "Not an absolute URI" uri)))
-                    (if port
-                        AI_NUMERICSERV
-                        0))
+                    0
+                    AF_UNSPEC
+                    IPPROTO_TCP)
        (lambda (ai1 ai2)
          (equal? (addrinfo:addr ai1) (addrinfo:addr ai2))))))
   (define https?
