From 81dbc58566794e14a48e1ac5add5cb7ffe2da8e8 Mon Sep 17 00:00:00 2001
From: Aurelien Larcher <aurelien.larcher@gmail.com>
Date: Wed, 20 Mar 2019 00:08:33 +0100
Subject: [PATCH 7/7] libstdc++-no-rt

---
 libstdc++-v3/acinclude.m4 | 2 +-
 libstdc++-v3/configure    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libstdc++-v3/acinclude.m4 b/libstdc++-v3/acinclude.m4
index e48cf960d9f..3ce85773770 100644
--- a/libstdc++-v3/acinclude.m4
+++ b/libstdc++-v3/acinclude.m4
@@ -1466,7 +1466,7 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME], [
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+        GLIBCXX_LIBS="$GLIBCXX_LIBS"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index fe2a7e4b338..bcf7135140d 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -20527,7 +20527,7 @@ $as_echo "$glibcxx_glibc217" >&6; }
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+        GLIBCXX_LIBS="$GLIBCXX_LIBS"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
-- 
2.20.1

