--- diff_cover-7.3.0/tests/test_diff_reporter.py.orig
+++ diff_cover-7.3.0/tests/test_diff_reporter.py
@@ -96,6 +96,7 @@
         ),
     ],
 )
+@pytest.mark.skip(reason="always fails: https://github.com/Bachmann1234/diff_cover/issues/308")
 def test_git_path_selection(mocker, diff, git_diff, include, exclude, expected):
     old_cwd = os.getcwd()
     with tempfile.TemporaryDirectory() as tmp_dir:
