--- sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/spot.~1~	2014-04-08 10:05:38.928112089 +0400
+++ sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/spot	2014-04-08 10:05:46.647317643 +0400
@@ -1,4 +1,4 @@
-#!/usr/perl5/bin/perl
+#!/usr/bin/perl
 
 # @(#)spot.ver.sh 1.114 09/04/13 SMI
 
--- sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/er_html.~1~	2014-04-08 10:08:54.710977860 +0400
+++ sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/er_html	2014-04-08 10:09:05.660104302 +0400
@@ -1,4 +1,4 @@
-#!/usr/perl5/bin/perl
+#!/usr/bin/perl
 
 # @(#)er_html.ver.sh 1.58 08/06/19 SMI
 
--- sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/spot_diff.~1~	2014-04-08 10:09:27.137728880 +0400
+++ sunstudio12u1/i386/opt/sunstudio12.1/prod/bin/spot_diff	2014-04-08 10:09:33.631094330 +0400
@@ -1,4 +1,4 @@
-#!/usr/perl5/bin/perl
+#!/usr/bin/perl
 
 # @(#)spot_diff.ver.sh 1.37 08/08/27 SMI
 
