https://github.com/zopefoundation/persistent/issues/183

--- persistent-6.1/src/persistent/tests/test_docs.py.orig
+++ persistent-6.1/src/persistent/tests/test_docs.py
@@ -33,7 +33,7 @@
         prev, here = here, os.path.dirname(here)
         if here == prev:
             # Let's avoid infinite loops at root
-            raise AssertionError('could not find my setup.py')
+            return unittest.TestSuite()
 
     docs = os.path.join(here, 'docs', 'api')
 
