--- gftp-2.0.19/lib/parse-dir-listing.c.1	2015-02-06 19:32:20.672635116 +0300
+++ gftp-2.0.19/lib/parse-dir-listing.c	2015-02-06 19:32:52.131865264 +0300
@@ -532,7 +532,8 @@
 #endif
 
   /* Skip the blanks till we get to the next entry */
-  startpos = goto_next_token (startpos);
+  /* We are already at the next token */
+  /* startpos = goto_next_token (startpos); */
 
   /* Parse the filename. If this file is a symbolic link, remove the -> part */
   if (S_ISLNK (fle->st_mode) && ((endpos = strstr (startpos, "->")) != NULL))
