--- pygtk-2.24.0/pango.defs.orig	2011-04-01 12:37:25.000000000 +0000
+++ pygtk-2.24.0/pango.defs	2022-04-15 18:53:10.437575712 +0000
@@ -1303,16 +1303,6 @@
   )
 )
 
-(define-method find_shaper
-  (of-object "PangoFont")
-  (c-name "pango_font_find_shaper")
-  (return-type "PangoEngineShape*")
-  (parameters
-    '("PangoLanguage*" "language")
-    '("guint32" "ch")
-  )
-)
-
 (define-method get_metrics
   (of-object "PangoFont")
   (c-name "pango_font_get_metrics")
@@ -1391,15 +1381,6 @@
   )
 )
 
-(define-virtual find_shaper
-  (of-object "PangoFont")
-  (return-type "PangoEngineShape*")
-  (parameters
-    '("PangoLanguage*" "lang")
-    '("guint32" "ch")
-  )
-)
-
 (define-virtual get_glyph_extents
   (of-object "PangoFont")
   (return-type "none")
