--- libcroco-0.6.8/configure.in	Tue Oct 16 00:55:43 2012
+++ libcroco-0.6.8/configure.in	Thu Jan 16 18:50:09 2014
@@ -133,7 +133,7 @@
   -Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
   -Wdeclaration-after-statement -Wold-style-definition \
   -Wno-missing-field-initializers -Wno-unused-parameter \
-  -fno-common -Wno-switch-enum])
+  -Wno-switch-enum])
 
 ##########################################
 
