diff -urp -x '*~' -x '*.orig' config/config.cpp config/config.cpp
--- a/config/config.cpp	2010-08-06 18:40:29.000000000 -0700
+++ b/config/config.cpp	2011-01-28 18:01:17.331211005 -0800
@@ -1,6 +1,13 @@
 # X font server configuration file
 # See xfs(__appmansuffix__) man page for more information.
 
+XCOMM   DO NOT EDIT /usr/lib/X11/fs/config directly.
+XCOMM
+XCOMM   /usr/lib/X11/fs/config is a factory-default file and will
+XCOMM   be unconditionally overwritten upon subsequent installation.
+XCOMM   Before making changes to the file, copy it to the system configuration 
+XCOMM   location, /etc/X11/fs/config and edit it there.
+    
 clone-self = on
 use-syslog = off
 catalogue = DEFAULTFONTPATH
