--- pyproject_fmt-0.4.1/tests/test_pyproject_toml_fmt.py.orig
+++ pyproject_fmt-0.4.1/tests/test_pyproject_toml_fmt.py
@@ -4,10 +4,13 @@
 import sys
 from pathlib import Path
 
+import pytest
+
 
 def test_help_invocation_as_module() -> None:
     subprocess.check_call([sys.executable, "-m", "pyproject_fmt", "--help"])
 
 
+@pytest.mark.skip(reason="pyproject-fmt is not installed yet at final location")
 def test_help_invocation_as_script() -> None:
     subprocess.check_call([str(Path(sys.executable).parent / "pyproject-fmt"), "--help"])
