--- pyproject_fmt-1.0.0/tests/test_pyproject_toml_fmt.py.orig	2023-08-23 10:04:06.351933206 +0200
+++ pyproject_fmt-1.0.0/tests/test_pyproject_toml_fmt.py	2023-08-23 10:04:13.901787350 +0200
@@ -11,5 +11,5 @@
 
 def test_help_invocation_as_script() -> None:
     subprocess.check_call(
-        [str(Path(sys.executable).parent / "pyproject-fmt"), "--help"],
+        [str("pyproject-fmt"), "--help"],
     )
