We test in proto area not in site-packages.

--- toml-fmt-pyproject-fmt-2.7.0/pyproject-fmt/tox.toml.orig
+++ toml-fmt-pyproject-fmt-2.7.0/pyproject-fmt/tox.toml
@@ -20,7 +20,7 @@
             "{work_dir}{/}junit.{env_name}.xml",
             "--no-cov-on-fail",
             "--cov",
-            "{env_site_packages_dir}{/}pyproject_fmt",
+            "pyproject_fmt",
             "--cov",
             "{tox_root}{/}tests",
             "--cov-config",
