We test in proto area not in site-packages.

--- pyproject_fmt-2.2.4/tox.ini.orig
+++ pyproject_fmt-2.2.4/tox.ini
@@ -25,7 +25,7 @@
     COVERAGE_FILE = {work_dir}/.coverage.{env_name}
 commands =
     python -m pytest {tty:--color=yes} {posargs: \
-      --junitxml {work_dir}{/}junit.{env_name}.xml --cov {env_site_packages_dir}{/}pyproject_fmt \
+      --junitxml {work_dir}{/}junit.{env_name}.xml --cov pyproject_fmt \
       --cov {tox_root}{/}tests --cov-fail-under=100 \
       --cov-config=pyproject.toml --no-cov-on-fail --cov-report term-missing:skip-covered --cov-context=test \
       --cov-report html:{env_tmp_dir}{/}htmlcov --cov-report xml:{work_dir}{/}coverage.{env_name}.xml \
