--- mate-desktop-1.24.0/mate-about/mate-about.desktop.in	2020-02-10 11:33:51.000000000 +0000
+++ mate-desktop-1.24.0/mate-about/mate-about.desktop.in.new	2020-02-15 21:22:14.881829091 +0000
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=About MATE
-Comment=Learn more about MATE
+Comment=MATE version, licence and useful links
 Exec=mate-about
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
 Icon=mate-desktop
