--- mate-desktop-1.14.1/schemas/org.mate.background.gschema.xml.in.1	2016-07-27 23:05:33.255959034 +0300
+++ mate-desktop-1.14.1/schemas/org.mate.background.gschema.xml.in	2016-07-27 23:07:44.108797523 +0300
@@ -34,7 +34,7 @@
       <description>Determines how the image set by wallpaper_filename is rendered.  Possible values are "wallpaper", "centered", "scaled", "stretched", "zoom", "spanned".</description>
     </key>
     <key name="picture-filename" type="s">
-      <default>'@datadir@/backgrounds/mate/desktop/Stripes.png'</default>
+      <default>'/usr/share/pixmaps/backgrounds/openindiana/openindiana-default.jpg'</default>
       <summary>Picture Filename</summary>
       <description>File to use for the background image.</description>
     </key>
