--- mate-desktop-1.14.1/schemas/org.mate.interface.gschema.xml.in.1	2016-07-28 19:39:39.383711974 +0300
+++ mate-desktop-1.14.1/schemas/org.mate.interface.gschema.xml.in	2016-07-28 19:40:28.371002978 +0300
@@ -61,12 +61,12 @@
       <description>Length of the cursor blink cycle, in milliseconds.</description>
     </key>
     <key name="icon-theme" type="s">
-      <default>'menta'</default>
+      <default>'nimbus'</default>
       <summary>Icon Theme</summary>
       <description>Icon theme to use for the panel, Caja etc.</description>
     </key>
     <key name="gtk-theme" type="s">
-      <default>'Menta'</default>
+      <default>'nimbus'</default>
       <summary>Gtk+ Theme</summary>
       <description>Basename of the default theme used by gtk+.</description>
     </key>
