--- valgrind-3.13.0/memcheck/tests/x86-solaris/scalar.c.orig	2017-06-22 21:30:42.294920737 +0000
+++ valgrind-3.13.0/memcheck/tests/x86-solaris/scalar.c	2017-06-22 21:30:53.619783465 +0000
@@ -1,6 +1,6 @@
 /* Basic syscall test for Solaris/x86 specific syscalls. */
 
-#include "scalar.h"
+#include "../solaris/scalar.h"
 
 #include <string.h>
 #include <sys/fcntl.h>
--- valgrind-3.13.0/memcheck/tests/amd64-solaris/scalar.c.orig	2017-06-22 22:56:07.060430433 +0000
+++ valgrind-3.13.0/memcheck/tests/amd64-solaris/scalar.c	2017-06-22 22:56:15.692164082 +0000
@@ -1,6 +1,6 @@
 /* Basic syscall test for Solaris/amd64 specific syscalls. */
 
-#include "scalar.h"
+#include "../solaris/scalar.h"
 
 #include <sys/lwp.h>
 
