--- citus-9.5.1/src/backend/distributed/Makefile.orig	2020-12-02 09:20:43.000000000 +0000
+++ citus-9.5.1/src/backend/distributed/Makefile	2020-12-31 20:43:48.616685901 +0000
@@ -18,7 +18,7 @@
 DATA_built = $(generated_sql_files)
 
 # directories with source files
-SUBDIRS = . commands connection ddl deparser executor metadata operations planner progress relay safeclib test transaction utils worker
+SUBDIRS = commands connection ddl deparser executor metadata operations planner progress relay safeclib test transaction utils worker
 
 # Symlinks are not copied over to the build directory if a separete build
 # directory is used during configure (such as on CI)
@@ -29,6 +29,8 @@
 OBJS += \
 	$(patsubst $(citus_abs_srcdir)/%.c,%.o,$(foreach dir,$(SUBDIRS), $(sort $(wildcard $(citus_abs_srcdir)/$(dir)/*.c))))
 
+OBJS += shared_library_init.o
+
 # be explicit about the default target
 all:
 
