--- nspr-4.21/nspr/config/nspr.pc.in	2019-03-07 11:04:05.000000000 +0000
+++ nspr-4.21/nspr/config/nspr.pc.in.new	2019-03-16 09:47:09.536584020 +0000
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: NSPR
 Description: The Netscape Portable Runtime
 Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
-Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
+Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ -lpthread -ldl -lposix4
 Cflags: -I@includedir@
