--- harfbuzz-1.7.1/src/check-defs.sh.orig	2017-11-27 16:36:03.821184237 +0000
+++ harfbuzz-1.7.1/src/check-defs.sh	2017-11-27 16:42:05.503578049 +0000
@@ -21,7 +21,7 @@
 	lib=`echo "$def" | sed 's/[.]def$//;s@.*/@@'`
 	so=.libs/lib${lib}.so
 
-	EXPORTED_SYMBOLS="`nm "$so" | grep ' [BCDGINRSTVW] .' | grep -v ' _fini\>\| _init\>\| _fdata\>\| _ftext\>\| _fbss\>\| __bss_start\>\| __bss_start__\>\| __bss_end__\>\| _edata\>\| _end\>\| _bss_end__\>\| __end__\>\| __gcov_flush\>\| llvm_' | cut -d' ' -f3`"
+	EXPORTED_SYMBOLS="`nm "$so" | grep ' [BCDGINRSTVW] .' | grep -v ' _fini\>\| _init\>\| _fdata\>\| _ftext\>\| _fbss\>\| __bss_start\>\| __bss_start__\>\| __bss_end__\>\| _edata\>\| _end\>\| _bss_end__\>\| __end__\>\| __gcov_flush\>\| llvm_\| _DYNAMIC\| _GLOBAL_OFFSET_TABLE_\| _PROCEDURE_LINKAGE_TABLE_\| _etext\| _lib_version' | cut -d' ' -f3`"
 
 	if test -f "$so"; then
 
--- harfbuzz-1.7.1/src/check-symbols.sh.orig	2017-11-27 16:44:38.419934197 +0000
+++ harfbuzz-1.7.1/src/check-symbols.sh	2017-11-27 16:44:43.235984364 +0000
@@ -20,7 +20,7 @@
 	so=.libs/libharfbuzz.$suffix
 	if ! test -f "$so"; then continue; fi
 
-	EXPORTED_SYMBOLS="`nm "$so" | grep ' [BCDGINRSTVW] .' | grep -v ' _fini\>\| _init\>\| _fdata\>\| _ftext\>\| _fbss\>\| __bss_start\>\| __bss_start__\>\| __bss_end__\>\| _edata\>\| _end\>\| _bss_end__\>\| __end__\>\| __gcov_flush\>\| ___gcov_flush\>\| llvm_\| _llvm_' | cut -d' ' -f3`"
+	EXPORTED_SYMBOLS="`nm "$so" | grep ' [BCDGINRSTVW] .' | grep -v ' _fini\>\| _init\>\| _fdata\>\| _ftext\>\| _fbss\>\| __bss_start\>\| __bss_start__\>\| __bss_end__\>\| _edata\>\| _end\>\| _bss_end__\>\| __end__\>\| __gcov_flush\>\| ___gcov_flush\>\| llvm_\| _llvm_\| _DYNAMIC\| _GLOBAL_OFFSET_TABLE_\| _PROCEDURE_LINKAGE_TABLE_\| _etext\| _lib_version' | cut -d' ' -f3`"
 
 	prefix=`basename "$so" | sed 's/libharfbuzz/hb/; s/-/_/g; s/[.].*//'`
 
