--- harfbuzz-2.3.1/src/check-symbols.sh	2018-12-14 20:54:21.000000000 +0000
+++ harfbuzz-2.3.1/src/check-symbols.sh.new	2019-03-13 11:02:30.434637366 +0000
@@ -7,7 +7,7 @@ test -z "$srcdir" && srcdir=.
 test -z "$libs" && libs=.libs
 stat=0
 
-IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_flush\|llvm_.*'
+IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_flush\|llvm_.*\|_DYNAMIC\|_GLOBAL_OFFSET_TABLE_\|_PROCEDURE_LINKAGE_TABLE_\|_etext\|_lib_version'
 
 if which nm 2>/dev/null >/dev/null; then
 	:
@@ -26,7 +26,7 @@ for soname in harfbuzz harfbuzz-subset h
 		symprefix=
 		if test $suffix = dylib; then symprefix=_; fi
 
-		EXPORTED_SYMBOLS="`nm "$so" | grep ' [BCDGINRST] .' | grep -v " $symprefix\\($IGNORED_SYMBOLS\\>\\)" | cut -d' ' -f3 | c++filt`"
+		EXPORTED_SYMBOLS=$(nm "$so" | grep ' [BCDGINRST] .' | grep -v " $symprefix\\($IGNORED_SYMBOLS\\>\\)" | cut -d' ' -f3 | c++filt)
 
 		prefix=$symprefix`basename "$so" | sed 's/libharfbuzz/hb/; s/-/_/g; s/[.].*//'`
