--- harfbuzz-11.2.1/src/check-symbols.py.orig	2025-05-12 18:51:07.000000000 +0200
+++ harfbuzz-11.2.1/src/check-symbols.py	2025-05-25 10:05:00.901854791 +0200
@@ -26,13 +26,18 @@
     "_end",
     "_bss_end__",
     "__end__",
-    "__gcov_.*",
+    "__gcov_flush.*",
     "llvm_.*",
     "flush_fn_list",
     "writeout_fn_list",
     "mangle_path",
     "lprofDirMode",
     "reset_fn_list",
+    "_DYNAMIC",
+    "_GLOBAL_OFFSET_TABLE_",
+    "_PROCEDURE_LINKAGE_TABLE_",
+    "_etext",
+    "_lib_version",
 ]
 
 # Rust
