--- harfbuzz-2.6.6/src/check-symbols.sh.orig	2020-05-29 23:45:09.650814291 +0000
+++ harfbuzz-2.6.6/src/check-symbols.sh	2020-05-29 23:47:49.450438349 +0000
@@ -8,7 +8,7 @@
 test -z "$libs" && libs=.libs
 stat=0
 
-IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_.*\|llvm_.*\|flush_fn_list\|writeout_fn_list'
+IGNORED_SYMBOLS='_fini\|_init\|_fdata\|_ftext\|_fbss\|__bss_start\|__bss_start__\|__bss_end__\|_edata\|_end\|_bss_end__\|__end__\|__gcov_flush\|llvm_.*\|_DYNAMIC\|_GLOBAL_OFFSET_TABLE_\|_PROCEDURE_LINKAGE_TABLE_\|_etext\|_lib_version'
 
 if which nm 2>/dev/null >/dev/null; then
 	:
@@ -27,7 +27,7 @@
 		symprefix=
 		if test $suffix = dylib; then symprefix=_; fi
 
-		EXPORTED_SYMBOLS=`nm "$so" | grep ' [BCDGIRST] .' | grep -v " $symprefix\\($IGNORED_SYMBOLS\\>\\)" | cut -d' ' -f3 | c++filt`
+		EXPORTED_SYMBOLS=$(nm "$so" | grep ' [BCDGINRST] .' | grep -v " $symprefix\\($IGNORED_SYMBOLS\\>\\)" | cut -d' ' -f3 | c++filt)
 
 		prefix=$symprefix`basename "$so" | sed 's/libharfbuzz/hb/; s/-/_/g; s/[.].*//'`
 
