--- rdesktop-1.5.0/keymaps/common.orig	2008-02-18 17:53:38.985920000 +0900
+++ rdesktop-1.5.0/keymaps/common	2008-02-18 19:17:09.696254000 +0900
@@ -183,8 +183,11 @@ Right 0xcd localstate
 #
 Num_Lock 0x45
 KP_Divide 0xb5 localstate
+F25 0xb5 localstate
 KP_Multiply 0x37 localstate
+F26 0x37 localstate
 KP_Subtract 0x4a localstate
+F24 0x4a localstate
 KP_Add 0x4e localstate
 KP_Enter 0x9c localstate
 
@@ -197,30 +200,39 @@ KP_Insert 0x52 localstate
 
 KP_1 0x4f numlock
 KP_End 0x4f localstate
+F33 0x4f localstate
 
 KP_2 0x50 numlock
 KP_Down 0x50 localstate
+Down 0x50 localstate
 
 KP_3 0x51 numlock
 KP_Next 0x51 localstate
+F35 0x51 localstate
 
 KP_4 0x4b numlock
 KP_Left 0x4b localstate
+Left 0x4b localstate
 
 KP_5 0x4c numlock
 KP_Begin 0x4c localstate
+F31 0x4c localstate
 
 KP_6 0x4d numlock
 KP_Right 0x4d localstate
+Right 0x4d localstate
 
 KP_7 0x47 numlock
 KP_Home 0x47 localstate
+F27 0x47 localstate
 
 KP_8 0x48 numlock
 KP_Up 0x48 localstate
+Up 0x48 localstate
 
 KP_9 0x49 numlock
 KP_Prior 0x49 localstate
+F29 0x49 localstate
 
 #
 # Inhibited keys
