--- gnome-keyring-3.28.2/docs/Makefile.am.orig	2017-10-17 14:01:18.000000000 +0000
+++ gnome-keyring-3.28.2/docs/Makefile.am	2023-01-20 16:46:43.102125142 +0000
@@ -26,7 +26,7 @@
 	--stringparam man.copyright.section.enabled 0
 
 XSLTPROC_XSL = \
-	http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl
+	file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl
 
 .xml.1:
 	$(AM_V_GEN) $(XSLTPROC) $(XSLTPROC_MAN) $(XSLTPROC_FLAGS) \
