--- fsexam-0.8.1/src/fsexam-encoding-dialog.h.1	2015-03-12 22:46:39.162780667 +0300
+++ fsexam-0.8.1/src/fsexam-encoding-dialog.h	2015-03-12 22:52:04.159795016 +0300
@@ -84,11 +84,12 @@
     FSEXAM_ENCODING_ISO_2022_JP,
     FSEXAM_ENCODING_ISO_2022_KR,
     FSEXAM_ENCODING_JOHAB,
+    FSEXAM_ENCODING_KOI8_R,
     FSEXAM_ENCODING_SJIS,
     FSEXAM_ENCODING_UTF_16,
     FSEXAM_ENCODING_UTF_16BE,
-    FSEXAM_ENCODING_UTF_16LE,
-    FSEXAM_ENCODING_UTF_8	/* 60 */
+    FSEXAM_ENCODING_UTF_16LE,	/* 60 */
+    FSEXAM_ENCODING_UTF_8
 }FsexamEncodingIndex;
 
 typedef struct {
--- fsexam-0.8.1/src/fsexam-encoding-dialog.c.~1~	2010-07-28 20:21:02.000000000 +0400
+++ fsexam-0.8.1/src/fsexam-encoding-dialog.c	2015-03-12 23:06:43.582272987 +0300
@@ -105,6 +105,7 @@
     { FSEXAM_ENCODING_8859_8,   "ISO-8859-8",   "ISO88598",     N_("Hebrew"),               NULL },
     { FSEXAM_ENCODING_8859_9,   "ISO-8859-9",   "ISO88599",     N_("Turkish"),              "tr_TR.ISO8859-9"},
     { FSEXAM_ENCODING_JOHAB,    "JOHAB",        "JOHAB",        N_("Korean"),               NULL },
+    { FSEXAM_ENCODING_KOI8_R,   "KOI8-R",        "KOI8R",         N_("Russian (KOI8-R)"),              NULL },
     { FSEXAM_ENCODING_SJIS,     "SJIS",         "SJIS",         N_("Japanese"),             "ja_JP.PCK" },
 /*   { FSEXAM_ENCODING_UTF_16,   "UTF-16",       "UTF16",        N_("Unicode"),              NULL },
     { FSEXAM_ENCODING_UTF_16BE, "UTF-16BE",     "UTF16BE",      N_("Unicode"),              NULL },
