--- nimbus-0.1.2/gtk-engine/gtk-2.0/Makefile.am.orig	2009-04-10 14:38:37.566262368 +0900
+++ nimbus-0.1.2/gtk-engine/gtk-2.0/Makefile.am	2009-04-10 14:39:09.424816844 +0900
@@ -64,6 +64,7 @@ theme_DATA =	gtkrc		\
 		media-next.png  \
 		media-pause.png  \
 		media-play.png  \
+		media-play-rtl.png  \
 		media-prev.png  \
 		media-record.png  \
 		media-rewind.png  \
--- nimbus-0.1.2/gtk-engine/gtk-2.0/iconrc.in.orig	2009-02-05 10:34:27.000000000 +0900
+++ nimbus-0.1.2/gtk-engine/gtk-2.0/iconrc.in	2009-04-10 11:04:39.081974338 +0900
@@ -36,12 +36,16 @@ style "icon-style"
    stock["gtk-floppy"] =		 {{ "floppy.png",*,*,*}}
    stock["gtk-fullscreen"] =		 {{ "fullscreen.png",*,*,*}}
    stock["gtk-goto-bottom"] =		 {{ "go-bottom.png",*,*,*}}
-   stock["gtk-goto-first"] =		 {{ "go-first.png",*,*,*}}
-   stock["gtk-goto-last"] =		 {{ "go-last.png",*,*,*}}
+   stock["gtk-goto-first"] =		 {{ "go-first.png", LTR,*,*},
+                            		  { "go-last.png", RTL,*,*}}
+   stock["gtk-goto-last"] =		 {{ "go-last.png", LTR,*,*},
+                           		  { "go-first.png", RTL,*,*}}
    stock["gtk-goto-top"] =		 {{ "go-top.png",*,*,*}}
-   stock["gtk-go-back"] =		 {{ "go-previous.png",*,*,*}}
+   stock["gtk-go-back"] =		 {{ "go-previous.png", LTR,*,*},
+                         		  { "go-next.png", RTL,*,*}}
    stock["gtk-go-down"] =		 {{ "go-down.png",*,*,*}}
-   stock["gtk-go-forward"] =		 {{ "go-next.png",*,*,*}}
+   stock["gtk-go-forward"] =		 {{ "go-next.png", LTR,*,*},
+                            		  { "go-previous.png", RTL,*,*}}
    stock["gtk-go-up"] =			 {{ "go-up.png",*,*,*}}
    stock["gtk-harddisk"] =		 {{ "harddisk.png",*,*,*}}
    stock["gtk-help"] =			 {{ "help.png",*,*,*}}
@@ -58,13 +62,18 @@ style "icon-style"
    stock["gtk-justify-right"] = 	 {{ "format-justify-right.png",*,*,*}}
    stock["gtk-leave-fullscreen"] = 	 {{ "leave_fullscreen.png",*,*,*}}
    stock["gtk-missing-image"] = 	 {{ "broken_image.png",*,*,*}}
-   stock["gtk-media-forward"] = 	 {{ "media-ffwd.png",*,*,*}}
-   stock["gtk-media-next"] = 		 {{ "media-next.png",*,*,*}}
+   stock["gtk-media-forward"] = 	 {{ "media-ffwd.png", LTR,*,*},
+                                	  { "media-rewind.png", RTL,*,*}}
+   stock["gtk-media-next"] = 		 {{ "media-next.png", LTR,*,*},
+                             		  { "media-prev.png", RTL,*,*}}
    stock["gtk-media-pause"] = 		 {{ "media-pause.png",*,*,*}}
-   stock["gtk-media-play"] = 		 {{ "media-play.png",*,*,*}}
-   stock["gtk-media-previous"] = 	 {{ "media-prev.png",*,*,*}}
+   stock["gtk-media-play"] = 		 {{ "media-play.png", LTR,*,*},
+                             		  { "media-play-rtl.png", RTL,*,*}}
+   stock["gtk-media-previous"] = 	 {{ "media-prev.png", LTR,*,*},
+                                 	  { "media-next.png", RTL,*,*}}
    stock["gtk-media-record"] =		 {{ "media-record.png",*,*,*}}
-   stock["gtk-media-rewind"] =	         {{ "media-rewind.png",*,*,*}}
+   stock["gtk-media-rewind"] =	         {{ "media-rewind.png", LTR,*,*},
+                              	          { "media-ffwd.png", RTL,*,*}}
    stock["gtk-media-stop"] = 		 {{ "media-stop.png",*,*,*}}
    stock["gtk-network"] = 	 	 {{ "network.png",*,*,*}}
    stock["gtk-new"] =		         {{ "document-new.png",*,*,*}}
