--- ntfs-3g_ntfsprogs-2014.2.15/src/usermap.c.~1~	2014-07-22 10:25:07.281871539 +0400
+++ ntfs-3g_ntfsprogs-2014.2.15/src/usermap.c	2014-07-22 10:25:36.122130939 +0400
@@ -73,8 +73,8 @@
 #else
 #define USESTUBS 0 /* direct calls to API, based on following definitions */
 #define ENVNTFS3G "NTFS3G"
-#define LIBFILE64 "/lib64/libntfs-3g.so.491"
-#define LIBFILE "/lib/libntfs-3g.so.491"
+#define LIBFILE64 "/usr/lib/64/libntfs-3g.so.491"
+#define LIBFILE "/usr/lib/libntfs-3g.so.491"
 #endif
 
 #define GET_FILE_SECURITY "ntfs_get_file_security"
--- ntfs-3g_ntfsprogs-2014.2.15/src/secaudit.h.~1~	2014-07-22 10:24:53.878319278 +0400
+++ ntfs-3g_ntfsprogs-2014.2.15/src/secaudit.h	2014-07-22 10:25:53.178788452 +0400
@@ -56,8 +56,8 @@
 #else
 #define USESTUBS 0 /* direct calls to API, based on following definitions */
 #define ENVNTFS3G "NTFS3G"
-#define LIBFILE64 "/lib64/libntfs-3g.so.4921"
-#define LIBFILE "/lib/libntfs-3g.so.4921"
+#define LIBFILE64 "/usr/lib/64/libntfs-3g.so.4921"
+#define LIBFILE "/usr/lib/libntfs-3g.so.4921"
 #endif
 
 #define MAPDIR ".NTFS-3G"
