diff -urN gnome-session-2.23.6/gnome-session/gsm-client.c ../SUNWgnome-session-2.23.6.hacked/gnome-session-2.23.6/gnome-session/gsm-client.c
--- gnome-session-2.23.6/gnome-session/gsm-client.c	2008-08-05 00:15:36.000000000 +0100
+++ ../SUNWgnome-session-2.23.6.hacked/gnome-session-2.23.6/gnome-session/gsm-client.c	2008-08-15 10:55:10.184547000 +0100
@@ -527,5 +527,5 @@
                                  const char *reason)
 {
         g_signal_emit (client, signals[END_SESSION_RESPONSE], 0,
-                       is_ok, do_last, cancel, reason);
+                       is_ok, do_last, cancel, reason ? reason :"");
 }
