diff -urN gnome-session-2.26.1/gnome-session/gsm-xsmp-server.c ../SUNWgnome-session-2.26.1.hacked/gnome-session-2.26.1/gnome-session/gsm-xsmp-server.c
--- gnome-session-2.26.1/gnome-session/gsm-xsmp-server.c        2009-04-14 15:31:45.000000000 +0000
+++ ../SUNWgnome-session-2.26.1.hacked/gnome-session-2.26.1/gnome-session/gsm-xsmp-server.c     2009-06-08 10:33:02.333381950 +0000
@@ -471,7 +471,7 @@
          * ask X what transport types it knows about, so we're forced to
          * guess.
          */
-        _IceTransNoListen ("tcp");
+        // _IceTransNoListen ("tcp");
 #endif
 
         /* Create the XSMP socket. Older versions of IceListenForConnections

