diff -r b857c6cff902 glchess/src/glchess.in.in
--- a/glchess/src/glchess.in.in	Tue Dec 08 15:10:20 2009 +0800
+++ b/glchess/src/glchess.in.in	Tue Dec 08 15:14:41 2009 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 # glChess is a 2D/3D chess game for GNOME. This is the startup
diff -r b857c6cff902 gnome-sudoku/src/gnome-sudoku.in.in
--- a/gnome-sudoku/src/gnome-sudoku.in.in	Tue Dec 08 15:10:20 2009 +0800
+++ b/gnome-sudoku/src/gnome-sudoku.in.in	Tue Dec 08 15:14:41 2009 +0800
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 # GNOME Sudoku is a simple sudoku generator and player. Sudoku is a 
diff -r b857c6cff902 gnome-sudoku/src/lib/number_box.py
--- a/gnome-sudoku/src/lib/number_box.py	Tue Dec 08 15:10:20 2009 +0800
+++ b/gnome-sudoku/src/lib/number_box.py	Tue Dec 08 15:14:41 2009 +0800
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-#!/usr/bin/python
+#!/usr/bin/python2.7
 
 import gtk, gobject, pango, cairo
 import math
