--- gnome-games-2.25.91/configure.in.ori	2009-02-18 07:16:15.169699531 +0000
+++ gnome-games-2.25.91/configure.in	2009-02-18 07:19:44.279511018 +0000
@@ -850,6 +850,9 @@
 # Check for library functions
 # ===========================
 
+dnl *** check for libz ***
+AC_CHECK_LIB(z,gzread)
+
 AC_CHECK_FUNCS([dup2 strcasecmp strchr strerror \
                 malloc vprintf], [],
                [AC_MSG_ERROR([cannot find required function])])
