diff -ruN gnome-games-2.30.0.orig/configure.in gnome-games-2.30.0/configure.in
--- gnome-games-2.30.0.orig/configure.in	2010-04-22 11:55:22.490189423 +0100
+++ gnome-games-2.30.0/configure.in	2010-04-22 11:55:30.717872863 +0100
@@ -23,7 +23,7 @@
 # we support and which features to check for
 
 # This is the canonical list of all game subdirectories.
-allgames="aisleriot quadrapassel gnect gnomine swell-foop mahjongg gtali gnotravex gnotski glines iagno glchess gnobots2 gnibbles gnome-sudoku lightsoff"
+allgames="aisleriot gnect gnomine swell-foop mahjongg gtali gnotravex gnotski glines iagno glchess gnobots2 gnibbles gnome-sudoku lightsoff"
 AC_SUBST([allgames])
 staginggames=""
 AC_SUBST([staginggames])
diff -ruN gnome-games-2.30.0.orig/configure.in gnome-games-2.30.0/configure.in
--- gnome-games-2.30.0.orig/configure.in	2010-04-22 12:12:55.347670789 +0100
+++ gnome-games-2.30.0/configure.in	2010-04-22 12:13:20.147603818 +0100
@@ -205,9 +205,9 @@
 AC_PROG_LN_S
 AC_PROG_SED
 
-if test "$need_cxx" = "yes"; then
-  AC_PROG_CXX
+AC_PROG_CXX
 
+if test "$need_cxx" = "yes"; then
   # Check whether a C++ was found (AC_PROG_CXX sets $CXX to "g++" even when it
   # doesn't exist)
   AC_LANG_PUSH([C++])
