diff -Nru qbittorrent-4.5.0.orig/src/app/stacktrace.cpp qbittorrent-4.5.0/src/app/stacktrace.cpp
--- qbittorrent-4.5.0.orig/src/app/stacktrace.cpp	2023-01-14 20:42:09.116787672 +0000
+++ qbittorrent-4.5.0/src/app/stacktrace.cpp	2023-01-14 20:50:57.742537489 +0000
@@ -26,6 +26,8 @@
  * exception statement from your version.
  */
 
+#define _GNU_SOURCE
+
 #include "stacktrace.h"
 
 #include <boost/stacktrace.hpp>