--- ansible-core-2.15.2/test/lib/ansible_test/_internal/commands/units/__init__.py.orig
+++ ansible-core-2.15.2/test/lib/ansible_test/_internal/commands/units/__init__.py
@@ -258,6 +258,10 @@
             '-n', str(args.num_workers) if args.num_workers else 'auto',
             '--color', 'yes' if args.color else 'no',
             '-p', 'no:cacheprovider',
+            '-p', 'no:black',
+            '-p', 'no:cov',
+            '-p', 'no:mypy',
+            '-p', 'no:relaxed',
             '-c', os.path.join(ANSIBLE_TEST_DATA_ROOT, 'pytest', 'config', config_name),
             '--junit-xml', os.path.join(ResultType.JUNIT.path, 'python%s-%s-units.xml' % (python.version, test_context)),
             '--strict-markers',  # added in pytest 4.5.0
