--- eog-2.32.1/src/eog-util.c.orig	Sun May  7 10:29:51 2017
+++ eog-2.32.1/src/eog-util.c	Sun May  7 10:30:38 2017
@@ -28,7 +28,9 @@
 
 #include <sys/time.h>
 #ifdef HAVE_STRPTIME
+#ifndef __sun
 #define _XOPEN_SOURCE
+#endif
 #endif /* HAVE_STRPTIME */
 
 #include <time.h>
--- eog-2.32.1/src/eog-exif-util.c.orig	Sun May  7 10:29:21 2017
+++ eog-2.32.1/src/eog-exif-util.c	Sun May  7 10:31:00 2017
@@ -29,8 +29,10 @@
 #endif
 
 #ifdef HAVE_STRPTIME
+#ifndef __sun
 #define _XOPEN_SOURCE
 #endif
+#endif
 #include <time.h>
 
 #include "eog-exif-util.h"
