https://github.com/neilb/Text-Diff/pull/4

--- Text-Diff-1.45/LICENSE.orig
+++ Text-Diff-1.45/LICENSE
@@ -25,7 +25,7 @@
 software is free for all its users. This General Public License applies to most of
 the Free Software Foundation's software and to any other program whose
 authors commit to using it. (Some other Free Software Foundation software is
-covered by the GNU Library General Public License instead.) You can apply it to
+covered by the GNU Lesser General Public License instead.) You can apply it to
 your programs, too.
 
 When we speak of free software, we are referring to freedom, not price. Our
