diff -ur nimbus-0.1.9.orig/dark-index.theme.in nimbus-0.1.9/dark-index.theme.in
--- nimbus-0.1.9.orig/dark-index.theme.in	2010-10-02 16:55:56.000000000 +0200
+++ nimbus-0.1.9/dark-index.theme.in	2010-10-03 11:44:43.000000000 +0200
@@ -8,3 +8,4 @@
 GtkTheme=dark-nimbus
 IconTheme=nimbus
 MetacityTheme=dark-nimbus
+CursorTheme=DMZ-White
diff -ur nimbus-0.1.9.orig/index.theme.in nimbus-0.1.9/index.theme.in
--- nimbus-0.1.9.orig/index.theme.in	2010-10-02 16:55:56.000000000 +0200
+++ nimbus-0.1.9/index.theme.in	2010-10-03 11:45:04.000000000 +0200
@@ -8,3 +8,4 @@
 GtkTheme=nimbus
 IconTheme=nimbus
 MetacityTheme=nimbus
+CursorTheme=DMZ-White
diff -ur nimbus-0.1.9.orig/light-index.theme.in nimbus-0.1.9/light-index.theme.in
--- nimbus-0.1.9.orig/light-index.theme.in	2010-10-02 16:55:56.000000000 +0200
+++ nimbus-0.1.9/light-index.theme.in	2010-10-03 11:45:16.000000000 +0200
@@ -8,3 +8,4 @@
 GtkTheme=light-nimbus
 IconTheme=nimbus
 MetacityTheme=light-nimbus
+CursorTheme=DMZ-White
