--- libusb-1.0.22/libusb/os/sunos_usb.c.~2~	2018-05-21 11:13:21.078217665 +0000
+++ libusb-1.0.22/libusb/os/sunos_usb.c	2018-05-21 11:14:34.298914302 +0000
@@ -1635,6 +1635,13 @@
 	return (status);
 }
 
+#ifdef USBI_TIMERFD_AVAILABLE
+static clockid_t op_get_timerfd_clockid(void)
+{
+       return CLOCK_MONOTONIC;
+}
+#endif
+
 const struct usbi_os_backend usbi_backend = {
         .name = "Solaris",
         .caps = 0,
@@ -1667,6 +1674,9 @@
         .clear_transfer_priv = sunos_clear_transfer_priv,
         .handle_transfer_completion = sunos_handle_transfer_completion,
         .clock_gettime = sunos_clock_gettime,
+#ifdef USBI_TIMERFD_AVAILABLE
+        .get_timerfd_clockid = op_get_timerfd_clockid,
+#endif
         .device_priv_size = sizeof(sunos_dev_priv_t),
         .device_handle_priv_size = sizeof(sunos_dev_handle_priv_t),
         .transfer_priv_size = sizeof(sunos_xfer_priv_t),
