illumos fts.h header requires sys/types.h

--- clamav-0.100.1/clamd/fts.c.1	2018-07-26 14:15:45.924793648 +0000
+++ clamav-0.100.1/clamd/fts.c	2018-07-26 14:16:02.590310495 +0000
@@ -57,6 +57,7 @@

 #include <sys/param.h>
 #include <sys/stat.h>
+#include <sys/types.h>
 #include <fcntl.h>
 #include <dirent.h>
 #include <errno.h>
--- clamav-0.100.1/m4/reorganization/code_checks/fts.m4.1	2018-07-26 14:16:33.115191143 +0000
+++ clamav-0.100.1/m4/reorganization/code_checks/fts.m4	2018-07-26 14:16:44.773148067 +0000
@@ -1,5 +1,6 @@
 AC_MSG_CHECKING([LFS safe fts implementation])
 AC_COMPILE_IFELSE( [
+#include <sys/types.h>
 #include <fts.h>

 int main(void) {
