ClamAV dislikes zlib 1.2.11, considering it zlib 1.2.1

--- clamav-0.99.2/m4/reorganization/libs/libz.m4.1	2017-08-17 21:07:10.571808149 +0000
+++ clamav-0.99.2/m4/reorganization/libs/libz.m4	2017-08-17 21:07:25.843268748 +0000
@@ -29,9 +29,9 @@
     AC_MSG_ERROR([Please install zlib and zlib-devel packages])
 else
 
-    vuln=`grep "ZLIB_VERSION \"1.2.0" $ZLIB_HOME/include/zlib.h`
+    vuln=`grep "ZLIB_VERSION \"1.2.0\"" $ZLIB_HOME/include/zlib.h`
     if test -z "$vuln"; then
-	vuln=`grep "ZLIB_VERSION \"1.2.1" $ZLIB_HOME/include/zlib.h`
+	vuln=`grep "ZLIB_VERSION \"1.2.1\"" $ZLIB_HOME/include/zlib.h`
     fi
 
     if test -n "$vuln"; then
