We have no trio packaged yet.

--- jinja2-3.1.6/requirements/tests.in.orig
+++ jinja2-3.1.6/requirements/tests.in
@@ -1,2 +1 @@
 pytest
-trio
--- jinja2-3.1.6/tests/test_async.py.orig
+++ jinja2-3.1.6/tests/test_async.py
@@ -1,7 +1,6 @@
 import asyncio
 
 import pytest
-import trio
 
 from jinja2 import ChainableUndefined
 from jinja2 import DictLoader
@@ -18,7 +17,7 @@
     return asyncio.run(async_fn(*args))
 
 
-@pytest.fixture(params=[_asyncio_run, trio.run], ids=["asyncio", "trio"])
+@pytest.fixture(params=[_asyncio_run], ids=["asyncio"])
 def run_async_fn(request):
     return request.param
 
--- jinja2-3.1.6/tests/test_async_filters.py.orig
+++ jinja2-3.1.6/tests/test_async_filters.py
@@ -3,7 +3,6 @@
 from collections import namedtuple
 
 import pytest
-import trio
 from markupsafe import Markup
 
 from jinja2 import Environment
@@ -33,7 +32,7 @@
     return asyncio.run(async_fn(*args))
 
 
-@pytest.fixture(params=[_asyncio_run, trio.run], ids=["asyncio", "trio"])
+@pytest.fixture(params=[_asyncio_run], ids=["asyncio"])
 def run_async_fn(request):
     return request.param
 
