--- php-8.2.13/TSRM/TSRM.h	2023-11-21 10:55:59.000000000 +0100
+++ php-8.2.13/TSRM/TSRM.h.new	2023-12-20 21:43:41.798746063 +0100
@@ -148,7 +148,7 @@
 # define __has_attribute(x) 0
 #endif
 
-#if !__has_attribute(tls_model) || defined(__FreeBSD__) || defined(__MUSL__) || defined(__HAIKU__)
+#if !__has_attribute(tls_model) || defined(__FreeBSD__) || defined(__MUSL__) || defined(__HAIKU__) || defined(__solaris__)
 # define TSRM_TLS_MODEL_ATTR
 #elif __PIC__
 # define TSRM_TLS_MODEL_ATTR __attribute__((tls_model("initial-exec")))
