diff -wpruN --no-dereference '--exclude=*.orig' a~/config.h.in a/config.h.in
--- a~/config.h.in	1970-01-01 00:00:00
+++ a/config.h.in	1970-01-01 00:00:00
@@ -315,8 +315,8 @@
 /* Define if you have the <libutil.h> header file. (FreeBSD: openpty()) */
 #undef HAVE_LIBUTIL_H
 
-/* Define if you have the <sys/stropts.h> header file. (stream opts on SunOS)*/
-#undef HAVE_SYS_STROPTS_H
+/* Define if you have the <stropts.h> header file. (stream opts on SunOS)*/
+#undef HAVE_STROPTS_H
 
 /* Define if you have the <regex.h> header file. */
 #undef HAVE_REGEX_H
diff -wpruN --no-dereference '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -86,7 +86,7 @@ AC_CHECK_HEADERS(linux/types.h)
 AC_CHECK_HEADER(linux/errqueue.h, AC_DEFINE(HAVE_LINUX_ERRQUEUE_H), [], [#include <sys/time.h>
 #include <linux/types.h>])
 AC_CHECK_HEADERS(sys/utsname.h sys/select.h sys/file.h)
-AC_CHECK_HEADERS(util.h bsd/libutil.h libutil.h sys/stropts.h regex.h)
+AC_CHECK_HEADERS(util.h bsd/libutil.h libutil.h stropts.h regex.h)
 AC_CHECK_HEADERS(linux/fs.h linux/ext2_fs.h)
 
 dnl Checks for setgrent, getgrent and endgrent.
diff -wpruN --no-dereference '--exclude=*.orig' a~/sysincludes.h a/sysincludes.h
--- a~/sysincludes.h	1970-01-01 00:00:00
+++ a/sysincludes.h	1970-01-01 00:00:00
@@ -167,8 +167,8 @@
 #elif HAVE_LIBUTIL_H
 #include <libutil.h>		/* FreeBSD openpty() */
 #endif
-#if HAVE_SYS_STROPTS_H
-#include <sys/stropts.h>	/* SunOS I_PUSH ... */
+#if HAVE_STROPTS_H
+#include <stropts.h>		/* SunOS I_PUSH ... */
 #endif
 #if HAVE_REGEX_H
 #include <regex.h>
