--- src/Makefile.am	Mon Dec  5 14:30:20 2011
+++ src/Makefile.am	Mon Dec  5 14:55:55 2011
@@ -33,6 +33,9 @@
 intel_drv_la_LDFLAGS = -module -avoid-version
 intel_drv_ladir = @moduledir@/drivers
 intel_drv_la_LIBADD = legacy/liblegacy.la @PCIACCESS_LIBS@
+intel_drv_la_LDFLAGS += -R $(moduledir) -R $(moduledir)/extensions
+intel_drv_la_LIBADD += -lfb -lextmod -lshadowfb -lXfont -lm \
+	-lvbe -lvgahw -lxaa -lexa -ldri -ldri2 -lglx $(XORG_LIBS)
 
 if SNA
 SUBDIRS += sna
