--- a/src/sna/kgem.c.orig	2016-05-12 07:18:33.189782540 +0300
+++ b/src/sna/kgem.c	2016-05-12 07:27:00.673235483 +0300
@@ -808,11 +808,7 @@ static void gem_close(int fd, uint32_t h
 constant inline static unsigned long __fls(unsigned long word)
 {
 #if defined(__GNUC__) && (defined(__i386__) || defined(__x86__) || defined(__x86_64__))
-	asm("bsr %1,%0"
-	    : "=r" (word)
-	    : "rm" (word));
-	return word;
-#else
+	
 	unsigned int v = 0;
 
 	while (word >>= 1)
