diff -ruN xf86-video-intel-4798e18b2b2c8b0a05dc967e6140fd9962bc1a73.orig/src/legacy/i810/i810.h xf86-video-intel-4798e18b2b2c8b0a05dc967e6140fd9962bc1a73/src/legacy/i810/i810.h
--- xf86-video-intel-4798e18b2b2c8b0a05dc967e6140fd9962bc1a73.orig/src/legacy/i810/i810.h	2020-05-14 13:03:03.617242838 +0000
+++ xf86-video-intel-4798e18b2b2c8b0a05dc967e6140fd9962bc1a73/src/legacy/i810/i810.h	2020-05-14 13:03:56.303502181 +0000
@@ -322,6 +322,6 @@
 extern const OptionInfoRec *I810AvailableOptions(int chipid, int busid);
 
 extern const int I810CopyROP[16];
-const int I810PatternROP[16];
+extern const int I810PatternROP[16];
 
 #endif /* _I810_H_ */
