--- src/i830_memory.c	Fri Dec  4 14:15:17 2009
+++ src/i830_memory.c	Fri Dec  4 14:15:44 2009
@@ -966,6 +966,9 @@
 	}
     }
 
+    if (IS_IGDNG(pI830))
+        mem->tiling = TILE_NONE;
+
     return mem;
 }
 
--- src/cpu_display.c	Fri Dec  4 14:25:07 2009
+++ src/cpu_display.c	Fri Dec  4 14:26:05 2009
@@ -431,9 +431,11 @@
     /* enable PCH clock reference source */
     temp = INREG(PCH_DREF_CONTROL);
     temp &= ~DREF_CONTROL_MASK;
-    /* XXX match bios dump, need to change for more outputs */
-    temp |= DREF_NONSPREAD_SOURCE_ENABLE;
-    OUTREG(PCH_DREF_CONTROL, temp);
+    if (!IS_IGDNG(pI830)) {
+        /* XXX match bios dump, need to change for more outputs */
+        temp |= DREF_NONSPREAD_SOURCE_ENABLE;
+        OUTREG(PCH_DREF_CONTROL, temp);
+    }
 
      /* enable PCH SSC modulator, looks not needed for CRT */
     /* enable PCH DPLL */
