--- src/Makefile.am.~1~	2015-02-09 13:13:13.145715337 +0300
+++ src/Makefile.am	2015-02-09 13:14:57.808764132 +0300
@@ -31,7 +31,9 @@
 
 intel_drv_la_LTLIBRARIES = intel_drv.la
 intel_drv_la_LDFLAGS = -module -avoid-version
-intel_drv_ladir = @moduledir@/drivers
+tmp = @moduledir@/drivers
+drvdir = $(tmp:amd64/xorg/modules/drivers=xorg/modules/drivers/amd64)
+intel_drv_ladir = $(drvdir)
 intel_drv_la_LIBADD = -lm @DRM_LIBS@ -ldrm_intel ../uxa/libuxa.la
 intel_drv_la_LIBADD += @PCIACCESS_LIBS@
 
