--- idzebra-2.0.56/doc/common/id.man.xsl.~1~	2015-07-25 16:09:42.152526630 +0300
+++ idzebra-2.0.56/doc/common/id.man.xsl	2015-07-25 16:11:12.471718582 +0300
@@ -1,7 +1,7 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
 	        version="1.0">
 
-<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"/>
+<xsl:import href="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"/>
 </xsl:stylesheet>
 
 
--- idzebra-2.0.56/doc/common/id.htmlhelp.xsl.~1~	2015-07-25 16:18:29.839177712 +0300
+++ idzebra-2.0.56/doc/common/id.htmlhelp.xsl	2015-07-25 16:19:37.859169311 +0300
@@ -1,7 +1,7 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
 	        version="1.0">
 
-<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/htmlhelp/htmlhelp.xsl" />
+<xsl:import href="file:///usr/share/sgml/docbook/xsl-stylesheets/htmlhelp/htmlhelp.xsl" />
 
 <xsl:template name="body.attributes">
    <link rel="stylesheet" type="text/css" href="common/style1.css"/>
--- idzebra-2.0.56/doc/common/id.tkl.xsl.~1~	2015-07-25 16:20:08.417438510 +0300
+++ idzebra-2.0.56/doc/common/id.tkl.xsl	2015-07-25 16:21:01.659230014 +0300
@@ -1,7 +1,7 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
 		version="1.0">
 
-  <xsl:include href="http://docbook.sourceforge.net/release/xsl/current/xhtml/chunk.xsl"/>
+  <xsl:include href="http:///usr/share/sgml/docbook/xsl-stylesheets/xhtml/chunk.xsl"/>
 
   <xsl:variable name="use.id.as.filename">1</xsl:variable>
   <xsl:variable name="html.ext">.tkl</xsl:variable>
