--- pycmd-1.2/pycmd/pylookup.py.orig
+++ pycmd-1.2/pycmd/pylookup.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 """\
 py.lookup [search_directory] SEARCH_STRING [options]
--- pycmd-1.2/pycmd/pycleanup.py.orig
+++ pycmd-1.2/pycmd/pycleanup.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 """\
 py.cleanup [PATH] ...
--- pycmd-1.2/pycmd/pycountloc.py.orig
+++ pycmd-1.2/pycmd/pycountloc.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 # hands on script to compute the non-empty Lines of Code
 # for tests and non-test code
--- pycmd-1.2/pycmd/pysvnwcrevert.py.orig
+++ pycmd-1.2/pycmd/pysvnwcrevert.py
@@ -1,4 +1,3 @@
-#! /usr/bin/env python
 """\
 py.svnwcrevert [options] WCPATH
 
