--- pycmd-1.2/pycmd/pyconvert_unittest.py.orig
+++ pycmd-1.2/pycmd/pyconvert_unittest.py
@@ -114,12 +114,12 @@
     else:
         message = None
 
-    if argnum is 0 or (argnum is 1 and argnum is message_pos): #unittest fail()
+    if argnum == 0 or (argnum == 1 and argnum == message_pos): #unittest fail()
         string = ''
         if message:
             message = ' ' + message
 
-    elif message_pos is 4:  # assertAlmostEqual & friends
+    elif message_pos == 4:  # assertAlmostEqual & friends
         try:
             pos = argl[2].lstrip()
         except IndexError:
