--- squeak-4-3797/platforms/unix/plugins/SqueakSSL/config.cmake	Tue Nov 10 23:25:35 2020
+++ p0/squeak-4-3797/platforms/unix/plugins/SqueakSSL/config.cmake	Sat Feb 20 20:22:44 2021
@@ -10,13 +10,13 @@
 
 # Traditional dynamic linking.
 #
-# PLUGIN_DEFINITIONS("-DSQSSL_OPENSSL_LINKED")
+PLUGIN_DEFINITIONS("-DSQSSL_OPENSSL_LINKED")
 
 # Runtime symbol and function lookup.
 # Portable across platforms because only required symbols are looked up.
 # Requires a Gnu extension for implementation of symbol lookup.
 #
-PLUGIN_DEFINITIONS("-D_GNU_SOURCE")
+# PLUGIN_DEFINITIONS("-D_GNU_SOURCE")
 
 ################################################################################
 #            Explanation from commit notice on opensmalltalk-vm:               #
