--- squeak4-3775/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c	Sun Apr 26 21:11:54 2020
+++ p0/squeak4-3775/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c	Sun Mar 29 17:26:21 2020
@@ -72,7 +72,7 @@
 #endif
 
 static sqInt sound_Stop(void);
-static int sound_AvailableSpace(void);
+static sqInt sound_AvailableSpace(void);
 
 static int auFd=	       -1;   /* open on /dev/audio */
 static int auCtlFd=	       -1;   /* open on /dev/audioctl */
