--- Parse-Yapp-1.21/Makefile.PL.orig
+++ Parse-Yapp-1.21/Makefile.PL
@@ -5,7 +5,7 @@
     'NAME'	=> 'Parse::Yapp',
     # CORRELATION #py001: $VERSION must be changed in both Parse::Yapp & Parse::Yapp::Driver
     'VERSION_FROM' => 'lib/Parse/Yapp/Driver.pm',
-    'MAN3PODS' => { 'lib/Parse/Yapp.pm' => '$(INST_MAN3DIR)/Parse::Yapp.3' },
+    'MAN3PODS' => { 'lib/Parse/Yapp.pm' => '$(INST_MAN3DIR)/Parse::Yapp.$(MAN3EXT)' },
     'PM' => {
               'lib/Parse/Yapp.pm' => '$(INST_LIBDIR)/Yapp.pm',
               'lib/Parse/Yapp/Options.pm' => '$(INST_LIBDIR)/Yapp/Options.pm',
