--- librsvg-2.55.3/Makefile.in	2023-07-20 21:32:11.000000000 +0200
+++ librsvg-2.55.3/Makefile.in.orig	2023-09-07 23:00:44.892173165 +0200
@@ -1536,7 +1536,7 @@
 	CARGO_TARGET_DIR=$(CARGO_TARGET_DIR)		\
 	LIBRSVG_BUILD_DIR=$(LIBRSVG_BUILD_DIR)		\
 	OUT_DIR=$(LIBRSVG_BUILD_DIR)/tests/output	\
-	$(CARGO) --locked test $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) -- --include-ignored -Z unstable-options
+	$(CARGO) --locked test $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) -- --include-ignored unstable-options
 
 clean-local:
 	cd $(top_srcdir) &&			\
--- librsvg-2.55.3/Makefile.am	2023-07-20 03:48:16.000000000 +0200
+++ librsvg-2.55.3/Makefile.am.orig	2023-09-07 23:02:48.417612652 +0200
@@ -160,7 +160,7 @@
 	CARGO_TARGET_DIR=$(CARGO_TARGET_DIR)		\
 	LIBRSVG_BUILD_DIR=$(LIBRSVG_BUILD_DIR)		\
 	OUT_DIR=$(LIBRSVG_BUILD_DIR)/tests/output	\
-	$(CARGO) --locked test $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) -- --include-ignored -Z unstable-options
+	$(CARGO) --locked test $(CARGO_VERBOSE) $(CARGO_TARGET_ARGS) $(CARGO_RELEASE_ARGS) -- --include-ignored unstable-options
 
 clean-local:
 	cd $(top_srcdir) &&			\
