--- gnome-session-2.30.2/configure.ac.2	2015-01-12 22:23:54.279933575 +0300
+++ gnome-session-2.30.2/configure.ac	2015-01-12 22:24:03.742768847 +0300
@@ -118,11 +118,6 @@
    PATH=$GCONF_SERVERDIR:$PATH
 fi
 
-AC_PATH_PROG(GCONF_SANITY_CHECK, gconf-sanity-check-2, no)
-if test x"$GCONF_SANITY_CHECK" = xno; then
-  AC_MSG_ERROR([gconf-sanity-check-2 executable not found in your path - should be installed with GConf])
-fi
-
 AC_SUBST(GCONF_SANITY_CHECK)
 PATH=$old_path
 
