--- joe-4.4/configure.ac.1	2017-04-25 17:48:20.379206860 +0300
+++ joe-4.4/configure.ac	2017-04-25 17:48:35.288321358 +0300
@@ -23,13 +23,13 @@
 AC_PROG_INSTALL
 AC_PROG_LN_S
 
-case "$host" in
-	*-*-solaris*)
-		CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-		LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
-		;;
-	*)	;;
-esac
+#case "$host" in
+#	*-*-solaris*)
+#		CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+#		LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
+#		;;
+#	*)	;;
+#esac
 
 search_libs="$search_libs m"
 
