--- variables.c	2009-01-04 14:32:46.000000000 -0500
+++ variables.c	2009-08-15 14:53:56.426776000 -0400
@@ -402,14 +402,10 @@
   /* Now make our own defaults in case the vars that we think are
      important are missing. */
   temp_var = set_if_not ("PATH", DEFAULT_PATH_VALUE);
-#if 0
   set_auto_export (temp_var);	/* XXX */
-#endif
 
   temp_var = set_if_not ("TERM", "dumb");
-#if 0
   set_auto_export (temp_var);	/* XXX */
-#endif
 
 #if defined (__QNX__)
   /* set node id -- don't import it from the environment */
