--- doc/bashref.info	2009-12-29 12:59:33.000000000 -0800
+++ doc/bashref.info	2010-09-01 10:49:49.070499702 -0700
@@ -6387,7 +6387,7 @@
 directory.  The name of this file is taken from the value of the shell
 variable `INPUTRC'.  If that variable is unset, the default is
 `~/.inputrc'.  If that file does not exist or cannot be read, the
-ultimate default is `/etc/inputrc'.
+ultimate default is `/etc/bash/inputrc'.
 
    When a program which uses the Readline library starts up, the init
 file is read, and the key bindings are set.
@@ -6812,8 +6812,8 @@
 `$include'
      This directive takes a single filename as an argument and reads
      commands and bindings from that file.  For example, the following
-     directive reads from `/etc/inputrc':
-          $include /etc/inputrc
+     directive reads from `/etc/bash/inputrc':
+          $include /etc/bash/inputrc
 
 
 File: bashref.info,  Node: Sample Init File,  Prev: Conditional Init Constructs,  Up: Readline Init File
