--- support/signames.c	2008-08-13 08:31:48.000000000 -0400
+++ support/signames.c	2009-08-15 16:40:01.729682000 -0400
@@ -212,6 +212,14 @@
   signal_names[SIGXRES] = "SIGXRES";
 #endif
 
+#if defined (SIGJVM1) /* Solaris Java Virtual Machine 1 */
+  signal_names[SIGJVM1] = "SIGJVM1";
+#endif
+
+#if defined (SIGJVM2) /* Solaris Java Virtual Machine 2 */
+  signal_names[SIGJVM2] = "SIGJVM2";
+#endif
+
 /* HP-UX */
 #if defined (SIGDIL)	/* DIL signal (?) */
   signal_names[SIGDIL] = "SIGDIL";
