--- unison-2.27.57/Makefile.OCaml.orig	Tue May 24 12:44:25 2011
+++ unison-2.27.57/Makefile.OCaml	Tue May 24 12:44:42 2011
@@ -42,9 +42,9 @@
 # to Unix '/' separators, and extraneous control-M's are deleted.
 # Unfortunately there is a literal control-M buried in this, I'd rather
 # get rid of it...
-# OCAMLLIBDIR=$(shell ocamlc -v | tail -n -1 | sed -e 's/.* //g' | sed -e 's/\\/\//g' | sed -e 's///g')
+# OCAMLLIBDIR=$(shell ocamlc -v | tail -1 | sed -e 's/.* //g' | sed -e 's/\\/\//g' | sed -e 's///g')
 # Better(?) version, June 2005:
-OCAMLLIBDIR=$(shell ocamlc -v | tail -n -1 | sed -e 's/.* //g' | sed -e 's/\\/\//g' | tr -d '\r')
+OCAMLLIBDIR=$(shell ocamlc -v | tail -1 | sed -e 's/.* //g' | sed -e 's/\\/\//g' | tr -d '\r')
 
 ## BCP (6/05) an alternative, but not quite working, version
 ## suggested by Nick Montfort:
@@ -274,7 +274,7 @@
            >>dot.tmp
 	echo '{ rank = same; "Tree"; "Safelist"; }'>>dot.tmp
 	echo '{ rank = same; "Uarg"; "Prefs"; }'>>dot.tmp
-	ocamldot .depend | tail -n +2 >> dot.tmp
+	ocamldot .depend | tail +2 >> dot.tmp
 	-dot -Tps -o DEPENDENCIES.ps dot.tmp
 endif
 
