--- guilt-0.36/Documentation/Makefile.1	2018-01-25 08:02:16.499853679 +0000
+++ guilt-0.36/Documentation/Makefile	2018-01-25 08:03:04.129788954 +0000
@@ -7,7 +7,7 @@
 DOC_MAN7=$(patsubst %.txt,%.7,$(MAN7_TXT))
 
 USAGE=$(patsubst %.txt,usage-%.txt,$(MAN1_TXT))
-VERSION=$(shell git describe 2> /dev/null || sed -n -e '/^GUILT_VERSION=/ { s/^GUILT_VERSION="/v/; s/"//; p; q; }' ../guilt)
+VERSION=$(sed -n -e '/^GUILT_VERSION=/ { s/^GUILT_VERSION="/v/; s/"//; p; q; }' ../guilt)
 
 prefix?=$(PREFIX)
 bindir?=$(prefix)/bin
