diff --git a/guilt b/guilt
index e143b3b..686dd26 100755
--- a/guilt
+++ b/guilt
@@ -120,8 +120,22 @@ else
 	disp "Guilt v$GUILT_VERSION"
 	disp ""
 	disp "Pick a command:"
-	guilt_commands | sort | column | column -t | sed -e 's/^/	/'
 
+	guilt_commands | sort | awk '
+BEGIN {
+	x=0
+}
+(x==0) {
+	print ""
+}
+{ 
+	printf "%-13s  ", $0
+}
+{
+	x=(x+1)%5
+}' | sed -e 's/^/	/'
+
+	disp ""
 	disp ""
 	disp "Example:"
 	disp "\tguilt push"
