--- guilt-0.36/regression/scaffold.1	2018-01-25 08:59:25.295258557 +0000
+++ guilt-0.36/regression/scaffold	2018-01-25 08:59:40.136969158 +0000
@@ -44,7 +44,7 @@
 
 function filter_dd
 {
-	sed -e 's:^\([0-9]*\) bytes (\([0-9]*\.[0-9]* .B\)) copied.*$:\1 bytes (\2) copied:'
+	sed -e 's:^\([0-9]*\) bytes (\([0-9]*\.[0-9]* .B\), \([0-9]*\.[0-9]* .*B\)) copied.*$:\1 bytes (\2) copied:'
 }
 
 # usage: cmd <cmd>..
