We need direct dependencies only.

--- coverage-7.6.1/tox.ini.orig
+++ coverage-7.6.1/tox.ini
@@ -16,9 +16,9 @@
 
 # PYVERSIONS
 deps =
-    -r requirements/pip.pip
-    -r requirements/pytest.pip
-    py3{8,9,10,11}: -r requirements/light-threads.pip
+    -r requirements/pip.in
+    -r requirements/pytest.in
+    py3{8,9,10,11}: -r requirements/light-threads.in
 
 # Windows can't update the pip version with pip running, so use Python
 # to install things.
