--- coverage-7.2.1/tox.ini.orig
+++ coverage-7.2.1/tox.ini
@@ -36,9 +36,7 @@
     # Create tests/zipmods.zip
     python igor.py zip_mods
 
-    # Build the C extension and test with the CTracer
-    python setup.py --quiet build_ext --inplace
-    python -m pip install {env:COVERAGE_PIP_ARGS} -q -e .
+    # Test with the CTracer
     python igor.py test_with_tracer c {posargs}
 
     # Remove the C extension so that we can test the PyTracer
