--- ./src/plugins/switch/cray/switch_cray.c.orig	2016-05-04 00:41:59.000000000 +0200
+++ ./src/plugins/switch/cray/switch_cray.c	2016-05-05 21:03:02.479050209 +0200
@@ -35,7 +35,7 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA.
 \*****************************************************************************/
 
-#if !(defined(__FreeBSD__) || defined(__NetBSD__))
+#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__sun))
 #if     HAVE_CONFIG_H
 #include "config.h"
 #endif
--- ./src/plugins/task/cgroup/task_cgroup_cpuset.c.orig	2016-05-05 21:05:50.907239807 +0200
+++ ./src/plugins/task/cgroup/task_cgroup_cpuset.c	2016-05-05 21:06:40.709081181 +0200
@@ -36,7 +36,7 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA.
 \*****************************************************************************/
 
-#if !(defined(__FreeBSD__) || defined(__NetBSD__))
+#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__sun))
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
