From 5c41fbfce407edd89cbddb6ce68ea0f7bc211f68 Mon Sep 17 00:00:00 2001
From: Aurelien Larcher <aurelien.larcher@gmail.com>
Date: Sat, 6 Feb 2021 00:19:20 +0100
Subject: [PATCH] Add OpenSSL 1.1 package fmri to pkglintrc

---
 exception_lists/packaging.deps | 1 +
 1 file changed, 1 insertion(+)

diff --git a/exception_lists/packaging.deps b/exception_lists/packaging.deps
index 5673f33843..304522b8aa 100644
--- a/exception_lists/packaging.deps
+++ b/exception_lists/packaging.deps
@@ -22,6 +22,7 @@ pkg:/library/nspr
 pkg:/library/nspr/header-nspr
 pkg:/library/perl-5/xml-parser
 pkg:/library/security/openssl
+pkg:/library/security/openssl-11
 pkg:/library/security/trousers
 pkg:/library/zlib
 pkg:/package/pkg
-- 
2.30.0

