--- postgresql-9.5.0/src/pl/plpython/Makefile.~1~	2016-01-05 00:29:34.000000000 +0300
+++ postgresql-9.5.0/src/pl/plpython/Makefile	2016-01-22 14:39:20.967007169 +0300
@@ -13,7 +13,7 @@
 
 override CPPFLAGS := -I. -I$(srcdir) $(python_includespec) $(CPPFLAGS)
 
-rpathdir = $(python_libdir)
+#rpathdir = $(python_libdir)
 
 PGFILEDESC = "PL/Python - procedural language"
 
--- postgresql-10.0/contrib/hstore_plpython/Makefile.1	2017-10-14 13:29:34.418055278 +0000
+++ postgresql-10.0/contrib/hstore_plpython/Makefile	2017-10-14 13:29:43.140780678 +0000
@@ -28,7 +28,7 @@
 # ... see silliness in plpython Makefile ...
 SHLIB_LINK += $(sort $(wildcard ../../src/pl/plpython/libpython*.a))
 else
-rpathdir = $(python_libdir)
+#rpathdir = $(python_libdir)
 SHLIB_LINK += $(python_libspec) $(python_additional_libs)
 endif
 
