--- idn-1.2c/config.m4.~1~	2010-03-28 19:28:12.000000000 +0400
+++ idn-1.2c/config.m4	2013-10-09 18:20:12.940661922 +0400
@@ -12,7 +12,7 @@
   PHP_EXTENSION(idn, $ext_shared)
 
   for i in /usr /usr/local $PHP_IDN; do
-    if test -f $i/include/idna.h; then
+    if test -f $i/include/idn/idna.h; then
       IDN_DIR=$i
     fi
   done
@@ -21,7 +21,7 @@
     AC_MSG_ERROR([Cannot find libidn])
   fi
 
-  IDN_INCDIR=$IDN_DIR/include
+  IDN_INCDIR=$IDN_DIR/include/idn
   IDN_LIBDIR=$IDN_DIR/lib
 
   AC_CHECK_LIB(idn,
