--- system-config-printer-1.0.9/Makefile.am	2008-10-21 10:02:45.000000000 +0100
+++ system-config-printer-1.0.9.hacked/Makefile.am	2008-11-06 10:26:32.144560000 +0000
@@ -106,10 +106,6 @@
 	system-config-printer-applet		\
 	my-default-printer
 
-man_MANS=					\
-	man/system-config-printer.1		\
-	man/system-config-printer-applet.1
-
 dbus_DATA =\
 	newprinternotification.conf
 dbusdir = $(sysconfdir)/dbus-1/system.d/
@@ -185,10 +181,6 @@
 
 @INTLTOOL_DESKTOP_RULE@
 
-# The man pages are generated from DocBook XML.
-$(man_MANS): $(top_srcdir)/man/system-config-printer.xml
-	xmlto man -o man $<
-
 html:	$(EXPORT_MODULES)
 	rm -rf html
 	epydoc -o html --html $(EXPORT_MODULES)
