Avoid passing -m64 to the linker

--- tomcat-connectors-1.2.42-src/native/configure.ac.~1~	2019-02-28 17:37:09.527852901 +0000
+++ tomcat-connectors-1.2.42-src/native/configure.ac	2019-02-28 17:38:03.399264651 +0000
@@ -671,7 +671,7 @@
 
 dnl Prefix all LDFLAGS with "-Wl,",
 dnl because we pass them to libtool
-JK_PREFIX_IF_MISSING(APXSLDFLAGS, [-Wl,])
+# JK_PREFIX_IF_MISSING(APXSLDFLAGS, [-Wl,])
 
 AC_SUBST(APXSLDFLAGS)
 
