--- xf86-video-r128-6.10.2/src/Makefile.am.orig	2017-01-17 23:42:44.000000000 +0100
+++ xf86-video-r128-6.10.2/src/Makefile.am	2023-02-05 16:08:25.729502556 +0100
@@ -38,7 +38,7 @@
 
 r128_drv_la_LTLIBRARIES = r128_drv.la
 r128_drv_la_LDFLAGS = -module -avoid-version
-r128_drv_ladir = @moduledir@/drivers
+r128_drv_ladir = @moduledir@
 r128_drv_la_SOURCES = \
 	r128_accel.c r128_cursor.c r128_driver.c \
 	r128_video.c r128_misc.c r128_crtc.c r128_output.c r128_probe.c \
