We have no a2wsgi packaged yet.

--- uvicorn-0.39.0/pyproject.toml.orig
+++ uvicorn-0.39.0/pyproject.toml
@@ -70,7 +70,6 @@
     # check dist
     "twine==6.1.0",
     # Explicit optionals,
-    "a2wsgi==1.10.8",
     "wsproto==1.2.0",
     "websockets==13.1",
 ]
--- uvicorn-0.39.0/tests/middleware/test_wsgi.py.orig
+++ uvicorn-0.39.0/tests/middleware/test_wsgi.py
@@ -5,7 +5,6 @@
 from collections.abc import AsyncGenerator
 from typing import Callable
 
-import a2wsgi
 import httpx
 import pytest
 
@@ -53,7 +52,7 @@
         return [output]
 
 
-@pytest.fixture(params=[wsgi._WSGIMiddleware, a2wsgi.WSGIMiddleware])
+@pytest.fixture(params=[wsgi._WSGIMiddleware])
 def wsgi_middleware(request: pytest.FixtureRequest) -> Callable:
     return request.param
 
