--- domdf_python_tools-3.9.0/tox.ini.orig
+++ domdf_python_tools-3.9.0/tox.ini
@@ -261,8 +261,4 @@
 extras = all
 ignore_errors = True
 commands =
-    python --version
-    python -m pip install pytz>=2019.1
     python -m pytest --cov=domdf_python_tools -r aR tests/ {posargs}
-    python -m pip uninstall pytz -y
-    python -m pytest --cov=domdf_python_tools -r aR tests/test_dates.py --cov-fail-under=0 --cov-append {posargs}
