--- a/Lib/ctypes/util.py	2018-08-31 20:16:48.909478118 +0000
+++ b/Lib/ctypes/util.py	2018-08-31 20:20:00.496166118 +0000
@@ -144,7 +144,7 @@ elif os.name == "posix":
                 return None
 
             try:
-                proc = subprocess.Popen(("/usr/ccs/bin/dump", "-Lpv", f),
+                proc = subprocess.Popen(("/usr/bin/dump", "-Lpv", f),
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.DEVNULL)
             except OSError:  # E.g. command not found
