--- Python-3.4.5/setup.py	2016-07-18 23:55:02.274359616 +0300
+++ Python-3.4.5/setup.py	2016-07-18 23:56:02.325232996 +0300
@@ -1290,6 +1290,9 @@
         curses_defines = []
         curses_includes = []
         panel_library = 'panel'
+        if host_platform == 'sunos5':
+            curses_library = 'ncurses'
+
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
             curses_includes.append('/usr/include/ncursesw')
