--- a/src/src/app.cpp	Sat Mar 28 12:36:04 2020
+++ b/src/src/app.cpp	Tue Aug 16 07:55:02 2022
@@ -1149,7 +1149,7 @@
     base = wxFileName(base).GetPath();
 #endif
 #if defined(sun) || defined(__sun)
-    base = wxString(getexecname(),wxConvCurrent);
+    base = wxString(getexecname(),*wxConvCurrent);
     base = wxFileName(base).GetPath();
 #endif
 #if defined(__APPLE__) && defined(__MACH__)
--- a/src/src/main.cpp	Tue Aug 16 08:12:37 2022
+++ b/src/src/main.cpp	Tue Aug 16 08:16:36 2022
@@ -2679,8 +2679,7 @@
     else
     {
         wxBusyCursor useless;
-        wxPaintEvent e;
-        ProcessEvent(e);
+        this->Refresh();
         for (unsigned int i = 0; i < files.GetCount(); ++i)
           success &= OpenGeneric(files[i]);
     }
--- a/src/src/watchesdlg.cpp	Sat Mar 28 12:36:04 2020
+++ b/src/src/watchesdlg.cpp	Tue Aug 16 09:09:48 2022
@@ -103,7 +103,7 @@
                                   cb_unused const wxPoint& pos,
                                   cb_unused const wxSize& sz) const override
     {
-        wxPGWindowList const list;
+        wxPGWindowList const list(nullptr);
         return list;
     }
     void UpdateControl(cb_unused wxPGProperty* property, cb_unused wxWindow* ctrl) const override {}
