--- codeblocks-20.03/src/src/watchesdlg.orig	Fri Feb 25 13:04:20 2022
+++ codeblocks-20.03/src/src/watchesdlg.cpp	Fri Feb 25 13:05:05 2022
@@ -103,7 +103,7 @@
                                   cb_unused const wxPoint& pos,
                                   cb_unused const wxSize& sz) const override
     {
-        wxPGWindowList const list;
+        wxPGWindowList const list(NULL);
         return list;
     }
     void UpdateControl(cb_unused wxPGProperty* property, cb_unused wxWindow* ctrl) const override {}
