--- a/src/sdk/configmanager.cpp	Sat Mar 28 12:36:00 2020
+++ b/src/sdk/configmanager.cpp	Thu Nov 17 05:36:35 2022
@@ -1576,7 +1576,7 @@
             if (!wxDirExists(ConfigManager::plugin_path_global) && wxIsPlatform64Bit())
             {
                 // if standard-path does not exist and we are on 64-bit system, use lib64 instead
-                ConfigManager::plugin_path_global = ((const wxStandardPaths&)wxStandardPaths::Get()).GetInstallPrefix() + _T("/lib64/codeblocks/plugins");
+                ConfigManager::plugin_path_global = ((const wxStandardPaths&)wxStandardPaths::Get()).GetInstallPrefix() + _T("/lib/amd64/codeblocks/plugins");
             }
 #endif // __WXGTK__
         }
