--- codeblocks_25.03/src/sdk/configmanager.cpp.orig
+++ codeblocks_25.03/src/sdk/configmanager.cpp
@@ -1511,7 +1511,7 @@
             if (!wxDirExists(ConfigManager::plugin_path_global) && wxIsPlatform64Bit())
             {
                 // if standard-path does not exist and we are on 64-bit system, use lib64 instead
-                ConfigManager::plugin_path_global = ((const wxStandardPaths&)wxStandardPaths::Get()).GetInstallPrefix() + _T("/lib64/codeblocks/plugins");
+                ConfigManager::plugin_path_global = ((const wxStandardPaths&)wxStandardPaths::Get()).GetInstallPrefix() + _T("/lib/amd64/codeblocks/plugins");
             }
 #endif // __WXGTK__
         }
