--- ruby-1.8.7-p334.orig/ext/extmk.rb	Thu Dec 24 01:01:58 2009
+++ ruby-1.8.7-p334/ext/extmk.rb	Fri Mar 25 10:45:30 2011
@@ -17,6 +17,15 @@
 $ignore = nil
 $message = nil
 
+#The $do_not_build variable explicitly prevents
+# extensions from being built.
+# This is an Oracle internal patch created to prevent
+# Ruby from automatically picking extensions
+# for libraries that people integrate into Userland.
+# This way, we can enable the extensions in our own
+# time.
+$do_not_build = %w(gdbm)
+
 $progname = $0
 alias $PROGRAM_NAME $0
 alias $0 $progname
@@ -440,6 +449,12 @@
 
 hdrdir = $hdrdir
 $hdrdir = $top_srcdir = relative_from(srcdir, $topdir = "..")
+
+$do_not_build.each do |d|
+ exts.delete(d)
+ puts "Extension #{d} has been disabled by the build\n"
+end
+
 exts.each do |d|
   extmake(d) or abort
 end
