Add solaris driver to list of available drivers

--- ORIGINAL/setup.cfg	2015-02-19 18:50:54.000000000 +0000
+++ ironic-2014.2.1/setup.cfg	2015-04-09 16:59:01.035921792 +0100
@@ -57,6 +57,7 @@
 	pxe_ilo = ironic.drivers.pxe:PXEAndIloDriver
 	pxe_drac = ironic.drivers.drac:PXEDracDriver
 	pxe_snmp = ironic.drivers.pxe:PXEAndSNMPDriver
+	solaris = ironic.drivers.solaris:SolarisAndIPMIToolDriver
 ironic.database.migration_backend = 
 	sqlalchemy = ironic.db.sqlalchemy.migration
 
