Add solaris driver to list of available drivers

--- ironic-5.1.0/setup.cfg.~2~	2016-03-31 01:17:58.933550758 -0800
+++ ironic-5.1.0/setup.cfg	2016-03-31 01:19:08.019805547 -0800
@@ -81,6 +81,7 @@ ironic.drivers =
 	pxe_wol = ironic.drivers.pxe:PXEAndWakeOnLanDriver
 	pxe_iscsi_cimc = ironic.drivers.pxe:PXEAndCIMCDriver
 	pxe_agent_cimc = ironic.drivers.agent:AgentAndCIMCDriver
+	solaris = ironic.drivers.solaris:SolarisAndIPMIToolDriver
 ironic.database.migration_backend = 
 	sqlalchemy = ironic.db.sqlalchemy.migration
 tempest.test_plugins = 
