Add solaris driver to list of available drivers

--- ORIGINAL/setup.cfg	2015-06-02 17:45:44.579172327 +0100
+++ ironic-2015.1.0/setup.cfg	2015-06-02 17:48:37.552116117 +0100
@@ -63,6 +63,7 @@
 	pxe_snmp = ironic.drivers.pxe:PXEAndSNMPDriver
 	pxe_irmc = ironic.drivers.pxe:PXEAndIRMCDriver
 	pxe_amt = ironic.drivers.pxe:PXEAndAMTDriver
+	solaris = ironic.drivers.solaris:SolarisAndIPMIToolDriver
 ironic.database.migration_backend = 
 	sqlalchemy = ironic.db.sqlalchemy.migration
 
