#
# prevent man page compress to *.gz,
#
diff -Naru ibus-1.5.13.org/bus/Makefile.am ibus-1.5.13/bus/Makefile.am
--- ibus-1.5.13.org/bus/Makefile.am	2016-02-22 17:46:23.000000000 +0900
+++ ibus-1.5.13/bus/Makefile.am	2016-09-09 19:34:19.088715498 +0900
@@ -165,7 +165,7 @@
 
 man_one_in_files = ibus-daemon.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA =$(man_one_files:.1=.1.gz) 
+man_one_DATA =$(man_one_files) 
 man_onedir = $(mandir)/man1
 %.1: %.1.in
 	$(AM_V_GEN) sed \
diff -Naru ibus-1.5.13.org/bus/Makefile.in ibus-1.5.13/bus/Makefile.in
--- ibus-1.5.13.org/bus/Makefile.in	2016-02-22 17:47:37.000000000 +0900
+++ ibus-1.5.13/bus/Makefile.in	2016-09-09 19:34:19.090235028 +0900
@@ -822,7 +822,7 @@
 
 man_one_in_files = ibus-daemon.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA = $(man_one_files:.1=.1.gz) 
+man_one_DATA = $(man_one_files)
 man_onedir = $(mandir)/man1
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
diff -Naru ibus-1.5.13.org/setup/Makefile.am ibus-1.5.13/setup/Makefile.am
--- ibus-1.5.13.org/setup/Makefile.am	2016-02-22 17:46:23.000000000 +0900
+++ ibus-1.5.13/setup/Makefile.am	2016-09-09 19:34:19.090754323 +0900
@@ -45,7 +45,7 @@
 
 man_one_in_files = ibus-setup.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA =$(man_one_files:.1=.1.gz) 
+man_one_DATA =$(man_one_files)
 man_onedir = $(mandir)/man1
 %.1: %.1.in
 	$(AM_V_GEN) sed \
diff -Naru ibus-1.5.13.org/setup/Makefile.in ibus-1.5.13/setup/Makefile.in
--- ibus-1.5.13.org/setup/Makefile.in	2016-02-22 17:47:39.000000000 +0900
+++ ibus-1.5.13/setup/Makefile.in	2016-09-09 19:34:19.091360317 +0900
@@ -453,7 +453,7 @@
 desktopdir = $(datadir)/applications
 man_one_in_files = ibus-setup.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA = $(man_one_files:.1=.1.gz) 
+man_one_DATA = $(man_one_files)
 man_onedir = $(mandir)/man1
 CLEANFILES = \
 	$(man_one_DATA) \
diff -Naru ibus-1.5.13.org/tools/Makefile.am ibus-1.5.13/tools/Makefile.am
--- ibus-1.5.13.org/tools/Makefile.am	2016-02-22 17:46:23.000000000 +0900
+++ ibus-1.5.13/tools/Makefile.am	2016-09-09 19:34:19.091723047 +0900
@@ -76,7 +76,7 @@
 
 man_one_in_files = ibus.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA =$(man_one_files:.1=.1.gz) 
+man_one_DATA =$(man_one_files) 
 man_onedir = $(mandir)/man1
 %.1: %.1.in
 	$(AM_V_GEN) sed \
diff -Naru ibus-1.5.13.org/tools/Makefile.in ibus-1.5.13/tools/Makefile.in
--- ibus-1.5.13.org/tools/Makefile.in	2016-02-22 17:47:40.000000000 +0900
+++ ibus-1.5.13/tools/Makefile.in	2016-09-09 19:34:19.092221004 +0900
@@ -537,7 +537,7 @@
 bash_completiondir = @datadir@/bash-completion/completions
 man_one_in_files = ibus.1.in
 man_one_files = $(man_one_in_files:.1.in=.1)
-man_one_DATA = $(man_one_files:.1=.1.gz) 
+man_one_DATA = $(man_one_files)
 man_onedir = $(mandir)/man1
 EXTRA_DIST = \
 	$(man_one_in_files) \
