diff -Nrau ibus-1.3.9.org/data/ibus.schemas.in ibus-1.3.9/data/ibus.schemas.in
--- ibus-1.3.9.org/data/ibus.schemas.in	2009-12-07 15:37:01.000000000 +0900
+++ ibus-1.3.9/data/ibus.schemas.in	2010-02-23 15:27:13.524842100 +0900
@@ -18,7 +18,7 @@
       <owner>ibus</owner>
       <type>list</type>
       <list_type>string</list_type>
-      <default>[Control+space,Zenkaku_Hankaku,Alt+Kanji,Alt+grave,Hangul,Alt+Release+Alt_R]</default>
+      <default>[Control+space,Zenkaku_Hankaku,Henkan_Mode,Alt+Kanji,Alt+grave,Hangul,Alt+Release+Alt_R]</default>
       <locale name="C">
         <short>Trigger shortcut keys</short>
 	    <long>The shortcut keys for turning input method on or off</long>
