--- python_dbusmock-0.37.0/dbusmock/testcase.py.orig
+++ python_dbusmock-0.37.0/dbusmock/testcase.py
@@ -170,7 +170,7 @@
 
     def start(self):
         """Start the D-Bus daemon"""
-        argv = ["dbus-daemon", f"--config-file={self._config}", "--print-pid=1"]
+        argv = ["/usr/libexec/dbus-daemon", f"--config-file={self._config}", "--print-pid=1"]
         # pylint: disable=consider-using-with
         self._daemon = subprocess.Popen(argv, stdout=subprocess.PIPE)
         # we don't need the value, but waiting for it ensures that the bus has
