diff -ruN GConf-2.28.0.orig/doc/gconf/Makefile.in GConf-2.28.0/doc/gconf/Makefile.in
--- GConf-2.28.0.orig/doc/gconf/Makefile.in	2010-01-12 15:49:14.146948482 +0000
+++ GConf-2.28.0/doc/gconf/Makefile.in	2010-01-12 15:49:45.537400278 +0000
@@ -239,7 +239,7 @@
 MKDB_OPTIONS = --sgml-mode --output-format=xml
 
 # Extra options to supply to gtkdoc-fixref
-FIXXREF_OPTIONS = --extra-dir=../glib/html
+FIXXREF_OPTIONS = --extra-dir=../glib/html --module=$(DOC_MODULE)
 
 # Used for dependencies
 HFILE_GLOB = $(top_srcdir)/gconf/*.h
