--- GConf-2.32.4/gconf/default.path.in.~1~	2015-01-05 20:47:10.241300126 +0300
+++ GConf-2.32.4/gconf/default.path.in	2015-01-05 20:49:50.174890995 +0300
@@ -21,7 +21,8 @@
 xml:readwrite:$(HOME)/.gconf
 
 # Location for system-wide settings that are set by the defaults mechanism
-xml:readonly:@sysgconfdir@/gconf.xml.system
+# We dont use this default mechanism on opensolaris
+# xml:readonly:@sysgconfdir@/gconf.xml.system
 
 # To read in any defaults settings that the Sys Admin may have created
 # prior to a desktop system upgrade. The SysAdmin can stick default values
