diff -ruN gnome-system-monitor-2.27.4.orig/gnome-system-monitor.desktop.in.in gnome-system-monitor-2.27.4/gnome-system-monitor.desktop.in.in
--- gnome-system-monitor-2.27.4.orig/gnome-system-monitor.desktop.in.in	2009-08-27 23:37:34.872724855 +0100
+++ gnome-system-monitor-2.27.4/gnome-system-monitor.desktop.in.in	2009-08-27 23:38:03.541965577 +0100
@@ -1,6 +1,8 @@
 [Desktop Entry]
-_Name=System Monitor
-_Comment=View current processes and monitor system state
+# SUN_BRANDING
+_Name=Performance Monitor
+# SUN_BRANDING
+_Comment=Monitor disk, memory and CPU usage of running processes
 TryExec=gnome-system-monitor
 Exec=gnome-system-monitor
 Icon=utilities-system-monitor
