--- gnome-system-monitor-2.28.2/src/interface.h	2010-02-09 13:23:26.000000000 +0000
+++ gnome-system-monitor-2.28.2/src/interface.h	2014-03-08 15:49:12.409847365 +0000
@@ -20,7 +20,7 @@
 #ifndef _PROCMAN_INTERFACE_H_
 #define _PROCMAN_INTERFACE_H_
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 #include "procman.h"
 
--- gnome-system-monitor-2.28.2/src/load-graph.h	2010-02-09 13:23:26.000000000 +0000
+++ gnome-system-monitor-2.28.2/src/load-graph.h	2014-03-08 15:49:12.413941133 +0000
@@ -1,7 +1,7 @@
 #ifndef _PROCMAN_LOAD_GRAPH_H_
 #define _PROCMAN_LOAD_GRAPH_H_
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glibtop/cpu.h>
 
 enum
--- gnome-system-monitor-2.28.2/src/lsof.h	2010-02-09 13:23:26.000000000 +0000
+++ gnome-system-monitor-2.28.2/src/lsof.h	2014-03-08 15:49:12.404614273 +0000
@@ -1,7 +1,7 @@
 #ifndef H_PROCMAN_LSOF_1161179202
 #define H_PROCMAN_LSOF_1161179202
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include "procman.h"
 
 void procman_lsof(ProcData *data);
--- gnome-system-monitor-2.28.2/src/openfiles.h	2010-02-09 13:23:26.000000000 +0000
+++ gnome-system-monitor-2.28.2/src/openfiles.h	2014-03-08 15:49:12.413345276 +0000
@@ -1,7 +1,7 @@
 #ifndef _OPENFILES_H_
 #define _OPENFILES_H_
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "procman.h"
 
--- gnome-system-monitor-2.28.2/src/procman_gnomesu.h	2010-02-09 13:23:26.000000000 +0000
+++ gnome-system-monitor-2.28.2/src/procman_gnomesu.h	2014-03-08 15:49:12.409212717 +0000
@@ -1,7 +1,7 @@
 #ifndef H_GNOME_SYSTEM_MONITOR_GNOMESU_H_1132171917
 #define H_GNOME_SYSTEM_MONITOR_GNOMESU_H_1132171917
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 gboolean
 procman_gnomesu_create_root_password_dialog(const char * message);
--- gnome-system-monitor-2.28.2/src/util.cpp	2014-03-08 15:50:21.308920627 +0000
+++ gnome-system-monitor-2.28.2/src/util.cpp	2014-03-08 15:49:12.408392689 +0000
@@ -1,7 +1,7 @@
 #include <config.h>
 
 #include <glib/gi18n.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include <glibtop/proctime.h>
