--- jupyter_core-5.7.2/tests/test_paths.py.orig
+++ jupyter_core-5.7.2/tests/test_paths.py
@@ -283,6 +283,7 @@
         assert path[-2:] == system_path
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/jupyter/jupyter_core/issues/231")
 def test_jupyter_path_user_site():
     with patch.object(site, "ENABLE_USER_SITE", True):
         path = jupyter_path()
@@ -301,6 +302,7 @@
             assert p == v
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/jupyter/jupyter_core/issues/231")
 def test_jupyter_path_no_user_site():
     with patch.object(site, "ENABLE_USER_SITE", False):
         path = jupyter_path()
@@ -308,6 +310,7 @@
         assert path[1] == paths.ENV_JUPYTER_PATH[0]
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/jupyter/jupyter_core/issues/231")
 def test_jupyter_path_prefer_env():
     with prefer_env:
         path = jupyter_path()
