diff -urN nautilus-2.30.1/data/nautilus-folder-handler.desktop.in.in ../SUNWgnome-file-mgr-2.30.1.hacked/nautilus-2.30.1/data/nautilus-folder-handler.desktop.in.in
--- nautilus-2.30.1/data/nautilus-folder-handler.desktop.in.in	2009-12-15 10:22:12.000000000 +0000
+++ ../SUNWgnome-file-mgr-2.30.1.hacked/nautilus-2.30.1/data/nautilus-folder-handler.desktop.in.in	2011-03-23 14:08:31.164403827 +0000
@@ -7,7 +7,7 @@
 Icon=folder-open
 StartupNotify=true
 Type=Application
-MimeType=x-directory/gnome-default-handler;x-directory/normal;inode/directory;application/x-gnome-saved-search;
+MimeType=inode/directory;application/x-gnome-saved-search;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=nautilus
