--- nautilus-2.30.1/eel/eel-editable-label.c	2010-04-12 11:43:44.000000000 +0100
+++ nautilus-2.30.1/eel/eel-editable-label.c	2014-03-01 21:04:34.205905810 +0000
@@ -682,7 +682,7 @@
  * Return value: the text in the label widget. This is the internal
  *   string used by the label, and must not be modified.
  **/
-G_CONST_RETURN gchar *
+const gchar *
 eel_editable_label_get_text (EelEditableLabel *label)
 {
   g_return_val_if_fail (EEL_IS_EDITABLE_LABEL (label), NULL);
@@ -4269,7 +4269,7 @@
   atk_object_set_role (accessible, ATK_ROLE_TEXT);
 }
 
-static G_CONST_RETURN gchar*
+static const gchar*
 eel_editable_label_accessible_get_name (AtkObject *accessible)
 {
   if (accessible->name != NULL)
--- nautilus-2.30.1/eel/eel-editable-label.h	2009-12-08 09:50:42.000000000 +0000
+++ nautilus-2.30.1/eel/eel-editable-label.h	2014-03-01 21:04:34.194514736 +0000
@@ -112,7 +112,7 @@
 GtkWidget*            eel_editable_label_new                  (const char       *str);
 void                  eel_editable_label_set_text             (EelEditableLabel *label,
 							       const char       *str);
-G_CONST_RETURN gchar* eel_editable_label_get_text             (EelEditableLabel *label);
+const gchar* eel_editable_label_get_text             (EelEditableLabel *label);
 void                  eel_editable_label_set_justify          (EelEditableLabel *label,
 							       GtkJustification  jtype);
 GtkJustification      eel_editable_label_get_justify          (EelEditableLabel *label);
--- nautilus-2.30.1/eel/eel-i18n.c	2009-04-20 11:57:19.000000000 +0100
+++ nautilus-2.30.1/eel/eel-i18n.c	2014-03-01 21:04:34.193310626 +0000
@@ -32,7 +32,7 @@
 
 
 
-G_CONST_RETURN char *
+const char *
 _eel_gettext (const char *str)
 {
 	static gboolean _eel_gettext_initialized = FALSE;
--- nautilus-2.30.1/eel/eel-i18n.h	2009-04-20 11:57:19.000000000 +0100
+++ nautilus-2.30.1/eel/eel-i18n.h	2014-03-01 21:04:34.206544660 +0000
@@ -28,7 +28,7 @@
 #ifdef ENABLE_NLS
 #include <glib.h>
 
-G_CONST_RETURN char *
+const char *
 _eel_gettext (const char *str) G_GNUC_FORMAT (1);
 
 #include <libintl.h>
--- nautilus-2.30.1/eel/eel-labeled-image.c	2010-04-12 11:43:44.000000000 +0100
+++ nautilus-2.30.1/eel/eel-labeled-image.c	2014-03-01 21:04:34.186708919 +0000
@@ -2143,7 +2143,7 @@
 	return EEL_LABELED_IMAGE (widget);
 }
 
-static G_CONST_RETURN gchar *
+static const gchar *
 eel_labeled_image_accessible_get_name (AtkObject *accessible)
 {
 	EelLabeledImage *labeled_image;
