--- pytest-8.3.3/testing/test_helpconfig.py.orig
+++ pytest-8.3.3/testing/test_helpconfig.py
@@ -6,6 +6,7 @@
 import pytest
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
 def test_version_verbose(pytester: Pytester, pytestconfig, monkeypatch) -> None:
     monkeypatch.delenv("PYTEST_DISABLE_PLUGIN_AUTOLOAD")
     result = pytester.runpytest("--version", "--version")
@@ -15,6 +16,7 @@
         result.stdout.fnmatch_lines(["*registered third-party plugins:", "*at*"])
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
 def test_version_less_verbose(pytester: Pytester, pytestconfig, monkeypatch) -> None:
     monkeypatch.delenv("PYTEST_DISABLE_PLUGIN_AUTOLOAD")
     result = pytester.runpytest("--version")
--- pytest-8.3.3/testing/test_junitxml.py.orig
+++ pytest-8.3.3/testing/test_junitxml.py
@@ -1356,6 +1356,7 @@
     result.stdout.fnmatch_lines(expected_lines)
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
 def test_random_report_log_xdist(
     pytester: Pytester, monkeypatch: MonkeyPatch, run_and_parse: RunAndParse
 ) -> None:
@@ -1413,6 +1414,7 @@
     assert first == second
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
 def test_runs_twice_xdist(
     pytester: Pytester, monkeypatch: MonkeyPatch, run_and_parse: RunAndParse
 ) -> None:
--- pytest-8.3.3/testing/test_terminal.py.orig
+++ pytest-8.3.3/testing/test_terminal.py
@@ -855,6 +855,7 @@
         result.stdout.fnmatch_lines(["test_passes.py ..*", "* 2 pass*"])
         assert result.ret == 0
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_header_trailer_info(
         self, monkeypatch: MonkeyPatch, pytester: Pytester, request
     ) -> None:
@@ -1063,6 +1064,7 @@
         )
         assert result.ret == 1
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_verbose_reporting_xdist(
         self,
         verbose_testfile,
@@ -2200,6 +2202,7 @@
             ]
         )
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_xdist_normal(
         self, many_tests_files, pytester: Pytester, monkeypatch
     ) -> None:
@@ -2208,6 +2211,7 @@
         output = pytester.runpytest("-n2")
         output.stdout.re_match_lines([r"\.{20} \s+ \[100%\]"])
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_xdist_normal_count(
         self, many_tests_files, pytester: Pytester, monkeypatch
     ) -> None:
@@ -2222,6 +2226,7 @@
         output = pytester.runpytest("-n2")
         output.stdout.re_match_lines([r"\.{20} \s+ \[20/20\]"])
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_xdist_verbose(
         self, many_tests_files, pytester: Pytester, monkeypatch
     ) -> None:
@@ -2366,6 +2371,7 @@
             )
         )
 
+    @pytest.mark.skip(reason="always fails: https://github.com/pytest-dev/pytest/issues/12073")
     def test_xdist_normal(self, many_files, pytester: Pytester, monkeypatch) -> None:
         pytest.importorskip("xdist")
         monkeypatch.delenv("PYTEST_DISABLE_PLUGIN_AUTOLOAD", raising=False)
