https://github.com/pytest-dev/pytest/pull/12616

--- pytest-8.2.2/src/_pytest/config/__init__.py.orig
+++ pytest-8.2.2/src/_pytest/config/__init__.py
@@ -835,7 +835,7 @@
     ) -> None:
         plugins = _get_plugin_specs_as_list(spec)
         for import_spec in plugins:
-            self.import_plugin(import_spec)
+            self.import_plugin(import_spec, consider_entry_points=True)
 
     def import_plugin(self, modname: str, consider_entry_points: bool = False) -> None:
         """Import a plugin with ``modname``.
