--- sudo-1.9.13p1/configure.ac.orig	2023-02-21 06:40:53.635621273 +0100
+++ sudo-1.9.13p1/configure.ac	2023-02-21 06:45:22.074929870 +0100
@@ -1817,9 +1817,9 @@
 		AS_IF([test "$with_AFS" = "yes"], [
 		    AFS_LIBS="-lc -lucb"
 		])
-		: ${mansectsu='1m'}
-		: ${mansectform='4'}
-		: ${mansectmisc='5'}
+		: ${mansectsu='8'}
+		: ${mansectform='5'}
+		: ${mansectmisc='7'}
 		test -z "$with_pam" && AUTH_EXCL_DEF="PAM"
 		AC_CHECK_FUNCS([priv_set], [PSMAN=1])
 		;;
