https://bugzilla.sudo.ws/show_bug.cgi?id=902

--- sudo-1.8.28p1/plugins/sudoers/sudoers.in	2019-10-10 18:32:33.000000000 +0000
+++ sudo-1.8.28p1/plugins/sudoers/sudoers.in.new	2019-10-16 19:40:32.874578851 +0000
@@ -94,4 +94,4 @@ root ALL=(ALL) ALL
 
 ## Read drop-in files from @sysconfdir@/sudoers.d
 ## (the '#' here does not indicate a comment)
-#includedir @sysconfdir@/sudoers.d
+# includedir @sysconfdir@/sudoers.d
