--- gdb-7.6/gdb/config/sparc/sol2.mh	2012-01-10 08:30:49.000000000 -0800
+++ gdb-7.6/gdb/config/sparc/sol2.mh	2013-10-05 16:57:25.851694100 -0700
@@ -1,5 +1,5 @@
 # Host: Solaris SPARC & UltraSPARC
 NAT_FILE= nm-sol2.h
-NATDEPFILES= sparc-sol2-nat.o \
-	core-regset.o fork-child.o \
-	procfs.o proc-api.o proc-events.o proc-flags.o proc-why.o
+NATDEPFILES= sol2-core-regset.o sparc-sol2-nat.o \
+	fork-child.o procfs.o proc-api.o proc-events.o \
+        proc-flags.o proc-why.o
