# Solaris-specific ELF information.
# Will contribute upstream if possible.

--- gdb-7.11-orig/bfd/elfxx-target.h	Tue Feb  9 19:19:39 2016
+++ gdb-7.11/bfd/elfxx-target.h	Thu Mar 10 13:12:46 2016
@@ -550,6 +550,9 @@
 #ifndef elf_backend_grok_psinfo
 #define elf_backend_grok_psinfo			NULL
 #endif
+#ifndef elf_backend_grok_lwpstatus
+#define elf_backend_grok_lwpstatus              NULL
+#endif
 #ifndef elf_backend_write_core_note
 #define elf_backend_write_core_note		NULL
 #endif
@@ -762,6 +765,7 @@
   elf_backend_sort_relocs_p,
   elf_backend_grok_prstatus,
   elf_backend_grok_psinfo,
+  elf_backend_grok_lwpstatus,
   elf_backend_write_core_note,
   elf_backend_lookup_section_flags_hook,
   elf_backend_reloc_type_class,
