# Patch required for Solaris.
# Will contribute upstream if possible.

--- gdb-7.11-orig/gdb/config/sparc/sol2.mh	Thu Feb 19 03:58:07 2015
+++ gdb-7.11/gdb/config/sparc/sol2.mh	Thu Mar 10 17:02:29 2016
@@ -1,6 +1,6 @@
 # Host: Solaris SPARC & UltraSPARC
 NAT_FILE= nm-sol2.h
-NATDEPFILES= sparc-sol2-nat.o \
+NATDEPFILES= sol2-core-regset.o sparc-sol2-nat.o \
 	fork-child.o \
 	procfs.o proc-api.o proc-events.o proc-flags.o proc-why.o
 HAVE_NATIVE_GCORE_HOST = 1
