# Solaris-specific ELF information
# Will contribute upstream if possible.

--- gdb-7.11-orig/bfd/elf-bfd.h	Wed Feb 24 01:55:15 2016
+++ gdb-7.11/bfd/elf-bfd.h	Thu Mar 10 13:12:46 2016
@@ -1190,6 +1190,11 @@
   bfd_boolean (*elf_backend_grok_psinfo)
     (bfd *, Elf_Internal_Note *);
 
+  /* This function, if defined, is called when an NT_LWPSTATUS
+     note is found in a core file.  */
+  bfd_boolean (*elf_backend_grok_lwpstatus)
+    (bfd *, Elf_Internal_Note *);
+
   /* This function, if defined, is called to write a note to a corefile.  */
   char *(*elf_backend_write_core_note)
     (bfd *abfd, char *buf, int *bufsiz, int note_type, ...);
