--- /dev/null	2013-08-23 08:27:09.000000000 -0700
+++ gdb-7.6/gdb/features/i386/32bit-solaris.c	2013-08-23 08:26:12.807778351 -0700
@@ -0,0 +1,16 @@
+/* THIS FILE IS GENERATED.  -*- buffer-read-only: t -*- vi:set ro:
+  Original: 32bit-solaris.xml */
+
+#include "defs.h"
+#include "osabi.h"
+#include "target-descriptions.h"
+
+struct target_desc *tdesc_32bit_solaris;
+static void
+initialize_tdesc_32bit_solaris (void)
+{
+  struct target_desc *result = allocate_target_description ();
+  struct tdesc_feature *feature;
+
+  tdesc_32bit_solaris = result;
+}
