# Patch required for Solaris.
# Will contribute upstream if possible.

--- gdb-7.11-orig/gdb/solib-svr4.c	Tue Feb  9 19:19:39 2016
+++ gdb-7.11/gdb/solib-svr4.c	Thu Mar 10 13:12:47 2016
@@ -174,6 +174,10 @@
       && strcmp (inferior_so_name, "/lib/ld.so.1") == 0)
     return 1;
 
+  if (strcmp (gdb_so_name, "/usr/lib/amd64/ld.so.1") == 0
+      && strcmp (inferior_so_name, "/lib/amd64/ld.so.1") == 0)
+    return 1;
+
   /* Similarly, we observed the same issue with sparc64, but with
      different locations.  */
   if (strcmp (gdb_so_name, "/usr/lib/sparcv9/ld.so.1") == 0
