--- gdb-6.8/gdb/solib-svr4.c.orig	Fri Jul 10 12:25:23 2009
+++ gdb-6.8/gdb/solib-svr4.c	Fri Jul 10 12:25:34 2009
@@ -1649,6 +1649,12 @@
   if (strcmp (gdb->so_original_name, "/usr/lib/ld.so.1") == 0
       && strcmp (inferior->so_original_name, "/lib/ld.so.1") == 0)
     return 1;
+  if (strcmp (gdb->so_original_name, "/usr/lib/amd64/ld.so.1") == 0
+      && strcmp (inferior->so_original_name, "/lib/amd64/ld.so.1") == 0)
+    return 1;
+  if (strcmp (gdb->so_original_name, "/usr/lib/sparcv9/ld.so.1") == 0
+      && strcmp (inferior->so_original_name, "/lib/sparcv9/ld.so.1") == 0)
+    return 1;
 
   return 0;
 }
