Change the interpreter to a specific version of python

--- fio-3.28/tools/hist/fio-histo-log-pctiles.py-orig	Wed Sep  8 09:59:48 2021
+++ fio-3.28/tools/hist/fio-histo-log-pctiles.py	Tue Nov 23 14:46:37 2021
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 
 # module to parse fio histogram log files, not using pandas
 # runs in python v2 or v3
--- fio-3.28/tools/plot/fio2gnuplot-orig	Wed Sep  8 09:59:48 2021
+++ fio-3.28/tools/plot/fio2gnuplot	Tue Nov 23 14:48:49 2021
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Note: this script is python2 and python3 compatible.
 #
 #  Copyright (C) 2013 eNovance SAS <licensing@enovance.com>
--- fio-3.28/tools/fio_jsonplus_clat2csv-orig	Wed Sep  8 09:59:48 2021
+++ fio-3.28/tools/fio_jsonplus_clat2csv	Tue Nov 23 14:50:43 2021
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Note: this script is python2 and python3 compatible.
 
 """
--- fio-3.28/tools/fiologparser.py-orig	Wed Sep  8 09:59:48 2021
+++ fio-3.28/tools/fiologparser.py	Tue Nov 23 15:10:36 2021
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3.9
 # Note: this script is python2 and python 3 compatible.
 #
 # fiologparser.py
