--- mod_perl-2.0.4/Apache-Test/lib/Apache/TestConfigParse.pm	Sun Nov 11 21:34:46 2007
+++ mod_perl-2.0.4/Apache-Test/lib/Apache/TestConfigParse.pm	Thu May 19 05:35:01 2011
@@ -211,6 +211,11 @@
     for my $args (@{ $c->{$directive} }) {
         my $modname = $args->[0];
         my $file = $self->server_file_rel2abs($args->[1]);
+	
+       unless ($file !~ m/\/64\//) {
+            debug "$file is 64 bit module (we don't want it)";
+            next;
+        }
 
         unless (-e $file) {
             debug "$file does not exist, skipping LoadModule";
