--- uthash-2.3.0/Makefile.orig	2021-07-07 03:57:18.876256617 +0000
+++ uthash-2.3.0/Makefile	2021-07-07 03:57:01.242117313 +0000
@@ -0,0 +1,21 @@
+PREFIX ?= /usr
+
+all:
+	@echo srcdir is $(srcdir)
+
+install:
+	@mkdir -p $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p utarray.h $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p uthash.h $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p utlist.h $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p utringbuffer.h $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p utstack.h $(DESTDIR)$(PREFIX)/include/uthash
+	@cd src/ && cp -p utstring.h $(DESTDIR)$(PREFIX)/include/uthash
+
+uninstall:
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/utarray.h
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/uthash.h
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/utlist.h
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/utringbuffer.h
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/utstack.h
+	@rm -rf $(DESTDIR)$(PREFIX)/include/uthash/utstring.h
