--- fbreader-0.12.10/makefiles/arch/unix.mk	Thu Apr  1 15:14:24 2010
+++ fbreader-0.12.10-patch/makefiles/arch/unix.mk	Sun Jul 11 21:10:23 2010
@@ -1,0 +1,1 @@
+ZLSHARED = no
--- fbreader-0.12.10/makefiles/subdir.mk.~1~	2014-01-12 17:23:52.322089454 +0400
+++ fbreader-0.12.10/makefiles/subdir.mk	2014-01-12 17:24:37.964815359 +0400
@@ -8,6 +8,8 @@
 SOURCES_OBJC = $(wildcard *.m)
 OBJECTS = $(patsubst %.cpp, %.o, $(SOURCES_CPP)) $(patsubst %.M, %.o, $(SOURCES_OBJCPP)) $(patsubst %.m, %.o, $(SOURCES_OBJC))
 
+CFLAGS += -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include
+
 .SUFFIXES: .cpp .M .m .o .h
 
 .cpp.o:
