--- fbreader-0.12.10/fbreader/src/library/Library.cpp	2010-04-01 15:14:24.000000000 +0200
+++ fbreader-0.12.10/fbreader/src/library/Library.cpp	2011-11-06 12:59:42.135088678 +0100
@@ -39,7 +39,7 @@
 shared_ptr<Library> Library::ourInstance;
 const size_t Library::MaxRecentListSize = 10;
 
-Library::Library &Library::Instance() {
+Library &Library::Instance() {
 	if (ourInstance.isNull()) {
 		ourInstance = new Library();
 	}
