--- ./src/bin/e.h.orig	2015-02-16 01:05:11.507185953 +0000
+++ ./src/bin/e.h	2015-02-16 01:08:30.887669995 +0000
@@ -122,7 +122,7 @@
 # include <Elementary.h>
 
 # ifdef HAVE_HAL
-#  include <E_Hal.h>
+//#  include <E_Hal.h>
 # endif
 
 # ifdef HAVE_WAYLAND
@@ -180,6 +180,9 @@
 typedef struct _E_Rect         E_Rect;
 
 /* convenience macro to compress code and avoid typos */
+#ifndef MIN
+# define MIN(x, y) (((x) > (y)) ? (y) : (x))
+#endif
 #ifndef MAX
 # define MAX(x, y) (((x) > (y)) ? (x) : (y))
 #endif
