diff -Nuraw stunnel-5.74_O/src/log.c stunnel-5.74/src/log.c
--- stunnel-5.74_O/src/log.c	2024-10-14 20:32:52.000000000 +0200
+++ stunnel-5.74/src/log.c	2025-01-01 15:36:52.000000000 +0100
@@ -36,6 +36,7 @@
  */

 #include "prototypes.h"
+#include <alloca.h>

 NOEXPORT void log_queue(SERVICE_OPTIONS *, int, char *, char *, char *);
 NOEXPORT void log_raw(SERVICE_OPTIONS *, int, char *, char *, char *);
