# Make the 'authpriv' syslog facility the default on Solaris
#

diff -u -r stunnel-4.55.orig/doc/stunnel.fr.pod stunnel-4.55/doc/stunnel.fr.pod
--- stunnel-4.55.orig/doc/stunnel.fr.pod	2012-12-02 11:00:24.000000000 -0800
+++ stunnel-4.55/doc/stunnel.fr.pod	2013-03-21 22:30:02.672293057 -0700
@@ -178,7 +178,7 @@
 B<debug = 7> donneront le maximum d'informations. La valeur par défaut
 est notice (5).
 
-La facilité syslog «E<nbsp>daemonE<nbsp>» est utilisée, sauf si un autre nom est spécifié
+La facilité syslog «E<nbsp>authprivE<nbsp>» est utilisée, sauf si un autre nom est spécifié
 (Win32 ne permet pas l'usage des facilités.)
 
 La casse est ignorée, aussi bien pour la facilité que pour le niveau.
diff -u -r stunnel-4.55.orig/doc/stunnel.pod stunnel-4.55/doc/stunnel.pod
--- stunnel-4.55.orig/doc/stunnel.pod	2013-01-13 09:25:20.000000000 -0800
+++ stunnel-4.55/doc/stunnel.pod	2013-03-21 22:28:04.473314299 -0700
@@ -184,7 +184,7 @@
 all levels numerically less than it will be shown.  Use I<debug = debug> or
 I<debug = 7> for greatest debugging output.  The default is notice (5).
 
-The syslog facility 'daemon' will be used unless a facility name is supplied.
+The syslog facility 'authpriv' will be used unless a facility name is supplied.
 (Facilities are not supported on Win32.)
 
 Case is ignored for both facilities and levels.
diff -u -r stunnel-4.55.orig/src/options.c stunnel-4.55/src/options.c
--- stunnel-4.55.orig/src/options.c	2013-02-02 08:20:32.000000000 -0800
+++ stunnel-4.55/src/options.c	2013-03-21 22:27:13.163038368 -0700
@@ -185,8 +185,12 @@
     case CMD_BEGIN:
         new_global_options.debug_level=LOG_NOTICE;
 #if !defined (USE_WIN32) && !defined (__vms)
+#if defined(LOG_AUTHPRIV)
+        new_global_options.facility=LOG_AUTHPRIV;
+#else
         new_global_options.facility=LOG_DAEMON;
 #endif
+#endif
         break;
     case CMD_EXEC:
         if(strcasecmp(opt, "debug"))
