--- firefox-9e7131f6df9112288f17666bddbe901dbc3527db/browser/app/profile/firefox.js.old	2025-05-19 22:16:00.116322321 -0400
+++ firefox-9e7131f6df9112288f17666bddbe901dbc3527db/browser/app/profile/firefox.js	2025-05-19 22:17:14.993000713 -0400
@@ -3401,3 +3401,9 @@
 pref("browser.ipProtection.variant", "");
 
 pref("browser.ipProtection.guardian.endpoint", "https://vpn.mozilla.org/");
+
+// Audio/video doesn't work with new Utility system
+pref("media.allow-audio-non-utility",true);
+
+// Disable privacy-perserving ads
+pref("dom.private-attribution.submission.enabled", false);
