--- firefox-130.0/browser/app/profile/firefox.js.old	2024-08-08 16:42:17.763558842 -0400
+++ firefox-130.0/browser/app/profile/firefox.js	2024-08-08 16:45:00.766707882 -0400
@@ -3198,3 +3198,9 @@
 pref("toolkit.contentRelevancy.ingestEnabled", false);
 // Pref to enable extra logging for the content relevancy feature
 pref("toolkit.contentRelevancy.log", false);
+
+// Audio/video doesn't work with new Utility system
+pref("media.allow-audio-non-utility",true);
+
+// Disable privacy-perserving ads
+pref("dom.private-attribution.submission.enabled", false);
