--- firefox-82e2435fd101e64854b8c56ac792c8a5522db4fe/browser/app/profile/firefox.js.~1~	2025-11-20 21:58:34.000000000 -0500
+++ firefox-82e2435fd101e64854b8c56ac792c8a5522db4fe/browser/app/profile/firefox.js	2025-11-21 08:04:37.248651027 -0500
@@ -3542,3 +3542,9 @@
 
 // When enabled, Firefox ignores the distribution.ini file if global.id is MozillaOnline.
 pref("distribution.mozillaonline.ignore", true);
+
+// Audio/video doesn't work with new Utility system
+pref("media.allow-audio-non-utility",true);
+
+// Disable privacy-perserving ads
+pref("dom.private-attribution.submission.enabled", false);
