--- firefox-44.0/media/libstagefright/moz.build.1	2016-01-28 11:27:57.499853402 +0300
+++ firefox-44.0/media/libstagefright/moz.build	2016-01-28 11:28:04.015635089 +0300
@@ -7,9 +7,6 @@
 DEFINES['ANDROID_SMP'] = 0
 DEFINES['LOG_NDEBUG'] = 1
 
-if CONFIG['OS_TARGET'] != 'WINNT':
-   DEFINES['_GLIBCXX_OS_DEFINES'] = True
-
 if CONFIG['OS_TARGET'] == 'WINNT':
     if CONFIG['_MSC_VER']:
         DEFINES['ssize_t'] = 'intptr_t'
