--- firefox-60.2.0/browser/installer/package-manifest.in.orig	2018-09-27 19:20:48.624770136 +0000
+++ firefox-60.2.0/browser/installer/package-manifest.in	2018-09-27 19:21:16.765444657 +0000
@@ -735,11 +735,11 @@
 #endif
 
 ; for Solaris SPARC
-#ifdef SOLARIS
-bin/libfreebl_32fpu_3.so
-bin/libfreebl_32int_3.so
-bin/libfreebl_32int64_3.so
-#endif
+;#ifdef SOLARIS
+;bin/libfreebl_32fpu_3.so
+;bin/libfreebl_32int_3.so
+;bin/libfreebl_32int64_3.so
+;#endif
 
 ; [Updater]
 ;
