--- firefox-52.1.0esr/browser/branding/official/pref/firefox-branding.js.~1~	2017-05-04 11:51:32.593399439 +0300
+++ firefox-52.1.0esr/browser/branding/official/pref/firefox-branding.js	2017-05-04 11:52:41.015959984 +0300
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 pref("startup.homepage_override_url", "");
-pref("startup.homepage_welcome_url", "https://www.mozilla.org/%LOCALE%/firefox/%VERSION%/firstrun/");
+pref("startup.homepage_welcome_url", "file:///usr/share/doc/openindiana-welcome/html/index.html");
 pref("startup.homepage_welcome_url.additional", "");
 // Interval: Time between checks for a new version (in seconds)
 pref("app.update.interval", 43200); // 12 hours
