--- firefox-60.2.0/ipc/chromium/src/third_party/libevent/solaris/event2/event-config.h.old  2018-10-17 13:57:31.238059933 +0000
+++ firefox-60.2.0/ipc/chromium/src/third_party/libevent/solaris/event2/event-config.h  2018-10-17 13:55:24.078008175 +0000
@@ -211,7 +211,7 @@
 #define EVENT__HAVE_SELECT 1
 
 /* Define to 1 if you have the `sendfile' function. */
-#define EVENT__HAVE_SENDFILE 1
+#undef EVENT__HAVE_SENDFILE
 
 /* Define to 1 if you have the `setenv' function. */
 #define EVENT__HAVE_SETENV 1

