--- firefox-68.2.0/browser/branding/official/pref/firefox-branding.js	2019-10-16 21:30:15.000000000 +0000
+++ firefox-68.2.0/browser/branding/official/pref/firefox-branding.js.new	2019-10-23 07:46:52.587177686 +0000
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 pref("startup.homepage_override_url", "");
-pref("startup.homepage_welcome_url", "about:welcome");
+pref("startup.homepage_welcome_url", "file:///usr/share/doc/openindiana-welcome/html/index.html");
 pref("startup.homepage_welcome_url.additional", "");
 // Interval: Time between checks for a new version (in seconds)
 pref("app.update.interval", 43200); // 12 hours
