Do not require extension's signing by default
--- firefox-43.0b6/browser/app/profile/firefox.js.1	2015-11-26 17:56:58.706663813 +0300
+++ firefox-43.0b6/browser/app/profile/firefox.js	2015-11-26 17:57:28.803833794 +0300
@@ -73,8 +73,8 @@
 // See the SCOPE constants in AddonManager.jsm for values to use here.
 pref("extensions.autoDisableScopes", 15);
 
-// Require signed add-ons by default
-pref("xpinstall.signatures.required", true);
+// Don't require signed add-ons by default
+pref("xpinstall.signatures.required", false);
 pref("xpinstall.signatures.devInfoURL", "https://wiki.mozilla.org/Addons/Extension_Signing");
 
 // Dictionary download preference
