--- firefox-43.0b7/browser/base/content/aboutDialog.xul.~1~	2015-11-26 23:11:19.000000000 +0300
+++ firefox-43.0b7/browser/base/content/aboutDialog.xul	2015-11-28 20:40:28.540383967 +0300
@@ -127,6 +127,9 @@
           <description class="text-blurb" id="communityDesc">
             &community.start2;<label class="text-link" href="http://www.mozilla.org/">&community.mozillaLink;</label>&community.middle2;<label class="text-link" href="about:credits">&community.creditsLink;</label>&community.end3;
           </description>
+          <description class="text-blurb" id="thanksMartinDesc">
+            OpenIndiana Hipster Firefox version is based on the <label class="text-link" href="https://github.com/OpenSXCE-org/FireFox-43-port-for-all-OpenSolaris-distros">work of Martin Bochnig</label> made for <label class="text-link" href="http://opensxce.org/">OpenSXCE</label>. 
+          </description>
           <description class="text-blurb" id="contributeDesc">
               &helpus.start;<label class="text-link" href="https://sendto.mozilla.org/page/contribute/Give-Now?source=mozillaorg_default_footer&#38;ref=firefox_about&#38;utm_campaign=firefox_about&#38;tm_source=firefox&#38;tm_medium=referral&#38;utm_content=20140929_FireFoxAbout">&helpus.donateLink;</label>&helpus.middle;<label class="text-link" href="http://www.mozilla.org/contribute/">&helpus.getInvolvedLink;</label>&helpus.end;
           </description>
