--- firefox-52.1.0esr/xpcom/ds/nsMathUtils.h.1	2017-05-04 16:36:03.330988035 +0300
+++ firefox-52.1.0esr/xpcom/ds/nsMathUtils.h	2017-05-04 16:35:45.887771790 +0300
@@ -103,7 +103,7 @@
 #ifdef WIN32
   // NOTE: '!!' casts an int to bool without spamming MSVC warning C4800.
   return !!_finite(aNum);
-#elif defined(XP_DARWIN)
+#elif defined(XP_DARWIN) || defined(SOLARIS)
   // Darwin has deprecated |finite| and recommends |isfinite|. The former is
   // not present in the iOS SDK.
   return std::isfinite(aNum);
