--- firefox-52.2.0esr/config/external/icu/defs.mozbuild.1	2017-06-19 15:19:43.071794106 +0000
+++ firefox-52.2.0esr/config/external/icu/defs.mozbuild	2017-06-19 15:26:00.268002102 +0000
@@ -33,6 +33,15 @@
 elif CONFIG['OS_TARGET'] == 'WINNT':
     CXXFLAGS += ['-GR']
 
+if CONFIG['OS_TARGET'] == 'Solaris':
+    CFLAGS += [ 
+        '-std=c99',
+        '-D_XOPEN_SOURCE=600',
+        '-D__EXTENSIONS__=1',
+        '-D_XPG6'
+    ]
+    CXXFLAGS += [ '-std=c++11' ]
+
 DISABLE_STL_WRAPPING = True
 ALLOW_COMPILER_WARNINGS = True
 
