--- firefox-60.5.2/browser/app/profile/firefox.js	2019-02-22 21:34:33.372446092 +0000
+++ firefox-60.5.2/browser/app/profile/firefox.js.new	2019-02-22 21:56:53.548484823 +0000
@@ -1581,6 +1581,9 @@ pref("dom.ipc.cpows.allow-cpows-in-compa
 // detection).
 pref("dom.ipc.processHangMonitor", true);
 
+// By default use only 1 content process - limits CPU and memory resources.
+pref("dom.ipc.processCount", 1);
+
 #ifdef DEBUG
 // Don't report hangs in DEBUG builds. They're too slow and often a
 // debugger is attached.
