https://www.illumos.org/issues/5876

--- firefox-52.1.0esr/js/src/wasm/WasmSignalHandlers.cpp.1	2017-05-05 15:46:14.362603660 +0300
+++ firefox-52.1.0esr/js/src/wasm/WasmSignalHandlers.cpp	2017-05-05 15:49:02.977093168 +0300
@@ -222,6 +222,11 @@
 # include <sys/ucontext.h> // for ucontext_t, mcontext_t
 #endif
 
+#if defined(__sun)
+# include <ucontext.h> // for ucontext_t, etc
+# include <sys/regset.h>   // for REG_PC, etc
+#endif
+
 #if defined(JS_CPU_X64)
 # if defined(__DragonFly__)
 #  include <machine/npx.h> // for union savefpu
