--- firefox-52.1.1esr/layout/base/nsLayoutUtils.cpp.1	2017-05-11 21:47:14.260439179 +0300
+++ firefox-52.1.1esr/layout/base/nsLayoutUtils.cpp	2017-05-11 21:47:58.995807653 +0300
@@ -3586,7 +3586,7 @@
       nsCString string("dump-");
       // Include the process ID in the dump file name, to make sure that in an
       // e10s setup different processes don't clobber each other's dump files.
-      string.AppendInt(getpid());
+      string.AppendInt((int)getpid());
       for (int paintCount : *gPaintCountStack) {
         string.AppendLiteral("-");
         string.AppendInt(paintCount);
