https://bugzilla.mozilla.org/show_bug.cgi?id=1391072

diff -r 883bf2620473 -r 7ab6305b7f40 build/autoconf/config.guess
--- a/build/autoconf/config.guess	Thu Jun 20 09:36:15 2019 +0200
+++ b/build/autoconf/config.guess	Tue Mar 26 00:55:36 2019 -0700
@@ -385,7 +385,7 @@
 	echo sparc-hal-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
     sun4*:SunOS:5.*:* | tadpole*:SunOS:5.*:*)
-	echo sparc-sun-solaris2"`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`"
+	echo sparc64-sun-solaris2"`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`"
 	exit ;;
     i86pc:AuroraUX:5.*:* | i86xen:AuroraUX:5.*:*)
 	echo i386-pc-auroraux"$UNAME_RELEASE"
