diff --git a/dom/media/systemservices/CamerasTypes.h b/dom/media/systemservices/CamerasTypes.h
--- a/dom/media/systemservices/CamerasTypes.h
+++ b/dom/media/systemservices/CamerasTypes.h
@@ -50,11 +50,11 @@
           mozilla::camera::CaptureEngine::InvalidEngine,
           mozilla::camera::CaptureEngine::MaxEngine> {};
 
 template <>
 struct ParamTraits<mozilla::camera::CamerasAccessStatus>
-    : public ContiguousEnumSerializer<
+    : public ContiguousEnumSerializerInclusive<
           mozilla::camera::CamerasAccessStatus,
           mozilla::camera::CamerasAccessStatus::Granted,
           mozilla::camera::CamerasAccessStatus::Error> {};
 }  // namespace IPC

