diff --git a/security/nss/lib/freebl/mpi/montmulf.c b/security/nss/lib/freebl/mpi/montmulf.c
index ce8fbc3..6d067d8 100644
--- a/security/nss/lib/freebl/mpi/montmulf.c
+++ b/security/nss/lib/freebl/mpi/montmulf.c
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #ifdef SOLARIS
-#define RF_INLINE_MACROS 1
+#undef RF_INLINE_MACROS
 #endif
 
 static const double TwoTo16 = 65536.0;
