--- fbreader-0.99.4/makefiles/arch/desktop.mk	2019-02-09 14:08:39.386527296 +0000
+++ fbreader-0.99.4/makefiles/arch/desktop.mk.new	2019-02-09 14:13:11.261765793 +0000
@@ -16,7 +16,7 @@ CFLAGS = -pipe -fno-exceptions -Wall -Wn
 LDFLAGS =
 EXTERNAL_INCLUDE = $(shell pkg-config --cflags fribidi)
 
-MOC = /usr/bin/moc-qt5
+MOC = /usr/lib/qt/5.15/bin/amd64/moc
 QTINCLUDE = $(shell pkg-config --cflags Qt5Gui Qt5Widgets Qt5Network)
 UILIBS = $(shell pkg-config --libs Qt5Gui Qt5Widgets Qt5Network)
 
