--- fbreader-0.99.4/makefiles/arch/desktop.mk.orig	2020-05-23 23:25:39.963338761 +0000
+++ fbreader-0.99.4/makefiles/arch/desktop.mk	2020-05-23 23:26:05.782706129 +0000
@@ -12,8 +12,8 @@
 AR = ar rsu
 LD = g++
 
-CFLAGS = -pipe -fno-exceptions -Wall -Wno-ctor-dtor-privacy -W -DLIBICONV_PLUG -fPIC
-LDFLAGS =
+CFLAGS += -pipe -fno-exceptions -Wall -Wno-ctor-dtor-privacy -W -DLIBICONV_PLUG -fPIC
+LDFLAGS+= 
 EXTERNAL_INCLUDE = $(shell pkg-config --cflags fribidi)
 
 MOC = /usr/lib/qt/5.15/bin/$(MACH64)/moc
