diff -ruN udpxy-1.0.23-9.orig/Makefile udpxy-1.0.23-9/Makefile
--- udpxy-1.0.23-9.orig/Makefile	2013-05-18 09:02:55.000000000 -0700
+++ udpxy-1.0.23-9/Makefile	2016-11-06 08:48:13.622135839 -0800
@@ -182,11 +182,11 @@
 	@cp $(EXEC) $(INSTALLROOT)/bin
 	@ls -l $(INSTALLROOT)/bin/$(EXEC)
 	@mkdir -p -m 755 $(MANPAGE_DIR)
-	@$(GZIP) -c $(UDPXY_MANPAGE_EN) > $(MANPAGE_DIR)/$(UDPXY_MAN1).gz
+	@cp $(UDPXY_MANPAGE_EN) $(MANPAGE_DIR)/$(UDPXY_MAN1)
 ifneq (yes, $(NO_UDPXREC))
 	@cp $(UDPXREC) $(INSTALLROOT)/bin/$(UDPXREC)
 	@ls -l $(INSTALLROOT)/bin/$(UDPXREC)
-	@$(GZIP) -c $(UDPXREC_MANPAGE_EN) > $(MANPAGE_DIR)/$(UDPXREC_MAN1).gz
+	@cp $(UDPXREC_MANPAGE_EN) $(MANPAGE_DIR)/$(UDPXREC_MAN1)
 endif
 	@echo "Installation of udpxy is complete"
 
