--- qt-everywhere-opensource-src-4.8.6/mkspecs/solaris-g++-64/qmake.conf.~1~	2014-04-10 22:37:12.000000000 +0400
+++ qt-everywhere-opensource-src-4.8.6/mkspecs/solaris-g++-64/qmake.conf	2015-01-17 22:12:42.284874397 +0300
@@ -36,7 +36,7 @@
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -m64 -D_XOPEN_SOURCE=500 -D__EXTENSIONS__
+QMAKE_CFLAGS		= -m64
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
@@ -59,14 +59,14 @@
 QMAKE_CXXFLAGS_YACC	= $$QMAKE_CFLAGS_YACC
 QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_THREAD
 
-QMAKE_INCDIR		= /usr/sfw/include
-QMAKE_LIBDIR		= /usr/sfw/lib/64
-QMAKE_INCDIR_X11	= /usr/openwin/include
-QMAKE_LIBDIR_X11	= /usr/openwin/lib/64
+QMAKE_INCDIR		= /usr/include
+QMAKE_LIBDIR		= /usr/lib/64
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib/64
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/openwin/include
-QMAKE_LIBDIR_OPENGL	= /usr/openwin/lib/64
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib/64
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
@@ -82,7 +82,7 @@
 QMAKE_LFLAGS_NOUNDEF	= -z defs
 QMAKE_LFLAGS_RPATH	= -Wl,-R,
 
-QMAKE_LIBS		=
+QMAKE_LIBS		= -m64
 QMAKE_LIBS_DYNLOAD	= -ldl
 QMAKE_LIBS_X11		= -lXext -lX11 -lresolv -lsocket -lnsl
 QMAKE_LIBS_X11SM	= -lSM -lICE
