--- trousers-0.3.11.2/src/include/trousers_types.h.~1~	2014-12-01 18:09:17.447737290 +0300
+++ trousers-0.3.11.2/src/include/trousers_types.h	2014-12-01 18:10:11.091939561 +0300
@@ -118,9 +118,9 @@
 	BYTE *encData;
 } TSS_KEY;
 
-#if (defined (__linux) || defined (linux) || defined (SOLARIS) || defined (__GLIBC__))
+#if (defined (__linux) || defined (linux) || defined (__GLIBC__))
 #define BSD_CONST
-#elif (defined (__OpenBSD__) || defined (__FreeBSD__))
+#elif (defined (__OpenBSD__) || defined (__FreeBSD__) || defined (__sun))
 #define BSD_CONST const
 #endif
 
