--- src/include/tss/tss_defines.h.old	Tue May 31 07:39:12 2011
+++ src/include/tss/tss_defines.h	Tue May 31 07:39:52 2011
@@ -1214,6 +1214,9 @@
 #define TSS_UUID_USK5 {0, 0, 0, 0, 0, {0, 0, 0, 0, 0, 9}} // user storage key 5
 #define TSS_UUID_USK6 {0, 0, 0, 0, 0, {0, 0, 0, 0, 0, 10}}// user storage key 6
 
+/* SOLARIS: Migratable Root Key UUID */
+#define TSS_UUID_MRK {0, 0, 0, 0, 0, {0, 0, 0, 0, 0, 11}}
+
 // macro to derive UUIDs for keys whose "OwnerEvict" key is set.
 #define TSS_UUID_OWNEREVICT(i) {0, 0, 0, 0, 0, {0, 0, 0, 0, 1, (i)}}
 
