--- foomatic-db-engine-4.0.13/foomatic-kitload.in.orig
+++ foomatic-db-engine-4.0.13/foomatic-kitload.in
@@ -54,7 +54,7 @@
 $ENV{'LANG'} = "C";
 
 # Copy the files
-my $retval = system("( cd $dir ; tar cf - --exclude CVS . ) | tar xv${dontoverwrite}Cf $destdir$libdir/db/source - 2> $destdir$libdir/kitload2.log | tee $destdir$libdir/kitload.log");
+my $retval = system("( cd $dir ; /usr/bin/gtar cf - --exclude CVS . ) | /usr/bin/gtar xv${dontoverwrite}Cf $destdir$libdir/db/source - 2> $destdir$libdir/kitload2.log | tee $destdir$libdir/kitload.log");
 #$retval = $retval / 256;
 #if ($retval) {
 #    die "Error copying files to $destdir$libdir/db/source: $?";
