*** a/lib/Foomatic/DB.pm.orig	Mon Oct 27 15:51:02 2008
--- b/lib/Foomatic/DB.pm	Mon Oct 27 15:50:37 2008
***************
*** 102,108 ****
      }
  
      # Build a new overview
!     my $otype = ($cupsppds ? '-C' : '-O');
      $otype .= ' -n' if ($cupsppds == 1);
      # Generate overview Perl data structure from database
      my $VAR1;
--- 102,108 ----
      }
  
      # Build a new overview
!     my $otype = ($cupsppds ? '-C' : '-C');
      $otype .= ' -n' if ($cupsppds == 1);
      # Generate overview Perl data structure from database
      my $VAR1;
