--- foomatic-db-engine-3.0-20080903/Makefile.in.orig	2022-08-09 09:53:42.484623215 +0000
+++ foomatic-db-engine-3.0-20080903/Makefile.in	2022-08-09 09:53:53.002643494 +0000
@@ -188,7 +188,7 @@
 #	$(DESTDIR) properly
 	( cd lib && \
 	  $(MAKE) PREFIX=$(PERLPREFIX) && \
-	  $(MAKE) PREFIX=$(DESTDIR)$(PERLPREFIX) pure_install \
+	  $(MAKE) PREFIX=$(DESTDIR)$(PERLPREFIX) install_vendor \
 	)
 	${SRC}/mkinstalldirs $(DESTDIR)$(BINDIR)
 	${SRC}/mkinstalldirs $(DESTDIR)$(SBINDIR)
