https://github.com/dabeaz/curio/issues/344

--- curio-1.6/tests/test_kernel.py.orig
+++ curio-1.6/tests/test_kernel.py
@@ -718,6 +718,7 @@
     kernel.run(main)
     assert results == [ 'child', 'cancelled', 'done cancel' ]
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_reentrant_kernel(kernel):
     async def child():
         pass
--- curio-1.6/tests/test_io.py.orig
+++ curio-1.6/tests/test_io.py
@@ -838,6 +838,7 @@
     assert results['handler'] == results['sender']
 
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_stream_bad_context(kernel, portno):
     done = Event()
     results = []
@@ -870,6 +871,7 @@
 
     assert results == [ True ]
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_stream_bad_iter(kernel, portno):
     done = Event()
     results = []
--- curio-1.6/tests/test_meta.py.orig
+++ curio-1.6/tests/test_meta.py
@@ -11,6 +11,7 @@
 
     assert meta.iscoroutinefunction(partial(spam, 1))
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_instantiate_coroutine():
     async def coro(x, y):
         pass
--- curio-1.6/tests/test_file.py.orig
+++ curio-1.6/tests/test_file.py
@@ -213,6 +213,7 @@
     kernel.run(main())
 
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_sync_with(kernel):
     async def main():
         f = aopen(testinput, 'r')
--- curio-1.6/tests/test_thread.py.orig
+++ curio-1.6/tests/test_thread.py
@@ -254,6 +254,7 @@
         assert t.result == 5
     kernel.run(main)
 
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_errors(kernel):
     # spawn_thread used on a coroutine
     async def main():
--- curio-1.6/setup.cfg.orig
+++ curio-1.6/setup.cfg
@@ -10,6 +10,8 @@
 	--ignore=setup.py --ignore=docs/conf.py
 markers = 
 	internet: mark tests as requiring internet connectivity (deselect with '-m "not internet"')
+filterwarnings =
+	ignore::pytest.PytestUnraisableExceptionWarning
 
 [egg_info]
 tag_build = 
