--- greenlet-3.1.0/src/greenlet/tests/test_leaks.py.orig
+++ greenlet-3.1.0/src/greenlet/tests/test_leaks.py
@@ -18,6 +18,8 @@
 from .leakcheck import ignores_leakcheck
 from .leakcheck import RUNNING_ON_MANYLINUX
 
+import unittest
+
 # pylint:disable=protected-access
 
 assert greenlet.GREENLET_USE_GC # Option to disable this was removed in 1.0
@@ -430,6 +432,7 @@
     @ignores_leakcheck
     # Because we're just trying to track raw memory, not objects, and running
     # the leakcheck makes an already slow test slower.
+    @unittest.skip("Always fails: https://github.com/python-greenlet/greenlet/issues/379")
     def test_untracked_memory_doesnt_increase_unfinished_thread_dealloc_in_thread(self):
         self._check_untracked_memory_thread(deallocate_in_thread=True)
 
