--- opensmalltalk-vm-sun-v5.0.66/platforms/Cross/vm/sqSCCSVersion.h	Sun Sep  8 15:04:33 2024
+++ p0/opensmalltalk-vm-sun-v5.0.66/platforms/Cross/vm/sqSCCSVersion.h	Sat Sep 14 11:00:23 2024
@@ -30,13 +30,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 202409081304 $";
 # define REV_START (SvnRawRevisionString + 6)
 
-static char SvnRawRevisionDate[] = "$Date$";
+static char SvnRawRevisionDate[] = "$Date: Sun Sep 8 15:04:33 2024 +0200 $";
 # define DATE_START (SvnRawRevisionDate + 7)
 
-static char SvnRawRepositoryURL[] = "$URL$";
+static char SvnRawRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (SvnRawRepositoryURL + 6)
 
 static char *
@@ -73,17 +73,17 @@
 # undef URL_START
 #elif GIT
 # define PREFIX ""
-static char GitRawRevisionString[] = "$Rev$";
+static char GitRawRevisionString[] = "$Rev: 202409081304 $";
 # define REV_START (GitRawRevisionString + 6)
 # define REV_TIME_START (GitRawRevisionString + 14)
 
-static char GitRawRevisionDate[] = "$Date$";
+static char GitRawRevisionDate[] = "$Date: Sun Sep 8 15:04:33 2024 +0200 $";
 # define DATE_START (GitRawRevisionDate + 7)
 
-static char GitRawRepositoryURL[] = "$URL$";
+static char GitRawRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (GitRawRepositoryURL + 6)
 
-static char GitRawRevisionShortHash[] = "$CommitHash$";
+static char GitRawRevisionShortHash[] = "$CommitHash: b7904564e $";
 # define SHORTHASH_START (GitRawRevisionShortHash + 13)
 
 static char *
