--- opensmalltalk-vm-sun-v5.0.32/platforms/Cross/plugins/sqPluginsSCCSVersion.h	2021-07-30 19:43:30.000000000 +0200
+++ p0/opensmalltalk-vm-sun-v5.0.32/platforms/Cross/plugins/sqPluginsSCCSVersion.h	2021-07-31 11:42:12.036247223 +0200
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202107301743 $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@mix:src/opensmalltalk-vm $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202107301743 $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@mix:src/opensmalltalk-vm $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
