--- opensmalltalk-vm-202603271636/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Fri Mar 27 17:36:22 2026
+++ p0/opensmalltalk-vm-202603271636/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Wed Apr 15 09:55:19 2026
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202603271636 $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202603271636 $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
