--- opensmalltalk-vm-sun-v5.0.52/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Thu Jul 21 19:35:46 2022
+++ p0/opensmalltalk-vm-sun-v5.0.52/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Fri Jul 22 09:44:19 2022
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202207211735 $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk-vm $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202207211735 $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk-vm $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
