--- opensmalltalk-vm-sun-v5.0.62/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sun Feb 25 21:18:14 2024
+++ p0/opensmalltalk-vm-sun-v5.0.62/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Fri Mar  1 18:48:00 2024
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202402252018 $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202402252018 $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
