--- opensmalltalk-vm-sun-v5.0.66/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sun Sep  8 15:04:33 2024
+++ p0/opensmalltalk-vm-sun-v5.0.66/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sat Sep 14 11:00:23 2024
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202409081304 $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202409081304 $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
