--- opensmalltalk-vm-sun-v5.0.45/src/spur64.stack/gcc3x-interp.c	Sun Feb  6 17:52:18 2022
+++ p0/opensmalltalk-vm-sun-v5.0.45/src/spur64.stack/gcc3x-interp.c	Fri Feb 11 18:36:48 2022
@@ -69381,7 +69381,7 @@
 		oop = GIV(nilObj);
 		longAtput((sp = GIV(stackPointer) + ((0) * BytesPerWord)), oop);
 		GIV(stackPointer) = sp;
-		return transferTo(wakeHighestPriority());
+		transferTo(wakeHighestPriority());
 	}
 	/* begin fetchPointer:ofObject: */
 	myList = longAt((process + BaseHeaderSize) + (((sqInt)((usqInt)(MyListIndex) << (shiftForWord())))));
