--- libtorrent-0.13.3/src/net/socket_fd.cc	2012-01-19 14:19:26.000000000 +0400
+++ libtorrent-0.13.3/src/net/socket_fd.cc	2013-10-05 00:58:51.031864245 +0400
@@ -127,6 +127,8 @@
 
 void
 SocketFd::close() {
+  if (::shutdown(m_fd,SHUT_RDWR) && errno == EBADF)
+    throw internal_error("SocketFd::close() called on an invalid file descriptor (shutdown)");
   if (::close(m_fd) && errno == EBADF)
     throw internal_error("SocketFd::close() called on an invalid file descriptor");
 }
