--- llvm-9.0.0.src/cmake/modules/AddLLVM.cmake.~1~	2019-11-05 18:34:04.588174189 +0000
+++ llvm-9.0.0.src/cmake/modules/AddLLVM.cmake	2019-11-05 18:37:01.191226044 +0000
@@ -224,7 +224,7 @@
                      LINK_FLAGS " -Wl,-dead_strip")
       elseif(${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
         set_property(TARGET ${target_name} APPEND_STRING PROPERTY
-                     LINK_FLAGS " -Wl,-z -Wl,discard-unused=sections")
+		     LINK_FLAGS " -Wl,-z -Wl,ignore")
       elseif(NOT WIN32 AND NOT LLVM_LINKER_IS_GOLD AND
              NOT ${CMAKE_SYSTEM_NAME} MATCHES "OpenBSD|AIX")
         # Object files are compiled with -ffunction-data-sections.
