diff -wpruN '--exclude=*.orig' a~/tools/lto/lto.exports a/tools/lto/lto.exports
--- a~/tools/lto/lto.exports	1970-01-01 00:00:00
+++ a/tools/lto/lto.exports	1970-01-01 00:00:00
@@ -41,12 +41,6 @@ lto_codegen_optimize
 lto_codegen_compile_optimized
 lto_codegen_set_should_internalize
 lto_codegen_set_should_embed_uselists
-LLVMCreateDisasm
-LLVMCreateDisasmCPU
-LLVMDisasmDispose
-LLVMDisasmInstruction
-LLVMSetDisasmOptions
-LLVMCreateDisasmCPUFeatures
 thinlto_create_codegen
 thinlto_codegen_dispose
 thinlto_codegen_add_module
diff -wpruN '--exclude=*.orig' a~/tools/remarks-shlib/Remarks.exports a/tools/remarks-shlib/Remarks.exports
--- a~/tools/remarks-shlib/Remarks.exports	1970-01-01 00:00:00
+++ a/tools/remarks-shlib/Remarks.exports	1970-01-01 00:00:00
@@ -1,24 +1,0 @@
-LLVMRemarkStringGetData
-LLVMRemarkStringGetLen
-LLVMRemarkDebugLocGetSourceFilePath
-LLVMRemarkDebugLocGetSourceLine
-LLVMRemarkDebugLocGetSourceColumn
-LLVMRemarkArgGetKey
-LLVMRemarkArgGetValue
-LLVMRemarkArgGetDebugLoc
-LLVMRemarkEntryDispose
-LLVMRemarkEntryGetType
-LLVMRemarkEntryGetPassName
-LLVMRemarkEntryGetRemarkName
-LLVMRemarkEntryGetFunctionName
-LLVMRemarkEntryGetDebugLoc
-LLVMRemarkEntryGetHotness
-LLVMRemarkEntryGetNumArgs
-LLVMRemarkEntryGetFirstArg
-LLVMRemarkEntryGetNextArg
-LLVMRemarkParserCreateYAML
-LLVMRemarkParserGetNext
-LLVMRemarkParserHasError
-LLVMRemarkParserGetErrorMessage
-LLVMRemarkParserDispose
-LLVMRemarkVersion
