To avoid dependency on encumbered gst-plugins-bad.

--- gst-python-1.26.0/meson.build.orig
+++ gst-python-1.26.0/meson.build
@@ -20,8 +20,7 @@
   fallback : ['gstreamer', 'gst_dep'])
 gstbase_dep = dependency('gstreamer-base-1.0', version : gst_req,
   fallback : ['gstreamer', 'gst_base_dep'])
-gstbad_dep = dependency('gstreamer-analytics-1.0', version :gst_req,
-  fallback : [])
+gstbad_dep = []
 gmodule_dep = dependency('gmodule-no-export-2.0')
 libdl = cc.find_library('dl', required: false)
 pygobject_dep = dependency('pygobject-3.0', fallback: ['pygobject', 'pygobject_dep'], version : '>= 3.8')
--- gst-python-1.26.0/gi/overrides/meson.build.orig
+++ gst-python-1.26.0/gi/overrides/meson.build
@@ -1,4 +1,4 @@
-pysources = ['Gst.py', 'GstPbutils.py', 'GstVideo.py', 'GstAudio.py','GstAnalytics.py']
+pysources = ['Gst.py', 'GstPbutils.py', 'GstVideo.py', 'GstAudio.py']
 python.install_sources(pysources,
   pure : false,
   subdir: 'gi/overrides',
@@ -9,7 +9,6 @@
   gst_dep_for_gi = gst_dep
 else
   gst_dep_for_gi = gst_dep.partial_dependency(compile_args: true, includes: true, sources: true)
-  gstanalytics_dep_for_gi = gstbad_dep.partial_dependency(compile_args:true, includes:true, sources:true)
 endif
 
 gstpython = python.extension_module('_gi_gst',
@@ -21,15 +20,6 @@
     dependencies : [gst_dep_for_gi, python_dep, pygobject_dep],
 )
 
-gstanalyticspython = python.extension_module ('_gi_gst_analytics',
-  sources: ['gstanalyticsmodule.c'],
-  install: true,
-  install_dir: pygi_override_dir,
-  install_tag: 'python-runtime',
-  include_directories : [configinc],
-  dependencies : [gst_dep_for_gi, python_dep, pygobject_dep, gstbad_dep],
-)
-
 env = environment()
 env.prepend('_GI_OVERRIDES_PATH', [
     meson.current_source_dir(),
--- gst-python-1.26.0/testsuite/meson.build.orig
+++ gst-python-1.26.0/testsuite/meson.build
@@ -4,7 +4,6 @@
     ['Test gst', 'test_gst.py'],
     ['Test fundamentals', 'test_types.py'],
     ['Test plugins', 'test_plugin.py'],
-    ['Test analytics', 'test_analytics.py', ['gst-plugins-bad/gst-libs/gst/analytics', 'gst-plugins-base/gst-libs/gst/video']],
 ]
 
 runcmd = run_command(python, '-c', '''with open("@0@/mesonconfig.py", "w") as f:
