--- gst-python-0.10.16/testsuite/runtests.py-orig	2009-08-14 18:54:17.361975000 -0500
+++ gst-python-0.10.16/testsuite/runtests.py	2009-08-14 18:54:24.633404000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 #
--- gst-python-0.10.16/codegen/scmexpr.py-orig	2009-08-14 18:54:35.145412000 -0500
+++ gst-python-0.10.16/codegen/scmexpr.py	2009-08-14 18:54:58.682431000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python; py-indent-offset: 4 -*-
 from __future__ import generators
 
--- gst-python-0.10.16/codegen/docgen.py-orig	2009-08-14 18:54:38.837983000 -0500
+++ gst-python-0.10.16/codegen/docgen.py	2009-08-14 18:55:07.958349000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python; py-indent-offset: 4 -*-
 import sys, os, string, re, getopt
 
--- gst-python-0.10.16/codegen/mkskel.py-orig	2009-08-14 18:54:43.915652000 -0500
+++ gst-python-0.10.16/codegen/mkskel.py	2009-08-14 18:55:14.481337000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python; py-indent-offset: 4 -*-
 
 import sys, os, getopt
--- gst-python-0.10.16/codegen/h2def.py-orig	2009-08-14 18:54:48.301740000 -0500
+++ gst-python-0.10.16/codegen/h2def.py	2009-08-14 18:55:23.015906000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python; py-indent-offset: 4 -*-
 # Search through a header file looking for function prototypes.
 # For each prototype, generate a scheme style definition.
--- gst-python-0.10.16/codegen/mergedefs.py-orig	2009-08-14 18:54:52.698798000 -0500
+++ gst-python-0.10.16/codegen/mergedefs.py	2009-08-14 18:55:30.730327000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python; py-indent-offset: 4 -*-
 
 import optparse
--- gst-python-0.10.16/examples/f2f.py-orig	2009-08-14 18:55:46.938466000 -0500
+++ gst-python-0.10.16/examples/f2f.py	2009-08-14 18:55:52.565455000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/play.py-orig	2009-08-14 18:56:55.853160000 -0500
+++ gst-python-0.10.16/examples/play.py	2009-08-14 18:57:00.227219000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/gst-discover-orig	2009-08-14 18:55:57.476573000 -0500
+++ gst-python-0.10.16/examples/gst-discover	2009-08-14 18:56:13.472288000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # gst-python
 # Copyright (C) 2006 Andy Wingo <wingo at pobox.com>
 #
--- gst-python-0.10.16/examples/cp.py-orig	2009-08-14 18:57:18.748646000 -0500
+++ gst-python-0.10.16/examples/cp.py	2009-08-14 18:57:23.101451000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/decodebin.py-orig	2009-08-14 18:56:22.530344000 -0500
+++ gst-python-0.10.16/examples/decodebin.py	2009-08-14 18:56:38.875853000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # decodebin.py - Audio autopluging example using 'decodebin' element
 # Copyright (C) 2006 Jason Gerard DeRose <jderose@jasonderose.org>
--- gst-python-0.10.16/examples/pipeline-tester-orig	2009-08-14 18:56:46.129670000 -0500
+++ gst-python-0.10.16/examples/pipeline-tester	2009-08-14 18:56:52.131786000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # gst-python
 # Copyright (C) 2005 Andy Wingo <wingo@pobox.com>
--- gst-python-0.10.16/examples/remuxer.py-orig	2009-08-14 18:57:09.854093000 -0500
+++ gst-python-0.10.16/examples/remuxer.py	2009-08-14 18:57:15.245427000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/bps.py-orig	2009-08-14 18:57:26.488755000 -0500
+++ gst-python-0.10.16/examples/bps.py	2009-08-14 18:57:34.401019000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/sinkelement.py-orig	2009-08-14 18:57:41.788694000 -0500
+++ gst-python-0.10.16/examples/sinkelement.py	2009-08-14 18:57:47.140585000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/gstfile.py-orig	2009-08-14 18:57:51.522131000 -0500
+++ gst-python-0.10.16/examples/gstfile.py	2009-08-14 18:57:56.859536000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/vumeter.py-orig	2009-08-14 18:58:00.761309000 -0500
+++ gst-python-0.10.16/examples/vumeter.py	2009-08-14 18:58:05.739350000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/audioconcat.py-orig	2009-08-14 18:58:12.386756000 -0500
+++ gst-python-0.10.16/examples/audioconcat.py	2009-08-14 18:58:19.274890000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/filesrc.py-orig	2009-08-14 18:58:23.506563000 -0500
+++ gst-python-0.10.16/examples/filesrc.py	2009-08-14 18:58:28.953294000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/debugslider.py-orig	2009-08-14 18:58:35.907476000 -0500
+++ gst-python-0.10.16/examples/debugslider.py	2009-08-14 18:58:41.095983000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/examples/audio-controller.py-orig	2009-08-14 18:58:49.714272000 -0500
+++ gst-python-0.10.16/examples/audio-controller.py	2009-08-14 18:59:01.872025000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 
--- gst-python-0.10.16/gst/extend/sources.py-orig	2009-08-14 18:59:13.213305000 -0500
+++ gst-python-0.10.16/gst/extend/sources.py	2009-08-14 18:59:17.737450000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 #
--- gst-python-0.10.16/gst/extend/__init__.py-orig	2009-08-14 18:59:27.059519000 -0500
+++ gst-python-0.10.16/gst/extend/__init__.py	2009-08-14 18:59:32.891686000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 #
--- gst-python-0.10.16/gst/extend/jukebox.py-orig	2009-08-14 18:59:37.457768000 -0500
+++ gst-python-0.10.16/gst/extend/jukebox.py	2009-08-14 18:59:43.027344000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 #
--- gst-python-0.10.16/gst/extend/pygobject.py-orig	2009-08-14 18:59:49.619610000 -0500
+++ gst-python-0.10.16/gst/extend/pygobject.py	2009-08-14 18:59:54.174125000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- Mode: Python -*-
 # vi:si:et:sw=4:sts=4:ts=4
 #
 
--- gst-python-0.10.21/plugin/Makefile.am-orig	2011-03-03 03:48:59.936450614 -0600
+++ gst-python-0.10.21/plugin/Makefile.am	2011-03-03 03:49:11.555970439 -0600
@@ -8,4 +8,4 @@ INCLUDES = $(PYGOBJECT_CFLAGS) $(GST_CFL
 
 libgstpython_la_SOURCES = gstpythonplugin.c
 libgstpython_la_LDFLAGS = -module -avoid-version
-libgstpython_la_LIBADD = $(GST_LIBS)
+libgstpython_la_LIBADD = $(GST_LIBS) -lpython2.6
