--- nss-3.43/nss/lib/freebl/verified/kremlib.h	2019-03-16 01:25:08.000000000 +0000
+++ nss-3.43/nss/lib/freebl/verified/kremlib.h.new	2019-04-13 13:46:57.808230329 +0000
@@ -182,7 +182,7 @@ typedef const char *Prims_string;
 /******************************************************************************/
 
 /* ... for Linux */
-#if defined(__linux__) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__CYGWIN__) || defined(__sun__)
 #include <endian.h>
 
 /* ... for OSX */
