--- nss-3.12.6/nss/coreconf/SunOS5.mk.orig	2012-02-02 14:46:03.520995829 +0000
+++ nss-3.12.6/nss/coreconf/SunOS5.mk	2012-02-02 14:46:43.710652404 +0000
@@ -15,7 +15,7 @@
       ARCHFLAG=-m64
   else
       ifeq ($(OS_TEST),i86pc)
-        ARCHFLAG=-xarch=amd64
+        ARCHFLAG=-m64
       else
         ARCHFLAG=-xarch=v9
       endif
@@ -106,7 +106,7 @@
 else
 ifeq ($(USE_64), 1)
 	ifeq ($(OS_TEST),i86pc)
-	    DSO_LDOPTS +=-xarch=amd64
+	    DSO_LDOPTS +=-m64
 	else
 	    DSO_LDOPTS +=-xarch=v9
 	endif
