diff -ur nss-3.12.6.orig/nss/tests/pkcs11/netscape/trivial/configure nss-3.12.6/nss/tests/pkcs11/netscape/trivial/configure
--- nss-3.12.6.orig/nss/tests/pkcs11/netscape/trivial/configure	2000-05-09 00:19:44.000000000 +0100
+++ nss-3.12.6/nss/tests/pkcs11/netscape/trivial/configure	2012-02-02 13:56:33.870847158 +0000
@@ -583,7 +583,7 @@
 
 
 NSPR_CFLAGS="-I$NSPR_INCLUDE"
-NSPR_LDFLAGS="-L$NSPR_LIBPATH -lnspr4 -lplc4 -lplds4"
+NSPR_LDFLAGS="-L$NSPR_LIBPATH -R$NSPR_LIBPATH -lnspr4 -lplc4 -lplds4"
 
 
 # Check whether --with-nss-dist or --without-nss-dist was given.
diff -ur nss-3.12.6.orig/nss/tests/pkcs11/netscape/trivial/configure.in nss-3.12.6/nss/tests/pkcs11/netscape/trivial/configure.in
--- nss-3.12.6.orig/nss/tests/pkcs11/netscape/trivial/configure.in	2006-03-02 22:48:55.000000000 +0000
+++ nss-3.12.6/nss/tests/pkcs11/netscape/trivial/configure.in	2012-02-02 13:57:06.895561687 +0000
@@ -28,7 +28,7 @@
 )
 
 NSPR_CFLAGS="-I$NSPR_INCLUDE"
-NSPR_LDFLAGS="-L$NSPR_LIBPATH -lnspr4 -lplc4 -lplds4"
+NSPR_LDFLAGS="-L$NSPR_LIBPATH -R$NSPR_LIBPATH -lnspr4 -lplc4 -lplds4"
 
 dnl This isn't optimum, but the mozilla build system and autoconf don't really mesh well..
 
