diff -ur nss-3.12.6.orig/nss/cmd/pk11mode/Makefile nss-3.12.6/nss/cmd/pk11mode/Makefile
--- nss-3.12.6.orig/nss/cmd/pk11mode/Makefile	2009-03-31 00:40:52.000000000 +0100
+++ nss-3.12.6/nss/cmd/pk11mode/Makefile	2012-02-02 14:38:31.494513429 +0000
@@ -38,6 +38,7 @@
 
 EXTRA_SHARED_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/cmd/platlibs.mk nss-3.12.6/nss/cmd/platlibs.mk
--- nss-3.12.6.orig/nss/cmd/platlibs.mk	2010-02-04 18:59:10.000000000 +0000
+++ nss-3.12.6/nss/cmd/platlibs.mk	2012-02-02 14:36:02.417536468 +0000
@@ -127,6 +127,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
@@ -179,6 +179,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/cmd/shlibsign/Makefile nss-3.12.6/nss/cmd/shlibsign/Makefile
--- nss-3.12.6.orig/nss/cmd/shlibsign/Makefile	2009-08-07 20:06:37.000000000 +0100
+++ nss-3.12.6/nss/cmd/shlibsign/Makefile	2012-02-02 14:38:02.243374759 +0000
@@ -36,6 +36,7 @@
 
 EXTRA_SHARED_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/cmd/shlibsign/mangle/Makefile nss-3.12.6/nss/cmd/shlibsign/mangle/Makefile
--- nss-3.12.6.orig/nss/cmd/shlibsign/mangle/Makefile	2008-11-20 15:44:24.000000000 +0000
+++ nss-3.12.6/nss/cmd/shlibsign/mangle/Makefile	2012-02-02 14:36:31.841732437 +0000
@@ -36,6 +36,7 @@
 
 EXTRA_SHARED_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/ckfw/builtins/Makefile nss-3.12.6/nss/lib/ckfw/builtins/Makefile
--- nss-3.12.6.orig/nss/lib/ckfw/builtins/Makefile	2009-12-17 22:00:47.000000000 +0000
+++ nss-3.12.6/nss/lib/ckfw/builtins/Makefile	2012-02-02 14:28:57.559752851 +0000
@@ -34,6 +34,7 @@
 
 EXTRA_SHARED_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/ckfw/capi/Makefile nss-3.12.6/nss/lib/ckfw/capi/Makefile
--- nss-3.12.6.orig/nss/lib/ckfw/capi/Makefile	2009-07-29 21:15:19.000000000 +0100
+++ nss-3.12.6/nss/lib/ckfw/capi/Makefile	2012-02-02 14:27:59.590451326 +0000
@@ -40,6 +40,7 @@
 
 EXTRA_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/ckfw/nssmkey/Makefile nss-3.12.6/nss/lib/ckfw/nssmkey/Makefile
--- nss-3.12.6.orig/nss/lib/ckfw/nssmkey/Makefile	2005-11-23 23:04:08.000000000 +0000
+++ nss-3.12.6/nss/lib/ckfw/nssmkey/Makefile	2012-02-02 14:28:33.862666008 +0000
@@ -35,6 +35,7 @@
 
 EXTRA_LIBS += \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/freebl/config.mk nss-3.12.6/nss/lib/freebl/config.mk
--- nss-3.12.6.orig/nss/lib/freebl/config.mk	2009-04-11 22:18:42.000000000 +0100
+++ nss-3.12.6/nss/lib/freebl/config.mk	2012-02-02 14:33:24.476440608 +0000
@@ -85,6 +85,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lnspr4 \
 	$(NULL)
 endif
diff -ur nss-3.12.6.orig/nss/lib/nss/config.mk nss-3.12.6/nss/lib/nss/config.mk
--- nss-3.12.6.orig/nss/lib/nss/config.mk	2009-12-15 22:22:31.000000000 +0000
+++ nss-3.12.6/nss/lib/nss/config.mk	2012-02-02 14:32:23.596086591 +0000
@@ -41,6 +41,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/smime/config.mk nss-3.12.6/nss/lib/smime/config.mk
--- nss-3.12.6.orig/nss/lib/smime/config.mk	2009-06-11 01:55:47.000000000 +0100
+++ nss-3.12.6/nss/lib/smime/config.mk	2012-02-02 14:32:04.860999052 +0000
@@ -43,6 +43,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/softoken/config.mk nss-3.12.6/nss/lib/softoken/config.mk
--- nss-3.12.6.orig/nss/lib/softoken/config.mk	2009-06-11 01:55:48.000000000 +0100
+++ nss-3.12.6/nss/lib/softoken/config.mk	2012-02-02 14:32:52.701287385 +0000
@@ -52,6 +52,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/softoken/legacydb/config.mk nss-3.12.6/nss/lib/softoken/legacydb/config.mk
--- nss-3.12.6.orig/nss/lib/softoken/legacydb/config.mk	2009-06-11 01:55:49.000000000 +0100
+++ nss-3.12.6/nss/lib/softoken/legacydb/config.mk	2012-02-02 14:33:09.164394761 +0000
@@ -50,6 +50,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/ssl/config.mk nss-3.12.6/nss/lib/ssl/config.mk
--- nss-3.12.6.orig/nss/lib/ssl/config.mk	2010-02-04 19:09:08.000000000 +0000
+++ nss-3.12.6/nss/lib/ssl/config.mk	2012-02-02 14:33:35.117601311 +0000
@@ -60,6 +60,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/sysinit/config.mk nss-3.12.6/nss/lib/sysinit/config.mk
--- nss-3.12.6.orig/nss/lib/sysinit/config.mk	2010-02-16 19:38:42.000000000 +0000
+++ nss-3.12.6/nss/lib/sysinit/config.mk	2012-02-02 14:29:31.994941326 +0000
@@ -46,6 +46,7 @@
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
diff -ur nss-3.12.6.orig/nss/lib/util/config.mk nss-3.12.6/nss/lib/util/config.mk
--- nss-3.12.6.orig/nss/lib/util/config.mk	2009-12-15 22:22:31.000000000 +0000
+++ nss-3.12.6/nss/lib/util/config.mk	2012-02-02 14:32:38.525180567 +0000
@@ -36,6 +36,7 @@
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
 	-L$(NSPR_LIB_DIR) \
+	-R$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
