--- libtecla/Makefile.rules	Tue Dec  3 22:26:56 2002
+++ libtecla/Makefile.rules	Tue Jan 24 14:03:33 2012
@@ -54,19 +54,19 @@
 demos: $(DEMO_PROGS)
 
 demo$(SUFFIX): $(OBJDIR)/demo.o
-	LD_RUN_PATH="$(LIBDIR):$$LD_RUN_PATH:`pwd`" $(CC) $(CFLAGS) -o $@ \
+	$(CC) $(CFLAGS) -o $@ \
  $(OBJDIR)/demo.o -L. -ltecla$(SUFFIX) $(LIBS)
 
 demo2$(SUFFIX): $(OBJDIR)/demo2.o
-	LD_RUN_PATH="$(LIBDIR):$$LD_RUN_PATH:`pwd`" $(CC) $(CFLAGS) -o $@ \
+	$(CC) $(CFLAGS) -o $@ \
  $(OBJDIR)/demo2.o -L. -ltecla$(SUFFIX) $(LIBS)
 
 demo3$(SUFFIX): $(OBJDIR)/demo3.o
-	LD_RUN_PATH="$(LIBDIR):$$LD_RUN_PATH:`pwd`" $(CC) $(CFLAGS) -o $@ \
+	$(CC) $(CFLAGS) -o $@ \
  $(OBJDIR)/demo3.o -L. -ltecla$(SUFFIX) $(LIBS)
 
 enhance$(SUFFIX): $(OBJDIR)/enhance.o
-	LD_RUN_PATH="$(LIBDIR):$$LD_RUN_PATH:`pwd`" $(CC) $(CFLAGS) -o $@ \
+	$(CC) $(CFLAGS) -o $@ \
  $(OBJDIR)/enhance.o -L. -ltecla$(SUFFIX) $(LIBS)
 
 #-----------------------------------------------------------------------
