diff -up xdg-utils-1.0.2/scripts/xdg-desktop-icon.orig xdg-utils-1.0.2/scripts/xdg-desktop-icon
--- xdg-utils-1.0.2/scripts/xdg-desktop-icon.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-desktop-icon	2008-02-13 12:16:12.568697000 +0000
@@ -31,6 +31,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 usage()
 {
 cat << _USAGE

diff -up xdg-utils-1.0.2/scripts/xdg-desktop-menu.orig xdg-utils-1.0.2/scripts/xdg-desktop-menu
--- xdg-utils-1.0.2/scripts/xdg-desktop-menu.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-desktop-menu	2008-02-13 12:16:24.126242000 +0000
@@ -30,6 +30,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

diff -up xdg-utils-1.0.2/scripts/xdg-email.orig xdg-utils-1.0.2/scripts/xdg-email
--- xdg-utils-1.0.2/scripts/xdg-email.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-email	2008-02-13 12:16:30.151625000 +0000
@@ -32,6 +32,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

diff -up xdg-utils-1.0.2/scripts/xdg-icon-resource.orig xdg-utils-1.0.2/scripts/xdg-icon-resource
--- xdg-utils-1.0.2/scripts/xdg-icon-resource.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-icon-resource	2008-02-13 12:16:34.848978000 +0000
@@ -31,6 +31,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

diff -up xdg-utils-1.0.2/scripts/xdg-mime.orig xdg-utils-1.0.2/scripts/xdg-mime
--- xdg-utils-1.0.2/scripts/xdg-mime.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-mime	2008-02-13 12:16:40.231289000 +0000
@@ -32,6 +32,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

diff -up xdg-utils-1.0.2/scripts/xdg-open.orig xdg-utils-1.0.2/scripts/xdg-open
--- xdg-utils-1.0.2/scripts/xdg-open.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-open	2008-02-13 12:16:45.207908000 +0000
@@ -31,6 +31,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

diff -up xdg-utils-1.0.2/scripts/xdg-screensaver.orig xdg-utils-1.0.2/scripts/xdg-screensaver
--- xdg-utils-1.0.2/scripts/xdg-screensaver.orig	2008-02-13 12:10:36.000000000 +0000
+++ xdg-utils-1.0.2/scripts/xdg-screensaver	2008-02-13 12:16:49.482362000 +0000
@@ -30,6 +30,9 @@
 #
 #---------------------------------------------
 
+PATH=/usr/bin:/usr/openwin/bin:/usr/X11/bin:/usr/X11R6/bin:${PATH}
+export PATH
+
 manualpage()
 {
 cat << _MANUALPAGE

