--- SPIRV-Tools-2024.3/CMakeLists.txt.old	2024-11-18 14:18:25.146940505 -0500
+++ SPIRV-Tools-2024.3/CMakeLists.txt	2024-11-18 14:19:01.876955989 -0500
@@ -78,6 +78,8 @@
   add_definitions(-DSPIRV_GNU)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "QNX")
   add_definitions(-DSPIRV_QNX)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS")
+  add_definitions(-DSPIRV_SUNOS)
 else()
   message(FATAL_ERROR "Your platform '${CMAKE_SYSTEM_NAME}' is not supported!")
 endif()
--- SPIRV-Tools-2024.3/source/print.cpp.old	2024-11-18 14:25:14.998945818 -0500
+++ SPIRV-Tools-2024.3/source/print.cpp	2024-11-18 14:25:48.464713809 -0500
@@ -17,7 +17,7 @@
 #if defined(SPIRV_ANDROID) || defined(SPIRV_LINUX) || defined(SPIRV_MAC) || \
     defined(SPIRV_IOS) || defined(SPIRV_TVOS) || defined(SPIRV_FREEBSD) ||  \
     defined(SPIRV_OPENBSD) || defined(SPIRV_EMSCRIPTEN) ||                  \
-    defined(SPIRV_FUCHSIA) || defined(SPIRV_GNU) || defined(SPIRV_QNX)
+    defined(SPIRV_FUCHSIA) || defined(SPIRV_GNU) || defined(SPIRV_QNX) || defined(SPIRV_SUNOS)
 namespace spvtools {
 
 clr::reset::operator const char*() { return "\x1b[0m"; }
