--- postgresql-11.1/contrib/ltree_plpython/Makefile.~1~	2018-11-07 02:56:26.000000000 +0000
+++ postgresql-11.1/contrib/ltree_plpython/Makefile	2019-01-24 16:24:50.209094474 +0000
@@ -30,7 +30,7 @@
 # ... see silliness in plpython Makefile ...
 SHLIB_LINK_INTERNAL += $(sort $(wildcard ../../src/pl/plpython/libpython*.a))
 else
-rpathdir = $(python_libdir)
+#rpathdir = $(python_libdir)
 SHLIB_LINK += $(python_libspec) $(python_additional_libs)
 endif

--- postgresql-10.1/src/pl/plpython/Makefile.~1~	2017-11-07 03:46:52.000000000 +0000
+++ postgresql-10.1/src/pl/plpython/Makefile	2018-01-27 18:51:34.322805485 +0000
@@ -13,7 +13,7 @@

 override CPPFLAGS := -I. -I$(srcdir) $(python_includespec) $(CPPFLAGS)

-rpathdir = $(python_libdir)
+#rpathdir = $(python_libdir)

 PGFILEDESC = "PL/Python - procedural language"

--- postgresql-11.1/contrib/hstore_plpython/Makefile.~1~	2018-11-07 02:56:26.000000000 +0000
+++ postgresql-11.1/contrib/hstore_plpython/Makefile	2019-01-24 16:25:56.012593981 +0000
@@ -30,7 +30,7 @@
 # ... see silliness in plpython Makefile ...
 SHLIB_LINK_INTERNAL += $(sort $(wildcard ../../src/pl/plpython/libpython*.a))
 else
-rpathdir = $(python_libdir)
+#rpathdir = $(python_libdir)
 SHLIB_LINK += $(python_libspec) $(python_additional_libs)
 endif
