--- igerman98-20151222/Makefile.2	2016-02-19 13:58:26.432573178 +0300
+++ igerman98-20151222/Makefile	2016-02-19 13:58:57.400287708 +0300
@@ -80,7 +80,7 @@
 HUNSPELL_BIN = hunspell
 # Hunspell starting with 1.1.9 (?) uses automatic charset transcoding,
 # we need to make sure we use latin1:
-HUNSPELL = $(HUNSPELL_BIN) `$(HUNSPELL_BIN) -h 2>&1 | grep -q "hunspell \\-i" && echo "-i latin1"`
+HUNSPELL = $(HUNSPELL_BIN) `$(HUNSPELL_BIN) -h 2>&1 | grep -q "hunspell \\-i" && echo "-i ISO8859-1"`
 
 
 SQ = sq
