--- a/configure.ac	2016-05-09 18:58:41.470160047 +0200
+++ b/configure.ac	2016-05-09 19:00:52.709784163 +0200
@@ -48,10 +48,10 @@
 	AC_MSG_CHECKING([for socklen_t equivalent])
 	AC_CACHE_VAL([dcc_cv_socklen_t_equiv],
 	[
-	# Systems have either "struct sockaddr *" or
+	# Systems have either "struct sockaddr{,_in} *" or
 	# "void *" as the second argument to getpeername
 	dcc_cv_socklen_t_equiv=
-	for arg2 in "struct sockaddr" void; do
+	for arg2 in "struct sockaddr_in" "struct sockaddr" void; do
 		for t in int size_t unsigned long "unsigned long"; do
 			AC_TRY_COMPILE([
 #include <sys/types.h>
