diff -Nru rxvt-unicode-9.31.orig/doc/Makefile.in rxvt-unicode-9.31/doc/Makefile.in
--- rxvt-unicode-9.31.orig/doc/Makefile.in	2022-12-29 02:31:09.000000000 +0100
+++ rxvt-unicode-9.31/doc/Makefile.in	2023-10-07 17:20:05.992831113 +0200
@@ -100,8 +100,8 @@
 	$(INSTALL_DATA) rxvtd.1.man   $(DESTDIR)$(man1dir)/$(RXVTNAME)d.$(man1ext)
 	$(INSTALL_DATA) rxvt.7.man    $(DESTDIR)$(man7dir)/$(RXVTNAME).$(man7ext)
 	$(INSTALL_DATA) rclock.1.man  $(DESTDIR)$(man1dir)/urclock.$(man1ext)
-	@TIC@ -x $(srcdir)/etc/rxvt-unicode.terminfo || \
-           @TIC@ $(srcdir)/etc/rxvt-unicode.terminfo
+#	@TIC@ -x $(srcdir)/etc/rxvt-unicode.terminfo || \
+#           @TIC@ $(srcdir)/etc/rxvt-unicode.terminfo
 
 distdepend: alldoc
 
