--- VirtualBox-6.1.22/Config.kmk.~1~	2021-04-28 18:21:25.000000000 +0000
+++ VirtualBox-6.1.22/Config.kmk	2021-05-08 17:35:29.372050156 +0000
@@ -2752,8 +2752,6 @@
 		$(VBOX_GCC32_PATH_CC) \
 		$(VBOX_GCC32_PATH_CXX) \
 		$(VBOX_GCC32_LIBGCC) \
-		$(if-expr "$(KBUILD_HOST).$(KBUILD_HOST_ARCH)" == "solaris.amd64" && $(KBUILD_HOST_VERSION_MINOR) >= 11 \
-			, /platform/i86pc/kernel/$(KBUILD_HOST_ARCH)/unix,) \
         	| $(PATH_OUT)/DynamicConfig.c $(PATH_OUT)/DynamicConfig.cpp
 	$(call MSG_GENERATE,,$@,)
 	$(QUIET)$(RM) -f '$@'
