--- VirtualBox-5.2.18/src/VBox/Storage/testcase/Makefile.kmk.1	2018-09-25 10:57:14.145027298 +0000
+++ VirtualBox-5.2.18/src/VBox/Storage/testcase/Makefile.kmk	2018-09-25 10:58:17.507384829 +0000
@@ -160,6 +160,9 @@
   vbox-img_SDKS.win = VBOX_NTDLL
  else ifeq ($(KBUILD_TARGET),solaris)
   vbox-img_LIBS += kstat
+  ifdef SDK_VBOX_LIBXML2_LIBS
+   vbox-img_LIBS += xml2
+  endif
  endif
 
 endif
--- VirtualBox-6.1.4/src/VBox/Runtime/Makefile.kmk	2020-02-18 18:20:34.000000000 +0000
+++ VirtualBox-6.1.4/src/VBox/Runtime/Makefile.kmk	2020-02-20 19:10:02.466580278 +0000
@@ -2212,6 +2212,7 @@ ifndef SDK_VBOX_LIBXML2_LIBS
  VBoxRT_LIBS                  += \
 	$(PATH_STAGE_LIB)/VBox-libxml2$(VBOX_SUFF_LIB)
 endif
+VBoxRT_LIBS                 += xml2
 VBoxRT_LIBS                   += \
 	$(SDK_VBOX_ZLIB_LIBS)
 ifndef SDK_VBOX_OPENSSL_LIBS
@@ -2366,6 +2367,7 @@ ifndef SDK_VBOX_OPENSSL_LIBS
 	$(PATH_STAGE_LIB)/VBox-libcrypto-x86$(VBOX_SUFF_LIB) \
 	$(PATH_STAGE_LIB)/VBox-libssl-x86$(VBOX_SUFF_LIB)
 endif
+VBoxRT-x86_LIBS                  += lxml2
 ifdef IPRT_WITH_LZO
  VBoxRT-x86_LIBS                  += lzo2
 endif
