--- VirtualBox-7.0.18/configure.orig
+++ VirtualBox-7.0.18/configure
@@ -560,8 +560,8 @@
       fi
       # Solaris sports a 32-bit gcc/g++.
       if [ "$OS" = "solaris" -a "$BUILD_MACHINE" = "amd64" ]; then
-        [ "$CC" = "gcc" ] && CC="gcc -m64"
-        [ "$CXX" = "g++" ] && CXX="g++ -m64"
+        CC="$CC -m64"
+        CXX="$CXX -m64"
       fi
     fi
   fi
--- VirtualBox-7.0.18/Config.kmk.orig
+++ VirtualBox-7.0.18/Config.kmk
@@ -2623,13 +2623,13 @@
 
 # The X11 LIBPATH. (The amd64 hack is for RHEL4.)
 ifeq ($(KBUILD_TARGET_ARCH),amd64)
- VBOX_LIBPATH_X11 ?= /usr/X11R6/lib64 /usr/X11R6/lib
+ VBOX_LIBPATH_X11 ?= /usr/X11R6/lib/amd64
 else
  VBOX_LIBPATH_X11 ?= /usr/X11R6/lib
 endif
 
 # 32bit X11 LIBPATH for building 32bit guest additions
-VBOX_LIBPATH32_X11 ?= /usr/X11R6/lib32 /usr/X11R6/lib
+VBOX_LIBPATH32_X11 ?= /usr/X11R6/lib
 
 # Good windows shell.
 VBOX_GOOD_COMSPEC = $(subst \,/,$(SystemRoot))/System32/cmd.exe
