--- VirtualBox-5.2.20/configure.orig	2018-10-28 02:18:52.277011628 +0000
+++ VirtualBox-5.2.20/configure	2018-10-28 02:21:56.272230984 +0000
@@ -511,8 +511,8 @@
       fi
       # Solaris sports a 32-bit gcc/g++.
       if [ "$OS" = "solaris" -a "$BUILD_MACHINE" = "amd64" ]; then
-        [ "$CC" = "gcc" ] && CC="gcc -m64"
-        [ "$CXX" = "g++" ] && CXX="g++ -m64"
+        CC="$CC -m64"
+        CXX="$CXX -m64"
       fi
     fi
   fi
