--- VirtualBox-6.0.4/src/VBox/Installer/solaris/vboxconfig.sh	2019-01-25 19:20:05.000000000 +0000
+++ VirtualBox-6.0.4/src/VBox/Installer/solaris/vboxconfig.sh.new	2019-02-27 06:55:46.286189698 +0000
@@ -663,9 +663,9 @@ install_drivers()
 {
     if test -f "$DIR_CONF/vboxdrv.conf"; then
         if test -n "_HARDENED_"; then
-            add_driver "$MOD_VBOXDRV" "$DESC_VBOXDRV" "$FATALOP" "not-$NULLOP" "'* 0600 root sys','vboxdrvu 0666 root sys'"
+            add_driver "$MOD_VBOXDRV" "$DESC_VBOXDRV" "$FATALOP" "not-$NULLOP" "* 0600 root sys,vboxdrvu 0666 root sys"
         else
-            add_driver "$MOD_VBOXDRV" "$DESC_VBOXDRV" "$FATALOP" "not-$NULLOP" "'* 0666 root sys','vboxdrvu 0666 root sys'"
+            add_driver "$MOD_VBOXDRV" "$DESC_VBOXDRV" "$FATALOP" "not-$NULLOP" "* 0666 root sys,vboxdrvu 0666 root sys"
         fi
         load_module "drv/$MOD_VBOXDRV" "$DESC_VBOXDRV" "$FATALOP"
     else
@@ -746,7 +746,7 @@ install_drivers()
         # All users which need host USB-passthrough support will have to be added to this group.
         groupadd vboxuser >/dev/null 2>&1
 
-        add_driver "$MOD_VBOXUSBMON" "$DESC_VBOXUSBMON" "$FATALOP" "not-$NULLOP" "'* 0666 root sys'"
+        add_driver "$MOD_VBOXUSBMON" "$DESC_VBOXUSBMON" "$FATALOP" "not-$NULLOP" "* 0666 root sys"
         load_module "drv/$MOD_VBOXUSBMON" "$DESC_VBOXUSBMON" "$FATALOP"
 
         chown root:vboxuser "/devices/pseudo/vboxusbmon@0:vboxusbmon"
