--- VirtualBox-5.2.22/src/VBox/Additions/x11/vboxvideo/Makefile.kmk.1	2018-12-07 08:31:39.028471393 +0000
+++ VirtualBox-5.2.22/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2018-12-07 08:32:25.607417992 +0000
@@ -410,6 +410,7 @@
         /usr/include/xorg \
         /usr/include/pixman-1
  vboxvideo_drv_system_SOURCES := $(vboxvideo_drv_17_SOURCES)
+ vboxvideo_drv_system_LIBS += $(vboxvideo_drv_70_LIBS)
 endif
