--- VirtualBox-5.2.22/src/VBox/Additions/solaris/Installer/vbox_vendor_select.1	2018-12-20 12:28:02.003452887 +0000
+++ VirtualBox-5.2.22/src/VBox/Additions/solaris/Installer/vbox_vendor_select	2018-12-20 12:28:57.298346649 +0000
@@ -31,7 +31,7 @@
 # DEALINGS IN THE SOFTWARE.
 #
 
-LINKDIR=/system/volatile/opengl
+LINKDIR=/var/run/opengl
 MESA_SELECT=/lib/opengl/ogl_select/mesa_vendor_select
 
 PATH=/usr/bin:/usr/sbin
