--- VirtualBox-5.2.22/configure.1	2018-12-21 14:50:56.273157765 +0000
+++ VirtualBox-5.2.22/configure	2018-12-21 14:53:09.904912027 +0000
@@ -2755,7 +2755,7 @@
 fi
 
 # the libraries
-if [ $ONLY_ADDITIONS -eq 0 ]; then
+#if [ $ONLY_ADDITIONS -eq 0 ]; then
   [ "$OS" != "darwin"  ] && check_pthread
   check_libxml2
   [ $WITH_LIBIDL -eq 1 ] && check_libidl
@@ -2791,7 +2791,7 @@
       cnf_append "VBOX_WITH_AUDIO_PULSE" ""
     fi
  fi
-fi
+#fi
 
 # Linux-specific
 if [ "$OS" = "linux" ]; then
