--- VirtualBox-7.0.18/Config.kmk.orig
+++ VirtualBox-7.0.18/Config.kmk
@@ -5433,10 +5433,10 @@
 
 ifeq ($(VBOX_LDR_FMT),elf)
  TEMPLATE_VBoxR0_TOOL               = $(VBOX_GCC_TOOL)
- TEMPLATE_VBoxR0_CFLAGS             = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_C) \
+ TEMPLATE_VBoxR0_CFLAGS             = -fno-pie -nostdinc -g -gdwarf-2 -gstrict-dwarf $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_C) \
  	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
  	$(VBOX_GCC_fno-stack-protector) -fno-common -ffreestanding $(VBOX_GCC_fvisibility-hidden) -std=gnu99 $(VBOX_GCC_IPRT_FMT_CHECK)
- TEMPLATE_VBoxR0_CXXFLAGS           = -fno-pie -nostdinc -g $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_CXX) \
+ TEMPLATE_VBoxR0_CXXFLAGS           = -fno-pie -nostdinc -g -gdwarf-2 -gstrict-dwarf $(VBOX_GCC_pipe) $(VBOX_GCC_WERR) $(VBOX_GCC_PEDANTIC_CXX) \
  	$(VBOX_GCC_Wno-variadic-macros) $(VBOX_GCC_R0_OPT) $(VBOX_GCC_R0_FP) -fno-strict-aliasing -fno-exceptions \
  	$(VBOX_GCC_fno-stack-protector) -fno-common $(VBOX_GCC_fvisibility-inlines-hidden) $(VBOX_GCC_fvisibility-hidden) \
  	-fno-rtti $(VBOX_GCC_std) $(VBOX_GCC_IPRT_FMT_CHECK)
@@ -5844,7 +5844,7 @@
  	-Wformat  -Wimplicit-int -Wimplicit-function-declaration -Werror-implicit-function-declaration -Wparentheses \
  	-O2 -fno-omit-frame-pointer -ffreestanding -nodefaultlibs -fno-strict-aliasing -fno-common -std=c99
  TEMPLATE_VBoxR0Drv_CFLAGS.x86         = -m32 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow
- TEMPLATE_VBoxR0Drv_CFLAGS.amd64       = -m64 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow $(VBOX_GCC_msave-args) \
+ TEMPLATE_VBoxR0Drv_CFLAGS.amd64       = -g -gdwarf-2 -gstrict-dwarf -m64 -mno-sse -mno-mmx -mno-sse2 -mno-3dnow $(VBOX_GCC_msave-args) \
  	-mno-red-zone -mcmodel=kernel $(if-expr defined(VBOX_GCC_USING_SOLARIS_AS),-Wa$(COMMA)-xmodel=kernel,) \
  	-fno-reorder-blocks -fno-asynchronous-unwind-tables -funit-at-a-time
  TEMPLATE_VBoxR0Drv_CXXFLAGS          = -Wpointer-arith \
