--- VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp	2019-03-15 10:53:19.491237602 +0000
+++ VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp	2019-03-15 10:53:46.415980120 +0000
@@ -61,7 +61,7 @@
 {
     struct utsname name;
 
-    if (uname(&name))
+    if (uname(&name) == -1)
         VBClFatalError(("Failed to get kernel name.\n"));
     if (strcmp(name.sysname, "Linux"))
         return false;
