"OpenIndiana" does not fit into 10 characters.

--- VirtualBox-6.0.6/Config.kmk	2019-04-27 22:40:07.002154519 +0000
+++ VirtualBox-6.0.6/Config.kmk.new	2019-04-28 14:44:08.047602298 +0000
@@ -7484,7 +7484,7 @@ endif
 #
 VBOX_VERSION_COMP_VALID_CHAR_SET = ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvwxyz()[]{}+-.,
 ifneq ($(VBOX_BUILD_PUBLISHER),)
- if !defined(VBOX_BUILD_TAG) && $(length $(VBOX_BUILD_PUBLISHER)) > 10
+ if !defined(VBOX_BUILD_TAG) && $(length $(VBOX_BUILD_PUBLISHER)) > 11
   $(error VBox: VBOX_BUILD_PUBLISHER must not be longer than 10 characters!)
  else if defined(VBOX_BUILD_TAG) && $(length $(VBOX_BUILD_PUBLISHER)) > 48
   $(error VBox: VBOX_BUILD_PUBLISHER must not be longer than 48 characters!)
