--- VirtualBox-7.1.10/src/VBox/Frontends/VirtualBox/Makefile.kmk.orig
+++ VirtualBox-7.1.10/src/VBox/Frontends/VirtualBox/Makefile.kmk
@@ -1660,6 +1660,7 @@
 ifdef VBOX_WITH_SCREENSAVER_CONTROL
  UICommon_QT_MODULES.solaris  += DBus
  UICommon_QT_MODULES.solaris  += Xml
+ UICommon_QT_MODULES.solaris  += PrintSupport
 endif
 ifdef VBOX_WITH_SCREENSAVER_CONTROL
  UICommon_QT_MODULES.freebsd  += DBus
