diff --git a/src/VBox/Main/Makefile.kmk b/src/VBox/Main/Makefile.kmk
index fd74244..4f0d5e2 100644
--- a/src/VBox/Main/Makefile.kmk
+++ b/src/VBox/Main/Makefile.kmk
@@ -794,6 +794,9 @@ if !defined(VBOX_ONLY_SDK) && !defined(VBOX_ONLY_EXTPACKS) # Note this goes on f
  ifdef VBOX_WITH_3D_ACCELERATION
   VBoxSVC_DEFS += VBOX_WITH_3D_ACCELERATION
  endif
+ ifdef VBOX_WITH_VIDEOHWACCEL
+  VBoxSVC_DEFS += VBOX_WITH_VIDEOHWACCEL
+ endif
 
  ifeq ($(KBUILD_TYPE),debug)
   VBoxSVC_LDFLAGS.linux  += -rdynamic # for backtrace_symbols()
