--- a/configure.ac	2016-10-24 16:59:06.225902806 +0200
+++ b/configure.ac	2016-10-24 17:00:35.592535133 +0200
@@ -514,7 +514,7 @@
 # We define the name of the OS so the code can act accordingly
 # We also need to add the right LDFLAGS
 AS_CASE([$host_os],
-	[solaris*], [
+	[solaris*|illumos*], [
 		AC_CHECK_HEADERS([kstat.h libdevinfo.h procfs.h sys/sockio.h])
 		AC_DEFINE(SOLARIS, 1, [Building on Solaris])
 		AM_CONDITIONAL(SETUIDBINS, true)
