coverage-pyver-pragma is not needed to run tests.

--- handy-archives-0.2.0/tests/requirements.txt.orig
+++ handy-archives-0.2.0/tests/requirements.txt
@@ -1,6 +1,5 @@
 coincidence>=0.2.0
 coverage>=5.1
-coverage-pyver-pragma>=0.2.1
 importlib-metadata>=3.6.0
 pytest>=6.0.0
 pytest-cov>=2.8.1
--- handy-archives-0.2.0/tox.ini.orig
+++ handy-archives-0.2.0/tox.ini
@@ -202,9 +202,6 @@
 unused-arguments-ignore-magic-methods = True
 unused-arguments-ignore-variadic-names = True
 
-[coverage:run]
-plugins = coverage_pyver_pragma
-
 [coverage:report]
 fail_under = 100
 exclude_lines =
