diff -up gnome-python-extras-2.19.1/gksu/gksu.defs.orig gnome-python-extras-2.19.1/gksu/gksu.defs
--- gnome-python-extras-2.19.1/gksu/gksu.defs.orig	2007-06-05 15:52:30.000000000 +0100
+++ gnome-python-extras-2.19.1/gksu/gksu.defs	2008-02-18 22:14:31.081280000 +0000
@@ -104,21 +104,6 @@
   (return-type "gboolean")
 )
 
-(define-method set_keep_env
-  (of-object "GksuContext")
-  (c-name "gksu_context_set_keep_env")
-  (return-type "none")
-  (parameters
-    '("gboolean" "value")
-  )
-)
-
-(define-method get_keep_env
-  (of-object "GksuContext")
-  (c-name "gksu_context_get_keep_env")
-  (return-type "gboolean")
-)
-
 (define-method set_debug
   (of-object "GksuContext")
   (c-name "gksu_context_set_debug")
@@ -134,21 +119,6 @@
   (return-type "gboolean")
 )
 
-(define-method set_ssh_fwd
-  (of-object "GksuContext")
-  (c-name "gksu_context_set_ssh_fwd")
-  (return-type "none")
-  (parameters
-    '("gboolean" "value")
-  )
-)
-
-(define-method get_ssh_fwd
-  (of-object "GksuContext")
-  (c-name "gksu_context_get_ssh_fwd")
-  (return-type "gboolean")
-)
-
 (define-function gksu_context_get_type
   (c-name "gksu_context_get_type")
   (return-type "GType")

