--- wxWidgets-3.1.5/include/wx/filename.h.orig	lun. janv. 24 16:18:50 2022
+++ wxWidgets-3.1.5/include/wx/filename.h	lun. janv. 24 16:19:05 2022
@@ -465,7 +465,7 @@
 
     // get the canonical path separator for this format
     static wxUniChar GetPathSeparator(wxPathFormat format = wxPATH_NATIVE)
-        { return GetPathSeparators(format)[0u]; }
+        { return GetPathSeparators(format)[0]; }
 
     // is the char a path separator for this format?
     static bool IsPathSeparator(wxChar ch, wxPathFormat format = wxPATH_NATIVE);
