--- wxWidgets-3.1.5/include/wx/event.orig	Fri Feb 25 14:24:03 2022
+++ wxWidgets-3.1.5/include/wx/event.h	Fri Feb 25 14:25:01 2022
@@ -2408,9 +2408,10 @@
 {
     // This ctor is only intended to be used by wxWidgets itself, so it's
     // intentionally declared as private when not building the library itself.
-#ifdef WXBUILDING
+// #ifdef WXBUILDING
+// we need this for building codeblocks-20.03
 public:
-#endif // WXBUILDING
+// #endif // WXBUILDING
     explicit wxNcPaintEvent(wxWindowBase* window = NULL);
 
 public:
