diff -x '*.out' -x '*.*~' -x '*.orig' -x '*.rej' -rup ../compiz-0.8.4/metadata/decoration.xml.in compiz-0.8.4/metadata/decoration.xml.in
--- ../compiz-0.8.4/metadata/decoration.xml.in	2009-10-14 15:28:57.100052463 +0200
+++ compiz-0.8.4/metadata/decoration.xml.in	2009-10-14 15:29:19.105380457 +0200
@@ -16,7 +16,7 @@
 	    <option name="shadow_radius" type="float">
 		<_short>Shadow Radius</_short>
 		<_long>Drop shadow radius</_long>
-		<default>8.0</default>
+		<default>9.0</default>
 		<min>0.1</min>
 		<max>18.0</max>
 		<precision>0.1</precision>
diff -x '*.out' -x '*.*~' -x '*.orig' -x '*.rej' -rup ../compiz-0.8.4/metadata/place.xml.in compiz-0.8.4/metadata/place.xml.in
--- ../compiz-0.8.4/metadata/place.xml.in	2009-10-14 15:28:57.098599661 +0200
+++ compiz-0.8.4/metadata/place.xml.in	2009-10-14 15:29:19.105729230 +0200
@@ -12,7 +12,7 @@
 	    <option name="mode" type="int">
 		<_short>Placement Mode</_short>
 		<_long>Algorithm to use for window placement</_long>
-		<default>0</default>
+		<default>2</default>
 		<min>0</min>
 		<max>5</max>
 		<desc>
