--- compiz-0.8.14/metadata/cube.xml.in.new	2018-10-18 08:41:19.480607349 +0000
+++ compiz-0.8.14/metadata/cube.xml.in	2018-10-18 08:39:07.726512162 +0000
@@ -96,9 +96,9 @@
 		    <_short>Cube Color</_short>
 		    <_long>Color of top and bottom sides of the cube</_long>
 		    <default>
-			<red>0xfefe</red>
+			<red>0xffff</red>
 			<green>0xffff</green>
-			<blue>0xc7c7</blue>
+			<blue>0xffff</blue>
 		    </default>
 		</option>
 		<subgroup>
@@ -106,7 +106,7 @@
 		    <option name="scale_image" type="bool">
 			<_short>Scale image</_short>
 			<_long>Scale images to cover top face of cube</_long>
-			<default>false</default>
+			<default>true</default>
 		    </option>
 		    <option name="images" type="list">
 			<_short>Image files</_short>
@@ -114,7 +114,7 @@
 			<type>string</type>
 			<hints>file;image;</hints>
 			<default>
-			    <value>freedesktop</value>
+			    <value>/usr/share/pixmaps/backgrounds/openindiana/openindiana-default.jpg</value>
 			</default>
 		    </option>
 		    <option name="adjust_image" type="bool">
@@ -165,7 +165,7 @@
 		<option name="active_opacity" type="float">
 		    <_short>Opacity During Rotation</_short>
 		    <_long>Opacity of desktop window during rotation.</_long>
-		    <default>70.0</default>
+		    <default>60.0</default>
 		    <min>0.0</min>
 		    <max>100.0</max>
 		    <precision>1.0</precision>
