--- compiz-0.8.18/metadata/decoration.xml.in.orig	2020-02-11 12:21:50.000000000 +0000
+++ compiz-0.8.18/metadata/decoration.xml.in	2020-09-20 19:43:46.307477778 +0000
@@ -17,7 +17,7 @@
 	    <option name="shadow_radius" type="float">
 		<short>Shadow Radius</short>
 		<long>Drop shadow radius</long>
-		<default>8.0</default>
+		<default>9.0</default>
 		<min>0.1</min>
 		<max>18.0</max>
 		<precision>0.1</precision>
--- compiz-0.8.18/metadata/place.xml.in.orig	2020-02-11 12:21:50.000000000 +0000
+++ compiz-0.8.18/metadata/place.xml.in	2020-09-20 19:44:35.532457447 +0000
@@ -13,7 +13,7 @@
 	    <option name="mode" type="int">
 		<short>Placement Mode</short>
 		<long>Algorithm to use for window placement</long>
-		<default>0</default>
+		<default>2</default>
 		<min>0</min>
 		<max>5</max>
 		<desc>
