--- compiz-0.8.0/metadata/Makefile.am.ori	2009-03-11 08:05:16.108609014 +0000
+++ compiz-0.8.0/metadata/Makefile.am	2009-03-11 08:06:46.529133933 +0000
@@ -38,8 +38,8 @@
 core.xml.in: core.xml.in.in Makefile
 	plugins=$(default_plugins); \
 	if [ -z $$plugins ]; then plugins="core"; fi; \
-	plugin_values=`echo $$plugins | sed 's/\([a-zA-Z0-9]*\)/<value>\1<\/value>/g' | tr ',' '\n' | sed 's/^/\t\t    /g'`; \
-	$(AWK) -v aa="$$plugin_values" '{gsub("\t\t    <default_plugins/>", aa); print }' <$(srcdir)/core.xml.in.in >core.xml.in
+	plugin_values=`echo $$plugins | sed -e 's#\([a-zA-Z0-9]*\)#<value>\1<\/value>#g' -e 's#,#\\\n\\\t\\\t    #g'`; \
+	$(AWK) -v aa="$$plugin_values" '{gsub("<default_plugins/>", aa); print }' <$(srcdir)/core.xml.in.in >core.xml.in
 
 @INTLTOOL_XML_RULE@
 
