--- compiz-0.8.18/metadata/cube.xml.in.orig	2020-02-11 12:21:50.000000000 +0000
+++ compiz-0.8.18/metadata/cube.xml.in	2020-09-20 19:40:58.242477420 +0000
@@ -96,9 +96,9 @@
 		    <short>Cube Color</short>
 		    <long>Color of top and bottom sides of the cube</long>
 		    <default>
-			<red>0xfefe</red>
+			<red>0xffff</red>
 			<green>0xffff</green>
-			<blue>0xc7c7</blue>
+			<blue>0xffff</blue>
 		    </default>
 		</option>
 		<subgroup>
@@ -106,7 +106,7 @@
 		    <option name="scale_image" type="bool">
 			<short>Scale image</short>
 			<long>Scale images to cover top face of cube</long>
-			<default>false</default>
+			<default>true</default>
 		    </option>
 		    <option name="images" type="list">
 			<short>Image files</short>
@@ -114,7 +114,7 @@
 			<type>string</type>
 			<hints>file;image;</hints>
 			<default>
-			    <value>freedesktop</value>
+			    <value>/usr/share/pixmaps/backgrounds/openindiana/openindiana-default.jpg</value>
 			</default>
 		    </option>
 		    <option name="adjust_image" type="bool">
@@ -165,7 +165,7 @@
 		<option name="active_opacity" type="float">
 		    <short>Opacity During Rotation</short>
 		    <long>Opacity of desktop window during rotation.</long>
-		    <default>70.0</default>
+		    <default>60.0</default>
 		    <min>0.0</min>
 		    <max>100.0</max>
 		    <precision>1.0</precision>
