--- mod_perl-2.0.4/lib/Apache2/SizeLimit.pm.orig	Mon Nov 23 02:17:25 2009
+++ mod_perl-2.0.4/lib/Apache2/SizeLimit.pm	Mon Nov 23 02:21:39 2009
@@ -52,7 +52,8 @@
         if Apache2::MPM->is_threaded();
 
     # decide at compile time how to check for a process' memory size.
-    if (SOLARIS && $Config{'osvers'} >= 2.6) {
+    my ($major,$minor) = split(/\./, $Config{'osvers'});
+    if (SOLARIS && (($major > 2) || ($major == 2 && $minor >= 6))) {
 
         $HOW_BIG_IS_IT = \&solaris_2_6_size_check;
 
