diff --git a/vsftpd.conf b/vsftpd.conf
index cc1c607..029b229 100644
--- a/vsftpd.conf
+++ b/vsftpd.conf
@@ -9,7 +9,7 @@
 # capabilities.
 #
 # Allow anonymous FTP? (Beware - allowed by default if you comment this out).
-anonymous_enable=YES
+anonymous_enable=NO
 #
 # Uncomment this to allow local users to log in.
 #local_enable=YES
@@ -62,7 +62,7 @@ connect_from_port_20=YES
 #
 # It is recommended that you define on your system a unique user which the
 # ftp server can use as a totally isolated and unprivileged user.
-#nopriv_user=ftpsecure
+nopriv_user=ftp
 #
 # Enable this and the server will recognise asynchronous ABOR requests. Not
 # recommended for security (the code is non-trivial). Not enabling it,
diff --git a/vsftpd.conf b/vsftpd.conf
index 6bbceab..68e7b46 100644
--- a/vsftpd.conf
+++ b/vsftpd.conf
@@ -110,6 +110,7 @@ nopriv_user=ftp
 # listens on IPv4 sockets. This directive cannot be used in conjunction
 # with the listen_ipv6 directive.
 listen=YES
+background=YES
 #
 # This directive enables listening on IPv6 sockets. To listen on IPv4 and IPv6
 # sockets, you must run two copies of vsftpd with two configuration files.
