--- blueprint-0.9.20/Theme/gtk-2.0/iconrc.in.orig	2009-04-10 10:29:49.989433970 +0900
+++ blueprint-0.9.20/Theme/gtk-2.0/iconrc.in	2009-04-10 10:31:35.575541424 +0900
@@ -34,23 +34,31 @@ style "icon-style"
   stock["gtk-floppy"] =		  {{ "stock_save_16.png", *, *, "gtk-menu"},
 				   { "stock_save_24.png", *, *, *}}
 			 
-  stock["gtk-go-back"] =	  {{ "stock_left_arrow_16.png", *, *, "gtk-menu"},
-				   { "stock_left_arrow_24.png", *, *, *}}
+  stock["gtk-go-back"] =	  {{ "stock_left_arrow_16.png", LTR, *, "gtk-menu"},
+                        	   { "stock_right_arrow_16.png", RTL, *, "gtk-menu"},
+				   { "stock_left_arrow_24.png", LTR, *, *},
+				   { "stock_right_arrow_24.png", RTL, *, *}}
 			  
   stock["gtk-go-down"] =	  {{ "stock_down_arrow_16.png", *, *, "gtk-menu"},
 				   { "stock_down_arrow_24.png", *, *, *}}
 			  
-  stock["gtk-go-forward"] =	  {{ "stock_right_arrow_16.png", *, *, "gtk-menu"},
-				   { "stock_right_arrow_24.png", *, *, *}}
+  stock["gtk-go-forward"] =	  {{ "stock_right_arrow_16.png", LTR, *, "gtk-menu"},
+                           	   { "stock_left_arrow_16.png", RTL, *, "gtk-menu"},
+				   { "stock_right_arrow_24.png", LTR, *, *},
+				   { "stock_left_arrow_24.png", RTL, *, *}}
 			     
   stock["gtk-goto-bottom"] =	  {{ "stock_bottom_16.png", *, *, "gtk-menu"},
 				   { "stock_bottom_24.png", *, *, *}}
 			      
-  stock["gtk-goto-first"] =	  {{ "stock_first_16.png", *, *, "gtk-menu"},
-				   { "stock_first_24.png", *, *, *}}
+  stock["gtk-goto-first"] =	  {{ "stock_first_16.png", LTR, *, "gtk-menu"},
+                           	   { "stock_last_16.png", RTL, *, "gtk-menu"},
+				   { "stock_first_24.png", LTR, *, *},
+				   { "stock_last_24.png", RTL, *, *}}
 			     
-  stock["gtk-goto-last"] =	  {{ "stock_last_16.png", *, *, "gtk-menu"},
-				   { "stock_last_24.png", *, *, *}}
+  stock["gtk-goto-last"] =	  {{ "stock_last_16.png", LTR, *, "gtk-menu"},
+                          	   { "stock_last_16.png", RTL, *, "gtk-menu"},
+				   { "stock_last_24.png", LTR, *, *},
+				   { "stock_last_24.png", RTL, *, *}}
 			    
   stock["gtk-goto-top"] =	  {{ "stock_top_16.png", *, *, "gtk-menu"},
 				   { "stock_top_24.png", *, *, *}}
