--- Cython-3.0.0/Cython/Build/Cythonize.py.orig
+++ Cython-3.0.0/Cython/Build/Cythonize.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 
 from __future__ import absolute_import, print_function
 
--- Cython-3.0.0/Cython/Debugger/libpython.py.orig
+++ Cython-3.0.0/Cython/Debugger/libpython.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 
 # NOTE: Most of this file is taken from the Python source distribution
 # It can be found under Tools/gdb/libpython.py. It is shipped with Cython
