Remove timing from test output

--- cython-3.1.3/runtests.py.orig
+++ cython-3.1.3/runtests.py
@@ -689,7 +689,6 @@
             lines.append("%-12s: %8.2f sec  (%4d, %6.3f / run) - slowest: %s\n" % (
                 metric, t, count, t / count,
                 ', '.join("'{2}:{1}' ({0:.2f}s)".format(*item) for item in heapq.nlargest(self.top_n, top))))
-        out.write(''.join(lines))
 
 
 class TestBuilder(object):
@@ -2643,7 +2642,6 @@
                     return
             if open_shards is not None:
                 waiting_for_shards = f" - waiting for {open_shards}"
-            write(f'\n#### {now()}{waiting_for_shards}\n')
 
     thread = threading.Thread(target=time_stamper, name='time_stamper')
     thread.daemon = True
