--- illumos-fusefs-5955f660e3eece3e9144ea1d990186cc9a88bc13/kernel/amd64/Makefile.orig
+++ illumos-fusefs-5955f660e3eece3e9144ea1d990186cc9a88bc13/kernel/amd64/Makefile
@@ -23,7 +23,7 @@
 # Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 
-CFLAGS=-xmodel=kernel -m64
+CFLAGS= -ffreestanding -mcmodel=kernel -mno-red-zone -m64
 LINTFLAGS = -m64
 include ../Makefile.com
 
