--- libvdpau-1.3/test/dlclose.c.orig	2020-02-03 00:38:21.922090389 +0000
+++ libvdpau-1.3/test/dlclose.c	2020-02-03 00:39:06.824734649 +0000
@@ -34,7 +34,7 @@
     // Generic Event Extension causes an identical bug to the one this program
     // is trying to test for.
     int nOpenFDs = countOpenFDs();
-    void *libXext = dlopen("libXext.so.6", RTLD_LAZY);
+    void *libXext = dlopen("libXext.so", RTLD_LAZY);
     void *libvdpau = dlopen("src/libvdpau.so", RTLD_LAZY);
     Display *dpy = XOpenDisplay(NULL);
     VdpDeviceCreateX11 *pvdp_device_create_x11;
@@ -43,7 +43,7 @@
     VdpStatus status;
 
     if (!libXext) {
-        fprintf(stderr, "Failed to open libXext.so.6: %s", dlerror());
+        fprintf(stderr, "Failed to open libXext.so: %s", dlerror());
         return SKIP;
     }
 
