Subject of ISC Ticket RT5819, This in-house hack does not address the
issue as such but allows for consistent failure results.

--- a/bin/tests/dst/Makefile.in	Thu Mar 03 13:53:44 2016 +0000
+++ b/bin/tests/dst/Makefile.in	Sat Mar 05 14:22:01 2016 +0000
@@ -49,7 +49,7 @@
 	${LIBTOOL_MODE_LINK} ${PURIFY} ${CC} ${CFLAGS} ${LDFLAGS} -o $@ \
 		dst_test.@O@ ${LIBS}
 
-t_dst@EXEEXT@: t_dst.@O@ ${DEPLIBS} ${TLIB} randomfile
+t_dst@EXEEXT@: t_dst.@O@ ${DEPLIBS} ${TLIB}
 	${LIBTOOL_MODE_LINK} ${PURIFY} ${CC} ${CFLAGS} ${LDFLAGS} -o $@ \
 		t_dst.@O@ ${TLIB} ${LIBS}
 
@@ -58,7 +58,6 @@
 		gsstest.@O@ ${LIBS}
 
 test: t_dst@EXEEXT@ randomfile
-	../genrandom@EXEEXT@ 100 randomfile
 	-@ ./t_dst@EXEEXT@ -q 1800 -a
 
 randomfile:
