--- a/bin/dig/nslookup.c	Fri Dec 21 12:02:39 2012
+++ b/bin/dig/nslookup.c	Fri Dec 21 12:03:43 2012
@@ -57,7 +57,6 @@
 static char defclass[MXRD] = "IN";
 static char deftype[MXRD] = "A";
 static isc_event_t *global_event = NULL;
-static int query_error = 1, print_error = 0;
 
 static char domainopt[DNS_NAME_MAXTEXT];
 
@@ -407,9 +406,6 @@
 printmessage(dig_query_t *query, dns_message_t *msg, isc_boolean_t headers) {
 	char servtext[ISC_SOCKADDR_FORMATSIZE];
 
-	/* I've we've gotten this far, we've reached a server. */
-	query_error = 0;
-
 	debug("printmessage()");
 
 	isc_sockaddr_format(&query->sockaddr, servtext, sizeof(servtext));
@@ -437,9 +433,6 @@
 		       (msg->rcode != dns_rcode_nxdomain) ? nametext :
 		       query->lookup->textname, rcode_totext(msg->rcode));
 		debug("returning with rcode == 0");
-
-		/* the lookup failed */
-		print_error |= 1;
 		return (ISC_R_SUCCESS);
 	}
 
@@ -910,5 +903,5 @@
 	destroy_libs();
 	isc_app_finish();
 
-	return (query_error | print_error);
+	return (0);
 }
