It is hard to normalize pytest-xdist test results

--- attrs-22.2.0/tox.ini.orig
+++ attrs-22.2.0/tox.ini
@@ -26,13 +26,13 @@
 
 [testenv]
 extras = tests
-commands = python -m pytest {posargs:-n auto}
+commands = python -m pytest {posargs}
 
 
 [testenv:py36]
 extras = cov
 setenv = COVERAGE_PROCESS_START={toxinidir}/pyproject.toml
-commands = coverage run -m pytest {posargs:-n auto}
+commands = coverage run -m pytest {posargs}
 
 
 [testenv:py310]
