--- attrs-25.4.0/tox.ini.orig
+++ attrs-25.4.0/tox.ini
@@ -20,7 +20,6 @@
 
 
 [testenv]
-runner = uv-venv-lock-runner
 package = wheel
 wheel_build_env = .pkg
 dependency_groups =
@@ -79,7 +78,6 @@
 
 
 [testenv:docs-{build,doctests,linkcheck}]
-runner = uv-venv-lock-runner
 # Keep base_python in-sync with .readthedocs.yaml.
 base_python = py313
 dependency_groups = docs
@@ -102,7 +100,6 @@
         docs
 
 [testenv:docs-sponsors]
-runner = uv-venv-runner
 skip_install = true
 description = Ensure sponsor logos are up to date.
 deps = cogapp
@@ -110,7 +107,6 @@
 
 
 [testenv:pre-commit]
-runner = uv-venv-runner
 skip_install = true
 deps = pre-commit-uv
 commands = pre-commit run --all-files
@@ -140,7 +136,6 @@
 
 
 [testenv:docset]
-runner = uv-venv-runner
 deps = doc2dash
 dependency_groups = docs
 allowlist_externals =
