--- attrs-26.1.0/tox.ini.orig
+++ attrs-26.1.0/tox.ini
@@ -19,7 +19,6 @@
 
 
 [testenv]
-runner = uv-venv-lock-runner
 package = wheel
 wheel_build_env = .pkg
 dependency_groups =
@@ -78,7 +77,6 @@
 
 
 [testenv:docs-{build,doctests,linkcheck}]
-runner = uv-venv-lock-runner
 # Keep base_python in-sync with .readthedocs.yaml.
 base_python = py313
 dependency_groups = docs
@@ -101,7 +99,6 @@
         docs
 
 [testenv:docs-sponsors]
-runner = uv-venv-runner
 skip_install = true
 description = Ensure sponsor logos are up to date.
 deps = cogapp
@@ -109,7 +106,6 @@
 
 
 [testenv:pre-commit]
-runner = uv-venv-runner
 skip_install = true
 deps = prek
 commands = prek run --all-files
@@ -139,7 +135,6 @@
 
 
 [testenv:docset]
-runner = uv-venv-runner
 deps = doc2dash
 dependency_groups = docs
 allowlist_externals =
