--- arj-3.10.22/gnu/configure.in	2005-06-21 22:27:20.000000000 +0400
+++ arj-3.10.22/gnu/configure.in.new	2014-07-23 18:08:47.613952260 +0400
@@ -12,7 +12,7 @@
 AC_PROG_INSTALL([e:/os2apps/autoconf/install-sh])
 
 dnl Configuration parameters
-AC_CONFIG_HEADER([../c_defs.h:config.h.in])
+AC_CONFIG_HEADER([c_defs.h:config.h.in])
 
 dnl Checks for the canonical system name
 AC_CANONICAL_HOST
@@ -198,5 +198,5 @@
 AC_SUBST(REGISTER)
 AC_SUBST(TOLERANT_FNMATCH)
 
-AC_CONFIG_FILES([../GNUmakefile:makefile.in])
+AC_CONFIG_FILES([GNUmakefile:makefile.in])
 AC_OUTPUT
