diff -ur hplip-3.19.5.orig/prnt/hpcups/HPCupsFilter.cpp hplip-3.19.5/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.19.5.orig/prnt/hpcups/HPCupsFilter.cpp	2019-04-08 06:04:41.000000000 +0000
+++ hplip-3.19.5/prnt/hpcups/HPCupsFilter.cpp	2019-05-21 19:42:07.984879710 +0000
@@ -39,6 +39,8 @@
 #include <time.h>
 #include "utils.h"
 
+extern "C" int getdomainname(char*, int);
+
 #define HP_FILE_VERSION_STR    "03.09.08.0"
 
 static HPCupsFilter    filter;
@@ -636,7 +638,7 @@
     char hpPreProcessedRasterFile[MAX_FILE_PATH_LEN]; //temp file needed to store raster data with swaped pages.
 
 
-    sprintf(hpPreProcessedRasterFile, "%s/hp_%s_cups_SwapedPagesXXXXXX",CUPS_TMP_DIR, m_JA.user_name);
+    snprintf(hpPreProcessedRasterFile, sizeof(hpPreProcessedRasterFile), "%s/hp_%s_cups_SwapedPagesXXXXXX",CUPS_TMP_DIR, m_JA.user_name);
     image_processor_t* imageProcessor = imageProcessorCreate();
 
     while (cupsRasterReadHeader2(cups_raster, &cups_header))
diff -ur hplip-3.19.5.orig/common/utils.h hplip-3.19.5/common/utils.h
--- hplip-3.19.5.orig/common/utils.h	2019-04-08 06:03:26.000000000 +0000
+++ hplip-3.19.5/common/utils.h	2019-05-21 19:41:20.327758369 +0000
@@ -26,7 +26,8 @@
 #define SCAN_PLUGIN_ORBLITE "bb_orblite.so"
 
 #define HPLIP_PLUGIN_STATE  "/var/lib/hp/hplip.state"
-#define CUPS_TMP_DIR   getenv("TMPDIR") ? : getenv("HOME") ?:"/tmp"
+//#define CUPS_TMP_DIR   getenv("TMPDIR") ? : getenv("HOME") ?:"/tmp"
+#define CUPS_TMP_DIR	"/tmp"
 
 enum UTILS_CONF_RESULT
 {
