https://github.com/pasteorg/paste/issues/99

--- Paste-3.10.0/tests/__init__.py.orig
+++ Paste-3.10.0/tests/__init__.py
@@ -1 +1,3 @@
 """Tests for Paste"""
+
+import pkg_resources
--- Paste-3.10.0/tests/test_cgiapp.py.orig
+++ Paste-3.10.0/tests/test_cgiapp.py
@@ -42,6 +42,7 @@
     assert res.full_status == '200 Okay'
     assert 'This is the body' in res
 
+@pytest.mark.skip(reason="Always fails: https://github.com/pasteorg/paste/issues/99")
 def test_form():
     app = TestApp(CGIApplication({}, script='form.cgi', path=[data_dir]))
     res = app.post('', params={'name': b'joe'},
