https://github.com/fedora-python/tox-current-env/pull/54

--- tox-current-env-0.0.12/tests/test_integration_tox3.py.orig
+++ tox-current-env-0.0.12/tests/test_integration_tox3.py
@@ -518,7 +518,7 @@
 
     # default tox produces output sorted by package names
     assert packages == sorted(
-        packages, key=lambda p: p.partition("==")[0].partition(" @ ")[0].lower()
+        packages, key=lambda p: p.partition("==")[0].partition(" @ ")[0]
     )
 
     # without a flag, the output must match tox defaults
