It is hard to normalize pytest-xdist test results

--- pytest-mypy-0.10.3/tox.ini.orig
+++ pytest-mypy-0.10.3/tox.ini
@@ -83,7 +83,7 @@
     pytest-randomly ~= 3.4
     pytest-xdist ~= 1.34
 
-commands = pytest -p no:mypy {posargs:--cov pytest_mypy --cov-fail-under 100 --cov-report term-missing -n auto}
+commands = pytest -p no:mypy {posargs:--cov pytest_mypy --cov-fail-under 100 --cov-report term-missing}
 
 [pytest]
 testpaths = tests
