--- pytest_mypy-1.0.1/tox.ini.orig
+++ pytest_mypy-1.0.1/tox.ini
@@ -27,14 +27,14 @@
     pytest7.0: pytest ~= 7.0.0
     pytest7.x: pytest ~= 7.0
     pytest8.0: pytest ~= 8.0.0
-    pytest8.x: pytest ~= 8.0
+    pytest ~= 8.0
     mypy1.0: mypy ~= 1.0.0
-    mypy1.x: mypy ~= 1.0
+    mypy ~= 1.0
     xdist1.x: pytest-xdist ~= 1.0
     xdist2.0: pytest-xdist ~= 2.0.0
     xdist2.x: pytest-xdist ~= 2.0
     xdist3.0: pytest-xdist ~= 3.0.0
-    xdist3.x: pytest-xdist ~= 3.0
+    pytest-xdist ~= 3.0
 
     packaging ~= 21.3
     pytest-cov ~= 4.1.0
