--- ptlib-2.16.2/make/my_macros.m4	2023-10-13 20:01:11.131817824 +0200
+++ ptlib-2.16.2/make/my_macros.m4.new	2023-10-13 20:06:47.829066845 +0200
@@ -433,7 +433,7 @@
 
 AC_CHECK_TOOL(STRIP, strip)
 AC_CHECK_TOOL(OBJCOPY, objcopy)   dnl GNU (Linux)
-AC_CHECK_TOOL(DSYMUTIL, dsymutil) dnl clang (Mac-OS)
+dnl AC_CHECK_TOOL(DSYMUTIL, dsymutil) dnl clang (Mac-OS)
 
 dnl Integer sizes, also defines HAVE_STDINT_H and HAVE_INTTYPES_H
 AC_CHECK_SIZEOF(int)
--- ptlib-2.16.2/make/ptlib_config.mak.in	2016-05-12 11:53:04.000000000 +0200
+++ ptlib-2.16.2/make/ptlib_config.mak.in.new	2023-10-13 20:18:08.138103288 +0200
@@ -66,7 +66,7 @@
 RANLIB  := @RANLIB@
 STRIP   := @STRIP@
 OBJCOPY := @OBJCOPY@
-DSYMUTIL:= @DSYMUTIL@
+#DSYMUTIL:= @DSYMUTIL@
 LN_S    := @LN_S@
 MKDIR_P := @MKDIR_P@
 BISON   := @BISON@
