diff -Nru dbus-test-runner-19.04.0.orig/tests/Makefile.am dbus-test-runner-19.04.0/tests/Makefile.am
--- dbus-test-runner-19.04.0.orig/tests/Makefile.am	2019-01-14 23:47:10.000000000 +0000
+++ dbus-test-runner-19.04.0/tests/Makefile.am	2023-01-08 09:43:23.649644571 +0000
@@ -159,9 +159,9 @@
 	@echo "#!/bin/sh -e" > $@
 	@echo "$(DBUS_RUNNER) --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat1\" --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat2\" > testcat.output.txt" >> $@
 	@echo "echo Finding cat1 data" >> $@
-	@echo "grep ^cat1: testcat.output.txt | tail -n +2 | head -n -1 > testcat.output.cat1.txt" >> $@
+	@echo "grep ^cat1: testcat.output.txt | tail -n +2 | ghead -n -1 > testcat.output.cat1.txt" >> $@
 	@echo "echo Finding cat2 data" >> $@
-	@echo "grep ^cat2: testcat.output.txt | tail -n +2 | head -n -1 > testcat.output.cat2.txt" >> $@
+	@echo "grep ^cat2: testcat.output.txt | tail -n +2 | ghead -n -1 > testcat.output.cat2.txt" >> $@
 	@echo "echo Filtering cat1 data" >> $@
 	@echo "sed -e s/cat1:\\ //g testcat.output.cat1.txt > testcat.output.cat1.filtered.txt" >> $@
 	@echo "echo Filtering cat2 data" >> $@
