--- MediaInfoLib-24.11/Project/GNU/Library/test/Makefile.am.orig	2024-11-07 19:46:33.000000000 +0100
+++ MediaInfoLib-24.11/Project/GNU/Library/test/Makefile.am	2024-11-17 18:53:17.766111739 +0100
@@ -16,6 +16,6 @@
 mil_analyze_SOURCES = \
     Source/analyze.cpp
 
-mil_analyze_CPPFLAGS = -I../../../../Source -I../../../../../ZenLib/Source
+mil_analyze_CPPFLAGS = -I$(VPATH)/../../../../Source -I../../../../../ZenLib/Source
 
 mil_analyze_LDFLAGS = ../libmediainfo.la
