diff -urN sound-juicer-2.28.2/src/sj-main.c ../SUNWgnome-cd-2.30.0.hacked/sound-juicer-2.28.2/src/sj-main.c
--- sound-juicer-2.28.2/src/sj-main.c	2010-04-16 16:13:55.490312130 +0100
+++ ../SUNWgnome-cd-2.30.0.hacked/sound-juicer-2.28.2/src/sj-main.c	2010-04-16 16:15:15.146658849 +0100
@@ -215,7 +215,7 @@
   /* first make sure we're not playing */
   stop_playback ();
 
-  brasero_drive_eject (drive, FALSE, NULL);
+  brasero_drive_eject (drive, TRUE, NULL);
 }
 
 gboolean on_delete_event (GtkWidget *widget, GdkEvent *event, gpointer user_data)
