--- apache-tomcat-6.0.18-src/bin/catalina.sh.orig	Fri Dec  5 06:54:48 2008
+++ apache-tomcat-6.0.18-src/bin/catalina.sh	Fri Dec  5 07:06:15 2008
@@ -78,6 +78,7 @@
 CYGWIN*) cygwin=true;;
 OS400*) os400=true;;
 Darwin*) darwin=true;;
+SunOS*) solaris=true;;
 esac
 
 # resolve links - $0 may be a softlink
@@ -127,6 +128,11 @@
   export QIBM_MULTI_THREADED=Y
 fi
 
+# For Solaris
+if $solaris; then
+  [ -z "$JAVA_HOME" ] && JAVA_HOME=/usr/java
+fi
+
 # Get standard Java environment variables
 if $os400; then
   # -r will Only work on the os400 if the files are:
--- apache-tomcat-6.0.18-src/bin/tool-wrapper.sh.orig	Fri Dec  5 07:01:29 2008
+++ apache-tomcat-6.0.18-src/bin/tool-wrapper.sh	Fri Dec  5 07:05:52 2008
@@ -35,8 +35,10 @@
 
 # OS specific support.  $var _must_ be set to either true or false.
 cygwin=false
+solaris=false
 case "`uname`" in
 CYGWIN*) cygwin=true;;
+SunOS*) solaris=true;;
 esac
 
 # resolve links - $0 may be a softlink
@@ -66,6 +68,11 @@
   [ -n "$CLASSPATH" ] && CLASSPATH=`cygpath --path --unix "$CLASSPATH"`
 fi
 
+# For Solaris
+if $solaris; then
+  [ -z "$JAVA_HOME" ] && JAVA_HOME=/usr/java
+fi
+
 # Get standard Java environment variables
 if [ -r "$CATALINA_HOME"/bin/setclasspath.sh ]; then
   BASEDIR="$CATALINA_HOME"
