--- apache-tomcat-6.0.37-src/bin/catalina.sh	Wed Jun 12 02:20:24 2013
+++ apache-tomcat-6.0.37-src/bin/catalina.sh	Wed Jun 12 02:20:24 2013
@@ -287,6 +287,7 @@
         -Dcatalina.base="$CATALINA_BASE" \
         -Dcatalina.home="$CATALINA_HOME" \
         -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+        -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
         org.apache.catalina.startup.Bootstrap "$@" start
     else
       exec "$_RUNJDB" "$LOGGING_CONFIG" $LOGGING_MANAGER $JAVA_OPTS $CATALINA_OPTS \
@@ -295,6 +296,7 @@
         -Dcatalina.base="$CATALINA_BASE" \
         -Dcatalina.home="$CATALINA_HOME" \
         -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+        -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
         org.apache.catalina.startup.Bootstrap "$@" start
     fi
   fi
@@ -314,6 +316,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start
   else
     exec "$_RUNJAVA" "$LOGGING_CONFIG" $LOGGING_MANAGER $JAVA_OPTS $CATALINA_OPTS \
@@ -321,6 +324,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start
   fi
 
@@ -378,6 +382,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start \
       >> "$CATALINA_OUT" 2>&1 &
 
@@ -387,6 +392,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start \
       >> "$CATALINA_OUT" 2>&1 &
 
@@ -437,6 +443,7 @@
     -Dcatalina.base="$CATALINA_BASE" \
     -Dcatalina.home="$CATALINA_HOME" \
     -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+    -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
     org.apache.catalina.startup.Bootstrap "$@" stop
 
   if [ ! -z "$CATALINA_PID" ]; then
