--- goaccess-1.0/src/sha1.h.orig	2016-06-09 19:20:03.424726826 +0300
+++ goaccess-1.0/src/sha1.h	2016-06-09 19:20:16.984277660 +0300
@@ -3,6 +3,13 @@

 #include <sys/types.h>

+#if defined(__sun__)
+#include <stdint.h>
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+#endif
+
 // From http://www.mirrors.wiretapped.net/security/cryptography/hashes/sha1/sha1.c

 typedef struct
