--- goaccess-1.0/src/websocket.h	2016-06-09 19:07:57.569813483 +0300
+++ goaccess-1.0/src/websocket.h.orig	2016-06-09 19:07:42.420698499 +0300
@@ -34,7 +34,7 @@
 #include <limits.h>
 #include <sys/select.h>

-#if defined(__linux__) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__CYGWIN__) || defined(__sun__)
 #  include <endian.h>
 #elif defined(__FreeBSD__) || defined(__NetBSD__)
 #  include <sys/endian.h>
