--- ./sysdep.h.~1~	2008-11-19 21:36:12.000000000 +0100
+++ ./sysdep.h	2016-11-17 23:20:51.470704173 +0100
@@ -123,8 +123,10 @@
 
 /* where is this defined? */
 #include <sys/socket.h>
+#ifndef HAVE_INET_NTOP
 const char *inet_ntop(int af, const void *src, char *dst, size_t cnt);
 #endif
+#endif
 /***************************************************************************/
 #if defined (__SKYOS__)
 #define HAVE_UNSETENV  1
