--- ./Makefile.~1~	2016-11-18 00:11:03.951763927 +0100
+++ ./Makefile	2016-11-18 00:11:27.320483682 +0100
@@ -47,8 +47,8 @@
 
 # Comment this in to obtain a binary with certificate support which is
 # GPL incompliant though.
-#OPENSSL_GPL_VIOLATION = -DOPENSSL_GPL_VIOLATION
-#OPENSSLLIBS = -lcrypto
+OPENSSL_GPL_VIOLATION = -DOPENSSL_GPL_VIOLATION
+OPENSSLLIBS = -lcrypto
 
 CC=gcc
 CFLAGS ?= -O3 -g
