--- mariadb-10.1.17/sql/item_cmpfunc.h.1	2016-08-29 11:38:54.000000000 -0300
+++ mariadb-10.1.17/sql/item_cmpfunc.h		2016-10-12 00:49:39.126614652 -0300
@@ -26,7 +26,7 @@
 #include "thr_malloc.h"                         /* sql_calloc */
 #include "item_func.h"             /* Item_int_func, Item_bool_func */
 #define PCRE_STATIC 1             /* Important on Windows */
-#include "pcre.h"                 /* pcre header file */
+#include "pcre/pcre.h"                 /* pcre header file */
 
 extern Item_result item_cmp_type(Item_result a,Item_result b);
 inline Item_result item_cmp_type(const Item *a, const Item *b)
--- mariadb-10.1.17/client/mysqltest.cc.1	2016-08-29 11:38:51.000000000 -0300
+++ mariadb-10.1.17/client/mysqltest.cc		2016-10-12 01:16:10.857059234 -0300
@@ -45,7 +45,7 @@
 #include <stdarg.h>
 #include <violite.h>
 #define PCRE_STATIC 1  /* Important on Windows */
-#include "pcreposix.h" /* pcreposix regex library */
+#include "pcre/pcreposix.h" /* pcreposix regex library */
 #ifdef HAVE_SYS_WAIT_H
 #include <sys/wait.h>
 #endif
