Fix for compiling.  This change can be sent upstream

--- libproxy-master/libproxy/modules/config_gnome3.cpp	Sat Sep  7 04:14:17 2013
+++ libproxy-master/libproxy/modules/config_gnome3.cpp	Mon May 11 17:56:50 2015
@@ -25,6 +25,7 @@
 #include <sys/stat.h>     // For stat()
 #include <unistd.h>       // For pipe(), close(), vfork(), dup(), execl(), _exit()
 #include <signal.h>       // For kill()
+#include <string.h>       // For memset()
 
 #include "../extension_config.hpp"
 using namespace libproxy;
