--- zabbix-6.2.7/src/zabbix_agent/Makefile.am.bak	Mon Feb 27 01:51:42 2023
+++ zabbix-6.2.7/src/zabbix_agent/Makefile.am	Mon Feb 27 03:33:08 2023
@@ -84,4 +84,4 @@
 install-data-hook:
 	$(MKDIR_P) "$(DESTDIR)$(AGENT_CONFIG_FILE).d"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "../../conf/zabbix_agentd.conf" "$(DESTDIR)$(AGENT_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "$(top_srcdir)/conf/zabbix_agentd.conf" "$(DESTDIR)$(AGENT_CONFIG_FILE)"
