We have no pytest-fixture-classes packaged.

--- pytest_lazy_fixtures-1.4.1/tests/tests.py.orig
+++ pytest_lazy_fixtures-1.4.1/tests/tests.py
@@ -3,7 +3,6 @@
 from operator import attrgetter
 
 import pytest
-from pytest_fixture_classes import fixture_class
 
 from pytest_lazy_fixtures import lf, lfc
 from pytest_lazy_fixtures.lazy_fixture import LazyFixtureWrapper
@@ -237,15 +236,6 @@
     return []
 
 
-@fixture_class(name="some")
-class Fixture:
-    data: list[str]
-
-    def __call__(self) -> list[str]:
-        self.data.append("some_value")
-        return self.data
-
-
 @pytest.mark.parametrize("value", [[lf("some")]])
 @pytest.mark.parametrize("value2", [True, False])
 def test_fixture_classes(value, value2):
