--- ./src/lib/MSPUBMetaData.cpp.orig	2015-07-29 03:17:19.837887752 +0200
+++ ./src/lib/MSPUBMetaData.cpp	2015-07-29 03:18:01.598301831 +0200
@@ -194,7 +194,7 @@
   uint32_t firstDirSectorLocation = readU32(input);
 
   // Seek to the Root Directory Entry
-  size_t sectorSize = pow(2, sectorShift);
+  size_t sectorSize = std::pow(2, sectorShift);
   input->seek((firstDirSectorLocation + 1) * sectorSize, librevenge::RVNG_SEEK_SET);
   // DirectoryEntryName: 64 bytes
   // DirectoryEntryNameLength: 2 bytes
