diff -Nru libgdiplus-6.1.orig/configure libgdiplus-6.1/configure
--- libgdiplus-6.1.orig/configure	2021-11-03 20:53:43.000000000 +0000
+++ libgdiplus-6.1/configure	2022-06-05 15:08:26.410072725 +0000
@@ -16676,13 +16676,8 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: *** cmake not found, won't build googletest-based tests." >&5
 $as_echo "$as_me: WARNING: *** cmake not found, won't build googletest-based tests." >&2;}
 fi
- if test x$CMAKE != "xno"; then
-  HAVE_CMAKE_TRUE=
-  HAVE_CMAKE_FALSE='#'
-else
-  HAVE_CMAKE_TRUE='#'
-  HAVE_CMAKE_FALSE=
-fi
+HAVE_CMAKE_TRUE='#'
+HAVE_CMAKE_FALSE=
 
 
 CAIRO_REQUIRED_VERSION="1.6.4"
