--- libetpan-1.7/rules.mk.orig	2016-04-26 10:24:44.995205817 +0300
+++ libetpan-1.7/rules.mk	2016-04-26 10:24:17.333337392 +0300
@@ -135,9 +135,9 @@
 	           echo "$(LN_S) -f ../../$(subdir)/$${hdr} ."; \
 	           $(LN_S) -f ../../$(subdir)/$${hdr} .; \
              else \
-	           echo "$(LN_S) -f ../../$(subdir)/$(srcdir)/$${hdr} ."; \
-	           $(LN_S) -f ../../$(subdir)/$(srcdir)/$${hdr} .; \
+	           echo "$(LN_S) -f $(srcdir)/$${hdr} ."; \
+	           $(LN_S) -f $(srcdir)/$${hdr} .; \
              fi; \
            fi; \
 	     done; \
 	fi
