--- stellarium-25.1/src/external/CMakeLists.txt.orig	2025-03-20 20:14:27.000000000 +0100
+++ stellarium-25.1/src/external/CMakeLists.txt	2025-04-26 11:05:16.830723940 +0200
@@ -111,6 +111,13 @@
 		VERSION 2.1.2.1
 		DOWNLOAD_ONLY YES)
 
+        # remove missing definitions
+	file(READ ${indiclient_SOURCE_DIR}/libs/indicore/indicom.c INDICOM_C)
+	string(REPLACE "case 576000: bps = B576000; break;" "" INDICOM_C "${INDICOM_C}")
+	string(REPLACE "case 921600: bps = B921600; break;" "" INDICOM_C "${INDICOM_C}")
+	file(WRITE ${indiclient_SOURCE_DIR}/libs/indicore/indicom.c.new "${INDICOM_C}")
+	configure_file(${indiclient_SOURCE_DIR}/libs/indicore/indicom.c.new ${indiclient_SOURCE_DIR}/libs/indicore/indicom.c COPYONLY)
+	
         # First, fix a missing include
 	file(READ ${indiclient_SOURCE_DIR}/libs/indicore/indidevapi.h INDIDEVAPI_H)
 	string(REGEX REPLACE "#include .lilxml.h." "#include \"lilxml.h\"\n#include <stdarg.h>" INDIDEVAPI_H "${INDIDEVAPI_H}")
@@ -119,7 +126,7 @@
 	
 	# Fix missing include
 	file(READ ${indiclient_SOURCE_DIR}/libs/sockets/select.h INDISOCKETSELECT_H)
-	string(REGEX REPLACE "#include .sys/socket.h." "#include <sys/socket.h>\n#include <sys/select.h>" INDISOCKETSELECT_H "${INDISOCKETSELECT_H}")
+	string(REGEX REPLACE "#include .sys/socket.h." "#include <cstring>\n#include <sys/socket.h>\n#include <sys/select.h>" INDISOCKETSELECT_H "${INDISOCKETSELECT_H}")
 	file(WRITE ${indiclient_SOURCE_DIR}/libs/sockets/select.h.new "${INDISOCKETSELECT_H}")
 	configure_file(${indiclient_SOURCE_DIR}/libs/sockets/select.h.new ${indiclient_SOURCE_DIR}/libs/sockets/select.h COPYONLY)
 
