--- stellarium-25.1/src/core/modules/StarMgr.cpp.orig	2025-03-23 06:15:57.000000000 +0100
+++ stellarium-25.1/src/core/modules/StarMgr.cpp	2025-04-26 11:11:18.236401127 +0200
@@ -1582,7 +1582,7 @@
 	int lv12_pix = source_id / 34359738368;
 	Vec3d v;
 	StelObjectP so;
-	healpix_pix2vec(pow(2, 12), lv12_pix, v.v);  // search which pixel the source is in and turn to coordinates
+	healpix_pix2vec(pow(static_cast<double>(2), 12), lv12_pix, v.v);  // search which pixel the source is in and turn to coordinates
 	Vec3f vf = v.toVec3f();
 
 	for (const auto* z : gridLevels)
--- stellarium-25.2/src/core/StelSkyCultureSkyPartition.cpp.orig	2025-06-22 13:52:45.000000000 +0200
+++ stellarium-25.2/src/core/StelSkyCultureSkyPartition.cpp	2025-06-25 11:46:41.768381716 +0200
@@ -484,7 +484,7 @@
 		else if (partitions.at(0)==27)
 		{
 			int padaInSign=int(floor((degreeInSign/widthOfSign)*4.));
-			return QString("%1: %2").arg(symbols.at(sign), QString::number(int(floor(padaInSign+1))));
+			return QString("%1: %2").arg(symbols.at(sign), QString::number(int(floor(static_cast<double>(padaInSign+1)))));
 		}
 		else
 		{
--- stellarium-25.2/src/core/modules/ZoneArray.cpp.orig	2025-06-22 13:52:45.000000000 +0200
+++ stellarium-25.2/src/core/modules/ZoneArray.cpp	2025-06-25 11:52:21.224696246 +0200
@@ -664,7 +664,7 @@
                                                   double &      RV) const
 {
    // loop through each zone in the level which is 20 * 4 ** level + 1 as index
-   for (int i = 0; i < 20 * pow(4, (level)) + 1; i++) {
+   for (int i = 0; i < 20 * pow(static_cast<double>(4), (level)) + 1; i++) {
       // get the zone data
       const SpecialZoneData<Star> * const z = getZones() + i;
       // loop through the stars in the zone
