--- doxygen-1.5.7.1/doc/Doxyfile	Sat Aug 16 10:07:19 2008
+++ doxygen-1.5.7.1/doc/Doxyfile.new	Fri Apr  1 11:21:12 2011
@@ -22,7 +22,7 @@
 EXTRACT_ALL       = NO
 EXTRACT_PRIVATE   = NO
 GENERATE_MAN      = NO
-GENERATE_LATEX    = YES
+GENERATE_LATEX    = NO
 GENERATE_HTML     = YES
 GENERATE_HTMLHELP = YES
 GENERATE_RTF      = NO
--- doxygen-1.5.7.1/tmake/lib/solaris-cc/tmake.conf	Fri Jul 21 12:54:14 2006
+++ doxygen-1.5.7.1/tmake/lib/solaris-cc/tmake.conf.new	Fri Apr  1 11:22:45 2011
@@ -17,7 +17,7 @@
 TMAKE_CFLAGS_YACC	=
 
 TMAKE_CXX		= CC
-TMAKE_CXXFLAGS		= $$TMAKE_CFLAGS
+TMAKE_CXXFLAGS		= -norunpath -D__USE_LEGACY_PROTOTYPES__
 TMAKE_CXXFLAGS_WARN_ON	= $$TMAKE_CFLAGS_WARN_ON
 TMAKE_CXXFLAGS_WARN_OFF	= $$TMAKE_CFLAGS_WARN_OFF
 TMAKE_CXXFLAGS_RELEASE	= -O2
@@ -37,7 +37,7 @@
 TMAKE_LINK		= CC
 # Jan Wortelboer <janw@wins.uva.nl> suggests avoiding $LD_LIBRARY_PATH:
 TMAKE_LINK_SHLIB	= CC -R$(QTDIR)/lib:/usr/openwin/lib
-TMAKE_LFLAGS		=
+TMAKE_LFLAGS		= -norunpath
 TMAKE_LFLAGS_RELEASE	=
 TMAKE_LFLAGS_DEBUG	=
 TMAKE_LFLAGS_SHLIB	= -G -h $(TARGET1)
--- doxygen-1.7.6.1/tmake/lib/solaris-g++/tmake.conf	2013-05-28 23:00:14.511817711 +0400
+++ doxygen-1.7.6.1/tmake/lib/solaris-g++/tmake.conf.new	2013-05-28 23:00:04.887062499 +0400
@@ -43,7 +43,7 @@
 TMAKE_LFLAGS_SHLIB	= -shared -h $(TARGET1)
 TMAKE_LFLAGS_SONAME	=
 
-TMAKE_LIBS		= -liconv
+TMAKE_LIBS		= 
 TMAKE_LIBS_X11		= -lXext -lX11 -lm
 TMAKE_LIBS_QT		= -lqt
 TMAKE_LIBS_QT_MT	= -lqt-mt
