--- mariadb-10.3.8/libmariadb/plugins/auth/gssapi_errmsg.c	2018-07-02 09:34:09.000000000 +0000
+++ mariadb-10.3.8/libmariadb/plugins/auth/gssapi_errmsg.c.new	2018-07-13 23:56:25.697238336 +0000
@@ -29,7 +29,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifdef  __FreeBSD__
 #include <gssapi/gssapi.h>
 #else
-#include <gssapi.h>
+#include <gssapi/gssapi.h>
 #endif
 #include <string.h>
 void gssapi_errmsg(OM_uint32 major, OM_uint32 minor, char *buf, size_t size)
--- mariadb-10.3.8/plugin/auth_gssapi/gssapi_errmsg.cc  2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/plugin/auth_gssapi/gssapi_errmsg.cc.new      2018-07-14 08:35:14.914309795 +0000
@@ -26,7 +26,7 @@ ARISING IN ANY WAY OUT OF THE USE OF THI
 POSSIBILITY OF SUCH DAMAGE.
 */
 
-#include <gssapi.h>
+#include <gssapi/gssapi.h>
 #include <string.h>
 
 void gssapi_errmsg(OM_uint32 major, OM_uint32 minor, char *buf, size_t size)
