--- mariadb-10.3.8/mysys/mf_iocache.c	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/mysys/mf_iocache.c.new	2018-07-14 00:02:58.493727848 +0000
@@ -1589,7 +1589,7 @@ int _my_b_async_read(IO_CACHE *info, uch
     Buffer+=length;
     Count-=length;
     left_length+=length;
-    info->read_end=info->rc_pos+read_length;
+    info->read_end=info->read_pos+read_length;
     info->read_pos+=length;
   }
   else
