--- mariadb-10.3.8/sql/sql_class.h	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/sql_class.h.new	2018-08-07 06:58:32.140512292 +0000
@@ -6391,7 +6391,7 @@ inline int handler::ha_read_first_row(uc
 inline int handler::ha_write_tmp_row(uchar *buf)
 {
   int error;
-  MYSQL_INSERT_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_INSERT_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   increment_statistics(&SSV::ha_tmp_write_count);
   TABLE_IO_WAIT(tracker, m_psi, PSI_TABLE_WRITE_ROW, MAX_KEY, 0,
                 { error= write_row(buf); })
@@ -6402,7 +6402,7 @@ inline int handler::ha_write_tmp_row(uch
 inline int handler::ha_delete_tmp_row(uchar *buf)
 {
   int error;
-  MYSQL_DELETE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_DELETE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   increment_statistics(&SSV::ha_tmp_delete_count);
   TABLE_IO_WAIT(tracker, m_psi, PSI_TABLE_DELETE_ROW, MAX_KEY, 0,
                 { error= delete_row(buf); })
@@ -6413,7 +6413,7 @@ inline int handler::ha_delete_tmp_row(uc
 inline int handler::ha_update_tmp_row(const uchar *old_data, uchar *new_data)
 {
   int error;
-  MYSQL_UPDATE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_UPDATE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   increment_statistics(&SSV::ha_tmp_update_count);
   TABLE_IO_WAIT(tracker, m_psi, PSI_TABLE_UPDATE_ROW, active_index, 0,
                 { error= update_row(old_data, new_data);})
--- mariadb-10.3.8/sql/filesort.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/filesort.cc.new	2018-08-07 07:30:45.020793273 +0000
@@ -165,7 +165,7 @@ SORT_INFO *filesort(THD *thd, TABLE *tab
   SORT_INFO *sort;
   TABLE_LIST *tab= table->pos_in_table_list;
   Item_subselect *subselect= tab ? tab->containing_subselect() : 0;
-  MYSQL_FILESORT_START(table->s->db.str, table->s->table_name.str);
+  MYSQL_FILESORT_START((char*)table->s->db.str, (char*)table->s->table_name.str);
   DEBUG_SYNC(thd, "filesort_start");
 
   if (!(sort= new SORT_INFO))
--- mariadb-10.3.8/storage/csv/ha_tina.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/storage/csv/ha_tina.cc.new	2018-08-07 18:47:07.864867214 +0000
@@ -1226,7 +1226,7 @@ int ha_tina::rnd_next(uchar *buf)
 {
   int rc;
   DBUG_ENTER("ha_tina::rnd_next");
-  MYSQL_READ_ROW_START(table_share->db.str, table_share->table_name.str,
+  MYSQL_READ_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str,
                        TRUE);
 
   if (share->crashed)
@@ -1281,7 +1281,7 @@ int ha_tina::rnd_pos(uchar * buf, uchar
 {
   int rc;
   DBUG_ENTER("ha_tina::rnd_pos");
-  MYSQL_READ_ROW_START(table_share->db.str, table_share->table_name.str,
+  MYSQL_READ_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str,
                        FALSE);
   current_position= my_get_ptr(pos,ref_length);
   rc= find_current_row(buf);
--- mariadb-10.3.8/sql/handler.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/handler.cc.new	2018-08-07 19:04:42.437141903 +0000
@@ -6099,18 +6099,18 @@ int handler::ha_external_lock(THD *thd,
   {
     if (lock_type == F_RDLCK)
     {
-      MYSQL_HANDLER_RDLOCK_START(table_share->db.str,
-                                 table_share->table_name.str);
+      MYSQL_HANDLER_RDLOCK_START((char*)table_share->db.str,
+                                 (char*)table_share->table_name.str);
     }
     else if (lock_type == F_WRLCK)
     {
-      MYSQL_HANDLER_WRLOCK_START(table_share->db.str,
-                                 table_share->table_name.str);
+      MYSQL_HANDLER_WRLOCK_START((char*)table_share->db.str,
+                                 (char*)table_share->table_name.str);
     }
     else if (lock_type == F_UNLCK)
     {
-      MYSQL_HANDLER_UNLOCK_START(table_share->db.str,
-                                 table_share->table_name.str);
+      MYSQL_HANDLER_UNLOCK_START((char*)table_share->db.str,
+                                 (char*)table_share->table_name.str);
     }
   }
 
@@ -6189,7 +6189,7 @@ int handler::ha_write_row(uchar *buf)
   DBUG_ENTER("handler::ha_write_row");
   DEBUG_SYNC_C("ha_write_row_start");
 
-  MYSQL_INSERT_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_INSERT_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   mark_trx_read_write();
   increment_statistics(&SSV::ha_write_count);
 
@@ -6221,7 +6221,7 @@ int handler::ha_update_row(const uchar *
   DBUG_ASSERT(new_data == table->record[0]);
   DBUG_ASSERT(old_data == table->record[1]);
 
-  MYSQL_UPDATE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_UPDATE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   mark_trx_read_write();
   increment_statistics(&SSV::ha_update_count);
 
@@ -6277,7 +6277,7 @@ int handler::ha_delete_row(const uchar *
   DBUG_ASSERT(buf == table->record[0] ||
               buf == table->record[1]);
 
-  MYSQL_DELETE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_DELETE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   mark_trx_read_write();
   increment_statistics(&SSV::ha_delete_count);
 
@@ -6311,7 +6311,7 @@ int handler::ha_direct_update_rows(ha_ro
 {
   int error;
 
-  MYSQL_UPDATE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_UPDATE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   mark_trx_read_write();
 
   error = direct_update_rows(update_rows);
@@ -6338,7 +6338,7 @@ int handler::ha_direct_delete_rows(ha_ro
   /* Ensure we are not using binlog row */
   DBUG_ASSERT(!table->in_use->is_current_stmt_binlog_format_row());
 
-  MYSQL_DELETE_ROW_START(table_share->db.str, table_share->table_name.str);
+  MYSQL_DELETE_ROW_START((char*)table_share->db.str, (char*)table_share->table_name.str);
   mark_trx_read_write();
 
   error = direct_delete_rows(delete_rows);
--- mariadb-10.3.8/sql/sp_head.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/sp_head.cc.new	2018-08-07 19:27:05.342039650 +0000
@@ -3585,7 +3585,7 @@ sp_instr_stmt::exec_core(THD *thd, uint
 {
   MYSQL_QUERY_EXEC_START(thd->query(),
                          thd->thread_id,
-                         thd->get_db(),
+                         (char*)thd->get_db(),
                          &thd->security_ctx->priv_user[0],
                          (char *)thd->security_ctx->host_or_ip,
                          3);
--- mariadb-10.3.8/sql/sql_cursor.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/sql_cursor.cc.new	2018-08-07 19:31:10.441044534 +0000
@@ -132,7 +132,7 @@ int mysql_open_cursor(THD *thd, select_r
 
   MYSQL_QUERY_EXEC_START(thd->query(),
                          thd->thread_id,
-                         thd->get_db(),
+                         (char*)thd->get_db(),
                          &thd->security_ctx->priv_user[0],
                          (char *) thd->security_ctx->host_or_ip,
                          2);
--- mariadb-10.3.8/sql/sql_parse.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/sql_parse.cc.new	2018-08-07 19:38:07.511539303 +0000
@@ -1823,7 +1823,7 @@ bool dispatch_command(enum enum_server_c
     if (unlikely(alloc_query(thd, packet, packet_length)))
       break;					// fatal error is set
     MYSQL_QUERY_START(thd->query(), thd->thread_id,
-                      thd->get_db(),
+                      (char*)thd->get_db(),
                       &thd->security_ctx->priv_user[0],
                       (char *) thd->security_ctx->host_or_ip);
     char *packet_end= thd->query() + thd->query_length();
@@ -1901,7 +1901,7 @@ bool dispatch_command(enum enum_server_c
 
       /* DTRACE begin */
       MYSQL_QUERY_START(beginning_of_next_stmt, thd->thread_id,
-                        thd->get_db(),
+                        (char*)thd->get_db(),
                         &thd->security_ctx->priv_user[0],
                         (char *) thd->security_ctx->host_or_ip);
 
@@ -8065,7 +8065,7 @@ void mysql_parse(THD *thd, char *rawbuf,
           lex->set_trg_event_type_for_tables();
           MYSQL_QUERY_EXEC_START(thd->query(),
                                  thd->thread_id,
-                                 thd->get_db(),
+                                 (char*)thd->get_db(),
                                  &thd->security_ctx->priv_user[0],
                                  (char *) thd->security_ctx->host_or_ip,
                                  0);
--- mariadb-10.3.8/sql/sql_prepare.cc	2018-07-02 09:34:11.000000000 +0000
+++ mariadb-10.3.8/sql/sql_prepare.cc.new	2018-08-07 19:40:20.438131200 +0000
@@ -4754,7 +4754,7 @@ bool Prepared_statement::execute(String
       PSI_statement_locker *parent_locker;
       MYSQL_QUERY_EXEC_START(thd->query(),
                              thd->thread_id,
-                             thd->get_db(),
+                             (char*)thd->get_db(),
                              &thd->security_ctx->priv_user[0],
                              (char *) thd->security_ctx->host_or_ip,
                              1);
