diff --git a/libs/mpfi/mpfi-src/src/z_sub.c b/libs/mpfi/mpfi-src/src/z_sub.c
index 79be969..d815084 100644
--- a/libs/mpfi/mpfi-src/src/z_sub.c
+++ b/libs/mpfi/mpfi-src/src/z_sub.c
@@ -27,7 +27,7 @@ MA 02110-1301, USA. */
 #include "mpfi_config.h"
 
 #ifndef HAVE_MPFR_Z_SUB
-static int
+int
 mpfr_z_sub (mpfr_ptr x, mpz_srcptr z, mpfr_srcptr y, mpfr_rnd_t rnd)
 {
   /* mpfr_z_sub does not exist (at least up to version 3.0) */
