diff --git a/texk/texlive/linked_scripts/memoize/memoize-clean.py b/texk/texlive/linked_scripts/memoize/memoize-clean.py
index 69beeb0..1516963 100755
--- a/texk/texlive/linked_scripts/memoize/memoize-clean.py
+++ b/texk/texlive/linked_scripts/memoize/memoize-clean.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 # This file is a part of Memoize, a TeX package for externalization of
 # graphics and memoization of compilation results in general, available at
diff --git a/texk/texlive/linked_scripts/texliveonfly/texliveonfly.py b/texk/texlive/linked_scripts/texliveonfly/texliveonfly.py
index fa809ff..94ef19d 100755
--- a/texk/texlive/linked_scripts/texliveonfly/texliveonfly.py
+++ b/texk/texlive/linked_scripts/texliveonfly/texliveonfly.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 # default options; feel free to change!
 defaultCompiler = "pdflatex"
