From 6e354dd9471ff7acaa87b2031f797e7b8694745f Mon Sep 17 00:00:00 2001
From: Bert JW Regeer <bertjw@regeer.org>
Date: Tue, 21 Jan 2020 22:46:35 -0800
Subject: [PATCH] s/isAlive/is_alive/

---
 tests/conftest.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index e84fabe3..675c0247 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -60,7 +60,7 @@
             log.debug("shutting server down")
             server.shutdown()
             worker.join(1)
-            if worker.isAlive():
+            if worker.is_alive():
                 log.warning('worker is hanged')
             else:
                 log.debug("server stopped")
