--- icu/source/runConfigureICU.~1~	2015-04-27 12:26:16.553280192 +0300
+++ icu/source/runConfigureICU	2015-04-27 12:28:01.133138890 +0300
@@ -155,25 +155,18 @@
     Solaris)
         THE_OS=SOLARIS
         THE_COMP="Sun's CC"
-        CC=`which cc`; export CC
-        CXX=`which CC`; export CXX
         RELEASE_CFLAGS="-xO1 -xlibmil"
         RELEASE_CXXFLAGS="-O4 -xlibmil"
         ;;
     Solaris/GCC)
         THE_OS=SOLARIS
         THE_COMP="the GNU C++"
-        CC=gcc; export CC
-        CXX=g++; export CXX
         RELEASE_CFLAGS=-O1
         RELEASE_CXXFLAGS=-O2
         ;;
     SolarisX86)
         THE_OS="SOLARIS X86"
         THE_COMP="Sun's CC"
-        CC=`which cc`; export CC
-        CXX=`which CC`; export CXX
-        LDFLAGS="${LDFLAGS} -lCrun";export LDFLAGS
         RELEASE_CFLAGS=-xO3
         RELEASE_CXXFLAGS=-O3
         ;;
