--- icu/source/i18n/decNumber.h	Fri Feb 26 03:29:00 2010
+++ icu/source/i18n/decNumber.h	Wed Apr 27 12:03:11 2011
@@ -84,7 +84,8 @@
                          /* range: -1999999997 through 999999999      */
     uint8_t bits;        /* Indicator bits (see above)                */
                          /* Coefficient, from least significant unit  */
-    decNumberUnit lsu[DECNUMUNITS];
+    /*CR 7035907*/
+    decNumberUnit lsu[DECNUMUNITS+2];
     } decNumber;
 
   /* Notes:                                                           */
