--- icu/source/test/cintltst/ccapitst.c.orig    2020-03-19 17:34:21.597185785 +0000
+++ icu/source/test/cintltst/ccapitst.c 2020-03-19 17:34:53.672058002 +0000
@@ -2923,8 +2923,8 @@
         { 3, "CESU-8"},
         { 3, "SCSU"},
         { 4, "UTF-32"},
-        { 4, "UTF-7"},
-        { 4, "IMAP-mailbox-name"},
+        { 5, "UTF-7"},
+        { 8, "IMAP-mailbox-name"},
         { 4, "BOCU-1"},
 
         { 1, "windows-1256"},
