https://github.com/pytest-dev/pytest-freezer/pull/11

diff --git a/pyproject.toml b/pyproject.toml
index bab07e6..b046152 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,7 +22,7 @@ Homepage = "https://github.com/pytest-dev/pytest-freezer"
 [project.entry-points.pytest11]
 freezer = "pytest_freezer"
 
-[tool.hatch.build]
+[tool.hatch.build.targets.sdist]
 include = [
     "LICENSE",
     "README.rst",
