--- libesmtp-1.1.0/meson.build.orig	2021-06-04 18:18:50.000000000 +0200
+++ libesmtp-1.1.0/meson.build	2023-12-31 21:46:53.063547449 +0100
@@ -204,7 +204,8 @@
 endif
 
 mapfile = 'libesmtp.map'
-vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
+#vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
+vflag = ''
 
 lib = library('esmtp', sources,
 	      link_args : vflag, link_depends : mapfile,
