--- moosefs-3.0.116/mfsscripts/mfscgiserv.py.in.bak	Mon Jan  9 10:53:27 2023
+++ moosefs-3.0.116/mfsscripts/mfscgiserv.py.in	Mon Jan  9 10:53:40 2023
@@ -12,12 +12,8 @@
 import traceback
 import datetime
 import mimetypes
-try:
-	from urllib.parse import urlparse
-	from urllib.parse import urlunparse
-except ImportError:
-	from urlparse import urlparse
-	from urlparse import urlunparse
+from urllib.parse import urlparse
+from urllib.parse import urlunparse
 import socket
 import select
 import subprocess
