In-house patch to remove gzipping of man pages

--- rabbitmq-server-3.6.1/Makefile.orig  2016-03-31 14:49:40.661997085 -0700
+++ rabbitmq-server-3.6.1/Makefile    2016-03-31 14:55:24.094181414 -0700
@@ -308,8 +308,8 @@ install-man: manpages
 	for section in $$sections; do \
 		mkdir -p $(DESTDIR)$(MANDIR)/man$$section; \
 		for manpage in $(DEPS_DIR)/rabbit/docs/*.$$section; do \
-			gzip < $$manpage \
-			 > $(DESTDIR)$(MANDIR)/man$$section/$$(basename $$manpage).gz; \
+			cp $$manpage \
+			 $(DESTDIR)$(MANDIR)/man$$section/$$(basename $$manpage); \
 		done; \
 	done

