diff -Nru nheko-0.11.3.orig/CMakeLists.txt nheko-0.11.3/CMakeLists.txt
--- nheko-0.11.3.orig/CMakeLists.txt	2023-02-23 00:50:45.000000000 +0100
+++ nheko-0.11.3/CMakeLists.txt	2023-10-07 16:13:53.668729180 +0200
@@ -750,7 +750,9 @@
 	nlohmann_json::nlohmann_json
 	lmdbxx::lmdbxx
 	liblmdb::lmdb
-	SingleApplication::SingleApplication)
+	SingleApplication::SingleApplication
+	nsl
+	socket)
 
 if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16.0")
 	target_precompile_headers(nheko
