--- core-b14583ba37a6d7ce398ccd3cf339f954785b03d8/include/comphelper/errcode.hxx.~1~	2023-10-24 23:34:49.856823014 -0400
+++ core-b14583ba37a6d7ce398ccd3cf339f954785b03d8/include/comphelper/errcode.hxx	2023-10-24 23:38:05.004249806 -0400
@@ -24,6 +24,10 @@
 #include <o3tl/typed_flags_set.hxx>
 #include <optional>
 
+#if defined(__sun)
+#include <climits>
+#endif
+
 #if defined(DBG_UTIL)
 #if __has_include(<version>)
 #include <version>
