--- core-aea53c0ed1527ed1f8233972a27128e14d645e8f/bridges/inc/vtablefactory.hxx.old	2023-11-11 13:37:54.344140873 -0500
+++ core-aea53c0ed1527ed1f8233972a27128e14d645e8f/bridges/inc/vtablefactory.hxx	2023-11-11 13:38:37.233549332 -0500
@@ -31,7 +31,7 @@
 /*See: http://people.redhat.com/drepper/selinux-mem.html*/
 #if defined(LINUX) || defined(OPENBSD) || defined(FREEBSD) \
     || defined(NETBSD) || defined(DRAGONFLY) || defined (ANDROID) \
-    || defined(HAIKU)
+    || defined(HAIKU) || defined(__sun)
 #define USE_DOUBLE_MMAP
 #endif
 
