--- libreoffice-7.2.5.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.~1~	Tue Feb  8 15:16:27 2022
+++ libreoffice-7.2.5.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	Tue Feb  8 16:34:47 2022
@@ -145,7 +145,7 @@
     bool collectBytes = false;
     int startOfScan = 0;
     int b1 = -1;
-    for (; ; )
+    for ( ; ; )
     {
         const int b2 = b1;
         b1 = str->getChar();
@@ -153,8 +153,8 @@
         if (b1 == -1)
             return false;
 
-        if (collectBytes)
-        {
+        if (collectBytes) {
+     
             outBuf.push_back(static_cast<Output_t>(b1));
 
             bytesToMarker--;
@@ -433,9 +433,9 @@
     int nSize = 0;
 
 #if POPPLER_CHECK_VERSION(20, 12, 0)
-    std::string familyName = gfxFont->getNameWithoutSubsetTag();
+    std::string familyName( gfxFont->getNameWithoutSubsetTag() );
 #else
-    std::string familyName = gfxFont->getName()->toStr();
+    std::string familyName( gfxFont->getName()->getCString() );
     if (familyName.length() > 7 && familyName.at(6) == '+')
     {
         familyName = familyName.substr(7);
@@ -444,7 +444,7 @@
     if( familyName != "" )
     {
         aNewFont.familyName.clear();
-        aNewFont.familyName.append( familyName );
+        aNewFont.familyName.append( familyName.c_str() );
     }
     else
     {
