--- libreoffice-25.2.0.0.alpha1/external/skia/Library_skia.mk.~1~	2024-11-22 15:05:31.000000000 -0500
+++ libreoffice-25.2.0.0.alpha1/external/skia/Library_skia.mk	2024-12-02 01:33:18.558633273 -0500
@@ -109,7 +109,7 @@
     libpng \
 ))
 
-ifeq ($(OS),LINUX)
+ifeq ($(OS), $(filter LINUX %BSD SOLARIS, $(OS)))
 $(eval $(call gb_Library_add_libs,skia,\
     -lm \
     -ldl \
@@ -296,7 +296,7 @@
 #
 $(eval $(call gb_Library_add_generated_exception_objects,skia,\
     UnpackedTarball/skia/tools/window/WindowContext \
-	$(if $(filter LINUX,$(OS)),UnpackedTarball/skia/tools/window/unix/RasterWindowContext_unix) \
+	$(if $(filter LINUX SOLARIS,$(OS)),UnpackedTarball/skia/tools/window/unix/RasterWindowContext_unix) \
     $(if $(filter WNT,$(OS)),UnpackedTarball/skia/tools/window/win/RasterWindowContext_win) \
 ))
 ifeq ($(SKIA_GPU),METAL)
