--- core-aea53c0ed1527ed1f8233972a27128e14d645e8f/sw/Module_sw.mk.old	2023-11-11 18:13:43.130006484 -0500
+++ core-aea53c0ed1527ed1f8233972a27128e14d645e8f/sw/Module_sw.mk	2023-11-11 18:14:10.749527565 -0500
@@ -83,7 +83,6 @@
     CppunitTest_sw_ooxmlexport7 \
     CppunitTest_sw_ooxmlexport8 \
     CppunitTest_sw_ooxmlexport9 \
-    CppunitTest_sw_ooxmlexport10 \
     CppunitTest_sw_ooxmlexport11 \
     CppunitTest_sw_ooxmlexport12 \
     CppunitTest_sw_ooxmlexport13 \
