--- core-92e5898fa9ffa4dd8742f5f26e790563feef286e/vcl/source/app/svapp.cxx.old	2023-11-15 11:54:01.059046071 -0500
+++ core-92e5898fa9ffa4dd8742f5f26e790563feef286e/vcl/source/app/svapp.cxx	2023-11-15 11:54:27.968184792 -0500
@@ -1225,7 +1225,7 @@
 #endif
             appendDetails(u"", Localize(SV_APP_DEFAULT, bLocalize));
 
-#if (defined LINUX || defined _WIN32 || defined MACOSX || defined __FreeBSD__ || defined EMSCRIPTEN)
+#if (defined LINUX || defined _WIN32 || defined MACOSX || defined __FreeBSD__ || defined(__sun) || defined EMSCRIPTEN)
         appendDetails(u"; ", SV_APP_VCLBACKEND + GetToolkitName());
 #endif
     }
