--- libreoffice-7.2.5.2/formula/source/core/api/FormulaCompiler.cxx.orig	Wed Feb  9 13:46:51 2022
+++ libreoffice-7.2.5.2/formula/source/core/api/FormulaCompiler.cxx	Wed Feb  9 13:47:13 2022
@@ -37,6 +37,10 @@
 #include <com/sun/star/sheet/FormulaMapGroupSpecialOffset.hpp>
 #include <algorithm>
 
+#ifdef __sun__
+#include <alloca.h>
+#endif
+
 namespace formula
 {
     using namespace ::com::sun::star;
