--- libreoffice-7.2.7.2/configure.ac.orig	Wed Oct 26 15:10:19 2022
+++ libreoffice-7.2.7.2/configure.ac	Wed Oct 26 14:46:28 2022
@@ -12305,7 +12305,7 @@
 elif test "$enable_fuzzers" = "yes"; then
     AC_MSG_CHECKING([whether gpgmepp should be disabled due to oss-fuzz])
     AC_MSG_RESULT([yes])
-elif test "$_os" = "Linux" -o "$_os" = "Darwin" -o "$_os" = "WINNT" ; then
+elif test "$_os" = "Linux" -o "$_os" = "Darwin" -o "$_os" = "WINNT" -o "$_os" = "SunOS" ; then
     dnl ===================================================================
     dnl Check for system gpgme
     dnl ===================================================================
