--- libreoffice-7.5.1.1/configure.ac.~4~	2023-02-20 21:13:55.822829904 -0500
+++ libreoffice-7.5.1.1/configure.ac	2023-02-20 21:21:10.761205247 -0500
@@ -12524,7 +12524,7 @@
     AC_MSG_RESULT([no (MPL only])
 elif test "$enable_fuzzers" = "yes"; then
     AC_MSG_RESULT([no (oss-fuzz)])
-elif test \( \( "$_os" = "Linux" -o "$_os" = "Darwin" \) -a "$ENABLE_NSS" = TRUE \) -o "$_os" = "WINNT" ; then
+elif test \( \( "$_os" = "SunOS" -o "$_os" = "Linux" -o "$_os" = "Darwin" \) -a "$ENABLE_NSS" = TRUE \) -o "$_os" = "WINNT" ; then
     AC_MSG_RESULT([yes])
     dnl ===================================================================
     dnl Check for system gpgme
