--- libreoffice-25.2.0.0.alpha1/external/libgpg-error/UnpackedTarball_libgpg-error.mk.~1~	2024-11-26 22:23:14.458693470 -0500
+++ libreoffice-25.2.0.0.alpha1/external/libgpg-error/UnpackedTarball_libgpg-error.mk	2024-11-26 22:26:26.990129311 -0500
@@ -20,7 +20,7 @@
 	$(if $(filter MSC,$(COM)),external/libgpg-error/w32-build-fixes-4.patch) \
 	$(if $(filter MSC,$(COM)),external/libgpg-error/w32-build-fixes-5.patch) \
 	$(if $(filter MSC,$(COM)),external/libgpg-error/w32-build-fixes-6.patch) \
-	$(if $(filter LINUX,$(OS)),external/libgpg-error/libgpgerror-bundled-soname.patch.1) \
+	$(if $(filter LINUX SOLARIS,$(OS)),external/libgpg-error/libgpgerror-bundled-soname.patch.1) \
 	external/libgpg-error/clang-cl.patch \
 	external/libgpg-error/undeclared-environ-macOS.patch \
 ))
