--- a/external/libgpg-error/UnpackedTarball_libgpg-error.mk	Fri May  6 16:49:44 2022
+++ b/external/libgpg-error/UnpackedTarball_libgpg-error.mk	Wed Oct 26 23:13:12 2022
@@ -20,7 +20,7 @@
 	$(if $(filter MSC,$(COM)),external/libgpg-error/w32-disable-dllinit.patch.1) \
 	external/libgpg-error/w32-build-fixes-4.patch \
 	$(if $(filter MSC,$(COM)),external/libgpg-error/w32-build-fixes-5.patch) \
-	$(if $(filter LINUX,$(OS)),external/libgpg-error/libgpgerror-bundled-soname.patch.1) \
+	$(if $(filter LINUX SOLARIS,$(OS)),external/libgpg-error/libgpgerror-bundled-soname.patch.1) \
 	external/libgpg-error/clang-cl.patch \
 ))
 
