--- a/vcl/qa/cppunit/text.cxx	Sun Nov  6 22:31:47 2022
+++ b/vcl/qa/cppunit/text.cxx	Sun Nov  6 22:32:10 2022
@@ -22,7 +22,7 @@
 {
     // if enabled - check the result images with:
     // "xdg-open ./workdir/CppunitTest/vcl_text_test.test.core/"
-    static constexpr const bool mbExportBitmap = false;
+    static constexpr const bool mbExportBitmap = true;
 
     void exportDevice(const OUString& filename, const VclPtr<VirtualDevice>& device)
     {
