--- libreoffice-24.8.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx.orig	2024-12-26 14:39:16.769371535 +0100
+++ libreoffice-24.8.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2024-12-26 14:42:30.340644731 +0100
@@ -1090,7 +1090,8 @@
 
     int bitsPerComponent = 1;
     StreamColorSpaceMode csMode = streamCSNone;
-    str->getImageParams( &bitsPerComponent, &csMode );
+    bool hasAlpha = false;
+    str->getImageParams( &bitsPerComponent, &csMode, &hasAlpha );
     if( bitsPerComponent == 1 && (csMode == streamCSNone || csMode == streamCSDeviceGray) )
     {
         GfxRGB oneColor = { dblToCol( 1.0 ), dblToCol( 1.0 ), dblToCol( 1.0 ) };
