--- schroedinger-1.0.11/schroedinger/schrolowdelay.c	Tue Jan 24 20:26:20 2012
+++ schroedinger-1.0.11/schroedinger/schrolowdelay.c	Tue Jan 24 20:25:30 2012
@@ -750,14 +750,14 @@
 
   if (SCHRO_FRAME_FORMAT_DEPTH (picture->transform_frame->format) ==
       SCHRO_FRAME_FORMAT_DEPTH_S32) {
-    return schro_decoder_decode_lowdelay_transform_data_slow_s32 (picture);
+    schro_decoder_decode_lowdelay_transform_data_slow_s32 (picture);
   } else if ((params->iwt_chroma_width >> params->transform_depth) %
       params->n_horiz_slices == 0 &&
       (params->iwt_chroma_height >> params->transform_depth) %
       params->n_vert_slices == 0) {
-    return schro_decoder_decode_lowdelay_transform_data_fast (picture);
+    schro_decoder_decode_lowdelay_transform_data_fast (picture);
   } else {
-    return schro_decoder_decode_lowdelay_transform_data_slow (picture);
+    schro_decoder_decode_lowdelay_transform_data_slow (picture);
   }
 }
 
