--- lcms-1.19/configure.ac.1	2018-01-30 13:25:20.589673969 +0000
+++ lcms-1.19/configure.ac	2018-01-30 13:26:27.107419291 +0000
@@ -94,7 +94,7 @@
   if test -n "$PYTHON"
   then
 	LCMS_PYTHON_PREFIX=`$PYTHON -c "import sys; print sys.prefix"`
-	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1)"`
+	LCMS_PYEXECDIR=`echo "${PYTHONMODULEDIR}"`
     LCMS_PYINCLUDE=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`	
 	LCMS_PYLIB=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,1)"`
 	AC_CHECK_FILE([$LCMS_PYINCLUDE/Python.h],PYTHON_OK='yes',PYTHON_OK='no')
