--- configure	2009-10-30 08:57:45.000000000 -0700
+++ configure	2011-03-29 17:53:58.890691776 -0700
@@ -5246,13 +5246,8 @@
 
 
 # Configure python. AM_PATH_PYTHON is basically braindead
-PYTHON=''
 if test "$with_python" = 'yes'
 then
-
-
-
-
         if test -n "$PYTHON"; then
       # If the user set $PYTHON, use it and don't search something else.
       { echo "$as_me:$LINENO: checking whether $PYTHON version >= 2.2" >&5
@@ -5400,8 +5395,7 @@
 if test "${am_cv_python_pythondir+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  am_cv_python_pythondir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,0,prefix='$PYTHON_PREFIX')" 2>/dev/null ||
-     echo "$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages"`
+  am_cv_python_pythondir=`echo "${PYTHONMODULEDIR}"`
 fi
 { echo "$as_me:$LINENO: result: $am_cv_python_pythondir" >&5
 echo "${ECHO_T}$am_cv_python_pythondir" >&6; }
@@ -5417,8 +5411,7 @@
 if test "${am_cv_python_pyexecdir+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  am_cv_python_pyexecdir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1,0,prefix='$PYTHON_EXEC_PREFIX')" 2>/dev/null ||
-     echo "${PYTHON_EXEC_PREFIX}/lib/python${PYTHON_VERSION}/site-packages"`
+  am_cv_python_pyexecdir=`echo "${PYTHONMODULEDIR}"`
 fi
 { echo "$as_me:$LINENO: result: $am_cv_python_pyexecdir" >&5
 echo "${ECHO_T}$am_cv_python_pyexecdir" >&6; }
@@ -5436,7 +5429,8 @@
   if test -n "$PYTHON"
   then
 	LCMS_PYTHON_PREFIX=`$PYTHON -c "import sys; print sys.prefix"`
-	LCMS_PYEXECDIR=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1)"`
+	LCMS_PYTHON_VERSION=`$PYTHON -c "import sys; print sys.version[:3]"`
+	LCMS_PYEXECDIR=`echo "${PYTHONMODULEDIR}"`
     LCMS_PYINCLUDE=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`
 	LCMS_PYLIB=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,1)"`
 	as_ac_File=`echo "ac_cv_file_$LCMS_PYINCLUDE/Python.h" | $as_tr_sh`
@@ -9481,7 +9475,7 @@
       ;;
 
     solaris*)
-      no_undefined_flag=' -z text'
+      no_undefined_flag=' -z text -z defs'
       if test "$GCC" = yes; then
 	wlarc='${wl}'
 	archive_cmds='$CC -shared ${wl}-h ${wl}$soname -o $lib $libobjs $deplibs $compiler_flags'
@@ -9489,11 +9483,11 @@
 	  $CC -shared ${wl}-M ${wl}$lib.exp ${wl}-h ${wl}$soname -o $lib $libobjs $deplibs $compiler_flags~$rm $lib.exp'
       else
 	wlarc=''
-	archive_cmds='$LD -G${allow_undefined_flag} -h $soname -o $lib $libobjs $deplibs $linker_flags'
+	archive_cmds='$CC $LTCFLAGS $LDFLAGS -G ${allow_undefined_flag} -h $soname -o $lib $libobjs $deplibs $linker_flags'
 	archive_expsym_cmds='$echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~$echo "local: *; };" >> $lib.exp~
-  	$LD -G${allow_undefined_flag} -M $lib.exp -h $soname -o $lib $libobjs $deplibs $linker_flags~$rm $lib.exp'
+  	$CC $LTCFLAGS $LDFLAGS -G ${allow_undefined_flag} -M $lib.exp -h $soname -o $lib $libobjs $deplibs $linker_flags~$rm $lib.exp'
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec=''
       hardcode_shlibpath_var=no
       case $host_os in
       solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -11270,7 +11264,7 @@
 echo "$as_me: creating $ofile" >&6;}
 
   cat <<__EOF__ >> "$cfgfile"
-#! $SHELL
+#! $CONFIG_SHELL
 
 # `$echo "$cfgfile" | sed 's%^.*/%%'` - Provide generalized library-building support services.
 # Generated automatically by $PROGRAM (GNU $PACKAGE $VERSION$TIMESTAMP)
