--- python/Makefile.in	2009-07-17 07:35:22.000000000 -0700
+++ python/Makefile.in	2011-03-29 18:16:55.269652847 -0700
@@ -62,7 +62,7 @@
 _lcms_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
 	$(CXXFLAGS) $(_lcms_la_LDFLAGS) $(LDFLAGS) -o $@
-@SUPPORT_PYTHON_TRUE@am__lcms_la_rpath = -rpath $(pkgdir)
+@SUPPORT_PYTHON_TRUE@am__lcms_la_rpath = -rpath $(libdir)
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -71,7 +71,7 @@
 LTCXXCOMPILE = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
 	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
-CXXLD = $(CXX)
+CXXLD = $(CXX) $(CXXFLAGS) $(LDFLAGS)
 CXXLINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
@@ -126,7 +126,7 @@
 LCMS_PYEXECDIR = @LCMS_PYEXECDIR@
 LCMS_PYINCLUDE = @LCMS_PYINCLUDE@
 LCMS_PYLIB = @LCMS_PYLIB@
-LD = @LD@
+LD = $(CXX) $(CXXFLAGS) $(LDFLAGS)
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBRARY_AGE = @LIBRARY_AGE@
