--- a/doc/Makefile.in	2010-05-22 04:29:07.000000000 -0700
+++ b/doc/Makefile.in	2010-11-30 10:05:27.172181211 -0800
@@ -1041,10 +1041,8 @@
 	        && { if test -f $@; then exit 0; else break; fi; }; \
 	      exit 1;; \
 	  esac; \
-	done; \
-	echo ' cd $(top_srcdir) && $(AUTOMAKE) --gnu doc/Makefile'; \
-	$(am__cd) $(top_srcdir) && \
-	  $(AUTOMAKE) --gnu doc/Makefile
+	done;
+
 .PRECIOUS: Makefile
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
 	@case '$?' in \
