--- a/java/Makefile.in	2010-05-22 04:28:19.000000000 -0700
+++ b/java/Makefile.in	2011-05-11 11:45:44.210697487 -0700
@@ -747,7 +747,7 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = gnu . misc
 INSTALL_TARGETS = $(am__append_1)
-@JAVA_TRUE@libidn_jardir = $(datadir)/java
+@JAVA_TRUE@libidn_jardir = $(datadir)/lib/idn
 @JAVA_TRUE@dist_libidn_jar_DATA = libidn-$(VERSION).jar
 all: all-recursive
 
@@ -1135,7 +1135,7 @@
 @JAVA_TRUE@		gnu/inet/encoding/Stringprep.class		\
 @JAVA_TRUE@		gnu/inet/encoding/StringprepException.class
 @JAVA_TRUE@install-jar:
-@JAVA_TRUE@	install -d -m 755 $(DESTDIR)$(libidn_jardir)
+@JAVA_TRUE@	$(INSTALL) -d -m 755 $(DESTDIR)$(libidn_jardir)
 @JAVA_TRUE@	ln -sf libidn-$(VERSION).jar $(DESTDIR)$(libidn_jardir)/libidn.jar
 
 @JAVA_TRUE@clean-local:
