--- libdnet-1.12/configure.in.~1~	2014-01-09 12:11:33.721228009 +0400
+++ libdnet-1.12/configure.in	2014-01-09 12:12:59.522849122 +0400
@@ -169,12 +169,16 @@
 	    sys/sysctl.h sys/time.h)
 	AC_CHECK_HEADERS(net/bpf.h net/if.h net/if_var.h \
 	    net/if_arp.h net/if_dl.h net/pfilt.h \
-	    net/pfvar.h net/radix.h net/raw.h net/route.h netinet/in_var.h \
+	    net/pfvar.h net/radix.h net/raw.h netinet/in.h netinet/in_var.h \
 	    net/if_tun.h linux/if_tun.h netinet/ip_fw.h linux/ip_fw.h \
 	    linux/ip_fwchains.h linux/netfilter_ipv4/ipchains_core.h)
 	AC_CHECK_HEADERS(ip_fil_compat.h netinet/ip_fil_compat.h ip_compat.h \
 	    netinet/ip_compat.h ip_fil.h netinet/ip_fil.h)
 	AC_CHECK_HEADERS(hpsecurity.h stropts.h)
+	AC_CHECK_HEADER([net/route.h],[],[], 
+[#ifdef HAVE_NETINET_IN_H
+#include <netinet/in.h>
+#endif])
 fi
 
 dnl Checks for typedefs, structures, and compiler characteristics.
