--- php-5.4.45/configure.1	2015-11-04 09:53:15.577344545 +0300
+++ php-5.4.45/configure	2015-11-04 09:53:42.020895672 +0300
@@ -55403,7 +55403,7 @@
 
 
   if test "$ext_shared" = "yes"; then
-    LDAP_SHARED_LIBADD="-lldap $LDAP_SHARED_LIBADD"
+    LDAP_SHARED_LIBADD="-lldap-2.4 $LDAP_SHARED_LIBADD"
     if test -n "$LDAP_LIBDIR"; then
 
   if test "$LDAP_LIBDIR" != "/usr/$PHP_LIBDIR" && test "$LDAP_LIBDIR" != "/usr/lib"; then
