We do not support obsoleted Python versions

--- wcwidth-0.2.14/tox.ini.orig
+++ wcwidth-0.2.14/tox.ini
@@ -1,9 +1,7 @@
 [tox]
 envlist = update, compile, autopep8, docformatter, isort, pylint, flake8, pydocstyle, docs, verify_tables, py{36, 37, 38, 39, 310, 311, 312, 313, 314}, pypy{36, 37, 38, 39, 310}
 skip_missing_interpreters = true
-# needed for Python 3.6
-# https://tox.wiki/en/4.11.3/faq.html#testing-end-of-life-python-versions
-requires = virtualenv<20.22.0
+requires = virtualenv
 
 [base]
 pip_compile_command = pip-compile --resolver=backtracking --strip-extras --no-emit-index-url --allow-unsafe --upgrade
