--- wcwidth-0.2.13/tox.ini.orig
+++ wcwidth-0.2.13/tox.ini
@@ -7,7 +7,7 @@
 pip_compile_command = pip-compile --resolver=backtracking --strip-extras --no-emit-index-url --allow-unsafe --upgrade
 
 [testenv]
-deps = -r requirements-tests39.txt
+deps = -r requirements-tests39.in
 commands = {envpython} -m pytest --cov-config={toxinidir}/tox.ini {posargs:\
              --verbose \
              --junit-xml=.tox/results.{envname}.xml \
