--- nautilus-2.30.1/libnautilus-private/nautilus-search-engine-tracker.c.orig	2010-04-12 11:43:44.000000000 +0100
+++ nautilus-2.30.1/libnautilus-private/nautilus-search-engine-tracker.c	2010-12-07 10:57:33.316942852 +0000
@@ -99,7 +99,9 @@
 	MAP (tracker_get_version, TRACKER_0_6),
 	MAP (tracker_cancel_last_call, TRACKER_0_6 | TRACKER_0_7 | TRACKER_0_8),
 	MAP (tracker_search_metadata_by_text_async, TRACKER_0_6 | TRACKER_0_7),
+	MAP (tracker_search_metadata_by_text_and_mime_async, TRACKER_0_6 | TRACKER_0_7),
 	MAP (tracker_search_metadata_by_text_and_location_async, TRACKER_0_6 | TRACKER_0_7),
+	MAP (tracker_search_metadata_by_text_and_mime_and_location_async, TRACKER_0_6 | TRACKER_0_7),
 	MAP (tracker_client_new, TRACKER_0_8),
 	MAP (tracker_sparql_escape, TRACKER_0_8),
 	MAP (tracker_resources_sparql_query_async, TRACKER_0_8)
