--- nautilus-2.32.2.1/eel/Makefile.am.~1~	2015-03-24 16:04:28.287313416 +0300
+++ nautilus-2.32.2.1/eel/Makefile.am	2015-03-24 16:05:16.305704239 +0300
@@ -10,8 +10,6 @@
 	-DDATADIR=\""$(datadir)"\"			\
 	-DSOURCE_DATADIR=\""$(top_srcdir)/data"\"	\
 	-DGNOMELOCALEDIR=\""$(prefix)/${DATADIRNAME}/locale"\"	\
-	-DG_DISABLE_DEPRECATED				\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGMENU_I_KNOW_THIS_IS_UNSTABLE			\
 	$(NULL)
 
--- nautilus-2.30.1/eel/eel-i18n.h.orig	2013-08-20 14:56:27.078452000 +0530
+++ nautilus-2.30.1/eel/eel-i18n.h	2013-08-20 14:56:51.445048000 +0530
@@ -28,6 +28,7 @@
 #ifdef ENABLE_NLS
 #include <glib.h>
 
+#define G_CONST_RETURN const
 G_CONST_RETURN char *
 _eel_gettext (const char *str) G_GNUC_FORMAT (1);
 
--- nautilus-2.30.1/eel/eel-labeled-image.c.orig	2013-08-20 14:59:00.884546000 +0530
+++ nautilus-2.30.1/eel/eel-labeled-image.c	2013-08-20 14:59:35.628023000 +0530
@@ -41,6 +41,7 @@
 #define DEFAULT_Y_PADDING 0
 #define DEFAULT_X_ALIGNMENT 0.5
 #define DEFAULT_Y_ALIGNMENT 0.5
+#define G_CONST_RETURN const
 
 /* Signals */
 enum
--- nautilus-2.30.1/eel/eel-editable-label.h.orig	2013-08-20 14:57:30.899459000 +0530
+++ nautilus-2.30.1/./eel/eel-editable-label.h	2013-08-20 14:58:02.312653000 +0530
@@ -34,6 +34,8 @@
 extern "C" {
 #endif /* __cplusplus */
 
+#define G_CONST_RETURN const
+
 #define EEL_TYPE_EDITABLE_LABEL eel_editable_label_get_type()
 #define EEL_EDITABLE_LABEL(obj) \
   (G_TYPE_CHECK_INSTANCE_CAST ((obj), EEL_TYPE_EDITABLE_LABEL, EelEditableLabel))
