--- stdcxx-4.2.1/include/rw/_config.h	2008-04-24 20:23:43.000000000 -0400
+++ stdcxx-4.2.1/include/rw/_config.h	2009-03-28 00:35:53.000000000 -0400
@@ -33,7 +33,11 @@
 #define _RWSTD_RW_CONFIG_H_INCLUDED
 
 // include generated configuration header
+#if defined(_IN_BUILD_STDCXX)
 #include <config.h>
+#else
+#include <rw/config/rwconfig_std.h>
+#endif
 
 /*** library version numbers and ids **************************************/
 
