--- stdcxx-4.2.1/include/ansi/ciso646	2008-04-24 20:23:56.000000000 -0400
+++ stdcxx-4.2.1/include/ansi/ciso646	2009-03-12 16:35:36.000000000 -0400
@@ -1,32 +1,39 @@
 // -*- C++ -*-
-/***************************************************************************
+/**
+ * CDDL HEADER START
  *
- * ciso646 - C++ Standard library interface to the ANSI C header iso646.h
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
  *
- * $Id: ciso646 550991 2007-06-26 23:58:07Z sebor $
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
  *
- ***************************************************************************
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
  *
- * Licensed to the Apache Software  Foundation (ASF) under one or more
- * contributor  license agreements.  See  the NOTICE  file distributed
- * with  this  work  for  additional information  regarding  copyright
- * ownership.   The ASF  licenses this  file to  you under  the Apache
- * License, Version  2.0 (the  "License"); you may  not use  this file
- * except in  compliance with the License.   You may obtain  a copy of
- * the License at
+ * CDDL HEADER END
  *
- * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the  License is distributed on an  "AS IS" BASIS,
- * WITHOUT  WARRANTIES OR CONDITIONS  OF ANY  KIND, either  express or
- * implied.   See  the License  for  the  specific language  governing
- * permissions and limitations under the License.
+ * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+ * Use is subject to license terms.
  *
- * Copyright 1994-2006 Rogue Wave Software.
- * 
- **************************************************************************/
+ * ident "@(#)ciso646.8.diff 1.1     09/08/21 SMI"
+ */
+
+#ifndef _ISO_646_H
+#define _ISO_646_H
+
+#include <iso646.h>
+
+#ifndef __STDC_ISO_10646__
+#define __STDC_ISO_10646__
+#endif
 
-#include <rw/_defs.h>
+#endif /* _ISO_646_H */
 
-#include _RWSTD_ANSI_C_ISO646_H
