--- stdcxx-4.2.1/include/new.h	2009-06-08 10:13:00.141832000 -0400
+++ stdcxx-4.2.1/include/new.h	2007-05-03 05:51:01.000000000 -0400
@@ -0,0 +1,19 @@
+/*
+        Copyright 06/08/98 Sun Microsystems, Inc. All Rights Reserved
+*/
+/* new.h
+
+   @(#)new.h	1.2  06/08/98 14:00:58
+
+   Compatibility declaration for ISO standard new interface
+*/
+
+#ifndef _NEW_H_
+#define _NEW_H_
+
+#include <new>
+
+using std::bad_alloc;
+using std::set_new_handler;
+
+#endif
