--- stdcxx-4.2.1/include/vector	2008-04-24 20:23:57.000000000 -0400
+++ stdcxx-4.2.1/include/vector	2011-01-06 18:05:47.301233744 -0500
@@ -1226,7 +1226,7 @@
 #ifdef _RWSTD_BOUNDS_CHECKING
 
         _RWSTD_REQUIRES (__n < size (),
-                         (_RWSTD_ERROR_LENGTH_ERROR,
+                         (_RWSTD_ERROR_OUT_OF_RANGE,
                           _RWSTD_FUNC ("vector<bool>::[](size_type)"),
                           __n, size ()));
 
@@ -1239,7 +1239,7 @@
 #ifdef _RWSTD_BOUNDS_CHECKING
 
         _RWSTD_REQUIRES (__n < size (),
-                         (_RWSTD_ERROR_LENGTH_ERROR,
+                         (_RWSTD_ERROR_OUT_OF_RANGE,
                           _RWSTD_FUNC ("vector<bool>::[](size_type)"),
                           __n, size ()));
 
@@ -1250,7 +1250,7 @@
     reference       at (size_type __n)               
     { 
         _RWSTD_REQUIRES (__n < size (),
-                         (_RWSTD_ERROR_LENGTH_ERROR,
+                         (_RWSTD_ERROR_OUT_OF_RANGE,
                           _RWSTD_FUNC ("vector<bool>::at(size_type)"),
                           __n, size ()));
       return *(begin() + __n); 
@@ -1258,7 +1258,7 @@
     const_reference at (size_type __n)   const 
     {
         _RWSTD_REQUIRES (__n < size (),
-                         (_RWSTD_ERROR_LENGTH_ERROR,
+                         (_RWSTD_ERROR_OUT_OF_RANGE,
                           _RWSTD_FUNC ("vector<bool>::at(size_type) const"),
                           __n, size ()));
 
