--- stdcxx-4.2.1/include/valarray	2011-01-13 16:59:43.694747005 -0500
+++ stdcxx-4.2.1/include/valarray	2011-01-18 06:58:17.433752356 -0500
@@ -776,7 +776,7 @@
 operator&& (const valarray<_TypeT> &__lhs, const _TypeT &__rhs)
 {
     return _RW::__rw_binary_function (__lhs,
-                                      bind2nd (equal_to<_TypeT>(), __rhs));
+                                      bind2nd (logical_and<_TypeT>(), __rhs));
 }
 
 template<class _TypeT>
