--- stdcxx-4.2.1/include/rw/_strref.h	2008-04-24 20:23:43.000000000 -0400
+++ stdcxx-4.2.1/include/rw/_strref.h	2011-03-04 12:01:21.821122729 -0500
@@ -245,8 +245,10 @@
 
     union {
         size_type _C_size;    // Number of actual data values stored
+#if defined(__HP_aCC)
         _CharT    _C_dummy;   // force the alignment of the first char
-    } _C_size;                // named to work around an HP aCC 3.30 bug
+#endif                        // named to work around an HP aCC 3.30 bug
+    } _C_size;
 };
 
 
