# Patch for CR 7098498
--- stdcxx-4.2.1/include/loc/_moneypunct.h	2011-07-10 11:57:53.407297720 -0700
+++ stdcxx-4.2.1/include/loc/_moneypunct.h	2011-10-06 09:50:48.591327947 -0700
@@ -67,7 +67,7 @@
     typedef
     basic_string<char_type, char_traits<char_type>, allocator<char_type> >
     string_type;
-    typedef moneypunct<_CharT, bool> _Type;
+    typedef moneypunct<_CharT, _Intl> _Type;
 
     _EXPLICIT moneypunct (_RWSTD_SIZE_T __refs = 0)
         : _RW::__rw_facet (__refs),
