--- stdcxx-4.2.1/include/string	2011-05-04 12:08:04.799700400 -0700
+++ stdcxx-4.2.1/include/string	2012-07-17 17:38:04.411236200 -0700
@@ -853,7 +853,7 @@
 
 #ifndef _RWSTD_NO_COLLAPSE_TEMPLATE_STATICS
 
-    static _RW::__null_ref<_CharT, _Traits, _Allocator> _C_null_ref;
+    __global static _RW::__null_ref<_CharT, _Traits, _Allocator> _C_null_ref;
 
     static _RW::__null_ref<_CharT, _Traits, _Allocator>* _C_nullref () {
         return &_C_null_ref;
