--- pytest-randomly-4.0.1/pyproject.toml.orig
+++ pytest-randomly-4.0.1/pyproject.toml
@@ -51,7 +51,6 @@
   "factory-boy",
   "faker",
   "importlib-metadata>=3.6; python_version<'3.10'",
-  "model-bakery>=1.13",
   "numpy",
   "pytest",
   "pytest-xdist",
--- pytest-randomly-4.0.1/tests/test_pytest_randomly.py.orig
+++ pytest-randomly-4.0.1/tests/test_pytest_randomly.py
@@ -633,6 +633,7 @@
     out.assert_outcomes(passed=2)
 
 
+@pytest.mark.skip(reason="model_bakery is not available")
 def test_model_bakery(ourtester):
     """
     Check the Model Bakery random generator is reset between tests.
