--- configure.ac.orig	2019-01-02 14:28:47.000000000 +0000
+++ configure.ac	2020-11-07 20:34:03.896580896 +0000
@@ -28,6 +28,8 @@
 fi
 
 # Checks for libraries.
+AC_SEARCH_LIBS([gethostbyname], [nsl])
+AC_SEARCH_LIBS([socket, connect], [socket])
 AC_SEARCH_LIBS([tgetent], [termcap curses ncurses], , AC_MSG_ERROR([terminal handling library missing]))
 AC_SEARCH_LIBS([basename], [gen])
 if test "x$with_pcre" != "xno"; then
@@ -37,6 +39,8 @@
 fi
 
 # Checks for header files.
+AC_CHECK_HEADERS([netdb.h])
+AC_CHECK_HEADERS([sys/socket.h])
 AC_CHECK_HEADERS([libgen.h paths.h termcap.h curses.h term.h sys/loadavg.h])
 if test "x$with_pcre" != "xno"; then
    AC_CHECK_HEADERS([pcre.h])
