--- src/shmux.c.orig	Thu Apr  2 11:13:09 2009
+++ src/shmux.c	Thu Apr  2 11:12:11 2009
@@ -46,7 +46,7 @@
 #define DEFAULT_ANALYSIS "regex"
 #define DEFAULT_ERRORCODES "1-"
 #define	DEFAULT_MAXWORKERS 10
-#define DEFAULT_PINGTIMEOUT "500"
+#define DEFAULT_PINGTIMEOUT "10"
 #define DEFAULT_RCMD "ssh"
 #define DEFAULT_SPAWNMODE "one"
 #define DEFAULT_TESTTIMEOUT 15
@@ -70,7 +70,7 @@
     fprintf(stderr, "  -M <max>      Maximum number of simultaneous processes (Default: %u).\n", DEFAULT_MAXWORKERS);
     fprintf(stderr, "  -r <rcmd>     Set the default method (Default: %s).\n", DEFAULT_RCMD);
     fprintf(stderr, "  -p            Ping targets to check for life.\n");
-    fprintf(stderr, "  -P <millisec> Initial target timeout given to fping (Default: %s).\n", DEFAULT_PINGTIMEOUT);
+    fprintf(stderr, "  -P <seconds>  Initial target timeout given to ping (Default: %s).\n", DEFAULT_PINGTIMEOUT);
     fprintf(stderr, "  -t            Send test command to verify target health.\n");
     fprintf(stderr, "  -T <seconds>  Time to wait for test answer (Default: %d).\n", DEFAULT_TESTTIMEOUT);
     fprintf(stderr, "\n");
