--- ./bin/utxlock.orig	Thu Jul 10 14:26:31 2025
+++ ./bin/utxlock	Thu Jul 10 14:31:03 2025
@@ -116,7 +116,7 @@
     # install them in different places.
 
     XSCREENLOCKCMD=`command -v xscreensaver-command`
-    GNOMESCREENLOCKCMD=`command -v gnome-screensaver-command`
+    GNOMESCREENLOCKCMD=`command -v mate-screensaver-command`
     XLOCKCMD=`command -v xlock`
     CDELOCKCMD=/usr/dt/bin/dtaction
 
