--- ekiga-4.0.1/configure.ac.orig	2013-02-20 21:18:09.000000000 +0100
+++ ekiga-4.0.1/configure.ac	2016-08-27 02:55:02.009466599 +0200
@@ -155,7 +155,7 @@
 if test "x${gm_platform}" = "xmingw" ; then
    PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.20.0])
 else
-   PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.20.0 gnome-icon-theme >= 3.0.0])
+   PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.20.0 gnome-icon-theme >= 2.31.0])
 fi
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.24.0 gmodule-2.0 gobject-2.0 gthread-2.0])
 AC_ARG_ENABLE([gtk-debug],
