--- ekiga-3.2.6/ekiga.desktop.in.in.ori	2009-09-24 09:25:09.116779419 +0100
+++ ekiga-3.2.6/ekiga.desktop.in.in	2009-09-24 09:25:48.319743715 +0100
@@ -1,8 +1,10 @@
 [Desktop Entry]
-_Name=Ekiga Softphone
+# SUN_BRANDING
+_Name=Video Conference
 TryExec=@PACKAGE_NAME@
 _GenericName=IP Telephony, VoIP and Video Conferencing
-_Comment=Talk to people over the Internet
+# SUN_BRANDING
+_Comment=Chat to friends and colleagues using a webcam
 StartupNotify=true
 Exec=@PACKAGE_NAME@
 Icon=@PACKAGE_NAME@
