--- bison-3.5.2/Makefile.in.~1~	2020-02-13 18:17:58.000000000 +0000
+++ bison-3.5.2/Makefile.in	2020-02-15 11:58:44.935070794 +0000
@@ -565,6 +565,12 @@ am__lib_liby_a_SOURCES_DIST = lib/main.c
 @ENABLE_YACC_TRUE@am_lib_liby_a_OBJECTS = lib/main.$(OBJEXT) \
 @ENABLE_YACC_TRUE@	lib/yyerror.$(OBJEXT)
 lib_liby_a_OBJECTS = $(am_lib_liby_a_OBJECTS)
+
+# doing this for just the OBJECTS macros doesn't seeem to work
+# and we rely on the approprate xregs/no-app-regs flag already
+# being in our CFLAGS from shared-macros.mk
+lib/yyerror.o lib/main.o: CFLAGS += $(CC_PIC)
+
 am__objects_7 = examples/c++/calc++/calc__-driver.$(OBJEXT) \
 	examples/c++/calc++/calc__-scanner.$(OBJEXT) \
 	examples/c++/calc++/calc__-calc++.$(OBJEXT)
@@ -3119,6 +3125,9 @@ WARN_ON_USE_H = $(top_srcdir)/lib/warn-o
 @ENABLE_YACC_TRUE@lib_LIBRARIES = lib/liby.a
 @ENABLE_YACC_TRUE@EXTRA_LIBRARIES = lib/liby.a
 @ENABLE_YACC_TRUE@lib_liby_a_SOURCES = lib/main.c lib/yyerror.c
+
+lib_LIBRARIES += lib/liby.so.1
+
 # Prettify Automake-computed names of compiled objects.
 src_bison_SHORTNAME = bison
 src_bison_CPPFLAGS = $(AM_CPPFLAGS) -DINSTALLDIR=\"$(bindir)\"
@@ -3268,6 +3277,10 @@ gen_start_date = 2012-01-16
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
+lib/liby.so.1:    $(lib_liby_a_OBJECTS) $(lib_liby_a_DEPENDENCIES)
+	-rm -f lib/liby.so.1
+	$(CC) $(CFLAGS) -shared -Wl,-h -Wl,liby.so.1 -o lib/liby.so.1 $(lib_liby_a_OBJECTS) $(lib_liby_a_LIBADD)
+
 .SUFFIXES:
 .SUFFIXES: .gv .eps .pdf .svg .yy .stamp .c .cc .dvi .l .ll .log .o .obj .ps .test .test$(EXEEXT) .trs .y
 am--refresh: Makefile
