Generated stdio.h doesn't work with g++

--- bison-2.7.1/tests/atlocal.in	2013-04-08 15:42:46.000000000 +0400
+++ bison-2.7.1/tests/atlocal.in	2013-10-09 15:34:31.321332409 +0400
@@ -17,7 +17,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # We need `config.h'.
-CPPFLAGS="-I$abs_top_builddir/lib @CPPFLAGS@"
+CPPFLAGS="-idirafter $abs_top_builddir/lib @CPPFLAGS@"
 
 # Don't just check if $POSIXLY_CORRECT is set, as Bash, when launched
 # as /bin/sh, sets the shell variable POSIXLY_CORRECT to y, but not
