--- Python-2.7.13/Modules/_multiprocessing/multiprocessing.h.orig	Fri Apr 28 18:23:17 2017
+++ Python-2.7.13/Modules/_multiprocessing/multiprocessing.h	Fri Apr 28 18:23:52 2017
@@ -6,8 +6,10 @@
 #ifdef __sun
 /* The control message API is only available on Solaris 
    if XPG 4.2 or later is requested. */
+#ifndef _XOPEN_SOURCE
 #define _XOPEN_SOURCE 500
 #endif
+#endif
 
 #include "Python.h"
 #include "structmember.h"
