--- pgpool-II-4.4.2/doc/src/sgml/stylesheet-man.xsl.orig	2023-01-21 16:51:47.000000000 +0000
+++ pgpool-II-4.4.2/doc/src/sgml/stylesheet-man.xsl	2023-02-04 13:36:28.611553269 +0000
@@ -4,7 +4,7 @@
                 version='1.0'
                 exclude-result-prefixes="exsl">
 
-<xsl:import href="http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl"/>
+<xsl:import href="file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl"/>
 <xsl:import href="stylesheet-common.xsl" />
 
 
