--- rrdtool-1.4.7/bindings/tcl/Makefile.am.1	2018-01-29 15:16:23.079377231 +0000
+++ rrdtool-1.4.7/bindings/tcl/Makefile.am	2018-01-29 15:16:38.902442207 +0000
@@ -27,8 +27,8 @@
 tclpkg_DATA = pkgIndex.tcl
 tclpkg_SCRIPTS = ifOctets.tcl
 else
-pkglib_DATA = pkgIndex.tcl
-pkglib_SCRIPTS = ifOctets.tcl
+pkgdata_DATA = pkgIndex.tcl
+pkgdata_SCRIPTS = ifOctets.tcl
 endif
 
 # Automake doen't like `tclrrd$(VERSION)$(TCL_SHLIB_SUFFIX)' as
