--- sntp/m4/ntp_libevent.m4	2018-02-27 16:15:25.000000000 +0000
+++ sntp/m4/ntp_libevent.m4	2018-08-08 07:26:52.827518999 +0000
@@ -108,9 +108,9 @@ case "$ntp_use_local_libevent" in
 	    done
 	    case "$LIBISC_PTHREADS_NOTHREADS" in
 	     pthreads)
-		LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_pthreads"
+		LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_pthreads-2.0"
 	    esac
-	    LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_core"
+	    LDADD_LIBEVENT="$LDADD_LIBEVENT -levent_core-2.0"
 	else
 	    ntp_use_local_libevent=yes
 	    # HMS: do we only need to do this if LIBISC_PTHREADS_NOTHREADS
