--- opensmalltalk-vm-sun-v5.0.77/platforms/Cross/vm/sqSCCSVersion.h	So. Sep.  7 13:40:30 2025
+++ p0/opensmalltalk-vm-sun-v5.0.77/platforms/Cross/vm/sqSCCSVersion.h	So. Sep.  7 16:06:45 2025
@@ -30,13 +30,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 202509071140-cog $";
 # define REV_START (SvnRawRevisionString + 6)
 
-static char SvnRawRevisionDate[] = "$Date$";
+static char SvnRawRevisionDate[] = "$Date: Sun Sep 7 13:40:30 2025 +0200 $";
 # define DATE_START (SvnRawRevisionDate + 7)
 
-static char SvnRawRepositoryURL[] = "$URL$";
+static char SvnRawRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (SvnRawRepositoryURL + 6)
 
 static char *
@@ -73,17 +73,17 @@
 # undef URL_START
 #elif GIT
 # define PREFIX ""
-static char GitRawRevisionString[] = "$Rev$";
+static char GitRawRevisionString[] = "$Rev: 202509071140-cog $";
 # define REV_START (GitRawRevisionString + 6)
 # define REV_TIME_START (GitRawRevisionString + 14)
 
-static char GitRawRevisionDate[] = "$Date$";
+static char GitRawRevisionDate[] = "$Date: Sun Sep 7 13:40:30 2025 +0200 $";
 # define DATE_START (GitRawRevisionDate + 7)
 
-static char GitRawRepositoryURL[] = "$URL$";
+static char GitRawRepositoryURL[] = "$URL: stes@openindiana:src/opensmalltalk-vm $";
 # define URL_START (GitRawRepositoryURL + 6)
 
-static char GitRawRevisionShortHash[] = "$CommitHash$";
+static char GitRawRevisionShortHash[] = "$CommitHash: 265c7e89e $";
 # define SHORTHASH_START (GitRawRevisionShortHash + 13)
 
 static char *
