--- opensmalltalk-vm-sun-v5.0.59/platforms/Cross/vm/sqSCCSVersion.h	Sat Apr  8 15:42:34 2023
+++ p0/opensmalltalk-vm-sun-v5.0.59/platforms/Cross/vm/sqSCCSVersion.h	Sat Apr  8 18:07:31 2023
@@ -30,13 +30,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 202304081342-cog $";
 # define REV_START (SvnRawRevisionString + 6)
 
-static char SvnRawRevisionDate[] = "$Date$";
+static char SvnRawRevisionDate[] = "$Date: Sat Apr 8 15:42:34 2023 +0200 $";
 # define DATE_START (SvnRawRevisionDate + 7)
 
-static char SvnRawRepositoryURL[] = "$URL$";
+static char SvnRawRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (SvnRawRepositoryURL + 6)
 
 static char *
@@ -73,17 +73,17 @@
 # undef URL_START
 #elif GIT
 # define PREFIX ""
-static char GitRawRevisionString[] = "$Rev$";
+static char GitRawRevisionString[] = "$Rev: 202304081342-cog $";
 # define REV_START (GitRawRevisionString + 6)
 # define REV_TIME_START (GitRawRevisionString + 14)
 
-static char GitRawRevisionDate[] = "$Date$";
+static char GitRawRevisionDate[] = "$Date: Sat Apr 8 15:42:34 2023 +0200 $";
 # define DATE_START (GitRawRevisionDate + 7)
 
-static char GitRawRepositoryURL[] = "$URL$";
+static char GitRawRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (GitRawRepositoryURL + 6)
 
-static char GitRawRevisionShortHash[] = "$CommitHash$";
+static char GitRawRevisionShortHash[] = "$CommitHash: 9ed3da99b $";
 # define SHORTHASH_START (GitRawRevisionShortHash + 13)
 
 static char *
