--- opensmalltalk-vm-sun-v5.0.42/platforms/Cross/vm/sqSCCSVersion.h	Sat Dec  4 15:19:55 2021
+++ p0/opensmalltalk-vm-sun-v5.0.42/platforms/Cross/vm/sqSCCSVersion.h	Sat Dec  4 16:57:57 2021
@@ -30,13 +30,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 202112041419-cog $";
 # define REV_START (SvnRawRevisionString + 6)
 
-static char SvnRawRevisionDate[] = "$Date$";
+static char SvnRawRevisionDate[] = "$Date: Sat Dec 4 15:19:55 2021 +0100 $";
 # define DATE_START (SvnRawRevisionDate + 7)
 
-static char SvnRawRepositoryURL[] = "$URL$";
+static char SvnRawRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (SvnRawRepositoryURL + 6)
 
 static char *
@@ -73,17 +73,17 @@
 # undef URL_START
 #elif GIT
 # define PREFIX ""
-static char GitRawRevisionString[] = "$Rev$";
+static char GitRawRevisionString[] = "$Rev: 202112041419-cog $";
 # define REV_START (GitRawRevisionString + 6)
 # define REV_TIME_START (GitRawRevisionString + 14)
 
-static char GitRawRevisionDate[] = "$Date$";
+static char GitRawRevisionDate[] = "$Date: Sat Dec 4 15:19:55 2021 +0100 $";
 # define DATE_START (GitRawRevisionDate + 7)
 
-static char GitRawRepositoryURL[] = "$URL$";
+static char GitRawRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (GitRawRepositoryURL + 6)
 
-static char GitRawRevisionShortHash[] = "$CommitHash$";
+static char GitRawRevisionShortHash[] = "$CommitHash: d544d004c $";
 # define SHORTHASH_START (GitRawRevisionShortHash + 13)
 
 static char *
