--- opensmalltalk-vm-sun-v5.0.43/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Fri Jan  7 17:40:18 2022
+++ p0/opensmalltalk-vm-sun-v5.0.43/platforms/Cross/plugins/sqPluginsSCCSVersion.h	Sat Jan  8 15:16:05 2022
@@ -9,10 +9,10 @@
  */
 
 #if SUBVERSION
-static char SvnRawPluginsRevisionString[] = "$Rev$";
+static char SvnRawPluginsRevisionString[] = "$Rev: 202201071640-cog $";
 # define PLUGINS_REV_START (SvnRawPluginsRevisionString + 6)
 
-static char SvnRawPluginsRepositoryURL[] = "$URL$";
+static char SvnRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (SvnRawPluginsRepositoryURL + 6)
 
 static char *
@@ -35,10 +35,10 @@
 # undef PLUGINS_REV_START
 # undef URL_START
 #elif GIT
-static char GitRawPluginsRevisionString[] = "$Rev$";
+static char GitRawPluginsRevisionString[] = "$Rev: 202201071640-cog $";
 # define PLUGINS_REV_START (GitRawPluginsRevisionString + 6)
 
-static char GitRawPluginsRepositoryURL[] = "$URL$";
+static char GitRawPluginsRepositoryURL[] = "$URL: stes@pallas:src/opensmalltalk $";
 # define URL_START (GitRawPluginsRepositoryURL + 6)
 
 static char *
