--- opensmalltalk-vm-sun-v5.0.70/platforms/unix/vm/sqUnixMain.c	Fri Jan 17 21:14:01 2025
+++ p0/opensmalltalk-vm-sun-v5.0.70/platforms/unix/vm/sqUnixMain.c	Sun Feb  2 09:59:54 2025
@@ -2103,7 +2103,7 @@
     struct stat sb;
 
 	// first check for an embedded image
-	void *handle = dlopen(NULL, RTLD_NOW);
+	void *handle = NULL;
 	void *embeddedImage;
 	if (handle && (embeddedImage = dlsym(handle,"embeddedImage"))) {
 		strcpy(shortImageName,dlsym(handle,"embeddedImageName"));
