--- opensmalltalk-vm-sun-v5.0.86/platforms/unix/vm-display-X11/sqUnixX11Scale.c	Sat Jul 18 14:53:58 2026
+++ p0/opensmalltalk-vm-sun-v5.0.86/platforms/unix/vm-display-X11/sqUnixX11Scale.c	Sat Jul 18 16:33:33 2026
@@ -32,18 +32,6 @@
 
 void* _use_lib(const char* name)
 {
-  /* 0. */
-  if (name == NULL) {
-    return  dlopen(name, DL_FLAGS);
-    DPRINTF(("Found self\n"));
-  }
-  void* handle = NULL;
-  /* 2. */
-  if ((handle = dlopen(name, DL_FLAGS)) != NULL) {
-    DPRINTF(("Found %s proper\n", name));
-    return handle;
-  }
-  // any?
   return NULL;
 }
 
