--- opensmalltalk-vm-sun-v5.0.89/src/plugins/UnixOSProcessPlugin/UnixOSProcessPlugin.c	Wed Sep  2 20:38:56 2026
+++ p0/opensmalltalk-vm-sun-v5.0.89/src/plugins/UnixOSProcessPlugin/UnixOSProcessPlugin.c	Sat Sep  5 10:49:31 2026
@@ -1186,7 +1186,7 @@
 needSigaltstack(void)
 {
     char * (*GetAttributeString)(int);
-    sqInt ifNilSqInt;
+    void *ifNilSqInt;
     sqInt SigStackSize;
     stack_t sigstack;
 
