--- opensmalltalk-vm-sun-v5.0.33/platforms/unix/vm/sqUnixSpurMemory.c	Sat Aug  7 09:21:27 2021
+++ p0/opensmalltalk-vm-sun-v5.0.33/platforms/unix/vm/sqUnixSpurMemory.c	Sun Aug  8 13:48:39 2021
@@ -100,7 +100,7 @@
 # if __OpenBSD__
 #	define MAP_FLAGS	(MAP_ANON | MAP_PRIVATE | MAP_STACK)
 # else
-#	define MAP_FLAGS	(MAP_ANON | MAP_PRIVATE)
+#	define MAP_FLAGS	(MAP_32BIT | MAP_ANON | MAP_PRIVATE)
 # endif
 
 static int min(int x, int y) { return (x < y) ? x : y; }
