--- xf86-video-mga-1.6.5/src/Makefile.am.orig	2023-02-05 11:40:18.413235231 +0100
+++ xf86-video-mga-1.6.5/src/Makefile.am	2023-02-05 16:29:38.602119426 +0100
@@ -26,7 +26,7 @@
 AM_CFLAGS = @XORG_CFLAGS@ @DRI_CFLAGS@
 mga_drv_la_LTLIBRARIES = mga_drv.la
 mga_drv_la_LDFLAGS = -module -avoid-version
-mga_drv_ladir = @moduledir@/drivers
+mga_drv_ladir = @moduledir@
 mga_drv_la_LIBADD = $(noinst_LTLIBRARIES) $(XORG_LIBS) $(PCIACCESS_LIBS)
 mga_drv_la_LIBADD += -L$(moduledir) -lfb -lpixman-1 -lshadowfb -lfbdevhw -lexa
 mga_drv_la_LIBADD += -lvgahw -lvbe -lint10
