--- xf86-video-openchrome-0.6.0/src/Makefile.am.~2~	2023-02-05 11:35:32.284104254 +0100
+++ xf86-video-openchrome-0.6.0/src/Makefile.am	2023-02-05 16:16:10.623172529 +0100
@@ -33,7 +33,7 @@
 openchrome_drv_la_LTLIBRARIES = openchrome_drv.la
 openchrome_drv_la_LDFLAGS = -module -avoid-version @LIBUDEV_LIBS@
 openchrome_drv_la_LDFLAGS += $(openchrome_drv_la_LDFLAGS_add)
-openchrome_drv_ladir = @moduledir@/drivers
+openchrome_drv_ladir = @moduledir@
 
 openchrome_drv_la_SOURCES = \
          compat-api.h \
