--- xf86-video-openchrome-0.3.3/src/Makefile.am.1	2015-02-13 15:29:41.087614304 +0300
+++ xf86-video-openchrome-0.3.3/src/Makefile.am	2015-02-13 15:30:45.548610067 +0300
@@ -33,7 +33,8 @@
 openchrome_drv_la_LTLIBRARIES = openchrome_drv.la
 openchrome_drv_la_LDFLAGS = -module -avoid-version @LIBUDEV_LIBS@
 openchrome_drv_la_LDFLAGS += $(openchrome_drv_la_LDFLAGS_add)
-openchrome_drv_ladir = @moduledir@/drivers
+tmp = @moduledir@/drivers
+openchrome_drv_ladir = $(tmp:amd64/xorg/modules/drivers=xorg/modules/drivers/amd64)
 
 openchrome_drv_la_SOURCES = \
          compat-api.h \
