--- xf86-video-openchrome-0.6.0/src/via_driver.h.orig	2020-05-14 12:36:33.294523059 +0000
+++ xf86-video-openchrome-0.6.0/src/via_driver.h	2020-05-14 12:37:04.363876672 +0000
@@ -394,8 +394,8 @@
 
 
 /* In via_display.c. */
-const xf86CrtcFuncsRec iga1_crtc_funcs;
-const xf86CrtcFuncsRec iga2_crtc_funcs;
+extern const xf86CrtcFuncsRec iga1_crtc_funcs;
+extern const xf86CrtcFuncsRec iga2_crtc_funcs;
 
 /* In via_exa.c. */
 Bool viaInitExa(ScreenPtr pScreen);
--- xf86-video-openchrome-0.6.0/src/via_vt1632.h.orig	2020-05-14 12:37:38.342133568 +0000
+++ xf86-video-openchrome-0.6.0/src/via_vt1632.h	2020-05-14 12:38:16.214803967 +0000
@@ -44,7 +44,7 @@
 } viaVT1632Rec, *viaVT1632RecPtr;
 
 
-const xf86OutputFuncsRec via_vt1632_funcs;
+extern const xf86OutputFuncsRec via_vt1632_funcs;
 
 Bool viaVT1632Init(ScrnInfoPtr pScrn, I2CBusPtr pI2CBus);
 
--- xf86-video-openchrome-0.6.0/src/via_sii164.h.orig	2020-05-14 12:39:02.450463330 +0000
+++ xf86-video-openchrome-0.6.0/src/via_sii164.h	2020-05-14 12:39:12.487884649 +0000
@@ -47,7 +47,7 @@
 } viaSiI164Rec, *viaSiI164RecPtr;
 
 
-const xf86OutputFuncsRec via_sii164_funcs;
+extern const xf86OutputFuncsRec via_sii164_funcs;
 
 Bool viaSiI164Init(ScrnInfoPtr pScrn, I2CBusPtr pI2CBus);
 
