--- LuaJIT-2.0.4/Makefile.orig	2016-06-14 11:21:36.062368360 +0200
+++ LuaJIT-2.0.4/Makefile	2016-06-14 11:23:33.116193457 +0200
@@ -25,11 +25,12 @@
 # the paths in src/luaconf.h, too. Note: PREFIX must be an absolute path!
 #
 export PREFIX= /usr/local
+export MULTILIB= bin
 export MULTILIB= lib
 ##############################################################################
 
 DPREFIX= $(DESTDIR)$(PREFIX)
-INSTALL_BIN=   $(DPREFIX)/bin
+INSTALL_BIN=   $(DPREFIX)/$(MULTIBIN)
 INSTALL_LIB=   $(DPREFIX)/$(MULTILIB)
 INSTALL_SHARE= $(DPREFIX)/share
 INSTALL_INC=   $(DPREFIX)/include/luajit-$(MAJVER).$(MINVER)
