--- binutils-2.23.1/bfd/aoutf1.h	2007-07-03 07:26:39.000000000 -0700
+++ binutils-2.23.1/bfd/aoutf1.h	2013-01-15 13:57:43.579172744 -0800
@@ -253,7 +253,7 @@
 /* But the reg structure can be gotten from the SPARC processor handbook.
   This really should be in a GNU include file though so that gdb can use
   the same info.  */
-struct regs
+struct sparc_regs
 {
   int r_psr;
   int r_pc;
@@ -287,7 +287,7 @@
   int c_magic;				/* Corefile magic number.  */
   int c_len;				/* Sizeof (struct core).  */
 #define	SPARC_CORE_LEN	432
-  struct regs c_regs;			/* General purpose registers -- MACHDEP SIZE.  */
+  struct sparc_regs c_regs;		/* General purpose registers -- MACHDEP SIZE.  */
   struct external_exec c_aouthdr;	/* A.out header.  */
   int c_signo;				/* Killing signal, if any.  */
   int c_tsize;				/* Text size (bytes).  */
@@ -310,7 +310,7 @@
   int c_magic;				/* Corefile magic number.  */
   int c_len;				/* Sizeof (struct core).  */
 #define	SOLARIS_BCP_CORE_LEN	456
-  struct regs c_regs;			/* General purpose registers -- MACHDEP SIZE.  */
+  struct sparc_regs c_regs;		/* General purpose registers -- MACHDEP SIZE.  */
   int c_exdata_vp;			/* Exdata structure.  */
   int c_exdata_tsize;
   int c_exdata_dsize;
