--- binutils-2.23.1/binutils/elfcomm.h	2010-11-21 13:27:15.000000000 -0800
+++ binutils-2.23.1/binutils/elfcomm.h	2012-08-28 10:45:54.494740712 -0700
@@ -30,7 +30,7 @@
 void error (const char *, ...) ATTRIBUTE_PRINTF_1;
 void warn (const char *, ...) ATTRIBUTE_PRINTF_1;
 
-#if __STDC_VERSION__ >= 199901L || (defined(__GNUC__) && __GNUC__ >= 2)
+#if __STDC_VERSION__ >= 199901L || (defined(__GNUC__) && __GNUC__ >= 2) || defined(__SUNPRO_C) || defined(__SUNPRO_CC)
 /* We can't use any bfd types here since readelf may define BFD64 and
    objdump may not.  */
 #define HOST_WIDEST_INT	long long
