--- binutils-2.23.1/binutils/readelf.c	2012-11-05 08:27:35.000000000 -0800
+++ binutils-2.23.1/binutils/readelf.c	2012-12-19 08:07:54.220639084 -0800
@@ -7284,9 +7284,15 @@
   if (!do_unwind)
     return;
 
+  /* why do some developers insist on returning from void */
   for (i = 0; handlers[i].handler != NULL; i++)
+  {
     if (elf_header.e_machine == handlers[i].machtype)
-      return handlers[i].handler (file);
+    {
+      handlers[i].handler (file);
+      return;
+    }
+  }
 
   printf (_("\nThe decoding of unwind sections for machine type %s is not currently supported.\n"),
 	  get_machine_name (elf_header.e_machine));
