https://github.com/bustawin/retry-requests/issues/2

--- retry-requests-2.0.0/setup.py.orig
+++ retry-requests-2.0.0/setup.py
@@ -22,8 +22,6 @@
     long_description=Path("README.rst").read_text("utf8"),
     install_requires=["requests", "urllib3>=1.26"],
     extras_require={"test": test_requires},
-    tests_require=test_requires,
-    setup_requires=["pytest-runner"],
     classifiers=[
         "Development Status :: 5 - Production/Stable",
         "Intended Audience :: Developers",
