https://github.com/bustawin/retry-requests/issues/3

--- retry-requests-2.0.0/tests/test_testing.py.orig
+++ retry-requests-2.0.0/tests/test_testing.py
@@ -27,7 +27,7 @@
     """Retries in a non-idempotent method such as a POST."""
     session = retry(session=RSession(), status_to_retry=(200,), retries=1)
     with pytest.raises(requests.exceptions.RetryError):
-        session.post("http://example.com")
+        session.post("http://python.org")
 
 
 def test_retry_success():
