--- ./doc/Makefile.am.orig	2014-10-22 00:06:43.308675070 +0000
+++ ./doc/Makefile.am	2014-10-22 00:07:55.597609984 +0000
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-docdir = $(prefix)/doc/@PACKAGE@
+docdir = $(datadir)/doc/@PACKAGE@
 
 doc_DATA = \
 	CUnit_doc.css \
@@ -13,4 +13,4 @@
 	test_registry.html \
 	writing_tests.html
 
-SUBDIRS = headers
\ No newline at end of file
+SUBDIRS = headers
3c3
--- ./doc/headers/Makefile.am.orig	2014-10-22 00:23:48.224801096 +0000
+++ ./doc/headers/Makefile.am	2014-10-22 00:24:03.103112030 +0000
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-dochdrdir = $(prefix)/doc/@PACKAGE@/headers
+dochdrdir = $(datadir)/doc/@PACKAGE@/headers
 
 INCLUDE_FILES = \
 	Automated.h \
