--- gobject-introspection-0.6.9/gobject-introspection-no-export-1.0.pc.in-orig	2010-04-09 14:41:12.605742721 -0500
+++ gobject-introspection-0.6.9/gobject-introspection-no-export-1.0.pc.in	2010-04-09 14:41:20.293242388 -0500
@@ -6,7 +6,7 @@ datarootdir=@datarootdir@
 datadir=@datadir@
 includedir=@includedir@
 
-g_ir_scanner=${bindir}/g-ir-scanner
+g_ir_scanner=/usr/bin/g-ir-scanner
 g_ir_compiler=${bindir}/g-ir-compiler
 g_ir_generate=${bindir}/g-ir-generate
 girdir=${datadir}/gir-1.0
diff -ruN gobject-introspection-0.9.6.orig/gobject-introspection-1.0.pc.in gobject-introspection-0.9.6/gobject-introspection-1.0.pc.in
--- gobject-introspection-0.9.6.orig/gobject-introspection-1.0.pc.in	2010-09-23 15:25:00.655916670 +0000
+++ gobject-introspection-0.9.6/gobject-introspection-1.0.pc.in	2010-09-23 15:25:40.828645974 +0000
@@ -6,7 +6,7 @@
 datadir=@datadir@
 includedir=@includedir@
 
-g_ir_scanner=${bindir}/g-ir-scanner
+g_ir_scanner=/usr/bin/g-ir-scanner
 g_ir_compiler=${bindir}/g-ir-compiler
 g_ir_generate=${bindir}/g-ir-generate
 gidatadir=${datadir}/gobject-introspection-1.0
