--- gobject-introspection-0.9.12/common.mk.~1~	2014-02-22 11:53:32.687079836 +0400
+++ gobject-introspection-0.9.12/common.mk	2014-02-22 11:53:49.738602174 +0400
@@ -8,6 +8,7 @@
 
 INTROSPECTION_SCANNER = \
     env LPATH=.libs \
+	CFLAGS="$(CFLAGS)" \
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir) \
