--- bluefish-2.2.8/src/bf_lib.c.~1~	2014-11-01 00:06:04.000000000 +0300
+++ bluefish-2.2.8/src/bf_lib.c	2016-05-10 14:47:38.505673779 +0300
@@ -21,14 +21,6 @@
 
 #include "config.h"
 
-/* this is needed for Solaris to comply with the latest POSIX standard
- * regarding the ctime_r() function
- */
-#ifdef PLATFORM_SOLARIS
-#define __EXTENSIONS__
-#define _POSIX_C_SOURCE 200312L
-#endif
-
 #include <gtk/gtk.h>
 #include <glib/gstdio.h>	/* g_mkdir */
 #include <ctype.h>     /* toupper */
