--- VirtualBox-5.2.0/./src/VBox/Additions/x11/vboxvideo/Makefile.kmk.orig	Fri Nov 10 07:51:07 2017
+++ VirtualBox-5.2.0/./src/VBox/Additions/x11/vboxvideo/Makefile.kmk	Fri Nov 10 07:47:12 2017
@@ -397,7 +397,7 @@
 
 ifdef VBOX_USE_SYSTEM_XORG_HEADERS
  # Build using local X.Org headers.  We assume X.Org Server 1.7 or later.
- DLLS    := $(filter-out vboxvideo_drv_%,$(DLLS)) vboxvideo_drv_system
+ DLLS    := $(filter-out vboxvideo_drv_%,$(DLLS))
  SYSMODS := $(filter-out vboxvideo_drv%,$(SYSMODS))
  vboxvideo_drv_system_TEMPLATE = VBOXGUESTR3XORGMOD
  vboxvideo_drv_system_CFLAGS := \
