https://github.com/pytest-dev/pytest-xdist/pull/1266

--- pytest_xdist-3.8.0/testing/test_newhooks.py.orig
+++ pytest_xdist-3.8.0/testing/test_newhooks.py
@@ -115,7 +115,7 @@
         res.stdout.fnmatch_lines_random(["*HOOK: pytest_handlecrashitem"])
         res.stdout.fnmatch_lines(
             [
-                "FAILED test_handlecrashitem_one.py::test_b",
-                "FAILED test_handlecrashitem_one.py::test_b",
+                "FAILED test_handlecrashitem_one.py::test_b*",
+                "FAILED test_handlecrashitem_one.py::test_b*",
             ]
         )
