--- dist/tcsd.conf.in	Thu Jan 28 08:27:50 2010
+++ dist/tcsd.conf.in.new	Fri Nov 18 10:35:22 2011
@@ -37,6 +37,7 @@
 #
 # firmware_log_file = /sys/kernel/security/tpm0/binary_bios_measurements
 #
+firmware_log_file=/var/tpm/system/pcrevent.log
 
 # Option: kernel_log_file
 # Values: Any absolute directory path
@@ -54,7 +55,7 @@
 # Description: A list of PCR indices that are manipulated only by the system
 #  firmware and therefore are not extended or logged by the TCSD.
 #
-# firmware_pcrs =
+firmware_pcrs =0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,20,21
 #
 
 # Option: kernel_pcrs
