diff -wpruN --no-dereference '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -10200,10 +10200,11 @@ printf "%s\n" "#define HAVE_SOLARIS 1" >
 		#
 		# Use System V conventions for man pages.
 		#
-		MAN_ADMIN_COMMANDS=1m
-		MAN_FILE_FORMATS=4
-		MAN_MISC_INFO=5
-		MAN_DEVICES=7D
+		#MAN_ADMIN_COMMANDS=1m
+		#MAN_FILE_FORMATS=4
+		#MAN_MISC_INFO=5
+		#MAN_DEVICES=7D
+		;;
 	esac
 	;;
 esac
diff -wpruN --no-dereference '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -2484,10 +2484,11 @@ solaris*)
 		#
 		# Use System V conventions for man pages.
 		#
-		MAN_ADMIN_COMMANDS=1m
-		MAN_FILE_FORMATS=4
-		MAN_MISC_INFO=5
-		MAN_DEVICES=7D
+		#MAN_ADMIN_COMMANDS=1m
+		#MAN_FILE_FORMATS=4
+		#MAN_MISC_INFO=5
+		#MAN_DEVICES=7D
+		;;
 	esac
 	;;
 esac
