We test in proto area not in site-packages.

--- tox-4.11.2/tox.ini.orig
+++ tox-4.11.2/tox.ini
@@ -29,7 +29,7 @@
     COVERAGE_PROCESS_START = {toxinidir}{/}pyproject.toml
 commands =
     pytest {posargs: \
-      --junitxml {toxworkdir}{/}junit.{envname}.xml --cov {envsitepackagesdir}{/}tox --cov {toxinidir}{/}tests \
+      --junitxml {toxworkdir}{/}junit.{envname}.xml --cov {env:PYTHONPATH}{/}tox --cov {toxinidir}{/}tests \
       --cov-config={toxinidir}{/}pyproject.toml --no-cov-on-fail --cov-report term-missing:skip-covered --cov-context=test \
       --cov-report html:{envtmpdir}{/}htmlcov \
       --cov-report xml:{toxworkdir}{/}coverage.{envname}.xml \
