We have no rednose packaged yet.

--- sure-2.0.1/tox.ini.orig
+++ sure-2.0.1/tox.ini
@@ -7,10 +7,9 @@
 envlist = py27, pypy, py34, py35, py36, py37
 
 [testenv]
-commands = nosetests --rednose -vv --with-coverage --cover-package=sure
+commands = nosetests -vv --with-coverage --cover-package=sure
 deps =
     six
     mock
     nose
-    rednose
     coverage
--- sure-2.0.1/setup.cfg.orig
+++ sure-2.0.1/setup.cfg
@@ -1,6 +1,5 @@
 [nosetests]
 verbosity = 3
-rednose = 1
 with-coverage = 1
 cover-inclusive = 1
 cover-package = sure
