From 0378ee248e0193990a23eae79762208e5d24d0ec Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Mon, 28 Nov 2022 14:52:31 +0000
Subject: illumos OFD locking must span the entire file

---
 util/osdep.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/osdep.c b/util/osdep.c
index 4a8b8b5a90..1e5a9e2e71 100644
--- a/util/osdep.c
+++ b/util/osdep.c
@@ -31,6 +31,8 @@
 #include "qemu/hw-version.h"
 #include "monitor/monitor.h"
 
+#undef F_OFD_SETLK
+
 int socket_set_cork(int fd, int v)
 {
 #if defined(SOL_TCP) && defined(TCP_CORK)
