From f123c03e547fea55d6ef9dc4e28641d7a9bcb000 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <illumos@fiddaman.net>
Date: Mon, 28 Nov 2022 14:53:22 +0000
Subject: Do not pass --dynamic-list to linker

---
 plugins/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plugins/meson.build b/plugins/meson.build
index 3be8245a69..a3a328b7f2 100644
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -20,8 +20,6 @@ if not enable_modules
   elif host_os == 'windows' and meson.get_compiler('c').get_id() == 'clang'
     # LLVM/lld does not support exporting specific symbols. However, it works
     # out of the box with dllexport/dllimport attribute we set in the code.
-  else
-    emulator_link_args += ['-Xlinker', '--dynamic-list=' + qemu_plugin_symbols.full_path()]
   endif
 endif
 
