diff --git a/src/librustc_trans/back/linker.rs b/src/librustc_trans/back/linker.rs
index aa29c3c..48a4e33 100644
--- a/src/librustc_trans/back/linker.rs
+++ b/src/librustc_trans/back/linker.rs
@@ -275,7 +275,7 @@ impl<'a> Linker for GccLinker<'a> {
 
     fn no_default_libraries(&mut self) {
         if !self.is_ld {
-            self.cmd.arg("-nodefaultlibs");
+            self.cmd.arg("-Wl,-znodefaultlib");
         }
     }
 
