diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index 8d6c2db..6f4392e 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -956,7 +956,7 @@ impl<'a> Builder<'a> {
                 rustflags.arg("-Zosx-rpath-install-name");
                 Some("-Wl,-rpath,@loader_path/../lib")
             } else if !target.contains("windows") {
-                Some("-Wl,-rpath,$ORIGIN/../lib")
+                Some("-Wl,-rpath,/usr/clang/9.0/lib")
             } else {
                 None
             };
