diff --git a/src/libcompiler_builtins/compiler-rt/lib/asan/CMakeLists.txt b/src/libcompiler_builtins/compiler-rt/lib/asan/CMakeLists.txt
index 5ac5708..c096a30 100644
--- src/libcompiler_builtins/compiler-rt/lib/asan/CMakeLists.txt
+++ src/libcompiler_builtins/compiler-rt/lib/asan/CMakeLists.txt
@@ -219,7 +219,7 @@ else()
       if(MSVC)
         list(APPEND DYNAMIC_RUNTIME_THUNK_CFLAGS "-Zl")
       elseif(CMAKE_C_COMPILER_ID MATCHES Clang)
-        list(APPEND DYNAMIC_RUNTIME_THUNK_CFLAGS "-nodefaultlibs")
+        list(APPEND DYNAMIC_RUNTIME_THUNK_CFLAGS "-Wl,-znodefaultlib")
       endif()
 
       add_compiler_rt_runtime(clang_rt.asan_dynamic_runtime_thunk
