# Copied over from the desktop consolidation, not for upstream

--- rrdtool-1.4.3/bindings/lua/rrdlua.c 2014-02-03 13:17:19.604959810 +0530
+++ rrdtool-1.4.3/bindings/lua/rrdlua.c 2014-02-03 13:18:21.442352916 +0530
@@ -342,7 +342,7 @@
 
 /**********************************************************/
 
-static const struct luaL_reg rrd[] = {
+static const struct luaL_Reg rrd[] = {
   {"create", lua_rrd_create},
   {"dump", lua_rrd_dump},
   {"fetch", lua_rrd_fetch},
@@ -373,7 +373,7 @@ luaopen_rrd (lua_State * L)
   /* luaL_module is defined in compat-5.1.c */
   luaL_module (L, "rrd", rrd, 0);
 #else
-  luaL_register (L, "rrd", rrd);
+  lua_newlib(L, rrd);
 #endif
   set_info (L);
   return 1;
