--- gqview-2.0.4/src/editors.c.orig	2008-07-17 16:30:04.750522000 +0800
+++ gqview-2.0.4/src/editors.c	2008-07-17 16:30:08.095793000 +0800
@@ -47,10 +47,21 @@
 
 
 static gchar *editor_slot_defaults[] = {
-	N_("The Gimp"), "gimp-remote -n %f",
+#ifdef EDITOR_GIMP
+	N_("The Gimp"), "gimp-remote %f",
+#else 
+        NULL, NULL,
+#endif
+#ifdef EDITOR_XV
 	N_("XV"), "xv %f",
+#else
+        NULL, NULL,
+#endif
+#ifdef EDITOR_XPAINT
 	N_("Xpaint"), "xpaint %f",
+#else
 	NULL, NULL,
+#endif
 	NULL, NULL,
 	NULL, NULL,
 	NULL, NULL,
