diff -urN gobject-introspection-1.40.0.orig/gobject-introspection-1.0.pc.in gobject-introspection-1.40.0/gobject-introspection-1.0.pc.in
--- gobject-introspection-1.40.0.orig/gobject-introspection-1.0.pc.in	2014-02-23 00:47:08.000000000 +0100
+++ gobject-introspection-1.40.0/gobject-introspection-1.0.pc.in	2016-03-31 22:49:18.655945484 +0200
@@ -6,7 +6,7 @@
 datadir=@datadir@
 includedir=@includedir@

-g_ir_scanner=${bindir}/g-ir-scanner
+g_ir_scanner=/usr/bin/g-ir-scanner
 g_ir_compiler=${bindir}/g-ir-compiler@EXEEXT@
 g_ir_generate=${bindir}/g-ir-generate@EXEEXT@
 gidatadir=${datadir}/gobject-introspection-1.0
