--- gobject-introspection-1.60.0/giscanner/ccompiler.py	2019-05-17 13:09:44.222832279 +0000
+++ gobject-introspection-1.60.0/giscanner/ccompiler.py.new	2019-05-17 13:15:25.749587042 +0000
@@ -195,7 +195,7 @@ class CCompiler(object):

                 # Ensure libraries are always linked as we are going to use ldd to work
                 # out their names later
-                if sys.platform != 'darwin':
+                if sys.platform != 'darwin' and sys.platform != 'sunos5':
                     args.append('-Wl,--no-as-needed')

         for library_path in libpaths:
