diff -urN gobject-introspection-1.40.0.orig/common.mk gobject-introspection-1.40.0/common.mk
--- gobject-introspection-1.40.0.orig/common.mk	2014-02-21 13:45:34.000000000 +0100
+++ gobject-introspection-1.40.0/common.mk	2016-03-31 23:10:48.054949788 +0200
@@ -8,6 +8,7 @@

 INTROSPECTION_SCANNER = \
     env PATH=".libs:$(PATH)" \
+        CFLAGS="$(CFLAGS)" \
         LPATH=.libs \
         CC="$(CC)" \
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
