--- gobject-introspection-1.43.3/giscanner/ccompiler.py.1	2017-10-10 14:29:35.253122852 +0000
+++ gobject-introspection-1.43.3/giscanner/ccompiler.py	2017-10-10 14:29:57.699699380 +0000
@@ -73,8 +73,8 @@
 
         # Ensure libraries are always linked as we are going to use ldd to work
         # out their names later
-        if not libtool and pkgconfig_msvc_flags == '':
-            args.append('-Wl,--no-as-needed')
+        # if not libtool and pkgconfig_msvc_flags == '':
+        #    args.append('-Wl,--no-as-needed')
 
         for library in libraries:
             # Visual C++: We have the needed .lib files now, and we need to link
