--- evolution-data-server-3.24.0/CMakeLists.txt.~2~	2018-04-04 18:29:35.202458488 +0000
+++ evolution-data-server-3.24.0/CMakeLists.txt	2018-04-04 18:35:19.746923141 +0000
@@ -130,7 +130,7 @@
 set(CMAKE_SKIP_RPATH OFF)
 set(CMAKE_SKIP_BUILD_RPATH OFF CACHE BOOL INTERNAL)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF CACHE BOOL INTERNAL)
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON CACHE BOOL INTERNAL)
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH OFF CACHE BOOL INTERNAL)
 # CMAKE_INSTALL_RPATH is set below
 
 # CMake modules
@@ -180,7 +180,7 @@
 set(privlibdir "${LIB_INSTALL_DIR}/${PROJECT_NAME}")
 set(privlibexecdir "${LIBEXEC_INSTALL_DIR}/${PROJECT_NAME}")
 
-SET(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}:${privlibdir}")
+SET(CMAKE_INSTALL_RPATH "${privlibdir}")
 
 if(WIN32)
 	# On Win32 there is no "rpath" mechanism. We install the private
