--- php-5.2.17/configure.in.orig	Thu Jan 19 05:56:12 2012
+++ php-5.2.17/configure.in	Thu Jan 19 05:56:27 2012
@@ -1,7 +1,6 @@
 ## $Id: configure.in 307193 2011-01-06 23:01:19Z iliaa $ -*- autoconf -*-
 dnl ## Process this file with autoconf to produce a configure script.
 
-divert(1)
 
 dnl ## Diversion 1 is the autoconf + automake setup phase. We also
 dnl ## set the PHP version, deal with platform-specific compile
@@ -256,7 +255,6 @@
 sinclude(TSRM/tsrm.m4)
 
 
-divert(2)
 
 dnl ## Diversion 2 is where we set PHP-specific options and come up
 dnl ## with reasonable default values for them. We check for pthreads here
@@ -295,7 +293,6 @@
   PTHREADS_FLAGS
 fi
 
-divert(3)
 
 dnl ## In diversion 3 we check for compile-time options to the PHP
 dnl ## core and how to deal with different system dependencies.
@@ -655,7 +652,6 @@
   PHP_CRYPT_R_STYLE
 fi
 
-divert(4)
 
 dnl ## In diversion 4 we check user-configurable general settings.
 
@@ -913,7 +909,6 @@
   AC_MSG_RESULT([using system default])
 fi
 
-divert(5)
 
 dnl ## In diversion 5 we check which extensions should be compiled.
 dnl ## All of these are normally in the extension directories.
--- php-5.2.17/ext/standard/config.m4.orig	Thu Jan 19 05:57:19 2012
+++ php-5.2.17/ext/standard/config.m4	Thu Jan 19 05:59:05 2012
@@ -1,6 +1,5 @@
 dnl $Id: config.m4 291364 2009-11-27 23:41:13Z rasmus $ -*- autoconf -*-
 
-divert(3)dnl
 
 dnl
 dnl Check if flush should be called explicitly after buffered io
@@ -205,7 +204,6 @@
 AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass isinf isnan)
 AC_FUNC_FNMATCH	
 
-divert(5)dnl
 
 dnl
 dnl Check for regex library type
--- php-5.2.17/scripts/phpize.m4.orig	Thu Jan 19 05:57:11 2012
+++ php-5.2.17/scripts/phpize.m4	Thu Jan 19 05:58:52 2012
@@ -1,6 +1,5 @@
 dnl This file becomes configure.in for self-contained extensions.
 
-divert(1)
 
 AC_PREREQ(2.13)
 AC_INIT(config.m4)
