--- janet-1.41.2/src/core/net.c.old	2026-03-03 22:54:19.741409004 -0500
+++ janet-1.41.2/src/core/net.c	2026-03-03 22:54:44.916560159 -0500
@@ -1122,7 +1122,7 @@
         val.v_int = janet_getboolean(argv, 2);
         optlen = sizeof(val.v_int);
     } else if (st->type == JANET_NUMBER) {
-#ifdef JANET_BSD
+#if defined(JANET_BSD) || defined(JANET_ILLUMOS)
         int v_int = janet_getinteger(argv, 2);
         if (st->optname == IP_MULTICAST_TTL) {
             val.v_uchar = v_int;
