--- pyproject_api-1.7.1/tests/test_frontend_setuptools.py.orig
+++ pyproject_api-1.7.1/tests/test_frontend_setuptools.py
@@ -56,6 +56,7 @@
     assert isinstance(result.err, str)
 
 
+@pytest.mark.skip(reason="always fails: https://github.com/tox-dev/pyproject-api/issues/142")
 def test_setuptools_get_requires_for_build_wheel(frontend_setuptools: SubprocessFrontend) -> None:
     result = frontend_setuptools.get_requires_for_build_wheel()
     assert not result.requires
--- pyproject_api-1.7.1/pyproject.toml.orig
+++ pyproject_api-1.7.1/pyproject.toml
@@ -140,6 +140,7 @@
 ]
 report.omit = [
 ]
+report.fail_under = 95
 run.parallel = true
 run.plugins = [
   "covdefaults",
