--- pyproject_api-1.4.0/tox.ini.orig
+++ pyproject_api-1.4.0/tox.ini
@@ -23,7 +23,7 @@
 extras =
     testing
 commands =
-    pytest {tty:--color=yes} {posargs: --no-cov-on-fail --cov-context=test \
+    pytest {posargs: --no-cov-on-fail --cov-context=test \
       --cov={envsitepackagesdir}{/}pyproject_api --cov={toxinidir}{/}tests --cov-config={toxinidir}{/}pyproject.toml \
       --cov-report=term-missing:skip-covered --cov-report=html:{envtmpdir}{/}htmlcov \
       --cov-report=xml:{toxworkdir}{/}coverage.{envname}.xml --junitxml={toxworkdir}{/}junit.{envname}.xml \
