We test in proto area not in site-packages.

--- pyproject_api-1.7.1/tox.ini.orig
+++ pyproject_api-1.7.1/tox.ini
@@ -28,7 +28,7 @@
     COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}{/}.coverage.{envname}}
 commands =
     pytest {tty:--color=yes} {posargs: --no-cov-on-fail --cov-context=test \
-      --cov={envsitepackagesdir}{/}pyproject_api --cov={toxinidir}{/}tests --cov-config={toxinidir}{/}pyproject.toml \
+      --cov=pyproject_api --cov={toxinidir}{/}tests --cov-config={toxinidir}{/}pyproject.toml \
       --cov-report=term-missing:skip-covered --cov-report=html:{envtmpdir}{/}htmlcov \
       --cov-report=xml:{toxworkdir}{/}coverage.{envname}.xml --junitxml={toxworkdir}{/}junit.{envname}.xml \
       tests}
