--- /tmp/Makefile	Mon Feb 14 14:59:22 2011
+++ openssl-1.0.0d/engines/Makefile	Mon Feb 14 15:00:35 2011
@@ -26,7 +26,8 @@
 APPS=
 
 LIB=$(TOP)/libcrypto.a
-LIBNAMES= 4758cca aep atalla cswift gmp chil nuron sureware ubsec padlock capi
+LIBNAMES= 4758cca aep atalla cswift gmp chil nuron sureware ubsec padlock capi \
+	  devcrypto
 
 LIBSRC=	e_4758cca.c \
 	e_aep.c \
@@ -38,7 +39,8 @@
 	e_sureware.c \
 	e_ubsec.c \
 	e_padlock.c \
-	e_capi.c
+	e_capi.c \
+	e_devcrypto.c
 LIBOBJ= e_4758cca.o \
 	e_aep.o \
 	e_atalla.o \
@@ -49,7 +51,8 @@
 	e_sureware.o \
 	e_ubsec.o \
 	e_padlock.o \
-	e_capi.o
+	e_capi.o \
+	e_devcrypto.o
 
 SRC= $(LIBSRC)
 
@@ -63,7 +66,8 @@
 	e_nuron_err.c e_nuron_err.h \
 	e_sureware_err.c e_sureware_err.h \
 	e_ubsec_err.c e_ubsec_err.h \
-	e_capi_err.c e_capi_err.h
+	e_capi_err.c e_capi_err.h \
+	e_devcrypto_err.c e_devcrypto_err.h
 
 ALL=    $(GENERAL) $(SRC) $(HEADER)
 
@@ -78,7 +82,7 @@
 		for l in $(LIBNAMES); do \
 			$(MAKE) -f ../Makefile.shared -e \
 				LIBNAME=$$l LIBEXTRAS=e_$$l.o \
-				LIBDEPS='-L.. -lcrypto $(EX_LIBS)' \
+				LIBDEPS='-L.. -lcrypto -lcryptoutil $(EX_LIBS)' \
 				link_o.$(SHLIB_TARGET); \
 		done; \
 	else \
