--- MEGAsync-4.9.5.0_Linux/src/MEGASync/gui/ProxySettings.cpp.orig
+++ MEGAsync-4.9.5.0_Linux/src/MEGASync/gui/ProxySettings.cpp
@@ -31,7 +31,7 @@
     connect(mUi->rProxyManual, &QRadioButton::clicked, this, [this]{setManualMode(true);});
     connect(mUi->cProxyRequiresPassword, &QCheckBox::toggled, this, [this]{setManualMode(true);});
     connect(mUi->rNoProxy, &QRadioButton::clicked, this, [this]{setManualMode(false);});
-#ifndef Q_OS_LINUX
+#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
     connect(mUi->rProxyAuto, &QRadioButton::clicked, this, [this]{setManualMode(false);});
 #endif
 }
@@ -45,7 +45,7 @@
 void ProxySettings::initialize()
 {
     mUi->rNoProxy->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_NONE);
-#ifndef Q_OS_LINUX
+#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
     mUi->rProxyAuto->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_AUTO);
 #endif
     mUi->rProxyManual->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_CUSTOM);
@@ -98,7 +98,7 @@
     {
         mPreferences->setProxyType(Preferences::PROXY_TYPE_NONE);
     }
-#ifndef Q_OS_LINUX
+#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
     else if (mUi->rProxyAuto->isChecked())
     {
         mPreferences->setProxyType(Preferences::PROXY_TYPE_AUTO);
@@ -148,7 +148,7 @@
             proxy.setPassword(mUi->eProxyPassword->text());
         }
     }
-#ifndef Q_OS_LINUX
+#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
     else if (mUi->rProxyAuto->isChecked())
     {
         MegaProxy *proxySettings = mApp->getMegaApi()->getAutoProxySettings();
