diff -Nru MEGAsync-4.7.3.0_Linux.orig/src/MEGASync/transfers/gui/TransfersSummaryWidget.cpp MEGAsync-4.7.3.0_Linux/src/MEGASync/transfers/gui/TransfersSummaryWidget.cpp
--- MEGAsync-4.7.3.0_Linux.orig/src/MEGASync/transfers/gui/TransfersSummaryWidget.cpp	2022-11-15 15:47:34.000000000 +0100
+++ MEGAsync-4.7.3.0_Linux/src/MEGASync/transfers/gui/TransfersSummaryWidget.cpp	2023-03-12 15:16:04.596792703 +0100
@@ -9,6 +9,7 @@
 
 #include "TransferItem.h"
 #include "Utilities.h"
+#include "math.h"
 
 TransfersSummaryWidget::TransfersSummaryWidget(QWidget *parent) :
     QWidget(parent),
