diff -Nru MEGAsync-4.9.5.0_Linux.orig/src/MEGASync/mega/src/gfx/freeimage.cpp MEGAsync-4.9.5.0_Linux/src/MEGASync/mega/src/gfx/freeimage.cpp
--- MEGAsync-4.9.5.0_Linux.orig/src/MEGASync/mega/src/gfx/freeimage.cpp	2023-06-08 09:24:13.000000000 +0200
+++ MEGAsync-4.9.5.0_Linux/src/MEGASync/mega/src/gfx/freeimage.cpp	2023-10-03 09:47:52.023993600 +0200
@@ -209,12 +209,6 @@
 
 #ifdef HAVE_FFMPEG
 
-#ifdef AV_CODEC_CAP_TRUNCATED
-#define CAP_TRUNCATED AV_CODEC_CAP_TRUNCATED
-#else
-#define CAP_TRUNCATED CODEC_CAP_TRUNCATED
-#endif
-
 const char *GfxProviderFreeImage::supportedformatsFfmpeg()
 {
     return  ".264.265.3g2.3gp.3gpa.3gpp.3gpp2.mp3"
@@ -330,10 +324,6 @@
 
     // Force seeking to key frames
     formatContext->seek2any = false;
-    if (decoder->capabilities & CAP_TRUNCATED)
-    {
-        codecContext->flags |= CAP_TRUNCATED;
-    }
 
     AVPixelFormat sourcePixelFormat = static_cast<AVPixelFormat>(codecParm->format);
     AVPixelFormat targetPixelFormat = AV_PIX_FMT_BGR24; //raw data expected by freeimage is in this format
