diff -Nru MEGAsync-4.7.3.0_Linux.orig/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc MEGAsync-4.7.3.0_Linux/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc
--- MEGAsync-4.7.3.0_Linux.orig/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc	2022-11-15 15:47:34.000000000 +0100
+++ MEGAsync-4.7.3.0_Linux/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc	2023-03-12 15:27:59.368465153 +0100
@@ -39,11 +39,13 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <pthread.h>
 
 #include <algorithm>
 #include <cassert>
 #include <cstdio>
 #include <cstdlib>
+#include <cstring>
 #include <functional>
 
 #include "client/solaris/handler/solaris_lwp.h"
