--- MEGAsync-4.9.5.0_Linux/src/MEGASync/MegaApplication.cpp.orig
+++ MEGAsync-4.9.5.0_Linux/src/MEGASync/MegaApplication.cpp
@@ -37,7 +37,7 @@
 
 #include <assert.h>
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
     #include <signal.h>
     #include <condition_variable>
     #include <QSvgRenderer>
@@ -73,7 +73,7 @@
 
 void MegaApplication::loadDataPath()
 {
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
     dataPath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
                + QString::fromUtf8("/data/") // appending "data" is non-standard behavior according to XDG Base Directory specification
                + organizationName() + QString::fromLatin1("/")
@@ -135,7 +135,7 @@
         args += QString::fromUtf8(argv[i]);
     }
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
 
     if (args.contains(QLatin1String("--version")))
     {
@@ -417,7 +417,7 @@
     paused = false;
     indexing = false;
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
     isLinux = true;
 #else
     isLinux = false;
@@ -744,7 +744,7 @@
     createTrayIcon();
 }
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
 void MegaApplication::setTrayIconFromTheme(QString icon)
 {
     QString name = QString(icon).replace(QString::fromUtf8("://images/"), QString::fromUtf8("mega")).replace(QString::fromUtf8(".svg"),QString::fromUtf8(""));
@@ -1007,7 +1007,7 @@
 
 void MegaApplication::start()
 {
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
     QSvgRenderer qsr; //to have svg library linked
 #endif
 
@@ -2143,7 +2143,7 @@
 
     if (trayIcon)
     {
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
         if (counter==4 && getenv("XDG_CURRENT_DESKTOP") && !strcmp(getenv("XDG_CURRENT_DESKTOP"),"XFCE"))
         {
             trayIcon->hide();
@@ -5744,7 +5744,7 @@
         whyAmIBlocked();
     }
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
     if (getenv("XDG_CURRENT_DESKTOP") && (
                 !strcmp(getenv("XDG_CURRENT_DESKTOP"),"ubuntu:GNOME")
                 || !strcmp(getenv("XDG_CURRENT_DESKTOP"),"LXDE")
