--- MEGAsync-4.9.5.0_Linux/src/MEGASync/gui/node_selector/model/NodeSelectorModel.cpp.orig
+++ MEGAsync-4.9.5.0_Linux/src/MEGASync/gui/node_selector/model/NodeSelectorModel.cpp
@@ -1127,6 +1127,8 @@
                             const QIcon thisDeviceIcon (QLatin1String("://images/icons/pc/pc-mac_24.png"));
 #elif defined(Q_OS_LINUX)
                             const QIcon thisDeviceIcon (QLatin1String("://images/icons/pc/pc-linux_24.png"));
+#else
+                            const QIcon thisDeviceIcon (QLatin1String("://images/icons/pc/pc_24.png"));
 #endif
                             return thisDeviceIcon;
                         }
