diff -Nru MEGAsync-4.9.0.0_Linux.orig/src/MEGASync/syncs/gui/SyncsMenu.cpp MEGAsync-4.9.0.0_Linux/src/MEGASync/syncs/gui/SyncsMenu.cpp
--- MEGAsync-4.9.0.0_Linux.orig/src/MEGASync/syncs/gui/SyncsMenu.cpp	2023-03-06 09:58:18.000000000 +0100
+++ MEGAsync-4.9.0.0_Linux/src/MEGASync/syncs/gui/SyncsMenu.cpp	2023-03-12 19:32:41.851864231 +0100
@@ -16,6 +16,8 @@
 const QLatin1String DEVICE_ICON ("://images/icons/pc/pc-mac_24.png");
 #elif defined(Q_OS_LINUX)
 const QLatin1String DEVICE_ICON ("://images/icons/pc/pc-linux_24.png");
+#else
+const QLatin1String DEVICE_ICON ("://images/icons/pc/pc_24.png");
 #endif
 
 SyncsMenu::SyncsMenu(mega::MegaSync::SyncType type, QObject *parent) : QObject(parent),
