--- MEGAsync-4.9.5.0_Linux/src/MEGASync/syncs/gui/SyncsMenu.cpp.orig
+++ MEGAsync-4.9.5.0_Linux/src/MEGASync/syncs/gui/SyncsMenu.cpp
@@ -16,6 +16,8 @@
 const QLatin1String DEVICE_ICON ("://images/icons/pc/pc-mac_24.png");
 #elif defined(Q_OS_LINUX)
 const QLatin1String DEVICE_ICON ("://images/icons/pc/pc-linux_24.png");
+#else
+const QLatin1String DEVICE_ICON ("://images/icons/pc/pc_24.png");
 #endif
 
 SyncsMenu::SyncsMenu(mega::MegaSync::SyncType type, QObject *parent) : QObject(parent),
