--- faac-1.28/common/mp4v2/rtphint.cpp.orig	2018-05-31 17:02:50.031955269 +0000
+++ faac-1.28/common/mp4v2/rtphint.cpp	2018-05-31 17:03:48.747513353 +0000
@@ -342,7 +342,7 @@
 			  pSlash = strchr(pSlash, '/');
 			  if (pSlash != NULL) {
 			    pSlash++;
-			    if (pSlash != '\0') {
+			    if (*pSlash != '\0') {
 			      length = strlen(pRtpMap) - (pSlash - pRtpMap);
 			      *ppEncodingParams = (char *)MP4Calloc(length + 1);
 			      strncpy(*ppEncodingParams, pSlash, length);
