--- gcc-4.9.3/libstdc++-v3/acinclude.m4.1	2015-11-10 23:02:48.144804023 +0300
+++ gcc-4.9.3/libstdc++-v3/acinclude.m4	2015-11-10 23:02:57.650586415 +0300
@@ -1233,7 +1233,7 @@
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+        GLIBCXX_LIBS="$GLIBCXX_LIBS"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
--- gcc-4.9.3/libstdc++-v3/configure.1	2015-11-10 23:01:27.375669063 +0300
+++ gcc-4.9.3/libstdc++-v3/configure	2015-11-10 23:02:02.171610827 +0300
@@ -19507,7 +19507,7 @@
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+        GLIBCXX_LIBS="$GLIBCXX_LIBS"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
