--- parmap-1.0-rc11/Makefile.in.orig	2020-05-28 22:59:05.872447418 +0000
+++ parmap-1.0-rc11/Makefile.in	2020-05-28 22:59:21.553964408 +0000
@@ -88,11 +88,11 @@
 	$(INSTALL) -patch-version $(VERSION) $(NAME) $(INSTALL_STUFF)
 	(cd _build; ocamldoc -man -man-mini parmap.ml parmap.mli)
 	test -d $(MANDIR) || mkdir -p $(MANDIR)
-	cp -p _build/Parmap.3o $(MANDIR)
+	cp -p _build/Parmap.3 $(MANDIR)
 
 uninstall:
 	$(UNINSTALL) $(NAME)
-	rm -f $(MANDIR)/Parmap.3o
+	rm -f $(MANDIR)/Parmap.3
 
 doc:
 	$(OCAMLBUILD) $(OBFLAGS) $(NAME).docdir/index.html $(NAME).docdir/index.dot
