--- gtk-4.16.1/testsuite/gtk/meson.build.orig
+++ gtk-4.16.1/testsuite/gtk/meson.build
@@ -1,5 +1,3 @@
-gtk_tests_export_dynamic_ldflag = []
-
 if cc.get_id() != 'msvc'
   if os_darwin
     gtk_tests_export_dynamic_ldflag = ['-Wl,-export_dynamic']
@@ -10,6 +8,8 @@
   endif
 endif
 
+gtk_tests_export_dynamic_ldflag = []
+
 # Available keys for each test:
 #
 #  - 'name': the test name; used for the test and to determine the base
