We have no pytest-dependency packaged yet.

--- codespell-2.4.1/codespell_lib/tests/test_dictionary.py.orig
+++ codespell-2.4.1/codespell_lib/tests/test_dictionary.py
@@ -293,7 +293,6 @@
 
 
 @fname_params
-@pytest.mark.dependency(name="dictionary loop")
 def test_dictionary_looping(
     fname: str,
     in_aspell: Tuple[bool, bool],
@@ -363,7 +362,6 @@
     global_err_dicts[short_fname] = this_err_dict
 
 
-@pytest.mark.dependency(depends=["dictionary loop"])
 def test_ran_all() -> None:
     """Test that all pairwise tests ran."""
     for f1, _, _ in _fnames_in_aspell:
--- codespell-2.4.1/pyproject.toml.orig
+++ codespell-2.4.1/pyproject.toml
@@ -38,7 +38,6 @@
     "pre-commit",
     "pytest",
     "pytest-cov",
-    "pytest-dependency",
     "Pygments",
     "ruff",
     "tomli",
