Add auditing support to agent execution

--- puppet-3.8.6/lib/puppet/agent.rb.orig	2016-04-19 14:23:14.948120246 -0700
+++ puppet-3.8.6/lib/puppet/agent.rb	2016-04-19 14:24:09.178244630 -0700
@@ -1,4 +1,11 @@
+#######################################################################
+# Oracle has modified the originally distributed contents of this file.
+#######################################################################
+
 require 'puppet/application'
+if Facter.value(:osfamily) == "Solaris"
+  require 'PuppetAudit'
+end
 
 # A general class for triggering a run of another
 # class.
@@ -37,6 +44,11 @@
       return
     end
 
+    if Facter.value(:osfamily) == "Solaris"
+        audit_handle = PuppetAudit.new
+        audit_handle.audit_start
+    end
+
     result = nil
     block_run = Puppet::Application.controlled_run do
       splay client_options.fetch :splay, Puppet[:splay]
@@ -52,6 +64,10 @@
           end
         end
       end
+      if Facter.value(:osfamily) == "Solaris"
+          audit_handle.audit_stop
+          audit_handle.audit_teardown
+      end
       true
     end
     Puppet.notice "Shutdown/restart in progress (#{Puppet::Application.run_status.inspect}); skipping run" unless block_run
