Not submitted to upstream

diff --git a/lib/puppet/application/apply.rb b/lib/puppet/application/apply.rb
--- a/lib/puppet/application/apply.rb
+++ b/lib/puppet/application/apply.rb
@@ -1,6 +1,9 @@
 require 'puppet/application'
 require 'puppet/configurer'
 require 'puppet/util/profiler/aggregate'
+if Facter.value(:osfamily) == "Solaris"
+  require 'PuppetAudit'
+end
 
 class Puppet::Application::Apply < Puppet::Application
   require 'puppet/util/splayer'
@@ -182,6 +185,12 @@ Copyright (c) 2011 Puppet Labs, LLC Licensed under the Apache 2.0 License
   end
 
   def main
+    # configure auditing on Solaris
+    if Facter.value(:osfamily) == "Solaris"
+        audit_handle = PuppetAudit.new
+        audit_handle.audit_start
+    end
+
     # Set our code or file to use.
     if options[:code] or command_line.args.length == 0
       Puppet[:code] = options[:code] || STDIN.read
@@ -287,6 +296,11 @@ Copyright (c) 2011 Puppet Labs, LLC Licensed under the Apache 2.0 License
     end
 
   ensure
+    # clean up auditing records on Solaris
+    if Facter.value(:osfamily) == "Solaris"
+      audit_handle.audit_stop
+      audit_handle.audit_teardown
+    end
     if @profiler
       Puppet::Util::Profiler.remove_profiler(@profiler)
       @profiler.shutdown
