--- git-2.8.1/gpg-interface.c.orig	2016-04-25 19:39:45.502780368 +0300
+++ git-2.8.1/gpg-interface.c	2016-04-25 19:39:55.149402601 +0300
@@ -5,7 +5,7 @@
 #include "sigchain.h"

 static char *configured_signing_key;
-static const char *gpg_program = "gpg";
+static const char *gpg_program = "gpg2";

 #define PGP_SIGNATURE "-----BEGIN PGP SIGNATURE-----"
 #define PGP_MESSAGE "-----BEGIN PGP MESSAGE-----"
