https://lore.kernel.org/git/ZkdZRAn6KGgxBnT5@telcontar/

--- git-2.45.1/t/t0600-reffiles-backend.sh.orig
+++ git-2.45.1/t/t0600-reffiles-backend.sh
@@ -428,7 +428,7 @@
 	test_when_finished "rm -rf subdir" &&
 	git init --bare subdir &&
 
-	rm -rfv subdir/refs subdir/objects subdir/packed-refs &&
+	rm -rf subdir/refs subdir/objects subdir/packed-refs &&
 	ln -s ../.git/refs subdir/refs &&
 	ln -s ../.git/objects subdir/objects &&
 	ln -s ../.git/packed-refs subdir/packed-refs &&
