--- git-2.56.0/t/test-lib-functions.sh.orig	2026-09-28 06:30:55.000000000 +0200
+++ git-2.56.0/t/test-lib-functions.sh	2026-09-30 22:25:37.821686700 +0200
@@ -1335,11 +1335,11 @@
 	if test "x!" = "x$1"
 	then
 		shift
-		! grep "$@" && return 0
+		! ggrep "$@" && return 0
 
 		echo >&4 "error: '! grep $@' did find a match in:"
 	else
-		grep "$@" && return 0
+		ggrep "$@" && return 0
 
 		echo >&4 "error: 'grep $@' didn't find a match in:"
 	fi
