Make sure configure recognizes solaris.

--- parted-1.8.8/configure.orig
+++ parted-1.8.8/configure
@@ -2758,6 +2758,7 @@
 	linux*) OS=linux ;;
 	gnu*)	OS=gnu ;;
 	beos*)	OS=beos ;;
+	solaris*)	OS=solaris ;;
 	*)	{ { echo "$as_me:$LINENO: error: Unknown or unsupported OS \"$host_os\".  Only \"linux\", \"gnu\" and \"beos\" are supported in this version of GNU Parted." >&5
 echo "$as_me: error: Unknown or unsupported OS \"$host_os\".  Only \"linux\", \"gnu\" and \"beos\" are supported in this version of GNU Parted." >&2;}
    { (exit 1); exit 1; }; } ;;
