We do not want to distribute gnu.h and linux.h.

--- parted-1.8.8/include/parted/Makefile.in.orig
+++ parted-1.8.8/include/parted/Makefile.in
@@ -81,7 +81,7 @@
 CONFIG_CLEAN_FILES =
 SOURCES =
 DIST_SOURCES =
-am__partedinclude_HEADERS_DIST = gnu.h linux.h constraint.h debug.h \
+am__partedinclude_HEADERS_DIST = constraint.h debug.h \
 	device.h disk.h exception.h filesys.h geom.h natmath.h timer.h \
 	unit.h parted.h fdasd.h vtoc.h solaris.h
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
@@ -368,9 +368,7 @@
 @COMPILE_FOR_S390_FALSE@S390_HDRS = 
 @COMPILE_FOR_S390_TRUE@S390_HDRS = fdasd.h vtoc.h
 partedincludedir = $(includedir)/parted
-partedinclude_HEADERS = gnu.h		\
-			linux.h		\
-			constraint.h	\
+partedinclude_HEADERS = constraint.h	\
 			debug.h		\
 			device.h	\
 			disk.h		\
