https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/3551

--- gstreamer-1.24.3/libs/gst/helpers/ptp/net.rs.orig
+++ gstreamer-1.24.3/libs/gst/helpers/ptp/net.rs
@@ -498,8 +498,6 @@
 
         #[cfg(any(target_os = "solaris", target_os = "illumos"))]
         {
-            use crate::error::Context;
-
             socket
                 .join_multicast_v4(addr, &iface.ip_addr)
                 .with_context(|| {
