https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/3552

--- gstreamer-1.24.3/libs/gst/helpers/ptp/ffi.rs.orig
+++ gstreamer-1.24.3/libs/gst/helpers/ptp/ffi.rs
@@ -284,9 +284,7 @@
         )]
         pub fn poll(fds: *mut pollfd, nfds: nfds_t, timeout: c_int) -> c_int;
 
-        #[cfg(not(any(target_os = "solaris", target_os = "illumos")))]
         pub fn if_nametoindex(name: *const c_char) -> c_uint;
-        #[cfg(not(any(target_os = "solaris", target_os = "illumos")))]
         pub fn setsockopt(
             socket: c_int,
             level: c_int,
