Some tests expects bash instead of POSIX shell.

--- fakeroot-1.31/test/echo_arg.orig
+++ fakeroot-1.31/test/echo_arg
@@ -1,2 +1,2 @@
-#!/bin/sh
+#!/bin/bash
 eval echo \$$1
--- fakeroot-1.31/test/cp-atest.orig
+++ fakeroot-1.31/test/cp-atest
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 set -e
 echo cp-atest:
 TMP="$tmp"
--- fakeroot-1.31/test/tartest.orig
+++ fakeroot-1.31/test/tartest
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 set -e
 echo tartest:
 TMP="$tmp"
--- fakeroot-1.31/test/compare-tar.orig
+++ fakeroot-1.31/test/compare-tar
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 ROOTGROUP=root
 
 SYSNAME="`uname -s 2>/dev/null`"
