--- lablgtk-2.10.1/src/gobject.ml.orig	Wed May 18 10:06:50 2011
+++ lablgtk-2.10.1/src/gobject.ml	Wed May 18 10:07:09 2011
@@ -118,8 +118,8 @@
   external type_transformable : g_type -> g_type -> bool
       = "ml_g_value_type_transformable"
   external transform : g_value -> g_value -> bool = "ml_g_value_transform"
-  external get : g_value -> data_get = "ml_g_value_get_variant"
-  external set : g_value -> 'a data_set -> unit = "ml_g_value_set_variant"
+  external get : g_value -> data_get = "ml_mylocal_g_value_get_variant"
+  external set : g_value -> 'a data_set -> unit = "ml_mylocal_g_value_set_variant"
   external get_pointer : g_value -> Gpointer.boxed = "ml_g_value_get_pointer"
   external get_nativeint : g_value -> nativeint = "ml_g_value_get_nativeint"
   external get_int32 : g_value -> int32 = "ml_g_value_get_int32"
