--- Python-3.4.7/configure.ac.~4~	2018-01-27 12:05:06.587996652 +0000
+++ Python-3.4.7/configure.ac	2018-01-27 12:05:47.597154011 +0000
@@ -951,7 +951,7 @@
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
