--- Python-3.4.3/Lib/ctypes/util.py.1	2016-03-21 18:07:17.114549607 +0300
+++ Python-3.4.3/Lib/ctypes/util.py	2016-03-21 18:07:32.679690657 +0300
@@ -117,7 +117,7 @@
         def _get_soname(f):
             if not f:
                 return None
-            cmd = "/usr/ccs/bin/dump -Lpv 2>/dev/null " + f
+            cmd = "/usr/bin/dump -Lpv 2>/dev/null " + f
             with contextlib.closing(os.popen(cmd)) as f:
                 data = f.read()
             res = re.search(r'\[.*\]\sSONAME\s+([^\s]+)', data)
