Without this patch 64-bit install removes 32-bit files

--- Python-3.4.0/Makefile.pre.in.1	2014-06-03 13:53:00.112643443 +0400
+++ Python-3.4.0/Makefile.pre.in	2014-06-03 13:53:39.319167697 +0400
@@ -1422,8 +1422,8 @@
 		--install-scripts=$(BINDIR) \
 		--install-platlib=$(DESTSHARED) \
 		--root=$(DESTDIR)/
-	-rm $(DESTDIR)$(DESTSHARED)/_sysconfigdata.py
-	-rm -r $(DESTDIR)$(DESTSHARED)/__pycache__
+#	-rm $(DESTDIR)$(DESTSHARED)/_sysconfigdata.py
+#	-rm -r $(DESTDIR)$(DESTSHARED)/__pycache__
 
 # Here are a couple of targets for MacOSX again, to install a full
 # framework-based Python. frameworkinstall installs everything, the
