diff -ur gpgme-1.13.1-1/lang/python/tests/Makefile.am gpgme-1.13.1/lang/python/tests/Makefile.am
--- gpgme-1.13.1-1/lang/python/tests/Makefile.am	2019-12-21 11:18:22.615345351 +0000
+++ gpgme-1.13.1/lang/python/tests/Makefile.am	2019-12-21 11:34:14.624975749 +0000
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with this program; if not, see <https://www.gnu.org/licenses/>.
 
-GPG = gpg
+GPG = gpg2
 GPG_AGENT = gpg-agent
 
 test_srcdir = $(top_srcdir)/tests/gpg
diff -ur gpgme-1.13.1-1/lang/qt/tests/Makefile.am gpgme-1.13.1/lang/qt/tests/Makefile.am
--- gpgme-1.13.1-1/lang/qt/tests/Makefile.am	2019-12-21 11:18:22.636945971 +0000
+++ gpgme-1.13.1/lang/qt/tests/Makefile.am	2019-12-21 11:34:14.593516328 +0000
@@ -19,7 +19,7 @@
 
 ## Process this file with automake to produce Makefile.in
 
-GPG = gpg
+GPG = gpg2
 
 GNUPGHOME=$(abs_builddir)
 TESTS_ENVIRONMENT = GNUPGHOME=$(GNUPGHOME)
diff -ur gpgme-1.13.1-1/tests/gpg/Makefile.am gpgme-1.13.1/tests/gpg/Makefile.am
--- gpgme-1.13.1-1/tests/gpg/Makefile.am	2019-12-21 11:18:22.579660360 +0000
+++ gpgme-1.13.1/tests/gpg/Makefile.am	2019-12-21 11:34:14.700754294 +0000
@@ -19,7 +19,7 @@
 
 ## Process this file with automake to produce Makefile.in
 
-GPG = gpg
+GPG = gpg2
 GPG_AGENT = gpg-agent
 
 GNUPGHOME=$(abs_builddir)
diff -ur gpgme-1.13.1-1/tests/json/Makefile.am gpgme-1.13.1/tests/json/Makefile.am
--- gpgme-1.13.1-1/tests/json/Makefile.am	2019-12-21 11:18:22.597363086 +0000
+++ gpgme-1.13.1/tests/json/Makefile.am	2019-12-21 11:34:14.690764211 +0000
@@ -20,7 +20,7 @@
 ## Process this file with automake to produce Makefile.in
 
 GPGME_JSON=$(abs_builddir)/../../src/gpgme-json
-GPG = gpg
+GPG = gpg2
 
 GNUPGHOME=$(abs_builddir)
 TESTS_ENVIRONMENT = GNUPGHOME=$(GNUPGHOME) LC_ALL=C GPG_AGENT_INFO= \
