--- lightdm-gtk-greeter-2.0.1/src/lightdm-gtk-greeter.c.~3~	2017-05-11 13:43:22.329648823 +0300
+++ lightdm-gtk-greeter-2.0.1/src/lightdm-gtk-greeter.c	2017-05-11 13:46:34.646072457 +0300
@@ -94,6 +94,9 @@
 static GtkLabel     *power_title, *power_text;
 static GtkImage     *power_icon;
 
+static const gchar *DEFAULT_LAYOUT[] = {"~spacer", "~spacer", "~host", "~spacer",
+                                     "~session", "~a11y", "~clock", NULL};
+
 static const gchar *POWER_WINDOW_DATA_LOOP = "power-window-loop";           /* <GMainLoop*> */
 static const gchar *POWER_WINDOW_DATA_RESPONSE = "power-window-response";   /* <GtkResponseType> */
 
@@ -460,6 +463,32 @@
 {
     gchar *new_message = NULL;
 
+    gchar **names;
+    gsize length;
+    gboolean power_present;
+    int i;
+
+    /* Do anything only when power indicator is present */
+    power_present = FALSE;
+    names = config_get_string_list (NULL, CONFIG_KEY_INDICATORS, NULL);
+    if (!names)
+       names = (gchar**)DEFAULT_LAYOUT;
+    length = g_strv_length (names);
+    for (i = 0; i < length; ++i)
+    {
+        if (g_strcmp0 (names[i], "~power") == 0)
+        {   /* Power menu is present */
+            power_present = TRUE;
+            break;
+        }
+    }
+    if (names && names != (gchar**)DEFAULT_LAYOUT)
+        g_strfreev (names);
+    
+    if (power_present != TRUE) {
+        return FALSE;
+    }
+
     /* Check if there are still users logged in, count them and if so, display a warning */
     gint logged_in_users = 0;
     GList *items = lightdm_user_list_get_users (lightdm_user_list_get_instance ());
@@ -1437,9 +1466,6 @@
     gboolean inited = FALSE;
     #endif
 
-    const gchar *DEFAULT_LAYOUT[] = {"~spacer", "~spacer", "~host", "~spacer",
-                                     "~session", "~a11y", "~clock", NULL};
-
     gchar **names = config_get_string_list (NULL, CONFIG_KEY_INDICATORS, NULL);
     if (!names)
         names = (gchar**)DEFAULT_LAYOUT;
