--- dia-0.97.2/configure.in.~1~	2014-01-27 23:38:24.892273071 +0400
+++ dia-0.97.2/configure.in	2014-01-27 23:38:46.823513987 +0400
@@ -437,7 +437,6 @@
 
   DIA_CHECK_CFLAG(-Wall)
   dnl DIA_CHECK_CFLAG(-Wunused)
-  DIA_CHECK_CFLAG(--std=c89)
   DIA_CHECK_CFLAG(-Wdeclaration-after-statement)
   dnl I'd rather like to enable this but it fails on isnan() with -std=c89
   dnl DIA_CHECK_CFLAG(-Werror-implicit-function-declaration)
