Use local docbook.xsl
--- dia-0.97.3/configure.in.~1~	2014-09-05 19:05:27.000000000 +0400
+++ dia-0.97.3/configure.in	2015-07-17 15:57:26.661210289 +0300
@@ -481,7 +481,7 @@
   xsltproc=false
 fi
 AM_CONDITIONAL(HAVE_XSLTPROC, $xsltproc)
-AC_SUBST(DB2MAN, "http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl")
+AC_SUBST(DB2MAN, "file:///usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl")
 
 dnl Handling for Python
 AC_ARG_WITH(python,
