--- aoo-4.1.0/main/solenv/bin/addsym.awk.~1~	2014-07-22 16:43:38.734625512 +0400
+++ aoo-4.1.0/main/solenv/bin/addsym.awk	2014-07-22 16:45:16.136091408 +0400
@@ -30,14 +30,14 @@
         print "# Weak RTTI symbols for C++ exceptions:"
         print "UDK_3_0_0 {"
         print "global:"
-        print "_ZTI*; _ZTS*; # weak RTTI symbols for C++ exceptions"
+#        print "_ZTI*; _ZTS*; # weak RTTI symbols for C++ exceptions"
         if (ENVIRON["USE_SYSTEM_STL"] != "YES")
             print "_ZN4_STL7num_put*; # for STLport"
         print "};"
     }
 }
 state == 2 {
-    print "_ZTI*; _ZTS*; # weak RTTI symbols for C++ exceptions"
+#    print "_ZTI*; _ZTS*; # weak RTTI symbols for C++ exceptions"
     if (ENVIRON["USE_SYSTEM_STL"] != "YES")
         print "_ZN4_STL7num_put*; # for STLport"
     state = 3
