--- aoo-4.1.0/main/libxmlsec/makefile.mk.~1~	2014-07-23 09:26:17.166822061 +0400
+++ aoo-4.1.0/main/libxmlsec/makefile.mk	2014-07-23 09:29:45.648901995 +0400
@@ -140,13 +140,16 @@
 .IF "$(OS)$(COM)"=="SOLARISC52"
 xmlsec_LDFLAGS+=-Wl,-R'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib'
 .ENDIF			# "$(OS)$(COM)"=="SOLARISC52"
+.IF "$(OS)$(COM)"=="SOLARISGCC"
+xmlsec_LDFLAGS+=-Wl,-R'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib'
+.ENDIF			# "$(OS)$(COM)"=="SOLARISGCC"
 
 LDFLAGS:=$(xmlsec_LDFLAGS)
 .EXPORT: LDFLAGS
 
 .ENDIF
 CONFIGURE_DIR=
-CONFIGURE_ACTION=.$/configure ADDCFLAGS="$(xmlsec_CFLAGS)" CPPFLAGS="$(xmlsec_CPPFLAGS)"
+CONFIGURE_ACTION=.$/configure ADDCFLAGS="$(xmlsec_CFLAGS)" CPPFLAGS="$(xmlsec_CPPFLAGS)" LDFLAGS="$(xmlsec_LDFLAGS)"
 CONFIGURE_FLAGS=--with-pic --disable-shared --disable-crypto-dl --with-libxslt=no --with-openssl=no --with-gnutls=no LIBXML2LIB="$(LIBXML2LIB)"
 # system-nss needs pkgconfig to get the information about nss
 # FIXME: This also will enable pkg-config usage for libxml2. It *seems*
