--- aoo-4.1.0/main/vcl/unx/generic/app/saldisp.cxx.~1~	2014-07-23 19:52:09.377265805 +0400
+++ aoo-4.1.0/main/vcl/unx/generic/app/saldisp.cxx	2014-07-23 19:59:26.094023931 +0400
@@ -53,10 +53,10 @@
 
 #include <X11/Xatom.h>
 
+#define USE_XINERAMA_XORG
 #ifdef USE_XINERAMA
 #ifdef USE_XINERAMA_XORG
 #include <X11/extensions/Xinerama.h>
-#elif defined USE_XINERAMA_XSUN
 #if defined(SOLARIS) && defined(INTEL) // missing extension header in standard installation
 #define MAXFRAMEBUFFERS       16
 Bool XineramaGetState(Display*, int);
