--- mate-control-center-1.20.3/capplets/about-me/mate-about-me-password.c.~2~	2018-11-08 15:25:28.053932491 +0000
+++ mate-control-center-1.20.3/capplets/about-me/mate-about-me-password.c	2018-11-08 15:31:15.902591210 +0000
@@ -562,6 +564,11 @@
 							  "match",
 							  "1 numeric or special",
 							  "failure",
+							  "must contain",
+							  "must differ",
+							  "are not allowed",
+							  "repeating characters",
+							  "history",
 							  NULL)) {
 
 				/* What response did we get? */
@@ -586,12 +593,17 @@
 						   g_strrstr (str->str, "longer") != NULL) {
 						msg = g_strdup (_("The password is too short."));
 					} else if (g_strrstr (str->str, "palindrome") != NULL ||
+						   g_strrstr (str->str, "must contain") != NULL ||
+						   g_strrstr (str->str, "are not allowed") != NULL ||
+						   g_strrstr (str->str, "repeating characters") != NULL ||
+						   g_strrstr (str->str, "history") != NULL ||
 						   g_strrstr (str->str, "simpl") != NULL ||
 						   g_strrstr (str->str, "dictionary") != NULL) {
 						msg = g_strdup (_("The password is too simple."));
 					} else if (g_strrstr (str->str, "similar") != NULL ||
 					           g_strrstr (str->str, "different") != NULL ||
 					           g_strrstr (str->str, "case") != NULL ||
+					           g_strrstr (str->str, "must differ") != NULL ||
 						   g_strrstr (str->str, "wrapped") != NULL) {
 						msg = g_strdup (_("The old and new passwords are too similar."));
 					} else if (g_strrstr (str->str, "1 numeric or special") != NULL) {
