--- mate-control-center-1.20.3/capplets/about-me/mate-about-me-password.c.~4~	2018-11-13 23:22:52.988738173 +0000
+++ mate-control-center-1.20.3/capplets/about-me/mate-about-me-password.c	2018-11-13 23:30:47.042446380 +0000
@@ -550,8 +550,10 @@
 	}
 
 	if (g_io_channel_read_chars (source, buf, BUFSIZE, &bytes_read, &error) != G_IO_STATUS_NORMAL) {
-		g_warning ("IO Channel read error: %s", error->message);
-		g_error_free (error);
+		if (error) {
+			g_warning ("IO Channel read error: %s", error->message);
+			g_error_free (error);
+		}
 
 		return TRUE;
 	}
