diff -ru mate-control-center-1.24.0/configure.ac mate-control-center-1.24.0.new/configure.ac
--- mate-control-center-1.24.0/configure.ac	Mon Feb 10 12:48:47 2020
+++ mate-control-center-1.24.0.new/configure.ac	Fri Jul 31 21:37:08 2020
@@ -136,7 +136,6 @@
 PKG_CHECK_MODULES(MATECC_SHELL, $COMMON_MODULES libmate-menu)
 
 PKG_CHECK_MODULES(DBUS, dbus-1 dbus-glib-1)
-PKG_CHECK_MODULES(POLKIT, polkit-gobject-1)
 PKG_CHECK_MODULES(MATE_DESKTOP, mate-desktop-2.0 >= $MATE_DESKTOP_REQUIRED)
 PKG_CHECK_MODULES(DEFAULT_APPLICATIONS_CAPPLET, libxml-2.0)
 PKG_CHECK_MODULES(MARCO, libmarco-private >= $MARCO_REQUIRED)
@@ -290,9 +289,6 @@
 capplets/keyboard/Makefile
 capplets/mouse/Makefile
 capplets/network/Makefile
-capplets/time-admin/Makefile
-capplets/time-admin/src/Makefile
-capplets/time-admin/data/Makefile
 capplets/windows/Makefile
 font-viewer/Makefile
 help/Makefile
diff -ru mate-control-center-1.24.0/capplets/Makefile.am mate-control-center-1.24.0.new/capplets/Makefile.am
--- mate-control-center-1.24.0/capplets/Makefile.am	Mon Feb 10 12:48:47 2020
+++ mate-control-center-1.24.0.new/capplets/Makefile.am	Fri Jul 31 21:43:21 2020
@@ -8,7 +8,6 @@
 	keybindings \
 	mouse \
 	network \
-	time-admin \
 	windows \
 	about-me
 
@@ -21,7 +20,6 @@
 	keyboard \
 	mouse \
 	network \
-	time-admin \
 	windows \
 	display \
 	about-me
