--- domoticz-2022.1/main/domoticz.cpp.bak	Tue Jun 14 06:28:02 2022
+++ domoticz-2022.1/main/domoticz.cpp	Tue Jun 14 06:28:15 2022
@@ -1109,7 +1109,7 @@
 
 		// _log.Log(LOG_STATUS, "syslog to %s (%x)", logfacname.c_str(), logfacility);
 		setlogmask(LOG_UPTO(LOG_INFO));
-		openlog(daemonname.c_str(), LOG_CONS | LOG_PERROR, logfacility);
+		openlog(daemonname.c_str(), LOG_CONS | LOG_ERROR, logfacility);
 
 		syslog(LOG_INFO, "Domoticz is starting up....");
 	}
