--- domoticz-2022.1/main/mainworker.cpp.bak	Tue Jun 14 07:21:45 2022
+++ domoticz-2022.1/main/mainworker.cpp	Tue Jun 14 07:27:39 2022
@@ -1099,7 +1099,7 @@
 
 bool MainWorker::Start()
 {
-	utsname my_uname;
+	struct utsname my_uname;
 	if (uname(&my_uname) == 0)
 	{
 		m_szSystemName = my_uname.sysname;
@@ -1249,7 +1249,7 @@
 		}
 	}
 
-	utsname my_uname;
+	struct utsname my_uname;
 	if (uname(&my_uname) < 0)
 		return false;
 
