--- pidgin-otr-4.0.0/configure.ac.~1~	2014-01-30 13:49:30.922049897 +0400
+++ pidgin-otr-4.0.0/configure.ac	2014-01-30 13:49:53.161737964 +0400
@@ -141,8 +141,6 @@
     if test x$have_clang = xyes; then
        OTR_CHECK_CFLAGS(-Qunused-arguments)
     fi
-    OTR_CHECK_CFLAGS(-fstack-protector-all)
-    OTR_CHECK_CFLAGS(-Wstack-protector)
     OTR_CHECK_CFLAGS(-fwrapv)
     OTR_CHECK_CFLAGS(--param ssp-buffer-size=1)
 
