diff --git Python-2.6.4/Include/pyport.h Python-2.6.4/Include/pyport.h
--- Python-2.6.4/Include/pyport.h
+++ Python-2.6.4/Include/pyport.h
@@ -449,11 +449,6 @@
 in platform-specific #ifdefs.
 **************************************************************************/
 
-#ifdef SOLARIS
-/* Unchecked */
-extern int gethostname(char *, int);
-#endif
-
 #ifdef __BEOS__
 /* Unchecked */
 /* It's in the libs, but not the headers... - [cjh] */
