--- readline-7.0/doc/Makefile.in	2018-11-30 07:41:04.950711458 +0000
+++ readline-7.0/doc/Makefile.in	2018-11-30 07:40:36.469953378 +0000
@@ -52,7 +52,11 @@ INSTALL_DATA = @INSTALL_DATA@
 BUILD_DIR = @BUILD_DIR@
 TEXINPUTDIR = $(srcdir)
 
-MAKEINFO    = LANGUAGE= makeinfo
+# modification for solaris nv sfw-workspace nightly build
+# MAKEINFO    = LANGUAGE= makeinfo
+MAKEINFO    = makeinfo
+# end of modification
+
 TEXI2DVI    = $(srcdir)/texi2dvi
 TEXI2HTML   = $(srcdir)/texi2html
 QUIETPS     = #set this to -q to shut up dvips
