--- libgdata-0.6.5/gdata/gdata.symbols.orig	Tue Sep 10 12:34:03 2013
+++ libgdata-0.6.5/gdata/gdata.symbols	Tue Sep 10 12:35:36 2013
@@ -98,6 +98,7 @@
 gdata_g_time_val_get_type
 gdata_youtube_standard_feed_type_get_type
 gdata_youtube_service_error_get_type
+gdata_youtube_service_error_quark
 gdata_youtube_service_get_type
 gdata_youtube_service_new
 gdata_youtube_service_query_standard_feed
@@ -648,6 +649,7 @@
 gdata_documents_spreadsheet_get_download_uri
 gdata_documents_folder_get_type
 gdata_documents_folder_new
+gdata_documents_service_error_quark
 gdata_documents_service_get_type
 gdata_documents_service_new
 gdata_documents_service_query_documents
