--- system-tools-backends-1.4.2/users-conf.in.orig	2009-04-01 23:08:32.346712410 +0530
+++ system-tools-backends-1.4.2/users-conf.in	2009-04-01 23:08:47.268314722 +0530
@@ -1431,17 +1431,16 @@
   }
   else
   {
-    if ( $$new_data[$groups_propmap{"name"}] eq $$old_data[$group_prop_map{"name"}] )
+    $command = "$cmd_groupmod " ;
+    if ( $$new_data[$groups_prop_map{"name"}] ne $$old_data[$groups_prop_map{"name"}] )
     {
-      $command = "$cmd_groupmod -g \'" . $$new_data[$groups_prop_map{"gid"}] .
-        "\' \'" . $$old_data[$groups_prop_map{"name"}] . "\'";
+      $command = $command . "-n \'" . $$new_data[$groups_prop_map{"name"}] . "\' ";
     }
-    else
+    if ( $$new_data[$groups_prop_map{"gid"}] ne $$old_data[$groups_prop_map{"gid"}] )
     {
-      $command = "$cmd_groupmod -g \'" . $$new_data[$groups_prop_map{"gid"}] .
-        "\' -n \'" . $$new_data[$groups_prop_map{"name"}] . "\' " .
-        "\'" . $$old_data[$groups_prop_map{"name"}] . "\'";
+      $command = $command . "-g \'" . $$new_data[$groups_prop_map{"gid"}] . "\' ";
     }
+    $command = $command . "\'" . $$old_data[$groups_prop_map{"name"}] . "\'";
 
     &gst_file_run ($command);
 	
