--- libnet-1.3/include/libnet/libnet-structures.h.orig	2023-10-02 18:48:48.000000000 +0200
+++ libnet-1.3/include/libnet/libnet-structures.h	2023-12-31 15:56:46.567687377 +0100
@@ -49,9 +49,15 @@
 /* libnet statistics structure */
 struct libnet_stats
 {
+#if (!defined(__WIN32__) || (__CYGWIN__))
+    uint64_t packets_sent;             /* packets sent */
+    uint64_t packet_errors;            /* packets errors */
+    uint64_t bytes_written;            /* bytes written */
+#else
     int64_t packets_sent;               /* packets sent */
     int64_t packet_errors;              /* packets errors */
     int64_t bytes_written;              /* bytes written */
+#endif
 };
 
 
