--- libnet-1.1.5/include/libnet/libnet-types.h.orig	Sat Dec 17 16:12:12 2011
+++ libnet-1.1.5/include/libnet/libnet-types.h	Sat Dec 17 16:12:36 2011
@@ -32,8 +32,14 @@
 #ifndef __LIBNET_TYPES_H
 #define __LIBNET_TYPES_H
 
-/* libnet should be using the standard type names, so mapping standard
- * to non-standard should not be necessary. */
+/* Solaris has messed up POSIX nomenclature for these */
+#ifndef __BIT_TYPES_DEFINED__
+#define __BIT_TYPES_DEFINED__
+typedef uint8_t  u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
 
 #endif  /* __LIBNET_TYPES_H */
 
