diff -wpruN '--exclude=*.orig' a~/src/backend/distributed/commands/multi_copy.c a/src/backend/distributed/commands/multi_copy.c
--- a~/src/backend/distributed/commands/multi_copy.c	1970-01-01 00:00:00
+++ a/src/backend/distributed/commands/multi_copy.c	1970-01-01 00:00:00
@@ -54,6 +54,7 @@
 
 #include <arpa/inet.h> /* for htons */
 #include <netinet/in.h> /* for htons */
+#include <sys/stat.h>
 #include <string.h>
 
 #include "distributed/pg_version_constants.h"
