--- libmemcached-1.0.17/libtest/memcached.cc.orig	2013-06-27 23:10:04.722658483 +0200
+++ libmemcached-1.0.17/libtest/memcached.cc	2013-06-27 23:21:03.129972641 +0200
@@ -60,6 +60,7 @@
 
 namespace libtest {
 
+#if defined(HAVE_MEMCACHED_BINARY) && HAVE_MEMCACHED_BINARY
 class Memcached : public libtest::Server
 {
   std::string _username;
@@ -215,23 +216,28 @@
 
   return true;
 }
+#endif
 
 libtest::Server *build_memcached(const std::string& hostname, const in_port_t try_port)
 {
+#if defined(HAVE_MEMCACHED_BINARY) && HAVE_MEMCACHED_BINARY
   if (HAVE_MEMCACHED_BINARY)
   {
     return new Memcached(hostname, try_port, false);
   }
+#endif
 
   return NULL;
 }
 
 libtest::Server *build_memcached_socket(const std::string& socket_file, const in_port_t try_port)
 {
+#if defined(HAVE_MEMCACHED_BINARY) && HAVE_MEMCACHED_BINARY
   if (HAVE_MEMCACHED_BINARY)
   {
     return new Memcached(socket_file, try_port, true);
   }
+#endif
 
   return NULL;
 }
