*** libmemcached-0.16/docs/Makefile.in.original	Sun Feb 17 23:59:27 2008
--- libmemcached-0.16/docs/Makefile.in	Thu Mar  6 01:29:19 2008
***************
*** 46,52 ****
  DIST_SOURCES =
  man1dir = $(mandir)/man1
  am__installdirs = "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(man3dir)"
! man3dir = $(mandir)/man3
  NROFF = nroff
  MANS = $(man_MANS)
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
--- 46,52 ----
  DIST_SOURCES =
  man1dir = $(mandir)/man1
  am__installdirs = "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(man3dir)"
! man3dir = $(mandir)/man3libmemcached
  NROFF = nroff
  MANS = $(man_MANS)
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
***************
*** 197,204 ****
  	memcached_version.pod\
  	memflush.pod
  
! man_MANS = libmemcached.3\
! 	libmemcached_examples.3\
  	memcat.1\
  	memcp.1\
  	memerror.1\
--- 197,204 ----
  	memcached_version.pod\
  	memflush.pod
  
! man_MANS = libmemcached.3libmemcached\
! 	libmemcached_examples.3libmemcached\
  	memcat.1\
  	memcp.1\
  	memerror.1\
***************
*** 206,258 ****
  	memrm.1\
  	memslap.1\
  	memstat.1\
! 	memcached_add.3\
! 	memcached_add_by_key.3\
! 	memcached_append.3\
! 	memcached_append_by_key.3\
! 	memcached_behavior_get.3\
! 	memcached_behavior_set.3\
! 	memcached_callback_get.3\
! 	memcached_callback_set.3\
! 	memcached_cas.3\
! 	memcached_cas_by_key.3\
! 	memcached_clone.3\
! 	memcached_create.3\
! 	memcached_decrement.3\
! 	memcached_delete.3\
! 	memcached_delete_by_key.3\
! 	memcached_fetch.3\
! 	memcached_fetch_execute.3\
! 	memcached_fetch_result.3\
! 	memcached_free.3\
! 	memcached_get.3\
! 	memcached_get_by_key.3\
! 	memcached_increment.3\
! 	memcached_mget.3\
! 	memcached_mget_by_key.3\
! 	memcached_prepend.3\
! 	memcached_prepend_by_key.3\
! 	memcached_replace.3\
! 	memcached_replace_by_key.3\
! 	memcached_server_add.3\
! 	memcached_server_count.3\
! 	memcached_server_list.3\
! 	memcached_server_list_append.3\
! 	memcached_server_list_count.3\
! 	memcached_server_list_free.3\
! 	memcached_server_push.3\
! 	memcached_servers_parse.3\
! 	memcached_set.3\
! 	memcached_set_by_key.3\
! 	memcached_stat.3\
! 	memcached_stat_get_keys.3\
! 	memcached_stat_get_value.3\
! 	memcached_stat_servername.3\
! 	memcached_strerror.3\
! 	memcached_quit.3\
! 	memcached_verbosity.3\
! 	memcached_lib_version.3\
! 	memcached_version.3
  
  all: all-am
  
--- 206,258 ----
  	memrm.1\
  	memslap.1\
  	memstat.1\
! 	memcached_add.3libmemcached\
! 	memcached_add_by_key.3libmemcached\
! 	memcached_append.3libmemcached\
! 	memcached_append_by_key.3libmemcached\
! 	memcached_behavior_get.3libmemcached\
! 	memcached_behavior_set.3libmemcached\
! 	memcached_callback_get.3libmemcached\
! 	memcached_callback_set.3libmemcached\
! 	memcached_cas.3libmemcached\
! 	memcached_cas_by_key.3libmemcached\
! 	memcached_clone.3libmemcached\
! 	memcached_create.3libmemcached\
! 	memcached_decrement.3libmemcached\
! 	memcached_delete.3libmemcached\
! 	memcached_delete_by_key.3libmemcached\
! 	memcached_fetch.3libmemcached\
! 	memcached_fetch_execute.3libmemcached\
! 	memcached_fetch_result.3libmemcached\
! 	memcached_free.3libmemcached\
! 	memcached_get.3libmemcached\
! 	memcached_get_by_key.3libmemcached\
! 	memcached_increment.3libmemcached\
! 	memcached_mget.3libmemcached\
! 	memcached_mget_by_key.3libmemcached\
! 	memcached_prepend.3libmemcached\
! 	memcached_prepend_by_key.3libmemcached\
! 	memcached_replace.3libmemcached\
! 	memcached_replace_by_key.3libmemcached\
! 	memcached_server_add.3libmemcached\
! 	memcached_server_count.3libmemcached\
! 	memcached_server_list.3libmemcached\
! 	memcached_server_list_append.3libmemcached\
! 	memcached_server_list_count.3libmemcached\
! 	memcached_server_list_free.3libmemcached\
! 	memcached_server_push.3libmemcached\
! 	memcached_servers_parse.3libmemcached\
! 	memcached_set.3libmemcached\
! 	memcached_set_by_key.3libmemcached\
! 	memcached_stat.3libmemcached\
! 	memcached_stat_get_keys.3libmemcached\
! 	memcached_stat_get_value.3libmemcached\
! 	memcached_stat_servername.3libmemcached\
! 	memcached_strerror.3libmemcached\
! 	memcached_quit.3libmemcached\
! 	memcached_verbosity.3libmemcached\
! 	memcached_lib_version.3libmemcached\
! 	memcached_version.3libmemcached
  
  all: all-am
  
***************
*** 514,698 ****
  	uninstall-man3
  
  
! libmemcached.3: libmemcached.pod 
! 	pod2man -c "libmemcached" -r "" -s 3 libmemcached.pod > libmemcached.3
  
! libmemcached_examples.3: libmemcached_examples.pod
! 	pod2man -c "libmemcached" -r "" -s 3 libmemcached_examples.pod > libmemcached_examples.3
  
! memcached_create.3: memcached_create.pod 
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_create.pod > memcached_create.3
  
! memcached_free.3: memcached_create.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_create.pod > memcached_free.3
  
! memcached_clone.3: memcached_create.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_create.pod > memcached_clone.3
  
! memcached_set.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_set.3
  
! memcached_set_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_set_by_key.3
  
! memcached_cas.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_cas.3
  
! memcached_cas_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_cas_by_key.3
  
! memcached_replace.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_replace.3
  
! memcached_replace_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_replace_by_key.3
  
! memcached_add.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_add.3
  
! memcached_add_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_add_by_key.3
  
! memcached_prepend.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_prepend.3
  
! memcached_prepend_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_prepend_by_key.3
  
! memcached_append.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_append.3
  
! memcached_append_by_key.3: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_set.pod > memcached_append_by_key.3
  
! memcached_delete.3: memcached_delete.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_delete.pod > memcached_delete.3
  
! memcached_delete_by_key.3: memcached_delete.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_delete.pod > memcached_delete_by_key.3
  
! memcached_increment.3:
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_auto.pod > memcached_increment.3
  
! memcached_decrement.3:
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_auto.pod > memcached_decrement.3
  
! memcached_flush.3: memcached_flush.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_flush.pod > memcached_flush.3
  
! memcached_get.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_get.3
  
! memcached_get_by_key.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_get_by_key.3
  
! memcached_fetch_result.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_fetch_result.3
  
! memcached_fetch_execute.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_fetch_execute.3
  
! memcached_mget.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_mget.3
  
! memcached_mget_by_key.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_mget_by_key.3
  
! memcached_fetch.3: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_get.pod > memcached_fetch.3
  
! memcached_quit.3: memcached_quit.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_quit.pod > memcached_quit.3
  
! memcached_strerror.3: memcached_strerror.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_strerror.pod > memcached_strerror.3
  
! memcached_server_count.3: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_servers.pod > memcached_server_count.3
  
! memcached_server_list.3: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_servers.pod > memcached_server_list.3
  
! memcached_server_add.3: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_servers.pod > memcached_server_add.3
  
! memcached_server_add_unix_socket.3: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_servers.pod > memcached_server_add_unix_socket.3
  
! memcached_server_push.3: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_servers.pod > memcached_server_push.3
  
! memcached_server_list_free.3: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_server_st.pod > memcached_server_list_free.3
  
! memcached_server_list_count.3: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_server_st.pod > memcached_server_list_count.3
  
! memcached_server_list_append.3: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_server_st.pod > memcached_server_list_append.3
  
! memcached_servers_parse.3: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_server_st.pod > memcached_servers_parse.3
  
! memcached_verbosity.3: memcached_verbosity.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_verbosity.pod > memcached_verbosity.3
  
! memcached_behavior_get.3: memcached_behavior.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_behavior.pod > memcached_behavior_get.3
  
! memcached_behavior_set.3: memcached_behavior.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_behavior.pod > memcached_behavior_set.3
  
! memcached_callback_get.3: memcached_callback.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_callback.pod > memcached_callback_get.3
  
! memcached_callback_set.3: memcached_callback.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_callback.pod > memcached_callback_set.3
  
! memcached_stat.3: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_stats.pod > memcached_stat.3
  
! memcached_stat_servername.3: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_stats.pod > memcached_stat_servername.3
  
! memcached_stat_get_value.3: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_stats.pod > memcached_stat_get_value.3
  
! memcached_stat_get_keys.3: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_stats.pod > memcached_stat_get_keys.3
  
! memcached_result_st.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_st.3
  
! memcached_result_create.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_create.3
  
! memcached_result_free.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_free.3
  
! memcached_result_key_value.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_key_value.3
  
! memcached_result_key_length.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_key_length.3
  
! memcached_result_value.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_value.3
  
! memcached_result_length.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_length.3
  
! memcached_result_flags.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_flags.3
  
  memcached_result_cas.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_result_st.pod > memcached_result_cas.3
  
! memcached_version.3: memcached_version.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_version.pod > memcached_version.3
  
! memcached_lib_version.3: memcached_version.pod
! 	pod2man -c "libmemcached" -r "" -s 3 memcached_version.pod  > memcached_lib_version.3
  
  memcp.1: memcp.pod
  	pod2man -c "libmemcached" -r "" -s 1 memcp.pod > memcp.1
--- 514,698 ----
  	uninstall-man3
  
  
! libmemcached.3libmemcached: libmemcached.pod 
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached libmemcached.pod > libmemcached.3libmemcached
  
! libmemcached_examples.3libmemcached: libmemcached_examples.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached libmemcached_examples.pod > libmemcached_examples.3libmemcached
  
! memcached_create.3libmemcached: memcached_create.pod 
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_create.pod > memcached_create.3libmemcached
  
! memcached_free.3libmemcached: memcached_create.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_create.pod > memcached_free.3libmemcached
  
! memcached_clone.3libmemcached: memcached_create.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_create.pod > memcached_clone.3libmemcached
  
! memcached_set.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_set.3libmemcached
  
! memcached_set_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_set_by_key.3libmemcached
  
! memcached_cas.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_cas.3libmemcached
  
! memcached_cas_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_cas_by_key.3libmemcached
  
! memcached_replace.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_replace.3libmemcached
  
! memcached_replace_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_replace_by_key.3libmemcached
  
! memcached_add.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_add.3libmemcached
  
! memcached_add_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_add_by_key.3libmemcached
  
! memcached_prepend.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_prepend.3libmemcached
  
! memcached_prepend_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_prepend_by_key.3libmemcached
  
! memcached_append.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_append.3libmemcached
  
! memcached_append_by_key.3libmemcached: memcached_set.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_set.pod > memcached_append_by_key.3libmemcached
  
! memcached_delete.3libmemcached: memcached_delete.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_delete.pod > memcached_delete.3libmemcached
  
! memcached_delete_by_key.3libmemcached: memcached_delete.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_delete.pod > memcached_delete_by_key.3libmemcached
  
! memcached_increment.3libmemcached:
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_auto.pod > memcached_increment.3libmemcached
  
! memcached_decrement.3libmemcached:
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_auto.pod > memcached_decrement.3libmemcached
  
! memcached_flush.3libmemcached: memcached_flush.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_flush.pod > memcached_flush.3libmemcached
  
! memcached_get.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_get.3libmemcached
  
! memcached_get_by_key.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_get_by_key.3libmemcached
  
! memcached_fetch_result.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_fetch_result.3libmemcached
  
! memcached_fetch_execute.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_fetch_execute.3libmemcached
  
! memcached_mget.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_mget.3libmemcached
  
! memcached_mget_by_key.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_mget_by_key.3libmemcached
  
! memcached_fetch.3libmemcached: memcached_get.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_get.pod > memcached_fetch.3libmemcached
  
! memcached_quit.3libmemcached: memcached_quit.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_quit.pod > memcached_quit.3libmemcached
  
! memcached_strerror.3libmemcached: memcached_strerror.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_strerror.pod > memcached_strerror.3libmemcached
  
! memcached_server_count.3libmemcached: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_servers.pod > memcached_server_count.3libmemcached
  
! memcached_server_list.3libmemcached: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_servers.pod > memcached_server_list.3libmemcached
  
! memcached_server_add.3libmemcached: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_servers.pod > memcached_server_add.3libmemcached
  
! memcached_server_add_unix_socket.3libmemcached: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_servers.pod > memcached_server_add_unix_socket.3libmemcached
  
! memcached_server_push.3libmemcached: memcached_servers.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_servers.pod > memcached_server_push.3libmemcached
  
! memcached_server_list_free.3libmemcached: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_server_st.pod > memcached_server_list_free.3libmemcached
  
! memcached_server_list_count.3libmemcached: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_server_st.pod > memcached_server_list_count.3libmemcached
  
! memcached_server_list_append.3libmemcached: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_server_st.pod > memcached_server_list_append.3libmemcached
  
! memcached_servers_parse.3libmemcached: memcached_server_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_server_st.pod > memcached_servers_parse.3libmemcached
  
! memcached_verbosity.3libmemcached: memcached_verbosity.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_verbosity.pod > memcached_verbosity.3libmemcached
  
! memcached_behavior_get.3libmemcached: memcached_behavior.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_behavior.pod > memcached_behavior_get.3libmemcached
  
! memcached_behavior_set.3libmemcached: memcached_behavior.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_behavior.pod > memcached_behavior_set.3libmemcached
  
! memcached_callback_get.3libmemcached: memcached_callback.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_callback.pod > memcached_callback_get.3libmemcached
  
! memcached_callback_set.3libmemcached: memcached_callback.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_callback.pod > memcached_callback_set.3libmemcached
  
! memcached_stat.3libmemcached: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_stats.pod > memcached_stat.3libmemcached
  
! memcached_stat_servername.3libmemcached: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_stats.pod > memcached_stat_servername.3libmemcached
  
! memcached_stat_get_value.3libmemcached: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_stats.pod > memcached_stat_get_value.3libmemcached
  
! memcached_stat_get_keys.3libmemcached: memcached_stats.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_stats.pod > memcached_stat_get_keys.3libmemcached
  
! memcached_result_st.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_st.3libmemcached
  
! memcached_result_create.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_create.3libmemcached
  
! memcached_result_free.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_free.3libmemcached
  
! memcached_result_key_value.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_key_value.3libmemcached
  
! memcached_result_key_length.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_key_length.3libmemcached
  
! memcached_result_value.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_value.3libmemcached
  
! memcached_result_length.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_length.3libmemcached
  
! memcached_result_flags.3libmemcached: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_flags.3libmemcached
  
  memcached_result_cas.3: memcached_result_st.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_result_st.pod > memcached_result_cas.3libmemcached
  
! memcached_version.3libmemcached: memcached_version.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_version.pod > memcached_version.3libmemcached
  
! memcached_lib_version.3libmemcached: memcached_version.pod
! 	pod2man -c "libmemcached" -r "" -s 3libmemcached memcached_version.pod  > memcached_lib_version.3libmemcached
  
  memcp.1: memcp.pod
  	pod2man -c "libmemcached" -r "" -s 1 memcp.pod > memcp.1
***************
*** 717,727 ****
  
  clean:
  	rm -f *.1
! 	rm -f *.3
  
  distclean:
  	rm -f *.1
! 	rm -f *.3
  	rm -f Makefile
  # Tell versions [3.59,3.63) of GNU make to not export all variables.
  # Otherwise a system limit (for SysV at least) may be exceeded.
--- 717,727 ----
  
  clean:
  	rm -f *.1
! 	rm -f *.3libmemcached
  
  distclean:
  	rm -f *.1
! 	rm -f *.3libmemcached
  	rm -f Makefile
  # Tell versions [3.59,3.63) of GNU make to not export all variables.
  # Otherwise a system limit (for SysV at least) may be exceeded.
