--- lgi-0.9.1/lgi/Makefile.orig	Wed May 31 11:25:15 2017
+++ lgi-0.9.1/lgi/Makefile	Wed May 31 11:25:40 2017
@@ -17,17 +17,17 @@
 VERSION_FILE = version.lua
 
 ifneq ($(filter cygwin% msys% mingw%, $(HOST_OS)),)
-CORE = corelgilua51.dll
+CORE = corelgilua53.dll
 LIBFLAG = -shared
 LIBS += -llua
 else
 ifeq ($(HOST_OS),darwin)
-CORE = corelgilua51.so
+CORE = corelgilua53.so
 LIBFLAG = -bundle -undefined dynamic_lookup
 CCSHARED = -fno-common
 GOBJECT_INTROSPECTION_LIBDIR = $(shell pkg-config --variable=libdir $(GINAME))
 else
-CORE = corelgilua51.so
+CORE = corelgilua53.so
 LIBFLAG = -shared
 CCSHARED = -fPIC
 endif
