--- lgi-0.9.1/lgi/core.lua.orig	Wed May 31 11:27:27 2017
+++ lgi-0.9.1/lgi/core.lua	Wed May 31 11:27:38 2017
@@ -11,7 +11,7 @@
 -- This module decides what kind of core routines should be loaded.
 -- Currently only one implementation exists, standard-Lua C-side
 -- implementation, LuaJIT-FFI-based one is planned.
-local core = require 'lgi.corelgilua51'
+local core = require 'lgi.corelgilua53'
 
 -- Helper methods for converting between CamelCase and uscore_delim
 -- names.
