--- docbook-xsl-1.78.1/manpages/block.xsl.~1~	2015-06-12 17:55:57.947661831 +0300
+++ docbook-xsl-1.78.1/manpages/block.xsl	2015-06-12 17:57:03.897575268 +0300
@@ -20,7 +20,7 @@
   <xsl:call-template name="roff-if-start">
     <xsl:with-param name="condition">n</xsl:with-param>
   </xsl:call-template>
-  <xsl:text>.sp&#10;</xsl:text>
+  <xsl:text>&#10;</xsl:text>
   <xsl:call-template name="roff-if-end"/>
   <xsl:text>.RS 4&#10;</xsl:text>
   <xsl:if test="not($man.output.better.ps.enabled = 0)">
