--- docbook-xsl-1.69.1/fo/lists.xsl.orig	2006-01-19 02:36:22.000000000 -0800
+++ docbook-xsl-1.69.1/fo/lists.xsl	2006-01-19 02:34:54.000000000 -0800
@@ -125,9 +130,18 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates/>
+       <xsl:apply-templates/>
       </fo:block>
+     </xsl:otherwise>
+    </xsl:choose>
     </fo:list-item-body>
   </xsl:variable>
 
@@ -287,9 +301,18 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates/>
+       <xsl:apply-templates/>
       </fo:block>
+     </xsl:otherwise>
+    </xsl:choose>
     </fo:list-item-body>
   </xsl:variable>
 
@@ -486,9 +509,18 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates select="listitem"/>
+       <xsl:apply-templates/>
       </fo:block>
+     </xsl:otherwise>
+    </xsl:choose>
     </fo:list-item-body>
   </fo:list-item>
 </xsl:template>
@@ -906,9 +938,18 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates/>
+       <xsl:apply-templates/>
       </fo:block>
+     </xsl:otherwise>
+    </xsl:choose>
     </fo:list-item-body>
   </fo:list-item>
 </xsl:template>
@@ -932,9 +973,18 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates/>
+       <xsl:apply-templates/>
       </fo:block>
+     </xsl:otherwise>
+    </xsl:choose>
     </fo:list-item-body>
   </fo:list-item>
 </xsl:template>
@@ -1112,10 +1162,19 @@
       </fo:block>
     </fo:list-item-label>
     <fo:list-item-body start-indent="body-start()">
+    <xsl:choose>
+     <xsl:when test="child::*[1][local-name()='para' or
+      local-name()='simpara' or
+      local-name()='formalpara']">
+      <xsl:apply-templates/>
+     </xsl:when>
+     <xsl:otherwise>
       <fo:block>
-        <xsl:apply-templates/>
+       <xsl:apply-templates/>
       </fo:block>
-    </fo:list-item-body>
+     </xsl:otherwise>
+    </xsl:choose>
+   </fo:list-item-body>
   </fo:list-item>
 </xsl:template>
