--- docbook-xsl-1.69.1/fo/pagesetup.xsl.orig	2005-12-10 14:08:28.000000000 -0600
+++ docbook-xsl-1.69.1/fo/pagesetup.xsl	2006-01-01 19:49:28.000000000 -0600
@@ -1491,32 +1491,29 @@
 
       <fo:table-column column-number="1">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">header</xsl:with-param>
             <xsl:with-param name="position" select="$column1"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
       <fo:table-column column-number="2">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">header</xsl:with-param>
             <xsl:with-param name="position" select="2"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
       <fo:table-column column-number="3">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">header</xsl:with-param>
             <xsl:with-param name="position" select="$column3"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
 
@@ -1812,32 +1809,29 @@
       </xsl:call-template>
       <fo:table-column column-number="1">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">footer</xsl:with-param>
             <xsl:with-param name="position" select="$column1"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
       <fo:table-column column-number="2">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">footer</xsl:with-param>
             <xsl:with-param name="position" select="2"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
       <fo:table-column column-number="3">
         <xsl:attribute name="column-width">
-          <xsl:text>proportional-column-width(</xsl:text>
           <xsl:call-template name="header.footer.width">
             <xsl:with-param name="location">footer</xsl:with-param>
             <xsl:with-param name="position" select="$column3"/>
           </xsl:call-template>
-          <xsl:text>)</xsl:text>
+          <xsl:text>%</xsl:text>
         </xsl:attribute>
       </fo:table-column>
 
