--- docbook-xsl-1.69.1/manpages/block.xsl.sp	2005-08-09 10:11:02.000000000 +0100
+++ docbook-xsl-1.69.1/manpages/block.xsl	2006-01-19 12:07:47.000000000 +0000
@@ -37,7 +37,7 @@
     <xsl:apply-templates/>
   </xsl:variable>
   <xsl:value-of select="normalize-space($content)"/>
-  <xsl:text>.sp&#10;</xsl:text>
+  <xsl:text>&#10;.sp&#10;</xsl:text>
 </xsl:template>
 
 <xsl:template match="address|literallayout|programlisting|screen|synopsis">
