Solaris specific change. No plans to send upstream. 

--- sox-14.3.0/src/formats.c	Wed Aug 19 16:15:15 2009
+++ sox-14.3.0/src/formats.c.new	Wed Aug 19 16:23:06 2009
@@ -415,6 +415,8 @@
   fp->_r += PIPE_AUTO_DETECT_SIZE;
 #elif defined __GLIBC__
   fp->_IO_read_ptr = fp->_IO_read_base;
+#elif defined sun
+  #define NO_REWIND_PIPE
 #elif defined _MSC_VER || defined _WIN32 || defined _WIN64 || \
      defined _ISO_STDIO_ISO_H || defined __sgi
   fp->_ptr = fp->_base;
