--- flex-2.5.35/Makefile.in	Tue Feb 26 21:34:10 2008
+++ flex-2.5.35/Makefile.in	Thu Mar 24 20:50:14 2011
@@ -274,8 +274,7 @@
 m4 = @M4@
 indent = @INDENT@
 lib_LIBRARIES = \
-	libfl.a \
-	libfl_pic.a
+	$(SONAME) 
 
 flex_SOURCES = \
 	ccl.c \
@@ -484,6 +485,8 @@
 	-rm -f libfl_pic.a
 	$(libfl_pic_a_AR) libfl_pic.a $(libfl_pic_a_OBJECTS) $(libfl_pic_a_LIBADD)
 	$(RANLIB) libfl_pic.a
+$(SONAME): $(libfl_pic_a_OBJECTS) $(libfl_pic_a_DEPENDENCIES) 
+	$(CC) $(CFLAGS) -G -h $(SONAME) -o $(SONAME) $(libfl_pic_a_OBJECTS) $(libfl_pic_a_LIBADD)
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
