--- flex-2.5.35/tests/test-pthread/Makefile.in	2008-02-26 13:34:09.000000000 -0800
+++ flex-2.5.35/tests/test-pthread/Makefile.in	2013-03-19 12:37:17.679581930 -0700
@@ -346,7 +346,7 @@
 	$(FLEX) $(LFLAGS) $<
 
 $(testname)$(EXEEXT): $(OBJS)
-	$(CC) -o $@ $(LDFLAGS) $(OBJS) $(LOADLIBES)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJS) $(LOADLIBES)
 
 test: $(testname)$(EXEEXT)
 	./$(testname) $(srcdir)/test-*.input
