Egg version is incorrect
--- pygobject-3.18.2/Makefile.am.1	2018-09-19 20:53:11.092593655 +0000
+++ pygobject-3.18.2/Makefile.am	2018-09-19 20:53:51.723767384 +0000
@@ -92,7 +92,7 @@
 
 # install pth file.
 
-EGG_NAME = $(PACKAGE)-$(PACKAGE_VERSION)-py$(PYTHON_VERSION)-$(PLATFORM)
+EGG_NAME = $(PACKAGE)-$(PACKAGE_VERSION)-py$(PYTHON_VERSION)
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(pyexecdir)
