--- libgksu-2.0.12/configure.ac-orig	2010-11-30 15:29:33.775883010 -0600
+++ libgksu-2.0.12/configure.ac	2010-11-30 15:29:43.766247643 -0600
@@ -89,6 +89,7 @@ AC_CONFIG_FILES([
 	po/Makefile.in
 	libgksu/Makefile
 	libgksu/libgksu2.pc
+	libgksu/libgksu2-uninstalled.pc
 	libgksuui/Makefile
 	gksu-properties/Makefile
 	docs/Makefile
> 
--- /dev/null	2010-11-30 15:17:29.000000000 -0600
+++ libgksu-2.0.12/libgksu/libgksu2-uninstalled.pc.in	2010-11-30 15:29:23.548664762 -0600
@@ -0,0 +1,8 @@
+
+Name: LibGKSu
+Version: @VERSION@
+Description: GKSu library version 2.
+Requires: glib-2.0, libstartup-notification-1.0, gnome-keyring-1, gconf-2.0
+Libs: ${pc_top_builddir}/libgksu/libgksu2.la
+Cflags: -I${pc_top_builddir}/libgksu
+
