--- sphinx-7.2.5/tests/test_build_gettext.py.orig
+++ sphinx-7.2.5/tests/test_build_gettext.py
@@ -79,7 +79,7 @@
 
         assert (app.outdir / 'en_US.po').is_file(), 'msginit failed'
         try:
-            args = ['msgfmt', 'en_US.po',
+            args = ['gmsgfmt', 'en_US.po',
                     '-o', os.path.join('en', 'LC_MESSAGES', 'test_root.mo')]
             subprocess.run(args, capture_output=True, check=True)
         except OSError:
