https://github.com/sphinx-doc/sphinx/commit/71b1a06024268e39f59e3772e2420555cb9bf174

--- sphinx-7.4.7/tests/test_builders/test_build_html_image.py.orig
+++ sphinx-7.4.7/tests/test_builders/test_build_html_image.py
@@ -52,17 +52,32 @@
     assert re.search('\n<img alt="_images/img.png" src="_images/img.png" />', context)
 
     # scaled_image_link
-    # Docutils 0.21 adds a newline before the closing </a> tag
-    closing_space = "\n" if docutils.__version_info__[:2] >= (0, 21) else ""
-    assert re.search('\n<a class="reference internal image-reference" href="_images/img.png">'
-                     '<img alt="_images/img.png" src="_images/img.png" style="[^"]+" />'
-                     f'{closing_space}</a>',
-                     context)
+    if docutils.__version_info__[:2] >= (0, 22):
+        assert re.search(
+            '\n<a class="reference internal image-reference" href="_images/img.png">'
+            '<img alt="_images/img.png" height="[^"]+" src="_images/img.png" width="[^"]+" />'
+            '\n</a>',
+            context,
+        )
+    else:
+        # Docutils 0.21 adds a newline before the closing </a> tag
+        closing_space = "\n" if docutils.__version_info__[:2] >= (0, 21) else ""
+        assert re.search('\n<a class="reference internal image-reference" href="_images/img.png">'
+                         '<img alt="_images/img.png" src="_images/img.png" style="[^"]+" />'
+                         f'{closing_space}</a>',
+                         context)
 
     # no-scaled-link class disables the feature
-    assert re.search('\n<img alt="_images/img.png" class="no-scaled-link"'
-                     ' src="_images/img.png" style="[^"]+" />',
-                     context)
+    if docutils.__version_info__[:2] >= (0, 22):
+        assert re.search(
+            '\n<img alt="_images/img.png" class="no-scaled-link"'
+            ' height="[^"]+" src="_images/img.png" width="[^"]+" />',
+            context,
+        )
+    else:
+        assert re.search('\n<img alt="_images/img.png" class="no-scaled-link"'
+                         ' src="_images/img.png" style="[^"]+" />',
+                         context)
 
 
 @pytest.mark.sphinx('html', testroot='images')
