diff --git a/data/db/open-phrase/Makefile.am b/data/db/open-phrase/Makefile.am
index f5be517..ccb167d 100644
--- a/data/db/open-phrase/Makefile.am
+++ b/data/db/open-phrase/Makefile.am
@@ -31,7 +31,7 @@ DBTAR = pyzy-database-$(DBVER).tar.bz2
 
 $(DBTAR):
 	$(AM_V_GEN) \
-	wget http://pyzy.googlecode.com/files/$(DBTAR) || \
+	wget https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/pyzy/$(DBTAR) || \
 	( $(RM) $@; exit 1)
 
 stamp-db: $(DBTAR)
