--- barman-1.6.1/barman/utils.py.1	2016-09-19 14:34:20.005132022 +0300
+++ barman-1.6.1/barman/utils.py	2016-09-19 14:39:01.640914460 +0300
@@ -253,7 +253,7 @@
     :param str dir_path: The directory to sync
     :raise OSError: If fail opening the directory
     """
-    dir_fd = os.open(dir_path, os.O_DIRECTORY)
+    dir_fd = os.open(dir_path, os.O_RDONLY)
     try:
         os.fsync(dir_fd)
     except OSError as e:
