*** emacs-23.4.orig/src/s/sol2-5.h	Mon Oct 17 19:20:46 2011
--- emacs-23.4/src/s/sol2-5.h	Tue Feb  7 12:38:02 2012
***************
*** 17,22 ****
--- 17,23 ----
  /* This is not always necessary.  Turned off at present for testers to
     identify any problems with gmalloc more accurately.  */
  /* #define SYSTEM_MALLOC */
+ #define SYSTEM_MALLOC
  
  /* There have problems reported with mmap at least on Solaris 2.6
     and 2.7.  For simplicity, let's not use mmap for anything >= 2.5.
