*** emacs-23.1/src/s/sol2-5.h.orig	Wed Oct 14 17:57:22 2009
--- emacs-23.1/src/s/sol2-5.h	Wed Oct 14 17:58:54 2009
***************
*** 17,22 ****
--- 17,23 ----
  /* This is not always necessary.  Turned off at present for testers to
     identify any problems with gmalloc more accurately.  */
  /* #define SYSTEM_MALLOC */
+ #define SYSTEM_MALLOC
  
  /* There have problems reported with mmap at least on Solaris 2.6
     and 2.7.  For simplicity, let's not use mmap for anything >= 2.5.
