*** emacs-23.4.orig/configure	Thu Jan 19 08:01:37 2012
--- emacs-23.4/configure	Tue Feb  7 12:38:02 2012
***************
*** 9283,9288 ****
--- 9283,9289 ----
              { $as_echo "$as_me:${as_lineno-$LINENO}: checking XFT_LIBS" >&5
  $as_echo_n "checking XFT_LIBS... " >&6; }
              XFT_LIBS=`$PKG_CONFIG --libs "xft >= 0.13.0"|sed -e 's,///*,/,g'`
+             XFT_LIBS='-lXft -lXrender -lfontconfig -lfreetype -lX11  '
              { $as_echo "$as_me:${as_lineno-$LINENO}: result: $XFT_LIBS" >&5
  $as_echo "$XFT_LIBS" >&6; }
          else
