--- emacs-23.1/configure.orig	Thu Mar  3 12:13:44 2011
+++ emacs-23.1/configure	Thu Mar  3 12:14:37 2011
@@ -12863,6 +12863,7 @@
             { echo "$as_me:$LINENO: checking XFT_LIBS" >&5
 echo $ECHO_N "checking XFT_LIBS... $ECHO_C" >&6; }
             XFT_LIBS=`$PKG_CONFIG --libs "xft >= 0.13.0"|sed -e 's,///*,/,g'`
+            XFT_LIBS='-R/usr/lib -lXft -lXrender -lfontconfig -lfreetype -lX11  '
             { echo "$as_me:$LINENO: result: $XFT_LIBS" >&5
 echo "${ECHO_T}$XFT_LIBS" >&6; }
         else
