--- redis-6.0-rc3/redis.conf.1	2020-04-12 12:20:07.055640741 +0000
+++ redis-6.0-rc3/redis.conf	2020-04-12 12:24:12.900331140 +0000
@@ -202,7 +202,7 @@
 
 # By default Redis does not run as a daemon. Use 'yes' if you need it.
 # Note that Redis will write a pid file in /var/run/redis.pid when daemonized.
-daemonize no
+daemonize yes
 
 # If you run Redis from upstart or systemd, Redis can interact with your
 # supervision tree. Options:
@@ -224,7 +224,7 @@
 #
 # Creating a pid file is best effort: if Redis is not able to create it
 # nothing bad happens, the server will start and run normally.
-pidfile /var/run/redis_6379.pid
+pidfile /tmp/redis_6379.pid
 
 # Specify the server verbosity level.
 # This can be one of:
@@ -237,7 +237,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Redis to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile "/var/redis/log/redis_6379.log"
 
 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
@@ -342,7 +342,7 @@
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/redis/db
 
 ################################# REPLICATION #################################
 
