diff --git a/tests/modules/atomicslotmigration.c b/tests/modules/atomicslotmigration.c
index 83393cd..4b91676 100644
--- a/tests/modules/atomicslotmigration.c
+++ b/tests/modules/atomicslotmigration.c
@@ -3,6 +3,7 @@
 #include <stdlib.h>
 #include <memory.h>
 #include <errno.h>
+#include <string.h>
 
 #define MAX_EVENTS 1024
 
diff --git a/tests/modules/eventloop.c b/tests/modules/eventloop.c
index c0cfdf0..98cf389 100644
--- a/tests/modules/eventloop.c
+++ b/tests/modules/eventloop.c
@@ -17,6 +17,7 @@
 #include <fcntl.h>
 #include <memory.h>
 #include <errno.h>
+#include <string.h>
 
 int fds[2];
 long long buf_size;
diff --git a/tests/modules/rdbloadsave.c b/tests/modules/rdbloadsave.c
index 687269a..1d85443 100644
--- a/tests/modules/rdbloadsave.c
+++ b/tests/modules/rdbloadsave.c
@@ -5,6 +5,7 @@
 #include <fcntl.h>
 #include <memory.h>
 #include <errno.h>
+#include <string.h>
 
 /* Sanity tests to verify inputs and return values. */
 int sanity(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
