diff -Naur lz4-1.7.3.orig/lib/Makefile lz4-1.7.3/lib/Makefile
--- lz4-1.7.3.orig/lib/Makefile	2016-11-16 20:37:05.000000000 +0100
+++ lz4-1.7.3/lib/Makefile	2016-11-17 01:04:47.587158038 +0100
@@ -53,7 +53,7 @@
 FLAGS   = $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 LIBDIR?= $(PREFIX)/lib
-INCLUDEDIR=$(PREFIX)/include
+INCLUDEDIR?=$(PREFIX)/include
 
 
 # OS X linker doesn't support -soname, and use different extension
@@ -100,11 +100,6 @@
 	@echo Cleaning library completed
 
 
-#------------------------------------------------------------------------
-#make install is validated only for Linux, OSX, kFreeBSD, Hurd and
-#FreeBSD targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU FreeBSD))
-
 liblz4.pc: liblz4.pc.in Makefile
 	@echo creating pkgconfig
 	@sed -e 's|@PREFIX@|$(PREFIX)|' \
@@ -137,5 +132,3 @@
 	@$(RM) $(DESTDIR)$(INCLUDEDIR)/lz4hc.h
 	@$(RM) $(DESTDIR)$(INCLUDEDIR)/lz4frame.h
 	@echo lz4 libraries successfully uninstalled
-
-endif
diff -Naur lz4-1.7.3.orig/Makefile lz4-1.7.3/Makefile
--- lz4-1.7.3.orig/Makefile	2016-11-16 20:37:05.000000000 +0100
+++ lz4-1.7.3/Makefile	2016-11-17 01:05:02.296921321 +0100
@@ -35,7 +35,7 @@
 VOID    := /dev/null
 
 LIBDIR ?= $(PREFIX)/lib
-INCLUDEDIR=$(PREFIX)/include
+INCLUDEDIR?=$(PREFIX)/include
 PRGDIR  = programs
 LZ4DIR  = lib
 TESTDIR = tests
@@ -71,10 +71,6 @@
 	@echo Cleaning completed
 
 
-#------------------------------------------------------------------------
-#make install is validated only for Linux, OSX, kFreeBSD, Hurd and
-#FreeBSD targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU FreeBSD))
 HOST_OS = POSIX
 
 install:
@@ -125,8 +121,6 @@
 	$(MAKE) -C $(PRGDIR) lz4
 	$(MAKE) -C examples test
 
-endif
-
 
 ifneq (,$(filter MSYS%,$(shell uname)))
 HOST_OS = MSYS
diff -Naur lz4-1.7.3.orig/programs/Makefile lz4-1.7.3/programs/Makefile
--- lz4-1.7.3.orig/programs/Makefile	2016-11-16 20:37:05.000000000 +0100
+++ lz4-1.7.3/programs/Makefile	2016-11-17 01:04:29.215212394 +0100
@@ -30,8 +30,8 @@
 DESTDIR ?=
 PREFIX  ?= /usr/local
 VOID    := /dev/null
-BINDIR  := $(PREFIX)/bin
-MANDIR  := $(PREFIX)/share/man/man1
+BINDIR  ?= $(PREFIX)/bin
+MANDIR  ?= $(PREFIX)/share/man/man1
 LZ4DIR  := ../lib
 
 CFLAGS  ?= -O3   # allows custom optimization flags. For example : CFLAGS="-O2 -g" make
@@ -76,11 +76,6 @@
 	@echo Cleaning completed
 
 
-#------------------------------------------------------------------------
-#make install is validated only for Linux, OSX, kFreeBSD, Hurd and
-#FreeBSD targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU FreeBSD))
-
 install: lz4$(EXT) lz4c$(EXT)
 	@echo Installing binaries
 	@install -d -m 755 $(DESTDIR)$(BINDIR)/ $(DESTDIR)$(MANDIR)/
@@ -105,5 +100,3 @@
 	@$(RM) $(DESTDIR)$(MANDIR)/lz4cat.1
 	@$(RM) $(DESTDIR)$(MANDIR)/unlz4.1
 	@echo lz4 programs successfully uninstalled
-
-endif
diff -Naur lz4-1.7.3.orig/tests/Makefile lz4-1.7.3/tests/Makefile
--- lz4-1.7.3.orig/tests/Makefile	2016-11-16 20:37:05.000000000 +0100
+++ lz4-1.7.3/tests/Makefile	2016-11-17 01:04:06.901218060 +0100
@@ -34,8 +34,8 @@
 DESTDIR ?=
 PREFIX  ?= /usr/local
 VOID    := /dev/null
-BINDIR  := $(PREFIX)/bin
-MANDIR  := $(PREFIX)/share/man/man1
+BINDIR  ?= $(PREFIX)/bin
+MANDIR  ?= $(PREFIX)/share/man/man1
 LZ4DIR  := ../lib
 PRGDIR  := ../programs
 TESTDIR := versionsTest
