--- avant-window-navigator-0.3.2.1/libawn/awn-config-client-gconf.c	Sat May  2 08:05:21 2009
+++ avant-window-navigator-0.3.2.1/libawn/awn-config-client-gconf.c	Sat Feb 22 16:06:43 2014
@@ -31,7 +31,6 @@
 #include <gconf/gconf-client.h>

 #if GLIB_CHECK_VERSION(2,15,0)
-#include <glib/gchecksum.h>
 #else
 #include "egg/eggchecksum.h"
 #endif
--- avant-window-navigator-0.3.2.1/libawn/awn-config-client-shared.c	Sat May  2 08:05:22 2009
+++ avant-window-navigator-0.3.2.1/libawn/awn-config-client-shared.c	Sat Feb 22 16:06:53 2014
@@ -27,7 +27,7 @@
 #include <string.h>

 #if GLIB_CHECK_VERSION(2,15,0)
-#include <glib/gchecksum.h>
+#include <glib.h>
 #else
 #include "egg/eggchecksum.h"
 #endif
