--- python-ldap-2.4.25/setup.cfg.1	2016-06-06 17:42:24.129177720 +0300
+++ python-ldap-2.4.25/setup.cfg	2016-06-06 17:42:32.478823709 +0300
@@ -1,5 +1,5 @@
 [_ldap]
-library_dirs = /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64
+library_dirs = 
 include_dirs = /usr/include /usr/include/sasl /usr/local/include /usr/local/include/sasl
 defines = HAVE_SASL HAVE_TLS HAVE_LIBLDAP_R
 extra_compile_args = 
