--- ocamlfind-findlib-1.9.1/Makefile.orig	2021-03-16 12:39:21.000000000 +0000
+++ ocamlfind-findlib-1.9.1/Makefile	2021-11-26 20:03:01.749503843 +0000
@@ -25,7 +25,7 @@
 	$(MAKE) install-meta
 	test ! -f 'site-lib-src/num-top/META' || { cd src/findlib; $(MAKE) install-num-top; }
 	test ! -f 'site-lib-src/camlp4/META' ||	cp tools/safe_camlp4 "$(prefix)$(OCAMLFIND_BIN)"
-	$(MAKE) install-doc
+	cp src/findlib/ocamlfind_opt "$(prefix)$(OCAMLFIND_BIN)";
 
 uninstall: check-installation
 	$(MAKE) uninstall-doc
