Recast for getpid() because it requires an int. 

diff --git a/xpcom/base/nsStatusReporterManager.cpp b/xpcom/base/nsStatusReporterManager.cpp
--- a/xpcom/base/nsStatusReporterManager.cpp
+++ b/xpcom/base/nsStatusReporterManager.cpp
@@ -148,7 +148,7 @@
   nsresult rv;
 
   nsCString filename("status-reports-");
-  filename.AppendInt(getpid());
+  filename.AppendInt((int)getpid());
   filename.Append('-');
   filename.AppendInt(number++);
   filename.AppendLiteral(".json");
