--- firefox-51.0b14/media/libpng/pngpriv.h.orig	2017-01-31 04:09:56.868180940 +0000
+++ firefox-51.0b14/media/libpng/pngpriv.h	2017-01-31 04:23:27.339895342 +0000
@@ -23,20 +23,6 @@
 #ifndef PNGPRIV_H
 #define PNGPRIV_H
 
-/* Feature Test Macros.  The following are defined here to ensure that correctly
- * implemented libraries reveal the APIs libpng needs to build and hide those
- * that are not needed and potentially damaging to the compilation.
- *
- * Feature Test Macros must be defined before any system header is included (see
- * POSIX 1003.1 2.8.2 "POSIX Symbols."
- *
- * These macros only have an effect if the operating system supports either
- * POSIX 1003.1 or C99, or both.  On other operating systems (particularly
- * Windows/Visual Studio) there is no effect; the OS specific tests below are
- * still required (as of 2011-05-02.)
- */
-#define _POSIX_SOURCE 1 /* Just the POSIX 1003.1 and C89 APIs */
-
 #ifndef PNG_VERSION_INFO_ONLY
 /* Standard library headers not required by png.h: */
 #  include <stdlib.h>
