http://tug.org/pipermail/tlbuild/2014q2/002976.html

--- firefox-51.0b14/intl/icu/source/config/mh-solaris-gcc.bck	2017-01-17 07:13:39.150928690 +0000
+++ firefox-51.0b14/intl/icu/source/config/mh-solaris-gcc	2017-01-17 07:14:26.256169948 +0000
@@ -9,7 +9,7 @@
 SHAREDLIBCPPFLAGS = -DPIC
 
 ## Commands to generate dependency files
-GEN_DEPS.c= 	$(CC) -E -MM $(DEFS) $(CPPFLAGS) $(CFLAGS)
+GEN_DEPS.c= 	$(CC) $(CFLAGS) -E -MM $(DEFS) $(CPPFLAGS) $(CFLAGS)
 GEN_DEPS.cc=	$(CXX) -E -MM $(DEFS) $(CPPFLAGS) $(CXXFLAGS)
 
 ## These are the library specific LDFLAGS
