diff -ur libcroco-0.6.12-1/tests/test-prop-ident.sh libcroco-0.6.12/tests/test-prop-ident.sh
--- libcroco-0.6.12-1/tests/test-prop-ident.sh	2017-05-03 16:49:10.409322809 +0300
+++ libcroco-0.6.12/tests/test-prop-ident.sh	2017-05-03 16:49:42.644635496 +0300
@@ -1,5 +1,5 @@
 #! /bin/sh
 
-. global-test-vars.sh
+. ./global-test-vars.sh
 
 $CSSLINT "$TEST_INPUTS_DIR"/prop-vendor-ident.css
diff -ur libcroco-0.6.12-1/tests/test-several-media.sh libcroco-0.6.12/tests/test-several-media.sh
--- libcroco-0.6.12-1/tests/test-several-media.sh	2017-05-03 16:49:10.405697105 +0300
+++ libcroco-0.6.12/tests/test-several-media.sh	2017-05-03 16:50:13.034832777 +0300
@@ -1,5 +1,5 @@
 #! /bin/sh
 
-. global-test-vars.sh
+. ./global-test-vars.sh
 
 $CSSLINT "$TEST_INPUTS_DIR"/several-media.css
