diff -urp -x '*~' -x '*.orig' man/scanpci.man man/scanpci.man
--- a/man/scanpci.man	2010-10-07 11:14:44.000000000 -0700
+++ b/man/scanpci.man	2011-02-02 20:56:47.527978008 -0800
@@ -22,7 +22,7 @@
 .\" dealings in this Software without prior written authorization from the
 .\" XFree86 Project.
 .\"
-.TH SCANPCI 1 __xorgversion__
+.TH SCANPCI 1
 .SH NAME
 scanpci - scan/probe PCI buses
 .SH SYNOPSIS
@@ -35,10 +35,16 @@ about the configuration space settings f
 On most platforms,
 .I scanpci
 can only be run by the root user.
+On OpenIndiana it may also be run using the 
+.I "Desktop Configuration"
+rights profile.  See \fBrbac\fR(5) and \fBuser_attr\fR(4).
 .SH OPTIONS
 .TP 8
 .B \-v
 Print the configuration space information for each device in a verbose
 format.  Without this option, only a brief description is printed for
 each device.
+.SH "SEE ALSO"
+.BR prtconf "(1M), " prtdiag "(1M), " smbios "(1M), "
+.BR user_attr "(4), " rbac "(5)."
 
