--- libpciaccess-0.13.3/scanpci/Makefile.am.~1~	2015-02-10 11:49:54.403800037 +0300
+++ libpciaccess-0.13.3/scanpci/Makefile.am	2015-02-10 11:50:58.881744223 +0300
@@ -21,7 +21,7 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-noinst_PROGRAMS = scanpci
+bin_PROGRAMS = scanpci
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src
 LDADD =  $(top_builddir)/src/libpciaccess.la
--- libpciaccess-0.13.3/man/Makefile.am.~1~	2015-02-10 12:24:16.798806091 +0300
+++ libpciaccess-0.13.3/man/Makefile.am	2015-02-10 12:24:23.319341941 +0300
@@ -22,7 +22,7 @@
 #
 
 appman_PRE = scanpci.man
-noinst_DATA = $(appman_PRE:man=$(APP_MAN_SUFFIX))
+man1_MANS = scanpci.man
 
 EXTRA_DIST = $(appman_PRE)
 CLEANFILES = $(noinst_DATA)
