--- vlc-3.0.1/configure.ac.orig	2018-03-08 19:48:12.925261388 +0000
+++ vlc-3.0.1/configure.ac	2018-03-08 19:48:57.227221131 +0000
@@ -1331,7 +1331,7 @@
     ])
   ])
 
-  AS_IF([test "${ac_cv_sse_inline}" != "no" -a "${SYS}" != "solaris"], [
+  AS_IF([test "${ac_cv_sse_inline}" != "no"], [
     AC_DEFINE(CAN_COMPILE_SSE, 1, [Define to 1 if SSE inline assembly is available.])
   ])
 
@@ -1346,7 +1346,7 @@
       ac_cv_sse2_inline=no
     ])
   ])
-  AS_IF([test "${ac_cv_sse2_inline}" != "no" -a "${SYS}" != "solaris"], [
+  AS_IF([test "${ac_cv_sse2_inline}" != "no"], [
     AC_DEFINE(CAN_COMPILE_SSE2, 1, [Define to 1 if SSE2 inline assembly is available.])
     have_sse2="yes"
   ])
