--- a/iaxmodem.c
+++ b/iaxmodem.c
@@ -516,6 +516,9 @@
 		    dialextra = strndup(nnump, 64);
 		    nnump[0] = '\0';
 		}
+		// reset jitter counter
+		s->at_state.p.s_regs[11] = 0;
+		s->at_state.p.s_regs[12] = 0;
 		printlog(LOG_INFO, "Dialing '%s'\n", nnum);
 		char ich[256];
 		snprintf(ich, sizeof(ich), "%s:%s@%s/%s", regpeer, regsecret, server, nnum);
@@ -549,6 +552,9 @@
 	    tioflags &= ~TIOCM_RI;
 	    ioctl(aslave, TIOCMSET, &tioflags);
 
+	    // reset jitter counter
+	    s->at_state.p.s_regs[11] = 0;
+	    s->at_state.p.s_regs[12] = 0;
 	    iax_answer(session[0]);
 	    if (!info) t31_call_event(s, AT_CALL_EVENT_ANSWERED);
 	    modemstate = MODEM_CONNECTED;
@@ -1563,6 +1569,13 @@
 						int16_t zerodata[1] = { 0 };
 						while (nulldata--) t31_rx(&t31_state, zerodata, 1);
 					    } else {
+						t31_state.at_state.p.s_regs[11] += (t31_state.at_state.p.s_regs[12] + nulldata) / 256;
+						t31_state.at_state.p.s_regs[12] = (t31_state.at_state.p.s_regs[12] + nulldata) % 256;
+						if (t31_state.at_state.p.s_regs[11] > 255) {
+						    /* This is as high as our counters go. */
+						    t31_state.at_state.p.s_regs[11] = 255;
+						    t31_state.at_state.p.s_regs[12] = 255;
+						}
 						printlog(LOG_INFO, "Skipped %d samples of null data (presumed upstream jitter).\n", nulldata);
 					    }
 					}
