--- a/lib/spandsp/src/at_interpreter.c
+++ b/lib/spandsp/src/at_interpreter.c
@@ -1079,6 +1079,22 @@
 }
 /*- End of function --------------------------------------------------------*/
 
+static const char *at_cmd_S11(at_state_t *s, const char *t)
+{
+    /* User assigned value */ 
+    t += 3;
+    return s_reg_handler(s, t, 11);
+}
+/*- End of function --------------------------------------------------------*/
+
+static const char *at_cmd_S12(at_state_t *s, const char *t)
+{
+    /* User assigned value */ 
+    t += 3;
+    return s_reg_handler(s, t, 12);
+}
+/*- End of function --------------------------------------------------------*/
+
 static const char *at_cmd_S3(at_state_t *s, const char *t)
 {
     /* V.250 6.2.1 - Command line termination character */ 
