--- a/lib/spandsp/src/make_at_dictionary.c
+++ b/lib/spandsp/src/make_at_dictionary.c
@@ -429,6 +429,8 @@
     "Q",        /* V.250 6.2.5 - Result code suppression */ 
     "S0",       /* V.250 6.3.8 - Automatic answer */ 
     "S10",      /* V.250 6.3.12 - Automatic disconnect delay */ 
+    "S11",      /* User assigned value */ 
+    "S12",      /* User assigned value */ 
     "S3",       /* V.250 6.2.1 - Command line termination character */ 
     "S4",       /* V.250 6.2.2 - Response formatting character */ 
     "S5",       /* V.250 6.2.3 - Command line editing character */ 
