--- db-5.1.25/dist/configure.orig	2011-08-04 15:02:16.786577345 -0700
+++ db-5.1.25/dist/configure	2011-08-04 15:02:45.349081448 -0700
@@ -25858,7 +25858,7 @@
   cd jdbc
   test ! -e Makefile.in.tmp && mv Makefile.in Makefile.in.tmp
   sed "s/@BDB_LIB@/$BDB_LIB/g" Makefile.in.tmp > Makefile.in
-  eval "\$SHELL ./configure --with-sqlite3=../../lang/sql/generated $jdbc_args $jdbc_flags"
+  eval "\$SHELL ./configure --with-sqlite3=/usr $jdbc_args $jdbc_flags"
 fi
 
 fi
