--- ./src/sun_kbd.c.orig	2016-05-21 22:33:46.628443784 +0200
+++ ./src/sun_kbd.c	2016-05-21 22:51:44.333174387 +0200
@@ -51,11 +51,15 @@
 
 #include <xorg-server.h>
 #include "xf86.h"
+#include <sys/param.h>
 #include "xf86Priv.h"
 #include "xf86_OSlib.h"
 #include "xf86OSKbd.h"
 #include "sun_kbd.h"
 
+#include <unistd.h>
+#include <stropts.h>
+
 #include <sys/stropts.h>
 #include <sys/vuid_event.h>
 #include <sys/kbd.h>
