diff -Nru NetHack-3.6.7.orig/include/unixconf.h NetHack-3.6.7/include/unixconf.h
--- NetHack-3.6.7.orig/include/unixconf.h	2023-02-15 22:52:57.000000000 +0100
+++ NetHack-3.6.7/include/unixconf.h	2023-10-08 15:00:52.851613365 +0200
@@ -100,13 +100,13 @@
  */
 
 /* #define NO_FILE_LINKS */                       /* if no hard links */
-/* #define LOCKDIR "/usr/games/lib/nethackdir" */ /* where to put locks */
+#define LOCKDIR "/usr/lib/nethack" /* where to put locks */
 
 /*
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-/* #define VAR_PLAYGROUND "/var/lib/games/nethack" */
+#define VAR_PLAYGROUND "/var/games/nethack"
 
 /*
  * Define DEF_PAGER as your default pager, e.g. "/bin/cat" or "/usr/ucb/more"
@@ -133,7 +133,7 @@
  * "extra output" method is used, but not all systems provide access to
  * a fine-grained timer.
  */
-/* #define TIMED_DELAY */ /* usleep() */
+#define TIMED_DELAY /* usleep() */
 #endif
 #if defined(MACOSX) && !defined(TIMED_DELAY)
 #define TIMED_DELAY
