--- slang-2.3.1/autoconf/aclocal.m4.1	2018-01-31 14:45:00.000382855 +0000
+++ slang-2.3.1/autoconf/aclocal.m4	2018-01-31 14:48:33.810700906 +0000
@@ -628,7 +627,7 @@
       ELF_CFLAGS="\$(CFLAGS) -fPIC"
       ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-ztext -Wl,-h,\$(ELFLIB_MAJOR)"
       ELF_DEP_LIBS="\$(DL_LIB) -lm -lc"
-      CC_SHARED_FLAGS="-G -fPIC"
+      CC_SHARED_FLAGS="-shared -fPIC"
       CC_SHARED="\$(CC) $CC_SHARED_FLAGS \$(CFLAGS)"
     else
       DYNAMIC_LINK_FLAGS=""
