Without this several quic tests fails.

--- openssl-3.4.1/Configurations/10-main.conf.orig
+++ openssl-3.4.1/Configurations/10-main.conf
@@ -230,6 +230,7 @@
     "solaris-common-gcc" => {
         inherit_from     => [ "solaris-common" ],
         template         => 1,
+        cppflags         => add(threads("-D_REENTRANT")),
         shared_target    => "solaris-gcc-shared", # The rest is on shared_info.pl
     },
 #### Solaris x86 with GNU C setups
