--- gnome-applets-2.30.0/null_applet/null_applet.c-orig	2012-03-16 05:36:48.821253969 -0500
+++ gnome-applets-2.30.0/null_applet/null_applet.c	2012-03-16 05:36:51.637248549 -0500
@@ -113,8 +113,10 @@ static char
 					g_error_free (error);
 					error = NULL;
 				}
-				g_string_append_printf (string,
-						"    • %s\n", name);
+                                if (strcmp (name, "Volume Control") != 0) {
+					g_string_append_printf (string,
+							"    • %s\n", name);
+				}
 			}
 			g_free (oafiid);
 		}
@@ -143,6 +145,10 @@ applet_factory (PanelApplet *applet,
 
 	applet_list = get_all_applets ();
 
+	if (applet_list[0] == '\0') {
+		return TRUE;
+	}
+
 	dialog = gtk_message_dialog_new_with_markup (NULL,
 			GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
 			GTK_MESSAGE_INFO, GTK_BUTTONS_OK,
