--- gnome-applets-2.32.1.1/gweather/gweather-applet.c.1	2016-04-13 22:10:42.339106563 +0300
+++ gnome-applets-2.32.1.1/gweather/gweather-applet.c	2016-04-13 22:11:28.871372960 +0300
@@ -471,8 +471,7 @@
 			 if (icon == NULL)
 				 icon = "stock-unknown";
 	           	 
-			 n = notify_notification_new (message, detail, icon,
-					 gw_applet->container);
+			 n = notify_notification_new (message, detail, icon);
 	
 		   	 notify_notification_show (n, &error);
 			 if (error)
--- gnome-applets-2.32.1.1/battstat/battstat_applet.c.1	2016-04-13 22:14:36.868507521 +0300
+++ gnome-applets-2.32.1.1/battstat/battstat_applet.c	2016-04-13 22:15:25.350527567 +0300
@@ -435,7 +435,7 @@
 			GTK_ICON_LOOKUP_USE_BUILTIN,
 			NULL);
 	
-	NotifyNotification *n = notify_notification_new (_("Your battery is now fully recharged"), "", /* "battery" */ NULL, applet);
+	NotifyNotification *n = notify_notification_new (_("Your battery is now fully recharged"), "", /* "battery" */ NULL);
 
 	/* XXX: it would be nice to pass this as a named icon */
 	notify_notification_set_icon_from_pixbuf (n, icon);
