--- rdiff-backup-2.0.5/setup.py-orig	Sat Jul 25 03:53:18 2020
+++ rdiff-backup-2.0.5/setup.py	Sat Mar 20 17:12:21 2021
@@ -136,7 +136,7 @@
 
 setup(
     name="rdiff-backup",
-    use_scm_version=True,
+    version="2.0.5",
     description="Backup and Restore utility, easy to use, efficient, locally and remotely usable",
     long_description="""
         rdiff-backup is a simple backup tool which can be used locally and remotely,
@@ -223,5 +223,4 @@
         'clean': clean,
     },
     install_requires=['setuptools'],
-    setup_requires=['setuptools_scm'],
 )
