Python 3.9 works well with cffi >= 2.

--- zstandard-0.25.0/pyproject.toml.orig
+++ zstandard-0.25.0/pyproject.toml
@@ -35,8 +35,7 @@
 
 [build-system]
 requires = [
-    "cffi~=1.17; platform_python_implementation != 'PyPy' and python_version < '3.14'",
-    "cffi>=2.0.0b; platform_python_implementation != 'PyPy' and python_version >= '3.14'",
+    "cffi",
     "packaging",
     "setuptools>=77.0.0",
 ]
