--- ipmitool-1.8.11/src/plugins/ipmi_intf.c	Mon Jul 30 16:02:31 2012
+++ ipmitool-1.8.11/src/plugins/ipmi_intf.c	Mon Jul 30 16:02:51 2012
@@ -64,6 +64,9 @@
 #endif
 
 struct ipmi_intf * ipmi_intf_table[] = {
+#ifdef IPMI_INTF_BMC
+	&ipmi_bmc_intf,
+#endif
 #ifdef IPMI_INTF_OPEN
 	&ipmi_open_intf,
 #endif
@@ -73,9 +76,6 @@
 #ifdef IPMI_INTF_LIPMI
 	&ipmi_lipmi_intf,
 #endif
-#ifdef IPMI_INTF_BMC
-	&ipmi_bmc_intf,
-#endif
 #ifdef IPMI_INTF_LAN
 	&ipmi_lan_intf,
 #endif
