Copyright (c) 2011, Oracle and/or its affiliates. All rights reserved

--- ipmitool-1.8.11.old/configure.in	Thu Aug  7 11:26:08 2008
+++ ipmitool-1.8.11/configure.in	Tue Oct 25 11:48:31 2011
@@ -63,12 +63,7 @@
 	# openssl libs are in /usr/sfw/lib on solaris 10
         LIBS="$LIBS -R/usr/sfw/lib"
 	# disable the linux-specific interfaces
-	# and bmc on SPARC (there is no bmc driver for SPARC)
-	if [[ `mach -p` = sparc ]]; then
-		enable_intf_bmc=no
-	else
-		enable_intf_bmc=yes
-	fi
+	enable_intf_bmc=yes
 	enable_intf_imb=no
 	enable_intf_open=no
 	enable_intf_lipmi=no
