--- ipmitool-1.8.10/lib/ipmi_sdr.c.orig	Fri Nov 21 13:00:00 2008
+++ ipmitool-1.8.10/lib/ipmi_sdr.c	Fri Nov 21 13:01:00 2008
@@ -1760,7 +1760,8 @@
 				printf("ns  | %2d.%1d | ",
 				       sensor->entity.id,
 				       sensor->entity.instance);
-				if (IS_SCANNING_DISABLED(rsp->data[1]))
+				if ((rsp != NULL) &&
+				    IS_SCANNING_DISABLED(rsp->data[1]))
 					printf("Disabled");
 				else
 					printf("No Reading");
