Copyright (c) 2011, Oracle and/or its affiliates. All rights reserved

--- ipmitool-1.8.11.old/configure.in	Tue Oct 25 12:43:52 2011
+++ ipmitool-1.8.11/configure.in	Tue Oct 25 12:46:49 2011
@@ -65,7 +65,7 @@
 	# disable the linux-specific interfaces
 	enable_intf_bmc=yes
 	enable_intf_imb=no
-	enable_intf_open=no
+	enable_intf_open=yes
 	enable_intf_lipmi=no
 	enable_ipmishell=no
 	enable_all_options=no
@@ -219,6 +219,10 @@
 		[Define to 1 if you have the <sys/ipmi.h> header file.])],
 		[echo "** Unable to find OpenIPMI header files.  Using internal version."])])
 
+AC_CHECK_HEADER([sys/ioccom.h],
+	[AC_DEFINE(HAVE_IOCCOM_H, [1],
+	[Define to 1 if you have the <sys/ioccom.h> header file.])])
+
 dnl look for FreeIPMI files
 AC_CHECK_LIB(freeipmi, ipmi_open_inband, [have_free=yes], [have_free=no])
 if test "x$have_free" != "xyes"; then
--- ipmitool-1.8.11.old/src/ipmievd.c	Wed Feb 25 12:38:52 2009
+++ ipmitool-1.8.11/src/ipmievd.c	Fri Oct 21 16:41:54 2011
@@ -66,6 +66,10 @@
 #  include <sys/poll.h>
 #endif /* IPMI_INTF_OPEN */
 
+#ifdef HAVE_IOCCOM_H
+#include <sys/ioccom.h>
+#endif
+
 #include <ipmitool/helper.h>
 #include <ipmitool/log.h>
 #include <ipmitool/ipmi.h>
--- ipmitool-1.8.11.old/src/plugins/open/open.c	Wed Feb 25 12:38:53 2009
+++ ipmitool-1.8.11/src/plugins/open/open.c	Fri Oct 21 16:41:08 2011
@@ -62,6 +62,10 @@
 # include "open.h"
 #endif
 
+#ifdef HAVE_IOCCOM_H
+#include <sys/ioccom.h>
+#endif
+
 extern int verbose;
 
 static int
