--- bison-3.7/Makefile.in.orig	2020-07-24 10:39:34.580098670 +0000
+++ bison-3.7/Makefile.in	2020-07-24 10:43:04.145242328 +0000
@@ -608,6 +608,12 @@
 @ENABLE_YACC_TRUE@am_lib_liby_a_OBJECTS = lib/main.$(OBJEXT) \
 @ENABLE_YACC_TRUE@	lib/yyerror.$(OBJEXT)
 lib_liby_a_OBJECTS = $(am_lib_liby_a_OBJECTS)
+
+#doing this for just the OBJECTS macros doesn't seeem to work
+#and we rely on the approprate xregs/no-app-regs flag already
+# being in our CFLAGS from shared-macros.mk
+lib/yyerror.o lib/main.o: CFLAGS += $(CC_PIC)
+
 am__objects_9 = examples/c++/calc++/calc__-driver.$(OBJEXT) \
 	examples/c++/calc++/calc__-scanner.$(OBJEXT) \
 	examples/c++/calc++/calc__-calc++.$(OBJEXT)
@@ -3318,6 +3324,9 @@
 @ENABLE_YACC_TRUE@lib_LIBRARIES = lib/liby.a
 @ENABLE_YACC_TRUE@EXTRA_LIBRARIES = lib/liby.a
 @ENABLE_YACC_TRUE@lib_liby_a_SOURCES = lib/main.c lib/yyerror.c
+
+lib_LIBRARIES += lib/liby.so.1
+
 # Prettify Automake-computed names of compiled objects.
 src_bison_SHORTNAME = bison
 src_bison_CPPFLAGS = $(AM_CPPFLAGS) -DINSTALLDIR=\"$(bindir)\"
@@ -3486,6 +3495,10 @@
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
+lib/liby.so.1:    $(lib_liby_a_OBJECTS) $(lib_liby_a_DEPENDENCIES)
+	-rm -f lib/liby.so.1
+	$(CC) $(CFLAGS) -shared -Wl,-h -Wl,liby.so.1 -o lib/liby.so.1 $(lib_liby_a_OBJECTS) $(lib_liby_a_LIBADD)
+
 .SUFFIXES:
 .SUFFIXES: .gv .eps .pdf .svg .yy .stamp .c .cc .dvi .i .l .ll .log .o .obj .ps .test .test$(EXEEXT) .trs .y
 am--refresh: Makefile
