--- brotli-0.6.0/CMakeLists.txt.1	2017-06-23 16:27:16.671072865 +0000
+++ brotli-0.6.0/CMakeLists.txt	2017-06-23 16:28:09.558278681 +0000
@@ -35,9 +35,9 @@
   set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
   set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
   list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_LIBDIR}" isSystemDir)
-  if ("${isSystemDir}" STREQUAL "-1")
-    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_LIBDIR}")
-  endif()
+#  if ("${isSystemDir}" STREQUAL "-1")
+#    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_LIBDIR}")
+#  endif()
 endif()
 
 # Parse version information from common/version.h. Normally we would
