--- support/shobj-conf.orig
+++ support/shobj-conf
@@ -111,6 +111,7 @@
 
 #	SHLIB_XLDFLAGS='-R $(libdir)'
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)'
+	SHLIB_LIBS='$(TERMCAP_LIB)'
 	;;
 
 sunos5*|solaris2*)
@@ -120,6 +121,7 @@
 
 #	SHLIB_XLDFLAGS='-R $(libdir)'
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)'
+	SHLIB_LIBS='$(TERMCAP_LIB)'
 	;;
 
 # All versions of Linux (including Gentoo/FreeBSD) or the semi-mythical GNU Hurd.
