--- hunspell-1.3.4/src/tools/hunspell.cxx.1	2016-02-18 14:49:57.536075973 +0300
+++ hunspell-1.3.4/src/tools/hunspell.cxx	2016-02-18 14:51:16.157456452 +0300
@@ -112,9 +112,8 @@
 #include "odfparser.hxx"
 
 #define LIBDIR                \
-  "/usr/share/hunspell:"      \
-  "/usr/share/myspell:"       \
-  "/usr/share/myspell/dicts:" \
+  "/usr/share/spell/hunspell:"      \
+  "/usr/share/spell/myspell:"       \
   "/Library/Spelling"
 #define USEROOODIR                    \
   ".openoffice.org/3/user/wordbook:"  \
