diff -uprN gtkimageview-1.6.4/configure.in gtkimageview-1.6.4-new/configure.in
--- gtkimageview-1.6.4/configure.in	2009-04-06 04:02:43.000000000 +0800
+++ gtkimageview-1.6.4-new/configure.in	2009-12-14 15:47:28.829003188 +0800
@@ -39,7 +39,9 @@ GTK_DOC_CHECK([1.8])
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+if test "x$GCC" = "xyes"; then
+    CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+fi
 
 
 ######################################################################
