--- gtkimageview-1.6.4/src/gtkiimagetool.c	2009-04-05 01:25:15.000000000 +0200
+++ gtkimageview-1.6.4-patch/src/gtkiimagetool.c	2009-10-11 12:43:10.463578007 +0200
@@ -169,7 +169,7 @@
                              GdkDrawable       *drawable)
 {
     GtkIImageToolClass *klass = GTK_IIMAGE_TOOL_GET_CLASS (tool);
-    return klass->paint_image (tool, opts, drawable);
+    klass->paint_image (tool, opts, drawable);
 }
 
 /*************************************************************/
