illumos defines TIOCGWINSZ in termios.h

--- CalcMySky-0.3.0/CalcMySky/cmdline.cpp.orig	2023-03-28 21:00:36.000000000 +0200
+++ CalcMySky-0.3.0/CalcMySky/cmdline.cpp	2023-04-26 19:39:34.858814701 +0200
@@ -11,6 +11,9 @@
 #   include <sys/ioctl.h>
 #   include <unistd.h>
 #   include <stdio.h>
+#ifdef __sun__
+#   include <termios.h>
+#endif
 #elif defined Q_OS_WIN
 #   include <windows.h>
 #endif
