From 76e51a6343000165747b39228c0070bbbfa346ba Mon Sep 17 00:00:00 2001
From: Arvind Umrao <arvind.umrao@oracle.com>
Date: Tue, 8 May 2012 11:58:06 +0530
Subject: [PATCH] 7162172

---
 Xtranssock.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git Xtranssock.c Xtranssock.c
--- a/Xtranssock.c
+++ b/Xtranssock.c
@@ -1770,7 +1770,7 @@ TRANS(SocketINETConnect) (XtransConnInfo ciptr, char *host, char *port)
 	 * only affect one of a set of addresses.  
 	 */
 
-	if (olderrno == ECONNREFUSED || olderrno == EINTR
+	if (olderrno == ECONNREFUSED 
 #if defined(IPv6) && defined(AF_INET6)
 	  || (((addrlist->addr->ai_next != NULL) ||
 	        (addrlist->addr != addrlist->firstaddr)) &&
@@ -1783,7 +1783,7 @@ TRANS(SocketINETConnect) (XtransConnInfo ciptr, char *host, char *port)
 #endif
 	    )
 	    res = TRANS_TRY_CONNECT_AGAIN;
-	else if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS)
+	else if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS || olderrno == EINTR)
 	    res = TRANS_IN_PROGRESS;
 	else
 	{
@@ -2092,7 +2092,7 @@ TRANS(SocketUNIXConnect) (XtransConnInfo ciptr, char *host, char *port)
 	    if (olderrno == EWOULDBLOCK || olderrno == EINPROGRESS)
 		return TRANS_IN_PROGRESS;
 	    else if (olderrno == EINTR)
-		return TRANS_TRY_CONNECT_AGAIN;
+		return TRANS_IN_PROGRESS;
 	    else if (olderrno == ENOENT || olderrno == ECONNREFUSED) {
 		/* If opening as abstract socket failed, try again normally */
 		if (abstract) {
1.7.3.2

