--- totem-2.30.2/src/backend/bacon-video-widget-gst-0.10.c.orig	2013-12-19 16:12:46.941778257 +0400
+++ totem-2.30.2/src/backend/bacon-video-widget-gst-0.10.c	2013-12-19 16:11:56.004703161 +0400
@@ -305,6 +305,8 @@
 static gboolean bvw_set_playback_direction (BaconVideoWidget *bvw, gboolean forward);
 static gboolean bacon_video_widget_seek_time_no_lock (BaconVideoWidget *bvw, gint64 _time, GError **error);
 
+static gboolean notify_volume_idle_cb (BaconVideoWidget *bvw);
+
 static GtkWidgetClass *parent_class = NULL;
 
 static int bvw_signals[LAST_SIGNAL] = { 0 };
