diff -urN totem-2.30.2/configure.in ../SUNWgnome-media-player-2.30.0.hacked/totem-2.30.2/configure.in
--- totem-2.30.2/configure.in	2011-05-31 10:16:24.959780907 +0000
+++ ../SUNWgnome-media-player-2.30.0.hacked/totem-2.30.2/configure.in	2011-06-01 16:04:56.621671714 +0000
@@ -156,7 +156,7 @@
   [AS_HELP_STRING([--with-smclient],[which smclient backend to use (xsmp/win32/quartz)])],
   [],
   [case "$GDK_TARGET" in
-    x11) with_smclient=xsmp SMCLIENT_PKGS="sm" ;;
+    x11) with_smclient=xsmp SMCLIENT_PKGS="sm ice" ;;
     win32|quartz) with_smclient=$GDK_TARGET ;;
     *) with_smclient=no ;;
    esac])
