--- totem-2.24.0-orig/data/handlers.sh	Fri Sep 19 13:43:46 2008
+++ totem-2.24.0/data/handlers.sh	Wed Oct  8 17:02:54 2008
@@ -29,7 +29,12 @@
 
 	KEY="command"
 	TYPE="string";
-	DEFAULT="totem \"%s\"";
+	if [ "$i" = "icyx" ];
+	then
+		DEFAULT="";
+	else	
+		DEFAULT="totem \"%s\"";
+	fi
 	schema;
 
 	KEY="needs_terminal"
--- totem-2.27.92/data/mime-type-list.txt-orig	2009-09-17 10:22:25.268130909 -0500
+++ totem-2.27.92/data/mime-type-list.txt	2009-09-17 10:22:36.191942733 -0500
@@ -5,7 +5,6 @@ application/sdp
 application/smil
 application/smil+xml
 application/vnd.ms-wpl
-application/vnd.rn-realmedia
 application/x-extension-m4a
 application/x-extension-mp4
 application/x-flac
@@ -47,12 +45,8 @@ audio/x-ms-wma
 audio/x-musepack
 audio/x-pn-aiff
 audio/x-pn-au
-audio/x-pn-realaudio
-audio/x-pn-realaudio-plugin
 audio/x-pn-wav
 audio/x-pn-windows-acm
-audio/x-realaudio
-audio/x-real-audio
 audio/x-sbc
 audio/x-scpls
 audio/x-speex
@@ -62,7 +56,6 @@ audio/x-wavpack
 audio/x-vorbis
 audio/x-vorbis+ogg
 audio/x-xm
-image/vnd.rn-realpix
 image/x-pict
 misc/ultravox
 text/google-video-pointer
@@ -79,7 +72,6 @@ video/ogg
 video/quicktime
 video/vivo
 video/vnd.divx
-video/vnd.rn-realvideo
 video/vnd.vivo
 video/x-anim
 video/x-avi
diff -ruN totem-2.30.0.orig/data/mime-type-list.txt totem-2.30.0/data/mime-type-list.txt
--- totem-2.30.0.orig/data/mime-type-list.txt	2010-04-12 22:01:14.632291275 +0100
+++ totem-2.30.0/data/mime-type-list.txt	2010-04-12 22:01:41.145205829 +0100
@@ -28,7 +28,6 @@
 audio/mpegurl
 audio/ogg
 audio/prs.sid
-audio/vnd.rn-realaudio
 audio/x-aiff
 audio/x-ape
 audio/x-flac
