--- totem-2.30.2/data/totem-bugreport.py-orig	2010-07-21 21:42:10.852225325 -0500
+++ totem-2.30.2/data/totem-bugreport.py	2010-07-21 21:42:17.116012732 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 import gtk
 import os
 from datetime import datetime
--- totem-2.30.2/src/plugins/iplayer/iplayer2.py-orig	2010-07-21 21:42:41.547098091 -0500
+++ totem-2.30.2/src/plugins/iplayer/iplayer2.py	2010-07-21 21:42:48.004682027 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 
 # Python libs
 import re, os
