diff -ruN totem-2.23.91.orig/configure.in totem-2.23.91/configure.in
--- totem-2.23.91.orig/configure.in	2008-09-01 20:53:57.953228730 +0100
+++ totem-2.23.91/configure.in	2008-09-01 20:54:35.835743891 +0100
@@ -613,7 +613,7 @@
 if test "$enable_browser_plugins" = "yes" ; then
 	AC_DEFINE([ENABLE_BROWSER_PLUGINS],[1],[Define to build the browser plugin])
 
-	BROWSER_PLUGIN_DIR="${BROWSER_PLUGIN_DIR:-"\${libdir}/mozilla/plugins"}"
+	BROWSER_PLUGIN_DIR="${BROWSER_PLUGIN_DIR:-"\${libdir}/firefox/plugins"}"
 	AC_ARG_VAR([BROWSER_PLUGIN_DIR],[Where to install the plugin to])
 
 	PKG_CHECK_MODULES([BROWSER_PLUGIN],
