--- cffi-1.17.1/testing/cffi0/test_ownlib.py.orig
+++ cffi-1.17.1/testing/cffi0/test_ownlib.py
@@ -174,7 +174,7 @@
                 unicode_name = u+'testownlib'
                 encoded = str(unicode_name)
             subprocess.check_call(
-                "cc testownlib.c -shared -fPIC -o '%s.so'" % (encoded,),
+                "gcc testownlib.c -shared -fPIC -o '%s.so'" % (encoded,),
                 cwd=str(udir), shell=True)
             cls.module = os.path.join(str(udir), unicode_name + (u+'.so'))
         print(repr(cls.module))
