The patch was taken from solaris-userland.

--- cffi-1.14.0/testing/cffi0/test_ownlib.py
+++ cffi-1.14.0/testing/cffi0/test_ownlib.py
@@ -160,7 +160,7 @@ class TestOwnLib(object):
                 unicode_name = u+'testownlib'
                 encoded = str(unicode_name)
             subprocess.check_call(
-                "cc testownlib.c -shared -fPIC -o '%s.so'" % (encoded,),
+                os.getenv('TESTOWNLIB_CC') % ('testownlib.c', encoded),
                 cwd=str(udir), shell=True)
             cls.module = os.path.join(str(udir), unicode_name + (u+'.so'))
         print(repr(cls.module))
--- cffi-1.14.0/testing/support.py
+++ cffi-1.14.0/testing/support.py
@@ -109,6 +109,8 @@ else:
         extra_compile_args.append('-Qunused-arguments')
     else:
         # assume a standard gcc
+        # We add the -Wno-unknown-pragmas because Solaris' python is build
+        # with Studio, which uses no_inline rather than noinline.
         extra_compile_args = ['-Werror', '-Wall', '-Wextra', '-Wconversion',
-                              '-Wno-unused-parameter',
+                              '-Wno-unused-parameter', '-Wno-unknown-pragmas',
                               '-Wno-unreachable-code']

