--- pdns-recursor-3.7.3/sysdeps/SunOS.inc.orig	Mon Nov 23 21:51:39 2015
+++ pdns-recursor-3.7.3/sysdeps/SunOS.inc	Mon Nov 23 21:51:57 2015
@@ -13,12 +13,8 @@
 	CXXFLAGS+= -DHAVE_STRCASESTR
 endif
 
-OPTIONALS:=optional/portsmplexer.o optional/devpollmplexer.o
+OPTIONALS:=optional/devpollmplexer.o
 	
-optional/portsmplexer.o: portsmplexer.cc
-	$(CXX) $(CXXFLAGS) -c $< -o $@  ; true
-
-
 optional/devpollmplexer.o: devpollmplexer.cc
 	$(CXX) $(CXXFLAGS) -c $< -o $@  ; true
 
