https://savannah.gnu.org/bugs/?63924

--- groff-1.23.0/tmac/tmac.am.orig
+++ groff-1.23.0/tmac/tmac.am
@@ -305,9 +305,6 @@
 	$(AM_V_at)if test -n "$(tmac_wrap)"; then \
 	  for m in ""$(tmac_wrap); do \
 	    $(RM) $(top_builddir)/tmac/$$m-wrap; \
-	    if test "$$m" = an; then \
-	      echo .do mso an.tmac >>$(top_builddir)/tmac/$$m-wrap; \
-	    fi; \
 	    echo .cp 1 >>$(top_builddir)/tmac/$$m-wrap; \
 	    echo .so $(sys_tmac_prefix)$$m \
 	      >>$(top_builddir)/tmac/$$m-wrap; \
