--- dconf-0.6/gsettings/Makefile.am	2015-02-23 02:50:52.604095304 +0300
+++ dconf-0.6/gsettings/Makefile.am	2015-02-23 02:50:40.781299799 +0300
@@ -4,7 +4,7 @@
 giomodules_PROGRAMS = libdconfsettings.so
 
 libdconfsettings_so_LDADD = $(gio_LIBS)
-libdconfsettings_so_LDFLAGS = -module -avoid-version -shared
+libdconfsettings_so_LDFLAGS = -shared
 libdconfsettings_so_SOURCES = \
 	../engine/dconf-engine.c	\
 	../common/dconf-shmdir.c	\
