--- RBTools-5.0/rbtools/clients/tests/test_git.py.orig
+++ RBTools-5.0/rbtools/clients/tests/test_git.py
@@ -23,6 +23,8 @@
                                    run_process,
                                    run_process_exec)
 
+import pytest
+
 
 class BaseGitClientTests(SCMClientTestCase):
     """Base class for unit tests for GitClient.
@@ -1908,6 +1910,7 @@
         self.assertEqual(repository_info.local_path, self.checkout_dir)
         self.assertEqual(client._type, client.TYPE_GIT_P4)
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff(self):
         """Testing GitClient.diff with git-p4"""
         client = self.build_client(needs_diff=True)
@@ -1966,6 +1969,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_spaces_in_filename(self):
         """Testing GitClient.diff with git-p4 with spaces in filename"""
         client = self.build_client(needs_diff=True)
@@ -2012,6 +2016,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_rename(self):
         """Testing GitClient.diff with renamed file"""
         client = self.build_client(needs_diff=True)
@@ -2041,6 +2046,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_rename_and_changes(self):
         """Testing GitClient.diff with renamed file and changes"""
         client = self.build_client(needs_diff=True)
@@ -2083,6 +2089,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_deletes(self):
         """Testing GitClient.diff with deleted files"""
         client = self.build_client(needs_diff=True)
@@ -2124,6 +2131,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_multiple_commits(self):
         """Testing GitClient.diff with git-p4 and multiple commits"""
         client = self.build_client(needs_diff=True)
@@ -2166,6 +2174,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_exclude_patterns(self):
         """Testing GitClient.diff with git-p4 and file exclusion"""
         client = self.build_client(needs_diff=True)
@@ -2199,6 +2208,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_exclude_in_subdir(self):
         """Testing GitClient simple diff with file exclusion in a subdir"""
         client = self.build_client(needs_diff=True)
@@ -2235,6 +2245,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_exclude_patterns_root_pattern_in_subdir(self):
         """Testing GitClient diff with file exclusion in the repo root"""
         client = self.build_client(needs_diff=True)
