--- rbtools-5.2.1/rbtools/clients/tests/test_git.py.orig
+++ rbtools-5.2.1/rbtools/clients/tests/test_git.py
@@ -24,6 +24,8 @@
                                    run_process,
                                    run_process_exec)
 
+import pytest
+
 
 class BaseGitClientTests(SCMClientTestCase):
     """Base class for unit tests for GitClient.
@@ -1936,6 +1938,7 @@
         self.assertEqual(repository_info.local_path, self.checkout_dir)
         self.assertEqual(client._type, client.TYPE_GIT_P4)
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff(self):
         """Testing GitClient.diff with git-p4"""
         client = self.build_client(needs_diff=True)
@@ -1994,6 +1997,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_spaces_in_filename(self):
         """Testing GitClient.diff with git-p4 with spaces in filename"""
         client = self.build_client(needs_diff=True)
@@ -2040,6 +2044,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_rename(self):
         """Testing GitClient.diff with renamed file"""
         client = self.build_client(needs_diff=True)
@@ -2069,6 +2074,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_rename_and_changes(self):
         """Testing GitClient.diff with renamed file and changes"""
         client = self.build_client(needs_diff=True)
@@ -2111,6 +2117,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_deletes(self):
         """Testing GitClient.diff with deleted files"""
         client = self.build_client(needs_diff=True)
@@ -2152,6 +2159,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_multiple_commits(self):
         """Testing GitClient.diff with git-p4 and multiple commits"""
         client = self.build_client(needs_diff=True)
@@ -2194,6 +2202,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_exclude_patterns(self):
         """Testing GitClient.diff with git-p4 and file exclusion"""
         client = self.build_client(needs_diff=True)
@@ -2227,6 +2236,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_exclude_in_subdir(self):
         """Testing GitClient simple diff with file exclusion in a subdir"""
         client = self.build_client(needs_diff=True)
@@ -2263,6 +2273,7 @@
                 'parent_diff': None,
             })
 
+    @pytest.mark.skip(reason="always fails: https://hellosplat.com/s/beanbag/tickets/4996/")
     def test_diff_with_exclude_patterns_root_pattern_in_subdir(self):
         """Testing GitClient diff with file exclusion in the repo root"""
         client = self.build_client(needs_diff=True)
