We have no trio pacakged yet.

--- httpx-0.27.2/pyproject.toml.orig
+++ httpx-0.27.2/pyproject.toml
@@ -120,8 +120,6 @@
 filterwarnings = [
   "error",
   "ignore: You seem to already have a custom sys.excepthook handler installed. I'll skip installing Trio's custom handler, but this means MultiErrors will not show full tracebacks.:RuntimeWarning",
-  # See: https://github.com/agronholm/anyio/issues/508
-  "ignore: trio.MultiError is deprecated since Trio 0.22.0:trio.TrioDeprecationWarning"
 ]
 markers = [
   "copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup",
--- httpx-0.27.2/requirements.txt.orig
+++ httpx-0.27.2/requirements.txt
@@ -23,7 +23,5 @@
 mypy==1.11.0
 pytest==8.3.1
 ruff==0.5.4
-trio==0.26.0
-trio-typing==0.10.0
 trustme==1.1.0
 uvicorn==0.30.3
--- httpx-0.27.2/tests/concurrency.py.orig
+++ httpx-0.27.2/tests/concurrency.py
@@ -5,11 +5,7 @@
 import asyncio
 
 import sniffio
-import trio
 
 
 async def sleep(seconds: float) -> None:
-    if sniffio.current_async_library() == "trio":
-        await trio.sleep(seconds)  # pragma: no cover
-    else:
-        await asyncio.sleep(seconds)
+    await asyncio.sleep(seconds)
