We have no eventlet packaged yet.

--- gunicorn-23.0.0/gunicorn/workers/__init__.py.orig
+++ gunicorn-23.0.0/gunicorn/workers/__init__.py
@@ -5,7 +5,6 @@
 # supported gunicorn workers.
 SUPPORTED_WORKERS = {
     "sync": "gunicorn.workers.sync.SyncWorker",
-    "eventlet": "gunicorn.workers.geventlet.EventletWorker",
     "gevent": "gunicorn.workers.ggevent.GeventWorker",
     "gevent_wsgi": "gunicorn.workers.ggevent.GeventPyWSGIWorker",
     "gevent_pywsgi": "gunicorn.workers.ggevent.GeventPyWSGIWorker",
--- gunicorn-23.0.0/requirements_test.txt.orig
+++ gunicorn-23.0.0/requirements_test.txt
@@ -1,5 +1,4 @@
 gevent
-eventlet
 coverage
 pytest>=7.2.0
 pytest-cov
