--- gimp-2.6.6/desktop/gimp.desktop.in.in.old	2009-04-27 16:35:45.492062000 +0800
+++ gimp-2.6.6/desktop/gimp.desktop.in.in	2009-04-27 16:39:09.991566000 +0800
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-_Name=GNU Image Manipulation Program
+# SUN_BRANDING
+_Name=Image Editor
 _GenericName=Image Editor
-_Comment=Create images and edit photographs
+# SUN_BRANDING
+_Comment=Create, edit or convert images between standard graphics formats
 Exec=@GIMP_COMMAND@ %U
 TryExec=gimp-@GIMP_APP_VERSION@
 Icon=gimp
