diff -urp -x '*~' -x '*.orig' src/radeon_dri.h src/radeon_dri.h
--- a/src/radeon_dri.h	2008-08-17 10:54:00.000000000 -0700
+++ b/src/radeon_dri.h	2009-01-10 19:09:08.415906000 -0800
@@ -75,14 +75,17 @@ typedef struct {
     /* MMIO register data */
     drm_handle_t     registerHandle;
     drmSize       registerSize;
+	int		padding0;
 
     /* CP in-memory status information */
     drm_handle_t     statusHandle;
     drmSize       statusSize;
+	int		padding1;
 
     /* CP GART Texture data */
     drm_handle_t     gartTexHandle;
     drmSize       gartTexMapSize;
+	int		padding2;
     int           log2GARTTexGran;
     int           gartTexOffset;
     unsigned int  sarea_priv_offset;

 
