--- squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/jmemdatasrc.c	Fri Oct  5 02:00:00 2018
+++ p0/squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/jmemdatasrc.c	Sat Dec 25 17:36:07 2021
@@ -151,7 +151,6 @@
   /* no work necessary here */
 }
 
-#if !defined(USE_LIBRARY_SHARED)
 /*
  * Prepare for input from a stdio stream.
  * The caller must have already opened the stream, and is responsible
@@ -189,7 +188,6 @@
   src->pub.bytes_in_buffer = 0; /* forces fill_input_buffer on first read */
   src->pub.next_input_byte = NULL; /* until buffer loaded */
 }
-#endif
 
 /* This function allows data to be moved if necessary */
 GLOBAL(int) jpeg_mem_src_newLocationOfData (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize) {
