--- squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/jmemdatadst.c	Fri Oct  5 02:00:00 2018
+++ p0/squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/jmemdatadst.c	Sat Dec 25 17:40:27 2021
@@ -115,7 +115,6 @@
   }
 }
 
-#if !defined(USE_LIBRARY_SHARED)
 /*
  * Prepare for output to a stdio stream.
  * The caller must have already opened the stream, and is responsible
@@ -145,5 +144,4 @@
   dest->pSpaceUsed = pDestinationSize;
   *(dest->pSpaceUsed) = 0;
 }
-#endif
 
