--- squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/JPEGReadWriter2Plugin.h	Fri Oct  5 02:00:00 2018
+++ p0/squeak-3814/platforms/Cross/plugins/JPEGReadWriter2Plugin/JPEGReadWriter2Plugin.h	Fri Dec 31 18:27:22 2021
@@ -12,9 +12,9 @@
 typedef struct error_mgr2* error_ptr2;
 
 void error_exit (j_common_ptr cinfo);
-GLOBAL(void) jpeg_mem_src (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
-GLOBAL(int) jpeg_mem_src_newLocationOfData (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
-GLOBAL(void) jpeg_mem_dest (j_compress_ptr cinfo, char * pDestination, unsigned *pDestinationSize);
+GLOBAL(void) sqJpeg_mem_src (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
+GLOBAL(int) sqJpeg_mem_src_newLocationOfData (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
+GLOBAL(void) sqJpeg_mem_dest (j_compress_ptr cinfo, char * pDestination, unsigned *pDestinationSize);
 void primJPEGWriteImageonByteArrayformqualityprogressiveJPEGerrorMgrWriteScanlines(
     unsigned int, 
     unsigned int, 
