--- squeak-platforms-Cross-3840/plugins/JPEGReadWriter2Plugin/JPEGReadWriter2Plugin.h	Mon Jun 16 08:44:09 2025
+++ p0/squeak-platforms-Cross-3840/plugins/JPEGReadWriter2Plugin/JPEGReadWriter2Plugin.h	Tue Jun 17 11:44:51 2025
@@ -12,9 +12,9 @@
 typedef struct error_mgr2* error_ptr2;
 
 void error_exit (j_common_ptr cinfo);
-GLOBAL(void) jpeg_mem_src (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
-GLOBAL(int) jpeg_mem_src_newLocationOfData (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
-GLOBAL(void) jpeg_mem_dest (j_compress_ptr cinfo, char * pDestination, unsigned *pDestinationSize);
+GLOBAL(void) sqJpeg_mem_src (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
+GLOBAL(int) sqJpeg_mem_src_newLocationOfData (j_decompress_ptr cinfo, char * pSourceData, unsigned sourceDataSize);
+GLOBAL(void) sqJpeg_mem_dest (j_compress_ptr cinfo, char * pDestination, unsigned *pDestinationSize);
 void primJPEGWriteImageonByteArrayformqualityprogressiveJPEGerrorMgrWriteScanlines(
     unsigned int, 
     unsigned int, 
