--- intltool-0.51.0/intltool.m4.orig
+++ intltool-0.51.0/intltool.m4
@@ -126,8 +126,7 @@
 fi
 xgversion="`$XGETTEXT --version|grep '(GNU ' 2> /dev/null`"
 mmversion="`$MSGMERGE --version|grep '(GNU ' 2> /dev/null`"
-mfversion="`$MSGFMT --version|grep '(GNU ' 2> /dev/null`"
-if test -z "$xgversion" -o -z "$mmversion" -o -z "$mfversion"; then
+if test -z "$xgversion" -o -z "$mmversion"; then
     AC_MSG_ERROR([GNU gettext tools not found; required for intltool])
 fi
 
