--- platforms/Cross/plugins/JPEGReadWriter2Plugin/sqJPEGReadWriter2Plugin.c	Mon Jun 16 08:44:10 2025
+++ p0/platforms/Cross/plugins/JPEGReadWriter2Plugin/sqJPEGReadWriter2Plugin.c	Tue Jun 17 11:44:51 2025
@@ -63,7 +63,7 @@
 
 	if (*destinationSizePtr) {
 		jpeg_create_compress(pcinfo);
-		jpeg_mem_dest(pcinfo, destination, destinationSizePtr);
+		sqJpeg_mem_dest(pcinfo, destination, destinationSizePtr);
 
 		pcinfo->image_width = width;
 		pcinfo->image_height = height;
@@ -159,7 +159,7 @@
 	}
 
 	if (ok)
-		ok = jpeg_mem_src_newLocationOfData(pcinfo, source, sourceSize);
+		ok = sqJpeg_mem_src_newLocationOfData(pcinfo, source, sourceSize);
 
 	if (ok) {
 		jpeg_start_decompress(pcinfo);
@@ -301,7 +301,7 @@
 
 	if (sourceSize) {
 		jpeg_create_decompress(pcinfo);
-		jpeg_mem_src(pcinfo, source, sourceSize);
+		sqJpeg_mem_src(pcinfo, source, sourceSize);
 		jpeg_read_header(pcinfo, TRUE);
 	}
 }
