--- swipl-9.3.24/desktop/swipl-win.desktop.in.orig	2025-04-25 20:36:34.000000000 +0200
+++ swipl-9.3.24/desktop/swipl-win.desktop.in	2025-05-24 21:00:40.900430664 +0200
@@ -6,3 +6,4 @@
 Terminal=false
 StartupWMClass=SWI-Prolog
 MimeType=application/x-prolog
+Categories=Development
--- swipl-9.3.24/desktop/swipl.desktop.in.orig	2025-04-25 20:36:34.000000000 +0200
+++ swipl-9.3.24/desktop/swipl.desktop.in	2025-05-24 21:00:23.636300374 +0200
@@ -6,3 +6,4 @@
 Terminal=true
 StartupWMClass=SWI-Prolog
 MimeType=application/x-prolog
+Categories=Development
