diff --git a/internal/disk/directio_unix.go b/internal/disk/directio_unix.go
index 883df95..cb4b13e 100644
--- a/internal/disk/directio_unix.go
+++ b/internal/disk/directio_unix.go
@@ -1,5 +1,5 @@
-//go:build !windows && !darwin && !openbsd && !plan9
-// +build !windows,!darwin,!openbsd,!plan9
+//go:build !windows && !darwin && !openbsd && !plan9 && !solaris
+// +build !windows,!darwin,!openbsd,!plan9,!solaris
 
 // Copyright (c) 2015-2023 MinIO, Inc.
 //
diff --git a/internal/disk/directio_unsupported.go b/internal/disk/directio_unsupported.go
index 6c579aa..c1471c4 100644
--- a/internal/disk/directio_unsupported.go
+++ b/internal/disk/directio_unsupported.go
@@ -1,4 +1,4 @@
-//go:build windows || openbsd || plan9
+//go:build windows || openbsd || plan9 || solaris
 
 // Copyright (c) 2015-2023 MinIO, Inc.
 //
