--- lzlib-1.15/Makefile.in.orig	2024-11-03 20:02:28.000000000 +0100
+++ lzlib-1.15/Makefile.in	2025-01-19 13:03:40.061598989 +0100
@@ -29,7 +29,7 @@
 	$(AR) $(ARFLAGS) $@ $<
 
 lib$(libname).so.$(soversion) : lzlib_sh.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -fpic -fPIC -shared -Wl,--soname=$@ -o $@ $< || \
+	$(CC) $(CFLAGS) $(LDFLAGS) -fpic -fPIC -shared -Wl,-h,$@ -o $@ $< || \
 	$(CC) $(CFLAGS) $(LDFLAGS) -fpic -fPIC -shared -o $@ $<
 
 bin : $(progname_static) $(progname_shared)
