--- pulseaudio-13.0/src/Makefile.am	2019-09-13 15:10:23.000000000 +0000
+++ pulseaudio-13.0/src/Makefile.am	2019-09-15 19:01:51.015465742 +0000
@@ -295,13 +295,10 @@ TESTS_norun = \
 		sig2str-test \
 		stripnul \
 		echo-cancel-test \
-		lo-latency-test \
-		atomic-test
+		lo-latency-test
 
 # These tests need a running pulseaudio daemon
 TESTS_daemon = \
-		extended-test \
-		passthrough-test \
 		sync-playback
 
 # These tests need a running daemon and take a while to complete
