--- pulseaudio-11.1/src/daemon/default.pa.in-orig	Tue Aug 23 07:50:10 2016
+++ pulseaudio-11.1/src/daemon/default.pa.in	Mon Aug  6 07:50:59 2018
@@ -41,6 +41,7 @@
 #load-module module-alsa-source device=hw:1,0
 ])dnl
 ifelse(@HAVE_OSS_OUTPUT@, 1, [dnl
+load-module module-oss
 #load-module module-oss device="/dev/dsp" sink_name=output source_name=input
 #load-module module-oss-mmap device="/dev/dsp" sink_name=output source_name=input
 ])dnl
