diff --git a/src/pulse/meson.build b/src/pulse/meson.build
index c132630..7729f3c 100644
--- a/src/pulse/meson.build
+++ b/src/pulse/meson.build
@@ -75,7 +75,7 @@ if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
     command : [ join_paths(meson.source_root(), 'scripts/generate-map-file.sh'), 'map-file', 'libpulse.def',
                 [ libpulse_headers, 'simple.h', join_paths(meson.build_root(), 'src', 'pulse', 'version.h') ] ])
 
-  versioning_link_args = ['-Wl,-version-script=' + join_paths(meson.source_root(), 'src', 'pulse', 'map-file')]
+  versioning_link_args = []
 else
   versioning_link_args = []
 endif
