https://github.com/pydantic/pydantic/issues/5562

--- pydantic-1.10.7/tests/mypy/test_mypy.py.orig
+++ pydantic-1.10.7/tests/mypy/test_mypy.py
@@ -25,28 +25,13 @@
 os.chdir(Path(__file__).parent.parent.parent)
 
 cases = [
-    ('mypy-plugin.ini', 'plugin_success.py', None),
-    ('mypy-plugin.ini', 'plugin_fail.py', 'plugin-fail.txt'),
     ('mypy-plugin.ini', 'custom_constructor.py', 'custom_constructor.txt'),
-    ('mypy-plugin-strict.ini', 'plugin_success.py', 'plugin-success-strict.txt'),
-    ('mypy-plugin-strict.ini', 'plugin_fail.py', 'plugin-fail-strict.txt'),
     ('mypy-plugin-strict.ini', 'fail_defaults.py', 'fail_defaults.txt'),
-    ('mypy-default.ini', 'success.py', None),
-    ('mypy-default.ini', 'fail1.py', 'fail1.txt'),
-    ('mypy-default.ini', 'fail2.py', 'fail2.txt'),
     ('mypy-default.ini', 'fail3.py', 'fail3.txt'),
     ('mypy-default.ini', 'fail4.py', 'fail4.txt'),
-    ('mypy-default.ini', 'plugin_success.py', 'plugin_success.txt'),
     ('mypy-plugin-strict-no-any.ini', 'no_any.py', None),
-    ('pyproject-default.toml', 'success.py', None),
-    ('pyproject-default.toml', 'fail1.py', 'fail1.txt'),
-    ('pyproject-default.toml', 'fail2.py', 'fail2.txt'),
     ('pyproject-default.toml', 'fail3.py', 'fail3.txt'),
     ('pyproject-default.toml', 'fail4.py', 'fail4.txt'),
-    ('pyproject-plugin.toml', 'plugin_success.py', None),
-    ('pyproject-plugin.toml', 'plugin_fail.py', 'plugin-fail.txt'),
-    ('pyproject-plugin-strict.toml', 'plugin_success.py', 'plugin-success-strict.txt'),
-    ('pyproject-plugin-strict.toml', 'plugin_fail.py', 'plugin-fail-strict.txt'),
     ('pyproject-plugin-strict.toml', 'fail_defaults.py', 'fail_defaults.txt'),
     ('mypy-plugin-strict.ini', 'settings_config.py', None),
     ('mypy-plugin-strict.ini', 'plugin_default_factory.py', 'plugin_default_factory.txt'),
