diff -ur resource-agents-4.0.1/heartbeat-1/anything resource-agents-4.0.1/heartbeat/anything
--- resource-agents-4.0.1/heartbeat-1/anything	2017-09-17 10:11:54.766134114 +0000
+++ resource-agents-4.0.1/heartbeat/anything	2017-09-17 10:14:53.328898532 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #       OCF Resource Agent compliant resource script.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/apache resource-agents-4.0.1/heartbeat/apache
--- resource-agents-4.0.1/heartbeat-1/apache	2017-09-17 10:11:54.773284307 +0000
+++ resource-agents-4.0.1/heartbeat/apache	2017-09-17 10:14:53.336868947 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	High-Availability Apache/IBMhttp control script
 # 
diff -ur resource-agents-4.0.1/heartbeat-1/asterisk resource-agents-4.0.1/heartbeat/asterisk
--- resource-agents-4.0.1/heartbeat-1/asterisk	2017-09-17 10:11:54.766435025 +0000
+++ resource-agents-4.0.1/heartbeat/asterisk	2017-09-17 10:14:53.348084171 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # Asterisk
diff -ur resource-agents-4.0.1/heartbeat-1/AudibleAlarm resource-agents-4.0.1/heartbeat/AudibleAlarm
--- resource-agents-4.0.1/heartbeat-1/AudibleAlarm	2017-09-17 10:11:54.762403684 +0000
+++ resource-agents-4.0.1/heartbeat/AudibleAlarm	2017-09-17 10:14:53.349687081 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Startup script for the Audible Alarm
 #
diff -ur resource-agents-4.0.1/heartbeat-1/awseip resource-agents-4.0.1/heartbeat/awseip
--- resource-agents-4.0.1/heartbeat-1/awseip	2017-09-17 10:11:54.766533991 +0000
+++ resource-agents-4.0.1/heartbeat/awseip	2017-09-17 10:14:53.351817992 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #    Manage Elastic IP with Pacemaker
diff -ur resource-agents-4.0.1/heartbeat-1/awsvip resource-agents-4.0.1/heartbeat/awsvip
--- resource-agents-4.0.1/heartbeat-1/awsvip	2017-09-17 10:11:54.766651827 +0000
+++ resource-agents-4.0.1/heartbeat/awsvip	2017-09-17 10:14:53.353725742 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #    Manage Secondary Private IP with Pacemaker
diff -ur resource-agents-4.0.1/heartbeat-1/ClusterMon resource-agents-4.0.1/heartbeat/ClusterMon
--- resource-agents-4.0.1/heartbeat-1/ClusterMon	2017-09-17 10:11:54.762630698 +0000
+++ resource-agents-4.0.1/heartbeat/ClusterMon	2017-09-17 10:14:53.355888293 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	ClusterMon OCF RA.
diff -ur resource-agents-4.0.1/heartbeat-1/CTDB resource-agents-4.0.1/heartbeat/CTDB
--- resource-agents-4.0.1/heartbeat-1/CTDB	2017-09-17 10:11:54.762537919 +0000
+++ resource-agents-4.0.1/heartbeat/CTDB	2017-09-17 10:14:53.368372923 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #             OCF Resource Agent for managing CTDB
 #
diff -ur resource-agents-4.0.1/heartbeat-1/db2 resource-agents-4.0.1/heartbeat/db2
--- resource-agents-4.0.1/heartbeat-1/db2	2017-09-17 10:11:54.766957138 +0000
+++ resource-agents-4.0.1/heartbeat/db2	2017-09-17 10:14:53.375526599 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # db2
 #
diff -ur resource-agents-4.0.1/heartbeat-1/Delay resource-agents-4.0.1/heartbeat/Delay
--- resource-agents-4.0.1/heartbeat-1/Delay	2017-09-17 10:11:54.762714930 +0000
+++ resource-agents-4.0.1/heartbeat/Delay	2017-09-17 10:14:53.378853000 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # Support:      users@clusterlabs.org
diff -ur resource-agents-4.0.1/heartbeat-1/dhcpd resource-agents-4.0.1/heartbeat/dhcpd
--- resource-agents-4.0.1/heartbeat-1/dhcpd	2017-09-17 10:11:54.767049570 +0000
+++ resource-agents-4.0.1/heartbeat/dhcpd	2017-09-17 10:14:53.384735671 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #   Resource Agent for managing dhcpd resources.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/docker resource-agents-4.0.1/heartbeat/docker
--- resource-agents-4.0.1/heartbeat-1/docker	2017-09-17 10:11:54.767223991 +0000
+++ resource-agents-4.0.1/heartbeat/docker	2017-09-17 10:14:53.390846414 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # The docker HA resource agent creates and launches a docker container
 # based off a supplied docker image. Containers managed by this agent
diff -ur resource-agents-4.0.1/heartbeat-1/Dummy resource-agents-4.0.1/heartbeat/Dummy
--- resource-agents-4.0.1/heartbeat-1/Dummy	2017-09-17 10:11:54.762796558 +0000
+++ resource-agents-4.0.1/heartbeat/Dummy	2017-09-17 10:14:53.392393180 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	Dummy OCF RA. Does nothing but wait a few seconds, can be
diff -ur resource-agents-4.0.1/heartbeat-1/ethmonitor resource-agents-4.0.1/heartbeat/ethmonitor
--- resource-agents-4.0.1/heartbeat-1/ethmonitor	2017-09-17 10:11:54.767402692 +0000
+++ resource-agents-4.0.1/heartbeat/ethmonitor	2017-09-17 10:14:53.400708684 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	   OCF Resource Agent compliant script.
 #	   Monitor the vitality of a local network interface.
diff -ur resource-agents-4.0.1/heartbeat-1/Evmsd resource-agents-4.0.1/heartbeat/Evmsd
--- resource-agents-4.0.1/heartbeat-1/Evmsd	2017-09-17 10:11:54.762972235 +0000
+++ resource-agents-4.0.1/heartbeat/Evmsd	2017-09-17 10:14:53.402124704 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	Evmsd OCF RA.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/EvmsSCC resource-agents-4.0.1/heartbeat/EvmsSCC
--- resource-agents-4.0.1/heartbeat-1/EvmsSCC	2017-09-17 10:11:54.762881009 +0000
+++ resource-agents-4.0.1/heartbeat/EvmsSCC	2017-09-17 10:14:53.404150984 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Support:      users@clusterlabs.org
 # License:      GNU General Public License (GPL)
diff -ur resource-agents-4.0.1/heartbeat-1/exportfs resource-agents-4.0.1/heartbeat/exportfs
--- resource-agents-4.0.1/heartbeat-1/exportfs	2017-09-17 10:11:54.767495104 +0000
+++ resource-agents-4.0.1/heartbeat/exportfs	2017-09-17 10:14:53.407707107 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 # exportfs
 #
 # Description: Manages nfs exported file system.
diff -ur resource-agents-4.0.1/heartbeat-1/Filesystem resource-agents-4.0.1/heartbeat/Filesystem
--- resource-agents-4.0.1/heartbeat-1/Filesystem	2017-09-17 10:11:54.763068000 +0000
+++ resource-agents-4.0.1/heartbeat/Filesystem	2017-09-17 10:14:53.414663372 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Support:      users@clusterlabs.org
 # License:      GNU General Public License (GPL)
diff -ur resource-agents-4.0.1/heartbeat-1/findif.sh resource-agents-4.0.1/heartbeat/findif.sh
--- resource-agents-4.0.1/heartbeat-1/findif.sh	2017-09-17 10:11:54.767597014 +0000
+++ resource-agents-4.0.1/heartbeat/findif.sh	2017-09-17 10:14:53.416843792 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 ipcheck_ipv4() {
   local r1_to_255="([1-9][0-9]?|1[0-9][0-9]|2[0-4][0-9]|25[0-5])"
   local r0_to_255="([0-9][0-9]?|1[0-9][0-9]|2[0-4][0-9]|25[0-5])"
diff -ur resource-agents-4.0.1/heartbeat-1/galera resource-agents-4.0.1/heartbeat/galera
--- resource-agents-4.0.1/heartbeat-1/galera	2017-09-17 10:11:54.767781322 +0000
+++ resource-agents-4.0.1/heartbeat/galera	2017-09-17 10:14:53.425662059 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Copyright (c) 2014 David Vossel <davidvossel@gmail.com>
 #                    All Rights Reserved.
diff -ur resource-agents-4.0.1/heartbeat-1/garbd resource-agents-4.0.1/heartbeat/garbd
--- resource-agents-4.0.1/heartbeat-1/garbd	2017-09-17 10:11:54.767868818 +0000
+++ resource-agents-4.0.1/heartbeat/garbd	2017-09-17 10:14:53.429039449 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Copyright (c) 2015 Damien Ciabrini <dciabrin@redhat.com>
 #                    All Rights Reserved.
diff -ur resource-agents-4.0.1/heartbeat-1/ICP resource-agents-4.0.1/heartbeat/ICP
--- resource-agents-4.0.1/heartbeat-1/ICP	2017-09-17 10:11:54.763153957 +0000
+++ resource-agents-4.0.1/heartbeat/ICP	2017-09-17 10:14:53.432687317 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 
 # ICP
diff -ur resource-agents-4.0.1/heartbeat-1/ids resource-agents-4.0.1/heartbeat/ids
--- resource-agents-4.0.1/heartbeat-1/ids	2017-09-17 10:11:54.768261300 +0000
+++ resource-agents-4.0.1/heartbeat/ids	2017-09-17 10:14:53.438554664 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # ids
diff -ur resource-agents-4.0.1/heartbeat-1/iface-bridge resource-agents-4.0.1/heartbeat/iface-bridge
--- resource-agents-4.0.1/heartbeat-1/iface-bridge	2017-09-17 10:11:54.768364430 +0000
+++ resource-agents-4.0.1/heartbeat/iface-bridge	2017-09-17 10:14:53.445413069 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #       OCF Resource Agent compliant iface-bridge script.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/iface-vlan resource-agents-4.0.1/heartbeat/iface-vlan
--- resource-agents-4.0.1/heartbeat-1/iface-vlan	2017-09-17 10:11:54.768455715 +0000
+++ resource-agents-4.0.1/heartbeat/iface-vlan	2017-09-17 10:14:53.449261703 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #       OCF Resource Agent compliant iface-vlan script.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/IPaddr resource-agents-4.0.1/heartbeat/IPaddr
--- resource-agents-4.0.1/heartbeat-1/IPaddr	2017-09-17 10:11:54.763246974 +0000
+++ resource-agents-4.0.1/heartbeat/IPaddr	2017-09-17 10:14:53.456116236 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # License:      GNU General Public License (GPL) 
 # Support:      users@clusterlabs.org
diff -ur resource-agents-4.0.1/heartbeat-1/IPaddr2 resource-agents-4.0.1/heartbeat/IPaddr2
--- resource-agents-4.0.1/heartbeat-1/IPaddr2	2017-09-17 10:11:54.763367322 +0000
+++ resource-agents-4.0.1/heartbeat/IPaddr2	2017-09-17 10:14:53.465072745 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	$Id: IPaddr2.in,v 1.24 2006/08/09 13:01:54 lars Exp $
 #
diff -ur resource-agents-4.0.1/heartbeat-1/IPsrcaddr resource-agents-4.0.1/heartbeat/IPsrcaddr
--- resource-agents-4.0.1/heartbeat-1/IPsrcaddr	2017-09-17 10:11:54.763463981 +0000
+++ resource-agents-4.0.1/heartbeat/IPsrcaddr	2017-09-17 10:14:53.469029319 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	Description:	IPsrcaddr - Preferred source address modification
 #
diff -ur resource-agents-4.0.1/heartbeat-1/iscsi resource-agents-4.0.1/heartbeat/iscsi
--- resource-agents-4.0.1/heartbeat-1/iscsi	2017-09-17 10:11:54.768546043 +0000
+++ resource-agents-4.0.1/heartbeat/iscsi	2017-09-17 10:14:53.481509539 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # iSCSI OCF resource agent
 # Description: manage iSCSI disks (add/remove) using open-iscsi
diff -ur resource-agents-4.0.1/heartbeat-1/jboss resource-agents-4.0.1/heartbeat/jboss
--- resource-agents-4.0.1/heartbeat-1/jboss	2017-09-17 10:11:54.768635663 +0000
+++ resource-agents-4.0.1/heartbeat/jboss	2017-09-17 10:14:53.503528430 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Description:  Manages a Jboss Server as an OCF High-Availability
 #               resource under Heartbeat/LinuxHA control
diff -ur resource-agents-4.0.1/heartbeat-1/LinuxSCSI resource-agents-4.0.1/heartbeat/LinuxSCSI
--- resource-agents-4.0.1/heartbeat-1/LinuxSCSI	2017-09-17 10:11:54.763842695 +0000
+++ resource-agents-4.0.1/heartbeat/LinuxSCSI	2017-09-17 10:14:53.511682611 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # LinuxSCSI
diff -ur resource-agents-4.0.1/heartbeat-1/LVM resource-agents-4.0.1/heartbeat/LVM
--- resource-agents-4.0.1/heartbeat-1/LVM	2017-09-17 10:11:54.763740968 +0000
+++ resource-agents-4.0.1/heartbeat/LVM	2017-09-17 10:14:53.517337232 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 
 # LVM
diff -ur resource-agents-4.0.1/heartbeat-1/MailTo resource-agents-4.0.1/heartbeat/MailTo
--- resource-agents-4.0.1/heartbeat-1/MailTo	2017-09-17 10:11:54.763928881 +0000
+++ resource-agents-4.0.1/heartbeat/MailTo	2017-09-17 10:14:53.522007612 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for MailTo
 #
diff -ur resource-agents-4.0.1/heartbeat-1/mysql resource-agents-4.0.1/heartbeat/mysql
--- resource-agents-4.0.1/heartbeat-1/mysql	2017-09-17 10:11:54.769002684 +0000
+++ resource-agents-4.0.1/heartbeat/mysql	2017-09-17 10:14:53.546316688 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # MySQL
diff -ur resource-agents-4.0.1/heartbeat-1/mysql-common.sh resource-agents-4.0.1/heartbeat/mysql-common.sh
--- resource-agents-4.0.1/heartbeat-1/mysql-common.sh	2017-09-17 10:11:54.769091794 +0000
+++ resource-agents-4.0.1/heartbeat/mysql-common.sh	2017-09-17 10:14:53.548845125 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 
 #######################################################################
 
diff -ur resource-agents-4.0.1/heartbeat-1/mysql-proxy resource-agents-4.0.1/heartbeat/mysql-proxy
--- resource-agents-4.0.1/heartbeat-1/mysql-proxy	2017-09-17 10:11:54.769183757 +0000
+++ resource-agents-4.0.1/heartbeat/mysql-proxy	2017-09-17 10:14:53.554627095 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for MySQL Proxy
 #
diff -ur resource-agents-4.0.1/heartbeat-1/nagios resource-agents-4.0.1/heartbeat/nagios
--- resource-agents-4.0.1/heartbeat-1/nagios	2017-09-17 10:11:54.769278834 +0000
+++ resource-agents-4.0.1/heartbeat/nagios	2017-09-17 10:14:53.556689662 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #  License:      GNU General Public License (GPL)
 #  (c) 2015 T.J. Yang, O. Albrigtsen
diff -ur resource-agents-4.0.1/heartbeat-1/named resource-agents-4.0.1/heartbeat/named
--- resource-agents-4.0.1/heartbeat-1/named	2017-09-17 10:11:54.769373009 +0000
+++ resource-agents-4.0.1/heartbeat/named	2017-09-17 10:14:53.560604168 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Description:  Manages a named (Bind) server as an OCF High-Availability
 #               resource
diff -ur resource-agents-4.0.1/heartbeat-1/nfsserver resource-agents-4.0.1/heartbeat/nfsserver
--- resource-agents-4.0.1/heartbeat-1/nfsserver	2017-09-17 10:11:54.769550983 +0000
+++ resource-agents-4.0.1/heartbeat/nfsserver	2017-09-17 10:14:53.569863273 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 # nfsserver
 #
 # Description: Manages nfs server as OCF resource
diff -ur resource-agents-4.0.1/heartbeat-1/nginx resource-agents-4.0.1/heartbeat/nginx
--- resource-agents-4.0.1/heartbeat-1/nginx	2017-09-17 10:11:54.769735288 +0000
+++ resource-agents-4.0.1/heartbeat/nginx	2017-09-17 10:14:53.579195945 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	High-Availability nginx OCF resource agent
 # 
diff -ur resource-agents-4.0.1/heartbeat-1/oracle resource-agents-4.0.1/heartbeat/oracle
--- resource-agents-4.0.1/heartbeat-1/oracle	2017-09-17 10:11:54.770474803 +0000
+++ resource-agents-4.0.1/heartbeat/oracle	2017-09-17 10:14:53.600434025 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 
 # oracle
diff -ur resource-agents-4.0.1/heartbeat-1/oralsnr resource-agents-4.0.1/heartbeat/oralsnr
--- resource-agents-4.0.1/heartbeat-1/oralsnr	2017-09-17 10:11:54.770560587 +0000
+++ resource-agents-4.0.1/heartbeat/oralsnr	2017-09-17 10:14:53.602771104 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 
 # oralsnr
diff -ur resource-agents-4.0.1/heartbeat-1/pgagent resource-agents-4.0.1/heartbeat/pgagent
--- resource-agents-4.0.1/heartbeat-1/pgagent	2017-09-17 10:11:54.770643126 +0000
+++ resource-agents-4.0.1/heartbeat/pgagent	2017-09-17 10:14:53.604207846 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # High-Availability pgagent OCF resource agent
 #
diff -ur resource-agents-4.0.1/heartbeat-1/pgsql resource-agents-4.0.1/heartbeat/pgsql
--- resource-agents-4.0.1/heartbeat-1/pgsql	2017-09-17 10:11:54.770812974 +0000
+++ resource-agents-4.0.1/heartbeat/pgsql	2017-09-17 10:14:53.621515077 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Description:  Manages a PostgreSQL Server as an OCF High-Availability
 #               resource
diff -ur resource-agents-4.0.1/heartbeat-1/pingd resource-agents-4.0.1/heartbeat/pingd
--- resource-agents-4.0.1/heartbeat-1/pingd	2017-09-17 10:11:54.770908340 +0000
+++ resource-agents-4.0.1/heartbeat/pingd	2017-09-17 10:14:53.623875541 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	pingd OCF Resource Agent
diff -ur resource-agents-4.0.1/heartbeat-1/portblock resource-agents-4.0.1/heartbeat/portblock
--- resource-agents-4.0.1/heartbeat-1/portblock	2017-09-17 10:11:54.770998088 +0000
+++ resource-agents-4.0.1/heartbeat/portblock	2017-09-17 10:14:53.628406915 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # portblock: iptables temporary portblocking control 
 #
diff -ur resource-agents-4.0.1/heartbeat-1/postfix resource-agents-4.0.1/heartbeat/postfix
--- resource-agents-4.0.1/heartbeat-1/postfix	2017-09-17 10:11:54.771086316 +0000
+++ resource-agents-4.0.1/heartbeat/postfix	2017-09-17 10:14:53.631788253 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for Postfix
 #
diff -ur resource-agents-4.0.1/heartbeat-1/pound resource-agents-4.0.1/heartbeat/pound
--- resource-agents-4.0.1/heartbeat-1/pound	2017-09-17 10:11:54.771173586 +0000
+++ resource-agents-4.0.1/heartbeat/pound	2017-09-17 10:14:53.634584414 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # Pound
diff -ur resource-agents-4.0.1/heartbeat-1/proftpd resource-agents-4.0.1/heartbeat/proftpd
--- resource-agents-4.0.1/heartbeat-1/proftpd	2017-09-17 10:11:54.771266067 +0000
+++ resource-agents-4.0.1/heartbeat/proftpd	2017-09-17 10:14:53.637168453 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for Proftpd
 #
diff -ur resource-agents-4.0.1/heartbeat-1/Pure-FTPd resource-agents-4.0.1/heartbeat/Pure-FTPd
--- resource-agents-4.0.1/heartbeat-1/Pure-FTPd	2017-09-17 10:11:54.764266431 +0000
+++ resource-agents-4.0.1/heartbeat/Pure-FTPd	2017-09-17 10:14:53.639297187 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for Pure-FTPd
 #
diff -ur resource-agents-4.0.1/heartbeat-1/rabbitmq-cluster resource-agents-4.0.1/heartbeat/rabbitmq-cluster
--- resource-agents-4.0.1/heartbeat-1/rabbitmq-cluster	2017-09-17 10:11:54.771468845 +0000
+++ resource-agents-4.0.1/heartbeat/rabbitmq-cluster	2017-09-17 10:14:53.643978064 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Copyright (c) 2014 David Vossel <davidvossel@gmail.com>
 #                    All Rights Reserved.
diff -ur resource-agents-4.0.1/heartbeat-1/Raid1 resource-agents-4.0.1/heartbeat/Raid1
--- resource-agents-4.0.1/heartbeat-1/Raid1	2017-09-17 10:11:54.773192643 +0000
+++ resource-agents-4.0.1/heartbeat/Raid1	2017-09-17 10:14:53.648457196 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # License:      GNU General Public License (GPL)
diff -ur resource-agents-4.0.1/heartbeat-1/Route resource-agents-4.0.1/heartbeat/Route
--- resource-agents-4.0.1/heartbeat-1/Route	2017-09-17 10:11:54.764615939 +0000
+++ resource-agents-4.0.1/heartbeat/Route	2017-09-17 10:14:53.662160351 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	Route OCF RA. Enables and disables network routes.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/rsyncd resource-agents-4.0.1/heartbeat/rsyncd
--- resource-agents-4.0.1/heartbeat-1/rsyncd	2017-09-17 10:11:54.771646348 +0000
+++ resource-agents-4.0.1/heartbeat/rsyncd	2017-09-17 10:14:53.664399823 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for rsync daemon
 #
diff -ur resource-agents-4.0.1/heartbeat-1/SAPDatabase resource-agents-4.0.1/heartbeat/SAPDatabase
--- resource-agents-4.0.1/heartbeat-1/SAPDatabase	2017-09-17 10:11:54.764704737 +0000
+++ resource-agents-4.0.1/heartbeat/SAPDatabase	2017-09-17 10:14:53.669530934 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # SAPDatabase
 #
diff -ur resource-agents-4.0.1/heartbeat-1/SAPInstance resource-agents-4.0.1/heartbeat/SAPInstance
--- resource-agents-4.0.1/heartbeat-1/SAPInstance	2017-09-17 10:11:54.764815156 +0000
+++ resource-agents-4.0.1/heartbeat/SAPInstance	2017-09-17 10:14:53.686042069 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # SAPInstance
 #
diff -ur resource-agents-4.0.1/heartbeat-1/scsi2reservation resource-agents-4.0.1/heartbeat/scsi2reservation
--- resource-agents-4.0.1/heartbeat-1/scsi2reservation	2017-09-17 10:11:54.772002539 +0000
+++ resource-agents-4.0.1/heartbeat/scsi2reservation	2017-09-17 10:14:53.687557564 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 # by hxinwei@gmail.com
 # License: GNU General Public License 2 (GPL2)
 
diff -ur resource-agents-4.0.1/heartbeat-1/SendArp resource-agents-4.0.1/heartbeat/SendArp
--- resource-agents-4.0.1/heartbeat-1/SendArp	2017-09-17 10:11:54.764905243 +0000
+++ resource-agents-4.0.1/heartbeat/SendArp	2017-09-17 10:14:53.690995870 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	Copyright (c) 2006, Huang Zhen <zhen.huang@gmail.com>
diff -ur resource-agents-4.0.1/heartbeat-1/ServeRAID resource-agents-4.0.1/heartbeat/ServeRAID
--- resource-agents-4.0.1/heartbeat-1/ServeRAID	2017-09-17 10:11:54.764989827 +0000
+++ resource-agents-4.0.1/heartbeat/ServeRAID	2017-09-17 10:14:53.694388521 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # ServeRAID
diff -ur resource-agents-4.0.1/heartbeat-1/sfex resource-agents-4.0.1/heartbeat/sfex
--- resource-agents-4.0.1/heartbeat-1/sfex	2017-09-17 10:11:54.772170308 +0000
+++ resource-agents-4.0.1/heartbeat/sfex	2017-09-17 10:14:53.696876570 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #       Shared Disk File EXclusiveness (SF-EX) OCF RA.
 #       prevent a destruction of data on shared disk file system
diff -ur resource-agents-4.0.1/heartbeat-1/SphinxSearchDaemon resource-agents-4.0.1/heartbeat/SphinxSearchDaemon
--- resource-agents-4.0.1/heartbeat-1/SphinxSearchDaemon	2017-09-17 10:11:54.765076588 +0000
+++ resource-agents-4.0.1/heartbeat/SphinxSearchDaemon	2017-09-17 10:14:53.709822880 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	Searchd OCF RA. 
diff -ur resource-agents-4.0.1/heartbeat-1/Stateful resource-agents-4.0.1/heartbeat/Stateful
--- resource-agents-4.0.1/heartbeat-1/Stateful	2017-09-17 10:11:54.765254403 +0000
+++ resource-agents-4.0.1/heartbeat/Stateful	2017-09-17 10:14:53.715018141 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #	Example of a stateful OCF Resource Agent.
diff -ur resource-agents-4.0.1/heartbeat-1/symlink resource-agents-4.0.1/heartbeat/symlink
--- resource-agents-4.0.1/heartbeat-1/symlink	2017-09-17 10:11:54.772524712 +0000
+++ resource-agents-4.0.1/heartbeat/symlink	2017-09-17 10:14:53.717145808 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #   An OCF RA that manages a symlink
diff -ur resource-agents-4.0.1/heartbeat-1/tomcat resource-agents-4.0.1/heartbeat/tomcat
--- resource-agents-4.0.1/heartbeat-1/tomcat	2017-09-17 10:11:54.772726796 +0000
+++ resource-agents-4.0.1/heartbeat/tomcat	2017-09-17 10:14:53.729218595 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Description:  Manages a Tomcat Server as an OCF High-Availability
 #               resource under Heartbeat/LinuxHA control
diff -ur resource-agents-4.0.1/heartbeat-1/varnish resource-agents-4.0.1/heartbeat/varnish
--- resource-agents-4.0.1/heartbeat-1/varnish	2017-09-17 10:11:54.772846388 +0000
+++ resource-agents-4.0.1/heartbeat/varnish	2017-09-17 10:14:53.732976937 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # Varnish
diff -ur resource-agents-4.0.1/heartbeat-1/VIPArip resource-agents-4.0.1/heartbeat/VIPArip
--- resource-agents-4.0.1/heartbeat-1/VIPArip	2017-09-17 10:11:54.765426428 +0000
+++ resource-agents-4.0.1/heartbeat/VIPArip	2017-09-17 10:14:53.735499295 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 	License:	GNU General Public License (GPL)
 # 	Support:	users@clusterlabs.org
diff -ur resource-agents-4.0.1/heartbeat-1/VirtualDomain resource-agents-4.0.1/heartbeat/VirtualDomain
--- resource-agents-4.0.1/heartbeat-1/VirtualDomain	2017-09-17 10:11:54.765521916 +0000
+++ resource-agents-4.0.1/heartbeat/VirtualDomain	2017-09-17 10:14:53.742895961 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Support:	  users@clusterlabs.org
 # License:	  GNU General Public License (GPL)
diff -ur resource-agents-4.0.1/heartbeat-1/vmware resource-agents-4.0.1/heartbeat/vmware
--- resource-agents-4.0.1/heartbeat-1/vmware	2017-09-17 10:11:54.772934581 +0000
+++ resource-agents-4.0.1/heartbeat/vmware	2017-09-17 10:14:53.746489004 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # VMware OCF resource agent
 #
diff -ur resource-agents-4.0.1/heartbeat-1/WAS resource-agents-4.0.1/heartbeat/WAS
--- resource-agents-4.0.1/heartbeat-1/WAS	2017-09-17 10:11:54.765609813 +0000
+++ resource-agents-4.0.1/heartbeat/WAS	2017-09-17 10:14:53.753167336 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # 
 # WAS
diff -ur resource-agents-4.0.1/heartbeat-1/WAS6 resource-agents-4.0.1/heartbeat/WAS6
--- resource-agents-4.0.1/heartbeat-1/WAS6	2017-09-17 10:11:54.765708025 +0000
+++ resource-agents-4.0.1/heartbeat/WAS6	2017-09-17 10:14:53.757523651 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 # WAS6 
 #
 # Description:  Manages a Websphere Application Server as an HA resource
diff -ur resource-agents-4.0.1/heartbeat-1/WinPopup resource-agents-4.0.1/heartbeat/WinPopup
--- resource-agents-4.0.1/heartbeat-1/WinPopup	2017-09-17 10:11:54.765793682 +0000
+++ resource-agents-4.0.1/heartbeat/WinPopup	2017-09-17 10:14:53.759469383 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Resource script for sending WinPopups using smbclient
 # derived from Alan Robertson's MailTo script
diff -ur resource-agents-4.0.1/heartbeat-1/Xen resource-agents-4.0.1/heartbeat/Xen
--- resource-agents-4.0.1/heartbeat-1/Xen	2017-09-17 10:11:54.765890614 +0000
+++ resource-agents-4.0.1/heartbeat/Xen	2017-09-17 10:14:53.764507696 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 # Support:      users@clusterlabs.org
diff -ur resource-agents-4.0.1/heartbeat-1/Xinetd resource-agents-4.0.1/heartbeat/Xinetd
--- resource-agents-4.0.1/heartbeat-1/Xinetd	2017-09-17 10:11:54.765978898 +0000
+++ resource-agents-4.0.1/heartbeat/Xinetd	2017-09-17 10:14:53.766614452 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 # Startup/shutdown script for services managed by xinetd.
 #
diff -ur resource-agents-4.0.1/heartbeat-1/zabbixserver resource-agents-4.0.1/heartbeat/zabbixserver
--- resource-agents-4.0.1/heartbeat-1/zabbixserver	2017-09-17 10:11:54.773102424 +0000
+++ resource-agents-4.0.1/heartbeat/zabbixserver	2017-09-17 10:14:53.769232644 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #
 #   zabbixserver OCF RA for zabbix_server daemon
diff -ur resource-agents-3.9.7-1/tools/ocf-tester.in resource-agents-3.9.7/tools/ocf-tester.in
--- resource-agents-4.0.1/tools/ocf-tester.in.~1~	2017-02-02 11:51:34.000000000 +0000
+++ resource-agents-4.0.1/tools/ocf-tester.in	2017-09-17 10:08:27.656506760 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #	$Id: ocf-tester,v 1.2 2006/08/14 09:38:20 andrew Exp $
 #
@@ -216,7 +216,7 @@
 test_permissions() {
     action=meta-data
     debug ${1:-"Testing permissions with uid nobody"}
-    su nobody -s /bin/sh -c "$agent $action" > /dev/null
+    su nobody -s /usr/bin/bash -c "$agent $action" > /dev/null
 }
 
 test_metadata() {
