https://www.illumos.org/issues/5751

--- resource-agents-4.1.0/tools/send_arp.libnet.c.orig
+++ resource-agents-4.1.0/tools/send_arp.libnet.c
@@ -47,6 +47,9 @@
 #include <clplumbing/cl_signal.h>
 #include <clplumbing/cl_log.h>
 
+typedef uint8_t u_int8_t;
+typedef uint32_t u_int32_t;
+
 #ifdef HAVE_LIBNET_1_0_API
 #	define	LTYPE	struct libnet_link_int
 	static u_char *mk_packet(u_int32_t ip, u_char *device, u_char *macaddr, u_char *broadcast, u_char *netmask, u_short arptype);
