--- toolame-02l/Makefile.~1~	2003-03-02 10:07:02.000000000 +0300
+++ toolame-02l/Makefile	2015-07-17 22:27:18.338939134 +0300
@@ -1,5 +1,5 @@
 
-CC = gcc
+# CC = gcc
 
 c_sources = \
 	common.c \
@@ -26,41 +26,20 @@
 
 OBJ = $(c_sources:.c=.o)
 
-#Uncomment this if you want to do some profiling/debugging
-#PG = -g -pg
-PG = -fomit-frame-pointer
-
-# Optimize flag. 3 is about as high as you can sanely go with GCC3.2.
-OPTIM = -O3
+ifeq ($(CC),gcc)
+   INLINE = inline
+endif
 
 # These flags are pretty much mandatory
-REQUIRED = -DNDEBUG -DINLINE=inline
-
-#pick your architecture
-ARCH = -march=pentium
-#Possible x86 architectures
-#gcc3.2 => i386, i486, i586, i686, pentium, pentium-mmx
-#          pentiumpro, pentium2, pentium3, pentium4, k6, k6-2, k6-3,
-#          athlon, athlon-tbird, athlon-4, athlon-xp and athlon-mp.
-
-#TWEAK the hell out of the compile. Some of these are real dodgy
-# and will cause program instability
-#TWEAKS = -finline-functions -fexpensive-optimizations -ffast-math \
-#	-malign-double \
-#	-mfancy-math-387 -funroll-loops -funroll-all-loops -pipe \
-#	-fschedule-insns2 -fno-strength-reduce
-
-#Set a stack of warnings to overcome my atrocious coding style . MFC.
-WARNINGS = -Wall
-WARNINGS2 = -Wstrict-prototypes -Wmissing-prototypes -Wunused -Wunused-function -Wunused-label -Wunused-parameter -Wunused-variable -Wunused-value -Wredundant-decls
+REQUIRED = -DNDEBUG -DINLINE=$(INLINE)
 
 NEW_02L_FIXES = -DNEWENCODE -DNEWATAN
 
-CC_SWITCHES = $(OPTIM) $(REQUIRED) $(ARCH) $(PG) $(TWEAKS) $(WARNINGS) $(NEW_02L_FIXES)
+CC_SWITCHES = $(CFLAGS) $(REQUIRED) $(NEW_02L_FIXES)
 
 PGM = toolame
 
-LIBS =  -lm
+LIBS =  $(LDFLAGS) -lm
 
 #nick burch's OS/2 fix  gagravarr@SoftHome.net
 UNAME = $(shell uname)
@@ -75,7 +54,7 @@
 	$(CC) $(CC_SWITCHES) -c $< -o $@
 
 $(PGM):	$(OBJ) Makefile
-	$(CC) $(PG) -o $(PGM) $(OBJ) $(LIBS)
+	$(CC) $(CC_SWITCHES) -o $(PGM) $(OBJ) $(LIBS)
 
 clean:
 	-rm $(OBJ) $(DEP)
@@ -91,3 +70,6 @@
 TAGS: ${c_sources}
 	etags -T ${c_sources}
 
+install:
+	install -d $(DESTDIR)/usr/bin
+	install -m 555 $(PGM) $(DESTDIR)/usr/bin
