--- sslh-v1.17/Makefile	2015-03-10 09:51:39.000000000 +1300
+++ tmp/sslh-v1.17/Makefile	2016-02-09 12:27:45.680775713 +1300
@@ -5,7 +5,7 @@
 USELIBWRAP?=	# Use libwrap?
 USELIBCAP=	# Use libcap?
 COV_TEST= 	# Perform test coverage?
-PREFIX=/usr/local
+PREFIX=/usr
 
 MAN=sslh.8.gz	# man page name
 
@@ -59,7 +59,7 @@
 	$(CC) $(CFLAGS) $(LDFLAGS) -o echosrv echosrv.o probe.o common.o $(LIBS)
 
 $(MAN): sslh.pod Makefile
-	pod2man --section=8 --release=$(VERSION) --center=" " sslh.pod | gzip -9 - > $(MAN)
+	/usr/perl5/5.22/bin/pod2man --section=8 --release=$(VERSION) --center=" " sslh.pod | gzip -9 - > $(MAN)
 
 # Create release: export clean tree and tag current
 # configuration
