--- libXt-1.1.5.orig/src/Makefile.am	Mon Jan 16 10:15:06 2017
+++ libXt-1.1.5/src/Makefile.am	Mon Jan 16 10:18:01 2017
@@ -74,7 +74,8 @@
 
 libXt_la_LIBADD = @XT_LIBS@
 
-libXt_la_LDFLAGS = -version-number 6:0:0 -no-undefined
+libXt_la_LDFLAGS = -version-number 6:0:0 -no-undefined \
+	-Wl,-M$(top_srcdir)/../mapfile-vers -lc
 
 #
 # The util directory contains a program that builds some of the sources.
