diff -ruN totem-pl-parser-2.30.0.orig/configure.in totem-pl-parser-2.30.0/configure.in
--- totem-pl-parser-2.30.0.orig/configure.in	2010-04-12 19:09:41.297986410 +0100
+++ totem-pl-parser-2.30.0/configure.in	2010-04-12 19:10:46.593613814 +0100
@@ -71,8 +71,8 @@
 
 if test "x$enable_gmime" = "xyes" ; then
    PKG_CHECK_MODULES(GMIME,  
-		     gmime-2.4,
-                     [have_gmime=yes GMIME=gmime-2.4], 
+		     gmime-2.6,
+                     [have_gmime=yes GMIME=gmime-2.6], 
                      [have_gmime=no])
    if test "x$have_gmime" != "xyes" ; then
      PKG_CHECK_MODULES(GMIME,  
