--- totem-pl-parser-2.26.0/plparse/totem-pl-parser.c.old	2009-03-22 16:21:17.866499000 +0800
+++ totem-pl-parser-2.26.0/plparse/totem-pl-parser.c	2009-03-22 16:23:05.474296000 +0800
@@ -1448,6 +1448,7 @@
 
 	if (g_file_has_uri_scheme (file, "mms") != FALSE
 			|| g_file_has_uri_scheme (file, "rtsp") != FALSE
+			|| g_file_has_uri_scheme (file, "cdda") != FALSE
 			|| g_file_has_uri_scheme (file, "rtmp") != FALSE
 			|| g_file_has_uri_scheme (file, "icy") != FALSE
 			|| g_file_has_uri_scheme (file, "pnm") != FALSE) {
