--- foomatic-filters-4.0.17/foomaticrip.c.orig	2020-05-24 07:33:29.073672991 +0000
+++ foomatic-filters-4.0.17/foomaticrip.c	2020-05-24 07:37:21.405068026 +0000
@@ -109,7 +109,7 @@
 }
 
 
-dstr_t *postpipe;  /* command into which the output of this filter should be piped */
+extern dstr_t *postpipe;  /* command into which the output of this filter should be piped */
 FILE *postpipe_fh = NULL;
 
 FILE * open_postpipe()
--- foomatic-filters-4.0.17/options.c.orig	2020-05-24 07:32:12.094682850 +0000
+++ foomatic-filters-4.0.17/options.c	2020-05-24 07:37:29.885449811 +0000
@@ -38,7 +38,7 @@
 } icc_mapping_entry_t;
 
 /* Values from foomatic keywords in the ppd file */
-char printer_model [256];
+extern char printer_model [256];
 char printer_id [256];
 char driver [128];
 char cmd [4096];
@@ -46,7 +46,7 @@
 dstr_t *postpipe = NULL;  /* command into which the output of this
                              filter should be piped */
 int ps_accounting = 1;
-char cupsfilter [256];
+extern char cupsfilter [256];
 int jobentitymaxlen = 0;
 int userentitymaxlen = 0;
 int hostentitymaxlen = 0;
