--- gnome-menus-2.30.4/libmenu/Makefile.am-orig	2010-10-20 20:16:07.075966453 -0500
+++ gnome-menus-2.30.4/libmenu/Makefile.am	2010-10-20 20:16:13.665158059 -0500
@@ -51,7 +51,7 @@ CLEANFILES =
 # Introspection
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
-INTROSPECTION_SCANNER_ARGS = --warn-all --add-include-path=$(srcdir)
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir)
 INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 
 if HAVE_INTROSPECTION
