--- gnome-menus-2.30.5/Makefile.am	2010-09-20 19:44:39.000000000 +0400
+++ gnome-menus-2.30.5/Makefile.am.1	2016-06-28 11:07:25.074471959 +0300
@@ -1,7 +1,7 @@
 SUBDIRS = libmenu desktop-directories layout util po
 
 if HAVE_PYTHON
-SUBDIRS += python simple-editor
+SUBDIRS += python-2.6 python simple-editor
 endif
 
 ACLOCAL_AMFLAGS = -I m4
--- gnome-menus-2.30.5/python-2.6/Makefile.am.1	2016-06-28 11:13:14.067079437 +0300
+++ gnome-menus-2.30.5/python-2.6/Makefile.am	2016-06-28 11:14:46.554760826 +0300
@@ -1,6 +1,6 @@
 AM_CPPFLAGS =				\
 	$(GLIB_CFLAGS)			\
-	$(PYTHON_INCLUDES)		\
+	-I/usr/include/python2.6		\
 	-I$(srcdir)/../libmenu		\
 	-DGMENU_I_KNOW_THIS_IS_UNSTABLE	\
 	$(DISABLE_DEPRECATED_CFLAGS)	\
@@ -10,7 +10,7 @@
 	$(WARN_CFLAGS)			\
 	$(NO_STRICT_ALIASING_CFLAGS)
 
-gmenudir = $(pyexecdir)
+gmenudir = /usr/lib/python2.6/vendor-packages
 gmenu_LTLIBRARIES = gmenu.la
 
 gmenu_la_SOURCES = gmenu.c
 
--- gnome-menus-2.30.5/configure.ac.~1~	2010-11-17 14:10:04.000000000 +0300
+++ gnome-menus-2.30.5/configure.ac	2016-06-28 11:28:12.775963235 +0300
@@ -140,6 +140,7 @@
 desktop-directories/Makefile
 util/Makefile
 python/Makefile
+python-2.6/Makefile
 simple-editor/Makefile
 simple-editor/GMenuSimpleEditor/Makefile
 po/Makefile.in
