--- src/libafs/MakefileProto.SOLARIS.in.orig	2013-11-21 16:50:55.910244871 +0100
+++ src/libafs/MakefileProto.SOLARIS.in	2013-11-21 16:52:06.569253338 +0100
@@ -55,9 +55,9 @@
 # Name of directory to hold object files and libraries.
 <all -sun4x_57 -sun4x_58 -sun4x_59 -sun4x_510 -sun4x_511 -sunx86_57 -sunx86_58 -sunx86_59 -sunx86_510 -sunx86_511>
 KOBJ = MODLOAD
-<sun4x_57 sun4x_58 sun4x_59 sunx86_57 sunx86_58 sunx86_59 sunx86_510>
+<sun4x_57 sun4x_58 sun4x_59 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511>
 KOBJ = MODLOAD32 MODLOAD64
-<sun4x_510 sun4x_511 sunx86_511>
+<sun4x_510 sun4x_511>
 KOBJ = MODLOAD64
 
 # This tells Makefile.common to use it's single directory build target.
