diff -ruN evolution-data-server-2.29.5.orig/docs/reference/addressbook/libebook/Makefile.am evolution-data-server-2.29.5/docs/reference/addressbook/libebook/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/addressbook/libebook/Makefile.am	2010-01-16 22:13:48.636472917 +0000
+++ evolution-data-server-2.29.5/docs/reference/addressbook/libebook/Makefile.am	2010-01-16 22:14:34.670083380 +0000
@@ -36,3 +36,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/addressbook/libedata-book/Makefile.am evolution-data-server-2.29.5/docs/reference/addressbook/libedata-book/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/addressbook/libedata-book/Makefile.am	2010-01-16 22:20:23.073897325 +0000
+++ evolution-data-server-2.29.5/docs/reference/addressbook/libedata-book/Makefile.am	2010-01-16 22:20:42.007358062 +0000
@@ -37,3 +37,5 @@
 
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/calendar/libecal/Makefile.am evolution-data-server-2.29.5/docs/reference/calendar/libecal/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/calendar/libecal/Makefile.am	2010-01-16 22:38:17.910205227 +0000
+++ evolution-data-server-2.29.5/docs/reference/calendar/libecal/Makefile.am	2010-01-16 22:38:39.425697727 +0000
@@ -37,3 +37,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/calendar/libedata-cal/Makefile.am evolution-data-server-2.29.5/docs/reference/calendar/libedata-cal/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/calendar/libedata-cal/Makefile.am	2010-01-16 22:45:35.860167899 +0000
+++ evolution-data-server-2.29.5/docs/reference/calendar/libedata-cal/Makefile.am	2010-01-16 22:45:56.857233280 +0000
@@ -43,3 +43,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/camel/Makefile.am evolution-data-server-2.29.5/docs/reference/camel/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/camel/Makefile.am	2010-01-16 22:57:03.432041277 +0000
+++ evolution-data-server-2.29.5/docs/reference/camel/Makefile.am	2010-01-16 22:57:19.342605903 +0000
@@ -75,3 +75,5 @@
 	camel.hierarchy
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/libedataserver/Makefile.am evolution-data-server-2.29.5/docs/reference/libedataserver/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/libedataserver/Makefile.am	2010-01-16 23:08:29.729108129 +0000
+++ evolution-data-server-2.29.5/docs/reference/libedataserver/Makefile.am	2010-01-16 23:08:45.407237634 +0000
@@ -30,3 +30,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/libedataserverui/Makefile.am evolution-data-server-2.29.5/docs/reference/libedataserverui/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/libedataserverui/Makefile.am	2010-01-16 23:19:58.856783683 +0000
+++ evolution-data-server-2.29.5/docs/reference/libedataserverui/Makefile.am	2010-01-16 23:20:14.635064995 +0000
@@ -36,3 +36,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/Makefile.am evolution-data-server-2.29.5/docs/reference/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/Makefile.am	2010-01-16 23:27:45.027480888 +0000
+++ evolution-data-server-2.29.5/docs/reference/Makefile.am	2010-01-16 23:27:54.257528834 +0000
@@ -3,3 +3,5 @@
 
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
diff -ruN evolution-data-server-2.29.5.orig/docs/reference/libebackend/Makefile.am evolution-data-server-2.29.5/docs/reference/libebackend/Makefile.am
--- evolution-data-server-2.29.5.orig/docs/reference/libebackend/Makefile.am	2010-01-16 23:38:21.088828647 +0000
+++ evolution-data-server-2.29.5/docs/reference/libebackend/Makefile.am	2010-01-16 23:38:39.754640877 +0000
@@ -30,3 +30,5 @@
 include $(top_srcdir)/gtk-doc.make
 
 -include $(top_srcdir)/git.mk
+
+FIXXREF_OPTIONS =--module=$(DOC_MODULE)
