--- evolution-data-server-2.32.3/camel/camel-folder.c.~1~	2015-01-11 12:17:16.854990098 +0300
+++ evolution-data-server-2.32.3/camel/camel-folder.c	2015-01-11 12:22:21.021230025 +0300
@@ -819,23 +819,25 @@
 	}
 	camel_folder_unlock (folder, CAMEL_FOLDER_CHANGE_LOCK);
 
-	if (session->junk_plugin && info->uid_changed->len) {
+	if (info->uid_changed->len) {
 		guint32 flags;
 
 		for (i = 0; i < info->uid_changed->len; i++) {
 			flags = camel_folder_get_message_flags (folder, info->uid_changed->pdata[i]);
-			if (flags & CAMEL_MESSAGE_JUNK_LEARN) {
-				if (flags & CAMEL_MESSAGE_JUNK) {
-					if (!junk)
-						junk = g_ptr_array_new ();
-					g_ptr_array_add (junk, g_strdup (info->uid_changed->pdata[i]));
-				} else {
-					if (!notjunk)
-						notjunk = g_ptr_array_new ();
-					g_ptr_array_add (notjunk, g_strdup (info->uid_changed->pdata[i]));
+			if (session->junk_plugin) {
+				if (flags & CAMEL_MESSAGE_JUNK_LEARN) {
+					if (flags & CAMEL_MESSAGE_JUNK) {
+						if (!junk)
+							junk = g_ptr_array_new ();
+						g_ptr_array_add (junk, g_strdup (info->uid_changed->pdata[i]));
+					} else {
+						if (!notjunk)
+							notjunk = g_ptr_array_new ();
+						g_ptr_array_add (notjunk, g_strdup (info->uid_changed->pdata[i]));
+					}
+					/* reset junk learn flag so that we don't process it again*/
+					camel_folder_set_message_flags (folder, info->uid_changed->pdata[i], CAMEL_MESSAGE_JUNK_LEARN, 0);
 				}
-				/* reset junk learn flag so that we don't process it again*/
-				camel_folder_set_message_flags (folder, info->uid_changed->pdata[i], CAMEL_MESSAGE_JUNK_LEARN, 0);
 			}
 		}
 	}
