--- evolution-data-server-2.30.3/camel/camel-gpg-context.c.orig	2011-05-16 12:16:15.858706245 +0800
+++ evolution-data-server-2.30.3/camel/camel-gpg-context.c	2011-05-16 12:16:42.627645629 +0800
@@ -638,7 +638,7 @@
 		}
 
 		/* run gpg */
-		execvp ("gpg", (gchar **) argv->pdata);
+		execvp ("gpg2", (gchar **) argv->pdata);
 		_exit (255);
 	} else if (gpg->pid < 0) {
 		g_ptr_array_free (argv, TRUE);
