https://github.com/wichert/lingua/issues/120

--- lingua-4.16.1/pyproject.toml.orig
+++ lingua-4.16.1/pyproject.toml
@@ -8,7 +8,7 @@
 version = "4.16.1"
 authors = [{ name = "Wichert Akkerman", email = "wichert@wiggy.net" }]
 readme = "README.rst"
-license = { file = "LICENSE" }
+license-files = [ "LICENSE" ]
 keywords = ["translation", "po", "gettext", "Babel"]
 classifiers = [
     "Intended Audience :: Developers",
