From a4c0c27410de75e92970b4eecfad59a8eaa57eb6 Mon Sep 17 00:00:00 2001
From: LeeYoung624 <liyiang@hotmail.com>
Date: Tue, 11 Jun 2019 11:29:18 +0800
Subject: [PATCH] Install zstdless & zstdgrep as 'PROGRAMS' in CMake

---
 build/cmake/programs/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/cmake/programs/CMakeLists.txt b/build/cmake/programs/CMakeLists.txt
index f6f7a3616..03194fa04 100644
--- a/build/cmake/programs/CMakeLists.txt
+++ b/build/cmake/programs/CMakeLists.txt
@@ -38,6 +38,8 @@ if (UNIX)
     add_custom_target(unzstd ALL ${CMAKE_COMMAND} -E create_symlink zstd unzstd DEPENDS zstd COMMENT "Creating unzstd symlink")
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/zstdcat DESTINATION "bin")
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unzstd DESTINATION "bin")
+    install(PROGRAMS ${PROGRAMS_DIR}/zstdgrep DESTINATION "bin")
+    install(PROGRAMS ${PROGRAMS_DIR}/zstdless DESTINATION "bin")
 
     add_custom_target(zstd.1 ALL
         ${CMAKE_COMMAND} -E copy ${PROGRAMS_DIR}/zstd.1 .
