--- zziplib-0.13.69/configure.ac	2018-03-17 13:46:45.000000000 +0000
+++ zziplib-0.13.69/configure.ac	2018-08-03 07:09:25.865903598 +0000
@@ -90,7 +90,7 @@ AX_PAX_TAR_EXTRACT
 AC_PATH_PROGS(PERL, perl5 perl, echo no perl found for)
 AC_PATH_PROGS(PYTHON, python, echo no python found for)
 AC_PATH_PROGS(MKZIP, zip pkzip, :)
-AC_PATH_PROGS(XMLTO, xmlto, :)
+AC_PATH_PROGS(XMLTO, :, :)
 
 AC_C_INLINE
 AC_C_CONST
