--- mailutils-3.18/libmailutils/base/fdwait.c.orig	2025-01-02 14:36:29.000000000 +0100
+++ mailutils-3.18/libmailutils/base/fdwait.c	2025-01-04 12:54:43.304398966 +0100
@@ -23,6 +23,7 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <string.h>
 #include <errno.h>
 #include <mailutils/stream.h>
 
--- mailutils-3.18/libmailutils/server/server.c.orig	2025-01-02 14:36:33.000000000 +0100
+++ mailutils-3.18/libmailutils/server/server.c	2025-01-04 12:57:59.475276788 +0100
@@ -24,6 +24,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <mailutils/server.h>
 #include <mailutils/errno.h>
 #include <mailutils/acl.h>
