--- hotspot/src/share/vm/utilities/hashtable.cpp.~1~	2014-01-06 21:44:28.173732825 +0400
+++ hotspot/src/share/vm/utilities/hashtable.cpp	2014-01-06 21:44:45.899087479 +0400
@@ -318,7 +318,7 @@
 template class Hashtable<Symbol*, mtSymbol>;
 template class Hashtable<klassOop, mtClass>;
 template class Hashtable<oop, mtClass>;
-#ifdef SOLARIS
+#ifdef NOTSOLARIS
 template class Hashtable<oop, mtSymbol>;
 #endif
 template class Hashtable<oopDesc*, mtSymbol>;
