--- ./configure.ac.orig	2009-05-16 22:38:55.926956000 +0800
+++ ./configure.ac	2009-05-16 22:39:26.689492000 +0800
@@ -15,7 +15,7 @@
 AC_HEADER_STDC
 AM_PROG_LIBTOOL
 
-CFLAGS="${CFLAGS} -Wall"
+CFLAGS="${CFLAGS}"
 
 GNOME_MAINTAINER_MODE_DEFINES
 GNOME_COMPILE_WARNINGS
@@ -68,7 +68,7 @@
 CAIRO_REQUIRED=1.4.0
 DBUS_REQUIRED=1.0
 DBUS_GLIB_REQUIRED=0.7
-HAL_REQUIRED=0.5.9
+HAL_REQUIRED=0.5.8
 PANGOCAIRO_REQUIRED=1.18.0
 LIBRSVG_REQUIRED=2.18.0
 
--- ./src/cheese-webcam.c.orig	2009-05-19 04:46:03.000000000 +0800
+++ ./src/cheese-webcam.c	2009-06-30 11:47:32.067516000 +0800
@@ -37,7 +37,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
+#include <sys/videodev2.h>
 
 #include "cheese-webcam.h"
 
diff -ruN cheese-2.27.4.orig/src/cheese-webcam.c cheese-2.27.4/src/cheese-webcam.c
--- cheese-2.27.4.orig/src/cheese-webcam.c	2009-07-19 21:48:47.753253823 +0100
+++ cheese-2.27.4/src/cheese-webcam.c	2009-07-19 21:51:57.393929931 +0100
@@ -335,7 +335,7 @@
     char                   *subsystem  = NULL;
     char                   *gstreamer_src, *product_name;
     struct v4l2_capability  v2cap;
-    struct video_capability v1cap;
+    //struct video_capability v1cap;
     gint                    vendor_id     = 0;
     gint                    product_id    = 0;
     gchar                  *property_name = NULL;
@@ -398,23 +398,7 @@
       continue;
     }
     ok = ioctl (fd, VIDIOC_QUERYCAP, &v2cap);
-    if (ok < 0)
-    {
-      ok = ioctl (fd, VIDIOCGCAP, &v1cap);
-      if (ok < 0)
-      {
-        g_warning ("Error while probing v4l capabilities for %s: %s",
-                   device, strerror (errno));
-        libhal_free_string (device);
-        close (fd);
-        continue;
-      }
-      g_print ("Detected v4l device: %s\n", v1cap.name);
-      g_print ("Device type: %d\n", v1cap.type);
-      gstreamer_src = "v4lsrc";
-      product_name  = v1cap.name;
-    }
-    else
+    if (ok >= 0)
     {
       guint cap = v2cap.capabilities;
       g_print ("Detected v4l2 device: %s\n", v2cap.card);
