--- libgnome-2.32.1/libgnome/Makefile.am.~1~	2015-09-22 11:37:01.578687667 +0300
+++ libgnome-2.32.1/libgnome/Makefile.am	2015-09-22 11:37:14.033765597 +0300
@@ -17,7 +17,6 @@
 	-I$(srcdir)/..						\
 	$(WARN_CFLAGS)						\
 	$(LIBGNOME_CFLAGS)					\
-	-DG_DISABLE_DEPRECATED					\
 	-DLIBGNOME_PREFIX=\""$(prefix)"\"			\
 	-DLIBGNOME_LIBDIR=\""$(libdir)"\"			\
 	-DLIBGNOME_DATADIR=\""$(datadir)"\"			\
