diff -rup libgnome-2.22.0/schemas/desktop_gnome_background.schemas.in.in libgnome-2.22.0-indiana/schemas/desktop_gnome_background.schemas.in.in
--- libgnome-2.22.0/schemas/desktop_gnome_background.schemas.in.in	2008-01-08 10:04:30.000000000 +0100
+++ libgnome-2.22.0-indiana/schemas/desktop_gnome_background.schemas.in.in	2008-06-01 10:20:08.608787124 +0200
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default>@datadir@/pixmaps/backgrounds/gnome/background-default.jpg</default>
+	<default>@datadir@/pixmaps/backgrounds/openindiana/openindiana-default.jpg</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image.</long>
diff -urN libgnome-2.20.1.1/schemas/desktop_gnome_peripherals_mouse.schemas.in libgnome-2.20.1.1-hacked/schemas/desktop_gnome_peripherals_mouse.schemas.in
--- libgnome-2.20.1.1/schemas/desktop_gnome_peripherals_mouse.schemas.in	2007-10-17 07:30:40.000000000 +1300
+++ libgnome-2.20.1.1-hacked/schemas/desktop_gnome_peripherals_mouse.schemas.in	2008-01-29 14:29:10.374614425 +1300
@@ -102,6 +102,7 @@
       <applyto>/desktop/gnome/peripherals/mouse/cursor_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
+      <default>DMZ-White</default>
       <locale name="C">
 	<short>Cursor theme</short>
 	<long>
diff -ur libgnome-2.23.4-orig//schemas/desktop_gnome_interface.schemas.in.in libgnome-2.23.4//schemas/desktop_gnome_interface.schemas.in.in
--- libgnome-2.23.4-orig//schemas/desktop_gnome_interface.schemas.in.in	Tue Aug 12 15:46:20 2008
+++ libgnome-2.23.4//schemas/desktop_gnome_interface.schemas.in.in	Tue Aug 12 15:49:53 2008
@@ -165,8 +165,9 @@
       <applyto>/desktop/gnome/interface/font_name</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>Sans Condensed 10</default>
       <locale name="C">
+	<default><!-- SUN_BRANDING -->Sans Condensed 10</default>
 	<short>Default font</short>
 	<long>Name of the default font used by gtk+.</long>
       </locale>
@@ -209,8 +209,9 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>Sans Condensed 10</default>
       <locale name="C">
+	<default><!-- SUN_BRANDING -->Sans Condensed 10</default>
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents.</long>
       </locale>
