--- libgnome-2.30.0/schemas/desktop_gnome_accessibility_keyboard.schemas.in.orig	2011-07-28 15:56:24.867839528 +0800
+++ libgnome-2.30.0/schemas/desktop_gnome_accessibility_keyboard.schemas.in	2011-07-28 15:56:38.635131934 +0800
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/accessibility/keyboard/enable</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short></short>
         <long></long>
