--- libgnome-2.23.5/schemas/desktop_gnome_interface.schemas.in.in.orig	2008-09-09 12:09:06.663288729 +0800
+++ libgnome-2.23.5/schemas/desktop_gnome_interface.schemas.in.in	2008-09-09 12:09:39.746415644 +0800
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/interface/accessibility</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>@A11Y_ENABLED@</default>
+      <default>false</default>
       <locale name="C">
         <short>Enable Accessibility</short>
         <long>Whether Applications should have accessibility support.</long>
