--- Python-2.6.8/Misc/python.man.orig	2012-04-10 08:32:09 -0700
+++ Python-2.6.8/Misc/python.man	2015-06-10 13:31:25 -0700
@@ -108,6 +108,11 @@
 viewed by running the 
 .B pydoc
 program.  
+.SH CAVEAT
+Python 2.6 support from python.org ended in October 2013.  Therefore this
+version of Python is now Obsolete, and will be removed from a future
+release of Solaris.  Please migrate any consumers to use Python version
+2.7 or 3.4 instead.
 .SH COMMAND LINE OPTIONS
 .TP
 .B \-B
