--- Python-2.6.8/configure.in.orig	2012-05-14 06:24:15.149791072 -0700
+++ Python-2.6.8/configure.in	2012-05-14 06:27:52.383479003 -0700
@@ -1416,36 +1416,9 @@
 ])
 AC_MSG_RESULT($was_it_defined)
 
-# Check whether using makedev requires defining _OSF_SOURCE
-AC_MSG_CHECKING(for makedev)
-AC_TRY_LINK([
-#if defined(MAJOR_IN_MKDEV)
-#include <sys/mkdev.h>
-#elif defined(MAJOR_IN_SYSMACROS)
-#include <sys/sysmacros.h>
-#else
-#include <sys/types.h>
-#endif ],
-	    [ makedev(0, 0) ],
-	    ac_cv_has_makedev=yes,
-	    ac_cv_has_makedev=no)
-if test "$ac_cv_has_makedev" = "no"; then
-    # we didn't link, try if _OSF_SOURCE will allow us to link
-    AC_TRY_LINK([
-#define _OSF_SOURCE 1
-#include <sys/types.h>
-    ],
-    [ makedev(0, 0) ],
-    ac_cv_has_makedev=yes,
-    ac_cv_has_makedev=no)
-    if test "$ac_cv_has_makedev" = "yes"; then
-        AC_DEFINE(_OSF_SOURCE, 1, [Define _OSF_SOURCE to get the makedev macro.])
-    fi
-fi
-AC_MSG_RESULT($ac_cv_has_makedev)
-if test "$ac_cv_has_makedev" = "yes"; then
-    AC_DEFINE(HAVE_MAKEDEV, 1, [Define this if you have the makedev macro.])
-fi
+# Solaris has makedev, and posixmodule already has all the right #include
+# statements.
+AC_DEFINE(HAVE_MAKEDEV, 1, [Define this if you have the makedev macro.])
 
 # Enabling LFS on Solaris (2.6 to 9) with gcc 2.95 triggers a bug in
 # the system headers: If _XOPEN_SOURCE and _LARGEFILE_SOURCE are
