diff -u Python-2.6.4/Lib/platform.py.orig Python-2.6.4/Lib/platform.py
--- Python-2.6.4/Lib/platform.py.orig	Mon Feb 21 10:46:30 2011
+++ Python-2.6.4/Lib/platform.py	Mon Feb 21 10:46:30 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """ This module tries to retrieve as much platform-identifying data as
     possible. It makes this information available via function APIs.
diff -u Python-2.6.4/Lib/profile.py.orig Python-2.6.4/Lib/profile.py
--- Python-2.6.4/Lib/profile.py.orig	Mon Feb 21 10:46:25 2011
+++ Python-2.6.4/Lib/profile.py	Mon Feb 21 10:46:25 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 #
 # Class for profiling python code. rev 1.0  6/2/94
 #
diff -u Python-2.6.4/Lib/trace.py.orig Python-2.6.4/Lib/trace.py
--- Python-2.6.4/Lib/trace.py.orig	Mon Feb 21 10:46:29 2011
+++ Python-2.6.4/Lib/trace.py	Mon Feb 21 10:46:29 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # portions copyright 2001, Autonomous Zones Industries, Inc., all rights...
 # err...  reserved and offered to the public under the terms of the
diff -u Python-2.6.4/Lib/cProfile.py.orig Python-2.6.4/Lib/cProfile.py
--- Python-2.6.4/Lib/cProfile.py.orig	Mon Feb 21 10:46:29 2011
+++ Python-2.6.4/Lib/cProfile.py	Mon Feb 21 10:46:29 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Python interface for the 'lsprof' profiler.
    Compatible with the 'profile' module.
diff -u Python-2.6.4/Lib/idlelib/PyShell.py.orig Python-2.6.4/Lib/idlelib/PyShell.py
--- Python-2.6.4/Lib/idlelib/PyShell.py.orig	Mon Feb 21 10:46:23 2011
+++ Python-2.6.4/Lib/idlelib/PyShell.py	Mon Feb 21 10:46:23 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 import os
 import os.path
diff -u Python-2.6.4/Lib/UserString.py.orig Python-2.6.4/Lib/UserString.py
--- Python-2.6.4/Lib/UserString.py.orig	Mon Feb 21 10:46:27 2011
+++ Python-2.6.4/Lib/UserString.py	Mon Feb 21 10:46:27 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 ## vim:ts=4:et:nowrap
 """A user-defined wrapper around string objects
 
diff -u Python-2.6.4/Lib/lib2to3/tests/pytree_idempotency.py.orig Python-2.6.4/Lib/lib2to3/tests/pytree_idempotency.py
--- Python-2.6.4/Lib/lib2to3/tests/pytree_idempotency.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/pytree_idempotency.py	Mon Feb 21 10:46:24 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -u Python-2.6.4/Lib/lib2to3/tests/test_all_fixers.py.orig Python-2.6.4/Lib/lib2to3/tests/test_all_fixers.py
--- Python-2.6.4/Lib/lib2to3/tests/test_all_fixers.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/test_all_fixers.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 """Tests that run all fixer modules over an input stream.
 
 This has been broken out into its own test module because of its
diff -u Python-2.6.4/Lib/lib2to3/tests/test_util.py.orig Python-2.6.4/Lib/lib2to3/tests/test_util.py
--- Python-2.6.4/Lib/lib2to3/tests/test_util.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/test_util.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 """ Test suite for the code in fixer_util """
 
 # Testing imports
diff -u Python-2.6.4/Lib/lib2to3/tests/test_parser.py.orig Python-2.6.4/Lib/lib2to3/tests/test_parser.py
--- Python-2.6.4/Lib/lib2to3/tests/test_parser.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/test_parser.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 """Test suite for 2to3's parser and grammar files.
 
 This is the place to add tests for changes to 2to3's grammar, such as those
diff -u Python-2.6.4/Lib/lib2to3/tests/test_pytree.py.orig Python-2.6.4/Lib/lib2to3/tests/test_pytree.py
--- Python-2.6.4/Lib/lib2to3/tests/test_pytree.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/test_pytree.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -u Python-2.6.4/Lib/lib2to3/tests/test_fixers.py.orig Python-2.6.4/Lib/lib2to3/tests/test_fixers.py
--- Python-2.6.4/Lib/lib2to3/tests/test_fixers.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/tests/test_fixers.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 """ Test suite for the fixer modules """
 # Author: Collin Winter
 
diff -u Python-2.6.4/Lib/lib2to3/refactor.py.orig Python-2.6.4/Lib/lib2to3/refactor.py
--- Python-2.6.4/Lib/lib2to3/refactor.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/lib2to3/refactor.py	Mon Feb 21 10:46:24 2011
@@ -1,3 +1,4 @@
+#!/usr/bin/python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -u Python-2.6.4/Lib/lib2to3/pgen2/token.py.orig Python-2.6.4/Lib/lib2to3/pgen2/token.py
--- Python-2.6.4/Lib/lib2to3/pgen2/token.py.orig	Mon Feb 21 10:46:25 2011
+++ Python-2.6.4/Lib/lib2to3/pgen2/token.py	Mon Feb 21 10:46:25 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Token constants (from "token.h")."""
 
diff -u Python-2.6.4/Lib/cgi.py.orig Python-2.6.4/Lib/cgi.py
--- Python-2.6.4/Lib/cgi.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/cgi.py	Mon Feb 21 10:46:24 2011
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#!/usr/bin/python2.6
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
diff -u Python-2.6.4/Lib/base64.py.orig Python-2.6.4/Lib/base64.py
--- Python-2.6.4/Lib/base64.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/base64.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """RFC 3548: Base16, Base32, Base64 Data Encodings"""
 
diff -u Python-2.6.4/Lib/encodings/rot_13.py.orig Python-2.6.4/Lib/encodings/rot_13.py
--- Python-2.6.4/Lib/encodings/rot_13.py.orig	Mon Feb 21 10:46:26 2011
+++ Python-2.6.4/Lib/encodings/rot_13.py	Mon Feb 21 10:46:26 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 """ Python Character Mapping Codec for ROT13.
 
     See http://ucsub.colorado.edu/~kominek/rot13/ for details.
diff -u Python-2.6.4/Lib/tabnanny.py.orig Python-2.6.4/Lib/tabnanny.py
--- Python-2.6.4/Lib/tabnanny.py.orig	Mon Feb 21 10:46:27 2011
+++ Python-2.6.4/Lib/tabnanny.py	Mon Feb 21 10:46:27 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """The Tab Nanny despises ambiguous indentation.  She knows no mercy.
 
diff -u Python-2.6.4/Lib/timeit.py.orig Python-2.6.4/Lib/timeit.py
--- Python-2.6.4/Lib/timeit.py.orig	Mon Feb 21 10:46:23 2011
+++ Python-2.6.4/Lib/timeit.py	Mon Feb 21 10:46:23 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Tool for measuring execution time of small code snippets.
 
diff -u Python-2.6.4/Lib/pydoc.py.orig Python-2.6.4/Lib/pydoc.py
--- Python-2.6.4/Lib/pydoc.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/pydoc.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: latin-1 -*-
 """Generate Python documentation in HTML or text for interactive use.
 
diff -u Python-2.6.4/Lib/quopri.py.orig Python-2.6.4/Lib/quopri.py
--- Python-2.6.4/Lib/quopri.py.orig	Mon Feb 21 10:46:29 2011
+++ Python-2.6.4/Lib/quopri.py	Mon Feb 21 10:46:29 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Conversions to/from quoted-printable transport encoding as per RFC 1521."""
 
diff -u Python-2.6.4/Lib/keyword.py.orig Python-2.6.4/Lib/keyword.py
--- Python-2.6.4/Lib/keyword.py.orig	Mon Feb 21 10:46:24 2011
+++ Python-2.6.4/Lib/keyword.py	Mon Feb 21 10:46:24 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Keywords (from "graminit.c")
 
diff -u Python-2.6.4/Lib/mailbox.py.orig Python-2.6.4/Lib/mailbox.py
--- Python-2.6.4/Lib/mailbox.py.orig	Mon Feb 21 10:46:27 2011
+++ Python-2.6.4/Lib/mailbox.py	Mon Feb 21 10:46:27 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Read/write support for Maildir, mbox, MH, Babyl, and MMDF mailboxes."""
 
diff -u Python-2.6.4/Lib/difflib.py.orig Python-2.6.4/Lib/difflib.py
--- Python-2.6.4/Lib/difflib.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/difflib.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 Module difflib -- helpers for computing deltas between objects.
diff -u Python-2.6.4/Lib/webbrowser.py.orig Python-2.6.4/Lib/webbrowser.py
--- Python-2.6.4/Lib/webbrowser.py.orig	Mon Feb 21 10:46:26 2011
+++ Python-2.6.4/Lib/webbrowser.py	Mon Feb 21 10:46:26 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Interfaces for launching and remotely controlling Web browsers."""
 # Maintained by Georg Brandl.
 
diff -u Python-2.6.4/Lib/unittest.py.orig Python-2.6.4/Lib/unittest.py
--- Python-2.6.4/Lib/unittest.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/unittest.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 '''
 Python unit testing framework, based on Erich Gamma's JUnit and Kent Beck's
 Smalltalk testing framework.
diff -u Python-2.6.4/Lib/bsddb/test/test_dbtables.py.orig Python-2.6.4/Lib/bsddb/test/test_dbtables.py
--- Python-2.6.4/Lib/bsddb/test/test_dbtables.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/bsddb/test/test_dbtables.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 #-----------------------------------------------------------------------
 # A test suite for the table interface built on bsddb.db
diff -u Python-2.6.4/Lib/bsddb/dbshelve.py.orig Python-2.6.4/Lib/bsddb/dbshelve.py
--- Python-2.6.4/Lib/bsddb/dbshelve.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/bsddb/dbshelve.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/usr/bin/python2.6
 #------------------------------------------------------------------------
 #           Copyright (c) 1997-2001 by Total Control Software
 #                         All Rights Reserved
diff -u Python-2.6.4/Lib/token.py.orig Python-2.6.4/Lib/token.py
--- Python-2.6.4/Lib/token.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/token.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Token constants (from "token.h")."""
 
diff -u Python-2.6.4/Lib/Cookie.py.orig Python-2.6.4/Lib/Cookie.py
--- Python-2.6.4/Lib/Cookie.py.orig	Mon Feb 21 10:46:25 2011
+++ Python-2.6.4/Lib/Cookie.py	Mon Feb 21 10:46:25 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 
 ####
diff -u Python-2.6.4/Lib/smtplib.py.orig Python-2.6.4/Lib/smtplib.py
--- Python-2.6.4/Lib/smtplib.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/smtplib.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 '''SMTP/ESMTP client class.
 
diff -u Python-2.6.4/Lib/smtpd.py.orig Python-2.6.4/Lib/smtpd.py
--- Python-2.6.4/Lib/smtpd.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/smtpd.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """An RFC 2821 smtp proxy.
 
 Usage: %(program)s [options] [localhost:localport [remotehost:remoteport]]
diff -u Python-2.6.4/Lib/test/regrtest.py.orig Python-2.6.4/Lib/test/regrtest.py
--- Python-2.6.4/Lib/test/regrtest.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/regrtest.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Regression test.
 
diff -u Python-2.6.4/Lib/test/test_sets.py.orig Python-2.6.4/Lib/test/test_sets.py
--- Python-2.6.4/Lib/test/test_sets.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_sets.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import warnings
 warnings.filterwarnings("ignore", "the sets module is deprecated",
diff -u Python-2.6.4/Lib/test/test_dl.py.orig Python-2.6.4/Lib/test/test_dl.py
--- Python-2.6.4/Lib/test/test_dl.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_dl.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test dlmodule.c
    Roger E. Masse  revised strategy by Barry Warsaw
 """
diff -u Python-2.6.4/Lib/test/test_multiprocessing.py.orig Python-2.6.4/Lib/test/test_multiprocessing.py
--- Python-2.6.4/Lib/test/test_multiprocessing.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_multiprocessing.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 #
 # Unit tests for the multiprocessing package
diff -u Python-2.6.4/Lib/test/test_cmd.py.orig Python-2.6.4/Lib/test/test_cmd.py
--- Python-2.6.4/Lib/test/test_cmd.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/test/test_cmd.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 """
 Test script for the 'cmd' module
 Original by Michael Schneider
diff -u Python-2.6.4/Lib/test/test_userstring.py.orig Python-2.6.4/Lib/test/test_userstring.py
--- Python-2.6.4/Lib/test/test_userstring.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_userstring.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # UserString is a wrapper around the native builtin string type.
 # UserString instances should behave similar to builtin string objects.
 
diff -u Python-2.6.4/Lib/test/test_popen.py.orig Python-2.6.4/Lib/test/test_popen.py
--- Python-2.6.4/Lib/test/test_popen.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_popen.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Basic tests for os.popen()
 
   Particularly useful for platforms that fake popen.
diff -u Python-2.6.4/Lib/test/test_cd.py.orig Python-2.6.4/Lib/test/test_cd.py
--- Python-2.6.4/Lib/test/test_cd.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_cd.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Whimpy test script for the cd module
    Roger E. Masse
 """
diff -u Python-2.6.4/Lib/test/test_cl.py.orig Python-2.6.4/Lib/test/test_cl.py
--- Python-2.6.4/Lib/test/test_cl.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_cl.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Whimpy test script for the cl module
    Roger E. Masse
 """
diff -u Python-2.6.4/Lib/test/test_imgfile.py.orig Python-2.6.4/Lib/test/test_imgfile.py
--- Python-2.6.4/Lib/test/test_imgfile.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_imgfile.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Simple test script for imgfile.c
    Roger E. Masse
diff -u Python-2.6.4/Lib/test/test_random.py.orig Python-2.6.4/Lib/test/test_random.py
--- Python-2.6.4/Lib/test/test_random.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_random.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 import random
diff -u Python-2.6.4/Lib/test/curses_tests.py.orig Python-2.6.4/Lib/test/curses_tests.py
--- Python-2.6.4/Lib/test/curses_tests.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/curses_tests.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # $Id: ncurses.py 36559 2004-07-18 05:56:09Z tim_one $
 #
diff -u Python-2.6.4/Lib/test/test_codecmaps_cn.py.orig Python-2.6.4/Lib/test/test_codecmaps_cn.py
--- Python-2.6.4/Lib/test/test_codecmaps_cn.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_codecmaps_cn.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecmaps_cn.py
 #   Codec mapping tests for PRC encodings
diff -u Python-2.6.4/Lib/test/test_whichdb.py.orig Python-2.6.4/Lib/test/test_whichdb.py
--- Python-2.6.4/Lib/test/test_whichdb.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/test/test_whichdb.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the whichdb module
    based on test_anydbm.py
 """
diff -u Python-2.6.4/Lib/test/test_eof.py.orig Python-2.6.4/Lib/test/test_eof.py
--- Python-2.6.4/Lib/test/test_eof.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_eof.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """test script for a few new invalid token catches"""
 
 import unittest
diff -u Python-2.6.4/Lib/test/test_imageop.py.orig Python-2.6.4/Lib/test/test_imageop.py
--- Python-2.6.4/Lib/test/test_imageop.py.orig	Mon Feb 21 10:46:31 2011
+++ Python-2.6.4/Lib/test/test_imageop.py	Mon Feb 21 10:46:31 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Test script for the imageop module.  This has the side
    effect of partially testing the imgfile module as well.
diff -u Python-2.6.4/Lib/test/test_codecmaps_tw.py.orig Python-2.6.4/Lib/test/test_codecmaps_tw.py
--- Python-2.6.4/Lib/test/test_codecmaps_tw.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_codecmaps_tw.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecmaps_tw.py
 #   Codec mapping tests for ROC encodings
diff -u Python-2.6.4/Lib/test/test_codecencodings_hk.py.orig Python-2.6.4/Lib/test/test_codecencodings_hk.py
--- Python-2.6.4/Lib/test/test_codecencodings_hk.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_codecencodings_hk.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecencodings_hk.py
 #   Codec encoding tests for HongKong encodings.
diff -u Python-2.6.4/Lib/test/test_codecencodings_jp.py.orig Python-2.6.4/Lib/test/test_codecencodings_jp.py
--- Python-2.6.4/Lib/test/test_codecencodings_jp.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_codecencodings_jp.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecencodings_jp.py
 #   Codec encoding tests for Japanese encodings.
diff -u Python-2.6.4/Lib/test/test_with.py.orig Python-2.6.4/Lib/test/test_with.py
--- Python-2.6.4/Lib/test/test_with.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_with.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """Unit tests for the with statement specified in PEP 343."""
 
diff -u Python-2.6.4/Lib/test/crashers/recursive_call.py.orig Python-2.6.4/Lib/test/crashers/recursive_call.py
--- Python-2.6.4/Lib/test/crashers/recursive_call.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/crashers/recursive_call.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # No bug report AFAIK, mail on python-dev on 2006-01-10
 
diff -u Python-2.6.4/Lib/test/test_codecmaps_kr.py.orig Python-2.6.4/Lib/test/test_codecmaps_kr.py
--- Python-2.6.4/Lib/test/test_codecmaps_kr.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_codecmaps_kr.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecmaps_kr.py
 #   Codec mapping tests for ROK encodings
diff -u Python-2.6.4/Lib/test/test_array.py.orig Python-2.6.4/Lib/test/test_array.py
--- Python-2.6.4/Lib/test/test_array.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_array.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test the arraymodule.
    Roger E. Masse
 """
diff -u Python-2.6.4/Lib/test/test_pep263.py.orig Python-2.6.4/Lib/test/test_pep263.py
--- Python-2.6.4/Lib/test/test_pep263.py.orig   Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_pep263.py        Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-# -*- coding: koi8-r -*-
+#!/usr/bin/python2.6
 
 import unittest
 from test import test_support
diff -u Python-2.6.4/Lib/test/test_anydbm.py.orig Python-2.6.4/Lib/test/test_anydbm.py
--- Python-2.6.4/Lib/test/test_anydbm.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_anydbm.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the anydbm module
    based on testdumbdbm.py
 """
diff -u Python-2.6.4/Lib/test/test_smtpnet.py.orig Python-2.6.4/Lib/test/test_smtpnet.py
--- Python-2.6.4/Lib/test/test_smtpnet.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_smtpnet.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 from test import test_support
diff -u Python-2.6.4/Lib/test/test_al.py.orig Python-2.6.4/Lib/test/test_al.py
--- Python-2.6.4/Lib/test/test_al.py.orig	Mon Feb 21 10:46:31 2011
+++ Python-2.6.4/Lib/test/test_al.py	Mon Feb 21 10:46:31 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Whimpy test script for the al module
    Roger E. Masse
 """
diff -u Python-2.6.4/Lib/test/test_urllib2net.py.orig Python-2.6.4/Lib/test/test_urllib2net.py
--- Python-2.6.4/Lib/test/test_urllib2net.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_urllib2net.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 from test import test_support
diff -u Python-2.6.4/Lib/test/test_marshal.py.orig Python-2.6.4/Lib/test/test_marshal.py
--- Python-2.6.4/Lib/test/test_marshal.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_marshal.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: iso-8859-1 -*-
 
 from test import test_support
diff -u Python-2.6.4/Lib/test/test_urlparse.py.orig Python-2.6.4/Lib/test/test_urlparse.py
--- Python-2.6.4/Lib/test/test_urlparse.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_urlparse.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 from test import test_support
 import unittest
diff -u Python-2.6.4/Lib/test/test_bz2.py.orig Python-2.6.4/Lib/test/test_bz2.py
--- Python-2.6.4/Lib/test/test_bz2.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_bz2.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 from test import test_support
 from test.test_support import TESTFN
 
diff -u Python-2.6.4/Lib/test/test_urllibnet.py.orig Python-2.6.4/Lib/test/test_urllibnet.py
--- Python-2.6.4/Lib/test/test_urllibnet.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_urllibnet.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 from test import test_support
diff -u Python-2.6.4/Lib/test/test_logging.py.orig Python-2.6.4/Lib/test/test_logging.py
--- Python-2.6.4/Lib/test/test_logging.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_logging.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # Copyright 2001-2004 by Vinay Sajip. All Rights Reserved.
 #
diff -u Python-2.6.4/Lib/test/test_socket.py.orig Python-2.6.4/Lib/test/test_socket.py
--- Python-2.6.4/Lib/test/test_socket.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_socket.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 from test import test_support
diff -u Python-2.6.4/Lib/test/pystone.py.orig Python-2.6.4/Lib/test/pystone.py
--- Python-2.6.4/Lib/test/pystone.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/pystone.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 "PYSTONE" Benchmark Program
diff -u Python-2.6.4/Lib/test/test_multibytecodec.py.orig Python-2.6.4/Lib/test/test_multibytecodec.py
--- Python-2.6.4/Lib/test/test_multibytecodec.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_multibytecodec.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_multibytecodec.py
 #   Unit test for multibytecodec itself
diff -u Python-2.6.4/Lib/test/test_popen2.py.orig Python-2.6.4/Lib/test/test_popen2.py
--- Python-2.6.4/Lib/test/test_popen2.py.orig	Mon Feb 21 10:46:33 2011
+++ Python-2.6.4/Lib/test/test_popen2.py	Mon Feb 21 10:46:33 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for popen2.py"""
 
 import warnings
diff -u Python-2.6.4/Lib/test/test_binhex.py.orig Python-2.6.4/Lib/test/test_binhex.py
--- Python-2.6.4/Lib/test/test_binhex.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_binhex.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the binhex C module
 
    Uses the mechanism of the python binhex module
diff -u Python-2.6.4/Lib/test/test_gzip.py.orig Python-2.6.4/Lib/test/test_gzip.py
--- Python-2.6.4/Lib/test/test_gzip.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/test/test_gzip.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the gzip module.
 """
 
diff -u Python-2.6.4/Lib/test/re_tests.py.orig Python-2.6.4/Lib/test/re_tests.py
--- Python-2.6.4/Lib/test/re_tests.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/re_tests.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- mode: python -*-
 
 # Re test suite and benchmark suite v1.5
diff -u Python-2.6.4/Lib/test/test_urllib2_localnet.py.orig Python-2.6.4/Lib/test/test_urllib2_localnet.py
--- Python-2.6.4/Lib/test/test_urllib2_localnet.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_urllib2_localnet.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import mimetools
 import threading
diff -u Python-2.6.4/Lib/test/test_codecencodings_kr.py.orig Python-2.6.4/Lib/test/test_codecencodings_kr.py
--- Python-2.6.4/Lib/test/test_codecencodings_kr.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_codecencodings_kr.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecencodings_kr.py
 #   Codec encoding tests for ROK encodings.
diff -u Python-2.6.4/Lib/test/test_multibytecodec_support.py.orig Python-2.6.4/Lib/test/test_multibytecodec_support.py
--- Python-2.6.4/Lib/test/test_multibytecodec_support.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_multibytecodec_support.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_multibytecodec_support.py
 #   Common Unittest Routines for CJK codecs
diff -u Python-2.6.4/Lib/test/test_tcl.py.orig Python-2.6.4/Lib/test/test_tcl.py
--- Python-2.6.4/Lib/test/test_tcl.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/test/test_tcl.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import unittest
 import os
diff -u Python-2.6.4/Lib/test/test_codecencodings_cn.py.orig Python-2.6.4/Lib/test/test_codecencodings_cn.py
--- Python-2.6.4/Lib/test/test_codecencodings_cn.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_codecencodings_cn.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecencodings_cn.py
 #   Codec encoding tests for PRC encodings.
diff -u Python-2.6.4/Lib/test/test_gl.py.orig Python-2.6.4/Lib/test/test_gl.py
--- Python-2.6.4/Lib/test/test_gl.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_gl.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Very simple test script for the SGI gl library extension module
     taken mostly from the documentation.
     Roger E. Masse
diff -u Python-2.6.4/Lib/test/test___future__.py.orig Python-2.6.4/Lib/test/test___future__.py
--- Python-2.6.4/Lib/test/test___future__.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test___future__.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 import unittest
 from test import test_support
 import __future__
diff -u Python-2.6.4/Lib/test/test_bsddb.py.orig Python-2.6.4/Lib/test/test_bsddb.py
--- Python-2.6.4/Lib/test/test_bsddb.py.orig	Mon Feb 21 10:46:36 2011
+++ Python-2.6.4/Lib/test/test_bsddb.py	Mon Feb 21 10:46:36 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the bsddb C module by Roger E. Masse
    Adapted to unittest format and expanded scope by Raymond Hettinger
 """
diff -u Python-2.6.4/Lib/test/test_errno.py.orig Python-2.6.4/Lib/test/test_errno.py
--- Python-2.6.4/Lib/test/test_errno.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_errno.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test the errno module
    Roger E. Masse
 """
diff -u Python-2.6.4/Lib/test/test_optparse.py.orig Python-2.6.4/Lib/test/test_optparse.py
--- Python-2.6.4/Lib/test/test_optparse.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_optparse.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 
 #
 # Test suite for Optik.  Supplied by Johannes Gijsbers
diff -u Python-2.6.4/Lib/test/test_codecencodings_tw.py.orig Python-2.6.4/Lib/test/test_codecencodings_tw.py
--- Python-2.6.4/Lib/test/test_codecencodings_tw.py.orig	Mon Feb 21 10:46:35 2011
+++ Python-2.6.4/Lib/test/test_codecencodings_tw.py	Mon Feb 21 10:46:35 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecencodings_tw.py
 #   Codec encoding tests for ROC encodings.
diff -u Python-2.6.4/Lib/test/test_codecmaps_jp.py.orig Python-2.6.4/Lib/test/test_codecmaps_jp.py
--- Python-2.6.4/Lib/test/test_codecmaps_jp.py.orig	Mon Feb 21 10:46:32 2011
+++ Python-2.6.4/Lib/test/test_codecmaps_jp.py	Mon Feb 21 10:46:32 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecmaps_jp.py
 #   Codec mapping tests for Japanese encodings
diff -u Python-2.6.4/Lib/test/test_codecmaps_hk.py.orig Python-2.6.4/Lib/test/test_codecmaps_hk.py
--- Python-2.6.4/Lib/test/test_codecmaps_hk.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_codecmaps_hk.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # test_codecmaps_hk.py
 #   Codec mapping tests for HongKong encodings
diff -u Python-2.6.4/Lib/test/test_dumbdbm.py.orig Python-2.6.4/Lib/test/test_dumbdbm.py
--- Python-2.6.4/Lib/test/test_dumbdbm.py.orig	Mon Feb 21 10:46:34 2011
+++ Python-2.6.4/Lib/test/test_dumbdbm.py	Mon Feb 21 10:46:34 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Test script for the dumbdbm module
    Original by Roger E. Masse
 """
diff -u Python-2.6.4/Lib/tarfile.py.orig Python-2.6.4/Lib/tarfile.py
--- Python-2.6.4/Lib/tarfile.py.orig	Mon Feb 21 10:46:38 2011
+++ Python-2.6.4/Lib/tarfile.py	Mon Feb 21 10:46:38 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: iso-8859-1 -*-
 #-------------------------------------------------------------------
 # tarfile.py
diff -u Python-2.6.4/Lib/uu.py.orig Python-2.6.4/Lib/uu.py
--- Python-2.6.4/Lib/uu.py.orig	Mon Feb 21 10:46:25 2011
+++ Python-2.6.4/Lib/uu.py	Mon Feb 21 10:46:25 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Copyright 1994 by Lance Ellinghouse
 # Cathedral City, California Republic, United States of America.
diff -u Python-2.6.4/Lib/symbol.py.orig Python-2.6.4/Lib/symbol.py
--- Python-2.6.4/Lib/symbol.py.orig	Mon Feb 21 10:46:27 2011
+++ Python-2.6.4/Lib/symbol.py	Mon Feb 21 10:46:27 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Non-terminal symbols of Python grammar (from "graminit.h")."""
 
diff -u Python-2.6.4/Lib/mimify.py.orig Python-2.6.4/Lib/mimify.py
--- Python-2.6.4/Lib/mimify.py.orig	Mon Feb 21 10:46:28 2011
+++ Python-2.6.4/Lib/mimify.py	Mon Feb 21 10:46:28 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Mimification and unmimification of mail messages.
 
diff -u Python-2.6.4/Lib/pdb.py.orig Python-2.6.4/Lib/pdb.py
--- Python-2.6.4/Lib/pdb.py.orig	Mon Feb 21 10:46:38 2011
+++ Python-2.6.4/Lib/pdb.py	Mon Feb 21 10:46:38 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """A Python debugger."""
 
diff -u Python-2.6.4/Lib/plat-mac/bundlebuilder.py.orig Python-2.6.4/Lib/plat-mac/bundlebuilder.py
--- Python-2.6.4/Lib/plat-mac/bundlebuilder.py.orig	Mon Feb 21 10:46:38 2011
+++ Python-2.6.4/Lib/plat-mac/bundlebuilder.py	Mon Feb 21 10:46:38 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """\
 bundlebuilder.py -- Tools to assemble MacOS X (application) bundles.
diff -u Python-2.6.4/Lib/plat-mac/appletrunner.py.orig Python-2.6.4/Lib/plat-mac/appletrunner.py
--- Python-2.6.4/Lib/plat-mac/appletrunner.py.orig	Mon Feb 21 10:46:38 2011
+++ Python-2.6.4/Lib/plat-mac/appletrunner.py	Mon Feb 21 10:46:38 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # This file is meant as an executable script for running applets.
 # BuildApplet will use it as the main executable in the .app bundle if
 # we are not running in a framework build.
diff -u Python-2.6.4/Demo/scripts/primes.py.orig Python-2.6.4/Demo/scripts/primes.py
--- Python-2.6.4/Demo/scripts/primes.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/primes.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Print prime numbers in a given range
 
diff -u Python-2.6.4/Demo/scripts/mkrcs.py.orig Python-2.6.4/Demo/scripts/mkrcs.py
--- Python-2.6.4/Demo/scripts/mkrcs.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/mkrcs.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # A rather specialized script to make sure that a symbolic link named
 # RCS exists pointing to a real RCS directory in a parallel tree
diff -u Python-2.6.4/Demo/scripts/queens.py.orig Python-2.6.4/Demo/scripts/queens.py
--- Python-2.6.4/Demo/scripts/queens.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/queens.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """N queens problem.
 
diff -u Python-2.6.4/Demo/scripts/pi.py.orig Python-2.6.4/Demo/scripts/pi.py
--- Python-2.6.4/Demo/scripts/pi.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/pi.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Print digits of pi forever.
 #
diff -u Python-2.6.4/Demo/scripts/mboxconvert.py.orig Python-2.6.4/Demo/scripts/mboxconvert.py
--- Python-2.6.4/Demo/scripts/mboxconvert.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/mboxconvert.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Convert  MH directories (1 message per file) or MMDF mailboxes (4x^A
 # delimited) to unix mailbox (From ... delimited) on stdout.
diff -u Python-2.6.4/Demo/scripts/eqfix.py.orig Python-2.6.4/Demo/scripts/eqfix.py
--- Python-2.6.4/Demo/scripts/eqfix.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/eqfix.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Fix Python source files to use the new equality test operator, i.e.,
 #       if x = y: ...
diff -u Python-2.6.4/Demo/scripts/find-uname.py.orig Python-2.6.4/Demo/scripts/find-uname.py
--- Python-2.6.4/Demo/scripts/find-uname.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/find-uname.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 For each argument on the command line, look for it in the set of all Unicode
diff -u Python-2.6.4/Demo/scripts/pp.py.orig Python-2.6.4/Demo/scripts/pp.py
--- Python-2.6.4/Demo/scripts/pp.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/pp.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Emulate some Perl command line options.
 # Usage: pp [-a] [-c] [-d] [-e scriptline] [-F fieldsep] [-n] [-p] [file] ...
diff -u Python-2.6.4/Demo/scripts/unbirthday.py.orig Python-2.6.4/Demo/scripts/unbirthday.py
--- Python-2.6.4/Demo/scripts/unbirthday.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/unbirthday.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Calculate your unbirthday count (see Alice in Wonderland).
 # This is defined as the number of days from your birth until today
diff -u Python-2.6.4/Demo/scripts/update.py.orig Python-2.6.4/Demo/scripts/update.py
--- Python-2.6.4/Demo/scripts/update.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/update.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Update a bunch of files according to a script.
 # The input file contains lines of the form <filename>:<lineno>:<text>,
diff -u Python-2.6.4/Demo/scripts/markov.py.orig Python-2.6.4/Demo/scripts/markov.py
--- Python-2.6.4/Demo/scripts/markov.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/markov.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 class Markov:
     def __init__(self, histsize, choice):
diff -u Python-2.6.4/Demo/scripts/from.py.orig Python-2.6.4/Demo/scripts/from.py
--- Python-2.6.4/Demo/scripts/from.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/from.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Print From and Subject of messages in $MAIL.
 # Extension to multiple mailboxes and other bells & whistles are left
diff -u Python-2.6.4/Demo/scripts/script.py.orig Python-2.6.4/Demo/scripts/script.py
--- Python-2.6.4/Demo/scripts/script.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/script.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # script.py -- Make typescript of terminal session.
 # Usage:
 #       -a      Append to typescript.
diff -u Python-2.6.4/Demo/scripts/lpwatch.py.orig Python-2.6.4/Demo/scripts/lpwatch.py
--- Python-2.6.4/Demo/scripts/lpwatch.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/lpwatch.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Watch line printer queue(s).
 # Intended for BSD 4.3 lpq.
diff -u Python-2.6.4/Demo/scripts/fact.py.orig Python-2.6.4/Demo/scripts/fact.py
--- Python-2.6.4/Demo/scripts/fact.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/fact.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Factorize numbers.
 # The algorithm is not efficient, but easy to understand.
diff -u Python-2.6.4/Demo/scripts/beer.py.orig Python-2.6.4/Demo/scripts/beer.py
--- Python-2.6.4/Demo/scripts/beer.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/beer.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # By GvR, demystified after a version by Fredrik Lundh.
 import sys
 n = 100
diff -u Python-2.6.4/Demo/scripts/newslist.py.orig Python-2.6.4/Demo/scripts/newslist.py
--- Python-2.6.4/Demo/scripts/newslist.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/newslist.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 #######################################################################
 # Newslist  $Revision: 66429 $
 #
diff -u Python-2.6.4/Demo/scripts/ftpstats.py.orig Python-2.6.4/Demo/scripts/ftpstats.py
--- Python-2.6.4/Demo/scripts/ftpstats.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/ftpstats.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Extract statistics from ftp daemon log.
 
diff -u Python-2.6.4/Demo/scripts/makedir.py.orig Python-2.6.4/Demo/scripts/makedir.py
--- Python-2.6.4/Demo/scripts/makedir.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/scripts/makedir.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Like mkdir, but also make intermediate directories if necessary.
 # It is not an error if the given directory already exists (as long
diff -u Python-2.6.4/Demo/turtle/tdemo_clock.py.orig Python-2.6.4/Demo/turtle/tdemo_clock.py
--- Python-2.6.4/Demo/turtle/tdemo_clock.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_clock.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 # -*- coding: cp1252 -*-
 """       turtle-example-suite:
 
diff -u Python-2.6.4/Demo/turtle/tdemo_planet_and_moon.py.orig Python-2.6.4/Demo/turtle/tdemo_planet_and_moon.py
--- Python-2.6.4/Demo/turtle/tdemo_planet_and_moon.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_planet_and_moon.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
         tdemo_planets_and_moon.py
diff -u Python-2.6.4/Demo/turtle/tdemo_lindenmayer_indian.py.orig Python-2.6.4/Demo/turtle/tdemo_lindenmayer_indian.py
--- Python-2.6.4/Demo/turtle/tdemo_lindenmayer_indian.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_lindenmayer_indian.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
         xtx_lindenmayer_indian.py
diff -u Python-2.6.4/Demo/turtle/tdemo_penrose.py.orig Python-2.6.4/Demo/turtle/tdemo_penrose.py
--- Python-2.6.4/Demo/turtle/tdemo_penrose.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_penrose.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       xturtle-example-suite:
 
           xtx_kites_and_darts.py
diff -u Python-2.6.4/Demo/turtle/turtleDemo.py.orig Python-2.6.4/Demo/turtle/turtleDemo.py
--- Python-2.6.4/Demo/turtle/turtleDemo.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/turtleDemo.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 import sys
 import os
 
diff -u Python-2.6.4/Demo/turtle/tdemo_yinyang.py.orig Python-2.6.4/Demo/turtle/tdemo_yinyang.py
--- Python-2.6.4/Demo/turtle/tdemo_yinyang.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_yinyang.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
             tdemo_yinyang.py
diff -u Python-2.6.4/Demo/turtle/tdemo_tree.py.orig Python-2.6.4/Demo/turtle/tdemo_tree.py
--- Python-2.6.4/Demo/turtle/tdemo_tree.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_tree.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """      turtle-example-suite:
 
              tdemo_tree.py
diff -u Python-2.6.4/Demo/turtle/tdemo_fractalcurves.py.orig Python-2.6.4/Demo/turtle/tdemo_fractalcurves.py
--- Python-2.6.4/Demo/turtle/tdemo_fractalcurves.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_fractalcurves.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """      turtle-example-suite:
 
         tdemo_fractalCurves.py
diff -u Python-2.6.4/Demo/turtle/turtledemo_two_canvases.py.orig Python-2.6.4/Demo/turtle/turtledemo_two_canvases.py
--- Python-2.6.4/Demo/turtle/turtledemo_two_canvases.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/turtledemo_two_canvases.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 ## DEMONSTRATES USE OF 2 CANVASES, SO CANNOT BE RUN IN DEMOVIEWER!
 """turtle example: Using TurtleScreen and RawTurtle
 for drawing on two distinct canvases.
diff -u Python-2.6.4/Demo/turtle/tdemo_minimal_hanoi.py.orig Python-2.6.4/Demo/turtle/tdemo_minimal_hanoi.py
--- Python-2.6.4/Demo/turtle/tdemo_minimal_hanoi.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_minimal_hanoi.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
          tdemo_minimal_hanoi.py
diff -u Python-2.6.4/Demo/turtle/tdemo_peace.py.orig Python-2.6.4/Demo/turtle/tdemo_peace.py
--- Python-2.6.4/Demo/turtle/tdemo_peace.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_peace.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
               tdemo_peace.py
diff -u Python-2.6.4/Demo/turtle/tdemo_bytedesign.py.orig Python-2.6.4/Demo/turtle/tdemo_bytedesign.py
--- Python-2.6.4/Demo/turtle/tdemo_bytedesign.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_bytedesign.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """      turtle-example-suite:
 
         tdemo_bytedesign.py
diff -u Python-2.6.4/Demo/turtle/tdemo_paint.py.orig Python-2.6.4/Demo/turtle/tdemo_paint.py
--- Python-2.6.4/Demo/turtle/tdemo_paint.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_paint.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
             tdemo_paint.py
diff -u Python-2.6.4/Demo/turtle/tdemo_I_dontlike_tiltdemo.py.orig Python-2.6.4/Demo/turtle/tdemo_I_dontlike_tiltdemo.py
--- Python-2.6.4/Demo/turtle/tdemo_I_dontlike_tiltdemo.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/turtle/tdemo_I_dontlike_tiltdemo.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """       turtle-example-suite:
 
      tdemo-I_dont_like_tiltdemo.py
diff -u Python-2.6.4/Demo/cgi/cgi1.py.orig Python-2.6.4/Demo/cgi/cgi1.py
--- Python-2.6.4/Demo/cgi/cgi1.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/cgi/cgi1.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2.6
 
 """CGI test 1 - check server setup."""
 
diff -u Python-2.6.4/Demo/cgi/cgi3.py.orig Python-2.6.4/Demo/cgi/cgi3.py
--- Python-2.6.4/Demo/cgi/cgi3.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/cgi/cgi3.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2.6
 
 """CGI test 3 (persistent data)."""
 
diff -u Python-2.6.4/Demo/cgi/cgi2.py.orig Python-2.6.4/Demo/cgi/cgi2.py
--- Python-2.6.4/Demo/cgi/cgi2.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/cgi/cgi2.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2.6
 
 """CGI test 2 - basic use of cgi module."""
 
diff -u Python-2.6.4/Demo/comparisons/sortingtest.py.orig Python-2.6.4/Demo/comparisons/sortingtest.py
--- Python-2.6.4/Demo/comparisons/sortingtest.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/comparisons/sortingtest.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # 2)  Sorting Test
 #
diff -u Python-2.6.4/Demo/comparisons/regextest.py.orig Python-2.6.4/Demo/comparisons/regextest.py
--- Python-2.6.4/Demo/comparisons/regextest.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/comparisons/regextest.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # 1)  Regular Expressions Test
 #
diff -u Python-2.6.4/Demo/comparisons/systemtest.py.orig Python-2.6.4/Demo/comparisons/systemtest.py
--- Python-2.6.4/Demo/comparisons/systemtest.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/comparisons/systemtest.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # 3)  System Test
 #
diff -u Python-2.6.4/Demo/pysvr/pysvr.py.orig Python-2.6.4/Demo/pysvr/pysvr.py
--- Python-2.6.4/Demo/pysvr/pysvr.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/pysvr/pysvr.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """A multi-threaded telnet-like server that gives a Python prompt.
 
diff -u Python-2.6.4/Demo/tkinter/guido/sortvisu.py.orig Python-2.6.4/Demo/tkinter/guido/sortvisu.py
--- Python-2.6.4/Demo/tkinter/guido/sortvisu.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/sortvisu.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Sorting algorithms visualizer using Tkinter.
 
diff -u Python-2.6.4/Demo/tkinter/guido/dialog.py.orig Python-2.6.4/Demo/tkinter/guido/dialog.py
--- Python-2.6.4/Demo/tkinter/guido/dialog.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/dialog.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # A Python function that generates dialog boxes with a text message,
 # optional bitmap, and any number of buttons.
diff -u Python-2.6.4/Demo/tkinter/guido/electrons.py.orig Python-2.6.4/Demo/tkinter/guido/electrons.py
--- Python-2.6.4/Demo/tkinter/guido/electrons.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/electrons.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Simulate "electrons" migrating across the screen.
 # An optional bitmap file in can be in the background.
diff -u Python-2.6.4/Demo/tkinter/guido/newmenubardemo.py.orig Python-2.6.4/Demo/tkinter/guido/newmenubardemo.py
--- Python-2.6.4/Demo/tkinter/guido/newmenubardemo.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/newmenubardemo.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Play with the new Tk 8.0 toplevel menu option."""
 
diff -u Python-2.6.4/Demo/tkinter/guido/canvasevents.py.orig Python-2.6.4/Demo/tkinter/guido/canvasevents.py
--- Python-2.6.4/Demo/tkinter/guido/canvasevents.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/canvasevents.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 from Tkinter import *
 from Canvas import Oval, Group, CanvasText
diff -u Python-2.6.4/Demo/tkinter/guido/tkman.py.orig Python-2.6.4/Demo/tkinter/guido/tkman.py
--- Python-2.6.4/Demo/tkinter/guido/tkman.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/tkman.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Tk man page browser -- currently only shows the Tcl/Tk man pages
 
diff -u Python-2.6.4/Demo/tkinter/guido/solitaire.py.orig Python-2.6.4/Demo/tkinter/guido/solitaire.py
--- Python-2.6.4/Demo/tkinter/guido/solitaire.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/solitaire.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Solitaire game, much like the one that comes with MS Windows.
 
diff -u Python-2.6.4/Demo/tkinter/guido/kill.py.orig Python-2.6.4/Demo/tkinter/guido/kill.py
--- Python-2.6.4/Demo/tkinter/guido/kill.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/kill.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # Tkinter interface to Linux `kill' command.
 
 from Tkinter import *
diff -u Python-2.6.4/Demo/tkinter/guido/MimeViewer.py.orig Python-2.6.4/Demo/tkinter/guido/MimeViewer.py
--- Python-2.6.4/Demo/tkinter/guido/MimeViewer.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/MimeViewer.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # View a single MIME multipart message.
 # Display each part as a box.
diff -u Python-2.6.4/Demo/tkinter/guido/mbox.py.orig Python-2.6.4/Demo/tkinter/guido/mbox.py
--- Python-2.6.4/Demo/tkinter/guido/mbox.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/mbox.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Scan MH folder, display results in window
 
diff -u Python-2.6.4/Demo/tkinter/guido/rmt.py.orig Python-2.6.4/Demo/tkinter/guido/rmt.py
--- Python-2.6.4/Demo/tkinter/guido/rmt.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/rmt.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # A Python program implementing rmt, an application for remotely
 # controlling other Tk applications.
diff -u Python-2.6.4/Demo/tkinter/guido/svkill.py.orig Python-2.6.4/Demo/tkinter/guido/svkill.py
--- Python-2.6.4/Demo/tkinter/guido/svkill.py.orig	Mon Feb 21 10:46:39 2011
+++ Python-2.6.4/Demo/tkinter/guido/svkill.py	Mon Feb 21 10:46:39 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Tkinter interface to SYSV `ps' and `kill' commands.
 
diff -u Python-2.6.4/Demo/pdist/rrcs.py.orig Python-2.6.4/Demo/pdist/rrcs.py
--- Python-2.6.4/Demo/pdist/rrcs.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/pdist/rrcs.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 "Remote RCS -- command line interface"
 
diff -u Python-2.6.4/Demo/pdist/rcvs.py.orig Python-2.6.4/Demo/pdist/rcvs.py
--- Python-2.6.4/Demo/pdist/rcvs.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/pdist/rcvs.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Remote CVS -- command line interface"""
 
diff -u Python-2.6.4/Demo/pdist/RCSProxy.py.orig Python-2.6.4/Demo/pdist/RCSProxy.py
--- Python-2.6.4/Demo/pdist/RCSProxy.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/pdist/RCSProxy.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """RCS Proxy.
 
diff -u Python-2.6.4/Demo/pdist/makechangelog.py.orig Python-2.6.4/Demo/pdist/makechangelog.py
--- Python-2.6.4/Demo/pdist/makechangelog.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/pdist/makechangelog.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Turn a pile of RCS log output into ChangeLog file entries.
 
diff -u Python-2.6.4/Demo/zlib/zlibdemo.py.orig Python-2.6.4/Demo/zlib/zlibdemo.py
--- Python-2.6.4/Demo/zlib/zlibdemo.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/zlib/zlibdemo.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # Takes an optional filename, defaulting to this file itself.
 # Reads the file and compresses the content using level 1 and level 9
diff -u Python-2.6.4/Demo/zlib/minigzip.py.orig Python-2.6.4/Demo/zlib/minigzip.py
--- Python-2.6.4/Demo/zlib/minigzip.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/zlib/minigzip.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # Demo program for zlib; it compresses or decompresses files, but *doesn't*
 # delete the original.  This doesn't support all of gzip's options.
 #
diff -u Python-2.6.4/Demo/curses/life.py.orig Python-2.6.4/Demo/curses/life.py
--- Python-2.6.4/Demo/curses/life.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/curses/life.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # life.py -- A curses-based version of Conway's Game of Life.
 # Contributed by AMK
 #
diff -u Python-2.6.4/Demo/curses/rain.py.orig Python-2.6.4/Demo/curses/rain.py
--- Python-2.6.4/Demo/curses/rain.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/curses/rain.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # $Id: rain.py 46625 2006-06-03 23:02:15Z andrew.kuchling $
 #
diff -u Python-2.6.4/Demo/curses/repeat.py.orig Python-2.6.4/Demo/curses/repeat.py
--- Python-2.6.4/Demo/curses/repeat.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/curses/repeat.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """repeat <shell-command>
 
diff -u Python-2.6.4/Demo/curses/ncurses.py.orig Python-2.6.4/Demo/curses/ncurses.py
--- Python-2.6.4/Demo/curses/ncurses.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/curses/ncurses.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # $Id: ncurses.py 66424 2008-09-13 01:22:08Z andrew.kuchling $
 #
diff -u Python-2.6.4/Demo/curses/tclock.py.orig Python-2.6.4/Demo/curses/tclock.py
--- Python-2.6.4/Demo/curses/tclock.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/curses/tclock.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # $Id: tclock.py 46626 2006-06-03 23:07:21Z andrew.kuchling $
 #
diff -u Python-2.6.4/Demo/parser/test_parser.py.orig Python-2.6.4/Demo/parser/test_parser.py
--- Python-2.6.4/Demo/parser/test_parser.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/parser/test_parser.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 #  (Force the script to use the latest build.)
 #
 #  test_parser.py
diff -u Python-2.6.4/Demo/sockets/gopher.py.orig Python-2.6.4/Demo/sockets/gopher.py
--- Python-2.6.4/Demo/sockets/gopher.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/gopher.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # A simple gopher client.
 #
diff -u Python-2.6.4/Demo/sockets/throughput.py.orig Python-2.6.4/Demo/sockets/throughput.py
--- Python-2.6.4/Demo/sockets/throughput.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/throughput.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Test network throughput.
 #
diff -u Python-2.6.4/Demo/sockets/rpython.py.orig Python-2.6.4/Demo/sockets/rpython.py
--- Python-2.6.4/Demo/sockets/rpython.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/rpython.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Remote python client.
 # Execute Python commands remotely and send output back.
diff -u Python-2.6.4/Demo/sockets/echosvr.py.orig Python-2.6.4/Demo/sockets/echosvr.py
--- Python-2.6.4/Demo/sockets/echosvr.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/echosvr.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Python implementation of an 'echo' tcp server: echo all data it receives.
 #
diff -u Python-2.6.4/Demo/sockets/telnet.py.orig Python-2.6.4/Demo/sockets/telnet.py
--- Python-2.6.4/Demo/sockets/telnet.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/telnet.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Minimal interface to the Internet telnet protocol.
 #
diff -u Python-2.6.4/Demo/sockets/finger.py.orig Python-2.6.4/Demo/sockets/finger.py
--- Python-2.6.4/Demo/sockets/finger.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/finger.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Python interface to the Internet finger daemon.
 #
diff -u Python-2.6.4/Demo/sockets/udpecho.py.orig Python-2.6.4/Demo/sockets/udpecho.py
--- Python-2.6.4/Demo/sockets/udpecho.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/udpecho.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Client and server for udp (datagram) echo.
 #
diff -u Python-2.6.4/Demo/sockets/rpythond.py.orig Python-2.6.4/Demo/sockets/rpythond.py
--- Python-2.6.4/Demo/sockets/rpythond.py.orig	Mon Feb 21 10:46:40 2011
+++ Python-2.6.4/Demo/sockets/rpythond.py	Mon Feb 21 10:46:40 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Remote python server.
 # Execute Python commands remotely and send output back.
diff -u Python-2.6.4/Tools/ssl/get-remote-certificate.py.orig Python-2.6.4/Tools/ssl/get-remote-certificate.py
--- Python-2.6.4/Tools/ssl/get-remote-certificate.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/ssl/get-remote-certificate.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # fetch the certificate that the server(s) are providing in PEM form
 #
diff -u Python-2.6.4/Tools/compiler/demo.py.orig Python-2.6.4/Tools/compiler/demo.py
--- Python-2.6.4/Tools/compiler/demo.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/compiler/demo.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Print names of all methods defined in module
 
diff -u Python-2.6.4/Tools/compiler/dumppyc.py.orig Python-2.6.4/Tools/compiler/dumppyc.py
--- Python-2.6.4/Tools/compiler/dumppyc.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/compiler/dumppyc.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 import marshal
 import dis
diff -u Python-2.6.4/Tools/webchecker/wsgui.py.orig Python-2.6.4/Tools/webchecker/wsgui.py
--- Python-2.6.4/Tools/webchecker/wsgui.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/webchecker/wsgui.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Tkinter-based GUI for websucker.
 
diff -u Python-2.6.4/Tools/webchecker/webchecker.py.orig Python-2.6.4/Tools/webchecker/webchecker.py
--- Python-2.6.4/Tools/webchecker/webchecker.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/webchecker/webchecker.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Original code by Guido van Rossum; extensive changes by Sam Bayer,
 # including code to check URL fragments.
diff -u Python-2.6.4/Tools/webchecker/websucker.py.orig Python-2.6.4/Tools/webchecker/websucker.py
--- Python-2.6.4/Tools/webchecker/websucker.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/webchecker/websucker.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """A variant on webchecker that creates a mirror copy of a remote site."""
 
diff -u Python-2.6.4/Tools/webchecker/wcgui.py.orig Python-2.6.4/Tools/webchecker/wcgui.py
--- Python-2.6.4/Tools/webchecker/wcgui.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/webchecker/wcgui.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """GUI interface to webchecker.
 
diff -u Python-2.6.4/Tools/faqwiz/faqw.py.orig Python-2.6.4/Tools/faqwiz/faqw.py
--- Python-2.6.4/Tools/faqwiz/faqw.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/faqwiz/faqw.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#!/usr/bin/python2.6
 
 """FAQ wizard bootstrap."""
 
diff -u Python-2.6.4/Tools/unicode/comparecodecs.py.orig Python-2.6.4/Tools/unicode/comparecodecs.py
--- Python-2.6.4/Tools/unicode/comparecodecs.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/unicode/comparecodecs.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """ Compare the output of two codecs.
 
diff -u Python-2.6.4/Tools/i18n/msgfmt.py.orig Python-2.6.4/Tools/i18n/msgfmt.py
--- Python-2.6.4/Tools/i18n/msgfmt.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/i18n/msgfmt.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: iso-8859-1 -*-
 # Written by Martin v. Löwis <loewis@informatik.hu-berlin.de>
 
diff -u Python-2.6.4/Tools/i18n/makelocalealias.py.orig Python-2.6.4/Tools/i18n/makelocalealias.py
--- Python-2.6.4/Tools/i18n/makelocalealias.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/i18n/makelocalealias.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 """
     Convert the X11 locale.alias file into a mapping dictionary suitable
     for locale.py.
diff -u Python-2.6.4/Tools/i18n/pygettext.py.orig Python-2.6.4/Tools/i18n/pygettext.py
--- Python-2.6.4/Tools/i18n/pygettext.py.orig	Mon Feb 21 10:46:41 2011
+++ Python-2.6.4/Tools/i18n/pygettext.py	Mon Feb 21 10:46:41 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: iso-8859-1 -*-
 # Originally written by Barry Warsaw <barry@zope.com>
 #
diff -u Python-2.6.4/Tools/freeze/freeze.py.orig Python-2.6.4/Tools/freeze/freeze.py
--- Python-2.6.4/Tools/freeze/freeze.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/freeze/freeze.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Freeze a Python script into a binary.
 
diff -u Python-2.6.4/Tools/modulator/Tkextra.py.orig Python-2.6.4/Tools/modulator/Tkextra.py
--- Python-2.6.4/Tools/modulator/Tkextra.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/modulator/Tkextra.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # A Python function that generates dialog boxes with a text message,
 # optional bitmap, and any number of buttons.
diff -u Python-2.6.4/Tools/modulator/modulator.py.orig Python-2.6.4/Tools/modulator/modulator.py
--- Python-2.6.4/Tools/modulator/modulator.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/modulator/modulator.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 #
 # Modulator - Generate skeleton modules.
 #
diff -u Python-2.6.4/Tools/pybench/pybench.py.orig Python-2.6.4/Tools/pybench/pybench.py
--- Python-2.6.4/Tools/pybench/pybench.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/pybench/pybench.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python -O
+#!/usr/bin/python2.6
 
 """ A Python Benchmark Suite
 
diff -u Python-2.6.4/Tools/pybench/Setup.py.orig Python-2.6.4/Tools/pybench/Setup.py
--- Python-2.6.4/Tools/pybench/Setup.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/pybench/Setup.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!python
+#!/usr/bin/python2.6
 
 # Setup file for pybench
 #
diff -u Python-2.6.4/Tools/pybench/clockres.py.orig Python-2.6.4/Tools/pybench/clockres.py
--- Python-2.6.4/Tools/pybench/clockres.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/pybench/clockres.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """ clockres - calculates the resolution in seconds of a given timer.
 
diff -u Python-2.6.4/Tools/pybench/systimes.py.orig Python-2.6.4/Tools/pybench/systimes.py
--- Python-2.6.4/Tools/pybench/systimes.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/pybench/systimes.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """ systimes() user and system timer implementations for use by
     pybench.
diff -u Python-2.6.4/Tools/scripts/fixheader.py.orig Python-2.6.4/Tools/scripts/fixheader.py
--- Python-2.6.4/Tools/scripts/fixheader.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/fixheader.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Add some standard cpp magic to a header file
 
diff -u Python-2.6.4/Tools/scripts/fixdiv.py.orig Python-2.6.4/Tools/scripts/fixdiv.py
--- Python-2.6.4/Tools/scripts/fixdiv.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/fixdiv.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """fixdiv - tool to fix division operators.
 
diff -u Python-2.6.4/Tools/scripts/which.py.orig Python-2.6.4/Tools/scripts/which.py
--- Python-2.6.4/Tools/scripts/which.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/which.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Variant of "which".
 # On stderr, near and total misses are reported.
diff -u Python-2.6.4/Tools/scripts/suff.py.orig Python-2.6.4/Tools/scripts/suff.py
--- Python-2.6.4/Tools/scripts/suff.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/suff.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # suff
 #
diff -u Python-2.6.4/Tools/scripts/pysource.py.orig Python-2.6.4/Tools/scripts/pysource.py
--- Python-2.6.4/Tools/scripts/pysource.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/pysource.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """\
 List python source files.
diff -u Python-2.6.4/Tools/scripts/lll.py.orig Python-2.6.4/Tools/scripts/lll.py
--- Python-2.6.4/Tools/scripts/lll.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/lll.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Find symbolic links and show where they point to.
 # Arguments are directories to search; default is current directory.
diff -u Python-2.6.4/Tools/scripts/pickle2db.py.orig Python-2.6.4/Tools/scripts/pickle2db.py
--- Python-2.6.4/Tools/scripts/pickle2db.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/pickle2db.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 Synopsis: %(prog)s [-h|-b|-g|-r|-a|-d] [ picklefile ] dbfile
diff -u Python-2.6.4/Tools/scripts/texi2html.py.orig Python-2.6.4/Tools/scripts/texi2html.py
--- Python-2.6.4/Tools/scripts/texi2html.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/texi2html.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Convert GNU texinfo files into HTML, one file per node.
 # Based on Texinfo 2.14.
diff -u Python-2.6.4/Tools/scripts/mkreal.py.orig Python-2.6.4/Tools/scripts/mkreal.py
--- Python-2.6.4/Tools/scripts/mkreal.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/mkreal.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # mkreal
 #
diff -u Python-2.6.4/Tools/scripts/ftpmirror.py.orig Python-2.6.4/Tools/scripts/ftpmirror.py
--- Python-2.6.4/Tools/scripts/ftpmirror.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/ftpmirror.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Mirror a remote ftp subtree into a local directory tree.
 
diff -u Python-2.6.4/Tools/scripts/lfcr.py.orig Python-2.6.4/Tools/scripts/lfcr.py
--- Python-2.6.4/Tools/scripts/lfcr.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/lfcr.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 "Replace LF with CRLF in argument files.  Print names of changed files."
 
diff -u Python-2.6.4/Tools/scripts/ptags.py.orig Python-2.6.4/Tools/scripts/ptags.py
--- Python-2.6.4/Tools/scripts/ptags.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/ptags.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # ptags
 #
diff -u Python-2.6.4/Tools/scripts/copytime.py.orig Python-2.6.4/Tools/scripts/copytime.py
--- Python-2.6.4/Tools/scripts/copytime.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/copytime.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Copy one file's atime and mtime to another
 
diff -u Python-2.6.4/Tools/scripts/finddiv.py.orig Python-2.6.4/Tools/scripts/finddiv.py
--- Python-2.6.4/Tools/scripts/finddiv.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/finddiv.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """finddiv - a grep-like tool that looks for division operators.
 
diff -u Python-2.6.4/Tools/scripts/combinerefs.py.orig Python-2.6.4/Tools/scripts/combinerefs.py
--- Python-2.6.4/Tools/scripts/combinerefs.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/combinerefs.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 combinerefs path
diff -u Python-2.6.4/Tools/scripts/cvsfiles.py.orig Python-2.6.4/Tools/scripts/cvsfiles.py
--- Python-2.6.4/Tools/scripts/cvsfiles.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/cvsfiles.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Print a list of files that are mentioned in CVS directories.
 
diff -u Python-2.6.4/Tools/scripts/reindent.py.orig Python-2.6.4/Tools/scripts/reindent.py
--- Python-2.6.4/Tools/scripts/reindent.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/reindent.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Released to the public domain, by Tim Peters, 03 October 2000.
 
diff -u Python-2.6.4/Tools/scripts/checkpyc.py.orig Python-2.6.4/Tools/scripts/checkpyc.py
--- Python-2.6.4/Tools/scripts/checkpyc.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/checkpyc.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # Check that all ".pyc" files exist and are up-to-date
 # Uses module 'os'
 
diff -u Python-2.6.4/Tools/scripts/fixps.py.orig Python-2.6.4/Tools/scripts/fixps.py
--- Python-2.6.4/Tools/scripts/fixps.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/fixps.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # Fix Python script(s) to reference the interpreter via /usr/bin/env python.
 # Warning: this overwrites the file without making a backup.
diff -u Python-2.6.4/Tools/scripts/crlf.py.orig Python-2.6.4/Tools/scripts/crlf.py
--- Python-2.6.4/Tools/scripts/crlf.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/crlf.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 "Replace CRLF with LF in argument files.  Print names of changed files."
 
 import sys, os
diff -u Python-2.6.4/Tools/scripts/methfix.py.orig Python-2.6.4/Tools/scripts/methfix.py
--- Python-2.6.4/Tools/scripts/methfix.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/methfix.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Fix Python source files to avoid using
 #       def method(self, (arg1, ..., argn)):
diff -u Python-2.6.4/Tools/scripts/classfix.py.orig Python-2.6.4/Tools/scripts/classfix.py
--- Python-2.6.4/Tools/scripts/classfix.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/classfix.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # This script is obsolete -- it is kept for historical purposes only.
 #
diff -u Python-2.6.4/Tools/scripts/xxci.py.orig Python-2.6.4/Tools/scripts/xxci.py
--- Python-2.6.4/Tools/scripts/xxci.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/xxci.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # xxci
 #
diff -u Python-2.6.4/Tools/scripts/findnocoding.py.orig Python-2.6.4/Tools/scripts/findnocoding.py
--- Python-2.6.4/Tools/scripts/findnocoding.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/findnocoding.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """List all those Python files that require a coding directive
 
diff -u Python-2.6.4/Tools/scripts/fixnotice.py.orig Python-2.6.4/Tools/scripts/fixnotice.py
--- Python-2.6.4/Tools/scripts/fixnotice.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/fixnotice.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """(Ostensibly) fix copyright notices in files.
 
diff -u Python-2.6.4/Tools/scripts/eptags.py.orig Python-2.6.4/Tools/scripts/eptags.py
--- Python-2.6.4/Tools/scripts/eptags.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/eptags.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Create a TAGS file for Python programs, usable with GNU Emacs.
 
 usage: eptags pyfiles...
diff -u Python-2.6.4/Tools/scripts/linktree.py.orig Python-2.6.4/Tools/scripts/linktree.py
--- Python-2.6.4/Tools/scripts/linktree.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/linktree.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # linktree
 #
diff -u Python-2.6.4/Tools/scripts/checkappend.py.orig Python-2.6.4/Tools/scripts/checkappend.py
--- Python-2.6.4/Tools/scripts/checkappend.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/checkappend.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Released to the public domain, by Tim Peters, 28 February 2000.
 
diff -u Python-2.6.4/Tools/scripts/pindent.py.orig Python-2.6.4/Tools/scripts/pindent.py
--- Python-2.6.4/Tools/scripts/pindent.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/pindent.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # This file contains a class and a main program that perform three
 # related (though complimentary) formatting operations on Python
diff -u Python-2.6.4/Tools/scripts/google.py.orig Python-2.6.4/Tools/scripts/google.py
--- Python-2.6.4/Tools/scripts/google.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/google.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 import sys, webbrowser
 
diff -u Python-2.6.4/Tools/scripts/nm2def.py.orig Python-2.6.4/Tools/scripts/nm2def.py
--- Python-2.6.4/Tools/scripts/nm2def.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/nm2def.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """nm2def.py
 
 Helpers to extract symbols from Unix libs and auto-generate
diff -u Python-2.6.4/Tools/scripts/md5sum.py.orig Python-2.6.4/Tools/scripts/md5sum.py
--- Python-2.6.4/Tools/scripts/md5sum.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/md5sum.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Python utility to print MD5 checksums of argument files.
 """
diff -u Python-2.6.4/Tools/scripts/logmerge.py.orig Python-2.6.4/Tools/scripts/logmerge.py
--- Python-2.6.4/Tools/scripts/logmerge.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/logmerge.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Consolidate a bunch of CVS or RCS logs read from stdin.
 
diff -u Python-2.6.4/Tools/scripts/svneol.py.orig Python-2.6.4/Tools/scripts/svneol.py
--- Python-2.6.4/Tools/scripts/svneol.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/svneol.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 SVN helper script.
diff -u Python-2.6.4/Tools/scripts/rgrep.py.orig Python-2.6.4/Tools/scripts/rgrep.py
--- Python-2.6.4/Tools/scripts/rgrep.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/rgrep.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Reverse grep.
 
diff -u Python-2.6.4/Tools/scripts/h2py.py.orig Python-2.6.4/Tools/scripts/h2py.py
--- Python-2.6.4/Tools/scripts/h2py.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/h2py.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Read #define's and translate to Python code.
 # Handle #include statements.
diff -u Python-2.6.4/Tools/scripts/pathfix.py.orig Python-2.6.4/Tools/scripts/pathfix.py
--- Python-2.6.4/Tools/scripts/pathfix.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/pathfix.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Change the #! line occurring in Python scripts.  The new interpreter
 # pathname must be given with a -i option.
diff -u Python-2.6.4/Tools/scripts/byext.py.orig Python-2.6.4/Tools/scripts/byext.py
--- Python-2.6.4/Tools/scripts/byext.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/byext.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Show file statistics by extension."""
 
diff -u Python-2.6.4/Tools/scripts/ifdef.py.orig Python-2.6.4/Tools/scripts/ifdef.py
--- Python-2.6.4/Tools/scripts/ifdef.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/ifdef.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Selectively preprocess #ifdef / #ifndef statements.
 # Usage:
diff -u Python-2.6.4/Tools/scripts/treesync.py.orig Python-2.6.4/Tools/scripts/treesync.py
--- Python-2.6.4/Tools/scripts/treesync.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/treesync.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """Script to synchronize two source trees.
 
diff -u Python-2.6.4/Tools/scripts/ndiff.py.orig Python-2.6.4/Tools/scripts/ndiff.py
--- Python-2.6.4/Tools/scripts/ndiff.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/ndiff.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Module ndiff version 1.7.0
 # Released to the public domain 08-Dec-2000,
diff -u Python-2.6.4/Tools/scripts/findlinksto.py.orig Python-2.6.4/Tools/scripts/findlinksto.py
--- Python-2.6.4/Tools/scripts/findlinksto.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/findlinksto.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # findlinksto
 #
diff -u Python-2.6.4/Tools/scripts/objgraph.py.orig Python-2.6.4/Tools/scripts/objgraph.py
--- Python-2.6.4/Tools/scripts/objgraph.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/objgraph.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # objgraph
 #
diff -u Python-2.6.4/Tools/scripts/untabify.py.orig Python-2.6.4/Tools/scripts/untabify.py
--- Python-2.6.4/Tools/scripts/untabify.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/untabify.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 "Replace tabs with spaces in argument files.  Print names of changed files."
 
diff -u Python-2.6.4/Tools/scripts/db2pickle.py.orig Python-2.6.4/Tools/scripts/db2pickle.py
--- Python-2.6.4/Tools/scripts/db2pickle.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/db2pickle.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """
 Synopsis: %(prog)s [-h|-g|-b|-r|-a] dbfile [ picklefile ]
diff -u Python-2.6.4/Tools/scripts/parseentities.py.orig Python-2.6.4/Tools/scripts/parseentities.py
--- Python-2.6.4/Tools/scripts/parseentities.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/parseentities.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2.6
 """ Utility for parsing HTML entity definitions available from:
 
       http://www.w3.org/ as e.g.
diff -u Python-2.6.4/Tools/scripts/dutree.py.orig Python-2.6.4/Tools/scripts/dutree.py
--- Python-2.6.4/Tools/scripts/dutree.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/dutree.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 # Format du output in a tree shape
 
 import os, sys, errno
diff -u Python-2.6.4/Tools/scripts/pdeps.py.orig Python-2.6.4/Tools/scripts/pdeps.py
--- Python-2.6.4/Tools/scripts/pdeps.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/pdeps.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # pdeps
 #
diff -u Python-2.6.4/Tools/scripts/cleanfuture.py.orig Python-2.6.4/Tools/scripts/cleanfuture.py
--- Python-2.6.4/Tools/scripts/cleanfuture.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/cleanfuture.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 """cleanfuture [-d][-r][-v] path ...
 
diff -u Python-2.6.4/Tools/scripts/hotshotmain.py.orig Python-2.6.4/Tools/scripts/hotshotmain.py
--- Python-2.6.4/Tools/scripts/hotshotmain.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/hotshotmain.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: iso-8859-1 -*-
 
 """
diff -u Python-2.6.4/Tools/scripts/byteyears.py.orig Python-2.6.4/Tools/scripts/byteyears.py
--- Python-2.6.4/Tools/scripts/byteyears.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/byteyears.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Print the product of age and size of each file, in suitable units.
 #
diff -u Python-2.6.4/Tools/scripts/gprof2html.py.orig Python-2.6.4/Tools/scripts/gprof2html.py
--- Python-2.6.4/Tools/scripts/gprof2html.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/gprof2html.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python2.3
+#!/usr/bin/python2.6
 
 """Transform gprof(1) output into useful HTML."""
 
diff -u Python-2.6.4/Tools/scripts/fixcid.py.orig Python-2.6.4/Tools/scripts/fixcid.py
--- Python-2.6.4/Tools/scripts/fixcid.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/fixcid.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 
 # Perform massive identifier substitution on C source files.
 # This actually tokenizes the files (to some extent) so it can
diff -u Python-2.6.4/Tools/scripts/reindent-rst.py.orig Python-2.6.4/Tools/scripts/reindent-rst.py
--- Python-2.6.4/Tools/scripts/reindent-rst.py.orig	Mon Feb 21 10:46:42 2011
+++ Python-2.6.4/Tools/scripts/reindent-rst.py	Mon Feb 21 10:46:42 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # Make a reST file compliant to our pre-commit hook.
 # Currently just remove trailing whitespace.
diff -u Python-2.6.4/Mac/BuildScript/build-installer.py.orig Python-2.6.4/Mac/BuildScript/build-installer.py
--- Python-2.6.4/Mac/BuildScript/build-installer.py.orig	Mon Feb 21 10:46:43 2011
+++ Python-2.6.4/Mac/BuildScript/build-installer.py	Mon Feb 21 10:46:43 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """
 This script is used to build the "official unofficial" universal build on
 Mac OS X. It requires Mac OS X 10.4, Xcode 2.2 and the 10.4u SDK to do its
diff -u Python-2.6.4/Mac/scripts/zappycfiles.py.orig Python-2.6.4/Mac/scripts/zappycfiles.py
--- Python-2.6.4/Mac/scripts/zappycfiles.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Mac/scripts/zappycfiles.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python2.6
 """Recursively zap all .pyc and .pyo files"""
 import os
 import sys
diff -u Python-2.6.4/Mac/scripts/buildpkg.py.orig Python-2.6.4/Mac/scripts/buildpkg.py
--- Python-2.6.4/Mac/scripts/buildpkg.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Mac/scripts/buildpkg.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """buildpkg.py -- Build OS X packages for Apple's Installer.app.
 
diff -u Python-2.6.4/Mac/Tools/fixapplepython23.py.orig Python-2.6.4/Mac/Tools/fixapplepython23.py
--- Python-2.6.4/Mac/Tools/fixapplepython23.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Mac/Tools/fixapplepython23.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 """fixapplepython23 - Fix Apple-installed Python 2.3 (on Mac OS X 10.3)
 
 Python 2.3 (and 2.3.X for X<5) have the problem that building an extension
diff -u Python-2.6.4/Misc/find_recursionlimit.py.orig Python-2.6.4/Misc/find_recursionlimit.py
--- Python-2.6.4/Misc/find_recursionlimit.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Misc/find_recursionlimit.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Find the maximum recursion limit that prevents interpreter termination.
 
 This script finds the maximum safe recursion limit on a particular
diff -u Python-2.6.4/Parser/asdl_c.py.orig Python-2.6.4/Parser/asdl_c.py
--- Python-2.6.4/Parser/asdl_c.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Parser/asdl_c.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python2.6
 """Generate C code from an ASDL description."""
 
 # TO DO
diff -u Python-2.6.4/Doc/includes/email-alternative.py.orig Python-2.6.4/Doc/includes/email-alternative.py
--- Python-2.6.4/Doc/includes/email-alternative.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Doc/includes/email-alternative.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python2.6
 
 import smtplib
 
diff -u Python-2.6.4/Doc/includes/email-unpack.py.orig Python-2.6.4/Doc/includes/email-unpack.py
--- Python-2.6.4/Doc/includes/email-unpack.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Doc/includes/email-unpack.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """Unpack a MIME message into a directory of files."""
 
diff -u Python-2.6.4/Doc/includes/email-dir.py.orig Python-2.6.4/Doc/includes/email-dir.py
--- Python-2.6.4/Doc/includes/email-dir.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Doc/includes/email-dir.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """Send the contents of a directory as a MIME message."""
 
diff -u Python-2.6.4/Doc/tools/rstlint.py.orig Python-2.6.4/Doc/tools/rstlint.py
--- Python-2.6.4/Doc/tools/rstlint.py.orig	Mon Feb 21 10:46:44 2011
+++ Python-2.6.4/Doc/tools/rstlint.py	Mon Feb 21 10:46:44 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: utf-8 -*-
 
 # Check for stylistic and formal issues in .rst and .py
--- Python-2.6.8/Lib/lib2to3/tests/data/different_encoding.py.orig	2012-05-11 08:48:22.819694540 -0700
+++ Python-2.6.8/Lib/lib2to3/tests/data/different_encoding.py	2012-05-11 08:48:38.042397815 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # -*- coding: utf-8 -*-
 print u'ßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞ'
 
--- Python-2.6.8/Lib/whichdb.py.orig	2012-05-11 08:51:07.506663732 -0700
+++ Python-2.6.8/Lib/whichdb.py	2012-05-11 08:51:22.897281496 -0700
@@ -1,4 +1,4 @@
-# !/usr/bin/env python
+#!/usr/bin/python2.6
 """Guess which db package to use to open a db file."""
 
 import os
