--- links-1.00/html_r.c_   Mon Oct 17 03:49:38 2011
+++ links-1.00/html_r.c    Mon Oct 17 04:02:38 2011
@@ -1292,6 +1292,7 @@ void html_interpret(struct session *ses)
 	struct document_options o;
 	struct f_data_c *fd, *cf = NULL;
 	/*debug("I");*/
+	memset(&o, 0, sizeof(struct document_options));
 	if (!ses->screen) {
 		ses->screen = mem_alloc(sizeof(struct f_data_c));
 		memset(ses->screen, 0, sizeof(struct f_data_c));
