--- bacula-5.2.13/configure	2013-12-13 19:04:43.309330055 +0400
+++ bacula-5.2.13/configure	2013-12-13 19:05:11.392156575 +0400
@@ -24331,19 +24331,19 @@
 #
 # This code is very broken on older systems
 #
-#           MYSQL_CONFIG=`which mysql_config 2>/dev/null`
-#           if test "x${MYSQL_CONFIG}" != x; then
-#              MYSQL_BINDIR="${MYSQL_CONFIG%/*}"
-#              ${MYSQL_CONFIG} --variable=pkglibdir > /dev/null 2>&1
-#              if test $? = 0 ; then
-#                 MYSQL_LIBDIR=`${MYSQL_CONFIG} --variable=pkglibdir`
-#                 MYSQL_INCDIR=`${MYSQL_CONFIG} --variable=pkgincludedir`
-#              else
-#                 MYSQL_LIBDIR=`${MYSQL_CONFIG} --libs | sed -e 's/.*-L//' -e 's/ .*//'`
-#                 MYSQL_INCDIR=`${MYSQL_CONFIG} --include | sed -e 's/-I//'`
-#              fi
-#           elif test -f /usr/local/mysql/include/mysql/mysql.h; then
-           if test -f /usr/local/mysql/include/mysql/mysql.h; then
+           MYSQL_CONFIG=`which mysql_config 2>/dev/null`
+           if test "x${MYSQL_CONFIG}" != x; then
+              MYSQL_BINDIR="${MYSQL_CONFIG%/*}"
+              ${MYSQL_CONFIG} --variable=pkglibdir > /dev/null 2>&1
+              if test $? = 0 ; then
+                 MYSQL_LIBDIR=`${MYSQL_CONFIG} --variable=pkglibdir`
+                 MYSQL_INCDIR=`${MYSQL_CONFIG} --variable=pkgincludedir`
+              else
+                 MYSQL_LIBDIR=`${MYSQL_CONFIG} --libs | sed -e 's/.*-L//' -e 's/ .*//'`
+                 MYSQL_INCDIR=`${MYSQL_CONFIG} --include | sed -e 's/-I//'`
+              fi
+           elif test -f /usr/local/mysql/include/mysql/mysql.h; then
+#           if test -f /usr/local/mysql/include/mysql/mysql.h; then
               MYSQL_INCDIR=/usr/local/mysql/include/mysql
               if test -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.a \
                       -o -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.so; then
