--- bacula-5.2.13/configure.~2~	2018-01-27 01:29:29.316516424 +0000
+++ bacula-5.2.13/configure	2018-01-27 01:35:30.709153914 +0000
@@ -24776,7 +24754,7 @@
      fi
      SQLITE_INCLUDE=-I$SQLITE_INCDIR
      if test x$use_libtool != xno; then
-        SQLITE_LIBS="-R $SQLITE_LIBDIR -L$SQLITE_LIBDIR -lsqlite3"
+        SQLITE_LIBS="-L$SQLITE_LIBDIR -lsqlite3"
      else
         SQLITE_LIBS="-L$SQLITE_LIBDIR -lsqlite3"
      fi
