--- bacula-5.2.13/src/lib/bsnprintf.c.orig	2018-02-15 07:23:04.366499688 +0000
+++ bacula-5.2.13/src/lib/bsnprintf.c	2018-02-15 07:24:42.580747872 +0000
@@ -628,7 +628,7 @@
    return result;
 }
 
-static int64_t round(LDOUBLE value)
+static int64_t iround10(LDOUBLE value)
 {
    int64_t intpart;
 
@@ -700,7 +700,7 @@
    /* We "cheat" by converting the fractional part to integer by
     * multiplying by a factor of 10
     */
-   fracpart = round((pow10(max)) * (ufvalue - intpart));
+   fracpart = iround10((pow10(max)) * (ufvalue - intpart));
 
    if (fracpart >= pow10(max)) {
       intpart++;
