--- bacula-15.0.2/src/lib/xxhash.c	2024-03-22 10:48:41.000000000 +0100
+++ bacula-15.0.2/src/lib/xxhash.c.new	2024-06-09 17:39:17.357047406 +0200
@@ -40,4 +40,10 @@
 #define XXH_STATIC_LINKING_ONLY   /* access advanced declarations */
 #define XXH_IMPLEMENTATION   /* access definitions */
 
+#ifdef __cplusplus
+    #ifndef _Static_assert
+        #define _Static_assert static_assert
+    #endif
+#endif
+
 #include "xxhash.h"
