--- jsonrpclib-pelix-0.4.3.2/jsonrpclib/SimpleJSONRPCServer.py.orig
+++ jsonrpclib-pelix-0.4.3.2/jsonrpclib/SimpleJSONRPCServer.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # -- Content-Encoding: UTF-8 --
 """
 Defines a request dispatcher, a HTTP request handler, a HTTP server and a
--- jsonrpclib-pelix-0.4.3.2/jsonrpclib/jsonlib.py.orig
+++ jsonrpclib-pelix-0.4.3.2/jsonrpclib/jsonlib.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # -- Content-Encoding: UTF-8 --
 """
 Loads the "best" Python library available for the current interpreter and
--- jsonrpclib-pelix-0.4.3.2/jsonrpclib/jsonrpc.py.orig
+++ jsonrpclib-pelix-0.4.3.2/jsonrpclib/jsonrpc.py
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # -- Content-Encoding: UTF-8 --
 """
 ============================
--- jsonrpclib-pelix-0.4.3.2/jsonrpclib/threadpool.py.orig
+++ jsonrpclib-pelix-0.4.3.2/jsonrpclib/threadpool.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # -- Content-Encoding: UTF-8 --
 """
 Cached thread pool, inspired from Pelix/iPOPO Thread Pool
