--- percona-server-5.7.10-3/cmake/dtrace.cmake.1	2016-03-15 16:06:38.112157312 +0300
+++ percona-server-5.7.10-3/cmake/dtrace.cmake	2016-03-15 16:07:11.039390983 +0300
@@ -19,7 +19,7 @@
  MARK_AS_ADVANCED(DTRACE)
 
  # On FreeBSD, dtrace does not handle userland tracing yet
- IF(ENABLE_DTRACE AND NOT CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+ IF(ENABLE_DTRACE AND CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
    MESSAGE(FATAL_ERROR "dtrace is not supported on this system")
  ENDIF()
 
