--- librsvg-2.34.0/doc/Makefile.am.orig	2011-01-01 12:18:27.000000000 +0000
+++ librsvg-2.34.0/doc/Makefile.am	2011-07-18 13:56:37.034445084 +0100
@@ -43,7 +43,7 @@
 
 # Extra options to supply to gtkdoc-fixref. Not normally needed.
 # e.g. FIXXREF_OPTIONS=--extra-dir=../gdk-pixbuf/html --extra-dir=../gdk/html
-FIXXREF_OPTIONS = \
+FIXXREF_OPTIONS = --module=$(DOC_MODULE) \
 	--extra-dir=$(GLIB_PREFIX)/share/gtk-doc/html/glib \
 	--extra-dir=$(GLIB_PREFIX)/share/gtk-doc/html/gio \
 	--extra-dir=$(GDK_PIXBUF_PREFIX)/share/gtk-doc/html/gdk-pixbuf \
