--- brltty-4.2/Drivers/Braille/VisioBraille/vstp_main.c	2010-05-04 20:42:36.000000000 +0800
+++ brltty-4.2/Drivers/Braille/VisioBraille/vstp_main.c.mod	2012-06-14 00:11:33.561690986 +0800
@@ -99,7 +99,7 @@
    } else if (!strcmp(s,"vbs_ext")) {
     visiobases_ext = (char *) malloc(5);
     strncpy(visiobases_ext,c,4);
-    visiobases_ext[5]=0;
+    visiobases_ext[4]=0;
    } else if (!strcmp(s,"vbs_dir")) {
     visiobases_dir = (char *) malloc(strlen(c)+1);
     strcpy(visiobases_dir,c);
