With flake8-2020 installed we get "YTT" too.

--- flake8-5.0.4/tests/integration/test_plugins.py.orig
+++ flake8-5.0.4/tests/integration/test_plugins.py
@@ -104,7 +104,7 @@
 
     args = aggregator.aggregate_options(option_manager, cfg, cfg_dir, argv)
 
-    assert args.extended_default_select == ["XE", "C90", "F", "E", "W"]
+    assert args.extended_default_select == ["YTT", "XE", "C90", "F", "E", "W"]
     assert args.anopt == "foo"
 
 
