--- unixODBC-2.3.13/odbcinst/SQLGetPrivateProfileString.c.orig
+++ unixODBC-2.3.13/odbcinst/SQLGetPrivateProfileString.c
@@ -39,7 +39,7 @@
     return pth_mutex_acquire( mutex, 0, NULL );
 }
 
-static int mutex_exit( pth_mutex_t *mutex )
+static int odbc_mutex_exit( pth_mutex_t *mutex )
 {
     return pth_mutex_release( mutex );
 }
@@ -55,7 +55,7 @@
     return pthread_mutex_lock( mutex );
 }
 
-static int mutex_exit( pthread_mutex_t *mutex )
+static int odbc_mutex_exit( pthread_mutex_t *mutex )
 {
     return pthread_mutex_unlock( mutex );
 }
@@ -71,7 +71,7 @@
     return mutex_lock( mutex );
 }
 
-static int mutex_exit( mutex_t *mutex )
+static int odbc_mutex_exit( mutex_t *mutex )
 {
     return mutex_unlock( mutex );
 }
@@ -79,7 +79,7 @@
 #else
 
 #define mutex_entry(x)
-#define mutex_exit(x)
+#define odbc_mutex_exit(x)
 
 #endif
 
@@ -296,7 +296,7 @@
 	rval = _check_ini_cache( ret, pszSection, pszEntry, pszDefault,
 			pRetBuffer, nRetBuffer, pszFileName );
 
-	mutex_exit( &mutex_ini );
+	odbc_mutex_exit( &mutex_ini );
 
 	return rval;
 }
@@ -324,7 +324,7 @@
 			pRetBuffer, nRetBuffer, pszFileName );
 	}
 
-	mutex_exit( &mutex_ini );
+	odbc_mutex_exit( &mutex_ini );
 }
 
 void __clear_ini_cache( void ) 
@@ -333,7 +333,7 @@
 
     _clear_ini_cache();
 
-	mutex_exit( &mutex_ini );
+	odbc_mutex_exit( &mutex_ini );
 }
 
 #else
