--- netatalk-3.0.5/configure.~1~	2013-08-14 10:56:37.000000000 +0100
+++ netatalk-3.0.5/configure	2013-08-31 16:41:15.299975383 +0100
@@ -16220,9 +16220,9 @@
 		for ssldir in "" $tryssldir /usr /usr/local/openssl /usr/$atalk_libname/openssl /usr/local/ssl /usr/$atalk_libname/ssl /usr/local /usr/pkg /opt /opt/openssl /usr/local/ssl ; do
 			if test -f "$ssldir/include/openssl/cast.h" ; then
 				SSL_CFLAGS="$SSL_CFLAGS -I$ssldir/include -I$ssldir/include/openssl"
-				SSL_LIBS="$SSL_LIBS -L$ssldir/$atalk_libname -L$ssldir -lcrypto"
+				SSL_LIBS="$SSL_LIBS -L$ssldir/$atalk_libname -lcrypto"
 				if test "x$need_dash_r" = "xyes"; then
-					SSL_LIBS="$SSL_LIBS -R$ssldir/$atalk_libname -R$ssldir"
+					SSL_LIBS="$SSL_LIBS -R$ssldir/$atalk_libname "
 				fi
 				{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ssldir (enabling RANDNUM and DHX support)" >&5
 $as_echo "$ssldir (enabling RANDNUM and DHX support)" >&6; }
@@ -17135,7 +17135,7 @@
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $KRB5_CONFIG" >&5
 $as_echo "$KRB5_CONFIG" >&6; }
       KRB5_CFLAGS="`$KRB5_CONFIG --cflags krb5`"
-      KRB5_LIBS="`$KRB5_CONFIG --libs krb5`"
+      KRB5_LIBS="-lkrb5"
 
 
       with_kerberos="yes"
