--- netatalk-3.0.5/distrib/initscripts/Makefile.in.orig	2013-08-31 15:09:16.887215295 +0100
+++ netatalk-3.0.5/distrib/initscripts/Makefile.in	2013-08-31 15:09:58.415438460 +0100
@@ -725,10 +725,10 @@
 @USE_NETBSD_TRUE@uninstall-startup: uninstall-am
 
 @USE_SOLARIS_TRUE@install-data-hook:
-@USE_SOLARIS_TRUE@	svccfg import netatalk.xml
+#@USE_SOLARIS_TRUE@	svccfg import netatalk.xml
 
 @USE_SOLARIS_TRUE@uninstall-startup:
-@USE_SOLARIS_TRUE@	svccfg delete network/netatalk
+#@USE_SOLARIS_TRUE@	svccfg delete network/netatalk
 
 @USE_GENTOO_TRUE@$(sysv_SCRIPTS): rc.gentoo
 @USE_GENTOO_TRUE@	cp -f rc.gentoo $(sysv_SCRIPTS)
