--- webkitgtk-2.10.4/Source/bmalloc/bmalloc/VMAllocate.h.1	2015-12-06 03:49:00.283191588 +0300
+++ webkitgtk-2.10.4/Source/bmalloc/bmalloc/VMAllocate.h	2015-12-06 03:51:25.142548675 +0300
@@ -133,6 +133,8 @@
     vmValidate(p, vmSize);
 #if BOS(DARWIN)
     SYSCALL(madvise(p, vmSize, MADV_FREE_REUSABLE));
+#elif defined(__sun__)
+    SYSCALL(posix_madvise(p, vmSize, POSIX_MADV_DONTNEED));
 #else
     SYSCALL(madvise(p, vmSize, MADV_DONTNEED));
 #endif
@@ -143,6 +145,8 @@
     vmValidate(p, vmSize);
 #if BOS(DARWIN)
     SYSCALL(madvise(p, vmSize, MADV_FREE_REUSE));
+#elif defined(__sun__)
+    SYSCALL(posix_madvise(p, vmSize, POSIX_MADV_NORMAL));
 #else
     SYSCALL(madvise(p, vmSize, MADV_NORMAL));
 #endif
