--- webkitgtk-2.10.4/Source/WTF/wtf/InlineASM.h.1	2015-12-06 10:33:06.234203770 +0300
+++ webkitgtk-2.10.4/Source/WTF/wtf/InlineASM.h	2015-12-06 10:34:39.441135763 +0300
@@ -81,6 +81,7 @@
 #define LOCAL_LABEL_STRING(name) "L" #name
 #elif   OS(LINUX)               \
      || OS(FREEBSD)             \
+     || OS(SOLARIS)             \
      || OS(OPENBSD)             \
      || OS(HURD)                \
      || OS(NETBSD)
