--- webkitgtk-2.10.4/Source/WebCore/platform/FileSystem.cpp.1	2015-12-06 19:00:51.231072285 +0300
+++ webkitgtk-2.10.4/Source/WebCore/platform/FileSystem.cpp	2015-12-06 19:01:04.552158882 +0300
@@ -164,7 +164,7 @@
         return;
     }
 
-    void* data = mmap(0, size, PROT_READ, MAP_FILE | MAP_SHARED, fd, 0);
+    void* data = mmap(0, size, PROT_READ, MAP_SHARED, fd, 0);
     close(fd);
 
     if (data == MAP_FAILED) {
