--- webkitgtk-2.32.3/Source/WebCore/platform/gtk/po/CMakeLists.txt.orig	2021-02-26 10:57:13.000000000 +0000
+++ webkitgtk-2.32.3/Source/WebCore/platform/gtk/po/CMakeLists.txt	2021-10-06 18:05:41.722968436 +0000
@@ -33,5 +33,5 @@
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/..
 )
 
-file(GLOB po_files *.po)
-GETTEXT_CREATE_TRANSLATIONS(${pot_file} ALL ${po_files})
+#file(GLOB po_files *.po)
+#GETTEXT_CREATE_TRANSLATIONS(${pot_file} ALL ${po_files})
