Disable for Solaris

Not suitable for upstream

diff --git a/Source/WTF/wtf/PlatformUse.h b/Source/WTF/wtf/PlatformUse.h
index ab14b07..2bf4a3c 100644
--- a/Source/WTF/wtf/PlatformUse.h
+++ b/Source/WTF/wtf/PlatformUse.h
@@ -96,8 +96,10 @@
 #endif
 
 #if PLATFORM(COCOA) || (PLATFORM(GTK) || PLATFORM(WPE))
+#if !OS(SOLARIS)
 #define USE_FILE_LOCK 1
 #endif
+#endif
 
 #if PLATFORM(COCOA)
 #define USE_FOUNDATION 1
