Fixes release build

--- webkitgtk-2.14.5/Source/WebCore/html/canvas/WebGLRenderingContext.cpp.1	2017-06-01 16:20:31.844077995 +0300
+++ webkitgtk-2.14.5/Source/WebCore/html/canvas/WebGLRenderingContext.cpp	2017-06-01 16:23:09.860282933 +0300
@@ -60,6 +60,7 @@
 #include <JavaScriptCore/GenericTypedArrayViewInlines.h>
 #include <JavaScriptCore/JSCJSValueInlines.h>
 #include <JavaScriptCore/JSCellInlines.h>
+#include <heap/HeapInlines.h>
 #include <JavaScriptCore/JSGenericTypedArrayViewInlines.h>
 
 namespace WebCore {
