--- webkitgtk-2.48.1/Source/WebKit/CMakeLists.txt.orig
+++ webkitgtk-2.48.1/Source/WebKit/CMakeLists.txt
@@ -1108,7 +1108,7 @@
 
     install(TARGETS WebKit WebProcess NetworkProcess
         LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
-        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}"
+        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}"
     )
     if (ENABLE_GPU_PROCESS)
         install(TARGETS GPUProcess
--- webkitgtk-2.48.1/Source/JavaScriptCore/shell/CMakeLists.txt.orig
+++ webkitgtk-2.48.1/Source/JavaScriptCore/shell/CMakeLists.txt
@@ -96,7 +96,7 @@
 WEBKIT_EXECUTABLE(jsc)
 
 if (SHOULD_INSTALL_JS_SHELL)
-    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}")
+    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}")
 endif ()
 
 if (DEVELOPER_MODE)
