--- webkitgtk-2.32.3/Source/WebKit/CMakeLists.txt.orig	2021-03-25 14:27:32.000000000 +0000
+++ webkitgtk-2.32.3/Source/WebKit/CMakeLists.txt	2021-09-18 10:31:34.558797510 +0000
@@ -595,6 +595,6 @@
 
     install(TARGETS WebKit WebProcess NetworkProcess
         LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
-        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}"
+        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}"
     )
 endif ()
--- webkitgtk-2.32.3/Source/JavaScriptCore/shell/CMakeLists.txt.orig	2021-02-26 10:57:05.000000000 +0000
+++ webkitgtk-2.32.3/Source/JavaScriptCore/shell/CMakeLists.txt	2021-09-18 10:33:58.433350845 +0000
@@ -93,7 +93,7 @@
 WEBKIT_EXECUTABLE(jsc)
 
 if (SHOULD_INSTALL_JS_SHELL)
-    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}")
+    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}")
 endif ()
 
 if (DEVELOPER_MODE)
