--- webkitgtk-2.53.4/Source/WebKit/CMakeLists.txt.old
+++ webkitgtk-2.53.4/Source/WebKit/CMakeLists.txt
@@ -1513,7 +1513,7 @@
 
     install(TARGETS WebKit WebProcess NetworkProcess
         LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
-        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}"
+        RUNTIME DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}"
         FRAMEWORK DESTINATION "${LIB_INSTALL_DIR}"
         BUNDLE DESTINATION "${LIBEXEC_INSTALL_DIR}"
     )
--- webkitgtk-2.48.1/Source/JavaScriptCore/shell/CMakeLists.txt.orig
+++ webkitgtk-2.48.1/Source/JavaScriptCore/shell/CMakeLists.txt
@@ -96,7 +96,7 @@
 WEBKIT_EXECUTABLE(jsc)
 
 if (SHOULD_INSTALL_JS_SHELL)
-    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}")
+    install(TARGETS jsc DESTINATION "${LIBEXEC_INSTALL_DIR}/webkit2gtk-${WEBKITGTK_API_VERSION}")
 endif ()
 
 if (DEVELOPER_MODE)
