--- webkitgtk-2.52.2/Source/WTF/wtf/glib/FileSystemGlib.cpp.old
+++ webkitgtk-2.52.2/Source/WTF/wtf/glib/FileSystemGlib.cpp
@@ -93,7 +93,13 @@
 CString currentExecutablePath()
 {
     static char readLinkBuffer[PATH_MAX];
+#if OS(SOLARIS)
+    static char path[PATH_MAX];
+    sprintf(path, "/proc/%ld/path/a.out", static_cast<long>(getpid()));
+    ssize_t result = readlink(path, readLinkBuffer, PATH_MAX);
+#else
     ssize_t result = readlink(_PROC_CURPROC_PATH, readLinkBuffer, PATH_MAX);
+#endif
     if (result <= 0)
         return { };
     return CString(unsafeMakeSpan(readLinkBuffer, static_cast<size_t>(result)));
