--- pyroma-4.1/pyroma/tests.py.orig
+++ pyroma-4.1/pyroma/tests.py
@@ -9,6 +9,8 @@
 from pyroma import projectdata, distributiondata, pypidata
 from pyroma.ratings import rate
 
+import pytest
+
 long_description = resource_string(__name__, os.path.join("testdata", "complete", "README.txt"))
 if not isinstance(long_description, str):
     long_description = long_description.decode()
@@ -161,6 +163,7 @@
             ),
         )
 
+    @pytest.mark.skip(reason="always fails: https://github.com/regebro/pyroma/issues/95")
     def test_minimal(self):
         rating = self._get_file_rating("minimal")
 
@@ -220,6 +223,7 @@
             ),
         )
 
+    @pytest.mark.skip(reason="always fails: https://github.com/regebro/pyroma/issues/95")
     def test_custom_test(self):
         rating = self._get_file_rating("custom_test")
 
