--- github/gnome_connection_manager.py	2020-04-15 11:56:25.772356289 +0000
+++ fixed/gnome_connection_manager.py	2020-04-15 12:39:05.024238615 +0000
@@ -1145,7 +1145,10 @@
             v.set_scrollback_lines(conf.BUFFER_LINES)
             #if v.get_emulation() != os.getenv("TERM"):
             #    os.environ['TERM'] = v.get_emulation()
+            ### As of Apr 2020 there are issues in OI use of xterm* in gcm
+            ### https://github.com/kuthulux/gnome-connection-manager/issues/5
-            os.environ['TERM'] = 'xterm-256color'
+            #os.environ['TERM'] = 'xterm-256color'
+            os.environ['TERM'] = 'vt220'
             
             if isinstance(host, str):
                 host = Host('', host) 
