--- confuse-2.6/tests/Makefile.in	Sat Dec 29 02:56:13 2007
+++ confuse-2.6/tests/Makefile.in.new	Tue Jul 12 10:27:06 2011
@@ -91,10 +91,12 @@
 suite_single_SOURCES = suite_single.c
 suite_single_OBJECTS = suite_single.$(OBJEXT)
 suite_single_LDADD = $(LDADD)
+suite_single_LIBADD = $(LIBADD)
 suite_single_DEPENDENCIES = ../src/libconfuse.la $(am__DEPENDENCIES_1)
 suite_validate_SOURCES = suite_validate.c
 suite_validate_OBJECTS = suite_validate.$(OBJEXT)
 suite_validate_LDADD = $(LDADD)
+suite_validate_LIBADD = $(LIBADD)
 suite_validate_DEPENDENCIES = ../src/libconfuse.la \
 	$(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
@@ -242,6 +244,7 @@
 	single_title_sections quote_before_print
 
 LDADD = -L../src ../src/libconfuse.la $(LTLIBINTL)
+LIBADD = -lsocket -lnsl
 EXTRA_DIST = a.conf b.conf check_confuse.h
 CLEANFILES = *~
 @cc_is_gcc_TRUE@AM_CFLAGS = -Wall -Werror
@@ -308,10 +311,10 @@
 	$(LINK) $(suite_list_LDFLAGS) $(suite_list_OBJECTS) $(suite_list_LDADD) $(LIBS)
 suite_single$(EXEEXT): $(suite_single_OBJECTS) $(suite_single_DEPENDENCIES) 
 	@rm -f suite_single$(EXEEXT)
-	$(LINK) $(suite_single_LDFLAGS) $(suite_single_OBJECTS) $(suite_single_LDADD) $(LIBS)
+	$(LINK) $(suite_single_LDFLAGS) $(suite_single_OBJECTS) $(suite_single_LIBADD) $(suite_single_LDADD) $(LIBS)
 suite_validate$(EXEEXT): $(suite_validate_OBJECTS) $(suite_validate_DEPENDENCIES) 
 	@rm -f suite_validate$(EXEEXT)
-	$(LINK) $(suite_validate_LDFLAGS) $(suite_validate_OBJECTS) $(suite_validate_LDADD) $(LIBS)
+	$(LINK) $(suite_validate_LDFLAGS) $(suite_validate_OBJECTS) $(suite_validate_LIBADD) $(suite_validate_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
