diff --git a/src/Makefile.am b/src/Makefile.am
index fbbca45..eb24a4d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -75,7 +75,7 @@ AM_CFLAGS = $(XI_CFLAGS) \
             $(CWARNFLAGS)
 
 
-libXi_la_LDFLAGS = -version-number 6:1:0 -no-undefined
+libXi_la_LDFLAGS = -version-number 5:0:0 -no-undefined
 
 libXiincludedir = $(includedir)/X11/extensions
 libXiinclude_HEADERS = $(top_srcdir)/include/X11/extensions/XInput.h \
diff --git a/src/Makefile.in b/src/Makefile.in
index 2f58bbb..6c0f927 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -442,7 +442,7 @@ AM_CFLAGS = $(XI_CFLAGS) \
             $(MALLOC_ZERO_CFLAGS) \
             $(CWARNFLAGS)
 
-libXi_la_LDFLAGS = -version-number 6:1:0 -no-undefined
+libXi_la_LDFLAGS = -version-number 5:0:0 -no-undefined
 libXiincludedir = $(includedir)/X11/extensions
 libXiinclude_HEADERS = $(top_srcdir)/include/X11/extensions/XInput.h \
 		       $(top_srcdir)/include/X11/extensions/XInput2.h
