diff -urN gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in gnome-mime-data-2.4.1/gnome-vfs.keys.in
--- gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in	2004-10-05 16:18:48.516302000 +0900
+++ gnome-mime-data-2.4.1/gnome-vfs.keys.in	2004-10-05 16:20:58.123719000 +0900
@@ -957,7 +957,12 @@
 
 application/x-shockwave-flash
 	_description=Macromedia Flash file
+	default_action_type=application
+	short_list_application_ids_for_novice_user_level=mozilla
+	short_list_application_ids_for_intermediate_user_level=mozilla
+	short_list_application_ids_for_advanced_user_level=mozilla
 	_category=Documents/World Wide Web
+	use_category_default=yes
 
 application/x-slp
 	_description=Stampede package
