diff -urN gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in gnome-mime-data-2.4.1/gnome-vfs.keys.in
--- gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in	2004-10-05 16:11:26.736741000 +0900
+++ gnome-mime-data-2.4.1/gnome-vfs.keys.in	2004-10-05 16:11:52.570389000 +0900
@@ -1869,14 +1869,14 @@
 text/html
 	_description=HTML page
 	default_action_type=application
-	default_application_id=epiphany
+	default_application_id=mozilla
 	short_list_component_iids=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:Nautilus_Text_View
 	short_list_component_iids_for_novice_user_level=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_intermediate_user_level=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_advanced_user_level=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
-	short_list_application_ids_for_novice_user_level=epiphany,galeon,mozilla,netscape
-	short_list_application_ids_for_intermediate_user_level=epiphany,galeon,mozilla,netscape
-	short_list_application_ids_for_advanced_user_level=epiphany,galeon,mozilla,netscape,lynx
+	short_list_application_ids_for_novice_user_level=mozilla,epiphany,galeon,netscape
+	short_list_application_ids_for_intermediate_user_level=mozilla,epiphany,galeon,netscape
+	short_list_application_ids_for_advanced_user_level=mozilla,epiphany,galeon,netscape,lynx
 	_category=Documents/World Wide Web
 	use_category_default=yes
 
