diff -urN gnome-mime-data-2.4.1-orig/gnome-vfs.applications gnome-mime-data-2.4.1/gnome-vfs.applications
--- gnome-mime-data-2.4.1-orig/gnome-vfs.applications	2004-10-05 16:22:37.503171000 +0900
+++ gnome-mime-data-2.4.1/gnome-vfs.applications	2004-10-05 16:23:01.255468000 +0900
@@ -250,6 +250,14 @@
 	requires_terminal=true
 	mime_types=application/x-java-byte-code
 
+java-jar
+	command=java -jar
+	name=Java Interpreter
+	can_open_multiple_files=false
+	expects_uris=false
+	requires_terminal=false
+	mime_types=application/x-java-archive
+
 javac
 	command=javac
 	name=Java compiler
diff -urN gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in gnome-mime-data-2.4.1/gnome-vfs.keys.in
--- gnome-mime-data-2.4.1-orig/gnome-vfs.keys.in	2004-10-05 16:22:37.459408000 +0900
+++ gnome-mime-data-2.4.1/gnome-vfs.keys.in	2004-10-05 16:23:01.267344000 +0900
@@ -715,6 +715,14 @@
 	_description=Java byte code
 	_category=Software Development
 
+application/x-java-archive
+	_description=Java archive
+	default_action_type=application
+	short_list_application_ids_for_novice_user_level=java-jar
+        short_list_application_ids_for_intermediate_user_level=java-jar
+        short_list_application_ids_for_advanced_user_level=java-jar
+	_category=Packages
+
 application/x-jbuilder-project
 	_description=JBuilder Project
 	default_action_type=application
