--- gnome-mime-data-2.4.1/gnome-vfs-mime-magic	2003-12-23 00:27:08.000000000 +0530
+++ gnome-mime-data-2.4.1-new/gnome-vfs-mime-magic	2005-05-10 19:22:34.758890736 +0530
@@ -228,6 +228,7 @@
 0	string		BMxxxx\000\000 &0xffff00000000ffff	image/bmp
 0	string		AT&TFORM				image/vnd.djvu
 0	string		Return-Path:				message/rfc822
+0	string		Return-path:				message/rfc822
 0	string		Path:					message/news
 0	string		Xref:					message/news
 0	string		From:					message/rfc822
