diff -pNaur gnome-mime-data-2.4.1.old/gnome-vfs.keys.in gnome-mime-data-2.4.1/gnome-vfs.keys.in
--- gnome-mime-data-2.4.1.old/gnome-vfs.keys.in	2004-09-07 17:50:46.000000000 +0530
+++ gnome-mime-data-2.4.1/gnome-vfs.keys.in	2004-09-07 17:49:24.000000000 +0530
@@ -67,9 +67,9 @@ application/pdf
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_PDF_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_PDF_Control
 	short_list_component_iids_for_advanced_user_level=OAFIID:GNOME_PDF_Control
-	short_list_application_ids_for_novice_user_level=acroread,gpdf,xpdf,ggv,gv
-	short_list_application_ids_for_intermediate_user_level=acroread,gpdf,xpdf,ggv,gv
-	short_list_application_ids_for_advanced_user_level=acroread,gpdf,xpdf,ggv,gv
+	short_list_application_ids_for_novice_user_level=acroread,gpdf,xpdf,gv
+	short_list_application_ids_for_intermediate_user_level=acroread,gpdf,xpdf,gv
+	short_list_application_ids_for_advanced_user_level=acroread,gpdf,xpdf,gv
 	_category=Documents/Published Materials
 	use_category_default=yes
 
