diff -urN sg3_utils-1.28/lib/Makefile.am sg3_utils-1.28.new/lib/Makefile.am
--- sg3_utils-1.28/lib/Makefile.am	2011-03-08 15:16:10.684395224 +0800
+++ sg3_utils-1.28.new/lib/Makefile.am	2011-03-08 15:16:52.811149215 +0800
@@ -133,7 +133,7 @@
 endif
 
 
-AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W
+AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -v
 # AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W -pedantic -std=c99
 
 lib_LTLIBRARIES = libsgutils2.la
diff -urN sg3_utils-1.28/lib/Makefile.in sg3_utils-1.28.new/lib/Makefile.in
--- sg3_utils-1.28/lib/Makefile.in	2011-03-08 15:16:10.683842532 +0800
+++ sg3_utils-1.28.new/lib/Makefile.in	2011-03-08 15:16:39.931893403 +0800
@@ -330,7 +330,7 @@
 @OS_WIN32_MINGW_TRUE@	sg_pt_solaris.c \
 @OS_WIN32_MINGW_TRUE@	getopt_long.c
 
-AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W
+AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -v
 # AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W -pedantic -std=c99
 lib_LTLIBRARIES = libsgutils2.la
 libsgutils2_la_LDFLAGS = -version-info 2:0:0
diff -urN sg3_utils-1.28/src/Makefile.am sg3_utils-1.28.new/src/Makefile.am
--- sg3_utils-1.28/src/Makefile.am	2011-03-08 15:16:10.677707201 +0800
+++ sg3_utils-1.28.new/src/Makefile.am	2011-03-08 15:17:03.471143277 +0800
@@ -119,7 +119,7 @@
 endif
 
 
-AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W
+AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -v
 # AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W -pedantic -std=c99
 
 sg_dd_SOURCES =	sg_dd.c
diff -urN sg3_utils-1.28/src/Makefile.in sg3_utils-1.28.new/src/Makefile.in
--- sg3_utils-1.28/src/Makefile.in	2011-03-08 15:16:10.676305953 +0800
+++ sg3_utils-1.28.new/src/Makefile.in	2011-03-08 15:17:13.216078187 +0800
@@ -584,7 +584,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W
+AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -v
 # AM_CFLAGS = -I ../include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wall -W -pedantic -std=c99
 sg_dd_SOURCES = sg_dd.c
 sg_dd_LDADD = ../lib/libsgutils2.la @os_libs@
