--- ./magick/property.c.orig	2015-08-01 00:48:57.387497777 +0200
+++ ./magick/property.c	2015-08-01 00:46:55.767596904 +0200
@@ -81,8 +81,8 @@
 #include "magick/version.h"
 #include "magick/xml-tree.h"
 #if defined(MAGICKCORE_LCMS_DELEGATE)
-#if defined(MAGICKCORE_HAVE_LCMS_LCMS2_H)
-#include <lcms/lcms2.h>
+#if defined(MAGICKCORE_HAVE_LCMS2_LCMS2_H)
+#include <lcms2/lcms2.h>
 #elif defined(MAGICKCORE_HAVE_LCMS2_H)
 #include "lcms2.h"
 #elif defined(MAGICKCORE_HAVE_LCMS_LCMS_H)
--- ./magick/profile.c.orig	2015-08-01 00:49:16.715396864 +0200
+++ ./magick/profile.c	2015-08-01 00:47:03.769820772 +0200
@@ -65,9 +65,9 @@
 #include "magick/token.h"
 #include "magick/utility.h"
 #if defined(MAGICKCORE_LCMS_DELEGATE)
-#if defined(MAGICKCORE_HAVE_LCMS_LCMS2_H)
+#if defined(MAGICKCORE_HAVE_LCMS2_LCMS2_H)
 #include <wchar.h>
-#include <lcms/lcms2.h>
+#include <lcms2/lcms2.h>
 #elif defined(MAGICKCORE_HAVE_LCMS2_H)
 #include <wchar.h>
 #include "lcms2.h"
