--- ./magick/method-attribute.h.orig	2015-08-01 00:16:05.432590217 +0200
+++ ./magick/method-attribute.h	2015-08-01 00:16:44.406557378 +0200
@@ -108,7 +108,7 @@
 
 #if defined(MAGICKCORE_HAVE___ATTRIBUTE__)
 #  define magick_aligned(x,y)  x __attribute__((aligned(y)))
-#  define magick_attribute  __attribute__
+#  define magick_attribute(x)
 #  define magick_unused(x)  magick_unused_ ## x __attribute__((unused))
 #  define magick_unreferenced(x)  /* nothing */
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__CYGWIN__)
