--- tcllib-1.17.orig/Makefile.in	2015-02-12 04:16:15.000000000 +0300
+++ tcllib-1.17/Makefile.in	2016-05-02 23:12:20.217047649 +0300
@@ -54,7 +54,7 @@
		-no-examples -no-html \
		-pkg-path   `$(CYGPATH) $(DESTDIR)$(pkglibdir)` \
		-app-path   `$(CYGPATH) $(DESTDIR)$(bindir)` \
-		-nroff-path `$(CYGPATH) $(DESTDIR)$(mandir)/mann` \
+		-nroff-path `$(CYGPATH) $(DESTDIR)$(mandir)/man1t` \
		-no-wait -no-gui

 install-libraries:
