--- Drivers/Postgre7.1/Makefile.in	2010-02-24 11:52:22.000000000 -0500
+++ Drivers/Postgre7.1/Makefile.in	2010-09-15 12:35:33.089913000 -0400
@@ -652,8 +652,6 @@
 	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
 	tags uninstall uninstall-am uninstall-libLTLIBRARIES
 
-	-module
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
