--- fontconfig-2.13.0/conf.d/30-metric-aliases.conf.orig	2017-11-20 13:02:20.000000000 +0000
+++ fontconfig-2.13.0/conf.d/30-metric-aliases.conf	2018-03-08 16:30:25.117636116 +0000
@@ -622,13 +622,22 @@
 	  <accept>
 	    <family>Carlito</family>
 	  </accept>
-  </alias>
+  	</alias>
 
 	<alias binding="same">
 	  <family>Symbol</family>
 	  <accept>
 	    <family>SymbolNeu</family>
 	  </accept>
-  </alias>
+  	</alias>
+
+        <!-- StarOffice formerly included Arial Narrow - Solaris includes Arial Narrow MT instead. -->
+        <alias>
+          <family>Arial Narrow</family>
+            <accept>
+              <family>Arial Narrow MT</family>
+              <family>Nimbus Sans L Condensed</family>
+            </accept>
+        </alias>
 
 </fontconfig>
