diff -urp -x '*~' -x '*.orig' conf.d/30-metric-aliases.conf conf.d/30-metric-aliases.conf
--- a/conf.d/30-metric-aliases.conf	2009-11-16 09:24:52.000000000 -0800
+++ b/conf.d/30-metric-aliases.conf	2010-05-10 15:09:19.723404099 -0700
@@ -207,5 +207,13 @@
 	  </accept>
 	</alias>
 
+	<!-- StarOffice formerly included Arial Narrow - Solaris includes Arial Narrow MT instead. -->
+	<alias>
+	  <family>Arial Narrow</family>
+	    <accept>
+	      <family>Arial Narrow MT</family>
+	      <family>Nimbus Sans L Condensed</family>
+	    </accept>
+	</alias>
 
 </fontconfig>
