--- a/conf.d/30-metric-aliases.conf	Wed Nov  5 20:13:44 2014
+++ b/conf.d/30-metric-aliases.conf	Fri Jul 31 11:22:23 2015
@@ -600,7 +600,7 @@
 	  <accept>
 	    <family>Carlito</family>
 	  </accept>
-  </alias>
+  	</alias>
 
 	<alias binding="same">
 	  <family>Symbol</family>
@@ -607,6 +607,15 @@
 	  <accept>
 	    <family>SymbolNeu</family>
 	  </accept>
-  </alias>
+  	</alias>
 
+        <!-- StarOffice formerly included Arial Narrow - Solaris includes Arial Narrow MT instead. -->
+        <alias>
+          <family>Arial Narrow</family>
+            <accept>
+              <family>Arial Narrow MT</family>
+              <family>Nimbus Sans L Condensed</family>
+            </accept>
+        </alias>
+
 </fontconfig>
