--- tifficc/tiffdiff.c	2009-10-30 08:57:46.000000000 -0700
+++ tifficc/tiffdiff.c	2013-08-05 16:21:59.504764730 -0700
@@ -633,7 +633,7 @@
     cmsIT8SetSheetType(hIT8, "TIFFDIFF");
     
    
-    sprintf(Buffer, "Differences between %s and %s", TiffName1, TiffName2);
+    snprintf(Buffer, sizeof(Buffer), "Differences between %s and %s", TiffName1, TiffName2);
   
     cmsIT8SetComment(hIT8, Buffer);
 
