--- sound-juicer-2.32.0/data/sound-juicer.desktop.in.in.1	2016-03-11 12:38:12.801760453 +0300
+++ sound-juicer-2.32.0/data/sound-juicer.desktop.in.in	2016-03-11 12:38:27.375942704 +0300
@@ -5,7 +5,7 @@
 _GenericName=CD Ripper
 # SUN_BRANDING
 _Comment=Convert audio CDs to sound files
-Exec=sound-juicer %u
+Exec=pfexec sound-juicer %u
 Icon=sound-juicer
 StartupNotify=true
 MimeType=x-content/audio-cdda;
