--- sound-juicer-2.26.1/data/sound-juicer.desktop.in.in.old	2009年  4月 27日 一 10:40:18
+++ sound-juicer-2.26.1/data/sound-juicer.desktop.in.in	2009年  4月 27日 一 10:41:37
@@ -1,6 +1,10 @@
 [Desktop Entry]
-_Name=Audio CD Extractor
-_Comment=Copy music from your CDs
+# SUN_BRANDING
+_Name=CD Ripper
+# SUN_BRANDING
+_GenericName=CD Ripper
+# SUN_BRANDING
+_Comment=Convert audio CDs to sound files
 Exec=sound-juicer %u
 Icon=sound-juicer
 StartupNotify=true
