--- sound-juicer-2.32.0/src/sj-main.c.1	2017-04-04 23:33:49.111768045 +0300
+++ sound-juicer-2.32.0/src/sj-main.c	2017-04-04 23:35:17.110210605 +0300
@@ -1135,6 +1135,7 @@
   GstEncodingProfile *profile;
   const char *media_type;
 
+  if (!entry) return;
   g_assert (strcmp (entry->key, GCONF_AUDIO_PROFILE_MEDIA_TYPE) == 0);
   if (!entry->value) return;
   media_type = gconf_value_get_string (entry->value);
