diff -ur id3lib-3.8.3-orig/src/tag_file.cpp id3lib-3.8.3/src/tag_file.cpp
--- id3lib-3.8.3-orig/src/tag_file.cpp	2003-03-02 07:23:00.000000000 +0700
+++ id3lib-3.8.3/src/tag_file.cpp	2006-05-08 19:32:35.073541180 +0700
@@ -282,7 +282,7 @@
 
     tmpOut.write(tagData, tagSize);
     file.seekg(tag.GetPrependedBytes(), ios::beg);
-    uchar tmpBuffer[BUFSIZ];
+    char tmpBuffer[BUFSIZ];
     while (file)
     {
       file.read(tmpBuffer, BUFSIZ);
