--- pytest-cov-6.0.0/tests/test_pytest_cov.py.orig
+++ pytest-cov-6.0.0/tests/test_pytest_cov.py
@@ -884,6 +884,7 @@
     assert result.ret == 0
 
 
+@pytest.mark.skip(reason="mangles PYTHONPATH and so it is not able to find pytest-cov in proto area")
 def test_central_subprocess_change_cwd_with_pythonpath(pytester, testdir, monkeypatch):
     stuff = testdir.mkdir('stuff')
     parent_script = stuff.join('parent_script.py')
--- pytest-cov-6.0.0/tox.ini.orig
+++ pytest-cov-6.0.0/tox.ini
@@ -31,7 +31,6 @@
     {bootstrap,clean,check,report,docs}: {env:TOXPYTHON:python3}
 extras = testing
 setenv =
-    PYTHONPATH={toxinidir}/tests
     PYTHONUNBUFFERED=yes
 
     # Use env vars for (optional) pinning of deps.
