--- libopenusb-1.0.1/src/Makefile.in	Thu Feb 21 17:54:43 2008
+++ Makefile.in	Thu Feb 21 17:59:12 2008
@@ -374,6 +374,9 @@
 	  else :; fi; \
 	done
 
+install-sunos-backend: $(usblib_LTLIBRARIES)
+	$(CC) $(CFLAGS) -shared -Wl,-h -Wl,sunos.so -o .libs/sunos.so  .libs/sunos_la-sunos.o  -R$(libdir) -L.libs $(sunos_la_LIBADD)
+
 uninstall-usblibLTLIBRARIES:
 	@$(NORMAL_UNINSTALL)
 	@list='$(usblib_LTLIBRARIES)'; for p in $$list; do \
