--- libvisual-plugins-0.4.0/plugins/actor/oinksie/table.h.orig	2020-05-17 07:59:54.778086869 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/oinksie/table.h	2020-05-17 08:00:16.895832306 +0000
@@ -27,10 +27,10 @@
 #define OINK_TABLE_NORMAL_SIZE	1200
 #define OINK_TABLE_LARGE_SIZE	12000
 
-float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
-float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
-float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
-float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
+extern float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
+extern float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
+extern float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
+extern float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
 
 void _oink_table_init ();
 
--- libvisual-plugins-0.4.0/plugins/actor/oinksie/table.c.orig	2020-05-17 08:00:02.245148497 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/oinksie/table.c	2020-05-17 08:00:47.858280170 +0000
@@ -26,6 +26,11 @@
 #include "common.h"
 #include "table.h"
 
+float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
+float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
+float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
+float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
+
 void _oink_table_init ()
 {
 	float adder;
