Our harfbuzz include files reside in a separate harfbuzz folder.

--- pango-1.48.0/pango/pango-font.h.orig	2020-11-08 15:25:10.861614200 +0000
+++ pango-1.48.0/pango/pango-font.h	2021-01-09 20:46:35.715122334 +0000
@@ -26,7 +26,7 @@
 #include <pango/pango-types.h>
 
 #include <glib-object.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
--- pango-1.48.0/pango/pango-coverage.h.orig	2020-11-08 15:25:10.861614200 +0000
+++ pango-1.48.0/pango/pango-coverage.h	2021-01-09 20:46:58.516545730 +0000
@@ -25,7 +25,7 @@
 #include <glib-object.h>
 
 #include <pango/pango-version-macros.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
--- pango-1.48.0/pango/pangofc-fontmap.h.orig	2020-11-08 15:25:10.870614300 +0000
+++ pango-1.48.0/pango/pangofc-fontmap.h	2021-01-09 20:47:37.813385285 +0000
@@ -26,7 +26,7 @@
 #include <fontconfig/fontconfig.h>
 #include <pango/pangofc-decoder.h>
 #include <pango/pangofc-font.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
