Our harfbuzz include files reside in a separate harfbuzz folder.

--- pango-1.56.1/pango/pango-font.h.orig
+++ pango-1.56.1/pango/pango-font.h
@@ -26,7 +26,7 @@
 #include <pango/pango-types.h>
 
 #include <glib-object.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
--- pango-1.56.1/pango/pango-coverage.h.orig
+++ pango-1.56.1/pango/pango-coverage.h
@@ -25,7 +25,7 @@
 #include <glib-object.h>
 
 #include <pango/pango-version-macros.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
--- pango-1.56.1/pango/pangofc-fontmap.h.orig
+++ pango-1.56.1/pango/pangofc-fontmap.h
@@ -26,7 +26,7 @@
 #include <fontconfig/fontconfig.h>
 #include <pango/pangofc-decoder.h>
 #include <pango/pangofc-font.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
