--- opensmalltalk-vm-sun-v5.0.6/platforms/unix/config/Makefile.in	Wed Apr 29 13:45:06 2020
+++ p0/opensmalltalk-vm-sun-v5.0.6/platforms/unix/config/Makefile.in	Fri May  1 10:46:03 2020
@@ -77,10 +77,10 @@
 
 # unix launch scripts
 squeak.sh: ${cfgdir}/squeak.sh.in getversion
-	sed "s|.expanded_relative_imgdir.|lib/squeak/`./getversion VERSION_TAG`|" <${cfgdir}/squeak.sh.in >squeak.sh
+	sed "s|.expanded_relative_imgdir.|lib/amd64/squeak/`./getversion VERSION_TAG`|" <${cfgdir}/squeak.sh.in >squeak.sh
 
 bin.squeak.sh: ${cfgdir}/bin.squeak.sh.in getversion
-	sed "s|.expanded_relative_imgdir.|lib/squeak/`./getversion VERSION_TAG`|" <${cfgdir}/bin.squeak.sh.in >bin.squeak.sh
+	sed "s|.expanded_relative_imgdir.|lib/amd64/squeak/`./getversion VERSION_TAG`|" <${cfgdir}/bin.squeak.sh.in >bin.squeak.sh
 
 # npsqueak
 
