--- opensmalltalk-vm-sun-v5.0.13/platforms/Cross/vm/sqSCCSVersion.h	Thu Sep  3 19:44:13 2020
+++ p0/opensmalltalk-vm-sun-v5.0.13/platforms/Cross/vm/sqSCCSVersion.h	Sat Sep  5 14:48:01 2020
@@ -28,13 +28,13 @@
 
 #if SUBVERSION
 # define PREFIX "r"
-static char SvnRawRevisionString[] = "$Rev$";
+static char SvnRawRevisionString[] = "$Rev: 202009031744 $";
 # define REV_START (SvnRawRevisionString + 6)
 
-static char SvnRawRevisionDate[] = "$Date$";
+static char SvnRawRevisionDate[] = "$Date: Thu Sep 3 19:44:13 2020 +0200 $";
 # define DATE_START (SvnRawRevisionDate + 7)
 
-static char SvnRawRepositoryURL[] = "$URL$";
+static char SvnRawRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (SvnRawRepositoryURL + 6)
 
 static char *
@@ -71,16 +71,16 @@
 # undef URL_START
 #elif GIT
 # define PREFIX ""
-static char GitRawRevisionString[] = "$Rev$";
+static char GitRawRevisionString[] = "$Rev: 202009031744 $";
 # define REV_START (GitRawRevisionString + 6)
 
-static char GitRawRevisionDate[] = "$Date$";
+static char GitRawRevisionDate[] = "$Date: Thu Sep 3 19:44:13 2020 +0200 $";
 # define DATE_START (GitRawRevisionDate + 7)
 
-static char GitRawRepositoryURL[] = "$URL$";
+static char GitRawRepositoryURL[] = "$URL: stes@deimos:src/opensmalltalk $";
 # define URL_START (GitRawRepositoryURL + 6)
 
-static char GitRawRevisionShortHash[] = "$CommitHash$";
+static char GitRawRevisionShortHash[] = "$CommitHash: f2f36342b $";
 # define SHORTHASH_START (GitRawRevisionShortHash + 13)
 
 static char *
