https://github.com/pallets/click/issues/3128

--- click-8.4.1/tests/test_termui.py.orig
+++ click-8.4.1/tests/test_termui.py
@@ -398,7 +398,7 @@
         named_tempfile.write("a\nb\n")
         named_tempfile.flush()
 
-        result = click.edit(filename=named_tempfile.name, editor="sed -i~ 's/$/Test/'")
+        result = click.edit(filename=named_tempfile.name, editor="gsed -i~ 's/$/Test/'")
         assert result is None
 
         # We need to reopen the file as it becomes unreadable after the edit.
