--- dbus-1.4.14/dbus/dbus-sysdeps-unix.c-orig	2011-09-13 16:41:50.420584386 -0500
+++ dbus-1.4.14/dbus/dbus-sysdeps-unix.c	2011-09-13 16:42:18.502907891 -0500
@@ -56,6 +56,7 @@
 #include <netdb.h>
 #include <grp.h>
 #include <arpa/inet.h>
+#include <alloca.h>
 
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
--- dbus-1.7.0/bus/Makefile.am.orig	Mon Apr 22 10:57:39 2013
+++ dbus-1.7.0/bus/Makefile.am	Mon Apr 22 11:13:30 2013
@@ -7,6 +7,7 @@
 	$(THREAD_LIBS) \
 	$(ADT_LIBS) \
 	$(NETWORK_libs) \
+	$(X_EXTRA_LIBS) \
 	$(NULL)
 
 DBUS_LAUNCHER_LIBS = \
@@ -13,6 +14,7 @@
 	$(XML_LIBS) \
 	$(THREAD_LIBS) \
 	$(NETWORK_libs) \
+	$(X_EXTRA_LIBS) \
 	$(NULL)
 
 AM_CPPFLAGS = \
--- dbus-1.10.0/dbus/Makefile.am.orig	Mon Oct 26 14:29:01 2015
+++ dbus-1.10.0/dbus/Makefile.am	Mon Oct 26 14:33:56 2015
@@ -291,7 +291,7 @@
 SYMBOL_EXPORT_LDFLAGS=
 endif
 
-libdbus_1_la_LIBADD= $(LIBDBUS_LIBS)
+libdbus_1_la_LIBADD= $(LIBDBUS_LIBS) $(X_EXTRA_LIBS)
 libdbus_1_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
