--- dbus-1.10.8/m4/libtool.m4.~1~	2016-03-11 00:21:28.159616976 +0300
+++ dbus-1.10.8/m4/libtool.m4	2016-03-11 00:28:29.226194478 +0300
@@ -1547,13 +1547,13 @@
 if test -n "$RANLIB"; then
   case $host_os in
   bitrig* | openbsd*)
-    old_postinstall_cmds="$old_postinstall_cmds~\$RANLIB -t \$tool_oldlib"
+    old_postinstall_cmds="$old_postinstall_cmds~\$RANLIB -t \$oldlib"
     ;;
   *)
-    old_postinstall_cmds="$old_postinstall_cmds~\$RANLIB \$tool_oldlib"
+    old_postinstall_cmds="$old_postinstall_cmds~\$RANLIB \$oldlib"
     ;;
   esac
-  old_archive_cmds="$old_archive_cmds~\$RANLIB \$tool_oldlib"
+  old_archive_cmds="$old_archive_cmds~\$RANLIB \$oldlib"
 fi
 
 case $host_os in
