--- dbus-1.12.8/test/Makefile.am	2018-02-23 11:25:17.000000000 +0000
+++ dbus-1.12.8/test/Makefile.am	2018-07-01 21:32:06.070657489 +0000
@@ -1,9 +1,3 @@
-## the "name-test" subdir in fact contains a bunch of tests now that need a temporary bus
-## to be running to do stuff with. The directory should be renamed.
-## We want to build the current directory first to pick up the testutils lib
-SUBDIRS= . name-test
-DIST_SUBDIRS=name-test
-
 CLEANFILES =
 EXTRA_DIST =
 
@@ -466,7 +460,6 @@ in_data = \
 EXTRA_DIST += $(in_data)
 
 static_data = \
-	name-test/tmp-session-like-system.conf \
 	data/auth/anonymous-client-successful.auth-script \
 	data/auth/anonymous-server-successful.auth-script \
 	data/auth/cancel.auth-script \
