--- yelp-tools-3.18.0/tools/yelp.m4.1	2015-12-06 02:23:18.003715081 +0300
+++ yelp-tools-3.18.0/tools/yelp.m4	2015-12-06 02:23:35.541552943 +0300
@@ -23,7 +23,7 @@
 AC_ARG_WITH([help-dir],
             AS_HELP_STRING([--with-help-dir=DIR],
                            [path where help files are installed]),,
-            [with_help_dir='${datadir}/help'])
+            [with_help_dir='${datadir}/gnome/help'])
 HELP_DIR="$with_help_dir"
 AC_SUBST(HELP_DIR)
 
