--- socat-1.7.2.4/configure.in.~1~	2014-03-09 18:51:39.000000000 +0400
+++ socat-1.7.2.4/configure.in	2014-11-25 18:49:20.622766510 +0300
@@ -296,7 +296,12 @@
   AC_CHECK_HEADER(netinet/if_ether.h,
     AC_DEFINE(HAVE_NETINET_IF_ETHER_H),
     [WITH_INTERFACE=;
-     AC_MSG_WARN([include file netinet/if_ether.h not found, disabling interface])])
+     AC_MSG_WARN([include file netinet/if_ether.h not found, disabling interface])],[AC_INCLUDES_DEFAULT
+	#ifdef HAVE_NET_IF_H
+        #include <net/if.h>
+	#endif]) # Solaris and illumos need this
+
+
 fi
 if test "$WITH_INTERFACE"; then
    AC_DEFINE(WITH_INTERFACE)
