--- unzip60/unix/configure.bkp	2011-07-28 07:07:08.691240200 -0700
+++ unzip60/unix/configure	2011-07-28 07:07:47.315977300 -0700
@@ -646,7 +646,7 @@
 
 
 echo CC=\"${CC}\" CF=\"${CFLAGSR} ${D_USE_BZ2}\" CRCA_O=\"${CRC32OA}\" \
-       AS=\"${CC} -c\" LFLAGS1=\"${LFLAGS1}\" LF2=\"${LFLAGS2}\" \
+       AS=\"${CC} -c\" LFLAGS1=\"${LFLAGS1} ${LD_OPTIONS}\" LF2=\"${LFLAGS2}\" \
        CC_BZ=\"${CC_BZ}\" CFLAGS_BZ=\"${CFLAGS_BZ}\" \
        IZ_BZIP2=\"${IZ_BZIP2}\" D_USE_BZ2=\"${D_USE_BZ2}\" \
        L_BZ2=\"${L_BZ2}\" LIBBZ2=\"${LIBBZ2}\"  > flags
