--- slib/slib.sh_orig	Thu Dec 20 20:17:37 2007
+++ slib/slib.sh	Tue Mar 29 14:16:10 2011
@@ -1,3 +1,11 @@
+#!/usr/bin/bash
+RELOC_ROOT=$(pkginfo -r SUNWslib)
+if [[ "$RELOC_ROOT" == "none" ]]; then
+       RELOC_ROOT=/
+fi
+SCHEME_LIBRARY_PATH=${SCHEME_LIBRARY_PATH:-${RELOC_ROOT}usr/lib/slib}
+export SCHEME_LIBRARY_PATH
+VERSION=3b1
 
 ##"slib" script; Find a Scheme implementation and initialize SLIB in it.
 #Copyright (C) 2003, 2004 Aubrey Jaffer
