--- openmpi-4.0.1/opal/datatype/opal_convertor_raw.c.orig	2019-07-23 15:56:59.775503017 +0000
+++ openmpi-4.0.1/opal/datatype/opal_convertor_raw.c	2019-07-23 15:57:42.526108675 +0000
@@ -30,6 +30,8 @@
 #define DO_DEBUG(INST)
 #endif /* OPAL_ENABLE_DEBUG */
 
+#define MIN(a,b) ((a)<(b)?(a):(b))
+
 /**
  * This function always work in local representation. This means no representation
  * conversion (i.e. no heterogeneity) has to be taken into account, and that all
