--- openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/src/mca/pshmem/mmap/pshmem_mmap.c.orig	2019-08-03 19:00:32.531143929 +0000
+++ openmpi-4.0.1/opal/mca/pmix/pmix3x/pmix/src/mca/pshmem/mmap/pshmem_mmap.c	2019-08-03 19:01:56.930931440 +0000
@@ -33,6 +33,11 @@
 #    define MAP_ANONYMOUS MAP_ANON
 #endif /* MAP_ANONYMOUS and MAP_ANON */
 
+// ZFS does not support posix_fallocate
+#if defined(__sun)
+#undef HAVE_POSIX_FALLOCATE
+#endif
+
 static int _mmap_init(void);
 static void _mmap_finalize(void);
 static int _mmap_segment_create(pmix_pshmem_seg_t *sm_seg, const char *file_name, size_t size);
