--- src/mesa/drivers/dri/Makefile.am.1	2018-12-20 14:54:53.075943388 +0000
+++ src/mesa/drivers/dri/Makefile.am	2018-12-20 14:55:01.701995407 +0000
@@ -7,11 +7,9 @@
 SUBDIRS+=common
 
 # On Android, we need to explicitly link to libglapi.so.
-if HAVE_ANDROID
 if HAVE_SHARED_GLAPI
 SHARED_GLAPI_LIB = $(top_builddir)/src/mapi/shared-glapi/libglapi.la
 endif
-endif
 
 if HAVE_I915_DRI
 SUBDIRS += i915
