--- src/mesa/drivers/dri/Makefile.template	2010-07-02 19:59:36.605863132 -0700
+++ src/mesa/drivers/dri/Makefile.template	2010-07-03 14:00:38.222593253 -0700
@@ -73,7 +73,7 @@ $(TOP)/$(LIB_DIR)/$(LIBNAME): $(LIBNAME)
 	$(INSTALL) $(LIBNAME) $(TOP)/$(LIB_DIR) 
 
 
-depend: $(C_SOURCES) $(ASM_SOURCES) $(SYMLINKS)
+depend: $(C_SOURCES) $(ASM_SOURCES) $(SYMLINKS) $(COMMON_SYMLINKS)
 	touch depend
 	$(MKDEP) $(MKDEP_OPTIONS) $(DRIVER_DEFINES) $(INCLUDES) $(C_SOURCES) \
 		$(ASM_SOURCES)
