Description: fix an issue.
Author: Lukáš Lalinský <lalinsky@gmail.com>
Last-Update: 2006-08-22
Index: libofa-0.9.3/examples/wavefile.cpp
===================================================================
--- libofa-0.9.3.orig/examples/wavefile.cpp
+++ libofa-0.9.3/examples/wavefile.cpp
@@ -42,7 +42,11 @@ AudioData* loadWaveFile(char *file) {
     int srate = 0;
     int channels = 0;
 
-    int fd = open(file, O_RDONLY | 0x8000);
+#ifdef WIN32
+    int fd = open(file, O_RDONLY | O_BINARY);
+#else
+    int fd = open(file, O_RDONLY);
+#endif
     if (fd == -1)
 	return 0;
 
