From: Martin Michlmayr <tbm@cyrius.com>
Subject: FTBFS with GCC 4.4: missing #include
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=504902
Forwarded: no
---
 examples/example.cpp |    1 +
 1 file changed, 1 insertion(+)

Index: libofa-0.9.3/examples/example.cpp
===================================================================
--- libofa-0.9.3.orig/examples/example.cpp
+++ libofa-0.9.3/examples/example.cpp
@@ -8,6 +8,7 @@
 -------------------------------------------------------------------*/
 
 #include "protocol.h"
+#include <cstdio>
 #include <cctype>
 #include <string.h>
 
