--- netaddr-0.7.10/netaddr/tools/netaddr.~1~	2014-01-23 20:36:19.593996884 +0400
+++ netaddr-0.7.10/netaddr/tools/netaddr	2014-01-23 20:36:32.947171706 +0400
@@ -24,7 +24,7 @@
 try:
     try:
         # ipython >= 0.11
-        from IPython.frontend.terminal.embed import InteractiveShellEmbed
+        from IPython.terminal.embed import InteractiveShellEmbed
         ipshell = InteractiveShellEmbed(banner1=banner, exit_msg=exit_msg)
     except ImportError:
         # ipython < 0.11
