diff -urN gvfs-1.3.2/monitor/hal/ghalmount.c ../SUNWgnome-gvfs-2.26.1.p5.hacked/gvfs-1.3.2/monitor/hal/ghalmount.c
--- gvfs-1.3.2/monitor/hal/ghalmount.c	2009-07-04 16:48:13.000000000 +0100
+++ ../SUNWgnome-gvfs-2.26.1.p5.hacked/gvfs-1.3.2/monitor/hal/ghalmount.c	2009-07-23 19:10:30.266575742 +0100
@@ -960,7 +960,7 @@
                      GAsyncReadyCallback  callback,
                      gpointer             user_data)
 {
-  return g_hal_mount_unmount_with_operation (mount, flags, NULL, cancellable, callback, user_data);
+  g_hal_mount_unmount_with_operation (mount, flags, NULL, cancellable, callback, user_data);
 }
 
 static gboolean
@@ -1045,7 +1045,7 @@
                    GAsyncReadyCallback  callback,
                    gpointer             user_data)
 {
-  return g_hal_mount_eject_with_operation (mount, flags, NULL, cancellable, callback, user_data);
+  g_hal_mount_eject_with_operation (mount, flags, NULL, cancellable, callback, user_data);
 }
 
 static gboolean
diff -urN gvfs-1.3.2/monitor/hal/ghalvolume.c ../SUNWgnome-gvfs-2.26.1.p5.hacked/gvfs-1.3.2/monitor/hal/ghalvolume.c
--- gvfs-1.3.2/monitor/hal/ghalvolume.c	2009-07-04 16:48:13.000000000 +0100
+++ ../SUNWgnome-gvfs-2.26.1.p5.hacked/gvfs-1.3.2/monitor/hal/ghalvolume.c	2009-07-23 19:00:30.546386028 +0100
@@ -908,7 +908,7 @@
                     GAsyncReadyCallback  callback,
                     gpointer             user_data)
 {
-  return g_hal_volume_eject_with_operation (volume, flags, NULL, cancellable, callback, user_data);
+  g_hal_volume_eject_with_operation (volume, flags, NULL, cancellable, callback, user_data);
 }
 
 static gboolean
