--- openldap-2.4.46/servers/slapd/slapd.conf	2018-12-19 22:36:40.151361833 +0000
+++ openldap-2.4.46/servers/slapd/slapd.conf.new	2018-12-20 06:37:20.828478604 +0000
@@ -15,8 +15,8 @@ argsfile	%LOCALSTATEDIR%/run/slapd.args
 
 # Load dynamic backend modules:
 # modulepath	%MODULEDIR%
-# moduleload	back_mdb.la
-# moduleload	back_ldap.la
+# moduleload	back_mdb.so
+# moduleload	back_ldap.so
 
 # Sample security restrictions
 #	Require integrity protection (prevent hijacking)
