--- Pgtcl-2.2.0/configure.~1~	2017-06-21 15:14:48.000000000 +0000
+++ Pgtcl-2.2.0/configure	2017-07-18 21:47:36.316575862 +0000
@@ -5395,8 +5395,8 @@
 LIBPG=""
 PG_INCLUDES="-I$PG_INC_DIR"
 PKG_INCLUDES="-I$PG_INC_DIR"
-PG_LIBS="-L$PG_LIB_DIR -lpq"
-PKG_LIBS="-L$PG_LIB_DIR -lpq"
+PG_LIBS="-L$PG_LIB_DIR -R$PG_LIB_DIR -lpq"
+PKG_LIBS="-L$PG_LIB_DIR -R$PG_LIB_DIR -lpq"
 if test ! "$PG_LIB_DIR" = ""; then
     LIBPG="libpq${SHLIB_SUFFIX}"
 fi
