--- Pgtcl-3b2b2f8e35ec3ae65d2fc295e6a6ce7afbd5e26d/Makefile.in.~1~	2013-06-14 17:17:03.000000000 +0400
+++ Pgtcl-3b2b2f8e35ec3ae65d2fc295e6a6ce7afbd5e26d/Makefile.in	2013-10-30 15:55:07.039896794 +0400
@@ -79,7 +79,7 @@
 
 DESTDIR		=
 
-PKG_DIR		= $(PACKAGE_NAME)$(PACKAGE_VERSION)
+PKG_DIR		= 
 pkgdatadir	= $(datadir)/$(PKG_DIR)
 pkglibdir	= $(libdir)/$(PKG_DIR)
 pkgincludedir	= $(includedir)/$(PKG_DIR)
@@ -194,7 +194,7 @@
 	@echo "        xml2nroff sample.xml > sample.n"
 	@echo "        xml2html sample.xml > sample.html"
 
-install: all install-binaries install-libraries install-doc
+install: all install-binaries install-libraries
 
 install-binaries: binaries install-lib-binaries install-bin-binaries
 
