--- notification-daemon-0.4.0/src/capplet/notification-properties.c-orig	   :: 
+++ notification-daemon-0.4.0/src/capplet/notification-properties.c	   :: 
@@ -431,12 +431,21 @@
 		g_object_unref(dialog->preview);
 		dialog->preview = NULL;
 	}
-
+#ifdef NOTIFY_CHECK_VERSION
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
 	dialog->preview = notify_notification_new(_("Notification Test"),
 											  _("Just a test"),
-											  "gnome-util",
-											  NULL);
-
+											  "gnome-util");
+#else
+	dialog->preview = notify_notification_new(_("Notification Test"),
+											  _("Just a test"),
+											  "gnome-util", NULL);
+#endif
+#else
+	dialog->preview = notify_notification_new(_("Notification Test"),
+											  _("Just a test"),
+											  "gnome-util", NULL);
+#endif
 	if (!notify_notification_show(dialog->preview, &error))
 	{
 		char *message = g_strdup_printf(
