diff -ur notification-daemon-0.4.0/src/themes/standard/theme.c notification-daemon-0.4.0-new/src/themes/standard/theme.c
--- notification-daemon-0.4.0/src/themes/standard/theme.c	2008-11-20 17:38:01.000000000 +0800
+++ notification-daemon-0.4.0-new/src/themes/standard/theme.c	2008-12-03 21:02:18.638728949 +0800
@@ -670,6 +670,7 @@
 	gtk_box_pack_start(GTK_BOX(hbox), windata->summary_label, TRUE, TRUE, 0);
 	gtk_misc_set_alignment(GTK_MISC(windata->summary_label), 0, 0);
 	gtk_label_set_line_wrap(GTK_LABEL(windata->summary_label), TRUE);
+	gtk_label_set_line_wrap_mode(GTK_LABEL(windata->summary_label), PANGO_WRAP_WORD_CHAR);
 
 	atkobj = gtk_widget_get_accessible(windata->summary_label);
 	atk_object_set_description(atkobj, "Notification summary text.");
@@ -726,6 +727,7 @@
 	gtk_box_pack_start(GTK_BOX(vbox), windata->body_label, TRUE, TRUE, 0);
 	gtk_misc_set_alignment(GTK_MISC(windata->body_label), 0, 0);
 	gtk_label_set_line_wrap(GTK_LABEL(windata->body_label), TRUE);
+	gtk_label_set_line_wrap_mode(GTK_LABEL(windata->body_label), PANGO_WRAP_WORD_CHAR);
 	g_signal_connect_swapped(G_OBJECT(windata->body_label), "url_activated",
 							 G_CALLBACK(windata->url_clicked), win);
 
