https://github.com/malthe/chameleon/issues/366

--- Chameleon-3.10.2.orig/tox.ini
+++ Chameleon-3.10.2/tox.ini
@@ -0,0 +1,29 @@
+[tox]
+envlist =
+    py27,py35,py36,py37,py38,py39,pypy,pypy3,cover
+
+[testenv]
+commands =
+   {envdir}/bin/zope-testrunner --path {toxinidir}/src {posargs:-vc}
+deps =
+    zope.testrunner
+    chameleon
+
+[testenv:cover]
+basepython =
+    python3
+commands =
+    coverage run -m zope.testrunner -m chameleon --path={toxinidir}/src []
+    coverage html -i
+    coverage report -i
+deps =
+    {[testenv]deps}
+    coverage
+
+[testenv:docs]
+commands =
+    sphinx-build -b html -d docs/_build/doctrees docs docs/_build/html
+deps =
+    .
+    Sphinx
+    sphinx_rtd_theme
