Index: trunk/configure.in
===================================================================
--- trunk/configure.in	(revision 63)
+++ trunk/configure.in	(working copy)
@@ -25,6 +25,9 @@
 
 dnl ==========================================================================
 
+dnl Check for bind in libsocket (needed on Solaris)
+AC_CHECK_LIB(socket, bind)
+
 set_debug=no
 AC_ARG_ENABLE(debug,
 [  --enable-debug       Lots of noisy debug messages],
