Index: trunk/src/connect.c
===================================================================
--- trunk/src/connect.c	(revision 68)
+++ trunk/src/connect.c	(working copy)
@@ -438,10 +438,12 @@
     
   miProtocol = gtk_menu_item_new_with_mnemonic ("XDMCP");
   gtk_container_add (GTK_CONTAINER (mnuProtocol), miProtocol);
-  gtk_tooltips_set_tip (tooltips, miProtocol, _("XDMCP protocol uses Xnest to connect to servers running X."), NULL);
+  gtk_tooltips_set_tip (tooltips, miProtocol, _("XDMCP protocol uses Xephyr or Xnest to connect to servers running X."), NULL);
   g_signal_connect (G_OBJECT (miProtocol), "activate", G_CALLBACK (on_protocol_changed), (int*)2);
   gtk_widget_ref (miProtocol);
-  if (!g_find_program_in_path ("Xnest")) { gtk_widget_set_sensitive (miProtocol, FALSE); }
+  if (!g_find_program_in_path ("Xephyr") &&
+      !g_find_program_in_path ("Xnest"))
+    { gtk_widget_set_sensitive (miProtocol, FALSE); }
 
   miProtocol = gtk_menu_item_new_with_mnemonic ("ICA");
   gtk_container_add (GTK_CONTAINER (mnuProtocol), miProtocol);
Index: trunk/src/support.c
===================================================================
--- trunk/src/support.c	(revision 68)
+++ trunk/src/support.c	(working copy)
@@ -643,14 +643,19 @@
     } else if (rdp->protocol == 2) {
       int display;
     
-      if (g_find_program_in_path ("Xnest")) {
+      if (g_find_program_in_path ("Xephyr")) {
         sflags += G_SPAWN_SEARCH_PATH;
         
+        sprintf(buffer, "Xephyr");
+        c_argv[c_argc++] = strdup(buffer);
+      } else if (g_find_program_in_path ("Xnest")) {
+        sflags += G_SPAWN_SEARCH_PATH;
+        
         sprintf(buffer, "Xnest");
         c_argv[c_argc++] = strdup(buffer);
       } else {
 	if(error) {
-	  *error = g_strdup(_("Xnest was not found in your path.\nPlease verify your Xnest installation."));
+	  *error = g_strdup(_("Xephyr and Xnest were not found in your path.\nPlease verify your Xephyr or Xnest installation."));
 	}
         return 1;
       }
Index: trunk/tsclient.1
===================================================================
--- trunk/tsclient.1	(revision 68)
+++ trunk/tsclient.1	(working copy)
@@ -33,7 +33,7 @@
 Remote Desktop Protocol (RDP). Using the rdesktop program as a backend,
 tsclient allows users to access and view their desktops as stored on remote
 Windows NT/2000/XP servers.  It also supports connections using other remote
-desktop tools such as: vnc, Xnest and the Citrix ICA client.
+desktop tools such as: vnc, Xephyr, Xnest and the Citrix ICA client.
 .PP
 The GNOME applet can be added to your panel by right-clicking the panel and
 choosing the following: \fBAdd to Panel -> Internet -> Terminal Server Client
