Index: trunk/tsclient.desktop.in
===================================================================
--- trunk/tsclient.desktop.in	(revision 72)
+++ trunk/tsclient.desktop.in	(working copy)
@@ -3,13 +3,12 @@
 _Comment=Connect to a remote computer
 TryExec=tsclient
 Exec=tsclient
-Icon=tsclient.png
+Icon=tsclient
 Terminal=false
 Type=Application
 Version=1.0
 Actions=Full;
 Categories=GTK;GNOME;X-GNOME-NetworkSettings;Network;
-Encoding=UTF-8
 
 [Desktop Action Full]
 Exec=tsclient -f
