--- tsclient-0.150/configure.in.ori	2009-02-10 13:16:49.015754000 +0000
+++ tsclient-0.150/configure.in	2009-02-10 13:17:17.598616800 +0000
@@ -54,8 +54,7 @@
    pkg_modules="gtk+-2.0"
 else
    AC_DEFINE([HAVE_GNOME],1,[gnome support])
-   pkg_modules="libgnomeui-2.0"
-   pkg_modules="libpanelapplet-2.0"   
+   pkg_modules="libgnomeui-2.0 libpanelapplet-2.0"   
 fi
 
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
