--- tsclient-0.150/src/support.c.orig	2012-03-29 13:53:11.439189792 +0800
+++ tsclient-0.150/src/support.c	2012-03-29 13:54:02.910424176 +0800
@@ -676,31 +676,31 @@
 
       switch (rdp->desktopwidth) {
       case 640:
-        sprintf(buffer, "-geometry");
+        sprintf(buffer, "-screen");
         c_argv[c_argc++] = strdup(buffer);
         sprintf(buffer, "640x480");
         c_argv[c_argc++] = strdup(buffer);
         break;
       case 800:
-        sprintf(buffer, "-geometry");
+        sprintf(buffer, "-screen");
         c_argv[c_argc++] = strdup(buffer);
         sprintf(buffer, "800x600");
         c_argv[c_argc++] = strdup(buffer);
         break;
       case 1024:
-        sprintf(buffer, "-geometry");
+        sprintf(buffer, "-screen");
         c_argv[c_argc++] = strdup(buffer);
         sprintf(buffer, "1024x768");
         c_argv[c_argc++] = strdup(buffer);
         break;
       case 1152:
-        sprintf(buffer, "-geometry");
+        sprintf(buffer, "-screen");
         c_argv[c_argc++] = strdup(buffer);
         sprintf(buffer, "1152x864");
         c_argv[c_argc++] = strdup(buffer);
         break;
       case 1280:
-        sprintf(buffer, "-geometry");
+        sprintf(buffer, "-screen");
         c_argv[c_argc++] = strdup(buffer);
         sprintf(buffer, "1280x960");
         c_argv[c_argc++] = strdup(buffer);
@@ -948,7 +948,7 @@
     }
 
     c_argv[c_argc++] = NULL;
-    
+
     // complete events in gtk queue
     while(gtk_events_pending())
       gtk_main_iteration();
