--- bracex-2.3.post1/tox.ini.orig
+++ bracex-2.3.post1/tox.ini
@@ -12,11 +12,11 @@
     -rrequirements/project.txt
     -rrequirements/test.txt
 commands=
-    {envbindir}/mypy
-    {envbindir}/py.test --cov bracex --cov-append tests
-    {envbindir}/coverage html -d {envtmpdir}/coverage
-    {envbindir}/coverage xml
-    {envbindir}/coverage report --show-missing
+    mypy
+    py.test --cov bracex --cov-append tests
+    coverage html -d {envtmpdir}/coverage
+    coverage xml
+    coverage report --show-missing
 
 [testenv:lint]
 passenv= *
