To bootstrap this we need to remove the tox version constraint.
See also https://github.com/sirosen/dependency-groups/issues/16

--- dependency_groups-1.3.0/tox.ini.orig
+++ dependency_groups-1.3.0/tox.ini
@@ -10,7 +10,6 @@
     ci = py{38,39,310,311,312,313}, covcombine, covreport
     ci-mypy = mypy-py38, mypy-py313
     ci-package-check = twine-check
-minversion = 4.22.0
 
 [testenv]
 package = wheel
