--- xf86-video-vmware-13.3.0/src/Makefile.am.orig	2018-05-17 09:42:24.000000000 +0200
+++ xf86-video-vmware-13.3.0/src/Makefile.am	2023-02-05 15:26:04.082188147 +0100
@@ -27,7 +27,7 @@
 vmware_drv_la_LTLIBRARIES = vmware_drv.la
 vmware_drv_la_LDFLAGS = -module -avoid-version
 vmware_drv_la_CFLAGS = $(CWARNFLAGS) @XORG_CFLAGS@
-vmware_drv_ladir = @moduledir@/drivers
+vmware_drv_ladir = @moduledir@
 
 if BUILD_VMWGFX
 vmware_drv_la_LIBADD = $(top_builddir)/vmwgfx/libvmwgfx.la
