https://github.com/rsyslog/rsyslog/issues/1009

Fixed in 8.19
--- rsyslog-8.15.0/configure.ac	2016-05-04 07:56:46.266980065 -0700
+++ rsyslog-8.15.0/configure.ac	2016-05-04 07:56:24.215900559 -0700
@@ -259,12 +259,18 @@ AC_ARG_ENABLE(gssapi_krb5,
          esac],
 	[enable_gssapi_krb5=no]
 )
+
+case "${os_type}" in
+solaris) GSSLIB=gss ;;
+*)       GSSLIB=gssapi_krb5 ;;
+esac
+  
 if test $enable_gssapi_krb5 = yes; then
-	AC_CHECK_LIB(gssapi_krb5, gss_acquire_cred, [
+	AC_CHECK_LIB($GSSLIB, gss_acquire_cred, [
 		AC_CHECK_HEADER(gssapi/gssapi.h, [
 			AC_DEFINE(USE_GSSAPI,,
 				  Define if you want to use GSSAPI)
-			GSS_LIBS="-lgssapi_krb5"
+			GSS_LIBS="-l$GSSLIB"
 			AC_SUBST(GSS_LIBS)
 		])
 	])
