--- rsyslog/plugins/imsolaris/imsolaris.c.orig	Mon Jan  9 06:05:17 2012
+++ rsyslog/plugins/imsolaris/imsolaris.c	Thu Mar  1 12:55:01 2012
@@ -392,9 +392,9 @@
 
 	/* register config file handlers */
 	CHKiRet(omsdRegCFSLineHdlr((uchar *)"resetconfigvariables", 1, eCmdHdlrCustomHandler,
-		resetConfigVariables, NULL, STD_LOADABLE_MODULE_ID));
+		resetConfigVariables, NULL, STD_LOADABLE_MODULE_ID, eConfObjGlobal));
         CHKiRet(omsdRegCFSLineHdlr((uchar *)"imsolarislogsocketname", 0, eCmdHdlrGetWord,
-                NULL, &LogName, STD_LOADABLE_MODULE_ID));
+                NULL, &LogName, STD_LOADABLE_MODULE_ID, eConfObjGlobal));
 ENDmodInit
 /* vim:set ai:
  */
