--- rsyslog/tools/Makefile.in	Mon Jan  9 06:15:51 2012
+++ rsyslog/tools/Makefile.in	Thu Feb 16 11:23:14 2012
@@ -339,7 +339,7 @@
 	../dirty.h
 
 rsyslogd_CPPFLAGS = $(PTHREADS_CFLAGS) $(RSRT_CFLAGS)
-rsyslogd_LDADD = $(ZLIB_LIBS) $(PTHREADS_LIBS) $(RSRT_LIBS) $(SOL_LIBS) $(LIBEE_LIBS) $(LIBLOGNORM_LIBS)
+rsyslogd_LDADD = $(ZLIB_LIBS) $(PTHREADS_LIBS) $(RSRT_LIBS) $(SOL_LIBS) $(LIBEE_LIBS) $(LIBESTR_LIBS) $(LIBLOGNORM_LIBS)
 rsyslogd_LDFLAGS = -export-dynamic
 @ENABLE_DIAGTOOLS_TRUE@rsyslog_diag_hostname_SOURCES = gethostn.c
 @ENABLE_DIAGTOOLS_TRUE@zpipe_SOURCES = zpipe.c
