https://github.com/ktosiek/pytest-freezegun/pull/38

diff --git a/pytest_freezegun.py b/pytest_freezegun.py
index a08133f..23fae90 100644
--- a/pytest_freezegun.py
+++ b/pytest_freezegun.py
@@ -2,7 +2,6 @@
 
 import pytest
 
-from distutils.version import LooseVersion
 from freezegun import freeze_time
 
 
@@ -10,16 +9,6 @@
 FIXTURE_NAME = 'freezer'
 
 
-def get_closest_marker(node, name):
-    """
-    Get our marker, regardless of pytest version
-    """
-    if LooseVersion(pytest.__version__) < LooseVersion('3.6.0'):
-        return node.get_marker('freeze_time')
-    else:
-        return node.get_closest_marker('freeze_time')
-
-
 @pytest.fixture(name=FIXTURE_NAME)
 def freezer_fixture(request):
     """
@@ -30,7 +19,7 @@ def freezer_fixture(request):
     ignore = []
 
     # If we've got a marker, use the arguments provided there
-    marker = get_closest_marker(request.node, MARKER_NAME)
+    marker = request.node.get_closest_marker('freeze_time')
     if marker:
         ignore = marker.kwargs.pop('ignore', [])
         args = marker.args
@@ -52,7 +41,7 @@ def pytest_collection_modifyitems(items):
     Inject our fixture into any tests with our marker
     """
     for item in items:
-        if get_closest_marker(item, MARKER_NAME):
+        if item.get_closest_marker('freeze_time'):
             item.fixturenames.insert(0, FIXTURE_NAME)
 
 
--- pytest-freezegun-0.4.2/setup.cfg.orig
+++ pytest-freezegun-0.4.2/setup.cfg
@@ -31,7 +31,7 @@
 	pytest_freezegun
 install_requires = 
 	freezegun>0.3
-	pytest>=3.0.0
+	pytest>=3.6
 
 [options.entry_points]
 pytest11 = 
