Silence a recurring warning during the build to reduce noise.

--- webkitgtk-2.48.1/Source/WTF/wtf/MathExtras.h.orig
+++ webkitgtk-2.48.1/Source/WTF/wtf/MathExtras.h
@@ -421,7 +421,7 @@
         value = 0;
     else {
         // -2^{64} < fmodValue < 2^{64}.
-        double fmodValue = fmod(trunc(d), maxPlusOne<unsigned long long>);
+	double fmodValue = fmod(trunc(d), static_cast<double>(maxPlusOne<unsigned long long>));
         if (fmodValue >= 0) {
             // 0 <= fmodValue < 2^{64}.
             // 0 <= value < 2^{64}. This cast causes no loss.
