--- webkitgtk-2.53.4/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/NonCompositedFrameRenderer.cpp.old	2026-07-26 22:58:08.715599138 -0400
+++ webkitgtk-2.53.4/Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/NonCompositedFrameRenderer.cpp	2026-07-26 22:58:26.512752838 -0400
@@ -328,7 +328,7 @@
 }
 #endif
 
-#if ENABLE(DAMAGE_TRACKING)
+#if 0
 void NonCompositedFrameRenderer::resetDamageHistoryForTesting()
 {
     m_frameDamageHistoryForTesting = std::make_optional<Vector<WebCore::Region>>();
