--- citus-8.0.1/src/backend/distributed/worker/worker_sql_task_protocol.c.1	2018-12-06 20:32:55.000573256 +0000
+++ citus-8.0.1/src/backend/distributed/worker/worker_sql_task_protocol.c	2018-12-06 20:33:52.003573975 +0000
@@ -20,6 +20,7 @@
 #include "utils/builtins.h"
 #include "utils/memutils.h"
 
+#include <sys/stat.h>
 
 /* TaskFileDestReceiver can be used to stream results into a file */
 typedef struct TaskFileDestReceiver
