--- xf86-video-savage-2.3.7/src/Makefile.am.~1~	2015-02-13 16:24:03.658976232 +0300
+++ xf86-video-savage-2.3.7/src/Makefile.am	2015-02-13 16:26:30.661628874 +0300
@@ -26,7 +26,8 @@
 AM_CFLAGS = @XORG_CFLAGS@ @DRI_CFLAGS@
 savage_drv_la_LTLIBRARIES = savage_drv.la
 savage_drv_la_LDFLAGS = -module -avoid-version
-savage_drv_ladir = @moduledir@/drivers
+tmp = @moduledir@/drivers
+savage_drv_ladir = $(tmp:amd64/xorg/modules/drivers=xorg/modules/drivers/amd64)
 
 savage_drv_la_SOURCES = \
          compat-api.h \
