--- dovecot-2.4.2/src/lib-sasl/gssapi-dummy.c	2025-10-29 07:58:41.000000000 +0100
+++ dovecot-2.4.2/src/lib-sasl/gssapi-dummy.c.new	2026-01-05 13:19:41.337180927 +0100
@@ -26,7 +26,7 @@
 	gss_OID_desc mech_type;
 };
 
-gss_OID GSS_C_NT_HOSTBASED_SERVICE = NULL;
+const gss_OID GSS_C_NT_HOSTBASED_SERVICE = NULL;
 const gss_OID GSS_KRB5_NT_PRINCIPAL_NAME = NULL;
 
 static void gss_alloc_buffer(gss_buffer_t buffer, size_t length);
