https://www.smartmontools.org/changeset/5357/

--- smartmontools/ChangeLog	(revision 5356)
+++ smartmontools/ChangeLog	(revision 5357)
@@ -1,3 +1,8 @@
 $Id$
+
+2022-04-24  Christian Franke  <franke@computer.org>
+
+	update-smart-drivedb.in: Don't use semicolon in sed scripts.
+	This also fixes a syntax error if sed requires ';' before '}'.
 
 2022-04-23  Gabriele Pohl  <contact@dipohl.de>
--- smartmontools/update-smart-drivedb.in	(revision 5356)
+++ smartmontools/update-smart-drivedb.in	(revision 5357)
@@ -543,6 +543,8 @@
 {
   local r v x
-  x=$(sed -n '/^[ {]*"VERSION: *[^"]*"/{s,^[ {]*"VERSION: \([1-9][./0-9]* [^"]*\)".*$,\1,p;q}' "$1") \
-  || return 1
+  x=`sed -n '/^[ {]*"VERSION: *[^"]*"/{
+       s,^[ {]*"VERSION: \([1-9][./0-9]* [^"]*\)".*$,\1,p
+       q
+     }' "$1"` || return 1
   v=${x%% *}
   test -n "$v" || return 0
