--- xchat-2.8.8/src/common/xchat.h.orig	2014-03-08 19:16:39.748313736 +0000
+++ xchat-2.8.8/src/common/xchat.h	2014-03-08 19:16:51.748856387 +0000
@@ -1,10 +1,6 @@
 #include "../../config.h"
 
-#include <glib/gslist.h>
-#include <glib/glist.h>
-#include <glib/gutils.h>
-#include <glib/giochannel.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <time.h>			/* need time_t */
 
 #ifndef XCHAT_H
--- xchat-2.8.8/src/common/util.c.orig	2014-03-08 19:20:42.525276492 +0000
+++ xchat-2.8.8/src/common/util.c	2014-03-08 19:20:54.846213510 +0000
@@ -39,7 +39,6 @@
 #include <errno.h>
 #include "xchat.h"
 #include "xchatc.h"
-#include <glib/gmarkup.h>
 #include <ctype.h>
 #include "util.h"
 #include "../../config.h"
--- xchat-2.8.8/src/common/text.c.orig	2014-03-08 19:19:37.259990918 +0000
+++ xchat-2.8.8/src/common/text.c	2014-03-08 19:20:19.703049166 +0000
@@ -28,7 +28,6 @@
 #include <sys/mman.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
 #include "cfgfiles.h"
 #include "chanopt.h"
 #include "plugin.h"
--- xchat-2.8.8/src/common/servlist.c.orig	2014-03-08 19:19:59.014305598 +0000
+++ xchat-2.8.8/src/common/servlist.c	2014-03-08 19:20:14.547518707 +0000
@@ -24,7 +24,6 @@
 #include <unistd.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
 
 #include "cfgfiles.h"
 #include "fe.h"
