--- xchat-2.8.8/src/fe-gtk/plugin-tray.c.1	2016-04-14 12:35:14.686020971 +0300
+++ xchat-2.8.8/src/fe-gtk/plugin-tray.c	2016-04-14 12:42:37.390147125 +0300
@@ -125,8 +125,7 @@
 /* prototypes */
 static gboolean (*nn_init) (char *);
 static void (*nn_uninit) (void);
-static void *(*nn_new_with_status_icon) (const gchar *summary, const gchar *message, const gchar *icon, GtkStatusIcon *status_icon);
-static void *(*nn_new) (const gchar *summary, const gchar *message, const gchar *icon, GtkWidget *attach);
+static void *(*nn_new) (const gchar *summary, const gchar *message, const gchar *icon);
 static gboolean (*nn_show) (void *noti, GError **error);
 static void (*nn_set_timeout) (void *noti, gint timeout);
 
@@ -151,7 +150,7 @@
 		nn_mod = g_module_open ("libnotify", G_MODULE_BIND_LAZY);
 		if (!nn_mod)
 		{
-			nn_mod = g_module_open ("libnotify.so.1", G_MODULE_BIND_LAZY);
+			nn_mod = g_module_open ("libnotify.so.4", G_MODULE_BIND_LAZY);
 			if (!nn_mod)
 				return FALSE;
 		}
@@ -160,8 +159,6 @@
 			goto bad;
 		if (!g_module_symbol (nn_mod, "notify_uninit", (gpointer)&nn_uninit))
 			goto bad;
-		if (!g_module_symbol (nn_mod, "notify_notification_new_with_status_icon", (gpointer)&nn_new_with_status_icon))
-			goto bad;
 		if (!g_module_symbol (nn_mod, "notify_notification_new", (gpointer)&nn_new))
 			goto bad;
 		if (!g_module_symbol (nn_mod, "notify_notification_show", (gpointer)&nn_show))
@@ -174,7 +171,7 @@
 
 	text = strip_color (text, -1, STRIP_ALL|STRIP_ESCMARKUP);
 	title = strip_color (title, -1, STRIP_ALL);
-	noti = nn_new (title, text, XCHATSHAREDIR"/pixmaps/xchat.png", NULL);
+	noti = nn_new (title, text, XCHATSHAREDIR"/pixmaps/xchat.png");
 	g_free ((char *)title);
 	g_free ((char *)text);
 
