https://savannah.gnu.org/bugs/?56831

--- findutils-4.7.0/tests/find/printf_inode.sh	2019-08-30 10:21:17.198324425 +0000
+++ findutils-4.7.0/tests/find/printf_inode.sh.new	2019-08-30 10:21:47.966762636 +0000
@@ -27,7 +27,7 @@ make_canonical() {
 > file || framework_failure_
 
 # Let ls(1) create the expected output.
-ls -i file | make_canonical > exp || framework_failure_
+/usr/gnu/bin/ls -i file | make_canonical > exp || framework_failure_
 
 for executable in oldfind find; do
   rm -f out out2
