--- util-macros-1.19.2/configure.ac.orig	2018-03-05 19:07:23.998065704 +0000
+++ util-macros-1.19.2/configure.ac	2018-03-05 19:07:45.498845440 +0000
@@ -22,10 +22,10 @@
 dnl Process this file with autoconf to create configure.
 
 AC_PREREQ([2.62])
-AC_INIT([util-macros],
+AC_INIT([xorg-macros],
         [1.19.2],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg],
-        [util-macros])
+        [xorg-macros])
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 
