--- util-macros-1.19.0/configure.ac.orig	2016-04-23 23:38:25.284091825 +0200
+++ util-macros-1.19.0/configure.ac	2016-04-23 23:38:45.603615907 +0200
@@ -22,10 +22,10 @@
 dnl Process this file with autoconf to create configure.
 
 AC_PREREQ([2.62])
-AC_INIT([util-macros],
+AC_INIT([xorg-macros],
         [1.19.0],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg],
-        [util-macros])
+        [xorg-macros])
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 
