--- util-macros-1.19.1/configure.ac.orig	2017-01-29 20:23:44.000000000 +0100
+++ util-macros-1.19.1/configure.ac	2017-01-30 04:39:48.117896370 +0100
@@ -22,10 +22,10 @@
 dnl Process this file with autoconf to create configure.
 
 AC_PREREQ([2.62])
-AC_INIT([util-macros],
+AC_INIT([xorg-macros],
         [1.19.1],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg],
-        [util-macros])
+        [xorg-macros])
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 
