--- samba-4.15.6/lib/addns/wscript_build	2021-08-09 15:38:36.075378000 +0000
+++ samba-4.15.6/lib/addns/wscript_build.new	2022-03-16 14:38:01.961166839 +0000
@@ -11,6 +11,6 @@
                        error.c
                        dnsquery_srv.c
                    ''',
-                   public_deps='samba-util gssapi ndr resolv dns_lookup',
+                   public_deps='samba-util gssapi krb5 ndr resolv dns_lookup',
                    private_library=True,
                    vars=locals())
