Don't be too clever about directories layout
--- samba-4.4.6/dynconfig/wscript.1	2016-11-11 02:18:57.484812873 +0300
+++ samba-4.4.6/dynconfig/wscript	2016-11-11 02:19:29.556837907 +0300
@@ -300,10 +300,10 @@
         flavor = 'FHS-PATH'
     else:
         flavor = 'STD-PATH'
-        if conf.env.PREFIX == '/usr' or conf.env.PREFIX == '/usr/local':
-           Logs.error("Don't install directly under /usr or /usr/local without using the FHS option (--enable-fhs)")
-           raise Utils.WafError("ERROR: invalid --prefix=%s value" % (conf.env.PREFIX))
-
+#        if conf.env.PREFIX == '/usr' or conf.env.PREFIX == '/usr/local':
+#           Logs.error("Don't install directly under /usr or /usr/local without using the FHS option (--enable-fhs)")
+#           raise Utils.WafError("ERROR: invalid --prefix=%s value" % (conf.env.PREFIX))
+#
     explicit_set ={}
 
     dyn_vars = {}
