--- openssl-1.1.1m/Configurations/10-main.conf.orig	2021-12-14 16:45:01.000000000 +0000
+++ openssl-1.1.1m/Configurations/10-main.conf	2022-02-23 17:24:28.860772169 +0000
@@ -311,7 +311,7 @@
         # -m32 should be safe to add as long as driver recognizes
         # -mcpu=ultrasparc
         inherit_from     => [ "solaris-sparcv7-gcc", asm("sparcv9_asm") ],
-        cflags           => add_before("-m32 -mcpu=ultrasparc"),
+        cflags           => add_before("-m32 -mcpu=ultrasparc -mvis -mfsmuld"),
     },
     "solaris64-sparcv9-gcc" => {
         inherit_from     => [ "solaris-sparcv9-gcc" ],
