--- os.hpp.old	Wed Mar 25 07:26:03 2009
+++ os.hpp	Wed Mar 25 07:40:39 2009
@@ -228,6 +228,19 @@
   #endif
 #endif
 
+#if !defined(LITTLE_ENDIAN) && !defined(BIG_ENDIAN)
+  #if defined(__i386) || defined(i386) || defined(__i386__)
+    #define LITTLE_ENDIAN
+  #elif defined(BYTE_ORDER) && BYTE_ORDER == LITTLE_ENDIAN
+    #define LITTLE_ENDIAN
+  #elif defined(BYTE_ORDER) && BYTE_ORDER == BIG_ENDIAN
+    #define BIG_ENDIAN
+  #else
+    #error "Neither LITTLE_ENDIAN nor BIG_ENDIAN are defined. Define one of them."
+  #endif
+#endif
+
+
 #if !defined(BIG_ENDIAN) && !defined(_WIN_CE) && defined(_WIN_32)
 /* allow not aligned integer access, increases speed in some operations */
 #define ALLOW_NOT_ALIGNED_INT
