--- consio.cpp.orig	Tue Jun  7 16:00:37 2011
+++ consio.cpp	Tue Jun  7 16:01:33 2011
@@ -124,7 +124,7 @@
   OemToChar(Str,Str);
   SetConsoleMode(hConIn,ConInMode);
   SetConsoleMode(hConOut,ConOutMode);
-#elif defined(_EMX) || defined(_BEOS) || defined(__sparc) || defined(sparc) || defined (__VMS)
+#elif defined(_EMX) || defined(_BEOS) || defined (__VMS)
   fgets(Str,MaxLength-1,stdin);
 #else
   strncpyz(Str,getpass(""),MaxLength);
