--- pdsh-pdsh-2.33/doc/pdsh.1.in	2017-06-29 02:28:26.000000000 +0000
+++ pdsh-pdsh-2.33/doc/pdsh.1.in.new	2018-07-22 18:53:20.717597754 +0000
@@ -82,21 +82,6 @@ across the files console.foo[0-10]:
 ssh
 Uses a variant of popen(3) to run multiple copies of the ssh(1)
 command.
-.TP
-mrsh
-This module uses the mrsh(1) protocol to execute jobs on remote hosts.
-The mrsh protocol uses a credential based authentication, forgoing
-the need to allocate reserved ports. In other aspects, it acts just
-like rsh. Remote nodes must be running mrshd(8) in order for the mrsh
-module to work.
-.TP
-krb4
-The krb4 module allows users to execute remote commands after 
-authenticating with kerberos. Of course, the remote rshd daemons
-must be kerberized.
-.TP
-xcpu
-The xcpu module uses the xcpu service to execute remote commands.
 
 .SH OPTIONS
 The list of available options is determined at runtime
