--- sip-4.19.25/configure.py	Fri Tue 26 23:28:41 2021
+++ sip-4.19.25_patched/configure.py	Wed Thu 27 06:34:07 2022
@@ -943,7 +943,8 @@
     # Set defaults.
     sip_bin_dir = plat_bin_dir
     sip_inc_dir = plat_py_venv_inc_dir
-    sip_root_dir = plat_py_site_dir
+    sip_root_dir = plat_py_site_dir+"/sip"
+    sip_module_dir = sip_root_dir+"/64"
     sip_sip_dir = plat_sip_dir
 
     if opts.config_file is not None:
