diff -ruN gmime-2.5.0.orig/docs/reference/Makefile.am gmime-2.5.0/docs/reference/Makefile.am
--- gmime-2.5.0.orig/docs/reference/Makefile.am	2010-02-02 20:04:19.066806600 +0000
+++ gmime-2.5.0/docs/reference/Makefile.am	2010-02-02 20:04:34.204663800 +0000
@@ -64,7 +64,7 @@
 	gmime-events.h
 
 # Extra options to supply to gtkdoc-fixref
-FIXXREF_OPTIONS = 
+FIXXREF_OPTIONS = --module=$(DOC_MODULE)
 
 include $(top_srcdir)/gtk-doc.make
 
