Description: Fix indentation to allow building with GCC 6.
Author: Johannes Brandstätter <jbrandst@2ds.eu>
Bug-Debian: https://bugs.debian.org/831117
Last-Update: <2016-09-24>

--- gtkimageview-1.6.4+dfsg.orig/src/gtkimagenav.c
+++ gtkimageview-1.6.4+dfsg/src/gtkimagenav.c
@@ -71,8 +71,8 @@ static Size
 gtk_image_nav_get_preview_size (GtkImageNav *nav)
 {
 	GdkPixbuf *pixbuf = gtk_image_view_get_pixbuf (nav->view);
-    if (!pixbuf)
-        return (Size){GTK_IMAGE_NAV_MAX_WIDTH, GTK_IMAGE_NAV_MAX_HEIGHT};
+	if (!pixbuf)
+		return (Size){GTK_IMAGE_NAV_MAX_WIDTH, GTK_IMAGE_NAV_MAX_HEIGHT};
 	int img_width = gdk_pixbuf_get_width (pixbuf);
 	int img_height = gdk_pixbuf_get_height (pixbuf);
 
