https://rt.cpan.org/Public/Bug/Display.html?id=118837

--- DBD-mysql-4.038/t/05dbcreate.t	st říj  5 14:44:00 2016
+++ DBD-mysql-4.037/t/05dbcreate.t	po říj  3 08:52:35 2016
@@ -10,7 +10,7 @@
 require 'lib.pl';
 
 # remove database from DSN
-$test_dsn =~ s/^DBI:mysql:([^:]+)(:?)/DBI:mysql:$2/;
+$test_dsn =~ s/^DBI:mysql:([^:;]+)([:;]?)/DBI:mysql:$2/;
 
 my $dbh;
 eval {$dbh= DBI->connect($test_dsn, $test_user, $test_password,
