https://rt.cpan.org/Public/Bug/Display.html?id=112072

--- DBD-mysql-4.033/t/05dbcreate.t	2016-02-13 13:29:56.467130717 -0800
+++ DBD-mysql-4.033/t/05dbcreate.t	2016-02-13 13:23:45.371688976 -0800
@@ -10,7 +10,7 @@ use lib 't', '.';
 require 'lib.pl';
 
 my $dbh;
-eval {$dbh= DBI->connect('DBI:mysql:', $test_user, $test_password,
+eval {$dbh= DBI->connect($test_dsn, $test_user, $test_password,
                       { RaiseError => 1, PrintError => 1, AutoCommit => 0 });};
 if ($@) {
     diag $@;
