--- xdg-user-dirs-gtk-0.12/meson.build.orig
+++ xdg-user-dirs-gtk-0.12/meson.build
@@ -28,13 +28,4 @@
                  po_dir: 'po',
                  type: 'desktop')
 
-# data: Also install desktop file into datadir
-# At least on GLib-based systems, autostart folders are not searched
-# for installed .desktop files. So in order for GNOME to pick up the
-# correct name and icon, install another copy into the datadir's
-# application directory.
-#
-# Mirrors commit: fe37c50488efe8ef9605df239d3e33537cf46332
-meson.add_install_script('meson_custom_install_desktop_file.sh', skip_if_destdir: false)
-
 subdir('po')
