We have no pylama packaged yet.

--- isort-6.0.1/tests/unit/test_importable.py.orig
+++ isort-6.0.1/tests/unit/test_importable.py
@@ -20,7 +20,6 @@
     import isort.parse
     import isort.place
     import isort.profiles
-    import isort.pylama_isort
     import isort.sections
     import isort.settings
     import isort.setuptools_commands
--- isort-6.0.1/pyproject.toml.orig
+++ isort-6.0.1/pyproject.toml
@@ -116,7 +116,6 @@
     "mypy>=1.14.1",
     "mypy-extensions>=0.4.3",
     "pip>=21.1.1",
-    "pylama>=7.7",
     "pytest>=7.4.2",
     "pytest-benchmark>=3.4.1",
     "pytest-mock>=1.10",
