--- mongodb-src-r3.4.9/SConstruct.~1~	2017-09-07 22:28:48.000000000 +0000
+++ mongodb-src-r3.4.9/SConstruct	2017-09-13 11:41:00.565704974 +0000
@@ -408,6 +408,11 @@
     nargs=0,
 )
 
+add_option('disable-ggdb',
+    help="Don't add -ggdb to compiler command line",
+    nargs=0,
+)
+
 add_option('variables-help',
     help='Print the help text for SCons variables',
     nargs=0,
@@ -1557,13 +1562,16 @@
     # -Winvalid-pch Warn if a precompiled header (see Precompiled Headers) is found in the search path but can't be used.
     env.Append( CCFLAGS=["-fno-omit-frame-pointer",
                          "-fno-strict-aliasing",
-                         "-ggdb",
                          "-pthread",
                          "-Wall",
                          "-Wsign-compare",
                          "-Wno-unknown-pragmas",
                          "-Winvalid-pch"] )
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
+
+    if not has_option("disable-ggdb"):
+        env.Append( CCFLAGS=["-ggdb"] )
+
     if env.TargetOSIs('linux', 'osx', 'solaris'):
         if not has_option("disable-warnings-as-errors"):
             env.Append( CCFLAGS=["-Werror"] )
