--- mongodb-src-r3.3.12/SConstruct.1	2016-09-08 23:44:50.673340313 +0300
+++ mongodb-src-r3.3.12/SConstruct	2016-09-08 23:45:08.701896471 +0300
@@ -1527,7 +1527,6 @@
     # -Winvalid-pch Warn if a precompiled header (see Precompiled Headers) is found in the search path but can't be used.
     env.Append( CCFLAGS=["-fno-omit-frame-pointer",
                          "-fno-strict-aliasing",
-                         "-ggdb",
                          "-pthread",
                          "-Wall",
                          "-Wsign-compare",
