--- gimp-3.0.0/app/config/gimpguiconfig.h.old	2025-03-19 00:36:13.187239252 -0400
+++ gimp-3.0.0/app/config/gimpguiconfig.h	2025-03-19 00:36:58.570470883 -0400
@@ -24,8 +24,8 @@
 #include "config/gimpdisplayconfig.h"
 
 
-#define GIMP_CONFIG_DEFAULT_THEME          "Default"
-#define GIMP_CONFIG_DEFAULT_ICON_THEME     "Default"
+#define GIMP_CONFIG_DEFAULT_THEME          "System"
+#define GIMP_CONFIG_DEFAULT_ICON_THEME     "Legacy"
 
 
 #define GIMP_TYPE_GUI_CONFIG            (gimp_gui_config_get_type ())
