--- gimp-2.10.20/app/config/gimpguiconfig.h.1	2020-08-01 18:06:00.362870946 +0000
+++ gimp-2.10.20/app/config/gimpguiconfig.h	2020-08-01 18:07:10.806828257 +0000
@@ -24,8 +24,8 @@
 #include "config/gimpdisplayconfig.h"
 
 
-#define GIMP_CONFIG_DEFAULT_THEME          "Dark"
-#define GIMP_CONFIG_DEFAULT_ICON_THEME     "Symbolic"
+#define GIMP_CONFIG_DEFAULT_THEME          "System"
+#define GIMP_CONFIG_DEFAULT_ICON_THEME     "Legacy"
 
 
 #define GIMP_TYPE_GUI_CONFIG            (gimp_gui_config_get_type ())
