diff -u -r gimp-2.8.14/app/composite/make-installer.py gimp-2.8.14-py/app/composite/make-installer.py
--- gimp-2.8.14/app/composite/make-installer.py	2014-08-20 19:27:57.000000000 +0400
+++ gimp-2.8.14-py/app/composite/make-installer.py	2015-01-04 09:51:43.600050227 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # Gimp image compositing
 # Copyright (C) 2003  Helvetix Victorinox, <helvetix@gimp.org>
diff -u -r gimp-2.8.14/app/composite/ns.py gimp-2.8.14-py/app/composite/ns.py
--- gimp-2.8.14/app/composite/ns.py	2014-08-20 19:27:57.000000000 +0400
+++ gimp-2.8.14-py/app/composite/ns.py	2015-01-04 09:51:43.595152610 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # Copyright (C) 2003  Helvetix Victorinox, a pseudonym, <helvetix@gimp.org>
 #
 # This program is free software: you can redistribute it and/or modify
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/benchmark-foreground-extract.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/benchmark-foreground-extract.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/benchmark-foreground-extract.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/benchmark-foreground-extract.py	2015-01-04 09:51:43.622063735 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Foreground Extraction Benchmark
 #   Copyright 2005  Sven Neumann <sven@gimp.org>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/clothify.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/clothify.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/clothify.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/clothify.py	2015-01-04 09:51:43.622893262 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/colorxhtml.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/colorxhtml.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/colorxhtml.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/colorxhtml.py	2015-01-04 09:51:43.603827682 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 2003, 2005  Manish Singh <yosh@gimp.org>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/file-openraster.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/file-openraster.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/file-openraster.py	2014-08-20 19:27:58.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/file-openraster.py	2015-01-04 09:51:43.608428979 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 # GIMP Plug-in for the OpenRaster file format
 # http://create.freedesktop.org/wiki/OpenRaster
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/foggify.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/foggify.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/foggify.py	2014-08-20 19:27:58.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/foggify.py	2015-01-04 09:51:43.620316138 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/gradients-save-as-css.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/gradients-save-as-css.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/gradients-save-as-css.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/gradients-save-as-css.py	2015-01-04 09:51:43.601901464 +0300
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.7
 # -*- coding: utf-8 -*-

 #   Allows saving (TODO: and loading) CSS gradient files
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-offset.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-offset.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-offset.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-offset.py	2015-01-04 09:51:43.600775242 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-
 #    This program is free software: you can redistribute it and/or modify
 #   it under the terms of the GNU General Public License as published by
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-sort.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-sort.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-sort.py	2014-08-20 19:27:58.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-sort.py	2015-01-04 09:51:43.609786965 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-to-gradient.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-to-gradient.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/palette-to-gradient.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/palette-to-gradient.py	2015-01-04 09:51:43.623809425 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/py-slice.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/py-slice.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/py-slice.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/py-slice.py	2015-01-04 09:51:43.613682276 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-

 #Copyright (c) Manish Singh
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/python-console.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/python-console.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/python-console.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/python-console.py	2015-01-04 09:51:43.605876466 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/python-eval.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/python-eval.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/python-eval.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/python-eval.py	2015-01-04 09:51:43.608987169 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 2006  Manish Singh <yosh@gimp.org>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/shadow_bevel.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/shadow_bevel.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/shadow_bevel.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/shadow_bevel.py	2015-01-04 09:51:43.624665563 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/sphere.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/sphere.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/sphere.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/sphere.py	2015-01-04 09:51:43.628882854 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/text-brush.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/text-brush.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/text-brush.py	2014-08-20 19:27:58.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/text-brush.py	2015-01-04 09:51:43.627770155 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # coding: utf-8

 # Author: João Sebastião de Oliveira Bueno
diff -u -r gimp-2.8.14/plug-ins/pygimp/plug-ins/whirlpinch.py gimp-2.8.14-py/plug-ins/pygimp/plug-ins/whirlpinch.py
--- gimp-2.8.14/plug-ins/pygimp/plug-ins/whirlpinch.py	2012-03-12 23:18:11.000000000 +0400
+++ gimp-2.8.14-py/plug-ins/pygimp/plug-ins/whirlpinch.py	2015-01-04 09:51:43.626745084 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 #   Gimp-Python - allows the writing of Gimp plugins in Python.
 #   Copyright (C) 1997  James Henstridge <james@daa.com.au>
diff -u -r gimp-2.8.14/tools/defcheck.py gimp-2.8.14-py/tools/defcheck.py
--- gimp-2.8.14/tools/defcheck.py	2012-03-12 23:18:16.000000000 +0400
+++ gimp-2.8.14-py/tools/defcheck.py	2015-01-04 09:51:43.592246665 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 """
 defcheck.py -- Consistency check for the .def files.
diff -u -r gimp-2.8.14/tools/gimppath2svg.py gimp-2.8.14-py/tools/gimppath2svg.py
--- gimp-2.8.14/tools/gimppath2svg.py	2012-03-12 23:18:16.000000000 +0400
+++ gimp-2.8.14-py/tools/gimppath2svg.py	2015-01-04 09:51:43.590194220 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7

 import sys,re
