--- gnome-utils-2.30.0/gsearchtool/libgnomeui-deprecated/Makefile.am	2014-03-08 17:59:21.497898141 +0000
+++ gnome-utils-2.30.0/gsearchtool/libgnomeui-deprecated/Makefile.am	2014-03-08 17:59:35.165008663 +0000
@@ -1,8 +1,3 @@
-INCLUDES = -DGTK_DISABLE_DEPRECATED \
-           -DGDK_DISABLE_DEPRECATED \
-           -DG_DISABLE_DEPRECATED
-
-
 noinst_LTLIBRARIES = libgnomeui-deprecated.la
 
 libgnomeui_deprecated_la_LIBADD = $(GNOME_UTILS_LIBS)
Files gnome-utils-2.30.0/gnome-dictionary/libgdict/.libs/libgdict_1_0_la-gdict-marshal.o and ../libgdict/.libs/libgdict_1_0_la-gdict-marshal.o differ
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-client-context.c	2009-09-08 13:46:49.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-client-context.c	2014-03-08 16:05:11.532152436 +0000
@@ -555,7 +555,7 @@
  *   owned by the #GdictClientContext object and should never be modified or
  *   freed.
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_client_context_get_hostname (GdictClientContext *context)
 {
   gchar *hostname;
@@ -636,7 +636,7 @@
  * Return value: the client name. The returned string is owned by the
  *   #GdictClientContext object and should never be modified or freed.
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_client_context_get_client (GdictClientContext *context)
 {
   gchar *client_name;
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-client-context.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-client-context.h	2014-03-08 16:05:11.539645675 +0000
@@ -87,12 +87,12 @@
 
 void                  gdict_client_context_set_hostname (GdictClientContext *context,
 						         const gchar        *hostname);
-G_CONST_RETURN gchar *gdict_client_context_get_hostname (GdictClientContext *context);
+const gchar *gdict_client_context_get_hostname (GdictClientContext *context);
 void                  gdict_client_context_set_port     (GdictClientContext *context,
 							 gint                port);
 guint                 gdict_client_context_get_port     (GdictClientContext *context);
 void                  gdict_client_context_set_client   (GdictClientContext *context,
 							 const gchar        *client);
-G_CONST_RETURN gchar *gdict_client_context_get_client   (GdictClientContext *context);
+const gchar *gdict_client_context_get_client   (GdictClientContext *context);
 
 #endif /* __GDICT_CLIENT_CONTEXT_H__ */
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-context.c	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-context.c	2014-03-08 16:05:11.540268922 +0000
@@ -494,7 +494,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_database_get_name (GdictDatabase *db)
 {
   g_return_val_if_fail (db != NULL, NULL);
@@ -513,7 +513,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_database_get_full_name (GdictDatabase *db)
 {
   g_return_val_if_fail (db != NULL, NULL);
@@ -603,7 +603,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_strategy_get_name (GdictStrategy *strat)
 {
   g_return_val_if_fail (strat != NULL, NULL);
@@ -621,7 +621,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_strategy_get_description (GdictStrategy *strat)
 {
   g_return_val_if_fail (strat != NULL, NULL);
@@ -710,7 +710,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_match_get_word (GdictMatch *match)
 {
   g_return_val_if_fail (match != NULL, NULL);
@@ -728,7 +728,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_match_get_database (GdictMatch *match)
 {
   g_return_val_if_fail (match != NULL, NULL);
@@ -842,7 +842,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_definition_get_word (GdictDefinition *def)
 {
   g_return_val_if_fail (def != NULL, NULL);
@@ -863,7 +863,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_definition_get_database (GdictDefinition *def)
 {
   g_return_val_if_fail (def != NULL, NULL);
@@ -883,7 +883,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_definition_get_text (GdictDefinition *def)
 {
   g_return_val_if_fail (def != NULL, NULL);
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-context.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-context.h	2014-03-08 16:05:11.537552454 +0000
@@ -82,8 +82,8 @@
 GType                 gdict_database_get_type        (void) G_GNUC_CONST;
 GdictDatabase *       gdict_database_ref             (GdictDatabase   *db);
 void                  gdict_database_unref           (GdictDatabase   *db);
-G_CONST_RETURN gchar *gdict_database_get_name        (GdictDatabase   *db);
-G_CONST_RETURN gchar *gdict_database_get_full_name   (GdictDatabase   *db);
+const gchar *gdict_database_get_name        (GdictDatabase   *db);
+const gchar *gdict_database_get_full_name   (GdictDatabase   *db);
 
 /**
  * GdictStrategy:
@@ -97,8 +97,8 @@
 GType                 gdict_strategy_get_type        (void) G_GNUC_CONST;
 GdictStrategy *       gdict_strategy_ref             (GdictStrategy   *strat);
 void                  gdict_strategy_unref           (GdictStrategy   *strat);
-G_CONST_RETURN gchar *gdict_strategy_get_name        (GdictStrategy   *strat);
-G_CONST_RETURN gchar *gdict_strategy_get_description (GdictStrategy   *strat);
+const gchar *gdict_strategy_get_name        (GdictStrategy   *strat);
+const gchar *gdict_strategy_get_description (GdictStrategy   *strat);
 
 /**
  * GdictMatch:
@@ -111,8 +111,8 @@
 GType                 gdict_match_get_type           (void) G_GNUC_CONST;
 GdictMatch *          gdict_match_ref                (GdictMatch      *match);
 void                  gdict_match_unref              (GdictMatch      *match);
-G_CONST_RETURN gchar *gdict_match_get_word           (GdictMatch      *match);
-G_CONST_RETURN gchar *gdict_match_get_database       (GdictMatch      *match);
+const gchar *gdict_match_get_word           (GdictMatch      *match);
+const gchar *gdict_match_get_database       (GdictMatch      *match);
 
 /**
  * GdictDefinition:
@@ -127,9 +127,9 @@
 GdictDefinition *     gdict_definition_ref           (GdictDefinition *def);
 void                  gdict_definition_unref         (GdictDefinition *def);
 gint                  gdict_definition_get_total     (GdictDefinition *def);
-G_CONST_RETURN gchar *gdict_definition_get_word      (GdictDefinition *def);
-G_CONST_RETURN gchar *gdict_definition_get_database  (GdictDefinition *def);
-G_CONST_RETURN gchar *gdict_definition_get_text      (GdictDefinition *def);
+const gchar *gdict_definition_get_word      (GdictDefinition *def);
+const gchar *gdict_definition_get_database  (GdictDefinition *def);
+const gchar *gdict_definition_get_text      (GdictDefinition *def);
 
 /**
  * GdictContextIface:
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-defbox.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-defbox.h	2014-03-08 16:05:11.535936068 +0000
@@ -74,8 +74,8 @@
 GdictContext *        gdict_defbox_get_context        (GdictDefbox  *defbox);
 void                  gdict_defbox_set_database       (GdictDefbox  *defbox,
 						       const gchar  *database);
-G_CONST_RETURN gchar *gdict_defbox_get_database       (GdictDefbox  *defbox);
-G_CONST_RETURN gchar *gdict_defbox_get_word           (GdictDefbox  *defbox);
+const gchar *gdict_defbox_get_database       (GdictDefbox  *defbox);
+const gchar *gdict_defbox_get_word           (GdictDefbox  *defbox);
 gchar *               gdict_defbox_get_text           (GdictDefbox  *defbox,
 						       gsize        *length) G_GNUC_MALLOC;
 void                  gdict_defbox_select_all         (GdictDefbox  *defbox);
@@ -94,7 +94,7 @@
 void                  gdict_defbox_find_previous      (GdictDefbox  *defbox);
 void                  gdict_defbox_set_font_name      (GdictDefbox  *defbox,
 						       const gchar  *font_name);
-G_CONST_RETURN gchar *gdict_defbox_get_font_name      (GdictDefbox  *defbox);
+const gchar *gdict_defbox_get_font_name      (GdictDefbox  *defbox);
 gchar *               gdict_defbox_get_selected_word  (GdictDefbox  *defbox) G_GNUC_MALLOC;
 
 G_END_DECLS
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source-loader.c	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source-loader.c	2014-03-08 16:05:11.540007250 +0000
@@ -301,7 +301,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN GSList *
+const GSList *
 gdict_source_loader_get_paths (GdictSourceLoader *loader)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE_LOADER (loader), NULL);
@@ -470,7 +470,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN GSList *
+const GSList *
 gdict_source_loader_get_sources (GdictSourceLoader *loader)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE_LOADER (loader), NULL);
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source-loader.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source-loader.h	2014-03-08 16:05:11.537956432 +0000
@@ -65,10 +65,10 @@
 void                   gdict_source_loader_update          (GdictSourceLoader *loader);
 void                   gdict_source_loader_add_search_path (GdictSourceLoader *loader,
 							    const gchar       *path);
-G_CONST_RETURN GSList *gdict_source_loader_get_paths       (GdictSourceLoader *loader);
+const GSList *gdict_source_loader_get_paths       (GdictSourceLoader *loader);
 gchar **               gdict_source_loader_get_names       (GdictSourceLoader *loader,
 							    gsize             *length) G_GNUC_MALLOC;
-G_CONST_RETURN GSList *gdict_source_loader_get_sources     (GdictSourceLoader *loader);
+const GSList *gdict_source_loader_get_sources     (GdictSourceLoader *loader);
 GdictSource *          gdict_source_loader_get_source      (GdictSourceLoader *loader,
 							    const gchar       *name);
 gboolean               gdict_source_loader_has_source      (GdictSourceLoader *loader,
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source.c	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source.c	2014-03-08 16:05:11.538870644 +0000
@@ -807,7 +807,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_source_get_name (GdictSource *source)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE (source), NULL);
@@ -869,7 +869,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_source_get_description (GdictSource *source)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE (source), NULL);
@@ -931,7 +931,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_source_get_database (GdictSource *source)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE (source), NULL);
@@ -993,7 +993,7 @@
  *
  * Since: 1.0
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_source_get_strategy (GdictSource *source)
 {
   g_return_val_if_fail (GDICT_IS_SOURCE (source), NULL);
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-source.h	2014-03-08 16:05:11.533210034 +0000
@@ -87,16 +87,16 @@
 						    
 void                  gdict_source_set_name        (GdictSource           *source,
 						    const gchar           *name);
-G_CONST_RETURN gchar *gdict_source_get_name        (GdictSource           *source);
+const gchar *gdict_source_get_name        (GdictSource           *source);
 void                  gdict_source_set_description (GdictSource           *source,
 						    const gchar           *description);
-G_CONST_RETURN gchar *gdict_source_get_description (GdictSource           *source);
+const gchar *gdict_source_get_description (GdictSource           *source);
 void                  gdict_source_set_database    (GdictSource           *source,
 						    const gchar           *database);
-G_CONST_RETURN gchar *gdict_source_get_database    (GdictSource           *source);
+const gchar *gdict_source_get_database    (GdictSource           *source);
 void                  gdict_source_set_strategy    (GdictSource           *source,
 						    const gchar           *strategy);
-G_CONST_RETURN gchar *gdict_source_get_strategy    (GdictSource           *source);
+const gchar *gdict_source_get_strategy    (GdictSource           *source);
 void                  gdict_source_set_transport   (GdictSource           *source,
 						    GdictSourceTransport   transport,
 						    const gchar           *first_transport_property,
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-speller.c	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-speller.c	2014-03-08 16:05:11.538333179 +0000
@@ -544,7 +544,7 @@
  *
  * Since: FIXME
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_speller_get_database (GdictSpeller *speller)
 {
   g_return_val_if_fail (GDICT_IS_SPELLER (speller), NULL);
@@ -590,7 +590,7 @@
  *
  * Since: FIXME
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_speller_get_strategy (GdictSpeller *speller)
 {
   g_return_val_if_fail (GDICT_IS_SPELLER (speller), NULL);
--- gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-speller.h	2009-04-27 12:37:51.000000000 +0100
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/gdict-speller.h	2014-03-08 16:05:11.535357208 +0000
@@ -69,10 +69,10 @@
 GdictContext *        gdict_speller_get_context      (GdictSpeller *speller);
 void                  gdict_speller_set_database     (GdictSpeller *speller,
 						      const gchar *database);
-G_CONST_RETURN gchar *gdict_speller_get_database     (GdictSpeller *speller);
+const gchar *gdict_speller_get_database     (GdictSpeller *speller);
 void                  gdict_speller_set_strategy     (GdictSpeller *speller,
 						      const gchar  *strategy);
-G_CONST_RETURN gchar *gdict_speller_get_strategy     (GdictSpeller *speller);
+const gchar *gdict_speller_get_strategy     (GdictSpeller *speller);
 
 void                  gdict_speller_clear            (GdictSpeller *speller);
 void                  gdict_speller_match            (GdictSpeller *speller,
--- gnome-utils-2.32.0/gnome-dictionary/libgdict/gdict-defbox.c.~1~	2010-07-16 19:18:51.000000000 +0400
+++ gnome-utils-2.32.0/gnome-dictionary/libgdict/gdict-defbox.c	2015-01-31 16:50:54.947253332 +0300
@@ -52,6 +52,12 @@
 #define QUERY_MARGIN	48
 #define ERROR_MARGIN	24
 
+#if GLIB_CHECK_VERSION (2, 30, 0) 
+#define SPACING_MARK G_UNICODE_SPACING_MARK 
+#else 
+#define SPACING_MARK G_UNICODE_COMBINING_MARK 
+#endif 
+
 typedef struct
 {
   GdictDefinition *definition;
@@ -393,7 +399,7 @@
 	/* If string contains prefix, check that prefix is not followed
 	 * by a unicode mark symbol, e.g. that trailing 'a' in prefix
 	 * is not part of two-char a-with-hat symbol in string. */
-	return type != G_UNICODE_COMBINING_MARK &&
+	return type != SPACING_MARK &&
 		type != G_UNICODE_ENCLOSING_MARK &&
 		type != G_UNICODE_NON_SPACING_MARK;
 }
@@ -2047,7 +2053,7 @@
  *
  * Since: 0.1
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_defbox_get_database (GdictDefbox *defbox)
 {
   g_return_val_if_fail (GDICT_IS_DEFBOX (defbox), NULL);
@@ -2067,7 +2073,7 @@
  *
  * Since: 0.12
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_defbox_get_word (GdictDefbox *defbox)
 {
   g_return_val_if_fail (GDICT_IS_DEFBOX (defbox), NULL);
@@ -2873,7 +2879,7 @@
  *
  * Since: 0.3
  */
-G_CONST_RETURN gchar *
+const gchar *
 gdict_defbox_get_font_name (GdictDefbox *defbox)
 {
   g_return_val_if_fail (GDICT_IS_DEFBOX (defbox), NULL);
