--- gnome-utils-2.30.0/gnome-dictionary/libgdict/Makefile.am	Mon Jun 15 13:45:27 2009
+++ gnome-utils-2.30.0/gnome-dictionary/libgdict/Makefile.am	Sat Sep  7 12:38:27 2013
@@ -13,8 +13,6 @@
 	-DGDICTSOURCESDIR=\""$(datadir)/gdict-1.0/sources"\" 	\
 	-DGDICT_ENABLE_INTERNALS=1 				\
 	-DG_DISABLE_DEPRECATED					\
-	-DGDK_DISABLE_DEPRECATED				\
-	-DGTK_DISABLE_DEPRECATED				\
 	-DPANGO_DISABLE_DEPRECATED				\
 	-DG_DISABLE_SINGLE_INCLUDES				\
 	-DGTK_DISABLE_SINGLE_INCLUDES				\
