coverage-pyver-pragma is not needed to run tests.

--- dom_toml-2.0.1/tests/requirements.txt.orig
+++ dom_toml-2.0.1/tests/requirements.txt
@@ -1,6 +1,5 @@
 coincidence>=0.2.0
 coverage>=5.1
-coverage-pyver-pragma>=0.2.1
 importlib-metadata>=3.6.0
 numpy>=1.19.3; python_version < "3.14"
 pytest>=6.0.0
--- dom_toml-2.0.1/tox.ini.orig
+++ dom_toml-2.0.1/tox.ini
@@ -216,9 +216,6 @@
 unused-arguments-ignore-magic-methods = True
 unused-arguments-ignore-variadic-names = True
 
-[coverage:run]
-plugins = coverage_pyver_pragma
-
 [coverage:report]
 fail_under = 98.5
 show_missing = True
