--- openal-soft-1.17.0/CMakeLists.txt.orig	2015-11-12 16:40:17.000000000 +0100
+++ openal-soft-1.17.0/CMakeLists.txt	2023-03-31 15:41:42.395268435 +0200
@@ -1023,7 +1023,7 @@
 SET(prefix ${CMAKE_INSTALL_PREFIX})
 SET(exec_prefix "\${prefix}")
 SET(libdir "\${exec_prefix}/lib${LIB_SUFFIX}")
-SET(bindir "\${exec_prefix}/bin")
+SET(bindir "\${exec_prefix}/bin${LIB_SUFFIX}")
 SET(includedir "\${prefix}/include")
 SET(PACKAGE_VERSION "${LIB_VERSION}")
 
@@ -1113,7 +1113,7 @@
 IF(ALSOFT_INSTALL)
     # Add an install target here
     INSTALL(TARGETS ${LIBNAME}
-            RUNTIME DESTINATION bin
+            RUNTIME DESTINATION "bin${LIB_SUFFIX}"
             LIBRARY DESTINATION "lib${LIB_SUFFIX}"
             ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
     )
@@ -1181,7 +1181,7 @@
 
     IF(ALSOFT_INSTALL)
         INSTALL(TARGETS openal-info makehrtf bsincgen
-                RUNTIME DESTINATION bin
+                RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                 LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                 ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
         )
@@ -1202,7 +1202,7 @@
 
         IF(ALSOFT_INSTALL)
             INSTALL(TARGETS altonegen
-                    RUNTIME DESTINATION bin
+                    RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                     LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                     ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
             )
@@ -1246,7 +1246,7 @@
 
         IF(ALSOFT_INSTALL)
             INSTALL(TARGETS alstream alreverb allatency alloopback
-                    RUNTIME DESTINATION bin
+                    RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                     LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                     ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
             )
@@ -1284,7 +1284,7 @@
 
             IF(ALSOFT_INSTALL)
                 INSTALL(TARGETS alffplay
-                        RUNTIME DESTINATION bin
+                        RUNTIME DESTINATION "bin${LIB_SUFFIX}"
                         LIBRARY DESTINATION "lib${LIB_SUFFIX}"
                         ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
                 )
