versioneer does not expect flake8-typing-imports plugin.

--- versioneer-0.29/tox.ini.orig
+++ versioneer-0.29/tox.ini
@@ -55,6 +55,6 @@
     # pyflakes all of src except src/git/long_header.py and src/header.py
     python test/run_pyflakes_src.py
     pyflakes test
-    flake8 git_version.py versioneer.py
+    flake8 --extend-ignore=TYP git_version.py versioneer.py
     pycodestyle --max-line-length=88 git_version.py versioneer.py
     !pypy3: mypy git_version.py versioneer.py src/installer.py setup.py
