--- ./src/live_effects/lpe-lattice.cpp.orig	2015-07-20 03:31:32.150515939 +0200
+++ ./src/live_effects/lpe-lattice.cpp	2015-07-20 03:39:10.713309563 +0200
@@ -150,7 +150,7 @@
                         // corner = actual corner of the rectangle
                         // origin = Upperleft point
                         double dl = dot((*handles[corner+4*i] - (base + origin)), dir)/dot(dir,dir);
-                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0,ui+vi);
+                        sb2[dim][i][corner] = dl/( dim ? height : width )*pow(4.0, (int) (ui+vi));
                     }
                 }
             }
