--- ./src/extension/internal/pdfinput/pdf-parser.cpp.orig	2015-07-20 02:35:06.558841073 +0200
+++ ./src/extension/internal/pdfinput/pdf-parser.cpp	2015-07-20 02:49:06.117595499 +0200
@@ -861,7 +861,7 @@
 	  isolated = knockout = gFalse;
 	  if (!obj4.dictLookup(const_cast<char*>("CS"), &obj5)->isNull()) {
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
-	    blendingColorSpace = GfxColorSpace::parse(&obj5, NULL, NULL);
+	    blendingColorSpace = GfxColorSpace::parse(NULL, &obj5, NULL, NULL);
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
 	    blendingColorSpace = GfxColorSpace::parse(&obj5, NULL);
 #else
@@ -1087,9 +1087,9 @@
   res->lookupColorSpace(args[0].getName(), &obj);
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
   if (obj.isNull()) {
-    colorSpace = GfxColorSpace::parse(&args[0], NULL, NULL);
+    colorSpace = GfxColorSpace::parse(NULL, &args[0], NULL, NULL);
   } else {
-    colorSpace = GfxColorSpace::parse(&obj, NULL, NULL);
+    colorSpace = GfxColorSpace::parse(NULL, &obj, NULL, NULL);
   }
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
   if (obj.isNull()) {
@@ -1128,9 +1128,9 @@
   res->lookupColorSpace(args[0].getName(), &obj);
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
   if (obj.isNull()) {
-    colorSpace = GfxColorSpace::parse(&args[0], NULL, NULL);
+    colorSpace = GfxColorSpace::parse(NULL, &args[0], NULL, NULL);
   } else {
-    colorSpace = GfxColorSpace::parse(&obj, NULL, NULL);
+    colorSpace = GfxColorSpace::parse(NULL, &obj, NULL, NULL);
   }
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
   if (obj.isNull()) {
@@ -2757,7 +2757,7 @@
     }
     if (!obj1.isNull()) {
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
-            colorSpace = GfxColorSpace::parse(&obj1, NULL, NULL);
+            colorSpace = GfxColorSpace::parse(NULL, &obj1, NULL, NULL);
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
       colorSpace = GfxColorSpace::parse(&obj1, NULL);
 #else
@@ -2848,7 +2848,7 @@
 	}
       }
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
-            GfxColorSpace *maskColorSpace = GfxColorSpace::parse(&obj1, NULL, NULL);
+            GfxColorSpace *maskColorSpace = GfxColorSpace::parse(NULL, &obj1, NULL, NULL);
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
       maskColorSpace = GfxColorSpace::parse(&obj1, NULL);
 #else
@@ -3041,7 +3041,7 @@
       transpGroup = gTrue;
       if (!obj1.dictLookup(const_cast<char*>("CS"), &obj3)->isNull()) {
 #if defined(POPPLER_EVEN_NEWER_COLOR_SPACE_API)
-	blendingColorSpace = GfxColorSpace::parse(&obj3, NULL, NULL);
+	blendingColorSpace = GfxColorSpace::parse(NULL, &obj3, NULL, NULL);
 #elif defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
 	blendingColorSpace = GfxColorSpace::parse(&obj3, NULL);
 #else
