--- ./src/widgets/ruler.cpp.orig	2015-07-20 04:56:22.700370597 +0200
+++ ./src/widgets/ruler.cpp	2015-07-20 04:58:01.658726699 +0200
@@ -359,7 +359,7 @@
                 (label_spacing_px > 6*digit_height || tick_index%2 == 0 || cur == 0) && 
                 (label_spacing_px > 3*digit_height || tick_index%4 == 0 || cur == 0))
             {
-                if (fabs((int)cur) >= 2000 && (((int) cur)/1000)*1000 == ((int) cur))
+                if (fabs(cur) >= 2000 && (((int) cur)/1000)*1000 == ((int) cur))
                     sprintf (unit_str, "%dk", ((int) cur)/1000);
                 else
                     sprintf (unit_str, "%d", (int) cur);
