--- ./src/sp-ellipse.cpp.orig	2015-07-20 02:25:05.300830641 +0200
+++ ./src/sp-ellipse.cpp	2015-07-20 02:25:20.595294221 +0200
@@ -147,7 +147,7 @@
         if (viewbox) {
             double const dx = viewbox->width();
             double const dy = viewbox->height();
-            double const dr = sqrt(dx*dx + dy*dy)/sqrt(2);
+            double const dr = sqrt(dx*dx + dy*dy)/sqrt(2.);
             double const em = style->font_size.computed;
             double const ex = em * 0.5; // fixme: get from pango or libnrtype
             ellipse->cx.update(em, ex, dx);
