--- ./src/ui/dialog/tile.cpp.orig	2015-07-20 05:02:37.760813965 +0200
+++ ./src/ui/dialog/tile.cpp	2015-07-20 05:03:45.198686120 +0200
@@ -581,8 +581,8 @@
                 prefs->setInt("/dialogs/gridtiler/NoOfCols", NoOfCols);
             }
         } else {
-            double PerRow = ceil(sqrt(selcount));
-            double PerCol = ceil(sqrt(selcount));
+            double PerRow = ceil(sqrt((double)selcount));
+            double PerCol = ceil(sqrt((double)selcount));
             NoOfRowsSpinner.set_value(PerRow);
             NoOfColsSpinner.set_value(PerCol);
             prefs->setInt("/dialogs/gridtiler/NoOfCols", static_cast<int>(PerCol));
@@ -647,8 +647,8 @@
 
     /*#### Number of Rows ####*/
 
-    double PerRow = ceil(sqrt(selcount));
-    double PerCol = ceil(sqrt(selcount));
+    double PerRow = ceil(sqrt((double)selcount));
+    double PerCol = ceil(sqrt((double)selcount));
 
     #ifdef DEBUG_GRID_ARRANGE
         g_print("/n PerRox = %f PerCol = %f selcount = %d",PerRow,PerCol,selcount);
