*** a2ps-4.14/sheets/sheets.map.orig	Wed May 12 15:37:37 2004
--- a2ps-4.14/sheets/sheets.map	Wed May 12 15:48:45 2004
***************
*** 237,242 ****
--- 237,243 ----
  # GIF Images
  gif:	/*.gif/i
  	<GIF image*>
+ 	<GIF file*>
  
  # Haskell programs
  haskell:
***************
*** 326,331 ****
--- 327,333 ----
  # PDF documents
  pdf:	/*.pdf/i
  	<PDF document*>
+ 	<* (PDF) *>
  
  # Perl files
  perl:	/*.p[ml]/
***************
*** 338,343 ****
--- 340,346 ----
  # PNG files
  png:	/*.png/i
  	<PNG image*>
+ 	<PNG file*>
  
  # Portable Object (Gettext)
  po:	/*.po/
***************
*** 474,481 ****
  	<*vtcl*>
  
  # TIFF images
! tiff:	/.tif/i  /.tiff/i
  	<TIFF image*>
  
  # tk files
  tk:	/*.tk/
--- 477,485 ----
  	<*vtcl*>
  
  # TIFF images
! tiff:	/*.tif/i  /*.tiff/i
  	<TIFF image*>
+ 	<TIFF file*>
  
  # tk files
  tk:	/*.tk/
