--- a2ps-4.14/tests/tstfiles/InsertBlock.java.orig
+++ a2ps-4.14/tests/tstfiles/InsertBlock.java
@@ -1,5 +1,5 @@
 /* This code is part of Freenet. It is distributed under the GNU General
- * Public License, version 2 (or at your option any later version). See
+ * Public License, version 3 (or at your option any later version). See
  * http://www.gnu.org/ for further details of the GPL. */
 package freenet.client;
 
--- a2ps-4.14/tests/tstfiles/bookie.idl.orig
+++ a2ps-4.14/tests/tstfiles/bookie.idl
@@ -4,7 +4,7 @@
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
-   the Free Software Foundation; either version 2, or (at your option)
+   the Free Software Foundation; either version 3, or (at your option)
    any later version.
 
    (c) Copyright 1999 Bob Phillips
