--- ./Makefile.am.orig	2016-06-01 03:36:07.046079960 +0200
+++ ./Makefile.am	2016-06-01 03:36:37.728290264 +0200
@@ -97,7 +97,7 @@
 	src/utf8.h \
 	src/utils.c \
 	src/utils.h
-libxkbcommon_la_LDFLAGS = -Wl,--version-script=${srcdir}/xkbcommon.map
+libxkbcommon_la_LDFLAGS = -Wl,-M${srcdir}/xkbcommon.map
 
 if ENABLE_X11
 pkgconfig_DATA += xkbcommon-x11.pc
@@ -111,7 +111,7 @@
 libxkbcommon_x11_la_CFLAGS = $(AM_CFLAGS) $(XCB_XKB_CFLAGS)
 libxkbcommon_x11_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/src/x11
 libxkbcommon_x11_la_LIBADD = libxkbcommon.la $(XCB_XKB_LIBS)
-libxkbcommon_x11_la_LDFLAGS = -Wl,--version-script=${srcdir}/xkbcommon-x11.map
+libxkbcommon_x11_la_LDFLAGS = -Wl,-M${srcdir}/xkbcommon-x11.map
 
 libxkbcommon_x11_la_SOURCES = \
 	src/x11/keymap.c \
