--- libxkbcommon-0.8.0/Makefile.am.orig	2018-03-14 21:34:18.363184296 +0000
+++ libxkbcommon-0.8.0/Makefile.am	2018-03-14 21:36:13.207491147 +0000
@@ -107,7 +107,7 @@
 	src/utils.c \
 	src/utils.h
 if HAVE_VERSION_SCRIPT
-libxkbcommon_la_LDFLAGS = -Wl,--version-script=${top_srcdir}/xkbcommon.map
+libxkbcommon_la_LDFLAGS = -Wl,-M${top_srcdir}/xkbcommon.map
 endif HAVE_VERSION_SCRIPT
 
 if ENABLE_X11
@@ -123,7 +123,7 @@
 libxkbcommon_x11_la_CPPFLAGS = $(AM_CPPFLAGS)
 libxkbcommon_x11_la_LIBADD = libxkbcommon.la $(XCB_XKB_LIBS)
 if HAVE_VERSION_SCRIPT
-libxkbcommon_x11_la_LDFLAGS = -Wl,--version-script=${top_srcdir}/xkbcommon-x11.map
+libxkbcommon_x11_la_LDFLAGS = -Wl,-M${top_srcdir}/xkbcommon-x11.map
 endif HAVE_VERSION_SCRIPT
 
 libxkbcommon_x11_la_SOURCES = \
