--- freeway-d563ac6c0b78f8179e38b4e4b9f4976f41f9001e/src/Makefile.1	2026-06-19 14:57:55.750811726 -0400
+++ freeway-d563ac6c0b78f8179e38b4e4b9f4976f41f9001e/src/Makefile	2026-06-19 14:58:08.742431367 -0400
@@ -9,11 +9,11 @@
 AR = ar
 PREFIX = /usr/
 BINDIR = ${PREFIX}/bin
-LIBDIR = ${PREFIX}/lib
+LIBDIR = ${PREFIX}/lib/MACH64
 INCLUDEDIR = ${PREFIX}/include
 DATADIR = ${PREFIX}/share
 PKGCONFIGDIR = ${LIBDIR}/pkgconfig
-PLATFORM = linux
+PLATFORM = sun
 
 PROTOCOL_XML = ../protocol/wayland.xml
 PROTOCOL_DTD = ../protocol/wayland.dtd
