--- vte-0.22.5/configure.in-orig	2010-04-07 18:02:09.412286645 -0500
+++ vte-0.22.5/configure.in	2010-04-07 18:02:17.857283149 -0500
@@ -152,7 +152,7 @@ MAYBE_WARN="-Wall -Wextra \
 -Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
 -Wdeclaration-after-statement -Wold-style-definition \
 -Wno-missing-field-initializers -Wno-unused-parameter \
--fno-common -Wno-switch-enum"
+-Wno-switch-enum"
 
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$vte_cv_warn_maybe" != "x$MAYBE_WARN"; then
