--- urllib3-2.6.3/test/with_dummyserver/test_socketlevel.py.orig
+++ urllib3-2.6.3/test/with_dummyserver/test_socketlevel.py
@@ -1899,6 +1899,7 @@
         "method",
         [_run_preload, _run_read_None, _run_read_amt, _run_read1_None, _run_read1_amt],
     )
+    @pytest.mark.skip(reason="needs a lot of memory")
     def test_requesting_large_resources_via_ssl(
         self, method: typing.Callable[[typing.Any, HTTPSConnectionPool, int], None]
     ) -> None:
