diff -uNr libgnome-java-2.12.6.orig/doc/examples/runExample.sh.in libgnome-java-2.12.6/doc/examples/runExample.sh.in
--- libgnome-java-2.12.6.orig/doc/examples/runExample.sh.in	2006-10-26 14:52:02.183551000 +0100
+++ libgnome-java-2.12.6/doc/examples/runExample.sh.in	2006-10-26 15:55:31.961544000 +0100
@@ -3,14 +3,6 @@
 # A small script to help run example apps
 #
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-classpath=@INSTALLED_CLASSPATH@
-
-GNOMEJAVA_LIB=$libdir
-GNOMEJAVA_JAR=$classpath
-
 # You should not need to modify anything below here
 
 error_message ()
@@ -30,8 +22,8 @@
 echo "Java-Gnome Example Application Launcher"
 echo
 
-GLIBJAVA_JAR=`pkg-config --variable classpath glib-java`
-if ! test -n "$GLIBJAVA_JAR"; then
+JGLIB_JAR=`pkg-config --variable classpath glib-java`
+if ! test -n "$JGLIB_JAR"; then
 	echo "Package glib-java was not found in the pkg-config search path."
 	echo "Perhaps you should add the directory containing 'glib-java.pc'"
 	echo "to the PKG_CONFIG_PATH environment variable"
@@ -39,35 +31,36 @@
 	exit 1
 fi
 
-GLIBJAVA_LIB=`pkg-config --variable libdir glib-java`
-if ! test -n "$GLIBJAVA_LIB"; then
-	echo "Library requirements (libglibjni >= 0.2) not met;"
-	echo "consider adjusting the PKG_CONFIG_PATH environment" 
-	echo "variable if your libraries are in a nonstandard prefix" 
-	echo "so pkg-config can find them."
+JGTK_JAR=`pkg-config --variable classpath gtk2-java`
+if ! test -n "$JGTK_JAR"; then
+	echo "Package gtk2-java was not found in the pkg-config search path."
+	echo "Perhaps you should add the directory containing 'gtk2-java.pc'"
+	echo "to the PKG_CONFIG_PATH environment variable"
+	echo "No package 'gtk2-java' found"
 	exit 1
 fi
 
-GTKJAVA_JAR=`pkg-config --variable classpath gtk2-java`
-if ! test -n "$GTKJAVA_JAR"; then
-	echo "Package gtk2-java was not found in the pkg-config search path."
-	echo "Perhaps you should add the directory containing 'gtk2-java.pc'"
+JGNOME_JAR=`pkg-config --variable classpath gnome2-java`
+if ! test -n "$JGNOME_JAR"; then
+	echo "Package gnome2-java was not found in the pkg-config search path."
+	echo "Perhaps you should add the directory containing 'gnome2-java.pc'"
 	echo "to the PKG_CONFIG_PATH environment variable"
 	echo "No package 'gtk2-java' found"
 	exit 1
 fi
 
-GTKJAVA_LIB=`pkg-config --variable libdir gtk2-java`
-if ! test -n "$GTKJAVA_LIB"; then
-	echo "Library requirements (libgtkjava2.4) not met;"
+JGNOME_LIBDIR=`pkg-config --variable libdir gnome2-java`
+if ! test -n "$JGNOME_LIBDIR"; then
+	echo "Library requirements for gnome2-java not met;"
 	echo "consider adjusting the PKG_CONFIG_PATH environment" 
 	echo "variable if your libraries are in a nonstandard prefix" 
 	echo "so pkg-config can find them."
 	exit 1
 fi
 
-NEED_LIB="$GLIBJAVA_LIB $GTKJAVA_LIB $GNOMEJAVA_LIB"
-NEED_JAR="$GLIBJAVA_JAR $GTKJAVA_JAR $GNOMEJAVA_JAR"
+
+NEED_LIB="$JGNOME_LIBDIR"
+NEED_JAR="$JGLIB_JAR $JGTK_JAR $JGNOME_JAR"
 
 echo "Checking for jvm and compiler availability..."
 
