diff -urN gconf.orig/gconf/default.path.in gconf.new/gconf/default.path.in
--- gconf.orig/gconf/default.path.in	2009-03-20 17:22:05.699831017 +0000
+++ gconf.new/gconf/default.path.in	2009-03-20 17:23:26.092463128 +0000
@@ -21,7 +21,8 @@
 xml:readwrite:$(HOME)/.gconf
 
 # Location for system-wide settings that are set by the defaults mechanism
-xml:readonly:/etc/gconf/gconf.xml.system
+# We dont use this default mechanism on opensolaris
+# xml:readonly:/etc/gconf/gconf.xml.system
 
 # To read in any defaults settings that the Sys Admin may have created
 # prior to a desktop system upgrade. The SysAdmin can stick default values
