--- GConf-2.32.4/gconf/gconf-client.c.~1~	2010-09-10 12:14:17.000000000 +0400
+++ GConf-2.32.4/gconf/gconf-client.c	2015-08-05 18:29:18.502454311 +0300
@@ -814,7 +814,7 @@
   
   gconf_listeners_remove(client->listeners, cnxn);
   
-  if (gconf_listeners_count(client->listeners) == 0)
+  if (client->listeners && gconf_listeners_count(client->listeners) == 0)
     {
       gconf_listeners_free(client->listeners);
       client->listeners = NULL;
