Upstream contribution is planned but has not yet occurred.

--- postfix-2.11.1/makedefs.orig	2014-01-16 11:49:11.000000000 -0800
+++ postfix-2.11.1/makedefs	2014-09-19 13:43:56.091926293 -0700
@@ -217,9 +217,13 @@
 		case $RELEASE in
 		    5.[0-8]|5.[0-8].*) CCARGS="$CCARGS -DNO_CLOSEFROM -DNO_DEV_URANDOM -DNO_FUTIMESAT -DSTREAM_CONNECTIONS";;
 		esac
-		# Somewhere NISPLUS went away.
+		# NISPLUS went away in Solaris 11.
 		case $RELEASE in
-		    5.[0-9][0-9]*) CCARGS="$CCARGS -DNO_NISPLUS";;
+		    5.1[1-9]*) CCARGS="$CCARGS -DNO_NISPLUS";;
+		esac
+		# libsocket & libnsl were merged into libc in Solaris 12.
+		case $RELEASE in
+		    5.1[2-9]*) SYSLIBS="-lresolv -ldl"
 		esac
 		# Work around broken str*casecmp(). Do it all here instead
 		# of having half the solution in the sys_defs.h file.
