From 0358269de147ea7a05c32a1c495e73ea373cb4f4 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Tue, 14 Oct 2014 19:42:28 +0300
Subject: [PATCH] Allow the dbus calls

---
 data/ConsoleKit.conf | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/data/ConsoleKit.conf b/data/ConsoleKit.conf
index 911f5b9..35c79a8 100644
--- a/data/ConsoleKit.conf
+++ b/data/ConsoleKit.conf
@@ -36,6 +36,27 @@
     <allow send_destination="org.freedesktop.ConsoleKit"
            send_interface="org.freedesktop.ConsoleKit.Manager"
            send_member="CanStop"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="Suspend"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="CanSuspend"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="AuthSuspend"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="Hibernate"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="CanHibernate"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="AuthHibernate"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="Inhibit"/>
     <allow send_destination="org.freedesktop.ConsoleKit"
            send_interface="org.freedesktop.ConsoleKit.Manager"
            send_member="OpenSession"/>
