--- python-lazy-object-proxy-1.6.0/setup.cfg-orig	Mon Mar 22 10:06:24 2021
+++ python-lazy-object-proxy-1.6.0/setup.cfg	Mon Jul 19 12:52:48 2021
@@ -24,6 +24,7 @@
     *_test.py
     tests.py
 markers =
+    benchmark: Baseline of pylint performance.
     xfail_subclass: Expected test to fail with a subclass of Proxy.
     xfail_simple: Expected test to fail on the `simple` implementation.
 addopts =
