https://github.com/pallets/werkzeug/pull/2773

--- werkzeug-2.3.7/src/werkzeug/debug/__init__.py.orig
+++ werkzeug-2.3.7/src/werkzeug/debug/__init__.py
@@ -116,6 +116,15 @@
             except OSError:
                 pass
 
+        # On illumos and Solaris use the hostid(1) tool.
+        if sys.platform == "sunos5":
+            from subprocess import Popen, PIPE
+            hostid = Popen(
+                ["/usr/bin/hostid"], stdout=PIPE
+            ).communicate()[0]
+            if hostid is not "":
+                return hostid
+
         return None
 
     _machine_id = _generate()
