--- websockets-12.0/tests/sync/test_connection.py.orig
+++ websockets-12.0/tests/sync/test_connection.py
@@ -660,6 +660,7 @@
         """Connection has a local_address attribute."""
         self.assertIsNotNone(self.connection.local_address)
 
+    @unittest.skip('always fails: https://github.com/python-websockets/websockets/issues/1427')
     def test_remote_address(self):
         """Connection has a remote_address attribute."""
         self.assertIsNotNone(self.connection.remote_address)
