diff -urN hexchat-2.10.2.orig/plugins/perl/Makefile.in hexchat-2.10.2/plugins/perl/Makefile.in
--- hexchat-2.10.2.orig/plugins/perl/Makefile.in	2015-12-13 01:32:05.598558376 +0100
+++ hexchat-2.10.2/plugins/perl/Makefile.in	2015-12-13 01:33:09.176245776 +0100
@@ -727,7 +727,7 @@
 hexchat.pm.h irc.pm.h: lib/HexChat.pm lib/Xchat.pm lib/HexChat/Embed.pm \
 	lib/HexChat/List/Network.pm lib/HexChat/List/Network/Entry.pm \
 	lib/HexChat/List/Network/AutoJoin.pm lib/IRC.pm
-	perl generate_header
+	cd $(srcdir) ; perl generate_header
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
