diff -urN hexchat-2.10.2.orig/src/common/Makefile.in hexchat-2.10.2/src/common/Makefile.in
--- hexchat-2.10.2.orig/src/common/Makefile.in	2014-11-25 19:04:10.000000000 +0100
+++ hexchat-2.10.2/src/common/Makefile.in	2015-12-13 01:27:15.251635609 +0100
@@ -1309,7 +1309,7 @@
 textenums.h: textevents.h
 
 textevents.h: textevents.in make-te
-	$(AM_V_GEN) ./make-te < textevents.in > textevents.h 2> textenums.h
+	$(AM_V_GEN) ./make-te < $(srcdir)/textevents.in > textevents.h 2> textenums.h
 
 marshal.h: marshalers.list
 	$(AM_V_GEN) $(GLIB_GENMARSHAL) --prefix=_hexchat_marshal --header $(srcdir)/marshalers.list > $@
diff -urN hexchat-2.10.2.orig/src/fe-gtk/Makefile.in hexchat-2.10.2/src/fe-gtk/Makefile.in
--- hexchat-2.10.2.orig/src/fe-gtk/Makefile.in	2014-11-25 19:04:10.000000000 +0100
+++ hexchat-2.10.2/src/fe-gtk/Makefile.in	2015-12-13 01:28:51.539657992 +0100
@@ -380,7 +380,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 unamepath = @unamepath@
-AM_CPPFLAGS = $(GUI_CFLAGS) -DLOCALEDIR=\"$(localedir)\"
+AM_CPPFLAGS = $(GUI_CFLAGS) -DLOCALEDIR=\"$(localedir)\" -I$(top_builddir)/src/common
 hexchat_LDADD = ../common/libhexchatcommon.a $(GUI_LIBS)
 EXTRA_DIST = \
 	ascii.h banlist.h chanlist.h chanview.h chanview-tabs.c \
