--- curl-7.60.0/configure.ac.1	2018-05-17 19:52:38.637214847 +0000
+++ curl-7.60.0/configure.ac	2018-05-17 19:56:10.531967993 +0000
@@ -1401,6 +1401,9 @@
            *-hp-hpux*)
               gss_libname="gss"
               ;;
+           *sunos*)
+              gss_libname="gss"
+              ;;
            *)
               gss_libname="gssapi"
               ;;
@@ -1422,7 +1425,7 @@
         LIBS="-lgss $LIBS"
         ;;
      *)
-        LIBS="-lgssapi $LIBS"
+        LIBS="-lgss $LIBS"
         ;;
      esac
   fi
