--- curl-8.11.0/libcurl.pc.in.~1~	2024-11-06 19:34:59.490073469 -0500
+++ curl-8.11.0/libcurl.pc.in	2024-11-06 19:37:54.461813779 -0500
@@ -35,7 +35,7 @@
 Version: @CURLVERSION@
 Requires: @LIBCURL_PC_REQUIRES@
 Requires.private: @LIBCURL_PC_REQUIRES_PRIVATE@
-Libs: -L${libdir} -lcurl @LIBCURL_PC_LIBS@
+Libs: -lcurl @LIBCURL_PC_LIBS@
 Libs.private: @LDFLAGS@ @LIBCURL_PC_LIBS_PRIVATE@
-Cflags: -I${includedir} @LIBCURL_PC_CFLAGS@
+Cflags: @LIBCURL_PC_CFLAGS@
 Cflags.private: @LIBCURL_PC_CFLAGS_PRIVATE@
