--- php-5.3.10/ext/mysql/config.m4_orig	Mon Nov 30 07:11:29 2009
+++ php-5.3.10/ext/mysql/config.m4	Wed Feb  8 11:25:25 2012
@@ -78,7 +78,9 @@
 Note that the MySQL client library is not bundled anymore!])
   fi
 
-  if test "$enable_maintainer_zts" = "yes"; then
+  use_reentrant_mysqlclient="yes"
+  if test "$enable_maintainer_zts" = "yes" -o \
+    "$use_reentrant_mysqlclient" = "yes" ; then
     MYSQL_LIBNAME=mysqlclient_r
   else
     MYSQL_LIBNAME=mysqlclient
@@ -132,7 +134,9 @@
   MYSQL_MODULE_TYPE=external
   MYSQL_LIBS="-L$MYSQL_LIB_DIR -l$MYSQL_LIBNAME $MYSQL_LIBS"
   MYSQL_INCLUDE=-I$MYSQL_INC_DIR
+  MYSQL_SHARED_LIBADD="$MYSQL_SHARED_LIBADD -R/usr/mysql/5.1/lib/mysql"
  
+  PHP_SUBST(MYSQL_SHARED_LIBADD)
   PHP_SUBST_OLD(MYSQL_MODULE_TYPE)
   PHP_SUBST_OLD(MYSQL_LIBS)
   PHP_SUBST_OLD(MYSQL_INCLUDE)
