--- php-5.3.10/ext/pdo_mysql/config.m4_orig	Sat May 14 22:49:34 2011
+++ php-5.3.10/ext/pdo_mysql/config.m4	Wed Feb  8 11:25:25 2012
@@ -64,7 +64,9 @@
       if test "x$SED" = "x"; then
         AC_PATH_PROG(SED, sed)
       fi
-      if test "$enable_maintainer_zts" = "yes"; then
+      use_reentrant_mysqlclient="yes"
+      if test "$enable_maintainer_zts" = "yes" -o \
+        "$use_reentrant_mysqlclient" = "yes" ; then
         PDO_MYSQL_LIBNAME=mysqlclient_r
         PDO_MYSQL_LIBS=`$PDO_MYSQL_CONFIG --libs_r | $SED -e "s/'//g"`
       else
@@ -167,6 +169,7 @@
     fi
   ])
   PDO_MYSQL_MODULE_TYPE=external
+  PDO_MYSQL_SHARED_LIBADD="$PDO_MYSQL_SHARED_LIBADD -R/usr/mysql/5.1/lib/mysql"
 
   PHP_SUBST(PDO_MYSQL_SHARED_LIBADD)
   PHP_SUBST_OLD(PDO_MYSQL_MODULE_TYPE)
