--- php-5.3.10/ext/mysqli/config.m4_orig	Fri Aug 20 05:25:17 2010
+++ php-5.3.10/ext/mysqli/config.m4	Wed Feb  8 11:25:25 2012
@@ -23,6 +23,7 @@
     MYSQL_CONFIG=$PHP_MYSQLI
   fi
 
+  use_reentrant_mysqlclient="yes"
   MYSQL_LIB_NAME='mysqlclient'
   if test "$PHP_EMBEDDED_MYSQLI" = "yes"; then
     AC_DEFINE(HAVE_EMBEDDED_MYSQLI, 1, [embedded MySQL support enabled])
@@ -29,7 +30,8 @@
     MYSQL_LIB_CFG='--libmysqld-libs'
     dnl mysqlnd doesn't support embedded, so we have to add some extra stuff
     mysqli_extra_sources="mysqli_embedded.c"
-  elif test "$enable_maintainer_zts" = "yes"; then
+  elif test "$enable_maintainer_zts" = "yes" -o \
+    "$use_reentrant_mysqlclient" = "yes" ; then
     MYSQL_LIB_CFG='--libs_r'
     MYSQL_LIB_NAME='mysqlclient_r'
   else
@@ -79,6 +81,7 @@
                   mysqli_fe.c mysqli_report.c mysqli_driver.c mysqli_warning.c \
                   mysqli_exception.c $mysqli_extra_sources"
   PHP_NEW_EXTENSION(mysqli, $mysqli_sources, $ext_shared)
+  MYSQLI_SHARED_LIBADD="$MYSQLI_SHARED_LIBADD -R/usr/mysql/5.1/lib/mysql"
   PHP_SUBST(MYSQLI_SHARED_LIBADD)
   PHP_INSTALL_HEADERS([ext/mysqli/php_mysqli_structs.h])
 
