--- php-5.3.10/sapi/apache/config.m4_orig	Wed Jul 11 16:20:37 2007
+++ php-5.3.10/sapi/apache/config.m4	Wed Feb  8 11:25:25 2012
@@ -18,6 +18,23 @@
 fi
 ])
 
+dnl PHP_AP2_EXTRACT_VERSION(/path/httpd)
+dnl
+dnl This macro is used to get a comparable
+dnl version for apache1/2.
+dnl
+AC_DEFUN([PHP_AP2_EXTRACT_VERSION],[
+  ac_output=`env LD_LIBRARY_PATH=$APXS_LIBDIR $1 -v 2>&1 | grep version`
+  ac_IFS=$IFS
+IFS="- /.
+"
+  set $ac_output
+  IFS=$ac_IFS
+
+  APACHE_VERSION=`expr [$]4 \* 1000000 + [$]5 \* 1000 + [$]6`
+])
+
+dnl
 dnl Apache 1.x shared module
 PHP_ARG_WITH(apxs,,
 [  --with-apxs[=FILE]      Build shared Apache 1.x module. FILE is the optional
@@ -52,6 +69,7 @@
 
   APXS_LDFLAGS="@SYBASE_LFLAGS@ @SYBASE_LIBS@ @SYBASE_CT_LFLAGS@ @SYBASE_CT_LIBS@"
   APXS_INCLUDEDIR=`$APXS -q INCLUDEDIR`
+  APXS_LIBDIR=`$APXS -q LIBDIR`
   APXS_CFLAGS=`$APXS -q CFLAGS`
   APXS_HTTPD=`$APXS -q SBINDIR`/`$APXS -q TARGET`
   APACHE_INCLUDE=-I$APXS_INCLUDEDIR
@@ -58,6 +76,7 @@
 
   # Test that we're trying to configure with apache 1.x
   PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
+  PHP_AP2_EXTRACT_VERSION($APXS_HTTPD)
   if test "$APACHE_VERSION" -ge 2000000; then
     AC_MSG_ERROR([You have enabled Apache 1.3 support while your server is Apache 2.  Please use the appropiate switch --with-apxs2]) 
   fi
