--- php-5.3.10_orig/Zend/zend_dtrace_gen.php	Wed Nov 23 11:34:50 2011
+++ php-5.3.10/Zend/zend_dtrace_gen.php	Wed Nov 23 11:36:10 2011
@@ -0,0 +1,31 @@
+<?php
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 1998-2009 Zend Technologies Ltd. (http://www.zend.com) |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: David Soria Parra <david.soriaparra@sun.com>                |
+   +----------------------------------------------------------------------+
+
+   $Id:$
+*/
+
+echo "regenerating zend_dtrace.h\n";
+system('dtrace -h -s zend_dtrace.d');
+
+echo "renaming probe definitions\n";
+$content = file_get_contents('zend_dtrace.h');
+$content = str_replace('#define        PHP_', '#define DTRACE_', $content);
+$content = str_replace('PHP_STABILITY', 'DTRACE_STABILITY', $content);
+$content = str_replace('PHP_TYPEDEFS', 'DTRACE_TYPEDEFS', $content);
+$content = str_replace('#if _DTRACE_VERSION', '#if _DTRACE_VERSION && HAVE_SYS_SDT_H', $content);
+file_put_contents('zend_dtrace.h', $content);
