--- php-5.3.10/Zend/zend_objects.c_orig	Sun Jan  1 05:15:04 2012
+++ php-5.3.10/Zend/zend_objects.c	Wed Feb  8 11:25:25 2012
@@ -25,6 +25,7 @@
 #include "zend_API.h"
 #include "zend_interfaces.h"
 #include "zend_exceptions.h"
+#include "zend_dtrace.h"
 
 ZEND_API void zend_object_std_init(zend_object *object, zend_class_entry *ce TSRMLS_DC)
 {
@@ -51,6 +52,16 @@
 {
 	zend_function *destructor = object ? object->ce->destructor : NULL;
 
+#ifdef HAVE_DTRACE
+       if (DTRACE_OBJECT_DESTROY_ENABLED()) {
+               char *filename;
+               int lineno;
+
+               filename = dtrace_get_executed_filename(TSRMLS_C);
+               lineno = zend_get_executed_lineno(TSRMLS_C);
+               DTRACE_OBJECT_DESTROY(object->ce->name, filename, lineno);
+       }
+#endif /* HAVE_DTRACE */
 	if (destructor) {
 		zval *old_exception;
 		zval *obj;
