--- php-5.3.10/Zend/zend_vm_def.h_orig	Sun Jan  1 05:15:04 2012
+++ php-5.3.10/Zend/zend_vm_def.h	Wed Feb  8 11:25:26 2012
@@ -2569,6 +2569,11 @@
 		ZEND_VM_CONTINUE(); /* CHECK_ME */
 	}
 	ce = Z_OBJCE_P(EG(exception));
+
+#ifdef HAVE_DTRACE
+       DTRACE_EXCEPTION_CAUGHT(ce->name);
+#endif /* HAVE_DTRACE */
+
 	if (ce != EX_T(opline->op1.u.var).class_entry) {
 		if (!instanceof_function(ce, EX_T(opline->op1.u.var).class_entry TSRMLS_CC)) {
 			if (opline->op1.u.EA.type) {
