We do not have these projects packaged yet.

--- elasticsearch-py-9.0.2/pyproject.toml.orig
+++ elasticsearch-py-9.0.2/pyproject.toml
@@ -64,26 +64,18 @@
     "coverage",
     "jinja2",
     "python-dateutil",
-    "unasync",
     "pyyaml>=5.4",
     "isort",
     "black",
     "twine",
     "build",
     "nox",
-    "orjson",
     "numpy",
-    "simsimd",
     "pyarrow",
-    "pandas",
-    "mapbox-vector-tile",
     "jinja2",
-    "sentence_transformers",
     "tqdm",
     "mypy",
-    "pyright",
     "types-python-dateutil",
-    "types-tqdm",
 ]
 docs = [
     "sphinx",
