We do not have these projects packaged yet.

--- elasticsearch-py-8.17.2/pyproject.toml.orig
+++ elasticsearch-py-8.17.2/pyproject.toml
@@ -61,19 +61,14 @@
     "coverage",
     "jinja2",
     "python-dateutil",
-    "unasync",
     "pyyaml>=5.4",
     "isort",
     "black",
     "twine",
     "build",
     "nox",
-    "orjson",
     "numpy",
-    "simsimd",
     "pyarrow",
-    "pandas",
-    "mapbox-vector-tile",
 ]
 docs = [
     "sphinx",
