We have no pyarrow packaged yet.

--- elasticsearch-py-8.15.1/pyproject.toml.orig
+++ elasticsearch-py-8.15.1/pyproject.toml
@@ -67,7 +67,6 @@
     "build",
     "nox",
     "numpy",
-    "pyarrow",
 ]
 docs = [
     "sphinx",
--- elasticsearch-py-8.15.1/test_elasticsearch/test_client/test_deprecated_options.py.orig
+++ elasticsearch-py-8.15.1/test_elasticsearch/test_client/test_deprecated_options.py
@@ -134,7 +134,6 @@
         "application/x-ndjson",
         "application/json",
         "text/*",
-        "application/vnd.apache.arrow.stream",
         "application/vnd.elasticsearch+json",
         "application/vnd.elasticsearch+x-ndjson",
     }
@@ -155,7 +154,6 @@
         "application/x-ndjson",
         "application/json",
         "text/*",
-        "application/vnd.apache.arrow.stream",
         "application/vnd.elasticsearch+json",
         "application/vnd.elasticsearch+x-ndjson",
         "application/cbor",
--- elasticsearch-py-8.15.1/test_elasticsearch/test_client/test_serializers.py.orig
+++ elasticsearch-py-8.15.1/test_elasticsearch/test_client/test_serializers.py
@@ -94,7 +94,6 @@
             "application/json",
             "text/*",
             "application/x-ndjson",
-            "application/vnd.apache.arrow.stream",
             "application/vnd.mapbox-vector-tile",
             "application/vnd.elasticsearch+json",
             "application/vnd.elasticsearch+x-ndjson",
@@ -122,7 +121,6 @@
             "application/json",
             "text/*",
             "application/x-ndjson",
-            "application/vnd.apache.arrow.stream",
             "application/vnd.mapbox-vector-tile",
             "application/vnd.elasticsearch+json",
             "application/vnd.elasticsearch+x-ndjson",
@@ -142,7 +140,6 @@
             "application/json",
             "text/*",
             "application/x-ndjson",
-            "application/vnd.apache.arrow.stream",
             "application/vnd.mapbox-vector-tile",
             "application/vnd.elasticsearch+json",
             "application/vnd.elasticsearch+x-ndjson",
