--- xdebug-2.0.5/debugclient/configure.in.orig	Thu Jan 19 06:06:16 2012
+++ xdebug-2.0.5/debugclient/configure.in	Thu Jan 19 06:06:28 2012
@@ -3,7 +3,6 @@
 dnl Process this file with autoconf to produce a configure script.
 dnl
 
-divert(1)
 
 dnl
 dnl Start
@@ -19,7 +18,6 @@
 # This is needed if SUBDIRS is set
 AC_PROG_MAKE_SET
 
-divert(2)
 dnl
 dnl Checks for working compiler and other programs needed.
 dnl
