--- uchardet-0.0.8/src/symbols.cmake.orig	2022-12-08 21:52:25.000000000 +0100
+++ uchardet-0.0.8/src/symbols.cmake	2023-07-06 16:48:44.116351028 +0200
@@ -24,7 +24,7 @@
 	set(_version_script "${CMAKE_CURRENT_BINARY_DIR}/version.script")
 	file(WRITE ${_version_script} "${_symbols}\n")
 
-	set(LINK_FLAGS "${LINK_FLAGS} -Wl,--version-script,\"${_version_script}\"")
+	set(LINK_FLAGS "${LINK_FLAGS} -W")
 endif (APPLE)
 
 set_target_properties(
