--- gdk-pixbuf-2.42.12/tests/meson.build.orig
+++ gdk-pixbuf-2.42.12/tests/meson.build
@@ -4,7 +4,7 @@
 # gdk-pixbuf-pixdata from build directory because it needs all DLL locations in
 # $PATH. Ideally we should use gnome.compile_resources() and let Meson deal with
 # this problem: See https://github.com/mesonbuild/meson/issues/8266.
-if enabled_loaders.contains('png') and host_system != 'windows'
+if enabled_loaders.contains('png') and host_system != 'windows' and host_system != 'sunos'
   # Resources; we cannot use gnome.compile_resources() here, because we need to
   # override the environment in order to use the utilities we just built instead
   # of the system ones
