Patch taken from https://www.illumos.org/issues/3853

--- math/i386/usr/include/floatingpoint.h	2013-07-11 10:15:18.170057704 +0400
+++ math/i386/usr/include/floatingpoint.h	2013-07-11 10:16:04.879688599 +0400
@@ -197,12 +197,21 @@
 extern char *fcvt __P((double, int, int *, int *));
 extern char *gcvt __P((double, int, char *));
 
+#if __cplusplus >= 199711L
+namespace std {
+#endif
 /*
  * ANSI C Standard says the following entry points should be
  * prototyped in <stdlib.h>.  They are now, but weren't before.
  */
 extern double atof __P((const char *));
 extern double strtod __P((const char *, char **));
+#if __cplusplus >= 199711L
+}
+
+using std::atof;
+using std::strtod;
+#endif /* end of namespace std */
 
 #undef	_FLOATINGPOINT_FILE
 
--- math/sparc/usr/include/floatingpoint.h	2013-07-11 10:16:19.869517751 +0400
+++ math/sparc/usr/include/floatingpoint.h	2013-07-11 10:16:48.967195870 +0400
@@ -197,12 +197,21 @@
 extern char *fcvt __P((double, int, int *, int *));
 extern char *gcvt __P((double, int, char *));
 
+#if __cplusplus >= 199711L
+namespace std {
+#endif
 /*
  * ANSI C Standard says the following entry points should be
  * prototyped in <stdlib.h>.  They are now, but weren't before.
  */
 extern double atof __P((const char *));
 extern double strtod __P((const char *, char **));
+#if __cplusplus >= 199711L
+} /* end of namespace std */
+
+using std::atof;
+using std::strtod;
+#endif
 
 #undef	_FLOATINGPOINT_FILE
 
