--- libmtp-1.1.8/configure.ac.~1~	2015-03-11 00:53:06.170636796 +0300
+++ libmtp-1.1.8/configure.ac	2015-03-11 00:54:22.763758794 +0300
@@ -132,6 +132,7 @@
          AC_CHECK_LIB([usb], [usb_open],[
             LIBUSB_VER=0.1
             AC_DEFINE(HAVE_LIBUSB0, [], [Have libusb 0.x])
+            LIBUSB_LIBS="-lusb"
             found_libusb0_pkg=yes
          ],[
             AC_MSG_ERROR([*** libusb not found! (missing -dev/-devel package?) ***\n*** prefer to install libusb-1 if you can! ***])
