diff -urN evince-2.30.3/backend/pdf/ev-poppler.cc ../SUNWgnome-pdf-viewer-2.30.3.hacked/evince-2.30.3/backend/pdf/ev-poppler.cc
--- evince-2.30.3/backend/pdf/ev-poppler.cc	2010-06-24 09:19:28.000000000 +0100
+++ ../SUNWgnome-pdf-viewer-2.30.3.hacked/evince-2.30.3/backend/pdf/ev-poppler.cc	2012-02-27 20:58:49.148832384 +0000
@@ -1990,6 +1990,8 @@
 					(PopplerSelectionStyle)style,
 					&r);
 
+	if (retval[strlen(retval)-1] == '\n')
+	  	retval[strlen(retval)-1] = '\0';
 	return retval;
 }
 
