--- evince-2.32.0/data/evince.desktop.in.in.~1~	2015-01-03 13:57:32.869441426 +0300
+++ evince-2.32.0/data/evince.desktop.in.in	2015-01-03 14:02:46.715634454 +0300
@@ -1,17 +1,20 @@
 [Desktop Entry]
-_Name=Document Viewer
-_Comment=View multi-page documents
+# SUN_BRANDING
+_Name=Evince Document Viewer
+# SUN_BRANDING
+_Comment=View and print PostScript and PDF documents
 TryExec=evince
 Exec=evince %U
 StartupNotify=true
 Terminal=false
 Type=Application
 Icon=evince
-NoDisplay=true
+# SUN_BRANDING
+NoDisplay=false
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
 X-GNOME-Bugzilla-Component=BugBuddyBugs
 X-GNOME-Bugzilla-Version=@VERSION@
-Categories=GNOME;GTK;Graphics;VectorGraphics;Viewer;
+Categories=GNOME;GTK;Application;Office;
 MimeType=@EVINCE_MIME_TYPES@
