diff -ruN evince-2.29.92.orig/data/evince.desktop.in.in evince-2.29.92/data/evince.desktop.in.in
--- evince-2.29.92.orig/data/evince.desktop.in.in	2010-03-13 07:17:44.503772460 +0000
+++ evince-2.29.92/data/evince.desktop.in.in	2010-03-13 07:19:11.221249740 +0000
@@ -1,18 +1,21 @@
 [Desktop Entry]
-_Name=Document Viewer
+# SUN_BRANDING
+_Name=Evince Document Viewer
 _GenericName=Document Viewer
-_Comment=View multi-page documents
+# SUN_BRANDING
+_Comment=View and print PostScript and PDF documents
 TryExec=evince
 Exec=evince %U
 StartupNotify=true
 Terminal=false
 Type=Application
 Icon=evince
-NoDisplay=true
+# SUN_BRANDING
+NoDisplay=false
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
 X-GNOME-Bugzilla-Component=BugBuddyBugs
 X-GNOME-Bugzilla-Version=@VERSION@
-Categories=GNOME;GTK;Graphics;VectorGraphics;Viewer;
+Categories=GNOME;GTK;Application;Office;
 MimeType=@EVINCE_MIME_TYPES@
