# 24314638 LLVM CommandLine subsystem is busted
# 3.9.X for upstrea.
--- tools/clang/utils/TableGen/TableGen.cpp	2016-01-03 23:51:46.000000000 -0500
+++ tools/clang/utils/TableGen/TableGen.cpp	2016-05-08 23:19:20.578431871 -0400
@@ -56,8 +56,7 @@
   GenAttrDocs
 };
 
-namespace {
-cl::opt<ActionType> Action(
+static cl::opt<ActionType> Action(
     cl::desc("Action to perform:"),
     cl::values(
         clEnumValN(GenClangAttrClasses, "gen-clang-attr-classes",
@@ -136,7 +135,7 @@
                    "Generate attribute documentation"),
         clEnumValEnd));
 
-cl::opt<std::string>
+static cl::opt<std::string>
 ClangComponent("clang-component",
                cl::desc("Only use warnings from specified component"),
                cl::value_desc("component"), cl::Hidden);
@@ -238,7 +237,6 @@
 
   return false;
 }
-}
 
 int main(int argc, char **argv) {
   sys::PrintStackTraceOnErrorSignal();
