# 24314638 LLVM CommandLine subsystem is busted
# 3.9.X for upstream.
--- utils/TableGen/AsmMatcherEmitter.cpp	2016-01-03 02:33:39.000000000 -0500
+++ utils/TableGen/AsmMatcherEmitter.cpp	2016-05-08 23:19:20.581431944 -0400
@@ -121,7 +121,7 @@
 #define DEBUG_TYPE "asm-matcher-emitter"
 
 static cl::opt<std::string>
-MatchPrefix("match-prefix", cl::init(""),
+MatchPrefix("match-prefix", cl::init(std::string("")),
             cl::desc("Only match instructions with the given prefix"));
 
 namespace {
--- utils/TableGen/TableGen.cpp	2016-01-03 23:51:51.000000000 -0500
+++ utils/TableGen/TableGen.cpp	2016-05-08 23:19:20.582431969 -0400
@@ -46,55 +46,55 @@
   GenAttributes
 };
 
-namespace {
-  cl::opt<ActionType>
-  Action(cl::desc("Action to perform:"),
-         cl::values(clEnumValN(PrintRecords, "print-records",
-                               "Print all records to stdout (default)"),
-                    clEnumValN(GenEmitter, "gen-emitter",
-                               "Generate machine code emitter"),
-                    clEnumValN(GenRegisterInfo, "gen-register-info",
-                               "Generate registers and register classes info"),
-                    clEnumValN(GenInstrInfo, "gen-instr-info",
-                               "Generate instruction descriptions"),
-                    clEnumValN(GenCallingConv, "gen-callingconv",
-                               "Generate calling convention descriptions"),
-                    clEnumValN(GenAsmWriter, "gen-asm-writer",
-                               "Generate assembly writer"),
-                    clEnumValN(GenDisassembler, "gen-disassembler",
-                               "Generate disassembler"),
-                    clEnumValN(GenPseudoLowering, "gen-pseudo-lowering",
-                               "Generate pseudo instruction lowering"),
-                    clEnumValN(GenAsmMatcher, "gen-asm-matcher",
-                               "Generate assembly instruction matcher"),
-                    clEnumValN(GenDAGISel, "gen-dag-isel",
-                               "Generate a DAG instruction selector"),
-                    clEnumValN(GenDFAPacketizer, "gen-dfa-packetizer",
-                               "Generate DFA Packetizer for VLIW targets"),
-                    clEnumValN(GenFastISel, "gen-fast-isel",
-                               "Generate a \"fast\" instruction selector"),
-                    clEnumValN(GenSubtarget, "gen-subtarget",
-                               "Generate subtarget enumerations"),
-                    clEnumValN(GenIntrinsic, "gen-intrinsic",
-                               "Generate intrinsic information"),
-                    clEnumValN(GenTgtIntrinsic, "gen-tgt-intrinsic",
-                               "Generate target intrinsic information"),
-                    clEnumValN(PrintEnums, "print-enums",
-                               "Print enum values for a class"),
-                    clEnumValN(PrintSets, "print-sets",
-                               "Print expanded sets for testing DAG exprs"),
-                    clEnumValN(GenOptParserDefs, "gen-opt-parser-defs",
-                               "Generate option definitions"),
-                    clEnumValN(GenCTags, "gen-ctags",
-                               "Generate ctags-compatible index"),
-                    clEnumValN(GenAttributes, "gen-attrs",
-                               "Generate attributes"),
-                    clEnumValEnd));
+static cl::opt<ActionType>
+Action(cl::desc("Action to perform:"),
+       cl::values(clEnumValN(PrintRecords, "print-records",
+                             "Print all records to stdout (default)"),
+                  clEnumValN(GenEmitter, "gen-emitter",
+                             "Generate machine code emitter"),
+                  clEnumValN(GenRegisterInfo, "gen-register-info",
+                             "Generate registers and register classes info"),
+                  clEnumValN(GenInstrInfo, "gen-instr-info",
+                             "Generate instruction descriptions"),
+                  clEnumValN(GenCallingConv, "gen-callingconv",
+                             "Generate calling convention descriptions"),
+                  clEnumValN(GenAsmWriter, "gen-asm-writer",
+                             "Generate assembly writer"),
+                  clEnumValN(GenDisassembler, "gen-disassembler",
+                             "Generate disassembler"),
+                  clEnumValN(GenPseudoLowering, "gen-pseudo-lowering",
+                             "Generate pseudo instruction lowering"),
+                  clEnumValN(GenAsmMatcher, "gen-asm-matcher",
+                             "Generate assembly instruction matcher"),
+                  clEnumValN(GenDAGISel, "gen-dag-isel",
+                             "Generate a DAG instruction selector"),
+                  clEnumValN(GenDFAPacketizer, "gen-dfa-packetizer",
+                             "Generate DFA Packetizer for VLIW targets"),
+                  clEnumValN(GenFastISel, "gen-fast-isel",
+                             "Generate a \"fast\" instruction selector"),
+                  clEnumValN(GenSubtarget, "gen-subtarget",
+                             "Generate subtarget enumerations"),
+                  clEnumValN(GenIntrinsic, "gen-intrinsic",
+                             "Generate intrinsic information"),
+                  clEnumValN(GenTgtIntrinsic, "gen-tgt-intrinsic",
+                             "Generate target intrinsic information"),
+                  clEnumValN(PrintEnums, "print-enums",
+                             "Print enum values for a class"),
+                  clEnumValN(PrintSets, "print-sets",
+                             "Print expanded sets for testing DAG exprs"),
+                  clEnumValN(GenOptParserDefs, "gen-opt-parser-defs",
+                             "Generate option definitions"),
+                  clEnumValN(GenCTags, "gen-ctags",
+                             "Generate ctags-compatible index"),
+                  clEnumValN(GenAttributes, "gen-attrs",
+                             "Generate attributes"),
+                  clEnumValEnd));
 
-  cl::opt<std::string>
-  Class("class", cl::desc("Print Enum list for this class"),
-          cl::value_desc("class name"));
+static cl::opt<std::string>
+Class("class", cl::desc("Print Enum list for this class"),
+      cl::value_desc("class name"));
 
+namespace {
 bool LLVMTableGenMain(raw_ostream &OS, RecordKeeper &Records) {
   switch (Action) {
   case PrintRecords:
