# 24314638 LLVM CommandLine subsystem is busted
# 3.9.X for upstream.
--- tools/bugpoint/OptimizerDriver.cpp	2015-04-14 23:14:06.000000000 -0400
+++ tools/bugpoint/OptimizerDriver.cpp	2016-05-08 23:19:20.539430923 -0400
@@ -47,14 +47,15 @@
     cl::desc("Preserve use-list order when writing LLVM bitcode."),
     cl::init(true), cl::Hidden);
 
-namespace {
-  // ChildOutput - This option captures the name of the child output file that
-  // is set up by the parent bugpoint process
-  cl::opt<std::string> ChildOutput("child-output", cl::ReallyHidden);
-  cl::opt<std::string> OptCmd("opt-command", cl::init(""),
-                              cl::desc("Path to opt. (default: search path "
-                                       "for 'opt'.)"));
-}
+// ChildOutput - This option captures the name of the child output file that
+// is set up by the parent bugpoint process
+static cl::opt<std::string>
+ChildOutput("child-output", cl::ReallyHidden);
+
+static cl::opt<std::string>
+OptCmd("opt-command", cl::init(std::string("")),
+       cl::desc("Path to opt. (default: search path "
+                "for 'opt'.)"));
 
 /// writeProgramToFile - This writes the current "Program" to the named bitcode
 /// file.  If an error occurs, true is returned.
