We have no pre-commit packaged.

--- ephemeral-port-reserve-1.1.4/requirements-dev.txt.orig
+++ ephemeral-port-reserve-1.1.4/requirements-dev.txt
@@ -1,5 +1,4 @@
 -e .
 coverage
 mock
-pre-commit>=0.7.0
 pytest
--- ephemeral-port-reserve-1.1.4/tox.ini.orig
+++ ephemeral-port-reserve-1.1.4/tox.ini
@@ -8,8 +8,6 @@
     coverage erase
     coverage run -m pytest {posargs:tests}
     coverage report --show-missing --fail-under 100
-    pre-commit install -f --install-hooks
-    pre-commit run --all-files
 
 [testenv:venv]
 envdir = venv-{[tox]project}
