Add wanboot to list of potentially supported boot devices.

--- ORIGINAL/ironicclient/v1/node_shell.py	2015-06-03 14:28:54.964047000 +0100
+++ python-ironicclient-0.5.1/ironicclient/v1/node_shell.py	2015-06-03 14:32:08.040441131 +0100
@@ -383,8 +383,8 @@
 @cliutils.arg(
     'device',
     metavar='<boot-device>',
-    choices=['pxe', 'disk', 'cdrom', 'bios', 'safe'],
-    help="'pxe', 'disk', 'cdrom', 'bios', or 'safe'.")
+    choices=['pxe', 'disk', 'cdrom', 'bios', 'safe', 'wanboot'],
+    help="'pxe', 'disk', 'cdrom', 'bios', 'safe' or 'wanboot'.")
 @cliutils.arg(
     '--persistent',
     dest='persistent',
