--- gzip-1.5/zgrep.in.orig	2012-09-05 15:31:20.730722910 -0700
+++ gzip-1.5/zgrep.in	2012-09-05 15:33:51.300884579 -0700
@@ -26,9 +26,13 @@
 case $1 in
 --__bindir) bindir=${2?}; shift; shift;;
 esac
-PATH=$bindir:$PATH
+PATH=/usr/xpg4/bin:$bindir:$PATH
 
-grep='${GREP-grep}'
+case "$1" in
+	-F) grep='${FGREP-fgrep}'; shift;;
+	-E) grep='${EGREP-egrep}'; shift;;
+	*)  grep='${GREP-grep}';;
+esac
 
 version='zgrep (gzip) @VERSION@
 Copyright (C) 2010-2012 Free Software Foundation, Inc.
