diff -ur soxr-0.1.2-Source-orig/src/soxr-lsr.pc.in soxr-0.1.2-Source/src/soxr-lsr.pc.in
--- soxr-0.1.2-Source-orig/src/soxr-lsr.pc.in	2017-12-12 07:17:40.863672849 +0000
+++ soxr-0.1.2-Source/src/soxr-lsr.pc.in	2017-12-12 07:18:28.813986981 +0000
@@ -1,5 +1,5 @@
 Name: ${LSR}
 Description: ${DESCRIPTION_SUMMARY} (with libsamplerate-like bindings)
 Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -l${LSR}
+Libs: -l${LSR}
 Cflags: -I${INCLUDE_INSTALL_DIR}
diff -ur soxr-0.1.2-Source-orig/src/soxr.pc.in soxr-0.1.2-Source/src/soxr.pc.in
--- soxr-0.1.2-Source-orig/src/soxr.pc.in	2017-12-12 07:17:40.865299127 +0000
+++ soxr-0.1.2-Source/src/soxr.pc.in	2017-12-12 07:18:14.994642464 +0000
@@ -1,5 +1,5 @@
 Name: ${PROJECT_NAME}
 Description: ${DESCRIPTION_SUMMARY}
 Version: ${PROJECT_VERSION}
-Libs: -L${LIB_INSTALL_DIR} -l${PROJECT_NAME}
+Libs: -l${PROJECT_NAME}
 Cflags: -I${INCLUDE_INSTALL_DIR}
