--- aalib-1.4.0/aalib-config.in.orig	Wed Jun  8 13:30:30 2011
+++ aalib-1.4.0/aalib-config.in	Wed Jun  8 13:33:14 2011
@@ -48,7 +48,7 @@
       echo -I@includedir@ @AALIB_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      libdirs="-L@libdir@ @AALIB_RLD_FLAGS@"
+@ENABLE_SHARED_TRUE@      libdirs="-L@libdir@"
 @ENABLE_SHARED_TRUE@      echo $libdirs @AALIB_LIBS@
 @ENABLE_SHARED_TRUE@      ;;
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
