--- aalib-1.4.0/configure.in.1	2018-01-27 21:29:12.878039200 +0000
+++ aalib-1.4.0/configure.in	2018-01-27 21:29:23.715982603 +0000
@@ -377,9 +377,9 @@
   AALIB_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
 fi
 
-if test $ARCH = solaris; then
-  AALIB_RLD_FLAGS="-R\${exec_prefix}/lib"
-fi
+#if test $ARCH = solaris; then
+#  AALIB_RLD_FLAGS="-R\${exec_prefix}/lib"
+#fi
 
 
 dnl Expand the cflags and libraries needed by apps using AALIB
