diff -urN aalib-1.4.0.orig/aalib.m4 aalib-1.4.0/aalib.m4
--- aalib-1.4.0/aalib.m4.orig	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0/aalib.m4	2017-02-22 21:56:53.505179073 +0100
@@ -9,7 +9,7 @@
 dnl AM_PATH_AALIB([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for AALIB, and define AALIB_CFLAGS and AALIB_LIBS
 dnl
-AC_DEFUN(AM_PATH_AALIB,
+AC_DEFUN([AM_PATH_AALIB],
 [dnl 
 dnl Get the cflags and libraries from the aalib-config script
 dnl
