--- hexedit/Makefile.in.orig	Mon Apr 25 12:20:14 2011
+++ hexedit/Makefile.in	Mon Apr 25 12:20:37 2011
@@ -40,7 +40,7 @@
 	rm -f Makefile config.h
 
 install: $(PRODUCT)
-	$(INSTALL) -d -m 755 $(bindir)
-	$(INSTALL) -s -m 755 $(PRODUCT) $(bindir)
-	$(INSTALL) -d -m 755 $(mandir)/man1
-	$(INSTALL) -m 644 $(PRODUCT).1 $(mandir)/man1
+	$(INSTALL) -d -m 755 $(DESTDIR)/$(bindir)
+	$(INSTALL) -s -m 755 $(PRODUCT) $(DESTDIR)/$(bindir)
+	$(INSTALL) -d -m 755 $(DESTDIR)/$(mandir)/man1
+	$(INSTALL) -m 644 $(PRODUCT).1 $(DESTDIR)/$(mandir)/man1
