### This patch will not be needed after libnet 1.1.6

--- libnet-1.1.5.orig/src/libnet_checksum.c	2010-11-03 11:35:12.000000000 -0700
+++ libnet-1.1.5/src/libnet_checksum.c	2012-05-27 15:48:47.202890574 -0700
@@ -207,6 +207,7 @@
     if (iph_p->ip_v == 6)
     {
         ip6h_p = (struct libnet_ipv6_hdr *)iph_p;
+        is_ipv6 = 1;
         iph_p = NULL;
         ip_hl   = 40;
         if((uint8_t*)(ip6h_p+1) > end)
