--- Makedefs.in.orig	Fri Apr 23 11:56:34 2010
+++ Makedefs.in	Tue Dec  7 00:56:35 2010
@@ -259,7 +259,7 @@
 #
 
 .SILENT:
-.SUFFIXES:	.1 .1.gz .1m .1m.gz .3 .3.gz .5 .5.gz .7 .7.gz .8 .8.gz .a .c .cxx .h .man .o .32.o .64.o .gz
+.SUFFIXES:	.1 .1.gz .1cups .1m .1m.gz .3 .3.gz .5 .5.gz .7 .7.gz .8 .8.gz .a .c .cxx .h .man .o .32.o .64.o .gz
 
 .c.o:
 	echo Compiling $<...
@@ -277,7 +277,7 @@
 	echo Compiling $<...
 	$(CXX) $(ARCHFLAGS) $(OPTIM) $(ALL_CXXFLAGS) -c -o $@ $<
 
-.man.1 .man.1m .man.3 .man.5 .man.7 .man.8:
+.man.1 .man.1cups .man.1m .man.3 .man.5 .man.7 .man.8:
 	echo Linking $<...
 	$(RM) $@
 	$(LN) $< $@
--- man/Makefile.orig	Tue Jun 22 14:42:44 2010
+++ man/Makefile	Tue Dec  7 00:56:35 2010
@@ -126,10 +126,10 @@
 #
 
 install-data: all
-	$(INSTALL_DIR) -m 755 $(MANDIR)/man1
+	$(INSTALL_DIR) -m 755 $(MANDIR)/man$(MAN1EXT)
 	for file in $(MAN1); do \
-		echo Installing $$file in $(MANDIR)/man1...; \
-		$(INSTALL_MAN) $$file $(MANDIR)/man1; \
+		echo Installing $$file in $(MANDIR)/man$(MAN1EXT)...; \
+		$(INSTALL_MAN) $$file $(MANDIR)/man$(MAN1EXT); \
 	done
 	$(INSTALL_DIR) -m 755 $(MANDIR)/man5
 	for file in $(MAN5); do \
