--- scheduler/dirsvc.c.orig	Fri Apr 23 11:56:34 2010
+++ scheduler/dirsvc.c	Mon Dec  6 20:58:17 2010
@@ -1335,7 +1335,7 @@
 #  if defined(HAVE_LDAP_SSL) && defined (HAVE_MOZILLA_LDAP)
     if (ldap_ssl && (rc == LDAP_SERVER_DOWN || rc == LDAP_CONNECT_ERROR))
     {
-      ssl_err = PORT_GetError();
+      ssl_err = rc;
       if (ssl_err != 0)
         cupsdLogMessage(CUPSD_LOG_ERROR, "LDAP SSL error %d: %s", ssl_err,
                         ldapssl_err2string(ssl_err));
