--- scheduler/Makefile.~1~	2014-04-22 13:43:25.093821305 +0400
+++ scheduler/Makefile	2014-04-22 13:46:05.791930438 +0400
@@ -295,7 +295,7 @@
 	$(CC) $(LDFLAGS) -o cupsd $(CUPSDOBJS) -L. -lcupsmime \
 		$(LIBZ) $(SSLLIBS) $(LIBSLP) $(LIBLDAP) $(PAMLIBS) \
 		$(LIBPAPER) $(LIBMALLOC) $(CUPSDLIBS) $(DNSSDLIBS) $(LIBS) \
-		$(LIBGSSAPI) $(LIBWRAP)
+		$(LIBGSSAPI) $(LIBWRAP) -lsecdb
 
 cupsd-static:	$(CUPSDOBJS) libcupsmime.a ../cups/libcups.a
 	echo Linking $@...
