diff --git a/man/xhost.man b/man/xhost.man
index aac0a0a..cb2f22d 100644
--- a/man/xhost.man
+++ b/man/xhost.man
@@ -162,7 +162,9 @@ For each name removed from the access control list,
 a line of the form "\fIname\fP being removed from access control list"
 is printed.
 .SH "SEE ALSO"
-X(__miscmansuffix__), Xsecurity(__miscmansuffix__), Xserver(__appmansuffix__), xdm(__appmansuffix__), xauth(__appmansuffix__), getaddrinfo(__libmansuffix__)
+X(__miscmansuffix__), Xsecurity(__miscmansuffix__), Xserver(__appmansuffix__),
+xdm(__appmansuffix__), xauth(__appmansuffix__), netservices(__adminmansuffix__),
+getaddrinfo(__libmansuffix__)
 .SH ENVIRONMENT
 .TP 8
 .B DISPLAY
@@ -181,6 +183,9 @@ the server-interpreted hostname type address.  If somehow you change a
 host's network address while the server is still running, and you are
 using a network-address based form of authentication, \fIxhost\fP must
 be used to add the new address and/or remove the old address.
+.SH WARNINGS
+__distroname__ is secure by default.  See Xserver(1) which details how to
+re-enable remote connections to the X server as in previous releases.
 .SH AUTHORS
 Bob Scheifler, MIT Laboratory for Computer Science,
 .br
