--- pgadmin3-1.22.2/pgadmin/frm/frmEditGrid.cpp.orig	2020-05-23 14:52:17.378935763 +0000
+++ pgadmin3-1.22.2/pgadmin/frm/frmEditGrid.cpp	2020-05-23 15:04:08.982095439 +0000
@@ -2386,7 +2386,8 @@
 			columns[i].typlen = colSet->GetLong(wxT("typlen"));
 			columns[i].typmod = colSet->GetLong(wxT("typmod"));
 
-			switch (columns[i].type)
+			// Who knows what happened here earlier... anyways OID is unsigned :S
+			switch (static_cast<long>(columns[i].type))
 			{
 				case PGOID_TYPE_BOOL:
 					columns[i].numeric = false;
