diff --git a/configure.ac b/configure.ac
index 6fa6d40..5d4000f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,5 @@
 # Vinagre autoconf
 
-AC_PREREQ([2.64])
 AC_INIT([vinagre],[2.30.3],[https://bugzilla.gnome.org/enter_bug.cgi?product=vinagre],[vinagre],[http://projects.gnome.org/vinagre/])
 AC_CONFIG_SRCDIR([vinagre/vinagre-main.c])
 AC_CONFIG_HEADERS([config.h])
@@ -32,7 +31,6 @@ AC_FUNC_MALLOC
 AC_CHECK_FUNCS([dup2 memset putenv select setlocale socket strchr strerror strstr getpt posix_openpt grantpt unlockpt ptsname ptsname_r])
 
 AM_INIT_AUTOMAKE([1.10 -Wno-portability])
-LT_PREREQ([2.2.6])
 AC_CONFIG_MACRO_DIR([m4])
 
 GNOME_COMPILE_WARNINGS([maximum])
@@ -54,7 +52,7 @@ AC_SEARCH_LIBS([strerror],[cposix])
 AC_PROG_CC
 AC_HEADER_STDC
 
-LT_INIT([disable-static])
+AM_PROG_LIBTOOL
 AM_PROG_CC_C_O
 
 #*******************************************************************************
