diff --git a/plugins/vnc/vinagre-vnc-tab.c b/plugins/vnc/vinagre-vnc-tab.c
index 2218c53..13925f0 100644
--- a/plugins/vnc/vinagre-vnc-tab.c
+++ b/plugins/vnc/vinagre-vnc-tab.c
@@ -324,7 +324,6 @@ static void
 vnc_server_cut_text_cb (VncDisplay *vnc, const gchar *text, VinagreVncTab *vnc_tab)
 {
   GtkClipboard *cb;
-  gsize a, b;
   GtkTargetEntry targets[] = {
 				{"UTF8_STRING", 0, 0},
 				{"COMPOUND_TEXT", 0, 0},
@@ -336,7 +335,7 @@ vnc_server_cut_text_cb (VncDisplay *vnc, const gchar *text, VinagreVncTab *vnc_t
     return;
 
   g_free (vnc_tab->priv->clipboard_str);
-  vnc_tab->priv->clipboard_str = g_convert (text, -1, "utf-8", "iso8859-1", &a, &b, NULL);
+  vnc_tab->priv->clipboard_str = g_strdup (text);
 
   if (vnc_tab->priv->clipboard_str)
     {
@@ -882,23 +881,9 @@ vinagre_vnc_tab_send_ctrlaltdel (VinagreVncTab *tab)
 void
 vinagre_vnc_tab_paste_text (VinagreVncTab *tab, const gchar *text)
 {
-  gchar *out;
-  gsize a, b;
-  GError *error = NULL;
-
   g_return_if_fail (VINAGRE_IS_VNC_TAB (tab));
 
-  out = g_convert_with_fallback (text, -1, "iso8859-1//TRANSLIT", "utf-8", NULL, &a, &b, &error);
-  if (out)
-    {
-      vnc_display_client_cut_text (VNC_DISPLAY (tab->priv->vnc), out);
-      g_free (out);
-    }
-  else
-    {
-      g_warning ("%s", error->message);
-      g_error_free (error);
-    }
+  vnc_display_client_cut_text (VNC_DISPLAY (tab->priv->vnc), text);
 }
 
 gboolean
