diff -Nura json-c-0.10.orig/tests/Makefile.in json-c-0.10/tests/Makefile.in
--- json-c-0.10.orig/tests/Makefile.in  2012-05-31 04:31:36.000000000 +0200
+++ json-c-0.10/tests/Makefile.in       2013-08-22 14:17:19.827822601 +0200
@@ -215,7 +215,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -Wall -Wwrite-strings -Werror -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
 LIBJSON_LA = $(top_builddir)/libjson.la
 test1_LDADD = $(LIBJSON_LA)
 test1Formatted_LDADD = $(LIBJSON_LA)
