--- nano-2.2.6/configure.orig	2014-02-09 16:50:34.868716078 +0100
+++ nano-2.2.6/configure	2014-02-09 16:46:24.262821000 +0100
@@ -7878,8 +7878,8 @@
 
 
     if test x$enable_utf8 != xno; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for get_wch in -lncursesw" >&5
-$as_echo_n "checking for get_wch in -lncursesw... " >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for getwch in -lncursesw" >&5
+$as_echo_n "checking for getwch in -lncursesw... " >&6; }
 if test "${ac_cv_lib_ncursesw_get_wch+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
@@ -7894,11 +7894,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char get_wch ();
+char getwch ();
 int
 main ()
 {
-return get_wch ();
+return getwch ();
   ;
   return 0;
 }
@@ -7979,8 +7979,8 @@
 
 
     if test x$enable_utf8 != xno; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for get_wch in -lcurses" >&5
-$as_echo_n "checking for get_wch in -lcurses... " >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for getwch in -lcurses" >&5
+$as_echo_n "checking for getwch in -lcurses... " >&6; }
 if test "${ac_cv_lib_curses_get_wch+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
@@ -7995,11 +7995,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char get_wch ();
+char getwch ();
 int
 main ()
 {
-return get_wch ();
+return getwch ();
   ;
   return 0;
 }
