diff -urN gok-1.0.5/launcher.kbd.in gok-1.0.5-hacked/launcher.kbd.in
--- gok-1.0.5/launcher.kbd.in	2003-11-11 10:17:54.000000000 +1300
+++ gok-1.0.5-hacked/launcher.kbd.in	2005-09-18 19:39:06.835481816 +1200
@@ -11,13 +11,25 @@
 	</GOK:key> 
 	<GOK:key left="3" right="6" top="0" bottom="1" type="normal"> 
 		<_GOK:label>Web Browser</_GOK:label> 
-		<GOK:output type="exec">mozilla</GOK:output>
+		<GOK:output type="exec">firefox</GOK:output>
 	</GOK:key> 
-	<GOK:key left="0" right="3" top="1" bottom="2" type="normal"> 
+ 	<GOK:key left="6" right="9" top="0" bottom="1" type="normal"> 
+ 		<_GOK:label>StarOffice</_GOK:label> 
+ 		<GOK:output type="exec">soffice</GOK:output>
+ 	</GOK:key> 
+  	<GOK:key left="0" right="3" top="1" bottom="2" type="normal"> 
+ 		<_GOK:label>Email and Calendar</_GOK:label> 
+ 		<GOK:output type="exec">evolution</GOK:output>
+ 	</GOK:key> 
+ 	<GOK:key left="3" right="5" top="1" bottom="2" type="normal"> 
 		<_GOK:label>Text Editor</_GOK:label> 
 		<GOK:output type="exec">gedit</GOK:output>
 	</GOK:key> 
-	<GOK:key left="3" right="6" top="1" bottom="2" type="normal"> 
+ 	<GOK:key left="5" right="7" top="1" bottom="2" type="normal"> 
+ 		<_GOK:label>Instant Messaging</_GOK:label> 
+ 		<GOK:output type="exec">pidgin</GOK:output>
+ 	</GOK:key> 
+ 	<GOK:key left="7" right="9" top="1" bottom="2" type="normal"> 
 		<_GOK:label>Help Browser</_GOK:label> 
 		<GOK:output type="exec">gnome-help</GOK:output>
 	</GOK:key> 
