--- atk-1.32.0/atk/Makefile.am.~1~	2015-03-24 14:11:47.352178199 +0300
+++ atk-1.32.0/atk/Makefile.am	2015-03-24 14:12:29.432085548 +0300
@@ -161,6 +161,7 @@
 Atk_1_0_gir_INCLUDES = GObject-2.0
 Atk_1_0_gir_CFLAGS = $(INCLUDES)
 Atk_1_0_gir_LIBS = libatk-1.0.la
+Atk_1_0_gir_LDFLAGS = -lgmodule-2.0
 Atk_1_0_gir_SCANNERFLAGS = --pkg-export atk
 Atk_1_0_gir_FILES = \
     $(addprefix $(srcdir)/, $(introspection_sources))	\
