None of the provided library names names load libgpgme.

--- thunderbird-144.0/comm/mail/extensions/openpgp/content/modules/GPGMELib.sys.mjs.old	2025-10-19 20:08:43.098166370 -0400
+++ thunderbird-144.0/comm/mail/extensions/openpgp/content/modules/GPGMELib.sys.mjs	2025-10-19 20:12:52.802836703 -0400
@@ -84,6 +84,10 @@
     if (!libgpgme) {
       tryLoadGPGME("gpgme.11", "");
     }
+
+    if (!libgpgme) {
+      tryLoadGPGME("gpgme.so.11", "");
+    }
   }
 
   return !!libgpgme;
