--- firefox-43.0b3/media/libpng/pngpriv.h.~1~	2015-11-18 23:47:44.987893422 +0300
+++ firefox-43.0b3/media/libpng/pngpriv.h	2015-11-18 23:48:07.360418176 +0300
@@ -36,6 +36,7 @@
  * still required (as of 2011-05-02.)
  */
 #define _POSIX_SOURCE 1 /* Just the POSIX 1003.1 and C89 APIs */
+#define _XOPEN_SOURCE 600
 
 #ifndef PNG_VERSION_INFO_ONLY
 /* Standard library headers not required by png.h: */
