--- firefox-52.1.1esr/toolkit/components/terminator/nsTerminator.cpp.1	2017-05-11 23:45:33.893236340 +0300
+++ firefox-52.1.1esr/toolkit/components/terminator/nsTerminator.cpp	2017-05-11 23:51:53.577453894 +0300
@@ -388,7 +388,7 @@
   }
 
   UniquePtr<Options> options(new Options());
-  const PRIntervalTime ticksDuration = PR_MillisecondsToInterval(1000);
+  const PRIntervalTime ticksDuration = 1000;
   options->crashAfterTicks = crashAfterMS / ticksDuration;
 
   DebugOnly<PRThread*> watchdogThread = CreateSystemThread(RunWatchdog,
