--- firefox-43.0b3/ipc/chromium/src/third_party/libevent/arc4random.c.1	2015-11-18 14:31:05.402888998 +0300
+++ firefox-43.0b3/ipc/chromium/src/third_party/libevent/arc4random.c	2015-11-18 14:33:10.482583357 +0300
@@ -487,6 +487,7 @@
 }
 #endif
 
+#ifdef _we_have_arc4random_buf
 ARC4RANDOM_EXPORT void
 arc4random_buf(void *_buf, size_t n)
 {
@@ -500,6 +501,7 @@
 	}
 	_ARC4_UNLOCK();
 }
+#endif
 
 #ifndef ARC4RANDOM_NOUNIFORM
 /*
