Shell translations. 

diff --git a/js/src/configure b/js/src/configure
--- a/js/src/configure
+++ b/js/src/configure
@@ -1230,7 +1230,7 @@
   fi
 
     fi
-    IS_GCC=$($CC -v 2>&1 | grep gcc)
+    IS_GCC=`$CC -v 2>&1 | grep gcc`
     if test -n "$IS_GCC"
     then
       echo gcc is known to be broken on OS X, please use clang.
@@ -4426,7 +4426,7 @@
 for base in $MOZILLA_CENTRAL_PATH $_topsrcdir $_topsrcdir/../..; do
   possible=$base/python/mozbuild/mozbuild/virtualenv.py
 
-  if test -e $possible; then
+  if test -f $possible; then
     _virtualenv_topsrcdir=$base
     _virtualenv_populate_path=$possible
     break
@@ -8743,12 +8743,12 @@
 
 if test "$GNU_CC" -a -n "$MOZ_FORCE_GOLD"; then
             if $CC -Wl,--version 2>&1 | grep -q "GNU ld"; then
-        GOLD=$($CC -print-prog-name=ld.gold)
+        GOLD=`$CC -print-prog-name=ld.gold`
         case "$GOLD" in
         /*)
             ;;
         *)
-            GOLD=$(which $GOLD)
+            GOLD=`which $GOLD`
             ;;
         esac
         if test -n "$GOLD"; then
@@ -16322,9 +16322,12 @@
 
 
 rm -fr confdefs* $ac_clean_files
-if test "$no_create" != yes && ! ${PYTHON} $CONFIG_STATUS; then
+if test "$no_create" != yes; then
+  ${PYTHON} $CONFIG_STATUS
+  if test "$?" != 0; then
     trap '' EXIT
     exit 1
-fi
-
-fi
+  fi
+fi
+
+fi
