diff --git a/js/src/wasm/WasmSignalHandlers.cpp b/js/src/wasm/WasmSignalHandlers.cpp
index 86ab0f4..5f46b87 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -109,6 +109,7 @@ using mozilla::DebugOnly;
 #    define R32_sig(p) ((p)->sc_frame.srr0)
 #  endif
 #elif defined(__linux__) || defined(__sun)
+#  include <sys/regset.h>
 #  if defined(__linux__)
 #    define EIP_sig(p) ((p)->uc_mcontext.gregs[REG_EIP])
 #    define EBP_sig(p) ((p)->uc_mcontext.gregs[REG_EBP])
