--- a/ldap/c-sdk/libraries/libldap/sort.c	2016-05-16 10:36:26.007446145 -0700
+++ b/ldap/c-sdk/libraries/libldap/sort.c.new	2016-05-16 12:56:49.799496038 -0700
@@ -58,11 +58,7 @@
  * We don't want to require use of libxp when linking with libldap, so
  * I'll leave use of xp_qsort as a MOZILLA_CLIENT-only thing for now. --mcs
  */
-#if defined(MOZILLA_CLIENT) && defined(SOLARIS)
-#include "xp_qsort.h"
-#else
 #define XP_QSORT qsort
-#endif
 
 typedef struct keycmp {
     void                 *kc_arg;
