diff -up enscript-1.6.4/src/main.c.doublefree enscript-1.6.4/src/main.c
--- enscript-1.6.4/src/main.c.doublefree	2008-08-08 11:10:07.000000000 +0200
+++ enscript-1.6.4/src/main.c	2008-08-08 11:11:51.000000000 +0200
@@ -1714,8 +1714,9 @@ name             width\theight\tllx\tlly
 	      is_close (&is);
 	    }
 
-	  /* Clean up toc file. */
-	  fclose (toc_fp);
+	  else
+	    /* Clean up toc file. */
+	    fclose (toc_fp);
 	}
 
       /* Give trailer a chance to dump itself. */
