--- enscript-1.6.1/src/psgen.c.CAN-2004-1186	2005-01-10 15:57:07.432740602 +0000
+++ enscript-1.6.1/src/psgen.c	2005-01-10 15:58:24.606016206 +0000
@@ -1914,8 +1914,9 @@
   else
     {
       ftail++;
-      strncpy (buf, fname, ftail - fname);
-      buf[ftail - fname] = '\0';
+      i = ftail - fname >= sizeof (buf)-1 ? sizeof (buf)-1 : ftail - fname;
+      strncpy (buf, fname, i);
+      buf[i] = '\0';
     }
 
   if (nup > 1)
