diff -up enscript-1.6.4/src/psgen.c.CAN-2004-1185 enscript-1.6.4/src/psgen.c
--- enscript-1.6.4/src/psgen.c.CAN-2004-1185	2008-08-08 11:18:06.000000000 +0200
+++ enscript-1.6.4/src/psgen.c	2008-08-08 11:17:57.000000000 +0200
@@ -2386,9 +2386,11 @@ recognize_eps_file (Token *token)
   MESSAGE (2, (stderr, "^@epsf=\"%s\"\n", token->u.epsf.filename));
 
   i = strlen (token->u.epsf.filename);
+
+  /*
   if (i > 0 && token->u.epsf.filename[i - 1] == '|')
     {
-      /* Read EPS data from pipe. */
+      / * Read EPS data from pipe. * /
       token->u.epsf.pipe = 1;
       token->u.epsf.filename[i - 1] = '\0';
       token->u.epsf.fp = popen (token->u.epsf.filename, "r");
@@ -2401,6 +2403,7 @@ recognize_eps_file (Token *token)
 	}
     }
   else
+  */
     {
       char *filename;
 
