diff -up enscript-1.6.4/afm/Makefile.am.config enscript-1.6.4/afm/Makefile.am
--- enscript-1.6.4/afm/Makefile.am.config	2003-03-05 08:26:16.000000000 +0100
+++ enscript-1.6.4/afm/Makefile.am	2009-11-30 14:38:34.410478926 +0100
@@ -36,11 +36,11 @@ public_fonts = matrix.afm matrix.pfa
 EXTRA_DIST = font.map $(default_afm) $(public_fonts)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/afm
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/afm
 	for f in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/afm/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/afm/$$f; \
 	done
 
 uninstall-local:
-	rm -rf $(datadir)/enscript/afm
+	rm -rf $(DESTDIR)/$(datadir)/enscript/afm
diff -up enscript-1.6.4/lib/Makefile.am.config enscript-1.6.4/lib/Makefile.am
--- enscript-1.6.4/lib/Makefile.am.config	2009-11-30 14:39:38.377977061 +0100
+++ enscript-1.6.4/lib/Makefile.am	2009-11-30 14:40:13.887977573 +0100
@@ -38,12 +38,12 @@ CLEANFILES = enscript.cfg
 all-local: enscript.cfg
 
 install-data-local: enscript.cfg
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
-	if test -r $(sysconfdir)/enscript.cfg; then \
-	  cp $(sysconfdir)/enscript.cfg $(sysconfdir)/enscript.cfg.old; \
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sysconfdir)
+	if test -r $(DESTDIR)$(sysconfdir)/enscript.cfg; then \
+	  cp $(DESTDIR)$(sysconfdir)/enscript.cfg $(DESTDIR)$(sysconfdir)/enscript.cfg.old; \
 	else :; \
 	fi
-	$(INSTALL_DATA) enscript.cfg $(sysconfdir)/enscript.cfg
+	$(INSTALL_DATA) enscript.cfg $(DESTDIR)$(sysconfdir)/enscript.cfg
 
 uninstall-local:
 	rm -f $(sysconfdir)/enscript.cfg
diff -up enscript-1.6.4/states/hl/Makefile.am.config enscript-1.6.4/states/hl/Makefile.am
--- enscript-1.6.4/states/hl/Makefile.am.config	2003-03-05 08:26:33.000000000 +0100
+++ enscript-1.6.4/states/hl/Makefile.am	2009-11-30 14:38:34.410478926 +0100
@@ -45,10 +45,10 @@ states = $(misc) $(styles) $(languages) 
 EXTRA_DIST = $(states)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/hl
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/hl
 	for f in $(states); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/hl/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/hl/$$f; \
 	done
 
 uninstall-local:
