diff -up enscript-1.6.4/src/main.c.rh457719 enscript-1.6.4/src/main.c
--- enscript-1.6.4/src/main.c.rh457719	2008-08-08 11:23:27.000000000 +0200
+++ enscript-1.6.4/src/main.c	2008-08-08 11:23:36.000000000 +0200
@@ -2001,7 +2001,7 @@ handle_options (int argc, char *argv[])
       int i;
 
       c = getopt_long (argc, argv,
-		       "#:123456789a:A:b:BcC::d:D:e::E::f:F:gGhH::i:I:jJ:kKlL:mM:n:N:o:Op:P:qrRs:S:t:T:u::U:vVW:X:zZ",
+		       "#:123456789a:A:b:BcC::d:D:e::E::f:F:gGhH::i:I:jJ:kKlL:mM:n:N:o:Op:P:qrRs:S:t:T:u::U:vVw:W:X:zZ",
 		       long_options, &option_index);
 
       if (c == -1)
