use 64-bit library path
--- c-icap-server-C_ICAP_0.6.3/configure.ac.old	2024-07-09 23:07:43.640827757 -0400
+++ c-icap-server-C_ICAP_0.6.3/configure.ac	2024-07-09 23:08:03.685797448 -0400
@@ -369,7 +369,7 @@
    ICFG_STATE_SAVE(OPENSSL)
    if test "a$opensslpath" != "a"; then
         CFLAGS="$CFLAGS -I$opensslpath/include"
-        LDFLAGS="$LDFLAGS -L$opensslpath/lib"
+        LDFLAGS="$LDFLAGS -L$opensslpath/lib/amd64"
    fi
    LIBS="-lssl -lcrypto $LIBS"
    (test -n "$opensslpath" && echo -n "checking for OpenSSL library under $opensslpath... ") || echo -n "checking for OpenSSL library... ";
