diff --git a/man/X.man b/man/X.man
index 16b4c96..4c8e0f6 100644
--- a/man/X.man
+++ b/man/X.man
@@ -110,6 +110,8 @@ web browser plug-ins, \fIlibxrx.so\fP and \fIlibxrxnest.so\fP;
 an RX MIME-type helper program, \fIxrx\fP;
 and a utility to cause part or all of the screen to be redrawn, \fIxrefresh\fP.
 .PP
+Many, but not all, of these programs are included in __distroname__.
+.PP
 Many other utilities, window managers, games, toolkits, etc. are included
 as user-contributed software in the X.Org Foundation distribution, or are
 available on the Internet.
@@ -122,11 +124,10 @@ operating system you are running and whether or not you use other window
 systems in addition to X.
 .TP 8
 .B "Display Manager"
-If you want to always have X running on your display, your site administrator
-can set your machine up to use a Display Manager such as \fIxdm\fP, \fIgdm\fP,
-or \fIkdm\fP.  This program
-is typically started by the system at boot time and takes care of keeping the
-server running and getting users logged in.  If you are running one of these
+The default configuration of __distroname__ starts X at boot time via
+the \fIgdm\fP(__adminmansuffix__) display manager.   The \fIxdm\fP(1) display manager is also
+provided as an option the system administrator can configure.
+If you are running one of these
 display managers, you will normally see a window on the screen welcoming you
 to the system and asking for your login information.  Simply type them in as
 you would at a normal terminal.  If you make a mistake, the display manager
@@ -362,13 +363,12 @@ to explicitly draw the window's region on the screen with the pointer, for
 example).
 .PP
 Since window managers are regular (albeit complex) client programs,
-a variety of different user interfaces can be built.  The X.Org Foundation
-provides a window manager named \fItwm\fP which supports overlapping windows,
-popup menus, point-and-click or click-to-type input models, title bars, nice
-icons (and an icon manager for those who don't like separate icon windows).
-.PP
-See your distro package manager for other
-popular window managers.
+a variety of different user interfaces can be built.
+__distroname__ includes several window managers, including the
+GNOME window manager
+.IR metacity (1),
+and the X.Org classic window manager
+.IR twm (1).
 .SH "FONT NAMES"
 Collections of characters for displaying text and symbols in X are known as
 \fIfonts\fP.  A font typically contains images that share a common appearance
@@ -1265,10 +1268,14 @@ the appropriate instance name can be placed before the asterisk:
 .BR Xdmx (__appmansuffix__),
 .BR Xephyr (__appmansuffix__),
 .BR Xnest (__appmansuffix__),
-.BR Xquartz (__appmansuffix__),
 .BR Xvfb (__appmansuffix__),
 .BR Xvnc (__appmansuffix__),
-.BR XWin (__appmansuffix__).
+.\" related software
+.BR gdm (__adminmansuffix__),
+.BR gnome-session (1),
+.BR metacity (1),
+.BR locale (__miscmansuffix__).
+
 .PP
 .\" specifications
 .I "Xlib \- C Language X Interface\fR,\fP"
