--- cairo-1.12.2/build/configure.ac.pthread.orig	2012-10-22 11:07:51.251654403 +0530
+++ cairo-1.12.2/build/configure.ac.pthread	2012-10-22 13:06:52.192850958 +0530
@@ -190,10 +190,10 @@
 	dnl Try to use the user's PTHREAD_LIBS/CFLAGS
 	dnl if they're available.
 	if test "x$PTHREAD_CFLAGS" = "x"; then
-		PTHREAD_CFLAGS="-D_REENTRANT"
+		PTHREAD_CFLAGS="-D_REENTRANT -mt -lpthread"
 	fi
 	if test "x$PTHREAD_LIBS" = "x"; then
-		PTHREAD_LIBS="-lpthread"
+		PTHREAD_LIBS="-mt -lpthread"
 	fi
 
 	dnl First try to find the real pthreads.
