--- totem-pl-parser-3.10.7-1/plparse/plparser.symbols.1	2017-03-11 14:38:19.321158524 +0300
+++ totem-pl-parser-3.10.7-1/plparse/plparser.symbols	2017-03-11 14:39:09.374126047 +0300
@@ -14,7 +13,6 @@
 totem_pl_parser_error_get_type
 totem_pl_parser_error_quark
 totem_pl_parser_get_type
-totemplparser_marshal_VOID__STRING_STRING_STRING
 totem_pl_parser_new
 totem_pl_parser_parse
 totem_pl_parser_parse_async
