--- totem-pl-parser-3.10.7/plparse/Makefile.am.1	2017-03-22 23:50:02.203889065 +0300
+++ totem-pl-parser-3.10.7/plparse/Makefile.am	2017-03-22 23:52:32.719138006 +0300
@@ -21,6 +21,7 @@
 	totem-pl-parser-features.h		\
 	totem-pl-parser.h			\
 	totem-pl-playlist.h			\
+	totem-disc.h				\
 	totem-pl-parser-mini.h
 
 plparser_sources =				\
@@ -155,7 +156,7 @@
 	$(CODE_COVERAGE_LDFLAGS)			\
 	$(AM_LDFLAGS)
 
-totem-pl-parser-builtins.h: totem-pl-parser.h
+totem-pl-parser-builtins.h: totem-pl-parser.h totem-disc.h
 	$(AM_V_GEN) ($(GLIB_MKENUMS) \
 			--fhead "#ifndef __TOTEM_PL_PARSER_BUILTINS_H__\n#define __TOTEM_PL_PARSER_BUILTINS_H__\n\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
 			--fprod "/* enumerations from \"@filename@\" */\n" \
@@ -163,9 +164,9 @@
 			--ftail "G_END_DECLS\n\n#endif /* __TOTEM_PL_PARSER_BUILTINS_H__ */" $^ > xgen-$(@F) \
 		&& mv -f xgen-$(@F) $@)
 
-totem-pl-parser-builtins.c: totem-pl-parser.h totem-pl-parser-builtins.h
+totem-pl-parser-builtins.c: totem-pl-parser.h totem-disc.h totem-pl-parser-builtins.h
 	$(AM_V_GEN) ($(GLIB_MKENUMS) \
-			--fhead "#include \"totem-pl-parser.h\"\n#include \"totem-pl-parser-builtins.h\"" \
+			--fhead "#include \"totem-pl-parser.h\"\n#include \"totem-disc.h\"\n#include \"totem-pl-parser-builtins.h\"" \
 			--fprod "\n/* enumerations from \"@filename@\" */" \
 			--vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {" \
 			--vprod "      { @VALUENAME@, \"@VALUENAME@\", \"@valuenick@\" }," \
