--- crmsh-3.0.1/hb_report/hb_report.in.orig	2017-09-23 22:45:05.286109072 +0000
+++ crmsh-3.0.1/hb_report/hb_report.in	2017-09-23 22:46:01.469785546 +0000
@@ -1421,12 +1421,12 @@
 	if [ -z "$SSH_USER" -a `id -u` != 0 ] ||
 			[ -n "$SSH_USER" -a "$SSH_USER" != root ]; then
 		debug "ssh user other than root, use sudo"
-		SUDO="sudo -u root"
+		SUDO="sudo - root"
 	fi
 	LOCAL_SUDO=""
 	if [ `id -u` != 0 ]; then
 		debug "local user other than root, use sudo"
-		LOCAL_SUDO="sudo -u root"
+		LOCAL_SUDO="sudo - root"
 	fi
 fi
 
