--- crmsh-4.1.0/hb_report/hb_report.in.~1~	2019-08-19 08:48:47.883167482 +0000
+++ crmsh-4.1.0/hb_report/hb_report.in	2019-08-19 08:49:58.845386940 +0000
@@ -255,10 +255,10 @@
         if ((not constants.SSH_USER) and (os.getuid() != 0)) or \
            constants.SSH_USER and constants.SSH_USER != "root":
             utillib.log_debug("ssh user other than root, use sudo")
-            constants.SUDO = "sudo -u root"
+            constants.SUDO = "sudo - root"
         if os.getuid() != 0:
             utillib.log_debug("local user other than root, use sudo")
-            constants.LOCAL_SUDO = "sudo -u root"
+            constants.LOCAL_SUDO = "sudo - root"
 
     #
     # find the logs and cut out the segment for the period
