--- a/hb_report/hb_report.in	2016-10-26 05:52:01.650755386 +0200
+++ b/hb_report/hb_report.in	2016-10-26 05:52:35.784870126 +0200
@@ -1399,12 +1399,12 @@
 	if [ -z "$SSH_USER" -a `id -u` != 0 ] ||
 			[ -n "$SSH_USER" -a "$SSH_USER" != root ]; then
 		debug "ssh user other than root, use sudo"
-		SUDO="sudo -u root"
+		SUDO="sudo - root"
 	fi
 	LOCAL_SUDO=""
 	if [ `id -u` != 0 ]; then
 		debug "local user other than root, use sudo"
-		LOCAL_SUDO="sudo -u root"
+		LOCAL_SUDO="sudo - root"
 	fi
 fi
 
