--- a/m4/gecko.m4.orig	2011-03-21 11:45:39.585790802 +0800
+++ b/m4/gecko.m4	2011-03-21 11:52:12.887346051 +0800
@@ -276,8 +277,10 @@
 	version = "1.9";
 } else if (strncmp (MOZILLA_VERSION, "1.8", strlen ("1.8")) == 0) {
 	version = "1.8";
-} else {
+} else if (strncmp (MOZILLA_VERSION, "1.", strlen ("1.")) == 0) {
 	version = "1.7";
+} else {
+	version = "2.0";
 }
 #endif
 fprintf (stream, "%s\n", version);
@@ -296,7 +299,7 @@
 
 gecko_cv_gecko_version_int="$(echo "$gecko_cv_gecko_version" | $AWK -F . '{print [$]1 * 1000000 + [$]2 * 1000 + [$]3}')"
 
-if test "$gecko_cv_gecko_version_int" -lt "1007000" -o "$gecko_cv_gecko_version_int" -gt "1009000"; then
+if test "$gecko_cv_gecko_version_int" -lt "1007000" -o "$gecko_cv_gecko_version_int" -gt "2000000"; then
 	AC_MSG_ERROR([Gecko version $gecko_cv_gecko_version is not supported!])
 fi
 
@@ -316,6 +319,10 @@
 	AC_DEFINE([HAVE_GECKO_1_9],[1],[Define if we have gecko 1.9])
 	gecko_cv_have_gecko_1_9=yes
 fi
+if test "$gecko_cv_gecko_version_int" -ge "2000000"; then
+	AC_DEFINE([HAVE_GECKO_2_0],[1],[Define if we have gecko 2.0])
+	gecko_cv_have_gecko_2_0=yes
+fi
 
 fi # if gecko_cv_have_gecko
 
