--- rsync-3.0.9/main.c.orig	2011-06-18 12:44:47.000000000 -0700
+++ rsync-3.0.9/main.c	2013-04-19 14:14:00.331981573 -0700
@@ -1100,7 +1100,7 @@
 	for (i = 0; argv[i]; i++) {
 		if (!(argv[i] = strdup(argv[i]))) {
 			rprintf (FERROR, "out of memory at %s(%d)\n",
-				 __FILE__, __LINE__);
+				 basename(__FILE__), __LINE__);
 			return RERR_MALLOC;
 		}
 	}
--- rsync-3.0.9/cleanup.c.orig	2011-02-21 10:20:58.000000000 -0800
+++ rsync-3.0.9/cleanup.c	2013-04-19 14:14:00.332420836 -0700
@@ -126,7 +126,7 @@
 		if (verbose > 3) {
 			rprintf(FINFO,
 				"[%s] _exit_cleanup(code=%d, file=%s, line=%d): entered\n",
-				who_am_i(), code, file, line);
+				who_am_i(), code, basename(file), line);
 		}
 
 		/* FALLTHROUGH */
@@ -202,7 +202,7 @@
 			rprintf(FINFO,
 				"[%s] _exit_cleanup(code=%d, file=%s, line=%d): "
 				"about to call exit(%d)\n",
-				who_am_i(), unmodified_code, file, line, code);
+				who_am_i(), unmodified_code, basename(file), line, code);
 		}
 
 		/* FALLTHROUGH */
--- rsync-3.0.9/t_stub.c.orig	2011-02-21 11:32:48.000000000 -0800
+++ rsync-3.0.9/t_stub.c	2013-04-19 14:14:00.332763107 -0700
@@ -54,7 +54,7 @@
  void _exit_cleanup(int code, const char *file, int line)
 {
 	fprintf(stderr, "exit(%d): %s(%d)\n",
-		code, file, line);
+		code, basename(file), line);
 	exit(code);
 }
 
--- rsync-3.0.9/log.c.orig	2011-01-29 19:25:53.000000000 -0800
+++ rsync-3.0.9/log.c	2013-04-19 14:14:00.333124681 -0700
@@ -829,10 +829,10 @@
 		/* VANISHED is not an error, only a warning */
 		if (code == RERR_VANISHED) {
 			rprintf(FWARNING, "rsync warning: %s (code %d) at %s(%d) [%s=%s]\n",
-				name, code, file, line, who_am_i(), RSYNC_VERSION);
+				name, code, basename(file), line, who_am_i(), RSYNC_VERSION);
 		} else {
 			rprintf(FERROR, "rsync error: %s (code %d) at %s(%d) [%s=%s]\n",
-				name, code, file, line, who_am_i(), RSYNC_VERSION);
+				name, code, basename(file), line, who_am_i(), RSYNC_VERSION);
 		}
 	}
 }
