*** cim-client/build.xml.orig	Wed Jul 13 13:42:20 2011
--- cim-client/build.xml	Wed Jul 13 13:43:08 2011
***************
*** 164,169 ****
--- 164,170 ----
                  dir="${basedir}"
                  includes="**/*.java"
              	excludes="org/sblim/wbem/cimclient/sample/**/*.java, org/sblim/slp/example/*.java"/>
+             <arg value="-notimestamp" />
          </javadoc>
      </target>
  
