diff -urp -x '*~' -x '*.orig' hw/xfree86/dixmods/Makefile.am hw/xfree86/dixmods/Makefile.am
--- hw/xfree86/dixmods/Makefile.am	2009-01-22 20:34:57.149681000 -0800
+++ hw/xfree86/dixmods/Makefile.am	2009-01-22 20:35:04.779066000 -0800
@@ -44,7 +44,7 @@ libwfb_la_LIBADD = $(top_builddir)/fb/li
 libwfb_la_SOURCES = $(top_builddir)/fb/fbcmap_mi.c fbmodule.c
 libwfb_la_CFLAGS = $(AM_CFLAGS) -DFB_ACCESS_WRAPPER
 
-libglx_la_LDFLAGS = -avoid-version
+libglx_la_LDFLAGS = -avoid-version -mimpure-text
 if AIGLX
 GLXDRI_LIBRARY = $(top_builddir)/glx/libglxdri.la
 endif
