diff -urp -x '*~' -x '*.orig' hw/xfree86/dri/dri.c hw/xfree86/dri/dri.c
--- hw/xfree86/dri/dri.c	2009-12-02 18:58:16.434496135 -0800
+++ hw/xfree86/dri/dri.c	2009-12-02 18:58:18.707182684 -0800
@@ -2490,7 +2490,11 @@ int drmInstallSIGIOHandler(int fd, void 
     entry     = drmGetEntry(fd);
     entry->f  = f;
 
+#if defined(__sun) && defined(__SVR4)
+    return 1;
+#else
     return xf86InstallSIGIOHandler(fd, drmSIGIOHandler, 0);
+#endif
 }
 
 int drmRemoveSIGIOHandler(int fd)
@@ -2499,5 +2503,9 @@ int drmRemoveSIGIOHandler(int fd)
 
     entry->f = NULL;
 
+#if defined(__sun) && defined(__SVR4)
+    return 1;
+#else
     return xf86RemoveSIGIOHandler(fd);
+#endif
 }
