--- hw/xfree86/dixmods/Makefile.am.~1~	2015-01-06 14:53:39.074104827 +0300
+++ hw/xfree86/dixmods/Makefile.am	2015-01-06 14:53:58.439108743 +0300
@@ -36,6 +36,7 @@ libdbe_la_SOURCES = dbemodule.c
 
 libfb_la_LDFLAGS = -avoid-version
 libfb_la_LIBADD = $(top_builddir)/fb/libfb.la
+libfb_la_LIBADD += -lpixman-1
 libfb_la_SOURCES = $(top_builddir)/fb/fbcmap_mi.c fbmodule.c
 libfb_la_CFLAGS = $(AM_CFLAGS)
 
@@ -47,6 +48,8 @@ libwfb_la_CFLAGS = $(AM_CFLAGS) -DFB_ACC
 libglx_la_LDFLAGS = -avoid-version -mimpure-text
 if AIGLX
 GLXDRI_LIBRARY = $(top_builddir)/glx/libglxdri.la
+libglx_la_LDFLAGS += -R$(extsmoduledir)
+libglx_la_LDFLAGS += -Wl,-N,libdri.so -Wl,-N,libdri2.so
 endif
 libglx_la_LIBADD = \
 	$(top_builddir)/glx/libglx.la \
diff -urp -x '*~' -x '*.orig' hw/xfree86/int10/Makefile.am hw/xfree86/int10/Makefile.am
--- hw/xfree86/int10/Makefile.am	2007-10-17 07:01:23.000000000 -0700
+++ hw/xfree86/int10/Makefile.am	2009-05-08 17:46:07.162854000 -0700
@@ -42,3 +42,5 @@ libint10_la_SOURCES = stub.c xf86int10mo
 endif
 
 EXTRA_DIST = xf86x86emu.h INT10.HOWTO
+
+libint10_la_LIBADD = $(PCIACCESS_LIBS)
--- hw/xfree86/shadowfb/Makefile.am.~1~	2008-04-16 10:14:18.000000000 +0400
+++ hw/xfree86/shadowfb/Makefile.am	2014-09-24 20:41:06.641419796 +0400
@@ -1,5 +1,5 @@
 module_LTLIBRARIES = libshadowfb.la
-libshadowfb_la_LDFLAGS = -avoid-version
+libshadowfb_la_LDFLAGS = -L/usr/lib/xorg$(SERVERMOD_SUBDIR) -R/usr/lib/xorg$(SERVERMOD_SUBDIR) -avoid-version
 libshadowfb_la_SOURCES = sfbmodule.c shadow.c
 
 sdk_HEADERS = shadowfb.h
@@ -7,3 +7,5 @@
 INCLUDES = $(XORG_INCS)
 
 AM_CFLAGS = $(DIX_CFLAGS) $(XORG_CFLAGS)
+
+libshadowfb_la_LIBADD = -lXfont
--- hw/xfree86/vbe/Makefile.am~	Tue Apr 15 23:14:18 2008
+++ hw/xfree86/vbe/Makefile.am	Tue Apr 20 16:01:20 2010
@@ -8,3 +8,5 @@
 
 INCLUDES = $(XORG_INCS) -I$(srcdir)/../ddc -I$(srcdir)/../i2c \
            -I$(srcdir)/../int10
+
+libvbe_la_LIBADD = ../int10/libint10.la
