diff -urp -x '*~' -x '*.orig' mi/micmap.c mi/micmap.c
--- mi/micmap.c	2009-10-27 23:56:24.000000000 -0700
+++ mi/micmap.c	2009-12-02 18:58:23.806599766 -0800
@@ -499,6 +499,11 @@ maskShift (Pixel p)
  * the set which can be used with this version of cfb.
  */
 
+#if defined(__sparc__) || defined(__sparc)
+_X_EXPORT miInitVisualsProcPtr miInitVisualsProc = NULL;
+#endif /* __sparc__ */
+
+
 Bool
 miInitVisuals(VisualPtr *visualp, DepthPtr *depthp, int *nvisualp,
 		int *ndepthp, int *rootDepthp, VisualID *defaultVisp,
@@ -518,6 +523,13 @@ miInitVisuals(VisualPtr *visualp, DepthP
     int		*preferredCVCs, *prefp;
     int		first_depth;
 
+#if defined(__sparc__) || defined(__sparc)
+    if (miInitVisualsProc) {
+       return (miInitVisualsProc(visualp, depthp, nvisualp, ndepthp,
+               rootDepthp, defaultVisp, sizes, bitsPerRGB, preferredVis));
+    }
+#endif /* __sparc__ */
+
     /* none specified, we'll guess from pixmap formats */
     if (!miVisuals) 
     {
