--- a/hw/xfree86/utils/cvt/cvt.c	Fri Jan 16 02:27:48 2015
+++ b/hw/xfree86/utils/cvt/cvt.c	Fri Jan 16 11:48:28 2015
@@ -31,6 +31,14 @@
 #include "regionstr.h"
 _X_EXPORT BoxRec RegionEmptyBox = { 0, 0, 0, 0 };
 _X_EXPORT RegDataRec RegionEmptyData = { 0, 0 };
+
+/* added to quieten undefined symbol error while compiling with Studio 12.4 */
+#include "dixstruct.h"
+_X_HIDDEN DevPrivateKey
+_dixGetScreenPrivateKey(const DevScreenPrivateKey key, ScreenPtr pScreen)
+{
+    abort();
+}
 #endif
 
 #include "xf86.h"
--- a/hw/dmx/config/xdmxconfig.c	Thu Jul 25 22:14:32 2013
+++ b/hw/dmx/config/xdmxconfig.c	Wed Jan 21 08:51:48 2015
@@ -55,6 +55,16 @@
 extern int yyparse(void);
 extern FILE *yyin;
 
+#if (defined(__sparc__) || defined(__sparc))
+/* added to quieten undefined symbol error while compiling with Studio 12.4 */
+#include "dixstruct.h"
+_X_HIDDEN DevPrivateKey
+_dixGetScreenPrivateKey(const DevScreenPrivateKey key, ScreenPtr pScreen)
+{
+    abort();
+}
+#endif
+
 #define DMX_INFO "xdmxconfig v0.9\nCopyright 2002 Red Hat Inc.\n"
 
 #define DMX_MAIN_WIDTH    800
