--- a/mi/miinitext.c	Fri Aug 30 09:09:00 2013
+++ b/mi/miinitext.c	Fri Aug 30 10:10:51 2013
@@ -317,6 +317,12 @@
 #ifdef RES
     {ResExtensionInit, XRES_NAME, &noResExtension},
 #endif
+#if defined(SolarisIAExtension) && !defined(XORGSERVER)
+    {IAExtensionInit, IANAME, &noIAExtension},
+#endif
+#if defined(TSOL) && !defined(XORGSERVER)
+    {TsolExtensionInit, TSOLNAME, &noXTSolExtension},
+#endif
 #ifdef XV
     {XvExtensionInit, XvName, &noXvExtension},
     {XvMCExtensionInit, XvMCName, &noXvExtension},
