--- exa/exa_render.c	Wed Jan  6 12:31:27 2010
+++ exa/exa_render.c	Wed Jan  6 12:32:59 2010
@@ -903,8 +903,9 @@
     Bool saveMaskRepeat = pMask ? pMask->repeat : 0;
     RegionRec region;
 
-    if (pExaScr->swappedOut)
-	goto fallback;
+    if (pExaScr->swappedOut || pSrc->pDrawable == NULL || 
+	(pMask != NULL && pMask->pDrawable == NULL))
+	    goto fallback;
 
     /* Remove repeat in source if useless */
     if (pSrc->pDrawable && pSrc->repeat && !pSrc->transform && xSrc >= 0 &&
