--- exa/exa.c	Tue Jan 12 15:07:10 2010
+++ exa/exa.c	Tue Jan 12 15:10:44 2010
@@ -64,7 +64,11 @@
     ExaScreenPriv (pPix->drawable.pScreen);
     ExaPixmapPriv (pPix);
 
-    return (CARD8 *)pExaPixmap->fb_ptr - pExaScr->info->memoryBase;
+    /* added in as a fix for 6912996 */
+    if (pExaPixmap->offscreen && pExaPixmap->fb_ptr)
+	return (CARD8 *)pExaPixmap->fb_ptr - pExaScr->info->memoryBase;
+    else
+	return (CARD8 *)pExaPixmap->sys_ptr - pExaScr->info->memoryBase;
 }
 
 void *
