https://bugs.freedesktop.org/show_bug.cgi?id=25418

--- configure.ac.~1~	2014-04-27 17:15:12.053881922 +0400
+++ configure.ac	2014-04-27 17:15:59.516733895 +0400
@@ -665,7 +665,7 @@
 dnl first and if it's not in the default location, that you set the ACLOCAL
 dnl environment variable to find it, such as:
 dnl	ACLOCAL="aclocal -I ${PREFIX}/share/aclocal"
-m4_pattern_forbid(XTRANS_CONNECTION_FLAGS)
+m4_pattern_forbid([^XTRANS_CONNECTION_FLAGS$])
 
 # Transport selection macro from xtrans.m4
 XTRANS_CONNECTION_FLAGS
