diff -urN gnome-vfs-2.16.2/schemas/system_http_proxy.schemas.in ../SUNWgnome-vfs-2.16.1.hacked/gnome-vfs-2.16.2/schemas/system_http_proxy.schemas.in
--- gnome-vfs-2.16.2/schemas/system_http_proxy.schemas.in	2005-01-31 01:51:04.000000000 +0000
+++ ../SUNWgnome-vfs-2.16.1.hacked/gnome-vfs-2.16.2/schemas/system_http_proxy.schemas.in	2006-11-10 15:54:40.903612000 +0000
@@ -170,6 +170,17 @@
       </locale>
     </schema>
     <schema>
+      <key>/schemas/system/proxy/socks_version</key>
+      <applyto>/system/proxy/socks_version</applyto>
+      <owner>gnome-vfs</owner>
+      <type>int</type>
+      <default>5</default>
+      <locale name="C">
+        <short>SOCKS version</short>
+        <long>The SOCKS version of the proxy server</long>
+      </locale>
+    </schema>
+    <schema>
       <key>/schemas/system/proxy/autoconfig_url</key>
       <applyto>/system/proxy/autoconfig_url</applyto>
       <owner>gnome-vfs</owner>
