--- gnome-vfs-2.20.0-orig/libgnomevfs/gnome-vfs-filesystem-type.c	2007-09-17 18:00:46.000000000 +0100
+++ gnome-vfs-2.20.0-migi/libgnomevfs/gnome-vfs-filesystem-type.c	2007-11-08 16:54:57.983327965 +0000
@@ -85,6 +85,8 @@
 	{ "xfs"      , N_("XFS Linux Volume"), 1 },
 	{ "xiafs"    , N_("XIAFS Volume"), 0 },
 	{ "cifs"     , N_("CIFS Volume"), 1 },
+	/* SUN_BRANDING */
+        { "zfs"     , N_("ZFS Volume"), 1 },
 };
 
 static struct FSInfo *
--- gnome-vfs-2.20.0-orig/libgnomevfs/gnome-vfs-volume.c	2007-11-08 16:53:39.848358165 +0000
+++ gnome-vfs-2.20.0-migi/libgnomevfs/gnome-vfs-volume.c	2007-11-08 16:56:01.527169813 +0000
@@ -416,6 +416,7 @@
  *    <row><entry>xfs</entry><entry>XFS Linux Volume</entry><entry>Yes</entry></row>
  *    <row><entry>xiafs</entry><entry>XIAFS Volume</entry><entry>No</entry></row>
  *    <row><entry>cifs</entry><entry>CIFS Volume</entry><entry>Yes</entry></row>
+ *    <row><entry>zfs</entry><entry>ZFS Volume</entry><entry>Yes</entry></row>
  *   </tbody>
  *  </tgroup>
  * </table>
