diff -ur gnome-vfs-2.16.1/libgnomevfs/gnome-vfs-xfer.c ../SUNWgnome-vfs-2.16.1.hacked/gnome-vfs-2.16.1/libgnomevfs/gnome-vfs-xfer.c
--- gnome-vfs-2.16.1/libgnomevfs/gnome-vfs-xfer.c	2006-08-21 14:49:48.000000000 +0000
+++ ../SUNWgnome-vfs-2.16.1.hacked/gnome-vfs-2.16.1/libgnomevfs/gnome-vfs-xfer.c	2006-10-20 15:28:26.367491000 +0000
@@ -2390,8 +2390,8 @@
 	merge_source_uri_list = NULL;
 	merge_target_uri_list = NULL;
 
-	if ((xfer_options & GNOME_VFS_XFER_USE_UNIQUE_NAMES) == 0) {
-		/* see if moved files are on the same file system so that we decide to do
+	/* if ((xfer_options & GNOME_VFS_XFER_USE_UNIQUE_NAMES) == 0) {
+		 * see if moved files are on the same file system so that we decide to do
 		 * a simple move or have to do a copy/remove
 		 */
 		for (source_uri = source_uri_list, target_uri = target_uri_list;
@@ -2410,7 +2410,7 @@
 
 			move &= same_fs;
 		}
-	}
+	/*}*/
 
 	if (target_dir_uri != NULL) {
 		gnome_vfs_uri_unref (target_dir_uri);
