We have no html5lib packaged yet.

--- soupsieve-2.6/pyproject.toml.orig
+++ soupsieve-2.6/pyproject.toml
@@ -110,17 +110,6 @@
     py{38,39,310,311,312},
     lint, nolxml, nohtml5lib
 
-[testenv]
-passenv = *
-deps =
-    -rrequirements/tests.txt
-commands =
-    mypy
-    pytest --cov soupsieve --cov-append {toxinidir}
-    coverage html -d {envtmpdir}/coverage
-    coverage xml
-    coverage report --show-missing
-
 [testenv:documents]
 passenv = *
 deps =
@@ -143,7 +132,7 @@
 commands =
     pytest {toxinidir}
 
-[testenv:nohtml5lib]
+[testenv]
 passenv = *
 deps =
     -rrequirements/tests-nohtml5lib.txt
