--- mozilla/security/jss/lib/config.mk.~1~	2014-09-25 17:23:36.240822508 +0400
+++ mozilla/security/jss/lib/config.mk	2014-09-25 17:25:06.955609145 +0400
@@ -102,7 +102,17 @@
 ifeq ($(OS_ARCH),SunOS)
 MAPFILE = $(OBJDIR)/jssmap.sun
 ALL_TRASH += $(MAPFILE)
-MKSHLIB += -M $(MAPFILE)
+
+ifdef NS_USE_GCC
+ifdef GCC_USE_GNU_LD
+    MKSHLIB += -Wl,--version-script,$(MAPFILE)
+else
+    MKSHLIB += -Wl,-M,$(MAPFILE)
+endif
+else
+    MKSHLIB += -M $(MAPFILE)
+endif
+
 #ifndef USE_64
 #ifeq ($(CPU_ARCH),sparc)
 # The -R '$ORIGIN' linker option instructs libnss3.so to search for its
--- mozilla/security/jss/config/config.mk.~1~	2014-09-25 17:28:31.124563545 +0400
+++ mozilla/security/jss/config/config.mk	2014-09-25 17:29:39.401199918 +0400
@@ -61,7 +61,9 @@
 # Stricter semantic checking for SunOS compiler. This catches calling
 # undeclared functions, a major headache during debugging.
 ifeq ($(OS_ARCH), SunOS)
+  ifndef NS_USE_GCC
     OS_CFLAGS += -v
+  endif
 endif

 ifeq ($(OS_ARCH), WINNT)
