--- modsecurity-2.9.2/modsecurity.conf-recommended.~1~	2017-07-18 21:45:57.000000000 +0000
+++ modsecurity-2.9.2/modsecurity.conf-recommended	2019-03-01 18:11:39.397303649 +0000
@@ -1,3 +1,12 @@
+# Load mod_unique_id if necessary
+<IfModule !unique_id_module>
+LoadModule unique_id_module libexec/mod_unique_id.so
+</IfModule>
+
+LoadModule security2_module libexec/mod_security2.so
+
+<IfModule mod_security2.c>
+
 # -- Rule engine initialization ----------------------------------------------
 
 # Enable ModSecurity, attaching it to every transaction. Use detection
@@ -135,13 +144,13 @@
 # This default setting is chosen due to all systems have /tmp available however, 
 # this is less than ideal. It is recommended that you specify a location that's private.
 #
-SecTmpDir /tmp/
+SecTmpDir /var/apache2/2.4/modsec/tmp/
 
 # The location where ModSecurity will keep its persistent data.  This default setting 
 # is chosen due to all systems have /tmp available however, it
 # too should be updated to a place that other users can't access.
 #
-SecDataDir /tmp/
+SecDataDir /var/apache2/2.4/modsec/tmp/
 
 
 # -- File uploads handling configuration -------------------------------------
@@ -150,7 +159,7 @@
 # location must be private to ModSecurity. You don't want other users on
 # the server to access the files, do you?
 #
-#SecUploadDir /opt/modsecurity/var/upload/
+#SecUploadDir /var/apache2/2.4/modsec/upload/
 
 # By default, only keep the files that were determined to be unusual
 # in some way (by an external inspection script). For this to work you
@@ -170,7 +179,7 @@
 # The default debug log configuration is to duplicate the error, warning
 # and notice messages from the error log.
 #
-#SecDebugLog /opt/modsecurity/var/log/debug.log
+#SecDebugLog /var/apache2/2.4/logs/modsec_debug.log
 #SecDebugLogLevel 3
 
 
@@ -190,10 +199,10 @@
 # assumes that you will use the audit log only ocassionally.
 #
 SecAuditLogType Serial
-SecAuditLog /var/log/modsec_audit.log
+SecAuditLog /var/apache2/2.4/logs/modsec_audit.log
 
 # Specify the path for concurrent audit logging.
-#SecAuditLogStorageDir /opt/modsecurity/var/audit/
+#SecAuditLogStorageDir /var/apache2/2.4/modsec/audit/
 
 
 # -- Miscellaneous -----------------------------------------------------------
@@ -224,3 +233,4 @@
 # version, Anonymous unique id for host.
 SecStatusEngine On
 
+</IfModule>
