--- pidgin-2.7.0/finch/gntsound.c.orig1	2010-05-13 15:42:22.086579807 +0800
+++ pidgin-2.7.0/finch/gntsound.c	2010-05-13 15:42:36.686243569 +0800
@@ -427,7 +427,6 @@
 		/* fall-through and clean up */
 	case GST_MESSAGE_EOS:
 		gst_element_set_state(play, GST_STATE_NULL);
-		gst_object_unref(GST_OBJECT(play));
 		break;
 	case GST_MESSAGE_WARNING:
 		gst_message_parse_warning(msg, &err, NULL);
--- pidgin-2.7.0/pidgin/gtksound.c.orig1	2010-05-13 15:41:30.889630112 +0800
+++ pidgin-2.7.0/pidgin/gtksound.c	2010-05-13 15:41:59.172381670 +0800
@@ -393,7 +393,6 @@
 		/* fall-through and clean up */
 	case GST_MESSAGE_EOS:
 		gst_element_set_state(play, GST_STATE_NULL);
-		gst_object_unref(GST_OBJECT(play));
 		break;
 	case GST_MESSAGE_WARNING:
 		gst_message_parse_warning(msg, &err, NULL);
