--- gtkperf/configure.in.orig	2020-01-24 01:34:26.382604417 +0000
+++ gtkperf/configure.in	2020-01-24 01:35:26.581890683 +0000
@@ -29,7 +29,7 @@
 
 dnl Languages which your application supports
 ALL_LINGUAS=""
-AM_GNU_GETTEXT
+AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION(0.10.40)
 
 dnl Set PACKAGE_LOCALE_DIR in config.h.
--- gtkperf/Makefile.am.orig	2020-01-24 01:35:55.619597255 +0000
+++ gtkperf/Makefile.am	2020-01-24 01:36:10.075107611 +0000
@@ -3,7 +3,7 @@
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
 
-SUBDIRS = intl po src pixmaps
+SUBDIRS = po src pixmaps
 
 gtkperfdocdir = ${prefix}/doc/gtkperf
 gtkperfdoc_DATA = \
