diff -Naur fail2ban-1.0.2/config.old/jail.conf fail2ban-1.0.2/config/jail.conf
--- fail2ban-1.0.2/config.old/jail.conf	2022-11-09 10:46:15.000000000 -0500
+++ fail2ban-1.0.2/config/jail.conf	2024-01-18 00:47:34.850887929 -0500
@@ -33,7 +33,7 @@
 [INCLUDES]
 
 #before = paths-distro.conf
-before = paths-debian.conf
+before = paths-openindiana.conf
 
 # The DEFAULT allows a global definition of the options. They can be overridden
 # in each jail afterwards.
