https://github.com/tox-dev/python-discovery/issues/75

--- python_discovery-1.3.1/src/python_discovery/_py_spec.py.orig
+++ python_discovery-1.3.1/src/python_discovery/_py_spec.py
@@ -17,7 +17,7 @@
     (?P<version>[0-9.]+)?           # version (e.g. 3.12, 3.12.1)
     (?P<threaded>t)?                # free-threaded flag
     (?:-(?P<arch>32|64))?           # architecture bitness
-    (?:-(?P<machine>[a-zA-Z0-9_]+))?  # ISA (e.g. arm64, x86_64)
+    (?:-(?P<machine>[a-zA-Z0-9_.]+))?  # ISA (e.g. arm64, x86_64)
     $
     """,
     re.VERBOSE,
