--- vobcopy-1.2.0/Makefile.orig	2016-09-10 00:22:09.805469536 +0200
+++ vobcopy-1.2.0/Makefile	2016-09-10 00:22:13.741619295 +0200
@@ -4,16 +4,13 @@
 #stuff places.
 DESTDIR = 
 CC     ?= gcc
-#PREFIX += /usr/local
-#BINDIR = ${PREFIX}/bin
-#MANDIR = ${PREFIX}/man
-PREFIX += /usr/local
-BINDIR = ${PREFIX}/bin
-MANDIR = ${PREFIX}/man
-DOCDIR = ${PREFIX}/share/doc/vobcopy
-LFS    = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CFLAGS += -I/usr/local//include
-LDFLAGS += -ldvdread -L/usr/local//lib
+PREFIX ?= /usr/local
+BINDIR ?= ${PREFIX}/bin
+MANDIR ?= ${PREFIX}/man
+DOCDIR ?= ${PREFIX}/share/doc
+LFS     = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+CFLAGS += -I/usr/include
+LDFLAGS+= -ldvdread 
 
 #This specifies the conversion from .c to .o 
 .c.o:
@@ -34,12 +31,6 @@
 	$(CC) -c dvd.c     -Wall -ggdb -pedantic $(CFLAGS) $(LFS)
 	$(CC) -o vobcopy vobcopy.o dvd.o -ldvdread
 
-deb:
-        
-	echo "this here is really really experimental..."
-	dpkg-buildpackage -rfakeroot -us -uc -tc
-		
-
 clean :
 	rm -f vobcopy vobcopy.o dvd.o
 
@@ -47,17 +38,14 @@
 	rm -f vobcopy.o dvd.o *~
 
 install:
-#	mkdir -p $(MANDIR)/man1
-#	cp vobcopy   $(BINDIR)/vobcopy
-#	cp vobcopy.1 $(MANDIR)/man1/vobcopy.1
 	install -d -m 755 $(DESTDIR)/$(BINDIR)
 	install -d -m 755 $(DESTDIR)/$(MANDIR)/man1
 	install -d -m 755 $(DESTDIR)/$(MANDIR)/de/man1
-	install -d -m 755 $(DESTDIR)/$(DOCDIR)
+	install -d -m 755 $(DESTDIR)/$(DOCDIR)/vobcopy
 	install -m 755 vobcopy $(DESTDIR)/$(BINDIR)/vobcopy
 	install -m 644 vobcopy.1 $(DESTDIR)/$(MANDIR)/man1/vobcopy.1
 	install -m 644 vobcopy.1.de $(DESTDIR)/$(MANDIR)/de/man1/vobcopy.1
-	install -m 644 COPYING Changelog README Release-Notes TODO $(DESTDIR)/$(DOCDIR)
+	install -m 644 COPYING Changelog README Release-Notes TODO $(DESTDIR)/$(DOCDIR)/vobcopy
 
 uninstall:
 	rm -f $(DESTDIR)/$(BINDIR)/vobcopy
