diff -Nru vobcopy-1.2.1.orig/vobcopy.h vobcopy-1.2.1/vobcopy.h
--- vobcopy-1.2.1.orig/vobcopy.h	2022-05-03 21:55:34.000000000 +0200
+++ vobcopy-1.2.1/vobcopy.h	2023-03-21 17:18:28.299893243 +0100
@@ -107,7 +107,6 @@
 
 #ifdef HAVE_SYS_VFS_H
 #include <sys/vfs.h>
-#define USE_STATFS
 /* #define USE_STATFS_FOR_DEV */
 #endif
 
@@ -115,9 +114,6 @@
 #include <sys/statvfs.h>
 #ifndef USE_STATFS
 #define USE_STATVFS
-#ifndef USE_STATFS_FOR_DEV
-#define USE_STATVFS_FOR_DEV
-#endif
 #endif
 #endif
 
