diff -ruN joyent-illumos-kvm-cmd-8fa43e8.orig/configure joyent-illumos-kvm-cmd-8fa43e8/configure
--- joyent-illumos-kvm-cmd-8fa43e8.orig/configure	2011-08-13 18:37:55.000000000 +0100
+++ joyent-illumos-kvm-cmd-8fa43e8/configure	2011-08-19 15:30:39.495506247 +0100
@@ -755,7 +749,7 @@
   ;;
   --enable-sunos-vnic) sunos_vnic="yes"
   ;;
-  *) echo "ERROR: unknown option $opt"; show_help="yes"
+  *) echo "ERROR: unknown option $opt"
   ;;
   esac
 done
