see https://bugs.archlinux.org/task/60489
--- libgnome-keyring-3.9.1/library/GnomeKeyring-1.0.metadata	2012-09-25 15:42:47.000000000 +0200
+++ libgnome-keyring-3.9.1/library/GnomeKeyring-1.0.metadata.new	2023-11-29 20:36:34.106292881 +0100
@@ -54,7 +54,7 @@
 set_network_password skip=false
 store_password skip=false
 store_password_sync skip=false
-unlock skip=false
+@unlock skip=false
 
 find_items*.callback type_arguments="GnomeKeyring.Found"
 find_network_password*.callback type_arguments="GnomeKeyring.NetworkPasswordData"
