--- gnome-settings-daemon-2.30.2/plugins/housekeeping/gsd-housekeeping-manager.c.orig	Fri Apr 11 16:43:13 2014
+++ gnome-settings-daemon-2.30.2/plugins/housekeeping/gsd-housekeeping-manager.c	Fri Apr 11 16:45:13 2014
@@ -193,22 +193,37 @@
 
         g_debug ("housekeeping: checking thumbnail cache size and freshness");
 
+#if GLIB_CHECK_VERSION (2, 34, 0)
+	path = g_build_filename (g_get_user_cache_dir (),
+				"thumbnails",
+#else
         path = g_build_filename (g_get_home_dir (),
                                  ".thumbnails",
+#endif
                                  "normal",
                                  NULL);
         files = read_dir_for_purge (path, NULL);
         g_free (path);
 
+#if GLIB_CHECK_VERSION (2, 34, 0)
+	path = g_build_filename (g_get_user_cache_dir (),
+				"thumbnails",
+#else
         path = g_build_filename (g_get_home_dir (),
                                  ".thumbnails",
+#endif
                                  "large",
                                  NULL);
         files = read_dir_for_purge (path, files);
         g_free (path);
 
+#if GLIB_CHECK_VERSION (2, 34, 0)
+	path = g_build_filename (g_get_user_cache_dir (),
+				"thumbnails",
+#else
         path = g_build_filename (g_get_home_dir (),
                                  ".thumbnails",
+#endif
                                  "fail",
                                  "gnome-thumbnail-factory",
                                  NULL);
