diff -rup ../SUNWgnome-desktop-prefs-2.28.0/gnome-settings-daemon-2.28.1/data/desktop_gnome_font_rendering.schemas.in gnome-settings-daemon-2.28.1/data/desktop_gnome_font_rendering.schemas.in
--- ../SUNWgnome-desktop-prefs-2.28.0/gnome-settings-daemon-2.28.1/data/desktop_gnome_font_rendering.schemas.in	2009-05-03 20:15:30.000000000 +0200
+++ gnome-settings-daemon-2.28.1/data/desktop_gnome_font_rendering.schemas.in	2009-10-30 15:51:55.336169575 +0100
@@ -6,6 +6,7 @@
 	    <applyto>/desktop/gnome/font_rendering/dpi</applyto>
             <owner>gnome</owner>
 	    <type>float</type>
+	    <default>96</default>
             <locale name="C">
 	        <short>DPI</short>
                 <long>
@@ -19,7 +20,7 @@
 	    <applyto>/desktop/gnome/font_rendering/antialiasing</applyto>
             <owner>gnome</owner>
 	    <type>string</type>
-            <default>grayscale</default>
+            <default>rgba</default>
             <locale name="C">
 	        <short>Antialiasing</short>
                 <long>
@@ -35,7 +36,7 @@
 	    <applyto>/desktop/gnome/font_rendering/hinting</applyto>
             <owner>gnome</owner>
 	    <type>string</type>
-            <default>medium</default>
+            <default>full</default>
             <locale name="C">
 	        <short>Hinting</short>
                 <long>
diff -rup ../SUNWgnome-desktop-prefs-2.28.0/gnome-settings-daemon-2.28.1/plugins/xsettings/gsd-xsettings-manager.c gnome-settings-daemon-2.28.1/plugins/xsettings/gsd-xsettings-manager.c
--- ../SUNWgnome-desktop-prefs-2.28.0/gnome-settings-daemon-2.28.1/plugins/xsettings/gsd-xsettings-manager.c	2009-09-21 12:24:02.000000000 +0200
+++ gnome-settings-daemon-2.28.1/plugins/xsettings/gsd-xsettings-manager.c	2009-10-30 15:51:30.463357438 +0100
@@ -268,7 +268,7 @@ get_dpi_from_gconf_or_x_server (GConfCli
         GConfValue *value;
         double      dpi;
 
-        value = gconf_client_get_without_default (client, FONT_DPI_KEY, NULL);
+        value = gconf_client_get (client, FONT_DPI_KEY, NULL);
 
         /* If the user has ever set the DPI preference in GConf, we use that.
          * Otherwise, we see if the X server reports a reasonable DPI value:  some X
