diff -up gnome-settings-daemon-2.21.5/config.h.in.orig gnome-settings-daemon-2.21.5/config.h.in
--- gnome-settings-daemon-2.21.5/config.h.in.orig	2008-01-24 07:29:51.136459000 +0000
+++ gnome-settings-daemon-2.21.5/config.h.in	2008-01-24 07:32:05.292550000 +0000
@@ -54,6 +54,9 @@
 /* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
 
+/* Define if Xft functionality is available */
+#undef HAVE_XFT2
+
 /* Define to 1 if you have the <X11/extensions/xf86misc.h> header file. */
 #undef HAVE_X11_EXTENSIONS_XF86MISC_H
 

--- gnome-settings-daemon-2.25.2/configure.ac.ori	2008-12-24 08:27:48.785545893 +0000
+++ gnome-settings-daemon-2.25.2/configure.ac	2008-12-24 08:30:24.733312829 +0000
@@ -113,6 +113,16 @@
 dnl - X11 stuff
 dnl ---------------------------------------------------------------------------
 
+dnl
+dnl Check for Xft version 2; we build in extra functionality to the font capplet
+dnl when we have it.
+dnl
+xft_modules=
+if $PKG_CONFIG --exists xft ; then
+  xft_modules="xft"
+  AC_DEFINE(HAVE_XFT2,,[Define if Xft functionality is available])
+fi
+
 # Like AC_CHECK_HEADERS, but it uses the already-computed -I directories.
 #
 AC_DEFUN([AC_CHECK_X_HEADERS], [
