--- ocaml-5.4.0/configure.orig	2025-10-09 11:14:48.000000000 +0200
+++ ocaml-5.4.0/configure	2025-10-15 19:26:32.409797284 +0200
@@ -8423,9 +8423,6 @@
       case $lt_cv_prog_gnu_ld in
       yes*)
         case $target in
-        i?86-*-solaris*|x86_64-*-solaris*)
-          LD="${LD-ld} -m elf_x86_64"
-          ;;
         sparc*-*-solaris*)
           LD="${LD-ld} -m elf64_sparc"
           ;;
