--- ocaml-4.06.0/asmrun/startup.c.orig	2020-05-25 17:48:40.752085284 +0000
+++ ocaml-4.06.0/asmrun/startup.c	2020-05-25 17:57:24.377786866 +0000
@@ -44,8 +44,8 @@
 #endif
 
 extern int caml_parser_trace;
-CAMLexport header_t caml_atom_table[256];
 char * caml_code_area_start, * caml_code_area_end;
+struct ext_table caml_code_fragments_table;
 
 /* Initialize the atom table and the static data and code area limits. */
 
--- ocaml-4.06.0/byterun/backtrace.c.orig	2020-05-25 17:52:33.951292237 +0000
+++ ocaml-4.06.0/byterun/backtrace.c	2020-05-25 17:53:03.334824259 +0000
@@ -27,9 +27,6 @@
 #include "caml/backtrace_prim.h"
 #include "caml/fail.h"
 
-/* The table of debug information fragments */
-struct ext_table caml_debug_info;
-
 CAMLexport int32_t caml_backtrace_active = 0;
 CAMLexport int32_t caml_backtrace_pos = 0;
 CAMLexport backtrace_slot * caml_backtrace_buffer = NULL;
--- ocaml-4.06.0/byterun/fix_code.c.orig	2020-05-25 17:49:26.850712856 +0000
+++ ocaml-4.06.0/byterun/fix_code.c	2020-05-25 17:49:42.927654228 +0000
@@ -38,6 +38,7 @@
 code_t caml_start_code;
 asize_t caml_code_size;
 unsigned char * caml_saved_code;
+struct ext_table caml_code_fragments_table;
 
 /* Read the main bytecode block from a file */
 
--- ocaml-4.06.0/byterun/caml/intext.h.orig	2020-05-25 17:46:17.875415077 +0000
+++ ocaml-4.06.0/byterun/caml/intext.h	2020-05-25 17:48:26.146522534 +0000
@@ -196,7 +196,7 @@
 
 CAMLextern struct code_fragment * caml_extern_find_code(char *addr);
 
-struct ext_table caml_code_fragments_table;
+extern struct ext_table caml_code_fragments_table;
 
 #endif /* CAML_INTERNALS */
 
--- ocaml-4.06.0/byterun/caml/major_gc.h.orig	2020-05-25 17:41:53.572400895 +0000
+++ ocaml-4.06.0/byterun/caml/major_gc.h	2020-05-25 17:42:42.281949692 +0000
@@ -64,9 +64,9 @@
 extern char *caml_gc_sweep_hp;
 
 extern int caml_major_window;
-double caml_major_ring[Max_major_window];
-int caml_major_ring_index;
-double caml_major_work_credit;
+extern double caml_major_ring[Max_major_window];
+extern int caml_major_ring_index;
+extern double caml_major_work_credit;
 extern double caml_gc_clock;
 
 /* [caml_major_gc_hook] is called just between the end of the mark
--- ocaml-4.06.0/byterun/debugger.c.orig	2020-05-25 17:45:48.074478150 +0000
+++ ocaml-4.06.0/byterun/debugger.c	2020-05-25 17:46:00.050116004 +0000
@@ -195,7 +195,7 @@
     sock_domain = PF_UNIX;
     sock_addr.s_unix.sun_family = AF_UNIX;
     strncpy(sock_addr.s_unix.sun_path, address,
-            sizeof(sock_addr.s_unix.sun_path));
+            sizeof(sock_addr.s_unix.sun_path) - 1);
     sock_addr_len =
       ((char *)&(sock_addr.s_unix.sun_path) - (char *)&(sock_addr.s_unix))
         + strlen(address);
