diff -wpruN --no-dereference '--exclude=*.orig' a~/Completion/Solaris/Command/_pkg5 a/Completion/Solaris/Command/_pkg5
--- a~/Completion/Solaris/Command/_pkg5	1970-01-01 00:00:00
+++ a/Completion/Solaris/Command/_pkg5	1970-01-01 00:00:00
@@ -260,7 +260,7 @@ _pkg5() {
 		;;
 
 	("contents")
-		_arguments -A "-*" \
+		_arguments -s -A "-*" \
 			'-H[Omit headers]' \
 			'-m[Print raw manifests]' \
 			'*-a[Attribute matching]:attribute=pattern:' \
@@ -268,6 +268,7 @@ _pkg5() {
 			'*-s[Sort key]:attribute:' \
 			'*-t[Action type]:action:_values -s , "action" $pkg5_actions' \
 			'-r[Fetch manifests from depot]:*:package:_pkg5_pkgs_a' \
+			'-g[Use alternate repo]:directory:_path_files -/' \
 			'*:package:_pkg5_pkgs'
 		;;
 
