diff -wpruN '--exclude=*.orig' a~/Functions/VCS_Info/test-repo-git-rebase-apply a/Functions/VCS_Info/test-repo-git-rebase-apply
--- a~/Functions/VCS_Info/test-repo-git-rebase-apply	1970-01-01 00:00:00
+++ a/Functions/VCS_Info/test-repo-git-rebase-apply	1970-01-01 00:00:00
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # This script creates a test repository for testing the git backend's behaviour during rebase-apply operations.
 #
diff -wpruN '--exclude=*.orig' a~/Functions/VCS_Info/test-repo-git-rebase-merge a/Functions/VCS_Info/test-repo-git-rebase-merge
--- a~/Functions/VCS_Info/test-repo-git-rebase-merge	1970-01-01 00:00:00
+++ a/Functions/VCS_Info/test-repo-git-rebase-merge	1970-01-01 00:00:00
@@ -1,4 +1,4 @@
-#!/usr/local/bin/zsh -f
+#!/usr/bin/zsh -f
 #
 # This script creates a test repository for testing the git backend's behaviour during rebase-apply operations.
 #
