--- a/src/hotspot/share/jfr/utilities/jfrBigEndian.hpp	   :: 
+++ b/src/hotspot/share/jfr/utilities/jfrBigEndian.hpp	   :: 
@@ -102,7 +102,7 @@
 inline bool JfrBigEndian::platform_supports_unaligned_reads(void) {
 #if defined(IA32) || defined(AMD64) || defined(PPC) || defined(S390)
   return true;
-#elif defined(ARM) || defined(AARCH64) || defined(RISCV)
+#elif defined(SPARC) || defined(ARM) || defined(AARCH64) || defined(RISCV)
   return false;
 #else
   #warning "Unconfigured platform"
