--- a/make/autoconf/toolchain.m4	   :: 
+++ b/make/autoconf/toolchain.m4	   :: 
@@ -57,7 +57,7 @@
 TOOLCHAIN_MINIMUM_VERSION_xlc=""
 
 # Minimum supported linker versions, empty means unspecified
-TOOLCHAIN_MINIMUM_LD_VERSION_gcc="2.18"
+TOOLCHAIN_MINIMUM_LD_VERSION_gcc=""
 
 # Prepare the system so that TOOLCHAIN_CHECK_COMPILER_VERSION can be called.
 # Must have CC_VERSION_NUMBER and CXX_VERSION_NUMBER.
