diff --git a/xinitrc.cpp b/xinitrc.cpp
index 049a8e4..e60ff12 100644
--- a/xinitrc.cpp
+++ b/xinitrc.cpp
@@ -39,6 +39,13 @@ if [ -f "$usermodmap" ]; then
     XMODMAP "$usermodmap"
 fi
 
+XCOMM
+XCOMM  If GNOME desktop is available, run it
+XCOMM
+if [ -x /etc/gdm/Xsession ] && [ -x /usr/bin/dtstart ] ; then
+  exec /etc/gdm/Xsession /usr/bin/dtstart jds
+fi
+
 XCOMM start some nice programs
 
 #if defined(__SCO__) || defined(__UNIXWARE__)
