diff --git a/man/xinit.man b/man/xinit.man
index f8005ce..9033b7e 100644
--- a/man/xinit.man
+++ b/man/xinit.man
@@ -120,7 +120,7 @@ arguments to the default \fIxterm\fP command.  It will ignore \fI\.xinitrc\fP.
 This will use the command \fI\.\/Xorg \-l \-c\fP to start the server and will
 append the arguments \fI\-e widgets\fP to the default \fIxterm\fP command.
 .TP 8
-.B "xinit /usr/ucb/rsh fasthost cpupig \-display ws:1 \-\^\-  :1 \-a 2 \-t 5"
+.B "xinit /usr/bin/rsh fasthost cpupig \-display ws:1 \-\^\-  :1 \-a 2 \-t 5"
 This will start a server named \fIX\fP on display 1 with the arguments
 \fI\-a 2 \-t 5\fP.  It will then start a remote shell on the machine
 \fBfasthost\fP in which it will run the command \fIcpupig\fP, telling it
