--- lua.pc.orig	1970-01-01 00:00:00.000000000 +0000
+++ lua.pc	2012-09-19 14:02:00.393620513 +0000
@@ -0,0 +1,15 @@
+# lua.pc -- pkg-config data for Lua
+
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: Lua
+Description: An Extensible Extension Language
+Version: 5.2.1
+Requires: 
+Libs: -L${libdir} -llua -lm
+Cflags: -I${includedir}
+
+# (end of lua.pc)
