# 7029500 Mercurial doesn't work through https in snv_161

--- mercurial-1.8.2/mercurial/keepalive.py	Thu May  3 21:06:28 2012
+++ mercurial-1.8.2/mercurial/keepalive.py	Thu May  3 21:06:28 2012
@@ -247,6 +251,8 @@
             else:
                 # no (working) free connections were found.  Create a new one.
                 h = http_class(host)
+                if req._tunnel_host:
+                        h._set_tunnel(req._tunnel_host)
                 if DEBUG:
                     DEBUG.info("creating new connection to %s (%d)",
                                host, id(h))
