# 7029500 Mercurial doesn't work through https in snv_161

--- mercurial-1.8.2/mercurial/keepalive.py.orig	Wed Jun  1 21:32:43 2011
+++ mercurial-1.8.2/mercurial/keepalive.py	Fri Apr  1 11:07:32 2011
@@ -249,6 +249,8 @@
             else:
                 # no (working) free connections were found.  Create a new one.
                 h = http_class(host)
+                if req._tunnel_host:
+                        h._set_tunnel(req._tunnel_host)
                 if DEBUG:
                     DEBUG.info("creating new connection to %s (%d)",
                                host, id(h))
