--- mercurial-1.3.1/contrib/hg-relink.orig	Thu Mar 24 21:42:58 2011
+++ mercurial-1.3.1/contrib/hg-relink	Thu Mar 24 21:44:18 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # Copyright (C) 2007 Brendan Cully <brendan@kublai.com>
 #
--- mercurial-1.3.1/contrib/hgdiff.orig	Thu Mar 24 21:43:35 2011
+++ mercurial-1.3.1/contrib/hgdiff	Thu Mar 24 21:44:23 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 import os, sys, struct, stat
 import difflib
--- mercurial-1.3.1/contrib/hg-ssh.orig	Thu Mar 24 21:47:46 2011
+++ mercurial-1.3.1/contrib/hg-ssh	Thu Mar 24 21:47:58 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # Copyright 2005-2007 by Intevation GmbH <intevation@intevation.de>
 #
