--- tklib-0.6/Makefile.in	2013-02-14 04:37:16.000000000 +0000
+++ tklib-0.6/Makefile.in.new	2018-12-14 13:18:10.018220550 +0000
@@ -54,7 +54,7 @@ install:
 		-no-examples -no-html \
 		-pkg-path   $(DESTDIR)$(pkglibdir) \
 		-app-path   $(DESTDIR)$(bindir) \
-		-nroff-path $(DESTDIR)$(mandir)/mann \
+		-nroff-path $(DESTDIR)$(mandir)/man1t \
 		-no-wait -no-gui
 
 install-libraries:
@@ -65,7 +65,7 @@ install-libraries:
 
 install-doc:
 	$(TCLSH_PROG) `$(CYGPATH) $(srcdir)/installer.tcl` \
-		-nroff-path $(DESTDIR)$(mandir)/mann \
+		-nroff-path $(DESTDIR)$(mandir)/man1t \
 		-no-apps -no-examples -no-pkgs -no-html \
 		-no-wait -no-gui
 
