--- tklib-0.5.orig/Makefile.in	2009-01-22 06:00:46.000000000 +0200
+++ tklib-0.5/Makefile.in	2016-05-05 23:01:23.908616567 +0300
@@ -52,7 +52,7 @@
 	$(TCLSH_PROG) `$(CYGPATH) $(srcdir)/installer.tcl` \
 		-no-examples -no-html \
 		-pkg-path   $(DESTDIR)$(pkglibdir) \
-		-nroff-path $(DESTDIR)$(mandir)/mann \
+		-nroff-path $(DESTDIR)$(mandir)/man1t \
 		-no-wait -no-gui
 
 install-libraries:
@@ -63,7 +63,7 @@
 
 install-doc:
 	$(TCLSH_PROG) `$(CYGPATH) $(srcdir)/installer.tcl` \
-		-nroff-path $(DESTDIR)$(mandir)/mann \
+		-nroff-path $(DESTDIR)$(mandir)/man1t \
 		-no-examples -no-pkgs -no-html \
 		-no-wait -no-gui
 
