--- tklib-0.5.orig/installer.tcl	2009-01-22 06:00:46.000000000 +0200
+++ tklib-0.5/installer.tcl	2016-05-05 23:02:34.638671617 +0300
@@ -219,7 +219,7 @@
 	    set mandir  {}
 	    set htmldir [file join $basedir tklib_doc]
 	} else {
-	    set mandir  [file join $basedir man mann]
+	    set mandir  [file join $basedir man man1t]
 	    set htmldir [file join $libdir  tklib${tklib_version} tklib_doc]
 	}
 
