--- tklib-0.6/installer.tcl	2013-03-15 02:55:52.000000000 +0000
+++ tklib-0.6/installer.tcl.new	2018-12-14 13:20:26.666459286 +0000
@@ -302,7 +302,7 @@ proc defaults {} {
 	    set mandir  {}
 	    set htmldir [file join $basedir ${package_name}_doc]
 	} else {
-	    set mandir  [file join $basedir man mann]
+	    set mandir  [file join $basedir man man1t]
 	    set htmldir [file join $libdir  ${package_name}${package_version} ${package_name}_doc]
 	}
 
