--- solaris/quagga.init.in
+++ solaris/quagga.init.in
@@ -63,6 +63,7 @@
 	case $1 in
 	zebra)
 		args="${args} `get_daemon_option_from_boolean_property $SMF_FMRI batch -b true`"
+		args="${args} `get_daemon_option_from_boolean_property $SMF_FMRI retain -r true`"
 		;;
 	ripd|ripngd)
 		args="${args} `get_daemon_option_from_boolean_property $SMF_FMRI retain -r true`"
--- solaris/quagga.xml.in
+++ solaris/quagga.xml.in
@@ -127,6 +127,7 @@
 		
 		<!-- Options specific to zebra -->
 		<propval name='batch' type='boolean' value='false' />
+		<propval name='retain' type='boolean' value='false' />
 	</property_group>
 	
 	<property_group name='general' type='framework'>
