--- solaris/quagga.xml.in
+++ solaris/quagga.xml.in
@@ -170,14 +170,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 
-	<dependency
-		name='ipv4-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv4-forwarding' />
-	</dependency>
-	
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
@@ -299,14 +291,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 	
-	<dependency
-		name='ipv6-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv6-forwarding' />
-	</dependency>
-
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
@@ -428,14 +412,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 	
-	<dependency
-		name='ipv4-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv4-forwarding' />
-	</dependency>
-
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
@@ -557,14 +533,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 	
-	<dependency
-		name='ipv6-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv6-forwarding' />
-	</dependency>
-
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
@@ -684,22 +652,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 	
-	<dependency
-		name='ipv6-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv6-forwarding' />
-	</dependency>
-
-	<dependency
-		name='ipv4-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv4-forwarding' />
-	</dependency>
-	
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
@@ -834,22 +786,6 @@
 			value='svc:/system/filesystem/usr:default' />
 	</dependency>
 
-	<dependency
-		name='ipv6-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv6-forwarding' />
-	</dependency>
-	
-	<dependency
-		name='ipv4-forwarding'
-		grouping='optional_all'
-		restart_on='refresh'
-		type='service'>
-		<service_fmri value='svc:/network/ipv4-forwarding' />
-	</dependency>
-
 	<!-- do not not run unless routing-setup has run -->
 	<dependency
 		name='network_routing_setup'
