--- solaris/quagga.init.in
+++ solaris/quagga.init.in
@@ -73,6 +73,9 @@
 		args="${args} `get_daemon_option_from_boolean_property $SMF_FMRI no_kernel -n true`"
 		args="${args} `get_daemon_option_from_property $SMF_FMRI bgp_port p 179`"
 		;;
+	ospfd)
+		args="${args} `get_daemon_option_from_boolean_property $SMF_FMRI apiserver -a true`"
+		;;
 	esac
 	echo ${args}
 }
--- solaris/quagga.xml.in
+++ solaris/quagga.xml.in
@@ -491,6 +491,9 @@
 		<!-- The pidfile to use, if not the default of
 		     @quagga_statedir@ -->
 		<propval name='pid_file' type='astring' value='' />
+
+		<!-- Options specific to ospfd -->
+		<propval name='apiserver' type='boolean' value='false' />
 	</property_group>
 
 	<property_group name='general' type='framework'>
