https://gitlab.freedesktop.org/xorg/lib/libxfont/-/issues/14

--- libXfont2-2.0.6/Makefile.am.orig
+++ libXfont2-2.0.6/Makefile.am
@@ -43,7 +43,6 @@
 	src/FreeType/ftfuncs.h       \
 	src/util/replace.h
 
-pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xfont2.pc
 
 lib_LTLIBRARIES = libXfont2.la
--- libXfont2-2.0.6/configure.ac.orig
+++ libXfont2-2.0.6/configure.ac
@@ -65,6 +65,7 @@
 # If the first PKG_CHECK_MODULES appears inside a conditional, pkg-config
 # must first be located explicitly.
 PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
 
 #
 # select libraries to include
