diff -Nru freeglut-3.4.0.orig/src/x11/fg_joystick_x11.c freeglut-3.4.0/src/x11/fg_joystick_x11.c
--- freeglut-3.4.0.orig/src/x11/fg_joystick_x11.c	2022-10-07 18:12:51.000000000 +0200
+++ freeglut-3.4.0/src/x11/fg_joystick_x11.c	2023-10-03 13:32:05.380485367 +0200
@@ -627,7 +627,7 @@
 
     do
     {
-        fghJoystickRawRead( joy, NULL, joy->center );
+        fgJoystickRawRead( joy, NULL, joy->center );
         counter++;
     } while( !joy->error &&
              counter < 100 &&
