--- vala-0.56.18/build-aux/test-driver.orig
+++ vala-0.56.18/build-aux/test-driver
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 # test-driver - basic testsuite driver script.
 
 scriptversion=2018-03-07.03; # UTC
--- vala-0.56.18/build-aux/testrunner.sh.orig
+++ vala-0.56.18/build-aux/testrunner.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 # testrunner.sh
 #
 # Copyright (C) 2006-2008  Jürg Billeter
@@ -22,6 +22,10 @@
 
 set -e
 
+# script requires GNUish behavior from some utils, and expects 'echo -n'
+# to suppress newlines
+PATH=/usr/gnu/bin:$PATH; export PATH
+
 export G_DEBUG=fatal-warnings
 
 EXTRA_ENVIRONMENT_FILE=tests-extra-environment.sh
