diff -ur vala-0.54.3.orig/build-aux/test-driver vala-0.54.3/build-aux/test-driver
--- vala-0.54.3.orig/build-aux/test-driver	2021-10-30 03:12:08.000000000 +0000
+++ vala-0.54.3/build-aux/test-driver	2021-11-24 14:30:06.819527273 +0000
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 # test-driver - basic testsuite driver script.
 
 scriptversion=2018-03-07.03; # UTC
diff -ur vala-0.54.3.orig/build-aux/testrunner.sh vala-0.54.3/build-aux/testrunner.sh
--- vala-0.54.3.orig/build-aux/testrunner.sh	2021-10-26 07:38:24.000000000 +0000
+++ vala-0.54.3/build-aux/testrunner.sh	2021-11-24 14:34:23.701349983 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 # testrunner.sh
 #
 # Copyright (C) 2006-2008  Jürg Billeter
@@ -22,6 +22,10 @@
 
 set -e
 
+# script requires GNUish behavior from some utils, and expects 'echo -n'
+# to suppress newlines
+PATH=/usr/gnu/bin:$PATH; export PATH
+
 export G_DEBUG=fatal-warnings
 
 EXTRA_ENVIRONMENT_FILE=tests-extra-environment.sh
