--- vala-0.40.14/valadoc/tests/testrunner.sh.~1~	Sun Mar  3 13:49:21 2019
+++ vala-0.40.14/valadoc/tests/testrunner.sh	Tue Mar  5 08:23:03 2019
@@ -48,15 +48,11 @@
 	-X -Werror=implicit \
 	-X -Werror=sequence-point \
 	-X -Werror=return-type \
-	-X -Werror=uninitialized \
 	-X -Werror=pointer-arith \
-	-X -Werror=int-to-pointer-cast \
-	-X -Werror=pointer-to-int-cast \
 	-X -Wformat \
 	-X -Werror=format-security \
 	-X -Werror=format-nonliteral \
 	-X -Werror=redundant-decls \
-	-X -Werror=int-conversion \
 	-X -L$topbuilddir/vala/.libs -X -lvala$PACKAGE_SUFFIX \
 	-X -L$topbuilddir/libvaladoc/.libs -X -lvaladoc$PACKAGE_SUFFIX \
 	-X -I$topsrcdir/gee \
