--- vala-0.40.9/valadoc/tests/testrunner.sh.~1~	Tue Aug 14 03:25:19 2018
+++ vala-0.40.9/valadoc/tests/testrunner.sh	Wed Aug 29 17:39:16 2018
@@ -47,15 +47,11 @@
 	-X -Werror=implicit \
 	-X -Werror=sequence-point \
 	-X -Werror=return-type \
-	-X -Werror=uninitialized \
 	-X -Werror=pointer-arith \
-	-X -Werror=int-to-pointer-cast \
-	-X -Werror=pointer-to-int-cast \
 	-X -Wformat \
 	-X -Werror=format-security \
 	-X -Werror=format-nonliteral \
 	-X -Werror=redundant-decls \
-	-X -Werror=int-conversion \
 	-X -L$topbuilddir/vala/.libs -X -lvala$PACKAGE_SUFFIX \
 	-X -L$topbuilddir/libvaladoc/.libs -X -lvaladoc$PACKAGE_SUFFIX \
 	-X -I$topsrcdir/gee \
