--- vala-0.40.9/tests/testrunner.sh.~1~	Tue Aug 14 03:25:19 2018
+++ vala-0.40.9/tests/testrunner.sh	Wed Aug 29 13:18:52 2018
@@ -43,15 +43,11 @@
 	-X -Werror=implicit \
 	-X -Werror=sequence-point \
 	-X -Werror=return-type \
-	-X -Werror=uninitialized \
 	-X -Werror=pointer-arith \
-	-X -Werror=int-to-pointer-cast \
-	-X -Werror=pointer-to-int-cast \
 	-X -Wformat \
 	-X -Werror=format-security \
 	-X -Werror=format-nonliteral \
-	-X -Werror=redundant-decls \
-	-X -Werror=int-conversion"
+	-X -Werror=redundant-decls"
 VAPIGEN=$topbuilddir/vapigen/vapigen$EXEEXT
 VAPIGENFLAGS="--vapidir $vapidir"
 
