--- netifaces-0.10.9/setup.py.orig	2020-03-16 21:08:12.054401184 +0000
+++ netifaces-0.10.9/setup.py	2020-03-16 21:08:39.614755253 +0000
@@ -148,7 +148,7 @@
               return 0;
             }
             """
-            if self.test_build(testrig):
+            if self.test_build(testrig,libraries=libraries):
                 result = True
             else:
                 result = False
