--- Python-3.12.1/configure.ac.~6~	Mon Feb  5 10:07:19 2024
+++ Python-3.12.1/configure.ac	Mon Feb  5 10:21:31 2024
@@ -6509,7 +6509,7 @@
 # first curses header check
 ac_save_cppflags="$CPPFLAGS"
 if test "$cross_compiling" = no; then
-  CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+  CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw -I/usr/include/ncurses"
 fi
 
 # On Solaris, term.h requires curses.h
--- Python-3.12.1/configure.ac.~7~	Mon Feb  5 17:08:56 2024
+++ Python-3.12.1/configure.ac	Mon Feb  5 17:19:15 2024
@@ -6470,6 +6470,7 @@
         have_panel=panelw
       ], [
         WITH_SAVE_ENV([
+	  AS_VAR_APPEND([LDFLAGS], [" -L/usr/gnu/lib/amd64"])
           AC_CHECK_LIB([panelw], [update_panels], [
             have_panel=panelw
             PANEL_CFLAGS=${PANEL_CFLAGS-""}
