--- mozilla/modules/libpref/src/init/all.js.~1~	2014-01-15 20:45:22.317881093 +0400
+++ mozilla/modules/libpref/src/init/all.js	2014-01-15 20:46:09.746852217 +0400
@@ -1399,7 +1399,7 @@
 pref("intl.charset.detector",               "chrome://global/locale/intl.properties");
 pref("intl.charset.default",                "chrome://global-platform/locale/intl.properties");
 pref("intl.ellipsis",                       "chrome://global-platform/locale/intl.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
