diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 5ef66c9..7dcd0f5 100644
--- mozilla/browser/app/profile/firefox.js
+++ mozilla/browser/app/profile/firefox.js
@@ -61,7 +61,7 @@ pref("extensions.hotfix.certs.1.sha1Fingerprint", "F1:DB:F9:6A:7B:B8:04:FA:48:3C
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 11);
 
 // Dictionary download preference
 pref("browser.dictionaries.download.url", "https://addons.mozilla.org/%LOCALE%/firefox/dictionaries/");
