--- mozilla/media/libcubeb/tests/test_audio.cpp	2015-04-29 12:37:04.490292415 +0300
+++ mozilla/media/libcubeb/tests/test_audio.cpp.~1~	2015-04-29 12:39:00.685448422 +0300
@@ -10,7 +10,7 @@
 #ifdef NDEBUG
 #undef NDEBUG
 #endif
-#define _XOPEN_SOURCE 500
+//#define _XOPEN_SOURCE 500
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
--- mozilla/media/libcubeb/tests/test_sanity.cpp	2015-04-29 12:38:42.325334438 +0300
+++ mozilla/media/libcubeb/tests/test_sanity.cpp.~1~	2015-04-29 12:38:29.285521666 +0300
@@ -7,7 +7,7 @@
 #ifdef NDEBUG
 #undef NDEBUG
 #endif
-#define _XOPEN_SOURCE 500
+//#define _XOPEN_SOURCE 500
 #include "cubeb/cubeb.h"
 #include <assert.h>
 #include <stdio.h>
--- mozilla/media/libcubeb/tests/test_tone.cpp	2015-04-29 10:18:42.013150946 +0300
+++ mozilla/media/libcubeb/tests/test_tone.cpp.~1~	2015-03-30 19:48:12.000000000 +0300
@@ -9,7 +9,7 @@
 #ifdef NDEBUG
 #undef NDEBUG
 #endif
-#define _XOPEN_SOURCE 500
+//#define _XOPEN_SOURCE 500
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
