--- mozilla/build/autoconf/compiler-opts.m4.~1~	2015-04-21 13:18:47.524663000 +0300
+++ mozilla/build/autoconf/compiler-opts.m4	2015-04-21 13:19:19.914182152 +0300
@@ -256,8 +256,8 @@
 
 if test "$GNU_CC"; then
     if test -z "$DEVELOPER_OPTIONS"; then
-        CFLAGS="$CFLAGS -ffunction-sections -fdata-sections"
-        CXXFLAGS="$CXXFLAGS -ffunction-sections -fdata-sections"
+        CFLAGS="$CFLAGS"
+        CXXFLAGS="$CXXFLAGS"
     fi
     CFLAGS="$CFLAGS -fno-math-errno"
     CXXFLAGS="$CXXFLAGS -fno-exceptions -fno-math-errno"
@@ -280,7 +280,7 @@
               'int main() {return foo() - bar();}' > conftest.${ac_ext}
         # If the linker supports ICF, foo and bar symbols will have
         # the same address
-        if AC_TRY_COMMAND([${CC-cc} -o conftest${ac_exeext} $LDFLAGS -Wl,--icf=safe -ffunction-sections conftest.${ac_ext} $LIBS 1>&2]) &&
+        if AC_TRY_COMMAND([${CC-cc} -o conftest${ac_exeext} $LDFLAGS -Wl,--icf=safe conftest.${ac_ext} $LIBS 1>&2]) &&
            test -s conftest${ac_exeext} &&
            objdump -t conftest${ac_exeext} | awk changequote(<<, >>)'{a[<<$>>6] = <<$>>1} END {if (a["foo"] && (a["foo"] != a["bar"])) { exit 1 }}'changequote([, ]); then
             LD_SUPPORTS_ICF=yes
