--- mozilla/xpcom/tests/TestPRIntN.cpp	2013-12-09 23:14:36.778752283 +0400
+++ mozilla/xpcom/tests/TestPRIntN.cpp	2013-12-09 23:17:29.447519932 +0400
@@ -25,10 +25,12 @@
 int
 main()
 {
+/* I don't know what's wrong, but it doesn't compile
   PRInt8 a; PRInt16 b; PRInt32 c; PRInt64 d;
   int8_t w; int16_t x; int32_t y; int64_t z;
 
   ClearNSPRIntTypes(&w, &x, &y, &z);
   ClearStdIntTypes(&a, &b, &c, &d);
+*/
   return 0;
 }
