--- mozilla/mfbt/lz4.c.~1~	2015-04-17 16:02:10.294387593 +0300
+++ mozilla/mfbt/lz4.c	2015-04-17 16:06:19.666820980 +0300
@@ -104,7 +104,7 @@
 /**************************************
  Compiler Options
 **************************************/
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)   /* C99 */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) && (!defined __cplusplus)   /* C99 */
 /* "restrict" is a known keyword */
 #else
 #  define restrict /* Disable restrict */
