--- mozilla/configure.in.~1~	2015-04-21 16:42:17.840858486 +0300
+++ mozilla/configure.in	2015-04-21 16:42:50.020325277 +0300
@@ -4129,7 +4129,7 @@
     _subconfigure_config_args="$ac_configure_args"
   }
   tmpscript=`$PYTHON -c 'import os, tempfile; print tempfile.mktemp(prefix="subscript.").replace(os.sep, "/")'` || exit 1
-  m4 "${srcdir}/build/autoconf/subconfigure.m4" \
+  /usr/gnu/bin/m4 "${srcdir}/build/autoconf/subconfigure.m4" \
      "${srcdir}/build/autoconf/altoptions.m4" \
      "${srcdir}/${MOZ_BUILD_APP}/configure.in" > $tmpscript
   . $tmpscript
