--- mozilla/js/src/frontend/BytecodeCompiler.cpp.~1~	2015-05-27 13:03:31.977594702 +0300
+++ mozilla/js/src/frontend/BytecodeCompiler.cpp	2015-05-27 13:29:36.558245287 +0300
@@ -544,7 +544,7 @@
 
     RootedScriptSource sourceObject(cx, CreateScriptSourceObject(cx, options));
     if (!sourceObject)
-        return nullptr;
+        return false;
     ScriptSource* ss = sourceObject->source();
 
     SourceCompressionTask sct(cx);
