--- im/configure.in.~1~	2015-04-21 13:33:07.082107480 +0300
+++ im/configure.in	2015-04-21 13:33:16.263179967 +0300
@@ -9,7 +9,7 @@
 
 if test -d $topsrcdir/extensions/purple ; then
   tmpscript=`$PYTHON -c 'import os, tempfile; print tempfile.mktemp(prefix="subscript.").replace(os.sep, "/")'` || exit 1
-  m4 "${topsrcdir}/build/autoconf/subconfigure.m4" \
+  /usr/gnu/bin/m4 "${topsrcdir}/build/autoconf/subconfigure.m4" \
    "${topsrcdir}/build/autoconf/altoptions.m4" \
    "${topsrcdir}/extensions/purple/configure.in" > $tmpscript
   . $tmpscript
