--- mozilla/media/libsoundtouch/src/SoundTouch.cpp.~1~	2015-04-17 17:38:03.999624587 +0300
+++ mozilla/media/libsoundtouch/src/SoundTouch.cpp	2015-04-17 17:38:21.556894401 +0300
@@ -71,6 +71,7 @@
 
 #include <assert.h>
 #include <stdlib.h>
+#include <alloca.h>
 #include <memory.h>
 #include <math.h>
 #include <stdio.h>
--- mozilla/media/libsoundtouch/src/FIRFilter.cpp.~1~	2015-04-17 17:30:05.478593256 +0300
+++ mozilla/media/libsoundtouch/src/FIRFilter.cpp	2015-04-17 17:34:43.624166703 +0300
@@ -43,6 +43,7 @@
 #include <assert.h>
 #include <math.h>
 #include <stdlib.h>
+#include <alloca.h>
 #include "FIRFilter.h"
 #include "cpu_detect.h"
 
