--- mailnews/mailnews.js.orig	2013-02-20 16:55:34.540100762 +0530
+++ mailnews/mailnews.js	2013-02-20 16:56:57.583240041 +0530
@@ -447,9 +447,9 @@ pref("mail.server.default.inhibitWhiteLi
 // should we inhibit whitelisting of the domain for a server's identities?
 pref("mail.server.default.inhibitWhiteListingIdentityDomain", false);
 
-// to activate auto-sync feature (preemptive message download for imap) by default
-pref("mail.server.default.autosync_offline_stores",true);
-pref("mail.server.default.offline_download",true);
+// disable auto-sync feature (preemptive message download for imap) by default
+pref("mail.server.default.autosync_offline_stores",false);
+pref("mail.server.default.offline_download",false);
 
 // -1 means no limit, no purging of offline stores.
 pref("mail.server.default.autosync_max_age_days", -1);
