diff -r 727595a78622 mail/app/profile/all-thunderbird.js
--- mail/app/profile/all-thunderbird.js	Fri Nov 04 22:12:51 2011 +0000
+++ mail/app/profile/all-thunderbird.js	Wed Nov 09 16:30:30 2011 +0800
@@ -138,7 +138,7 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 11);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
