--- gnome-media-2.32.0/gst-mixer/src/main.c.~1~	2015-03-30 00:16:00.370917666 +0300
+++ gnome-media-2.32.0/gst-mixer/src/main.c	2015-03-30 00:16:17.087528201 +0300
@@ -36,7 +36,8 @@
 static gchar* page = NULL;
 static GOptionEntry entries[] =
 {
-  { "page", 'p', 0, G_OPTION_ARG_STRING, &page, N_("Startup page"), "playback|recording|switches|options" }
+  { "page", 'p', 0, G_OPTION_ARG_STRING, &page, N_("Startup page"), "playback|recording|switches|options" },
+  { NULL, 0, 0, 0, NULL, NULL, NULL }
 };
 
 /*
