--- gnome-media-2.30.0/gnome-volume-control/src/Makefile.am-orig	2012-03-15 14:18:52.374146796 -0500
+++ gnome-media-2.30.0/gnome-volume-control/src/Makefile.am	2012-03-15 14:19:00.870250231 -0500
@@ -2,7 +2,6 @@ NULL =
 
 bin_PROGRAMS = 					\
 	gnome-volume-control-applet		\
-	gnome-volume-control			\
 	$(NULL)
 
 AM_CPPFLAGS =					\
--- gnome-media-2.30.0/gnome-volume-control/data/Makefile.am-orig	2012-03-15 14:56:22.893512844 -0500
+++ gnome-media-2.30.0/gnome-volume-control/data/Makefile.am	2012-03-15 14:56:28.302731750 -0500
@@ -1,7 +1,6 @@
 NULL =
 
 SUBDIRS =			\
-	icons			\
 	$(NULL)
 
 @INTLTOOL_DESKTOP_RULE@
