--- gnome-media-2.27.91/gst-mixer/gnome-volume-control.desktop.in.in-orig	2011-05-19 18:24:29.785648093 -0500
+++ gnome-media-2.27.91/gst-mixer/gnome-volume-control.desktop.in.in	2011-05-19 18:24:53.357246842 -0500
@@ -1,13 +1,16 @@
 [Desktop Entry]
-_Name=Volume Control
-_Comment=Change sound volume and sound events
+# SUN_BRANDING
+_Name=Sound
+# SUN_BRANDING
+_Comment=Adjust volume levels and speaker balance
 Exec=gnome-volume-control
 Icon=multimedia-volume-control
 StartupNotify=true
 X-Info=Gnome Mixer
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;AudioVideo;Settings;HardwareSettings;
+# SUN_BRANDING
+Categories=GNOME;GTK;Settings;
 X-GNOME-DocPath=gst-mixer/gnome-volume-control.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-media
