--- rich-13.9.1/tox.ini.orig
+++ rich-13.9.1/tox.ini
@@ -8,8 +8,6 @@
 
 [testenv]
 description = Run unit-testing
-deps =
-    poetry
 # do not put * in passenv as it may break builds due to reduced isolation
 passenv =
     CI
@@ -22,7 +20,6 @@
     PYTHONDONTWRITEBYTECODE=1
     PYTHONUNBUFFERED=1
 commands =
-    poetry install
     pytest --cov-report term-missing --cov=rich tests/ {posargs}
 
 [testenv:lint]
