*** gcc-3.4.3/gcc/cp/g++spec.c.orig	Wed Nov 17 09:59:39 2004
--- gcc-3.4.3/gcc/cp/g++spec.c	Mon Jan 28 13:40:34 2008
***************
*** 310,318 ****
    /* Add `-lstdc++' if we haven't already done so.  */
    if (library > 0)
      {
-       arglist[j++] = saw_m64_flag
- 	? "-R" LIBDIR "/amd64"
- 	: "-R" LIBDIR;
        arglist[j++] = saw_profile_flag ? LIBSTDCXX_PROFILE : LIBSTDCXX;
        added_libraries++;
      }
--- 310,315 ----
