--- dasher-4.11/Data/dasher.desktop.in.in.orig	2008-05-28 16:59:51.166363292 +0100
+++ dasher-4.11/Data/dasher.desktop.in.in	2008-05-28 17:01:00.173213483 +0100
@@ -1,13 +1,16 @@
 [Desktop Entry]
-_Name=Dasher
-_GenericName=Predictive text entry
-_Comment=Enter text without a keyboard
+# SUN_BRANDING
+_Name=Predictive Text Entry
+# SUN_BRANDING
+_GenericName=Predictive Text Entry 
+# SUN_BRANDING
+_Comment=Navigate applications and type using alternative input devices
 Exec=dasher
 Icon=dasher
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Utility;Accessibility;
+Categories=GNOME;GTK;Application;Accessibility;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=dasher
 X-GNOME-Bugzilla-Component=general
