--- dasher-4.11/Src/DasherCore/LanguageModelling/CTWLanguageModel.cpp-orig	2010-03-15 16:43:24.533121441 -0500
+++ dasher-4.11/Src/DasherCore/LanguageModelling/CTWLanguageModel.cpp	2010-03-15 18:56:05.517829627 -0500
@@ -358,7 +358,7 @@ void CCTWLanguageModel::GetProbs(Context
 	// calculate probabilities of all possible symbols. Again assume all 2^NrPhases
 	int *Index = new int[LocalContext.Context.size()+1]; // +1 for the rootnode
 
-	vector <unsigned short int>Interval((1<<(NrPhases+1))-1); // number of rootnodes*2 (1 prob for bit 0 and 1 each)
+	vector <unsigned int>Interval((1<<(NrPhases+1))-1); // number of rootnodes*2 (1 prob for bit 0 and 1 each)
 	if (Norm>65535)
 	{
 		Interval[0]=65535; // to prevent overflow
