--- dasher-4.11/Src/DasherCore/DasherModel.h-orig	2010-03-15 19:04:18.619960775 -0500
+++ dasher-4.11/Src/DasherCore/DasherModel.h	2010-03-15 19:03:58.416196316 -0500
@@ -56,7 +56,7 @@ namespace Dasher {
 ///             knows the current viewpoint
 ///             knows how to evolve the viewpoint
 ///
-class Dasher::CDasherModel:public CFrameRate, private NoClones
+class Dasher::CDasherModel:public Dasher::CFrameRate, private NoClones
 {
  public:
 
