diff -Nru pidgin-2.10.12.orig/libpurple/purple-notifications-example pidgin-2.10.12/libpurple/purple-notifications-example
--- pidgin-2.10.12.orig/libpurple/purple-notifications-example	2016-01-01 02:19:41.000000000 +0300
+++ pidgin-2.10.12/libpurple/purple-notifications-example	2016-05-14 11:43:11.196955183 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 
 # This is a simple purple notification server.
 # It shows notifications when your buddy signs on or you get an IM message.
diff -Nru pidgin-2.10.12.orig/libpurple/purple-remote pidgin-2.10.12/libpurple/purple-remote
--- pidgin-2.10.12.orig/libpurple/purple-remote	2016-01-01 02:19:41.000000000 +0300
+++ pidgin-2.10.12/libpurple/purple-remote	2016-05-14 11:42:49.004649836 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 
 import codecs
 import dbus
diff -Nru pidgin-2.10.12.orig/libpurple/purple-url-handler pidgin-2.10.12/libpurple/purple-url-handler
--- pidgin-2.10.12.orig/libpurple/purple-url-handler	2016-01-01 02:19:41.000000000 +0300
+++ pidgin-2.10.12/libpurple/purple-url-handler	2016-05-14 11:41:52.549139167 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 
 from __future__ import absolute_import, division, print_function
 
