--- pidgin-ori/configure.ac	2008-05-30 12:07:21.986527000 +0800
+++ pidgin-new/configure.ac	2008-05-30 11:56:25.655174000 +0800
@@ -1850,7 +1850,7 @@
 
 			if test "x$ac_cv_moz_nspr_libs" != "xno"; then
 				have_nspr_libs="yes"
-				NSPR_LIBS="-L$with_nspr_libs $nsprlibs"
+				NSPR_LIBS="-L$with_nspr_libs $nsprlibs -R$with_nspr_libs"
 			else
 				NSPR_CFLAGS=""
 				enable_nss="no"
@@ -1948,7 +1948,7 @@
 				AC_DEFINE(HAVE_NSS)
 				AC_DEFINE(HAVE_SSL)
 
-				NSS_LIBS="-L$with_nss_libs $nsslibs"
+				NSS_LIBS="-L$with_nss_libs $nsslibs -R$with_nss_libs"
 
 				if test "$enable_nss" = "static"; then
 					msg_nss="Mozilla NSS (static)"
