--- pidgin-2.10.12/finch/gntsound.c.~1~	2016-05-14 21:18:30.004305362 +0300
+++ pidgin-2.10.12/finch/gntsound.c	2016-05-14 21:19:45.388996242 +0300
@@ -448,7 +448,6 @@
 		/* fall-through and clean up */
 	case GST_MESSAGE_EOS:
 		gst_element_set_state(play, GST_STATE_NULL);
-		gst_object_unref(GST_OBJECT(play));
 		return FALSE;
 		break;
 	case GST_MESSAGE_WARNING:
--- pidgin-2.10.12/pidgin/gtksound.c.~1~	2016-05-14 21:18:30.010407898 +0300
+++ pidgin-2.10.12/pidgin/gtksound.c	2016-05-14 21:20:33.972700327 +0300
@@ -393,7 +393,6 @@
 		/* fall-through and clean up */
 	case GST_MESSAGE_EOS:
 		gst_element_set_state(play, GST_STATE_NULL);
-		gst_object_unref(GST_OBJECT(play));
 		return FALSE;
 		break;
 	case GST_MESSAGE_WARNING:
