--- pidgin-2.6.0/pidgin/gtkmain.c.orig2	2009-08-19 07:36:41.935460000 +0800
+++ pidgin-2.6.0/pidgin/gtkmain.c	2009-08-19 07:37:42.090824000 +0800
@@ -731,6 +731,10 @@
 	gtk_rc_add_default_file(search_path);
 	g_free(search_path);
 
+#if GLIB_CHECK_VERSION(2,2,0)
+       g_set_application_name(_("Pidgin"));
+#endif /* glib-2.0 >= 2.2.0 */
+
 #ifdef  GAIM_ENABLE_KEYRING
         GnomeKeyringResult rtn = gnome_keyring_unlock_sync(NULL, NULL);
         // if (rtn == GNOME_KEYRING_RESULT_DENIED)
