--- pidgin-2.14.10/pidgin/data/pidgin.desktop.in.orig	2022-06-03 04:07:55.000000000 +0000
+++ pidgin-2.14.10/pidgin/data/pidgin.desktop.in	2022-06-25 14:14:29.535039293 +0000
@@ -2,7 +2,7 @@
 _Name=Pidgin
 _GenericName=Internet Messenger
 _X-GNOME-FullName=Pidgin Internet Messenger
-_Comment=Chat over IM.  Supports XMPP, IRC, and more
+_Comment=Send instant messages to contacts on Jabber, Google Talk, ICQ and other networks
 # TRANSLATORS: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 _Keywords=chat;talk;im;message;bonjour;gadu-gadu;irc;groupwise;jabber;sametime;silc;simple;xmpp;zephyr
 Exec=pidgin
