--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/osc/osc/OscOutboundPacketStream.cpp.orig	2022-08-10 16:52:57.488229245 +0000
+++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/osc/osc/OscOutboundPacketStream.cpp	2022-08-10 16:57:14.472287562 +0000
@@ -37,6 +37,10 @@
 #include <malloc.h> // for alloca
 #endif
 
+#if defined(__sun) && (__SVR4)
+#include <alloca.h> // for alloca
+#endif
+
 #include "OscHostEndianness.h"
 
 
