https://github.com/pypa/twine/issues/1116

--- twine-5.1.1/tests/test_package.py.orig
+++ twine-5.1.1/tests/test_package.py
@@ -384,11 +384,6 @@
     "read_data, missing_fields",
     [
         pytest.param(
-            b"Metadata-Version: 102.3\nName: test-package\nVersion: 1.0.0\n",
-            "Name, Version",
-            id="unsupported Metadata-Version",
-        ),
-        pytest.param(
             b"Metadata-Version: 2.3\nName: UNKNOWN\nVersion: UNKNOWN\n",
             "Name, Version",
             id="missing Name and Version",
