sshd_config customizations for OpenIndiana.

--- hpn-ssh-hpn-18.3.1/sshd_config.orig
+++ hpn-ssh-hpn-18.3.1/sshd_config
@@ -34,6 +34,7 @@
 
 #LoginGraceTime 2m
 #PermitRootLogin prohibit-password
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
@@ -78,10 +79,15 @@
 #AllowTcpForwarding yes
 #GatewayPorts no
 #X11Forwarding no
+X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
+
 #PrintMotd yes
+# On OpenIndiana it is assumed that the login shell will print /etc/motd
+PrintMotd no
+
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
