diff -Nru IcedTea-Web-icedtea-web-1.8.8.orig/configure.ac IcedTea-Web-icedtea-web-1.8.8/configure.ac
--- IcedTea-Web-icedtea-web-1.8.8.orig/configure.ac	2021-10-27 22:13:18.000000000 +0200
+++ IcedTea-Web-icedtea-web-1.8.8/configure.ac	2023-06-14 12:22:07.249396534 +0200
@@ -72,7 +72,7 @@
 build_windows=no
 build_macos=no
 case "${host_os}" in
-    linux*|freebsd*)
+    linux*|freebsd*|solaris*)
         build_linux=yes
         ;;
     cygwin*)
