--- net-snmp-5.4.4/snmplib/scapi.c.~1~	2011-01-07 16:24:31.000000000 +0300
+++ net-snmp-5.4.4/snmplib/scapi.c	2015-08-17 20:50:11.401118436 +0300
@@ -407,8 +407,8 @@
 {
 #if defined(NETSNMP_USE_OPENSSL) || defined(NETSNMP_USE_PKCS11)
     int            rval = SNMPERR_SUCCESS;
-    unsigned int   tmp_len;
 #endif
+    unsigned int   tmp_len;
     int            ret;
 
 #ifdef NETSNMP_USE_OPENSSL
@@ -482,13 +482,11 @@
 
 #ifndef NETSNMP_DISABLE_MD5
     if (ISTRANSFORM(hashtype, HMACMD5Auth)) {
-	rval = pkcs_digest(CKM_MD5, buf, buf_len, MAC, &tmp_len);
-        *MAC_len = tmp_len;
+	rval = pkcs_digest(CKM_MD5, buf, buf_len, MAC, MAC_len);
     } else
 #endif
         if (ISTRANSFORM(hashtype, HMACSHA1Auth)) {
-	rval = pkcs_digest(CKM_SHA_1, buf, buf_len, MAC, &tmp_len);
-        *MAC_len = tmp_len;
+	rval = pkcs_digest(CKM_SHA_1, buf, buf_len, MAC, MAC_len);
     } else {
         return (SNMPERR_GENERR);
     }
