Index: net-snmp/snmplib/asn1.c
===================================================================
--- net-snmp/snmplib/asn1.c	(revision 16862)
+++ net-snmp/snmplib/asn1.c	(revision 16863)
@@ -2735,7 +2735,7 @@
     }
 
     CHECK_OVERFLOW_S(integer,10);
-    testvalue = (*intp < 0) ? -1 : 0;
+    testvalue = (integer < 0) ? -1 : 0;
 
     if (((*pkt_len - *offset) < 1) && !(r && asn_realloc(pkt, pkt_len))) {
         return 0;
