--- patch/net-snmp-5.4.1/agent/mibgroup/kernel_sunos5.c	Tue Oct 12 06:55:48 2010
+++ net-snmp-5.4.1/agent/mibgroup/kernel_sunos5.c	Tue Oct 12 07:03:49 2010
@@ -82,7 +82,7 @@
 static
 mibcache        Mibcache[MIBCACHE_SIZE+1] = {
     {MIB_SYSTEM, 0, (void *) -1, 0, 0, 0, 0},
-    {MIB_INTERFACES, 10 * sizeof(mib2_ifEntry_t), (void *) -1, 0, 30, 0,
+    {MIB_INTERFACES, 50 * sizeof(mib2_ifEntry_t), (void *) -1, 0, 30, 0,
      0},
     {MIB_AT, 0, (void *) -1, 0, 0, 0, 0},
     {MIB_IP, sizeof(mib2_ip_t), (void *) -1, 0, 60, 0, 0},
