--- net-snmp-5.4.1/Makefile.in.orig	2012-03-21 09:36:23.148069679 -0700
+++ net-snmp-5.4.1/Makefile.in	2012-03-21 09:36:56.194195238 -0700
@@ -207,7 +207,7 @@
         fi
 
 pythoninstall:
-	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir) ; \
+	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir --root $(DESTDIR) --install-lib=$(PYTHON_VENDOR_PACKAGES)) ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
