Fix for https://www.illumos.org/issues/2245

--- ./include/net-snmp/net-snmp-config.h.in.bak	2012-03-04 08:33:01.874796059 +0000
+++ ./include/net-snmp/net-snmp-config.h.in	2012-03-04 08:50:14.401107234 +0000
@@ -5,8 +5,9 @@
 /* Portions of this file are subject to the following copyright(s).  See
  * the Net-SNMP's COPYING file for more details and other copyrights
  * that may apply:
- */
-/*
+ *
+ * Copyright (c) 2012 Nexenta Systems, Inc.  All rights reserved.
+ *
  * Portions of this file are copyrighted by:
  * Copyright © 2003 Sun Microsystems, Inc. All rights reserved.
  * Use is subject to license terms specified in the COPYING file
@@ -66,9 +67,6 @@
 /* define if you have <sys/disklabel.h> */
 #undef HAVE_SYS_DISKLABEL_H
 
-/* define if your compiler (processor) defines __FUNCTION__ for you */
-#undef HAVE_CPP_UNDERBAR_FUNCTION_DEFINED
-
 /* on aix, if you have perfstat */
 #undef HAVE_PERFSTAT
 
--- ./include/net-snmp/library/snmp_assert.h.bak	2012-03-04 08:38:03.367318968 +0000
+++ ./include/net-snmp/library/snmp_assert.h	2012-03-04 08:46:46.097465403 +0000
@@ -34,7 +34,7 @@
  *  if asserts weren't requested, just log, unless NETSNMP_NO_DEBUGGING specified
  */
 #   ifndef NETSNMP_NO_DEBUGGING
-#      ifdef  HAVE_CPP_UNDERBAR_FUNCTION_DEFINED
+#      ifdef __FUNCTION__
 #         define netsnmp_assert(x)  do { \
                  if ( x ) \
                     ; \
--- ./include/net-snmp/library/snmp_debug.h.bak	2012-03-04 08:42:55.614024798 +0000
+++ ./include/net-snmp/library/snmp_debug.h	2012-03-04 08:46:13.774100176 +0000
@@ -121,7 +121,7 @@
 #define __DBGMSGL_NC(x)  __DBGTRACE; debugmsg x
 #define __DBGMSGTL_NC(x) __DBGTRACE; debug_combo_nc x
 
-#ifdef  HAVE_CPP_UNDERBAR_FUNCTION_DEFINED
+#ifdef  __FUNCTION__
 #define __DBGTRACE       __DBGMSGT(("trace","%s(): %s, %d:\n",__FUNCTION__,\
                                  __FILE__,__LINE__))
 #else
--- ./win32/net-snmp/net-snmp-config.h.in.bak	2012-03-04 08:34:19.879485815 +0000
+++ ./win32/net-snmp/net-snmp-config.h.in	2012-03-04 08:37:30.951096411 +0000
@@ -1105,9 +1105,6 @@
 /* define if SIOCGIFADDR exists in sys/ioctl.h */
 /* #undef SYS_IOCTL_H_HAS_SIOCGIFADDR */
 
-/* define if your compiler (processor) defines __FUNCTION__ for you */
-/* #undef HAVE_CPP_UNDERBAR_FUNCTION_DEFINED */
-
 /* Mib-2 tree Info */
 /* These are the system information variables. */
 
--- ./win32/net-snmp/net-snmp-config.h.bak	2012-03-04 08:33:47.741984111 +0000
+++ ./win32/net-snmp/net-snmp-config.h	2012-03-04 08:34:01.718179986 +0000
@@ -1105,9 +1105,6 @@
 /* define if SIOCGIFADDR exists in sys/ioctl.h */
 /* #undef SYS_IOCTL_H_HAS_SIOCGIFADDR */
 
-/* define if your compiler (processor) defines __FUNCTION__ for you */
-/* #undef HAVE_CPP_UNDERBAR_FUNCTION_DEFINED */
-
 /* Mib-2 tree Info */
 /* These are the system information variables. */
 
--- ./win32/config.h.borland.bak	2012-03-04 08:32:01.298279084 +0000
+++ ./win32/config.h.borland	2012-03-04 08:32:33.881252737 +0000
@@ -543,9 +543,6 @@
 /* type check for in_addr_t */
 #define in_addr_t u_long
 
-/* define if your compiler (processor) defines __FUNCTION__ for you */
-#undef HAVE_CPP_UNDERBAR_FUNCTION_DEFINED
-
 /* mib pointer to the top of the extensible tree.  This has been
  assigned to UCDavis by the iana group.  Optionally, point this to the
  location in the tree your company/organization has been allocated. */
--- ./configure.in.bak	2012-03-04 08:30:33.615179399 +0000
+++ ./configure.in	2012-03-04 08:52:52.022427962 +0000
@@ -2,6 +2,8 @@
 # the Net-SNMP's COPYING file for more details and other copyrights
 # that may apply:
 #
+# Copyright (c) 2012 Nexenta Systems, Inc.  All rights reserved.
+#
 # Portions Copyright © 2003 Sun Microsystems, Inc. All rights reserved.
 # Use is subject to license terms specified in the COPYING file
 # distributed with the Net-SNMP package.
@@ -3717,17 +3719,6 @@
   AC_DEFINE(STAT_STATFS_FS_DATA)
 fi
 
-# check if compiler pre-processor defines __FUNCTION__
-AC_CACHE_CHECK(if __FUNCTION__ is defined,ac_cv_HAVE_CPP_UNDERBAR_FUNCTION_DEFINED,
-[
-AC_TRY_COMPILE(,[
-    char *cp = __FUNCTION__;
-], ac_cv_HAVE_CPP_UNDERBAR_FUNCTION_DEFINED=yes, ac_cv_HAVE_CPP_UNDERBAR_FUNCTION_DEFINED=no)])
-
-if test "x$ac_cv_HAVE_CPP_UNDERBAR_FUNCTION_DEFINED" = "xyes"; then
-  AC_DEFINE(HAVE_CPP_UNDERBAR_FUNCTION_DEFINED)
-fi
-
 #--------------------------------------------------------------
 # on SCO Unixware 7.1.4 (SCO compiler), "static inline" functions
 # are not supported, so this disables the whole inline thing if it
--- ./acconfig.h.bak	2012-03-04 08:29:16.612477343 +0000
+++ ./acconfig.h	2012-03-04 08:30:13.617896753 +0000
@@ -4,8 +4,9 @@
 /* Portions of this file are subject to the following copyright(s).  See
  * the Net-SNMP's COPYING file for more details and other copyrights
  * that may apply:
- */
-/*
+ *
+ * Copyright (c) 2012 Nexenta Systems, Inc.  All rights reserved.
+ *
  * Portions of this file are copyrighted by:
  * Copyright © 2003 Sun Microsystems, Inc. All rights reserved.
  * Use is subject to license terms specified in the COPYING file
@@ -65,9 +66,6 @@
 /* define if you have <sys/disklabel.h> */
 #undef HAVE_SYS_DISKLABEL_H
 
-/* define if your compiler (processor) defines __FUNCTION__ for you */
-#undef HAVE_CPP_UNDERBAR_FUNCTION_DEFINED
-
 /* on aix, if you have perfstat */
 #undef HAVE_PERFSTAT
 
