--- net-snmp-5.4.1/configure.in.old	2013-10-12 19:15:59.351569129 +0400
+++ net-snmp-5.4.1/configure.in	2013-10-12 19:17:35.426716226 +0400
@@ -2968,7 +2968,7 @@
     if test "xenable_perl_cc_checks" != "xno" ; then
       AC_MSG_CHECKING([for Perl cc])
       changequote(, )
-      PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-\w\s\/]+).;\s*/$1/);'`
+      PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-\w\s\/ -m]+).;\s*/$1/);'`
       changequote([, ])
       if test "x$PERLCC" != "x" ; then
         AC_MSG_RESULT([$PERLCC])
