--- gksu-2.0.2/gksu/gksu.c-orig	2010-12-14 14:16:49.208124140 -0600
+++ gksu-2.0.2/gksu/gksu.c	2010-12-14 14:18:20.242890080 -0600
@@ -439,11 +439,13 @@ request_command_and_user (GksuContext *c
   image = gtk_image_new_from_file (DATA_DIR"/pixmaps/gksu-icon.png");
   gtk_box_pack_start (GTK_BOX(rvbox), image, TRUE, TRUE, 0);
 
+#if 0
   /* advanced button */
   advanced_button = gtk_button_new_with_mnemonic (_("_Advanced"));
   g_signal_connect (G_OBJECT(advanced_button), "clicked",
 		    G_CALLBACK(show_hide_advanced), context);
   gtk_box_pack_start (GTK_BOX(rvbox), advanced_button, TRUE, FALSE, 0);
+#endif
 
   /* let the magic begin! */
   gtk_widget_show_all (dialog);
