diff -ur libmng-2.0.3.orig/configure.ac libmng-2.0.3/configure.ac
--- libmng-2.0.3.orig/configure.ac	2013-03-05 05:59:27.000000000 +0000
+++ libmng-2.0.3/configure.ac	2019-05-30 16:48:07.511714702 +0000
@@ -3,7 +3,7 @@
 #
 define([MAJ_NR],[2])
 define([MIN_NR],[0])
-define([REL_NR],[2])
+define([REL_NR],[3])
 #
 #AC_INIT (PACKAGE, VERSION, [BUG-REPORT], [TARNAME], [URL])
 AC_INIT( [libmng],[MAJ_NR.MIN_NR.REL_NR],
