The status of this patch is unknown.

--- mercurial-1.8.2/contrib/check-code.py.orig	Fri Apr  1 10:41:52 2011
+++ mercurial-1.8.2/contrib/check-code.py	Fri Apr  1 11:10:58 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # check-code - a style and portability checker for Mercurial
 #
--- mercurial-1.8.2/contrib/dumprevlog.orig	Fri Apr  1 10:41:52 2011
+++ mercurial-1.8.2/contrib/dumprevlog	Fri Apr  1 11:11:03 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # Dump revlogs as raw data stream
 # $ find .hg/store/ -name "*.i" | xargs dumprevlog > repo.dump
 
--- mercurial-1.8.2/contrib/hg-ssh.orig	Fri Apr  1 10:41:52 2011
+++ mercurial-1.8.2/contrib/hg-ssh	Fri Apr  1 11:11:05 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # Copyright 2005-2007 by Intevation GmbH <intevation@intevation.de>
 #
--- mercurial-1.8.2/contrib/hgweb.fcgi.orig	Fri Apr  1 10:41:52 2011
+++ mercurial-1.8.2/contrib/hgweb.fcgi	Fri Apr  1 11:11:07 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 #
 # An example FastCGI script for use with flup, edit as necessary
 
--- mercurial-4.4/contrib/simplemerge.~1~	2017-11-02 09:04:30.833847375 +0000
+++ mercurial-4.4/contrib/simplemerge	2017-11-02 09:05:16.774306731 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 from __future__ import absolute_import
 
 import getopt
--- mercurial-1.8.2/contrib/undumprevlog.orig	Fri Apr  1 10:41:52 2011
+++ mercurial-1.8.2/contrib/undumprevlog	Fri Apr  1 11:11:10 2011
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.7
 # Undump a dump from dumprevlog
 # $ hg init
 # $ undumprevlog < repo.dump
