illumos doesn't have O_DIRECTORY and python doesn't expose it
--- mercurial-5.3.2/mercurial/posix.py.~1~	2020-04-13 02:12:46.960961632 +0000
+++ mercurial-5.3.2/mercurial/posix.py	2020-04-13 02:21:12.249080096 +0000
@@ -772,7 +772,7 @@
     dirname, basename = os.path.split(path)
     bakwdfd = None
     if dirname:
-        bakwdfd = os.open(b'.', os.O_DIRECTORY)
+        bakwdfd = os.open(b'.', os.O_RDONLY)
         os.chdir(dirname)
     sock.bind(basename)
     if bakwdfd:
