https://github.com/Lucretiel/autocommand/pull/27

--- autocommand-2.2.2/test/test_autoasync.py.orig
+++ autocommand-2.2.2/test/test_autoasync.py
@@ -102,6 +102,8 @@
     assert data == {1, 2, 3, 4}
 
 
+@pytest.mark.filterwarnings(
+    "ignore:coroutine 'test_custom_loop.<locals>.bad_coro' was never awaited")
 def test_custom_loop(context_loop, new_loop):
     did_bad_coro_run = False
 
@@ -110,8 +112,6 @@
         did_bad_coro_run = True
         await YieldOnce()
 
-    # TODO: this fires a "task wasn't awaited" warning; figure out how to
-    # supress
     context_loop.create_task(bad_coro())
 
     @autoasync(loop=new_loop)
