--- ./contrib/perl/Build.PL
+++ ./contrib/perl/Build.PL
@@ -9,7 +9,7 @@
     dist_version_from   => 'lib/Text/Gr2.pm',
     extra_compiler_flags => $^O ne 'MSWin32' ? '-Wall -Wno-unused-variable' : '',
     extra_linker_flags  => $^O eq 'MSWin32' ? '../../build/src/graphite2.lib' : '-lgraphite2',
-    include_dirs        => ['.'],
+    include_dirs        => ['.', '../include'],
     requires => {
         'Test::More' => 0,
     },
