--- jdk23u-jdk-23.0.1-ga/src/hotspot/share/runtime/osThread.hpp.orig
+++ jdk23u-jdk-23.0.1-ga/src/hotspot/share/runtime/osThread.hpp
@@ -32,7 +32,7 @@
 #include "runtime/suspendedThreadTask.hpp"
 #include "utilities/macros.hpp"
 
-#if defined(LINUX) || defined(AIX) || defined(BSD)
+#if defined(LINUX) || defined(AIX) || defined(BSD) || defined(SOLARIS)
 #include "suspendResume_posix.hpp"
 #endif
 
