--- jdk23u-jdk-23.0.1-ga/src/java.base/unix/classes/sun/nio/fs/UnixUserPrincipals.java.orig
+++ jdk23u-jdk-23.0.1-ga/src/java.base/unix/classes/sun/nio/fs/UnixUserPrincipals.java
@@ -67,6 +67,10 @@
             throw new AssertionError();
         }
 
+        boolean isSpecial() {
+            return id == -1;
+        }
+
         @Override
         public String getName() {
             return name;
