--- jdk23u-jdk-23.0.1-ga/src/hotspot/os/posix/signals_posix.cpp.orig
+++ jdk23u-jdk-23.0.1-ga/src/hotspot/os/posix/signals_posix.cpp
@@ -542,6 +542,8 @@
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_aix_signal
 #elif defined(LINUX)
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_linux_signal
+#elif defined(SOLARIS)
+#define JVM_HANDLE_XXX_SIGNAL JVM_handle_solaris_signal
 #else
 #error who are you?
 #endif
