--- jdk23u-jdk-23.0.1-ga/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg.orig
+++ jdk23u-jdk-23.0.1-ga/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg
@@ -19,5 +19,10 @@
 disabledMechanisms = {
   CKM_DSA_KEY_PAIR_GEN
   SecureRandom
+# the following mechanisms are disabled due to lack of digest cloning support
+  CKM_MD5
+  CKM_SHA256
+  CKM_SHA384
+  CKM_SHA512
 }
 
