--- a/src/hotspot/cpu/zero/stubGenerator_zero.cpp	Wed Aug 19 21:41:08 2020
+++ b/src/hotspot/cpu/zero/stubGenerator_zero.cpp	Tue Sep  1 21:46:36 2020
@@ -42,6 +42,9 @@
 #ifdef COMPILER2
 #include "opto/runtime.hpp"
 #endif
+#ifdef __sun__
+#include <alloca.h>
+#endif
 
 // Declaration and definition of StubGenerator (no .hpp file).
 // For a more detailed description of the stub routine structure
