--- ConsoleKit-0.4.1/src/ck-vt-monitor.c-orig	2011-02-23 15:43:10.541619433 -0600
+++ ConsoleKit-0.4.1/src/ck-vt-monitor.c	2011-02-23 15:43:21.779875852 -0600
@@ -470,7 +470,7 @@ ck_vt_monitor_init (CkVtMonitor *vt_moni
                 res = ck_get_active_console_num (fd, &active);
                 if (! res) {
                         /* FIXME: handle failure */
-                        g_warning ("Could not determine active console");
+                        g_debug ("Could not determine active console");
                         active = 0;
                 }
 
