--- idn-0.2.0/config.m4_orig	Tue Jan  6 06:56:21 2009
+++ idn-0.2.0/config.m4	Wed Jan  4 11:20:49 2012
@@ -6,7 +6,7 @@
 
 if test "$PHP_IDN" != "no"; then
   SEARCH_PATH="/usr/local /usr"
-  SEARCH_FOR="/include/idna.h"
+  SEARCH_FOR="include/idn/idna.h"
 
   if test -r $PHP_IDN/$SEARCH_FOR; then # path given as parameter
     IDN_DIR=$PHP_IDN
@@ -33,7 +33,7 @@
     AC_MSG_RESULT([no])
   fi
 
-  PHP_ADD_INCLUDE($IDN_DIR/include)
+  PHP_ADD_INCLUDE(/usr/include/idn)
 
   dnl # --with-idn -> check for lib and symbol presence
   LIBNAME=idn
