--- meld-1.4.0/Makefile	Sun Sep 26 15:25:02 2010
+++ meld-1.4.0/new.Makefile	Thu Apr 21 11:55:26 2011
@@ -47,7 +47,6 @@
 		$(DESTDIR)$(libdir_)/meld/vc \
 		$(DESTDIR)$(sharedir_)/ui \
 		$(DESTDIR)$(sharedir_)/icons \
-		$(DESTDIR)$(docdir_) \
 		$(DESTDIR)$(sharedir)/applications \
 		$(DESTDIR)$(sharedir)/pixmaps \
 		$(DESTDIR)$(sharedir)/icons/hicolor/16x16/apps \
@@ -56,43 +55,44 @@
 		$(DESTDIR)$(sharedir)/icons/hicolor/48x48/apps \
 		$(DESTDIR)$(sharedir)/icons/hicolor/scalable/apps \
 		$(DESTDIR)$(helpdir_)
-	install -m 755 bin/meld.install \
+#		$(DESTDIR)$(docdir_)
+	ginstall -m 555 bin/meld.install \
 		$(DESTDIR)$(bindir)/meld
-	install -m 644 meld/*.py \
+	ginstall -m 444 meld/*.py \
 		$(DESTDIR)$(libdir_)/meld
-	install -m 644 meld/ui/*.py \
+	ginstall -m 444 meld/ui/*.py \
 		$(DESTDIR)$(libdir_)/meld/ui
-	install -m 644 meld/util/*.py \
+	ginstall -m 444 meld/util/*.py \
 		$(DESTDIR)$(libdir_)/meld/util
-	install -m 644 meld/vc/*.py \
+	ginstall -m 444 meld/vc/*.py \
 		$(DESTDIR)$(libdir_)/meld/vc
-	install -m 644 meld/paths.py.install \
+	ginstall -m 444 meld/paths.py.install \
 		$(DESTDIR)$(libdir_)/meld/paths.py
-	install -m 644 data/meld.desktop \
+	ginstall -m 444 data/meld.desktop \
 		$(DESTDIR)$(sharedir)/applications
 	$(PYTHON)    -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)",10,"$(libdir_)")'
-	$(PYTHON) -O -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)",10,"$(libdir_)")'
-	install -m 644 \
+#	$(PYTHON) -O -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)",10,"$(libdir_)")'
+	ginstall -m 444 \
 		data/ui/*.glade \
 		$(DESTDIR)$(sharedir_)/ui
-	install -m 644 \
+	ginstall -m 444 \
 		data/ui/*.xml \
 		$(DESTDIR)$(sharedir_)/ui
-	install -m 644 \
+	ginstall -m 444 \
 		data/icons/*.xpm \
 		data/icons/*.png \
 		$(DESTDIR)$(sharedir_)/icons
-	install -m 644 data/icons/icon.png \
+	ginstall -m 444 data/icons/icon.png \
 		$(DESTDIR)$(sharedir)/pixmaps/meld.png
-	install -m 644 data/icons/16x16/meld.png \
+	ginstall -m 444 data/icons/16x16/meld.png \
 		$(DESTDIR)$(sharedir)/icons/hicolor/16x16/apps/meld.png
-	install -m 644 data/icons/22x22/meld.png \
+	ginstall -m 444 data/icons/22x22/meld.png \
 		$(DESTDIR)$(sharedir)/icons/hicolor/22x22/apps/meld.png
-	install -m 644 data/icons/32x32/meld.png \
+	ginstall -m 444 data/icons/32x32/meld.png \
 		$(DESTDIR)$(sharedir)/icons/hicolor/32x32/apps/meld.png
-	install -m 644 data/icons/48x48/meld.png \
+	ginstall -m 444 data/icons/48x48/meld.png \
 		$(DESTDIR)$(sharedir)/icons/hicolor/48x48/apps/meld.png
-	install -m 644 data/icons/48x48/meld.svg \
+	ginstall -m 444 data/icons/48x48/meld.svg \
 		$(DESTDIR)$(sharedir)/icons/hicolor/scalable/apps/meld.svg
 	$(MAKE) -C po install
 	$(MAKE) -C help install
