--- pecl.orig	Fri Dec  9 14:38:56 2011
+++ pecl	Mon Dec 12 14:11:35 2011
@@ -25,4 +25,15 @@
   fi
 fi
 
+# define CC if not defined
+if test -x "/usr/bin/gcc"; then
+  CC="/usr/bin/gcc"
+  LD=$CC
+  export CC LD
+elif test -x "/usr/sfw/bin/gcc"; then
+  CC="/usr/sfw/bin/gcc"
+  LD=$CC
+  export CC LD
+fi
+
 exec $PHP -C -n -q $INCARG -d date.timezone=UTC -d output_buffering=1 -d variables_order=EGPCS -d safe_mode=0 -d register_argc_argv="On" $INCDIR/peclcmd.php "$@"
