--- gnome-keyring-2.32.1/configure.in	2014-03-01 08:46:25.632749386 +0000
+++ gnome-keyring-2.32.1/configure.in	2014-03-01 08:47:30.706208786 +0000
@@ -82,6 +82,10 @@
 AC_SUBST(GIO_CFLAGS)
 AC_SUBST(GIO_LIBS)
 
+PKG_CHECK_MODULES(GMODULE, glib-2.0 >= 2.25.0 gmodule-2.0)
+AC_SUBST(GMODULE_CFLAGS)
+AC_SUBST(GMODULE_LIBS)
+
 AC_MSG_CHECKING([which GTK+ version to compile against])
 AC_ARG_WITH([gtk],
 	[AS_HELP_STRING([--with-gtk=2.0|3.0],[which GTK+ version to compile against (default: 2.0)])],
--- gnome-keyring-2.32.1/gp11/Makefile.am	2014-03-01 08:47:46.568976322 +0000
+++ gnome-keyring-2.32.1/gp11/Makefile.am	2014-03-01 08:48:22.753971163 +0000
@@ -8,7 +8,8 @@
     	-I$(top_srcdir) \
     	$(GOBJECT_CFLAGS) \
 	$(GTHREAD_CFLAGS) \
-	$(GLIB_CFLAGS)
+	$(GLIB_CFLAGS) \
+	$(GMODULE_CFLAGS)
 
 BUILT_SOURCES = \
 	gp11-marshal.c gp11-marshal.h
@@ -34,7 +35,8 @@
 	$(GOBJECT_LIBS) \
 	$(GTHREAD_LIBS) \
 	$(GIO_LIBS) \
-	$(GLIB_LIBS)
+	$(GLIB_LIBS) \
+	$(GMODULE_LIBS)
 
 gp11-marshal.h: gp11-marshal.list $(GLIB_GENMARSHAL)
 	$(GLIB_GENMARSHAL) $< --header --prefix=_gp11_marshal > $@
@@ -64,4 +66,4 @@
 
 SUBDIRS = . \
 	$(TESTS_DIR)
-	
\ No newline at end of file
+	
