diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-gpg.desktop.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-gpg.desktop.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-gpg.desktop.in	2010-10-26 02:47:49.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-gpg.desktop.in	2016-08-01 13:11:24.992405097 +0300
@@ -3,7 +3,7 @@
 _Name=GPG Password Agent
 _Comment=GNOME Keyring: GPG Agent
 Exec= gnome-keyring-daemon --start --components=gpg
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-gpg.desktop.in.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-gpg.desktop.in.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-gpg.desktop.in.in	2010-10-19 06:11:38.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-gpg.desktop.in.in	2016-08-01 13:11:24.992678186 +0300
@@ -3,7 +3,7 @@
 _Name=GPG Password Agent
 _Comment=GNOME Keyring: GPG Agent
 Exec=@VALGRIND_RUN@ gnome-keyring-daemon --start --components=gpg
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-pkcs11.desktop.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-pkcs11.desktop.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-pkcs11.desktop.in	2010-10-26 02:47:49.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-pkcs11.desktop.in	2016-08-01 13:11:24.992935621 +0300
@@ -3,7 +3,7 @@
 _Name=Certificate and Key Storage
 _Comment=GNOME Keyring: PKCS#11 Component
 Exec=/usr/bin/gnome-keyring-daemon --start --components=pkcs11
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-pkcs11.desktop.in.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-pkcs11.desktop.in.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-pkcs11.desktop.in.in	2010-10-19 06:11:45.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-pkcs11.desktop.in.in	2016-08-01 13:11:24.993213916 +0300
@@ -3,7 +3,7 @@
 _Name=Certificate and Key Storage
 _Comment=GNOME Keyring: PKCS#11 Component
 Exec=@VALGRIND_RUN@@BINDIR@/gnome-keyring-daemon --start --components=pkcs11
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-secrets.desktop.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-secrets.desktop.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-secrets.desktop.in	2010-10-26 02:47:49.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-secrets.desktop.in	2016-08-01 13:11:24.993478319 +0300
@@ -3,7 +3,7 @@
 _Name=Secret Storage Service
 _Comment=GNOME Keyring: Secret Service
 Exec=/usr/bin/gnome-keyring-daemon --start --components=secrets
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-secrets.desktop.in.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-secrets.desktop.in.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-secrets.desktop.in.in	2010-10-19 06:11:45.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-secrets.desktop.in.in	2016-08-01 13:11:24.993738931 +0300
@@ -3,7 +3,7 @@
 _Name=Secret Storage Service
 _Comment=GNOME Keyring: Secret Service
 Exec=@VALGRIND_RUN@@BINDIR@/gnome-keyring-daemon --start --components=secrets
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-ssh.desktop.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-ssh.desktop.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-ssh.desktop.in	2010-10-26 02:47:49.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-ssh.desktop.in	2016-08-01 13:11:24.994029879 +0300
@@ -3,7 +3,7 @@
 _Name=SSH Key Agent
 _Comment=GNOME Keyring: SSH Agent
 Exec=/usr/bin/gnome-keyring-daemon --start --components=ssh
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
diff -ur gnome-keyring-2.32.1/daemon/gnome-keyring-ssh.desktop.in.in gnome-keyring-2.32.1-m/daemon/gnome-keyring-ssh.desktop.in.in
--- gnome-keyring-2.32.1/daemon/gnome-keyring-ssh.desktop.in.in	2010-10-19 06:11:45.000000000 +0400
+++ gnome-keyring-2.32.1-m/daemon/gnome-keyring-ssh.desktop.in.in	2016-08-01 13:11:24.994286369 +0300
@@ -3,7 +3,7 @@
 _Name=SSH Key Agent
 _Comment=GNOME Keyring: SSH Agent
 Exec=@VALGRIND_RUN@@BINDIR@/gnome-keyring-daemon --start --components=ssh
-OnlyShowIn=GNOME;LXDE;
+OnlyShowIn=GNOME;LXDE;MATE;
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
