We test in proto area not in site-packages.

--- virtualenv-20.35.3/tox.ini.orig
+++ virtualenv-20.35.3/tox.ini
@@ -33,7 +33,7 @@
     COVERAGE_FILE = {toxworkdir}/.coverage.{envname}
     COVERAGE_PROCESS_START = {toxinidir}/pyproject.toml
     PYTHONWARNDEFAULTENCODING = 1
-    _COVERAGE_SRC = {envsitepackagesdir}/virtualenv
+    _COVERAGE_SRC = virtualenv
 commands =
     !graalpy: coverage erase
     !graalpy: coverage run -m pytest {posargs:--junitxml "{toxworkdir}/junit.{envname}.xml" tests --int}
