We test in proto area not in site-packages.

--- virtualenv-21.3.3/tox.toml.orig
+++ virtualenv-21.3.3/tox.toml
@@ -26,7 +26,7 @@
 wheel_build_env = ".pkg"
 dependency_groups = [ "test" ]
 pass_env = [ "CI_RUN", "PYTEST_*", "TERM" ]
-set_env._COVERAGE_SRC = "{env_site_packages_dir}{/}virtualenv"
+set_env._COVERAGE_SRC = "virtualenv"
 set_env.COVERAGE_FILE = "{work_dir}{/}.coverage.{env_name}"
 set_env.COVERAGE_PROCESS_START = "{tox_root}{/}pyproject.toml"
 set_env.PYTHONWARNDEFAULTENCODING = "1"
