--- virtualenv-21.5.0/tasks/make_zipapp.py.orig
+++ virtualenv-21.5.0/tasks/make_zipapp.py
@@ -103,7 +103,7 @@
         into.mkdir(parents=True)
         self.into = into
         self.collected = defaultdict(lambda: defaultdict(dict))
-        self.pip_cmd = [str(Path(sys.executable).parent / "pip")]
+        self.pip_cmd = ["python", "-m", "pip"]
         self._cmd = [*self.pip_cmd, "download", "-q", "--no-deps", "--no-cache-dir", "--dest", str(self.into)]
 
     def run(self, target: Path, versions: list[str]) -> None:
