--- virtualenv-21.7.2/tox.toml.orig
+++ virtualenv-21.7.2/tox.toml
@@ -32,8 +32,9 @@
 set_env.COVERAGE_PROCESS_START = "{tox_root}{/}pyproject.toml"
 set_env.PYTHONWARNDEFAULTENCODING = "1"
 commands = [
-  [ "coverage", "erase" ],
+  [ "python", "-m", "coverage", "erase" ],
   [
+    "python", "-m",
     "coverage",
     "run",
     "-m",
@@ -49,10 +50,11 @@
       "loadfile",
     ], extend = true },
   ],
-  [ "coverage", "combine" ],
-  [ "coverage", "report", "--skip-covered", "--show-missing" ],
-  [ "coverage", "xml", "-o", "{work_dir}{/}coverage.{env_name}.xml" ],
+  [ "python", "-m", "coverage", "combine" ],
+  [ "python", "-m", "coverage", "report", "--skip-covered", "--show-missing" ],
+  [ "python", "-m", "coverage", "xml", "-o", "{work_dir}{/}coverage.{env_name}.xml" ],
   [
+    "python", "-m",
     "coverage",
     "html",
     "-d",
