XXX file upstream bug

--- vte-0.40.2/configure.ac	2015-05-12 05:19:21.000000000 -0700
+++ vte-0.40.2/configure.ac	2015-06-24 12:32:35.681360861 -0700
@@ -137,6 +137,17 @@
 CC_CHECK_FLAGS_APPEND([AM_CXXFLAGS],[CXXFLAGS],[compiler_flags_common compiler_flags_only_cxx])
 AC_LANG_POP([C++])
 
+# If these flags are supported for compilation, they must also be used when
+# linking.  To ensure that libtool doesn't strip them out, they're
+# explicitly added to LDFLAGS using -Wc,$OPTION
+if test "$cc_cv_c_CFLAGS__fstack_protector" != "no"; then
+  eval "AM_LDFLAGS='${AM_LDFLAGS} -Wc,-fstack-protector'"
+fi
+
+if test "$cc_cv_c_CFLAGS__fstack_protector_strong" != "no"; then
+  eval "AM_LDFLAGS='${AM_LDFLAGS} -Wc,-fstack-protector-strong'"
+fi
+
 ##########################################################
 # Check for -Bsymbolic-functions linker flag used to avoid
 # intra-library PLT jumps, if available.
