Avoid dependency on gstreamer-basevideo plugin from /hipster-encumbered

--- clutter-gst-1.6.0/configure.ac.1	2018-01-23 18:10:28.973942021 +0000
+++ clutter-gst-1.6.0/configure.ac	2018-01-23 18:11:18.262306211 +0000
@@ -211,17 +211,6 @@
       ])
 
 dnl ========================================================================
-dnl Experimental support for hardware accelerated decoders.
-
-PKG_CHECK_MODULES(HW,
-                 [gstreamer-basevideo-$GST_MAJORMINOR >= $GST_PLUGINS_BAD_REQ_VERSION],
-                 [
-                   have_hw_decoder_support=yes
-                   AC_DEFINE([HAVE_HW_DECODER_SUPPORT], [1],
-                             ["Defined if building Clutter with HW decoder support"])
-                 ],[ have_hw_decoder_support=no ])
-
-dnl ========================================================================
 
 m4_define([maintainer_default], [m4_if(clutter_gst_release_status,
                                        [git], [yes], [no])])
