Solaris-specific; not suitable for upstream

--- ghostscript-9.20/configure.ac	2016-09-26 03:41:28.000000000 -0700
+++ ghostscript-9.20/configure.ac 	2016-09-30 02:35:51.247500625 -0700
@@ -2229,6 +2229,10 @@
         # These font directories are used by Solaris...
         fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type1"
         fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/TrueType"
+        fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type3"
+        fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type1"
+        fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type3"
+        fontpath="${fontpath}:/usr/X11/lib/X11/fonts/TrueType"
 
         # This font directory is used by CUPS...
         if test "x$CUPSCONFIG" != x; then
