*** ghostscript-8.64/base/configure.ac.orig	Tue Aug 12 14:31:40 2008
--- ghostscript-8.64/base/configure.ac	Mon Sep  8 17:20:11 2008
***************
*** 960,966 ****
  		DYNAMIC_FLAGS="-DGS_DEVS_SHARED -DGS_DEVS_SHARED_DIR=\\\"\$(gssharedir)\\\""
  		DYNAMIC_LDFLAGS="-G"
  		DYNAMIC_LIBS=""
- 		X_DEVS=""
  		;;
  		*)
  		AC_MSG_ERROR([Sorry, dynamic driver support not available on this platform!])
--- 960,965 ----
***************
*** 1001,1006 ****
--- 1000,1009 ----
  	# These font directories are used by Solaris...
  	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type1"
  	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/TrueType"
+ 	fontpath="${fontpath}:/usr/openwin/lib/X11/fonts/Type3"
+ 	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type1"
+ 	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/Type3"
+ 	fontpath="${fontpath}:/usr/X11/lib/X11/fonts/TrueType"
  
  	# This font directory is used by CUPS...
  	if test "x$CUPSCONFIG" != x; then
