Solaris-specific; not suitable for upstream

--- ghostscript-9.20/Resource/Init/gs_init.ps	2016-09-26 03:41:28.000000000 -0700
+++ ghostscript-9.20/Resource/Init/gs_init.ps	2016-10-04 23:24:03.811258506 -0700
@@ -2063,7 +2063,7 @@
   SAFETY /safe get not {
     <<
       /PermitFileReading [ ]
-      /PermitFileWriting [ ]
+      /PermitFileWriting [ (/usr/openwin/lib/*) (/usr/X11/lib/*) (/usr/share/ghostscript/*) ]
       /PermitFileControl [ ]
     >> setuserparams
   }
