Solaris-specific; not suitable for upstream

--- ghostscript-9.20/ijs/autogen.sh	2016-09-26 03:41:29.000000000 -0700
+++ ghostscript-9.20/ijs/autogen.sh	2016-10-05 02:32:21.301952517 -0700
@@ -33,7 +33,7 @@
 # do we need automake?
 if test "x$USE_OLD" = "xyes" ; then
   (grep "^AM_INIT_AUTOMAKE" "$CONFIG_AC" >/dev/null) && {
-    AM_NEEDED=`fgrep AUTOMAKE_OPTIONS Makefile.am | $VERSIONGREP`
+    AM_NEEDED=`fgrep AUTOMAKE_OPTIONS Makefile.am | nawk '{print $3}'`
     AM_NMAJOR=`echo $AM_NEEDED |cut -d. -f1`
     AM_NMINOR=`echo $AM_NEEDED |cut -d. -f2`
     AM_NPATCHLEVEL=`echo $AM_NEEDED |cut -d. -f3`
