Solaris-specific; not suitable for upstream

--- ghostscript-9.20/base/ugcclib.mak	2016-09-26 03:41:28.000000000 -0700
+++ ghostscript-9.20/base/ugcclib.mak	2016-10-12 23:12:09.441179744 -0700
@@ -47,7 +47,7 @@
 PGDIRPREFIX=
 
 JSRCDIR=jpeg
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 PNGSRCDIR=libpng
--- ghostscript-9.20/base/unix-gcc.mak	2016-09-26 03:41:28.000000000 -0700
+++ ghostscript-9.20/base/unix-gcc.mak	2016-10-12 23:14:22.153002400 -0700
@@ -230,7 +230,7 @@
 # some older JPEG streams that violate the standard. If the JPEG
 # library built from local sources, the patch will be applied.
 
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 # Define the directory where the PNG library sources are stored,
@@ -238,7 +238,7 @@
 # You may need to change this if the libpng version changes.
 # See png.mak for more information.
 
-SHARE_LIBPNG=0
+SHARE_LIBPNG=1
 PNGSRCDIR=./libpng
 LIBPNG_NAME=png
 
@@ -253,7 +253,7 @@
 # Define the directory where the zlib sources are stored.
 # See zlib.mak for more information.
 
-SHARE_ZLIB=0
+SHARE_ZLIB=1
 ZSRCDIR=./zlib
 #ZLIB_NAME=gz
 ZLIB_NAME=z
--- ghostscript-9.20/base/unixansi.mak	2016-09-26 03:41:28.000000000 -0700
+++ ghostscript-9.20/base/unixansi.mak	2016-10-12 23:16:28.621061891 -0700
@@ -129,7 +129,7 @@
 # some older JPEG streams that violate the standard. If the JPEG
 # library built from local sources, the patch will be applied.
 
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 # Define the directory where the PNG library sources are stored,
@@ -143,7 +143,7 @@
 # what its name is.
 # See gs.mak and Make.htm for more information.
 
-SHARE_LIBPNG=0
+SHARE_LIBPNG=1
 LIBPNG_NAME=png
 
 # Define whether to use a shared version of libtiff and where
@@ -165,7 +165,7 @@
 # what its name is (usually libz, but sometimes libgz).
 # See gs.mak and Make.htm for more information.
 
-SHARE_ZLIB=0
+SHARE_ZLIB=1
 #ZLIB_NAME=gz
 ZLIB_NAME=z
 
