--- ghostscript-9.00/base/ugcclib.mak.orig	Tue Jul 26 01:47:52 2011
+++ ghostscript-9.00/base/ugcclib.mak	Tue Jul 26 01:55:16 2011
@@ -42,7 +42,7 @@
 PGRELDIR=.
 
 JSRCDIR=jpeg
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 PNGSRCDIR=libpng
--- ghostscript-9.00/base/unix-gcc.mak.orig	Tue Jul 26 01:48:14 2011
+++ ghostscript-9.00/base/unix-gcc.mak	Tue Jul 26 01:56:23 2011
@@ -138,7 +138,7 @@
 # some older JPEG streams that violate the standard. If the JPEG
 # library built from local sources, the patch will be applied.
 
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 # Define the directory where the PNG library sources are stored,
@@ -152,7 +152,7 @@
 # what its name is.
 # See gs.mak and Make.htm for more information.
 
-SHARE_LIBPNG=0
+SHARE_LIBPNG=1
 LIBPNG_NAME=png
 
 # Define whether to use a shared version of libtiff and where
@@ -173,7 +173,7 @@
 # what its name is (usually libz, but sometimes libgz).
 # See gs.mak and Make.htm for more information.
 
-SHARE_ZLIB=0
+SHARE_ZLIB=1
 #ZLIB_NAME=gz
 ZLIB_NAME=z
 
--- ghostscript-9.00/base/unixansi.mak.orig	Tue Jul 26 01:48:19 2011
+++ ghostscript-9.00/base/unixansi.mak	Tue Jul 26 01:57:00 2011
@@ -123,7 +123,7 @@
 # some older JPEG streams that violate the standard. If the JPEG
 # library built from local sources, the patch will be applied.
 
-SHARE_JPEG=0
+SHARE_JPEG=1
 JPEG_NAME=jpeg
 
 # Define the directory where the PNG library sources are stored,
@@ -137,7 +137,7 @@
 # what its name is.
 # See gs.mak and Make.htm for more information.
 
-SHARE_LIBPNG=0
+SHARE_LIBPNG=1
 LIBPNG_NAME=png
 
 # Define whether to use a shared version of libtiff and where
@@ -158,7 +158,7 @@
 # what its name is (usually libz, but sometimes libgz).
 # See gs.mak and Make.htm for more information.
 
-SHARE_ZLIB=0
+SHARE_ZLIB=1
 #ZLIB_NAME=gz
 ZLIB_NAME=z
 
