*** ghostscript-8.64/base/Makefile.in.orig	Mon Nov 29 15:18:29 2010
--- ghostscript-8.64/base/Makefile.in	Mon Nov 29 15:20:15 2010
***************
*** 241,247 ****
  # Define the added flags for standard, debugging, profiling 
  # and shared object builds.
  
! CFLAGS_STANDARD=@OPT_CFLAGS@
  CFLAGS_DEBUG=-g -O0
  CFLAGS_PROFILE=-pg @OPT_CFLAGS@
  CFLAGS_SO=@DYNAMIC_CFLAGS@
--- 241,247 ----
  # Define the added flags for standard, debugging, profiling 
  # and shared object builds.
  
! CFLAGS_STANDARD=-O2
  CFLAGS_DEBUG=-g -O0
  CFLAGS_PROFILE=-pg @OPT_CFLAGS@
  CFLAGS_SO=@DYNAMIC_CFLAGS@
