--- ./configure.ac	Fri Jul 10 11:23:57 2015
+++ ./configure.ac	Fri Jul 10 11:31:10 2015
@@ -84,6 +84,12 @@
         has_znow=no
         EPOXY_LINK_LIBS=""
         ;;
+    solaris*)
+        build_egl=no
+        build_glx=yes
+        build_wgl=no
+        EPOXY_LINK_LIBS=""
+        ;;
     *)
         build_egl=yes
         build_glx=yes
--- ./src/dispatch_common.h	Fri Jul 10 11:57:22 2015
+++ ./src/dispatch_common.h	Fri Jul 10 11:58:48 2015
@@ -33,6 +33,11 @@
 #define PLATFORM_HAS_GLX 1
 #define PLATFORM_HAS_WGL 0
 #define EPOXY_IMPORTEXPORT
+#elif defined(__sun) && defined(__SVR4)
+#define PLATFORM_HAS_EGL 0
+#define PLATFORM_HAS_GLX 1
+#define PLATFORM_HAS_WGL 0
+#define EPOXY_IMPORTEXPORT
 #else
 #define PLATFORM_HAS_EGL 1
 #define PLATFORM_HAS_GLX 1
