--- a/src/java.base/unix/classes/sun/nio/fs/UnixUserPrincipals.java	Thu May 28 00:44:10 2020
+++ b/src/java.base/unix/classes/sun/nio/fs/UnixUserPrincipals.java	Wed May 20 19:29:11 2020
@@ -67,6 +67,10 @@
             throw new AssertionError();
         }
 
+        boolean isSpecial() {
+            return id == -1;
+        }
+
         @Override
         public String getName() {
             return name;
