diff -Nru jdk17u-jdk-17.0.7-ga.orig/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg jdk17u-jdk-17.0.7-ga/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg
--- jdk17u-jdk-17.0.7-ga.orig/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg	2023-08-19 17:26:03.629148428 +0200
+++ jdk17u-jdk-17.0.7-ga/src/jdk.crypto.cryptoki/solaris/conf/security/sunpkcs11-solaris.cfg	2023-08-19 17:29:14.784751746 +0200
@@ -19,5 +19,10 @@
 disabledMechanisms = {
   CKM_DSA_KEY_PAIR_GEN
   SecureRandom
+# the following mechanisms are disabled due to lack of digest cloning support
+  CKM_MD5
+  CKM_SHA256
+  CKM_SHA384
+  CKM_SHA512
 }
 
