--- gettext-0.16.1.orig/gettext-tools/tests/lang-bash
+++ gettext-0.16.1/gettext-tools/tests/lang-bash
@@ -74,9 +74,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
--- gettext-0.16.1.orig/gettext-tools/tests/lang-c
+++ gettext-0.16.1/gettext-tools/tests/lang-c
@@ -107,9 +107,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
--- gettext-0.16.1.orig/gettext-tools/tests/lang-clisp
+++ gettext-0.16.1/gettext-tools/tests/lang-clisp
@@ -74,9 +74,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
--- gettext-0.16.1.orig/gettext-tools/tests/lang-gawk
+++ gettext-0.16.1/gettext-tools/tests/lang-gawk
@@ -78,9 +78,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
--- gettext-0.16.1.orig/gettext-tools/tests/lang-guile
+++ gettext-0.16.1/gettext-tools/tests/lang-guile
@@ -77,9 +77,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
--- gettext-0.16.1.orig/gettext-tools/tests/lang-sh
+++ gettext-0.16.1/gettext-tools/tests/lang-sh
@@ -72,9 +72,11 @@
 : ${DIFF=diff}
 ${DIFF} fr.po fr.po.new || exit 1
 
-tmpfiles="$tmpfiles fr"
+tmpfiles="$tmpfiles fr_FR.ISO8859-1 fr_FR.UTF-8"
 test -d fr || mkdir fr
 test -d fr/LC_MESSAGES || mkdir fr/LC_MESSAGES
+test -L fr_FR.ISO8859-1 || ln -s fr fr_FR.ISO8859-1
+test -L fr_FR.UTF-8 || ln -s fr fr_FR.UTF-8
 
 : ${MSGFMT=msgfmt}
 ${MSGFMT} -o fr/LC_MESSAGES/prog.mo fr.po
