Fix out-of-tree builds.

XXX file upstream bug
--- librsvg-2.40.9/Makefile.am	2015-02-04 14:17:14.000000000 -0800
+++ librsvg-2.40.9/Makefile.am	2015-05-22 16:26:28.059926082 -0700
@@ -254,7 +254,7 @@
 
 librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo
 librsvg_@RSVG_API_VERSION_U@_vapi_METADATADIRS = $(srcdir)
-librsvg_@RSVG_API_VERSION_U@_vapi_FILES = Rsvg-$(RSVG_API_VERSION).gir Rsvg-$(RSVG_API_VERSION)-custom.vala
+librsvg_@RSVG_API_VERSION_U@_vapi_FILES = Rsvg-$(RSVG_API_VERSION).gir $(srcdir)/Rsvg-$(RSVG_API_VERSION)-custom.vala
 
 vapidir = $(datadir)/vala/vapi
 vapi_DATA = $(VAPIGEN_VAPIS)
