--- nrpe-4.1.0/startup/solaris-init.xml.in.orig	2022-09-04 08:25:13.012334859 +0000
+++ nrpe-4.1.0/startup/solaris-init.xml.in	2022-09-04 08:25:32.429454220 +0000
@@ -91,9 +91,6 @@
 			name='start'
 			exec='@sbindir@/nrpe -c @pkgsysconfdir@/nrpe.cfg -d'
 			timeout_seconds='5'>
-			<method_context>
-				<method_credential user='@nrpe_user@' group='@nrpe_group@'/>
-			</method_context>
 		</exec_method>
 
 		<exec_method
