--- configure	2007-10-09 14:28:30.000000000 -0700
+++ configure	2011-03-15 10:44:19.669826790 -0700
@@ -2474,13 +2474,13 @@
   if test "$GXX" = yes; then
     CXXFLAGS="-g -O2"
   else
-    CXXFLAGS="-g"
+    CXXFLAGS="${CXXFLAGS}"
   fi
 else
   if test "$GXX" = yes; then
     CXXFLAGS="-O2"
   else
-    CXXFLAGS=
+    CXXFLAGS="${CXXFLAGS}"
   fi
 fi
 for ac_declaration in \
@@ -3304,13 +3304,13 @@
   if test "$GCC" = yes; then
     CFLAGS="-g -O2"
   else
-    CFLAGS="-g"
+    CFLAGS="${CFLAGS}"
   fi
 else
   if test "$GCC" = yes; then
     CFLAGS="-O2"
   else
-    CFLAGS=
+    CFLAGS="${CFLAGS}"
   fi
 fi
 echo "$as_me:$LINENO: checking for $CC option to accept ANSI C" >&5
@@ -4416,7 +4416,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CPP needs to be expanded
-    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib/cpp"
+    for CPP in "$CPP" "$CC $CPPFLAGS $CFLAGS -E"
     do
       ac_preproc_ok=false
 for ac_c_preproc_warn_flag in '' yes
@@ -5040,7 +5040,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CXXCPP needs to be expanded
-    for CXXCPP in "$CXX -E" "/lib/cpp"
+    for CXXCPP in "$CXXCPP" "$CXX $CPPFLAGS $CXXFLAGS -E"
     do
       ac_preproc_ok=false
 for ac_cxx_preproc_warn_flag in '' yes
@@ -7533,17 +7533,17 @@
       ;;
 
     solaris*)
-      no_undefined_flag=' -z text'
+      no_undefined_flag=' -z text -z defs -z ignore'
       if test "$GCC" = yes; then
 	archive_cmds='$CC -shared ${wl}-h ${wl}$soname -o $lib $libobjs $deplibs $compiler_flags'
 	archive_expsym_cmds='$echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~$echo "local: *; };" >> $lib.exp~
 	  $CC -shared ${wl}-M ${wl}$lib.exp ${wl}-h ${wl}$soname -o $lib $libobjs $deplibs $compiler_flags~$rm $lib.exp'
       else
-	archive_cmds='$LD -G${allow_undefined_flag} -h $soname -o $lib $libobjs $deplibs $linker_flags'
+	archive_cmds='$CC $CFLAGS $LDFLAGS -G ${allow_undefined_flag} -h $soname -o $lib $libobjs $deplibs $linker_flags'
 	archive_expsym_cmds='$echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~$echo "local: *; };" >> $lib.exp~
-  	$LD -G${allow_undefined_flag} -M $lib.exp -h $soname -o $lib $libobjs $deplibs $linker_flags~$rm $lib.exp'
+  	$CC $CFLAGS $LDFLAGS -G ${allow_undefined_flag} -M $lib.exp -h $soname -o $lib $libobjs $deplibs $linker_flags~$rm $lib.exp'
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec=''
       hardcode_shlibpath_var=no
       case $host_os in
       solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -10747,11 +10747,11 @@
       CC)
 	# Sun C++ 4.2, 5.x and Centerline C++
 	no_undefined_flag_CXX=' -zdefs'
-	archive_cmds_CXX='$CC -G${allow_undefined_flag} -nolib -h$soname -o $lib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags'
+	archive_cmds_CXX='$CC $CFLAGS $LDFLAGS -G ${allow_undefined_flag} -nolib -h$soname -o $lib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags'
 	archive_expsym_cmds_CXX='$echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~$echo "local: *; };" >> $lib.exp~
-	$CC -G${allow_undefined_flag} -nolib ${wl}-M ${wl}$lib.exp -h$soname -o $lib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags~$rm $lib.exp'
+	$CC $CFLAGS $LDFLAGS -G ${allow_undefined_flag} -nolib ${wl}-M ${wl}$lib.exp -h$soname -o $lib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags~$rm $lib.exp'
 
-	hardcode_libdir_flag_spec_CXX='-R$libdir'
+	hardcode_libdir_flag_spec_CXX=''
 	hardcode_shlibpath_var_CXX=no
 	case $host_os in
 	  solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -10773,13 +10773,13 @@
 	# explicitly linking system object files so we need to strip them
 	# from the output so that they don't get included in the library
 	# dependencies.
-	output_verbose_link_cmd='templist=`$CC -G $CFLAGS -v conftest.$objext 2>&1 | grep "\-[LR]"`; list=""; for z in $templist; do case $z in conftest.$objext) list="$list $z";; *.$objext);; *) list="$list $z";;esac; done; echo $list'
+	output_verbose_link_cmd='templist=`$CC $CFLAGS $LDFLAGS -G -v conftest.$objext 2>&1 | grep "\-[LR]"`; list=""; for z in $templist; do case $z in conftest.$objext) list="$list $z";; *.$objext);; *) list="$list $z";;esac; done; echo $list'
 
 	# Archives containing C++ object files must be created using
 	# "CC -xar", where "CC" is the Sun C++ compiler.  This is
 	# necessary to make sure instantiated templates are included
 	# in the archive.
-	old_archive_cmds_CXX='$CC -xar -o $oldlib $oldobjs'
+	old_archive_cmds_CXX='$CC $CFLAGS -xar -o $oldlib $oldobjs'
 	;;
       gcx)
 	# Green Hills C++ Compiler
@@ -16858,7 +16858,7 @@
       ;;
 
     solaris*)
-      no_undefined_flag_GCJ=' -z text'
+      no_undefined_flag_GCJ=' -z text-z defs -z ignore'
       if test "$GCC" = yes; then
 	archive_cmds_GCJ='$CC -shared ${wl}-h ${wl}$soname -o $lib $libobjs $deplibs $compiler_flags'
 	archive_expsym_cmds_GCJ='$echo "{ global:" > $lib.exp~cat $export_symbols | $SED -e "s/\(.*\)/\1;/" >> $lib.exp~$echo "local: *; };" >> $lib.exp~
@@ -19531,7 +19531,7 @@
 # which indicates that we try without any flags at all, and "pthread-config"
 # which is a program returning the flags for the Pth emulation library.
 
-acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+acx_pthread_flags="-mt"
 
 # The ordering *is* (sometimes) important.  Some notes on the
 # individual items follow:
@@ -19847,7 +19847,7 @@
 	LIBS="$PTHREAD_LIBS $LIBS"
 	CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 	CXXFLAGS="$CXXFLAGS $PTHREAD_CFLAGS"
-	CC="$PTHREAD_CC"
+	CC="$CC"
 
 
 # Check whether --enable-posix-sem or --disable-posix-sem was given.
@@ -20974,7 +20974,7 @@
 echo "$as_me:$LINENO: checking for gcc optimization flags" >&5
 echo $ECHO_N "checking for gcc optimization flags... $ECHO_C" >&6
 old_cflags=$CFLAGS
-CFLAGS="$CFLAGS -pipe"
+CFLAGS="$CFLAGS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21012,7 +21012,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-   EXTRA_OPT_CFLAGS="-pipe"
+   EXTRA_OPT_CFLAGS=""
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
