Append -pipe to CFLAGS.
Will not send upstream.

--- configure	2016-11-14 16:35:20.860772554 +0000
+++ configure	2016-11-14 16:35:42.301383478 +0000
@@ -17146,7 +17146,7 @@
 { $as_echo "$as_me:$LINENO: checking for gcc optimization flags" >&5
 $as_echo_n "checking for gcc optimization flags... " >&6; }
 old_cflags=$CFLAGS
-CFLAGS="$CFLAGS -pipe"
+CFLAGS="$CFLAGS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -17180,7 +17180,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest.$ac_objext; then
-   EXTRA_OPT_CFLAGS="-pipe"
+   EXTRA_OPT_CFLAGS=""
 else
   $as_echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
